Index: sarab-0.2.4/etc/sarab.conf
===================================================================
--- sarab-0.2.4.orig/etc/sarab.conf
+++ sarab-0.2.4/etc/sarab.conf
@@ -75,10 +75,10 @@ SECURITY_CONFIG=$SARAB_ETC/sarab.dcf.sec
 BASENAME="$(date +'%m-%d-%Y_%H%M')"
 
 # The location of the DAR executable file
-DAR_BINARY="/usr/local/bin/dar"
+DAR_BINARY="/usr/bin/dar"
 
 # The location of the statically-compiled DAR executable file
-DAR_STATIC="/usr/local/bin/dar_static"
+DAR_STATIC="/usr/bin/dar_static"
 
 # The temporary directory name that backups will be created in. This will be created underneath $DESTINATION.
 WORK_DIR=".sarab_temp_workdir"
@@ -91,4 +91,4 @@ DAR_CREATE="$DESTINATION/$WORK_DIR/$BASE
 DAR_COMMAND="$DAR_BINARY --batch $SECURITY_CONFIG --batch $SARAB_ETC/$SARAB_DCF -c $DAR_CREATE --noconf"
 
 # Location and name of temporary log file
-TEMP_LOG="$SARAB_ETC/sarab_temp.log"
+TEMP_LOG="/var/log/sarab_temp.log"
