--- tribes/barbarians/deep_goldmine/conf.old	2011-07-12 17:25:38.038927012 +0200
+++ tribes/barbarians/deep_goldmine/conf	2011-07-12 17:26:40.067926999 +0200
@@ -1,7 +1,7 @@
 size=mine
 buildable=no
 enhanced_building=yes
-output=gold
+output=goldstone
 enhancement=deeper_goldmine
 
 [buildcost]
@@ -26,11 +26,11 @@
 sleep=42000
 consume=snack
 animate=working 18000
-mine=coal 2 66 5
-produce=gold:2
+mine=gold 2 66 5
+produce=goldstone:2
 animate=working 18000
 mine=gold 2 66 5
-produce=gold:2
+produce=goldstone:2
 
 [idle]
 pics=deep_goldmine_i_??.png
