--- include/csutil/blockallocator.h.old	2012-07-17 09:08:57.203478402 +0200
+++ include/csutil/blockallocator.h	2012-07-17 09:10:34.556642885 +0200
@@ -166,7 +166,7 @@
   ~csBlockAllocator()
   {
     ObjectDispose dispose (*this, false);
-    DisposeAll (dispose);
+    this->DisposeAll (dispose);
   }
 
   /**
@@ -177,7 +177,7 @@
   void Empty()
   {
     ObjectDispose dispose (*this, true);
-    DisposeAll (dispose);
+    this->DisposeAll (dispose);
   }
 
   /**
--- include/csutil/parray.h.old	2012-07-17 13:02:11.524886456 +0200
+++ include/csutil/parray.h	2012-07-17 13:02:43.531639147 +0200
@@ -106,7 +106,7 @@
   {
     CS_ASSERT (this->GetSize () > 0);
     T* ret = GetAndClear (this->GetSize () - 1); // see *1*
-    Truncate (this->GetSize () - 1);
+    this->Truncate (this->GetSize () - 1);
     return ret;
   }
 
--- include/csutil/refarr.h.old	2012-07-17 09:16:33.084469347 +0200
+++ include/csutil/refarr.h	2012-07-17 09:17:04.008298074 +0200
@@ -106,7 +106,7 @@
   {
     CS_ASSERT (this->GetSize () > 0);
     csRef<T> ret = this->Get (this->GetSize () - 1); // see *1*
-    SetSize (this->GetSize () - 1);
+    this->SetSize (this->GetSize () - 1);
     return csPtr<T> (ret);
   }
 };
