--- configure.ac.old	2010-06-19 21:50:21.000000000 +0200
+++ configure.ac	2010-06-19 22:00:50.000000000 +0200
@@ -1318,8 +1318,7 @@
       btDefaultCollisionConfiguration* c;
       btDynamicsWorld* world = new btDiscreteDynamicsWorld (d, b, s, c)])],
     [], [C++], [], [],
-    [], [], [-lbulletdynamics -lbulletcollision -lbulletmath],
-    [bulletdynamics])
+    [])
 
 AS_IF([test $cs_cv_libbullet = yes],
     [CS_CHECK_BUILD([if bullet is sufficiently recent (>=2.68)], [cs_cv_bullet_recent],
