diff -urN hwsetup-1.2.orig/hwsetup.c hwsetup-1.2/hwsetup.c
--- hwsetup-1.2.orig/hwsetup.c	2005-12-06 08:42:39.000000000 -0500
+++ hwsetup-1.2/hwsetup.c	2005-12-06 08:42:18.000000000 -0500
@@ -23,14 +23,15 @@
 #endif
 
 #define VERSION "HWSETUP 1.2, an automatic hardware configuration tool\n" \
-                "(C) 2006 Klaus Knopper <knoppix@knopper.net>\n\n"
+                "(C) 2006 Klaus Knopper <knoppix@knopper.net>\n" \
+				"Modified for Gentoo Linux\n\n"
 
 #define CARDSDB "/usr/share/hwdata/Cards"
-#define XPATH "/usr/X11R6/bin/"
-#define XMODPATH "/usr/X11R6/lib/modules/drivers/"
+#define XPATH "/usr/bin/"
+#define XMODPATH "/usr/lib/xorg/modules/drivers/"
 
 #define VERBOSE_PRINT 1
-#define VERBOSE_PROMPT 2
+#define VERBOSE_PROMPT 0
 
 #define SKIP_AUDIO 1
 #define SKIP_SCSI 2
@@ -255,7 +256,7 @@
              char xmodule[32];
              char fullpath[128];
              sscanf(&buffer[7],"%31s",xmodule);
-             sprintf(fullpath,XMODPATH"%.31s_drv.o",xmodule);
+             sprintf(fullpath,XMODPATH"%.31s_drv.so",xmodule);
              if(exists(fullpath))
               {
                strncpy(xi.xmodule,xmodule,sizeof(xi.xmodule));
@@ -517,7 +518,7 @@
 int writeconfig(char *name,struct device *d,int verbose)
 {
  FILE *f,*k;
- const char *kconfig="/etc/sysconfig/knoppix";
+ const char *kconfig="/etc/sysconfig/gentoo";
  char *desc;
  unlink(name);
  if((f=fopen(name,"w"))==NULL)
