--- eggtimer-0.9.5/i18n.c.orig  2006-09-28 08:46:26.000000000 -0700
+++ eggtimer-0.9.5/i18n.c       2006-09-28 08:54:38.000000000 -0700
@@ -7,7 +7,7 @@
     "Eieruhr", /*Deutsch*/
     "", // TODO /*Slovenski*/
     "", // TODO /*Italiano*/
-    "Eierwekker", /*Nederlands*
+    "Eierwekker", /*Nederlands*/
     "", // TODO /*Português*/
     "Programmateur", /*Français*/
     "", // TODO /*Norsk*/
@@ -31,7 +31,7 @@
     "Eieruhr für VDR",  /*Deutsch*/
     "", // TODO /*Slovenski*/
     "", // TODO /*Italiano*/
-    "Eierwekker voor VDR", /*Nederlands*
+    "Eierwekker voor VDR", /*Nederlands*/
     "", // TODO /*Português*/
     "Programmateur pour VDR", /*Français*/
     "", // TODO /*Norsk*/
@@ -55,7 +55,7 @@
     "Eieruhr: Zeit ist abgelaufen",    /*Deutsch*/
     "", // TODO /*Slovenski*/
     "", // TODO /*Italiano*/
-    "Eierwekker: tijd is afgelopen", /*Nederlands*
+    "Eierwekker: tijd is afgelopen", /*Nederlands*/
     "", // TODO /*Português*/
     "Programmateur: le temps est écoulé", /*Français*/
     "", // TODO /*Norsk*/
@@ -79,7 +79,7 @@
     "Eieruhr ausschalten",     /*Deutsch*/
     "", // TODO /*Slovenski*/
     "", // TODO /*Italiano*/
-    "Stop eierwekker", /*Nederlands*
+    "Stop eierwekker", /*Nederlands*/
     "", // TODO /*Português*/
     "Arrêter le programmateur", /*Français*/
     "", // TODO /*Norsk*/
@@ -103,7 +103,7 @@
     "Vorlage", /*Deutsch*/
     "", // TODO /*Slovenski*/
     "", // TODO /*Italiano*/
-    "Voorgave", /*Nederlands*
+    "Voorgave", /*Nederlands*/
     "", // TODO /*Português*/
     "Modèle", /*Français*/
     "", // TODO /*Norsk*/
@@ -127,7 +127,7 @@
     "Nachricht im OSD",        /*Deutsch*/
     "", // TODO /*Slovenski*/
     "", // TODO /*Italiano*/
-    "OSD bericht", /*Nederlands*
+    "OSD bericht", /*Nederlands*/
     "", // TODO /*Português*/
     "Message OSD", /*Français*/
     "", // TODO /*Norsk*/
@@ -151,7 +151,7 @@
     "Umschalten",      /*Deutsch*/
     "", // TODO /*Slovenski*/
     "", // TODO /*Italiano*/
-    "Verwissel kanaal", /*Nederlands*
+    "Verwissel kanaal", /*Nederlands*/
     "", // TODO /*Português*/
     "Changer de chaine", /*Français*/
     "", // TODO /*Norsk*/
@@ -175,7 +175,7 @@
     "Eieruhr stellen", /*Deutsch*/
     "", // TODO /*Slovenski*/
     "", // TODO /*Italiano*/
-    "Stel eierwekker in", // TODO /*Nederlands*
+    "Stel eierwekker in", // TODO /*Nederlands*/
     "", // TODO /*Português*/
     "Valider programmateur", /*Français*/
     "", // TODO /*Norsk*/
@@ -199,7 +199,7 @@
     "Aktion",  /*Deutsch*/
     "", // TODO /*Slovenski*/
     "", // TODO /*Italiano*/
-    "Aktie", // TODO /*Nederlands*
+    "Aktie", // TODO /*Nederlands*/
     "", // TODO /*Português*/
     "Action", /*Français*/
     "", // TODO /*Norsk*/
@@ -223,7 +223,7 @@
     "Abbrechen",       /*Deutsch*/
     "", // TODO /*Slovenski*/
     "", // TODO /*Italiano*/
-    "Afbreken", /*Nederlands*
+    "Afbreken", /*Nederlands*/
     "", // TODO /*Português*/
     "Annuler", /*Français*/
     "", // TODO /*Norsk*/
@@ -247,7 +247,7 @@
     "Zeit-Modus",      /*Deutsch*/
     "", // TODO /*Slovenski*/
     "", // TODO /*Italiano*/
-    "Tijd-modus", /*Nederlands*
+    "Tijd-modus", /*Nederlands*/
     "", // TODO /*Português*/
     "Echelle de temps", /*Français*/
     "", // TODO /*Norsk*/
@@ -271,7 +271,7 @@
     "Sekunden",        /*Deutsch*/
     "", // TODO /*Slovenski*/
     "", // TODO /*Italiano*/
-    "Seconden", /*Nederlands*
+    "Seconden", /*Nederlands*/
     "", // TODO /*Português*/
     "Secondes", /*Français*/
     "", // TODO /*Norsk*/
@@ -295,7 +295,7 @@
     " Sekunden",       /*Deutsch*/
     "", // TODO /*Slovenski*/
     "", // TODO /*Italiano*/
-    " Seconden", /*Nederlands*
+    " Seconden", /*Nederlands*/
     "", // TODO /*Português*/
     " Secondes", /*Français*/
     "", // TODO /*Norsk*/
@@ -319,7 +319,7 @@
     " Datum",  /*Deutsch*/
     "", // TODO /*Slovenski*/
     "", // TODO /*Italiano*/
-    " Datum",   /*Nederlands*
+    " Datum",   /*Nederlands*/
     "", // TODO /*Português*/
     " Date", /*Français*/
     "", // TODO /*Norsk*/
@@ -343,7 +343,7 @@
     " Zeit",   /*Deutsch*/
     "", // TODO /*Slovenski*/
     "", // TODO /*Italiano*/
-    " Tijd",    /*Nederlands*
+    " Tijd",    /*Nederlands*/
     "", // TODO /*Português*/
     " Heure", /*Français*/
     "", // TODO /*Norsk*/
@@ -367,7 +367,7 @@
     "Datum/Zeit",      /*Deutsch*/
     "", // TODO /*Slovenski*/
     "", // TODO /*Italiano*/
-    "Datum/TIJD",    /*Nederlands*
+    "Datum/TIJD",    /*Nederlands*/
     "", // TODO /*Português*/
     "Date/Heure", /*Français*/
     "", // TODO /*Norsk*/
@@ -391,7 +391,7 @@
     "Minuten", /*Deutsch*/
     "", // TODO /*Slovenski*/
     "", // TODO /*Italiano*/
-    "Minuten",  /*Nederlands*
+    "Minuten",  /*Nederlands*/
     "", // TODO /*Português*/
     "Minutes", /*Français*/
     "", // TODO /*Norsk*/
@@ -415,7 +415,7 @@
     " Minuten",        /*Deutsch*/
     "", // TODO /*Slovenski*/
     "", // TODO /*Italiano*/
-    " Minuten", /*Nederlands*
+    " Minuten", /*Nederlands*/
     "", // TODO /*Português*/
     " Minutes", /*Français*/
     "", // TODO /*Norsk*/
@@ -439,7 +439,7 @@
     " Kanal",  /*Deutsch*/
     "", // TODO /*Slovenski*/
     "", // TODO /*Italiano*/
-    " Kanaal",  /*Nederlands*
+    " Kanaal",  /*Nederlands*/
     "", // TODO /*Português*/
     " Chaine", /*Français*/
     "", // TODO /*Norsk*/
@@ -464,7 +464,7 @@
     "Eieruhr: Konnte nicht umschalten",        /*Deutsch*/
     "", // TODO /*Slovenski*/
     "", // TODO /*Italiano*/
-    "Eierwekker: kon niet van kanaal wisselen", /*Nederlands*
+    "Eierwekker: kon niet van kanaal wisselen", /*Nederlands*/
     "", // TODO /*Português*/
     "Programmateur: impossible de changer de chaine", /*Français*/
     "", // TODO /*Norsk*/
@@ -489,7 +489,7 @@
     " Befehl", /*Deutsch*/
     "", // TODO /*Slovenski*/
     "", // TODO /*Italiano*/
-    " Commando", /*Nederlands*
+    " Commando", /*Nederlands*/
     "", // TODO /*Português*/
     " Commande", /*Français*/
     "", // TODO /*Norsk*/
@@ -514,7 +514,7 @@
     "Eggtimer: Befehl nicht vorhanden",        /*Deutsch*/
     "", // TODO /*Slovenski*/
     "", // TODO /*Italiano*/
-    "Eierwekker: commando niet gevonden", // TODO /*Nederlands*
+    "Eierwekker: commando niet gevonden", // TODO /*Nederlands*/
     "", // TODO /*Português*/
     "Programmateur; commande introuvable", /*Français*/
     "", // TODO /*Norsk*/
@@ -539,7 +539,7 @@
     "Befehl ausführen",/*Deutsch*/
     "", // TODO /*Slovenski*/
     "", // TODO /*Italiano*/
-    "Commando uitgevoerd", /*Nederlands*
+    "Commando uitgevoerd", /*Nederlands*/
     "", // TODO /*Português*/
     "Exécution d'une commande", /*Français*/
     "", // TODO /*Norsk*/

