--- a/src/knot/common.h
+++ b/src/knot/common.h
@@ -42,7 +42,7 @@ typedef unsigned int uint; /*!< \brief U
 
 #define PROJECT_EXEC SBINDIR "/" "knotd" /*!< \brief  Project executable. */
 #define ZONEPARSER_EXEC LIBEXECDIR "/" "knot-zcompile" /*!< \brief  Zoneparser executable. */
-#define PID_FILE "knot.pid" /*!< \brief Server PID file name. */
+#define PID_FILE "/var/run/knotd.pid" /*!< \brief Server PID file name. */
 
 /*
  * Server.
--- a/src/knot/conf/conf.c
+++ b/src/knot/conf/conf.c
@@ -198,7 +198,7 @@ static int conf_process(conf_t *conf)
 
 	// Create PID file
 	if (conf->pidfile == NULL) {
-		conf->pidfile = strcdup(conf->storage, "/" PID_FILE);
+		conf->pidfile = strdup(PID_FILE);
 		if (conf->pidfile == NULL) {
 			return KNOTD_ENOMEM;
 		}
