--- netgen-1.4.40/scripts/configure
+++ netgen-1.4.40/scripts/configure
@@ -6244,11 +6244,11 @@
       ;;
 
     *-linux*)
-      LDDL_FLAGS='-shared -Wl,-soname,$@'
+      LDDL_FLAGS+=' -shared -Wl,-soname,$@'
       if test "x${loader_run_path}" != "x" ; then
         LD_RUN_PATH="${LD_RUN_PATH} -Wl,-rpath,${loader_run_path}"
       fi
-      LDFLAGS=""
+      LDFLAGS+=" "
       EXTRA_LIB_SPECS="-ldl"
       ;;
 
--- netgen-1.4.40/tcltk/Makefile
+++ netgen-1.4.40/tcltk/Makefile
@@ -23,7 +23,7 @@
 	${CP} netgenexec${EXEEXT} $(DESTDIR)${TCLDIR}/netgenexec${EXEEXT}
 
 netgenexec${EXEEXT}: netgenexec.c
-	${CC} ${CFLAGS} ${CPPFLAGS} ${DFLAGS} netgenexec.c -o netgenexec${EXEEXT} \
+	${CC} ${CFLAGS} ${CPPFLAGS} ${DFLAGS} netgenexec.c -o netgenexec${EXEEXT} ${LDFLAGS} \
 		${LIBS} ${LIB_SPECS}
 
 netgen.tcl: netgen.tcl.in
--- netgen-1.4.40/rules.mak
+++ netgen-1.4.40/rules.mak
@@ -28,7 +28,7 @@
 ${MODULE}: lib${MODULE}.o ${EXTRA_LIBS}
 	@echo --- building main ${MODULE}
 	${RM} ${MODULE}
-	${CC} ${CFLAGS} ${CPPFLAGS} ${DFLAGS} lib${MODULE}.o ${EXTRA_LIBS} -o ${MODULE} ${LIBS}
+	${CC} ${CFLAGS} ${CPPFLAGS} ${DFLAGS} lib${MODULE}.o ${EXTRA_LIBS} -o ${MODULE} ${LIBS} ${LDFLAGS}
 
 clean:
 	${RM} ${CLEANS}
--- netgen-1.4.40/netgen/Makefile
+++ netgen-1.4.40/netgen/Makefile
@@ -10,7 +10,7 @@
 DFLAGS += ${GR_DFLAGS}
 DFLAGS += -DNETGEN_DATE="\"`date`\""
 
-LIBS += ${GR_LIBS} -lm
+LIBS += ${GR_LIBS} -lm -ltcl
 CFLAGS += ${GR_CFLAGS} -I${NETGENDIR}/base
 CLEANS += netgen netcomp ntk2adl inetcomp ntk2xnf
 
@@ -22,7 +22,7 @@
 	@echo --- making netgen Tcl library \(tclnetgen${SHDLIB_EXT}\)
 	${RM} tclnetgen${SHDLIB_EXT}
 	${CC} ${CFLAGS} ${CPPFLAGS} -o $@ ${LDDL_FLAGS} \
-		${LD_RUN_PATH} ${EXTRA_LIBS} ${LD_EXTRA_LIBS} -lc ${LIBS}
+		${LD_RUN_PATH} ${EXTRA_LIBS} ${LD_EXTRA_LIBS} -lc ${LIBS} ${LDFLAGS}
 
 $(DESTDIR)${BINDIR}/netgen:
 	${RM} $@
@@ -46,19 +46,19 @@
 
 netcomp: netcomp.c
 	${CC} ${CFLAGS} ${CPPFLAGS} ${DFLAGS} netcomp.c -o $@ ${EXTRA_LIBS} \
-		${LIBS}
+		${LIBS} ${LDFLAGS}
 
 inetcomp: inetcomp.c
 	${CC} ${CFLAGS} ${CPPFLAGS} ${DFLAGS} inetcomp.c -o $@ ${EXTRA_LIBS} \
-		${LIBS}
+		${LIBS} ${LDFLAGS}
 
 ntk2adl: ntk2adl.c
 	${CC} ${CFLAGS} ${CPPFLAGS} ${DFLAGS} ntk2adl.c -o $@ ${EXTRA_LIBS} \
-		${LIBS}
+		${LIBS} ${LDFLAGS}
 
 ntk2xnf: ntk2xnf.c
 	${CC} ${CFLAGS} ${CPPFLAGS} ${DFLAGS} ntk2xnf.c -o $@ ${EXTRA_LIBS} \
-		${LIBS}
+		${LIBS} ${LDFLAGS}
 
 install: $(DESTDIR)${BINDIR}/netgen $(DESTDIR)${BINDIR}/netcomp \
 		$(DESTDIR)${BINDIR}/inetcomp $(DESTDIR)${BINDIR}/ntk2adl \
--- netgen-1.4.40/scripts/defs.mak.in
+++ netgen-1.4.40/scripts/defs.mak.in
@@ -47,9 +47,9 @@
 
 RM                     = rm -f
 CP                     = cp
-AR                     = ar
+AR                     = @AR@
 ARFLAGS                = crv
-LINK		       = ld -r
+LINK		       = @LD@ -r
 LD                     = @LD@
 M4		       = @M4@
 RANLIB                 = @RANLIB@
