diff -ru tvtv-0.3.3-orig/update.c tvtv-0.3.3/update.c
--- tvtv-0.3.3-orig/update.c	2008-04-02 15:29:07.317447833 +0200
+++ tvtv-0.3.3/update.c	2008-04-02 15:30:37.827449337 +0200
@@ -519,7 +519,11 @@
 #else
           oss << "1:";
 #endif
+#if VDRVERSNUM >= 10600
+	  oss << *(oTVTVChannel->GetChannelID().ToString()) << ":";
+#else
 	  oss << oTVTVChannel->GetChannelID().ToString() << ":";
+#endif
 
 // Starting with VDR 1.3.23 timer entries support a full date in ISO notation. 
 #if VDRVERSNUM >= 10323
