diff -ur tipcutils-1.0.4.orig/Makefile tipcutils-1.0.4/Makefile
--- tipcutils-1.0.4.orig/Makefile	2006-10-12 17:46:49.000000000 +0300
+++ tipcutils-1.0.4/Makefile	2008-01-21 18:38:02.000000000 +0200
@@ -2,12 +2,12 @@
 # Makefile for tipcutils
 #
 
-ifndef KERNELDIR
-	KERNELDIR = /usr/src/linux
+ifndef KERNEL_DIR
+	KERNEL_DIR = /usr/src/linux
 endif
 
 VERSION = 1.0.4
-CFLAGS = -Wall -O2 -I${KERNELDIR}/include -D VERSION=\"${VERSION}\"
+CFLAGS += -Wall -D VERSION=\"${VERSION}\"
 
 all: tipc-config
 
