From f71b86486f5b42821fad50f8cada03edd3eb07e2 Mon Sep 17 00:00:00 2001
From: Dan Winship <danw@gnome.org>
Date: Thu, 23 Sep 2010 19:58:12 +0000
Subject: Run gtk-update-icon-cache after installing

https://bugzilla.gnome.org/show_bug.cgi?id=630465
---
diff --git a/Makefile.am b/Makefile.am
index 693a3cc..2097ee2 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -80,6 +80,9 @@ install-data-local:
 		touch $(DESTDIR)$(themedir); \
 	fi
 
+install-data-hook:
+	$(GTK_UPDATE_ICON_CACHE) -q $(DESTDIR)$(themedir)
+
 pkgconfigdir = $(datadir)/pkgconfig
 pkgconfig_DATA = gnome-icon-theme.pc
 
diff --git a/configure.ac b/configure.ac
index 6074f2c..0e7e3b1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -57,6 +57,11 @@ fi
 
 AC_SUBST(ICONMAP)
 
+AC_PATH_PROG(GTK_UPDATE_ICON_CACHE, gtk-update-icon-cache)
+if test -z "$GTK_UPDATE_ICON_CACHE"; then
+    AC_MSG_ERROR([Could not find gtk-update-icon-cache])
+fi
+
 AC_CONFIG_FILES([
 Makefile
 gnome-icon-theme.pc
--
cgit v0.8.3.4
