--- Lib/__init__.py
+++ Lib/__init__.py
@@ -34,8 +34,8 @@
 # DAMAGE.
 #
 #-----------------------------------------------------------------------------
-print """CXX is installed. 
+print("""CXX is installed. 
 The support files you need are in the PYTHON/etc/CXX directory. 
 The include files are in the distutils include path already.
 Just refer to them as "CXX/CXX_Objects.h", etc.
-"""
+""")
--- setup.py
+++ setup.py
@@ -12,13 +12,13 @@
 class my_install (install):
 
     def finalize_options (self):
-	if not self.install_data or (len(self.install_data) < 8) :
-	    self.install_data = "$base/share/python$py_version_short"
+        if not self.install_data or (len(self.install_data) < 8) :
+            self.install_data = "$base/share/python$py_version_short"
         install.finalize_options (self)
 
     def run (self):
-	self.distribution.data_files = [("CXX", sources)]
-	self.distribution.headers = headers
+        self.distribution.data_files = [("CXX", sources)]
+        self.distribution.headers = headers
         install.run (self)
 
 
