https://github.com/rack/rack/issues/492

--- test/spec_sendfile.rb.~1~	2013-01-07 19:49:40.000000000 +0100
+++ test/spec_sendfile.rb	2013-01-13 09:50:37.950890803 +0100
@@ -10,10 +10,10 @@
 end
 
 describe Rack::Sendfile do
   def sendfile_body
-    FileUtils.touch "/tmp/rack_sendfile"
+    FileUtils.touch "#{ENV['TMPDIR']}/rack_sendfile"
     res = ['Hello World']
-    def res.to_path ; "/tmp/rack_sendfile" ; end
+    def res.to_path ; "#{ENV['TMPDIR']}/rack_sendfile" ; end
     res
   end
 
@@ -44,7 +46,7 @@
       response.should.be.ok
       response.body.should.be.empty
       response.headers['Content-Length'].should.equal '0'
-      response.headers['X-Sendfile'].should.equal '/tmp/rack_sendfile'
+      response.headers['X-Sendfile'].should.equal "#{ENV['TMPDIR']}/rack_sendfile"
     end
   end
 
@@ -53,7 +55,7 @@
       response.should.be.ok
       response.body.should.be.empty
       response.headers['Content-Length'].should.equal '0'
-      response.headers['X-Lighttpd-Send-File'].should.equal '/tmp/rack_sendfile'
+      response.headers['X-Lighttpd-Send-File'].should.equal "#{ENV['TMPDIR']}/rack_sendfile"
     end
   end
 
@@ -60,7 +60,7 @@
   it "sets X-Accel-Redirect response header and discards body" do
     headers = {
       'HTTP_X_SENDFILE_TYPE' => 'X-Accel-Redirect',
-      'HTTP_X_ACCEL_MAPPING' => '/tmp/=/foo/bar/'
+      'HTTP_X_ACCEL_MAPPING' => "#{ENV['TMPDIR']}/=/foo/bar/"
     }
     request headers do |response|
       response.should.be.ok
