--- bin/wrapper.orig	2006-09-13 01:39:17.000000000 +0200
+++ bin/wrapper	2006-09-13 01:39:38.000000000 +0200
@@ -221,12 +221,6 @@
     exec >/dev/null 2>&1
 fi
 
-# FIXME - Only US English has been well tested to this point,
-#         so we unset all LOCALE strings, causing us to default to Posix and hence US English
- for x in `locale 2>/dev/null | sed 's/=.*$//'` ; do 
-    unset $x 
-done
-
 # Apply the .inf files (reapply if they've changed)
 for infcmd in wine.inf/DefaultInstall mozctl.inf/Mozctl picasa.inf/DefaultInstall ; do
     inffile=`echo $infcmd | cut -d '/' -f 1`
