From upstream CVS
http://bugs.gentoo.org/show_bug.cgi?id=353287

--- Makefile.comm
+++ Makefile.comm
@@ -21,7 +21,7 @@
 #
 INCLUDES=-I. -I$(srcdir) \
   -I$(top_builddir)/src/include -I$(top_srcdir)/src/include \
-  -I$(tob_builddir)/src/libs/gnulib/lib
+  -I$(top_builddir)/src/libs/gnulib/lib
 ALL_CCFLAGS=$(INCLUDES) $(CCDEFINES) $(CCFLAGS) $(CPPFLAGS)
 COMPILE.cpp=$(CCC) $(ALL_CCFLAGS) -c
 ALL_CFLAGS=$(INCLUDES) $(CDEFINES) $(CFLAGS) $(CPPFLAGS)
