diff -Naur lsat-0.9.5.orig/checkpkgs.c lsat-0.9.5/checkpkgs.c
--- lsat-0.9.5.orig/checkpkgs.c	2007-04-27 00:48:31.000000000 +0900
+++ lsat-0.9.5/checkpkgs.c	2007-04-28 10:11:17.000000000 +0900
@@ -113,14 +113,14 @@
     }
 
 
-    /* if distro = gentoo, use pkglist */
+    /* if distro = gentoo, use qpkg */
     if (distribution == 4)
     {
         if (verbose > 0)
         {
             printf(" Generating list of pkgs on system.\n");
         }
-        shellcode = "/usr/lib/portage/bin/pkglist 2>/dev/null >>/tmp/lsat1.lsat";
+        shellcode = "/usr/bin/qpkg -I -nc 2>/dev/null >>/tmp/lsat1.lsat";
         if ((dostuff(tempfile, 0, shellcode, 0, html)) < 0)
         {
             /* rhut-rho...something bad happened */
