https://bugs.gentoo.org/show_bug.cgi?id=546992#c10
--- a/print-cc.sh
+++ b/print-cc.sh
@@ -45,4 +45,5 @@ case "$cc:$systype" in
 esac
 
 cat warn-auto.sh
+echo echo "$cc" "$gccssl" '-c ${1+"$@"}'
 echo exec "$cc" "$gccssl" '-c ${1+"$@"}'
--- a/print-ld.sh
+++ b/print-ld.sh
@@ -3,4 +3,5 @@ systype="`cat systype`"
 
 cat warn-auto.sh
 echo 'main="$1"; shift'
+echo echo "$ld" '-o "$main" "$main".o ${1+"$@"}'
 echo exec "$ld" '-o "$main" "$main".o ${1+"$@"}'

