--- src/Makefile.in	2009-03-05 12:45:12.000000000 +0000
+++ src/Makefile.in	2009-03-05 12:52:44.000000000 +0000
@@ -133,7 +133,7 @@
 USE_NLS = @USE_NLS@
 VERSION = @VERSION@
 
-INCLUDES =  	@DRWRIGHT_CFLAGS@				-DGLADEDIR=\"$(datadir)/drwright\"		-DIMAGEDIR=\"$(datadir)/drwright\"		-Werror 
+INCLUDES =  	@DRWRIGHT_CFLAGS@				-DGLADEDIR=\"$(datadir)/drwright\"		-DIMAGEDIR=\"$(datadir)/drwright\"
 
 
 bin_PROGRAMS = drwright
