diff -burN mediatomb-0.12.1.old//src/metadata/ffmpeg_handler.cc mediatomb-0.12.1/src/metadata/ffmpeg_handler.cc
--- mediatomb-0.12.1.old//src/metadata/ffmpeg_handler.cc	2012-11-25 15:47:25.001638849 +0100
+++ mediatomb-0.12.1/src/metadata/ffmpeg_handler.cc	2012-11-25 15:52:15.926628259 +0100
@@ -374,7 +374,7 @@
     free(path_temp);
     return ret;
 }
-
+#ifdef HAVE_FFMPEGTHUMBNAILER
 static String getThumbnailCacheFilePath(String& movie_filename, bool create)
 {
     Ref<ConfigManager> cfg = ConfigManager::getInstance();
@@ -421,6 +421,8 @@
     fclose(fp);
 }
 
+#endif
+
 Ref<IOHandler> FfmpegHandler::serveContent(Ref<CdsItem> item, int resNum, off_t *data_size)
 {
     *data_size = -1;
Binary files mediatomb-0.12.1.old//src/metadata/.ffmpeg_handler.cc.swp and mediatomb-0.12.1/src/metadata/.ffmpeg_handler.cc.swp differ
