--- a/src/backend/Makefile.in	2007-12-27 17:05:38.000000000 +0100
+++ b/src/backend/Makefile.in	2009-04-02 09:07:53.000000000 +0200
@@ -1,11 +1,10 @@
-CC=@CC@ -O0
 AR=ar
 RANLIB=@RANLIB@
 
 system=@system@
 HAVE_NVCONTROL=@HAVE_NVCONTROL@
 OBJECTS=backend.o bios.o config.o error.o nv30.o nv40.o nv50.o info.o overclock.o utils.o i2c.o xf86i2c.o adt7473.o f75375.o lm99.o w83781d.o w83l785r.o libc_wrapper.o
-CFLAGS=@X11_CFLAGS@ -I../.. -I../nvcontrol
+CFLAGS+=@X11_CFLAGS@ -I../.. -I../nvcontrol
 
 ifeq ($(system), FreeBSD)
     OBJECTS+=back_bsd.o
@@ -36,4 +35,4 @@
 	rm -f Makefile
 
 install:
-uninstall:
\ No newline at end of file
+uninstall:
--- a/src/gtk/Makefile.in	2008-05-11 12:25:46.000000000 +0200
+++ b/src/gtk/Makefile.in	2009-04-02 09:12:10.000000000 +0200
@@ -1,7 +1,6 @@
-CC=@CC@
 INSTALL=@INSTALL@
 
-CFLAGS=@GTK_CFLAGS@ -I../backend -I../nvcontrol -I../..
+CFLAGS+=@GTK_CFLAGS@ -I../backend -I../nvcontrol -I../..
 HAVE_GTK2=@HAVE_GTK2@
 HAVE_NVCONTROL=@HAVE_NVCONTROL@
 libbackend=../backend/libbackend.a
@@ -34,7 +33,7 @@
 all: $(PROGRAM)
 
 nvclock_gtk: $(DEPS) $(OBJECTS)
-	$(CC) -o nvclock_gtk $(OBJECTS) $(libbackend) $(LIBS)
+	$(CC) -o nvclock_gtk $(LDFLAGS) $(OBJECTS) $(libbackend) $(LIBS)
 
 clean:
 	rm -f nvclock_gtk *.core *.o *.exe
--- a/src/Makefile.in	2009-01-03 18:02:54.000000000 +0100
+++ b/src/Makefile.in	2009-04-02 09:12:24.000000000 +0200
@@ -1,4 +1,3 @@
-CC=@CC@
 INSTALL=@INSTALL@
 
 nvclock_SOURCES=nvclock.c
@@ -31,10 +30,10 @@
 
 
 nvclock: $(DEPS) $(LIB) nvclock.c
-	$(CC) -o nvclock $(nvclock_SOURCES) $(INCLUDES) $(LIBS)
+	$(CC) -o nvclock $(CFLAGS) $(LDFLAGS) $(nvclock_SOURCES) $(INCLUDES) $(LIBS)
 
 smartdimmer: $(DEPS) $(LIB) smartdimmer.c
-	$(CC) -o smartdimmer $(smartdimmer_SOURCES) $(INCLUDES) $(LIBS)
+	$(CC) -o smartdimmer $(CFLAGS) $(LDFLAGS) $(smartdimmer_SOURCES) $(INCLUDES) $(LIBS)
 
 
 clean: clean-recursive clean-app
@@ -50,8 +49,8 @@
 install: install-recursive install-app
 install-app:
 	mkdir -p $(bindir)
-	$(INSTALL) -c nvclock $(DESTDIR)$(bindir)/nvclock
-	$(INSTALL) -c smartdimmer $(DESTDIR)$(bindir)/smartdimmer
+	$(INSTALL) -c nvclock $(bindir)/nvclock
+	$(INSTALL) -c smartdimmer $(bindir)/smartdimmer
 
 
 uninstall: uninstall-recursive uninstall-app
--- a/src/nvcontrol/Makefile.in	2005-12-09 20:46:05.000000000 +0100
+++ b/src/nvcontrol/Makefile.in	2009-04-02 09:07:53.000000000 +0200
@@ -1,8 +1,7 @@
-CC=@CC@
 AR=ar
 RANLIB=@RANLIB@
 system=@system@
-CFLAGS=@X11_CFLAGS@ -I../backend
+CFLAGS+=@X11_CFLAGS@ -I../backend
 HAVE_X11=@HAVE_NVCONTROL@
 OBJECTS=libnvcontrol.o nvcontrol.o
 
@@ -26,4 +25,4 @@
 	rm -f Makefile
 
 install:
-uninstall:
\ No newline at end of file
+uninstall:
--- a/src/qt/Makefile.in	2008-05-11 12:25:46.000000000 +0200
+++ b/src/qt/Makefile.in	2009-04-02 09:11:51.000000000 +0200
@@ -1,9 +1,8 @@
-CXX=@CXX@
 INSTALL=@INSTALL@
 MOC=@MOC@
 
 OBJECTS=main.o qt_nvclock.o qt_nvclock.moc.o qt_xfree.o qt_xfree.moc.o
-CXXFLAGS=@QT_CFLAGS@ -I../backend
+CXXFLAGS+=@QT_CFLAGS@ -I../backend
 HAVE_NVCONTROL=@HAVE_NVCONTROL@
 HAVE_QT=@HAVE_QT@
 libbackend=../backend/libbackend.a
@@ -32,7 +31,7 @@
 all: $(PROGRAM)
 
 nvclock_qt: $(DEPS) qt_nvclock.moc qt_xfree.moc $(OBJECTS)
-	$(CXX) -o nvclock_qt $(CXXFLAGS) $(OBJECTS) $(LIBS) 
+	$(CXX) -o nvclock_qt $(CXXFLAGS) $(LDFLAGS) $(OBJECTS) $(LIBS) 
 
 clean:
 	rm -f nvclock_qt *.core *.moc *.o
