--- a/src/backend/bios.c
+++ b/src/backend/bios.c
@@ -891,6 +891,8 @@ int main(int argc, char **argv)
 
 
 #else
+int load_bios_pramin(char *data);
+int load_bios_prom(char *data);
 void dump_bios(const char *filename)
 {
 	int i;

--- a/src/backend/libc_wrapper.c
+++ b/src/backend/libc_wrapper.c
@@ -25,6 +25,7 @@
  *
  */
 
+#include <config.h>
 #include <stdio.h>
 #include "xfree.h"
 

--- a/src/backend/nv40.c
+++ b/src/backend/nv40.c
@@ -68,6 +68,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <unistd.h>
 #include "backend.h"
 
 /*

--- a/src/gtk/banner.c
+++ b/src/gtk/banner.c
@@ -112,7 +112,7 @@ void banner_set_text(Banner *banner, char *text, int type)
 	if(banner->text)
 		g_free(banner->text);
 
-	banner->text = g_strdup_printf(text);
+	banner->text = g_strdup(text);
 	banner->type = type;
 	/* Refresh the banner */
 	banner_expose(GTK_WIDGET(banner), NULL, banner);

--- a/src/gtk/main.c
+++ b/src/gtk/main.c
@@ -25,6 +25,7 @@
 
 #include <stdlib.h>
 #include <gtk/gtk.h>
+#include <X11/Xlib.h>
 
 #include "banner.h"
 #include "interface.h"

--- a/src/smartdimmer.c
+++ b/src/smartdimmer.c
@@ -38,6 +38,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <getopt.h>
+#include <ctype.h>
 #include "nvclock.h"
 
 /* for command line arguments */
