--- configure.ac.orig	2010-10-05 18:28:18.000000000 +0200
+++ configure.ac	2011-03-24 18:35:57.000000000 +0100
@@ -20,7 +20,7 @@
 AC_CHECK_HEADERS([libintl.h netdb.h netinet/in.h stdlib.h string.h sys/ioctl.h sys/socket.h sys/time.h sys/timeb.h unistd.h])
 
 # We need GTK
-pkg_modules="gtk+-2.0 >= 2.4 glib-2.0 >= 2.4 gthread-2.0 >= 2.4"
+pkg_modules="gtk+-2.0 >= 2.4 glib-2.0 >= 2.4 gthread-2.0 >= 2.4 gdk-pixbuf-2.0"
 PKG_CHECK_MODULES(DEPS, [$pkg_modules])
 
 # Checks for library functions.
