From: Julian Ospald <hasufell@gentoo.org>
Date: Tue Apr 22 12:10:46 UTC 2014
Subject: fix build failure

--- a/src/antimicro.pro
+++ b/src/antimicro.pro
@@ -11,8 +11,12 @@
         packagesExist(sdl2) {
             USE_SDL_2 = 1
             FOUND_SDL = 1
+			CONFIG+=link_pkgconfig
+			PKGCONFIG+=sdl2
         } else:packageExist(sdl) {
             FOUND_SDL = 1
+			CONFIG+=link_pkgconfig
+			PKGCONFIG+=sdl
         } else {
             error("SDL library was not found")
         }
