From: Julian Ospald <hasufell@gentoo.org>
Date: Sat Dec 14 21:56:15 UTC 2013
Subject: fix desktop file

--- a/other/antimicro.desktop
+++ b/other/antimicro.desktop
@@ -4,7 +4,7 @@
 Name[sr]=Анти-микро
 Comment[sr]=Користите џојстик или играћу тастатуру за управљање различитим програмима
 Name[fr]=AntiMicro
-Name[fr]=Utilisez une manette de jeu pour commander un logiciel
+Comment[fr]=Utilisez une manette de jeu pour commander un logiciel
 Exec=antimicro
 Icon=antimicro
 StartupNotify=true
