--- libjacksms/DataTypesApi.cpp.old	2012-05-10 19:09:21.114626223 +0200
+++ libjacksms/DataTypesApi.cpp	2012-05-10 19:10:37.047647425 +0200
@@ -25,6 +25,9 @@
 #include <QString>
 #include <QRegExp>
 
+#undef major
+#undef minor
+
 namespace libJackSMS {
 
     namespace dataTypesApi {
--- libjacksms/netclientqhttp.cpp.old	2012-08-09 18:53:14.936366832 +0200
+++ libjacksms/netclientqhttp.cpp	2012-08-09 18:54:19.539559180 +0200
@@ -276,6 +276,7 @@
             if (userAgentSetted && !overwrite)
                 r.setRawHeader("User-Agent",ua.toUtf8());
             makeQueryString(false);
+             r.setHeader(QNetworkRequest::ContentTypeHeader, "application/x-www-form-urlencoded");
             reply=request.post(r, lastQueryString.toUtf8());
             loop->exec(QEventLoop::ExcludeUserInputEvents);
             errorPage=error;
