diff -ruN SDK.orig/samples/binarySegmentation/Makefile SDK/samples/binarySegmentation/Makefile
--- SDK.orig/samples/binarySegmentation/Makefile	2010-09-17 22:12:00.000000000 +0200
+++ SDK/samples/binarySegmentation/Makefile	2011-05-14 20:00:00.000000000 +0200
@@ -103,7 +103,7 @@
 endif
 
 all:
-	$(CXX) $(INC) $(LIB) -o binarySegmentation binarySegmentation.cpp -lUtilNPP$(LIBARCH) -lfreeimage$(FREEIMAGELIBARCH)
+	$(CXX) $(INC) -o binarySegmentation binarySegmentation.cpp -lUtilNPP$(LIBARCH) -lfreeimage$(FREEIMAGELIBARCH) $(LIB)
 
 clean:
 	@rm -f binarySegmentation
diff -ruN SDK.orig/samples/boxFilter/Makefile SDK/samples/boxFilter/Makefile
--- SDK.orig/samples/boxFilter/Makefile	2010-09-17 20:07:19.000000000 +0200
+++ SDK/samples/boxFilter/Makefile	2011-05-14 19:58:34.000000000 +0200
@@ -103,7 +103,7 @@
 endif
 
 all:
-	$(CXX) $(INC) $(LIB) -o boxFilter boxFilter.cpp -lUtilNPP$(LIBARCH) -lfreeimage$(FREEIMAGELIBARCH)
+	$(CXX) $(INC) -o boxFilter boxFilter.cpp -lUtilNPP$(LIBARCH) -lfreeimage$(FREEIMAGELIBARCH) $(LIB)
 
 clean:
 	@rm -f boxFilter
diff -ruN SDK.orig/samples/freeImageInterop/Makefile SDK/samples/freeImageInterop/Makefile
--- SDK.orig/samples/freeImageInterop/Makefile	2010-09-17 20:26:32.000000000 +0200
+++ SDK/samples/freeImageInterop/Makefile	2011-05-14 19:59:21.000000000 +0200
@@ -103,7 +103,7 @@
 endif
 
 all:
-	$(CXX) $(INC) $(LIB) -o freeImageInterop freeImageInterop.cpp -lUtilNPP$(LIBARCH) -lfreeimage$(FREEIMAGELIBARCH)
+	$(CXX) $(INC) -o freeImageInterop freeImageInterop.cpp -lUtilNPP$(LIBARCH) -lfreeimage$(FREEIMAGELIBARCH) $(LIB)
 
 clean:
 	@rm -f freeImageInterop
diff -ruN SDK.orig/samples/histEqualization/Makefile SDK/samples/histEqualization/Makefile
--- SDK.orig/samples/histEqualization/Makefile	2010-09-17 22:05:24.000000000 +0200
+++ SDK/samples/histEqualization/Makefile	2011-05-14 20:00:23.000000000 +0200
@@ -105,7 +105,7 @@
 
 
 all:
-	$(CXX) $(INC) $(LIB) -o histEqualization histEqualization.cpp -lUtilNPP$(LIBARCH) -lfreeimage$(FREEIMAGELIBARCH)
+	$(CXX) $(INC) -o histEqualization histEqualization.cpp -lUtilNPP$(LIBARCH) -lfreeimage$(FREEIMAGELIBARCH) $(LIB)
 
 clean:
 	@rm -f histEqualization
