Index: res-3.2.0/lib/Makefile
===================================================================
--- res-3.2.0.orig/lib/Makefile
+++ res-3.2.0/lib/Makefile
@@ -11,7 +11,7 @@ LIBINSTALL_FILES = res.mli strat.mli pre
                    strat.cmi pres_intf.cmi nopres_intf.cmi weak_intf.cmi \
 		   res.cmi res.cma res.cmxa res.a
 
-all:	byte-code-library native-code-library
+all:	byte-code-library
 
 install:	libinstall
 uninstall:	libuninstall
