diff -Naur stunnel-4.56.orig/src/options.c stunnel-4.56/src/options.c
--- stunnel-4.56.orig/src/options.c	2013-03-13 09:41:12.000000000 -0400
+++ stunnel-4.56/src/options.c	2013-06-16 11:17:49.000000000 -0400
@@ -1913,6 +1913,24 @@
         break;
     }
 
+    /* listenqueue */
+    switch(cmd) {
+    case CMD_BEGIN:
+        section->listenqueue=SOMAXCONN;
+        break;
+    case CMD_EXEC:
+        if(strcasecmp(opt, "listenqueue"))
+            break;
+        section->listenqueue=atoi(arg);
+        return (section->listenqueue?NULL:"Bad verify level");
+    case CMD_DEFAULT:
+        s_log(LOG_NOTICE, "%-15s = %d", "listenqueue", SOMAXCONN);
+        break;
+    case CMD_HELP:
+        s_log(LOG_NOTICE, "%-15s = defines the maximum length the queue of pending connections may grow to", "listenqueue");
+        break;
+    }
+
     if(cmd==CMD_EXEC)
         return option_not_found;
 
diff -Naur stunnel-4.56.orig/src/prototypes.h stunnel-4.56/src/prototypes.h
--- stunnel-4.56.orig/src/prototypes.h	2013-03-19 13:30:55.000000000 -0400
+++ stunnel-4.56/src/prototypes.h	2013-06-16 11:17:49.000000000 -0400
@@ -183,6 +183,7 @@
     int timeout_close;                          /* maximum close_notify time */
     int timeout_connect;                           /* maximum connect() time */
     int timeout_idle;                        /* maximum idle connection time */
+    int listenqueue;										/* Listen baklog */
     enum {FAILOVER_RR, FAILOVER_PRIO} failover;         /* failover strategy */
 
         /* service-specific data for protocol.c */
diff -Naur stunnel-4.56.orig/src/stunnel.c stunnel-4.56/src/stunnel.c
--- stunnel-4.56.orig/src/stunnel.c	2013-03-19 13:30:34.000000000 -0400
+++ stunnel-4.56/src/stunnel.c	2013-06-16 11:17:49.000000000 -0400
@@ -388,7 +388,7 @@
                 str_free(local_address);
                 return 1;
             }
-            if(listen(opt->fd, SOMAXCONN)) {
+            if(listen(opt->fd, opt->listenqueue)) {
                 sockerror("listen");
                 closesocket(opt->fd);
                 str_free(local_address);
