diff -up tkimg1.4/zlib/zlibtcl.decls.gzgetc_fix tkimg1.4/zlib/zlibtcl.decls
--- tkimg1.4/zlib/zlibtcl.decls.gzgetc_fix	2012-07-31 14:54:11.104302862 -0400
+++ tkimg1.4/zlib/zlibtcl.decls	2012-07-31 14:54:24.846303790 -0400
@@ -138,7 +138,7 @@ declare 38 {
     int gzputc(gzFile file, int c)
 }
 declare 39 {
-    int gzgetc(gzFile file)
+    int gzgetc_(gzFile file)
 }
 declare 40 {
     int gzflush(gzFile file, int flush)
diff -up tkimg1.4/zlib/zlibtclDecls.h.gzgetc_fix tkimg1.4/zlib/zlibtclDecls.h
--- tkimg1.4/zlib/zlibtclDecls.h.gzgetc_fix	2012-07-31 14:55:12.754307116 -0400
+++ tkimg1.4/zlib/zlibtclDecls.h	2012-07-31 14:55:17.520307957 -0400
@@ -179,7 +179,7 @@ typedef struct ZlibtclStubs {
     int (*gzputsPtr) (gzFile file, const char *s); /* 36 */
     char * (*gzgetsPtr) (gzFile file, char *buf, int len); /* 37 */
     int (*gzputcPtr) (gzFile file, int c); /* 38 */
-    int (*gzgetcPtr) (gzFile file); /* 39 */
+    int (*gzgetc_Ptr) (gzFile file); /* 39 */
     int (*gzflushPtr) (gzFile file, int flush); /* 40 */
     z_off_t (*gzseekPtr) (gzFile file, z_off_t offset, int whence); /* 41 */
     int (*gzrewindPtr) (gzFile file); /* 42 */
@@ -273,8 +273,8 @@ ZEXTERN const ZlibtclStubs *zlibtclStubs
 	(zlibtclStubsPtr->gzgetsPtr) /* 37 */
 #define gzputc \
 	(zlibtclStubsPtr->gzputcPtr) /* 38 */
-#define gzgetc \
-	(zlibtclStubsPtr->gzgetcPtr) /* 39 */
+#define gzgetc_ \
+	(zlibtclStubsPtr->gzgetc_Ptr) /* 39 */
 #define gzflush \
 	(zlibtclStubsPtr->gzflushPtr) /* 40 */
 #define gzseek \
