diff -up tkimg1.4/libtiff/configure.libtiff tkimg1.4/libtiff/configure
--- tkimg1.4/libtiff/configure.libtiff	2010-07-07 04:09:46.000000000 -0400
+++ tkimg1.4/libtiff/configure	2011-08-01 15:49:04.039079729 -0400
@@ -5183,17 +5183,6 @@ mv -f tifftcl_confdefs.h confdefs.h
 
 
     vars="
-	../compat/libtiff/libtiff/tif_aux.c		../compat/libtiff/libtiff/tif_close.c		../compat/libtiff/libtiff/tif_codec.c
-	../compat/libtiff/libtiff/tif_compress.c	../compat/libtiff/libtiff/tif_dir.c		../compat/libtiff/libtiff/tif_dirinfo.c
-	../compat/libtiff/libtiff/tif_dirread.c	../compat/libtiff/libtiff/tif_dirwrite.c	../compat/libtiff/libtiff/tif_dumpmode.c
-	../compat/libtiff/libtiff/tif_error.c		../compat/libtiff/libtiff/tif_fax3.c
-	../compat/libtiff/libtiff/tif_getimage.c	../compat/libtiff/libtiff/tif_flush.c		../compat/libtiff/libtiff/tif_luv.c
-	../compat/libtiff/libtiff/tif_lzw.c		../compat/libtiff/libtiff/tif_next.c		../compat/libtiff/libtiff/tif_open.c
-	../compat/libtiff/libtiff/tif_packbits.c	../compat/libtiff/libtiff/tif_predict.c	../compat/libtiff/libtiff/tif_print.c
-	../compat/libtiff/libtiff/tif_read.c		../compat/libtiff/libtiff/tif_swab.c		../compat/libtiff/libtiff/tif_strip.c
-	../compat/libtiff/libtiff/tif_thunder.c	../compat/libtiff/libtiff/tif_tile.c		../compat/libtiff/libtiff/tif_version.c
-	../compat/libtiff/libtiff/tif_warning.c	../compat/libtiff/libtiff/tif_write.c
-	../compat/libtiff/libtiff/tif_color.c		../compat/libtiff/libtiff/tif_extension.c
 "
     for i in $vars; do
 	case $i in
@@ -5574,7 +5563,7 @@ if test "${TEA_PLATFORM}" = "windows" ; 
 else
     CLEANFILES=""
 
-    vars="../compat/libtiff/libtiff/tif_unix.c"
+    vars=""
     for i in $vars; do
 	case $i in
 	    \$*)
@@ -6053,7 +6042,7 @@ $as_echo "$tcl_cv_cc_visibility_hidden" 
     if test $tcl_cv_cc_visibility_hidden = yes; then :
 
 
-$as_echo "#define MODULE_SCOPE extern __attribute__((__visibility__(\"hidden\")))" >>confdefs.h
+$as_echo "#define MODULE_SCOPE extern" >>confdefs.h
 
 
 fi
diff -up tkimg1.4/libtiff/Makefile.in.libtiff tkimg1.4/libtiff/Makefile.in
--- tkimg1.4/libtiff/Makefile.in.libtiff	2010-07-07 07:08:08.000000000 -0400
+++ tkimg1.4/libtiff/Makefile.in	2011-08-01 15:49:04.040079716 -0400
@@ -35,8 +35,8 @@
 
 # tif_fax3sm is special. It is generated.
 
-PKG_SOURCES	= @PKG_SOURCES@ tif_fax3sm.c
-PKG_OBJECTS	= @PKG_OBJECTS@ tif_fax3sm.$(OBJEXT)
+PKG_SOURCES	= @PKG_SOURCES@
+PKG_OBJECTS	= @PKG_OBJECTS@
 
 PKG_STUB_SOURCES = @PKG_STUB_SOURCES@
 PKG_STUB_OBJECTS = @PKG_STUB_OBJECTS@
@@ -109,7 +109,7 @@ RANLIB		= @RANLIB@
 RANLIB_STUB	= @RANLIB_STUB@
 SHLIB_CFLAGS	= @SHLIB_CFLAGS@
 SHLIB_LD	= @SHLIB_LD@
-SHLIB_LD_LIBS	= @SHLIB_LD_LIBS@
+SHLIB_LD_LIBS	= @SHLIB_LD_LIBS@ -ltiff
 STLIB_LD	= @STLIB_LD@
 TCL_DEFS	= @TCL_DEFS@
 TCL_BIN_DIR	= @TCL_BIN_DIR@
diff -up tkimg1.4/libtiff/tifftclDecls.h.libtiff tkimg1.4/libtiff/tifftclDecls.h
--- tkimg1.4/libtiff/tifftclDecls.h.libtiff	2010-07-07 07:08:08.000000000 -0400
+++ tkimg1.4/libtiff/tifftclDecls.h	2011-08-01 15:49:04.041079703 -0400
@@ -34,9 +34,13 @@ EXTERN int Tifftcl_SafeInit(Tcl_Interp *
  */
 #undef INLINE
 
-#include "../compat/libtiff/libtiff/tiffio.h"
-#include "../compat/libtiff/libtiff/tiffiop.h"
-#include "../compat/libtiff/libtiff/tif_predict.h"
+#include <@GENTOO_PORTAGE_EPREFIX@/usr/include/tiffio.h>
+
+/*
+ * Typedefs for ``method pointers'' used internally.
+ */
+typedef unsigned char tidataval_t;	/* internal image data value type */
+typedef tidataval_t* tidata_t;          /* reference to internal image data */
 
 /* !BEGIN!: Do not edit below this line. */
 
diff -up tkimg1.4/tiff/configure.libtiff tkimg1.4/tiff/configure
--- tkimg1.4/tiff/configure.libtiff	2010-07-13 11:44:38.000000000 -0400
+++ tkimg1.4/tiff/configure	2011-08-01 15:49:04.044079664 -0400
@@ -5753,7 +5753,7 @@ fi
 #-----------------------------------------------------------------------
 
 
-    vars="tiff.c tiffJpeg.c tiffZip.c tiffPixar.c"
+    vars="tiff.c"
     for i in $vars; do
 	case $i in
 	    \$*)
diff -up tkimg1.4/tiff/tiff.c.libtiff tkimg1.4/tiff/tiff.c
--- tkimg1.4/tiff/tiff.c.libtiff	2010-07-08 06:41:55.000000000 -0400
+++ tkimg1.4/tiff/tiff.c	2011-08-01 15:51:27.451272526 -0400
@@ -124,14 +124,14 @@ SetupTiffLibrary (interp)
 	  if (Zlibtcl_InitStubs(interp, ZLIBTCL_VERSION, 0) == NULL) {
 	    return TCL_ERROR;
 	  }
-	  TIFFRegisterCODEC (COMPRESSION_DEFLATE,  "Deflate",  TkimgTIFFInitZip);
-	  TIFFRegisterCODEC (COMPRESSION_ADOBE_DEFLATE, "AdobeDeflate", TkimgTIFFInitZip);
+	  TIFFRegisterCODEC (COMPRESSION_DEFLATE,  "Deflate", TIFFInitZIP);
+	  TIFFRegisterCODEC (COMPRESSION_ADOBE_DEFLATE, "AdobeDeflate", TIFFInitZIP);
 
 	  if (Jpegtcl_InitStubs(interp, JPEGTCL_VERSION, 0) == NULL) {
 	    return TCL_ERROR;
 	  }
-	  TIFFRegisterCODEC (COMPRESSION_JPEG,     "JPEG",     TkimgTIFFInitJpeg);
-	  TIFFRegisterCODEC (COMPRESSION_PIXARLOG, "PixarLog", TkimgTIFFInitPixar);
+	  TIFFRegisterCODEC (COMPRESSION_JPEG,     "JPEG",     TIFFInitJPEG);
+	  TIFFRegisterCODEC (COMPRESSION_PIXARLOG, "PixarLog", TIFFInitPixarLog);
 	}
     }
     return TCL_OK;
