diff -urN libzip-0.10.old/libzip.pc.in libzip-0.10/libzip.pc.in
--- libzip-0.10.old/libzip.pc.in	2011-04-29 12:54:23.000000000 +0200
+++ libzip-0.10/libzip.pc.in	2011-04-29 12:55:28.000000000 +0200
@@ -2,7 +2,6 @@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
 includedir=@includedir@
-libincludedir=@libdir@/libzip/include
 
 zipcmp=@prefix@/bin/zipcmp
 
@@ -10,5 +9,5 @@
 Description: library for handling zip archives
 Version: @VERSION@
 Libs: -L${libdir} -lzip @LIBS@
-Cflags: -I${includedir} -I${libincludedir}
+Cflags: -I${includedir}
 
