--- modules/afsmtp/Makefile.am.orig	2014-11-09 02:47:00.110145300 -0500
+++ modules/afsmtp/Makefile.am	2014-11-09 02:47:13.996684666 -0500
@@ -12,7 +12,7 @@
 	modules/afsmtp/afsmtp-parser.c		\
 	modules/afsmtp/afsmtp-parser.h
 modules_afsmtp_libafsmtp_la_LIBADD	=	\
-	$(LIBESMTP_LIBS) $(MODULE_DEPS_LIBS) $(CRYPTO_LIBS)
+	$(MODULE_DEPS_LIBS) $(LIBESMTP_LIBS) $(CRYPTO_LIBS)
 modules_afsmtp_libafsmtp_la_LDFLAGS	=	\
 	$(MODULE_LDFLAGS)
 modules_afsmtp_libafsmtp_la_DEPENDENCIES=	\
