From a160eedd0b176cfa3da755762a8e40a38b753114 Mon Sep 17 00:00:00 2001
From: Paul Varner <fuzzyray@gentoo.org>
Date: Mon, 19 Dec 2011 13:09:42 -0600
Subject: [PATCH] Fix setup.py to work with EPREFIX changes in Portage.

---
 setup.py |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/setup.py b/setup.py
index 2990e33..b24cd8f 100755
--- a/setup.py
+++ b/setup.py
@@ -22,7 +22,7 @@ cwd = os.getcwd()
 try: 
 	from portage.const import EPREFIX 
 except ImportError: 
-	EPREFIX='/' 
+	EPREFIX='' 
 
 # Python files that need `__version__ = ""` subbed, relative to this dir:
 python_scripts = [os.path.join(cwd, path) for path in (
@@ -80,9 +80,9 @@ core.setup(
 	packages=packages,
 	scripts=(glob('bin/*')),
 	data_files=(
-		(os.path.join(EPREFIX, 'usr/share/man/man1'), glob('man/en/*')),
-		(os.path.join(EPREFIX, 'usr/share/man/fr/man1'), glob('man/fr/*')),
-		(os.path.join(EPREFIX, 'usr/share/man/it/man1'), glob('man/it/*')),
+		(os.path.join(os.sep, EPREFIX.lstrip(os.sep), 'usr/share/man/man1'), glob('man/en/*')),
+		(os.path.join(os.sep, EPREFIX.lstrip(os.sep), 'usr/share/man/fr/man1'), glob('man/fr/*')),
+		(os.path.join(os.sep, EPREFIX.lstrip(os.sep), 'usr/share/man/it/man1'), glob('man/it/*')),
 	),
 	cmdclass={
 		'set_version': set_version,
-- 
1.7.8

