From 8399831cd2f5968fc45d0f1dc4e131368161e39a Mon Sep 17 00:00:00 2001
From: Brian Dolbec <dolsen@gentoo.org>
Date: Fri, 9 Dec 2011 23:58:52 -0800
Subject: [PATCH] apply zmedico's patch to fix EPREFIX capability.

---
 esearch/search.py |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/esearch/search.py b/esearch/search.py
index 8498687..f296cb7 100755
--- a/esearch/search.py
+++ b/esearch/search.py
@@ -15,7 +15,7 @@ from os import listdir, getenv, system
 from os.path import isdir, exists
 import re
 
-sys.path.insert(0, "/usr/lib/portage/pym")
+#sys.path.insert(0, "/usr/lib/portage/pym")
 # commented out so it can run from the git checkout
 #sys.path.insert(0, "/usr/lib/esearch")
 
-- 
1.7.8

