--- semantik-0.8.2.orig/wscript	2012-09-18 00:01:40.000000000 +0300
+++ semantik-0.8.2/wscript	2012-09-26 20:25:48.988264096 +0300
@@ -213,9 +213,6 @@
 	opt.add_option('--use64', action='store_true', default=False, help='set the installation into lib+64 (configuration)')
 
 def post_build(bld):
-	if bld.cmd == 'install':
-		try: bld.exec_command('/sbin/ldconfig 2> /dev/null')
-		except: pass
 	if Options.options.exe:
 		#os.popen('export LD_LIBRARY_PATH=out/default/:$LD_LIBRARY_PATH; PATH=plugins:$PATH out/default/src/semantik')
 		bld.exec_command('LD_LIBRARY_PATH=build/:$LD_LIBRARY_PATH build/src/semantik --style plastique', stdout=None, stderr=None)
