diff -U 3 -dHrN TOPCOM-0.16.2.orig/configure.in TOPCOM-0.16.2/configure.in
--- TOPCOM-0.16.2.orig/configure.in	2009-10-15 09:00:51.000000000 +0200
+++ TOPCOM-0.16.2/configure.in	2010-11-09 18:02:19.000000000 +0100
@@ -17,12 +17,8 @@
 dnl AC_CHECK_LIB(stdc, main)
 dnl Replace `main' with a function in -lstdc++:
 dnl AC_CHECK_LIB(stdc++, main)
-dnl Make libgmp:
-make -C external gmp
-dnl Make libcddgmp:
-make -C external cdd
 dnl Check for soplex:
-AC_CHECK_LIB(soplex, main, , csh -c 'echo soplex not found: --soplex will not work.')
+AC_CHECK_LIB(soplex, main, , /bin/sh -c 'echo soplex not found: --soplex will not work.')
 
 dnl Checks for header files.
 AC_HEADER_STDC
diff -U 3 -dHrN TOPCOM-0.16.2.orig/Makefile.am TOPCOM-0.16.2/Makefile.am
--- TOPCOM-0.16.2.orig/Makefile.am	2009-10-15 07:36:24.000000000 +0200
+++ TOPCOM-0.16.2/Makefile.am	2010-11-09 18:02:19.000000000 +0100
@@ -1,4 +1,2 @@
 SUBDIRS      = wrap-gmp-gmpxx lib-src-reg lib-src src-reg src examples
-EXTRA_DIST   = external/Makefile
-EXTRA_DIST  += external/gmp-4.3.0.tar.gz
-EXTRA_DIST  += external/cddlib-0.94f-TOPCOM.tar.gz
+
diff -U 3 -dHrN TOPCOM-0.16.2.orig/src/Makefile.am TOPCOM-0.16.2/src/Makefile.am
--- TOPCOM-0.16.2.orig/src/Makefile.am	2009-07-03 12:56:30.000000000 +0200
+++ TOPCOM-0.16.2/src/Makefile.am	2010-11-09 18:03:25.000000000 +0100
@@ -76,10 +76,7 @@
 santos_22_triang_SOURCES           = santos_22_triang.cc
 
 LDADD           = ../lib-src/libTOPCOM.a \
-                  ../lib-src-reg/libCHECKREG.a \
-                  ../external/lib/libcddgmp.a \
-                  ../external/lib/libgmpxx.a \
-                  ../external/lib/libgmp.a
+                  ../lib-src-reg/libCHECKREG.a 
 
 INCLUDES        = -I../lib-src
 INCLUDES       += -I../lib-src-reg
diff -U 3 -dHrN TOPCOM-0.16.2.orig/src-reg/Makefile.am TOPCOM-0.16.2/src-reg/Makefile.am
--- TOPCOM-0.16.2.orig/src-reg/Makefile.am	2009-07-03 12:56:33.000000000 +0200
+++ TOPCOM-0.16.2/src-reg/Makefile.am	2010-11-09 18:02:40.000000000 +0100
@@ -3,10 +3,7 @@
 checkregularity_SOURCES = checkregularity.cc
 
 LDADD           = ../lib-src-reg/libCHECKREG.a \
-                  ../lib-src/libTOPCOM.a \
-                  ../external/lib/libcddgmp.a \
-                  ../external/lib/libgmpxx.a \
-                  ../external/lib/libgmp.a
+                  ../lib-src/libTOPCOM.a
 
 INCLUDES        = -I../lib-src
 INCLUDES       += -I../lib-src-reg
