diff -urN libsmtp-0.8.orig/examples/multipart.c libsmtp-0.8/examples/multipart.c
--- libsmtp-0.8.orig/examples/multipart.c	2004-12-22 08:22:05.415981248 -0500
+++ libsmtp-0.8/examples/multipart.c	2004-12-22 08:33:32.158580504 -0500
@@ -341,4 +341,5 @@
   /* Free the allocated struct mem */
   libsmtp_free (mailsession);
   return 0;
-}
\ No newline at end of file
+}
+
diff -urN libsmtp-0.8.orig/examples/singlepart.c libsmtp-0.8/examples/singlepart.c
--- libsmtp-0.8.orig/examples/singlepart.c	2004-12-22 08:22:05.434978360 -0500
+++ libsmtp-0.8/examples/singlepart.c	2004-12-22 08:33:32.159580352 -0500
@@ -293,4 +293,5 @@
   /* Free the allocated struct mem */
   free (mailsession);
   return 0;
-}
\ No newline at end of file
+}
+
diff -urN libsmtp-0.8.orig/examples/smtpmail.c libsmtp-0.8/examples/smtpmail.c
--- libsmtp-0.8.orig/examples/smtpmail.c	2004-12-22 08:22:05.511966656 -0500
+++ libsmtp-0.8/examples/smtpmail.c	2004-12-22 08:33:32.160580200 -0500
@@ -247,4 +247,5 @@
   /* Free the allocated struct mem */
   libsmtp_free (mailsession);
   return 0;
-}
\ No newline at end of file
+}
+
diff -urN libsmtp-0.8.orig/include/libsmtp_mime.h libsmtp-0.8/include/libsmtp_mime.h
--- libsmtp-0.8.orig/include/libsmtp_mime.h	2004-12-22 08:22:05.592954344 -0500
+++ libsmtp-0.8/include/libsmtp_mime.h	2004-12-22 08:33:22.892989088 -0500
@@ -191,4 +191,5 @@
   #define LIBSMTP_BADCHARSET	2055
   
   #define LIBSMTP_MAX_MIME_ERRNO	2055
-/* #endif LIBSMTP_USE_MIME */
\ No newline at end of file
+/* #endif LIBSMTP_USE_MIME */
+
diff -urN libsmtp-0.8.orig/mime/comm.c libsmtp-0.8/mime/comm.c
--- libsmtp-0.8.orig/mime/comm.c	2004-12-22 08:22:04.856066368 -0500
+++ libsmtp-0.8/mime/comm.c	2004-12-22 08:33:05.151686176 -0500
@@ -158,7 +158,7 @@
     /* Text and message parts will have a charset setting */
     if ((libsmtp_temp_part->Type==LIBSMTP_MIME_TEXT) ||
         (libsmtp_temp_part->Type==LIBSMTP_MIME_MESSAGE))
-      if ((libsmtp_temp_string=libsmtp_int_lookup_mime_charset(libsmtp_temp_part)))
+      if ((libsmtp_temp_string=(char *)libsmtp_int_lookup_mime_charset(libsmtp_temp_part)))
       {
         g_string_sprintf (libsmtp_temp_gstring, "; charset=\"%s\"", \
            libsmtp_temp_string);
diff -urN libsmtp-0.8.orig/mime/parts.c libsmtp-0.8/mime/parts.c
--- libsmtp-0.8.orig/mime/parts.c	2004-12-22 08:22:04.905058920 -0500
+++ libsmtp-0.8/mime/parts.c	2004-12-22 08:33:05.213676752 -0500
@@ -396,7 +396,7 @@
           /* Text and message parts will have a charset setting */
           if ((libsmtp_temp_part->Type==LIBSMTP_MIME_TEXT) ||
               (libsmtp_temp_part->Type==LIBSMTP_MIME_MESSAGE))
-            if ((libsmtp_temp_string=libsmtp_int_lookup_mime_charset(libsmtp_temp_part)))
+            if ((libsmtp_temp_string=(char *)libsmtp_int_lookup_mime_charset(libsmtp_temp_part)))
             {
               g_string_sprintf (libsmtp_temp_gstring, ";\r\n charset=\"%s\"", \
                  libsmtp_temp_string);
diff -urN libsmtp-0.8.orig/smtp/libsmtp_comm.c libsmtp-0.8/smtp/libsmtp_comm.c
--- libsmtp-0.8.orig/smtp/libsmtp_comm.c	2004-12-22 08:22:05.102028976 -0500
+++ libsmtp-0.8/smtp/libsmtp_comm.c	2004-12-22 08:32:35.081257576 -0500
@@ -82,13 +82,13 @@
       g_string_assign (libsmtp_gstring_read, libsmtp_int_rec_buffer);
 
       /* Ok, take the first part of the response ... */
-      libsmtp_int_temp_buffer = strtok ((char *)libsmtp_int_rec_buffer, " ");
+      libsmtp_int_temp_buffer = (char *)strtok ((char *)libsmtp_int_rec_buffer, " ");
   
       /* and extract the response code */
       libsmtp_session->LastResponseCode = atoi(libsmtp_int_temp_buffer);
   
       /* Then fetch the rest of the string and save it */
-      libsmtp_int_temp_buffer = strtok (NULL, "\0");
+      libsmtp_int_temp_buffer = (char *)strtok (NULL, "\0");
       libsmtp_session->LastResponse = g_string_new (libsmtp_int_temp_buffer);
       break;
   }
@@ -718,4 +718,5 @@
   /* Wrong stage, dude ! */
   libsmtp_session->ErrorCode = LIBSMTP_BADSTAGE;
   return LIBSMTP_BADSTAGE;
-}
\ No newline at end of file
+}
+
diff -urN libsmtp-0.8.orig/smtp/libsmtp_connection.c libsmtp-0.8/smtp/libsmtp_connection.c
--- libsmtp-0.8.orig/smtp/libsmtp_connection.c	2004-12-22 08:22:05.188015904 -0500
+++ libsmtp-0.8/smtp/libsmtp_connection.c	2004-12-22 08:32:35.082257424 -0500
@@ -206,4 +206,5 @@
   }
 
   return LIBSMTP_NOERR;
-}
\ No newline at end of file
+}
+
diff -urN libsmtp-0.8.orig/smtp/libsmtp_data.c libsmtp-0.8/smtp/libsmtp_data.c
--- libsmtp-0.8.orig/smtp/libsmtp_data.c	2004-12-22 08:22:05.078032624 -0500
+++ libsmtp-0.8/smtp/libsmtp_data.c	2004-12-22 08:32:35.090256208 -0500
@@ -168,4 +168,5 @@
   }
   
   return LIBSMTP_NOERR;
-}
\ No newline at end of file
+}
+
diff -urN libsmtp-0.8.orig/smtp/libsmtp_error.c libsmtp-0.8/smtp/libsmtp_error.c
--- libsmtp-0.8.orig/smtp/libsmtp_error.c	2004-12-22 08:22:05.146022288 -0500
+++ libsmtp-0.8/smtp/libsmtp_error.c	2004-12-22 08:32:35.090256208 -0500
@@ -110,4 +110,5 @@
     return LIBSMTP_UNDEFERR;
 
   return libsmtp_session->ErrorCode;
-}
\ No newline at end of file
+}
+
