--- a/modules/text_renderer/freetype.c
+++ b/modules/text_renderer/freetype.c
@@ -67,10 +67,10 @@
 # define SYSTEM_DEFAULT_MONOSPACE_FONT_FILE "/system/fonts/DroidSansMono.ttf"
 # define SYSTEM_DEFAULT_MONOSPACE_FAMILY "Droid Sans Mono"
 #else
-# define SYSTEM_DEFAULT_FONT_FILE "/usr/share/fonts/truetype/freefont/FreeSerifBold.ttf"
-# define SYSTEM_DEFAULT_FAMILY "Serif Bold"
-# define SYSTEM_DEFAULT_MONOSPACE_FONT_FILE "/usr/share/fonts/truetype/freefont/FreeMono.ttf"
-# define SYSTEM_DEFAULT_MONOSPACE_FAMILY "Monospace"
+# define SYSTEM_DEFAULT_FONT_FILE "/usr/share/fonts/dejavu/DejaVuSans.ttf"
+# define SYSTEM_DEFAULT_FAMILY "DejaVu Sans"
+# define SYSTEM_DEFAULT_MONOSPACE_FONT_FILE "/usr/share/fonts/dejavu/DejaVuSansMono.ttf"
+# define SYSTEM_DEFAULT_MONOSPACE_FAMILY "DejaVu Sans Mono"
 #endif
 
 #ifndef DEFAULT_FONT_FILE
