--- Bundles/PGP/PGPController.h.orig    2007-09-21 13:51:25.000000000 -0700
+++ Bundles/PGP/PGPController.h 2007-09-21 13:58:22.000000000 -0700
@@ -69,6 +69,7 @@
 // other methods
 //
 - (void) updateAndRestartTimer;
+- (NSString *) myVersion;
 
 @end
 
--- Bundles/PGP/PGPController.m.orig    2007-09-21 13:51:37.000000000 -0700
+++ Bundles/PGP/PGPController.m 2007-09-21 13:58:19.000000000 -0700
@@ -274,7 +274,7 @@
 //
 //
 //
-- (NSString *) version
+- (NSString *) myVersion
 {
   return @"v0.9.1";
 }
--- Bundles/PGP/PGPViewController.m.orig        2007-09-25 11:42:10.000000000 -0700
+++ Bundles/PGP/PGPViewController.m     2007-09-25 11:42:30.000000000 -0700
@@ -71,7 +71,7 @@
 
   // We set the version label value
   [versionLabel setStringValue: [NSString stringWithFormat: _(@"Version: %@"),
-					  [[PGPController singleInstance] version]]];
+					  [[PGPController singleInstance] myVersion]]];
 
   return self;
 }

