From: Julian Ospald <hasufell@gentoo.org>
Date: Thu May  9 11:51:38 UTC 2013
Subject: fix build

--- CGenius-1.4.2-Release-Source/install.cmake
+++ CGenius-1.4.2-Release-Source/install.cmake
@@ -20,11 +20,13 @@
 	DESTINATION ${APPDIR})
 
 # This will copy the resources files to the proper directory
-INSTALL(DIRECTORY vfsroot/ 
-	DESTINATION ${DATADIR})
+IF(IS_DIRECTORY "${CMAKE_SOURCE_DIR}/vfsroot")
+	INSTALL(DIRECTORY vfsroot/ 
+		DESTINATION ${DATADIR})
+ENDIF(IS_DIRECTORY "${CMAKE_SOURCE_DIR}/vfsroot")
 
 # This will copy the readme file. 
-INSTALL(FILES README 
+INSTALL(FILES "${CMAKE_BINARY_DIR}/README"
 	DESTINATION ${DOCDIR})
 
 # This will copy the readme file. 
