From: Julian Ospald <hasufell@gentoo.org>
Date: Tue Jul  1 22:23:59 UTC 2014
Subject: fix build

--- lib/GsKit/CMakeLists.txt
+++ lib/GsKit/CMakeLists.txt
@@ -48,7 +48,6 @@
 #    pkg_check_modules(SDL sdl)
 #ENDIF(BUILD_TARGET STREQUAL WIN32)
 
-Find_Package ( SDL_image REQUIRED )
 
 
 
@@ -92,6 +91,7 @@
     set(SDL_IMAGE_LIBRARIES "SDL2_image")
     target_link_libraries(GsKit ${SDL_IMAGE_LIBRARIES})
 else(SDL2_FOUND)
+	Find_Package ( SDL_image REQUIRED )
     target_link_libraries(GsKit ${SDL_LIBRARY})
     target_link_libraries(GsKit ${SDLIMAGE_LIBRARY})
 endif(SDL2_FOUND)
