From b9aa302416387b5e31459c8dde21069e72001e5d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Amadeusz=20=C5=BBo=C5=82nowski?= <aidecoe@aidecoe.name>
Date: Thu, 20 Jun 2013 20:33:14 +0200
Subject: [PATCH 3/3] LatArCyrHeb-16 as a default i18n font for Gentoo

---
 dracut.conf.d/gentoo.conf.example | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dracut.conf.d/gentoo.conf.example b/dracut.conf.d/gentoo.conf.example
index b78e5c6..f9617f0 100644
--- a/dracut.conf.d/gentoo.conf.example
+++ b/dracut.conf.d/gentoo.conf.example
@@ -11,5 +11,6 @@ ro_mnt=yes
 
 # i18n
 i18n_vars="/etc/conf.d/keymaps:keymap-KEYMAP,extended_keymaps-EXT_KEYMAPS /etc/conf.d/consolefont:consolefont-FONT,consoletranslation-FONT_MAP /etc/rc.conf:unicode-UNICODE"
+i18n_default_font="LatArCyrHeb-16"
 
 omit_drivers+=" i2o_scsi"
-- 
1.8.3.2

