diff --git a/kio/kio/kprotocolmanager.cpp b/kio/kio/kprotocolmanager.cpp
index 3f1cadd..502a3e6 100644
--- a/kio/kio/kprotocolmanager.cpp
+++ b/kio/kio/kprotocolmanager.cpp
@@ -47,7 +47,6 @@
 #include <ksharedconfig.h>
 #include <kstandarddirs.h>
 #include <kurl.h>
-#include <kmimetype.h>
 #include <kmimetypetrader.h>
 #include <kprotocolinfofactory.h>
 
@@ -1190,23 +1189,7 @@ QString KProtocolManager::protocolForArchiveMimetype( const QString& mimeType )
             }
         }
     }
-    const QString prot = d->protocolForArchiveMimetypes.value(mimeType);
-    if (!prot.isEmpty())
-        return prot;
-
-    // Check parent mimetypes
-    KMimeType::Ptr mime = KMimeType::mimeType(mimeType);
-    if (mime) {
-        const QStringList parentMimeTypes = mime->allParentMimeTypes();
-        Q_FOREACH(const QString& parentMimeType, parentMimeTypes) {
-            const QString res = d->protocolForArchiveMimetypes.value(parentMimeType);
-            if (!res.isEmpty()) {
-                return res;
-            }
-        }
-    }
-
-    return QString();
+    return d->protocolForArchiveMimetypes.value(mimeType);
 }
 
 #undef PRIVATE_DATA
