# drop longrunning and unneeded search for python library
--- macros/python.m4.orig	2010-03-12 07:11:49.000000000 +0100
+++ macros/python.m4	2010-03-12 07:12:21.000000000 +0100
@@ -56,21 +56,6 @@
 
 	if test "$cf_with_python_devel" != "no" ; then
 
-	# Check for Python library path
-	AC_MSG_CHECKING([for Python library path])
-	python_path=`echo $PYTHON | sed "s,/bin.*$,,"`
-	for i in "$python_path/lib/python$PYTHON_VERSION/config/" "$python_path/lib/python$PYTHON_VERSION/" "$python_path/lib/python/config/" "$python_path/lib/python/" "$python_path/" ; do
-		python_path=`find $i -type f -name libpython$PYTHON_VERSION.* -print | sed "1q"`
-		if test -n "$python_path" ; then
-			break
-		fi
-	done
-	python_path=`echo $python_path | sed "s,/libpython.*$,,"`
-	AC_MSG_RESULT([$python_path])
-	if test -z "$python_path" ; then
-		AC_MSG_WARN([cannot find Python library path])
-	fi
-	AC_SUBST([PYTHON_LDFLAGS],["-L$python_path -lpython$PYTHON_VERSION"])
 	#
 	python_site=`echo $python_path | sed "s/config/site-packages/"`
 	AC_SUBST([PYTHON_SITE_PKG],[$python_site])
Index: bindings/Makefile.am
# fix tcl and python install path
===================================================================
--- bindings/Makefile.am	(revision 2854)
+++ bindings/Makefile.am	(working copy)
@@ -69,7 +69,7 @@
 
 tcl_ltlib = hamlibtcl.la
 
-tcldir = $(prefix)/lib/tcl
+tcldir = @tclpackagedir@/hamlib
 
 PKG_VER=1.0
 DLL=hamlibtcl-$(PKG_VER)@TCL_SHLIB_SUFFIX@
@@ -105,7 +105,7 @@
 
 python_ltlib = _Hamlib.la
 
-pythondir = $(prefix)/lib/python
+pythondir = @pythondir@
 
 nodist__Hamlib_la_SOURCES = hamlibpy_wrap.c
 _Hamlib_la_LDFLAGS = -no-undefined -module -avoid-version
@@ -131,8 +131,8 @@
 uninstall-py: 
 
 ##########################################
-
-lib_LTLIBRARIES = $(tcl_ltlib) $(python_ltlib)
+_Hamlib_la_LTLIBRARIES = $(python_ltlib)
+lib_LTLIBRARIES = $(tcl_ltlib)
 # lib_LTLIBRARIES = @BINDING_LIB_TARGETS@
 
 all-local: @BINDING_ALL@
Index: configure.ac
# add macro to determine path to tcl packages
===================================================================
--- configure.ac	(revision 2854)
+++ configure.ac	(working copy)
@@ -330,6 +330,8 @@
 AC_SUBST(TCL_LIB_SPEC)
 AC_SUBST(TCL_INCLUDE_SPEC)
 AC_SUBST(TCL_SHLIB_SUFFIX)
+tclpackagedir=`echo $TCL_PACKAGE_PATH | cut -d ' ' -f 1 | sed "s,$TCL_PREFIX,\\${prefix},"`
+AC_SUBST(tclpackagedir)
 
 # TODO: require "${ac_cv_header_sys_socket_h}" = "no"
 
