Index: atscepg-0.1.1/structs.cpp
===================================================================
--- atscepg-0.1.1.orig/structs.cpp
+++ atscepg-0.1.1/structs.cpp
@@ -1,4 +1,5 @@
 #include <string.h>
+#include <stdlib.h>
 
 #include "structs.h"
 
@@ -159,6 +160,7 @@ const Channel& Channel::operator= (const
   minorChannelNumber  = arg.minorChannelNumber;
   free(short_name);
   short_name          = strdup(arg.short_name);
+  return *this;
 }
 
 
