diff -ru openvpn-2.1_rc20~/easy-rsa/2.0/openssl.cnf openvpn-2.1_rc20/easy-rsa/2.0/openssl.cnf
--- openvpn-2.1_rc20~/easy-rsa/2.0/openssl.cnf	2009-10-11 01:04:36.000000000 +0200
+++ openvpn-2.1_rc20/easy-rsa/2.0/openssl.cnf	2009-10-11 01:05:53.000000000 +0200
@@ -283,9 +283,9 @@
 #pkcs11 = pkcs11_section
 
 [ pkcs11_section ]
-engine_id = pkcs11
-dynamic_path = /usr/lib/engines/engine_pkcs11.so
-MODULE_PATH = $ENV::PKCS11_MODULE_PATH
-PIN = $ENV::PKCS11_PIN
-init = 0
+#engine_id = pkcs11
+#dynamic_path = /usr/lib/engines/engine_pkcs11.so
+#MODULE_PATH = $ENV::PKCS11_MODULE_PATH
+#PIN = $ENV::PKCS11_PIN
+#init = 0
 
