diff -ru cuyo-2.0.0.orig/icons/Makefile.am cuyo-2.0.0/icons/Makefile.am
--- cuyo-2.0.0.orig/icons/Makefile.am	2012-01-12 12:51:05.000000000 -0500
+++ cuyo-2.0.0/icons/Makefile.am	2015-02-18 17:19:55.320961943 -0500
@@ -17,7 +17,7 @@
 #   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 #
 
-desktopdir = $(datarootdir)/applications
+desktopdir = /usr/share/applications
 desktop_DATA = cuyo.desktop
 
 
@@ -32,7 +32,7 @@
 #	echo "Exec=$(bindir)/cuyo" >> cuyo.desktop
 
 
-iconsdir = $(datarootdir)/icons/hicolor
+iconsdir = /usr/share/icons/hicolor
 nobase_icons_DATA = 32x32/apps/cuyo.png 64x64/apps/cuyo.png
 
 EXTRA_DIST = $(nobase_icons_DATA) cuyo.desktop
diff -ru cuyo-2.0.0.orig/icons/cuyo.desktop cuyo-2.0.0/icons/cuyo.desktop
--- cuyo-2.0.0.orig/icons/cuyo.desktop	2011-03-26 11:20:01.000000000 -0400
+++ cuyo-2.0.0/icons/cuyo.desktop	2015-02-18 17:19:26.506082349 -0500
@@ -4,7 +4,6 @@
 Type=Application
 Terminal=false
 Categories=Game;BlocksGame;
-Encoding=UTF-8
 Comment=Play a tetris like game with many levels
 Comment[de]=Ein Tetris-artiges Spiel mit vielen Leveln spielen
 Exec=cuyo
