--- po/Makefile.am
+++ po/Makefile.am
@@ -60,7 +60,7 @@
 
 # By virtue of this rule, more precisely by virtue of genpot, this Makefile
 # probably only works with srcdir=.
-$(DOMAIN).pot: $(pot_sources)
++$(DOMAIN).pot: $(pot_sources)
 	./genpot
 
 %.mo: %.po
--- po/genpot
+++ po/genpot
@@ -27,10 +27,10 @@
     -o - \
     src/* data/* \
   | sed \
-    -e "1 s/SOME DESCRIPTIVE TITLE/Message catalog for the game Cuyo/" \
-    -e "2 s/Copyright (C) YEAR THE PACKAGE'S COPYRIGHT HOLDER/This is a partly generated file. The copyright of the original messages is identical to the one of the source files they are generated from. The copyright of the translations is by the translators./" \
-    -e "3,9 s/PACKAGE/Cuyo/" \
-    -e "16 s/CHARSET/ASCII/" \
+    -e "s/SOME DESCRIPTIVE TITLE/Message catalog for the game Cuyo/" \
+    -e "s/Copyright (C) YEAR THE PACKAGE'S COPYRIGHT HOLDER/This is a partly generated file. The copyright of the original messages is identical to the one of the source files they are generated from. The copyright of the translations is by the translators./" \
+    -e "s/PACKAGE/Cuyo/" \
+    -e "s/CHARSET/ASCII/" \
   | msgconv -t ISO-8859-1 \
   > cuyo.pot

--- icons/cuyo.desktop
+++ icons/cuyo.desktop
@@ -5,7 +5,6 @@
 Terminal=false
 MimeType=application/x-executable;
 Categories=Game;BlocksGame;
-Encoding=UTF-8
 Comment=Play a tetris like game with many levels
 Comment[de]=Ein Tetris-artiges Spiel mit vielen Leveln spielen
 Exec=cuyo 
--- icons/Makefile.am
+++ icons/Makefile.am
@@ -1,4 +1,4 @@
-desktopdir = $(datarootdir)/applications
+desktopdir = /usr/share/applications
 desktop_DATA = cuyo.desktop
 
 
@@ -25,7 +25,7 @@
 # Das vorige war falsch: die png sollen alle cuyo.png heissen
 # und nicht cuyo-xxxx.png. Deshalb nachfolgend neue Version:
 
-iconsdir = $(datarootdir)/icons/hicolor
+iconsdir = /usr/share/icons/hicolor
 icons_DATA = cuyo-32x32.png cuyo-64x64.png
 
 install-iconsDATA:
