don't install the pam.d file if pam is disabled

--- a/configure
+++ b/configure
@@ -28572,12 +28572,16 @@
 		LIBS="$savedLIBS"
 	fi
 
-	netatalk_cv_install_pam=yes
-	if test x"$pam_found" = "xyes" -a "x$PAMDIR" = "xNONE"; then
+	if test "x$require_pam" != "xnever"; then
+		netatalk_cv_install_pam=yes
+	if test x"$pam_found" = "xyes" -a "x$PAMDIR" = "xNONE" ; then
 		{ echo "$as_me:$LINENO: WARNING: PAM support can be compiled, but the install location for the netatalk.pamd file could not be determined. Either install this file by hand or specify the install path." >&5
 echo "$as_me: WARNING: PAM support can be compiled, but the install location for the netatalk.pamd file could not be determined. Either install this file by hand or specify the install path." >&2;}
 		netatalk_cv_install_pam=no
 	fi
+	else
+		netatalk_cv_install_pam=no
+	fi
 
 	{ echo "$as_me:$LINENO: checking whether to enable PAM support" >&5
 echo $ECHO_N "checking whether to enable PAM support... $ECHO_C" >&6; }
