Index: netatalk-2.2.1/etc/afpd/afp_avahi.c
===================================================================
--- netatalk-2.2.1.orig/etc/afpd/afp_avahi.c
+++ netatalk-2.2.1/etc/afpd/afp_avahi.c
@@ -176,8 +176,8 @@ static void register_stuff(void) {
     return;
 
 fail:
-    time(NULL);
 //    avahi_threaded_poll_quit(ctx->threaded_poll);
+    return;
 }
 
 /* Called when publishing of service data completes */
Index: netatalk-2.2.1/etc/uams/Makefile.am
===================================================================
--- netatalk-2.2.1.orig/etc/uams/Makefile.am
+++ netatalk-2.2.1/etc/uams/Makefile.am
@@ -97,16 +97,16 @@ uams_dhx_pam_la_CFLAGS     = @SSL_CFLAGS
 uams_dhx2_pam_la_CFLAGS    = @LIBGCRYPT_CFLAGS@ @PAM_CFLAGS@
 uams_gss_la_CFLAGS   	   = @GSSAPI_CFLAGS@
 
-uams_guest_la_LDFLAGS      = -module -avoid-version
-uams_randnum_la_LDFLAGS    = -module -avoid-version @SSL_LIBS@
-uams_passwd_la_LDFLAGS     = -module -avoid-version @CRYPT_LIBS@
-uams_pam_la_LDFLAGS        = -module -avoid-version @PAM_LIBS@
-uams_pgp_la_LDFLAGS        = -module -avoid-version @SSL_LIBS@
-uams_dhx_passwd_la_LDFLAGS = -module -avoid-version @CRYPT_LIBS@ @SSL_LIBS@
-uams_dhx_pam_la_LDFLAGS		= -module -avoid-version @CRYPT_LIBS@ @SSL_LIBS@ @PAM_LIBS@
-uams_dhx2_passwd_la_LDFLAGS	= -module -avoid-version @CRYPT_LIBS@ @LIBGCRYPT_LIBS@
-uams_dhx2_pam_la_LDFLAGS	= -module -avoid-version @LIBGCRYPT_LIBS@ @PAM_LIBS@
-uams_gss_la_LDFLAGS   	   = -module -avoid-version @GSSAPI_LIBS@ 
+uams_guest_la_LDFLAGS      = -shared -module -avoid-version
+uams_randnum_la_LDFLAGS    = -shared -module -avoid-version @SSL_LIBS@
+uams_passwd_la_LDFLAGS     = -shared -module -avoid-version @CRYPT_LIBS@
+uams_pam_la_LDFLAGS        = -shared -module -avoid-version @PAM_LIBS@
+uams_pgp_la_LDFLAGS        = -shared -module -avoid-version @SSL_LIBS@
+uams_dhx_passwd_la_LDFLAGS = -shared -module -avoid-version @CRYPT_LIBS@ @SSL_LIBS@
+uams_dhx_pam_la_LDFLAGS		= -shared -module -avoid-version @CRYPT_LIBS@ @SSL_LIBS@ @PAM_LIBS@
+uams_dhx2_passwd_la_LDFLAGS	= -shared -module -avoid-version @CRYPT_LIBS@ @LIBGCRYPT_LIBS@
+uams_dhx2_pam_la_LDFLAGS	= -shared -module -avoid-version @LIBGCRYPT_LIBS@ @PAM_LIBS@
+uams_gss_la_LDFLAGS   	   = -shared -module -avoid-version @GSSAPI_LIBS@ 
 
 #
 # module compilation
Index: netatalk-2.2.1/libatalk/Makefile.am
===================================================================
--- netatalk-2.2.1.orig/libatalk/Makefile.am
+++ netatalk-2.2.1/libatalk/Makefile.am
@@ -37,5 +37,3 @@ libatalk_la_DEPENDENCIES = \
 	unicode/libunicode.la \
 	vfs/libvfs.la
 
-libatalk_la_LDFLAGS = -static
-
Index: netatalk-2.2.1/libatalk/vfs/Makefile.am
===================================================================
--- netatalk-2.2.1.orig/libatalk/vfs/Makefile.am
+++ netatalk-2.2.1/libatalk/vfs/Makefile.am
@@ -6,4 +6,5 @@ libvfs_la_SOURCES = vfs.c unix.c ea.c sy
 
 if HAVE_ACLS
 libvfs_la_SOURCES += acl.c
+libvfs_la_LIBADD = $(ACL_LIBS)
 endif
Index: netatalk-2.2.1/libatalk/dsi/Makefile.am
===================================================================
--- netatalk-2.2.1.orig/libatalk/dsi/Makefile.am
+++ netatalk-2.2.1/libatalk/dsi/Makefile.am
@@ -7,5 +7,6 @@ LIBS = @LIBS@
 noinst_LTLIBRARIES = libdsi.la
 
 libdsi_la_SOURCES = dsi_attn.c dsi_close.c dsi_cmdreply.c dsi_getsess.c dsi_getstat.c dsi_init.c dsi_opensess.c dsi_read.c dsi_tcp.c dsi_tickle.c dsi_write.c dsi_stream.c
+libdsi_la_LIBADD = $(WRAP_LIBS)
 
 noinst_HEADERS = dsi_private.h
