--- config/binmake.gmk.orig
+++ config/binmake.gmk
@@ -234,7 +234,7 @@ ifdef GLOBALLIBS
              -lG4intercoms \
              -lG4global
 
-  ifdef G4LIB_USE_ZLIB
+  ifdef G4LIB_BUILD_ZLIB
     LDLIBS2 += -lG4zlib
   endif
 

--- config/architecture.gmk.orig	2011-01-16 00:22:05.976079082 +0000
+++ config/architecture.gmk	2011-01-16 00:23:18.346082379 +0000
@@ -214,8 +214,10 @@
 else
   LDFLAGS   += -L$(CLHEP_LIB_DIR)
   LOADLIBS  := -l$(CLHEP_LIB) -lm
-  ifndef G4LIB_USE_ZLIB
-    LOADLIBS += -lz
+  ifdef G4LIB_USE_ZLIB
+    ifndef G4LIB_BUILD_ZLIB
+      LOADLIBS += -lz
+    endif
   endif
 endif
 
