--- lib-src/enigma-core/ecl_alist.hh.old	2012-09-28 12:28:01.465223434 +0200
+++ lib-src/enigma-core/ecl_alist.hh	2012-09-28 12:28:41.195739039 +0200
@@ -61,7 +61,7 @@
 	VAL &operator[] (const key_type &key) { 
             iterator i=find(key);
             if (i==this->end())
-                i=insert(this->end(), make_pair(key, VAL()));
+                i=this->insert(this->end(), make_pair(key, VAL()));
             return i->second;
         }
     };
