--- msp430mcu-20111205/scripts/install.sh.orig
+++ msp430mcu-20111205/scripts/install.sh
@@ -32,14 +32,3 @@
 # Install MCU-specific memory and periph maps
 cp -pr ${ANALYSIS}/ldscripts ${LIBPATH}
 chmod -R og+rX ${LIBPATH}/ldscripts
-
-# Install utility that tells where everything got installed
-cat bin/msp430mcu-config.in \
-| sed \
-    -e 's!@PREFIX@!'"${PREFIX}"'!g' \
-    -e 's!@SCRIPTPATH@!'"${LIBPATH}/ldscripts"'!g' \
-    -e 's!@INCPATH@!'"${INCPATH}"'!g' \
-    -e 's!@VERSION@!'"${VERSION}"'!g' \
-    -e 's!@UPSTREAM_VERSION@!'"${UPSTREAM_VERSION}"'!g' \
-> ${BINPATH}/msp430mcu-config \
-&& chmod 0755 ${BINPATH}/msp430mcu-config
