commit 0246d8e24c49e2ed12c183f0cde08d2e16b4c299
Author: hasufell <hasufell@gentoo.org>
Date:   Mon Sep 30 00:10:54 2013 +0200

    allow building with libpng-1.6

diff --git a/configure.ac b/configure.ac
index 2a8cba9..334b96e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -47,7 +47,7 @@ LW6_CODENAME="Blackbeard"
 AC_SUBST(LW6_CODENAME)
 
 dnl Headers, some are optional
-AC_CHECK_HEADERS(arpa/inet.h ctype.h dirent.h errno.h execinfo.h expat.h fcntl.h gmp.h GL/gl.h GL/glu.h iconv.h ifaddrs.h jpeglib.h langinfo.h libintl.h libpng12/png.h libpng13/png.h libpng14/png.h libpng15/png.h lmcons.h limits.h locale.h ltdl.h mach/vm_statistics.h mach/mach_types.h mach/mach_init.h mach/mach_host.h math.h netdb.h netinet/in.h omp.h pthread.h readline/history.h readline/readline.h SDL/SDL.h SDL/SDL_image.h SDL/SDL_mixer.h SDL/SDL_ttf.h signal.h stdarg.h stdio.h stdlib.h string.h syslog.h sys/select.h sys/signal.h sys/socket.h sys/stat.h sys/sysinfo.h sys/time.h sys/types.h sys/utsname.h time.h unistd.h winbase.h windows.h winsock2.h)
+AC_CHECK_HEADERS(arpa/inet.h ctype.h dirent.h errno.h execinfo.h expat.h fcntl.h gmp.h GL/gl.h GL/glu.h iconv.h ifaddrs.h jpeglib.h langinfo.h libintl.h libpng12/png.h libpng13/png.h libpng14/png.h libpng15/png.h libpng16/png.h lmcons.h limits.h locale.h ltdl.h mach/vm_statistics.h mach/mach_types.h mach/mach_init.h mach/mach_host.h math.h netdb.h netinet/in.h omp.h pthread.h readline/history.h readline/readline.h SDL/SDL.h SDL/SDL_image.h SDL/SDL_mixer.h SDL/SDL_ttf.h signal.h stdarg.h stdio.h stdlib.h string.h syslog.h sys/select.h sys/signal.h sys/socket.h sys/stat.h sys/sysinfo.h sys/time.h sys/types.h sys/utsname.h time.h unistd.h winbase.h windows.h winsock2.h)
 
 AM_MISSING_PROG(PERL, perl, ${missing_dir})
 AM_MISSING_PROG(AWK, awk, ${missing_dir})
@@ -319,7 +319,8 @@ AC_CHECK_HEADER(libpng12/png.h, HAVE_LIBPNG12_PNG_H=1)
 AC_CHECK_HEADER(libpng13/png.h, HAVE_LIBPNG13_PNG_H=1)
 AC_CHECK_HEADER(libpng14/png.h, HAVE_LIBPNG14_PNG_H=1)
 AC_CHECK_HEADER(libpng15/png.h, HAVE_LIBPNG15_PNG_H=1)
-if test x${HAVE_LIBPNG12_PNG_H} = x1 || test x${HAVE_LIBPNG13_PNG_H} = x1 || test x${HAVE_LIBPNG14_PNG_H} = x1 || test x${HAVE_LIBPNG15_PNG_H} = x1; then
+AC_CHECK_HEADER(libpng16/png.h, HAVE_LIBPNG16_PNG_H=1)
+if test x${HAVE_LIBPNG12_PNG_H} = x1 || test x${HAVE_LIBPNG13_PNG_H} = x1 || test x${HAVE_LIBPNG14_PNG_H} = x1 || test x${HAVE_LIBPNG15_PNG_H} = x1 || test x${HAVE_LIBPNG16_PNG_H} = x1; then
    if test x${HAVE_LIBPNG12_PNG_H} = x1; then
       LIBPNG_1X_H=libpng12/png.h
    fi
@@ -332,9 +333,12 @@ if test x${HAVE_LIBPNG12_PNG_H} = x1 || test x${HAVE_LIBPNG13_PNG_H} = x1 || tes
    if test x${HAVE_LIBPNG15_PNG_H} = x1; then
       LIBPNG_1X_H=libpng15/png.h
    fi
+   if test x${HAVE_LIBPNG16_PNG_H} = x1; then
+      LIBPNG_1X_H=libpng16/png.h
+   fi
 else
    AC_MSG_ERROR([
-*** Liquid War 6 needs libpng 1.2 1.3 1.4 or 1.5 (http://www.libpng.org/pub/png/libpng.html)
+*** Liquid War 6 needs libpng 1.2 1.3 1.4 1.5 or 1.6 (http://www.libpng.org/pub/png/libpng.html)
 ])
 fi
 AC_SUBST(LIBPNG_1X_H)
