commit e204ce228f5aafc7efd2c09fb4b0bfefe9708dfe
Author: hasufell <hasufell@gentoo.org>
Date:   Mon Sep 30 00:19:48 2013 +0200

    make paths modifiable

diff --git a/configure.ac b/configure.ac
index 334b96e..d846210 100644
--- a/configure.ac
+++ b/configure.ac
@@ -698,6 +698,20 @@ else
 fi
 AC_SUBST(VALGRIND_CFLAGS)
 
+AC_ARG_WITH([iconsdir],
+			[AS_HELP_STRING([--with-iconsdir=DIR],
+							[path where icons get installed (default: $datadir/icons/hicolor/48x48/apps)])],
+			[ICONSDIR="$withval"],
+			[ICONSDIR='$(datadir)/pixmaps'])
+AC_SUBST(ICONSDIR)
+
+AC_ARG_WITH([desktopdir],
+			[AS_HELP_STRING([--with-desktopdir=DIR],
+							[path where desktop files get installed (default: $datadir/applications)])],
+			[DESKTOPDIR="$withval"],
+			[DESKTOPDIR='$(datadir)/applications'])
+AC_SUBST(DESKTOPDIR)
+
 if test x${USE_INCLUDED_LIBINTL} = xyes ; then
    LW6_LIBINTL_LDFLAGS=
    LW6_LIBINTL_LDADD=${LTLIBINTL}
diff --git a/data/icon/Makefile.am b/data/icon/Makefile.am
index b4d5ae4..f9b738c 100644
--- a/data/icon/Makefile.am
+++ b/data/icon/Makefile.am
@@ -19,6 +19,6 @@
 # Contact author        : ufoot@ufoot.org
 
 pkgdata_DATA = liquidwar6.png liquidwar6.xpm
-pkgdatadir = $(datadir)/pixmaps
+pkgdatadir = $(ICONSDIR)
 EXTRA_DIST = $(pkgdata_DATA) liquidwar6.ico liquidwar6.icns
 
diff --git a/pkg/Makefile.am b/pkg/Makefile.am
index 37f73ee..1980369 100644
--- a/pkg/Makefile.am
+++ b/pkg/Makefile.am
@@ -19,7 +19,7 @@
 # Contact author        : ufoot@ufoot.org
 
 pkgdata_DATA = liquidwar6.desktop
-pkgdatadir = $(datadir)/applications
+pkgdatadir = $(DESKTOPDIR)
 EXTRA_DIST =  $(pkgdata_DATA) liquidwar6.spec liquidwar6d
 
 distclean-local:
