commit b69515e24f9ea066a7e393fa7b364e0701a8f58c
Author: hasufell <hasufell@gentoo.org>
Date:   Mon Sep 30 00:11:04 2013 +0200

    don't run ldconfig

diff --git a/src/Makefile.am b/src/Makefile.am
index da78107..a0961b8 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -55,7 +55,6 @@ EXTRA_DIST = scriptpo.c indent.sh scriptpo-update.sh stamp.sh clean.sh dtou.sh u
 SUBDIRS = lib
 
 install-exec-local:
-	if which id && which ldconfig && test `id -u` -eq 0 && test "$(prefix)" != "/usr"; then ldconfig; fi # Run ldconfig automatically if we're root
 	if test -x $(DESTDIR)$(bindir)/liquidwar6; then mv $(DESTDIR)$(bindir)/liquidwar6 $(DESTDIR)$(bindir)/@PACKAGE_TARNAME@-@PACKAGE_VERSION@; ln -s @PACKAGE_TARNAME@-@PACKAGE_VERSION@ $(DESTDIR)$(bindir)/liquidwar6; fi
 
 uninstall-local:
