From 5a51812b7c1457feb8acaa106f43d7d5252b3980 Mon Sep 17 00:00:00 2001
From: Peter Breitenlohner <peb@mppmu.mpg.de>
Date: Tue, 12 Mar 2013 20:25:50 +0100
Subject: [PATCH] Fix compile when not using libjpeg

---
 poppler/Stream.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/poppler/Stream.h b/poppler/Stream.h
index 7a5ff1c..76d6a1f 100644
--- a/poppler/Stream.h
+++ b/poppler/Stream.h
@@ -22,6 +22,7 @@
 // Copyright (C) 2011, 2012 William Bader <williambader@hotmail.com>
 // Copyright (C) 2012, 2013 Thomas Freitag <Thomas.Freitag@alfa.de>
 // Copyright (C) 2012 Fabio D'Urso <fabiodurso@hotmail.it>
+// Copyright (C) 2013 Peter Breitenlohner <peb@mppmu.mpg.de>
 //
 // To see a description of the changes please see the Changelog file that
 // came with your tarball or type make ChangeLog if you are building from git
@@ -850,7 +851,7 @@ struct DCTHuffTable {
 class DCTStream: public FilterStream {
 public:
 
-  DCTStream(Stream *strA, int colorXformA);
+  DCTStream(Stream *strA, int colorXformA, Object *dict);
   virtual ~DCTStream();
   virtual StreamKind getKind() { return strDCT; }
   virtual void reset();
-- 
1.8.1.5

