diff --git a/libopendkim/dkim-cache.c b/libopendkim/dkim-cache.c
index 1321b74..1bb10e8 100644
--- a/libopendkim/dkim-cache.c
+++ b/libopendkim/dkim-cache.c
@@ -20,6 +20,7 @@
 #include <errno.h>
 #include <pthread.h>
 #include <string.h>
+#include <stdlib.h>
 
 /* libdb includes */
 #include <db.h>
diff --git a/opendkim/util.c b/opendkim/util.c
index e9de05e..37cdf46 100644
--- a/opendkim/util.c
+++ b/opendkim/util.c
@@ -1013,8 +1013,13 @@ dkimf_mkpath(char *path, size_t pathlen, char *root, char *file)
 	}
 	else if (root[0] == '\0')			/* no root, use cwd */
 	{
-		(void) getcwd(path, pathlen);
-		strlcat(path, "/", pathlen);
+		char *p;
+
+		p = getcwd(path, pathlen);
+		if (p == NULL)
+			strlcpy(path, "./", pathlen);
+		else
+			strlcat(path, "/", pathlen);
 		strlcat(path, file, pathlen);
 	}
 	else						/* use root */
