--- libopendkim/tests/t-test153.c	2013-02-12 23:28:23.000000000 +0200
+++ libopendkim/tests/t-test153.c	2013-02-18 14:31:56.012827510 +0200
@@ -14,6 +14,16 @@
 #include <string.h>
 #include <stdio.h>
 
+/* libbsd if found */
+#ifdef USE_BSD_H
+# include <bsd/string.h>
+#endif /* USE_BSD_H */
+
+/* libstrl if needed */
+#ifdef USE_STRL_H
+# include <strl.h>
+#endif /* USE_STRL_H */
+
 #ifdef USE_GNUTLS
 # include <gnutls/gnutls.h>
 #endif /* USE_GNUTLS */
