only in patch2:
unchanged:
--- opendkim-2.8.0~beta5.orig/configure.ac
+++ opendkim-2.8.0~beta5/configure.ac
@@ -1713,7 +1713,7 @@
 			AC_MSG_RESULT($ldappath)
 		        ldap_found="yes"
 			OPENLDAP_CPPFLAGS="-I$ldappath/include"
-			OPENLDAP_LIBS="-L$ldappath/lib -lldap"
+			OPENLDAP_LIBS="-L$ldappath/lib -lldap -llber"
 		else
 			AC_MSG_ERROR(not found at $ldappath)
 		fi
@@ -1727,7 +1727,7 @@
 				AC_MSG_RESULT($d)
 		        	ldap_found="yes"
 				OPENLDAP_CPPFLAGS="-I$ldappath/include"
-				OPENLDAP_LIBS="-L$ldappath/lib -lldap"
+				OPENLDAP_LIBS="-L$ldappath/lib -lldap -llber"
 				break
 			fi
 		done
only in patch2:
unchanged:
--- opendkim-2.8.0~beta5.orig/opendkim/Makefile.am
+++ opendkim-2.8.0~beta5/opendkim/Makefile.am
@@ -28,7 +28,7 @@
 opendkim_CFLAGS = $(PTHREAD_CFLAGS) $(LIBCRYPTO_CFLAGS) $(COV_CFLAGS)
 opendkim_CPPFLAGS = -I$(srcdir)/../libopendkim $(LIBCRYPTO_CPPFLAGS)
 opendkim_LDFLAGS = $(LIBCRYPTO_LIBDIRS) $(LIBMILTER_LIBDIRS) $(PTHREAD_CFLAGS) $(COV_LDFLAGS)
-opendkim_LDADD = ../libopendkim/libopendkim.la $(LIBMILTER_LIBS) $(LIBCRYPTO_LIBS) $(PTHREAD_LIBS) $(COV_LIBADD)
+opendkim_LDADD = ../libopendkim/libopendkim.la $(LIBMILTER_LIBS) $(LIBCRYPTO_LIBS) $(PTHREAD_LIBS) $(COV_LIBADD) $(LIBRESOLV)
 if STRL
 opendkim_CFLAGS += -I$(top_srcdir)/libstrl
 opendkim_LDADD += ../libstrl/libstrl.la
only in patch2:
unchanged:
--- opendkim-2.8.0~beta5.orig/libopendkim/tests/Makefile.am
+++ opendkim-2.8.0~beta5/libopendkim/tests/Makefile.am
@@ -6,7 +6,7 @@
 AM_CFLAGS += -g
 endif
 
-LDADD = ../libopendkim.la $(COV_LIBADD) $(LIBCRYPTO_LIBS)
+LDADD = ../libopendkim.la $(COV_LIBADD) $(LIBCRYPTO_LIBS) $(LIBRESOLV)
 AM_CPPFLAGS = -I..
 
 if STRL
