--- configure.ac	2013-01-22 07:59:36.312240111 +0000
+++ configure.ac	2013-01-22 07:59:23.694397856 +0000
@@ -240,11 +240,14 @@
 # we need to include <strl.h> if:
 # (a) strlcat and strlcpy weren't found, since we'll be rolling our own; OR
 # (b) an installed strl.h was found and no installed bsd/string.h was found
-if test x"$strl_h_found" = x"yes" -o \( x"$strl_found" = x"no" -a x"$libstrl_found" = x"no" \) -a x"$bsdstrl_h_found" = x"no"
-then
-	AC_DEFINE([USE_STRL_H], 1,
-	          [Define to 1 if you need to include <strl.h> to get the `strlcat()' and `strlcpy()' functions.])
-fi
+
+# Gentoo specific: Don't include strl.h - breaks upgrading from
+# <=mail-filter/opendkim-2.7.2.  We link against libbsd always.
+#if test x"$strl_h_found" = x"yes" -o \( x"$strl_found" = x"no" -a x"$libstrl_found" = x"no" \) -a x"$bsdstrl_h_found" = x"no"
+#then
+#	AC_DEFINE([USE_STRL_H], 1,
+#	          [Define to 1 if you need to include <strl.h> to get the `strlcat()' and `strlcpy()' functions.])
+#fi
 
 if test x"$bsdstrl_h_found" = x"yes"
 then
