See https://github.com/h5py/h5py/issues/263
diff -ur h5py-2.1.2.orig/h5py/_hl/tests/test_base.py h5py-2.1.2/h5py/_hl/tests/test_base.py
--- h5py/_hl/tests/test_base.py	2013-02-07 23:40:26.000000000 +0800
+++ h5py/_hl/tests/test_base.py	2013-03-07 16:35:25.774142098 +0800
@@ -62,6 +62,7 @@
         typ = self.f['type']
         self._check_type(typ)
 
+    @unittest.skipUnless(sys.getfilesystemencoding() == 'UTF-8', "only passes for locale utf-8")
     def test_file(self):
         """ File object repr() with unicode """
         fname = tempfile.mktemp(self.USTRING+u'.hdf5')
diff -ur h5py-2.1.2.orig/h5py/_hl/tests/test_file.py h5py-2.1.2/h5py/_hl/tests/test_file.py
--- h5py/_hl/tests/test_file.py	2013-02-07 23:40:26.000000000 +0800
+++ h5py/_hl/tests/test_file.py	2013-03-07 16:34:59.006143430 +0800
@@ -323,6 +323,7 @@
             self.assertTrue(fid)
         self.assertTrue(not fid)
 
+@unittest.skipUnless(sys.getfilesystemencoding() == 'UTF-8', "only passes for locale utf-8")
 class TestUnicode(TestCase):
 
     """
