diff -ur cyphesis-0.5.20.orig/configure.ac cyphesis-0.5.20/configure.ac
--- cyphesis-0.5.20.orig/configure.ac	2009-05-11 15:23:20.000000000 +0300
+++ cyphesis-0.5.20/configure.ac	2009-08-15 12:33:01.000000000 +0300
@@ -342,26 +342,6 @@
 libraries required and where to obtain them.])
     ])
 
-PKG_CHECK_MODULES(AVAHI, avahi-client,
-    [
-        CPPFLAGS="$CPPFLAGS $AVAHI_CFLAGS"
-        MDNS_LIBS="$MDNS_LIBS $AVAHI_LIBS"
-        AC_DEFINE(HAVE_AVAHI, 1, [Define to 1 if you have avahi libs.])
-    ],
-    [
-        AC_MSG_RESULT([no])
-        PKG_CHECK_MODULES(HOWL, howl,
-            [
-                CPPFLAGS="$CPPFLAGS $HOWL_CFLAGS"
-                MDNS_LIBS="$MDNS_LIBS $HOWL_LIBS"
-                AC_DEFINE(HAVE_LIBHOWL, 1, [Define to 1 if you have howl libs.])
-            ],
-            [
-                AC_MSG_RESULT([no])
-                AC_MSG_NOTICE([Ommiting DNS-SD functionality.])
-            ])
-    ])
-
 AM_PATH_PSQL(7.1.0,
     [
         if test "$PG_CFLAGS" != "-I/usr/include"; then
diff -ur cyphesis-0.5.20.orig/Makefile.am cyphesis-0.5.20/Makefile.am
--- cyphesis-0.5.20.orig/Makefile.am	2009-05-11 15:23:20.000000000 +0300
+++ cyphesis-0.5.20/Makefile.am	2009-08-15 12:33:45.000000000 +0300
@@ -3,7 +3,6 @@
           tools tests man dox
 
 docdir = $(datadir)/doc/cyphesis-$(VERSION)
-dist_doc_DATA = README COPYING AUTHORS THANKS NEWS FIXME
 
 rulesetdir = $(datadir)/cyphesis/rulesets
 
