--- Makefile.am.orig	2012-11-12 21:44:40.625023121 -0500
+++ Makefile.am	2012-11-12 21:44:52.862604752 -0500
@@ -9,18 +9,6 @@
 socketdir = $(localstatedir)/tmp
 socket_DATA =
 
-install-data-hook:
-	chmod 1777 $(DESTDIR)$(socketdir)
-	@echo 
-	@echo "Cyphesis is not yet ready to run."
-	@echo "Unless you have configured the server to run without a database, Cyphesis"
-	@echo "requires access to a PostgreSQL database to store rules and account data."
-	@echo "Please run 'sudo $(top_srcdir)/scripts/cyphesis-setup.sh' to create the"
-	@echo "PostgreSQL account and database required, or run it as root."
-	@echo 
-	@echo "For detailed instructions please see the README file."
-	@echo 
-
 EXTRA_DIST = cyphesis.spec Doxyfile autogen.sh \
              cyphesis.init cyclient.init cyphesis.sysconfig \
              scripts/compile_python.py scripts/gen_op.py \
