commit a8e97f672325557b3fbc84b987299350ec5ac10b
Author: Rex Dieter <rdieter@fedoraproject.org>
Date:   Mon Jan 9 11:13:25 2012 -0600

    gcc47 fix, unistd.h header required unconditionally for 'sleep'

diff --git a/strigidaemon/bin/daemon/eventlistener/eventlistenerqueue.cpp b/strigidaemon/bin/daemon/eventlistener/eventlistenerqueue.cpp
index 9fcdcb1..1e97599 100644
--- a/strigidaemon/bin/daemon/eventlistener/eventlistenerqueue.cpp
+++ b/strigidaemon/bin/daemon/eventlistener/eventlistenerqueue.cpp
@@ -27,9 +27,7 @@
 #include <iostream>
 #include <stdio.h>
 
-#if defined(__SUNPRO_CC)
 #include <unistd.h>
-#endif
 
 using namespace std;
 
