Index: yagf-0.9/CMakeLists.txt
===================================================================
--- yagf-0.9.orig/CMakeLists.txt
+++ yagf-0.9/CMakeLists.txt
@@ -21,6 +21,7 @@ qt4_add_resources(yagf_RCC_SRCS ${yagf_R
 include_directories( ${CMAKE_BINARY_DIR} ${PROJECT_SOURCE_DIR})
 add_executable(yagf ${SOURCES} ${UI_HEADERS} ${MOC_SRCS} ${yagf_RCC_SRCS} ${QM_FILES})
 add_library(xspreload SHARED ${INT_SOURCES})
+target_link_libraries(xspreload dl)
 target_link_libraries(yagf ${QT_LIBRARIES} ${ASPELL_LIBRARIES})
 add_custom_target(translations DEPENDS ${QM_FILES})
 qt4_add_translation(${CMAKE_BINARY_DIR}/yagf_ru.qm ${PROJECT_SOURCE_DIR}/src/mainform.cpp ${PROJECT_SOURCE_DIR}/src/popplerdialog.cpp ${MOC_SRCS} ${UI_HEADERS} ${PROJECT_SOURCE_DIR}/src/translations/yagf_ru.ts)
