--- common/lib/modules/fglrx/build_mod/kcl_acpi.c.ORIG	2014-01-21 10:38:08.951838343 +0100
+++ common/lib/modules/fglrx/build_mod/kcl_acpi.c	2014-01-21 10:42:25.728845582 +0100
@@ -775,10 +775,10 @@
 unsigned int ATI_API_CALL KCL_ACPI_GetHandles(kcl_match_info_t *pInfo)
 {
 #if LINUX_VERSION_CODE > KERNEL_VERSION(2,6,12)
-    #if LINUX_VERSION_CODE > KERNEL_VERSION(2,6,19)
-        pInfo->video_handle = pInfo->pcidev->dev.archdata.acpi_handle;
-    #else 
-        pInfo->video_handle = pInfo->pcidev->dev.firmware_data;
+    #if LINUX_VERSION_CODE >= KERNEL_VERSION(3,13,0)
+        pInfo->video_handle = (acpi_handle)ACPI_HANDLE(&pInfo->pcidev->dev);
+    #else
+        pInfo->video_handle = DEVICE_ACPI_HANDLE(&pInfo->pcidev->dev);
     #endif    
     if ( pInfo->video_handle &&
         (KCL_ACPI_videoDevice(pInfo->video_handle) != KCL_ACPI_OK) )
