diff -ru pg_top-3.6.2~/configure.ac pg_top-3.6.2/configure.ac
--- pg_top-3.6.2~/configure.ac	2008-12-21 14:43:47.000000000 +0100
+++ pg_top-3.6.2/configure.ac	2008-12-21 14:44:21.000000000 +0100
@@ -90,7 +90,7 @@
 
 DBCFLAGS="-I$INCLUDEDIR"
 LIBDIR=`$PG_CONFIG --libdir`
-DBLDFLAGS="-L$LIBDIR -lpq"
+DBLDFLAGS="-L$LIBDIR"
 
 AC_SUBST(DBCFLAGS)
 AC_SUBST(DBLDFLAGS)
@@ -109,6 +109,8 @@
 AC_SEARCH_LIBS(inet_aton, resolv, AC_DEFINE(HAVE_INET_ATON, 1,
         [inet_aton is already defined]))
 
+AC_SEARCH_LIBS(PQexec, pq)
+
 # check for libraries required by extension
 extlibs=""
 if test -n "$with_ext" -a -f "${srcdir}/ext/$with_ext.libs"; then
Only in pg_top-3.6.2~: .configure.ac.swp
Only in pg_top-3.6.2~: .configure.swp
Only in pg_top-3.6.2~: .Makefile.am.swp
