--- mg-20140220-orig/sysdef.h
+++ mg-20140220/sysdef.h
@@ -15,7 +15,7 @@
 #  define __USE_GNU
 #endif
 
-#ifdef __APPLE__
+#if defined(__APPLE__) || defined(__FreeBSD__)
 #  define LOGIN_NAME_MAX _POSIX_LOGIN_NAME_MAX
 #endif
 
--- mg-20140220-orig/tags.c
+++ mg-20140220/tags.c
@@ -15,9 +15,12 @@
 #include <err.h>
 #include <stdlib.h>
 #include <string.h>
+#include <stdio.h>
 
-#ifndef LIBBSD_OVERLAY
-#include <util.h>
+#if defined(LIBBSD_OVERLAY) || defined(__FreeBSD__)
+#  include <libutil.h>
+#else
+#  include <util.h>
 #endif
 
 #include "def.h"
