http://bugs.gentoo.org/479946

--- vcl/source/filter/jpeg/JpegReader.cxx
+++ vcl/source/filter/jpeg/JpegReader.cxx
@@ -71,7 +71,7 @@
      * but we don't clear the input buffer.
      * This is correct behavior for reading a series of images from one source.
      */
-    source->start_of_file = sal_True;
+    source->start_of_file = boolean(sal_True);
 }
 
 long StreamRead( SvStream* pStream, void* pBuffer, long nBufferSize )
@@ -119,9 +119,9 @@
 
     source->pub.next_input_byte = source->buffer;
     source->pub.bytes_in_buffer = nbytes;
-    source->start_of_file = sal_False;
+    source->start_of_file = boolean(sal_False);
 
-    return sal_True;
+    return boolean(sal_True);
 }
 
 extern "C" void skip_input_data (j_decompress_ptr cinfo, long numberOfBytes)
--- vcl/source/filter/jpeg/JpegWriter.cxx
+++ vcl/source/filter/jpeg/JpegWriter.cxx
@@ -86,7 +86,7 @@
     destination->pub.next_output_byte = destination->buffer;
     destination->pub.free_in_buffer = BUFFER_SIZE;
 
-    return sal_True;
+    return boolean(sal_True);
 }
 
 extern "C" void term_destination (j_compress_ptr cinfo)
