--- libreoffice-core-3.6.4.3-orig/comphelper/inc/comphelper/scoped_disposing_ptr.hxx	2012-11-28 19:29:53.000000000 +1100
+++ libreoffice-core-3.6.4.3/comphelper/inc/comphelper/scoped_disposing_ptr.hxx	2013-02-09 00:09:13.608271375 +1100
@@ -78,7 +78,7 @@
 
     operator bool () const
     {
-        return m_aItem;
+        return static_cast< bool >(m_aItem);
     }
 
     virtual ~scoped_disposing_ptr()
--- libreoffice-core-3.6.4.3-orig/slideshow/source/engine/animatedsprite.cxx	2012-11-28 19:29:53.000000000 +1100
+++ libreoffice-core-3.6.4.3/slideshow/source/engine/animatedsprite.cxx	2013-02-09 00:09:13.610271414 +1100
@@ -160,7 +160,7 @@
                 }
             }
 
-            return mpSprite;
+            return static_cast< bool >(mpSprite);
         }
 
         void AnimatedSprite::setPixelOffset( const ::basegfx::B2DSize& rPixelOffset )
--- libreoffice-core-3.6.4.3-orig/slideshow/source/inc/shapeattributelayer.hxx	2012-11-28 19:29:53.000000000 +1100
+++ libreoffice-core-3.6.4.3/slideshow/source/inc/shapeattributelayer.hxx	2013-02-09 00:09:13.613271472 +1100
@@ -476,7 +476,7 @@
             // ShapeAttributeLayer(const ShapeAttributeLayer&);
             // ShapeAttributeLayer& operator=( const ShapeAttributeLayer& );
 
-            bool haveChild() const { return mpChild; }
+            bool haveChild() const { return static_cast< bool >(mpChild); }
             void updateStateIds();
 
             template< typename T > T calcValue( const T&                   rCurrValue,
--- libreoffice-core-3.6.4.3-orig/slideshow/source/inc/shapeattributelayerholder.hxx	2012-11-28 19:29:53.000000000 +1100
+++ libreoffice-core-3.6.4.3/slideshow/source/inc/shapeattributelayerholder.hxx	2013-02-09 00:09:13.615271511 +1100
@@ -92,7 +92,7 @@
                 if( mpShape )
                     mpAttributeLayer = mpShape->createAttributeLayer();
 
-                return mpAttributeLayer;
+                return static_cast< bool >(mpAttributeLayer);
             }
 
             ShapeAttributeLayerSharedPtr get() const
--- libreoffice-core-3.6.4.3-orig/slideshow/source/engine/shapesubset.cxx	2012-11-28 19:29:53.000000000 +1100
+++ libreoffice-core-3.6.4.3/slideshow/source/engine/shapesubset.cxx	2013-02-09 00:09:13.617271550 +1100
@@ -113,7 +113,7 @@
                     maTreeNode );
             }
 
-            return mpSubsetShape;
+            return static_cast< bool >(mpSubsetShape);
         }
 
         void ShapeSubset::disableSubsetShape()
--- libreoffice-core-3.6.4.3-orig/slideshow/source/engine/shapes/viewshape.cxx	2012-11-28 19:29:53.000000000 +1100
+++ libreoffice-core-3.6.4.3/slideshow/source/engine/shapes/viewshape.cxx	2013-02-09 00:09:13.621271628 +1100
@@ -187,7 +187,7 @@
                 }
             }
 
-            return io_rCacheEntry.mpRenderer;
+            return static_cast< bool >(io_rCacheEntry.mpRenderer);
         }
 
         bool ViewShape::draw( const ::cppcanvas::CanvasSharedPtr&   rDestinationCanvas,
--- libreoffice-core-3.6.4.3-orig/slideshow/source/engine/slide/slideanimations.cxx	2012-11-28 19:29:53.000000000 +1100
+++ libreoffice-core-3.6.4.3/slideshow/source/engine/slide/slideanimations.cxx	2013-02-09 00:09:13.623271666 +1100
@@ -83,7 +83,7 @@
 
             SHOW_NODE_TREE( mpRootNode );
 
-            return mpRootNode;
+            return static_cast< bool >(mpRootNode);
         }
 
         bool SlideAnimations::isAnimated() const
--- libreoffice-core-3.6.4.3-orig/sc/source/core/data/colorscale.cxx	2012-11-28 19:29:53.000000000 +1100
+++ libreoffice-core-3.6.4.3/sc/source/core/data/colorscale.cxx	2013-02-09 00:09:13.627271744 +1100
@@ -219,7 +219,7 @@
 
 bool ScColorScaleEntry::HasFormula() const
 {
-    return mpCell;
+    return static_cast< bool >(mpCell);
 }
 
 void ScColorScaleEntry::SetMin(bool bMin)
--- libreoffice-core-3.6.4.3-orig/sw/source/core/inc/bookmrk.hxx	2012-11-28 19:29:53.000000000 +1100
+++ libreoffice-core-3.6.4.3/sw/source/core/inc/bookmrk.hxx	2013-02-09 00:09:13.629271783 +1100
@@ -84,7 +84,7 @@
 
             virtual bool IsCoveringPosition(const SwPosition& rPos) const;
             virtual bool IsExpanded() const
-                { return m_pPos2; }
+                { return static_cast< bool >(m_pPos2); }
 
             virtual void SetName(const ::rtl::OUString& rName)
                 { m_aName = rName; }
--- libreoffice-core-3.6.4.3-orig/sd/source/ui/slidesorter/view/SlsButtonBar.cxx	2012-11-28 19:29:53.000000000 +1100
+++ libreoffice-core-3.6.4.3/sd/source/ui/slidesorter/view/SlsButtonBar.cxx	2013-02-09 00:09:13.632271841 +1100
@@ -401,7 +401,7 @@
 
 bool ButtonBar::IsMouseOverButton (void) const
 {
-    return mpButtonUnderMouse;
+    return static_cast< bool >(mpButtonUnderMouse);
 }
 
 
--- libreoffice-core-3.6.4.3-orig/writerfilter/source/dmapper/StyleSheetTable.cxx	2012-11-28 19:29:53.000000000 +1100
+++ libreoffice-core-3.6.4.3/writerfilter/source/dmapper/StyleSheetTable.cxx	2013-02-09 00:10:20.949580346 +1100
@@ -169,7 +169,7 @@
             pStack->push_back(pEntry);
 
         TableStyleSheetEntry* pParent = static_cast<TableStyleSheetEntry *>( pEntry.get( ) );
-            pProps->insert( pParent->GetProperties( nMask ), pStack );
+            pProps->insert( pParent->GetProperties( nMask ), static_cast< bool >(pStack) );
 
             pStack->pop_back();
     }
--- libreoffice-core-3.6.4.3-orig/sc/source/filter/excel/xechart.cxx	2012-11-28 19:29:53.000000000 +1100
+++ libreoffice-core-3.6.4.3/sc/source/filter/excel/xechart.cxx	2013-02-09 00:09:13.645272093 +1100
@@ -691,7 +691,7 @@
 
 bool XclExpChEscherFormat::IsValid() const
 {
-    return maData.mxEscherSet;
+    return static_cast< bool >(maData.mxEscherSet);
 }
 
 void XclExpChEscherFormat::Save( XclExpStream& rStrm )
--- libreoffice-core-3.6.4.3-orig/sc/source/filter/excel/xehelper.cxx	2012-11-28 19:29:53.000000000 +1100
+++ libreoffice-core-3.6.4.3/sc/source/filter/excel/xehelper.cxx	2013-02-09 00:09:13.648272151 +1100
@@ -312,7 +312,7 @@
     if( GetBiff() == EXC_BIFF8 )    // no HLINK records in BIFF2-BIFF7
     {
         // there was/is already a HLINK record
-        mbMultipleUrls = mxLinkRec;
+        mbMultipleUrls = static_cast< bool >(mxLinkRec);
 
         mxLinkRec.reset( new XclExpHyperlink( GetRoot(), rUrlField, maScPos ) );
 
--- libreoffice-core-3.6.4.3-orig/sc/source/filter/excel/xichart.cxx	2012-11-28 19:29:53.000000000 +1100
+++ libreoffice-core-3.6.4.3/sc/source/filter/excel/xichart.cxx	2013-02-09 00:09:13.656272306 +1100
@@ -2698,7 +2698,7 @@
     maType.Finalize( bStockChart );
 
     // extended type info
-    maTypeInfo.Set( maType.GetTypeInfo(), mxChart3d, false );
+    maTypeInfo.Set( maType.GetTypeInfo(), static_cast< bool >(mxChart3d), false );
 
     // reverse series order for some unstacked 2D chart types
     if( maTypeInfo.mbReverseSeries && !Is3dChart() && !maType.IsStacked() && !maType.IsPercent() )
--- libreoffice-core-3.6.4.3-orig/sc/source/filter/inc/xichart.hxx	2012-11-28 19:29:53.000000000 +1100
+++ libreoffice-core-3.6.4.3/sc/source/filter/inc/xichart.hxx	2013-02-09 00:09:13.661272403 +1100
@@ -1240,9 +1240,9 @@
     /** Returns true, if the axis contains caption labels. */
     inline bool         HasLabels() const { return !mxTick || mxTick->HasLabels(); }
     /** Returns true, if the axis shows its major grid lines. */
-    inline bool         HasMajorGrid() const { return mxMajorGrid; }
+    inline bool         HasMajorGrid() const { return static_cast< bool >(mxMajorGrid); }
     /** Returns true, if the axis shows its minor grid lines. */
-    inline bool         HasMinorGrid() const { return mxMinorGrid; }
+    inline bool         HasMinorGrid() const { return static_cast< bool >(mxMinorGrid); }
 
     /** Creates an API axis object. */
     XAxisRef            CreateAxis( const XclImpChTypeGroup& rTypeGroup, const XclImpChAxis* pCrossingAxis ) const;
