diff -Nur muparser.orig/configure muparser/configure
--- muparser.orig/configure	2008-06-09 22:51:39.000000000 +0100
+++ muparser/configure	2009-03-09 16:12:02.335504093 +0000
@@ -1927,10 +1927,6 @@
     NONENONEs,x,x, &&
   program_prefix=${target_alias}-
 
-CFLAGS=
-CXXFLAGS=
-CPPFLAGS=
-
 for ac_prog in gawk mawk nawk awk
 do
   # Extract the first word of "$ac_prog", so it can be a program name with args.
diff -Nur muparser.orig/Makefile.in muparser/Makefile.in
--- muparser.orig/Makefile.in	2008-06-09 22:51:39.000000000 +0100
+++ muparser/Makefile.in	2009-03-09 16:07:55.871757201 +0000
@@ -192,8 +192,8 @@
 all: $(__muParser_lib___depname) $(__muParser_dll___depname) $(__example1___depname)
 
 install: all $(__install_muParser_lib___depname) $(__install_muParser_lib_headers___depname) $(__install_muParser_dll___depname) $(__install_muParser_dll_headers___depname)
-	$(INSTALL_DIR) $(libdir)/pkgconfig
-	$(INSTALL_DATA) build/autoconf/muparser.pc $(libdir)/pkgconfig
+	$(INSTALL_DIR) $(DESTDIR)$(libdir)/pkgconfig
+	$(INSTALL_DATA) build/autoconf/muparser.pc $(DESTDIR)$(libdir)/pkgconfig
 
 uninstall: $(__uninstall_muParser_lib___depname) $(__uninstall_muParser_lib_headers___depname) $(__uninstall_muParser_dll___depname) $(__uninstall_muParser_dll_headers___depname)
 
