diff -ur razorqt-0.5.1.orig/razorqt-panel/CMakeLists.txt razorqt-0.5.1/razorqt-panel/CMakeLists.txt
--- razorqt-0.5.1.orig/razorqt-panel/CMakeLists.txt	2012-10-29 14:43:48.237180971 +0100
+++ razorqt-0.5.1/razorqt-panel/CMakeLists.txt	2012-10-29 15:10:39.384567282 +0100
@@ -33,7 +33,11 @@
 setByDefault(CPULOAD_PLUGIN Yes)
 setByDefault(NETWORKMONITOR_PLUGIN Yes)
 setByDefault(SENSORS_PLUGIN Yes)
+
 setByDefault(VOLUME_PLUGIN Yes)
+# Options for the volume plugin.
+setByDefault(VOLUME_USE_PULSEAUDIO Yes)
+setByDefault(VOLUME_USE_ALSA Yes)
 # *******************************************************************
 
 
@@ -162,8 +166,13 @@
 endif (SENSORS_PLUGIN)
 
 if (VOLUME_PLUGIN)
-  FIND_PACKAGE(PulseAudio)
-  FIND_PACKAGE(ALSA)
+  if (VOLUME_USE_PULSEAUDIO)
+    FIND_PACKAGE(PulseAudio)
+  endif (VOLUME_USE_PULSEAUDIO)
+
+  if (VOLUME_USE_ALSA)
+    FIND_PACKAGE(ALSA)
+  endif (VOLUME_USE_ALSA)
 
   if (PULSEAUDIO_FOUND OR ALSA_FOUND)
     set(ENABLED_PLUGINS ${ENABLED_PLUGINS}   "Volume")
