diff -Naur paxtest-0.9.9.orig/Makefile paxtest-0.9.9/Makefile
--- paxtest-0.9.9.orig/Makefile	2010-02-22 18:47:19.000000000 -0500
+++ paxtest-0.9.9/Makefile	2010-08-09 07:50:53.000000000 -0400
@@ -52,11 +52,13 @@
 endif
 
 # for some reason the .c files need it, else GNU_STACK=RWE
-OPT_FLAGS  := -O2 $(ASFLAGS)
+# Gentoo's CFLAGS should be honored
+OPT_FLAGS  := $(CFLAGS) $(ASFLAGS)
 PTHREAD := -lpthread
 # define stripping of binaries/libs here, or set these on make's commandline,
 # else you'll loose the chpax flags!
-LDFLAGS :=
-SHLDFLAGS :=
+# Gentoo's LDFLAGS should be honored
+LDFLAGS := $(LDFLAGS)
+SHLDFLAGS := $(LDFLAGS)
 ifndef RUNDIR
 RUNDIR := .
@@ -136,7 +138,7 @@
 
 $(EXEC_TESTS) $(MPROT_TESTS): body.o
 	$(CC) $(CFLAGS) -o $@.o -c $@.c
-	$(CC) $(LDFLAGS) $(PTHREAD) -o $@ $< $@.o
+	$(CC) $(LDFLAGS) -o $@ $< $@.o $(PTHREAD)
 	
 $(RAND_TESTS): randbody.o
 	$(CC) $(CFLAGS) -o $@.o -c $@.c
@@ -185,7 +187,7 @@
 	
 $(MPROTSH_TESTS): body.o shlibtest.so
 	$(CC) $(CFLAGS) -o $@.o -c $@.c
-	$(CC) $(LDFLAGS) $(DL) $(PTHREAD) -o $@ $@.o $^
+	$(CC) $(LDFLAGS) -o $@ $@.o $^ $(DL) $(PTHREAD)
 
 # used for RANDEXEC'd binaries
 retbody.o: body.c
@@ -194,12 +196,12 @@
 # build as ET_EXEC (recommended by PaX Team, not really a requirement)
 $(RET_TESTS): retbody.o
 	$(CC) $(CFLAGS) $(CC_ETEXEC) -o $@.o -c $@.c
-	$(CC) $(LDFLAGS) $(LD_ETEXEC) $(PTHREAD) -o $@ $< $@.o
+	$(CC) $(LDFLAGS) $(LD_ETEXEC) -o $@ $< $@.o $(PTHREAD)
 
 # build as ET_EXEC (not in Adamantix's Makefile)
 $(RETX_TESTS): retbody.o
 	$(CC) $(CFLAGS) $(CC_ETEXEC) -o $@.o -c $@.c
-	$(CC) $(LDFLAGS) $(LD_ETEXEC) $(PTHREAD) -o $@ $< $@.o
+	$(CC) $(LDFLAGS) $(LD_ETEXEC) -o $@ $< $@.o $(PTHREAD)
 	-$(PAXBIN) -C $@
 	$(PAXBIN) -SPXM $@
 	
@@ -212,4 +214,4 @@
 	$(CC) $(SHLDFLAGS) -shared -o $@ $<
 
 $(SHLIB_TESTS): body.o $(SHLIBS) shlibbss.o shlibdata.o
-	$(CC) $(LDFLAGS) $(PTHREAD) -o $@ body.o $@.o $(SHLIBS) $(DL)
+	$(CC) $(LDFLAGS) -o $@ body.o $@.o $(SHLIBS) $(DL) $(PTHREAD)
