diff -ruN cups-filters-1.0.29.orig/Makefile.am cups-filters-1.0.29/Makefile.am
--- cups-filters-1.0.29.orig/Makefile.am	2013-01-02 19:12:39.000000000 +0100
+++ cups-filters-1.0.29/Makefile.am	2013-02-02 21:18:22.000000000 +0100
@@ -602,7 +602,7 @@
 # UTILS
 # =====
 
-EXTRA_DIST += utils/cups-browsed.in
+# EXTRA_DIST += utils/cups-browsed.in
 sbin_PROGRAMS =
 if ENABLE_AVAHI
 sbin_PROGRAMS += \
@@ -655,28 +655,9 @@
 	rm -rf *.cache *~
 
 install-data-hook:
-if ENABLE_AVAHI
-if RCLINKS
-	for level in $(RCLEVELS); do \
-		$(INSTALL) -d -m 755 $(DESTDIR)$(INITDIR)/rc$${level}.d; \
-		$(LN_S) ../init.d/cups-browsed $(DESTDIR)$(INITDIR)/rc$${level}.d/S$(RCSTART)cups-browsed; \
-		$(LN_S) ../init.d/cups-browsed $(DESTDIR)$(INITDIR)/rc$${level}.d/K$(RCSTOP)cups-browsed; \
-	done; \
-	$(INSTALL) -d -m 755 $(DESTDIR)$(INITDIR)/rc0.d; \
-	$(LN_S) ../init.d/cups-browsed $(DESTDIR)$(INITDIR)/rc0.d/K$(RCSTOP)cups-browsed;
-endif
-endif
 	$(LN_S) pdf.utf-8.simple \
 		$(DESTDIR)$(pkgcharsetdir)/pdf.utf-8
 
 
 uninstall-hook:
-if ENABLE_AVAHI
-if RCLINKS
-	if test "x$(INITDIR)" != x; then \
-		$(RM) $(DESTDIR)$(BUILDROOT)$(INITDIR)/rc?.d/[SK]??cups-browsed || :; \
-		rmdir $(DESTDIR)$(BUILDROOT)$(INITDIR)/rc?.d || :;\
-	fi
-endif
-endif
 	$(RM) $(DESTDIR)$(pkgcharsetdir)/pdf.utf-8
