diff --git a/test/testall.py b/test/testall.py
index 3e12d78..6031fc3 100755
--- a/test/testall.py
+++ b/test/testall.py
@@ -24,6 +24,12 @@ pkgpath = os.path.dirname(__file__) or '.'
 sys.path.append(pkgpath)
 os.chdir(pkgpath)
 
+try:
+    del sys.modules["beetsplug"]
+except KeyError:
+    pass
+
+
 def suite():
     s = unittest.TestSuite()
     # Get the suite() of every module in this directory beginning with
