diff -ur cowbell-0.2.7.1.orig/resources/cowbell.desktop.in cowbell-0.2.7.1/resources/cowbell.desktop.in
--- cowbell-0.2.7.1.orig/resources/cowbell.desktop.in	2006-04-10 00:58:03.000000000 +0300
+++ cowbell-0.2.7.1/resources/cowbell.desktop.in	2008-12-19 22:00:39.000000000 +0200
@@ -1,5 +1,4 @@
 [Desktop Entry]
-Encoding=UTF-8
 _Name=Cowbell Music Organizer
 _GenericName=Music Organizer
 _Comment=An elegant music organizer
@@ -7,5 +6,5 @@
 Icon=cowbell
 Terminal=false
 Type=Application
-Categories=GNOME;Application;AudioVideo;X-Ximian-Main;X-Red-Hat-Base;
+Categories=GNOME;AudioVideo;X-Ximian-Main;X-Red-Hat-Base;
 StartupNotify=true
