diff -ru softplay-cvs-20080421-orig/Makefile softplay-cvs-20080421/Makefile
--- softplay-cvs-20080421-orig/Makefile	2008-04-21 12:13:43.118686617 +0200
+++ softplay-cvs-20080421/Makefile	2008-04-21 12:14:07.148687085 +0200
@@ -62,7 +62,7 @@
 
 ### recent ffmpegs require -lavutil: 
 
-LIBS += $(shell $(CXX) $(CXXFLAGS) $(INCLUDES) getFFmpegLibs.c -o getFFmpegLibs ; ./getFFmpegLibs)
+LIBS += -lavutil
 
 ### Implicit rules:
 
