diff -ru softplay-cvs-20080421-orig/SoftHandles.h softplay-cvs-20080421/SoftHandles.h
--- softplay-cvs-20080421-orig/SoftHandles.h	2008-04-21 12:05:08.228686856 +0200
+++ softplay-cvs-20080421/SoftHandles.h	2008-04-21 12:05:49.088686674 +0200
@@ -13,7 +13,7 @@
 #define __SOFTHANDLES_H
 
 extern "C" {
-#include <avformat.h>
+#include <libavformat/avformat.h>
 }
 
 #if VDRVERSNUM >= 10330
diff -ru softplay-cvs-20080421-orig/SoftPlayer.h softplay-cvs-20080421/SoftPlayer.h
--- softplay-cvs-20080421-orig/SoftPlayer.h	2008-04-21 12:05:08.228686856 +0200
+++ softplay-cvs-20080421/SoftPlayer.h	2008-04-21 12:05:30.288687156 +0200
@@ -16,7 +16,7 @@
 #include <vdr/player.h>
 
 extern "C" {
-#include <avformat.h>
+#include <libavformat/avformat.h>
 }
 
 #include "SoftHandles.h"
