diff --git a/configure.ac b/configure.ac
index ebec29f..0311cc1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -227,6 +227,11 @@ PKG_CHECK_MODULES([sqlite3], sqlite3)
 AC_SUBST(sqlite3_LIBS)
 AC_SUBST(sqlite3_CFLAGS)
 
+#libx11
+PKG_CHECK_MODULES([x11], x11)
+AC_SUBST(x11_LIBS)
+AC_SUBST(x11_CFLAGS)
+
 # libsexy 
 AC_ARG_ENABLE([system-libsexy],
               [--enable-system-libsexy Use system libsexy instead of bundles.], 
diff --git a/src/Makefile.am b/src/Makefile.am
index 6ef60fd..d63cca7 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -118,7 +118,8 @@ LIBS= \
 	@libgio_LIBS@\
 	@SPIFF_LIBS@\
 	@libxspf_LIBS@\
-    @sqlite3_LIBS@
+	@sqlite3_LIBS@\
+	@x11_LIBS@
 
 if USE_SYSTEM_LIBSEXY
 LIBS += @libsexy_LIBS@
@@ -142,12 +143,13 @@ AM_CFLAGS = @CFLAGS@\
 	@gobject_CFLAGS@\
 	@gtk_CFLAGS@\
 	@gmodule_CFLAGS@\
-    @libxspf_CFLAGS@\
+	@libxspf_CFLAGS@\
 	@libglade_CFLAGS@\
 	@gthread_CFLAGS@\
 	@libsoup_CFLAGS@\
 	@libgio_CFLAGS@\
-    @sqlite3_CFLAGS@\
+	@sqlite3_CFLAGS@\
+	@x11_CFLAGS@\
 	@EXTRA_CFLAGS@\
 	-DPACKAGE_LIB_DIR=\""$(pkglibdir)"\"\
 	-DPACKAGE_DATA_DIR=\""$(datarootdir)"\"
