--- a/fsDialog.cpp	2011-06-12 17:26:23.000000000 +0300
+++ b/fsDialog.cpp	2012-12-09 22:51:40.329612830 +0200
@@ -132,7 +132,7 @@
         QStringList arg;
         arg << cr << pm << "-S";
         p->start(prg,arg);
-        if (! QFile::exists(cr+"/.encfs6.xml")) p->write("\n");
+        if (! (QFile::exists(cr+"/.encfs6.xml") || QFile::exists(cr+"/.encfs5")) ) p->write("\n");
         text.append("\n");
         p->write(text.toLocal8Bit().data());
         p->waitForFinished();
