--- src/doconfig.h.old	2007-01-16 16:17:17.000000000 +0100
+++ src/doconfig.h	2007-01-16 16:17:58.000000000 +0100
@@ -40,7 +40,7 @@
 
 /** Filename of the configuration file.
  *  @sa DATA_DIR */
-#define CFG_FILE_NAME      (user_dir + DATA_DIR "config.cfg").c_str()
+#define CFG_FILE_NAME      (user_dir + "config.cfg").c_str()
 
 /** Default configuration's screen resolution. From this value the
  *  TCONFIG::width and TCONFIG::height are set. */
--- src/dologs.h.old	2007-01-16 16:17:26.000000000 +0100
+++ src/dologs.h	2007-01-16 16:18:12.000000000 +0100
@@ -37,7 +37,7 @@
  *   Directory, where log files are created.
  *   @note Macro is only available if #LOG_TO_LOGFILES is specified.
  */
-# define LOG_PATH        (user_dir + DATA_DIR "logs/").c_str()
+# define LOG_PATH        (user_dir + "logs/").c_str()
 /**
  *   Filename of the error log #err_log.
  *   @note Macro is only available if #LOG_TO_LOGFILES is specified.
--- src/Makefile.old	2007-01-16 16:50:19.000000000 +0100
+++ src/Makefile	2007-01-16 16:51:39.000000000 +0100
@@ -1,12 +1,15 @@
 SOUND ?= 0
 DEBUG ?= 0
 
-CPP = g++ $(CPPFLAGS) $(INCLUDES) $(DEFINES)
+CPP = $(CXX) $(CXXFLAGS) $(INCLUDES) $(DEFINES)
 CPPFLAGS = -g -Wall -O
-DEFINES = -DDATA_DIR='""' -DUNIX=1 -DSOUND=$(SOUND) -DDEBUG=$(DEBUG)
+DEFINES = -DDATA_DIR='"$(GAMES_DATADIR)/$(PN)/"' -DUNIX=1 -DSOUND=$(SOUND) -DDEBUG=$(DEBUG)
 INCLUDES = -I/usr/X11R6/include -I/usr/X11R6/include/GL -I../libs
 LIBPATHS = -L/usr/X11R6/lib -L/usr/lib -L/usr/local/lib -L../libs
-LIBRARIES = -pthread -lglfw -lGL -lX11 -lGLU
+LIBRARIES = -pthread -lglfw -lGL -lX11 -lGLU -lXxf86vm
+ifneq ($(SOUND),0)
+LIBRARIES += -lfmod
+endif
 OBJECTS = doalloc.o doberon.o dobuildings.o doconfig.o dodata.o dodraw.o doengine.o doevents.o dofactories.o dofight.o dofile.o dofollower.o doforces.o dohost.o doipc.o dolayout.o doleader.o dologs.o domap.o domapunits.o domouse.o donet.o doplayers.o doraces.o doschemes.o doselection.o dosimpletypes.o dosound.o dosources.o dounits.o dowalk.o doworkers.o glfont.o glgui.o tga.o utils.o
 TARGETS = ../dark-oberon
 
