fix modpython compilation with forced --as-needed

--- znc-0.098-orig/modules/modpython/Makefile.inc
+++ znc-0.098/modules/modpython/Makefile.inc
@@ -1,7 +1,7 @@
 # vim: filetype=make
 
 ifneq "$(PYCFG)" ""
-PYTHONCOMMON := $(shell $(PYCFG) --includes) $(shell $(PYCFG) --ldflags)
+PYTHONCOMMON := $(shell $(PYCFG) --includes)
 PYTHONCOMMON += -DSWIG_TYPE_TABLE=znc
 PYTHONCOMMON += -Wno-missing-field-initializers -Wno-unused -Wno-shadow
 PYTHONCOMMON += -Wno-missing-declarations -Wno-uninitialized
@@ -56,7 +56,7 @@
 
 modpython/compiler: modpython/compiler.cpp Makefile
 	$(E) Building optimizer for python files...
-	$(Q)$(CXX) $(PYTHONCOMMON) -o $@ $<
+	$(Q)$(CXX) $(PYTHONCOMMON) -o $@ $< $(shell $(PYCFG) --ldflags)
 
 modpython_install: create_install_dir install_metadirs modpython_all
 	-for i in *.pyc; do \
