--- Makefile.in.orig	2012-03-20 14:59:48.308531073 -0400
+++ Makefile.in	2012-03-20 15:03:26.135623522 -0400
@@ -76,8 +76,7 @@ QUILT :=	$(QUILT_IN)
 SRC +=		$(QUILT_SRC:%=quilt/%)
 DIRT +=		$(QUILT_IN:%=quilt/%)
 
-SCRIPTS_IN :=	patchfns inspect dependency-graph edmail	\
-		remove-trailing-ws backup-files
+SCRIPTS_IN :=	patchfns inspect edmail	remove-trailing-ws backup-files
 
 SCRIPTS_SRC :=	$(SCRIPTS_IN:%=%.in)
 SCRIPTS :=	$(SCRIPTS_IN)
--- bash_completion.orig	2012-03-20 15:04:06.994453283 -0400
+++ bash_completion	2012-03-20 15:07:37.335589189 -0400
@@ -95,7 +95,7 @@ _quilt_completion()
     prev=${COMP_WORDS[COMP_CWORD-1]}
 
     # quilt sub commands 
-    cmds='add annotate applied delete diff edit files fold fork graph \
+    cmds='add annotate applied delete diff edit files fold fork \
           grep header import mail new next patches pop previous push refresh \
 	  remove rename revert series setup snapshot top unapplied'
 
@@ -189,19 +189,6 @@ _quilt_completion()
 	        ;;
 	   esac
 	   ;;
-	graph)
-	   case $prev in
-	     -T)
-	     	COMPREPLY=( $( compgen -W "ps" -- $cur ) )
-		;;
-	     --edge-labels)
-	        COMPREPLY=( $( compgen -W "files" -- $cur ) )
-		;;
-	     *)
-		COMPREPLY=( $( compgen -W "-T -h --all --reduce --lines --edge-labels $(quilt applied 2>/dev/null)" -- $cur ) )
-		;;
-	   esac
-	   ;;
 	grep)
 	   _longopt grep
 	   COMPREPLY=( ${COMPREPLY[@]:-} $( compgen -W "-h" -- $cur ) )
