diff -Naur FreeWnn-1.10-pl020.orig/configure FreeWnn-1.10-pl020/configure
--- FreeWnn-1.10-pl020.orig/configure	2002-09-02 02:48:20.000000000 +0900
+++ FreeWnn-1.10-pl020/configure	2004-06-26 11:29:16.742279625 +0900
@@ -2514,7 +2514,7 @@
 CDEBUGFLAGS=${CDEBUGFLAGS:-"-O"}
 case $host in
 *-*-linux*)
-    CCOPTIONS="-Dlinux -D_POSIX_SOURCE -D_BSD_SOURCE"
+    CCOPTIONS="-Dlinux -D_POSIX_SOURCE -D_BSD_SOURCE -D_GNU_SOURCE"
     CDEBUGFLAGS="-O2 -fno-strength-reduce"
     ;;
 *-*-freebsd*|*-*-netbsd*|*-*-openbsd*)
diff -Naur FreeWnn-1.10-pl020.orig/configure.in FreeWnn-1.10-pl020/configure.in
--- FreeWnn-1.10-pl020.orig/configure.in	2002-09-02 02:48:19.000000000 +0900
+++ FreeWnn-1.10-pl020/configure.in	2004-06-26 11:30:21.886225442 +0900
@@ -261,7 +261,7 @@
 CDEBUGFLAGS=${CDEBUGFLAGS:-"-O"}
 case $host in
 *-*-linux*)
-    CCOPTIONS="-Dlinux -D_POSIX_SOURCE -D_BSD_SOURCE"
+    CCOPTIONS="-Dlinux -D_POSIX_SOURCE -D_BSD_SOURCE -D_GNU_SOURCE"
     CDEBUGFLAGS="-O2 -fno-strength-reduce"
     ;;
 *-*-freebsd*|*-*-netbsd*|*-*-openbsd*)
