diff -ur amSynth-1.2.2.orig/configure.in amSynth-1.2.2/configure.in
--- amSynth-1.2.2.orig/configure.in	2009-07-12 19:40:37.000000000 +0300
+++ amSynth-1.2.2/configure.in	2009-08-28 21:13:09.000000000 +0300
@@ -5,7 +5,7 @@
 
 
 addCXXFLAGS="-D_REENTRANT"
-CXXFLAGS="-Wall"
+CXXFLAGS="$CXXFLAGS -Wall"
 
 AC_PROG_CXX
 AC_PROG_INSTALL
@@ -117,11 +117,6 @@
 	profiling=no
 fi
 
-if test $profiling = "no"; then 
-  if test $debugging = "no"; then
-    CXXFLAGS="$CXXFLAGS -O3 -ffast-math -fno-exceptions -fomit-frame-pointer"
-  fi 
-fi
 
 
 dnl
@@ -153,17 +148,19 @@
 fi
 if test "$with_alsa" = "yes"; then 
 	CXXFLAGS="$CXXFLAGS -Dwith_alsa"
-	LDFLAGS="$LDFLAGS -lasound"
+	LIBS="$LIBS -lasound"
 fi
 if test "$with_jack" = "yes"; then 
 	CXXFLAGS="$CXXFLAGS -Dwith_jack"
-	LDFLAGS="$LDFLAGS -ljack"
+	LIBS="$LIBS -ljack"
 fi
 if test "$with_sndfile" = "yes"; then
 	CXXFLAGS="$CXXFLAGS -Dwith_sndfile"
-	LDFLAGS="$LDFLAGS -lsndfile"
+	LIBS="$LIBS -lsndfile"
 fi
 
+AC_SUBST(LIBS)
+
 AC_OUTPUT([
 Makefile
 src/Makefile 
diff -ur amSynth-1.2.2.orig/src/Makefile.am amSynth-1.2.2/src/Makefile.am
--- amSynth-1.2.2.orig/src/Makefile.am	2009-07-12 19:40:37.000000000 +0300
+++ amSynth-1.2.2/src/Makefile.am	2009-08-28 21:14:08.000000000 +0300
@@ -16,12 +16,10 @@
 
 SUBDIRS = drivers VoiceBoard GUI Effects
 
-amSynth_LDADD = drivers/libdrivers.a VoiceBoard/libVoiceBoard.a GUI/libGUI.a Effects/libEffects.a
+amSynth_LDADD = drivers/libdrivers.a VoiceBoard/libVoiceBoard.a GUI/libGUI.a Effects/libEffects.a -lstdc++ -lm -lpthread $(DEPS_LIBS) $(LIBS)
 
 amSynth_CFLAGS = $(CFLAGS) -DENABLE_BINRELOC
 amSynth_CXXFLAGS = $(CXXFLAGS) -DENABLE_BINRELOC
-amSynth_LIBS = -lstdc++ -lm -lpthread
-amSynth_LDFLAGS = $(DEPS_LIBS)
 
 install-exec-hook):
 	chown root $(DESTDIR)$(bindir)/amSynth;\
