Index: amSynth-1.2.0/configure.in
===================================================================
--- amSynth-1.2.0.orig/configure.in
+++ amSynth-1.2.0/configure.in
@@ -5,7 +5,7 @@ AM_CONFIG_HEADER(config.h)
 
 
 addCXXFLAGS="-D_REENTRANT"
-CXXFLAGS="-Wall"
+CXXFLAGS="$CXXFLAGS -Wall"
 
 AC_PROG_CXX
 AC_PROG_INSTALL
@@ -117,12 +117,6 @@ else
 	profiling=no
 fi
 
-if test $profiling = "no"; then 
-  if test $debugging = "no"; then
-    CXXFLAGS="$CXXFLAGS -O3 -ffast-math -fno-exceptions -fomit-frame-pointer"
-  fi 
-fi
-
 
 dnl
 dnl
