--- doc/Makefile.in.orig	2005-07-21 15:36:11.000000000 +0900
+++ doc/Makefile.in	2006-11-05 02:44:15.000000000 +0900
@@ -36,15 +36,15 @@
 	    ../recmkdir $(infodir); fi
 	@for f in `echo $(DOC_INFO)*`; do \
 	   ../$(INSTALL_DATA) $$f  $(infodir); done
-	-@if [ X-`../X-whereisit install-info` != "X-" ] ; then \
-	   install-info $(DOC_INFO) \
-	     --info-dir=$(infodir) \
-	     --section='Miscellaneous' \
-	     --item="* VFlib 3.6: (VFlib-36.info).		A font library VFlib."; \
-	 else \
-	   echo "NOTICE: No 'install-info' program!"; \
-	   echo "        Edit $(infodir)/dir file by yourself."; \
-	 fi
+	#-@if [ X-`../X-whereisit install-info` != "X-" ] ; then \
+	#   install-info $(DOC_INFO) \
+	#     --info-dir=$(infodir) \
+	#     --section='Miscellaneous' \
+	#     --item="* VFlib 3.6: (VFlib-36.info).		A font library VFlib."; \
+	# else \
+	#   echo "NOTICE: No 'install-info' program!"; \
+	#   echo "        Edit $(infodir)/dir file by yourself."; \
+	# fi
 	@if [ ! -d $(runtimedir)/doc ] ; then \
 	    ../recmkdir $(runtimedir)/doc; fi
 	@for f in `echo *.texi *.dvi *.ps *.html`; do \
