--- Rakefile.~1~	2010-02-09 18:58:29.879967846 +0100
+++ Rakefile	2010-02-09 19:00:19.247218285 +0100
@@ -1,23 +1,6 @@
 require 'rubygems'
 require 'hoe'
 
-class Hoe 
-   # Dirty hack to eliminate Hoe from gem dependencies
-   def extra_deps 
-      @extra_deps.delete_if{ |x| x.first == 'hoe' }
-   end
-   
-   # Dirty hack to package only the required files per platform
-   def spec= s 
-      if ENV['PLATFORM'] =~ /win32/
-         s.files = s.files.reject! {|f| f =~ /extconf\.rb/}
-      else
-         s.files = s.files.reject! {|f| f =~ /win\//}
-      end
-      @spec = s
-   end
-end
-
 version = /^== *(\d+\.\d+\.\d+)/.match( File.read( 'History.txt' ) )[1]
 
 Hoe.new('oniguruma', version) do |p|
@@ -34,7 +17,6 @@
   else
      p.spec_extras[:extensions] = ["ext/extconf.rb"]
   end
-  p.rdoc_pattern = /^(lib|bin|ext)|txt$/
   p.changes = p.paragraphs_of('History.txt', 0).join("\n\n")
 end
 
