diff --git a/ccpnmr2.1/c/Makefile b/ccpnmr2.1/c/Makefile
index 9fa8500..7a1a186 100644
--- a/ccpnmr2.1/c/Makefile
+++ b/ccpnmr2.1/c/Makefile
@@ -7,16 +7,16 @@ compile: global structure analysis clouds dynamics
 global:
 	cd memops/global && $(MAKE)
 
-structure:
+structure: global
 	cd ccp/structure && $(MAKE)
 
-analysis:
+analysis: global
 	cd ccpnmr/analysis && $(MAKE)
 
-clouds:
+clouds: global
 	cd ccpnmr/clouds && $(MAKE)
 
-dynamics:
+dynamics: global
 	cd ccpnmr/dynamics && $(MAKE)
 
 clean: global_clean structure_clean analysis_clean clouds_clean dynamics_clean
