diff --git a/ccpnmr2.1/python/ccpnmr/c/linkSharedObjs b/ccpnmr2.1/python/ccpnmr/c/linkSharedObjs
index 8f56168..4490407 100755
--- a/ccpnmr2.1/python/ccpnmr/c/linkSharedObjs
+++ b/ccpnmr2.1/python/ccpnmr/c/linkSharedObjs
@@ -5,7 +5,7 @@ ln -s ../../../c/ccpnmr/analysis/ContourFile.so ContourFile.so
 ln -s ../../../c/ccpnmr/analysis/ContourLevels.so ContourLevels.so
 ln -s ../../../c/ccpnmr/analysis/ContourStyle.so ContourStyle.so
 ln -s ../../../c/ccpnmr/analysis/PeakCluster.so PeakCluster.so
-ln -s ../../../c/ccpnmr/analysis/PeakClusterList.so PeakClusterList.so
+#ln -s ../../../c/ccpnmr/analysis/PeakClusterList.so PeakClusterList.so
 ln -s ../../../c/ccpnmr/analysis/PeakList.so PeakList.so
 ln -s ../../../c/ccpnmr/analysis/SliceFile.so SliceFile.so
 ln -s ../../../c/ccpnmr/analysis/WinPeakList.so WinPeakList.so
