diff --git a/ccpnmr2.1/c/ccpnmr/analysis/py_slice_file.c b/ccpnmr2.1/c/ccpnmr/analysis/py_slice_file.c
index 451b32a..2f37c34 100644
--- a/ccpnmr2.1/c/ccpnmr/analysis/py_slice_file.c
+++ b/ccpnmr2.1/c/ccpnmr/analysis/py_slice_file.c
@@ -46,6 +46,7 @@ Development of a Software Pipeline. Proteins 59, 687 - 696.
 ===========================REFERENCE END===============================
 */
 #include "py_slice_file.h"
+#include "utility.h"
 
 #include "py_block_file.h"
 #include "py_draw_handler.h"
diff --git a/ccpnmr2.1/c/memops/global/gl_handler.c b/ccpnmr2.1/c/memops/global/gl_handler.c
index 48a6a34..278134f 100644
--- a/ccpnmr2.1/c/memops/global/gl_handler.c
+++ b/ccpnmr2.1/c/memops/global/gl_handler.c
@@ -46,6 +46,7 @@ Development of a Software Pipeline. Proteins 59, 687 - 696.
 ===========================REFERENCE END===============================
 */
 #include "gl_handler.h"
+#include "utility.h"
 
 #ifndef IGNORE_GL
 
diff --git a/ccpnmr2.1/c/memops/global/py_block_file.c b/ccpnmr2.1/c/memops/global/py_block_file.c
index 0d3f863..2f2b0f4 100644
--- a/ccpnmr2.1/c/memops/global/py_block_file.c
+++ b/ccpnmr2.1/c/memops/global/py_block_file.c
@@ -49,6 +49,7 @@ Development of a Software Pipeline. Proteins 59, 687 - 696.
 
 #include "py_mem_cache.h"
 #include "python_util.h"
+#include "utility.h"
 
 static PyObject *ErrorObject;   /* locally-raised exception */
 
diff --git a/ccpnmr2.1/c/memops/global/py_draw_handler.c b/ccpnmr2.1/c/memops/global/py_draw_handler.c
index 0d354d1..1d0ec59 100644
--- a/ccpnmr2.1/c/memops/global/py_draw_handler.c
+++ b/ccpnmr2.1/c/memops/global/py_draw_handler.c
@@ -46,6 +46,7 @@ Development of a Software Pipeline. Proteins 59, 687 - 696.
 ===========================REFERENCE END===============================
 */
 #include "py_draw_handler.h"
+#include "utility.h"
 
 #ifndef IGNORE_GL
 #include "py_gl_handler.h"
diff --git a/ccpnmr2.1/c/memops/global/py_shape_file.c b/ccpnmr2.1/c/memops/global/py_shape_file.c
index 32673f5..0e62683 100644
--- a/ccpnmr2.1/c/memops/global/py_shape_file.c
+++ b/ccpnmr2.1/c/memops/global/py_shape_file.c
@@ -46,6 +46,7 @@ Development of a Software Pipeline. Proteins 59, 687 - 696.
 ===========================REFERENCE END===============================
 */
 #include "py_shape_file.h"
+#include "utility.h"
 
 #include "py_mem_cache.h"
 #include "python_util.h"
diff --git a/ccpnmr2.1/c/memops/global/py_store_file.c b/ccpnmr2.1/c/memops/global/py_store_file.c
index 6a66a8c..8fa5973 100644
--- a/ccpnmr2.1/c/memops/global/py_store_file.c
+++ b/ccpnmr2.1/c/memops/global/py_store_file.c
@@ -46,6 +46,7 @@ Development of a Software Pipeline. Proteins 59, 687 - 696.
 ===========================REFERENCE END===============================
 */
 #include "py_store_file.h"
+#include "utility.h"
 
 #include "python_util.h"
 
