Index: webalizer-2.01-10/configure.in
===================================================================
--- webalizer-2.01-10.orig/configure.in
+++ webalizer-2.01-10/configure.in
@@ -71,7 +71,7 @@ if test "$USE_DNS" = "yes"; then
   if test "$HAVE_DB" = ""; then
     AC_CHECK_FUNCS(dbopen)
     if test $ac_cv_func_dbopen = no; then
-      AC_SEARCH_LIBS(dbopen, db db1, OPTS="-DUSE_DNS ${OPTS}"; HAVE_DB="yes")
+      OPTS="-DUSE_DNS ${OPTS}"; HAVE_DB="yes"; LIBS="-ldb ${LIBS}"
       if test "$HAVE_DB" = ""; then
         AC_MSG_WARN(You must have a V1.85 compatable DB library!)
         AC_MSG_WARN(DNS lookup code will be disabled...)
