From 9ca4d9b29caf37f2a35c088f963722234cd48bc5 Mon Sep 17 00:00:00 2001
From: Sebastian Pipping <sebastian@pipping.org>
Date: Wed, 10 Mar 2010 17:28:45 +0100
Subject: [PATCH] etc-webalizer-xtended-conf

---
 sample.conf |    4 ++--
 webalizer.1 |    6 +++---
 webalizer.c |    6 +++---
 3 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/sample.conf b/sample.conf
index 9c984ac..46c1abc 100644
--- a/sample.conf
+++ b/sample.conf
@@ -18,8 +18,8 @@
 # at least one space or tab between the keyword and its value.
 #
 # As of version 0.98, The Webalizer will look for a 'default' configuration
-# file named "webalizer.conf" in the current directory, and if not found
-# there, will look for "/etc/webalizer.conf".
+# file named "webalizer-xtended.conf" in the current directory, and if not found
+# there, will look for "/etc/webalizer-xtended.conf".
 
 
 # LogFile defines the web server log file to use.  If not specified
diff --git a/webalizer.1 b/webalizer.1
index 4756c28..b987429 100644
--- a/webalizer.1
+++ b/webalizer.1
@@ -43,10 +43,10 @@ as a \fBcrond(8)\fP job. Once executed, the general flow of the program is:
 .TP 8
 .B o
 A default configuration file is scanned for.  A file named
-\fIwebalizer.conf\fP is searched for in the current directory, and if
+\fIwebalizer-xtended.conf\fP is searched for in the current directory, and if
 found, and is owned by the invoking user, then its configuration data is parsed.  
 If the file is not present in the current directory, the file 
-\fI/etc/webalizer.conf\fP is searched for and, if found, is used instead.
+\fI/etc/webalizer-xtended.conf\fP is searched for and, if found, is used instead.
 .TP 8
 .B o
 Any command line arguments given to the program are parsed.  This
@@ -785,7 +785,7 @@ Specifies GeoIP database file. Defaults to libGeoIP precompiled one
 (normally \fI/usr/local/share/GeoIP/GeoIP.dat\fP).
 .SH FILES
 .TP 20
-.I webalizer.conf
+.I webalizer-xtended.conf
 Default configuration file.  Is searched for in the current directory
 and if not found, in the \fI/etc/\fP directory.
 .TP 20
diff --git a/webalizer.c b/webalizer.c
index f3a78d2..f837d52 100644
--- a/webalizer.c
+++ b/webalizer.c
@@ -335,11 +335,11 @@ int main(int argc, char *argv[])
    add_nlist("index.",&index_alias);
 
 #ifndef WIN32
-   snprintf(tmp_buf, sizeof(tmp_buf),"%s/webalizer.conf",ETCDIR);
+   snprintf(tmp_buf, sizeof(tmp_buf),"%s/webalizer-xtended.conf",ETCDIR);
 #endif	/* WIN32 */
    /* check for default config file */
-   if (stat("webalizer.conf", &st) == 0 && st.st_uid == getuid())
-     get_config("webalizer.conf");
+   if (stat("webalizer-xtended.conf", &st) == 0 && st.st_uid == getuid())
+     get_config("webalizer-xtended.conf");
 #ifndef WIN32
    else if (!access(tmp_buf,F_OK))
       get_config(tmp_buf);
-- 
1.7.0.1.61.gdc05d.dirty

