diff --git a/vmblock-only/linux/filesystem.c b/vmblock-only/linux/filesystem.c
index b7c535e..a36a01e 100644
--- a/vmblock-only/linux/filesystem.c
+++ b/vmblock-only/linux/filesystem.c
@@ -525,7 +525,11 @@ FsOpReadSuper(struct super_block *sb, // OUT: Superblock object
       return -EINVAL;
    }
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 4, 0)
    rootDentry = d_alloc_root(rootInode);
+#else
+   rootDentry = d_make_root(rootInode);
+#endif
    if (!rootDentry) {
       iput(rootInode);
       return -ENOMEM;
