--- engines/togaII1.2.1a/src/Makefile.old	2012-05-06 20:56:56.190522872 +0200
+++ engines/togaII1.2.1a/src/Makefile	2012-05-06 20:58:20.995568403 +0200
@@ -18,23 +18,16 @@
 
 # general
 
-CXX      = g++
-CXXFLAGS = -pipe -Wall
-LDFLAGS  = -lm
 
 # C++
 
-CXXFLAGS += -fno-exceptions -fno-rtti
 
 # optimisation
 
-CXXFLAGS += -O3 -fstrict-aliasing
-CXXFLAGS += -fomit-frame-pointer
 # CXXFLAGS += -march=athlon-xp # SELECT ME
 
 # strip
 
-LDFLAGS += -s
 
 # dependencies
 
--- tcl/config.tcl.old	2012-05-06 20:59:24.475111275 +0200
+++ tcl/config.tcl	2012-05-06 21:03:04.075692323 +0200
@@ -5,7 +5,7 @@
 if {$windowsOS} {
   set scidShareDir $scidExeDir
 } else {
-  set scidShareDir [file normalize [file join $scidExeDir "../share/scid"]]
+  set scidShareDir "@GENTOO_DATADIR@"
 }
 
 if {![info exists scidBooksDir]} {
--- tcl/start.tcl.old	2012-05-06 21:03:43.770905179 +0200
+++ tcl/start.tcl	2012-05-06 21:04:51.429154649 +0200
@@ -993,7 +993,7 @@
 
 ### Audio move announcement options:
 
-set ::utils::sound::soundFolder [file nativename [file join $::scidExeDir sounds]]
+set ::utils::sound::soundFolder "@GENTOO_DATADIR@/sounds"
 set ::utils::sound::announceNew 0
 set ::utils::sound::announceForward 0
 set ::utils::sound::announceBack 0
@@ -1002,7 +1002,7 @@
 if {$windowsOS} {
   set spellCheckFile [file join $scidDataDir "spelling.ssp"]
 } else {
-  set spellCheckFile "/usr/local/share/scid/spelling.ssp"
+  set spellCheckFile "@GENTOO_DATADIR@/spelling.ssp"
 }
 
 # book configuration
--- tcl/utils/sound.tcl.old	2012-05-06 21:05:31.475342861 +0200
+++ tcl/utils/sound.tcl	2012-05-06 21:06:07.066843851 +0200
@@ -41,7 +41,7 @@
   variable soundFolder
   
   ::splash::add "Setting up audio move announcement..."
-  if {[catch {package require snack 2.0}]} {
+  if {[catch {package require snack}]} {
     set hasSnackPackage 0
     ::splash::add "    Move speech disabled - Snack sound package not found"
     return
--- Makefile.conf.old	2012-05-06 21:06:39.825543746 +0200
+++ Makefile.conf	2012-05-06 21:07:22.668535585 +0200
@@ -212,7 +212,7 @@
 # Small extra programs. Most are written in Tcl using tcscid, but
 # a few contributed ones may be in Python or other languages.
 #
-SCRIPTS= sc_addmove sc_epgn sc_spell sc_eco sc_import sc_remote sc_tree scidpgn pgnfix spliteco spf2spi
+SCRIPTS= sc_addmove sc_epgn sc_spell sc_eco sc_import sc_remote sc_tree scidpgn pgnfix eco2pgn spliteco spf2spi
 
 ####################
 
