--- Makefile.conf.old	2012-02-22 14:12:51.280071808 +0100
+++ Makefile.conf	2012-02-22 14:13:15.432372323 +0100
@@ -379,7 +379,7 @@
 
 scmerge: src/scmerge.o src/misc.o src/index.o src/date.o src/namebase.o \
           src/gfile.o src/bytebuf.o src/textbuf.o src/myassert.o \
-          src/stralloc.o src/position.o
+          src/stralloc.o src/position.o $(OBJS)
 	$(LINK) -o scmerge src/scmerge.o $(OBJS) $(ZLIB) $(TCL_LIBRARY)
 
 pgnscid: src/pgnscid.o $(OBJS)
