diff -ru scid.orig/engines/togaII1.2.1a/src/Makefile scid/engines/togaII1.2.1a/src/Makefile
--- scid.orig/engines/togaII1.2.1a/src/Makefile	2009-07-08 04:28:03.000000000 -0400
+++ scid/engines/togaII1.2.1a/src/Makefile	2009-10-16 02:55:19.130622645 -0400
@@ -16,26 +16,6 @@
 clean:
 	$(RM) *.o .depend gmon.out fruit togaII
 
-# general
-
-CXX      = g++
-CXXFLAGS = -pipe -Wall
-LDFLAGS  = -lm
-
-# C++
-
-CXXFLAGS += -fno-exceptions -fno-rtti
-
-# optimisation
-
-CXXFLAGS += -O3 -fstrict-aliasing
-CXXFLAGS += -fomit-frame-pointer
-# CXXFLAGS += -march=athlon-xp # SELECT ME
-
-# strip
-
-LDFLAGS += -s
-
 # dependencies
 
 $(EXE): $(OBJS)
diff -ru scid.orig/src/scidlet.cpp scid/src/scidlet.cpp
--- scid.orig/src/scidlet.cpp	2009-01-06 03:02:08.000000000 -0500
+++ scid/src/scidlet.cpp	2009-10-16 02:58:56.289745835 -0400
@@ -232,7 +232,7 @@
 };
 
 const configT defaultConfig = {
-    (char *) "scidlet.sbk",  // bookFileName
+    (char *) "@GENTOO_DATADIR@/scidlet.sbk",  // bookFileName
     (char *) "scidlet.log",  // logFileName
     16,    // hashTableMB
     1,     // pawnTableMB
diff -ru scid.orig/tcl/config.tcl scid/tcl/config.tcl
--- scid.orig/tcl/config.tcl	2008-05-05 03:04:12.000000000 -0400
+++ scid/tcl/config.tcl	2009-10-16 02:56:10.800616475 -0400
@@ -5,7 +5,7 @@
 if {$windowsOS} {
   set scidShareDir $scidExeDir
 } else {
-  set scidShareDir [file normalize [file join $scidExeDir "../share/scid"]]
+  set scidShareDir "@GENTOO_DATADIR@"
 }
 
 if {![info exists scidBooksDir]} {
diff -ru scid.orig/tcl/start.tcl scid/tcl/start.tcl
--- scid.orig/tcl/start.tcl	2009-08-25 03:32:55.000000000 -0400
+++ scid/tcl/start.tcl	2009-10-16 02:57:53.763745315 -0400
@@ -922,7 +922,7 @@
 
 ### Audio move announcement options:
 
-set ::utils::sound::soundFolder [file nativename [file join $::scidExeDir sounds]]
+set ::utils::sound::soundFolder "@GENTOO_DATADIR@/sounds"
 set ::utils::sound::announceNew 0
 set ::utils::sound::announceForward 0
 set ::utils::sound::announceBack 0
@@ -931,7 +931,7 @@
 if {$windowsOS} {
   set spellCheckFile [file join $scidDataDir "spelling.ssp"]
 } else {
-  set spellCheckFile "/usr/local/share/scid/spelling.ssp"
+  set spellCheckFile "@GENTOO_DATADIR@/spelling.ssp"
 }
 
 # book configuration
diff -ru scid.orig/tcl/utils/sound.tcl scid/tcl/utils/sound.tcl
--- scid.orig/tcl/utils/sound.tcl	2009-08-23 06:08:39.000000000 -0400
+++ scid/tcl/utils/sound.tcl	2009-10-16 02:56:32.590882174 -0400
@@ -41,7 +41,7 @@
   variable soundFolder
   
   ::splash::add "Setting up audio move announcement..."
-  if {[catch {package require snack 2.0}]} {
+  if {[catch {package require snack}]} {
     set hasSnackPackage 0
     ::splash::add "    Move speech disabled - Snack sound package not found"
     return
--- scid/Makefile.conf.old	2010-04-28 11:02:18.000000000 +0200
+++ scid/Makefile.conf		2010-04-28 11:03:17.000000000 +0200
@@ -199,7 +199,7 @@
 # Small extra programs. Most are written in Tcl using tcscid, but
 # a few contributed ones may be in Python or other languages.
 #
-SCRIPTS= sc_addmove sc_epgn sc_spell sc_eco sc_import sc_remote sc_tree scidpgn pgnfix spliteco spf2spi
+SCRIPTS= sc_addmove sc_epgn sc_spell sc_eco sc_import sc_remote sc_tree scidpgn pgnfix eco2pgn spliteco spf2spi
 
 ####################
 
