From 99c64b99d6cafcccf3a88c8ab2a4283f7dd74376 Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Mon, 27 Sep 2010 00:10:01 +0200
Subject: [PATCH] Add missing definition of pygsfgnome_register_classes

---
 python/gsfgnomemodule.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/python/gsfgnomemodule.c b/python/gsfgnomemodule.c
index 665a003..dbb2c46 100644
--- a/python/gsfgnomemodule.c
+++ b/python/gsfgnomemodule.c
@@ -27,6 +27,8 @@
 extern PyMethodDef pygsfgnome_functions[];
 extern DL_EXPORT(void) initgnome (void);
 
+extern void pygsfgnome_register_classes(PyObject *d);
+
 DL_EXPORT(void)
 initgnome (void)
 {
-- 
1.7.3

