--- gutenprint-5.2.7.orig/src/cups/commandtocanon.c
+++ gutenprint-5.2.7/src/cups/commandtocanon.c
@@ -14,6 +14,7 @@
 #include <cups/cups.h>
 #include <ctype.h>
 #include <stdlib.h>
+#include <stdio.h>
 
 
 /*
--- gutenprint-5.2.7.orig/src/cups/commandtoepson.c
+++ gutenprint-5.2.7/src/cups/commandtoepson.c
@@ -32,6 +32,7 @@
 #include <ctype.h>
 #include <stdlib.h>
 #include <string.h>
+#include <stdio.h>
 
 /*
  * Macros...
