--- QuiteRSS-0.12.2-src.orig/QuiteRSS.pro	2013-02-20 13:26:34.000000000 +0100
+++ QuiteRSS-0.12.2-src/QuiteRSS.pro	2013-03-10 23:32:27.244729948 +0100
@@ -104,7 +104,7 @@
 MOC_DIR = $${BUILD_DIR}/moc/
 RCC_DIR = $${BUILD_DIR}/rcc/
 
-include(3rdparty/sqlite.pri)
+LIBS += -lsqlite3
 include(3rdparty/qtsingleapplication/qtsingleapplication.pri)
 include(3rdparty/qyursqltreeview/qyursqltreeview.pri)
 include(lang/lang.pri)
--- QuiteRSS-0.12.2-src.orig/src/aboutdialog.cpp	2013-02-20 13:26:34.000000000 +0100
+++ QuiteRSS-0.12.2-src/src/aboutdialog.cpp	2013-03-10 23:33:21.156730887 +0100
@@ -1,10 +1,6 @@
 #include "aboutdialog.h"
 #include "VersionNo.h"
-#if QT_VERSION >= 0x040800
-#include <sqlite_qt48x/sqlite3.h>
-#else
-#include <sqlite_qt47x/sqlite3.h>
-#endif
+#include <sqlite3.h>
 #include <qyursqltreeview.h>
 
 AboutDialog::AboutDialog(const QString &lang, QWidget *parent) :
--- QuiteRSS-0.12.2-src.orig/src/dbmemfilethread.cpp	2013-02-20 13:26:34.000000000 +0100
+++ QuiteRSS-0.12.2-src/src/dbmemfilethread.cpp	2013-03-10 23:33:56.868731509 +0100
@@ -1,9 +1,5 @@
 #include "dbmemfilethread.h"
-#if QT_VERSION >= 0x040800
-#include <sqlite_qt48x/sqlite3.h>
-#else
-#include <sqlite_qt47x/sqlite3.h>
-#endif
+#include <sqlite3.h>
 #include <QDebug>
 
 DBMemFileThread::DBMemFileThread(QString filename, QObject *pParent)
