diff -Naur QuiteRSS-0.11.0-src.orig/QuiteRSS.pro QuiteRSS-0.11.0-src/QuiteRSS.pro
--- QuiteRSS-0.11.0-src.orig/QuiteRSS.pro	2013-01-10 02:39:15.021798989 -0800
+++ QuiteRSS-0.11.0-src/QuiteRSS.pro	2013-01-10 02:48:20.244982761 -0800
@@ -90,7 +90,7 @@
 MOC_DIR = $${BUILD_DIR}/moc/
 RCC_DIR = $${BUILD_DIR}/rcc/
 
-include(3rdparty/sqlite.pri)
+LIBS += -lsqlite3
 include(3rdparty/qtsingleapplication/qtsingleapplication.pri)
 include(3rdparty/qyursqltreeview/qyursqltreeview.pri)
 include(lang/lang.pri)
diff -Naur QuiteRSS-0.11.0-src.orig/src/aboutdialog.cpp QuiteRSS-0.11.0-src/src/aboutdialog.cpp
--- QuiteRSS-0.11.0-src.orig/src/aboutdialog.cpp	2013-01-10 02:39:15.029798889 -0800
+++ QuiteRSS-0.11.0-src/src/aboutdialog.cpp	2013-01-10 02:41:44.171934356 -0800
@@ -1,11 +1,7 @@
 #include "aboutdialog.h"
 #include "VersionNo.h"
 #include "VersionRev.h"
-#if QT_VERSION >= 0x040800
-#include <sqlite_qt48x/sqlite3.h>
-#else
-#include <sqlite_qt47x/sqlite3.h>
-#endif
+#include <sqlite3.h>
 #include <qyursqltreeview.h>
 
 AboutDialog::AboutDialog(const QString &lang, QWidget *parent) :
diff -Naur QuiteRSS-0.11.0-src.orig/src/dbmemfilethread.cpp QuiteRSS-0.11.0-src/src/dbmemfilethread.cpp
--- QuiteRSS-0.11.0-src.orig/src/dbmemfilethread.cpp	2013-01-10 02:39:15.030798877 -0800
+++ QuiteRSS-0.11.0-src/src/dbmemfilethread.cpp	2013-01-10 02:42:07.203646420 -0800
@@ -1,9 +1,5 @@
 #include "dbmemfilethread.h"
-#if QT_VERSION >= 0x040800
-#include <sqlite_qt48x/sqlite3.h>
-#else
-#include <sqlite_qt47x/sqlite3.h>
-#endif
+#include <sqlite3.h>
 #include <QDebug>
 
 DBMemFileThread::DBMemFileThread(QObject *pParent)
