Index: haml-3.1.4/metadata
===================================================================
--- haml-3.1.4.orig/metadata
+++ haml-3.1.4/metadata
@@ -20,6 +20,17 @@ date: 2011-11-28 00:00:00 -08:00
 default_executable: 
 dependencies: 
 - !ruby/object:Gem::Dependency 
+  name: sass
+  prerelease: false
+  requirement: &id003 !ruby/object:Gem::Requirement 
+    none: false
+    requirements: 
+    - - "~>"
+      - !ruby/object:Gem::Version 
+        version: 3.1.11
+  type: :runtime
+  version_requirements: *id003
+- !ruby/object:Gem::Dependency 
   name: yard
   prerelease: false
   requirement: &id001 !ruby/object:Gem::Requirement 
