--- src/Makefile.orig	2002-09-22 12:47:57.000000000 -0400
+++ src/Makefile	2003-08-14 01:38:09.610424944 -0400
@@ -3,7 +3,7 @@
 BUILD = 6
 PLUGIN = libpadXwin-${VERSION}.${BUILD}.so
 CFG = cfgPadXwin
-CFLAGS = -fPIC -Wall -O2 -fomit-frame-pointer
+CFLAGS = -fPIC -Wall $(OPTFLAGS)
 OBJECTS = pad.o
 CFGOBJS = conf.o interface.o support.o
 LIBS = -lX11
@@ -14,7 +14,7 @@
 
 plugin: ${OBJECTS}
 	rm -f ${PLUGIN}
-	gcc -shared -Wl,-soname,${PLUGIN} ${CFLAGS} ${OBJECTS} -o ${PLUGIN} ${LIBS}
+	gcc -shared -Wl,-soname,${PLUGIN} ${CFLAGS} ${OBJECTS} -o ${PLUGIN} ${LIBS} ${CFGLIBS}
 	strip --strip-unneeded --strip-debug ${PLUGIN}
 
 cfg: ${CFGOBJS}
