From 2966d3603259430800cb5880491d28a1c5ea2314 Mon Sep 17 00:00:00 2001
From: Behdad Esfahbod <behdad@behdad.org>
Date: Tue, 28 May 2013 21:34:37 +0000
Subject: Fix test build

---
diff --git a/test/api/Makefile.am b/test/api/Makefile.am
index bef7716..4ff14fa 100644
--- a/test/api/Makefile.am
+++ b/test/api/Makefile.am
@@ -28,12 +28,13 @@ TEST_PROGS = \
 	$(NULL)
 
 test_unicode_CPPFLAGS = $(AM_CPPFLAGS)
+test_unicode_LDADD = $(LDADD)
 if HAVE_GLIB
 test_unicode_CPPFLAGS += $(GLIB_CFLAGS)
 endif
 if HAVE_ICU
 test_unicode_CPPFLAGS += $(ICU_CFLAGS)
-test_unicode_LDADD = $(LDADD) $(top_builddir)/src/libharfbuzz-icu.la
+test_unicode_LDADD += $(top_builddir)/src/libharfbuzz-icu.la
 endif
 
 
--
cgit v0.9.0.2-2-gbebe
