From 95dacce13e704e83d640e488ff50ccd16ca8411b Mon Sep 17 00:00:00 2001
From: Ask Solem <ask@celeryproject.org>
Date: Tue, 25 Jun 2013 14:22:49 +0100
Subject: [PATCH] Fixes test suite failures when logging enabled.  Closes #227

diff -ur kombu-2.5.10.orig/kombu/tests/transport/test_pyamqp.py 
kombu-2.5.10/kombu/tests/transport/test_pyamqp.py
--- kombu/tests/transport/test_pyamqp.py	2013-04-12 00:57:56.000000000 +0800
+++ kombu/tests/transport/test_pyamqp.py	2013-06-27 23:39:30.545684336 +0800
@@ -3,8 +3,10 @@
 
 import sys
 
+from functools import partial
 from mock import patch
 from nose import SkipTest
+from itertools import count
 
 try:
     import amqp    # noqa
@@ -13,6 +15,7 @@
 else:
     from kombu.transport import pyamqp
 from kombu import Connection
+from kombu.utils.compat import next
 
 from kombu.tests.utils import TestCase
 from kombu.tests.utils import mask_modules, Mock
@@ -43,6 +46,7 @@
                 pass
 
         self.conn = Mock()
+        self.conn._get_free_channel_id.side_effect = partial(next, count(0))	
         self.conn.channels = {}
         self.channel = Channel(self.conn, 0)
 
