http://bugs.gentoo.org/show_bug.cgi?id=293541

--- Makefile.in
+++ Makefile.in
@@ -46,11 +46,13 @@ LIBKPATHSEADEP=@LIBKPATHSEADEP@
 x_link = $(LDLIBT1) $(LDLIBKPATHSEA) $(x_ldflags) $(x_tool_libs) $(x_xmu_lib) -lXt $(x_pre_libs) $(x_ext_lib) $(x_xpm_libs) -lX11 $(x_extra_libs)
 
 TESTS=./tests/run_tests
-TESTS_DEP=$(TESTS)
+TESTS_DEP=tests
+.PHONY: $(TESTS_DEP)
 
 # various xdvik GUI elements
 LIBGUI=./gui/libgui.a
-LIBGUIDEP=$(LIBGUI)
+LIBGUIDEP=gui
+.PHONY: $(LIBGUIDEP)
 
 
 # Extra xdvi-specific compiler options.
@@ -106,18 +108,14 @@ objects = \
 perlprog = t1mapper
 manpage = xdvi
 
-default all: libguibuild @final_exec_name@ $(manpage).1
+default all: $(LIBGUIDEP) @final_exec_name@ $(manpage).1
 
-$(LIBGUI):
+$(LIBGUIDEP):
 	cd gui; $(MAKE) $(makeargs) libgui.a
 
-$(TESTS):
+$(TESTS_DEP):
 	cd tests; $(MAKE) $(makeargs)
 
-### we need this additional target so that libgui is always checked to be up-to-date ...
-libguibuild:
-	cd gui; $(MAKE) $(makeargs) libgui.a
-
 test:
 	cd tests; $(MAKE) $(makeargs) test
 
