--- a/Makefile	2009-11-20 00:20:32.000000000 +0100
+++ b/Makefile	2010-09-09 22:15:06.917610521 +0200
@@ -9,13 +9,13 @@
 # autodetect GNOME prefix, change this if you want it elsewhere
 PREFIX ?= `pkg-config libgnome-2.0 --variable=prefix || echo /usr`
 
-BINDIR = $(PREFIX)/bin
-LIBDIR = $(PREFIX)/share/blogtk2/lib/blogtk2/
-DATADIR = $(PREFIX)/share/blogtk2/glade
-RESDIR = $(PREFIX)/share/blogtk2/res
-I18NDIR = $(PREFIX)/share/blogtk2/i18n
-APPLICATIONSDIR = $(PREFIX)/share/applications
-ICONDIR = $(PREFIX)/share/pixmaps
+BINDIR = $(DESTDIR)$(PREFIX)/bin
+LIBDIR = $(DESTDIR)$(PREFIX)/share/blogtk2/lib/blogtk2/
+DATADIR = $(DESTDIR)$(PREFIX)/share/blogtk2/glade
+RESDIR = $(DESTDIR)$(PREFIX)/share/blogtk2/res
+I18NDIR = $(DESTDIR)$(PREFIX)/share/blogtk2/i18n
+APPLICATIONSDIR = $(DESTDIR)$(PREFIX)/share/applications
+ICONDIR = $(DESTDIR)$(PREFIX)/share/pixmaps
 
 PYFILES := $(shell $(FIND) . -name "*.py" -print)
 
