diff -NrU5 gloox-0.9.9.5.original/src/md5.cpp gloox-0.9.9.5/src/md5.cpp
--- gloox-0.9.9.5.original/src/md5.cpp	2009-06-18 15:19:28.000000000 -0600
+++ gloox-0.9.9.5/src/md5.cpp	2009-06-18 15:22:11.000000000 -0600
@@ -70,10 +70,11 @@
  */
 
 #include "md5.h"
 
 #include <string.h>
+#include <cstdio> 
 
 namespace gloox
 {
 // #undef BYTE_ORDER	/* 1 = big-endian, -1 = little-endian, 0 = unknown */
 // #ifdef ARCH_IS_BIG_ENDIAN
diff -NrU5 gloox-0.9.9.5.original/src/sha.cpp gloox-0.9.9.5/src/sha.cpp
--- gloox-0.9.9.5.original/src/sha.cpp	2009-06-18 15:19:28.000000000 -0600
+++ gloox-0.9.9.5/src/sha.cpp	2009-06-18 15:31:29.000000000 -0600
@@ -10,10 +10,12 @@
   This software is distributed without any warranty.
 */
 
 #include "sha.h"
 
+#include <cstdio>
+
 namespace gloox
 {
 
   SHA::SHA()
   {
