--- gloox-0.9.9.5b/src/tlsopenssl.cpp	2009-01-21 22:43:06.000000000 -0400
+++ gloox-0.9.9.5/src/tlsopenssl.cpp	2009-01-22 00:42:53.000000000 -0400
@@ -223,7 +223,8 @@
       m_certInfo.date_to = openSSLTime2UnixTime( (char*)(peer->cert_info->validity->notAfter->data) );
       std::string p;
       p.assign( peer_CN );
-      std::transform( p.begin(), p.end(), p.begin(), std::tolower );
+      /* http://lists.debian.org/debian-gcc/2002/04/msg00092.html */
+      std::transform( p.begin(), p.end(), p.begin(), (int(*)(int))std::tolower );
       if( p != m_server )
         m_certInfo.status |= CertWrongPeer;
 
