diff -uarN mysql-connector-odbc-5.3.4-src.orig/cmake/FindMySQL.cmake mysql-connector-odbc-5.3.4-src/cmake/FindMySQL.cmake
--- mysql-connector-odbc-5.3.4-src.orig/cmake/FindMySQL.cmake	2014-06-18 18:50:16.000000000 -0400
+++ mysql-connector-odbc-5.3.4-src/cmake/FindMySQL.cmake	2014-08-20 10:05:00.191485304 -0400
@@ -105,16 +105,16 @@
   # Finally remove the dot
   STRING(REGEX REPLACE "[.]" "" MYSQL_NUM_VERSION "${MYSQL_NUM_VERSION}")
 
-  IF(MYSQL_CXX_LINKAGE OR MYSQL_NUM_VERSION GREATER 50603)
+  IF(MYSQL_CXX_LINKAGE)
     EXECUTE_PROCESS(COMMAND ${MYSQL_CONFIG_EXECUTABLE} "--cxxflags"
             OUTPUT_VARIABLE _mysql_config_output
             )
     SET(MYSQL_CXX_LINKAGE 1)
-  ELSE (MYSQL_CXX_LINKAGE OR MYSQL_NUM_VERSION GREATER 50603)
+  ELSE (MYSQL_CXX_LINKAGE)
     EXECUTE_PROCESS(COMMAND ${MYSQL_CONFIG_EXECUTABLE} "--cflags"
             OUTPUT_VARIABLE _mysql_config_output
             )
-  ENDIF(MYSQL_CXX_LINKAGE OR MYSQL_NUM_VERSION GREATER 50603)
+  ENDIF(MYSQL_CXX_LINKAGE)
 
   # Remove the stl4port dependency
   STRING(REGEX REPLACE "-library=stlport4" "" _mysql_config_output "${_mysql_config_output}")
