diff -Nuar mysql-connector-odbc-5.1.6.orig/qt.m4 mysql-connector-odbc-5.1.6/qt.m4
--- mysql-connector-odbc-5.1.6.orig/qt.m4	2009-11-05 13:34:53.000000000 -0800
+++ mysql-connector-odbc-5.1.6/qt.m4	2010-01-30 22:12:02.105400463 -0800
@@ -91,6 +91,10 @@
   AC_SUBST(QTVERSION)
   AC_SUBST(MOC)
   AC_SUBST(UIC)
+  if expr "$QTVERSION" '>=' "040000" > /dev/null ; then
+    QT_CXXFLAGS="-I$qt_includes -I$qt_includes/QtCore -I$qt_includes/QtGui"
+  else
+    QT_CXXFLAGS="-I$qt_includes"
+  fi
-  QT_CXXFLAGS="-I$qt_includes"
   AC_SUBST(QT_CXXFLAGS)
   QT_LDFLAGS="-L$qt_libraries"
