--- mgl/Makefile.am
+++ mgl/Makefile.am
@@ -16,7 +16,7 @@
 WIDGETS += -DWITH_GLUT
 endif
 
-libmgl_la_LIBADD = $(PNG_LIBS) $(GL_LIBS) $(LOPENMP)
+libmgl_la_LIBADD = $(PNG_LIBS) -lz $(GL_LIBS) $(LOPENMP)
 libmgl_la_LDFLAGS = -no-undefined -version-info $(MGL_CURRENT):$(MGL_REVISION):$(MGL_AGE)
 
 libmgl_la_CXXFLAGS = $(WIDGETS) -DMGL_FONT_PATH=\"$(datadir)/$(PACKAGE)/fonts/\" \
--- mgl/mgl_data_png.cpp
+++ mgl/mgl_data_png.cpp
@@ -18,6 +18,7 @@
  *   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.             *
  ***************************************************************************/
 #include <stdlib.h>
+#include <zlib.h>
 #ifndef NO_PNG
 #include <png.h>
 #endif
