diff -ru a/Makefile b/Makefile
--- a/Makefile	2009-01-23 22:23:43.000000000 -0600
+++ b/Makefile	2009-01-23 22:27:06.000000000 -0600
@@ -1,7 +1,7 @@
 # Makefile for IFStatus
 # Gabriel Montenegro 
 
-LDFLAGS = -lncurses
+LDLIBS = -lncurses
 
 BIN = ifstatus
 OBJ = BorderDecorator.o\
@@ -28,7 +28,7 @@
 all : ifstatus printdone
 
 ifstatus : $(OBJ)
-	$(CXX) $(CXXFLAGS) $(LDFLAGS) $^ -o $@
+	$(LINK.cc) $^ $(LDLIBS) -o $@
 
 printdone:
 	@if [ -e "./${BIN}" ]; then echo "Compiled.."; else echo "Error compiling!"; fi
