Index: openbox-3.5.0/configure.ac
===================================================================
--- openbox-3.5.0.orig/configure.ac
+++ openbox-3.5.0/configure.ac
@@ -168,7 +168,7 @@ AC_ARG_ENABLE(imlib2,
     [--disable-imlib2],
     [disable use of Imlib2 image library for loading icons. [default=enabled]]
   ),
-  [enable_imlib2=$enableeval],
+  [enable_imlib2=$enableval],
   [enable_imlib2=yes]
 )
 
