--- Makefile
+++ Makefile
@@ -6,7 +6,7 @@
 CFLAGS ?= -Wall -O2 -g
 CFLAGS += -DHAVE_GETOPT_LONG -DUSE_SSL
 CFLAGS += -DSETPROCTITLE -DSPT_TYPE=1
-LDFLAGS += -lssl -lcrypto
+LDFLAGS += -lssl $(shell pkg-config libcrypto --libs)
 
 PREFIX =/usr/local
 BINDIR = $(PREFIX)/bin
@@ -27,7 +27,7 @@
 	ntlm.o
 
 proxytunnel: $(OBJ)
-	$(CC) -o $(PROGNAME) $(CFLAGS) $(LDFLAGS) $(OBJ)
+	$(CC) -o $(PROGNAME) $(CFLAGS) $(OBJ) $(LDFLAGS)
 
 clean:		
 	@rm -f $(PROGNAME) $(OBJ)
@@ -34,4 +34,4 @@
 
 install:
 		install -D -m755 $(PROGNAME) $(DESTDIR)$(BINDIR)/$(PROGNAME)
-		install -D -m644 debian/$(PROGNAME).1 $(DESTDIR)$(MANDIR))/$(PROGNAME).1
+		install -D -m644 debian/$(PROGNAME).1 $(DESTDIR)$(MANDIR)/man1/$(PROGNAME).1
