diff -ru taste-0.0.2d-orig/control.h taste-0.0.2d/control.h
--- taste-0.0.2d-orig/control.h	2006-12-03 15:48:00.521763680 +0100
+++ taste-0.0.2d/control.h	2006-12-03 15:48:15.469333968 +0100
@@ -15,7 +15,7 @@
 	const cChannel       *mChannel;
 	const cEvent         *mPresent;
 	const cEvent         *mFollowing;
-	uint64                mStart;
+	uint64_t              mStart;
 	bool                  mSwitch;
   cSkinDisplayChannel  *mOsd;
 
