diff --git a/setup.py b/setup.py
index c2eba94..1e032d7 100644
--- a/setup.py
+++ b/setup.py
@@ -16,15 +16,12 @@ if sys.version_info[:3] < (2, 6, 0):
     sys.exit("Error: Electrum requires Python version >= 2.6.0...")
 
 usr_share = '/usr/share'
-if not os.access(usr_share, os.W_OK):
-    usr_share = os.getenv("XDG_DATA_HOME", os.path.join(os.getenv("HOME"), ".local", "share"))
 
 data_files = []
 if (len(sys.argv) > 1 and (sys.argv[1] == "sdist")) or (platform.system() != 'Windows' and platform.system() != 'Darwin'):
     print "Including all files"
     data_files += [
         (os.path.join(usr_share, 'applications/'), ['electrum.desktop']),
-        (os.path.join(usr_share, 'app-install', 'icons/'), ['icons/electrum.png'])
     ]
     if not os.path.exists('locale'):
         os.mkdir('locale')
@@ -37,7 +34,6 @@ if not os.access(appdata_dir, os.W_OK):
     appdata_dir = os.path.join(usr_share, "electrum")
 
 data_files += [
-    (appdata_dir, ["data/README"]),
     (os.path.join(appdata_dir, "cleanlook"), [
         "data/cleanlook/name.cfg",
         "data/cleanlook/style.css"
