diff -ru catcodec-1.0.0.orig/Makefile catcodec-1.0.0/Makefile
--- catcodec-1.0.0.orig/Makefile	2009-11-05 17:33:48.000000000 -0500
+++ catcodec-1.0.0/Makefile	2010-04-07 12:52:09.839777068 -0400
@@ -19,18 +19,18 @@
 OBJS = catcodec.o io.o sample.o rev.o
 OS = unknown
 
-CFLAGS += -Wall -Wcast-qual -Wwrite-strings -Wno-multichar
+CXXFLAGS += -Wall -Wcast-qual -Wwrite-strings -Wno-multichar
 ifdef DEBUG
-	CFLAGS += -g -ggdb
+	CXXFLAGS += -g -ggdb
 endif
 
 all: $(CATCODEC)
 
 %.o: %.cpp
-	$(CXX) $(CFLAGS) -c -o $@ $^
+	$(CXX) $(CXXFLAGS) -c -o $@ $^
 
 $(CATCODEC): $(OBJS)
-	$(CXX) $(CFLAGS) -o $@ $^
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -o $@ $^
 
 VERSION := $(shell ./findversion.sh | cut -f 1 -d'	')
 RES := $(shell if [ "`cat version.cache 2>/dev/null`" != "$(VERSION)" ]; then echo "$(VERSION)" > version.cache; fi )
