--- m4/ost_socket.m4-orig	2008-11-16 15:24:05.000000000 +0100
+++ m4/ost_socket.m4	2008-11-16 15:24:10.000000000 +0100
@@ -95,6 +95,9 @@
 				AC_CHECK_HEADERS([linux/netfilter_ipv4.h linux/netfilter_ipv6.h],,,
 					[#ifdef HAVE_LIMITS_H
 					#include <limits.h>
+					#endif
+					#ifdef HAVE_NETINET_IN_H
+					#include <netinet/in.h>
 					#endif])
 				if test "$ac_cv_header_linux_netfilter_ipv4_h" = "yes" && 
 				   [ test "$ac_cv_header_linux_netfilter_ipv6_h" = "yes" &&
