diff -Naurp commoncpp2-1.6.3-orig/src/cidr.cpp commoncpp2-1.6.3/src/cidr.cpp
--- commoncpp2-1.6.3-orig/src/cidr.cpp	2008-06-23 07:02:09.000000000 -0600
+++ commoncpp2-1.6.3/src/cidr.cpp	2009-07-30 01:17:49.276744991 -0600
@@ -198,7 +198,7 @@ void IPV4Cidr::set(const char *cp)
 	memset(&netmask, 0, sizeof(netmask));
 	bitset((bit_t *)&netmask, getMask(cp));
 	setString(cbuf, sizeof(cbuf), cp);
-	ep = strchr(cp, '/');
+	ep = (char *)strchr(cp, '/');
 	if(ep)
 		*ep = 0;
 
@@ -326,7 +326,7 @@ void IPV6Cidr::set(const char *cp)
 	memset(&netmask, 0, sizeof(netmask));
 	bitset((bit_t *)&netmask, getMask(cp));
 	setString(cbuf, sizeof(cbuf), cp);
-	ep = strchr(cp, '/');
+	ep = (char *)strchr(cp, '/');
 	if(ep)
 		*ep = 0;
 
