Index: darksnow-0.6.1/Makefile.in
===================================================================
--- darksnow-0.6.1.orig/Makefile.in
+++ darksnow-0.6.1/Makefile.in
@@ -38,7 +38,7 @@ CFLAGS+= -DUSE_TOOLTIP
 all: darksnow
 
 darksnow: darksnow.o interface.o config_files.o tooltips.o
-	$(CC) $(CFLAGS) @GTK_LIBS@ @GTK_CFLAGS@ -DVERSION="\"$(VERSION)\"" darksnow.o interface.o config_files.o tooltips.o -o darksnow 
+	$(CC) $(CFLAGS) $(LDFLAGS) -DVERSION="\"$(VERSION)\"" darksnow.o interface.o config_files.o tooltips.o -o darksnow @GTK_LIBS@ @GTK_CFLAGS@
 
 darksnow.o: darksnow.c
 	$(CC) $(FUNNY) @GTK_CFLAGS@ -DINTLPREFIX="\"$(INTLPREFIX)\"" -DVERSION="\"$(VERSION)\"" -c darksnow.c -o darksnow.o
