--- xwindow/Makefile.in.orig	2006-11-03 13:14:34.000000000 +0900
+++ xwindow/Makefile.in	2006-11-03 13:14:39.000000000 +0900
@@ -42,9 +42,9 @@
 	-DLIBEXECDIR=\"$(LIBEXECDIR)\" -DDATADIR=\"$(DATADIR)\"\
 	-I$(top_srcdir)/mlterm -I${top_builddir}/xwindow -I/usr/local/include
 
-LIBS=$(LIBS_LOCAL) @IND_LIBS@ @FRIBIDI_LIBS@ @IMAGELIB_LIBS@ @AA_LIBS@ \
+LIBS=$(LIBS_LOCAL) @IND_LIBS@ @IMAGELIB_LIBS@ @AA_LIBS@ \
 	@DL_LIBS_IM@ @DL_LIBS_SB@ \
-	@X_LIBS@ @X_PRE_LIBS@ -lX11 @X_EXTRA_LIBS@ $(LMLTERM) $(LMKF) $(LKIK) \
+	@X_LIBS@ @X_PRE_LIBS@ -lX11 @X_EXTRA_LIBS@ $(LMLTERM) @FRIBIDI_LIBS@ $(LMKF) $(LKIK) \
 	-L/usr/local/lib -R/usr/local/lib
 
 PROG = mlterm
