diff -Naur mlterm-3.0.5.orig/contrib/scrollbar/extra/Makefile.in mlterm-3.0.5/contrib/scrollbar/extra/Makefile.in
--- mlterm-3.0.5.orig/contrib/scrollbar/extra/Makefile.in	2011-06-04 01:26:17.000000000 +0900
+++ mlterm-3.0.5/contrib/scrollbar/extra/Makefile.in	2011-06-09 19:35:38.745471044 +0900
@@ -20,7 +20,7 @@
 LIBTOOL = @LIBTOOL@
 
 LIBTOOL_CC = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 .SUFFIXES: .o .c
diff -Naur mlterm-3.0.5.orig/contrib/scrollbar/pixmap_engine/Makefile.in mlterm-3.0.5/contrib/scrollbar/pixmap_engine/Makefile.in
--- mlterm-3.0.5.orig/contrib/scrollbar/pixmap_engine/Makefile.in	2011-06-04 01:26:17.000000000 +0900
+++ mlterm-3.0.5/contrib/scrollbar/pixmap_engine/Makefile.in	2011-06-09 19:35:38.745471044 +0900
@@ -20,7 +20,7 @@
 LIBTOOL = @LIBTOOL@
 
 LIBTOOL_CC = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 .SUFFIXES: .o .c
diff -Naur mlterm-3.0.5.orig/contrib/tool/mlcc/Makefile.in mlterm-3.0.5/contrib/tool/mlcc/Makefile.in
--- mlterm-3.0.5.orig/contrib/tool/mlcc/Makefile.in	2011-06-04 01:26:17.000000000 +0900
+++ mlterm-3.0.5/contrib/tool/mlcc/Makefile.in	2011-06-09 23:29:02.851741057 +0900
@@ -20,7 +20,7 @@
 all: $(PROG)
 
 $(PROG): $(OBJECTS)
-	$(CC) -g -o $(PROG) $(OBJECTS)
+	$(CC) -g -o $(PROG) $(OBJECTS) $(LDFLAGS)
 
 .SUFFIXES: .c.o
 
diff -Naur mlterm-3.0.5.orig/contrib/tool/mlterm-menu/Makefile.in mlterm-3.0.5/contrib/tool/mlterm-menu/Makefile.in
--- mlterm-3.0.5.orig/contrib/tool/mlterm-menu/Makefile.in	2011-06-04 01:26:17.000000000 +0900
+++ mlterm-3.0.5/contrib/tool/mlterm-menu/Makefile.in	2011-06-09 19:35:38.745471044 +0900
@@ -30,7 +30,7 @@
 
 LIBS = $(LIBS1) $(LIBS2@WIN32TAG@)
 
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 PACKAGE = @PACKAGE@
diff -Naur mlterm-3.0.5.orig/contrib/tool/mlterm-zoom/Makefile.in mlterm-3.0.5/contrib/tool/mlterm-zoom/Makefile.in
--- mlterm-3.0.5.orig/contrib/tool/mlterm-zoom/Makefile.in	2011-06-04 01:26:17.000000000 +0900
+++ mlterm-3.0.5/contrib/tool/mlterm-zoom/Makefile.in	2011-06-09 19:35:38.745471044 +0900
@@ -21,7 +21,7 @@
 LIBS = $(LIBS_LOCAL) @X_LIBS@ -lX11 @X_EXTRA_LIBS@
 #LIBS = $(LIBS_LOCAL) @X_LIBS@ @X_PRE_LIBS@ -lX11 @X_EXTRA_LIBS@
 
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 PACKAGE = @PACKAGE@
diff -Naur mlterm-3.0.5.orig/gtk/Makefile.in mlterm-3.0.5/gtk/Makefile.in
--- mlterm-3.0.5.orig/gtk/Makefile.in	2011-06-04 01:26:09.000000000 +0900
+++ mlterm-3.0.5/gtk/Makefile.in	2011-06-09 19:35:38.746471084 +0900
@@ -18,7 +18,7 @@
 	-DSYSCONFDIR=\"$(SYSCONFDIR)\" @VTE_CFLAGS@
 
 LIBTOOL_CC = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 OBJ = vte.o marshal.o vtetypebuiltins.o reaper.o
diff -Naur mlterm-3.0.5.orig/inputmethod/ibus/Makefile.in mlterm-3.0.5/inputmethod/ibus/Makefile.in
--- mlterm-3.0.5.orig/inputmethod/ibus/Makefile.in	2011-06-04 01:26:34.000000000 +0900
+++ mlterm-3.0.5/inputmethod/ibus/Makefile.in	2011-06-09 19:35:38.746471084 +0900
@@ -30,7 +30,7 @@
 LIBTOOL = @LIBTOOL@
 
 LIBTOOL_CC      = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK    = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK    = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 .SUFFIXES: .o .c
diff -Naur mlterm-3.0.5.orig/inputmethod/iiimf/Makefile.in mlterm-3.0.5/inputmethod/iiimf/Makefile.in
--- mlterm-3.0.5.orig/inputmethod/iiimf/Makefile.in	2011-06-04 01:26:35.000000000 +0900
+++ mlterm-3.0.5/inputmethod/iiimf/Makefile.in	2011-06-09 19:35:38.746471084 +0900
@@ -30,7 +30,7 @@
 LIBTOOL = @LIBTOOL@
 
 LIBTOOL_CC      = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK    = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK    = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 .SUFFIXES: .o .c
diff -Naur mlterm-3.0.5.orig/inputmethod/kbd/Makefile.in mlterm-3.0.5/inputmethod/kbd/Makefile.in
--- mlterm-3.0.5.orig/inputmethod/kbd/Makefile.in	2011-06-04 01:26:35.000000000 +0900
+++ mlterm-3.0.5/inputmethod/kbd/Makefile.in	2011-06-09 19:35:38.746471084 +0900
@@ -28,7 +28,7 @@
 LIBTOOL = @LIBTOOL@
 
 LIBTOOL_CC      = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK    = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK    = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 .SUFFIXES: .o .c
diff -Naur mlterm-3.0.5.orig/inputmethod/m17nlib/Makefile.in mlterm-3.0.5/inputmethod/m17nlib/Makefile.in
--- mlterm-3.0.5.orig/inputmethod/m17nlib/Makefile.in	2011-06-04 01:26:37.000000000 +0900
+++ mlterm-3.0.5/inputmethod/m17nlib/Makefile.in	2011-06-09 19:35:38.746471084 +0900
@@ -30,7 +30,7 @@
 LIBTOOL = @LIBTOOL@
 
 LIBTOOL_CC      = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK    = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK    = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 .SUFFIXES: .o .c
diff -Naur mlterm-3.0.5.orig/inputmethod/scim/Makefile.in mlterm-3.0.5/inputmethod/scim/Makefile.in
--- mlterm-3.0.5.orig/inputmethod/scim/Makefile.in	2011-06-04 01:26:34.000000000 +0900
+++ mlterm-3.0.5/inputmethod/scim/Makefile.in	2011-06-09 19:35:38.746471084 +0900
@@ -33,7 +33,7 @@
 
 LIBTOOL_CC      = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
 LIBTOOL_CXX     = $(LIBTOOL) --mode=compile $(CXX) $(CFLAGS)
-LIBTOOL_LINK    = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK    = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 .SUFFIXES: .o .c .cpp
diff -Naur mlterm-3.0.5.orig/inputmethod/uim/Makefile.in mlterm-3.0.5/inputmethod/uim/Makefile.in
--- mlterm-3.0.5.orig/inputmethod/uim/Makefile.in	2011-06-04 01:26:35.000000000 +0900
+++ mlterm-3.0.5/inputmethod/uim/Makefile.in	2011-06-09 19:35:38.747471138 +0900
@@ -30,7 +30,7 @@
 LIBTOOL = @LIBTOOL@
 
 LIBTOOL_CC      = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK    = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK    = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 .SUFFIXES: .o .c
diff -Naur mlterm-3.0.5.orig/kiklib/src/Makefile.in mlterm-3.0.5/kiklib/src/Makefile.in
--- mlterm-3.0.5.orig/kiklib/src/Makefile.in	2011-06-04 01:26:11.000000000 +0900
+++ mlterm-3.0.5/kiklib/src/Makefile.in	2011-06-09 19:35:38.747471138 +0900
@@ -30,7 +30,7 @@
 INC = $(COMOBJ:.o=.h) kik_list.h kik_iterator.h kik_types.h kik_util.h kik_def.h kik_net.h kik_pty.h kik_utmp.h kik_dlfcn.h
 
 LIBTOOL_CC = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LIBS)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LIBS) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 LIBNAME=libkik
diff -Naur mlterm-3.0.5.orig/mkf/lib/Makefile.in mlterm-3.0.5/mkf/lib/Makefile.in
--- mlterm-3.0.5.orig/mkf/lib/Makefile.in	2011-06-04 01:26:16.000000000 +0900
+++ mlterm-3.0.5/mkf/lib/Makefile.in	2011-06-09 19:35:38.747471138 +0900
@@ -52,7 +52,7 @@
 MINOR = 0
 
 LIBTOOL_CC = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 
diff -Naur mlterm-3.0.5.orig/mkf/libtbl/Makefile.in mlterm-3.0.5/mkf/libtbl/Makefile.in
--- mlterm-3.0.5.orig/mkf/libtbl/Makefile.in	2011-06-04 01:26:15.000000000 +0900
+++ mlterm-3.0.5/mkf/libtbl/Makefile.in	2011-06-09 19:35:38.747471138 +0900
@@ -24,7 +24,7 @@
 OBJ = $(JAJP_TABLE_OBJ) $(KOKR_TABLE_OBJ) $(ZH_TABLE_OBJ) $(BYTE_TABLE_OBJ)
 
 LIBTOOL_CC = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 all: libmkf_jajp.la libmkf_kokr.la libmkf_zh.la libmkf_8bits.la
diff -Naur mlterm-3.0.5.orig/mlterm/Makefile.in mlterm-3.0.5/mlterm/Makefile.in
--- mlterm-3.0.5.orig/mlterm/Makefile.in	2011-06-04 01:26:09.000000000 +0900
+++ mlterm-3.0.5/mlterm/Makefile.in	2011-06-09 19:35:38.747471138 +0900
@@ -33,7 +33,7 @@
 MINOR = 0
 
 LIBTOOL_CC = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 
diff -Naur mlterm-3.0.5.orig/scrollbar/sample/Makefile.in mlterm-3.0.5/scrollbar/sample/Makefile.in
--- mlterm-3.0.5.orig/scrollbar/sample/Makefile.in	2011-06-04 01:26:11.000000000 +0900
+++ mlterm-3.0.5/scrollbar/sample/Makefile.in	2011-06-09 19:35:38.747471138 +0900
@@ -33,7 +33,7 @@
 LIBTOOL = @LIBTOOL@
 
 LIBTOOL_CC = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 .SUFFIXES: .o .c
diff -Naur mlterm-3.0.5.orig/tool/mlclient/Makefile.in mlterm-3.0.5/tool/mlclient/Makefile.in
--- mlterm-3.0.5.orig/tool/mlclient/Makefile.in	2011-06-04 01:26:11.000000000 +0900
+++ mlterm-3.0.5/tool/mlclient/Makefile.in	2011-06-09 23:27:57.251467757 +0900
@@ -20,7 +20,7 @@
 all: $(PROG)
 
 $(PROG): $(OBJ)
-	$(CC) -o $(PROG) $(OBJ) $(LIBS)
+	$(CC) -o $(PROG) $(OBJ) $(LIBS) $(LDFLAGS)
 	cp $(PROG) $(PROG)x
 
 .SUFFIXES: .c.o
diff -Naur mlterm-3.0.5.orig/tool/mlconfig/Makefile.in mlterm-3.0.5/tool/mlconfig/Makefile.in
--- mlterm-3.0.5.orig/tool/mlconfig/Makefile.in	2011-06-04 01:26:11.000000000 +0900
+++ mlterm-3.0.5/tool/mlconfig/Makefile.in	2011-06-09 19:35:38.748471200 +0900
@@ -42,7 +42,7 @@
 
 LIBS = $(LIBS1) $(LIBS2@WIN32TAG@)
 
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 PACKAGE = @PACKAGE@
diff -Naur mlterm-3.0.5.orig/tool/mlimgloader/Makefile.in mlterm-3.0.5/tool/mlimgloader/Makefile.in
--- mlterm-3.0.5.orig/tool/mlimgloader/Makefile.in	2011-06-04 01:26:11.000000000 +0900
+++ mlterm-3.0.5/tool/mlimgloader/Makefile.in	2011-06-09 19:35:38.748471200 +0900
@@ -32,7 +32,7 @@
 INSTALL_OPT = @INSTALL_OPT@
 
 LIBTOOL_CC = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 all: $(PROG)
diff -Naur mlterm-3.0.5.orig/tool/mlmenu/Makefile.in mlterm-3.0.5/tool/mlmenu/Makefile.in
--- mlterm-3.0.5.orig/tool/mlmenu/Makefile.in	2011-06-04 01:26:11.000000000 +0900
+++ mlterm-3.0.5/tool/mlmenu/Makefile.in	2011-06-09 19:35:38.748471200 +0900
@@ -18,7 +18,7 @@
 LIBS = $(LIBS_LOCAL) @X_LIBS@ -lX11 @X_EXTRA_LIBS@
 #LIBS = $(LIBS_LOCAL) @X_LIBS@ @X_PRE_LIBS@ -lX11 @X_EXTRA_LIBS@
 
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 PACKAGE = @PACKAGE@
diff -Naur mlterm-3.0.5.orig/xwindow/Makefile.in mlterm-3.0.5/xwindow/Makefile.in
--- mlterm-3.0.5.orig/xwindow/Makefile.in	2011-06-04 01:26:10.000000000 +0900
+++ mlterm-3.0.5/xwindow/Makefile.in	2011-06-09 19:35:38.748471200 +0900
@@ -64,7 +64,7 @@
 INSTALL_OPT = @INSTALL_OPT@
 
 LIBTOOL_CC = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(CFLAGS)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(CFLAGS) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 all: $(PROG)
diff --git a/xwindow/libtype/Makefile.in b/xwindow/libtype/Makefile.in
index 391c1fa..fff76f6 100644
--- a/xwindow/libtype/Makefile.in
+++ b/xwindow/libtype/Makefile.in
@@ -37,7 +37,7 @@ LIBS = $(LIBS_LOCAL) $(LMKF) $(LKIK) -L/usr/local/lib -R/usr/local/lib
 INSTALL_OPT = -m 755
 
 LIBTOOL_CC = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(CFLAGS)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(CFLAGS) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 TYPE_LIBS = @TYPE_LIBS@
