diff -Naur mlterm-3.0.5.orig/contrib/tool/mlcc/Makefile.in mlterm-3.0.5/contrib/tool/mlcc/Makefile.in
--- mlterm-3.0.5.orig/contrib/tool/mlcc/Makefile.in	2011-06-04 01:26:17.000000000 +0900
+++ mlterm-3.0.5/contrib/tool/mlcc/Makefile.in	2011-06-09 23:29:02.851741057 +0900
@@ -20,7 +20,7 @@
 all: $(PROG)
 
 $(PROG): $(OBJECTS)
-	$(CC) -g -o $(PROG) $(OBJECTS)
+	$(CC) -g -o $(PROG) $(OBJECTS) $(LDFLAGS)
 
 .SUFFIXES: .c.o
 
diff -Naur mlterm-3.0.5.orig/tool/mlclient/Makefile.in mlterm-3.0.5/tool/mlclient/Makefile.in
--- mlterm-3.0.5.orig/tool/mlclient/Makefile.in	2011-06-04 01:26:11.000000000 +0900
+++ mlterm-3.0.5/tool/mlclient/Makefile.in	2011-06-09 23:27:57.251467757 +0900
@@ -20,7 +20,7 @@
 all: $(PROG)
 
 $(PROG): $(OBJ)
-	$(CC) -o $(PROG) $(OBJ) $(LIBS)
+	$(CC) -o $(PROG) $(OBJ) $(LIBS) $(LDFLAGS)
 	cp $(PROG) $(PROG)x
 
 .SUFFIXES: .c.o
