diff -Naurb mlterm-3.0.3.orig/contrib/scrollbar/extra/Makefile.in mlterm-3.0.3/contrib/scrollbar/extra/Makefile.in
--- mlterm-3.0.3.orig/contrib/scrollbar/extra/Makefile.in	2011-03-21 07:36:11.000000000 +0900
+++ mlterm-3.0.3/contrib/scrollbar/extra/Makefile.in	2011-05-12 01:34:08.930300204 +0900
@@ -20,7 +20,7 @@
 LIBTOOL = @LIBTOOL@
 
 LIBTOOL_CC = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 .SUFFIXES: .o .c
diff -Naurb mlterm-3.0.3.orig/contrib/scrollbar/pixmap_engine/Makefile.in mlterm-3.0.3/contrib/scrollbar/pixmap_engine/Makefile.in
--- mlterm-3.0.3.orig/contrib/scrollbar/pixmap_engine/Makefile.in	2011-03-21 07:36:11.000000000 +0900
+++ mlterm-3.0.3/contrib/scrollbar/pixmap_engine/Makefile.in	2011-05-12 01:34:08.930300204 +0900
@@ -20,7 +20,7 @@
 LIBTOOL = @LIBTOOL@
 
 LIBTOOL_CC = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 .SUFFIXES: .o .c
diff -Naurb mlterm-3.0.3.orig/contrib/tool/mlterm-menu/Makefile.in mlterm-3.0.3/contrib/tool/mlterm-menu/Makefile.in
--- mlterm-3.0.3.orig/contrib/tool/mlterm-menu/Makefile.in	2011-03-21 07:36:11.000000000 +0900
+++ mlterm-3.0.3/contrib/tool/mlterm-menu/Makefile.in	2011-05-12 01:34:08.926300233 +0900
@@ -30,7 +30,7 @@
 
 LIBS = $(LIBS1) $(LIBS2@WIN32TAG@)
 
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 PACKAGE = @PACKAGE@
diff -Naurb mlterm-3.0.3.orig/contrib/tool/mlterm-zoom/Makefile.in mlterm-3.0.3/contrib/tool/mlterm-zoom/Makefile.in
--- mlterm-3.0.3.orig/contrib/tool/mlterm-zoom/Makefile.in	2011-03-21 07:36:11.000000000 +0900
+++ mlterm-3.0.3/contrib/tool/mlterm-zoom/Makefile.in	2011-05-12 01:34:08.926300233 +0900
@@ -21,7 +21,7 @@
 LIBS = $(LIBS_LOCAL) @X_LIBS@ -lX11 @X_EXTRA_LIBS@
 #LIBS = $(LIBS_LOCAL) @X_LIBS@ @X_PRE_LIBS@ -lX11 @X_EXTRA_LIBS@
 
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 PACKAGE = @PACKAGE@
diff -Naurb mlterm-3.0.3.orig/gtk/Makefile.in mlterm-3.0.3/gtk/Makefile.in
--- mlterm-3.0.3.orig/gtk/Makefile.in	2011-03-21 07:35:56.000000000 +0900
+++ mlterm-3.0.3/gtk/Makefile.in	2011-05-12 01:34:08.938300163 +0900
@@ -18,7 +18,7 @@
 	`pkg-config vte --cflags`
 
 LIBTOOL_CC = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 OBJ = vte.o marshal.o vtetypebuiltins.o reaper.o
diff -Naurb mlterm-3.0.3.orig/inputmethod/ibus/Makefile.in mlterm-3.0.3/inputmethod/ibus/Makefile.in
--- mlterm-3.0.3.orig/inputmethod/ibus/Makefile.in	2011-03-21 07:37:01.000000000 +0900
+++ mlterm-3.0.3/inputmethod/ibus/Makefile.in	2011-05-12 01:34:08.926300233 +0900
@@ -30,7 +30,7 @@
 LIBTOOL = @LIBTOOL@
 
 LIBTOOL_CC      = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK    = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK    = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 .SUFFIXES: .o .c
diff -Naurb mlterm-3.0.3.orig/inputmethod/iiimf/Makefile.in mlterm-3.0.3/inputmethod/iiimf/Makefile.in
--- mlterm-3.0.3.orig/inputmethod/iiimf/Makefile.in	2011-03-21 07:37:01.000000000 +0900
+++ mlterm-3.0.3/inputmethod/iiimf/Makefile.in	2011-05-12 01:34:08.926300233 +0900
@@ -30,7 +30,7 @@
 LIBTOOL = @LIBTOOL@
 
 LIBTOOL_CC      = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK    = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK    = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 .SUFFIXES: .o .c
diff -Naurb mlterm-3.0.3.orig/inputmethod/kbd/Makefile.in mlterm-3.0.3/inputmethod/kbd/Makefile.in
--- mlterm-3.0.3.orig/inputmethod/kbd/Makefile.in	2011-03-21 07:37:01.000000000 +0900
+++ mlterm-3.0.3/inputmethod/kbd/Makefile.in	2011-05-12 01:34:08.926300233 +0900
@@ -28,7 +28,7 @@
 LIBTOOL = @LIBTOOL@
 
 LIBTOOL_CC      = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK    = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK    = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 .SUFFIXES: .o .c
diff -Naurb mlterm-3.0.3.orig/inputmethod/m17nlib/Makefile.in mlterm-3.0.3/inputmethod/m17nlib/Makefile.in
--- mlterm-3.0.3.orig/inputmethod/m17nlib/Makefile.in	2011-03-21 07:37:01.000000000 +0900
+++ mlterm-3.0.3/inputmethod/m17nlib/Makefile.in	2011-05-12 01:34:08.926300233 +0900
@@ -30,7 +30,7 @@
 LIBTOOL = @LIBTOOL@
 
 LIBTOOL_CC      = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK    = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK    = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 .SUFFIXES: .o .c
diff -Naurb mlterm-3.0.3.orig/inputmethod/scim/Makefile.in mlterm-3.0.3/inputmethod/scim/Makefile.in
--- mlterm-3.0.3.orig/inputmethod/scim/Makefile.in	2011-03-21 07:37:01.000000000 +0900
+++ mlterm-3.0.3/inputmethod/scim/Makefile.in	2011-05-12 01:34:08.922300253 +0900
@@ -33,7 +33,7 @@
 
 LIBTOOL_CC      = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
 LIBTOOL_CXX     = $(LIBTOOL) --mode=compile $(CXX) $(CFLAGS)
-LIBTOOL_LINK    = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK    = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 .SUFFIXES: .o .c .cpp
diff -Naurb mlterm-3.0.3.orig/inputmethod/uim/Makefile.in mlterm-3.0.3/inputmethod/uim/Makefile.in
--- mlterm-3.0.3.orig/inputmethod/uim/Makefile.in	2011-03-21 07:37:01.000000000 +0900
+++ mlterm-3.0.3/inputmethod/uim/Makefile.in	2011-05-12 01:34:08.926300233 +0900
@@ -30,7 +30,7 @@
 LIBTOOL = @LIBTOOL@
 
 LIBTOOL_CC      = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK    = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK    = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 .SUFFIXES: .o .c
diff -Naurb mlterm-3.0.3.orig/kiklib/src/Makefile.in mlterm-3.0.3/kiklib/src/Makefile.in
--- mlterm-3.0.3.orig/kiklib/src/Makefile.in	2011-03-21 07:35:59.000000000 +0900
+++ mlterm-3.0.3/kiklib/src/Makefile.in	2011-05-12 01:34:08.934300183 +0900
@@ -30,7 +30,7 @@
 INC = $(COMOBJ:.o=.h) kik_list.h kik_iterator.h kik_types.h kik_util.h kik_def.h kik_net.h kik_pty.h kik_utmp.h kik_dlfcn.h
 
 LIBTOOL_CC = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LIBS)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LIBS) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 LIBNAME=libkik
diff -Naurb mlterm-3.0.3.orig/mkf/lib/Makefile.in mlterm-3.0.3/mkf/lib/Makefile.in
--- mlterm-3.0.3.orig/mkf/lib/Makefile.in	2011-03-21 07:36:07.000000000 +0900
+++ mlterm-3.0.3/mkf/lib/Makefile.in	2011-05-12 01:34:08.938300163 +0900
@@ -54,7 +54,7 @@
 MINOR = 0
 
 LIBTOOL_CC = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 
diff -Naurb mlterm-3.0.3.orig/mkf/libtbl/Makefile.in mlterm-3.0.3/mkf/libtbl/Makefile.in
--- mlterm-3.0.3.orig/mkf/libtbl/Makefile.in	2011-03-21 07:36:06.000000000 +0900
+++ mlterm-3.0.3/mkf/libtbl/Makefile.in	2011-05-12 01:34:08.934300183 +0900
@@ -24,7 +24,7 @@
 OBJ = $(JAJP_TABLE_OBJ) $(KOKR_TABLE_OBJ) $(ZH_TABLE_OBJ) $(BYTE_TABLE_OBJ)
 
 LIBTOOL_CC = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 all: libmkf_jajp.la libmkf_kokr.la libmkf_zh.la libmkf_8bits.la
diff -Naurb mlterm-3.0.3.orig/mlterm/Makefile.in mlterm-3.0.3/mlterm/Makefile.in
--- mlterm-3.0.3.orig/mlterm/Makefile.in	2011-03-21 07:35:54.000000000 +0900
+++ mlterm-3.0.3/mlterm/Makefile.in	2011-05-12 01:34:08.930300204 +0900
@@ -33,7 +33,7 @@
 MINOR = 0
 
 LIBTOOL_CC = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 
diff -Naurb mlterm-3.0.3.orig/scrollbar/sample/Makefile.in mlterm-3.0.3/scrollbar/sample/Makefile.in
--- mlterm-3.0.3.orig/scrollbar/sample/Makefile.in	2011-03-21 07:35:58.000000000 +0900
+++ mlterm-3.0.3/scrollbar/sample/Makefile.in	2011-05-12 01:34:08.934300183 +0900
@@ -33,7 +33,7 @@
 LIBTOOL = @LIBTOOL@
 
 LIBTOOL_CC = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 .SUFFIXES: .o .c
diff -Naurb mlterm-3.0.3.orig/tool/mlconfig/Makefile.in mlterm-3.0.3/tool/mlconfig/Makefile.in
--- mlterm-3.0.3.orig/tool/mlconfig/Makefile.in	2011-03-21 07:35:58.000000000 +0900
+++ mlterm-3.0.3/tool/mlconfig/Makefile.in	2011-05-12 01:34:08.930300204 +0900
@@ -42,7 +42,7 @@
 
 LIBS = $(LIBS1) $(LIBS2@WIN32TAG@)
 
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 PACKAGE = @PACKAGE@
diff -Naurb mlterm-3.0.3.orig/tool/mlimgloader/Makefile.in mlterm-3.0.3/tool/mlimgloader/Makefile.in
--- mlterm-3.0.3.orig/tool/mlimgloader/Makefile.in	2011-03-21 07:35:58.000000000 +0900
+++ mlterm-3.0.3/tool/mlimgloader/Makefile.in	2011-05-12 01:34:08.934300183 +0900
@@ -32,7 +32,7 @@
 INSTALL_OPT = @INSTALL_OPT@
 
 LIBTOOL_CC = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 all: $(PROG)
diff -Naurb mlterm-3.0.3.orig/tool/mlmenu/Makefile.in mlterm-3.0.3/tool/mlmenu/Makefile.in
--- mlterm-3.0.3.orig/tool/mlmenu/Makefile.in	2011-03-21 07:35:58.000000000 +0900
+++ mlterm-3.0.3/tool/mlmenu/Makefile.in	2011-05-12 01:34:08.934300183 +0900
@@ -18,7 +18,7 @@
 LIBS = $(LIBS_LOCAL) @X_LIBS@ -lX11 @X_EXTRA_LIBS@
 #LIBS = $(LIBS_LOCAL) @X_LIBS@ @X_PRE_LIBS@ -lX11 @X_EXTRA_LIBS@
 
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 PACKAGE = @PACKAGE@
diff -Naurb mlterm-3.0.3.orig/xwindow/Makefile.in mlterm-3.0.3/xwindow/Makefile.in
--- mlterm-3.0.3.orig/xwindow/Makefile.in	2011-03-21 07:35:57.000000000 +0900
+++ mlterm-3.0.3/xwindow/Makefile.in	2011-05-12 01:34:08.930300204 +0900
@@ -64,7 +64,7 @@
 INSTALL_OPT = @INSTALL_OPT@
 
 LIBTOOL_CC = $(LIBTOOL) --mode=compile $(CC) $(CFLAGS)
-LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC)
+LIBTOOL_LINK = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS)
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install $(INSTALL)
 
 all: $(PROG)
