--- src/generator68k/Makefile.am.old	2006-10-26 21:52:24.000000000 +0200
+++ src/generator68k/Makefile.am	2006-10-26 21:53:45.000000000 +0200
@@ -66,7 +66,10 @@
 
 #def68k : def68k.c
 #	gcc def68k.c tab68k.c -o def68k -I .. -I ../.. -I . `sdl-config --cflags`
-def68k.o gen68k.o: %.o : %.c
+def68k.o: %.o : %.c
+	$(CC_FOR_BUILD) -c $< -o $@ -I .. -I ../.. -I . `sdl-config --cflags`
+
+gen68k.o: gen68k.c def68k-iibs.h
 	$(CC_FOR_BUILD) -c $< -o $@ -I .. -I ../.. -I . `sdl-config --cflags`
 
 def68k : def68k.o
