--- configure.in
+++ configure.in
@@ -17,13 +17,14 @@
 AC_SUBST(CC_FOR_BUILD)
 
 # Humm is this a good idea?
-CFLAGS=""
+CFLAGS="$CFLAGS -Wall -Wno-unused"
 
  #echo $CC
 
 # Checks for programs.
 AC_PROG_CC
 AC_PROG_RANLIB
+AM_PROG_AS
 
 
 
@@ -213,33 +214,14 @@
     i?86)  AC_MSG_RESULT(Turning on x86 processor optimisations)
            AC_DEFINE(PROCESSOR_INTEL, 1, [Define if you have an x86 processor])
            ALIGNLONGS=0
-	   if test $CC_MAJOR -ge 3 ; then
-		CFLAGS="$CFLAGS -O3 -fstrength-reduce -frerun-loop-opt -Wall -Wno-unused -funroll-loops \
-                    -ffast-math -falign-functions=2 -falign-jumps=2 -fexpensive-optimizations \
-                    -falign-loops=2 -fschedule-insns2 -malign-double -fomit-frame-pointer -g \
-		    -fno-strict-aliasing -pipe -mtune=$target_cpu -DCPU=$target_cpu"
-	   else
-		CFLAGS="$CFLAGS -O3 -fstrength-reduce -frerun-loop-opt -Wall -Wno-unused -funroll-loops \
-                    -ffast-math -malign-functions=2 -malign-jumps=2 -fexpensive-optimizations \
-                    -malign-loops=2 -fschedule-insns2 -malign-double -fomit-frame-pointer -g \
-		    -fno-strict-aliasing -pipe -mcpu=$target_cpu -DCPU=$target_cpu"
-           fi
            ;;
     sparc*) AC_MSG_RESULT(Turning on sparc processor optimisations)
            AC_DEFINE(PROCESSOR_SPARC, 1, [Define if you have a sparc processor])
 	   ALIGNLONGS=1
-	   if test $target_cpu = sparc64 && test $CC_MAJOR -ge 3 ; then
-		CFLAGS="$CFLAGS -mcpu=v9"
-	   fi
-	   CFLAGS="$CFLAGS -O3 -Wall -Wno-unused -funroll-loops \
-                    -ffast-math -fomit-frame-pointer -g"
            ;;
     arm)   AC_MSG_RESULT(Turning on arm processor optimisations)
            #AC_DEFINE(PROCESSOR_ARM, 1, [Define if you have an ARM processor])
            ALIGNLONGS=1
-	   CFLAGS="$CFLAGS -O3 -Wall -ftracer -fstrength-reduce -Wno-unused -funroll-loops \
-                    -fomit-frame-pointer -fstrict-aliasing -ffast-math \
-	   	    "
 	   #CFLAGS="$CFLAGS -O2 -Wall -ftracer -fstrength-reduce -Wno-unused  \
            #         -fomit-frame-pointer -fstrict-aliasing -ffast-math \
 	   #	    "
@@ -248,8 +230,6 @@
     *)     AC_MSG_WARN(Processor type not known - Use generic optimisations)
            ALIGNLONGS=1
            optimum=no
-	   CFLAGS="$CFLAGS -O3 -Wall -Wno-unused -funroll-loops \
-                    -ffast-math -g"
            ;;
   esac
   
