diff -burN SOAP-Lite-0.60/Makefile.PL.old SOAP-Lite-0.60/Makefile.PL
--- SOAP-Lite-0.60/Makefile.PL.old	2004-10-29 11:07:34.665925280 -0400
+++ SOAP-Lite-0.60/Makefile.PL	2004-10-29 11:08:02.491695120 -0400
@@ -22,7 +22,7 @@
   ["MIME messages","SOAP::MIMEParser",{"MIME::Parser" => 0},1],
   ["SSL Support for TCP Transport","SOAP::Transport::TCP",{"IO::Socket::SSL" => 0},0],
   ["Compression support for HTTP","SOAP::Transport::HTTP",{"Compress::Zlib" => 0},0],
-  ["MIME interoperability w/ Axis","SOAP::Lite",{"MIME::Parser" => "6.106"},0],
+  ["MIME interoperability w/ Axis","SOAP::Lite",{"MIME::Parser" => "5.413"},0],
 );
 
 use Getopt::Long;
