diff -ur darkice-0.18.1.orig/src/JackDspSource.cpp darkice-0.18.1/src/JackDspSource.cpp
--- darkice-0.18.1.orig/src/JackDspSource.cpp	2007-02-19 09:56:49.000000000 +0200
+++ darkice-0.18.1/src/JackDspSource.cpp	2008-03-27 12:29:53.000000000 +0200
@@ -31,6 +31,8 @@
 
 /* ============================================================ include files */
 
+#include <climits>
+
 #include "AudioSource.h"
 
 #ifdef SUPPORT_JACK_DSP
