diff -ur darkice-0.19.orig/src/SerialUlaw.cpp darkice-0.19/src/SerialUlaw.cpp
--- darkice-0.19.orig/src/SerialUlaw.cpp	2007-04-26 11:10:32.000000000 +0300
+++ darkice-0.19/src/SerialUlaw.cpp	2009-07-30 20:07:30.000000000 +0300
@@ -40,6 +40,8 @@
 #include "config.h"
 #endif
 
+#include <stdio.h>
+
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
 #else
diff -ur darkice-0.19.orig/src/Util.cpp darkice-0.19/src/Util.cpp
--- darkice-0.19.orig/src/Util.cpp	2007-02-25 18:38:33.000000000 +0200
+++ darkice-0.19/src/Util.cpp	2009-07-30 20:06:44.000000000 +0300
@@ -296,7 +296,7 @@
     unsigned int    size;
     char          * s;
     char          * strdate;
-    char          * last;
+    const char    * last;
     time_t          now;
 
     if ( !str ) {
