From c0ca954710f83a9204ea01c2d6db717b728d0313 Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <gilles.dartiguelongue@esiee.org>
Date: Sat, 10 Mar 2012 16:00:12 +0100
Subject: [PATCH 2/3] Add missing dependency for e-map intermediate library

---
 configure.in |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/configure.in b/configure.in
index 2d60648..3a7d1a6 100644
--- a/configure.in
+++ b/configure.in
@@ -79,6 +79,7 @@ dnl =====================================================
 
 PKG_CHECK_MODULES(GST_DEPENDS,[
 			   gtk+-2.0 >= $GTK_REQUIRED
+			   atk
 			   gmodule-export-2.0
 			   gio-2.0 >= $GLIB_REQUIRED
 			   pango
-- 
1.7.8.5

