diff --git a/Makefile.in b/Makefile.in
index 434b8e5..a384b50 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -54,7 +54,7 @@ libmix++.a: mix/net.h $(OBJS_PP_A)
 	${RANLIB} libmix++.a
 
 libtest: libtest.o
-	$(CXX) $(CXXFLAGS) $(CCOPTS) -static -I. -L. libtest++.o -o libtest -lmix++  ${CLIB}
+	$(CXX) $(CXXFLAGS) $(CCOPTS) -static -I. -L. libtest++.o -o libtest -lmix++  $(CLIB)
 
 libtest.o:
 	$(CC) $(CCOPTS) $(CFLAGS) -c libtest.c -o libtest.o
diff --git a/configure.in b/configure.in
index 55fb7dd..e9bd924 100644
--- a/configure.in
+++ b/configure.in
@@ -28,7 +28,7 @@ AC_ARG_WITH([net2],
 AS_IF([test "x$with_net2" = "xyes"], [
   AC_DEFINE(LM_USE_NET2)
   CLIB="-lnet-1.0 -lpcap"
-])
+], [CLIB=""])
 
 AC_ARG_ENABLE([static],
 	AS_HELP_STRING([--enable-static], [Build static libs]))
