--- Makefile.orig	2010-02-23 11:13:31.000000000 +0100
+++ Makefile	2010-02-23 11:13:37.000000000 +0100
@@ -1,6 +1,4 @@
 UNAME = $(shell uname)
-CC = gcc
-CFLAGS = -g -Wall -O2
 LIBS = -lz
 OFILES = util.o keywords.o flasm.o unflasm.o lex.yy.o assembler.tab.o
 GARBAGE = assembler.tab.* lex.yy.c memwatch.o gmon.out memwatch.log core
@@ -9,8 +7,6 @@
 ifneq (,$(findstring debug,$(MAKECMDGOALS)))
 	CFLAGS += -DMEMWATCH -pg -p -pedantic -W -Wcast-align -Wcast-qual -Wshadow -Wnested-externs -Wstrict-prototypes -Waggregate-return -Wmissing-prototypes -Wpointer-arith
 	OFILES += memwatch.o
-else
-	CFLAGS += -s
 endif
 
 # executable should not depend on cygwin.dll
@@ -26,13 +22,15 @@
 	-rm -f ${OFILES} ${GARBAGE}
 
 flasm:	${OFILES}
-	${CC} $(CFLAGS) -o flasm ${OFILES} ${LIBS}
+	${CC} $(CFLAGS) $(LDFLAGS) -o flasm ${OFILES} ${LIBS}
 
-assembler.tab.c assembler.tab.h: assembler.y
+assembler.tab.c: assembler.tab.h
+
+assembler.tab.h: assembler.y
 	bison --defines --debug assembler.y
 
 lex.yy.c: assembler.flex assembler.tab.h
 	flex -i assembler.flex
 
 keywords.c: keywords.gperf assembler.tab.h
-	gperf --language=ANSI-C -t -T -E -o -k 1,$$,2,5 -S8 keywords.gperf > keywords.c
\ No newline at end of file
+	gperf --language=ANSI-C -t -T -E -o -k 1,$$,2,5 -S8 keywords.gperf > keywords.c
