--- dos2unix-5.2/Makefile
+++ dos2unix-5.2/Makefile
@@ -77,7 +77,7 @@
 	MOFILES			= $(DOS2UNIX_MOFILES) $(UNIX2DOS_MOFILES)
 endif
 
-DOCFILES	= $(PACKAGE).txt $(PACKAGE).ps $(PACKAGE).pdf $(PACKAGE).htm
+DOCFILES	= $(PACKAGE).txt $(PACKAGE).ps $(PACKAGE).htm
 INSTALL_OBJS_DOC = README.txt NEWS.txt ChangeLog.txt COPYING.txt TODO.txt BUGS.txt $(DOCFILES)
 
 #PODFILES	= man/man1/dos2unix.pod $(wildcard man/*/man1/dos2unix.pod)
@@ -246,9 +246,6 @@
 %.ps : man/man1/%.1
 	groff -man $< -T ps > $@
 
-%.pdf: %.ps
-	ps2pdf $< $@
-
 %.htm : man/man1/%.pod
 	pod2html --title="$(PACKAGE) $(DOS2UNIX_VERSION) - DOS/MAC to UNIX and vice versa text file format converter" $< > $@
 
