diff -c -r dos2unix-6.0.4.orig/man/es/man1/dos2unix.pod dos2unix-6.0.4/man/es/man1/dos2unix.pod
*** dos2unix-6.0.4.orig/man/es/man1/dos2unix.pod	2013-07-27 20:08:12.000000000 +0200
--- dos2unix-6.0.4/man/es/man1/dos2unix.pod	2014-01-02 14:19:52.944130621 +0100
***************
*** 50,55 ****
--- 50,57 ----
  
  =pod
  
+ =encoding latin1
+ 
  =head1 NOMBRE
  
  dos2unix - Convertidor de archivos de texto de formato DOS/Mac a Unix y viceversa
diff -c -r dos2unix-6.0.4.orig/man/nl/man1/dos2unix.pod dos2unix-6.0.4/man/nl/man1/dos2unix.pod
*** dos2unix-6.0.4.orig/man/nl/man1/dos2unix.pod	2013-07-27 20:08:12.000000000 +0200
--- dos2unix-6.0.4/man/nl/man1/dos2unix.pod	2014-01-02 14:19:36.510059975 +0100
***************
*** 46,51 ****
--- 46,53 ----
  */
  =pod
  
+ =encoding latin1
+ 
  =head1 NAAM
  
  dos2unix - DOS/Mac naar Unix en vice versa tekstbestand formaat omzetter
