 Makefile.in |    2 +-
 configure   |    8 ++++----
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 11908cd..cf34df0 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -67,7 +67,7 @@ MAN3EXT		=	@MAN3EXT@
 # Install commands...
 #
 
-INSTALL_BIN	=	$(LIBTOOL) $(INSTALL) -m 755 -s
+INSTALL_BIN	=	$(LIBTOOL) $(INSTALL) -m 755
 INSTALL_DATA	=	$(INSTALL) -m 644
 INSTALL_DIR	=	$(INSTALL) -d
 INSTALL_LIB	=	$(LIBTOOL) $(INSTALL) -m 755
diff --git a/configure b/configure
index da3cca7..77a6e90 100755
--- a/configure
+++ b/configure
@@ -1280,7 +1280,7 @@ CFLAGS="${CFLAGS:=}"
 CXXFLAGS="${CXXFLAGS:=}"
 LDFLAGS="${LDFLAGS:=}"
 
-OPTIM="-O"
+OPTIM=""
 
 
 
@@ -1296,9 +1296,9 @@ fi;
 if test "${enable_debug+set}" = set; then
   enableval="$enable_debug"
   if eval "test x$enable_debug = xyes"; then
- 	OPTIM="-g"
+ 	OPTIM=""
 else
-	LDFLAGS="$LDFLAGS -s"
+	LDFLAGS="$LDFLAGS"
 fi
 fi;
 
@@ -3350,7 +3350,7 @@ echo "${ECHO_T}yes" >&6
 echo "${ECHO_T}yes" >&6
 			LIBMXML="libmxml.so.1.0"
 			DSO="\$(CC)"
-			DSOFLAGS="$DSOFLAGS -Wl,-soname,libmxml.so.1,-rpath,\$(libdir) -shared \$(OPTIM)"
+			DSOFLAGS="$LDFLAGS $DSOFLAGS -Wl,-soname,libmxml.so.1,-rpath,\$(libdir) -shared \$(OPTIM)"
                         LDFLAGS="$LDFLAGS -Wl,-rpath,\$(libdir)"
 			;;
 
