--- a/src/lftp_ssl.cc
+++ b/src/lftp_ssl.cc
@@ -560,7 +560,11 @@
    {
       if(res==GNUTLS_E_AGAIN || res==GNUTLS_E_INTERRUPTED)
 	 return RETRY;
-      else if(res==GNUTLS_E_UNEXPECTED_PACKET_LENGTH || res==GNUTLS_E_PREMATURE_TERMINATION)
+      else if(res==GNUTLS_E_UNEXPECTED_PACKET_LENGTH
+#if LFTP_LIBGNUTLS_VERSION_CODE >= 0x030000
+                 || res==GNUTLS_E_PREMATURE_TERMINATION
+#endif /* LFTP_LIBGNUTLS_VERSION_CODE */
+             )
       {
 	 Log::global->Format(7,"gnutls_record_recv: %s Assuming EOF.\n",gnutls_strerror(res));
 	 return 0;
