From db25961b46c591aecaa18e824e57a2af3d379d08 Mon Sep 17 00:00:00 2001
From: "Alexander V. Lukyanov" <lavv17f@gmail.com>
Date: Wed, 15 Oct 2014 08:46:09 +0400
Subject: [PATCH] ftpclass.cc: fixed compilation without SSL (again).

---
 src/ftpclass.cc | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/ftpclass.cc b/src/ftpclass.cc
index 130704d..bc148bc 100644
--- a/src/ftpclass.cc
+++ b/src/ftpclass.cc
@@ -1610,12 +1610,12 @@ int   Ftp::Do()
 
       if(expect->Has(Expect::FEAT)
       || expect->Has(Expect::OPTS_UTF8)
-      || expect->Has(Expect::LANG)
-      || expect->Has(Expect::PROT))
+      || expect->Has(Expect::LANG))
 	 goto usual_return;
 
 #if USE_SSL
-      if(expect->Has(Expect::CCC))
+      if(expect->Has(Expect::CCC)
+      || expect->Has(Expect::PROT))
 	 goto usual_return;
 #endif // USE_SSL
 
