From 21823aa2b496509a7bc9bed2141d06df35d25b63 Mon Sep 17 00:00:00 2001
From: "Alexander V. Lukyanov" <lavv17f@gmail.com>
Date: Wed, 15 Oct 2014 08:50:19 +0400
Subject: [PATCH] lftp_ssl.cc: fixed compilation with gnutls version < 3.0

---
 src/lftp_ssl.cc | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/lftp_ssl.cc b/src/lftp_ssl.cc
index 100b90b..8ef8a8d 100644
--- a/src/lftp_ssl.cc
+++ b/src/lftp_ssl.cc
@@ -547,6 +547,11 @@ int lftp_ssl_gnutls::do_handshake()
 
    return DONE;
 }
+
+#ifndef GNUTLS_E_PREMATURE_TERMINATION // for gnutls < 3.0
+# define GNUTLS_E_PREMATURE_TERMINATION GNUTLS_E_UNEXPECTED_PACKET_LENGTH
+#endif
+
 int lftp_ssl_gnutls::read(char *buf,int size)
 {
    if(error)
