diff -urN qx11grab-0.4.3.orig/modules/FindFFmpeg.cmake qx11grab-0.4.3/modules/FindFFmpeg.cmake
--- qx11grab-0.4.3.orig/modules/FindFFmpeg.cmake	2012-04-22 17:37:13.146507207 +0200
+++ qx11grab-0.4.3/modules/FindFFmpeg.cmake	2012-04-22 17:37:47.640075978 +0200
@@ -156,9 +156,9 @@
 
 MACRO (FFMPEG_FILTER_EXISTS FILTER VARIABLE)
   TRY_RUN (filter_arg _filter_supported
-    ${CMAKE_BINARY_DIR}
-    ${CMAKE_MODULE_PATH}/check_ffmpeg_filter.c
-    COMPILE_DEFINITIONS ${CMAKE_REQUIRED_DEFINITIONS}
+    ${CMAKE_BINARY_DIR}/CMakeTmp
+    ${CMAKE_SOURCE_DIR}/modules/check_ffmpeg_filter.c
+    COMPILE_DEFINITIONS "${CMAKE_REQUIRED_DEFINITIONS} -std=gnu89"
     CMAKE_FLAGS
       "-DINCLUDE_DIRECTORIES:STRING=${FFMPEG_INCLUDEDIR}"
       "-DLINK_LIBRARIES:STRING=${FFMPEG_AVUTIL_LIBRARY};${FFMPEG_AVFILTER_LIBRARY}"
diff -urN qx11grab-0.4.3.orig/modules/check_ffmpeg_filter.c qx11grab-0.4.3/modules/check_ffmpeg_filter.c
--- qx11grab-0.4.3.orig/modules/check_ffmpeg_filter.c	2012-04-22 17:37:13.147507195 +0200
+++ qx11grab-0.4.3/modules/check_ffmpeg_filter.c	2012-04-22 17:37:18.619438786 +0200
@@ -27,7 +27,7 @@
 
 int main(int argc, char **argv)
 {
-  if( ( argc < 1 ) || strlen(argv[1]) < 3 )
+  if( ( argc < 2 ) || strlen(argv[1]) < 3 )
   {
     fprintf ( stderr, "Missing Filter arguments\n" );
     return EXIT_FAILURE;
