diff -Naur cddb.bundle.orig/Cddb/GNUmakefile cddb.bundle/Cddb/GNUmakefile
--- cddb.bundle.orig/Cddb/GNUmakefile	2008-01-25 00:40:03.000000000 +0100
+++ cddb.bundle/Cddb/GNUmakefile	2008-01-25 00:40:09.000000000 +0100
@@ -3,8 +3,6 @@
 BUNDLE_NAME = Cddb
 BUNDLE_EXTENSION = .bundle
 
-BUNDLE_INSTALL_DIR = $(GNUSTEP_INSTALLATION_DIR)/Library/Bundles
-
 Cddb_RESOURCE_FILES =
 
 Cddb_OBJC_FILES = Cddb.m
