diff -r -U2 udav-0.6.orig/src/src.pro udav-0.6/src/src.pro
--- udav-0.6.orig/src/src.pro	2009-12-24 18:38:09.000000000 +0100
+++ udav-0.6/src/src.pro	2010-01-04 18:42:09.000000000 +0100
@@ -1,3 +1,4 @@
-DEFINES += MGL_DOC_DIR=\\\"$(MGLDOCDIR)\\\"
+DEFINES += MGL_DOC_DIR=\\\"/usr/share/doc/$(PF)/\\\"
+DEFINES += H5_USE_16_API
 TEMPLATE = app
 CONFIG += release \
@@ -42,5 +43,5 @@
     ../INSTALL
 RC_FILE = udav.rc
-target.path = /usr/local/bin/
+target.path = /usr/bin/
 INSTALLS += target \
     help \
@@ -56,8 +57,8 @@
 help.files += ../help/*.html
 pics.files += ../help/pics/*.png
-help.path = /usr/local/share/doc/udav/
-pics.path = /usr/local/share/doc/udav/pics/
-icon.path = /usr/local/share/udav/
-desc.path = /usr/local/share/applications/
+help.path = /usr/share/doc/${PF}/
+pics.path = /usr/share/doc/${PF}/pics/
+icon.path = /usr/share/icons/hicolor/64x64/apps/
+desc.path = /usr/share/applications/
 OTHER_FILES += ../ChangeLog.txt \
     ../TODO
diff -r -U2 udav-0.6.orig/UDAV.desktop udav-0.6/UDAV.desktop
--- udav-0.6.orig/UDAV.desktop	2009-12-24 18:38:13.000000000 +0100
+++ udav-0.6/UDAV.desktop	2010-01-04 18:35:30.000000000 +0100
@@ -4,5 +4,5 @@
 Type=Application
 Terminal=false
-Icon=/usr/local/share/udav/udav.png
+Icon=/usr/share/icons/hicolor/64x64/apps/udav.png
 Name=UDAV
 Exec=udav
