Index: camlimages-3.0.2/src/Makefile.am
===================================================================
--- camlimages-3.0.2.orig/src/Makefile.am
+++ camlimages-3.0.2/src/Makefile.am
@@ -185,7 +185,10 @@ CSOURCES += ftintf.c
 AM_CFLAGS += $(INCFREETYPE)
 endif
 
-ocamlsitelib_DATA += units.cmi oXpm.cmi
+ocamlsitelib_DATA += units.cmi
+if HAVE_XPM
+ocamlsitelib_DATA += oXpm.cmi
+endif
 if HAVE_TIFF
 ocamlsitelib_DATA += oTiff.cmi
 endif
