diff -Naur pyme-0.8.1/pyme/core.py pyme-0.8.1.new/pyme/core.py
--- pyme-0.8.1/pyme/core.py	2008-04-02 08:12:57.000000000 +0200
+++ pyme-0.8.1.new/pyme/core.py	2012-05-18 11:09:48.737522830 +0200
@@ -58,6 +58,7 @@
             self.own = False
         else:
             tmp = pygpgme.new_gpgme_ctx_t_p()
+            check_version(None)
             errorcheck(pygpgme.gpgme_new(tmp))
             self.wrapped = pygpgme.gpgme_ctx_t_p_value(tmp)
             pygpgme.delete_gpgme_ctx_t_p(tmp)
