--- yatex1.73-orig/yatexhlp.el	2006-12-24 07:25:29.000000000 +0100
+++ yatex1.73/yatexhlp.el	2008-05-12 12:19:02.000000000 +0200
@@ -9,12 +9,7 @@
 			 (cond (YaTeX-japan "jp")
 			       (t "eng"))))
       (help-dir
-       (cond
-	((and (boundp 'site-directory) site-directory) site-directory)
-	((string-match "\\.app/" doc-directory)	;For Emacs.app(Darwin)
-	 (expand-file-name "../site-lisp" doc-directory))
-	(YaTeX-emacs-19 (expand-file-name "../../site-lisp" doc-directory))
-	(t exec-directory))))
+       "/usr/share/emacs/etc/yatex"))
   (defvar YaTeX-help-file
     (expand-file-name help-file help-dir)
     "*Help file of LaTeX/TeX commands or macros.")
--- yatex1.73-orig/docs/yahtmle	2006-12-24 07:28:10.000000000 +0100
+++ yatex1.73/docs/yahtmle	2008-05-12 12:19:40.000000000 +0200
@@ -3,8 +3,9 @@
 from file `yahtmle.tex'
 using `texinfmt.el' version 2.32 of 19 November 1993.
 
-
-
+START-INFO-DIR-ENTRY
+* yahtml: (yahtml). Yet Another HTML-mode for Emacs. (English).
+END-INFO-DIR-ENTRY
 
 
 File: yahtmle, Node: Top, Next: Intro, Prev: (dir), Up: (dir)
--- yatex1.73-orig/docs/yahtmlj	2006-12-24 07:28:10.000000000 +0100
+++ yatex1.73/docs/yahtmlj	2008-05-12 12:21:03.000000000 +0200
@@ -3,8 +3,9 @@
 from file `yahtmlj.tex'
 using `texinfmt.el' version 2.32 of 19 November 1993.
 
-
-
+START-INFO-DIR-ENTRY
+* yahtml-ja: (yahtml-ja). Yet Another HTML-mode for Emacs. (Japanese).
+END-INFO-DIR-ENTRY
 
 
 File: yahtmlj, Node: Top, Next: Intro, Prev: (dir), Up: (dir)
--- yatex1.73-orig/docs/yatexe	2006-12-24 07:28:10.000000000 +0100
+++ yatex1.73/docs/yatexe	2008-05-12 12:20:30.000000000 +0200
@@ -3,8 +3,9 @@
 from file `yatexe.tex'
 using `texinfmt.el' version 2.32 of 19 November 1993.
 
-
-
+START-INFO-DIR-ENTRY
+* YaTeX: (yatex). Yet Another tex-mode for Emacs. (English).
+END-INFO-DIR-ENTRY
 
 
 File: yatexe, Node: Top, Next: What is YaTeX?, Prev: (dir), Up: (dir)
--- yatex1.73-orig/docs/yatexj	2006-12-24 07:28:10.000000000 +0100
+++ yatex1.73/docs/yatexj	2008-05-12 12:20:56.000000000 +0200
@@ -3,8 +3,9 @@
 from file `yatexj.tex'
 using `texinfmt.el' version 2.32 of 19 November 1993.
 
-
-
+START-INFO-DIR-ENTRY
+* YaTeX-ja: (yatex-ja). Yet Another tex-mode for Emacs. (Japanese).
+END-INFO-DIR-ENTRY
 
 
 File: yatexj, Node: Top, Next: Intro, Prev: (dir), Up: (dir)
