--- work.orig/makefile
+++ work/makefile
@@ -431,9 +431,6 @@
 CPPONLYFLAGS += -x c++ -std=gnu++98
 COBJFLAGS += -x objective-c++
 
-# this speeds it up a bit by piping between the preprocessor/compiler/assembler
-CCOMFLAGS += -pipe
-
 # add -g if we need symbols, and ensure we have frame pointers
 ifdef SYMBOLS
 CCOMFLAGS += -g$(SYMLEVEL) -fno-omit-frame-pointer
@@ -449,23 +446,6 @@
 CCOMFLAGS += -pg
 endif
 
-# add the optimization flag
-CCOMFLAGS += -O$(OPTIMIZE)
-
-# if we are optimizing, include optimization options
-# and make all errors into warnings
-ifneq ($(OPTIMIZE),0)
-ifneq ($(TARGETOS),os2)
-ifndef NOWERROR
-CCOMFLAGS += -Werror -fno-strict-aliasing $(ARCHOPTS)
-else
-CCOMFLAGS += -fno-strict-aliasing $(ARCHOPTS)
-endif
-else
-CCOMFLAGS += -fno-strict-aliasing $(ARCHOPTS)
-endif
-endif
-
 # add a basic set of warnings
 CCOMFLAGS += \
 	-Wall \
@@ -524,16 +504,6 @@
 # linking flags
 #-------------------------------------------------
 
-# LDFLAGS are used generally; LDFLAGSEMULATOR are additional
-# flags only used when linking the core emulator
-LDFLAGS =
-ifneq ($(TARGETOS),macosx)
-ifneq ($(TARGETOS),os2)
-ifneq ($(TARGETOS),solaris)
-LDFLAGS = -Wl,--warn-common
-endif
-endif
-endif
 LDFLAGSEMULATOR =
 
 # add profiling information for the linker
@@ -541,13 +511,6 @@
 LDFLAGS += -pg
 endif
 
-# strip symbols and other metadata in non-symbols and non profiling builds
-ifndef SYMBOLS
-ifneq ($(TARGETOS),macosx)
-LDFLAGS += -s
-endif
-endif
-
 # output a map file (emulator only)
 ifdef MAP
 LDFLAGSEMULATOR += -Wl,-Map,$(FULLNAME).map
