Index: kanyremote-5.10/kanyremote
===================================================================
--- kanyremote-5.10.orig/kanyremote
+++ kanyremote-5.10/kanyremote
@@ -3777,8 +3777,10 @@ def main():
         usepybluez = False
     
     if pybluez == False and usepybluez == True:
-        print _tr('Install PyBluez first !\nOr run with --npybluez option')
-	return
+#        print _tr('Install PyBluez first !\nOr run with --npybluez option')
+	print _tr('Bluetooth support is disabled')
+	usepybluez = False
+#	return
 	
     if debug: print 'Use port ' + port
    
