diff -Nuar LVM2.2.02.84.orig/configure.in LVM2.2.02.84/configure.in
--- LVM2.2.02.84.orig/configure.in	2011-02-04 22:17:54.000000000 +0000
+++ LVM2.2.02.84/configure.in	2011-04-09 23:42:27.134617541 +0000
@@ -803,9 +803,18 @@
 AC_MSG_RESULT($UDEV_SYNC)
 
 if test x$UDEV_SYNC = xyes; then
-	AC_CHECK_LIB(udev, udev_queue_get_udev_is_active,
-		     [UDEV_PC="libudev"; UDEV_LIBS="-ludev"],
-		     [AC_MSG_ERROR([bailing out... libudev library is required])])
+	if  test x$PKGCONFIG_INIT != x1; then
+		pkg_config_init
+	fi
+	PKG_CHECK_MODULES(UDEV, libudev, [UDEV_PC="libudev" HAVE_UDEV=yes],
+		[NOTFOUND=0
+		AC_CHECK_HEADERS(libudev.h,,$bailout)
+		check_lib_no_libs udev udev_queue_get_udev_is_active
+		if test $NOTFOUND = 0; then
+			AC_CHECK_LIB(udev, udev_queue_get_udev_is_active,
+				     [UDEV_PC="libudev"; UDEV_LIBS="-ludev"],
+				     [AC_MSG_ERROR([bailing out... libudev library is required])])
+		fi])
 	AC_DEFINE([UDEV_SYNC_SUPPORT], 1, [Define to 1 to enable synchronisation with udev processing.])
 fi
 
