commit 44ec153bf6212044fd8d2e3f59cb509cc30f8318
Author: Gregory <seacreature@anatta.local>
Date:   Tue Jun 8 18:21:16 2010 -0400

    Fix some minor 1.9.2 issues

diff --git a/spec/document_spec.rb b/spec/document_spec.rb
index 7707e56..71d5751 100644
--- a/spec/document_spec.rb
+++ b/spec/document_spec.rb
@@ -374,9 +374,14 @@ end
 
 describe "The :optimize_objects option" do
   before(:all) do
-    @wasteful_doc = lambda do
-      transaction { start_new_page; text "Hidden text"; rollback }
-      text "Hello world"
+    @wasteful_doc = lambda do |pdf|
+      pdf.transaction do 
+        pdf.start_new_page
+        pdf.text "Hidden text"
+        pdf.rollback 
+      end
+
+      pdf.text "Hello world"
     end
   end
 
