commit 6adb8ceb844f57f50dcfa6e457a2cd37b933d40e
Author: Brad Ediger <brad.ediger@madriska.com>
Date:   Thu Jan 20 16:46:37 2011 -0600

    Green font_spec test for 1.9.2p136

diff --git a/spec/font_spec.rb b/spec/font_spec.rb
index 9304b01..2b3368a 100644
--- a/spec/font_spec.rb
+++ b/spec/font_spec.rb
@@ -224,7 +224,8 @@ describe "TTF fonts" do
   
   it "should encode text without kerning by default" do
     @activa.encode_text("To").should == [[0, "To"]]
-    @activa.encode_text("Télé").should == [[0, "T\216l\216"]]
+    @activa.encode_text("Télé").should == 
+      [[0, "T\216l\216".force_encoding("US-ASCII")]]
     @activa.encode_text("Technology").should == [[0, "Technology"]]
     @activa.encode_text("Technology...").should == [[0, "Technology..."]]
     @activa.encode_text("Teχnology...").should == [[0, "Te"], [1, "!"], [0, "nology..."]]
