diff -ur gbuffy-0.2.6/imap.c gbuffy-tinker/imap.c
--- gbuffy-0.2.6/imap.c	2003-10-10 11:06:53.000000000 +0200
+++ gbuffy-tinker/imap.c	2005-03-16 12:56:10.000000000 +0100
@@ -437,7 +437,7 @@
 
   /* Open mailbox (EXAMINE) and find the first UNSEEN message */
   imap_make_sequence (seq, sizeof (seq));
-  snprintf (buf, sizeof (buf), "%s EXAMINE %s\r\n", seq, ibox->path);
+  snprintf (buf, sizeof (buf), "%s EXAMINE \"%s\"\r\n", seq, ibox->path);
   msocket_write (conn, buf);
 
   do 
@@ -529,7 +529,7 @@
    * think we want that, so use UNSEEN instead */
   /* Go back to using RECENT, as it shouldn't be updated on EXAMINE */
   imap_make_sequence (seq, sizeof (seq));
-  snprintf (buf, sizeof (buf), "%s STATUS %s (MESSAGES RECENT)\r\n", seq, 
+  snprintf (buf, sizeof (buf), "%s STATUS \"%s\" (MESSAGES RECENT)\r\n", seq, 
       ibox->path);
 
   msocket_write (conn, buf);


