--- tools/rwhois_repack/Makefile.in.orig	2009-10-16 11:30:40.000000000 -0700
+++ tools/rwhois_repack/Makefile.in		2009-10-16 11:31:20.000000000 -0700
@@ -52,9 +52,9 @@
 # procedural
 
 install:
-	if [ ! -d $(exec_prefix) ]; then mkdir $(exec_prefix); fi
-	if [ ! -d $(bindir) ]; then mkdir $(bindir); fi
-	$(INSTALL) rwhois_repack $(bindir)
+	if [ ! -d $(DESTDIR)$(exec_prefix) ]; then mkdir $(DESTDIR)$(exec_prefix); fi
+	if [ ! -d $(DESTDIR)$(bindir) ]; then mkdir $(DESTDIR)$(bindir); fi
+	$(INSTALL) rwhois_repack $(DESTDIR)$(bindir)
 
 uninstall:
	$(RM) $(bindir)/rwhois_repack

--- sample.data/Makefile.in.orig	2009-10-16 11:37:54.000000000 -0700
+++ sample.data/Makefile.in		2009-10-16 11:38:22.000000000 -0700
@@ -10,7 +10,7 @@
 prefix      = @prefix@
 exec_prefix = @exec_prefix@
 bindir      = $(exec_prefix)/bin
-etcdir      = $(exec_prefix)/etc
+etcdir      = /etc
 
 srcdir	    = @srcdir@
 VPATH       = @srcdir@
