diff -Naur ATLAS/makes/Make.Flp ATLAS.new/makes/Make.Flp
--- ATLAS/makes/Make.Flp	2008-07-17 21:24:46.000000000 -0400
+++ ATLAS.new/makes/Make.Flp	2008-07-22 10:54:45.000000000 -0400
@@ -6,7 +6,7 @@
 sobj = ATL_f77wrap_ilaenv.o ATL_f77wrap_sgesv.o ATL_f77wrap_sgetnb.o \
        ATL_f77wrap_sgetrf.o ATL_f77wrap_sgetri.o ATL_f77wrap_sgetrs.o \
        ATL_f77wrap_slauum.o ATL_f77wrap_sposv.o ATL_f77wrap_spotrf.o \
-       ATL_f77wrap_spotrs.o ATL_f77wrap_strtri.o ieeeck.o ilaenv.o sgesv.o \
+       ATL_f77wrap_spotrs.o ATL_f77wrap_strtri.o ieeeck.o sgesv.o \
        sgetrf.o sgetri.o sgetrs.o slauum.o sposv.o spotrf.o spotri.o spotrs.o \
        strtri.o
 
@@ -64,8 +64,8 @@
        ATL_f77wrap_dgetri.o ATL_f77wrap_dgetrs.o ATL_f77wrap_dlauum.o \
        ATL_f77wrap_dposv.o ATL_f77wrap_dpotrf.o ATL_f77wrap_dpotrs.o \
        ATL_f77wrap_dtrtri.o ATL_f77wrap_ilaenv.o dgesv.o dgetrf.o dgetri.o \
-       dgetrs.o dlauum.o dposv.o dpotrf.o dpotri.o dpotrs.o dtrtri.o ieeeck.o \
-       ilaenv.o
+       dgetrs.o dlauum.o dposv.o dpotrf.o dpotri.o dpotrs.o dtrtri.o ieeeck.o 
+      
 
 dlaobj = dbdsdc.o dbdsqr.o ddisna.o dgbbrd.o dgbcon.o dgbequ.o dgbsv.o \
          dgbsvx.o dgbtf2.o dgbtrf.o dgbtrs.o dgebak.o dgebal.o dgebd2.o \
@@ -121,8 +121,8 @@
        ATL_f77wrap_cgetri.o ATL_f77wrap_cgetrs.o ATL_f77wrap_clauum.o \
        ATL_f77wrap_cposv.o ATL_f77wrap_cpotrf.o ATL_f77wrap_cpotrs.o \
        ATL_f77wrap_ctrtri.o ATL_f77wrap_ilaenv.o cgesv.o cgetrf.o cgetri.o \
-       cgetrs.o clauum.o cposv.o cpotrf.o cpotri.o cpotrs.o ctrtri.o ieeeck.o \
-       ilaenv.o
+       cgetrs.o clauum.o cposv.o cpotrf.o cpotri.o cpotrs.o ctrtri.o ieeeck.o 
+     
 
 claobj = cbdsqr.o cgbbrd.o cgbcon.o cgbequ.o cgbrfs.o cgbsv.o cgbsvx.o \
          cgbtf2.o cgbtrf.o cgbtrs.o cgebak.o cgebal.o cgebd2.o cgebrd.o \
@@ -172,7 +172,7 @@
 zobj = ATL_f77wrap_ilaenv.o ATL_f77wrap_zgesv.o ATL_f77wrap_zgetnb.o \
        ATL_f77wrap_zgetrf.o ATL_f77wrap_zgetri.o ATL_f77wrap_zgetrs.o \
        ATL_f77wrap_zlauum.o ATL_f77wrap_zposv.o ATL_f77wrap_zpotrf.o \
-       ATL_f77wrap_zpotrs.o ATL_f77wrap_ztrtri.o ieeeck.o ilaenv.o zgesv.o \
+       ATL_f77wrap_zpotrs.o ATL_f77wrap_ztrtri.o ieeeck.o zgesv.o \
        zgetrf.o zgetri.o zgetrs.o zlauum.o zposv.o zpotrf.o zpotri.o zpotrs.o \
        ztrtri.o
 
@@ -226,7 +226,7 @@
 lib : dlib slib zlib clib
 clean : dclean sclean zclean cclean iclean
 iclean :
-	rm -f $(ilaobj) ilaenv.o ieeeck.o
+	rm -f $(ilaobj) ieeeck.o
 
 sclean :
 	rm -f $(sobj) slib.grd slalib.grd $(slaobj)
@@ -462,8 +462,6 @@
 	$(ICC) $(ICCFLAGS) -o $@ -c $(mySRCdir)/f77wrap/ATL_f77wrap_ilaenv.c
 ieeeck.o : $(mySRCdir)/ieeeck.f
 	$(F77) -o $@ -c $(F77FLAGS) $(mySRCdir)/ieeeck.f
-ilaenv.o : $(mySRCdir)/ilaenv.f
-	$(F77) -o $@ -c $(F77FLAGS) $(mySRCdir)/ilaenv.f
 
 #
 #  All the rest of these files are compiled directly from netlib LAPACK
