diff -Naur ATLAS/makes/Make.Flp ATLAS.new/makes/Make.Flp
--- ATLAS/makes/Make.Flp	2010-01-11 13:19:20.000000000 -0500
+++ ATLAS.new/makes/Make.Flp	2010-01-20 12:24:06.000000000 -0500
@@ -9,7 +9,7 @@
        ATL_f77wrap_slarfb.o ATL_f77wrap_slarft.o ATL_f77wrap_slauum.o \
        ATL_f77wrap_sposv.o ATL_f77wrap_spotrf.o ATL_f77wrap_spotrs.o \
        ATL_f77wrap_strtri.o ieeeck.o sgelqf.o sgeqlf.o sgeqrf.o sgerqf.o \
-       sgesv.o sgetrf.o sgetri.o sgetrs.o slamc3.o slamch.o slarfb.o slarft.o \
+       sgesv.o sgetrf.o sgetri.o sgetrs.o slarfb.o slarft.o \
        slauum.o sposv.o spotrf.o spotri.o spotrs.o strtri.o
 dobj = ATL_f77wrap_dgelqf.o ATL_f77wrap_dgeqlf.o ATL_f77wrap_dgeqrf.o \
        ATL_f77wrap_dgerqf.o ATL_f77wrap_dgesv.o ATL_f77wrap_dgetnb.o \
@@ -17,7 +17,7 @@
        ATL_f77wrap_dlarfb.o ATL_f77wrap_dlarft.o ATL_f77wrap_dlauum.o \
        ATL_f77wrap_dposv.o ATL_f77wrap_dpotrf.o ATL_f77wrap_dpotrs.o \
        ATL_f77wrap_dtrtri.o dgelqf.o dgeqlf.o dgeqrf.o dgerqf.o dgesv.o \
-       dgetrf.o dgetri.o dgetrs.o dlamc3.o dlamch.o dlarfb.o dlarft.o \
+       dgetrf.o dgetri.o dgetrs.o dlarfb.o dlarft.o \
        dlauum.o dposv.o dpotrf.o dpotri.o dpotrs.o dtrtri.o ieeeck.o
 cobj = ATL_f77wrap_cgelqf.o ATL_f77wrap_cgeqlf.o ATL_f77wrap_cgeqrf.o \
        ATL_f77wrap_cgerqf.o ATL_f77wrap_cgesv.o ATL_f77wrap_cgetnb.o \
@@ -331,16 +331,6 @@
 ATL_f77wrap_zgetnb.o : $(mySRCdir)/f77wrap/ATL_f77wrap_getnb.c
 	$(ICC) $(ICCFLAGS) -o $@ -c -DDCPLX $(mySRCdir)/f77wrap/ATL_f77wrap_getnb.c
 
-slamch.o : $(mySRCdir)/lamch.c
-	$(ICC) $(ICCFLAGS) -o $@ -c -DSREAL $(mySRCdir)/lamch.c
-slamc3.o : $(mySRCdir)/slamc3.f
-	$(F77) $(F77FLAGS) $(F77NOOPT) -o $@ -c $(mySRCdir)/slamc3.f
-dlamch.o : $(mySRCdir)/lamch.c
-	$(ICC) $(ICCFLAGS) -o $@ -c -DDREAL $(mySRCdir)/lamch.c
-dlamc3.o : $(mySRCdir)/dlamc3.f
-	$(F77) $(F77FLAGS) $(F77NOOPT) -o $@ -c $(mySRCdir)/dlamc3.f
-
-
 ieeeck.o : $(mySRCdir)/ieeeck.f
 	$(F77) -o $@ -c $(F77FLAGS) $(mySRCdir)/ieeeck.f
 
