commit fdedc830ca5061f45cd2412c11566ca84806c427
Author: Andras Mantia <amantia@kde.org>
Date:   Fri Jul 12 15:03:36 2013 +0300

    Unbreak akonadi on Postgres

diff --git a/server/src/storage/datastore.cpp b/server/src/storage/datastore.cpp
index f5a1bc6..93e1ff2 100644
--- a/server/src/storage/datastore.cpp
+++ b/server/src/storage/datastore.cpp
@@ -208,8 +208,9 @@ bool DataStore::setItemsFlags( const PimItem::List &items, const QVector<Flag> &
   }
 
   QueryBuilder qb2( PimItemFlagRelation::tableName(), QueryBuilder::Insert );
-  qb2.setColumnValue( PimItemFlagRelation::leftFullColumnName(), insIds );
-  qb2.setColumnValue( PimItemFlagRelation::rightFullColumnName(), insFlags );
+  qb2.setColumnValue( PimItemFlagRelation::leftColumn(), insIds );
+  qb2.setColumnValue( PimItemFlagRelation::rightColumn(), insFlags );
+  qb2.setIdentificationColumn( QString() );
   if ( !qb2.exec() ) {
     return false;
   }
