commit 607b9338a40cdde705f91c5cd95d715e5514f4cb
Author: Jan Grulich <jgrulich@redhat.com>
Date:   Fri Oct 11 16:56:49 2013 +0200

    Fix build without libmm-qt

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f6b0fcb..b660b38 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -218,7 +218,12 @@ qt4_automoc(${DBUS_INTERFACE_SRCS})
 
 add_library(NetworkManagerQt SHARED ${NetworkManagerQt_PART_SRCS} ${DBUS_INTERFACE_SRCS} ${CONNECTIONS_SETTINGS_SRCS} ${NetworkManagerQt_MOC} )
 
+if (MODEMMANAGERQT_FOUND)
 target_link_libraries(NetworkManagerQt ModemManagerQt ${QT_QTCORE_LIBRARY} ${QT_QTDBUS_LIBRARY} ${QT_QTNETWORK_LIBRARY})
+else()
+target_link_libraries(NetworkManagerQt ${QT_QTCORE_LIBRARY} ${QT_QTDBUS_LIBRARY} ${QT_QTNETWORK_LIBRARY})
+endif()
+
 install(TARGETS NetworkManagerQt DESTINATION lib${LIB_SUFFIX})
 
 install( FILES
