commit c1fa14e924066c540b28765f9ee58a44a876562e
Author: Lamarque V. Souza <lamarque@kde.org>
Date:   Sat May 24 14:23:15 2014 -0300

    Respect external CMAKE_CXX_FLAGS values.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 73e92a4..8e36337 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -33,7 +33,7 @@ add_definitions(-DWITH_MODEMMANAGERQT=${WITH_MODEMMANAGERQT} ${QT_DEFINITIONS})
 include(CheckCXXCompilerFlag)
 check_cxx_compiler_flag("-Werror=return-type" HAVE_GCC_ERROR_RETURN_TYPE)
 if (HAVE_GCC_ERROR_RETURN_TYPE)
-    set(CMAKE_CXX_FLAGS "-Werror=return-type")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror=return-type")
 endif()
 
 include_directories(
