--- vpnclient-orig/linuxcniapi.h	2008-06-23 11:59:12.000000000 -0500
+++ vpnclient/linuxcniapi.h	2009-11-12 15:34:34.555673500 -0600
@@ -30,7 +30,11 @@
 #else
                           struct packet_type *);
 #endif
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,29)
+    const struct net_device_ops *Inject_ops;
+#else
     int (*InjectSend) (struct sk_buff * skb, struct net_device * dev);
+#endif
 
     int recv_real_hh_len;
     int send_real_hh_len;
