--- vpnclient-orig/GenDefs.h	2008-06-23 11:59:12.000000000 -0500
+++ vpnclient/GenDefs.h	2009-11-12 15:16:46.235673252 -0600
@@ -105,6 +105,15 @@
 #define _INTPTR_T_DEFINED
 #endif
 
+/* uintptr_t has been defined in include/linux/types.h in 2.6.24.
+ * No need to define it here again (will only lead to compile errors)
+ *
+ * by Alexander Griesser <work@tuxx-home.at>, 2008-01-11
+ */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,24)
+#define _UINTPTR_T_DEFINED
+#endif
+
 #ifndef _UINTPTR_T_DEFINED
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,24)
 #if defined(_LP64)
