--- vpnclient-orig/linuxkernelapi.c	2008-06-23 11:59:12.000000000 -0500
+++ vpnclient/linuxkernelapi.c	2009-11-12 15:35:04.947674515 -0600
@@ -9,7 +9,10 @@
     void*rc = kmalloc(size, GFP_ATOMIC);
     if(NULL == rc)
     {
-        printk("<1> os_malloc size %d failed\n",size);
+/* Allow for 32- or 64-bit size
+ *        printk("<1> os_malloc size %d failed\n",size);
+ */
+        printk("<1> os_malloc size %ld failed\n", (long)size);
     }
 
     return rc;
