--- client/distros/gentoo/main.py	2011-04-10 20:15:47.957683728 +0200
+++ client/distros/gentoo/main.py	2011-04-10 20:16:52.504183850 +0200
@@ -61,7 +61,7 @@
         """
         Returns True if we run on top of Gentoo, else False.
         """
-        return os.path.exists('/etc/gentoo-release')
+        return False  # sping: Too early to enable
 
     def gather(self, debug=False):
         def _stage(text):
