=== modified file 'helpers/external_acl/wbinfo_group/ext_wbinfo_group_acl.pl.in'
--- helpers/external_acl/wbinfo_group/ext_wbinfo_group_acl.pl.in	2015-01-13 07:25:36 +0000
+++ helpers/external_acl/wbinfo_group/ext_wbinfo_group_acl.pl.in	2015-02-19 03:00:51 +0000
@@ -121,6 +121,11 @@
 #
 use vars qw/ %opt /;
 
+my $user;
+my $group;
+my @groups;
+my $ans;
+
 # Disable output buffering
 $|=1;
 
@@ -132,7 +137,11 @@
 # Check if a user belongs to a group
 #
 sub check {
-        local($user, $group) = @_;
+	my $groupSID;
+	my $groupGID;
+	my @tmpuser;
+
+	our($user, $group) = @_;
 	if ($opt{K} && ($user =~ m/\@/)) {
 		@tmpuser = split(/\@/, $user);
 		$user = "$tmpuser[1]\\$tmpuser[0]";

