diff -Naur libmicrohttpd-0.9.26.orig/doc/examples/Makefile.am libmicrohttpd-0.9.26/doc/examples/Makefile.am
--- libmicrohttpd-0.9.26.orig/doc/examples/Makefile.am	2013-03-29 12:21:07.000000000 -0400
+++ libmicrohttpd-0.9.26/doc/examples/Makefile.am	2013-03-30 17:04:34.000000000 -0400
@@ -18,15 +18,17 @@
   basicauthentication \
   hellobrowser \
   logging \
-  responseheaders \
-  sessions 
+  responseheaders
 
 if ENABLE_HTTPS
 noinst_PROGRAMS += \
   tlsauthentication
 endif
 if HAVE_POSTPROCESSOR
-noinst_PROGRAMS += simplepost largepost
+noinst_PROGRAMS += \
+  simplepost \
+  largepost \
+  sessions 
 endif
 
 if HAVE_W32
