diff -Naur libmicrohttpd-0.9.28.orig/src/microhttpd/daemon.c libmicrohttpd-0.9.28/src/microhttpd/daemon.c
--- libmicrohttpd-0.9.28.orig/src/microhttpd/daemon.c	2013-07-23 21:10:14.000000000 -0400
+++ libmicrohttpd-0.9.28/src/microhttpd/daemon.c	2013-07-23 21:11:32.000000000 -0400
@@ -3112,9 +3112,11 @@
 	  int sk_flags = fcntl (socket_fd, F_GETFL);
 	  if (0 != fcntl (socket_fd, F_SETFL, sk_flags | O_NONBLOCK))
 	    {
+#if HAVE_MESSAGES
 	      MHD_DLOG (daemon,
 			"Failed to make listen socket non-blocking: %s\n", 
 			STRERROR (errno));
+#endif
 	      if (0 != CLOSE (socket_fd))
 		MHD_PANIC ("close failed\n");	      
 	      goto free_and_fail;	      
