--- sdk/angelscript/projects/gnuc/makefile
+++ sdk/angelscript/projects/gnuc/makefile
@@ -3,6 +3,7 @@
 
 # For 'make install' to work, set LOCAL according to your system configuration
 LOCAL = /usr/local
+INSTALL_LIBDIR = lib
 
 # If you want to build a shared library, then run make with SHARED=1 and VERSION=version
 ifdef SHARED
@@ -44,10 +45,10 @@
 else
 	OBJDIR = obj
 	LIBDIR = ../../lib
-	CXX ?= gcc
+	CXX ?= g++
 	# On i686 architecture you may need to add -march=i686 if you get 
 	# an undefined symbol for __sync_sub_and_fetch_4 in as_atomic.cpp.
-	CXXFLAGS += -g -Wall -fPIC -fno-strict-aliasing
+	CXXFLAGS += -Wall -fPIC -fno-strict-aliasing
 endif
 
 AR ?= ar
@@ -106,7 +107,7 @@
 
 $(BIN): $(OBJDIR) $(LIBDIR) $(OBJ)
 ifdef SHARED
-	$(CXX) -shared -Wl,-soname,$(LIB) -o $(BIN) $(OBJ)
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -shared -Wl,-soname,$(LIB) -o $(BIN) $(OBJ)
 	( cd $(LIBDIR); ln -s $(LIB) $(DEVLIB) )
 else
 	$(AR) r $(BIN) $(OBJ)
@@ -135,18 +136,18 @@
 	$(DELETER) $(OBJ) $(BIN) $(LIBDIR)/$(DEVLIB)
 
 install: $(BIN)
-	@echo Installing to: $(LOCAL)/lib and $(LOCAL)/include...
+	@echo Installing to: $(LOCAL)/$(INSTALL_LIBDIR) and $(LOCAL)/include...
 	@echo -------------------------------------------------------------------
 ifdef SHARED
-	$(COPIER) $(LIBDIR)/$(DEVLIB) $(LOCAL)/lib
+	$(COPIER) $(LIBDIR)/$(DEVLIB) $(LOCAL)/$(INSTALL_LIBDIR)
 endif
-	$(COPIER) $(BIN) $(LOCAL)/lib
+	$(COPIER) $(BIN) $(LOCAL)/$(INSTALL_LIBDIR)
 	$(COPIER) $(INCDIR)/$(INC) $(LOCAL)/include
 	@echo -------------------------------------------------------------------
 	@echo Angelscript library installed. Enjoy!
 
 uninstall:
-	$(DELETER) $(LOCAL)/include/$(INC) $(LOCAL)/lib/$(LIB) $(LOCAL)/lib/$(DEVLIB)
+	$(DELETER) $(LOCAL)/include/$(INC) $(LOCAL)/$(INSTALL_LIBDIR)/$(LIB) $(LOCAL)/$(INSTALL_LIBDIR)/$(DEVLIB)
 	@echo -------------------------------------------------------------------
 	@echo Angelscript library uninstalled.
 
