--- tong.cpp.ori
+++ tong.cpp
@@ -54,6 +54,8 @@
 #define MENUMOUSE_THRESHOLD 40
 #define DEMO_TIME           8000
 
+#define MIN_TIME            45
+
 bool drop_piece(Tetrad*, Tetrad*, Bucket*, Uint32&, bool&, bool&, Option*,
 		int&);
 
@@ -1979,6 +1981,13 @@
     SDL_Flip(screen);
     lastlastupdate=lastupdate;
     lastupdate=SDL_GetTicks();
+    
+    // limit to 1000/MIN_TIME fps
+    if (lastupdate-lastlastupdate < MIN_TIME) {
+      SDL_Delay(MIN_TIME - (lastupdate - lastlastupdate));
+    }
+    lastupdate = SDL_GetTicks();
+    
   }  //main game loop
 
   if(Mix_PlayingMusic()) {
