--- Makefile.orig	2005-02-01 11:49:23.000000000 -0500
+++ Makefile	2005-02-01 11:51:07.000000000 -0500
@@ -1,11 +1,10 @@
 #hello world
 #my first attempt at making a makefile
 
-SDL_CFLAGS := $(shell sdl-config --cflags)
+CXXFLAGS += $(shell sdl-config --cflags)
 SDL_LDFLAGS := $(shell sdl-config --libs)
 MIXER_FLAGS := -lSDL_mixer -lpthread
 IMG_FLAGS := -lSDL_image
-CC=g++ -O3 -Wall #-ggdb3 
 SOURCES = tetris.cpp pong.cpp text.cpp option.cpp media.cpp
 #these sources need to be linked to the SDL libs so may have special flag settings
 SDL_SOURCES = tong.cpp
@@ -16,13 +15,7 @@
 #builds all, builds the target
 
 all:	$(OBJS)
-	$(CC) $(SDL_LDFLAGS) $(MIXER_FLAGS) $(IMG_FLAGS) $(OBJS) -o $(TARGET)
-
-%.o: %.cpp
-	$(CC) $(SDL_CFLAGS) -c -o $@ $<
-
-tong.o: $(SDL_SOURCES)
-	$(CC) $(SDL_CFLAGS) -c -o $@ $<
+	$(CXX) $(LDFLAGS) $(OBJS) $(SDL_LDFLAGS) $(MIXER_FLAGS) $(IMG_FLAGS) -o $(TARGET)
 
 # clean out the mess
 clean:
