diff --git a/ureadseq.c b/ureadseq.c
index 337b16f..3f8e7f6 100644
--- a/ureadseq.c
+++ b/ureadseq.c
@@ -1578,7 +1578,7 @@ short writeSeq(FILE *outf, const char *seq, const long seqlen,
 
   short linesout = 0, seqtype = kNucleic;
   long  i, j, l, l1, ibase;
-  char  idword[31], endstr[10];
+  char  idword[31], endstr[32];
   char  seqnamestore[128], *seqname = seqnamestore;
   char  s[kMaxseqwidth], *cp;
   char  nameform[10], numform[10], nocountsymbols[10];
