diff --git a/readseq.c b/readseq.c
index 305d23a..9b02a17 100644
--- a/readseq.c
+++ b/readseq.c
@@ -171,6 +171,7 @@ link -w -o readseq -t MPST -c 'MPS ' 
 #include <stdio.h>
 #include <string.h>
 #include <ctype.h>
+#include <stdlib.h>
 
 #include "ureadseq.h"
 
diff --git a/ureadseq.c b/ureadseq.c
index 3f8e7f6..799b315 100644
--- a/ureadseq.c
+++ b/ureadseq.c
@@ -20,6 +20,7 @@
 #include <stdio.h>
 #include <ctype.h>
 #include <string.h>
+#include <stdlib.h>
 
 #define UREADSEQ_G
 #include "ureadseq.h"
