--- a/configure.ac
+++ b/configure.ac
@@ -1,14 +1,13 @@
 AC_INIT(tsemgr, 0.08)
-VERSION=0.08
 AC_PREREQ(2.53)
 
-AM_INIT_AUTOMAKE(tsemgr, $VERSION)
+AM_INIT_AUTOMAKE
 
 AC_PROG_CC
 AC_PROG_INSTALL
 AC_PROG_MAKE_SET
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 # Defines
 AC_DEFINE(TSEMGR_BASE_CONFIG_DIR, [".tsemgr"], [Base config directory.])
@@ -16,34 +16,17 @@ AC_DEFINE(REMOTE_CTREE_FILE, ["tsemgr_re
 AC_DEFINE(BASE_CONFIG_FILE, ["tsemgr.conf"], [Default config file.])
 
 # Add flags for Gtk. 
-AC_CHECK_PROG(GTK_CONFIG, gtk-config, yes, no, $PATH)
-if test $GTK_CONFIG = "no" ; then
-  AC_MSG_ERROR(gtk-config not found. Gtk is required for $PACKAGE)
-fi
+PKG_CHECK_MODULES([GTK], [gtk+ >= 1.2])
 
-CFLAGS="$CFLAGS `gtk-config --cflags`"
-LIBS="$LIBS `gtk-config --libs`"
+CFLAGS="$CFLAGS $GTK_CFLAGS"
+LIBS="$LIBS $GTK_LIBS"
 
 
 # Add flags for OpenObex. 
-AC_CHECK_PROG(OPENOBEX_CONFIG, openobex-config, yes, no, $PATH)
-if test $GTK_CONFIG = "no" ; then
-  AC_MSG_ERROR(openobex-config not found. OpenObex is requried for $PACKAGE)
-fi
+PKG_CHECK_MODULES([OPENOBEX], [openobex >= 1.0])
 
-AC_MSG_CHECKING(OpenObex version)
-OO_VERSION="`openobex-config --version`"
-AC_MSG_RESULT($OO_VERSION)
-
-case $OO_VERSION in
-
-1.*.*) ;;
-*) AC_MSG_ERROR(OpenObex library version is too old.)
-
-esac
-
-CFLAGS="$CFLAGS `openobex-config --cflags`"
-LIBS="$LIBS `openobex-config --libs`"
+CFLAGS="$CFLAGS $OPENOBEX_CFLAGS"
+LIBS="$LIBS $OPENOBEX_LIBS"
 
 
 # Look for the ezV24 library. 
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -1,4 +1,4 @@
-INCLUDES = -I$(top_srcdir)
+AM_CPPFLAGS = -I$(top_srcdir)
 
 bin_PROGRAMS = tsemgr
 
