http://www.ogre3d.org/mantis/view.php?id=425

--- ogre_src_v1-7-4/OgreMain/include/OgreAtomicWrappers.h
+++ ogre_src_v1-7-4/OgreMain/include/OgreAtomicWrappers.h
@@ -31,6 +31,7 @@
 #include <signal.h>
 #include "OgrePrerequisites.h"
 #include "OgreException.h"
+#include "Threading/OgreThreadHeaders.h"
 
 namespace Ogre {
 
--- ogre_src_v1-7-4/OgreMain/include/OgreCommon.h
+++ ogre_src_v1-7-4/OgreMain/include/OgreCommon.h
@@ -42,6 +42,8 @@
 #   pragma GCC visibility pop
 #endif
 
+#include "Threading/OgreThreadHeaders.h"
+
 namespace Ogre {
 	/** \addtogroup Core
 	*  @{
--- ogre_src_v1-7-4/OgreMain/include/OgreConvexBody.h
+++ ogre_src_v1-7-4/OgreMain/include/OgreConvexBody.h
@@ -31,6 +31,7 @@
 
 #include "OgrePrerequisites.h"
 #include "OgrePolygon.h"
+#include "Threading/OgreThreadHeaders.h"
 
 
 namespace Ogre
@@ -58,9 +59,7 @@
 
 		// Static 'free list' of polygons to save reallocation, shared between all bodies
 		static PolygonList msFreePolygons;
-#if OGRE_THREAD_SUPPORT
 		OGRE_STATIC_MUTEX(msFreePolygonsMutex)
-#endif
 
 	public:
 		ConvexBody();
--- ogre_src_v1-7-4/OgreMain/include/OgreLog.h
+++ ogre_src_v1-7-4/OgreMain/include/OgreLog.h
@@ -32,6 +32,8 @@
 #include "OgrePrerequisites.h"
 #include "OgreString.h"
 
+#include "Threading/OgreThreadHeaders.h"
+
 namespace Ogre {
 
 	/** \addtogroup Core
--- ogre_src_v1-7-4/OgreMain/include/OgreSharedPtr.h
+++ ogre_src_v1-7-4/OgreMain/include/OgreSharedPtr.h
@@ -30,6 +30,8 @@
 
 #include "OgrePrerequisites.h"
 
+#include "Threading/OgreThreadHeaders.h"
+
 namespace Ogre {
 	/** \addtogroup Core
 	*  @{
--- ogre_src_v1-7-4/OgreMain/src/OgreConvexBody.cpp
+++ ogre_src_v1-7-4/OgreMain/src/OgreConvexBody.cpp
@@ -44,9 +44,7 @@
 	// Statics
 	//-----------------------------------------------------------------------
 	ConvexBody::PolygonList ConvexBody::msFreePolygons;
-#if OGRE_THREAD_SUPPORT
 	OGRE_STATIC_MUTEX_INSTANCE(ConvexBody::msFreePolygonsMutex)
-#endif
 	//-----------------------------------------------------------------------
 	void ConvexBody::_initialisePool()
 	{
