https://bitbucket.org/sinbad/ogre/commits/6e2e797877491fa9bea169e063a9663834977d8e

--- RenderSystems/GLES2/src/GLSLES/src/OgreGLSLESProgramPipeline.cpp
+++ RenderSystems/GLES2/src/GLSLES/src/OgreGLSLESProgramPipeline.cpp
@@ -138,7 +138,7 @@
 				GLint binaryLength = 0;
 
 #if GL_OES_get_program_binary
-				glGetProgramiv(mGLHandle, GL_PROGRAM_BINARY_LENGTH_OES, &binaryLength);
+				glGetProgramiv(mGLProgramHandle, GL_PROGRAM_BINARY_LENGTH_OES, &binaryLength);
                 GL_CHECK_ERROR;
 #endif
 
@@ -148,7 +148,7 @@
 
 #if GL_OES_get_program_binary
 				// Get binary
-				glGetProgramBinaryOES(mGLHandle, binaryLength, NULL, (GLenum *)newMicrocode->getPtr(), newMicrocode->getPtr() + sizeof(GLenum));
+				glGetProgramBinaryOES(mGLProgramHandle, binaryLength, NULL, (GLenum *)newMicrocode->getPtr(), newMicrocode->getPtr() + sizeof(GLenum));
                 GL_CHECK_ERROR;
 #endif
 
