Index: activesupport-2.3.5/lib/active_support/testing/setup_and_teardown.rb
===================================================================
--- activesupport-2.3.5.orig/lib/active_support/testing/setup_and_teardown.rb
+++ activesupport-2.3.5/lib/active_support/testing/setup_and_teardown.rb
@@ -45,12 +45,7 @@ module ActiveSupport
           return if @method_name.to_s == "default_test"
 
           if using_mocha = respond_to?(:mocha_verify)
-            assertion_counter_klass = if defined?(Mocha::TestCaseAdapter::AssertionCounter)
-                                        Mocha::TestCaseAdapter::AssertionCounter
-                                      else
-                                        Mocha::Integration::TestUnit::AssertionCounter
-                                      end
-            assertion_counter = assertion_counter_klass.new(result)
+            assertion_counter = Mocha::Integration::TestUnit::AssertionCounter.new(result)
           end
 
           yield(Test::Unit::TestCase::STARTED, name)
