--- lib/active_support/testing/setup_and_teardown.rb.~1~	2010-09-10 09:22:52.828721874 +0200
+++ lib/active_support/testing/setup_and_teardown.rb	2010-09-10 09:29:01.825098824 +0200
@@ -97,11 +97,7 @@
 
         def retrieve_mocha_counter(result) #:nodoc:
           if respond_to?(:mocha_verify) # using mocha
-            if defined?(Mocha::TestCaseAdapter::AssertionCounter)
-              Mocha::TestCaseAdapter::AssertionCounter.new(result)
-            else
-              Mocha::Integration::TestUnit::AssertionCounter.new(result)
-            end
+            Mocha::Integration::TestUnit::AssertionCounter.new(result)
           end
         end
       end
