From 996cea18a7710aadd006fff772e8a64759ba178d Mon Sep 17 00:00:00 2001
From: Ralph Sennhauser <sera@gentoo.org>
Date: Sun, 20 Oct 2013 16:51:33 +0200
Subject: [PATCH] Fix parallel install. BGO 440906

---
 Makefile.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile.am b/Makefile.am
index 9d13451..918d2eb 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -246,6 +246,7 @@ endif
 install-data-local:
 	${mkinstalldirs} -d $(DESTDIR)$(mandir)/man1
 	${INSTALL_DATA} $(NETX_SRCDIR)/javaws.1 $(DESTDIR)$(mandir)/man1
+	${mkinstalldirs} $(DESTDIR)$(datarootdir)/$(PACKAGE_NAME)/
 	${INSTALL_DATA} $(NETX_RESOURCE_DIR)/about.jnlp $(DESTDIR)$(datarootdir)/$(PACKAGE_NAME)
 if ENABLE_DOCS
 	${mkinstalldirs} $(DESTDIR)$(htmldir)
-- 
1.8.1.5

