diff -r -u tgif-QPL-4.1.45.orig/Makefile.noimake tgif-QPL-4.1.45/Makefile.noimake
--- tgif-QPL-4.1.45.orig/Makefile.noimake	2006-06-14 00:29:58.000000000 +0200
+++ tgif-QPL-4.1.45/Makefile.noimake	2009-06-05 01:20:32.000000000 +0200
@@ -318,7 +318,11 @@
 		$(TGIFDIR)/tgificon.eps $(TGIFDIR)/tgificon.obj \
 		$(TGIFDIR)/tgificon.xbm $(TGIFDIR)/tgificon.xpm \
 		$(TGIFDIR)/tangram.sym $(TGIFDIR)/tgif.Xdefaults \
-		$(TGIFDIR)/keys.obj
+		$(TGIFDIR)/eq4-2x.sym $(TGIFDIR)/eq4-epstool.sym \
+		$(TGIFDIR)/eq4-lyx-ps2epsi.sym $(TGIFDIR)/eq4-ps2epsi.sym \
+		$(TGIFDIR)/eq4.sym $(TGIFDIR)/eq4xpm.sym \
+		$(TGIFDIR)/flip-flop.sym $(TGIFDIR)/nand2.sym \
+		$(TGIFDIR)/pin.sym $(TGIFDIR)/keys.obj
 	@echo Making install ...
 
 $(BINDIR)/tgif: tgif
@@ -339,6 +343,33 @@
 $(TGIFDIR)/tgificon.xpm: tgificon.xpm
 	$(INSTALL) -c tgificon.xpm $(TGIFDIR)/tgificon.xpm
 
+$(TGIFDIR)/eq4-2x.sym: eq4-2x.sym
+	$(INSTALL) -c eq4-2x.sym $(TGIFDIR)/eq4-2x.sym
+
+$(TGIFDIR)/eq4-epstool.sym: eq4-epstool.sym
+	$(INSTALL) -c eq4-epstool.sym $(TGIFDIR)/eq4-epstool.sym
+
+$(TGIFDIR)/eq4-lyx-ps2epsi.sym: eq4-lyx-ps2epsi.sym
+	$(INSTALL) -c eq4-lyx-ps2epsi.sym $(TGIFDIR)/eq4-lyx-ps2epsi.sym
+
+$(TGIFDIR)/eq4-ps2epsi.sym: eq4-ps2epsi.sym
+	$(INSTALL) -c eq4-ps2epsi.sym $(TGIFDIR)/eq4-ps2epsi.sym
+
+$(TGIFDIR)/eq4.sym: eq4.sym
+	$(INSTALL) -c eq4.sym $(TGIFDIR)/eq4.sym
+
+$(TGIFDIR)/eq4xpm.sym: eq4xpm.sym
+	$(INSTALL) -c eq4xpm.sym $(TGIFDIR)/eq4xpm.sym
+
+$(TGIFDIR)/flip-flop.sym: flip-flop.sym
+	$(INSTALL) -c flip-flop.sym $(TGIFDIR)/flip-flop.sym
+
+$(TGIFDIR)/nand2.sym: nand2.sym
+	$(INSTALL) -c nand2.sym $(TGIFDIR)/nand2.sym
+
+$(TGIFDIR)/pin.sym: pin.sym
+	$(INSTALL) -c pin.sym $(TGIFDIR)/pin.sym
+
 $(TGIFDIR)/tangram.sym: tangram.sym
 	$(INSTALL) -c tangram.sym $(TGIFDIR)/tangram.sym
 
