--- a/main.c
+++ b/main.c
@@ -148,7 +148,7 @@
 	fprintf(stderr, "--persistent-connections	-Q\n");
 	fprintf(stderr, "--no-cache		-Z\n");
 	fprintf(stderr, "--version		-V\n");
-	fprintf(stderr, "--help			-h\n");
+	fprintf(stderr, "--help			-H\n");
 }
 
 void emit_error()
@@ -306,7 +306,7 @@
 		{"password",	1, NULL, 'P' },
 		{"cookie",	1, NULL, 'C' },
 		{"version",	0, NULL, 'V' },
-		{"help",	0, NULL, 'h' },
+		{"help",	0, NULL, 'H' },
 		{NULL,		0, NULL, 0   }
 	};
 
