diff -NrU5 bzip2-1.0.3.original/Makefile bzip2-1.0.3/Makefile
--- bzip2-1.0.3.original/Makefile	2009-08-02 13:31:08.000000000 -0600
+++ bzip2-1.0.3/Makefile	2009-08-02 13:38:59.000000000 -0600
@@ -3,14 +3,14 @@
 
 # To assist in cross-compiling
 CC=gcc
 AR=ar
 RANLIB=ranlib
-LDFLAGS=
+#LDFLAGS=
 
 BIGFILES=-D_FILE_OFFSET_BITS=64
-CFLAGS=-Wall -Winline -O -g $(BIGFILES)
+CFLAGS+=-Wall -Winline -O -g $(BIGFILES)
 
 # Where you want it installed when you do 'make install'
 PREFIX=/usr
 
 
@@ -100,27 +100,27 @@
 	sample1.rb2 sample2.rb2 sample3.rb2 \
 	sample1.tst sample2.tst sample3.tst
 
 blocksort.o: blocksort.c
 	@cat words0
-	$(CC) $(CFLAGS) -c blocksort.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -c blocksort.c
 huffman.o: huffman.c
-	$(CC) $(CFLAGS) -c huffman.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -c huffman.c
 crctable.o: crctable.c
-	$(CC) $(CFLAGS) -c crctable.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -c crctable.c
 randtable.o: randtable.c
-	$(CC) $(CFLAGS) -c randtable.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -c randtable.c
 compress.o: compress.c
-	$(CC) $(CFLAGS) -c compress.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -c compress.c
 decompress.o: decompress.c
-	$(CC) $(CFLAGS) -c decompress.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -c decompress.c
 bzlib.o: bzlib.c
-	$(CC) $(CFLAGS) -c bzlib.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -c bzlib.c
 bzip2.o: bzip2.c
-	$(CC) $(CFLAGS) -c bzip2.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -c bzip2.c
 bzip2recover.o: bzip2recover.c
-	$(CC) $(CFLAGS) -c bzip2recover.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -c bzip2recover.c
 
 
 distclean: clean
 	rm -f manual.ps manual.html manual.pdf
 
