diff -NrU5 deltup-0.4.4.original/Makefile deltup-0.4.4/Makefile
--- deltup-0.4.4.original/Makefile	2009-08-02 13:41:03.000000000 -0600
+++ deltup-0.4.4/Makefile	2009-08-02 15:29:10.000000000 -0600
@@ -1,26 +1,26 @@
 DESTDIR=
 PREFIX=/usr
 BINDIR=$(PREFIX)/bin
 
-CC = g++
+CXX = g++
 
 objfiles=bpatch.o bzip2.o file.o system.o tmpstore.o deltup.o 
 libs=-lstdc++ -lz -lbz2 -lssl
 
 #edelta: edelta.cpp deltup
 #	gcc edelta.cpp ${CXXFLAGS} -o edelta
 %.o : %.cpp
-	${CC} -c $< -o $@ -g #-Wall -pedantic #-DFIXED_FRAMERATE
+	$(CXX) $(CFLAGS) $(LDFLAGS) -c $< -o $@ -g #-Wall -pedantic #-DFIXED_FRAMERATE
 
 all: deltup
 
 depend:
-	gcc -MM *.cpp > .depend
+	$(CXX) -MM *.cpp > .depend
 	
 deltup: $(objfiles)
-	${CC} $(objfiles) ${CXXFLAGS} -o deltup $(libs)
+	$(CXX) $(objfiles) $(CFLAGS) $(LDFLAGS) -o deltup $(libs)
 
 install: deltup
 	mkdir -p $(DESTDIR)$(BINDIR)
 	install -m 755 deltup $(DESTDIR)$(BINDIR)
 
