diff -NrU5 deltup-0.4.4.original/Makefile deltup-0.4.4/Makefile
--- deltup-0.4.4.original/Makefile	2009-08-02 15:38:27.000000000 -0600
+++ deltup-0.4.4/Makefile	2009-08-02 15:38:37.000000000 -0600
@@ -3,11 +3,11 @@
 BINDIR=$(PREFIX)/bin
 
 CXX = g++
 
 objfiles=bpatch.o bzip2.o file.o system.o tmpstore.o deltup.o 
-libs=-lstdc++ -lz -lbz2 -lssl
+libs=-lstdc++ -lz -lbz2 -lcrypto
 
 #edelta: edelta.cpp deltup
 #	gcc edelta.cpp ${CXXFLAGS} -o edelta
 %.o : %.cpp
 	$(CXX) $(CFLAGS) $(LDFLAGS) -c $< -o $@ -g #-Wall -pedantic #-DFIXED_FRAMERATE
