http://bugs.gentoo.org/238803

--- configure.ac
+++ configure.ac
@@ -181,7 +181,7 @@
 	[LIBMING_CFLAGS="-DHAVE_LIBMING"])
 if test -n "${LIBMING_LDFLAGS}" && test -n "${LIBMING_CFLAGS}"; then
    HAVE_LIBMING=yes
-   AC_CHECK_HEADER(util/png2dbl.c,[LIBMING_CFLAGS="$LIBMING_CFLAGS -DHAVE_LIBMINGUTIL"])
+   AC_CHECK_HEADER(util/png2dbl.c,[LIBMING_CFLAGS="$LIBMING_CFLAGS -DHAVE_LIBMINGUTIL -DUSE_PNG"])
 else
    LIBMING_LDFLAGS=
    LIBMING_CFLAGS=
--- src/drvswf.cpp
+++ src/drvswf.cpp
@@ -31,8 +31,7 @@
 //static bool trace = false;
 //static bool cubic = false;
 
-#include "mingpp.h"
-#include "ming_config.h"
+#include <mingpp.h>
 
 #if defined(_WIN32)
 extern "C" {
