diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -75,9 +75,6 @@
 if HAVE_HASKELL
 SUBDIRS += haskell
 endif
-if HAVE_PHP
-SUBDIRS += php
-endif
 if HAVE_ERLANG
 SUBDIRS += erlang erlang/examples
 endif
diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -356,6 +356,11 @@
         DISTRO=UBUNTU
     fi
 fi
+if test -f /etc/gentoo-release; then
+    DISTRO=GENTOO
+fi
+
+
 if test -f /etc/arch-release; then
     DISTRO=ARCHLINUX
 fi
@@ -368,6 +373,7 @@
 AC_CHECK_LIB([portablexdr],[xdrmem_create],[],[
         AC_SEARCH_LIBS([xdrmem_create],[rpc xdr nsl])
         ])
+AS_IF([test "xRPCGEN" = "xno"], [AC_MSG_ERROR([rpcgen not installed])],[])
 
 dnl Check for Augeas (optional).
 PKG_CHECK_MODULES([AUGEAS], [augeas],
@@ -381,9 +387,11 @@
          AC_CHECK_FUNCS([aug_load aug_defvar aug_defnode])
          LIBS="$old_LIBS"
         ],
-        [AC_MSG_WARN([augeas not found, some core features will be disabled])])
+ [AC_MSG_ERROR([augeas not found, some core features will be disabled])])
 
 dnl Check for libselinux (optional).
+dnl TODO selinux triplet
+
 AC_CHECK_HEADERS([selinux/selinux.h])
 AC_CHECK_LIB([selinux],[setexeccon],[
         have_libselinux="$ac_cv_header_selinux_selinux_h"
@@ -409,7 +417,7 @@
 AS_IF([test "x$enable_probes" != "xno"],[
     dnl http://sourceware.org/systemtap/wiki/AddingUserSpaceProbingToApps
     AC_CHECK_HEADERS([sys/sdt.h])
-    dnl AC_CHECK_PROG([DTRACE],[dtrace],[dtrace],[no])
+    AC_CHECK_PROG([DTRACE],[dtrace],[dtrace],[no])
     AS_IF([test "x$ac_cv_header_sys_sdt_h" = "xyes"],[
         AC_DEFINE([ENABLE_PROBES],[1],[enable systemtap/DTrace userspace probes])
     ])
@@ -442,7 +450,7 @@
     AC_MSG_RESULT([yes])
     POD2_STDERR_OPTION="--stderr"
 else
-    AC_MSG_RESULT([no])
+    AC_MSG_RESULT([no]
     POD2_STDERR_OPTION=""
 fi
 AC_SUBST([POD2_STDERR_OPTION])
@@ -464,11 +472,20 @@
 
 dnl Check for optional xmllint.
 AC_CHECK_PROG([XMLLINT],[xmllint],[xmllint],[no])
+AS_IF([test "xXMLLINT" = "xno"], [AC_MSG_ERROR([xmllint not installed])],[])
+
 AM_CONDITIONAL([HAVE_XMLLINT],[test "x$XMLLINT" != "xno"])
 
 dnl po4a for translating man pages and POD files (optional).
 AC_CHECK_PROG([PO4A],[po4a],[po4a],[no])
-AM_CONDITIONAL([HAVE_PO4A], [test "x$PO4A" != "xno"])
+AC_ARG_ENABLE([doc],
+       AS_HELP_STRING([--enable-doc], [Enable generation translating man pages and doc]),
+       [enable_doc=no],
+       [enable_doc=yes])
+AS_IF([test "xPO4A" = "xno"], [AC_MSG_WARN([po4a not installed])],[])  
+AM_CONDITIONAL([HAVE_PO4A],[test "x$PO4A" != "xno" && test "x$enable_doc" != "xno"])
+
+
 
 dnl Check for db_dump, db_load (optional).
 AC_CHECK_PROGS([DB_DUMP],
@@ -641,6 +658,9 @@
         [])
 
 dnl Readline.
+dnl TODO - add ncurses LDFLGS separately
+dnl build --without-redline in fact broken with -lncurses not inherited
+
 AC_ARG_WITH([readline],
     [AS_HELP_STRING([--with-readline],
         [support fancy command line editing @<:@default=check@:>@])],
@@ -894,11 +914,13 @@
     [test "x$RAKE" != "xno" && test -n "$HAVE_LIBRUBY"])
 
 dnl Check for Java.
-AC_ARG_WITH(java_home,
-    [AS_HELP_STRING([--with-java-home],
-        [specify path to JDK directory @<:@default=check@:>@])],
-    [],
-    [with_java_home=check])
+AC_ARG_ENABLE([java],
+       AS_HELP_STRING([--enable-java], [Disable Java language bindings]),
+        [enable_java=no],
+       [enable_java=yes])
+
+AS_IF([test "x$enable_java" = "xyes"],
+       [
 
 if test "x$with_java_home" != "xno"; then
     if test "x$with_java_home" != "xyes" && test "x$with_java_home" != "xcheck"
@@ -1034,7 +1056,9 @@
 AC_SUBST(JNI_INSTALL_DIR)
 AC_SUBST(JNI_VERSION_INFO)
 
-AM_CONDITIONAL([HAVE_JAVA],[test -n "$JAVAC"])
+
+])
+AM_CONDITIONAL([HAVE_JAVA],[test -n "$JAVAC" && test "x$enable_java" != "xno"])
 
 dnl Check for Haskell (GHC).
 GHC=no
@@ -1153,7 +1177,7 @@
 
 dnl Replace libtool with a wrapper that clobbers dependency_libs in *.la files
 dnl http://lists.fedoraproject.org/pipermail/devel/2010-November/146343.html
-LIBTOOL='bash $(top_srcdir)/libtool-kill-dependency_libs.sh $(top_builddir)/libtool'
+dnl LIBTOOL='bash $(top_srcdir)/libtool-kill-dependency_libs.sh $(top_builddir)/libtool'
 AC_SUBST([LIBTOOL])
 
 dnl Produce output files.
