--- OroboROX/AppRun.orig	2006-03-04 22:19:16.000000000 +0100
+++ OroboROX/AppRun	2008-01-21 12:51:26.000000000 +0100
@@ -77,16 +77,6 @@
 		rm -f "$APP_DIR/src/config.cache"
 		[ ! -d "$APP_DIR/=build" ] && mkdir "$APP_DIR/=build"
 		
-		CFLAGS="-Os"
-		COMPILER=`cc -dumpversion` 2>/dev/null	
-		if [ "$ARCH" = ix86 ] && [ ! x"$COMPILER" = x ] ; then			
-			MAJOR=${COMPILER%.?.?}
-			if [ "$MAJOR" = 4 ] ; then
-				CFLAGS="$CFLAGS -march=i486 -mtune=i686 -m32"
-			else
-				CFLAGS="$CFLAGS -march=i486 -mcpu=i686 -m32"
-			fi
-		fi
 		export CFLAGS
 
 		cd "$APP_DIR/=build" && "$APP_DIR/src/configure" --enable-rox --disable-startup-notification --with-platform="$PLATFORM" \
--- OroboROX/src/Makefile.in.orig	2006-03-04 22:19:13.000000000 +0100
+++ OroboROX/src/Makefile.in	2008-01-21 13:05:41.000000000 +0100
@@ -3,7 +3,7 @@
 VPATH = @srcdir@
 PLATFORM = @with_platform@
 PLATFORM_DIR = $(srcdir)/../${PLATFORM}
-CC = @CC@
+CC ?= @CC@
 .SUFFIXES:
 .SUFFIXES: .c .o
 
@@ -12,9 +12,9 @@
 USE_LIBS= @USE_LIBS@
 #PROF = -fprofile-arcs -pg
 
-CFLAGS = -g -I. -I${srcdir} ${PROF} @CFLAGS@  \
+CFLAGS += -I. -I${srcdir} ${PROF} @CFLAGS@  \
 	 `${PKG_CONFIG} --cflags glib-2.0 `
-LDFLAGS = ${PROF} @LDFLAGS@ `${PKG_CONFIG} --libs glib-2.0 ${USE_LIBS}` ${LIBS}
+LDFLAGS += ${PROF} @LDFLAGS@ `${PKG_CONFIG} --libs glib-2.0 ${USE_LIBS}` ${LIBS}
 EXEC = ${PLATFORM_DIR}/${PROG}
 
 ############ Things to change for different programs
@@ -35,10 +35,6 @@
 	${CC} -o "${PROG}" ${OBJECTS} ${LDFLAGS}
 	[ -d "${PLATFORM_DIR}" ] || mkdir "${PLATFORM_DIR}"
 	mv "${PROG}" "${PLATFORM_DIR}"
-	-(cd "${PLATFORM_DIR}" && \
-	 objcopy --only-keep-debug OroboROX OroboROX.dbg && \
-	 strip OroboROX && \
-	 objcopy --add-gnu-debuglink=OroboROX.dbg OroboROX)
 
 clean:
 	rm -f *.o Makefile.bak
