diff -ur f1/freemind/freemind/controller/Controller.java f2/freemind/freemind/controller/Controller.java
--- f1/freemind/freemind/controller/Controller.java	2005-06-14 22:38:06.000000000 +0200
+++ f2/freemind/freemind/controller/Controller.java	2007-02-09 21:13:16.000000000 +0100
@@ -1191,7 +1191,7 @@
 			options.buildPanel();
 			options.setProperties(getFrame().getProperties());
 			dialog.setTitle("Freemind Properties");
-			dialog.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
+			dialog.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);
 			dialog.addWindowListener(new WindowAdapter(){
 			    public void windowClosing(WindowEvent event) {
 			        options.closeWindow();
