--- usr/sbin/dxr3config.orig	2008-07-07 16:31:12.073930409 +0200
+++ usr/sbin/dxr3config	2008-07-07 16:32:24.277256585 +0200
@@ -148,7 +148,7 @@
                                 modprobed="/etc/modprobe.d";;
                 gentoo)         genModprobe="true";
                                 modify="false";
-                                modprobed="/etc/modules.d";;
+                                modprobed="/etc/modprobe.d";;
         esac
 }
 
