--- setup/setup.ml.old	2007-08-07 19:25:13.000000000 +0200
+++ setup/setup.ml	2007-08-07 19:24:59.000000000 +0200
@@ -5,8 +5,8 @@
 
 value port = ref 2316;
 value default_lang = ref "en";
-value setup_dir = ref ".";
-value bin_dir = ref "";
+value setup_dir = ref "/usr/share/geneweb";
+value bin_dir = ref "/usr/bin";
 value lang_param = ref "";
 value only_file = ref "";
 
--- tools/Makefile.inc.unix.old	2008-04-23 15:22:50.000000000 +0200
+++ tools/Makefile.inc.unix	2008-04-23 15:23:04.000000000 +0200
@@ -5,11 +5,11 @@
 CAMLP5F=-DUNIX
 CAMLP5D=+camlp5
 LIBUNIX=-cclib -lunix
-OCAMLC=ocamlc.opt -warn-error A
-OCAMLOPT=ocamlopt.opt -warn-error A
+OCAMLC=ocamlc.opt
+OCAMLOPT=ocamlopt.opt
 OCAMLI=
 OCAMLD=
-STRIP=strip
+STRIP=true
 RM=/bin/rm -f
 EXE=
 STATIC=-ccopt -static
--- Makefile.old	2013-04-18 22:13:39.285809438 +0200
+++ Makefile	2013-04-18 22:14:10.073514138 +0200
@@ -18,7 +18,6 @@
 	cd gwb2ged; $(MAKE) all
 	cd setup; $(MAKE) all
 	cd gwtp; $(MAKE) all
-	cd gui; $(MAKE) all
 
 opt::
 	cd wserver; $(MAKE) opt
@@ -28,7 +27,6 @@
 	cd gwb2ged; $(MAKE) opt
 	cd setup; $(MAKE) opt
 	cd gwtp; $(MAKE) opt
-	cd gui; $(MAKE) opt
 
 install:
 	mkdir -p $(PREFIX)/bin
@@ -111,7 +109,6 @@
 	cp setup/lang/lexicon.txt $(DESTDIR)/gw/setup/lang/.
 	cp setup/gwsetup $(DESTDIR)/gw/gwsetup$(EXE)
 	cp gui/gw/gui_lex.txt $(DESTDIR)/gw/.
-	cp gui/gui $(DESTDIR)/gw/gui$(EXE)
 	cp LICENSE $(DESTDIR)/LICENSE.txt
 	cp etc/START.htm $(DESTDIR)/.
 	cp CHANGES $(DESTDIR)/CHANGES.txt
