--- util/fluxbox-generate_menu.in.orig	2007-12-26 13:46:13.000000000 -0500
+++ util/fluxbox-generate_menu.in	2007-12-26 13:54:59.000000000 -0500
@@ -1685,6 +1685,7 @@
        append_submenu "${GRAPHICMENU}"
                normal_find gimp gimp2 gimp-2.2 inkscape sodipodi xv gqview showimg xpaint kpaint kiconedit \
                    ee xzgv xscreensaver-demo xlock gphoto tuxpaint krita skencil
+					find_it display          append "[exec] (ImageMagick Display) {display}"	
                find_it xnview           append "[exec] (xnview browser) {xnview -browser}"
                find_it blender          append "[exec] (blender) {blender -w}"
                find_it gears            append "[exec] (Mesa gears) {gears}"
@@ -1718,9 +1719,10 @@
     find_it gcalc           append "[exec] (gcalc) {gcalc}" || \
         find_it gnome-calculator append "[exec] (gcalc) {gnome-calculator}"
     find_it ical            append "[exec] (Calendar)   {ical}"
+	 find_it sunbird         append "[exec] (Sunbird)    {sunbird}"
 
     # older <=1.1.3 apparently have stuff like swriter, not sowriter
-    for ext in s so oo xoo; do
+    for ext in oo xoo; do
         find_it ${ext}ffice2 && (
             find_it ${ext}ffice2        append "[exec] (Open Office 2)  {${ext}ffice2}"
             find_it ${ext}base2         append "[exec] (OO Base 2)      {${ext}base2}"
