--- util/fluxbox-generate_menu.in.orig	2007-07-13 13:47:11.000000000 -0600
+++ util/fluxbox-generate_menu.in	2007-07-13 13:51:03.000000000 -0600
@@ -1803,12 +1803,8 @@
 append_submenu "${FBSETTINGSMENU}"
     append "[config] (${CONFIGUREMENU})"
 
-    append_menu "[submenu] (${SYSTEMSTYLES}) {${STYLEMENUTITLE}}"
-        append "[stylesdir] (${PREFIX}/share/fluxbox/styles)"
-    append_menu_end
-
-    append_menu "[submenu] (${USERSTYLES}) {${STYLEMENUTITLE}}"
-        append "[stylesdir] (~/.@pkgprefix@fluxbox@pkgsuffix@/styles)"
+    append_submenu "Styles"
+        append "[include] (@GENTOO_PORTAGE_EPREFIX@/usr/share/fluxbox/menu.d/styles/)"
     append_menu_end
 
     # Backgroundmenu
