--- ufed-0.40/ufed.pl
+++ ufed-0.40/ufed.pl
@@ -23,7 +23,7 @@
 
 delete @use_descriptions{qw(bootstrap build)};
 
-$Portage::all_flags{'-*'} = 1 if defined $Portage::make_conf_flags{'*'} && !$Portage::make_conf_flags{'*'};
+$Portage::make_conf_flags{'-*'} = 1 if defined $Portage::make_conf_flags{'*'} && !$Portage::make_conf_flags{'*'};
 
 for(keys %Portage::all_flags) {
 	@{$use_descriptions{$_}} = "(Unknown)"
@@ -39,21 +39,13 @@
 flags_dialog;
 
 sub finalise(@) {
-	my %flags;
-	@flags{@_} = ();
-	if(exists $flags{'-*'}) {
-		return sort keys %flags;
-	} else {
-		my(@enabled, @disabled);
-		my %all_flags;
-		@all_flags{keys %flags, keys %Portage::default_flags} = ();
-		for(sort keys %all_flags) {
-			next if $_ eq '*';
-			push @enabled,    $_  if exists $flags{$_} && !$Portage::default_flags{$_};
-			push @disabled, "-$_" if $Portage::default_flags{$_} && !exists $flags{$_};
-		}
-		return @enabled, @disabled;
-	}
+	return sort {
+		($a ne '-*') <=> ($b ne '-*')
+		||
+		($a =~ /^-/) <=> ($b =~ /^-/)
+		||
+		$a cmp $b
+	} @_;
 }
 
 sub flags_dialog() {
@@ -82,7 +74,8 @@
 		my %descriptions;
 		for(my $flag=0; $flag<@flags; $flag++) {
 			my $flag = $flags[$flag];
-			print $fh $flag, $Portage::all_flags{$flag} ? ' on ' : ' off ';
+			print $fh $flag;
+			print $fh defined $Portage::make_conf_flags{$flag} ? $Portage::make_conf_flags{$flag} ? ' on ' : ' off ' : ' def ';
 			print $fh exists $Portage::make_defaults_flags{$flag} ? $Portage::make_defaults_flags{$flag} ? '(+' :'(-' :'( ' ;
 			print $fh exists  $Portage::use_defaults_flags{$flag} ?  $Portage::use_defaults_flags{$flag} ?  '+' : '-' : ' ' ;
 			print $fh exists     $Portage::make_conf_flags{$flag} ?     $Portage::make_conf_flags{$flag} ?  '+)': '-)': ' )';
@@ -99,8 +92,8 @@
 	if(POSIX::WIFEXITED($?)) {
 		my $rc = POSIX::WEXITSTATUS($?);
 		if($rc==0) {
-			my @flags = do { local $/; split /\n/, <$fh> };
+			my @flags = grep { $_ ne '--*' } do { local $/; split /\n/, <$fh> };
-			save_flags finalise sort @flags;
+			save_flags finalise @flags;
 		} elsif($rc==1)
 		{ print "Cancelled, not saving changes.\n" }
 		exit $rc;
--- ufed-0.40/ufed-curses-checklist.c
+++ ufed-0.40/ufed-curses-checklist.c
@@ -95,8 +95,10 @@
 
 		line[on.end] = '\0';
 		if(!strcmp(&line[on.start], "on"))
-			flag->on = 'X';
+			flag->on = '+';
 		else if(!strcmp(&line[on.start], "off"))
+			flag->on = '-';
+		else if(!strcmp(&line[on.start], "def"))
 			flag->on = ' ';
 		else
 			exit(-1);
@@ -179,8 +181,10 @@
 		goto descriptiononly;
 	}
 	wmove(win(List), y, 0);
-	sprintf(buf, " [%c] %-*s %-5.5s ",
-		flag->on,
+	sprintf(buf, " %c%c%c %-*s %-5.5s ",
+		flag->on == ' ' ? '(' : '[',
+		flag->on == ' ' ? flag->state[1] : flag->on,
+		flag->on == ' ' ? ')' : ']',
 		minwidth-12, flag->name,
 		flag->state);
 	d = &flag->descr[0];
@@ -297,12 +301,23 @@
 		if(yesno("Cancel? (Y/N) "))
 			return 1;
 		break;
-	case ' ':
-		((struct flag *) *currentitem)->on ^= ' '^'X';
-		wattrset(win(List), COLOR_PAIR(3) | A_BOLD | A_REVERSE);
-		mvwhline(win(List), (*currentitem)->top-topy, 2, ((struct flag *) *currentitem)->on, 1);
+	case ' ': {
+		switch (((struct flag *) *currentitem)->on) {
+		case '+':
+			((struct flag *) *currentitem)->on = '-';
+			break;
+		case '-':
+			((struct flag *) *currentitem)->on = ' ';
+			break;
+		default:
+			((struct flag *) *currentitem)->on = '+';
+			break;
+		}
+		drawflag(*currentitem, TRUE);
+		wmove(win(List), (*currentitem)->top-topy, 2);
 		wrefresh(win(List));
 		break;
+	}
 	case KEY_LEFT:
 		if(descriptionleft>0)
 			descriptionleft--;
@@ -318,7 +333,20 @@
 		break;
 #ifdef NCURSES_MOUSE_VERSION
 	case KEY_MOUSE:
-		((struct flag *) *currentitem)->on ^= ' '^'X';
+		switch (((struct flag *) *currentitem)->on) {
+		case '+':
+			((struct flag *) *currentitem)->on = '-';
+			break;
+		case '-':
+			((struct flag *) *currentitem)->on = ' ';
+			break;
+		default:
+			((struct flag *) *currentitem)->on = '+';
+			break;
+		}
+		drawflag(*currentitem, TRUE);
+		wmove(win(List), (*currentitem)->top-topy, 2);
+		wrefresh(win(List));
 		break;
 #endif
 	case '?':
@@ -347,8 +375,15 @@
 		FILE *output = fdopen(4, "w");
 		struct flag *flag = flags;
 		do {
-			if(flag->on=='X')
+			switch(flag->on)
+			{
+			case '+':
 				fprintf(output, "%s\n", flag->name);
+				break;
+			case '-':
+				fprintf(output, "-%s\n", flag->name);
+				break;
+			}
 			flag = (struct flag *) flag->item.next;
 		} while(flag!=flags);
 		fclose(output);
