--- ufed-0.40/ufed.pl
+++ ufed-0.40/ufed.pl
@@ -161,6 +161,7 @@
 		close $makeconf;
 	}
 
+	my $sourcing = 0;
 	eval {
 		# USE comment start/end (start/end of newline character at the end, specifically)
 		# default to end of make.conf, to handle make.confs without #USE=
@@ -180,8 +181,12 @@
 			/\G$IDENT/gc or die;
 			my $name = $1;
 			/\G$BLANK/gc;
+			if($name ne 'source') {
 			/\G$ASSIG/gc or die;
 			/\G$BLANK/gc;
+			} else {
+				$sourcing = 1;
+			}
 			die if pos == length;
 			if($name ne 'USE') {
 				/\G(?:$UQVAL|$SQVAL|$DQVAL)+/gc or die;
@@ -265,8 +270,12 @@
 				/\G$IDENT/gc or die;
 				my $name = $1;
 				/\G$BLANK/gc;
+				if($name ne 'source') {
 				/\G$ASSIG/gc or die;
 				/\G$BLANK/gc;
+				} else {
+					$sourcing = 1;
+				}
 				/\G(?:$UQVAL|$SQVAL|$DQVAL)+/gc or die;
 				my $end = pos;
 				if($name eq 'USE') {
@@ -278,6 +287,10 @@
 	};
 	die "Parse error when writing make.conf - did you modify it while ufed was running?\n" if $@;
 
+	print STDERR <<EOF if $sourcing;
+Warning: source command found in /etc/make.conf. Flags may
+be saved incorrectly if the sourced file modifies them.
+EOF
 	{
 		open my $makeconf, '>', '/etc/make.conf' or die "Couldn't open /etc/make.conf\n";
 		print $makeconf $_;
--- ufed-0.40/Portage.pm
+++ ufed-0.40/Portage.pm
@@ -237,8 +237,10 @@
 				/\G$IDENT/gc or die;
 				my $name = $1;
 				/\G$BLANK/gc;
+				if($name ne 'source') {
 				/\G$ASSIG/gc or die;
 				/\G$BLANK/gc;
+				}
 				die if pos == length;
 				my $value = '';
 				for(;;) {
@@ -262,7 +264,20 @@
 						last
 					}
 				}
+				if($name eq 'source') {
+					open my $f, '<', $value or die;
+					my $pos = pos;
+					substr($_, pos, 0) = do {
+						local $/;
+						my $text = <$f>;
+						die if not defined $text;
+						$text;
+					};
+					pos = $pos;
+					close $f or die;
+				} else {
 				$env{$name} = $value;
+				}
 			}
 		};
 		die "Parse error in $fname\n" if $@;
