--- ufed-0.40/ufed.pl
+++ ufed-0.40/ufed.pl
@@ -33,8 +33,13 @@
 }
 @{$use_descriptions{'-*'}} = 'Never enable any flags other than those specified in /etc/make.conf';
 
-for(keys %Portage::use_masked_flags)
-{ delete $use_descriptions{$_} if $Portage::use_masked_flags{$_} }
+for(keys %Portage::use_masked_flags) {
+	if($Portage::use_masked_flags{$_}) {
+		delete $Portage::default_flags{$_};
+		delete $Portage::all_flags{$_};
+		delete $use_descriptions{$_};
+	}
+}
 
 flags_dialog;
 
