--- src/Makefile.old	2010-10-18 14:44:19.000000000 +0200
+++ src/Makefile	2010-10-18 14:45:05.000000000 +0200
@@ -34,7 +34,7 @@
 
 plugin: $(OBJECTS)
 	rm -f $(PLUGIN)
-	$(CC) -shared -Wl,-soname,$(PLUGIN) $(CFLAGS) $(OBJECTS) -o $(PLUGIN) $(LIBS)
+	$(CC) $(LDFLAGS) -shared -Wl,-soname,$(PLUGIN) $(CFLAGS) $(OBJECTS) -o $(PLUGIN) $(LIBS)
 
 
 cfg:
--- src/cfg-gtk2/Makefile.old	2010-10-18 14:45:38.000000000 +0200
+++ src/cfg-gtk2/Makefile	2010-10-18 14:46:06.000000000 +0200
@@ -12,8 +12,7 @@
 
 cfg:	${CFG_OBJECTS}
 	${RM} ${CFG}
-	${CC} `pkg-config --cflags --libs gtk+-2.0` ${CFLAGS} ${CFG_OBJECTS} -o ${CFG} ${LIBS}
-	${STRIP} ${CFG}
+	${CC} $(LDFLAGS) ${CFLAGS} ${CFG_OBJECTS} -o ${CFG} ${LIBS}
 
 clean:
 	${RM} *.o ${CFG}
