--- gtk-v2/src/png.c
+++ gtk-v2/src/png.c
@@ -571,7 +571,7 @@
         png_destroy_read_struct (&png_ptr, NULL, NULL);
         return PNGX_OUTOFMEM;
     }
-    if (setjmp (png_ptr->jmpbuf)) {
+    if (setjmp (png_jmpbuf(png_ptr))) {
         png_destroy_read_struct (&png_ptr, &info_ptr,NULL);
         return PNGX_DATA;
     }
--- x11/png.c
+++ x11/png.c
@@ -88,7 +88,7 @@
 	png_destroy_read_struct (&png_ptr, NULL, NULL);
 	return NULL;
     }
-    if (setjmp (png_ptr->jmpbuf)) {
+    if (setjmp (png_jmpbuf(png_ptr))) {
 	png_destroy_read_struct (&png_ptr, &info_ptr, NULL);
 	return NULL;
     }
@@ -600,7 +600,7 @@
 	png_destroy_read_struct (&png_ptr, NULL, NULL);
 	return PNGX_OUTOFMEM;
     }
-    if (setjmp (png_ptr->jmpbuf)) {
+    if (setjmp (png_jmpbuf(png_ptr))) {
 	png_destroy_read_struct (&png_ptr, &info_ptr, NULL);
 	return PNGX_DATA;
     }
