--- mlton-20100608-orig/runtime/gc/int-inf.c	2010-06-11 23:35:14.000000000 +1000
+++ mlton-20100608/runtime/gc/int-inf.c	2013-04-25 16:45:45.528297232 +1000
@@ -177,8 +177,8 @@
 
 static inline objptr binary (objptr lhs, objptr rhs, size_t bytes,
                              void(*binop)(__mpz_struct *resmpz,
-                                          __gmp_const __mpz_struct *lhsspace,
-                                          __gmp_const __mpz_struct *rhsspace)) {
+                                          const __mpz_struct *lhsspace,
+                                          const __mpz_struct *rhsspace)) {
   __mpz_struct lhsmpz, rhsmpz, resmpz;
   mp_limb_t lhsspace[LIMBS_PER_OBJPTR + 1], rhsspace[LIMBS_PER_OBJPTR + 1];
 
@@ -254,7 +254,7 @@
 
 static objptr unary (objptr arg, size_t bytes,
                      void(*unop)(__mpz_struct *resmpz,
-                                 __gmp_const __mpz_struct *argspace)) {
+                                 const __mpz_struct *argspace)) {
   __mpz_struct argmpz, resmpz;
  mp_limb_t argspace[LIMBS_PER_OBJPTR + 1];
 
@@ -280,7 +280,7 @@
 
 static objptr shary (objptr arg, Word32_t shift, size_t bytes,
                      void(*shop)(__mpz_struct *resmpz,
-                                 __gmp_const __mpz_struct *argspace,
+                                 const __mpz_struct *argspace,
                                  unsigned long shift))
 {
   __mpz_struct argmpz, resmpz;
