--- libs/base/trunk/Source/Additions/GSXML.m	2012/10/21 05:25:31	35729
+++ libs/base/trunk/Source/Additions/GSXML.m	2012/10/22 04:22:25	35730
@@ -983,7 +983,11 @@
 	1,
 	"utf-8");
       xmlOutputBufferFlush(buf);
+#if LIBXML_VERSION < 20900
       string = UTF8StrLen(buf->buffer->content, buf->buffer->use);
+#else
+      string = UTF8StrLen(xmlBufContent(buf->buffer), xmlBufUse(buf->buffer));
+#endif
       xmlOutputBufferClose(buf);
     }
   return string;
