From 6b4e0a3ebcc2f43e1918ff84ddb25f87accbcd78 Mon Sep 17 00:00:00 2001
From: Albert Zeyer <albert.zeyer@rwth-aachen.de>
Date: Tue, 12 Jan 2010 16:49:35 +0100
Subject: [PATCH] option in CMake to disable joystick support

---
 CMakeOlxCommon.cmake |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/CMakeOlxCommon.cmake b/CMakeOlxCommon.cmake
index e4279ca..815d310 100644
--- a/CMakeOlxCommon.cmake
+++ b/CMakeOlxCommon.cmake
@@ -33,6 +33,7 @@ OPTION(PYTHON_DED_EMBEDDED "Python embedded in dedicated server"  No)
 OPTION(OPTIM_PROJECTILES "Enable optimisations for projectiles" Yes)
 OPTION(MEMSTATS "Enable memory statistics and debugging" No)
 OPTION(BREAKPAD "Google Breakpad support" Yes)
+OPTION(DISABLE_JOYSTICK "Disable joystick support" No)
 
 IF (DEBUG)
 	SET(CMAKE_BUILD_TYPE Debug)
@@ -161,6 +162,10 @@ ELSE (BREAKPAD)
 	ADD_DEFINITIONS(-DNBREAKPAD)
 ENDIF (BREAKPAD)
 
+IF (DISABLE_JOYSTICK)
+	ADD_DEFINITIONS(-DDISABLE_JOYSTICK)
+ENDIF (DISABLE_JOYSTICK)
+
 IF (GCOREDUMPER)
 	INCLUDE_DIRECTORIES(${OLXROOTDIR}/libs/coredumper/src)
 	ADD_DEFINITIONS(-DGCOREDUMPER)
-- 
1.6.3

