--- mkinitcpio-0.9.2/install/base
+++ mkinitcpio-0.9.2/install/base
@@ -12,9 +12,9 @@
     add_symlink /bin usr/bin
     add_symlink /usr/sbin bin
 
-    add_binary /usr/lib/initcpio/busybox /bin/busybox
+    add_binary /bin/busybox
 
-    for applet in $(/usr/lib/initcpio/busybox --list); do
+    for applet in $(/bin/busybox --list); do
         add_symlink "/usr/bin/$applet" busybox
     done
 
@@ -35,7 +35,7 @@
 
     add_file "/usr/lib/initcpio/init_functions" "/init_functions"
     add_file "/usr/lib/initcpio/init" "/init"
-    add_file "/usr/lib/modprobe.d/usb-load-ehci-first.conf"
+    add_file "/lib/modprobe.d/usb-load-ehci-first.conf"
 }
 
 help() {
