--- mkinitcpio-modules-0_p20120704/install/udev
+++ mkinitcpio-modules-0_p20120704/install/udev
@@ -4,14 +4,14 @@
     local rules tool
 
     add_file "/etc/udev/udev.conf"
-    add_binary /usr/lib/systemd/systemd-udevd /usr/bin/udevd
+    add_binary /sbin/udevd /usr/bin/udevd
     add_binary /usr/bin/udevadm
 
     for rules in 50-udev-default.rules 60-persistent-storage.rules 80-drivers.rules; do
-        add_file "/usr/lib/udev/rules.d/$rules"
+        add_file "/lib/udev/rules.d/$rules"
     done
     for tool in ata_id scsi_id; do
-        add_file "/usr/lib/udev/$tool"
+        add_file "/lib/udev/$tool"
     done
 
     add_runscript
