diff -Naur a/usr/bin/SpiderOak b/usr/bin/SpiderOak
--- a/usr/bin/SpiderOak	2011-12-26 16:27:35.000000000 -0500
+++ b/usr/bin/SpiderOak	2012-01-09 10:04:03.000000000 -0500
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-LD_LIBRARY_PATH="/usr/lib/SpiderOak:$LD_LIBRARY_PATH"
+LD_LIBRARY_PATH="/opt/SpiderOak:$LD_LIBRARY_PATH"
 export LD_LIBRARY_PATH
 QT_PLUGIN_PATH= ; export QT_PLUGIN_PATH
-exec "/usr/lib/SpiderOak/SpiderOak" "$@"
+exec "/opt/SpiderOak/SpiderOak" "$@"
