diff -Naur sane-backends-1.0.20-vanilla/tools/sane-desc.c sane-backends-1.0.20/tools/sane-desc.c
--- sane-backends-1.0.20-vanilla/tools/sane-desc.c	2009-07-04 19:05:52.000000000 +0200
+++ sane-backends-1.0.20/tools/sane-desc.c	2009-07-04 19:07:17.000000000 +0200
@@ -3475,7 +3475,7 @@
 	    }
 	}
       printf ("\n");
-      printf ("ATTR{idVendor}==\"%s\", ATTR{idProduct}==\"%s\", MODE=\"%s\", GROUP=\"%s\", ENV{libsane_matched}=\"yes\"\n",
+      printf ("ATTRS{idVendor}==\"%s\", ATTRS{idProduct}==\"%s\", MODE=\"%s\", GROUP=\"%s\", ENV{libsane_matched}=\"yes\"\n",
 	      usbid->usb_vendor_id + 2,  usbid->usb_product_id + 2, DEVMODE, DEVGROUP);
       usbid = usbid->next;
     }
