--- backend/coolscan2.c	2006-06-11 20:39:40.000000000 +0200
+++ backend/coolscan2.c.new	2007-02-26 12:29:15.000000000 +0100
@@ -1482,6 +1482,7 @@
       if (xfer_len_out == 0)	/* no more data */
 	{
 	  *len = 0;
+	  s->scanning = SANE_FALSE;
 	  return SANE_STATUS_EOF;
 	}
 
@@ -1533,6 +1534,7 @@
   if (xfer_len_line == 0)	/* no more data */
     {
       *len = 0;
+      s->scanning = SANE_FALSE;
       return SANE_STATUS_EOF;
     }
 
