--- backend/fujitsu.c	2008-01-23 02:21:21.000000000 +0100
+++ backend/fujitsu.c.new	2008-04-08 10:08:09.000000000 +0200
@@ -1221,7 +1221,6 @@
 init_ms(struct fujitsu *s) 
 {
   int ret;
-  int oldDbg=DBG_LEVEL;
   unsigned char buffer[1];
   size_t inLen = sizeof(buffer);
 
@@ -1232,10 +1231,6 @@
     return SANE_STATUS_GOOD;
   }
 
-  if(DBG_LEVEL < 35){
-    DBG_LEVEL = 0;
-  }
-
   set_MSEN_xfer_length (mode_senseB.cmd, inLen);
 
   /* prepick */
@@ -1391,8 +1386,6 @@
     s->has_MS_jobsep=1;
   }
 
-  DBG_LEVEL = oldDbg;
-
   DBG (15, "  prepick: %d\n", s->has_MS_prepick);
   DBG (15, "  sleep: %d\n", s->has_MS_sleep);
   DBG (15, "  duplex: %d\n", s->has_MS_duplex);
