commit 980f83da6c61577dca7789c025af1a4cbb06b1c1
Author: Pierre Willenbrock <pierre@pirsoft.dnsalias.org>
Date:   Wed Jun 3 00:16:16 2009 +0200

    Always send registers before trying to acquire a line

diff --git a/backend/genesys_gl841.c b/backend/genesys_gl841.c
index 5d7addc..d1d20c1 100644
--- a/backend/genesys_gl841.c
+++ b/backend/genesys_gl841.c
@@ -4948,9 +4948,6 @@ gl841_offset_calibration (Genesys_Device * dev)
       return status;
     }
 
-  RIE (gl841_bulk_write_register
-       (dev, dev->calib_reg, GENESYS_GL841_MAX_REGS));
-
   used_res = dev->current_setup.xres;
   num_pixels = dev->current_setup.pixels;
 
@@ -4999,6 +4996,9 @@ gl841_offset_calibration (Genesys_Device * dev)
 
   do {
 
+      RIE (gl841_bulk_write_register
+	   (dev, dev->calib_reg, GENESYS_GL841_MAX_REGS));
+
       for (j=0; j < channels; j++) {
 	  off[j] = (offh[j]+offl[j])/2;
 	  dev->frontend.offset[j] = off[j];
