--- backend/Makefile.in	2008-08-12 21:50:12.000000000 +0200
+++ backend/Makefile.in.new	2008-08-12 21:58:24.000000000 +0200
@@ -234,8 +234,10 @@
 
 install: $(INSTALL_LOCKPATH) install-be@BE_INSTALL_FLAVOR@ install-libsane install-becfg
 
-install-be:
+install-paths:
 	$(MKDIR) $(DESTDIR)$(libdir) $(DESTDIR)$(libsanedir) $(DESTDIR)$(configdir)
+
+install-be: install-paths
 	@# Install all backends in $(prefix)/lib/sane/
 	@list="$(LIBS)"; for be in $$list; do \
 	  echo installing $${be} in $(DESTDIR)$(libsanedir)/$${be}...; \
@@ -267,13 +269,13 @@
 	  cp -p .libs/libsane-$${be}.so $(DESTDIR)$(BEOS_ADDONDIR)/$${be} || exit 1; \
 	done
 
-install-libsane:
+install-libsane: install-paths
 	@echo installing libsane.la in $(DESTDIR)$(libdir)/libsane.la...
 	@$(LIBTOOL) $(MINST) $(INSTALL_PROGRAM) libsane.la \
           $(DESTDIR)$(libdir)/libsane.la >/dev/null
 	@$(LIBTOOL) $(MINST) --finish $(DESTDIR)$(libdir)
 
-install-becfg:
+install-becfg: install-paths
 	@list="$(CONFIGS)"; for cfg in $$list; do \
 	  if test ! -r $${cfg}; then continue; fi; \
 	  if test -f $(DESTDIR)$(configdir)/$${cfg}; then \
