diff --git a/configure.in b/configure.in
index b62d4d3..76e37b2 100644
--- a/configure.in
+++ b/configure.in
@@ -129,7 +129,16 @@ SANE_CHECK_LOCKING
 SANE_CHECK_GPHOTO2
 
 
-PKG_CHECK_MODULES(LIBV4L, libv4l1, have_libv4l1=yes, have_libv4l1=no)
+AC_ARG_WITH(v4l,
+	    AC_HELP_STRING([--with-v4l],
+			   [include the v4l backend @<:@default=yes@:>@]),
+	    [# If --with-v4l=no or --without-v4l, disable backend
+             # as "$with_v4l" will be set to "no"])
+
+if test "$with_v4l" != "no" ; then
+  PKG_CHECK_MODULES(LIBV4L, libv4l1, have_libv4l1=yes, have_libv4l1=no)
+fi
+
 AC_SUBST(LIBV4L_LIBS)
 AC_SUBST(LIBV4L_CFLAGS)
 
