diff -Naur sane-backends-1.0.23/configure.in sane-backends-1.0.23.new/configure.in
--- sane-backends-1.0.23/configure.in	2012-08-19 13:31:34.000000000 -0700
+++ sane-backends-1.0.23.new/configure.in	2013-08-26 14:41:22.613012270 -0700
@@ -326,16 +326,24 @@
     AC_DEFINE(HAVE_DEV_URANDOM, 1, [Is /dev/urandom available?])
 fi
 
-nl added by PN 3/2/12 to detect cups
-$as_echo "checking for cups"
-if test -e /usr/include/cups/cups.h ; then
-    AC_DEFINE(HAVE_CUPS, 1, [Is /usr/include/cups/cups.h available?])
+AC_ARG_ENABLE(cups,
+	AC_HELP_STRING([--enable-cups], [enable CUPS support]),
+	[enable_cups=$enableval], [enable_cups=no])
+
+if test "$enable_cups" != no; then
+    dnl added by PN 3/2/12 to detect cups
+    $as_echo "checking for cups"
+    if test -e /usr/include/cups/cups.h ; then
+	AC_DEFINE(HAVE_CUPS, 1, [Is /usr/include/cups/cups.h available?])
 	with_cups="yes"
 	LIBS="-lcups  $LIBS"
-else
+    else
 	$as_echo "cups.h not found, you may want to install a cups development package"
 	$as_echo "in order to autodetect network scanners in kodakaio."
 	with_cups="no"
+    fi
+else
+    with_cups=no
 fi
 
 dnl ***********
