diff --git a/backend/xerox_mfp.c b/backend/xerox_mfp.c
index cda6e82..ab0f773 100644
--- a/backend/xerox_mfp.c
+++ b/backend/xerox_mfp.c
@@ -386,6 +386,7 @@ static int doc_source_to_code[] = {
 
 static SANE_String_Const scan_modes[] = {
   SANE_VALUE_SCAN_MODE_LINEART,
+  SANE_VALUE_SCAN_MODE_HALFTONE,
   SANE_VALUE_SCAN_MODE_GRAY,
   SANE_VALUE_SCAN_MODE_COLOR,
   NULL
@@ -940,8 +941,8 @@ SANE_Status
 sane_init (SANE_Int * version_code, SANE_Auth_Callback cb)
 {
   DBG_INIT ();
-  DBG (2, "sane_init: Xerox backend (build %d) %p, %p\n", BACKEND_BUILD,
-       (void *)version_code, (void *)cb);
+  DBG (2, "sane_init: Xerox backend (build %d), version %s null, authorize %s null\n", BACKEND_BUILD,
+       (version_code) ? "!=" : "==", (cb) ? "!=" : "==");
 
   if (version_code)
     *version_code = SANE_VERSION_CODE (V_MAJOR, V_MINOR, BACKEND_BUILD);
