diff -Nru openobex-1.3.orig/ircp/Makefile.am openobex-1.3/ircp/Makefile.am
--- openobex-1.3.orig/ircp/Makefile.am	2006-01-03 20:36:15.000000000 +0200
+++ openobex-1.3/ircp/Makefile.am	2006-08-20 09:39:38.940095250 +0300
@@ -11,7 +11,7 @@
 
 bin_PROGRAMS = ircp 
 
-LDADD = $(top_builddir)/lib/libopenobex.la libircp.a
+LDADD = libircp.a $(top_builddir)/lib/libopenobex.la
 
 INCLUDES = -I$(top_builddir)/include
 endif
