--- trunk/Makedefs.in
+++ trunk/Makedefs.in
@@ -96,11 +96,11 @@
 
 # Rule for libtool controlled C objects
 %.lo : %.c
-	$(LIBTOOL) --mode=compile $(CC) -c $(CPPFLAGS) $(CFLAGS) $<
+	$(LIBTOOL) --mode=compile --tag=CC $(CC) -c $(CPPFLAGS) $(CFLAGS) $<
 
 # Rule for dlopen'able C objects
 %.la : %.c
-	$(LIBTOOL) --mode=compile $(CC) -c $(CPPFLAGS) $(CFLAGS) $<
+	$(LIBTOOL) --mode=compile --tag=CC $(CC) -c $(CPPFLAGS) $(CFLAGS) $<
 	$(rep_DL_LD) $(CPPFLAGS) $(CFLAGS) -o $@ $*.lo
 
 # Build dependancy files from C source files.
--- trunk/intl/Makefile.in
+++ trunk/intl/Makefile.in
@@ -77,7 +77,7 @@
 .c.o:
 	$(COMPILE) $<
 .c.lo:
-	$(LIBTOOL) --mode=compile $(COMPILE) $<
+	$(LIBTOOL) --mode=compile --tag=CC $(COMPILE) $<
 
 INCLUDES = -I.. -I. -I$(top_srcdir)/intl -I$(top_srcdir)/lib
 
@@ -92,7 +92,7 @@
 	$(RANLIB) $@
 
 libintl.la: $(OBJECTS)
-	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o $@ $(OBJECTS) \
+	$(LIBTOOL) --mode=link --tag=CC $(CC) $(LDFLAGS) -o $@ $(OBJECTS) \
 		   -version-info 1:0 -rpath $(libdir)
 
 ../po/cat-id-tbl.$lo: ../po/cat-id-tbl.c $(top_srcdir)/po/$(PACKAGE).pot
--- trunk/rules.mk.sh
+++ trunk/rules.mk.sh
@@ -19,7 +19,7 @@
 # foo.la : foo.lo bar.lo
 #	\$(rep_DL_LD) link-opts...
 
-rep_DL_LD=\$(rep_LIBTOOL) --mode=link \$(CC) -avoid-version -module \
+rep_DL_LD=\$(rep_LIBTOOL) --mode=link --tag=CC \$(CC) -avoid-version -module \
 	  -rpath \$(rpath_repcommonexecdir)
 
 rep_DL_INSTALL=\$(rep_LIBTOOL) --mode=install \$(INSTALL)
@@ -27,6 +27,6 @@
 
 # Rule for libtool controlled C objects
 %.lo : %.c
-	\$(rep_LIBTOOL) --mode=compile \$(CC) -c \$(CPPFLAGS) \$(CFLAGS) \$<
+	\$(rep_LIBTOOL) --mode=compile --tag=CC \$(CC) -c \$(CPPFLAGS) \$(CFLAGS) \$<
 
 EOF
--- trunk/src/Makefile.in
+++ trunk/src/Makefile.in
@@ -56,16 +56,16 @@
 all : librep.la $(DL_OBJS) check-dl rep rep-config rep-remote rep-xgettext .libexec
 
 librep.la : $(OBJS) $(LIBOBJS) $(ALLOCA)
-	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) \
+	$(LIBTOOL) --mode=link --tag=CC $(CC) $(LDFLAGS) \
 	  -version-info $(libversion) -export-symbols $(srcdir)/librep.sym \
 	  -o $@ $^ -rpath $(libdir) $(LIBS) $(GMP_LIBS)
 
 rep : $(REP_OBJS) $(EXTRA_LIBOBJS) librep.la
-	$(LIBTOOL) --mode=link $(CC) -export-dynamic $(CPPFLAGS) $(CFLAGS) -o $@ \
+	$(LIBTOOL) --mode=link --tag=CC $(CC) -export-dynamic $(CPPFLAGS) $(CFLAGS) -o $@ \
 	  $(REP_OBJS) librep.la $(EXTRA_LIBOBJS) $(LIBS) $(GMP_LIBS)
 
 srep : $(REP_OBJS) $(EXTRA_LIBOBJS) librep.la
-	$(LIBTOOL) --mode=link $(CC) -static -export-dynamic $(CPPFLAGS) $(CFLAGS) -o $@ \
+	$(LIBTOOL) --mode=link --tag=CC $(CC) -static -export-dynamic $(CPPFLAGS) $(CFLAGS) -o $@ \
 	  $(REP_OBJS) librep.la $(EXTRA_LIBOBJS) $(LIBS) $(GMP_LIBS)
 
 rep-remote : rep-remote.c
@@ -132,7 +132,7 @@
 	chmod +x $@
 
 repdoc : repdoc.o
-	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o $@ $^ $(GDBM_LIBS)
+	$(LIBTOOL) --mode=link --tag=CC $(CC) $(LDFLAGS) -o $@ $^ $(GDBM_LIBS)
 
 sdbm.la : $(SDBM_LOBJS) repsdbm.lo
 	$(rep_DL_LD) $(LDFLAGS) -o $@ $^
