--- librep-0.17-orig/Makefile.in	2003-09-04 07:57:43.000000000 +0200
+++ librep-0.17/Makefile.in	2008-08-16 13:16:41.000000000 +0200
@@ -53,7 +53,6 @@
 	$(INSTALL_SCRIPT) libtool $(DESTDIR)$(repcommonexecdir)
 	$(INSTALL_DATA) rules.mk $(DESTDIR)$(repcommonexecdir)
 	$(INSTALL_SCRIPT) install-aliases $(DESTDIR)$(repcommonexecdir)
-	$(INSTALL_DATA) rep-debugger.el $(DESTDIR)$(emacssitelispdir)
 
 	@echo "---------------------------------------------------------"
 	@echo "Trying to install rep.m4 in $(DESTDIR)$(aclocaldir);"
@@ -64,8 +63,7 @@
 installdirs : mkinstalldirs
 	$(SHELL) $< $(DESTDIR)$(repdir)/$(version) \
 	  $(DESTDIR)$(bindir) $(DESTDIR)$(aclocaldir) \
-	  $(DESTDIR)$(repcommonexecdir) $(DESTDIR)$(repexecdir) \
-	  $(DESTDIR)$(emacssitelispdir)
+	  $(DESTDIR)$(repcommonexecdir) $(DESTDIR)$(repexecdir)
 
 uninstall :
 	-for dir in $(INSTALL_SUBDIRS); do \
