--- librep-0.17/Makedefs.in
+++ librep-0.17/Makedefs.in
@@ -101,7 +101,7 @@
 # Rule for dlopen'able C objects
 %.la : %.c
 	$(LIBTOOL) --mode=compile --tag=CC $(CC) -c $(CPPFLAGS) $(CFLAGS) $<
-	$(rep_DL_LD) $(CPPFLAGS) $(CFLAGS) -o $@ $*.lo
+	$(rep_DL_LD) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) -o $@ $*.lo
 
 # Build dependancy files from C source files.
 .%.d : %.c
--- librep-0.17/src/Makefile.in
+++ librep-0.17/src/Makefile.in
@@ -61,11 +61,11 @@
 	  -o $@ $^ -rpath $(libdir) $(LIBS) $(GMP_LIBS)
 
 rep : $(REP_OBJS) $(EXTRA_LIBOBJS) librep.la
-	$(LIBTOOL) --mode=link --tag=CC $(CC) -export-dynamic $(CPPFLAGS) $(CFLAGS) -o $@ \
+	$(LIBTOOL) --mode=link --tag=CC $(CC) -export-dynamic $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) -o $@ \
 	  $(REP_OBJS) librep.la $(EXTRA_LIBOBJS) $(LIBS) $(GMP_LIBS)
 
 srep : $(REP_OBJS) $(EXTRA_LIBOBJS) librep.la
-	$(LIBTOOL) --mode=link --tag=CC $(CC) -static -export-dynamic $(CPPFLAGS) $(CFLAGS) -o $@ \
+	$(LIBTOOL) --mode=link --tag=CC $(CC) -static -export-dynamic $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) -o $@ \
 	  $(REP_OBJS) librep.la $(EXTRA_LIBOBJS) $(LIBS) $(GMP_LIBS)
 
 rep-remote : rep-remote.c
