--- librep-0.17.3/Makefile.in
+++ librep-0.17.3/Makefile.in
@@ -53,15 +53,13 @@
 	$(INSTALL_SCRIPT) libtool $(DESTDIR)$(repcommonexecdir)
 	$(INSTALL_DATA) rules.mk $(DESTDIR)$(repcommonexecdir)
 	$(INSTALL_SCRIPT) install-aliases $(DESTDIR)$(repcommonexecdir)
-	$(INSTALL_DATA) rep-debugger.el $(DESTDIR)$(emacssitelispdir)
 	mkdir -p $(DESTDIR)$(libdir)/pkgconfig
 	$(INSTALL_DATA) $(top_srcdir)/librep.pc $(DESTDIR)$(libdir)/pkgconfig/
 
 installdirs : mkinstalldirs
 	$(SHELL) $< $(DESTDIR)$(repdir)/$(version) \
 	  $(DESTDIR)$(bindir) $(DESTDIR)$(aclocaldir) \
-	  $(DESTDIR)$(repcommonexecdir) $(DESTDIR)$(repexecdir) \
+	  $(DESTDIR)$(repcommonexecdir) $(DESTDIR)$(repexecdir)
-	  $(DESTDIR)$(emacssitelispdir)
 
 uninstall :
 	-for dir in $(INSTALL_SUBDIRS); do \
