diff -ruN dibbler-0.4.1.orig/CfgMgr/CfgMgr.h dibbler-0.4.1/CfgMgr/CfgMgr.h
--- dibbler-0.4.1.orig/CfgMgr/CfgMgr.h	2006-05-14 02:45:22.000000000 -0500
+++ dibbler-0.4.1/CfgMgr/CfgMgr.h	2006-05-14 02:49:20.000000000 -0500
@@ -40,7 +40,7 @@
 {
  public:
     TCfgMgr(SmartPtr<TIfaceMgr> IfaceMgr);
-    virtual TCfgMgr::~TCfgMgr();
+    virtual ~TCfgMgr();
 
     bool compareConfigs(const string cfgFile, const string oldCfgFile);
     void copyFile(const string cfgFile, const string oldCfgFile);
diff -ruN dibbler-0.4.1.orig/ClntCfgMgr/ClntCfgMgr.h dibbler-0.4.1/ClntCfgMgr/ClntCfgMgr.h
--- dibbler-0.4.1.orig/ClntCfgMgr/ClntCfgMgr.h	2006-05-14 02:45:22.000000000 -0500
+++ dibbler-0.4.1/ClntCfgMgr/ClntCfgMgr.h	2006-05-14 02:51:15.000000000 -0500
@@ -66,7 +66,7 @@
     int countAddrForIA(long IAID);
     
     SmartPtr<TClntCfgGroup> getGroupForIA(long IAID);
-    SmartPtr<TClntCfgIface> TClntCfgMgr::getIfaceByIAID(int iaid);
+    SmartPtr<TClntCfgIface> getIfaceByIAID(int iaid);
     bool isDone();
 
 private:
diff -ruN dibbler-0.4.1.orig/Makefile dibbler-0.4.1/Makefile
--- dibbler-0.4.1.orig/Makefile	2006-05-14 02:45:22.000000000 -0500
+++ dibbler-0.4.1/Makefile	2006-05-14 02:53:38.000000000 -0500
@@ -54,10 +54,10 @@
 	-L$(SRVOPTIONS)  -lSrvOptions \
 	-L$(SRVTRANSMGR) -lSrvTransMgr \
 	-L$(SRVCFGMGR)   -lSrvCfgMgr \
-	-L$(CFGMGR)      -lCfgMgr\
+	-L$(CFGMGR)      -lCfgMgr \
 	-L$(SRVIFACEMGR) -lSrvIfaceMgr \
 	-L$(IFACEMGR)     -lIfaceMgr \
-	-L$(MISC)        -lMisc\
+	-L$(MISC)        -lMisc \
 	-lSrvIfaceMgr -lSrvMsg -lSrvCfgMgr \
 	-L$(SRVADDRMGR)  -lSrvAddrMgr \
 	                 -lAddrMgr \
@@ -81,10 +81,10 @@
 	-L$(RELIFACEMGR) -lRelIfaceMgr \
 	-L$(RELOPTIONS)  -lRelOptions \
 	-L$(RELMESSAGES) -lRelMsg \
-	-L$(LOWLEVEL)    -lLowLevel\
-	-L$(CFGMGR)      -lCfgMgr\
+	-L$(LOWLEVEL)    -lLowLevel \
+	-L$(CFGMGR)      -lCfgMgr \
 	-L$(IFACEMGR)     -lIfaceMgr \
-	-L$(MISC)        -lMisc\
+	-L$(MISC)        -lMisc \
 	-L$(MESSAGES)    -lMsg \
 	-L$(MISC)        -lMisc \
 	-L$(OPTIONS)     -lOptions \
diff -ruN dibbler-0.4.1.orig/Misc/DUID.h dibbler-0.4.1/Misc/DUID.h
--- dibbler-0.4.1.orig/Misc/DUID.h	2006-05-14 02:45:22.000000000 -0500
+++ dibbler-0.4.1/Misc/DUID.h	2006-05-14 02:52:47.000000000 -0500
@@ -38,8 +38,8 @@
     TDUID(char* Plain); // plain
     TDUID(const TDUID &duid);
     TDUID& operator=(const TDUID& duid);
-    bool TDUID::operator==(const TDUID &duid);
-    bool TDUID::operator<=(const TDUID &duid);
+    bool operator==(const TDUID &duid);
+    bool operator<=(const TDUID &duid);
     int getLen();
     char * storeSelf(char* buf);
     const string getPlain();
diff -ruN dibbler-0.4.1.orig/Options/Opt.h dibbler-0.4.1/Options/Opt.h
--- dibbler-0.4.1.orig/Options/Opt.h	2006-05-14 02:45:22.000000000 -0500
+++ dibbler-0.4.1/Options/Opt.h	2006-05-14 02:49:49.000000000 -0500
@@ -41,7 +41,7 @@
     int getOptType();
     int getSubOptSize();
     
-    char* TOpt::storeSubOpt(char* buf);
+    char* storeSubOpt(char* buf);
     SmartPtr<TOpt> getOption(int optType);
 
     // suboptions management
