From 6e3f338f8285583b8c0548607a8c9f6d55c50c3d Mon Sep 17 00:00:00 2001
From: Tomek Mrugalski <thomson@klub.com.pl>
Date: Tue, 8 May 2012 15:22:23 +0200
Subject: [PATCH] Missing header added (debian bug #672003)

---
 CHANGELOG                       |    1 +
 ClntIfaceMgr/ClntIfaceIface.cpp |    1 +
 2 files changed, 2 insertions(+)

diff --git a/ClntIfaceMgr/ClntIfaceIface.cpp b/ClntIfaceMgr/ClntIfaceIface.cpp
index fbf9834..dab0ed6 100644
--- a/ClntIfaceMgr/ClntIfaceIface.cpp
+++ b/ClntIfaceMgr/ClntIfaceIface.cpp
@@ -8,6 +8,7 @@
  *
  */
 
+#include <unistd.h>
 #include <iostream>
 #include "ClntIfaceIface.h"
 #include "Portable.h"
-- 
1.7.10

