diff -ru remotetimers-0.0.1-orig/menu.c remotetimers-0.0.1/menu.c
--- remotetimers-0.0.1-orig/menu.c	2007-10-07 14:07:19.565912321 +0200
+++ remotetimers-0.0.1/menu.c	2007-10-07 14:08:13.062429433 +0200
@@ -206,7 +206,11 @@
 {
   cString day, name("");
   if (timer->WeekDays())
+#if APIVERSNUM >= 10503
+     day = timer->PrintDay(0, timer->WeekDays(), false);
+#else
      day = timer->PrintDay(0, timer->WeekDays());
+#endif
   else if (timer->Day() - time(NULL) < 28 * SECSINDAY) {
      day = itoa(timer->GetMDay(timer->Day()));
      name = WeekDayName(timer->Day());
