Index: code/tools/asm/cmdlib.c
===================================================================
--- code/tools/asm/cmdlib.c	(revision 1098)
+++ code/tools/asm/cmdlib.c	(revision 1099)
@@ -312,7 +312,7 @@ char *ExpandArg (const char *path)
 char *ExpandPath (const char *path)
 {
 	static char full[1024];
-	if (!qdir)
+	if (!qdir[0])
 		Error ("ExpandPath called without qdir set");
 	if (path[0] == '/' || path[0] == '\\' || path[1] == ':') {
 		strcpy( full, path );
@@ -325,7 +325,7 @@ char *ExpandPath (const char *path)
 char *ExpandGamePath (const char *path)
 {
 	static char full[1024];
-	if (!qdir)
+	if (!qdir[0])
 		Error ("ExpandGamePath called without qdir set");
 	if (path[0] == '/' || path[0] == '\\' || path[1] == ':') {
 		strcpy( full, path );
