--- a/configure.ac
+++ b/configure.ac
@@ -257,9 +257,10 @@
 # This is referenced via AC_REQUIRE([PKG_PROG_PKG_CONFIG] in some macros
 # like PKG_CHECK_MODULES. If the first call to such a macro is under an
 # "if" statement, it's safer to call PKG_PROG_PKG_CONFIG directly, see
-# the comments in acolocal.m4
+# the comments in aclocal.m4
 #
 PKG_PROG_PKG_CONFIG
+PKG_INSTALLDIR
 
 AC_ARG_ENABLE(osx-deploy-target,
   AC_HELP_STRING( [--enable-osx-deploy-target],
@@ -2808,6 +2808,7 @@
 AC_OUTPUT(
   Makefile
   doxygen.cfg
+  wireshark.pc
   asn1/Makefile
   _CUSTOM_ASN1_AC_OUTPUT_
   asn1/acp133/Makefile
--- a/Makefile.am
+++ b/Makefile.am
@@ -56,6 +56,12 @@
 	smi_modules ipmap.html pdml2html.xsl
 
 #
+# Install a pkgconfig file
+#
+
+pkgconfig_DATA = wireshark.pc
+
+#
 # Install global profiles in the "profiles" subdirectory
 #
 profilesdir = $(pkgdatadir)
