--- a/acinclude.m4
+++ b/acinclude.m4
@@ -416,7 +416,7 @@
 	  #
 	  CFLAGS="$CFLAGS -I$pcap_dir/include"
 	  CPPFLAGS="$CPPFLAGS -I$pcap_dir/include"
-	  AC_WIRESHARK_ADD_DASH_L(LDFLAGS, $pcap_dir/lib)
+dnl	  AC_WIRESHARK_ADD_DASH_L(LDFLAGS, $pcap_dir/lib)
 	fi
 
 	# Pcap header check
@@ -642,7 +642,7 @@
 	  wireshark_save_CPPFLAGS="$CPPFLAGS"
 	  CPPFLAGS="$CPPFLAGS -I$zlib_dir/include"
 	  wireshark_save_LIBS="$LIBS"
-	  AC_WIRESHARK_ADD_DASH_L(LIBS, $zlib_dir/lib)
+dnl	  AC_WIRESHARK_ADD_DASH_L(LIBS, $zlib_dir/lib)
 	fi
 
 	#
@@ -695,7 +695,7 @@
 			# LIBS.
 			#
 			LIBS=""
-			AC_WIRESHARK_ADD_DASH_L(LIBS, $zlib_dir/lib)
+dnl			AC_WIRESHARK_ADD_DASH_L(LIBS, $zlib_dir/lib)
 			LIBS="$LIBS -lz $wireshark_save_LIBS"
 		else
 			LIBS="-lz $LIBS"
@@ -1268,7 +1268,8 @@
 		if test "x$ac_cv_enable_usr_local" = "xyes" ; then
 			withval=/usr/local
 			if test -d "$withval"; then
-				AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+dnl				AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+				true
 			fi
 		fi
 	fi
@@ -1299,7 +1300,8 @@
 		if test "x$ac_cv_enable_usr_local" = "xyes" ; then
 			withval=/usr/local
 			if test -d "$withval"; then
-				AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+dnl				AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+				true
 			fi
 		fi
 	fi
@@ -1330,7 +1332,8 @@
 		if test "x$ac_cv_enable_usr_local" = "xyes" ; then
 			withval=/usr/local
 			if test -d "$withval"; then
-				AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+dnl				AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+				true
 			fi
 		fi
 	fi
@@ -1619,7 +1622,8 @@
 		if test "x$ac_cv_enable_usr_local" = "xyes" ; then
 			withval=/usr/local
 			if test -d "$withval"; then
-				AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+dnl				AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+				true
 			fi
 		fi
 	fi
--- a/configure.in
+++ b/configure.in
@@ -616,7 +616,8 @@
 		#
 		# Arrange that we search for libraries in "$prefix/lib".
 		#
-		AC_WIRESHARK_ADD_DASH_L(LDFLAGS, $prefix/lib)
+#		AC_WIRESHARK_ADD_DASH_L(LDFLAGS, $prefix/lib)
+		true
 	else
 		AC_MSG_RESULT(no)
 	fi
@@ -647,7 +648,8 @@
 		#
 		# Arrange that we search for libraries in "/usr/local/lib".
 		#
-		AC_WIRESHARK_ADD_DASH_L(LDFLAGS, /usr/local/lib)
+#		AC_WIRESHARK_ADD_DASH_L(LDFLAGS, /usr/local/lib)
+		true
 	else
 		AC_MSG_RESULT(no)
 	fi
@@ -1468,7 +1470,8 @@
 	want_libcap=yes
 elif test -d "$withval"; then
 	want_libcap=yes
-	AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+#	AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+	true
 fi
 ])
 if test "x$with_libcap" = "xno" ; then
@@ -1499,7 +1502,8 @@
 	want_ssl=yes
 elif test -d "$withval"; then
 	want_ssl=yes
-	AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+#	AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+	true
 fi
 ],[
 	want_ssl=no
@@ -1564,7 +1568,8 @@
 	want_c_ares=yes
 elif test -d "$withval"; then
 	want_c_ares=yes
-	AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+#	AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+	true
 fi
 ])
 if test "x$want_c_ares" = "xno" ; then
@@ -1589,7 +1594,8 @@
 	want_adns=yes
 elif test -d "$withval"; then
 	want_adns=yes
-	AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+#	AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+	true
 fi
 ])
 if test "x$want_adns" = "xno" -o "x$have_good_c_ares" = "xyes" ; then
@@ -1614,7 +1620,8 @@
 	want_geoip=yes
 elif test -d "$withval"; then
 	want_geoip=yes
-	AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+#	AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+	true
 fi
 ])
 if test "x$want_geoip" = "xno"; then
