--- a/configure.ac
+++ b/configure.ac
@@ -245,6 +245,8 @@
 # the comments in acolocal.m4
 #
 PKG_PROG_PKG_CONFIG
+PKG_INSTALLDIR
+
 if test -z "$PKG_CONFIG"; then
	AC_MSG_ERROR(I couldn't find pkg-config; make sure it's installed and in your path)
 fi
@@ -2808,6 +2808,7 @@
 AC_OUTPUT(
   Makefile
   doxygen.cfg
+  wireshark.pc
   asn1/Makefile
   _CUSTOM_ASN1_AC_OUTPUT_
   asn1/acp133/Makefile
--- a/Makefile.am
+++ b/Makefile.am
@@ -56,6 +56,12 @@
 	smi_modules ipmap.html pdml2html.xsl
 
 #
+# Install a pkgconfig file
+#
+
+pkgconfig_DATA = wireshark.pc
+
+#
 # Install global profiles in the "profiles" subdirectory
 #
 profilesdir = $(pkgdatadir)
