--- a/configure.in
+++ b/configure.in
@@ -652,7 +652,8 @@
 		#
 		# Arrange that we search for libraries in "$prefix/lib".
 		#
-		AC_WIRESHARK_ADD_DASH_L(LDFLAGS, $prefix/lib)
+# We do not need this - JeR (possibly bug #394479)
+#		AC_WIRESHARK_ADD_DASH_L(LDFLAGS, $prefix/lib)
 	else
 		AC_MSG_RESULT(no)
 	fi
@@ -648,7 +648,7 @@
 		#
 		# Arrange that we search for libraries in "/usr/local/lib".
 		#
-		AC_WIRESHARK_ADD_DASH_L(LDFLAGS, /usr/local/lib)
+#		AC_WIRESHARK_ADD_DASH_L(LDFLAGS, /usr/local/lib)
 	else
 		AC_MSG_RESULT(no)
 	fi
@@ -1469,7 +1469,7 @@
 	want_libcap=yes
 elif test -d "$withval"; then
 	want_libcap=yes
-	AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+#	AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
 fi
 ])
 if test "x$with_libcap" = "xno" ; then
@@ -1500,7 +1500,7 @@
 	want_ssl=yes
 elif test -d "$withval"; then
 	want_ssl=yes
-	AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+#	AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
 fi
 ],[
 	want_ssl=no
@@ -1565,7 +1565,7 @@
 	want_c_ares=yes
 elif test -d "$withval"; then
 	want_c_ares=yes
-	AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+#	AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
 fi
 ])
 if test "x$want_c_ares" = "xno" ; then
@@ -1590,7 +1590,7 @@
 	want_adns=yes
 elif test -d "$withval"; then
 	want_adns=yes
-	AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+#	AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
 fi
 ])
 if test "x$want_adns" = "xno" -o "x$have_good_c_ares" = "xyes" ; then
@@ -1615,7 +1615,7 @@
 	want_geoip=yes
 elif test -d "$withval"; then
 	want_geoip=yes
-	AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+#	AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
 fi
 ])
 if test "x$want_geoip" = "xno"; then
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -435,7 +435,7 @@
 	  # searches the specified directory.
 	  #
 	  CPPFLAGS="$CPPFLAGS -I$pcap_dir/include"
-	  AC_WIRESHARK_ADD_DASH_L(LDFLAGS, $pcap_dir/lib)
+dnl	  AC_WIRESHARK_ADD_DASH_L(LDFLAGS, $pcap_dir/lib)
 	fi
 
 	# Pcap header check
@@ -659,7 +659,7 @@
 	  wireshark_save_CPPFLAGS="$CPPFLAGS"
 	  CPPFLAGS="$CPPFLAGS -I$zlib_dir/include"
 	  wireshark_save_LIBS="$LIBS"
-	  AC_WIRESHARK_ADD_DASH_L(LIBS, $zlib_dir/lib)
+dnl	  AC_WIRESHARK_ADD_DASH_L(LIBS, $zlib_dir/lib)
 	fi
 
 	#
@@ -711,7 +711,7 @@
 			# of LIBS.
 			#
 			LIBS=""
-			AC_WIRESHARK_ADD_DASH_L(LIBS, $zlib_dir/lib)
+dnl			AC_WIRESHARK_ADD_DASH_L(LIBS, $zlib_dir/lib)
 			LIBS="$LIBS -lz $wireshark_save_LIBS"
 		else
 			LIBS="-lz $LIBS"
@@ -1162,7 +1162,8 @@
 		if test "x$ac_cv_enable_usr_local" = "xyes" ; then
 			withval=/usr/local
 			if test -d "$withval"; then
-				AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+dnl				AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+				true
 			fi
 		fi
 	fi
@@ -1193,7 +1194,8 @@
 		if test "x$ac_cv_enable_usr_local" = "xyes" ; then
 			withval=/usr/local
 			if test -d "$withval"; then
-				AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+dnl				AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+				true
 			fi
 		fi
 	fi
@@ -1224,7 +1226,8 @@
 		if test "x$ac_cv_enable_usr_local" = "xyes" ; then
 			withval=/usr/local
 			if test -d "$withval"; then
-				AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+dnl				AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+				true
 			fi
 		fi
 	fi
@@ -1508,7 +1511,8 @@
 		if test "x$ac_cv_enable_usr_local" = "xyes" ; then
 			withval=/usr/local
 			if test -d "$withval"; then
-				AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+dnl				AC_WIRESHARK_ADD_DASH_L(LDFLAGS, ${withval}/lib)
+				true
 			fi
 		fi
 	fi
