--- a/configure.in
+++ b/configure.in
@@ -1091,19 +1091,6 @@
         wireshark_SUBDIRS="codecs ui/gtk"
 	# Don't use GLIB_CFLAGS
 	AM_PATH_GLIB_2_0(2.14.0, , AC_MSG_ERROR(GLib 2.14 or later distribution not found.), gthread gmodule)
-
-	CPPFLAGS="-DGTK_DISABLE_SINGLE_INCLUDES $CPPFLAGS"
-	## Define GTK_DISABLE_DEPRECATED only if GTK lt 3.2
-	## GTK 3.2 deprecates GtkVBox & GtkHBox which are currently used extensively by Wireshark.
-	if test $gtk_config_major_version -eq 2 -o $gtk_config_minor_version -lt 2; then
-		CPPFLAGS="-DGTK_DISABLE_DEPRECATED $CPPFLAGS"
-	fi
-	# CPPFLAGS="-DGDK_DISABLE_DEPRECATED $CPPFLAGS"
-	if test $gtk_config_major_version -eq 2 -a $gtk_config_minor_version -ge 20; then
-		# Enable GSEAL when building with GTK > 2.20 and < 3.0
-		# (Versions prior to 2.22 lacked some necessary accessors.)
-		CPPFLAGS="-DGSEAL_ENABLE $CPPFLAGS"
-	fi
 fi
 
 # Error out if a glib header other than a "top level" header
