 pastebin |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/pastebin b/pastebin
index 1e4c65c..c2aaa2d 100755
--- a/pastebin
+++ b/pastebin
@@ -224,7 +224,7 @@ sub name {
 }
 
 sub language {
-    my $lang = lc $options->{l} || return 'text';
+    my $lang = defined $options->{l} ? lc $options->{l} : return 'text';
 
     for my $language (keys %languages) {
         my $check = Other::escape($languages{$language});
