 EXAMPLES/SIMPLE/makefile | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/EXAMPLES/SIMPLE/makefile b/EXAMPLES/SIMPLE/makefile
index 9eb51db..0764b44 100644
--- a/EXAMPLES/SIMPLE/makefile
+++ b/EXAMPLES/SIMPLE/makefile
@@ -7,21 +7,21 @@
 #\SCCS Information: @(#) 
 # FILE: makefile   SID: 2.2   DATE OF SID: 9/24/96   RELEASE: 2
 #
-include ../../ARmake.inc
+#include ../../ARmake.inc
 #
 # simple drivers
 #
 simple: sssimp dssimp snsimp dnsimp cnsimp znsimp
 #
 sssimp: sssimp.o
-	$(FC) $(FFLAGS) sssimp.o $(ALIBS) -o sssimp
+	$(FC) $(FFLAGS) $(LDFLAGS) sssimp.o $(ALIBS) -o sssimp `$(PKG_CONFIG) --libs blas lapack` -larpack
 dssimp: dssimp.o
-	$(FC) $(FFLAGS) dssimp.o $(ALIBS) -o dssimp
+	$(FC) $(FFLAGS) $(LDFLAGS) dssimp.o $(ALIBS) -o dssimp `$(PKG_CONFIG) --libs blas lapack` -larpack
 snsimp: snsimp.o
-	$(FC) $(FFLAGS) snsimp.o $(ALIBS) -o snsimp
+	$(FC) $(FFLAGS) $(LDFLAGS) snsimp.o $(ALIBS) -o snsimp `$(PKG_CONFIG) --libs blas lapack` -larpack
 dnsimp: dnsimp.o
-	$(FC) $(FFLAGS) dnsimp.o $(ALIBS) -o dnsimp
+	$(FC) $(FFLAGS) $(LDFLAGS) dnsimp.o $(ALIBS) -o dnsimp `$(PKG_CONFIG) --libs blas lapack` -larpack
 cnsimp: cnsimp.o
-	$(FC) $(FFLAGS) cnsimp.o $(ALIBS) -o cnsimp 
+	$(FC) $(FFLAGS) $(LDFLAGS) cnsimp.o $(ALIBS) -o cnsimp `$(PKG_CONFIG) --libs blas lapack` -larpack
 znsimp: znsimp.o
-	$(FC) $(FFLAGS) znsimp.o $(ALIBS) -o znsimp  
+	$(FC) $(FFLAGS) $(LDFLAGS) znsimp.o $(ALIBS) -o znsimp `$(PKG_CONFIG) --libs blas lapack` -larpack
