diff -Naur --exclude configure --exclude dnsmasq --exclude libtool --exclude Makefile --exclude autom4te.cache --exclude Makefile.in --exclude config.status --exclude aclocal.m4 --exclude ltmain.sh --exclude config.sub --exclude config.guess qenv-0.1/Makefile.am qenv-0.1.mod/Makefile.am
--- qenv-0.1/Makefile.am	2004-07-29 02:18:57.000000000 -0700
+++ qenv-0.1.mod/Makefile.am	2005-12-12 11:00:54.000000000 -0800
@@ -1 +1 @@
-SUBDIRS = src dnsmasq
+SUBDIRS = src
diff -Naur --exclude configure --exclude dnsmasq --exclude libtool --exclude Makefile --exclude autom4te.cache --exclude Makefile.in --exclude config.status --exclude aclocal.m4 --exclude ltmain.sh --exclude config.sub --exclude config.guess qenv-0.1/README qenv-0.1.mod/README
--- qenv-0.1/README	2004-08-10 11:17:15.000000000 -0700
+++ qenv-0.1.mod/README	2005-12-12 11:12:24.000000000 -0800
@@ -73,7 +73,7 @@
 
 Setting up the guests
 ---------------------
-The script 'qenv-mkguest' creates the COW HD image for the QEMU guest system
+The script 'qenv-mkguest' creates the QCOW HD image for the QEMU guest system
 and sets up several other QEMU-related parameters:
 
 Usage: qenv-mkguest --name=<node name> [options]
diff -Naur --exclude configure --exclude dnsmasq --exclude libtool --exclude Makefile --exclude autom4te.cache --exclude Makefile.in --exclude config.status --exclude aclocal.m4 --exclude ltmain.sh --exclude config.sub --exclude config.guess qenv-0.1/configure.in qenv-0.1.mod/configure.in
--- qenv-0.1/configure.in	2004-08-06 08:24:10.000000000 -0700
+++ qenv-0.1.mod/configure.in	2005-12-12 11:11:44.000000000 -0800
@@ -27,22 +27,22 @@
 
 
 dnl
-dnl qemu-mkcow
+dnl qemu-img
 dnl
-QEMU_MKCOW=""
-AC_ARG_WITH(qemu-mkcow,
-	AC_HELP_STRING([--with-qemu-mkcow],
-		[location of 'qemu-mkcow' (default is somewhere in your $PATH)]),
-[QEMU_MKCOW=$withval]
+QEMU_IMG=""
+AC_ARG_WITH(qemu-img,
+	AC_HELP_STRING([--with-qemu-img],
+		[location of 'qemu-img' (default is somewhere in your $PATH)]),
+[QEMU_IMG=$withval]
 )
-if test "x$QEMU_MKCOW" = "x"; then
-	AC_PATH_PROG(QEMU_MKCOW, qemu-mkcow)
+if test "x$QEMU_IMG" = "x"; then
+	AC_PATH_PROG(QEMU_IMG, qemu-img)
 fi
-if test "x$QEMU_MKCOW" = "x"; then
-	AC_MSG_ERROR([Cannot find 'qemu-mkcow' executable])
+if test "x$QEMU_IMG" = "x"; then
+	AC_MSG_ERROR([Cannot find 'qemu-img' executable])
 fi
-test -f "$QEMU_MKCOW" || AC_MSG_ERROR(['qemu-mkcow' executable '$QEMU_MKCOW' is not a file])
-test -x "$QEMU_MKCOW" || AC_MSG_ERROR([Cannot execute 'qemu-mkcow' executable '$QEMU_MKCOW'])
+test -f "$QEMU_IMG" || AC_MSG_ERROR(['qemu-img' executable '$QEMU_IMG' is not a file])
+test -x "$QEMU_IMG" || AC_MSG_ERROR([Cannot execute 'qemu-img' executable '$QEMU_IMG'])
 
 dnl
 dnl sudo
diff -Naur --exclude configure --exclude dnsmasq --exclude libtool --exclude Makefile --exclude autom4te.cache --exclude Makefile.in --exclude config.status --exclude aclocal.m4 --exclude ltmain.sh --exclude config.sub --exclude config.guess qenv-0.1/src/qenv-mkguest.in qenv-0.1.mod/src/qenv-mkguest.in
--- qenv-0.1/src/qenv-mkguest.in	2004-08-10 11:03:00.000000000 -0700
+++ qenv-0.1.mod/src/qenv-mkguest.in	2005-12-12 11:10:49.000000000 -0800
@@ -82,7 +82,7 @@
 # Create COW image
 # 
 mkdir @VAR@/$NODE_NAME
-@QEMU_MKCOW@ @VAR@/$NODE_NAME/hda.img $HDA_SIZE || exit 1
+@QEMU_IMG@ create -f qcow @VAR@/$NODE_NAME/hda.img $HDA_SIZE || exit 1
 
 echo "Node '$NODE_NAME' created. Start it with:"
 echo
--- qenv-0.1/src/Makefile.am.orig	2005-12-12 11:44:03.000000000 -0800
+++ qenv-0.1/src/Makefile.am	2005-12-12 11:43:50.000000000 -0800
@@ -3,5 +3,5 @@
 libexec_SCRIPTS = tunconfig tunconfig-root
 
 install-data-hook:
-	mkdir -p $(ETC)
-	mkdir -p $(VAR)/lib
+	mkdir -p $(DESTDIR)$(ETC)
+	mkdir -p $(DESTDIR)$(VAR)/lib
