--- src/Makefile.in	2011-11-11 19:19:12.000000000 +0100
+++ src/Makefile.in.new	2012-01-24 02:57:15.523607932 +0100
@@ -18,7 +18,7 @@
 CFLAGS	:= @CFLAGS@ @NEED_FNESTED_FUNCTIONS@
 CFLAGS	+= -Wall -funsigned-char -Os -DFSVS_VERSION='"$(VERSION)"'
 LDFLAGS	:= @LDFLAGS@
-FSVS_LDFLAGS = $(LDFLAGS) -lsvn_subr-1 -lsvn_delta-1 -lsvn_ra-1 -lpcre -lgdbm 
+FSVS_LIBS = -lsvn_subr-1 -lsvn_delta-1 -lsvn_ra-1 -lpcre -lgdbm
 EXTRALIBS	:= @EXTRALIBS@
 WAA_CHARS?= @WAA_WC_MD5_CHARS@
 
@@ -152,7 +152,7 @@
 # change, too.
 $(DEST): $(C_FILES:%.c=%.o)
 	@echo "     Link $@"
-	@$(CC) $(FSVS_LDFLAGS) $(LDLIBS) $(LIBS) $(EXTRALIBS) -o $@ $^
+	@$(CC) -o $@ $^ $(LDFLAGS) $(LDLIBS) $(LIBS) $(FSVS_LIBS) $(EXTRALIBS)
 ifeq (@ENABLE_RELEASE@, 1)
 	-strip $@
 endif
