--- src/plugins/CopyTitle/copytitle.cpp.orig	2014-01-29 21:43:11.073465642 +0400
+++ src/plugins/CopyTitle/copytitle.cpp	2014-01-29 21:43:33.079464399 +0400
@@ -61,7 +61,7 @@

 bool CopyTitle::testPlugin()
 {
-    return (QupZilla::VERSION == QLatin1String("1.6.1"));
+    return (QupZilla::VERSION == QLatin1String("1.6.6"));
 }

 QTranslator* CopyTitle::getTranslator(const QString &locale)
--- src/plugins/MailHandle/mailhandle_plugin.cpp.orig	2014-01-29 21:45:27.312457943 +0400
+++ src/plugins/MailHandle/mailhandle_plugin.cpp	2014-01-29 21:45:37.127457388 +0400
@@ -66,7 +66,7 @@
 {
     // Let's be sure, require latest version of QupZilla

-    return (QupZilla::VERSION == QLatin1String("1.6.1"));
+    return (QupZilla::VERSION == QLatin1String("1.6.6"));
 }

 QTranslator* MailHandle_Plugin::getTranslator(const QString &locale)
--- src/plugins/Videoner/videoner_plugin.cpp.orig	2014-01-29 21:44:59.552459512 +0400
+++ src/plugins/Videoner/videoner_plugin.cpp	2014-01-29 21:45:07.055459088 +0400
@@ -62,7 +62,7 @@
 {
     // Let's be sure, require latest version of QupZilla

-    return (QupZilla::VERSION == QLatin1String("1.6.1"));
+    return (QupZilla::VERSION == QLatin1String("1.6.6"));
 }

 QTranslator* Videoner_Plugin::getTranslator(const QString &locale)
