diff -ur linpopup-2.0.4.orig/Makefile.am linpopup-2.0.4/Makefile.am
--- linpopup-2.0.4.orig/Makefile.am	2003-08-17 03:18:55.000000000 +0300
+++ linpopup-2.0.4/Makefile.am	2009-08-05 10:10:40.000000000 +0300
@@ -17,6 +17,5 @@
 	   $(mkinstalldirs) $(DATA_FILE); \
 	   rm -fR $(DATA_FILE); \
 	   touch $(DATA_FILE); \
-	   chgrp nobody $(DATA_FILE); \
-	   chmod 0666 $(DATA_FILE); \
+	   chmod 0600 $(DATA_FILE); \
 	fi
diff -ur linpopup-2.0.4.orig/src/Makefile.am linpopup-2.0.4/src/Makefile.am
--- linpopup-2.0.4.orig/src/Makefile.am	2003-08-17 03:18:40.000000000 +0300
+++ linpopup-2.0.4/src/Makefile.am	2009-08-05 10:09:53.000000000 +0300
@@ -56,8 +56,6 @@
 
 install-data-local:
 	@$(NORMAL_INSTALL)
-	ln -fs $(PACKAGE) $(DESTDIR)$(bindir)/$(PROGNAME_LNK)
 
 uninstall-local:
 	@$(NORMAL_UNINSTALL)
-	rm -f $(DESTDIR)$(bindir)/$(PROGNAME_LNK)
