diff -ur linpopup-2.1.0.orig/Makefile.am linpopup-2.1.0/Makefile.am
--- linpopup-2.1.0.orig/Makefile.am	2007-09-09 22:50:05.000000000 +0300
+++ linpopup-2.1.0/Makefile.am	2009-08-05 11:35:06.000000000 +0300
@@ -21,6 +21,5 @@
 	   $(mkinstalldirs) $(DATA_FILE); \
 	   rm -fR $(DATA_FILE); \
 	   touch $(DATA_FILE); \
-	   chgrp nobody $(DATA_FILE); \
-	   chmod 0666 $(DATA_FILE); \
+	   chmod 0600 $(DATA_FILE); \
 	fi
diff -ur linpopup-2.1.0.orig/src/Makefile.am linpopup-2.1.0/src/Makefile.am
--- linpopup-2.1.0.orig/src/Makefile.am	2007-09-09 22:39:00.000000000 +0300
+++ linpopup-2.1.0/src/Makefile.am	2009-08-05 11:35:06.000000000 +0300
@@ -39,8 +39,6 @@
 
 install-data-local:
 	@$(NORMAL_INSTALL)
-	ln -fs $(PACKAGE) $(DESTDIR)$(bindir)/$(PROGNAME_LNK)
 
 uninstall-local:
 	@$(NORMAL_UNINSTALL)
-	rm -f $(DESTDIR)$(bindir)/$(PROGNAME_LNK)
