diff -uNr dynamips-0.2.8-RC2.ORIG//Makefile dynamips-0.2.8-RC2/Makefile
--- dynamips-0.2.8-RC2.ORIG//Makefile	2010-10-17 13:50:01.000000000 +0100
+++ dynamips-0.2.8-RC2/Makefile	2010-10-17 13:50:57.000000000 +0100
@@ -3,7 +3,7 @@
 
 # Replace x86 by amd64 for a build on x86_64.
 # Use "nojit" for architectures that are not x86 or x86_64.
-DYNAMIPS_ARCH?=x86
+DYNAMIPS_ARCH?=nojit
 
 # Change this to 0 if your system doesn't support RFC 2553 extensions
 HAS_RFC2553?=1
@@ -13,7 +13,7 @@
 HAS_PCAP?=1
 
 # Change this to 1 if your system has posix_memalign
-HAS_POSIX_MEMALIGN?=0
+HAS_POSIX_MEMALIGN?=1
 
 # Current dynamips release
 VERSION_TRAIN=0.2.8
@@ -35,7 +35,7 @@
 MIPS64_ARCH_INC_FILE=\"mips64_$(DYNAMIPS_ARCH)_trans.h\"
 PPC32_ARCH_INC_FILE=\"ppc32_$(DYNAMIPS_ARCH)_trans.h\"
 
-CFLAGS+=-g -Wall -O3 -fomit-frame-pointer \
+CFLAGS+=-g -Wall \
 	-DJIT_ARCH=\"$(DYNAMIPS_ARCH)\" -DJIT_CPU=CPU_$(DYNAMIPS_ARCH) \
 	-DMIPS64_ARCH_INC_FILE=$(MIPS64_ARCH_INC_FILE) \
 	-DPPC32_ARCH_INC_FILE=$(PPC32_ARCH_INC_FILE) \
@@ -215,10 +215,9 @@
 .PHONY: all
 all: $(PROG) nvram_export
 
-$(PROG): mips64_microcode_dump.inc ppc32_microcode_dump.inc \
-	$(LEX_C) $(C_OBJS) $(A_OBJS)
+$(PROG): $(LEX_C) $(C_OBJS) $(A_OBJS)
 	@echo "Linking $@"
-	@$(CC) -o $@ $(C_OBJS) $(A_OBJS) $(LIBS)
+	@$(CC) $(LDFLAGS) -o $@ $(C_OBJS) $(A_OBJS) $(LIBS)
 
 udp_send$(BIN_EXT): udp_send.c net.c crc.c
 	@echo "Linking $@"
@@ -246,9 +245,11 @@
 	@echo "Building assembly definitions header file"
 	@./asmdefs
 
+dev_rom.o: mips64_microcode_dump.inc ppc32_microcode_dump.inc
+
 nvram_export$(BIN_EXT): nvram_export.c
 	@echo "Linking $@"
-	@$(CC) -Wall $(CFLAGS) -o $@ nvram_export.c
+	@$(CC) $(LDFLAGS) -Wall $(CFLAGS) -o $@ nvram_export.c
 
 install: $(PROG) nvram_export
 	@echo "Installing"
