Index: common/toolcommon/Makefile.am
===================================================================
--- common/toolcommon/Makefile.am	(revision 9592)
+++ common/toolcommon/Makefile.am	(working copy)
@@ -8,7 +8,7 @@
 	-I$(top_srcdir)/common/miscutil \
 	-I$(top_srcdir)/common/parsecommon \
 	-I$(top_srcdir)/common/portability \
-	-I$(top_srcdir)/libipmidetect \
+	-I$(top_builddir)/libipmidetect \
 	-I$(top_builddir)/libfreeipmi/include \
 	-I$(top_srcdir)/libfreeipmi/include \
 	-D_GNU_SOURCE \
Index: ipmipower/Makefile.am
===================================================================
--- ipmipower/Makefile.am	(revision 9592)
+++ ipmipower/Makefile.am	(working copy)
@@ -6,7 +6,7 @@
 	-I$(top_srcdir)/common/miscutil \
 	-I$(top_srcdir)/common/parsecommon \
 	-I$(top_srcdir)/common/portability \
-	-I$(top_srcdir)/libipmidetect \
+	-I$(top_builddir)/libipmidetect \
 	-I$(top_builddir)/libfreeipmi/include \
 	-I$(top_srcdir)/libfreeipmi/include \
 	-DWITH_LSD_FATAL_ERROR_FUNC=1 \
