From 54010c34747e410f8332d1f4f85dcc2914836a88 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tiziano=20M=C3=BCller?= <tm@dev-zero.ch>
Date: Fri, 26 Jul 2013 13:34:40 +0200
Subject: [PATCH] Add workaround for broken parallel install with _LTLIBRARIES.

---
 api/src/Makefile.am | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/api/src/Makefile.am b/api/src/Makefile.am
index 8980c02..0782435 100644
--- a/api/src/Makefile.am
+++ b/api/src/Makefile.am
@@ -17,6 +17,10 @@ libgfapi_la_CPPFLAGS = $(GF_CPPFLAGS) -D__USE_FILE_OFFSET64 \
 
 xlator_LTLIBRARIES = api.la
 xlatordir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/xlator/mount
+# workaround for broken parallel install support in automake with LTLIBRARIES
+# http://debbugs.gnu.org/cgi/bugreport.cgi?bug=7328
+install_xlatorLTLIBRARIES = install-xlatorLTLIBRARIES
+$(install_xlatorLTLIBRARIES): install-libLTLIBRARIES
 
 api_la_SOURCES = glfs-master.c
 api_la_DEPENDENCIES = libgfapi.la
-- 
1.8.2.1

