--- gsview-4.7/src/cdll.c	2002-02-10 14:46:00.000000000 +0100
+++ gsview-4.7_new/src/cdll.c	2006-02-17 12:45:47.000000000 +0100
@@ -106,6 +106,10 @@
     }
     dll->revision_number = rv.revision;
 
+    /* Get rid of "wrong version" insanity for three-part GS version numbers */
+    while (dll->revision_number >= 1000)
+	dll->revision_number /= 10;
+
     if ( (dll->revision_number < GS_REVISION_MIN) || 
 	(dll->revision_number > GS_REVISION_MAX) ) {
 	gs_addmessf("\nWrong version of DLL found.\n\
