diff -ruN gsview-4.9.orig/srcunx/unx.mak gsview-4.9/srcunx/unx.mak
--- gsview-4.9.orig/srcunx/unx.mak	2004-11-27 16:51:26.000000000 +0100
+++ gsview-4.9/srcunx/unx.mak	2012-01-31 00:36:50.000000000 +0100
@@ -80,7 +80,7 @@
 CFLAGS=-O -Wall -Wstrict-prototypes -Wmissing-declarations -Wmissing-prototypes -fno-builtin -fno-common -Wcast-qual -Wwrite-strings $(CDEBUG) -DX11 -DUNIX -DNONAG $(RPM_OPT_FLAGS) `gtk-config --cflags` $(XINCLUDE) $(PFLAGS)
 
 # Linker flags
-LFLAGS=$(LDEBUG) $(PLINK) `gtk-config --libs`
+LFLAGS=$(LDEBUG) $(PLINK) `gtk-config --libs` -ldl
 
 COMP=$(CC) $(CFLAGS) -I. -I$(SRCDIR) -I$(SRCUNXDIR) -I$(OBJDIR)
 CCAUX=$(CC) $(CFLAGS) -I$(SRCDIR)
