diff -Nau HTTP.php.orig HTTP.php
--- HTTP.php.orig	23 Apr 2006 14:20:52 -0000
+++ HTTP.php	12 Dec 2007 14:16:25 -0000
@@ -332,6 +332,8 @@
         if (isset($_SERVER['PATH_INFO']) && strlen($_SERVER['PATH_INFO']) && 
                 $_SERVER['PHP_SELF'] != $_SERVER['PATH_INFO']) {
             $path = dirname(substr($_SERVER['PHP_SELF'], 0, -strlen($_SERVER['PATH_INFO'])));
+        } else if (substr($_SERVER['PHP_SELF'], -1) == "/") {
+            $path = $_SERVER['PHP_SELF'];
         } else {
             $path = dirname($_SERVER['PHP_SELF']);
         }
