 src/Makefile | 19 ++++++++++---------
 1 file changed, 10 insertions(+), 9 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index d0a273e..06a0a57 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -48,14 +48,14 @@ endif
 
 include version.mk
 
-CC		= gcc
+CC		?= gcc
 STATIC		=
-STRIP		= strip
+STRIP		= 
 
 ENABLE_NLS = 1
 
 PACKAGE		= wcd
-EXT		= .exe
+EXT		= 
 PROGRAM         = $(PACKAGE)$(EXT)
 BIN		= $(PROGRAM)
 
@@ -183,14 +183,14 @@ endif
 # possible values: ncurses, curses, pdcurses, pdcursesw or <empty>
 
 ifdef UCS
-	CURSES  = ncursesw
+	CURSES  = $(shell pkg-config --libs ncursesw)
 else
-	CURSES = ncurses
+	CURSES = $(shell pkg-config --libs ncurses)
 endif
 
 ifneq (,$(CURSES))
 	DEFS_CURSES = -DWCD_USECURSES
-	LIB_CURSES  = -l$(CURSES)
+	LIB_CURSES  = $(CURSES)
 endif
 
 
@@ -337,7 +337,7 @@ endif
 # (often found as /usr/lib/libncurses.a) to the OBJS1 list.
 
 CFLAGS_USER	=
-CFLAGS	= -O2 -Wall -Wextra -Wno-unused-parameter -Wconversion $(RPM_OPT_FLAGS) $(CPPFLAGS) $(CFLAGS_USER)
+CFLAGS	+= -Wall -Wextra -Wno-unused-parameter -Wconversion $(RPM_OPT_FLAGS) $(CPPFLAGS) $(CFLAGS_USER)
 
 EXTRA_CFLAGS	= -Ic3po \
 		  $(CFLAGS_VERSION) \
@@ -365,7 +365,7 @@ ifdef ASCII_TREE
 endif
 
 LDFLAGS_USER   =
-LDFLAGS		= $(RPM_OPT_FLAGS) \
+LIBS		= $(RPM_OPT_FLAGS) \
 	          $(LIB_CURSES) \
 	          $(LIB_UNISTRING) \
 	          $(LDFLAGS_EXTRA) \
@@ -418,6 +418,7 @@ status:
 	@echo "DEFS_CURSES   = $(DEFS_CURSES)"
 	@echo "EXTRA_DEFS    = $(EXTRA_DEFS)"
 	@echo "LDFLAGS       = $(LDFLAGS)"
+	@echo "LIBS          = $(LIBS)"
 	@echo "LDFLAGS_EXTRA = $(LDFLAGS_EXTRA)"
 	@echo "CFLAGS        = $(CFLAGS)"
 	@echo "CFLAGS_OS     = $(CFLAGS_OS)"
@@ -426,7 +427,7 @@ status:
 
 $(BIN): $(OBJS1)
 	$(MAKE) status
-	$(CC) $(OBJS1) $(LDFLAGS) -o $@
+	$(CC) $(LDFLAGS) $(OBJS1) -o $@ $(LIBS)
 
 %.o: %.c
 	$(CC) $(CFLAGS) $(EXTRA_CFLAGS) $(DEFS) -c $< -o $@
