--- aqualung-0.9beta11/configure.ac.old	2013-05-02 18:20:45.918789345 +0200
+++ aqualung-0.9beta11/configure.ac	2013-05-02 18:21:51.471466816 +0200
@@ -1,9 +1,9 @@
 # Process this file with autoconf to produce a configure script.
 
-AC_INIT(aqualung, 0.9beta11, http://aqualung.factorial.hu/mantis)
-AM_INIT_AUTOMAKE(aqualung, 0.9beta11)
+AC_INIT([aqualung],0.9beta11,[http://aqualung.factorial.hu/mantis])
+AM_INIT_AUTOMAKE
 AC_CONFIG_SRCDIR([src/core.c])
-AM_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADERS([config.h])
 
 
 
