Index: aqualung-0.9beta11/configure.ac
===================================================================
--- aqualung-0.9beta11.orig/configure.ac
+++ aqualung-0.9beta11/configure.ac
@@ -660,7 +660,7 @@ else
 		fi
 	fi
 
-	AC_CHECK_LIB(avformat, av_open_input_file, [avf_lib=yes], [avf_lib=no], [-lavcodec -lavutil -lz])
+	AC_CHECK_LIB(avformat, avformat_open_input, [avf_lib=yes], [avf_lib=no], [-lavcodec -lavutil -lz])
 	AC_CHECK_LIB(avcodec, avcodec_open, [avc_lib=yes], [avc_lib=no], [-lavformat -lavutil -lz])
 
 	if test "$avc_hdr" = "yes" -a "$avf_hdr" = "yes" -a "$avc_lib" = "yes" -a "$avf_lib" = "yes" ; then
Index: aqualung-0.9beta11/src/decoder/dec_lavc.c
===================================================================
--- aqualung-0.9beta11.orig/src/decoder/dec_lavc.c
+++ aqualung-0.9beta11/src/decoder/dec_lavc.c
@@ -111,7 +111,7 @@ lavc_decoder_open(decoder_t * dec, char
 	file_decoder_t * fdec = dec->fdec;
 	int i;
 
-	if (av_open_input_file(&pd->avFormatCtx, filename, NULL, 0, NULL) != 0)
+	if (avformat_open_input(&pd->avFormatCtx, filename, NULL, NULL) != 0)
 		return DECODER_OPEN_BADLIB;
 
 	if (av_find_stream_info(pd->avFormatCtx) < 0)
