diff -rC2 libtrash-2.2/cleanTrash/ct2.pl libtrash-2.0-gentoo/cleanTrash/ct2.pl
*** libtrash-2.2/cleanTrash/ct2.pl	Tue Aug 20 13:25:09 2002
--- libtrash-2.2-gentoo/cleanTrash/ct2.pl	Sun Mar 16 10:48:05 2003
***************
*** 40,44 ****
  ##########################################################################
  # Trash-directory relative to home-dir
! $TRASH_DIR       = '/Desktop/Trash';
  
  # Trash-history file relative to home-dir
--- 40,44 ----
  ##########################################################################
  # Trash-directory relative to home-dir
! $TRASH_DIR       = '/Trash';
  
  # Trash-history file relative to home-dir
***************
*** 48,57 ****
  $MAX_TRASH_SIZE  = 5000;
  
! # files to ignore in Trash dir (KDE friendly!)
! @IGNORE_TRASH    = ('.directory');
  
  # minimum user number (by convention on many Linux systems, 'real' users
  # have UIDs >= 500).  Set to zero to process all users
! $MIN_USER        = 500;
  
  ##########################################################################
--- 48,57 ----
  $MAX_TRASH_SIZE  = 5000;
  
! # files to ignore in Trash dir
! @IGNORE_TRASH    = ();
  
  # minimum user number (by convention on many Linux systems, 'real' users
  # have UIDs >= 500).  Set to zero to process all users
! $MIN_USER        = 1000;
  
  ##########################################################################
diff -rC2 libtrash-2.2/libtrash.conf libtrash-2.2-gentoo/libtrash.conf
*** libtrash-2.2/libtrash.conf	Wed Mar  5 15:39:45 2003
--- libtrash-2.2-gentoo/libtrash.conf	Sun Mar 16 10:50:42 2003
***************
*** 287,291 ****
  # below).
  
! UNREMOVABLE_DIRS =
  #UNREMOVABLE_DIRS=/bin;/boot;/dev;/etc;/lib;/sbin;/usr
  
--- 287,291 ----
  # below).
  
! UNREMOVABLE_DIRS = /bin;/boot;/dev;/etc;/lib;/opt;/sbin;/usr
  #UNREMOVABLE_DIRS=/bin;/boot;/dev;/etc;/lib;/sbin;/usr
  
***************
*** 375,379 ****
  # NOTE: This variable is ignored if global_protection is disabled.
  
! REMOVABLE_MEDIA_MOUNT_POINTS = /mnt
  
  
--- 375,379 ----
  # NOTE: This variable is ignored if global_protection is disabled.
  
! REMOVABLE_MEDIA_MOUNT_POINTS = /mnt/floppy;/mnt/cdrom
  
  
Only in libtrash-2.0-gentoo/src: .Makefile.swp
diff -rC2 libtrash-2.2/src/Makefile libtrash-2.0-gentoo/src/Makefile
*** libtrash-2.2/src/Makefile	Wed Mar  5 01:34:02 2003
--- libtrash-2.2-gentoo/src/Makefile	Sun Mar 16 10:52:26 2003
***************
*** 2,7 ****
  CFLAGS=-Wall -W -Wmissing-prototypes -D_REENTRANT
  
! INSTLIBDIR=/usr/local/lib
! SYSCONFFILE=/etc/libtrash.conf
  
  MAJOR =2
--- 2,7 ----
  CFLAGS=-Wall -W -Wmissing-prototypes -D_REENTRANT
  
! INSTLIBDIR=${DESTDIR}/usr/lib
! SYSCONFFILE=${DESTDIR}/etc/libtrash.conf
  
  MAJOR =2
