diff -ur a/cleanTrash/ct2.pl ./cleanTrash/ct2.pl
--- a/cleanTrash/ct2.pl	2003-12-07 14:11:49.000000000 -0500
+++ b/cleanTrash/ct2.pl	2005-10-04 17:30:15.000000000 -0400
@@ -39,7 +39,8 @@
 ### CONFIGURATION
 ##########################################################################
 # Trash-directory relative to home-dir
-$TRASH_DIR       = '/Desktop/Trash';
+#$TRASH_DIR       = '/Desktop/Trash';
+$TRASH_DIR       = '/Trash';
 
 # Trash-history file relative to home-dir
 $TRASH_HIST_FILE = '/.trashhist';
@@ -48,11 +49,13 @@
 $MAX_TRASH_SIZE  = 5000;
 
 # files to ignore in Trash dir (KDE friendly!)
-@IGNORE_TRASH    = ('.directory');
+#@IGNORE_TRASH    = ('.directory');
+@IGNORE_TRASH    = ( );
 
 # minimum user number (by convention on many Linux systems, 'real' users
 # have UIDs >= 500).  Set to zero to process all users
-$MIN_USER        = 500;
+#$MIN_USER        = 500;
+$MIN_USER        = 1000;
 
 ##########################################################################
 ### PROGRAM
diff -ur a/libtrash.conf ./libtrash.conf
--- a/libtrash.conf	2005-09-24 23:28:17.000000000 -0400
+++ b/libtrash.conf	2005-10-04 17:10:30.000000000 -0400
@@ -286,8 +286,8 @@
 # through the use of the UNCOVER_DIRS environmental variable (see
 # below).
 
-UNREMOVABLE_DIRS =
-#UNREMOVABLE_DIRS=/bin;/boot;/dev;/etc;/lib;/sbin;/usr
+#UNREMOVABLE_DIRS =
+UNREMOVABLE_DIRS=/bin;/boot;/dev;/etc;/lib;/sbin;/usr;/opt
 
 # This is a convenience configuration variable: it simply instructs
 # libtrash to temporarily remove the listed directory(ies) from the
@@ -374,7 +374,7 @@
 
 # NOTE: This variable is ignored if global_protection is disabled.
 
-REMOVABLE_MEDIA_MOUNT_POINTS = /mnt
+REMOVABLE_MEDIA_MOUNT_POINTS = /mnt/floppy;/mnt/cdrom
 
 
 # This setting is meant to allow certain programs to function correctly
diff -ur a/src/Makefile ./src/Makefile
--- a/src/Makefile	2005-09-24 22:41:06.000000000 -0400
+++ b/src/Makefile	2005-10-04 17:12:09.000000000 -0400
@@ -1,8 +1,8 @@
 CC=gcc
 CFLAGS=-Wall -W -Wmissing-prototypes -D_REENTRANT
 
-INSTLIBDIR=/usr/local/lib
-SYSCONFFILE=/etc/libtrash.conf
+INSTLIBDIR=${DESTDIR}/usr/lib
+SYSCONFFILE=${DESTDIR}/etc/libtrash.conf
 
 MAJOR =2
 VERSION =2.4
@@ -29,7 +29,7 @@
 	TRASH_OFF=YES install libtrash.so.${VERSION} ${INSTLIBDIR}/libtrash.so.${VERSION}
 	TRASH_OFF=YES ln -sf libtrash.so.${VERSION} ${INSTLIBDIR}/libtrash.so
 	TRASH_OFF=YES install libtrash.conf.sys --mode a=r ${SYSCONFFILE}
-	ldconfig	
+#	ldconfig	
 
 clean:
 	rm -f *~
