--- src/game/gameplay_action.cpp.old
+++ src/game/gameplay_action.cpp
@@ -268,7 +268,7 @@
  ** 
  ** @version	0.6.8
  **/
-GameplayAction::GameplayAction&
+GameplayAction&
 GameplayAction::operator=(GameplayAction const& action)
 {
   this->type = action.type;
--- src/game/gameplay_actions/pure.cpp.old
+++ src/game/gameplay_actions/pure.cpp
@@ -90,7 +90,7 @@
  ** 
  ** @version	0.7.3
  **/
-GameplayAction::Pure::Pure&
+GameplayAction::Pure&
 GameplayAction::Pure::operator=(Pure const& action)
 {
   this->GameplayAction::operator=(action);
--- src/game/gameplay_actions/player1.cpp.old
+++ src/game/gameplay_actions/player1.cpp
@@ -103,7 +103,7 @@
  ** 
  ** @version	0.7.3
  **/
-GameplayAction::Player1::Player1&
+GameplayAction::Player1&
 GameplayAction::Player1::operator=(Player1 const& action)
 {
   this->GameplayAction::operator=(action);
--- src/game/gameplay_actions/base_cards.cpp.old
+++ src/game/gameplay_actions/base_cards.cpp
@@ -142,7 +142,7 @@
  ** 
  ** @version	0.7.3
  **/
-GameplayAction::BaseCards::BaseCards&
+GameplayAction::BaseCards&
 GameplayAction::BaseCards::operator=(BaseCards const& poverty_shift)
 {
   this->Player1::operator=(poverty_shift);
--- src/game/gameplay_actions/reservation.cpp.old
+++ src/game/gameplay_actions/reservation.cpp
@@ -110,7 +110,7 @@
  ** 
  ** @version	0.7.3
  **/
-GameplayAction::Reservation::Reservation&
+GameplayAction::Reservation&
 GameplayAction::Reservation::operator=(Reservation const& reservation)
 {
   this->Player1::operator=(reservation);
--- src/game/gameplay_actions/marriage.cpp.old
+++ src/game/gameplay_actions/marriage.cpp
@@ -121,7 +121,7 @@
  ** 
  ** @version	0.7.1
  **/
-GameplayAction::Marriage::Marriage&
+GameplayAction::Marriage&
 GameplayAction::Marriage::operator=(Marriage const& marriage)
 {
   this->Player1::operator=(marriage);
--- src/game/gameplay_actions/genscher.cpp.old
+++ src/game/gameplay_actions/genscher.cpp
@@ -120,7 +120,7 @@
  ** 
  ** @version	0.7.1
  **/
-GameplayAction::Genscher::Genscher&
+GameplayAction::Genscher&
 GameplayAction::Genscher::operator=(Genscher const& genscher)
 {
   this->Player1::operator=(genscher);
--- src/game/gameplay_actions/announcement.cpp.old
+++ src/game/gameplay_actions/announcement.cpp
@@ -132,7 +132,7 @@
  ** 
  ** @version	0.7.1
  **/
-GameplayAction::Announcement::Announcement&
+GameplayAction::Announcement&
 GameplayAction::Announcement::operator=(Announcement const& announcement)
 {
   this->Player1::operator=(announcement);
--- src/game/gameplay_actions/card_played.cpp.old
+++ src/game/gameplay_actions/card_played.cpp
@@ -178,7 +178,7 @@
  ** 
  ** @version	0.7.3
  **/
-GameplayAction::CardPlayed::CardPlayed&
+GameplayAction::CardPlayed&
 GameplayAction::CardPlayed::operator=(CardPlayed const& card_played)
 {
   this->Player1::operator=(card_played);
--- src/game/gameplay_actions/trick_full.cpp.old
+++ src/game/gameplay_actions/trick_full.cpp
@@ -224,7 +224,7 @@
  ** 
  ** @version	0.7.1
  **/
-GameplayAction::TrickFull::TrickFull&
+GameplayAction::TrickFull&
 GameplayAction::TrickFull::operator=(TrickFull const& trick_full)
 {
   this->GameplayAction::operator=(trick_full);
--- src/game/gameplay_actions/check.cpp.old
+++ src/game/gameplay_actions/check.cpp
@@ -104,7 +104,7 @@
  ** 
  ** @version	0.7.6
  **/
-GameplayAction::Check::Check&
+GameplayAction::Check&
 GameplayAction::Check::operator=(Check const& action)
 {
   this->Pure::operator=(action);
