--- FreeDoko_0.7.8.orig/src/Makefile
+++ FreeDoko_0.7.8/src/Makefile
@@ -211,15 +211,7 @@
 # compile FreeDoko
 $(PROGNAME) : show_settings objects $(DIRECTORIES)
 	@echo "compiling $(PROGNAME)..."
-# Gentoo users do want to see the real compile line.
-# So remove the next five lines (the echo lines)
-# and remove the '@' in the line after.
-	@echo '$(CXX) $(CXXFLAGS)'
-	@echo '        $$(OBJECTS)'
-	@echo '        $(INCLUDE_ALL)'
-	@echo '        $(LIBS_ALL)'
-	@echo '        -o $@'
-	@$(CXX) $(CXXFLAGS) \
+	$(CXX) $(LDFLAGS) $(CXXFLAGS) \
 	  $(OBJECTS_ALL:%.o=$(FREEDOKO_WORKING_DIRECTORY)/%.o) \
 	  $(INCLUDE_ALL) \
 	  $(LIBS_ALL) \
--- FreeDoko_0.7.8.orig/src/Makefile.os
+++ FreeDoko_0.7.8/src/Makefile.os
@@ -1,6 +1,5 @@
 include $(DEPTH)/../Makefile.os
 
-include $(DEPTH)/Makefile.local
 include $(DEPTH)/Makefile.modules
 
 ifeq ($(OPERATING_SYSTEM), Linux)
--- FreeDoko_0.7.8.orig/src/Makefile.rules
+++ FreeDoko_0.7.8/src/Makefile.rules
@@ -81,9 +81,7 @@
 		  $(DEPTH)/Makefile.local \
 		  $(DEPTH)/Makefile.os
 # Gentoo users do want to see the real compile line.
-# So remove the next line and remove the '@' in the line after.
-	@echo $(CXX) -c $(SUBDIR)/$<
-	@$(CXX) $(CPPFLAGS) $(CXXFLAGS) $(INCLUDE) $(DEPGEN_FLAGS) -o $@ -c $<
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) $(INCLUDE) $(DEPGEN_FLAGS) -o $@ -c $<
 
 
 -include $(OBJECTS:%.o=$(TARGET_DIR_LOCAL)%.d)
--- FreeDoko_0.7.8.orig/src/ui/help.cpp
+++ FreeDoko_0.7.8/src/ui/help.cpp
@@ -142,14 +142,14 @@
     for (list<string>::const_iterator d = datadir.begin();
          d != datadir.end();
          d++)
-      directories.push_back(*d + "/"
+      directories.push_back("/"
                             + ::setting(Setting::MANUAL_DIRECTORY) + "/"
                             + ::translator.token());
     if (::translator.token() != ::setting(Setting::LANGUAGE))
       for (list<string>::const_iterator d = datadir.begin();
            d != datadir.end();
            d++)
-        directories.push_back(*d + "/"
+        directories.push_back("/"
                               + ::setting(Setting::MANUAL_DIRECTORY) + "/"
                               + ::setting(Setting::LANGUAGE));
   } // fill the directories list
