diff -Nru mcpp-2.7/src/internal.H mcpp-2.7-1.1/src/internal.H
--- mcpp-2.7/src/internal.H	2008-03-11 17:04:07.000000000 +0100
+++ mcpp-2.7-1.1/src/internal.H	2008-04-04 10:58:37.000000000 +0200
@@ -526,7 +526,7 @@
                 /* Do the final commands        */
 extern void     print_heap( void);
                 /* Print blocks of heap memory  */
-#if ! HOST_HAVE_STPCPY || HOST_COMPILER == GNUC
+#if ! HOST_HAVE_STPCPY
 extern char *   stpcpy( char * dest, const char * src);
                 /* Non-Standard library function*/
 #endif
