--- skstream/skstreamconfig.h.in.old	2008-11-06 21:15:03.000000000 +0100
+++ skstream/skstreamconfig.h.in	2008-11-06 21:15:21.000000000 +0100
@@ -38,6 +38,7 @@
 #if @SKSTREAM_UNIX_HEADER_FLAG@
   #include <sys/socket.h>
   #include <netinet/in.h>
+  #include <cstring>
 #endif
 #if @SKSTREAM_BEOS_HEADER_FLAG@
   #include <net/socket.h>
--- ping/ping.cpp.old   2009-01-23 16:35:39.000000000 +0100
+++ ping/ping.cpp       2009-01-23 16:35:59.000000000 +0100
@@ -10,6 +10,7 @@
 #include <utility>

 #include <cstdio>
+#include <cstdlib>

 #include <time.h>
 #include <signal.h> // to trap Ctrl+C : SIGINT

