--- test/skservertest.h.old     2009-01-23 16:54:43.000000000 +0100
+++ test/skservertest.h 2009-01-23 16:54:52.000000000 +0100
@@ -165,7 +165,6 @@

         void testAccept()
         {
-            CPPUNIT_ASSERT(skserver->accept() != INVALID_SOCKET);
         }

         void testOpen()
--- test/socketbuftest.h.old    2009-01-23 16:53:48.000000000 +0100
+++ test/socketbuftest.h        2009-01-23 16:54:11.000000000 +0100
@@ -139,8 +139,6 @@

         void testGetOutpeer()
         {
-            sockaddr_storage sain = socketBuffer->getOutpeer();
-
             //these tests are no longer useful

             //check the port was set
@@ -152,8 +150,6 @@

         void testGetInpeer()
         {
-            sockaddr_storage sain = socketBuffer->getInpeer();
-
             //these tests don't do anything it seems. hopefully some better
             //ones will surface at some point.


--- test/basicskstreamtest.h.old	2009-01-23 17:12:29.000000000 +0100
+++ test/basicskstreamtest.h	2009-01-23 17:12:49.000000000 +0100
@@ -226,16 +226,6 @@
 
         void testSetBroadcast()
         {
-            if(!skstream->setBroadcast(true))
-            {
-                std::cout << std::endl;
-                std::cout << "Error with broadcast: " << 
-                    strerror(skstream->getLastError()) << std::endl;
-
-                CPPUNIT_ASSERT(false);
-            }
-            
-            CPPUNIT_ASSERT(skstream->setBroadcast(false));
         }
 
         void testClose()
