diff -ru xineliboutput-cvs-20090810-orig/xine_fbfe_frontend.c xineliboutput-cvs-20090810/xine_fbfe_frontend.c
--- xineliboutput-cvs-20090810-orig/xine_fbfe_frontend.c	2009-08-10 13:17:33.000000000 +0200
+++ xineliboutput-cvs-20090810/xine_fbfe_frontend.c	2009-08-10 13:55:13.996629416 +0200
@@ -139,9 +139,10 @@
                              int xpos, int ypos,
                              int width, int height, int fullscreen, int hud,
                              int modeswitch, const char *modeline, int aspect,
-                             fe_keypress_f keyfunc, int no_x_kbd,
+                             fe_keypress_f keyfunc, int no_x_kbd, int gui_hotkeys,
                              const char *video_port,
-                             int scale_video, int field_order)
+                             int scale_video, int field_order,
+			     const char *aspect_controller, int window_id)
 {
   fe_t *this = (fe_t*)this_gen;
 
@@ -207,7 +208,7 @@
  *
  * configure windows
  */
-static int fbfe_display_config(frontend_t *this_gen, int width, int height, int fullscreen, 
+static int fbfe_display_config(frontend_t *this_gen, int xpos, int ypos, int width, int height, int fullscreen, 
 			       int modeswitch, const char *modeline, int aspect, 
 			       int scale_video, int field_order) 
 {
diff -ru xineliboutput-cvs-20090810-orig/xine_frontend_main.c xineliboutput-cvs-20090810/xine_frontend_main.c
--- xineliboutput-cvs-20090810-orig/xine_frontend_main.c	2009-08-10 13:17:34.000000000 +0200
+++ xineliboutput-cvs-20090810/xine_frontend_main.c	2009-08-10 13:59:13.453270355 +0200
@@ -632,7 +632,14 @@
 
   /* Initialize display */
   if(!fe->fe_display_open(fe, xpos, ypos, width, height, fullscreen, hud, 0,
-			  "", aspect, NULL, noxkbd, video_port, scale_video, 0)) {
+			  "", aspect, NULL, noxkbd, 0, video_port, scale_video, 0,
+			  aspect_controller,
+#ifndef IS_FBFE
+			  window_id
+#else
+			  0
+#endif
+			  )) {
     fprintf(stderr, "Error opening display\n");
     fe->fe_free(fe);
     return -4;
diff -ru xineliboutput-cvs-20090810-orig/xine_sxfe_frontend.c xineliboutput-cvs-20090810/xine_sxfe_frontend.c
--- xineliboutput-cvs-20090810-orig/xine_sxfe_frontend.c	2009-08-10 13:17:34.000000000 +0200
+++ xineliboutput-cvs-20090810/xine_sxfe_frontend.c	2009-08-10 13:55:13.999963596 +0200
@@ -903,8 +903,10 @@
                              int xpos, int ypos,
                              int width, int height, int fullscreen, int hud,
                              int modeswitch, const char *modeline, int aspect,
-                             fe_keypress_f keyfunc, int no_x_kbd, const char *video_port,
-                             int scale_video, int field_order)
+                             fe_keypress_f keyfunc, int no_x_kbd, int gui_hotkeys,
+			     const char *video_port,
+                             int scale_video, int field_order,
+			     const char *aspect_controller, int window_id)
 {
   sxfe_t    *this = (sxfe_t*)this_gen;
 
@@ -1156,6 +1158,7 @@
  * configure windows
  */
 static int sxfe_display_config(frontend_t *this_gen, 
+			       int xpos, int ypos,
 			       int width, int height, int fullscreen, 
 			       int modeswitch, const char *modeline, 
 			       int aspect, int scale_video, 
@@ -1250,7 +1253,7 @@
     this->ypos = this->origypos;
   }
 
-  this->fe.fe_display_config((frontend_t*)this, this->origwidth, this->origheight,
+  this->fe.fe_display_config((frontend_t*)this, -1, -1, this->origwidth, this->origheight,
 			     this->fullscreen ? 0 : 1, 
 			     this->vmode_switch, this->modeline, 
 			     this->aspect, this->scale_video, this->field_order);
