--- xineliboutput-1.0.5/Makefile.orig	2010-10-05 15:26:49.000000000 +0200
+++ xineliboutput-1.0.5/Makefile	2010-10-05 15:29:49.000000000 +0200
@@ -111,11 +111,11 @@
 ifeq ($(APPLE_DARWIN), 1)
     CXXFLAGS   ?= -O3 -pipe -Wall -Woverloaded-virtual -fPIC -g -fno-common -bundle -flat_namespace -undefined suppress
     CFLAGS     ?= -O3 -pipe -Wall -fPIC -g -fno-common -bundle -flat_namespace -undefined suppress
-    LDFLAGS_SO ?= -fvisibility=hidden
+    LDFLAGS_SO ?= $(LDFLAGS) -fvisibility=hidden
 else
     CXXFLAGS   ?= -O3 -pipe -Wall -Woverloaded-virtual -fPIC -g 
     CFLAGS     ?= -O3 -pipe -Wall -fPIC -g
-    LDFLAGS_SO ?= -shared -fvisibility=hidden 
+    LDFLAGS_SO ?= $(LDFLAGS) -shared -fvisibility=hidden 
 endif
 LIBS_VDR ?= 
 
@@ -482,7 +482,7 @@
 	@-rm -rf $(LIBDIR)/$(VDRPLUGIN_SXFE_SO).$(VERSION)
 	@cp $@ $(LIBDIR)/$(VDRPLUGIN_SXFE_SO).$(VERSION)
 $(VDRSXFE): $(OBJS_SXFE)
-	$(CC) -g $(OBJS_SXFE) $(LIBS_X11) -ljpeg $(LIBS_XINE) -o $@
+	$(CC) -g $(LDFLAGS) $(OBJS_SXFE) $(LIBS_X11) -ljpeg $(LIBS_XINE) -o $@
 endif
 
 ifeq ($(XINELIBOUTPUT_FB), 1)
@@ -491,7 +491,7 @@
 	@-rm -rf $(LIBDIR)/$(VDRPLUGIN_FBFE_SO).$(VERSION)
 	@cp $@ $(LIBDIR)/$(VDRPLUGIN_FBFE_SO).$(VERSION)
 $(VDRFBFE): $(OBJS_FBFE)
-	$(CC) -g $(OBJS_FBFE) $(LIBS_XINE) -ljpeg -o $@
+	$(CC) -g $(LDFLAGS) $(OBJS_FBFE) $(LIBS_XINE) -ljpeg -o $@
 endif
 
 ifeq ($(XINELIBOUTPUT_XINEPLUGIN), 1)
