Index: Makefile
===================================================================
RCS file: /cvsroot/xineliboutput/vdr-xineliboutput/Makefile,v
retrieving revision 1.106
diff -u -b -B -r1.106 Makefile
--- Makefile	23 Jan 2011 19:42:08 -0000	1.106
+++ Makefile	28 Feb 2011 21:16:10 -0000
@@ -323,7 +323,7 @@
 install : XINELIBOUTPUT_INSTALL_MSG =
 
 .PHONY: all
-all: config $(TARGETS_VDR) frontends i18n
+all: config $(TARGETS_VDR) frontends
 
 frontends: config $(TARGETS_FE) $(TARGETS_XINE)
 	$(XINELIBOUTPUT_INSTALL_MSG)
@@ -389,7 +389,7 @@
 # install
 #
 
-install: all
+install: all i18n
 ifeq ($(XINELIBOUTPUT_XINEPLUGIN), yes)
 	@mkdir -p $(DESTDIR)/$(XINEPLUGINDIR)/post
 	@echo Installing $(DESTDIR)/$(XINEPLUGINDIR)/$(XINEINPUTVDR)
