diff -ur wmmixer-1.5.orig/Makefile wmmixer-1.5/Makefile
--- wmmixer-1.5.orig/Makefile	2002-06-25 23:13:11.000000000 +0300
+++ wmmixer-1.5/Makefile	2008-12-23 00:34:28.000000000 +0200
@@ -1,19 +1,12 @@
 # $Id: wmmixer-1.5-respect_flags.patch,v 1.1 2008/12/22 22:51:03 ssuominen Exp $
 
-prefix      = /usr/local
+prefix      = /usr
 exec_prefix = ${prefix}
 bindir      = ${exec_prefix}/bin
 mandir      = ${prefix}/share/man
 
-DESTDIR     =
-
-CXX	    = g++
-CXXFLAGS    = -O -Wall
-EXTRA_LIBS  = -L/usr/X11R6/lib -lX11 -lXpm -lXext
-
-
-LD 	    = g++
-LDFLAGS     = -o $(EXECUTABLE) $(EXTRA_LIBDIRS) $(EXTRA_LIBS) $(CXXFLAGS)
+CXX	    ?= g++
+LIBS	    = -lX11 -lXpm -lXext
 
 EXECUTABLE  = wmmixer
 OBJS	    = xhandler.o mixctl.o wmmixer.o exception.o
@@ -26,7 +19,7 @@
 	$(CXX) $(CXXFLAGS) -c -o $@ $<
 
 $(EXECUTABLE): $(OBJS)
-	$(LD) $(LDFLAGS) $(OBJS)
+	$(CXX) $(LDFLAGS) $(OBJS) $(LIBS) -o $(EXECUTABLE)
 
 all: $(EXECUTABLE)
 
