--- gtkatlantic-0.4.3.orig/configure.in
+++ gtkatlantic-0.4.3/configure.in
@@ -48,15 +48,7 @@
 dnl fi
 
 dnl Set PACKAGE_DATA_DIR in config.h.
-if test "x${datarootdir}" = 'x${prefix}/share'; then
-  if test "x${prefix}" = "xNONE"; then
-    AC_DEFINE_UNQUOTED(PACKAGE_DATA_DIR, "${ac_default_prefix}/share/${PACKAGE}", [Path where data will be installed])
-  else
-    AC_DEFINE_UNQUOTED(PACKAGE_DATA_DIR, "${prefix}/share/${PACKAGE}", [Path where data will be installed])
-  fi
-else
-  AC_DEFINE_UNQUOTED(PACKAGE_DATA_DIR, "${datadir}/${PACKAGE}", [Path where data will be installed])
-fi
+AC_DEFINE_UNQUOTED(PACKAGE_DATA_DIR, "${datadir}/${PACKAGE}", [Path where data will be installed])
 
 dnl Set PACKAGE SOURCE DIR in config.h.
 packagesrcdir=`cd $srcdir && pwd`
@@ -67,10 +59,6 @@
 #       CFLAGS="-g";  \ 
 # fi
 
-if test "x$GCC" = "xyes"; then  \
-      CFLAGS="-Wall -Werror -ggdb";  \ 
-fi
-
 AC_OUTPUT([
 Makefile
 src/Makefile
