--- tnt-2.6-orig/tnt.el
+++ tnt-2.6/tnt.el
@@ -4132,7 +4132,7 @@
        (tnt-sound-exec
         (with-output-to-string
           (call-process shell-file-name nil t nil shell-command-switch (concat tnt-sound-exec " " tnt-sound-exec-args " " sound-file))))
-       (tnt-running-xemacs (play-sound-file sound-file))
+       ((fboundp 'play-sound-file) (play-sound-file sound-file))
        (t (message "Warning: tnt-sound-exec is not set")))
   (message "Warning: %s is not a readable file" sound-file)))
 
