diff -Naur msgpack-0.4.2.orig//cpp/Makefile.am msgpack-0.4.2//cpp/Makefile.am
--- msgpack-0.4.2.orig//cpp/Makefile.am	2010-02-18 14:31:42.000000000 +0900
+++ msgpack-0.4.2//cpp/Makefile.am	2010-03-02 01:43:12.000000000 +0900
@@ -39,6 +39,6 @@
 
 msgpack_test_SOURCES = test.cpp
 msgpack_test_CXXFLAGS = -I$(top_srcdir) -I$(top_srcdir)/c -I$(top_srcdir)/cpp
-msgpack_test_LDADD = libmsgpack.la -lgtest_main
+msgpack_test_LDADD = -L../c -lmsgpackc libmsgpack.la -lgtest_main
 
 TESTS = $(check_PROGRAMS)
