diff -ur clisp-2.38.orig/modules/fastcgi/Makefile.in clisp-2.38/modules/fastcgi/Makefile.in
--- clisp-2.38.orig/modules/fastcgi/Makefile.in	2004-08-12 09:38:36.000000000 -0500
+++ clisp-2.38/modules/fastcgi/Makefile.in	2006-04-24 15:14:38.000000000 -0500
@@ -23,10 +23,10 @@
 	$(CLISP) -c fastcgi.lisp
 
 fastcgi.o:       fastcgi.c
-	$(CC) $(CPPFLAGS) $(CFLAGS) $(INCLUDES) -I.. -c fastcgi.c
+	$(CC) $(CPPFLAGS) $(CFLAGS) -I$(INCLUDES) -I.. -c fastcgi.c
 
 fastcgi_wrappers.o:       fastcgi_wrappers.c
-	$(CC) $(CPPFLAGS) $(CFLAGS) $(INCLUDES) -I.. -c fastcgi_wrappers.c
+	$(CC) $(CPPFLAGS) $(CFLAGS) -I$(INCLUDES) -I.. -c fastcgi_wrappers.c
 
 # Make a module
 clisp-module : all
