diff -urN qtrans-0.2.2.6.old/src/plugins/stardict/stardict.cpp qtrans-0.2.2.6/src/plugins/stardict/stardict.cpp
--- qtrans-0.2.2.6.old/src/plugins/stardict/stardict.cpp	2011-04-28 23:08:08.000000000 +0200
+++ qtrans-0.2.2.6/src/plugins/stardict/stardict.cpp	2011-04-28 23:11:33.000000000 +0200
@@ -1123,7 +1123,7 @@
 
 QString StarDict::parseData(const char *data)
 {
-
+	return "Not implemented";
 }
 
 
