--- foosball/foosball.hxx
+++ foosball/foosball.hxx
@@ -15,9 +15,10 @@
  *                                                                         *
  ***************************************************************************/
 
- #include <stdio.h>
- #include <stdlib.h>
- #include <math.h>
+ #include <cstdio>
+ #include <cstdlib>
+ #include <cmath>
+ #include <cstring>
  #include "sgw.hxx"
  
  #define PLAYER1    (0)                         // Left to Right Player
