diff -ruN pyaim-t-0.8a/src/main.py pyaim-t-0.8a-fixed/src/main.py
--- pyaim-t-0.8a/src/main.py	2007-03-26 04:23:32.000000000 +0200
+++ pyaim-t-0.8a-fixed/src/main.py	2007-06-04 22:09:34.569889294 +0200
@@ -89,7 +89,10 @@
 	import signal
 	signal.signal(signal.SIGHUP, reloadConfig)
 	# Load scripts for PID and daemonizing
-	from twisted.scripts import twistd
+	try:
+		from twisted.scripts import _twistd_unix as twistd
+	except:
+		from twisted.scripts import twistd
 
 selectWarning = "Unable to install any good reactors (kqueue, cf, epoll, poll).\nWe fell back to using select. You may have scalability problems.\nThis reactor will not support more than 1024 connections +at a time.  You may silence this message by choosing 'select' or 'default' as your reactor in the transport config."
 if config.reactor and len(config.reactor) > 0:
