From 09cfb7f14f3766c80137a00586579dd2280ffe65 Mon Sep 17 00:00:00 2001
From: Alexandre Rostovtsev <tetromino@gentoo.org>
Date: Fri, 10 Aug 2012 14:34:03 -0400
Subject: [PATCH] Make vala bindings non-automagic

---
 configure.ac |    8 ++++++--
 1 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index b51b7d2..e174c4b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -134,7 +134,9 @@ PKG_CHECK_MODULES(LCMS, lcms2 >= 2.2)
 dnl ---------------------------------------------------------------------------
 dnl - Build VALA support
 dnl ---------------------------------------------------------------------------
-if test x$found_introspection != xno; then
+AC_ARG_ENABLE(vala, AS_HELP_STRING([--enable-vala],[build vala bindings]),
+	      enable_vala=$enableval,enable_vala=no)
+if test x$found_introspection != xno -a x$enable_vala != xno; then
 	AC_PATH_PROG([VAPIGEN], [vapigen], [])
 	if test "x$VAPIGEN" = "x"; then
 		has_vapigen="no"
@@ -144,7 +146,9 @@ if test x$found_introspection != xno; then
 	fi
 else
 	has_vapigen="no"
-	AC_MSG_WARN([Not introspection found, will not build Vala binding])
+	if test x$found_introspection = xno; then
+		AC_MSG_WARN([introspection not found, will not build Vala binding])
+	fi
 fi
 AM_CONDITIONAL(HAVE_VAPIGEN, [test "x$has_vapigen" = "xyes"])
 
-- 
1.7.8.6

