From 5ec05b3d7cec24c73c68bad4608a4ce15da0423f Mon Sep 17 00:00:00 2001
From: Glenn Randers-Pehrson <glennrp at users.sourceforge.net>
Date: Sat, 13 Apr 2013 20:01:09 -0500
Subject: [PATCH 3/5] [libpng16] Removed extra recently-inserted line from
 pngwutil.c

---
 pngwutil.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/pngwutil.c b/pngwutil.c
index 755ccef..f481683 100644
--- a/pngwutil.c
+++ b/pngwutil.c
@@ -1738,7 +1738,6 @@ png_write_iTXt(png_structrp png_ptr, int compression, png_const_charp key,
    {
       if (png_text_compress(png_ptr, png_iTXt, &comp, prefix_len) != Z_OK)
          png_error(png_ptr, png_ptr->zstream.msg);
-      png_write_chunk_header(png_ptr, png_iTXt, comp.output_len + prefix_len);
    }
 
    else
-- 
1.8.1.5

