https://sourceforge.net/p/gnuplot/feature-requests/399/
Index: configure.in
===================================================================
RCS file: /cvsroot/gnuplot/gnuplot/configure.in,v
retrieving revision 1.370
diff -u -r1.370 configure.in
--- configure.in	20 Jun 2014 22:58:46 -0000	1.370
+++ configure.in	21 Jul 2014 23:33:43 -0000
@@ -80,19 +80,25 @@
 AC_FUNC_FSEEKO
 AC_SYS_LARGEFILE
 
-dnl The libcerf library contains a set of complex-valued special functions
-dnl related to the error function (cerf, erfi, Dawson's integral, Voigt profile).
-dnl If libcerf is installed then we can use them.
-PKG_CHECK_MODULES_NOFAIL([LIBCERF], [libcerf],
-    [
-    CPPFLAGS="$CPPFLAGS $LIBCERF_CFLAGS"
-    LDFLAGS="$LDFLAGS $LIBCERF_LIBS"
-    ],
-)
-AC_SEARCH_LIBS([cdawson], [cerf],
-    [AC_DEFINE([HAVE_LIBCERF], 1, [define if you have libcerf])
-    have_libcerf=yes]
-)
+AC_ARG_WITH(libcerf,dnl
+[  --without-libcerf      build without special functions from libcerf (default enabled)],,
+  [test -z "${with_libcerf}" && with_libcerf=yes])
+
+if test "${with_libcerf}" = yes ; then
+  dnl The libcerf library contains a set of complex-valued special functions
+  dnl related to the error function (cerf, erfi, Dawson's integral, Voigt profile).
+  dnl If libcerf is installed then we can use them.
+  PKG_CHECK_MODULES_NOFAIL([LIBCERF], [libcerf],
+      [
+      CPPFLAGS="$CPPFLAGS $LIBCERF_CFLAGS"
+      LDFLAGS="$LDFLAGS $LIBCERF_LIBS"
+      ],
+  )
+  AC_SEARCH_LIBS([cdawson], [cerf],
+      [AC_DEFINE([HAVE_LIBCERF], 1, [define if you have libcerf])
+      have_libcerf=yes]
+  )
+fi
 
 dnl Various programs
 dnl X/Emacs for building .texi version of docs
