diff -Nur RandomGUID/RandomGUIDdemo.java RandomGUID_patched/RandomGUIDdemo.java
--- RandomGUID/RandomGUIDdemo.java	2002-01-30 04:19:53.000000000 +0200
+++ RandomGUID_patched/RandomGUIDdemo.java	2008-09-26 00:06:43.000000000 +0300
@@ -4,7 +4,7 @@
     // Generate 20 of 'em!
     public static void main(String[] args) {
 	for(int i=1; i<=20; i++) {
-	    RandomGUID myguid = new RandomGUID(false);
+	    com.javaexchange.RandomGUID myguid = new com.javaexchange.RandomGUID(false);
 	    System.out.println(i + " " + myguid.toString());
 	}
     }
