--- build/src/mn-evolution-server.c.~1~	2008-05-22 17:47:48.000000000 +0200
+++ build/src/mn-evolution-server.c	2009-03-30 20:02:17.936196624 +0200
@@ -595,7 +595,8 @@
 
 	for (i = 0; i < summary->len; i++)
 	  {
-	    CamelMessageInfo *info = summary->pdata[i];
+		char *uid = summary->pdata[i];
+	    CamelMessageInfo *info = camel_folder_get_message_info(folder, uid);
 
 	    if ((camel_message_info_flags(info) & CAMEL_MESSAGE_SEEN) == 0)
 	      g_ptr_array_add(*ret, self_camel_message_info_to_dbus_struct(info));
