Index: quilt-el/quilt.el
===================================================================
--- quilt-el.orig/quilt.el
+++ quilt-el/quilt.el
@@ -25,7 +25,7 @@
 
 (defun quilt-find-dir (fn)
   "find the top level dir for quilt from fn"
-  (if (or (not fn) (equal fn "/"))
+  (if (or (not fn) (equal fn (file-name-directory fn)))
       nil
     (let ((d (file-name-directory fn)))
       (if (file-accessible-directory-p (concat d "/.pc"))
