diff --git a/configure.ac b/configure.ac
index 1beab82..243b693 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2,7 +2,7 @@ dnl intro
 AC_INIT(pdns/receiver.cc)
 AM_INIT_AUTOMAKE(pdns, 3.2)
 AC_CANONICAL_HOST
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS([config.h])
 AC_C_BIGENDIAN 
 AC_PREREQ(2.52)
 : ${CXXFLAGS="-Wall -O2"}
@@ -180,7 +180,7 @@ AC_ARG_WITH([system-polarssl],
  [system_polarssl=$withval],
  [system_polarssl=yes])
 AC_MSG_RESULT($system_polarssl)
-AM_CONDITIONAL(HAVE_LIBPOLARSSL, false)
+AM_CONDITIONAL(HAVE_LIBPOLARSSL, test x$system_polarssl = xyes)
 if test x$system_polarssl = xyes; then
  AC_MSG_CHECKING([PolarSSL version >= 1.1])
  AC_COMPILE_IFELSE(
