--- a/modules/luabackend/Makefile.am	2013-12-10 11:57:48.597065748 +0100
+++ b/modules/luabackend/Makefile.am	2013-12-10 12:00:05.855127730 +0100
@@ -1,4 +1,4 @@
-AM_CPPFLAGS=-I/usr/include/lua5.1 @THREADFLAGS@
+AM_CPPFLAGS=$(LUA_CFLAGS) @THREADFLAGS@
 #AM_CPPFLAGS=-I/usr/local/include/luajit-2.0 -DUSE_LUAJIT @THREADFLAGS@
 EXTRA_DIST=OBJECTFILES OBJECTLIBS 
 
@@ -8,5 +8,6 @@
 libluabackend_la_SOURCES=luabackend.cc luabackend.hh minimal.cc reload.cc lua_functions.cc master.cc private.cc slave.cc supermaster.cc dnssec.cc \
 	lua_functions.hh 
 
-libluabackend_la_LDFLAGS=-module -avoid-version -llua5.1
+libluabackend_la_LDFLAGS=-module -avoid-version
+libluabackend_la_LIBADD=$(LUA_LIBS)
 #-lluajit-5.1
