--- data/CMakeLists.txt.orig	2012-10-29 16:15:27.836112518 +0800
+++ data/CMakeLists.txt	2012-10-29 16:15:57.249433574 +0800
@@ -46,7 +46,6 @@
 add_custom_target(pinyin_data ALL DEPENDS ${PY_DATA})
 
 add_custom_command(OUTPUT ${PY_ORGDATA}
-                   COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/download.sh "${WGET}" "${PY_URL}" "${PY_TAR}"
                    COMMAND ${TAR} xzmvf ${PY_TAR})
 add_custom_command(OUTPUT ${PY_DATA}
                    DEPENDS ${PY_ORGDATA}
@@ -80,8 +79,6 @@
 add_custom_target(spell_en_dict ALL DEPENDS ${SPELL_EN_DICT})
 add_custom_command(
   OUTPUT ${SPELL_EN_DICT_SRC}
-  COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/download.sh "${WGET}"
-  "${SPELL_EN_DICT_URL}" "${SPELL_EN_DICT_TAR}"
   COMMAND "${TAR}" xzmvf "${SPELL_EN_DICT_TAR}")
 
 set(COMP_SPELL_DICT
