diff -Naur fcitx-3.6.3.orig/lib/Makefile.am fcitx-3.6.3/lib/Makefile.am
--- fcitx-3.6.3.orig/lib/Makefile.am	2004-01-31 05:43:30.000000000 +0900
+++ fcitx-3.6.3/lib/Makefile.am	2010-02-14 13:22:28.000000000 +0900
@@ -1,5 +1,5 @@
 noinst_LIBRARIES = libXimd.a
-AM_CFLAGS = -c -O2 -fno-strength-reduce -fno-strict-aliasing
+AM_CFLAGS = -fno-strength-reduce -fno-strict-aliasing
 libXimd_a_SOURCES = \
   FrameMgr.c FrameMgr.h i18nAttr.c i18nClbk.c i18nIc.c \
   i18nIMProto.c i18nMethod.c i18nPtHdr.c i18nUtil.c i18nX.c \
diff -Naur fcitx-3.6.3.orig/src/Makefile.am fcitx-3.6.3/src/Makefile.am
--- fcitx-3.6.3.orig/src/Makefile.am	2010-01-16 15:17:12.000000000 +0900
+++ fcitx-3.6.3/src/Makefile.am	2010-02-14 13:23:28.000000000 +0900
@@ -1,7 +1,7 @@
 DEFS = -DPKGDATADIR=\"$(pkgdatadir)\"
 
 INCLUDES = -I. -I$(top_srcdir)/lib -I$(top_srcdir)/xpm
-AM_CFLAGS = -O2 -fno-strength-reduce
+AM_CFLAGS = -fno-strength-reduce
 
 
 bin_PROGRAMS = fcitx
@@ -36,5 +36,4 @@
 		internalVersion.c \
 		ImeRemote.c
 		
-fcitx_LDADD = ../lib/libXimd.a
-fcitx_LDFLAGS = -lXpm -lXtst -lpthread
+fcitx_LDADD = ../lib/libXimd.a -lXpm -lXtst -lXext -lpthread
