--- Rakefile.orig	2010-06-24 09:07:45.490965359 +0200
+++ Rakefile	2010-06-24 09:08:01.968715044 +0200
@@ -344,10 +344,8 @@
   system("gtk-update-icon-cache -f -t /usr/share/icons/hicolor") # HACK
 end
 
-task :post_install => [:scrollkeeper, :gconf, :update_icon_cache]
-
 desc "Install Alexandria"
-task :install => [:pre_install, :install_package, :post_install]
+task :install => [:pre_install, :install_package]
 
 desc "Uninstall Alexandria"
 task :uninstall => [:uninstall_package] # TODO gconf etc...
