--- inventor/lib/nodekits/include/Inventor/nodekits/SoSubKit.h.orig	2008-12-21 10:34:25.000000000 +0100
+++ inventor/lib/nodekits/include/Inventor/nodekits/SoSubKit.h	2008-12-21 10:34:33.000000000 +0100
@@ -192,7 +192,7 @@
 //
 
 #define SO_KIT_SOURCE(className)	                                      \
-    SO_NODE_SOURCE(className);	                                              \
+    SO_NODE_SOURCE(className)	                                              \
     SO__KIT_CATALOG_VARS(className);	                                      \
     SO__KIT_CATALOG_METHODS(className)
 
