--- ./inventor/lib/database/include/Inventor/SoPath.h.old	2005-09-15 22:19:05.000000000 +0000
+++ ./inventor/lib/database/include/Inventor/SoPath.h	2005-09-15 22:19:08.000000000 +0000
@@ -61,6 +61,8 @@
 #include <Inventor/nodes/SoNode.h>
 #include <Inventor/SoLists.h>
 
+class SoTempPath;
+
 //////////////////////////////////////////////////////////////////////////////
 //
 //  Class: SoPath
diff -urN ./inventor-old/lib/database/src/so/SoOutput.c++ ./inventor/lib/database/src/so/SoOutput.c++
--- ./inventor-old/lib/database/src/so/SoOutput.c++	2001-09-25 00:45:35.000000000 +0000
+++ ./inventor/lib/database/src/so/SoOutput.c++	2005-09-15 22:57:00.000000000 +0000
@@ -1462,11 +1462,7 @@
 
     // Generates a CC warning. Ho hum.
     if (refDict->find((unsigned long) base, ref))
-#if (_MIPS_SZPTR == 64 || __ia64)
         referenceId = (int) ((unsigned long) ref);
-#else
-	referenceId = (int)ref;
-#endif
 
     return referenceId;
 }
diff -urN ./inventor-old/lib/database/src/so/SoType.c++ ./inventor/lib/database/src/so/SoType.c++
--- ./inventor-old/lib/database/src/so/SoType.c++	2003-07-11 22:40:53.000000000 +0000
+++ ./inventor/lib/database/src/so/SoType.c++	2005-09-15 22:57:06.000000000 +0000
@@ -331,11 +331,7 @@
     if (b == NULL)
 	return SoType::badType();
 
-#if (_MIPS_SZPTR == 64 || __ia64)
     SoType result = typeData[(int) ((unsigned long) b)].type;
-#else
-    SoType result = typeData[(int)b].type;
-#endif
 
     if (result.storage.isPublic == 0) {
 #ifdef DEBUG
diff -urN ./inventor-old/lib/interaction/src/draggers/SoDragger.c++ ./inventor/lib/interaction/src/draggers/SoDragger.c++
--- ./inventor-old/lib/interaction/src/draggers/SoDragger.c++	2001-09-25 00:45:44.000000000 +0000
+++ ./inventor/lib/interaction/src/draggers/SoDragger.c++	2005-09-15 22:58:56.000000000 +0000
@@ -686,11 +686,7 @@
 		int indexInPath = tempPathToThis->getIndex(numFmHead+1);
 
 		int numKidsNow    = children->getLength();
-#if (_MIPS_SZPTR == 64 || __ia64)
 		int numKidsBefore = (int) ((long) (*tempPathNumKidsHack)[numFmHead]);
-#else
-		int numKidsBefore = (int) (*tempPathNumKidsHack)[numFmHead];
-#endif
 
 		// To be correct, the childNode has to be the correct numbered
 		// child under the parent, and the parent should still
diff -urN ./inventor-old/lib/nodekits/src/nodekits/SoNkCatalog.c++ ./inventor/lib/nodekits/src/nodekits/SoNkCatalog.c++
--- ./inventor-old/lib/nodekits/src/nodekits/SoNkCatalog.c++	2001-09-25 00:45:46.000000000 +0000
+++ ./inventor/lib/nodekits/src/nodekits/SoNkCatalog.c++	2005-09-15 22:59:09.000000000 +0000
@@ -336,11 +336,7 @@
     void *castPNum;
 
     if ( partNameDict.find( (unsigned long) theName.getString(), castPNum ) )
-#if (_MIPS_SZPTR == 64 || __ia64)
 	return ( (int) ((long) castPNum) );  // System long
-#else
-	return ( (int) castPNum );
-#endif
     else 
 	return SO_CATALOG_NAME_NOT_FOUND;
 }
diff -urN ./inventor-old/lib/nodekits/src/upgraders/SoV1NkCatalog.c++ ./inventor/lib/nodekits/src/upgraders/SoV1NkCatalog.c++
--- ./inventor-old/lib/nodekits/src/upgraders/SoV1NkCatalog.c++	2001-09-25 00:45:47.000000000 +0000
+++ ./inventor/lib/nodekits/src/upgraders/SoV1NkCatalog.c++	2005-09-15 22:59:03.000000000 +0000
@@ -330,11 +330,7 @@
     void *castPNum;
 
     if ( partNameDict.find( (unsigned long) theName.getString(), castPNum ) )
-#if (_MIPS_SZPTR == 64 || __ia64)
 	return ( (int) ((long) castPNum) );  // System long
-#else
-	return ( (int) castPNum );
-#endif
     else 
 	return SO_V1_CATALOG_NAME_NOT_FOUND;
 }
diff -urN ./inventor-old/libSoXt/src/SoXtRsrc.c++ ./inventor/libSoXt/src/SoXtRsrc.c++
--- ./inventor-old/libSoXt/src/SoXtRsrc.c++	2001-09-25 00:45:48.000000000 +0000
+++ ./inventor/libSoXt/src/SoXtRsrc.c++	2005-09-15 22:59:19.000000000 +0000
@@ -145,13 +145,8 @@
     classList = new XrmQuark[listSize];
     for (q = 0, s = len - 1;
     	 s >= 0; q++, s--) {
-#if (_MIPS_SZPTR == 64 || __ia64)
 	 nameList[q]  = (XrmQuark) ((long) nameplist[s]);
 	 classList[q] = (XrmQuark) ((long) classplist[s]);
-#else
-	 nameList[q]  = (XrmQuark) nameplist[s];
-	 classList[q] = (XrmQuark) classplist[s];
-#endif
     }
 
     // make the last entry the NULL sentinel
--- ./inventor-old/apps/demos/drop/drop.c++	2008-06-06 10:44:36.000000000 +0100
+++ ./inventor/apps/demos/drop/drop.c++	2008-06-04 15:22:05.000000000 +0100
@@ -93,5 +93,6 @@
     // Enter the Xt event loop
     SoXt::show( appWindow );
     SoXt::mainLoop();
+    return 0;
 }
 
--- ./inventor-old/apps/demos/maze/maze.c++	2008-06-06 10:44:36.000000000 +0100
+++ ./inventor/apps/demos/maze/maze.c++	2008-06-04 15:22:05.000000000 +0100
@@ -1332,5 +1332,6 @@
 
     SoXt::show(mainWindow);
     SoXt::mainLoop();
+    return 0;
 }
 
--- ./inventor-old/apps/demos/noodle/noodle.c++	2008-06-06 10:44:36.000000000 +0100
+++ ./inventor/apps/demos/noodle/noodle.c++	2008-06-04 15:22:05.000000000 +0100
@@ -679,5 +679,6 @@
     SoXt::show(mainWindow);
 
     SoXt::mainLoop();
+    return 0;
 }
 
--- ./inventor-old/apps/demos/SceneViewer/SceneViewer.c++	2008-06-06 10:44:36.000000000 +0100
+++ ./inventor/apps/demos/SceneViewer/SceneViewer.c++	2008-06-04 15:22:04.000000000 +0100
@@ -230,4 +230,5 @@
     // Loop forever
     //
     SoXt::mainLoop();
+    return 0;
 }
--- ./inventor-old/apps/samples/widgets/componentTest.c++	2008-06-06 10:44:36.000000000 +0100
+++ ./inventor/apps/samples/widgets/componentTest.c++	2008-06-04 15:22:05.000000000 +0100
@@ -60,7 +60,7 @@
 }
 
 int
-main(unsigned int argc, char *argv[])
+main(int argc, char *argv[])
 {
     if (argc != 2)
 	usage(argv[0]);
@@ -85,4 +85,5 @@
     XtRealizeWidget(mainWindow);
     
     SoXt::mainLoop();
+    return 0;
 }
--- ./inventor-old/doc/man/ivman/ClassDef.c++	2008-06-06 10:44:38.000000000 +0100
+++ ./inventor/doc/man/ivman/ClassDef.c++	2008-06-04 15:22:05.000000000 +0100
@@ -237,7 +237,7 @@
 
 #define NUM_KEY_WORDS (sizeof(keywords) / sizeof(keywords[0]))
 
-    int	i;
+    size_t i;
 
     if (names == NULL) {
 	names = new SbName[NUM_KEY_WORDS];
--- ./inventor-old/doc/man/ivman/Main.c++	2008-06-06 10:44:38.000000000 +0100
+++ ./inventor/doc/man/ivman/Main.c++	2008-06-04 15:22:05.000000000 +0100
@@ -89,7 +89,7 @@
 //
 /////////////////////////////////////////////////////////////////////////////
 
-main(int argc, char *argv[])
+int main(int argc, char *argv[])
 {
     SbBool	ok = TRUE;
     int		c, curArg;
--- ./inventor-old/doc/man/ivman/Man.c++	2008-06-06 10:44:38.000000000 +0100
+++ ./inventor/doc/man/ivman/Man.c++	2008-06-04 15:22:05.000000000 +0100
@@ -806,7 +806,7 @@
 
 #define NUM_OP_NAMES (sizeof(opNames) / sizeof(opNames[0]))
 
-		int op;
+		size_t op;
 		for (op = 0; op < NUM_OP_NAMES; op++) {
 		    if (name == opNames[op][0]) {
 			name = opNames[op][1];
--- ./inventor-old/doc/man/ivman/Man.h	2008-06-06 10:44:38.000000000 +0100
+++ ./inventor/doc/man/ivman/Man.h	2008-06-04 15:22:05.000000000 +0100
@@ -131,6 +131,7 @@
     virtual void	write(SbBool runIn) = 0;
     virtual SbBool	read()  = 0;
     virtual int		getNumPreTabChars() const;
+    virtual             ~Item() {}
   protected:
     Item();
   private:
