diff -u -ur inventor/doc/man/man1/GNUmakefile /var/tmp/portage/openinventor-2.1.5.10/work/inventor/doc/man/man1/GNUmakefile
--- a/inventor/doc/man/man1/GNUmakefile	2000-09-18 11:11:13.000000000 +0200
+++ b/inventor/doc/man/man1/GNUmakefile	2005-09-14 15:35:20.000000000 +0200
@@ -11,6 +11,8 @@
 	ivinfo.1 \
 	ivview.1
 
-all install: install_man1
+all:
+
+install: install_man1
 
 include $(IVCOMMONRULES)
diff -u -ur inventor/doc/man/man3/GNUmakefile /var/tmp/portage/openinventor-2.1.5.10/work/inventor/doc/man/man3/GNUmakefile
--- a/inventor/doc/man/man3/GNUmakefile	2000-09-18 11:11:13.000000000 +0200
+++ b/inventor/doc/man/man3/GNUmakefile	2005-09-14 15:35:37.000000000 +0200
@@ -3,7 +3,9 @@
 
 SUBDIRS = ivm
 
-all install:
+all:
+
+install:
 	$(SUBDIRS_MAKERULE)
 
 COMMONPREF = xxx
diff -u -ur inventor/make/commondefs /var/tmp/portage/openinventor-2.1.5.10/work/inventor/make/commondefs
--- a/inventor/make/commondefs	2001-06-16 01:27:00.000000000 +0200
+++ b/inventor/make/commondefs	2005-09-14 15:27:17.000000000 +0200
@@ -65,7 +65,7 @@
 STRIP	= $(TOOLROOT)/usr/bin/strip
 TAG	= $(TOOLROOT)/usr/sbin/tag
 TLINK	= $(TOOLROOT)/usr/sbin/tlink
-YACC    = $(TOOLROOTSAFE) $(TOOLROOT)/usr/bin/yacc
+YACC    = $(TOOLROOTSAFE) $(TOOLROOT)/usr/bin/byacc
 ifeq ($(usingIrix), 1)
 YACC	+= -P $(ROOT)/usr/lib/yaccpar
 endif
