--- configure.old	2009-02-10 17:14:47.000000000 -0500
+++ configure	2009-02-10 17:14:18.000000000 -0500
@@ -3979,11 +3979,6 @@
 fi
 
 
-if test "x$pgsql" != "xno" -o "x$mysql" != "xno"; then
-	SOURCES="$SOURCES sqlauth.cxx sqlacct.cxx gksql.cxx"
-	HEADERS="$HEADERS sqlacct.h gksql.h"
-fi
-
 # Check whether --enable-firebird or --disable-firebird was given.
 if test "${enable_firebird+set}" = set; then
   enableval="$enable_firebird"
@@ -4242,6 +4237,11 @@
 fi
 
 
+if test "x$pgsql" != "xno" -o "x$mysql" != "xno" -o "x$firebird" != "xno"; then
+	SOURCES="$SOURCES sqlauth.cxx sqlacct.cxx gksql.cxx"
+	HEADERS="$HEADERS sqlacct.h gksql.h"
+fi
+
 
 GNUGK_MAJOR_VERSION=`cat version.h | grep "define GNUGK_MAJOR_VERSION" | cut -f4 -d' '`
 GNUGK_MINOR_VERSION=`cat version.h | grep "define GNUGK_MINOR_VERSION" | cut -f4 -d' '`
