--- gksql_firebird.cxx.old	2009-02-11 01:31:53.000000000 +0100
+++ gksql_firebird.cxx	2009-02-11 01:31:59.000000000 +0100
@@ -370,7 +370,7 @@
 			char errormsg[512];
 			if (errcode == -999) {
 				errcode = status[1];
-				long *pvector = status;
+				ISC_STATUS *pvector = status;
 				// TODO: replace all isc_interprete() with fb_interpret()
 				errormsg[isc_interprete(errormsg, &pvector)] = 0;
 			} else {
@@ -417,7 +417,7 @@
 			char errormsg[512];
 			if (errcode == -999) {
 				errcode = status[1];
-				long *pvector = status;
+				ISC_STATUS *pvector = status;
 				errormsg[isc_interprete(errormsg, &pvector)] = 0;
 			} else {
 				strcpy(errormsg, "SQL:");
@@ -497,7 +497,7 @@
 	
 	isc_attach_database(status, 0, const_cast<char*>(dbname.c_str()), &conn, dpb_offset, &(dpb[0]));
 	if (status[0] == 1 && status[1] != 0) {
-		long *pvector = status;
+		ISC_STATUS *pvector = status;
 		char errormsg[512];
 		errormsg[isc_interprete(errormsg, &pvector)] = 0;
 		PTRACE(2, GetName() << "\tFirebird connection to " << m_username << '@' << dbname 
@@ -530,7 +530,7 @@
 	
 	isc_start_transaction(status, &tr, 1, &conn, 0, NULL);
 	if (status[0] == 1 && status[1] != 0) {
-		long *pvector = status;
+		ISC_STATUS *pvector = status;
 		char errormsg[512];
 		errormsg[isc_interprete(errormsg, &pvector)] = 0;
 		return new GkIBSQLResult(status[1], errormsg);
@@ -541,7 +541,7 @@
 		long errorcode = isc_sqlcode(status);
 		if (errorcode == -999) {
 			errorcode = status[1];
-			long *pvector = status;
+			ISC_STATUS *pvector = status;
 			errormsg[isc_interprete(errormsg, &pvector)] = 0;
 		} else {
 			strcpy(errormsg, "SQL:");
@@ -562,7 +562,7 @@
 		long errorcode = isc_sqlcode(status);
 		if (errorcode == -999) {
 			errorcode = status[1];
-			long *pvector = status;
+			ISC_STATUS *pvector = status;
 			errormsg[isc_interprete(errormsg, &pvector)] = 0;
 		} else {
 			strcpy(errormsg, "SQL:");
@@ -585,7 +585,7 @@
 			long errorcode = isc_sqlcode(status);
 			if (errorcode == -999) {
 				errorcode = status[1];
-				long *pvector = status;
+				ISC_STATUS *pvector = status;
 				errormsg[isc_interprete(errormsg, &pvector)] = 0;
 			} else {
 				strcpy(errormsg, "SQL:");
@@ -603,7 +603,7 @@
 		long errorcode = isc_sqlcode(status);
 		if (errorcode == -999) {
 			errorcode = status[1];
-			long *pvector = status;
+			ISC_STATUS *pvector = status;
 			errormsg[isc_interprete(errormsg, &pvector)] = 0;
 		} else {
 			strcpy(errormsg, "SQL:");
