--- gksql_firebird.cxx.old	2009-02-10 21:43:26.000000000 +0100
+++ gksql_firebird.cxx	2009-02-10 21:58:06.000000000 +0100
@@ -342,11 +342,11 @@
 	if (status[0] == 1 && status[1] != 0) {
 		m_numRows = m_sqlRow;
 		if (retval != 100) {
-			long errcode = isc_sqlcode(status);
+			ISC_STATUS errcode = isc_sqlcode(status);
 			char errormsg[512];
 			if (errcode == -999) {
 				errcode = status[1];
-				long *pvector = status;
+				ISC_STATUS *pvector = status;
 				errormsg[isc_interprete(errormsg, &pvector)] = 0;
 			} else {
 				strcpy(errormsg, "SQL:");
@@ -392,7 +392,7 @@
 			char errormsg[512];
 			if (errcode == -999) {
 				errcode = status[1];
-				long *pvector = status;
+				ISC_STATUS *pvector = status;
 				errormsg[isc_interprete(errormsg, &pvector)] = 0;
 			} else {
 				strcpy(errormsg, "SQL:");
@@ -472,7 +472,7 @@
 	
 	isc_attach_database(status, 0, const_cast<char*>(dbname.c_str()), &conn, dpb_offset, &(dpb[0]));
 	if (status[0] == 1 && status[1] != 0) {
-		long *pvector = status;
+		ISC_STATUS *pvector = status;
 		char errormsg[512];
 		errormsg[isc_interprete(errormsg, &pvector)] = 0;
 		PTRACE(2, GetName() << "\tFirebird connection to " << m_username << '@' << dbname 
@@ -505,7 +505,7 @@
 	
 	isc_start_transaction(status, &tr, 1, &conn, 0, NULL);
 	if (status[0] == 1 && status[1] != 0) {
-		long *pvector = status;
+		ISC_STATUS *pvector = status;
 		char errormsg[512];
 		errormsg[isc_interprete(errormsg, &pvector)] = 0;
 		return new GkIBSQLResult(status[1], errormsg);
@@ -516,7 +516,7 @@
 		long errorcode = isc_sqlcode(status);
 		if (errorcode == -999) {
 			errorcode = status[1];
-			long *pvector = status;
+			ISC_STATUS *pvector = status;
 			errormsg[isc_interprete(errormsg, &pvector)] = 0;
 		} else {
 			strcpy(errormsg, "SQL:");
@@ -536,7 +536,7 @@
 		long errorcode = isc_sqlcode(status);
 		if (errorcode == -999) {
 			errorcode = status[1];
-			long *pvector = status;
+			ISC_STATUS *pvector = status;
 			errormsg[isc_interprete(errormsg, &pvector)] = 0;
 		} else {
 			strcpy(errormsg, "SQL:");
@@ -558,7 +558,7 @@
 			long errorcode = isc_sqlcode(status);
 			if (errorcode == -999) {
 				errorcode = status[1];
-				long *pvector = status;
+				ISC_STATUS *pvector = status;
 				errormsg[isc_interprete(errormsg, &pvector)] = 0;
 			} else {
 				strcpy(errormsg, "SQL:");
@@ -575,7 +575,7 @@
 		long errorcode = isc_sqlcode(status);
 		if (errorcode == -999) {
 			errorcode = status[1];
-			long *pvector = status;
+			ISC_STATUS *pvector = status;
 			errormsg[isc_interprete(errormsg, &pvector)] = 0;
 		} else {
 			strcpy(errormsg, "SQL:");
