diff -urN 1/Mono.Profiler/Mono.Profiler.Widgets/Makefile.am 2/Mono.Profiler/Mono.Profiler.Widgets/Makefile.am
--- 1/Mono.Profiler/Mono.Profiler.Widgets/Makefile.am	2011-02-12 10:32:47.000000000 -0600
+++ 2/Mono.Profiler/Mono.Profiler.Widgets/Makefile.am	2012-02-14 23:45:06.048330154 -0600
@@ -9,7 +9,8 @@
 ASSEMBLY_MDB = 
 endif
 
-pkglib_DATA = $(ASSEMBLY) $(ASSEMBLY_MDB)
+programfilesdir = $(pkglibdir)
+programfiles_DATA = $(ASSEMBLY) $(ASSEMBLY_MDB)
 
 CLEANFILES = $(ASSEMBLY) $(ASSEMBLY_MDB)
 
diff -urN 1/Mono.Profiler/heap-snapshot-explorer/Makefile.am 2/Mono.Profiler/heap-snapshot-explorer/Makefile.am
--- 1/Mono.Profiler/heap-snapshot-explorer/Makefile.am	2011-02-12 10:32:47.000000000 -0600
+++ 2/Mono.Profiler/heap-snapshot-explorer/Makefile.am	2012-02-14 23:46:21.117195189 -0600
@@ -9,7 +9,8 @@
 ASSEMBLY_MDB = 
 endif
 
-pkglib_DATA = $(ASSEMBLY) $(ASSEMBLY_MDB)
+programfilesdir = $(pkglibdir)
+programfiles_DATA = $(ASSEMBLY) $(ASSEMBLY_MDB)
 
 CLEANFILES = $(ASSEMBLY) $(ASSEMBLY_MDB)
 
diff -urN 1/Mono.Profiler/heap-snapshot-viewer/Makefile.am 2/Mono.Profiler/heap-snapshot-viewer/Makefile.am
--- 1/Mono.Profiler/heap-snapshot-viewer/Makefile.am	2011-02-12 10:32:47.000000000 -0600
+++ 2/Mono.Profiler/heap-snapshot-viewer/Makefile.am	2012-02-14 23:47:14.819383799 -0600
@@ -8,7 +8,8 @@
 ASSEMBLY_MDB = 
 endif
 
-pkglib_DATA = $(ASSEMBLY) $(ASSEMBLY_MDB)
+programfilesdir = $(pkglibdir)
+programfiles_DATA = $(ASSEMBLY) $(ASSEMBLY_MDB)
 bin_SCRIPTS = mprof-heap-viewer
 man_MANS=man/man1/mprof-heap-viewer.1
 
diff -urN 1/Mono.Profiler/mprof-gui/Makefile.am 2/Mono.Profiler/mprof-gui/Makefile.am
--- 1/Mono.Profiler/mprof-gui/Makefile.am	2011-02-12 10:32:47.000000000 -0600
+++ 2/Mono.Profiler/mprof-gui/Makefile.am	2012-02-14 23:48:03.341651249 -0600
@@ -8,7 +8,8 @@
 ASSEMBLY_MDB = 
 endif
 
-pkglib_DATA = $(ASSEMBLY) $(ASSEMBLY_MDB)
+programfilesdir = $(pkglibdir)
+programfiles_DATA = $(ASSEMBLY) $(ASSEMBLY_MDB)
 bin_SCRIPTS = emveepee
 
 CLEANFILES = $(ASSEMBLY) $(ASSEMBLY_MDB)
diff -urN 1/Mono.Profiler/profiler-decoder-library/Makefile.am 2/Mono.Profiler/profiler-decoder-library/Makefile.am
--- 1/Mono.Profiler/profiler-decoder-library/Makefile.am	2011-02-12 10:32:47.000000000 -0600
+++ 2/Mono.Profiler/profiler-decoder-library/Makefile.am	2012-02-14 23:48:48.164974430 -0600
@@ -9,7 +9,8 @@
 ASSEMBLY_MDB = 
 endif
 
-pkglib_DATA = $(ASSEMBLY) $(ASSEMBLY_MDB)
+programfilesdir = $(pkglibdir)
+programfiles_DATA = $(ASSEMBLY) $(ASSEMBLY_MDB)
 
 CLEANFILES = $(ASSEMBLY) $(ASSEMBLY_MDB)
 
diff -urN 1/Mono.Profiler/profiler-file-decoder/Makefile.am 2/Mono.Profiler/profiler-file-decoder/Makefile.am
--- 1/Mono.Profiler/profiler-file-decoder/Makefile.am	2011-02-12 10:32:47.000000000 -0600
+++ 2/Mono.Profiler/profiler-file-decoder/Makefile.am	2012-02-14 23:49:37.892224361 -0600
@@ -9,7 +9,8 @@
 ASSEMBLY_MDB = 
 endif
 
-pkglib_DATA = $(ASSEMBLY) $(ASSEMBLY_MDB)
+programfilesdir = $(pkglibdir)
+programfiles_DATA = $(ASSEMBLY) $(ASSEMBLY_MDB)
 bin_SCRIPTS = mprof-decoder
 man_MANS = man/man1/mprof-decoder.1 
 
