diff -ur Ted-2.21.orig//Makefile Ted-2.21/Makefile
--- Ted-2.21.orig//Makefile	2009-10-16 23:20:58.000000000 +0800
+++ Ted-2.21/Makefile	2011-10-04 00:22:30.615975269 +0800
@@ -198,7 +198,7 @@
 	: To install Ted, you can now run 'make install' AS ROOT
 
 package.shared: tedPackage/makefile compile.shared
-	cd tedPackage && make package.shared
+	cd tedPackage && $(MAKE) package.shared
 	:
 	: Dynamically linked package ready.
 	: To install Ted, you can now run 'make install' AS ROOT
