From: Alexis Ballier <aballier@gentoo.org>
Date: Thu, 5 Nov 2009 19:48:41 +0100
Subject: Include sys/wait.h instead of only wait.h as described in POSIX specs. Fixes build on FreeBSD.

---
 src/core/main.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

--- a/src/core/main.c
+++ b/src/core/main.c
@@ -58,7 +58,7 @@
 
 #include <stdlib.h>
 #include <sys/types.h>
-#include <wait.h>
+#include <sys/wait.h>
 #include <stdio.h>
 #include <string.h>
 #include <signal.h>
