diff -Nuar libdbi-drivers-0.8.3-1.orig/configure.in libdbi-drivers-0.8.3-1/configure.in
--- libdbi-drivers-0.8.3-1.orig/configure.in	2008-10-09 17:09:09.303141877 -0700
+++ libdbi-drivers-0.8.3-1/configure.in	2008-10-09 17:10:48.100635797 -0700
@@ -149,7 +149,8 @@
-ac_docs="YES"
+ac_docs="no"
 AC_ARG_ENABLE(docs,
 	[  --disable-docs	  do not build and install the documentation.],
-	[  ac_docs="NO" ])
+	[ ac_docs="$enableval" ],
+	[ ac_docs="no" ])
 
-if test "$ac_docs" = "YES"; then
+if test "$ac_docs" = "yes"; then
 dnl   AC_PATH_PROGS(myjade, openjade jade)
