 base/CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/base/CMakeLists.txt b/base/CMakeLists.txt
index 013b097..c9080c4 100644
--- a/base/CMakeLists.txt
+++ b/base/CMakeLists.txt
@@ -59,7 +59,10 @@ if (MINGW)
 else (MINGW)
   set (FILES array.c credentials.c openvas_hosts.c cvss.c drop_privileges.c
        nvti.c nvticache.c openvas_compress.c openvas_file.c openvas_networking.c
-       openvas_string.c pidfile.c settings.c gpgme_util.c pwpolicy.c osp.c)
+       openvas_string.c pidfile.c settings.c gpgme_util.c pwpolicy.c osp.c
+       ${CMAKE_SOURCE_DIR}/misc/openvas_server.c
+       ${CMAKE_SOURCE_DIR}/omp/xml.c
+       )
   set (HEADERS array.h credentials.h openvas_hosts.h cvss.h drop_privileges.h
        nvticache.h nvti.h settings.h openvas_compress.h openvas_file.h osp.h
        openvas_networking.h openvas_string.h pidfile.h gpgme_util.h pwpolicy.h)
