--- libart/src/rbart_canvas.c
+++ libart/src/rbart_canvas.c
@@ -15,6 +15,7 @@
 
 #include <jpeglib.h>
 #include <png.h>
+#include <zlib.h> /* Z_BEST_COMPRESSION */
 
 #define ART_CANVAS_PNG_INTERLACE_ADAM7 1
 #define ART_CANVAS_ALPHA_MASK 2
@@ -352,7 +353,7 @@
         return Qnil;
     }
 
-    if (setjmp(png_ptr->jmpbuf))
+    if (setjmp(png_jmpbuf(png_ptr)))
     {
         png_destroy_write_struct(&png_ptr, &info_ptr);
         return Qnil;
