diff --git a/bindings/Makefile.am b/bindings/Makefile.am
index 3876cc2..6d1411d 100644
--- a/bindings/Makefile.am
+++ b/bindings/Makefile.am
@@ -48,7 +48,7 @@ perl: buildperl/Makefile

 buildperl/Makefile: .perl-directory-stamp buildperl/Makefile.PL \
 	$(top_builddir)/config.status
-	cd buildperl && @PERL@ Makefile.PL INSTALL_BASE=$(prefix) @PERL_BINDINGS_OPTIONS@
+	cd buildperl && @PERL@ Makefile.PL @PERL_BINDINGS_OPTIONS@

 buildperl/Makefile.PL: .perl-directory-stamp $(top_builddir)/config.status

diff --git a/configure.in b/configure.in
index 98395ed..5345a1d 100644
--- a/configure.in
+++ b/configure.in
@@ -5062,6 +5062,7 @@ then
 fi

 dnl Perl bindings
+PERL_BINDINGS_OPTIONS="PREFIX=${prefix}"
 AC_ARG_WITH(perl-bindings, [AS_HELP_STRING([--with-perl-bindings@<:@=OPTIONS@:>@], [Options passed to "perl Makefile.PL".])],
 [
 	if test "x$withval" != "xno" && test "x$withval" != "xyes"
@@ -5069,12 +5070,10 @@ AC_ARG_WITH(perl-bindings, [AS_HELP_STRING([--with-perl-bindings@<:@=OPTIONS@:>@
 		PERL_BINDINGS_OPTIONS="$withval"
 		with_perl_bindings="yes"
 	else
-		PERL_BINDINGS_OPTIONS=""
 		with_perl_bindings="$withval"
 	fi
 ],
 [
-	PERL_BINDINGS_OPTIONS=""
 	if test -n "$perl_interpreter"
 	then
 		with_perl_bindings="yes"