--- collectd-5.0.2/src/netlink.c~	2012-01-22 12:10:04.000000000 +0100
+++ collectd-5.0.2/src/netlink.c	2012-01-23 14:59:51.180465183 +0100
@@ -570,8 +570,7 @@
     return (-1);
   }
 
-  if (rtnl_dump_filter (&rth, link_filter, /* arg1 = */ NULL,
-	NULL, NULL) != 0)
+  if (rtnl_dump_filter (&rth, link_filter, /* arg1 = */ NULL) != 0)
   {
     ERROR ("netlink plugin: ir_read: rtnl_dump_filter failed.");
     return (-1);
@@ -608,8 +607,7 @@
 	continue;
       }
 
-      if (rtnl_dump_filter (&rth, qos_filter, (void *) &ifindex,
-	    NULL, NULL) != 0)
+      if (rtnl_dump_filter (&rth, qos_filter, (void *) &ifindex) != 0)
       {
 	ERROR ("netlink plugin: ir_read: rtnl_dump_filter failed.");
 	continue;
