--- babl-0.1.0.orig/extensions/Makefile.am	2009-04-17 12:05:15.000000000 -0500
+++ babl-0.1.0/extensions/Makefile.am	2009-08-19 00:53:21.435970261 -0500
@@ -1,7 +1,7 @@
 if OS_WIN32
 AM_LDFLAGS = -module -no-undefined
 else
-AM_LDFLAGS = -module
+AM_LDFLAGS = -module -avoid-version
 endif
 
 noinst_HEADERS = util.h
@@ -12,7 +12,7 @@
 	-I$(top_srcdir)/babl		\
 	-I$(top_srcdir)/extensions
 
-extdir = $(DESTDIR)$(libdir)/babl-@BABL_API_VERSION@
+extdir = $(libdir)/babl-@BABL_API_VERSION@
 ext_LTLIBRARIES = \
 	CIE-Lab.la      \
 	gegl-fixups.la  \
