diff --git a/babl/Makefile.am b/babl/Makefile.am
index 6abf148..6b3e270 100644
--- a/babl/Makefile.am
+++ b/babl/Makefile.am
@@ -92,10 +92,9 @@ if HAVE_INTROSPECTION
 Babl-$(BABL_API_VERSION).gir: $(G_IR_SCANNER) $(library_include_HEADERS) $(c_sources) $(srcdir)/Makefile.am
 	$(INTROSPECTION_SCANNER) -v --namespace Babl --nsversion=$(BABL_API_VERSION) \
             --add-include-path=$(srcdir) --add-include-path=. \
-            --library=babl-$(BABL_API_VERSION) \
+            --library=libbabl-$(BABL_API_VERSION).la \
             --libtool="$(LIBTOOL)" \
             --output $@ \
-            --pkg babl \
             -DBABL_IS_BEING_COMPILED \
 	    -I$(top_srcdir) \
 	    -I$(top_builddir) \
