--- babl/Makefile.am	2012-01-01 13:24:26.837016532 +0100
+++ babl/Makefile.am	2012-01-01 13:24:38.047876380 +0100
@@ -95,7 +95,7 @@
 Babl-$(BABL_API_VERSION).gir: $(G_IR_SCANNER) $(library_include_HEADERS) $(c_sources) $(srcdir)/Makefile.am
 	$(INTROSPECTION_SCANNER) -v --namespace Babl --nsversion=$(BABL_API_VERSION) \
             --add-include-path=$(srcdir) --add-include-path=. \
-            --library=babl-$(BABL_API_VERSION) \
+            --library=libbabl-$(BABL_API_VERSION).la \
             --libtool="$(LIBTOOL)" \
             --output $@ \
             -DBABL_IS_BEING_COMPILED \
