diff -Nurpa jrtplib-3.7.1.orig/src/rtcpcompoundpacketbuilder.cpp jrtplib-3.7.1/src/rtcpcompoundpacketbuilder.cpp
--- jrtplib-3.7.1.orig/src/rtcpcompoundpacketbuilder.cpp	2008-12-16 08:58:06.000000000 +0300
+++ jrtplib-3.7.1/src/rtcpcompoundpacketbuilder.cpp	2008-12-16 08:59:54.000000000 +0300
@@ -42,6 +42,8 @@
 
 #include "rtpdebug.h"
 
+#include <cstring>
+
 RTCPCompoundPacketBuilder::RTCPCompoundPacketBuilder(RTPMemoryManager *mgr) : RTCPCompoundPacket(mgr), report(mgr), sdes(mgr)
 {
 	byesize = 0;
diff -Nurpa jrtplib-3.7.1.orig/src/rtppacket.cpp jrtplib-3.7.1/src/rtppacket.cpp
--- jrtplib-3.7.1.orig/src/rtppacket.cpp	2008-12-16 08:58:06.000000000 +0300
+++ jrtplib-3.7.1/src/rtppacket.cpp	2008-12-16 08:58:52.000000000 +0300
@@ -45,6 +45,8 @@
 
 #include "rtpdebug.h"
 
+#include <cstring>
+
 void RTPPacket::Clear()
 {
 	hasextension = false;
