Index: gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpegprotocol.c
===================================================================
--- gst-ffmpeg-0.10.4.orig/ext/ffmpeg/gstffmpegprotocol.c
+++ gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpegprotocol.c
@@ -175,8 +175,8 @@ gst_ffmpegdata_write (URLContext * h, un
   return size;
 }
 
-static offset_t
-gst_ffmpegdata_seek (URLContext * h, offset_t pos, int whence)
+static int64_t
+gst_ffmpegdata_seek (URLContext * h, int64_t pos, int whence)
 {
   GstProtocolInfo *info;
   guint64 newpos;
