diff --git a/src/lbunzip2.c b/src/lbunzip2.c
index 113be47..5d8a456 100644
--- a/src/lbunzip2.c
+++ b/src/lbunzip2.c
@@ -604,7 +604,7 @@ work_decompr(struct w2w_blk *w2w_blk, struct w2m_q *w2m_q,
     w2m_blk->id.decompr_blk_id = decompr_blk_id++;
     w2m_blk->id.last_decompr = (YB_OK == ybret);
     w2m_blk->produced = sizeof w2m_blk->decompr - oleft;
-    w2m_blk->bs100k = w2w_blk->bs100k;
+    w2m_blk->bs100k = (YB_OK == ybret) ? w2w_blk->bs100k : 0u;
     w2m_blk->crc = w2w_blk->crc;
     w2m_blk->crc1 = crc;
     w2m_blk->bs100k1 = bs100k;
