diff --git a/common/gsm-sms.c b/common/gsm-sms.c
index 2163757..f93170d 100644
--- a/common/gsm-sms.c
+++ b/common/gsm-sms.c
@@ -923,8 +923,8 @@ gn_error gn_sms_pdu2raw(gn_sms_raw *rawsms, unsigned char *pdu, int pdu_len, int
 		/* TP-ST   TP-Status */
 		rawsms->report              = pdu[offset++];
 		/* TP-PI   TP-Parameter-Indicator */
-		dprintf("TP-Parameter-Indicator: 0x%02x\n", pdu[offset++]);
-		parameter_indicator         = pdu[offset];
+		dprintf("TP-Parameter-Indicator: 0x%02x\n", pdu[offset]);
+		parameter_indicator         = pdu[offset++];
 		/* handle the "extension bit" skipping the following octects, if any (see 9.2.3.27 TP-Parameter-Indicator) */
 		while (pdu[offset++] & 0x80)
 			;

