--- a/textblock.cc
+++ b/textblock.cc
@@ -18,6 +18,7 @@
 #include <algorithm>
 #include <cstdio>
 #include <vector>
+#include <string>
 
 #include "common.h"
 #include "rational.h"
--- a/page_image_layout.cc
+++ b/page_image_layout.cc
@@ -18,6 +18,8 @@
 #include <algorithm>
 #include <cstdio>
 #include <vector>
+#include <cstdlib>
+#include <string>
 
 #include "common.h"
 #include "rational.h"
--- a/track.cc
+++ b/track.cc
@@ -17,6 +17,7 @@
 
 #include <algorithm>
 #include <vector>
+#include <cstdio>
 
 #include "common.h"
 #include "rectangle.h"
--- a/textline.cc
+++ b/textline.cc
@@ -19,6 +19,7 @@
 #include <cctype>
 #include <cstdio>
 #include <vector>
+#include <string>
 
 #include "common.h"
 #include "rational.h"
--- a/common.cc
+++ b/common.cc
@@ -17,6 +17,7 @@
 
 #include <algorithm>
 #include <cstdio>
+#include <cstring>
 #include <cstdlib>
 #include "common.h"
 
--- a/textpage.cc
+++ b/textpage.cc
@@ -18,6 +18,7 @@
 #include <cstdio>
 #include <string>
 #include <vector>
+#include <algorithm>
 
 #include "common.h"
 #include "rectangle.h"
--- a/textline_r2.cc
+++ b/textline_r2.cc
@@ -18,6 +18,7 @@
 #include <cctype>
 #include <cstdio>
 #include <vector>
+#include <cstdlib>
 
 #include "common.h"
 #include "rectangle.h"
--- a/page_image.cc
+++ b/page_image.cc
@@ -19,6 +19,8 @@
 #include <cctype>
 #include <cstdio>
 #include <vector>
+#include <climits>
+#include <string>
 
 #include "common.h"
 #include "rational.h"
--- a/character_r12.cc
+++ b/character_r12.cc
@@ -17,6 +17,7 @@
 
 #include <cstdio>
 #include <vector>
+#include <cstdlib>
 #include "common.h"
 #include "rectangle.h"
 #include "ucs.h"
