Index: src/ddcpci/ddcpci.h
===================================================================
RCS file: /cvsroot/ddccontrol/ddccontrol/src/ddcpci/ddcpci.h,v
retrieving revision 1.9
retrieving revision 1.10
diff -u -r1.9 -r1.10
--- src/ddcpci/ddcpci.h	21 Sep 2005 20:45:37 -0000	1.9
+++ src/ddcpci/ddcpci.h	14 Feb 2006 19:32:48 -0000	1.10
@@ -22,6 +22,9 @@
 
 #include "i2c-algo-bit.h"
 
+#include <sys/types.h>
+#define PCI_HAVE_Uxx_TYPES
+
 #include <pci/pci.h>
 
 #include <libintl.h>
