Index: configure.ac
===================================================================
RCS file: /cvsroot/ddccontrol/ddccontrol/configure.ac,v
retrieving revision 1.32
retrieving revision 1.33
diff -u -r1.32 -r1.33
--- configure.ac	18 Nov 2005 16:07:22 -0000	1.32
+++ configure.ac	20 Feb 2006 17:27:55 -0000	1.33
@@ -132,6 +132,8 @@
 if test x$support_gnome_applet = xyes; then
 	GNOME_APPLET="gnome-ddcc-applet"
 	PKG_CHECK_MODULES(GNOME, libpanelapplet-2.0 >= 2.10,,GNOME_APPLET="")
+    GNOME_LDFLAGS="$LIBXML2_LDFLAGS `pkg-config --libs gtk+-2.0 gthread-2.0 libpanelapplet-2.0`"
+    GNOME_CFLAGS="$LIBXML2_CFLAGS `pkg-config --cflags gtk+-2.0 gthread-2.0 libpanelapplet-2.0`"
 fi
 
 
