Gentoo Bug #122640 --as-needed fixes
Signed-off-by: Ben Skeggs <darktama@iinet.net.au>
Acked-by: Robin H. Johnson <robbat2@gentoo.org>

diff -Nur ddccontrol-0.3.old/src/lib/Makefile.am ddccontrol-0.3/src/lib/Makefile.am
--- ddccontrol-0.3.old/src/lib/Makefile.am	2005-10-26 05:22:44.000000000 +1100
+++ ddccontrol-0.3/src/lib/Makefile.am	2006-02-13 14:29:01.000000000 +1100
@@ -8,4 +8,5 @@
 
 libddccontrol_la_SOURCES = ddcci.c ddcci.h monitor_db.c monitor_db.h \
    ddcpci-ipc.h i2c-dev.h conf.c conf.h
+libddccontrol_la_LIBADD = $(LIBXML2_LDFLAGS)
 AM_CFLAGS = $(LIBXML2_CFLAGS)
