Index: htmlc-2.21.0/compiler/Makefile
===================================================================
--- htmlc-2.21.0.orig/compiler/Makefile
+++ htmlc-2.21.0/compiler/Makefile
@@ -37,12 +37,12 @@ bin: $(PROG).bin
 
 $(PROG).bin: $(BINOBJS)
 	$(CAMLBIN) -o $(PROG).bin str.cmxa unix.cmxa $(BINOBJS) \
-		-cclib -lunix -cclib -lstr && \
+		-cclib -lunix -cclib -lcamlstr && \
 	$(RM) $(PROG) && ln -s $(PROG).bin $(PROG)
 
 $(PROG).byt: $(BYTOBJS)
 	$(CAMLBYT) -g -custom -o $(PROG).byt str.cma unix.cma $(BYTOBJS) \
-		-cclib -lunix -cclib -lstr && \
+		-cclib -lunix -cclib -lcamlstr && \
 	$(RM) $(PROG) && ln -s $(PROG).byt $(PROG)
 
 installbyt:
