diff -Naur rskkserv-2.95.4.orig/Makefile.in rskkserv-2.95.4/Makefile.in
--- rskkserv-2.95.4.orig/Makefile.in	2005-06-19 19:53:43.000000000 +0900
+++ rskkserv-2.95.4/Makefile.in	2010-05-29 17:22:16.000000000 +0900
@@ -29,7 +29,7 @@
 
 
 all: $(SKKSERV) $(CONFFILE) ext/Makefile # README INSTALL
-	cd ext; make $@; cd ..
+	cd ext; $(MAKE) $@; cd ..
 
 $(SKKSERV): $(SKKSERV).in @CONFIG_STATUS@
 	$(RUBY) -p -e 'sub %r%@RUBY\@%, %q%$(RUBY)%' \
@@ -61,7 +61,7 @@
 	cd ext; $(RUBY) extconf.rb
 
 distclean: clean
-	cd ext; make $@; cd ..
+	cd ext; $(MAKE) $@; cd ..
 	rm -f ext/extconf.rb ext/depend
 	rm -f config.* configure Makefile README INSTALL rskkserv.spec
 	rm -rf autom4te.cache
@@ -69,11 +69,11 @@
 
 clean:
 	find ./ -name "*~" -exec rm {} \;
-	cd ext; make $@; cd ..
+	cd ext; $(MAKE)$@; cd ..
 	rm -f $(SKKSERV) $(CONFFILE)
 
 install: all installdirs
-	cd ext; make DESTDIR=$(DESTDIR) rss-install; cd ..
+	cd ext; $(MAKE) DESTDIR=$(DESTDIR) rss-install; cd ..
 	$(INSTALL) -m0755 $(SKKSERV) $(DESTDIR)$(sbindir)/$(SKKSERV)
 	if [ ! -f $(DESTDIR)$(sysconfdir)/$(CONFFILE) ]; then \
 	    $(INSTALL) -m0644 $(CONFFILE) $(DESTDIR)$(sysconfdir)/; \
