=== modified file 'Makefile.am'
--- Makefile.am	2008-11-16 20:06:01 +0000
+++ Makefile.am	2008-11-16 20:20:44 +0000
@@ -74,23 +74,23 @@
 # to depend on the nb executable use "nb${EXEEXT}"
 
 nb_mod_simple.${NB_MOD_SUFFIX}: nbapi.h nb_mod_simple.c
-	${CC} ${NB_MOD_CFLAGS} nb_mod_simple.c -o nb_mod_simple.${NB_MOD_SUFFIX}
+	${CC} $(CFLAGS) ${NB_MOD_CFLAGS} $(LDFLAGS) nb_mod_simple.c -o nb_mod_simple.${NB_MOD_SUFFIX}
 nb_mod_tree.${NB_MOD_SUFFIX}: nbapi.h nb_mod_tree.c
-	${CC} ${NB_MOD_CFLAGS} nb_mod_tree.c -o nb_mod_tree.${NB_MOD_SUFFIX}
+	${CC} $(CFLAGS) ${NB_MOD_CFLAGS} $(LDFLAGS) nb_mod_tree.c -o nb_mod_tree.${NB_MOD_SUFFIX}
 nb_mod_console.${NB_MOD_SUFFIX}: nb${EXEEXT} nb.h nb_mod_console.c
-	${CC} ${NB_MOD_CFLAGS} nb_mod_console.c -o nb_mod_console.${NB_MOD_SUFFIX}
+	${CC} $(CFLAGS) ${NB_MOD_CFLAGS} $(LDFLAGS) nb_mod_console.c -o nb_mod_console.${NB_MOD_SUFFIX}
 nb_mod_netflow.${NB_MOD_SUFFIX}: nbapi.h nb_mod_netflow.c
-	${CC} ${NB_MOD_CFLAGS} nb_mod_netflow.c -o nb_mod_netflow.${NB_MOD_SUFFIX}
+	${CC} $(CFLAGS) ${NB_MOD_CFLAGS} $(LDFLAGS) nb_mod_netflow.c -o nb_mod_netflow.${NB_MOD_SUFFIX}
 nb_mod_snmptrap.${NB_MOD_SUFFIX}: nbapi.h nb_mod_snmptrap.c
-	${CC} ${NB_MOD_CFLAGS} nb_mod_snmptrap.c -o nb_mod_snmptrap.${NB_MOD_SUFFIX}
+	${CC} $(CFLAGS) ${NB_MOD_CFLAGS} $(LDFLAGS) nb_mod_snmptrap.c -o nb_mod_snmptrap.${NB_MOD_SUFFIX}
 nb_mod_syslog.${NB_MOD_SUFFIX}: nbapi.h nb_mod_syslog.c
-	${CC} ${NB_MOD_CFLAGS} nb_mod_syslog.c -o nb_mod_syslog.${NB_MOD_SUFFIX}
+	${CC} $(CFLAGS) ${NB_MOD_CFLAGS} $(LDFLAGS) nb_mod_syslog.c -o nb_mod_syslog.${NB_MOD_SUFFIX}
 nb_mod_translate.${NB_MOD_SUFFIX}: nbapi.h nb_mod_translate.c
-	${CC} ${NB_MOD_CFLAGS} nb_mod_translate.c -o nb_mod_translate.${NB_MOD_SUFFIX}
+	${CC} $(CFLAGS) ${NB_MOD_CFLAGS} $(LDFLAGS) nb_mod_translate.c -o nb_mod_translate.${NB_MOD_SUFFIX}
 nb_mod_cache.${NB_MOD_SUFFIX}: nb${EXEEXT} nb.h nb_mod_cache.c
-	${CC} ${NB_MOD_CFLAGS} nb_mod_cache.c -o nb_mod_cache.${NB_MOD_SUFFIX}
+	${CC} $(CFLAGS) ${NB_MOD_CFLAGS} $(LDFLAGS) nb_mod_cache.c -o nb_mod_cache.${NB_MOD_SUFFIX}
 nb_mod_servant.${NB_MOD_SUFFIX}: nb${EXEEXT} nbapi.h nb_mod_servant.c
-	${CC} ${NB_MOD_CFLAGS} nb_mod_servant.c -o nb_mod_servant.${NB_MOD_SUFFIX}
+	${CC} $(CFLAGS) ${NB_MOD_CFLAGS} $(LDFLAGS) nb_mod_servant.c -o nb_mod_servant.${NB_MOD_SUFFIX}
 
 # CLEANFILES = bin/nb_mod_<module>.${NB_MOD_SUFFIX}
 

=== modified file 'configure.ac'
--- configure.ac	2008-11-16 20:06:01 +0000
+++ configure.ac	2008-11-16 20:13:16 +0000
@@ -117,7 +117,6 @@
 NB_MOD_PATH="/usr/local/lib/${PACKAGE_NAME}"
 NB_MOD_PATH_SEPARATOR=":"
 NB_MOD_SUFFIX="so"
-CFLAGS=
 case "$host" in
   *-netbsd*)
     AC_DEFINE(NETBSD,,[Define if NetBSD])
@@ -144,7 +143,8 @@
   *-linux*)
     AC_DEFINE(LINUX,,[Define if LINUX])
     CFLAGS="${CFLAGS} -rdynamic"
-    LDFLAGS="${LDFLAGS} -ldl -lm"
+    LDFLAGS="${LDFLAGS}"
+    LIBS=" -ldl -lm"
     ;;
   *-hpux*)
     AC_DEFINE(HPUX,,[Define if HPUX])

