--- work/json_simple/src/org/json/simple/JSONArray.java	2006-04-16 04:04:24.000000000 +0300
+++ work/json_simple/src/org/json/simple/JSONArray.java	2007-06-04 17:43:56.000000000 +0300
@@ -12,6 +13,18 @@
  * @author FangYidong<fangyidong@yahoo.com.cn>
  */
 public class JSONArray extends ArrayList {
+	public JSONArray() {
+		super();
+	}
+
+	public JSONArray(java.util.Collection arg0) {
+		super(arg0);
+	}
+
+	public JSONArray(int initialCapacity) {
+		super(initialCapacity);
+	}
+
 	public String toString(){
 		ItemList list=new ItemList();
 		
--- work/json_simple/src/org/json/simple/JSONObject.java	2006-04-16 11:26:38.000000000 +0300
+++ work/json_simple/src/org/json/simple/JSONObject.java	2007-06-04 17:43:56.000000000 +0300
@@ -13,6 +13,22 @@
  */
 public class JSONObject extends HashMap{
 	
+	public JSONObject() {
+		super();
+	}
+
+	public JSONObject(int initialCapacity, float loadFactor) {
+		super(initialCapacity, loadFactor);
+	}
+
+	public JSONObject(int initialCapacity) {
+		super(initialCapacity);
+	}
+
+	public JSONObject(Map arg0) {
+		super(arg0);
+	}
+
 	public String toString(){
 		ItemList list=new ItemList();
 		Iterator iter=entrySet().iterator();
