--- Makefile
+++ Makefile
@@ -134,8 +134,8 @@
 
 .PHONY: install-prog
 install-prog:
-	$(TEST) -d $(INSTALL_DIR) || $(MKDIR) $(INSTALL_DIR)
-	$(TEST) -d $(INSTALL_BIN_DIR) || $(MKDIR) $(INSTALL_BIN_DIR)
+	$(TEST) -d $(INSTALL_DIR) || $(MKDIR) -p $(INSTALL_DIR)
+	$(TEST) -d $(INSTALL_BIN_DIR) || $(MKDIR) -p $(INSTALL_BIN_DIR)
 	$(CP) -f ./$(FROMDIR)/$(PROG)$(EXE_EXT) $(TARGET)
 	-$(RM) -f $(MTARGET) > /dev/null 2>&1
 	-$(LN) -s $(PROG)$(EXE_EXT) $(MTARGET)
