diff -Nuar adjtimex-1.20.orig/adjtimex.c adjtimex-1.20/adjtimex.c
--- adjtimex-1.20.orig/adjtimex.c	2007-03-07 11:57:07.767901000 -0800
+++ adjtimex-1.20/adjtimex.c	2007-03-07 11:57:34.649921951 -0800
@@ -42,7 +42,7 @@
 extern int adjtimex(struct timex *);
 #else
 #ifndef __PIC__
-_syscall1(int, adjtimex, struct timex *, txcp)
+#define adjtimex(txcp) syscall(SYS_adjtimex,txcp)
 #endif
 #endif
 #endif
