--- adjtimex-1.20/debian/adjtimexconfig	2006-03-25 16:38:26.817571030 -0800
+++ adjtimex-1.20/debian/adjtimexconfig.new	2006-03-25 16:40:57.028173528 -0800
@@ -3,6 +3,11 @@
 conffile=/etc/conf.d/adjtimex
 startfile=/etc/init.d/adjtimex
 
+# Get UTC setting
+source /etc/conf.d/clock
+params=''
+[ "$CLOCK" == "UTC" ] && params='--utc'
+
 echo -n "Comparing clocks (this will take 70 sec)..."
 
 # Get the parameters
@@ -11,7 +16,7 @@
 baseline=`awk '/<= tick/{print ($1+$NF)/2}' /etc/adj.adjust`
 hz=`awk '/USER_HZ/{print $3}' /etc/adj.adjust`
 /usr/sbin/adjtimex  --tick $baseline  --frequency 0
-/usr/sbin/adjtimex  --adjust  >/etc/adj.adjust
+/usr/sbin/adjtimex  ${params} --adjust  >/etc/adj.adjust
 echo "done."
 ticks=`tail -n 1 /etc/adj.adjust|awk '{print $6}'`
 freq=`tail -n 1 /etc/adj.adjust|awk '{print $7}'`
