From ef5b6b87f200a3cdf1cd9fd936724677e70c568d Mon Sep 17 00:00:00 2001
From: Alessandro Pignotti <a.pignotti@sssup.it>
Date: Sun, 10 Jun 2012 18:57:48 +0200
Subject: [PATCH] [Build] Avoid new for each syntax to keep GCC 4.5
 compatibility

---
 src/swf.cpp |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/swf.cpp b/src/swf.cpp
index b735081..8501407 100644
--- a/src/swf.cpp
+++ b/src/swf.cpp
@@ -83,8 +83,8 @@ ParseThread* lightspark::getParseThread()
 
 RootMovieClip::~RootMovieClip()
 {
-	for(DictionaryTag* it: dictionary)
-		delete it;
+	for(auto it=dictionary.begin();it!=dictionary.end();++it)
+		delete *it;
 }
 
 void RootMovieClip::parsingFailed()
-- 
1.7.10

