From 631cb37ae66c0150a148a29df7a3c3e26dc03fb4 Mon Sep 17 00:00:00 2001
From: Alessandro Pignotti <a.pignotti@sssup.it>
Date: Sun, 10 Jun 2012 19:55:40 +0200
Subject: [PATCH] Fix a couple of stale reference assignment

---
 src/swftypes.cpp |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/swftypes.cpp b/src/swftypes.cpp
index 69de488..a2d05e9 100644
--- a/src/swftypes.cpp
+++ b/src/swftypes.cpp
@@ -761,13 +761,13 @@ inline RGBA medianColor(const RGBA& a, const RGBA& b, float factor)
 			{
 				//Thrown if the bitmapId does not exists in dictionary
 				LOG(LOG_ERROR,"Exception in FillStyle parsing: " << e.what());
-				v.bitmap=NullRef;
+				v.bitmap.reset();
 			}
 		}
 		else
 		{
 			//The bitmap might be invalid, the style should not be used
-			v.bitmap=NullRef;
+			v.bitmap.reset();
 		}
 	}
 	else
-- 
1.7.10

