From 2f54ea1ee730c8bb250d360949badb500d4e1687 Mon Sep 17 00:00:00 2001
From: Alessandro Pignotti <a.pignotti@sssup.it>
Date: Sun, 10 Jun 2012 20:20:01 +0200
Subject: [PATCH] Do not use nullptr for compatibility with GCC 4.5

---
 src/scripting/abc.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/scripting/abc.cpp b/src/scripting/abc.cpp
index 8110666..8c71a07 100644
--- a/src/scripting/abc.cpp
+++ b/src/scripting/abc.cpp
@@ -170,7 +170,7 @@ void ScriptLimitsTag::execute(RootMovieClip* root) const
 
 void ABCVm::registerClasses()
 {
-	Global* builtin=Class<Global>::getInstanceS(nullptr, 0);
+	Global* builtin=Class<Global>::getInstanceS((ABCContext*)NULL, 0);
 	//Register predefined types, ASObject are enough for not implemented classes
 	builtin->registerBuiltin("Object","",Class<ASObject>::getRef());
 	builtin->registerBuiltin("Class","",Class_object::getRef());
-- 
1.7.10

