Index: ulogd-1.23/extensions/ulogd_SYSLOG.c
===================================================================
--- ulogd-1.23.orig/extensions/ulogd_SYSLOG.c
+++ ulogd-1.23/extensions/ulogd_SYSLOG.c
@@ -136,7 +136,7 @@ static ulog_output_t syslog_op = { 
 	.name = "syslog", 
 	.init = &syslog_init,
 	.fini = &syslog_fini,
-	.output &_output_syslog
+	.output = &_output_syslog
 };
 
 
Index: ulogd-1.23/include/ulogd/ulogd.h
===================================================================
--- ulogd-1.23.orig/include/ulogd/ulogd.h
+++ ulogd-1.23/include/ulogd/ulogd.h
@@ -56,8 +56,6 @@
 #define ULOGD_ERROR	7	/* error condition, requires user action */
 #define ULOGD_FATAL	8	/* fatal, program aborted */
 
-extern FILE *logfile;
-
 typedef struct ulog_iret {
 	/* next interpreter return (key) in the global list */
 	struct ulog_iret *next;
