diff -u ulogd-1.24.orig/ulogd.conf.in ulogd-1.24/ulogd.conf.in
--- ulogd-1.24.orig/ulogd.conf.in	2009-08-11 22:56:31.000000000 -0700
+++ ulogd-1.24/ulogd.conf.in	2009-08-11 23:07:35.000000000 -0700
@@ -50,7 +50,10 @@
 #plugin="@libdir@/ulogd_PGSQL.so"
 #plugin="@libdir@/ulogd_SQLITE3.so"
 #plugin="@libdir@/ulogd_PCAP.so"
+#plugin="@libdir@/ulogd_SYSLOG.so"
 
+## Note: if you change the paths of log files from defaults, you probably 
+##       will want to update paths in /etc/logrotate.d/ulogd to match
 
 [LOGEMU]
 file="/var/log/ulogd/ulogd.syslogemu"
@@ -83,3 +86,6 @@
 file="/var/log/ulogd/ulogd.pcap"
 sync=1
 
+[SYSLOG]
+facility=LOG_DAEMON
+level=LOG_INFO
diff -u ulogd-1.24.orig/ulogd.logrotate ulogd-1.24/ulogd.logrotate
--- ulogd-1.24.orig/ulogd.logrotate	2009-08-11 22:56:32.000000000 -0700
+++ ulogd-1.24/ulogd.logrotate	2009-08-11 23:06:26.000000000 -0700
@@ -1,6 +1,8 @@
-/var/log/ulogd.log /var/log/ulogd.syslogemu /var/log/ulogd.pktlog /var/log/ulogd.pcap {
+/var/log/ulogd/ulogd.log /var/log/ulogd/ulogd.syslogemu /var/log/ulogd/ulogd.pktlog /var/log/ulogd/ulogd.pcap {
     missingok
     sharedscripts
+    weekly
+    compress
     postrotate
 	/bin/killall -HUP ulogd 2> /dev/null || true
     endscript
