From: Julian Ospald <hasufell@gentoo.org>
Date: Sun Mar  3 00:09:56 UTC 2013
Subject: build system

don't error out when build/locale is non-existent

--- arandr-0.1.7/setup.py
+++ arandr-0.1.7/setup.py
@@ -120,12 +120,13 @@
 
 class install_data(_install_data):
     def run(self):
-        for lang in os.listdir('build/locale/'):
-            lang_dir = os.path.join('share', 'locale', lang, 'LC_MESSAGES')
-            lang_file = os.path.join('build', 'locale', lang, 'LC_MESSAGES', 'arandr.mo')
-            self.data_files.append((lang_dir, [lang_file]))
+        if (os.path.exists('build/locale/')):
+            for lang in os.listdir('build/locale/'):
+                lang_dir = os.path.join('share', 'locale', lang, 'LC_MESSAGES')
+                lang_file = os.path.join('build', 'locale', lang, 'LC_MESSAGES', 'arandr.mo')
+                self.data_files.append((lang_dir, [lang_file]))
 
-        _install_data.run(self)
+            _install_data.run(self)
 
 class sdist(_sdist):
     def run(self):
