diff -Naur boxes-1.1-orig/Makefile boxes-1.1/Makefile
--- boxes-1.1-orig/Makefile	2007-03-30 14:59:03.000000000 -0600
+++ boxes-1.1/Makefile	2007-03-30 15:01:38.000000000 -0600
@@ -73,7 +73,7 @@
 
 
 # The following line (GLOBALCONF) is the only line you should need to edit!
-GLOBALCONF = /usr/share/boxes
+GLOBALCONF = /usr/share/boxes/boxes-config
 BVERSION   = 1.1
 
 SNAPFILE   = boxes-SNAP-$(shell date +%Y%m%d)
diff -Naur boxes-1.1-orig/src/Makefile boxes-1.1/src/Makefile
--- boxes-1.1-orig/src/Makefile	2007-03-30 14:59:03.000000000 -0600
+++ boxes-1.1/src/Makefile	2007-03-30 15:02:37.000000000 -0600
@@ -86,7 +86,7 @@
 LEX        = flex
 YACC       = bison
 CC         = gcc
-CFLAGS     = -ansi -I. -Iregexp -Wall -W $(CFLAGS_ADDTL)
+CFLAGS     += -ansi -I. -Iregexp -Wall -W $(CFLAGS_ADDTL)
 LDFLAGS    = -Lregexp
 
 GEN_HDR    = parser.h
@@ -106,8 +106,7 @@
 
 
 build:
-	$(MAKE) CFLAGS_ADDTL=-O boxes
-	strip boxes
+	$(MAKE) boxes
 debug:
 	$(MAKE) CFLAGS_ADDTL=-g boxes
 
diff -Naur boxes-1.1-orig/src/regexp/Makefile boxes-1.1/src/regexp/Makefile
--- boxes-1.1-orig/src/regexp/Makefile	2007-03-30 14:59:03.000000000 -0600
+++ boxes-1.1/src/regexp/Makefile	2007-03-30 15:03:10.000000000 -0600
@@ -42,7 +42,7 @@
 #============================================================================
 
 
-CFLAGS   = -O -I. $(CFLAGS_ADDTL)
+CFLAGS   += -I. $(CFLAGS_ADDTL)
 
 ALL_CL   = regexp/regexp.c regexp/regsub.c
 C_SRC    = $(notdir $(ALL_CL))
