--- setup.py.orig	2006-08-11 06:24:49.000000000 +0300
+++ setup.py	2007-07-02 04:25:32.000000000 +0300
@@ -1359,29 +1359,7 @@
     # 2005.12.08:
     cmdclass={'config': ConfigCommand, 'build_ext': BuildExtCommand,},
 
-    ext_modules=extensionModules,
-    data_files = [
-      # documentation:
-      (
-        # This path will be interpreted by distutils as being relative to
-        # sys.exec_prefix.
-        ((platformIsWindows and 'Lib') or 'lib/python'+sys.version[:3])
-          + '/site-packages/kinterbasdb/docs',
-        [
-          'docs/changelog.txt',
-          'docs/firebird-client-library-thread-safety-overview.html',
-          'docs/global.css',
-          'docs/index.html',
-          'docs/installation-binary.html',
-          'docs/installation-source.html',
-          'docs/license.txt',
-          'docs/links.html',
-          'docs/Python-DB-API-2.0.html',
-          'docs/usage.html',
-          'docs/w3c.png',
-        ]
-      )
-    ]
+    ext_modules=extensionModules
   )
 
 
