Index: pykde4-4.9.2/CMakeLists.txt
===================================================================
--- pykde4-4.9.2.orig/CMakeLists.txt
+++ pykde4-4.9.2/CMakeLists.txt
@@ -33,7 +33,7 @@ INCLUDE(SIPMacros)
 
 FIND_PACKAGE(PyQt4 REQUIRED)
 IF(PYQT4_VERSION STRLESS "040900")  # These version numbers also appear in ../CMakeLists.txt
-    MESSAGE(FATAL_ERROR "The version of PyQt found is too old. 4.9 or later is required.")
+    MESSAGE(FATAL_ERROR "The version of PyQt found is too old. 4.9.5 or later is required.")
 ENDIF(PYQT4_VERSION STRLESS "040900")
 
 
@@ -95,9 +95,19 @@ SET(SIP_DISABLE_FEATURES VendorID PyQt_N
 
 # Use an extra option when compiling on Python 3.
 IF (PYTHON_3)
-    SET(SIP_EXTRA_OPTIONS -g -x Py_v3)
+    IF(PYQT4_VERSION STRGREATER "040904")
+	    #Disable for newer PyQt
+	    SET(SIP_EXTRA_OPTIONS -g -x PyKDE_QVector -x Py_v3)
+    ELSE(PYQT4_VERSION STRGREATER "040904")
+	    SET(SIP_EXTRA_OPTIONS -g -x Py_v3)
+    ENDIF(PYQT4_VERSION STRGREATER "040904")
 ELSE (PYTHON_3)
-    SET(SIP_EXTRA_OPTIONS -g)
+    IF(PYQT4_VERSION STRGREATER "040904")
+	    #Disable for newer PyQt
+	    SET(SIP_EXTRA_OPTIONS -g -x PyKDE_QVector)
+    ELSE(PYQT4_VERSION STRGREATER "040904")
+	    SET(SIP_EXTRA_OPTIONS -g -x Py_v3)
+    ENDIF(PYQT4_VERSION STRGREATER "040904")
 ENDIF (PYTHON_3)
 
 ADD_DEFINITIONS(-D_REENTRANT -DQT_CORE_LIB -DQT_GUI_LIB -DUSING_SOPRANO_NRLMODEL_UNSTABLE_API)
Index: pykde4-4.9.2/sip/kdecore/typedefs.sip
===================================================================
--- pykde4-4.9.2.orig/sip/kdecore/typedefs.sip
+++ pykde4-4.9.2/sip/kdecore/typedefs.sip
@@ -951,6 +951,9 @@ template <TYPE1>
 %End
 };
 
+%Feature PyKDE_QVector
+
+%If(PyKDE_QVector)
 %MappedType QVector<int>
 {
 %TypeHeaderCode
@@ -1025,3 +1028,4 @@ template <TYPE1>
     return sipGetState(sipTransferObj);
 %End
 };
+%End
