From 4da4792844017b7b7e5f78cf106e34a68a08af96 Mon Sep 17 00:00:00 2001
From: Radomir Stevanovic <radomir.stevanovic@gmail.com>
Date: Fri, 10 Aug 2012 17:51:01 +0200
Subject: [PATCH] tests: python2.6 compat (`assertIn` added in 2.7)

---
 tests/test_requests_ext.py |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tests/test_requests_ext.py b/tests/test_requests_ext.py
index 883bdce..7645d8c 100644
--- a/tests/test_requests_ext.py
+++ b/tests/test_requests_ext.py
@@ -110,7 +110,7 @@ def test_cookies_on_redirects(self):
         s = requests.session()
         s.get(url='http://tinyurl.com/preview.php?disable=1')
         # we should have set a cookie for tinyurl: preview=0
-        self.assertIn('preview', s.cookies)
+        self.assertTrue('preview' in s.cookies)
         self.assertEqual(s.cookies['preview'], '0')
         self.assertEqual(list(s.cookies)[0].name, 'preview')
         self.assertEqual(list(s.cookies)[0].domain, 'tinyurl.com')
@@ -118,13 +118,13 @@ def test_cookies_on_redirects(self):
         # get cookies on another domain
         r2 = s.get(url='http://httpbin.org/cookies')
         # the cookie is not there
-        self.assertNotIn('preview', json.loads(r2.text)['cookies'])
+        self.assertTrue('preview' not in json.loads(r2.text)['cookies'])
 
         # this redirects to another domain, httpbin.org
         # cookies of the first domain should NOT be sent to the next one
         r3 = s.get(url='http://tinyurl.com/7zp3jnr')
         assert r3.url == 'http://httpbin.org/cookies'
-        self.assertNotIn('preview', json.loads(r2.text)['cookies'])
+        self.assertTrue('preview' not in json.loads(r2.text)['cookies'])
 
 if __name__ == '__main__':
     unittest.main()
-- 
1.7.10

