Index: gigedit-0.1.1/src/gigedit/mainwindow.cpp
===================================================================
--- gigedit-0.1.1.orig/src/gigedit/mainwindow.cpp
+++ gigedit-0.1.1/src/gigedit/mainwindow.cpp
@@ -18,6 +18,7 @@
  */
 
 #include <iostream>
+#include <cstring>
 
 #include <gtkmm/filechooserdialog.h>
 #include <gtkmm/messagedialog.h>
Index: gigedit-0.1.1/src/gigedit/paramedit.cpp
===================================================================
--- gigedit-0.1.1.orig/src/gigedit/paramedit.cpp
+++ gigedit-0.1.1/src/gigedit/paramedit.cpp
@@ -18,6 +18,7 @@
  */
 
 #include "paramedit.h"
+#include <cstring>
 
 namespace {
     const char* const controlChangeTexts[] = {
