--- configure	2004-06-10 20:59:32.970436153 -0500
+++ configure	2004-06-10 20:38:51.655962203 -0500
@@ -4627,9 +4627,9 @@
 
 LDFLAGS_old="$LDFLAGS"
 LDFLAGS="$LDFLAGS $LIBRARY_SEARCH_PATH"
-echo "$as_me:$LINENO: checking for DAInitialize in -ldockapp" >&5
-echo $ECHO_N "checking for DAInitialize in -ldockapp... $ECHO_C" >&6
-if test "${ac_cv_lib_dockapp_DAInitialize+set}" = set; then
+echo "$as_me:$LINENO: checking for DAOpenDisplay in -ldockapp" >&5
+echo $ECHO_N "checking for DAOpenDisplay in -ldockapp... $ECHO_C" >&6
+if test "${ac_cv_lib_dockapp_DAOpenDisplay+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
@@ -4648,11 +4648,11 @@
 #endif
 /* We use char because int might match the return type of a gcc2
    builtin and then its argument prototype would still apply.  */
-char DAInitialize ();
+char DAOpenDisplay ();
 int
 main ()
 {
-DAInitialize ();
+DAOpenDisplay ();
   ;
   return 0;
 }
@@ -4669,19 +4669,19 @@
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; }; then
-  ac_cv_lib_dockapp_DAInitialize=yes
+  ac_cv_lib_dockapp_DAOpenDisplay=yes
 else
   echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
 
-ac_cv_lib_dockapp_DAInitialize=no
+ac_cv_lib_dockapp_DAOpenDisplay=no
 fi
 rm -f conftest.$ac_objext conftest$ac_exeext conftest.$ac_ext
 LIBS=$ac_check_lib_save_LIBS
 fi
-echo "$as_me:$LINENO: result: $ac_cv_lib_dockapp_DAInitialize" >&5
-echo "${ECHO_T}$ac_cv_lib_dockapp_DAInitialize" >&6
-if test $ac_cv_lib_dockapp_DAInitialize = yes; then
+echo "$as_me:$LINENO: result: $ac_cv_lib_dockapp_DAOpenDisplay" >&5
+echo "${ECHO_T}$ac_cv_lib_dockapp_DAOpenDisplay" >&6
+if test $ac_cv_lib_dockapp_DAOpenDisplay = yes; then
   yes=yes
 else
   no=no
@@ -4689,7 +4689,7 @@
 
 LDFLAGS="$LDFLAGS_old"
 
-if test "x$ac_cv_lib_dockapp_DAInitialize" = xyes; then
+if test "x$ac_cv_lib_dockapp_DAOpenDisplay" = xyes; then
 
 CPPFLAGS_old="$CPPFLAGS"
 CPPFLAGS="$CPPFLAGS $HEADER_SEARCH_PATH $XCFLAGS"
