diff --git a/data/sound-juicer.desktop.in.in b/data/sound-juicer.desktop.in.in
index ecd42e0..3a7dc28 100644
--- a/data/sound-juicer.desktop.in.in
+++ b/data/sound-juicer.desktop.in.in
@@ -1,5 +1,7 @@
 [Desktop Entry]
-_Name=Audio CD Extractor
+_Name=Sound Juicer
+_GenericName=Audio CD Extractor
+_X-GNOME-FullName=Sound Juicer Audio CD Extractor
 _Comment=Copy music from your CDs
 Exec=sound-juicer %u
 Icon=sound-juicer
