diff --git a/cola/app.py b/cola/app.py
index a689fdb..9618c3e 100644
--- a/cola/app.py
+++ b/cola/app.py
@@ -52,10 +52,10 @@ def setup_environment():
     # Provide an SSH_ASKPASS fallback
     if sys.platform == 'darwin':
         os.environ.setdefault('SSH_ASKPASS',
-                              resources.share('bin', 'ssh-askpass-darwin'))
+                              resources.prefix('bin', 'ssh-askpass-darwin'))
     else:
         os.environ.setdefault('SSH_ASKPASS',
-                              resources.share('bin', 'ssh-askpass'))
+                              resources.prefix('bin', 'x11-ssh-askpass'))
 
     # Setup the path so that git finds us when we run 'git cola'
     path_entries = os.environ.get('PATH').split(os.pathsep)
diff --git a/setup.py b/setup.py
index d639eda..90d5fd0 100755
--- a/setup.py
+++ b/setup.py
@@ -101,10 +101,6 @@ def cola_data_files():
     data.extend([_app_path(localedir, 'git-cola.mo')
                  for localedir in glob('share/locale/*/LC_MESSAGES')])
 
-    if sys.platform == 'darwin':
-        data.append(_app_path('share/git-cola/bin', 'ssh-askpass-darwin'))
-    else:
-        data.append(_app_path('share/git-cola/bin', 'ssh-askpass'))
     return data
 
 
