diff -Nuar --exclude '*~' --exclude others-menu spectools-2009-02-R1.orig/99-wispy.rules spectools-2009-02-R1/99-wispy.rules
--- spectools-2009-02-R1.orig/99-wispy.rules	2007-05-19 09:49:21.000000000 -0700
+++ spectools-2009-02-R1/99-wispy.rules	2009-03-05 18:10:22.984352068 -0800
@@ -1,4 +1,10 @@
-# Two vid/pid pairs for Wispy1 USB dongle
-ATTRS{idVendor}=="1781", ATTRS{idProduct}=="083e", MODE="0660", GROUP="plugdev"
+# Wispy1 rev a
 ATTRS{idVendor}=="04b4", ATTRS{idProduct}=="0bad", MODE="0660", GROUP="plugdev"
+# Wispy1 rev b
+ATTRS{idVendor}=="1781", ATTRS{idProduct}=="083e", MODE="0660", GROUP="plugdev"
+# WiSpy 2.4x
 ATTRS{idVendor}=="1781", ATTRS{idProduct}=="083f", MODE="0660", GROUP="plugdev"
+# WiSpy DBx
+ATTRS{idVendor}=="1dd5", ATTRS{idProduct}=="5000", MODE="0660", GROUP="plugdev"
+
+# vim: ft=udevrules:
diff -Nuar --exclude '*~' --exclude others-menu spectools-2009-02-R1.orig/pkg-hildon/etc/udev/rules.d/99-wispy.rules spectools-2009-02-R1/pkg-hildon/etc/udev/rules.d/99-wispy.rules
--- spectools-2009-02-R1.orig/pkg-hildon/etc/udev/rules.d/99-wispy.rules	2007-12-21 23:05:49.000000000 -0800
+++ spectools-2009-02-R1/pkg-hildon/etc/udev/rules.d/99-wispy.rules	2009-03-05 18:10:19.800286330 -0800
@@ -1,4 +1,10 @@
-# Two vid/pid pairs for Wispy1 USB dongle
-ATTRS{idVendor}=="1781", ATTRS{idProduct}=="083e", MODE="0660", OWNER="user"
+# Wispy1 rev a
 ATTRS{idVendor}=="04b4", ATTRS{idProduct}=="0bad", MODE="0660", OWNER="user"
+# Wispy1 rev b
+ATTRS{idVendor}=="1781", ATTRS{idProduct}=="083e", MODE="0660", OWNER="user"
+# WiSpy 2.4x
 ATTRS{idVendor}=="1781", ATTRS{idProduct}=="083f", MODE="0660", OWNER="user"
+# WiSpy DBx
+ATTRS{idVendor}=="1dd5", ATTRS{idProduct}=="5000", MODE="0660", OWNER="user"
+
+# vim: ft=udevrules:
