From: Julian Ospald <hasufell@gentoo.org>
Date: Sun Jun  9 10:23:38 UTC 2013
Subject: 

--- higan_v092-source/higan/Makefile
+++ higan_v092-source/higan/Makefile
@@ -16,8 +16,8 @@
 # compiler
 c       := $(compiler) -std=gnu99
 cpp     := $(subst cc,++,$(compiler)) -std=gnu++0x
-flags   := -I. -O3 -fomit-frame-pointer
-link    := -s
+flags   := -I.
+link    := $(LDFLAGS)
 objects := libco
 
 # profile-guided optimization mode
@@ -33,7 +33,6 @@
 
 # platform
 ifeq ($(platform),x)
-  flags += -march=native
   link += -Wl,-export-dynamic -ldl -lX11 -lXext
 else ifeq ($(platform),win)
   ifeq ($(arch),win32)
@@ -57,9 +56,9 @@
 compile = \
   $(strip \
     $(if $(filter %.c,$<), \
-      $(c) $(flags) $1 -c $< -o $@, \
+      $(c) $(flags) $(CFLAGS) $1 -c $< -o $@, \
       $(if $(filter %.cpp,$<), \
-        $(cpp) $(flags) $1 -c $< -o $@ \
+        $(cpp) $(flags) $(CXXFLAGS) $1 -c $< -o $@ \
       ) \
     ) \
   )
--- higan_v092-source/higan/target-ethos/Makefile
+++ higan_v092-source/higan/target-ethos/Makefile
@@ -67,7 +67,7 @@
 # targets
 build: $(objects)
 ifeq ($(platform),x)
-	$(strip $(cpp) -o out/$(name) $(objects) $(link))
+	$(strip $(cpp) $(CXXFLAGS) -o out/$(name) $(objects) $(link))
 else ifeq ($(platform),win)
 	$(strip $(cpp) -shared -o out/phoenix.dll obj/phoenix.o $(phoenixlink))
 	$(strip $(cpp) -o out/$(name) $(subst obj/phoenix.o,,$(objects)) $(link) -Lout -lphoenix)
