http://bugs.gentoo.org/371887

--- configure
+++ configure
@@ -121,6 +121,7 @@
 ######################### QT
 subdirs="/qt /qt4"
 search_includes QtCore/QObject || err "The QT Library headerfiles were not found. Set QTDIR appropriately."
+search_lib QtCore4 QtCore || err "The QT library was not found."
 search_lib QtGui4 QtGui || err "The QT library was not found."
 search_lib ltdl || err "Libtool ltdl not found."
 search_lib c_r || true
