diff -ru ngrep-1.45~/configure.in ngrep-1.45/configure.in
--- ngrep-1.45~/configure.in	2006-12-23 23:44:17.000000000 +0100
+++ ngrep-1.45/configure.in	2006-12-24 00:32:59.000000000 +0100
@@ -130,9 +130,10 @@
   echo 'Configuring Perl-Compatible Regular Expression (PCRE) library ...' 
   echo 
 
-  REGEX_DIR='pcre-5.0'
-  REGEX_OBJS="$REGEX_DIR/pcre.o $REGEX_DIR/study.o"
+  REGEX_DIR=''
+  REGEX_OBJS=""
   USE_PCRE="1"
+  EXTRA_LIBS="$EXTRA_LIBS `pcre-config --libs`"
 
 else
 
@@ -161,7 +162,7 @@
 dnl OS-specific options
 dnl
 
-STRIPFLAG="-s"
+STRIPFLAG=""
 HAVE_DUMB_UDPHDR="0"
 
 case "$target_os" in
diff -ru ngrep-1.45~/ngrep.c ngrep-1.45/ngrep.c
--- ngrep-1.45~/ngrep.c	2006-12-23 23:44:17.000000000 +0100
+++ ngrep-1.45/ngrep.c	2006-12-24 00:34:37.000000000 +0100
@@ -92,7 +92,7 @@
 #endif
 
 #if USE_PCRE
-#include "pcre-5.0/pcre.h"
+#include <pcre.h>
 #else
 #include "regex-0.12/regex.h"
 #endif
