--- libebt/Makefile.am.orig	2006-11-17 11:44:49.000000000 +0100
+++ libebt/Makefile.am	2006-11-17 11:47:28.000000000 +0100
@@ -23,17 +23,15 @@
 the_tests += libebt_boost_threads_TEST
 
 libebt_boost_threads_TEST_SOURCES = libebt_boost_threads_TEST.cc $(test_headers) $(libinclude_HEADERS)
-libebt_boost_threads_TEST_LDADD = libebttests.a
-libebt_boost_threads_TEST_LDFLAGS = -lboost_thread
-libebt_boost_threads_TEST_CXXFLAGS = -pthread $(AM_CXXFLAGS)
+libebt_boost_threads_TEST_LDADD = libebttests.a -lboost_thread
+libebt_boost_threads_TEST_LDFLAGS = -pthread
 endif
 
 if USE_ZTHREAD_TESTS
 the_tests += libebt_zthread_threads_TEST
 
 libebt_zthread_threads_TEST_SOURCES = libebt_zthread_threads_TEST.cc $(test_headers) $(libinclude_HEADERS)
-libebt_zthread_threads_TEST_LDADD = libebttests.a
-libebt_zthread_threads_TEST_LDFLAGS = -lZThread
+libebt_zthread_threads_TEST_LDADD = libebttests.a -lZThread
 endif
 
 if USE_PTHREAD_TESTS
@@ -41,7 +39,7 @@
 
 libebt_pthread_threads_TEST_SOURCES = libebt_pthread_threads_TEST.cc $(test_headers) $(libinclude_HEADERS)
 libebt_pthread_threads_TEST_LDADD = libebttests.a
-libebt_pthread_threads_TEST_LDFLAGS = -lpthread
+libebt_pthread_threads_TEST_LDFLAGS = -pthread
 endif
 
 TESTS = $(the_tests)
