--- Makefile.orig	2007-11-08 18:07:44.000000000 +0100
+++ Makefile	2007-11-08 18:09:35.000000000 +0100
@@ -19,10 +19,13 @@
 OCAMLOPT=ocamlopt
 OCAMLDEP=ocamldep
 OCAMLMKLIB=ocamlmklib
+OCAMLFIND=ocamlfind 
 
 OBJS=zlib.cmo zip.cmo gzip.cmo
 C_OBJS=zlibstubs.o
 
+LIBINSTALL_FILES = $(wildcard *.mli *.cmi *.cma *.cmxa *.a *.so)
+
 all: libcamlzip.a zip.cma
 
 allopt: libcamlzip.a zip.cmxa
@@ -55,18 +58,7 @@
 	rm -f *.o *.a
 
 install:
-	mkdir -p $(INSTALLDIR)
-	cp zip.cma zip.cmi gzip.cmi zip.mli gzip.mli libcamlzip.a $(INSTALLDIR)
-	if test -f dllcamlzip.so; then \
-	  cp dllcamlzip.so $(INSTALLDIR); \
-          ldconf=`$(OCAMLC) -where`/ld.conf; \
-          installdir=$(INSTALLDIR); \
-          if test `grep -s -c $$installdir'$$' $$ldconf || :` = 0; \
-          then echo $$installdir >> $$ldconf; fi \
-        fi
-
-installopt:
-	cp zip.cmxa zip.a zip.cmx gzip.cmx $(INSTALLDIR)
+	$(OCAMLFIND) install camlzip META $(LIBINSTALL_FILES)
 
 depend:
 	gcc -MM -I$(ZLIB_INCLUDE) *.c > .depend
