--- src/DBwrapper.cpp.old	2010-05-18 09:47:25.000000000 +0200
+++ src/DBwrapper.cpp	2010-05-18 09:47:50.000000000 +0200
@@ -96,7 +96,7 @@
 		cerr << "sqlite3_compile error:" <<endl;
 		cerr << sqlite3_errmsg( m_db ) <<endl;
 		cerr << "on insert: " << query <<endl;
-		values = list<string>::list();
+		values = list<string>();
 	} else {
 		int busyCnt = 0;
 		int number = sqlite3_column_count( stmt );
@@ -138,7 +138,7 @@
 			cerr <<  "sqlite_step error.\n";
 			cerr << sqlite3_errmsg( m_db ) << endl;
 			cerr << "on query: " << query << endl;
-			values = list<string>::list();
+			values = list<string>();
         }
     }
 
