diff -Naur gtest-1.5.0.orig//Makefile.am gtest-1.5.0//Makefile.am
--- gtest-1.5.0.orig//Makefile.am	2010-04-16 07:02:08.000000000 +0900
+++ gtest-1.5.0//Makefile.am	2010-05-18 17:27:48.000000000 +0900
@@ -247,7 +247,7 @@
 TESTS += test/gtest_all_test
 check_PROGRAMS += test/gtest_all_test
 test_gtest_all_test_SOURCES = test/gtest_all_test.cc
-test_gtest_all_test_LDADD = lib/libgtest_main.la
+test_gtest_all_test_LDADD = lib/libgtest.la
 
 # Tests that fused gtest files compile and work.
 FUSED_GTEST_SRC = \
