From c361e4eaccfaed4f5c5d64a206ea7941e3ef8a57 Mon Sep 17 00:00:00 2001
From: Kai Willadsen <kai.willadsen@gmail.com>
Date: Sat, 4 Oct 2014 10:01:05 +1000
Subject: build_helpers: Make sure that we build help for C locale (bgo#737833)


diff --git a/meld/build_helpers.py b/meld/build_helpers.py
index 385e3dc..69ad5bf 100644
--- a/meld/build_helpers.py
+++ b/meld/build_helpers.py
@@ -117,6 +117,9 @@ class build_help(distutils.cmd.Command):
         else:
             self.selected_languages = os.listdir(self.help_dir)
 
+        if 'C' not in self.selected_languages:
+            self.selected_languages.append('C')
+
         self.C_PAGES = glob.glob(os.path.join(self.help_dir, 'C', '*.page'))
         self.C_EXTRA = glob.glob(os.path.join(self.help_dir, 'C', '*.xml'))
 
-- 
cgit v0.10.1

