--- x2goserver/Makefile.orig	2013-01-08 10:44:54.134787510 +0100
+++ x2goserver/Makefile	2013-01-08 10:46:06.721676563 +0100
@@ -3,6 +3,10 @@
 SRC_DIR=$(CURDIR)
 SHELL=/bin/bash
 
+CC?=gcc
+CFLAGS+=-fPIE
+LDFLAGS+=-pie
+
 INSTALL_DIR=install -d -o root -g root -m 755
 INSTALL_FILE=install -o root -g root -m 644
 INSTALL_PROGRAM=install -o root -g root -m 755
@@ -37,7 +41,7 @@
 build-arch: build_setgidwrappers
 
 build_setgidwrappers:
-	gcc -fPIE -pie $(CFLAGS) -o x2gosqlitewrapper x2gosqlitewrapper.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -o x2gosqlitewrapper x2gosqlitewrapper.c
 
 build-indep: build_man2html
 
