https://savannah.gnu.org/patch/index.php?6904

--- configure.ac	2009-08-28 16:57:41 +0000
+++ configure.ac	2009-08-28 17:04:44 +0000
@@ -582,9 +582,9 @@
   enable_cddb=yes
   AC_ARG_ENABLE(cddb,
 	[  --enable-cddb           include CDDB lookups in cd_info (default enabled)],	
-	enable_cddb=yes, enable_cddb=no)
+	[], enable_cddb=check)
 fi
-if test x$enable_cddb != "no" ; then
+if test x$enable_cddb != x"no" ; then
    PKG_CHECK_MODULES(CDDB, libcddb >= 1.0.1, [
     HAVE_CDDB=yes 
     AC_DEFINE(HAVE_CDDB, [], [Define this if you have libcddb installed])

