Index: g-cpan-0.16.4/lib/Gentoo/CPAN.pm
===================================================================
--- g-cpan-0.16.4.orig/lib/Gentoo/CPAN.pm
+++ g-cpan-0.16.4/lib/Gentoo/CPAN.pm
@@ -11,7 +11,6 @@ use YAML::Node;
 use Memoize;
 use Cwd qw(getcwd abs_path cwd);
 use File::Basename;
-use Shell qw(perl);
 
 memoize('transformCPAN');
 memoize('FindDeps');
@@ -197,12 +196,12 @@ sub unpackModule {
 
     # If we have a Makefile.PL, run it to generate Makefile
     if ( -f "Makefile.PL" ) {
-        perl("Makefile.PL",'</dev/null');
+        system("perl Makefile.PL </dev/null");
     }
 
     # If we have a Build.PL, run it to generate the Build script
     if ( -f "Build.PL" ) {
-        perl("Build.PL",'</dev/null');
+        system("perl Build.PL </dev/null");
     }
 
     # Return whence we came
Index: g-cpan-0.16.4/lib/Gentoo/Portage.pm
===================================================================
--- g-cpan-0.16.4.orig/lib/Gentoo/Portage.pm
+++ g-cpan-0.16.4/lib/Gentoo/Portage.pm
@@ -47,7 +47,7 @@ sub getEnv {
         return($var =~ /^$envvar$/ );
     };
 
-foreach my $file ( "$ENV{HOME}/.gcpanrc", "/etc/make.conf", "/etc/make.globals" ) {
+foreach my $file ( "$ENV{HOME}/.gcpanrc", "/etc/make.conf", "/etc/portage/make.conf", "/etc/make.globals" ) {
     if ( -f $file) {
     	my $importer = Shell::EnvImporter->new(
     		file => $file,
