diff --git a/src/client/Makefile.am b/src/client/Makefile.am
index 8281555..994904c 100644
--- a/src/client/Makefile.am
+++ b/src/client/Makefile.am
@@ -43,5 +43,5 @@ endif
 # Completions
 bashcompletiondir = $(sysconfdir)/bash_completion.d
 bashcompletion_DATA = lldpcli.bash-completion
-zshcompletiondir = $(datadir)/zsh/vendor-completions
+zshcompletiondir = $(datadir)/zsh/site-functions
 zshcompletion_DATA = _lldpcli
