diff -ruN at76_usb-0.14beta1.orig/at76c503.c at76_usb-0.14beta1/at76c503.c
--- at76_usb-0.14beta1.orig/at76c503.c	2007-02-09 22:34:44.000000000 -0600
+++ at76_usb-0.14beta1/at76c503.c	2007-02-09 22:36:01.000000000 -0600
@@ -18,10 +18,14 @@
  * Some iw_handler code was taken from airo.c, (C) 1999 Benjamin Reed
  */
 
+#include <linux/version.h>
+#if KERNEL_VERSION(2,6,19) > LINUX_VERSION_CODE
 #include <linux/config.h>
+#else
+#include <linux/autoconf.h>
+#endif
 #include <linux/init.h>
 #include <linux/kernel.h>
-#include <linux/version.h>
 #include <linux/sched.h>
 #include <linux/errno.h>
 #include <linux/poll.h>
diff -ruN at76_usb-0.14beta1.orig/at76_usbdfu.c at76_usb-0.14beta1/at76_usbdfu.c
--- at76_usb-0.14beta1.orig/at76_usbdfu.c	2007-02-09 22:34:44.000000000 -0600
+++ at76_usb-0.14beta1/at76_usbdfu.c	2007-02-09 22:36:40.000000000 -0600
@@ -24,7 +24,12 @@
  * - Firmware upload (device-to-host) support
  */
 
+#include <linux/version.h>
+#if KERNEL_VERSION(2,6,19) > LINUX_VERSION_CODE
 #include <linux/config.h>
+#else
+#include <linux/autoconf.h>
+#endif
 #include <linux/slab.h>
 #include <linux/usb.h>
 #include "at76c503.h"
