--- at76c503.c~	2007-06-22 00:35:09.000000000 -0400
+++ at76c503.c	2007-06-22 00:34:28.000000000 -0400
@@ -2299,9 +2299,15 @@
 		    dev->netdev->name, flag);
 }
 
+#if KERNEL_VERSION(2,6,20) > LINUX_VERSION_CODE
 static void kevent(void *data)
 {
 	struct at76c503 *dev = data;
+#else
+static void kevent(struct work_struct *work)
+{
+	struct at76c503 *dev = container_of(work, struct at76c503, kevent);
+#endif
 	int ret;
 	unsigned long flags;
 
@@ -6371,7 +6377,11 @@
 	dev->netdev = netdev;
 
 	init_MUTEX (&dev->sem);
+#if KERNEL_VERSION(2,6,20) > LINUX_VERSION_CODE
 	INIT_WORK (&dev->kevent, kevent, dev);
+#else
+	INIT_WORK (&dev->kevent, kevent);
+#endif
 
 	dev->open_count = 0;
 
