 gtkdatabox/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gtkdatabox/Makefile.am b/gtkdatabox/Makefile.am
index 503f753..d972de9 100644
--- a/gtkdatabox/Makefile.am
+++ b/gtkdatabox/Makefile.am
@@ -1,6 +1,6 @@
 # $Id: Makefile.am 4 2008-06-22 09:19:11Z rbock $
 
-lib_LTLIBRARIES		= libgtkdatabox.la
+noinst_LTLIBRARIES		= libgtkdatabox.la
 libgtkdatabox_la_SOURCES= \
 			gtkdatabox.c\
 			gtkdatabox_marshal.c\
