 src/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index b73ccc2..990f835 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -29,7 +29,7 @@ AM_LDFLAGS = -static
 
 klavaro_LDADD = @GTK_LIBS@ \
 		-lgmodule-2.0 \
-	       	$(top_srcdir)/gtkdatabox/libgtkdatabox.la
+	       	$(top_builddir)/gtkdatabox/libgtkdatabox.la
 
 if IS_POSIX
 AM_CFLAGS += -export-dynamic
