--- a/Makefile.in
+++ b/Makefile.in
@@ -290,64 +290,14 @@
 
 install_nls: locales
 ifdef REQUIRE_NLS
-	mkdir -p $(DESTDIR)$(NLS_DIR)/af/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/cs/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/da/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/de/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/eo/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/es/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/eu/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/fi/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/fr/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/id/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/it/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/ja/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/ky/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/lv/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/mt/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/nb/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/nl/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/nn/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/pl/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/pt/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/pt_BR/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/ro/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/ru/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/sk/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/sv/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/uk/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/vi/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/zh_CN/LC_MESSAGES
-	mkdir -p $(DESTDIR)$(NLS_DIR)/zh_TW/LC_MESSAGES
-	$(INSTALL_DATA) nls/af.gmo $(DESTDIR)$(NLS_DIR)/af/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/cs.gmo $(DESTDIR)$(NLS_DIR)/cs/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/da.gmo $(DESTDIR)$(NLS_DIR)/da/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/de.gmo $(DESTDIR)$(NLS_DIR)/de/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/eo.gmo $(DESTDIR)$(NLS_DIR)/eo/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/es.gmo $(DESTDIR)$(NLS_DIR)/es/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/eu.gmo $(DESTDIR)$(NLS_DIR)/eu/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/fi.gmo $(DESTDIR)$(NLS_DIR)/fi/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/fr.gmo $(DESTDIR)$(NLS_DIR)/fr/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/id.gmo $(DESTDIR)$(NLS_DIR)/id/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/it.gmo $(DESTDIR)$(NLS_DIR)/it/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/ja.gmo $(DESTDIR)$(NLS_DIR)/ja/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/ky.gmo $(DESTDIR)$(NLS_DIR)/ky/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/lv.gmo $(DESTDIR)$(NLS_DIR)/lv/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/mt.gmo $(DESTDIR)$(NLS_DIR)/mt/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/nb.gmo $(DESTDIR)$(NLS_DIR)/nb/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/nl.gmo $(DESTDIR)$(NLS_DIR)/nl/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/nn.gmo $(DESTDIR)$(NLS_DIR)/nn/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/pl.gmo $(DESTDIR)$(NLS_DIR)/pl/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/pt.gmo $(DESTDIR)$(NLS_DIR)/pt/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/pt_BR.gmo $(DESTDIR)$(NLS_DIR)/pt_BR/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/ro.gmo $(DESTDIR)$(NLS_DIR)/ro/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/ru.gmo $(DESTDIR)$(NLS_DIR)/ru/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/sk.gmo $(DESTDIR)$(NLS_DIR)/sk/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/sv.gmo $(DESTDIR)$(NLS_DIR)/sv/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/uk.gmo $(DESTDIR)$(NLS_DIR)/uk/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/vi.gmo $(DESTDIR)$(NLS_DIR)/vi/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/zh_CN.gmo $(DESTDIR)$(NLS_DIR)/zh_CN/LC_MESSAGES/$(PACKAGE).mo
-	$(INSTALL_DATA) nls/zh_TW.gmo $(DESTDIR)$(NLS_DIR)/zh_TW/LC_MESSAGES/$(PACKAGE).mo
+	catalogs='$(NLSGMO)'; \
+	for cat in $$catalogs; do \
+		cat=`basename $$cat`; \
+		cat=`echo $$cat | sed -e 's/\.gmo$$//'`; \
+		mkdir -p $(DESTDIR)$(NLS_DIR)/$$cat/LC_MESSAGES; \
+		$(INSTALL_DATA) nls/$$cat.gmo \
+		$(DESTDIR)$(NLS_DIR)/$$cat/LC_MESSAGES/$(PACKAGE).mo; \
+	done
 endif
 
 install_base: all sa1 sa2 sysstat.sysconfig install_man install_nls \
