--- yablex228.py.orig	2003-11-26 00:01:43.000000000 +1300
+++ yablex228.py	2003-11-25 23:31:45.000000000 +1300
@@ -79,20 +79,9 @@
 #File System Operations
 
 #RENAME
-YABLEROOT = "/mnt/gigadue/archivio/linux/grafica/datigrafica/yafraydata"
-#YABLEROOT = "D:\Grafica\datileo\yafraydata"
+YABLEROOT = os.path.expanduser("~/yafraydata")
 
-#YABLEROOT = "G:/tmp/YFexport"
-
-"""
-Examples:
-
-YABLEROOT = "C:\\Documents\\YFexport"
-YABLEROOT"/home/username/YFexport"
-
-"""
-#VIEWERAPP = "c:\windows\explorer"
-VIEWERAPP = "/usr/X11R6/bin/xview"
+VIEWERAPP = "/usr/bin/xview"
 CRAFTERFILE = "C:\crafter\data\preview.mat"
 #####################################################################
 
