diff -ur mISDN.orig/capi.c mISDN/capi.c
--- mISDN.orig/capi.c	2007-10-22 20:29:42.000000000 +0200
+++ mISDN/capi.c	2007-10-22 20:31:18.000000000 +0200
@@ -258,7 +258,7 @@
 #else
 				sizeof(_cmsg),
 #endif
-				0, 0, NULL, NULL);
+				0, 0, NULL);
 	if (!mISDN_cmsg_cp) {
 		CapiCachesFree();
 		return(-ENOMEM);
@@ -269,7 +269,7 @@
 #else
 				sizeof(AppPlci_t),
 #endif
-				0, 0, NULL, NULL);
+				0, 0, NULL);
 	if (!mISDN_AppPlci_cp) {
 		CapiCachesFree();
 		return(-ENOMEM);
@@ -280,7 +280,7 @@
 #else
 				sizeof(Ncci_t),
 #endif
-				0, 0, NULL, NULL);
+				0, 0, NULL);
 	if (!mISDN_ncci_cp) {
 		CapiCachesFree();
 		return(-ENOMEM);
@@ -291,7 +291,7 @@
 #else
 				sizeof(SSProcess_t),
 #endif
-				0, 0, NULL, NULL);
+				0, 0, NULL);
 	if (!mISDN_sspc_cp) {
 		CapiCachesFree();
 		return(-ENOMEM);
diff -ur mISDN.orig/udevice.c mISDN/udevice.c
--- mISDN.orig/udevice.c	2007-10-22 20:29:42.000000000 +0200
+++ mISDN/udevice.c	2007-10-22 20:33:35.000000000 +0200
@@ -2025,9 +2025,7 @@
 	if ((err = mISDN_unregister(&udev_obj))) {
 		printk(KERN_ERR "Can't unregister UserDevice(%d)\n", err);
 	}
-	if ((err = unregister_chrdev(mISDN_MAJOR, "mISDN"))) {
-		printk(KERN_WARNING "mISDN: devices busy on remove\n");
-	}
+	unregister_chrdev(mISDN_MAJOR, "mISDN");
 #ifdef CONFIG_DEVFS_FS
 	devfs_remove("mISDN");
 #endif
