diff -uNr sniffit.0.3.7.beta/sn_structs.h sniffit.0.3.7.beta-gcc33/sn_structs.h
--- sniffit.0.3.7.beta/sn_structs.h	1998-07-16 12:17:10.000000000 -0400
+++ sniffit.0.3.7.beta-gcc33/sn_structs.h	2003-07-26 11:53:46.229204088 -0400
@@ -67,7 +67,7 @@
 struct snif_mask                                         /* struct for mask */
   {
   _32_bit source_ip, destination_ip;
-  _32_bit short source_port, destination_port;
+  _32_bit source_port, destination_port;
   };
 
 /* (packet generation) */
