--- test/abstract_unit.rb.~1~	2011-02-21 20:35:21.775255045 +0100
+++ test/abstract_unit.rb	2011-02-21 20:39:26.051025768 +0100
@@ -1,6 +1,8 @@
 require 'rubygems'
 require 'test/unit'
 
+gem 'activesupport', '~>2.3'
+gem 'actionpack', '~>2.3'
 $:.unshift File.expand_path('../../lib', __FILE__)
 $:.unshift File.expand_path('../../../activesupport/lib', __FILE__)
 $:.unshift File.expand_path('../../../actionpack/lib', __FILE__)
