--- docs/CMakeLists.txt
+++ docs/CMakeLists.txt
@@ -11,7 +11,7 @@
 CONFIGURE_FILE(doxyfile.in.cmake ${DOXYFILE} @ONLY)
 
 # now build documentation
-ADD_CUSTOM_TARGET(doc
+ADD_CUSTOM_TARGET(doc ALL
   COMMAND "${DOXYGEN_EXECUTABLE}"
   DEPENDS doxyfile.in.cmake manual-main.txt ../include/lensfun/lensfun.h.in
   WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
@@ -19,5 +19,4 @@
 )
 
 # install documentation  
-FILE(GLOB doc_files "${CMAKE_DOC_OUT}/*.*")
-INSTALL(FILES ${doc_files} DESTINATION ${DOCDIR})
+INSTALL(DIRECTORY "${CMAKE_BINARY_DIR}/doc_doxygen/" DESTINATION ${DOCDIR})
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -30,7 +30,7 @@
 
 # options controlling the build process
 OPTION(BUILD_STATIC "Build static library" OFF)
-OPTION(BUILD_TESTS "Build test programs" ON)
+OPTION(BUILD_TESTS "Build test programs" OFF)
 OPTION(BUILD_AUXFUN "Build auxfun library (requires libpng)" OFF)
 OPTION(BUILD_FOR_SSE "Build with support for SSE" OFF)
 OPTION(BUILD_FOR_SSE2 "Build with support for SSE2" OFF)
@@ -71,13 +71,13 @@
 IF(BUILD_STATIC)
   ADD_DEFINITIONS(-DCONF_LENSFUN_STATIC)
 ENDIF()
- 
+
 # define destination dirs (relative to prefix)
 SET(BINDIR bin)
-SET(LIBDIR lib)
+# SET(LIBDIR lib)
 SET(INCLUDEDIR include)
 SET(DATADIR share/lensfun)
-SET(DOCDIR share/doc/lensfun-${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_MICRO}.${VERSION_BUGFIX})
+# SET(DOCDIR share/doc/lensfun-${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_MICRO}.${VERSION_BUGFIX})
 
 IF(WIN32)
   SET(PLATFORM_WINDOWS 1)
@@ -125,12 +125,12 @@
 ENDIF()
 
 # install include files
-INSTALL(FILES ${CMAKE_BINARY_DIR}/lensfun.h DESTINATION ${INCLUDEDIR}/lensfun)
+INSTALL(FILES ${CMAKE_BINARY_DIR}/lensfun.h DESTINATION ${INCLUDEDIR})
 IF(NOT HAVE_REGEX_H)
   INSTALL(FILES include/regex/regex.h DESTINATION ${INCLUDEDIR}/regex)
 ENDIF()
 IF(BUILD_AUXFUN)
-  INSTALL(FILES include/auxfun/image.h include/auxfun/rgbpixel.h DESTINATION ${INCLUDEDIR}/auxfun)
+  INSTALL(FILES include/auxfun/image.h include/auxfun/rgbpixel.h DESTINATION ${INCLUDEDIR})
 ENDIF()
 
 # install db files
