--- a/src/Makefile.am	2007-01-05 01:35:49.000000000 -0600
+++ b/src/Makefile.am	2007-01-05 01:35:58.000000000 -0600
@@ -31,6 +31,5 @@
 gringotts_LDADD = @POPT@ @GTK2_LIBS@ @GDKPIX_LIBS@ @LIBGRG_LIBS@
 
 install-exec-local:
-	strip $(DESTDIR)$(bindir)/gringotts
 	chmod u+s $(DESTDIR)$(bindir)/gringotts
 	
--- a/src/Makefile.in	2009-09-12 11:03:53.000000000 +0200
+++ b/src/Makefile.in	2009-09-12 11:04:05.000000000 +0200
@@ -486,7 +486,6 @@
 
 
 install-exec-local:
-	strip $(DESTDIR)$(bindir)/gringotts
 	chmod u+s $(DESTDIR)$(bindir)/gringotts
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
