--- gringotts.desktop.in	2007-01-05 01:52:16.000000000 -0600
+++ gringotts.desktop.in	2007-01-05 01:51:25.000000000 -0600
@@ -4,12 +4,12 @@
 Comment=An electronic strongbox
 Version=1.0
 Encoding=UTF-8
-Icon=/usr/share/pixmaps/gringotts.xpm
+Icon=gringotts.xpm
 Exec=gringotts
 Terminal=false
 Type=Application
-TryExec=
-X-GNOME-DocPath=
+TryExec=gringotts
+Categories=Application;Cryptography;
 Name[it_IT]=@GRG_CAP_NAME@
 GenericName[it_IT]="Cassaforte" elettronica
 Comment[it_IT]=Una cassaforte elettronica
