diff -ur emms-2.0.orig/Makefile emms-2.0/Makefile
--- emms-2.0.orig/Makefile	2006-04-29 14:28:28.000000000 -0500
+++ emms-2.0/Makefile	2006-05-14 23:03:58.000000000 -0500
@@ -41,7 +41,7 @@
 	makeinfo --html --no-split $<
 
 emms-print-metadata: emms-print-metadata.c
-	$(CC) -o $@ $< -I/usr/include/taglib -L/usr/lib -ltag_c
+	$(CC) -o $@ $< `taglib-config --libs --cflags` -ltag_c
 
 install:
 	test -d $(SITELISP) || mkdir -p $(SITELISP)
