http://bugzilla.gnome.org/show_bug.cgi?id=592784

diff -NrU5 gnome-subtitles-0.9.1.orig/configure.ac gnome-subtitles-0.9.1/configure.ac
--- gnome-subtitles-0.9.1.orig/configure.ac	2009-08-23 03:15:48.000000000 +0200
+++ gnome-subtitles-0.9.1/configure.ac	2009-08-23 03:32:40.000000000 +0200
@@ -4,19 +4,20 @@
 GNOME_COMMON_INIT
 
 GNOME_DOC_INIT
 GTK_DOC_CHECK([1.0])
 
-
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_SRCDIR([configure.ac])
 
 AM_INIT_AUTOMAKE([1.9 tar-ustar])
 AM_MAINTAINER_MODE
 
 IT_PROG_INTLTOOL([0.35.0])
 
+AC_PROG_LIBTOOL
+
 AC_PROG_INSTALL
 AC_PREFIX_PROGRAM(mono)
 
 dnl Gettext stuff.
 
diff -NrU5 gnome-subtitles-0.9.1.orig/Makefile.am gnome-subtitles-0.9.1/Makefile.am
--- gnome-subtitles-0.9.1.orig/Makefile.am	2009-08-23 03:15:48.000000000 +0200
+++ gnome-subtitles-0.9.1/Makefile.am	2009-08-23 03:35:59.000000000 +0200
@@ -1,14 +1,10 @@
 SUBDIRS = src data po help
 
 ACLOCAL_AMFLAGS = -I m4
 
 gstpbdir = $(pkglibdir)
-gstpb_SCRIPTS = build/libgstreamer_playbin.so
-
-$(gstpb_SCRIPTS):
-	cp -f src/libgstreamer_playbin.so build
 
 dist-hook:
 	@if test -d "$(srcdir)/.git"; \
 	then \
 		echo Creating ChangeLog && \
diff -NrU5 gnome-subtitles-0.9.1.orig/src/Makefile.am gnome-subtitles-0.9.1/src/Makefile.am
--- gnome-subtitles-0.9.1.orig/src/Makefile.am	2009-08-23 03:15:48.000000000 +0200
+++ gnome-subtitles-0.9.1/src/Makefile.am	2009-08-23 04:03:41.000000000 +0200
@@ -1,18 +1,20 @@
 gnomesubtitlesdir = $(pkglibdir)
 gnomesubtitles_SCRIPTS = $(ASSEMBLY)
 gnomesubtitles_DATA = $(ASSEMBLY_CONFIG)
+
 GS_BUILDDIR = $(srcdir)/../build
 GS_DATADIR = $(srcdir)/../data
 
 ASSEMBLY = $(GS_BUILDDIR)/gnome-subtitles.exe
 ASSEMBLY_CONFIG = $(ASSEMBLY).config
 
 AM_CFLAGS = $(gstreamer_CFLAGS) -Wall -g -fPIC
-noinst_PROGRAMS = libgstreamer_playbin.so
-libgstreamer_playbin_so_SOURCES = $(srcdir)/External/GStreamerPlaybin/main.c
-libgstreamer_playbin_so_LDFLAGS = $(gstreamer_LIBS) -lgstinterfaces-0.10 -shared -Wl,--as-needed,-soname,libgstreamer_playbin.so
+gnomesubtitles_LTLIBRARIES = libgstreamer_playbin.la
+libgstreamer_playbin_la_SOURCES = $(srcdir)/External/GStreamerPlaybin/main.c
+libgstreamer_playbin_la_LIBADD = $(gstreamer_LIBS) -lgstinterfaces-0.10
+libgstreamer_playbin_la_LIBTOOLFLAGS = --tag=disable-static 
 
 GS_SOURCES = \
 	$(srcdir)/External/GStreamerPlaybin/*.cs \
 	$(srcdir)/External/NCharDet/*.cs \
 	$(srcdir)/GnomeSubtitles/Core/*.cs \
