Index: sh.h
===================================================================
RCS file: /cvs/src/bin/mksh/sh.h,v
retrieving revision 1.579
retrieving revision 1.580
diff -u -r1.579 -r1.580
--- sh.h	24 Aug 2012 19:03:21 -0000	1.579
+++ sh.h	24 Aug 2012 19:09:34 -0000	1.580
@@ -959,15 +959,16 @@
 
 /* Determine the location of the system (common) profile */
 
-/* This is deliberately not configurable via CPPFLAGS */
+#ifndef MKSH_DEFAULT_PROFILEDIR
 #if defined(ANDROID)
-#define MKSH_ETC_LOCATION	"/system/etc"
+#define MKSH_DEFAULT_PROFILEDIR	"/system/etc"
 #else
-#define MKSH_ETC_LOCATION	"/etc"
+#define MKSH_DEFAULT_PROFILEDIR	"/etc"
+#endif
 #endif
 
-#define MKSH_SYSTEM_PROFILE	MKSH_ETC_LOCATION "/profile"
-#define MKSH_SUID_PROFILE	MKSH_ETC_LOCATION "/suid_profile"
+#define MKSH_SYSTEM_PROFILE	MKSH_DEFAULT_PROFILEDIR "/profile"
+#define MKSH_SUID_PROFILE	MKSH_DEFAULT_PROFILEDIR "/suid_profile"
 
 
 /* Used by v_evaluate() and setstr() to control action when error occurs */
