diff -ru opensmtpd-201306071637p1/smtpd/smtpd.h opensmtpd-201306071637p1-modified/smtpd/smtpd.h
--- opensmtpd-201306071637p1/smtpd/smtpd.h	2013-06-07 16:37:02.000000000 +0200
+++ opensmtpd-201306071637p1-modified/smtpd/smtpd.h	2013-06-11 00:54:00.218211581 +0200
@@ -1194,6 +1194,7 @@
 void mfa_filter(uint64_t, int);
 void mfa_filter_event(uint64_t, int);
 void mfa_filter_data(uint64_t, const char *);
+void mfa_filter_prepare(void);
 
 /* mproc.c */
 int mproc_fork(struct mproc *, const char*, const char *);

