From ff902cb60e1ace5ca4af09f9c9f7d76517c8e5d5 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Wed, 09 Jun 2010 17:18:15 +0000
Subject: Remove never_unload from the plugin struct

Seeing as we never want to unload them now.
---
diff --git a/src/nautilus-sendto-plugin.h b/src/nautilus-sendto-plugin.h
index 216f051..14fb7f5 100644
--- a/src/nautilus-sendto-plugin.h
+++ b/src/nautilus-sendto-plugin.h
@@ -61,7 +61,6 @@ typedef enum {
  * @id: A unique ID representing the plugin
  * @description: The label used in the plugin selection drop-down
  * @gettext_package: The domain to use to translate the description, %NULL if the plugin is part of nautilus-sendto
- * @never_unload: Whether to unload the plugin on exit. Enable this if your plugin registers a new #GType
  * @capabilities: a bitmask of #NstPluginCapabilities
  * @init: Check for dependencies, and return %FALSE if dependencies such as programs are missing.
  * @get_contacts_widget: Return the contact widget, the widget to select the destination of the files
@@ -71,13 +70,12 @@ typedef enum {
  *
  * A structure representing a nautilus-sendto plugin. You should also call NST_INIT_PLUGIN() on the plugin structure to export it.
  **/
-struct _NstPluginInfo 
+struct _NstPluginInfo
 {
 	gchar                             *icon;
 	gchar                             *id;
 	gchar                             *description;
 	gchar                             *gettext_package;
-	gboolean                           never_unload;
 	NstPluginCapabilities              capabilities;
 	gboolean (*init)                  (NstPlugin *plugin);
 	GtkWidget* (*get_contacts_widget) (NstPlugin *plugin);
diff --git a/src/plugins/bluetooth/bluetooth.c b/src/plugins/bluetooth/bluetooth.c
index f2db2dd..160619d 100644
--- a/src/plugins/bluetooth/bluetooth.c
+++ b/src/plugins/bluetooth/bluetooth.c
@@ -549,7 +549,6 @@ NstPluginInfo plugin_info = {
 	"bluetooth",
 	N_("Bluetooth (OBEX Push)"),
 	NULL,
-	FALSE,
 	NAUTILUS_CAPS_NONE,
 	init,
 	get_contacts_widget,
diff --git a/src/plugins/evolution/evolution.c b/src/plugins/evolution/evolution.c
index 8e1dd0c..7feb305 100644
--- a/src/plugins/evolution/evolution.c
+++ b/src/plugins/evolution/evolution.c
@@ -377,7 +377,6 @@ NstPluginInfo plugin_info = {
 	"evolution",
 	N_("Email"),
 	NULL,
-	FALSE,
 	NAUTILUS_CAPS_NONE,
 	init,
 	get_contacts_widget,
diff --git a/src/plugins/gajim/gajim.c b/src/plugins/gajim/gajim.c
index 1943b3d..088c69b 100644
--- a/src/plugins/gajim/gajim.c
+++ b/src/plugins/gajim/gajim.c
@@ -504,7 +504,6 @@ NstPluginInfo plugin_info = {
 	"gajim",
 	N_("Instant Message (Gajim)"),
 	NULL,
-	TRUE,
 	NAUTILUS_CAPS_NONE,
 	init,
 	get_contacts_widget,
diff --git a/src/plugins/nautilus-burn/nautilus-burn.c b/src/plugins/nautilus-burn/nautilus-burn.c
index bdc80c1..cdaa09e 100644
--- a/src/plugins/nautilus-burn/nautilus-burn.c
+++ b/src/plugins/nautilus-burn/nautilus-burn.c
@@ -177,7 +177,6 @@ NstPluginInfo plugin_info = {
 	"nautilus-burn",
 	N_("CD/DVD Creator"),
 	NULL,
-	FALSE,
 	NAUTILUS_CAPS_SEND_DIRECTORIES,
 	init,
 	get_contacts_widget,
diff --git a/src/plugins/pidgin/pidgin.c b/src/plugins/pidgin/pidgin.c
index 319fdbd..5d9b52e 100644
--- a/src/plugins/pidgin/pidgin.c
+++ b/src/plugins/pidgin/pidgin.c
@@ -463,7 +463,6 @@ NstPluginInfo plugin_info = {
 	"pidgin",
 	N_("Instant Message (Pidgin)"),
 	NULL,
-	TRUE,
 	NAUTILUS_CAPS_NONE,
 	init,
 	get_contacts_widget,
diff --git a/src/plugins/removable-devices/removable-devices.c b/src/plugins/removable-devices/removable-devices.c
index 4a735a3..5984ec7 100644
--- a/src/plugins/removable-devices/removable-devices.c
+++ b/src/plugins/removable-devices/removable-devices.c
@@ -223,7 +223,6 @@ NstPluginInfo plugin_info = {
 	"folder-remote",
 	N_("Removable disks and shares"),
 	NULL,
-	FALSE,
 	NAUTILUS_CAPS_SEND_DIRECTORIES,
 	init,
 	get_contacts_widget,
diff --git a/src/plugins/upnp/upnp.c b/src/plugins/upnp/upnp.c
index da5669e..5d2448e 100644
--- a/src/plugins/upnp/upnp.c
+++ b/src/plugins/upnp/upnp.c
@@ -308,7 +308,6 @@ NstPluginInfo plugin_info = {
 	"upnp",
 	N_("UPnP Media Server"),
 	NULL,
-	FALSE,
 	NAUTILUS_CAPS_NONE,
 	init,
 	get_contacts_widget,
--
cgit v0.8.3.1
