From 6d9294461935516991082e535a09b28d27105abe Mon Sep 17 00:00:00 2001
From: Andre Klapper <a9016009@gmx.de>
Date: Sat, 29 May 2010 01:42:30 +0000
Subject: Compile with -DGSEAL_ENABLE. Fixes bug 612493.

---
diff --git a/src/nautilus-sendto-command.c b/src/nautilus-sendto-command.c
index 094fe62..674d774 100644
--- a/src/nautilus-sendto-command.c
+++ b/src/nautilus-sendto-command.c
@@ -335,7 +335,7 @@ static void
 send_if_no_pack_cb (GtkWidget *widget, NS_ui *ui)
 {
 	if (gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (ui->pack_checkbutton))) {
-		if (GTK_WIDGET_IS_SENSITIVE (ui->pack_entry)) {
+		if (gtk_widget_is_sensitive (ui->pack_entry)) {
 			gtk_widget_grab_focus (ui->pack_entry);
 		} else {
 			gtk_widget_grab_focus (ui->pack_checkbutton);
diff --git a/src/plugins/evolution/e-contact-entry.c b/src/plugins/evolution/e-contact-entry.c
index 69ffa3d..e190290 100644
--- a/src/plugins/evolution/e-contact-entry.c
+++ b/src/plugins/evolution/e-contact-entry.c
@@ -385,7 +385,7 @@ entry_changed_cb (GtkEditable *editable, gpointer user_data)
   EContactEntry *entry;
   entry = E_CONTACT_ENTRY (editable);
 
-  if (GTK_ENTRY (editable)->text_length >= entry->priv->lookup_length) {
+  if (gtk_entry_get_text_length (GTK_ENTRY (editable)) >= entry->priv->lookup_length) {
     GList *l;
     EBookQuery *query;
 
--
cgit v0.8.3.1
