--- lib/userSpec.cpp.orig	2008-07-15 22:01:23.000000000 +0200
+++ lib/userSpec.cpp	2008-07-15 22:02:22.000000000 +0200
@@ -41,7 +41,8 @@
 
 
 /////////////////////// Std includes
-//#include <stdlib.h>
+#include <cstdlib>
+
 
 
 /////////////////////// Local includes


--- gui/isotopicPatternCalculationDlg.cpp.orig	2008-07-15 22:09:07.000000000 +0200
+++ gui/isotopicPatternCalculationDlg.cpp	2008-07-15 22:09:56.000000000 +0200
@@ -61,8 +61,10 @@
 
 
 /////////////////////// Std includes
-#include <math.h>
+#include <cmath>
 #include <algorithm>
+#include <limits>
+
 
 
 /////////////////////// Local includes
