commit 066ac461a3864a014623c8670593c67ca6aa7a76
Author: hasufell <hasufell@gentoo.org>
Date:   Thu Jun 20 20:35:34 2013 +0200

    unbundle colamd

diff --git a/extern/CMakeLists.txt b/extern/CMakeLists.txt
index 941f69e..b21cdc3 100644
--- a/extern/CMakeLists.txt
+++ b/extern/CMakeLists.txt
@@ -26,7 +26,6 @@
 # Otherwise we get warnings here that we cant fix in external projects
 remove_strict_flags()
 
-add_subdirectory(colamd)
 add_subdirectory(rangetree)
 add_subdirectory(wcwidth)
 
diff --git a/source/blenderplayer/CMakeLists.txt b/source/blenderplayer/CMakeLists.txt
index c775f7d..c0b4e2d 100644
--- a/source/blenderplayer/CMakeLists.txt
+++ b/source/blenderplayer/CMakeLists.txt
@@ -170,8 +170,6 @@ endif()
 		list(APPEND BLENDER_SORTED_LIBS extern_ceres)
 	endif()
 
-	list(APPEND BLENDER_SORTED_LIBS extern_colamd)
-
 	if(WITH_MOD_BOOLEAN)
 		list(APPEND BLENDER_SORTED_LIBS bf_intern_bsp)
 		list(APPEND BLENDER_SORTED_LIBS bf_intern_moto)
@@ -214,7 +212,7 @@ endif()
 		list(SORT REM_MSG)
 		message(STATUS "Player Skipping: (${REM_MSG})")
 	endif()
-	target_link_libraries(blenderplayer ${BLENDER_SORTED_LIBS})
+	target_link_libraries(blenderplayer ${BLENDER_SORTED_LIBS} colamd)
 
 	unset(SEARCHLIB)
 	unset(SORTLIB)
diff --git a/source/creator/CMakeLists.txt b/source/creator/CMakeLists.txt
index d5407fe..cd8106a 100644
--- a/source/creator/CMakeLists.txt
+++ b/source/creator/CMakeLists.txt
@@ -900,7 +900,6 @@ endif()
 		bf_intern_smoke
 		extern_minilzo
 		extern_lzma
-		extern_colamd
 		ge_logic_ketsji
 		extern_recastnavigation
 		ge_logic
@@ -1014,7 +1013,7 @@ endif()
 		list(SORT REM_MSG)
 		message(STATUS "Blender Skipping: (${REM_MSG})")
 	endif()
-	target_link_libraries(blender ${BLENDER_SORTED_LIBS})
+	target_link_libraries(blender ${BLENDER_SORTED_LIBS} colamd)
 
 	unset(SEARCHLIB)
 	unset(SORTLIB)
