commit 6e212fef06cf18ce57a1f613e7c7a8e9ee72662e
Author: hasufell <hasufell@gentoo.org>
Date:   Fri Jul 19 18:37:46 2013 +0200

    remove binreloc references

diff --git a/extern/CMakeLists.txt b/extern/CMakeLists.txt
index 2b40c22..2823c7a 100644
--- a/extern/CMakeLists.txt
+++ b/extern/CMakeLists.txt
@@ -41,10 +41,6 @@ endif()
 #	add_subdirectory(eltopo)
 #endif()
 
-if(WITH_BINRELOC)
-	add_subdirectory(binreloc)
-endif()
-
 if(NOT WITH_SYSTEM_GLEW)
 	add_subdirectory(glew)
 endif()
diff --git a/source/blenderplayer/CMakeLists.txt b/source/blenderplayer/CMakeLists.txt
index bc3baee..2fa78b9 100644
--- a/source/blenderplayer/CMakeLists.txt
+++ b/source/blenderplayer/CMakeLists.txt
@@ -81,10 +81,6 @@ list(APPEND BLENDER_LINK_LIBS
 	blenkernel_blc
 )
 
-if(CMAKE_SYSTEM_NAME MATCHES "Linux")
-	list(APPEND BLENDER_LINK_LIBS extern_binreloc)
-endif()
-
 # if(UNIX)
 	# Sort libraries
 	set(BLENDER_SORTED_LIBS
