http://bugs.gentoo.org/452694

--- extern/bullet2/src/BulletSoftBody/btSoftBodyInternals.h
+++ extern/bullet2/src/BulletSoftBody/btSoftBodyInternals.h
@@ -18,6 +18,7 @@
 #define _BT_SOFT_BODY_INTERNALS_H
 
 #include "btSoftBody.h"
+#include <string.h>
 
 #include "LinearMath/btQuickprof.h"
 #include "BulletCollision/BroadphaseCollision/btBroadphaseInterface.h"
@@ -171,8 +172,7 @@
 template <typename T>
 static inline void			ZeroInitialize(T& value)
 {
-	static const T	zerodummy;
-	value=zerodummy;
+	memset(&value,0,sizeof(T));
 }
 //
 template <typename T>
--- intern/memutil/MEM_Allocator.h
+++ intern/memutil/MEM_Allocator.h
@@ -24,6 +24,7 @@
 #ifndef __MEM_Allocator_h_included__
 #define __MEM_Allocator_h_included__ 1
 
+#include <cstddef>
 #include "guardedalloc/MEM_guardedalloc.h"
 #include "guardedalloc/BLO_sys_types.h"
 
