commit eefa71abd4c13b28e0c42041240754c4ec1fda6e
Author: hasufell <hasufell@gentoo.org>
Date:   Thu Jun 20 20:32:46 2013 +0200

    remove hhc.exe reference

diff --git a/doc/doxygen/Doxyfile b/doc/doxygen/Doxyfile
index 8444ea7..745ab77 100644
--- a/doc/doxygen/Doxyfile
+++ b/doc/doxygen/Doxyfile
@@ -952,7 +952,7 @@ CHM_FILE               = blender.chm
 # the HTML help compiler (hhc.exe). If non-empty doxygen will try to run 
 # the HTML help compiler on the generated index.hhp.
 
-HHC_LOCATION           = "C:/Program Files (x86)/HTML Help Workshop/hhc.exe"
+HHC_LOCATION           = 
 
 # If the GENERATE_HTMLHELP tag is set to YES, the GENERATE_CHI flag 
 # controls if a separate .chi index file is generated (YES) or that 
