From: Julian Ospald <hasufell@gentoo.org>
Date: Fri Apr 18 21:42:45 UTC 2014
Subject: fix unguarded calls to omp_get_thread_num()

--- blender-v2.70a/source/blender/editors/sculpt_paint/paint_cursor.c
+++ blender-v2.70a/source/blender/editors/sculpt_paint/paint_cursor.c
@@ -248,8 +248,13 @@ static int load_tex(Brush *br, ViewConte
 
 					if (col) {
 						float rgba[4];
-
-						paint_get_tex_pixel_col(mtex, x, y, rgba, pool, omp_get_thread_num());
+						int thread_num;
+#ifdef _OPENMP
+						thread_num = omp_get_thread_num();
+#else
+						thread_num = 0;
+#endif
+						paint_get_tex_pixel_col(mtex, x, y, rgba, pool, thread_num);
 
 						buffer[index * 4]     = rgba[0] * 255;
 						buffer[index * 4 + 1] = rgba[1] * 255;
@@ -257,7 +262,14 @@ static int load_tex(Brush *br, ViewConte
 						buffer[index * 4 + 3] = rgba[3] * 255;
 					}
 					else {
-						float avg = paint_get_tex_pixel(mtex, x, y, pool, omp_get_thread_num());
+						float avg;
+						int thread_num;
+#ifdef _OPENMP
+						thread_num = omp_get_thread_num();
+#else
+						thread_num = 0;
+#endif
+						avg = paint_get_tex_pixel(mtex, x, y, pool, thread_num);
 
 						avg += br->texture_sample_bias;
 
