--- SConstruct	2011-10-29 05:18:43.149564401 +0200
+++ SConstruct	2011-10-29 05:19:37.332887017 +0200
@@ -165,6 +165,9 @@
 if crossbuild and platform not in ('win32-vc', 'win64-vc'):
     platform = 'linuxcross'
 
+if platform == 'linux3':
+    platform = 'linux2'  # Workaround for Linux 3.x
+
 env['OURPLATFORM'] = platform
 
 configfile = os.path.join("build_files", "scons", "config", platform + "-config.py")
