--- tools/Blender.py	2011-10-29 06:32:11.706449854 +0200
+++ tools/Blender.py	2011-10-29 06:32:15.923397138 +0200
@@ -219,7 +219,10 @@
 	"""
 	build_date = time.strftime ("%Y-%m-%d")
 	build_time = time.strftime ("%H:%M:%S")
-	build_rev = os.popen('svnversion').read()[:-1] # remove \n
+	# svn 1.6 says "exported"
+	# svn 1.7 says "Unversioned directory"
+	# Latter breaks the build somewhere, former doesn't.
+	build_rev = 'exported'  
 
 	obj = []
 	if lenv['BF_BUILDINFO']:
