diff -Naur GWMetadata.old/gmds/mdextractor/GNUmakefile.preamble GWMetadata/gmds/mdextractor/GNUmakefile.preamble
--- GWMetadata.old/gmds/mdextractor/GNUmakefile.preamble	2007-02-23 14:14:52.000000000 +0100
+++ GWMetadata/gmds/mdextractor/GNUmakefile.preamble	2007-05-03 14:21:57.000000000 +0200
@@ -12,6 +12,7 @@
 
 ADDITIONAL_LIB_DIRS += -L../../MDKit/MDKit.framework/Versions/Current/$(GNUSTEP_TARGET_LDIR) 
 ADDITIONAL_LIB_DIRS += -L../../../DBKit/$(GNUSTEP_OBJ_DIR)   
+ADDITIONAL_LIB_DIRS += -Wl,-rpath-link ../../../FSNode/FSNode.framework/Versions/Current/$(GNUSTEP_TARGET_LDIR)
 
 # Additional LDFLAGS to pass to the linker
 # ADDITIONAL_LDFLAGS += 
diff -Naur GWMetadata.old/gmds/mdfind/GNUmakefile.preamble GWMetadata/gmds/mdfind/GNUmakefile.preamble
--- GWMetadata.old/gmds/mdfind/GNUmakefile.preamble	2007-02-23 14:14:50.000000000 +0100
+++ GWMetadata/gmds/mdfind/GNUmakefile.preamble	2007-05-03 14:21:57.000000000 +0200
@@ -16,6 +16,8 @@
 
 # Additional library directories the linker should search
 ADDITIONAL_LIB_DIRS += -L../../MDKit/MDKit.framework/Versions/Current/$(GNUSTEP_TARGET_LDIR) 
+ADDITIONAL_LIB_DIRS += -Wl,-rpath-link ../../../FSNode/FSNode.framework/Versions/Current/$(GNUSTEP_TARGET_LDIR)
+ADDITIONAL_LIB_DIRS += -Wl,-rpath-link ../../../DBKit/$(GNUSTEP_OBJ_DIR)
                       
 
 ADDITIONAL_TOOL_LIBS +=
diff -Naur GWMetadata.old/GNUmakefile.preamble GWMetadata/GNUmakefile.preamble
--- GWMetadata.old/GNUmakefile.preamble	2007-02-23 14:14:55.000000000 +0100
+++ GWMetadata/GNUmakefile.preamble	2007-05-03 14:21:57.000000000 +0200
@@ -15,7 +15,7 @@
 ADDITIONAL_LDFLAGS +=  
 
 # Additional library directories the linker should search
-ADDITIONAL_LIB_DIRS += 
+ADDITIONAL_LIB_DIRS += -Wl,-rpath-link ../../../FSNode/FSNode.framework/Versions/0
                       
 
 ADDITIONAL_TOOL_LIBS +=
diff -Naur GWMetadata.old/MDFinder/GNUmakefile.preamble GWMetadata/MDFinder/GNUmakefile.preamble
--- GWMetadata.old/MDFinder/GNUmakefile.preamble	2007-02-23 14:14:44.000000000 +0100
+++ GWMetadata/MDFinder/GNUmakefile.preamble	2007-05-03 14:21:57.000000000 +0200
@@ -13,6 +13,7 @@
 
 ADDITIONAL_LIB_DIRS += -L../MDKit/MDKit.framework/Versions/Current/$(GNUSTEP_TARGET_LDIR) 
 ADDITIONAL_LIB_DIRS += -L../../FSNode/FSNode.framework/Versions/Current/$(GNUSTEP_TARGET_LDIR)
+ADDITIONAL_LIB_DIRS += -Wl,-rpath-link ../../DBKit/$(GNUSTEP_OBJ_DIR)
 
 # Additional LDFLAGS to pass to the linker
 # ADDITIONAL_LDFLAGS += 
