From: Julian Ospald <hasufell@gentoo.org>
Date: Tue Jul 24 15:08:47 UTC 2012
Subject: build system

--- openbox-menu-0.3.6.7/Makefile
+++ openbox-menu-0.3.6.7/Makefile
@@ -1,9 +1,8 @@
-LIBS= `pkg-config --libs glib-2.0 gtk+-2.0 libmenu-cache`
-CFLAGS+= -Wall -g `pkg-config --cflags glib-2.0 gtk+-2.0 libmenu-cache`
+LIBS= $(shell pkg-config --libs glib-2.0 gtk+-2.0 libmenu-cache)
+CFLAGS+= $(shell pkg-config --cflags glib-2.0 gtk+-2.0 libmenu-cache)
 
-prefix= /usr
-DESTDIR ?= $(prefix)
-BINDIR= /bin/
+prefix ?= /usr
+BINDIR = $(prefix)/bin/
 
 SRC= $(shell ls *.c 2> /dev/null)
 OBJ= $(SRC:.c=.o)
