--- gui/curiositylist_stats_t.cc.old	2012-10-04 08:46:47.623621825 +0200
+++ gui/curiositylist_stats_t.cc	2012-10-04 08:47:12.429854860 +0200
@@ -151,7 +151,8 @@
 		get_unique_attractions( sortby, sortreverse );
 	}
 
-	for (uint32 i=0; i<attractions.get_count()  &&  yoff<end; i++) {
+	uint32 i;
+	for (i=0; i<attractions.get_count()  &&  yoff<end; i++) {
 		const gebaeude_t* geb = attractions[i];
 
 		int xoff = offset.x+10;
--- tpl/hashtable_tpl.h.old	2012-10-03 13:51:30.558014602 +0200
+++ tpl/hashtable_tpl.h	2012-10-03 17:18:43.988237496 +0200
@@ -48,7 +48,7 @@
 		while(iter.next()) {
 			node_t node = iter.get_current();
 
-			if(comp(node.key, key) == 0) {
+			if(this->comp(node.key, key) == 0) {
 				return node.object;
 			}
 		}
@@ -61,7 +61,7 @@
 		while(iter.next()) {
 			node_t &node = iter.access_current();
 
-			if(comp(node.key, key) == 0) {
+			if(this->comp(node.key, key) == 0) {
 				return &node.object;
 			}
 		}
@@ -84,7 +84,7 @@
 		while(iter.next()) {
 			node_t &node = iter.access_current();
 
-			if(comp(node.key, key) == 0) {
+			if(this->comp(node.key, key) == 0) {
 				// duplicate
 				return false;
 			}
@@ -111,7 +111,7 @@
 
 		while(iter.next()) {
 			node_t &node = iter.access_current();
-			if(comp(node.key, key) == 0) {
+			if(this->comp(node.key, key) == 0) {
 				value_t value = node.object;
 				node.object = object;
 				return value;
@@ -138,7 +138,7 @@
 
 		while(iter.next()) {
 			node_t node = iter.get_current();
-			if(comp(node.key, key) == 0) {
+			if(this->comp(node.key, key) == 0) {
 				bags[code].remove( node );
 
 				return node.object;
