--- Makefile.orig	2013-01-22 16:53:11.574351933 -0500
+++ Makefile	2013-01-22 16:57:38.801999641 -0500
@@ -90,7 +90,6 @@
     endif
   endif
 else
-  CFLAGS += -O
 endif
 
 ifdef DEBUG
@@ -109,6 +108,7 @@
   endif
 else
   CFLAGS += -DNDEBUG
+  CXXFLAGS += -DNDEBUG
 endif
 
 ifneq ($(PROFILE),)
@@ -121,6 +121,7 @@
 
 ifneq  ($(MULTI_THREAD),)
   CFLAGS += -DMULTI_THREAD=$(MULTI_THREAD)
+  CXXFLAGS += -DMULTI_THREAD=$(MULTI_THREAD)
   ifneq  ($(MULTI_THREAD),1)
     ifeq ($(OSTYPE),mingw)
 #use lpthreadGC2d for debug alternatively
@@ -135,10 +136,12 @@
   REV = $(shell svnversion)
   ifneq ($(REV),)
     CFLAGS  += -DREVISION="$(REV)"
+    CXXFLAGS  += -DREVISION="$(REV)"
   endif
 endif
 
 CFLAGS   += -Wall -W -Wcast-qual -Wpointer-arith -Wcast-align $(FLAGS)
+CXXFLAGS   += -Wall -W -Wcast-qual -Wpointer-arith -Wcast-align $(FLAGS)
 CCFLAGS  += -Wstrict-prototypes
 
 
@@ -450,6 +453,7 @@
 ifeq ($(BACKEND),sdl)
   SOURCES += simsys_s.cc
   CFLAGS  += -DUSE_16BIT_DIB
+  CXXFLAGS  += -DUSE_16BIT_DIB
   ifeq ($(OSTYPE),mac)
     # Core Audio (Quicktime) base sound system routines
     SOURCES += sound/core-audio_sound.mm
@@ -482,6 +486,7 @@
     endif
   endif
   CFLAGS += $(SDL_CFLAGS)
+  CXXFLAGS += $(SDL_CFLAGS)
   LIBS   += $(SDL_LDFLAGS)
 endif
 
@@ -491,6 +496,7 @@
   SOURCES += sound/sdl_mixer_sound.cc
   SOURCES += music/sdl_midi.cc
   CFLAGS  += -DUSE_16BIT_DIB
+  CXXFLAGS  += -DUSE_16BIT_DIB
   ifeq ($(SDL_CONFIG),)
     SDL_CFLAGS  := -I$(MINGDIR)/include/SDL -Dmain=SDL_main
     SDL_LDFLAGS := -lmingw32 -lSDLmain -lSDL
@@ -505,12 +511,14 @@
     endif
   endif
   CFLAGS += $(SDL_CFLAGS)
+  CXXFLAGS += $(SDL_CFLAGS)
   LIBS   += $(SDL_LDFLAGS) -lSDL_mixer
 endif
 
 ifeq ($(BACKEND),opengl)
   SOURCES += simsys_opengl.cc
   CFLAGS  += -DUSE_16BIT_DIB
+  CXXFLAGS  += -DUSE_16BIT_DIB
   ifeq ($(OSTYPE),mac)
     # Core Audio (Quicktime) base sound system routines
     SOURCES += sound/core-audio_sound.mm
@@ -538,6 +546,7 @@
     endif
   endif
   CFLAGS += $(SDL_CFLAGS)
+  CXXFLAGS += $(SDL_CFLAGS)
   LIBS   += $(SDL_LDFLAGS)
   ifeq ($(OSTYPE),mingw)
     LIBS += -lopengl32
@@ -553,6 +562,7 @@
 endif
 
 CFLAGS += -DCOLOUR_DEPTH=$(COLOUR_DEPTH)
+CXXFLAGS += -DCOLOUR_DEPTH=$(COLOUR_DEPTH)
 
 ifneq ($(findstring $(OSTYPE), cygwin mingw),)
   SOURCES += simres.rc
@@ -560,7 +570,6 @@
 endif
 
 CCFLAGS  += $(CFLAGS)
-CXXFLAGS += $(CFLAGS)
 
 BUILDDIR ?= build/$(CFG)
 PROGDIR  ?= $(BUILDDIR)
