--- src/common/classes/alloc.h.old	2009-07-09 13:04:14.000000000 +0200
+++ src/common/classes/alloc.h	2009-07-09 13:05:37.000000000 +0200
@@ -48,6 +48,8 @@
 					   stdlib.h (EKU) */
 #endif
 
+#include <new>
+
 #ifdef _MSC_VER
 #define THROW_BAD_ALLOC
 #else
@@ -448,15 +450,6 @@
 	);
 }
 
-inline void* operator new(size_t, void* ptr) throw() 
-{
-	return ptr;
-}
-inline void* operator new[](size_t, void* ptr) throw() 
-{
-	return ptr;
-}
-
 inline void operator delete(void* mem) throw()
 {
 	Firebird::MemoryPool::globalFree(mem);
