diff -Naur Firebird-2.1.2.18118-0_orig/builds/posix/Makefile.in.firebird Firebird-2.1.2.18118-0/builds/posix/Makefile.in.firebird
--- Firebird-2.1.2.18118-0_orig/builds/posix/Makefile.in.firebird	2008-12-10 17:22:15.000000000 +0200
+++ Firebird-2.1.2.18118-0/builds/posix/Makefile.in.firebird	2009-05-20 15:13:47.673202327 +0300
@@ -185,14 +185,7 @@
 # 
 
 external_libraries:
-ifeq ($(STD_ICU),false)
-	$(MAKE) -f $(GEN_ROOT)/Makefile.extern.icu
-endif
-ifeq ($(STD_EDITLINE),false)
-ifeq ($(EDITLINE_FLG),Y)
-	$(MAKE) -f $(GEN_ROOT)/Makefile.extern.editline
-endif
-endif
+
 
 btyacc_binary: $(BTYACC)
 
diff -Naur Firebird-2.1.2.18118-0_orig/builds/posix/make.defaults Firebird-2.1.2.18118-0/builds/posix/make.defaults
--- Firebird-2.1.2.18118-0_orig/builds/posix/make.defaults	2008-09-04 14:47:18.000000000 +0300
+++ Firebird-2.1.2.18118-0/builds/posix/make.defaults	2009-05-20 15:15:31.088202976 +0300
@@ -232,11 +232,7 @@
 LIBFBSTATIC_A = $(LIB)/libfbstatic.a
 
 ifeq ($(EDITLINE_FLG),Y)
-  ifeq ($(STD_EDITLINE), true)
-	LIBEDITLINE := -leditline
-  else
-	LIBEDITLINE := $(LIB)/libeditline.a
-  endif
+  LIBEDITLINE := -ledit
 endif
 
 LIBICU_SO = $(LIB)/$(LIB_PREFIX)icuuc.$(SHRLIB_EXT) \
diff -Naur Firebird-2.1.2.18118-0_orig/builds/posix/make.rules Firebird-2.1.2.18118-0/builds/posix/make.rules
--- Firebird-2.1.2.18118-0_orig/builds/posix/make.rules	2008-12-10 03:03:31.000000000 +0200
+++ Firebird-2.1.2.18118-0/builds/posix/make.rules	2009-05-20 15:21:04.114225351 +0300
@@ -33,9 +33,6 @@
 
 # Please don't use compiler/platform specific flags here - nmcc 02-Nov-2002
 CFLAGS:= $(CFLAGS) -I$(SRC_ROOT)/include/gen -I$(SRC_ROOT)/include -I$(SRC_ROOT)/vulcan -DNAMESPACE=Vulcan
-ifeq ($(STD_ICU),false)
-  CFLAGS:= $(CFLAGS) -I$(ROOT)/extern/icu/source/common -I$(ROOT)/extern/icu/source/i18n
-endif
 
 ifeq ($(IsProdTypeBuild),Y)
   CFLAGS:= $(CFLAGS) $(PROD_FLAGS) -DPROD_BUILD
@@ -69,7 +66,7 @@
 
 .SUFFIXES: .c .e .epp .cpp
 
-.e.c:
+.e.c: $(GPRE_CURRENT)
 	$(GPRE_CURRENT) $(GPRE_FLAGS) $< $@
 
 
diff -Naur Firebird-2.1.2.18118-0_orig/builds/posix/make.shared.targets Firebird-2.1.2.18118-0/builds/posix/make.shared.targets
--- Firebird-2.1.2.18118-0_orig/builds/posix/make.shared.targets	2008-12-10 03:03:31.000000000 +0200
+++ Firebird-2.1.2.18118-0/builds/posix/make.shared.targets	2009-05-20 15:21:59.299203496 +0300
@@ -49,7 +49,7 @@
 
 # gpre_meta needs a special boot build since there is no database.
 
-$(SRC_ROOT)/gpre/gpre_meta.cpp:  $(SRC_ROOT)/gpre/gpre_meta.epp
+$(SRC_ROOT)/gpre/gpre_meta.cpp:  $(SRC_ROOT)/gpre/gpre_meta.epp $(GPRE_BOOT)
 	$(GPRE_BOOT) -lang_internal $(GPRE_FLAGS) $< $@
 
 
diff -Naur Firebird-2.1.2.18118-0_orig/builds/posix/prefix.linux Firebird-2.1.2.18118-0/builds/posix/prefix.linux
--- Firebird-2.1.2.18118-0_orig/builds/posix/prefix.linux	2007-05-03 14:24:50.000000000 +0300
+++ Firebird-2.1.2.18118-0/builds/posix/prefix.linux	2009-05-20 15:23:17.286523276 +0300
@@ -20,12 +20,10 @@
 
 # -fno-builtin is used because GCC 3.0-3.2.2 had bug with builtins expansion
 # you may remove it if engine is getting compiled with any other GCC version
-
-COMMON_FLAGS=-ggdb -DFB_SEND_FLAGS=MSG_NOSIGNAL -DLINUX -pipe -MMD -fPIC -fmessage-length=0
-OPTIMIZE_FLAGS=-O3 -march=i586 -mcpu=i686 -fno-omit-frame-pointer -fno-builtin
+COMMON_FLAGS=-ggdb -DLINUX -pipe -MMD -fPIC
 WARN_FLAGS=-Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable
 
-PROD_FLAGS=-DNDEBUG $(COMMON_FLAGS) $(OPTIMIZE_FLAGS)
+PROD_FLAGS=-DNDEBUG $(COMMON_FLAGS)
 DEV_FLAGS=-DUSE_VALGRIND -p $(COMMON_FLAGS) $(WARN_FLAGS)
 #DEV_FLAGS=-p $(COMMON_FLAGS) $(WARN_FLAGS)
 
diff -Naur Firebird-2.1.2.18118-0_orig/configure Firebird-2.1.2.18118-0/configure
--- Firebird-2.1.2.18118-0_orig/configure	2009-03-30 09:42:44.000000000 +0300
+++ Firebird-2.1.2.18118-0/configure	2009-05-20 15:26:30.856202564 +0300
@@ -4027,13 +4027,13 @@
 # not need editline in default libs, but need to test for its presence
 saveLIBS=$LIBS
 if test "$STD_EDITLINE" = "true"; then
-  { echo "$as_me:$LINENO: checking for readline in -leditline" >&5
-echo $ECHO_N "checking for readline in -leditline... $ECHO_C" >&6; }
+  { echo "$as_me:$LINENO: checking for readline in -ledit" >&5
+echo $ECHO_N "checking for readline in -ledit... $ECHO_C" >&6; }
 if test "${ac_cv_lib_editline_readline+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-leditline  $LIBS"
+LIBS="-ledit  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
