Index: firebird-2.0.0.12748/builds/posix/make.shared.targets
===================================================================
--- firebird-2.0.0.12748.orig/builds/posix/make.shared.targets
+++ firebird-2.0.0.12748/builds/posix/make.shared.targets
@@ -57,7 +57,7 @@ $(SRC_ROOT)/dsql/parse.cpp: $(SRC_ROOT)/
 
 # gpre_meta needs a special boot build since there is no database.
 
-$(SRC_ROOT)/gpre/gpre_meta.cpp:  $(SRC_ROOT)/gpre/gpre_meta.epp
+$(SRC_ROOT)/gpre/gpre_meta.cpp:  $(SRC_ROOT)/gpre/gpre_meta.epp $(GPRE_BOOT)
 	$(GPRE_BOOT) -lang_internal $(GPRE_FLAGS) $< $@
 
 
Index: firebird-2.0.0.12748/builds/posix/make.rules
===================================================================
--- firebird-2.0.0.12748.orig/builds/posix/make.rules
+++ firebird-2.0.0.12748/builds/posix/make.rules
@@ -103,19 +103,19 @@ DSQL_GPRE_FLAGS =  -lang_internal -r -m 
 
 .SUFFIXES: .c .e .epp .cpp
 
-.e.c:
+.e.c: $(GPRE_CURRENT)
 	$(GPRE_CURRENT) $(GPRE_FLAGS) $< $@
 
 
-.epp.cpp:
+.epp.cpp: $(GPRE_CURRENT)
 	$(GPRE_CURRENT) $(GPRE_FLAGS) $< $@
 
 
-$(SRC_ROOT)/jrd/%.cpp:: $(SRC_ROOT)/jrd/%.epp 
+$(SRC_ROOT)/jrd/%.cpp:: $(SRC_ROOT)/jrd/%.epp  $(GPRE_CURRENT)
 	$(GPRE_CURRENT) $(JRD_GPRE_FLAGS) $< $@
 
 
-$(SRC_ROOT)/dsql/%.cpp:: $(SRC_ROOT)/dsql/%.epp 
+$(SRC_ROOT)/dsql/%.cpp:: $(SRC_ROOT)/dsql/%.epp  $(GPRE_CURRENT)
 	$(GPRE_CURRENT) $(DSQL_GPRE_FLAGS) $< $@
 
 
