diff -Naur Firebird-2.0.3.12981-0_orig/builds/posix/prefix.linux Firebird-2.0.3.12981-0/builds/posix/prefix.linux
--- Firebird-2.0.3.12981-0_orig/builds/posix/prefix.linux	2007-11-05 21:07:48.000000000 -0500
+++ Firebird-2.0.3.12981-0/builds/posix/prefix.linux	2007-11-05 21:23:36.000000000 -0500
@@ -20,7 +20,7 @@
 
 # -fno-builtin is used because GCC 3.0-3.2.2 had bug with builtins expansion
 # you may remove it if engine is getting compiled with any other GCC version
-PROD_FLAGS=-ggdb -O3 -march=i586 -mcpu=i686 -fno-omit-frame-pointer -fno-builtin -DNDEBUG -DLINUX -pipe -MMD -fPIC -fmessage-length=0
+PROD_FLAGS=-ggdb -DNDEBUG -DLINUX -pipe -MMD -fPIC
 DEV_FLAGS=-DUSE_VALGRIND -ggdb -DLINUX -pipe -MMD -p -fPIC -Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable -fmessage-length=0
 #DEV_FLAGS=-ggdb -DLINUX -pipe -MMD -p -fPIC -Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable -fmessage-length=0
 
