--- src.old/common/classes/alloc.cpp 2009-08-20 09:41:15.000000000 -0700
+++ src/common/classes/alloc.cpp   2009-08-19 21:47:10.000000000 -0700
@@ -42,7 +42,11 @@

 #ifdef USE_VALGRIND
 #include <valgrind/memcheck.h>
-#endif
+#ifndef VALGRIND_MAKE_WRITABLE // VALGRIND 3.3
+#define VALGRIND_MAKE_WRITABLE VALGRIND_MAKE_MEM_UNDEFINED
+#define VALGRIND_MAKE_NOACCESS VALGRIND_MAKE_MEM_NOACCESS
+#endif  // VALGRIND 3.3
+#endif  // USE_VALGRIND

 // Fill blocks with patterns
 #define FREE_PATTERN 0xDEADBEEF

