Index: firebird-2.0.0.12748/builds/posix/Makefile.in.firebird
===================================================================
--- firebird-2.0.0.12748.orig/builds/posix/Makefile.in.firebird
+++ firebird-2.0.0.12748/builds/posix/Makefile.in.firebird
@@ -186,10 +186,10 @@ $(GEN_ROOT)/Make.Version : $(SRC_ROOT)/m
 # 
 
 external_libraries:
-	$(MAKE) -f $(GEN_ROOT)/Makefile.extern.icu
-ifeq ($(EDITLINE_FLG),Y)
-	$(MAKE) -f $(GEN_ROOT)/Makefile.extern.editline
-endif
+#	$(MAKE) -f $(GEN_ROOT)/Makefile.extern.icu
+#ifeq ($(EDITLINE_FLG),Y)
+#	$(MAKE) -f $(GEN_ROOT)/Makefile.extern.editline
+#endif
 
 #---------------------------------------------------------------------------
 # Phase1: make a gpre_static from scratch so we can parse .e and .epp files
Index: firebird-2.0.0.12748/configure.in
===================================================================
--- firebird-2.0.0.12748.orig/configure.in
+++ firebird-2.0.0.12748/configure.in
@@ -445,7 +445,7 @@
			AC_CHECK_LIB(termcap, tgetent, , \
				AC_CHECK_LIB(tinfo, tgetent, , \
					AC_MSG_ERROR(termcap support not found)))))  
-	AC_SUBST(EDITLINE_A, -leditline)
+	AC_SUBST(EDITLINE_A, -ledit)
 fi
 AC_CHECK_LIB(stdc++, main,,, $LIBS)
 
Index: firebird-2.0.0.12748/builds/posix/make.rules
===================================================================
--- firebird-2.0.0.12748.orig/builds/posix/make.rules
+++ firebird-2.0.0.12748/builds/posix/make.rules
@@ -33,7 +33,7 @@
 
 
 # Please don't use compiler/platform specific flags here - nmcc 02-Nov-2002
-CFLAGS:= $(CFLAGS) -I$(SRC_ROOT)/include/gen -I$(SRC_ROOT)/include -I$(SRC_ROOT)/vulcan -I$(ROOT)/extern/icu/source/common -I$(ROOT)/extern/icu/source/i18n -DNAMESPACE=Vulcan
+CFLAGS:= $(CFLAGS) -I$(SRC_ROOT)/include/gen -I$(SRC_ROOT)/include -I$(SRC_ROOT)/vulcan -DNAMESPACE=Vulcan
 
 ifeq ($(IsProdTypeBuild),Y)
   CFLAGS:= $(CFLAGS) $(PROD_FLAGS) -DPROD_BUILD
