diff -ur cutecom-0.20.0.orig/CMakeLists.txt cutecom-0.20.0/CMakeLists.txt
--- cutecom-0.20.0.orig/CMakeLists.txt	2008-03-12 21:09:49.000000000 +0000
+++ cutecom-0.20.0/CMakeLists.txt	2008-08-15 18:52:27.000000000 +0000
@@ -17,7 +17,7 @@
 
 install(TARGETS cutecom DESTINATION bin )
 
-install(FILES cutecom.1 DESTINATION man/man1 )
+install(FILES cutecom.1 DESTINATION share/man/man1 )
 
 find_program(KDECONFIG_EXECUTABLE NAMES kde-config )
 
