--- config.lib.orig	2014-09-03 23:16:49.535086210 -0400
+++ config.lib	2014-09-03 23:18:12.922857589 -0400
@@ -1450,7 +1450,6 @@
 			LDFLAGS="$LDFLAGS -noixemul"
 		fi
 
-		CFLAGS="-O2 -fomit-frame-pointer $CFLAGS"
 	else
 		OBJS_SUBDIR="debug"
 
@@ -1461,14 +1460,6 @@
 				CFLAGS="$CFLAGS -G0"
 			fi
 		fi
-		if [ $enable_debug -ge 2 ]; then
-			CFLAGS="$CFLAGS -fno-inline"
-		fi
-		if [ $enable_debug -ge 3 ]; then
-			CFLAGS="$CFLAGS -O0"
-		else
-			CFLAGS="$CFLAGS -O2"
-		fi
 	fi
 
 	if [ $enable_debug -le 2 ]; then
@@ -1493,11 +1484,6 @@
 		CFLAGS="$CFLAGS -fno-expensive-optimizations"
 	fi
 
-	if [ "$enable_profiling" != "0" ]; then
-		CFLAGS="$CFLAGS -p"
-		LDFLAGS="$LDFLAGS -pg"
-	fi
-
 	if [ "$with_threads" = "0" ]; then
 		CFLAGS="$CFLAGS -DNO_THREADS"
 	fi
