diff -Nur java-access-bridge-1.6.0/at-client/Makefile.am java-access-bridge-1.6.0_patched/at-client/Makefile.am
--- java-access-bridge-1.6.0/at-client/Makefile.am	2003-08-25 12:17:11.000000000 +0300
+++ java-access-bridge-1.6.0_patched/at-client/Makefile.am	2009-01-28 21:05:46.000000000 +0200
@@ -9,7 +9,7 @@
 
 ../gnome-java-bridge.jar : jar-stamp $(JAVA_FILES) $(DEP_CLASSES) 
 	touch ../jar-stamp
-	$(JAR) uf ../gnome-java-bridge.jar org/GNOME/Accessibility/*.class
+	$(JAR) uf ../gnome-java-bridge.jar org/GNOME/Accessibility/
 
 $(DEP_CLASSES) : 
 	@(cd `dirname $@`; $(MAKE) )
diff -Nur java-access-bridge-1.6.0/bridge/Makefile.am java-access-bridge-1.6.0_patched/bridge/Makefile.am
--- java-access-bridge-1.6.0/bridge/Makefile.am	2003-08-25 12:17:11.000000000 +0300
+++ java-access-bridge-1.6.0_patched/bridge/Makefile.am	2009-01-28 21:05:57.000000000 +0200
@@ -15,7 +15,7 @@
 
 ../gnome-java-bridge.jar : jar-stamp $(JAVA_FILES) $(DEP_CLASSES) 
 	touch ../jar-stamp
-	$(JAR) uf ../gnome-java-bridge.jar org/GNOME/Accessibility/*.class
+	$(JAR) uf ../gnome-java-bridge.jar org/GNOME/Accessibility/
 
 $(DEP_CLASSES) : 
 	@(cd `dirname $@`; $(MAKE) )
diff -Nur java-access-bridge-1.6.0/idlgen/Makefile.am java-access-bridge-1.6.0_patched/idlgen/Makefile.am
--- java-access-bridge-1.6.0/idlgen/Makefile.am	2003-05-06 11:51:38.000000000 +0300
+++ java-access-bridge-1.6.0_patched/idlgen/Makefile.am	2009-01-28 21:07:12.000000000 +0200
@@ -16,7 +16,7 @@
 
 ../gnome-java-bridge.jar : org/GNOME/Accessibility/Accessible.class $(CLASSES) ../jar-stamp
 	touch ../jar-stamp
-	$(JAR) cf ../gnome-java-bridge.jar org/GNOME/Bonobo/*.class org/GNOME/Accessibility/*.class
+	$(JAR) cf ../gnome-java-bridge.jar org/GNOME/Bonobo/ org/GNOME/Accessibility/
 
 all-local: org/GNOME/Accessibility/Accessible.class $(JARFILES)
 
diff -Nur java-access-bridge-1.6.0/util/Makefile.am java-access-bridge-1.6.0_patched/util/Makefile.am
--- java-access-bridge-1.6.0/util/Makefile.am	2003-08-25 12:17:14.000000000 +0300
+++ java-access-bridge-1.6.0_patched/util/Makefile.am	2009-01-28 21:08:22.000000000 +0200
@@ -9,7 +9,7 @@
 
 ../gnome-java-bridge.jar : jar-stamp $(JAVA_FILES) $(DEP_CLASSES) 
 	touch ../jar-stamp
-	$(JAR) uf ../gnome-java-bridge.jar org/GNOME/Accessibility/*.class
+	$(JAR) uf ../gnome-java-bridge.jar org/GNOME/Accessibility/
 
 $(DEP_CLASSES) : 
 	@(cd `dirname $@`; $(MAKE) )
