http://sourceware.org/ml/binutils/2005-12/msg00008.html
--- hints/solaris_2.sh
+++ hints/solaris_2.sh
@@ -289,7 +289,7 @@
 
 END
 		ccdlflags="$ccdlflags -Wl,-E"
-		lddlflags="$lddlflags -Wl,-E -G"
+		lddlflags="$lddlflags -shared"
 	    fi
 	fi
 
@@ -555,7 +555,7 @@
 		    : #ccflags="$ccflags -Wa,`getconf XBS5_LP64_OFF64_CFLAGS 2>/dev/null`"
 		fi
 		ldflags="$ldflags -m64"
-		lddlflags="$lddlflags -G -m64"
+		lddlflags="$lddlflags -m64"
 		;;
 	    *)
 		ccflags="$ccflags `getconf XBS5_LP64_OFF64_CFLAGS 2>/dev/null`"
