--- lib/Pod/Perldoc.pm.orig	2003-10-22 13:02:15.000000000 -0700
+++ lib/Pod/Perldoc.pm	2003-10-22 13:02:36.000000000 -0700
@@ -1513,6 +1513,7 @@
     $self->{'target'} = (splitdir $s)[-1];  # XXX: why not use File::Basename?
     for ($i=0; $i<@dirs; $i++) {
 	$dir = $dirs[$i];
+	next unless -d $dir;
 	($dir = VMS::Filespec::unixpath($dir)) =~ s!/\z!! if IS_VMS;
 	if (       (! $self->opt_m && ( $ret = $self->check_file($dir,"$s.pod")))
 		or ( $ret = $self->check_file($dir,"$s.pm"))
