--- perl-5.8.8/ext/B/B/C.pm.orig	2007-06-07 21:12:04.000000000 +0300
+++ perl-5.8.8/ext/B/B/C.pm	2007-06-07 21:12:21.000000000 +0300
@@ -647,7 +647,7 @@
     return $sym if defined $sym;
     my $val= $sv->NVX;
     $val .= '.00' if $val =~ /^-?\d+$/;
-    $xpvnvsect->add(sprintf("0, 0, 0, %d, %s", $sv->IVX, $val));
+    $xpvnvsect->add(sprintf("0, 0, 0, %d, %s", $sv->NVX, $val));
     $svsect->add(sprintf("&xpvnv_list[%d], %lu, 0x%x",
 			 $xpvnvsect->index, $sv->REFCNT , $sv->FLAGS));
     return savesym($sv, sprintf("&sv_list[%d]", $svsect->index));
