--- Configure.orig	2009-11-04 11:09:08.934965499 +0100
+++ Configure	2009-11-04 11:13:19.055305854 +0100
@@ -7617,7 +7617,7 @@
 			cat <<EOM
 You appear to have ELF support.  I'll use $cc to build dynamic libraries.
 EOM
-			dflt="$cc"
+			dflt="$cc $optimize"
 		else
 			echo "I'll use ld to build dynamic libraries."
 			dflt='ld'
--- Makefile.SH.orig	2009-11-04 15:34:22.643322823 +0100
+++ Makefile.SH	2009-11-04 15:34:30.110147884 +0100
@@ -201,7 +201,7 @@
 LD = $ld
 
 LDFLAGS = $ldflags
-CLDFLAGS = $ldflags
+CLDFLAGS = \$(OPTIMIZE) $ldflags
 
 mallocsrc = $mallocsrc
 mallocobj = $mallocobj
--- x2p/Makefile.SH.orig	2009-11-04 15:54:14.146759212 +0100
+++ x2p/Makefile.SH	2009-11-04 15:54:19.963253154 +0100
@@ -107,7 +107,7 @@
 	$(REALPERL) -I../lib ../utils/perlcc -I .. -L .. $(plextract) -v -log ../compilelog;  
 
 a2p$(EXE_EXT): $(obj) a2p$(OBJ_EXT)
-	$(CC) -o a2p $(LDFLAGS) $(obj) a2p$(OBJ_EXT) $(libs)
+	$(CC) -o a2p $(OPTIMIZE) $(LDFLAGS) $(obj) a2p$(OBJ_EXT) $(libs)
 
 # I now supply a2p.c with the kits, so the following section is
 # used only if you force byacc to run by saying
