--- perl-5.8.0-RC2/utils/h2ph_patched.pix	2002-06-24 12:52:31.000000000 +0200
+++ perl-5.8.0-RC2/utils/h2ph_patched	2002-06-24 12:54:32.000000000 +0200
@@ -39,6 +39,8 @@
 my ($t, $tab, %curargs, $new, $eval_index, $dir, $name, $args, $outfile);
 my ($incl, $next);
 while (defined (my $file = next_file())) {
+    next if $file eq 'machine/ansi.h';
+
     if (-l $file and -d $file) {
         link_if_possible($file) if ($opt_l);
         next;
