--- a/Configure
+++ b/Configure
@@ -1255,12 +1255,12 @@ libnames=''
 : change the next line if compiling for Xenix/286 on Xenix/386
 xlibpth='/usr/lib/386 /lib/386'
 : Possible local library directories to search.
-loclibpth="/usr/local/lib /opt/local/lib /usr/gnu/lib"
-loclibpth="$loclibpth /opt/gnu/lib /usr/GNU/lib /opt/GNU/lib"
+loclibpth="/usr/local/lib32 /opt/local/lib32 /usr/gnu/lib32"
+loclibpth="$loclibpth /opt/gnu/lib32 /usr/GNU/lib32 /opt/GNU/lib32"
 
 : general looking path for locating libraries
-glibpth="/lib /usr/lib $xlibpth"
-glibpth="$glibpth /usr/ccs/lib /usr/ucblib /usr/local/lib"
+glibpth="/lib32 /usr/lib32 $xlibpth"
+glibpth="$glibpth /usr/ccs/lib32 /usr/ucblib /usr/local/lib32"
 test -f /usr/shlib/libc.so && glibpth="/usr/shlib $glibpth"
 test -f /shlib/libc.so     && glibpth="/shlib $glibpth"
 
@@ -5945,8 +5945,8 @@ fi
 : Reproduce behavior of 5.005 and earlier, maybe drop that in 5.7.
 case "$installstyle" in
 '')	case "$prefix" in
-		*perl*) dflt='lib';;
-		*) dflt='lib/perl5' ;;
+		*perl*) dflt='lib32';;
+		*) dflt='lib32/perl5' ;;
 	esac
 	;;
 *)	dflt="$installstyle" ;;
@@ -5962,8 +5962,8 @@ installstyle=$dflt
 : /opt/perl/lib/perl5... would be redundant.
 : The default "style" setting is made in installstyle.U
 case "$installstyle" in
-*lib/perl5*) set dflt privlib lib/$package/$version ;;
-*)	 set dflt privlib lib/$version ;;
+*lib32/perl5*) set dflt privlib lib32/$package/$version ;;
+*)	 set dflt privlib lib32/$version ;;
 esac
 eval $prefixit
 $cat <<EOM
@@ -6465,8 +6465,8 @@ siteprefixexp="$ansexp"
 prog=`echo $package | $sed 's/-*[0-9.]*$//'`
 case "$sitelib" in
 '') case "$installstyle" in
-	*lib/perl5*) dflt=$siteprefix/lib/$package/site_$prog/$version ;;
-	*)	 dflt=$siteprefix/lib/site_$prog/$version ;;
+	*lib32/perl5*) dflt=$siteprefix/lib32/$package/site_$prog/$version ;;
+	*)	 dflt=$siteprefix/lib32/site_$prog/$version ;;
 	esac
 	;;
 *)	dflt="$sitelib"
@@ -6592,8 +6592,8 @@ case "$vendorprefix" in
 	'')
 		prog=`echo $package | $sed 's/-*[0-9.]*$//'`
 		case "$installstyle" in
-		*lib/perl5*) dflt=$vendorprefix/lib/$package/vendor_$prog/$version ;;
-		*)	     dflt=$vendorprefix/lib/vendor_$prog/$version ;;
+		*lib32/perl5*) dflt=$vendorprefix/lib32/$package/vendor_$prog/$version ;;
+		*)	     dflt=$vendorprefix/lib32/vendor_$prog/$version ;;
 		esac
 		;;
 	*)	dflt="$vendorlib"
@@ -10550,9 +10550,9 @@ else
 		else
 			for net in net socket
 			do
-				if test -f /usr/lib/lib$net$_a; then
-					( ($nm $nm_opt /usr/lib/lib$net$_a | eval $nm_extract) ||  \
-					$ar t /usr/lib/lib$net$_a) 2>/dev/null >> libc.list
+				if test -f /usr/lib32/lib$net$_a; then
+					( ($nm $nm_opt /usr/lib32/lib$net$_a | eval $nm_extract) ||  \
+					$ar t /usr/lib32/lib$net$_a) 2>/dev/null >> libc.list
 					if $contains socket libc.list >/dev/null 2>&1; then
 						d_socket="$define"
 						socketlib="-l$net"
