diff -Nutr mock-0.9.10.old/py/mock.py mock-0.9.10/py/mock.py
--- mock-0.9.10.old/py/mock.py  2008-05-20 16:41:20.000000000 +0200
+++ mock-0.9.10/py/mock.py  2009-03-27 17:40:15.223434449 +0100
@@ -522,7 +522,7 @@
         try:
             chroot._mountall()
             cmd = ' '.join(args)
-            status = os.system("PS1='mock-chroot> ' /usr/sbin/chroot %s %s" % (chroot.makeChrootPath(), cmd))
+            status = os.system("PS1='mock-chroot> ' /usr/bin/chroot %s %s" % (chroot.makeChrootPath(), cmd))
             ret['exitStatus'] = os.WEXITSTATUS(status)
 
         finally:

