diff -Nrbu kbd-1.12/man/man8/setfont.8.in kbd-1.12-OK/man/man8/setfont.8.in
--- kbd-1.12/man/man8/setfont.8.in	2004-01-03 07:15:59.000000000 +0300
+++ kbd-1.12-OK/man/man8/setfont.8.in	2004-11-26 19:20:15.000000000 +0300
@@ -50,6 +50,10 @@
 Load a given font (here a 512-glyph font combining several
 character sets) and indicate that one's local character set
 is ISO 8859-2.
+.LP
+Note: if a font has more than 256 glyphs, only 8 out of 16 colors
+can be used simultaneously. It can make console perception worse
+(loss of intensity and even some colors).
 
 .SH "FONT FORMATS"
 The standard Linux font format is the PSF font.
