Output terminal control sequences only when output is a terminal.

--- kbd-1.08/src/unicode_start	2002-11-22 19:23:00.000000000 -0500
+++ kbd-1.08/src/unicode_start	2002-11-22 19:24:34.000000000 -0500
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 # Enables Unicode processing in the current console.
 #
 # 1. The input side: the keyboard driver.
@@ -25,7 +25,9 @@
 # Tell the console output driver that the bytes arriving are UTF-8
 # encoded multibyte sequences.
 
-echo -n -e '\033%G'
+if test -t 1 -a -t 2 ; then
+	echo -n -e '\033%G'
+fi
 
 # Tell the graphics card how to display Unicode characters not
 # contained in the IBM 437 character set (on PCs). The font should
--- kbd-1.08/src/unicode_stop	2002-11-22 19:23:30.000000000 -0500
+++ kbd-1.08/src/unicode_stop	2002-11-22 19:24:15.000000000 -0500
@@ -1,4 +1,6 @@
 #!/bin/sh
 # stop unicode
 kbd_mode -a
-echo -n -e '\033%@'
+if test -t ; then
+	echo -n -e '\033%@'
+fi
