Index: ezbounce-1.99.14/src/irc/server_info.cc
===================================================================
--- ezbounce-1.99.14.orig/src/irc/server_info.cc
+++ ezbounce-1.99.14/src/irc/server_info.cc
@@ -131,7 +131,7 @@ int server_info::parse_numeric(const tex
 /* static */ int server_info::get_005_token(const char * text,
 			const char * token, char * buf, int bufflen)
 {
-        char * tok = strstr(text, token);
+        const char * tok = strstr(text, token);
 	int c = 0;
 	if (!tok) {
 	        return 0;
Index: ezbounce-1.99.14/src/util/tokenizer.cc
===================================================================
--- ezbounce-1.99.14.orig/src/util/tokenizer.cc
+++ ezbounce-1.99.14/src/util/tokenizer.cc
@@ -45,7 +45,7 @@ char * modifying_tokenizer::next()
 {
 	int state = 0;
 	char * p;
-	char * d;
+	const char * d;
 
 	if (end) {
 		if (end_char == 0)
