diff -ur a/Makefile.in b/Makefile.in
--- a/Makefile.in	2008-04-09 23:12:05.000000000 -0500
+++ b/Makefile.in	2009-01-25 16:25:43.000000000 -0600
@@ -91,7 +91,7 @@
 
 $(EXE):		$(OBJECTS)
 	@echo "  [LINK]    " $@ "    (options: $(CXX_LINKER_FLAGS) $(CXX_LIBS))"
-	@$(CXX_LINKER) -o $@ $(CXX_LINKER_FLAGS) $(CXX_LIBS) $(CXX_OPTIMIZATIONS) $(OBJECTS)
+	@$(CXX_LINKER) -o $@ $(CXX_LINKER_FLAGS) $(CXX_OPTIMIZATIONS) $(OBJECTS) $(CXX_LIBS)
 
 $(SYMLINK):	$(EXE)
 	@ln -sf $(EXE) $(SYMLINK)
