http://bugs.gentoo.org/251445

--- include/fs/file_system.h
+++ include/fs/file_system.h
@@ -12,6 +12,7 @@
 
 #include <vector>
 #include <cstring>
+#include <limits.h>
 #include "util/hash.h"
 #include "fs/fwd.h"
 #include "fs/entry_data.h"
--- src/io/filters.cc
+++ src/io/filters.cc
@@ -10,6 +10,8 @@
 
 #include "autoconf.h"
 
+#include <cstring>
+
 #include "io/filters.h"
 #include "debug.h"
 
--- src/main.cc
+++ src/main.cc
@@ -191,7 +191,7 @@
 		/* Write out pid-file if needed */
 		const string& pidfile = proxy->config().get<string>(proxy_config::PID_FILE);
 		if (!pidfile.empty()) {
-			int f = open(pidfile.c_str(), O_CREAT | O_WRONLY);
+			int f = open(pidfile.c_str(), O_CREAT | O_WRONLY, 0644);
 			io::fdprintf(f, "%d", getpid());
 			fchmod(f, 0644);
 			close(f);
