diff -Naur libpcap-1.3.0-orig/configure.in libpcap-1.3.0/configure.in
--- libpcap-1.3.0-orig/configure.in	2013-02-08 00:11:41.888270763 -0500
+++ libpcap-1.3.0/configure.in	2013-02-08 00:13:04.664272980 -0500
@@ -449,14 +449,14 @@
 		],
 		[
 			#
-			# No, we don't; do we have libnl 1.x?
+			# No, we don't; do we have libnl 1.x? (hijacked, linking 3.x no matter what)
 			#
 			AC_CHECK_LIB(nl, nl_handle_alloc,
 			[
 				#
 				# Yes.
 				#
-				LIBS="-lnl $LIBS"
+				LIBS="-lnl-genl-3 -lnl-3 $LIBS"
 				AC_DEFINE(HAVE_LIBNL,1,[if libnl exists])
 			],
 			[
