--- ./src/gtkui.c.orig	2007-08-12 18:18:21.000000000 +0400
+++ ./src/gtkui.c	2007-08-12 18:22:19.000000000 +0400
@@ -752,8 +752,8 @@
 	return;
 
     write_comm_pipe_int (&from_gui_pipe, 0, 0);
-    write_comm_pipe_int (&from_gui_pipe, (int)data, 1);
-    gtk_label_set_text (GTK_LABEL (disk_text_widget[(int)data]), "");
+    write_comm_pipe_int (&from_gui_pipe, GPOINTER_TO_INT(data), 1);
+    gtk_label_set_text (GTK_LABEL (disk_text_widget[GPOINTER_TO_INT(data)]), "");
 }
 
 static void pause_uae (void)
@@ -841,7 +841,7 @@
 
 static void did_insert (GtkWidget *w, gpointer data)
 {
-    int n = (int)data;
+    int n = GPOINTER_TO_INT(data);
     if (filesel_active != -1)
 	return;
     filesel_active = n;
@@ -1188,13 +1188,13 @@
 	gtk_box_pack_start (GTK_BOX (buttonbox), thing, FALSE, TRUE, 0);
 	gtk_widget_show (thing);
 	disk_eject_widget[i] = thing;
-	gtk_signal_connect (GTK_OBJECT (thing), "clicked", (GtkSignalFunc) did_eject, (gpointer) i);
+	gtk_signal_connect (GTK_OBJECT (thing), "clicked", (GtkSignalFunc) did_eject, GINT_TO_POINTER(i));
 
 	thing = gtk_button_new_with_label ("Insert");
 	gtk_box_pack_start (GTK_BOX (buttonbox), thing, FALSE, TRUE, 0);
 	gtk_widget_show (thing);
 	disk_insert_widget[i] = thing;
-	gtk_signal_connect (GTK_OBJECT (thing), "clicked", (GtkSignalFunc) did_insert, (gpointer) i);
+	gtk_signal_connect (GTK_OBJECT (thing), "clicked", (GtkSignalFunc) did_insert, GINT_TO_POINTER(i));
     }
 
     add_empty_vbox (vbox);
