--- bitlbee-3.0.5/Makefile
+++ bitlbee-3.0.5/Makefile
@@ -130,7 +130,7 @@
 	install -m 0644 $(_SRCDIR_)protocols/skype/skyped.cnf $(DESTDIR)$(ETCDIR)/../skyped/skyped.cnf
 	install -m 0644 $(_SRCDIR_)protocols/skype/skyped.conf.dist $(DESTDIR)$(ETCDIR)/../skyped/skyped.conf
 	install -m 0755 $(_SRCDIR_)protocols/skype/skyped.py $(DESTDIR)$(BINDIR)/skyped
-	make -C protocols/skype install-doc
+	$(MAKE) -C protocols/skype install-doc
 endif
 
 systemd:
