--- bitlbee-3.2.1/tests/check_nick.c
+++ bitlbee-3.2.1/tests/check_nick.c
@@ -30,7 +30,7 @@
 	for (i = 0; get[i]; i++) {
 		char copy[60];
 		strcpy(copy, get[i]);
-		nick_strip(copy);
+		nick_strip(NULL, copy);
 		fail_unless (strcmp(copy, expected[i]) == 0, 
 					 "(%d) nick_strip broken: %s -> %s (expected: %s)", 
 					 i, get[i], copy, expected[i]);
@@ -45,7 +45,7 @@
 	int i;
 
 	for (i = 0; nicks[i]; i++) {
-		fail_unless (nick_ok(nicks[i]) == 1,
+		fail_unless (nick_ok(NULL, nicks[i]) == 1,
 					 "nick_ok() failed: %s", nicks[i]);
 	}
 }
@@ -58,7 +58,7 @@
 	int i;
 
 	for (i = 0; nicks[i]; i++) {
-		fail_unless (nick_ok(nicks[i]) == 0,
+		fail_unless (nick_ok(NULL, nicks[i]) == 0,
 					 "nick_ok() succeeded for invalid: %s", nicks[i]);
 	}
 }
