Index: rednotebook-1.1.0/rednotebook/gui/keepnote/gui/richtext/__init__.py
===================================================================
--- rednotebook-1.1.0.orig/rednotebook/gui/keepnote/gui/richtext/__init__.py
+++ rednotebook-1.1.0/rednotebook/gui/keepnote/gui/richtext/__init__.py
@@ -17,11 +17,7 @@ pygtk.require('2.0')
 import gtk, gobject, pango
 from gtk import gdk
 
-# try to import spell check
-try:
-    import gtkspell
-except ImportError:
-    gtkspell = None
+gtkspell = None
 
 
 # textbuffer_tools imports
Index: rednotebook-1.1.0/rednotebook/gui/main_window.py
===================================================================
--- rednotebook-1.1.0.orig/rednotebook/gui/main_window.py
+++ rednotebook-1.1.0/rednotebook/gui/main_window.py
@@ -30,11 +30,7 @@ import gtk
 import gobject
 import pango
 
-# try to import gtkspell
-try:
-    import gtkspell
-except ImportError:
-    gtkspell = None
+gtkspell = None
 
 
 # Initialize the gtk thread engine
