diff -ur libselinux-2.1.9.orig/src/Makefile libselinux-2.1.9/src/Makefile
--- libselinux-2.1.9.orig/src/Makefile	2012-10-06 21:11:43.140757973 +0200
+++ libselinux-2.1.9/src/Makefile	2012-10-06 21:12:54.896758126 +0200
@@ -15,7 +15,7 @@
 PYLIBDIR ?= $(LIBDIR)/$(PYLIBVER)
 RUBYLIBVER ?= $(shell ruby -e 'print RUBY_VERSION.split(".")[0..1].join(".")')
 RUBYPLATFORM ?= $(shell ruby -e 'print RUBY_PLATFORM')
-RUBYINC ?= $(LIBDIR)/ruby/$(RUBYLIBVER)/$(RUBYPLATFORM)
+RUBYINC ?= $(shell pkg-config --cflags ruby-$(RUBYLIBVER))
 RUBYINSTALL ?= $(LIBDIR)/ruby/site_ruby/$(RUBYLIBVER)/$(RUBYPLATFORM)
 LIBBASE=$(shell basename $(LIBDIR))
 
