=== modified file 'src/terminal.c'
--- src/terminal.c	2008-11-17 18:29:55 +0000
+++ src/terminal.c	2008-11-17 18:33:17 +0000
@@ -190,7 +190,9 @@
 #endif
 
    pthread_mutex_destroy(&terms->admin_listen_th.finished);
+#ifdef HAS_CURSES
    pthread_mutex_destroy(&terms->gui_th.finished);
+#endif
    pthread_mutex_destroy(&terms->pcap_listen_th.finished);
    pthread_mutex_destroy(&terms->uptime_th.finished);
 

