diff -ur sfftobmp3_1_1.orig/src/common.cpp sfftobmp3_1_1/src/common.cpp
--- sfftobmp3_1_1.orig/src/common.cpp	2007-10-07 22:01:20.000000000 +0300
+++ sfftobmp3_1_1/src/common.cpp	2008-04-14 23:32:51.000000000 +0300
@@ -51,6 +51,7 @@
 
 #include <cassert>
 #include <iostream>
+#include <cerrno>
 
 #ifdef WIN32
 #include <io.h>   // _mktemp
diff -ur sfftobmp3_1_1.orig/src/input.cpp sfftobmp3_1_1/src/input.cpp
--- sfftobmp3_1_1.orig/src/input.cpp	2007-10-07 22:01:20.000000000 +0300
+++ sfftobmp3_1_1/src/input.cpp	2008-04-14 23:30:50.000000000 +0300
@@ -36,6 +36,7 @@
 
 ---RCS-Info--------------------------------------------------*/
 
+#include <cstring>
 #include <vector>
 #include <iostream>
 
diff -ur sfftobmp3_1_1.orig/src/output.cpp sfftobmp3_1_1/src/output.cpp
--- sfftobmp3_1_1.orig/src/output.cpp	2007-10-07 22:01:21.000000000 +0300
+++ sfftobmp3_1_1/src/output.cpp	2008-04-14 23:30:50.000000000 +0300
@@ -37,6 +37,7 @@
 
 ---RCS-Info--------------------------------------------------*/
 
+#include <cstring>
 #include <cassert>
 #include <iostream>
 
