diff -Naur notmuch-0.10.2/configure notmuch-0.10.2.new/configure
--- notmuch-0.10.2/configure	2011-12-06 04:06:50.000000000 +0100
+++ notmuch-0.10.2.new/configure	2012-04-22 12:04:52.227816674 +0200
@@ -275,7 +275,7 @@
 
 printf "Checking for GMime development files... "
 have_gmime=0
-for gmimepc in gmime-2.6 gmime-2.4; do
+for gmimepc in gmime-2.4; do
     if pkg-config --exists $gmimepc; then
 	printf "Yes ($gmimepc).\n"
 	have_gmime=1
