--- Makefile.orig	2003-07-31 22:49:19.000000000 -0400
+++ Makefile	2003-07-31 22:50:10.000000000 -0400
@@ -9,13 +9,10 @@
 #
 linux:
 	+ $(MAKE) -j2 -C src freecnc-bin \
-	"EXTRACFLAGS = -ansi -pedantic -Werror" \
 	"LIBS = -lSDL_net"
 	+ $(MAKE) -j2 -C src plugins \
-	"EXTRACFLAGS = -ansi -pedantic -fPIC -Werror" \
-	"LDFLAGS = -shared"
-	+ $(MAKE) -j2 -C tools \
-	"EXTRACFLAGS = -ansi -pedantic"
+	"LDFLAGS = ${LDFLAGS} -shared"
+	+ $(MAKE) -j2 -C tools
 
 linux-nonet:
 	+ $(MAKE) -j2 -C src freecnc-bin \
