--- src/freecnc.cpp.orig	2003-07-31 22:51:34.000000000 -0400
+++ src/freecnc.cpp	2003-07-31 22:51:45.000000000 -0400
@@ -62,6 +62,7 @@
 #elif defined(__BEOS__)
     // BeOS runs everything as root
 #else
+/*
     if (getuid() == 0) {
         fprintf(stderr,"WARNING WARNING WARNING WARNING!\n"
                 "\tYOU ARE RUNNING FREECNC AS ROOT.\n"
@@ -74,6 +75,7 @@
                 "PLEASE DO NOT DO SO, ROOT PRIVILEGES ARE NOT NEEDED.\n");
         exit(1);
     }
+*/
 #endif
     args = new Args();
     binpath = determineBinaryLocation(argv[0]);
