--- tools/shpview/shpview.cpp
+++ tools/shpview/shpview.cpp
@@ -19,7 +19,7 @@
 
 Logger* logger;
 
-void loadPal(char* palname);
+void loadPal(const char* palname);
 
 void usage(char* biname)
 {
@@ -92,7 +92,7 @@
     exit(0);
 }
 
-void loadPal(char *palname)
+void loadPal(const char *palname)
 {
     VFile *mixfile;
     SDL_Color palette[256];
--- tools/tmpinied/tmpinied.cpp
+++ tools/tmpinied/tmpinied.cpp
@@ -13,7 +13,7 @@
 
 #define TEMPLATES 216
 
-void loadPal(char *palname);
+void loadPal(const char *palname);
 
 
 TemplateImage *TemplateData::getImage()
@@ -316,7 +316,7 @@
     exit(0);
 }
 
-void loadPal(char *palname)
+void loadPal(const char *palname)
 {
     VFile *mixfile;
     //Uint32 offset, size;
