--- tools/tmpinied/Makefile	2009-12-07 18:18:23.985443151 +0000
+++ tools/tmpinied/Makefile	2009-12-07 18:19:45.869476911 +0000
@@ -13,7 +13,7 @@
 CFLAGS = -g -Wall -Werror -I../../src/include `sdl-config --cflags`
 
 ../../tmpinied: $(LOCALOBJS) $(OBJECTS)
-	$(CXX) -o ../../tmpinied -g `sdl-config --libs` $(LOCALOBJS) $(OBJECTS) $(LIBS) $(LDFLAGS)
+	$(CXX) -g $(LDFLAGS) -o ../../tmpinied $(LOCALOBJS) $(OBJECTS) $(LIBS) `sdl-config --libs`
 clean:
 	\rm -f $(LOCALOBJS) ../../tmpinied
 
--- tools/audplay/Makefile.old	2010-09-08 10:26:48.000000000 +0200
+++ tools/audplay/Makefile	2010-09-08 10:27:19.000000000 +0200
@@ -12,7 +12,7 @@
 CFLAGS = -g  -Wall -I. -I../../src/include `sdl-config --cflags` $(EXTRACFLAGS)
 
 ../../audplay: $(LOCALOBJS) $(OBJECTS)
-	$(CXX) -o ../../audplay -g $(LOCALOBJS) $(OBJECTS) `sdl-config --libs` $(LIBS) $(LDFLAGS)
+	$(CXX) -o ../../audplay -g $(LDFLAGS) $(LOCALOBJS) $(OBJECTS) `sdl-config --libs` $(LIBS)
 clean:
 	\rm -f $(LOCALOBJS) ../../audplay
 
--- tools/mixshell/Makefile.old	2010-09-08 10:28:25.000000000 +0200
+++ tools/mixshell/Makefile	2010-09-08 10:28:53.000000000 +0200
@@ -6,7 +6,7 @@
 CFLAGS = -g -Wall -I. -I../../src/include `sdl-config --cflags` $(EXTRACFLAGS)
 
 mixshell: $(LOCALOBJ) $(OBJ)
-	$(CXX) -o mixshell $(LOCALOBJ) $(OBJ) `sdl-config --libs` $(LIBS) $(LDFLAGS)
+	$(CXX) -o mixshell $(LDFLAGS) $(LOCALOBJ) $(OBJ) `sdl-config --libs` $(LIBS)
 
 clean:
 	\rm -rf $(LOCALOBJ) mixshell
--- tools/shpview/Makefile.old	2010-09-08 10:30:15.000000000 +0200
+++ tools/shpview/Makefile	2010-09-08 10:30:28.000000000 +0200
@@ -13,7 +13,7 @@
 CFLAGS = -g  -Wall -I. -I../../src/include `sdl-config --cflags` $(EXTRACFLAGS)
 
 ../../shpview: $(LOCALOBJS) $(OBJECTS)
-	$(CXX) -o ../../shpview -g $(LDFLAGS) $(LOCALOBJS) $(OBJECTS) `sdl-config --libs` $(LIBS) $(LDFLAGS)
+	$(CXX) -o ../../shpview -g $(LDFLAGS) $(LOCALOBJS) $(OBJECTS) `sdl-config --libs` $(LIBS)
 clean:
 	\rm -f $(LOCALOBJS) ../../shpview
 
