--- src/Makefile.old	2011-06-22 22:57:59.000000000 +0200
+++ src/Makefile	2011-06-22 22:58:27.000000000 +0200
@@ -35,7 +35,7 @@
 all: freecnc-bin plugins
 
 ../freecnc: $(OBJECTS)
-	$(CXX) $(LDFLAGS) -o ../freecnc $(DEBUG_FLAGS) $(OBJECTS) `sdl-config --libs` $(LIBS) $(EFENCE)
+	$(CXX) $(LDFLAGS) -o ../freecnc $(DEBUG_FLAGS) $(OBJECTS) `sdl-config --libs` $(LIBS) -lm -ldl $(EFENCE)
 
 freecnc-bin: ../freecnc
 
--- tools/shpview/Makefile.old	2011-06-22 23:00:10.000000000 +0200
+++ tools/shpview/Makefile	2011-06-22 23:00:32.000000000 +0200
@@ -13,7 +13,7 @@
 CFLAGS = -g  -Wall -I. -I../../src/include `sdl-config --cflags` $(EXTRACFLAGS)
 
 ../../shpview: $(LOCALOBJS) $(OBJECTS)
-	$(CXX) -o ../../shpview -g $(LDFLAGS) $(LOCALOBJS) $(OBJECTS) `sdl-config --libs` $(LIBS)
+	$(CXX) -o ../../shpview -g $(LDFLAGS) $(LOCALOBJS) $(OBJECTS) `sdl-config --libs` $(LIBS) -lm -ldl
 clean:
 	\rm -f $(LOCALOBJS) ../../shpview
 
--- tools/tmpinied/Makefile.old	2011-06-22 23:01:44.000000000 +0200
+++ tools/tmpinied/Makefile	2011-06-22 23:02:12.000000000 +0200
@@ -13,7 +13,7 @@
 CFLAGS = -g -Wall  -I../../src/include `sdl-config --cflags`
 
 ../../tmpinied: $(LOCALOBJS) $(OBJECTS)
-	$(CXX) -g $(LDFLAGS) -o ../../tmpinied $(LOCALOBJS) $(OBJECTS) $(LIBS) `sdl-config --libs`
+	$(CXX) -g $(LDFLAGS) -o ../../tmpinied $(LOCALOBJS) $(OBJECTS) $(LIBS) `sdl-config --libs` -lm -ldl
 clean:
 	\rm -f $(LOCALOBJS) ../../tmpinied
 
--- tools/audplay/Makefile.old	2011-06-22 23:10:33.000000000 +0200
+++ tools/audplay/Makefile	2011-06-22 23:10:49.000000000 +0200
@@ -12,7 +12,7 @@
 CFLAGS = -g  -Wall -I. -I../../src/include `sdl-config --cflags` $(EXTRACFLAGS)
 
 ../../audplay: $(LOCALOBJS) $(OBJECTS)
-	$(CXX) -o ../../audplay -g $(LDFLAGS) $(LOCALOBJS) $(OBJECTS) `sdl-config --libs` $(LIBS)
+	$(CXX) -o ../../audplay -g $(LDFLAGS) $(LOCALOBJS) $(OBJECTS) `sdl-config --libs` $(LIBS) -lm -ldl
 clean:
 	\rm -f $(LOCALOBJS) ../../audplay
 
--- tools/mixshell/Makefile.old	2011-06-22 23:12:13.000000000 +0200
+++ tools/mixshell/Makefile	2011-06-22 23:12:32.000000000 +0200
@@ -6,7 +6,7 @@
 CFLAGS = -g -Wall -I. -I../../src/include `sdl-config --cflags` $(EXTRACFLAGS)
 
 mixshell: $(LOCALOBJ) $(OBJ)
-	$(CXX) -o mixshell $(LDFLAGS) $(LOCALOBJ) $(OBJ) `sdl-config --libs` $(LIBS)
+	$(CXX) -o mixshell $(LDFLAGS) $(LOCALOBJ) $(OBJ) `sdl-config --libs` $(LIBS) -lm
 
 clean:
 	\rm -rf $(LOCALOBJ) mixshell
