diff -Naur freecnc++.old/src/include/structure.h freecnc++/src/include/structure.h
--- freecnc++.old/src/include/structure.h	2006-04-02 00:15:35.000000000 -0500
+++ freecnc++/src/include/structure.h	2006-04-02 00:17:01.000000000 -0500
@@ -160,6 +160,7 @@
     std::vector<UnitType*> specificTypeAllow;
 };
 
+class BuildingAnimEvent;
 class BAttackAnimEvent;
 
 class Structure : public UnitOrStructure
