--- configure.ac
+++ configure.ac
@@ -66,9 +66,9 @@
 AC_MSG_RESULT($want_debug)
 if test "x$want_debug" = "xyes"
 then
-        export CFLAGS="-O2 -g -Wall -Wmissing-prototypes -pedantic -pg $extra_C_flags"
+        export CFLAGS="${CFLAGS} -Wall -Wmissing-prototypes -pedantic -pg $extra_C_flags"
 else
-        export CFLAGS="-O3 -std=gnu99 -Wmissing-prototypes -pedantic -ffast-math -fomit-frame-pointer $extra_C_flags"
+        export CFLAGS="${CFLAGS} -std=gnu99 -Wmissing-prototypes -pedantic $extra_C_flags"
 fi
 
 
