diff -urN root.c.old root.c
--- root.c.old	2007-07-10 17:06:55.000000000 +0200
+++ root.c	2007-07-10 17:15:27.000000000 +0200
@@ -26,6 +26,8 @@
 
 #include <linux/module.h>
 
+#include <linux/sched.h>
+
 
 /*++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*
  * Robert W. Boone (November, 2002)                                           *
diff -urN daemon.c.old daemon.c
--- daemon.c.old	2007-07-10 17:07:58.000000000 +0200
+++ daemon.c	2007-07-10 17:15:03.000000000 +0200
@@ -29,6 +29,8 @@
 
 #include "cdfs.h"
 
+#include <linux/sched.h>
+
 int kcdfsd_pid = 0;
 static int kcdfsd_running = 0;
 static DECLARE_WAIT_QUEUE_HEAD(kcdfsd_wait);
