Index: latex2rtf-1.9.19/Makefile
===================================================================
--- latex2rtf-1.9.19.orig/Makefile
+++ latex2rtf-1.9.19/Makefile
@@ -5,7 +5,7 @@ MKDIR=mkdir -p
 TAR=gnutar
 RM=rm
 
-CFLAGS:=-DUNIX
+CFLAGS:=$(CFLAGS) -DUNIX
 #CFLAGS:=-DMSDOS         #Windows/DOS
 #CFLAGS:=-DMAC_CLASSIC   #MacOS 8/9
 #CFLAGS:=-DOS2           #OS/2
@@ -39,7 +39,7 @@ CFG_INSTALL=$(PREFIX)/share/latex2rtf/cf
 
 CFLAGS:=$(CFLAGS) -g -Wall -fsigned-char
 
-LIBS=
+LIBS= -lm
 #LIBS=-lMallocDebug -force_flat_namespace
 
 VERSION:="`scripts/version`"
@@ -141,7 +141,7 @@ OBJS=fonts.o direct.o encodings.o comman
 all : checkdir latex2rtf
 
 latex2rtf: $(OBJS) $(HDRS)
-	$(CC) $(CFLAGS) $(LINK_FLAGS) $(OBJS)	$(LIBS) -o $(BINARY_NAME)
+	$(CC) $(CFLAGS) $(LDFLAGS) $(OBJS)	$(LIBS) -o $(BINARY_NAME)
 
 cfg.o: Makefile cfg.c
 	$(CC) $(CFLAGS) -DCFGDIR=\"$(CFG_INSTALL)\" -c cfg.c -o cfg.o
