From: Julian Ospald <hasufell@gentoo.org>
Date: Tue Mar  4 03:07:38 UTC 2014
Subject: set correct version

--- a/configure.ac
+++ b/configure.ac
@@ -9,7 +9,7 @@
 dnl m4_if(2.65, AC_AUTOCONF_VERSION, [m4_fatal([autoconf version 2.65 is broken and unusable])])
 
 dnl Update version info on each release
-AC_INIT([XPaint], [2.9.9], [Jean-Pierre.demailly@ujf-grenoble.fr])
+AC_INIT([XPaint], [2.9.10], [Jean-Pierre.demailly@ujf-grenoble.fr])
 
 AC_CONFIG_MACRO_DIR([m4])
 AM_INIT_AUTOMAKE([-Wall -Werror])
