From 7d9c1a49b87df15cdb78c529db554ea82c77033f Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Thu, 17 Nov 2011 20:53:51 +0000
Subject: Check ABI major not encoded ABI

Signed-off-by: Adam Jackson <ajax@redhat.com>
---
diff --git a/src/smi_driver.c b/src/smi_driver.c
index 86644c7..6508fae 100644
--- a/src/smi_driver.c
+++ b/src/smi_driver.c
@@ -442,7 +442,7 @@ SMI_PreInit(ScrnInfoPtr pScrn, int flags)
 	    LEAVE(FALSE);
 
 	hwp = VGAHWPTR(pScrn);
-#if ABI_VIDEODRV_VERSION < 12
+#if GET_ABI_MAJOR(ABI_VIDEODRV_VERSION) < 12
 	pSmi->PIOBase = hwp->PIOOffset;
 #else
 	pSmi->PIOBase = 0;
--
cgit v0.9.0.2-2-gbebe
