--- powerpc-utils-1.1.0.orig/Makefile	2006-03-28 14:40:10.000000000 -0600
+++ powerpc-utils-1.1.0/Makefile	2006-04-19 14:44:55.000000000 -0500
@@ -12,19 +12,12 @@
 TARBALL_FILES += $(FILES) Changelog
 
 all:
-	@echo "Building cmds files..."
-	@$(MAKE) -C cmds
-	@echo "Building scripts files..."
-	@$(MAKE) -C scripts
-	@echo "Building doc files..."
-	@$(MAKE) -C doc
+	@$(foreach d,$(SUBDIRS), $(MAKE) -C $d;)
 
 install: all
-	@$(call install_doc,$(FILES),$(DESTDIR))
 	@$(foreach d,$(SUBDIRS), $(MAKE) -C $d install;)
 
 uninstall: 
-	@$(call uninstall_doc,$(FILES),$(DESTDIR))
 	@$(foreach d,$(SUBDIRS), $(MAKE) -C $d uninstall;)
 
 rpm: all
