commit dcc2ad6952e334b91f0f044a1f1083c99949d840
Author: Lukas Oberhuber <lukaso@gmail.com>
Date:   Wed Dec 21 16:55:19 2011 +0000

    Substitute Pathname to get tests to pass in ruby 1.9.3

diff --git a/lib/fakefs/base.rb b/lib/fakefs/base.rb
index 7e83e76..6c1741a 100644
--- a/lib/fakefs/base.rb
+++ b/lib/fakefs/base.rb
@@ -2,6 +2,7 @@ RealFile            = File
 RealFileTest        = FileTest
 RealFileUtils       = FileUtils
 RealDir             = Dir
+RealPathname        = Pathname
 
 module FakeFS
   def self.activate!
@@ -10,11 +11,14 @@ module FakeFS
       remove_const(:File)
       remove_const(:FileTest)
       remove_const(:FileUtils)
+      remove_const(:Pathname) if RUBY_VERSION >= "1.9.3"
+      
 
       const_set(:Dir,       FakeFS::Dir)
       const_set(:File,      FakeFS::File)
       const_set(:FileUtils, FakeFS::FileUtils)
       const_set(:FileTest,  FakeFS::FileTest)
+      const_set(:Pathname,  FakeFS::Pathname) if RUBY_VERSION >= "1.9.3"
     end
     true
   end
@@ -25,11 +29,13 @@ module FakeFS
       remove_const(:File)
       remove_const(:FileTest)
       remove_const(:FileUtils)
+      remove_const(:Pathname) if RUBY_VERSION >= "1.9.3"
 
       const_set(:Dir,       RealDir)
       const_set(:File,      RealFile)
       const_set(:FileTest,  RealFileTest)
       const_set(:FileUtils, RealFileUtils)
+      const_set(:Pathname,  RealPathname) if RUBY_VERSION >= "1.9.3"
     end
     true
   end
diff --git a/lib/fakefs/safe.rb b/lib/fakefs/safe.rb
index a1b804c..783de74 100644
--- a/lib/fakefs/safe.rb
+++ b/lib/fakefs/safe.rb
@@ -9,3 +9,5 @@ require 'fakefs/fileutils'
 require 'fakefs/file'
 require 'fakefs/file_test'
 require 'fakefs/dir'
+require 'fakefs/pathname' if RUBY_VERSION >= "1.9.3"
+
