diff --git a/src/ch-ccmx.c b/src/ch-ccmx.c
index d674b16..e33e6d6 100644
--- a/src/ch-ccmx.c
+++ b/src/ch-ccmx.c
@@ -510,9 +510,15 @@ ch_ccmx_get_device_download_kind (ChCcmxPrivate *priv)
 	case CH_DEVICE_MODE_FIRMWARE:
 		str = "colorhug";
 		break;
+#if CD_CHECK_VERSION(1,2,0)
+	case CH_DEVICE_MODE_BOOTLOADER_PLUS:
+	case CH_DEVICE_MODE_FIRMWARE_PLUS:
+		str = "colorhug-plus";
+#else
 	case CH_DEVICE_MODE_BOOTLOADER_SPECTRO:
 	case CH_DEVICE_MODE_FIRMWARE_SPECTRO:
 		str = "colorhug-spectro";
+#endif
 		break;
 	default:
 		str = "unknown";
diff --git a/src/ch-flash.c b/src/ch-flash.c
index 53ef123..82b2c03 100644
--- a/src/ch-flash.c
+++ b/src/ch-flash.c
@@ -721,9 +721,15 @@ ch_flash_get_device_download_kind (ChFlashPrivate *priv)
 	case CH_DEVICE_MODE_FIRMWARE:
 		str = "colorhug";
 		break;
+#if CD_CHECK_VERSION(1,2,0)
+	case CH_DEVICE_MODE_BOOTLOADER_PLUS:
+	case CH_DEVICE_MODE_FIRMWARE_PLUS:
+		str = "colorhug-plus";
+#else
 	case CH_DEVICE_MODE_BOOTLOADER_SPECTRO:
 	case CH_DEVICE_MODE_FIRMWARE_SPECTRO:
 		str = "colorhug-spectro";
+#endif
 		break;
 	default:
 		str = "unknown";
diff --git a/src/ch-spectro-util.c b/src/ch-spectro-util.c
index 7f5c989..f03ed5c 100644
--- a/src/ch-spectro-util.c
+++ b/src/ch-spectro-util.c
@@ -476,7 +476,12 @@ ch_util_device_added_cb (GUsbDeviceList *list,
 	g_debug ("Added: %i:%i",
 		 g_usb_device_get_vid (device),
 		 g_usb_device_get_pid (device));
+
+#if CD_CHECK_VERSION(1,2,0)
+	if (ch_device_get_mode (device) == CH_DEVICE_MODE_FIRMWARE_PLUS) {
+#else
 	if (ch_device_get_mode (device) == CH_DEVICE_MODE_FIRMWARE_SPECTRO) {
+#endif
 		priv->device = g_object_ref (device);
 		ch_util_got_device (priv);
 	}
