	provide an option for libVLC

--- megaglest-3.7.0/CMakeLists.txt
+++ megaglest-3.7.0/CMakeLists.txt
@@ -44,6 +44,7 @@
 OPTION(BUILD_MEGAGLEST_MODEL_VIEWER "Build model viewer" ON)
 OPTION(BUILD_MEGAGLEST_MAP_EDITOR "Build map editor" ON)
 OPTION(BUILD_MEGAGLEST "Build MegaGlest" ON)
+OPTION(WITH_VLC "Use libVLC to play videos" ON)
 
 FIND_PROGRAM(HELP2MAN "help2man")
 
--- megaglest-3.7.0/source/shared_lib/CMakeLists.txt
+++ megaglest-3.7.0/source/shared_lib/CMakeLists.txt
@@ -221,7 +221,7 @@
         SET(LIBVLC_MIN_VERSION "1.1.0")
 	FIND_PACKAGE(LIBVLC)
 
-        IF(LIBVLC_FOUND AND NOT LIBVLC_VERSION STRLESS "${LIBVLC_MIN_VERSION}")
+	IF(LIBVLC_FOUND AND WITH_VLC AND NOT LIBVLC_VERSION STRLESS "${LIBVLC_MIN_VERSION}")
                 MESSAGE(STATUS "**NOTE: LIBVLC found so videos are supported.")
 	        INCLUDE_DIRECTORIES(${LIBVLC_INCLUDE_DIR})
 	        IF(UNIX)
@@ -236,7 +236,11 @@
 		ENDIF()
 		
         ELSE()
-                MESSAGE(STATUS "**WARNING: LIBVLC NOT found so videos are NOT supported.")
+		IF(WITH_VLC)
+	                MESSAGE(STATUS "**WARNING: LIBVLC NOT found so videos are NOT supported.")
+		ELSE()
+			MESSAGE(STATUS "LIBVLC disabled so videos are NOT supported.")
+		ENDIF()
         ENDIF()
 
 	#########################################################################################
