diff --git a/uuagc.cabal b/uuagc.cabal
index 5012e56..281e68f 100644
--- a/uuagc.cabal
+++ b/uuagc.cabal
@@ -3,7 +3,7 @@ version: 0.9.1
 license: GPL
 license-file: LICENSE
 maintainer: Arthur Baars <arthurb@cs.uu.nl>
-Build-Depends: base, haskell98, uulib
+Build-Depends: base, haskell98, uulib, directory
 
 Executable:     uuagc
 Main-Is:        Ag.hs
@@ -31,5 +31,5 @@ Other-Modules: AbstractSyntax,
       Streaming,
       TokenDef,
       Transform
-Extensions: TypeSynonymInstances       
+Extensions: TypeSynonymInstances, MultiParamTypeClasses
 Hs-Source-Dir: src
diff --git a/uuagc.cabal.in b/uuagc.cabal.in
index 754b4ba..9693be5 100644
--- a/uuagc.cabal.in
+++ b/uuagc.cabal.in
@@ -3,7 +3,7 @@ version: @VERSION@
 license: GPL
 license-file: LICENSE
 maintainer: Arthur Baars <arthurb@cs.uu.nl>
-Build-Depends: base, haskell98, uulib
+Build-Depends: base, haskell98, uulib, directory
 
 Executable:     uuagc
 Main-Is:        Ag.hs
@@ -31,5 +31,5 @@ Other-Modules: AbstractSyntax,
       Streaming,
       TokenDef,
       Transform
-Extensions: TypeSynonymInstances       
+Extensions: TypeSynonymInstances, MultiParamTypeClasses
 Hs-Source-Dir: src
