Index: projectM-qt-2.1.0-Source/CMakeLists.txt
===================================================================
--- projectM-qt-2.1.0-Source.orig/CMakeLists.txt
+++ projectM-qt-2.1.0-Source/CMakeLists.txt
@@ -131,8 +131,8 @@ include_directories(${PROJECTM_INCLUDE}
 target_link_libraries(projectM-qt projectM GLEW GL ${QT_QTGUI_LIBRARIES} ${QT_QTOPENGL_LIBRARIES} ${QT_QTXML_LIBRARIES} ${QT_LIBRARIES})
 
 install(FILES qprojectm_mainwindow.hpp DESTINATION include/libprojectM-qt)
-install(TARGETS projectM-qt DESTINATION lib)
-install(FILES "${CMAKE_CURRENT_BINARY_DIR}/libprojectM-qt.pc" DESTINATION lib/pkgconfig)
+install(TARGETS projectM-qt DESTINATION lib${LIB_SUFFIX})
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/libprojectM-qt.pc" DESTINATION lib${LIB_SUFFIX}/pkgconfig)
 install(FILES "images/icons/prjm16-transparent.svg" DESTINATION	${CMAKE_INSTALL_PREFIX}/share/pixmaps)
 
 MESSAGE(STATUS "<<< Gentoo configuration >>>
