Index: rdoc-2.5.6/test/test_rdoc_markup_pre_process.rb
===================================================================
--- rdoc-2.5.6.orig/test/test_rdoc_markup_pre_process.rb
+++ rdoc-2.5.6/test/test_rdoc_markup_pre_process.rb
@@ -7,10 +7,10 @@ class TestRDocMarkupPreProcess < MiniTes
 
   def setup
     @tempfile = Tempfile.new 'test_rdoc_markup_pre_process'
-    @name = File.basename @tempfile.path
-    @dir  = File.dirname @tempfile.path
+    @tempname = File.basename @tempfile.path
+    @tempdir  = File.dirname @tempfile.path
 
-    @pp = RDoc::Markup::PreProcess.new __FILE__, [@dir]
+    @pp = RDoc::Markup::PreProcess.new __FILE__, [@tempdir]
   end
 
   def teardown
@@ -28,7 +28,7 @@ contents of a string.
     @tempfile.flush
     @tempfile.rewind
 
-    content = @pp.include_file @name, ''
+    content = @pp.include_file @tempname, ''
 
     expected = <<-EXPECTED
 Regular expressions (<i>regexp</i>s) are patterns which describe the
