--- setup.py
+++ setup.py
@@ -205,7 +205,7 @@
 copy_all_files = 1
 
 def copy_extra_files():
-    destroot = glob(os.path.join('build','lib'))[0]
+    destroot = glob(os.path.join('build-' + os.environ['EPYTHON'][6:],'lib'))[0]
 
     # go through MANIFEST to see what is supposed to be under build directory
     print "Copying extra files to %s ..." % destroot
@@ -413,18 +413,6 @@
 
     sys.exit(0)
     
-if 'install' in sys.argv:
-    
-    #if not os.path.isdir('build'):
-        #print "** Please run build command first **"
-        #sys.exit(1)
-    #	os.system('%s setup.py build' % sys.executable)
-
-    # make sure it's a clean build!
-    os.system('%s setup.py build' % sys.executable)	
-
-    # fall through to do_setup()
-    
 if 'install_all' in sys.argv:
     from disthelper.find_python import get_python_verlist
 
