diff -ru mysql++-2.1.1.orig/lib/qparms.h mysql++-2.1.1/lib/qparms.h
--- mysql++-2.1.1.orig/lib/qparms.h	2006-04-05 06:44:49.000000000 +0200
+++ mysql++-2.1.1/lib/qparms.h	2006-11-28 21:40:36.000000000 +0100
@@ -232,7 +232,7 @@
 	/// \param b the 'before' value
 	/// \param o the 'option' value
 	/// \param n the 'num' value
-	SQLParseElement(std::string b, char o, char n) :
+	SQLParseElement(std::string b, char o, signed char n) :
 	before(b),
 	option(o),
 	num(n)
@@ -241,7 +241,7 @@
 	
 	std::string before;		///< string inserted before the parameter
 	char option;			///< the parameter option, or blank if none
-	char num;				///< the parameter position to use
+	signed char num;		///< the parameter position to use
 };
 
 } // end namespace mysqlpp
diff -ru mysql++-2.1.1.orig/lib/query.cpp mysql++-2.1.1/lib/query.cpp
--- mysql++-2.1.1.orig/lib/query.cpp	2006-04-05 06:44:49.000000000 +0200
+++ mysql++-2.1.1/lib/query.cpp	2006-11-28 21:40:52.000000000 +0100
@@ -185,7 +185,7 @@
 				else {
 					num[1] = 0;
 				}
-				short int n = atoi(num);
+				signed char n = atoi(num);
 
 				// Look for option character following position value.
 				char option = ' ';
@@ -220,7 +220,7 @@
 				}
 
 				// Finished parsing parameter; save it.
-				parse_elems_.push_back(SQLParseElement(str, option, char(n)));
+				parse_elems_.push_back(SQLParseElement(str, option, n));
 				str = "";
 				name = "";
 			}
@@ -304,7 +304,7 @@
 {
 	sbuffer_.str("");
 
-	char num;
+	signed char num;
 	SQLString* ss;
 	SQLQueryParms* c;
 
