--- bindings/perl/src/CMakeLists.txt~	2010-09-01 22:23:34.000000000 +0200
+++ bindings/perl/src/CMakeLists.txt	2011-06-05 12:04:19.000000000 +0200
@@ -10,9 +10,6 @@
 
 set(Libproxy_LIB_SRCS Libproxy.c)
 
-set(CMAKE_C_COMPILER ${CMAKE_CXX_COMPILER})
-set(CMAKE_C_FLAGS ${PERL_CXX_FLAGS} )
-
 set(LIBRARY_OUTPUT_PATH ${CMAKE_BINARY_DIR}/perl/blib/arch/auto/Net)
 add_library(PLlibproxy SHARED ${Libproxy_LIB_SRCS})
 
