diff -Naru obs-build-20131015.orig/build obs-build-20131015/build
--- obs-build-20131015.orig/build	2014-01-28 12:31:51.126824118 +0000
+++ obs-build-20131015/build	2014-01-28 12:31:23.973477260 +0000
@@ -12,7 +12,7 @@
    exec /.build/build "$@"
 fi
 
-test -z "$BUILD_DIR" && BUILD_DIR=/usr/lib/build
+test -z "$BUILD_DIR" && BUILD_DIR=/usr/libexec/suse-build
 test -z "$BUILD_ROOT" && BUILD_ROOT=/var/tmp/build-root
 test -z "$CONFIG_DIR" && CONFIG_DIR="$BUILD_DIR/configs"
 
diff -Naru obs-build-20131015.orig/createarchdeps obs-build-20131015/createarchdeps
--- obs-build-20131015.orig/createarchdeps	2014-01-28 12:31:51.128824144 +0000
+++ obs-build-20131015/createarchdeps	2014-01-28 12:31:23.976477299 +0000
@@ -3,7 +3,7 @@
 # Archlinux support, based on the GSoC work of Nikolay Rysev <mad.f3ka@gmail.com>
 
 BEGIN {
-  unshift @INC, ($::ENV{"BUILD_DIR"} || "/usr/lib/build");
+  unshift @INC, ($::ENV{"BUILD_DIR"} || "/usr/libexec/suse-build");
 }
 
 use strict;
diff -Naru obs-build-20131015.orig/createrepomddeps obs-build-20131015/createrepomddeps
--- obs-build-20131015.orig/createrepomddeps	2014-01-28 12:31:51.128824144 +0000
+++ obs-build-20131015/createrepomddeps	2014-01-28 12:31:23.976477299 +0000
@@ -1,7 +1,7 @@
 #!/usr/bin/perl -w
 
 BEGIN {
-  unshift @INC, ($::ENV{'BUILD_DIR'} || '/usr/lib/build');
+  unshift @INC, ($::ENV{'BUILD_DIR'} || '/usr/libexec/suse-build');
 }
 
 use strict;
diff -Naru obs-build-20131015.orig/createrpmdeps obs-build-20131015/createrpmdeps
--- obs-build-20131015.orig/createrpmdeps	2014-01-28 12:31:51.129824157 +0000
+++ obs-build-20131015/createrpmdeps	2014-01-28 12:31:23.976477299 +0000
@@ -1,7 +1,7 @@
 #!/usr/bin/perl -w
 
 BEGIN {
-  unshift @INC, ($::ENV{'BUILD_DIR'} || '/usr/lib/build');
+  unshift @INC, ($::ENV{'BUILD_DIR'} || '/usr/libexec/suse-build');
 }
 
 use Build;
diff -Naru obs-build-20131015.orig/createyastdeps obs-build-20131015/createyastdeps
--- obs-build-20131015.orig/createyastdeps	2014-01-28 12:31:51.127824131 +0000
+++ obs-build-20131015/createyastdeps	2014-01-28 12:31:23.974477273 +0000
@@ -1,7 +1,7 @@
 #!/usr/bin/perl -w
 
 BEGIN {
-  unshift @INC, ($::ENV{'BUILD_DIR'} || '/usr/lib/build');
+  unshift @INC, ($::ENV{'BUILD_DIR'} || '/usr/libexec/suse-build');
 }
 
 use Build;
diff -Naru obs-build-20131015.orig/download obs-build-20131015/download
--- obs-build-20131015.orig/download	2014-01-28 12:31:51.129824157 +0000
+++ obs-build-20131015/download	2014-01-28 12:31:23.976477299 +0000
@@ -3,7 +3,7 @@
 use Net::SSL ();
 BEGIN {
   $ENV{PERL_LWP_SSL_VERIFY_HOSTNAME} = 0,
-  unshift @INC, ($::ENV{'BUILD_DIR'} || '/usr/lib/build');
+  unshift @INC, ($::ENV{'BUILD_DIR'} || '/usr/libexec/suse-build');
 }
 
 use LWP::UserAgent;
diff -Naru obs-build-20131015.orig/expanddeps obs-build-20131015/expanddeps
--- obs-build-20131015.orig/expanddeps	2014-01-28 12:31:51.128824144 +0000
+++ obs-build-20131015/expanddeps	2014-01-28 12:31:23.975477286 +0000
@@ -1,7 +1,7 @@
 #!/usr/bin/perl -w
 
 BEGIN {
-  unshift @INC, ($::ENV{'BUILD_DIR'} || '/usr/lib/build');
+  unshift @INC, ($::ENV{'BUILD_DIR'} || '/usr/libexec/suse-build');
 }
 
 use strict;
diff -Naru obs-build-20131015.orig/getbinaryid obs-build-20131015/getbinaryid
--- obs-build-20131015.orig/getbinaryid	2014-01-28 12:31:51.129824157 +0000
+++ obs-build-20131015/getbinaryid	2014-01-28 12:31:23.976477299 +0000
@@ -1,7 +1,7 @@
 #!/usr/bin/perl
 
 BEGIN {
-  unshift @INC, ($::ENV{'BUILD_DIR'} || '/usr/lib/build');
+  unshift @INC, ($::ENV{'BUILD_DIR'} || '/usr/libexec/suse-build');
 }
 
 use Build;
diff -Naru obs-build-20131015.orig/getchangetarget obs-build-20131015/getchangetarget
--- obs-build-20131015.orig/getchangetarget	2014-01-28 12:31:51.129824157 +0000
+++ obs-build-20131015/getchangetarget	2014-01-28 12:31:23.976477299 +0000
@@ -1,7 +1,7 @@
 #!/usr/bin/perl -w
 
 BEGIN {
-  unshift @INC, ($::ENV{'BUILD_DIR'} || '/usr/lib/build');
+  unshift @INC, ($::ENV{'BUILD_DIR'} || '/usr/libexec/suse-build');
 }
 
 use strict;
diff -Naru obs-build-20131015.orig/getmacros obs-build-20131015/getmacros
--- obs-build-20131015.orig/getmacros	2014-01-28 12:31:51.129824157 +0000
+++ obs-build-20131015/getmacros	2014-01-28 12:31:23.976477299 +0000
@@ -1,7 +1,7 @@
 #!/usr/bin/perl -w
 
 BEGIN {
-  unshift @INC, ($::ENV{'BUILD_DIR'} || '/usr/lib/build');
+  unshift @INC, ($::ENV{'BUILD_DIR'} || '/usr/libexec/suse-build');
 }
 
 use Build;
diff -Naru obs-build-20131015.orig/getoptflags obs-build-20131015/getoptflags
--- obs-build-20131015.orig/getoptflags	2014-01-28 12:31:51.128824144 +0000
+++ obs-build-20131015/getoptflags	2014-01-28 12:31:23.976477299 +0000
@@ -1,7 +1,7 @@
 #!/usr/bin/perl -w
 
 BEGIN {
-  unshift @INC, ($::ENV{'BUILD_DIR'} || '/usr/lib/build');
+  unshift @INC, ($::ENV{'BUILD_DIR'} || '/usr/libexec/suse-build');
 }
 
 use strict;
diff -Naru obs-build-20131015.orig/gettype obs-build-20131015/gettype
--- obs-build-20131015.orig/gettype	2014-01-28 12:31:51.127824131 +0000
+++ obs-build-20131015/gettype	2014-01-28 12:31:23.974477273 +0000
@@ -1,7 +1,7 @@
 #!/usr/bin/perl -w
 
 BEGIN {
-  unshift @INC, ($::ENV{'BUILD_DIR'} || '/usr/lib/build');
+  unshift @INC, ($::ENV{'BUILD_DIR'} || '/usr/libexec/suse-build');
 }
 
 use strict;
diff -Naru obs-build-20131015.orig/changelog2spec obs-build-20131015/changelog2spec
--- obs-build-20131015.orig/changelog2spec	2014-01-28 12:31:51.128824144 +0000
+++ obs-build-20131015/changelog2spec	2014-01-28 12:31:23.975477286 +0000
@@ -5,7 +5,7 @@
 #
 
 BEGIN {
-  unshift @INC, ($::ENV{'BUILD_DIR'} || '/usr/lib/build');
+  unshift @INC, ($::ENV{'BUILD_DIR'} || '/usr/libexec/suse-build');
 }
 
 use Date::Parse;
diff -Naru obs-build-20131015.orig/init_buildsystem obs-build-20131015/init_buildsystem
--- obs-build-20131015.orig/init_buildsystem	2014-01-28 12:31:51.125824105 +0000
+++ obs-build-20131015/init_buildsystem	2014-01-28 12:31:23.973477260 +0000
@@ -16,7 +16,7 @@
 export YAST_IS_RUNNING="instsys"
 export DEBIAN_FRONTEND=noninteractive
 export DEBIAN_PRIORITY=critical
-export BUILD_DIR=${BUILD_DIR:-/usr/lib/build}
+export BUILD_DIR=${BUILD_DIR:-/usr/libexec/suse-build}
 # need to restore build root owner for non-root builds
 browner=0
 definesnstuff=()
diff -Naru obs-build-20131015.orig/initvm.c obs-build-20131015/initvm.c
--- obs-build-20131015.orig/initvm.c	2014-01-28 12:31:51.128824144 +0000
+++ obs-build-20131015/initvm.c	2014-01-28 12:31:23.975477286 +0000
@@ -53,9 +53,10 @@
 #define SYSFS_BINFMT_MISC_REG	"/proc/sys/fs/binfmt_misc/register"
 #define SYSFS_BINFMT_MISC_STAT	"/proc/sys/fs/binfmt_misc/status"
 
-/* /usr/lib/build/x paths are copied to /.build inside a virtual machine */
+/* /usr/libexec/suse-build/x paths are copied to /.build inside a virtual machine */
 #define BINFMT_REGF_0		"/.build/qemu-reg"
-#define BINFMT_REGF_1		"/usr/lib/build/qemu-reg"
+#define BINFMT_REGF_1		"/usr/libexec/suse-build/qemu-reg"
+#define BINFMT_REGF_2		"/usr/lib/build/qemu-reg"
 #define BUILD			"/.build/build"
 
 /* useful constant arrays */
@@ -307,9 +308,13 @@
 			fprintf(stderr, "%s: failed. Trying alternate binfmt file\n",
 				BINFMT_REGF_0);
 			if (!binfmt_register(BINFMT_REGF_1, SYSFS_BINFMT_MISC_REG)) {
-				fprintf(stderr, "%s: binfmt registration failed\n",
+				fprintf(stderr, "%s: failed. Trying alternate binfmt file\n",
 					BINFMT_REGF_1);
-				exit(1);
+				if (!binfmt_register(BINFMT_REGF_2, SYSFS_BINFMT_MISC_REG)) {
+					fprintf(stderr, "%s: binfmt registration failed\n",
+						BINFMT_REGF_2);
+					exit(1);
+				}
 			}
 		}
 	}
diff -Naru obs-build-20131015.orig/mkdrpms obs-build-20131015/mkdrpms
--- obs-build-20131015.orig/mkdrpms	2014-01-28 12:31:51.127824131 +0000
+++ obs-build-20131015/mkdrpms	2014-01-28 12:31:23.974477273 +0000
@@ -1,7 +1,7 @@
 #!/usr/bin/perl -w
 
 BEGIN {
-  unshift @INC, '/usr/lib/build';
+  unshift @INC, '/usr/libexec/suse-build';
   unshift @INC, $::ENV{'BUILD_DIR'} if $::ENV{'BUILD_DIR'};
 }
 
diff -Naru obs-build-20131015.orig/order obs-build-20131015/order
--- obs-build-20131015.orig/order	2014-01-28 12:31:51.128824144 +0000
+++ obs-build-20131015/order	2014-01-28 12:31:23.975477286 +0000
@@ -1,7 +1,7 @@
 #!/usr/bin/perl -w
 
 BEGIN {
-  unshift @INC, ($::ENV{'BUILD_DIR'} || '/usr/lib/build');
+  unshift @INC, ($::ENV{'BUILD_DIR'} || '/usr/libexec/suse-build');
 }
 
 use Build;
diff -Naru obs-build-20131015.orig/spectool obs-build-20131015/spectool
--- obs-build-20131015.orig/spectool	2014-01-28 12:31:51.128824144 +0000
+++ obs-build-20131015/spectool	2014-01-28 12:31:23.975477286 +0000
@@ -116,7 +116,7 @@
 my $builddir;
 
 BEGIN {
-  $builddir = ($::ENV{'BUILD_DIR'} || '/usr/lib/build');
+  $builddir = ($::ENV{'BUILD_DIR'} || '/usr/libexec/suse-build');
   unshift @INC, $builddir;
 }
 
diff -Naru obs-build-20131015.orig/substitutedeps obs-build-20131015/substitutedeps
--- obs-build-20131015.orig/substitutedeps	2014-01-28 12:31:51.128824144 +0000
+++ obs-build-20131015/substitutedeps	2014-01-28 12:31:23.976477299 +0000
@@ -1,7 +1,7 @@
 #!/usr/bin/perl -w
 
 BEGIN {
-  unshift @INC, ($::ENV{'BUILD_DIR'} || '/usr/lib/build');
+  unshift @INC, ($::ENV{'BUILD_DIR'} || '/usr/libexec/suse-build');
 }
 
 use strict;
