diff -ruN configure.in configure.in
--- configure.in	1970-01-01 01:00:00.000000000 +0100
+++ configure.in	2006-07-18 08:21:30.110325984 +0200
@@ -0,0 +1,10 @@
+AC_INIT(configure.in)
+
+AM_CONFIG_HEADER(config.h)
+AM_INIT_AUTOMAKE(ibpp, 2.5.2.2)
+
+AC_LANG_CPLUSPLUS
+AC_PROG_CXX
+AM_PROG_LIBTOOL
+
+AC_OUTPUT(Makefile core/Makefile)
diff -ruN core/Makefile.am core/Makefile.am
--- core/Makefile.am	1970-01-01 01:00:00.000000000 +0100
+++ core/Makefile.am	2006-07-18 08:21:30.111325832 +0200
@@ -0,0 +1,12 @@
+libibpp_la_LIBADD = -lfbclient
+
+INCLUDES = $(all_includes) -I./fbheaders
+METASOURCES = AUTO
+lib_LTLIBRARIES = libibpp.la
+libibpp_la_SOURCES = array.cpp blob.cpp database.cpp date.cpp dbkey.cpp _dpb.cpp \
+    events.cpp exception.cpp ibase.h iberror.h _ibpp.cpp ibpp.h _ibpp.h _ibs.cpp \
+    _rb.cpp row.cpp service.cpp _spb.cpp statement.cpp time.cpp _tpb.cpp transaction.cpp \
+    user.cpp
+include_HEADERS = ibpp.h
+libibpp_la_LDFLAGS = -version-info 2:0:0 -no-undefined
+AM_CXXFLAGS = -DIBPP_GCC -DIBPP_LINUX
diff -ruN Makefile.am Makefile.am
--- Makefile.am	1970-01-01 01:00:00.000000000 +0100
+++ Makefile.am	2006-07-18 08:21:30.145320664 +0200
@@ -0,0 +1,6 @@
+# not a GNU package. You can remove this line, if
+# have all needed files, that a GNU package needs
+AUTOMAKE_OPTIONS = foreign 1.4
+
+SUBDIRS = core
+AM_CXXFLAGS = -DIBPP_GCC -DIBPP_LINUX -fPIC
