diff -Nru havp-0.90.orig/havp/connectiontobrowser.cpp havp-0.90/havp/connectiontobrowser.cpp
--- havp-0.90.orig/havp/connectiontobrowser.cpp	2008-08-03 19:51:45.000000000 +0200
+++ havp-0.90/havp/connectiontobrowser.cpp	2009-07-05 02:13:58.000000000 +0200
@@ -21,6 +21,7 @@
 #include "utils.h"
 
 #include <arpa/inet.h>
+#include <stdio.h>
 
 //Prepare Header for Server
 string ConnectionToBrowser::PrepareHeaderForServer( bool ScannerOff, bool UseParentProxy )
diff -Nru havp-0.90.orig/havp/connectiontohttp.cpp havp-0.90/havp/connectiontohttp.cpp
--- havp-0.90.orig/havp/connectiontohttp.cpp	2007-03-22 19:20:33.000000000 +0100
+++ havp-0.90/havp/connectiontohttp.cpp	2009-07-05 02:16:20.000000000 +0200
@@ -19,6 +19,8 @@
 #include "logfile.h"
 #include "utils.h"
 
+#include <stdio.h>
+
 extern int LL; //LogLevel
 
 //Prepare Header for Browser
diff -Nru havp-0.90.orig/havp/httphandler.cpp havp-0.90/havp/httphandler.cpp
--- havp-0.90.orig/havp/httphandler.cpp	2008-02-13 05:35:22.000000000 +0100
+++ havp-0.90/havp/httphandler.cpp	2009-07-05 02:15:13.000000000 +0200
@@ -20,6 +20,8 @@
 #include "utils.h"
 #include "default.h"
 
+#include <stdio.h>
+
 //Read header
 bool HTTPHandler::ReadHeader( string &headerT )
 {
diff -Nru havp-0.90.orig/havp/logfile.cpp havp-0.90/havp/logfile.cpp
--- havp-0.90.orig/havp/logfile.cpp	2008-11-26 12:07:49.000000000 +0100
+++ havp-0.90/havp/logfile.cpp	2009-07-05 02:07:51.000000000 +0200
@@ -26,6 +26,7 @@
 #include <stdarg.h>
 #include <fcntl.h>
 #include <syslog.h>
+#include <stdio.h>
 #include <map>
 #include <cstdlib>
 #include <cstring>
diff -Nru havp-0.90.orig/havp/scanners/clamlibscanner.cpp havp-0.90/havp/scanners/clamlibscanner.cpp
--- havp-0.90.orig/havp/scanners/clamlibscanner.cpp	2009-03-13 11:35:00.000000000 +0100
+++ havp-0.90/havp/scanners/clamlibscanner.cpp	2009-07-05 02:20:31.000000000 +0200
@@ -19,6 +19,7 @@
 
 #include "clamlibscanner.h"
 
+#include <stdio.h>
 
 bool ClamLibScanner::InitDatabase()
 {
diff -Nru havp-0.90.orig/havp/scanners/drwebscanner.cpp havp-0.90/havp/scanners/drwebscanner.cpp
--- havp-0.90.orig/havp/scanners/drwebscanner.cpp	2007-10-10 15:17:42.000000000 +0200
+++ havp-0.90/havp/scanners/drwebscanner.cpp	2009-07-05 02:19:12.000000000 +0200
@@ -17,6 +17,8 @@
 
 #include "drwebscanner.h"
 
+#include <stdio.h>
+
 /* drweb-clients-4.33-sources */
 
 /* -- SCAN_COMMANDS -------------------------------------------------- */
diff -Nru havp-0.90.orig/havp/scanners/f-protscanner.cpp havp-0.90/havp/scanners/f-protscanner.cpp
--- havp-0.90.orig/havp/scanners/f-protscanner.cpp	2007-09-21 05:11:11.000000000 +0200
+++ havp-0.90/havp/scanners/f-protscanner.cpp	2009-07-05 02:17:41.000000000 +0200
@@ -17,6 +17,7 @@
 
 #include "f-protscanner.h"
 
+#include <stdio.h>
 
 bool FProtScanner::InitDatabase()
 {
