diff -urN QScintilla-gpl-2.6.2.orig/designer-Qt4/designer.pro QScintilla-gpl-2.6.2/designer-Qt4/designer.pro
--- QScintilla-gpl-2.6.2.orig/designer-Qt4/designer.pro	2012-06-27 15:22:31.980241792 +0200
+++ QScintilla-gpl-2.6.2/designer-Qt4/designer.pro	2012-06-27 15:23:39.802393898 +0200
@@ -12,4 +12,5 @@
 target.path = $$[QT_INSTALL_PLUGINS]/designer
 INSTALLS += target
 
-LIBS += -lqscintilla2
+INCLUDEPATH = ../Qt4Qt5
+LIBS += -L../Qt4Qt5 -lqscintilla2
