diff -urN kudzu-1.1.62.orig/sbus.c kudzu-1.1.62/sbus.c
--- kudzu-1.1.62.orig/sbus.c	2003-03-24 17:39:14.000000000 -0500
+++ kudzu-1.1.62/sbus.c	2005-07-12 09:50:06.000000000 -0400
@@ -162,7 +162,7 @@
 		    	devClass = CLASS_NETWORK;
 		} else if (!strcmp(prop, "le")) {
 			type = "Sun Lance Ethernet";
-			module = "ignore";
+			module = "sunlance";
 		    	devClass = CLASS_NETWORK;
 		} else if (!strcmp(prop, "qe")) {
 			prop = prom_getproperty("channel#", &len);
