diff -urN kudzu-1.1.62.orig/pci.c kudzu-1.1.62/pci.c
--- kudzu-1.1.62.orig/pci.c	2004-07-20 11:58:27.000000000 -0400
+++ kudzu-1.1.62/pci.c	2005-11-16 09:38:31.000000000 -0500
@@ -30,6 +30,7 @@
 static int numPciDevices = 0;
 static struct pci_access *pacc=NULL;
 static jmp_buf pcibuf;
+typedef unsigned char byte;
 
 static int devCmp(const void * a, const void * b) {
     const struct pciDevice * one = a;
