diff -Naur topal-64.orig/MIME-tool/Makefile topal-64/MIME-tool/Makefile
--- topal-64.orig/MIME-tool/Makefile	2008-10-24 12:16:45.000000000 +0200
+++ topal-64/MIME-tool/Makefile	2009-04-22 09:36:34.000000000 +0200
@@ -1,10 +1,13 @@
 .PHONY:	all clean realclean distrib
 
+CC=gcc
+CFLAGS=-Wall -O2
+LDFLAGS=
+
 all:	mime-tool
 
 mime-tool:	mime.c
-	gcc -Wall -O2 -o mime-tool mime.c
-	-strip mime-tool
+	$(CC) $(CFLAGS) $(LDFLAGS) -o mime-tool mime.c
 
 realclean:	clean
 	-rm mime-tool
diff -Naur topal-64.orig/Makefile topal-64/Makefile
--- topal-64.orig/Makefile	2008-10-24 12:16:45.000000000 +0200
+++ topal-64/Makefile	2009-04-22 09:37:18.000000000 +0200
@@ -36,26 +36,29 @@
 INSTALLPATHDOC ?= $(INSTALLPATH)/share/doc/topal
 INSTALLPATHPATCHES ?= $(INSTALLPATH)/share/topal/patches
 
+CC=gcc
+CFLAGS=-Wall -O2
+LDFLAGS=
+
 # The default action.
 all:	topal mime-tool README.txt
 
 all2:	package distrib
 
 ada-readline-c.o:	ada-readline-c.c
-	gcc -c -Wall -O2 $(TOPALDEBUG) ada-readline-c.c
+	$(CC) -c $(CFLAGS) $(TOPALDEBUG) ada-readline-c.c
 
 ada-echo-c.o:	ada-echo-c.c
-	gcc -c -Wall -O2 $(TOPALDEBUG) ada-echo-c.c
+	$(CC) -c $(CFLAGS) $(TOPALDEBUG) ada-echo-c.c
 
 externals-c.o:	externals-c.c
-	gcc -c -Wall -O2 $(TOPALDEBUG) externals-c.c
+	$(CC) -c $(CFLAGS) $(TOPALDEBUG) externals-c.c
 
 mime-tool:	
-	make -C MIME-tool
+	$(MAKE) -C MIME-tool
 
 topal:	$(TOPALDEPS)
-	gnatmake -gnatwa -gnato -O2 $(TOPALDEBUG) topal
-	-strip topal
+	gnatmake -gnatwa -gnato -O2 $(TOPALDEBUG) topal -largs "$(LDFLAGS)"
 
 topal.gz:	topal
 	gzip -c9 topal > topal.gz
@@ -87,11 +89,11 @@
 
 install:	all
 	install -d $(INSTALLPATHBIN) $(INSTALLPATHDOC) $(INSTALLPATHMAN)/man1 $(INSTALLPATHPATCHES)
-	install -m 755 -s topal $(INSTALLPATHBIN)
-	install -m 644 README.html README.txt COPYING $(INSTALLPATHDOC)
+	install -m 755 topal $(INSTALLPATHBIN)
+	install -m 644 README.txt $(INSTALLPATHDOC)
 	install -m 644 topal.man $(INSTALLPATHMAN)/man1/topal.1
-	install -m 644 pine-*.patch alpine-*.patch $(INSTALLPATHPATCHES)
-	install -m 755 -s MIME-tool/mime-tool $(INSTALLPATHBIN)
+	install -m 644 alpine-*.patch $(INSTALLPATHPATCHES)
+	install -m 755 MIME-tool/mime-tool $(INSTALLPATHBIN)
 	install -m 644 MIME-tool/mime-tool.man $(INSTALLPATHMAN)/man1/mime-tool.1
 
 clean:
