diff -uraN topal-72-orig/MIME-tool/Makefile topal-72/MIME-tool/Makefile
--- topal-72-orig/MIME-tool/Makefile	2010-02-25 21:28:39.000000000 +0100
+++ topal-72/MIME-tool/Makefile	2010-03-04 13:24:53.691055969 +0100
@@ -1,10 +1,14 @@
 .PHONY:	all clean realclean distrib
 
+CC=gcc
+CFLAGS=-Wall -O2
+LDFLAGS=
+
+
 all:	mime-tool
 
 mime-tool:	mime.c
-	gcc -Wall -O2 -o mime-tool mime.c
-	-strip mime-tool
+	$(CC) $(CFLAGS) $(LDFLAGS) -o mime-tool mime.c
 
 realclean:	clean
 	-rm mime-tool
diff -uraN topal-72-orig/Makefile topal-72/Makefile
--- topal-72-orig/Makefile	2010-02-25 21:28:39.000000000 +0100
+++ topal-72/Makefile	2010-03-04 13:44:58.220101516 +0100
@@ -15,6 +15,10 @@
 
 .PHONY:	all clean realclean distclean package install distrib all2 FORCE
 
+CC=gcc
+CFLAGS=-Wall -O2
+LDFLAGS=
+
 RELEASECODE=$(shell grep '^<dt>' README.html | tail -1 | sed 's/.* release //; sx</a>.*$$xx')
 BUILDDATE=$(shell date '+%Y-%m-%dT%H%M%S%Z')
 
@@ -42,20 +46,19 @@
 all2:	package distrib
 
 ada-readline-c.o:	ada-readline-c.c
-	gcc -c -Wall -O2 $(TOPALDEBUG) ada-readline-c.c
+	$(CC) -c $(CFLAGS) $(TOPALDEBUG) ada-readline-c.c
 
 ada-echo-c.o:	ada-echo-c.c
-	gcc -c -Wall -O2 $(TOPALDEBUG) ada-echo-c.c
+	$(CC) -c $(CFLAGS) $(TOPALDEBUG) ada-echo-c.c
 
 externals-c.o:	externals-c.c
-	gcc -c -Wall -O2 $(TOPALDEBUG) externals-c.c
+	$(CC) -c $(CFLAGS) $(TOPALDEBUG) externals-c.c
 
 mime-tool:	
-	make -C MIME-tool
+	$(MAKE) -C MIME-tool
 
 topal:	$(TOPALDEPS)
-	gnatmake -gnatwa -gnato -O2 $(TOPALDEBUG) topal
-	-strip topal
+	gnatmake -gnatwa -gnato -O2 $(TOPALDEBUG) topal -largs $(LDFLAGS)
 
 topal.gz:	topal
 	gzip -c9 topal > topal.gz
@@ -85,13 +88,13 @@
 README.txt:	README.html
 	lynx -dont_wrap_pre -underscore -nolist -dump README.html > README.txt
 
-install:	all
+install:
 	install -d $(INSTALLPATHBIN) $(INSTALLPATHDOC) $(INSTALLPATHMAN)/man1 $(INSTALLPATHPATCHES)
-	install -m 755 -s topal $(INSTALLPATHBIN)
-	install -m 644 README.html README.txt COPYING $(INSTALLPATHDOC)
+	install -m 755 topal $(INSTALLPATHBIN)
+	install -m 644 README.txt $(INSTALLPATHDOC)
 	install -m 644 topal.man $(INSTALLPATHMAN)/man1/topal.1
-	install -m 644 pine-*.patch alpine-*.patch alpine-*.patch-* $(INSTALLPATHPATCHES)
-	install -m 755 -s MIME-tool/mime-tool $(INSTALLPATHBIN)
+	install -m 644 alpine-*.patch alpine-*.patch-* $(INSTALLPATHPATCHES)
+	install -m 755 MIME-tool/mime-tool $(INSTALLPATHBIN)
 	install -m 644 MIME-tool/mime-tool.man $(INSTALLPATHMAN)/man1/mime-tool.1
 
 clean:
