diff -Nur clients/map.orig/map.c clients/map/map.c
--- clients/map.orig/map.c	2007-09-11 15:56:24.838414157 +0100
+++ clients/map/map.c	2007-09-11 15:52:24.148698032 +0100
@@ -18,7 +18,7 @@
 *****************************************************************************/
 /* Form definition file generated with fdesign. */
 
-#include "forms.h"
+#include <forms.h>
 #include <stdlib.h>
 #include "map.h"
 #include "world_fill.h"
diff -Nur clients/map.orig/map_cb.c clients/map/map_cb.c
--- clients/map.orig/map_cb.c	2007-09-11 15:56:24.838414157 +0100
+++ clients/map/map_cb.c	2007-09-11 15:54:05.190456074 +0100
@@ -18,9 +18,11 @@
 *****************************************************************************/
 #include <stdio.h>
 #include <stdlib.h>
+#include <unistd.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
+#include <arpa/inet.h>
 #include <netdb.h>
 #include <signal.h>
 #include <time.h>
@@ -39,7 +41,7 @@
 	/* This function implements the arccosine function,
 	   returning a value between 0 and two pi. */
 
-	double result, fraction;
+	double result=0.0, fraction;
 
 	fraction=x/y;
 
@@ -76,7 +78,7 @@
 	bzero((char *)&sin,sizeof(struct sockaddr_in));
 	sin.sin_family = AF_INET;
 	
-	if (pse=getservbyname(service,protocol))
+	if ((pse=getservbyname(service,protocol)))
 		sin.sin_port=pse->s_port;
 
 	else if ((sin.sin_port=htons((unsigned short)atoi(service)))==0)
@@ -85,7 +87,7 @@
 	   return -1;
 	}
 
-	if (phe=gethostbyname(host))
+	if ((phe=gethostbyname(host)))
 		bcopy(phe->h_addr,(char *)&sin.sin_addr,phe->h_length);
 
 	else if ((sin.sin_addr.s_addr = inet_addr(host))==INADDR_NONE)
diff -Nur clients/map.orig/map_main.c clients/map/map_main.c
--- clients/map.orig/map_main.c	2007-09-11 15:56:24.838414157 +0100
+++ clients/map/map_main.c	2007-09-11 15:55:58.208896628 +0100
@@ -16,9 +16,11 @@
 * for more details.                                                         *
 *                                                                           *
 *****************************************************************************/
-#include "forms.h"
+#include <forms.h>
 #include "map.h"
 #include <sys/timeb.h>
+#include <unistd.h>
+#include <stdlib.h>
 
 extern int connectsock();
 extern void handler();
@@ -29,7 +31,7 @@
 int main(int argc, char *argv[])
 {
 	int i, skt, x, y, z;
-	char cmd[16], bufr[625], satnamelist[26][26];
+	char bufr[625], satnamelist[26][26];
 	const char *ptrsat;
 	FD_map *fd_map;
 	struct timeb tptr;
