--- a/adodb.c	2005-11-25 20:00:56.000000000 +0100
+++ b/adodb.c	2011-03-06 22:27:06.000000000 +0100
@@ -159,11 +159,7 @@
 
 static void adodb_init_zval(zval *v, char *s)
 {
-	v->type = IS_STRING;
-	v->value.str.val = s;
-	v->value.str.len = strlen(s);
-	v->is_ref = 0;
-	v->refcount = 1;
+	ZVAL_STRING(v, s, 0);
 }
 
 /* {{{ PHP_MINIT_FUNCTION
@@ -277,7 +273,7 @@
 	fci.function_table = EG(function_table);
 	fci.function_name = &adodb_zvals[zval_ocifetch];
 	fci.symbol_table = NULL;
-	fci.object_pp = NULL;
+	/* fci.object_pp = NULL; */
 	fci.retval_ptr_ptr = &retval;
 	fci.param_count = 2;
 	fci.params = &params[1];
