--- coccinelle-1.0.0-rc14/configure.ac
+++ coccinelle-1.0.0-rc14/configure.ac
@@ -3,6 +3,8 @@
 AC_INIT([coccinelle], m4_esyscmd([cat ./version | tr -d '\n']), [cocci@diku.dk], [], [http://coccinelle.lip6.fr/])
 AC_CONFIG_MACRO_DIR([setup])
 AC_CONFIG_AUX_DIR([setup])
+AM_INIT_AUTOMAKE
+AC_CONFIG_FILES([Makefile])
 
 AC_SUBST([CONFIGURE_FLAGS], ["$*"])
 AC_SUBST([COCCI_VERSION], ["$PACKAGE_VERSION"])
--- coccinelle-1.0.0-rc14/setup/cocci.m4
+++ coccinelle-1.0.0-rc14/setup/cocci.m4
@@ -285,7 +285,7 @@
     AC_SUBST([$1], [no])
   ],
   [dnl  find the tool
-    AC_COCCI_FINDTOOL([$1],[[$]$1])
+    AC_COCCI_FINDTOOL([$1],[$2])
   ])
 
   AS_IF([test -z "[$]$1" -o "x[$]$1" = xno],
