--- nscemake/md3.h.old  2009-01-23 11:25:56.000000000 +0100
+++ nscemake/md3.h      2009-01-23 11:26:12.000000000 +0100
@@ -32,6 +32,8 @@
 #ifndef __NSCEMAKE_MD3_H
 #define __NSCEMAKE_MD3_H

+#include <cstdlib>
+#include <cstring>
 #include "base.h"
 #include "converter.h"

--- nscemake/md5mesh.cpp.old    2009-01-23 11:36:28.000000000 +0100
+++ nscemake/md5mesh.cpp        2009-01-23 11:37:04.000000000 +0100
@@ -30,6 +30,7 @@
  ***************************************************************************/

 #include "md5mesh.h"
+#include <cstdlib>

 #include <neoengine/file.h>
 #include <neoengine/nemath.h>
--- nscemake/q3map.cpp.old      2009-01-23 11:39:43.000000000 +0100
+++ nscemake/q3map.cpp  2009-01-23 11:40:18.000000000 +0100
@@ -30,6 +30,7 @@
  ***************************************************************************/

 #include "q3map.h"
+#include <cstdlib>
 #include "q3maptypes.h"

 #include <neoengine/vertex.h>
--- nscemake/ut2.cpp.old        2009-01-23 11:41:29.000000000 +0100
+++ nscemake/ut2.cpp    2009-01-23 11:41:52.000000000 +0100
@@ -30,6 +30,7 @@
  ***************************************************************************/

 #include "ut2.h"
+#include <cstring>

 #include <neoengine/file.h>
 #include <neoengine/logstream.h>

