diff -Nurpa neoengine.orig/neoengine/core.cpp neoengine/neoengine/core.cpp
--- neoengine.orig/neoengine/core.cpp	2008-12-16 09:43:18.000000000 +0300
+++ neoengine/neoengine/core.cpp	2008-12-16 09:48:05.000000000 +0300
@@ -64,6 +64,8 @@
 #  include <sys/sysctl.h>
 #endif
 
+#include <cstdlib>
+
 using namespace std;
 
 
diff -Nurpa neoengine.orig/neoengine/hash.h neoengine/neoengine/hash.h
--- neoengine.orig/neoengine/hash.h	2008-12-16 09:43:18.000000000 +0300
+++ neoengine/neoengine/hash.h	2008-12-16 09:46:50.000000000 +0300
@@ -30,7 +30,7 @@
 #include "base.h"
 
 #include <string>
-
+#include <cstring>
  
  /**
   * \file hash.h
diff -Nurpa neoengine.orig/neoengine/particle.cpp neoengine/neoengine/particle.cpp
--- neoengine.orig/neoengine/particle.cpp	2008-12-16 09:43:17.000000000 +0300
+++ neoengine/neoengine/particle.cpp	2008-12-16 09:50:37.000000000 +0300
@@ -29,6 +29,8 @@
 #include "timer.h"
 #include "logstream.h"
 
+#include <cstdlib>
+
 using namespace std;
 
 
diff -Nurpa neoengine.orig/neoengine/shader.cpp neoengine/neoengine/shader.cpp
--- neoengine.orig/neoengine/shader.cpp	2008-12-16 09:43:18.000000000 +0300
+++ neoengine/neoengine/shader.cpp	2008-12-16 09:53:32.000000000 +0300
@@ -32,6 +32,8 @@
 #include "logstream.h"
 #include "strutil.h"
 
+#include <cstdlib>
+
 using namespace std;
 
 
diff -Nurpa neoengine.orig/neoengine/strutil.cpp neoengine/neoengine/strutil.cpp
--- neoengine.orig/neoengine/strutil.cpp	2008-12-16 09:43:18.000000000 +0300
+++ neoengine/neoengine/strutil.cpp	2008-12-16 09:54:23.000000000 +0300
@@ -25,6 +25,7 @@
  ***************************************************************************/
 
 #include "strutil.h"
+#include <cstring>
 
 using namespace std;
 
diff -Nurpa neoengine.orig/neoengine/tcp.cpp neoengine/neoengine/tcp.cpp
--- neoengine.orig/neoengine/tcp.cpp	2008-12-16 09:43:18.000000000 +0300
+++ neoengine/neoengine/tcp.cpp	2008-12-16 09:56:21.000000000 +0300
@@ -54,6 +54,9 @@
 #  define SOCKET_ERROR    -1
 #endif /* __APPLE__ */
 
+#include <cstring>
+#include <cstdlib>
+
 using namespace std;
 
 
diff -Nurpa neoengine.orig/neoengine/udp.cpp neoengine/neoengine/udp.cpp
--- neoengine.orig/neoengine/udp.cpp	2008-12-16 09:43:18.000000000 +0300
+++ neoengine/neoengine/udp.cpp	2008-12-16 09:57:05.000000000 +0300
@@ -57,6 +57,8 @@
 #  define SOCKET_ERROR    -1
 #endif /* __APPLE__ */
 
+#include <cstring>
+
 using namespace std;
 
 
diff -Nurpa neoengine.orig/neochunkio/texlayer.cpp neoengine/neochunkio/texlayer.cpp
--- neoengine.orig/neochunkio/texlayer.cpp	2004-03-27 22:25:18.000000000 +0300
+++ neoengine/neochunkio/texlayer.cpp	2008-12-17 01:27:41.000000000 +0300
@@ -40,6 +40,7 @@
 #include <vector>
 #include <sstream>
 #include <iomanip>
+#include <cstdlib>
 
 using namespace std;
 using namespace NeoEngine;
diff -Nurpa neoengine.orig/neochunkio/vertexbuffer.cpp neoengine/neochunkio/vertexbuffer.cpp
--- neoengine.orig/neochunkio/vertexbuffer.cpp	2004-03-15 10:35:50.000000000 +0300
+++ neoengine/neochunkio/vertexbuffer.cpp	2008-12-17 01:28:12.000000000 +0300
@@ -37,7 +37,7 @@
 
 #include <vector>
 #include <string>
-
+#include <cstdlib>
 
 using namespace std;
 using namespace NeoEngine;
diff -Nurpa neoengine.orig/neodevopengl/extensions.h neoengine/neodevopengl/extensions.h
--- neoengine.orig/neodevopengl/extensions.h	2004-06-12 20:47:42.000000000 +0400
+++ neoengine/neodevopengl/extensions.h	2008-12-16 23:17:49.000000000 +0300
@@ -486,8 +486,8 @@ extern fpglGetProgramivARB              
 typedef GLuint     (NEOGLAPIENTRY *                          fpglGenFragmentShadersATI)( GLuint );
 typedef void       (NEOGLAPIENTRY *                          fpglBindFragmentShaderATI)( GLuint );
 typedef void       (NEOGLAPIENTRY *                          fpglDeleteFragmentShaderATI)( GLuint );
-typedef void       (NEOGLAPIENTRY *                          fpglBeginFragmentShaderATI)( GLvoid );
-typedef void       (NEOGLAPIENTRY *                          fpglEndFragmentShaderATI)( GLvoid );
+typedef void       (NEOGLAPIENTRY *                          fpglBeginFragmentShaderATI)( void );
+typedef void       (NEOGLAPIENTRY *                          fpglEndFragmentShaderATI)( void );
 typedef void       (NEOGLAPIENTRY *                          fpglPassTexCoordATI)( GLuint, GLuint, GLenum );
 typedef void       (NEOGLAPIENTRY *                          fpglSampleMapATI)( GLuint, GLuint, GLenum );
 typedef void       (NEOGLAPIENTRY *                          fpglColorFragmentOp1ATI)( GLenum, GLuint, GLuint, GLuint, GLuint, GLuint, GLuint );
diff -Nurpa neoengine.orig/neodevopengl/open.cpp neoengine/neodevopengl/open.cpp
--- neoengine.orig/neodevopengl/open.cpp	2004-02-11 19:33:37.000000000 +0300
+++ neoengine/neodevopengl/open.cpp	2008-12-16 23:09:49.000000000 +0300
@@ -32,6 +32,8 @@
 
 #include <neoengine/logstream.h>
 
+#include <cstdlib>
+
 using namespace std;
 using namespace NeoEngine;
 
diff -Nurpa neoengine.orig/neodevopengl/shader-ati.cpp neoengine/neodevopengl/shader-ati.cpp
--- neoengine.orig/neodevopengl/shader-ati.cpp	2004-06-29 20:59:10.000000000 +0400
+++ neoengine/neodevopengl/shader-ati.cpp	2008-12-16 23:13:51.000000000 +0300
@@ -29,6 +29,7 @@
 #include <neoengine/scenenode.h>
 #include <neoengine/strutil.h>
 #include <neoengine/logstream.h>
+#include <cstdlib>
 
 using namespace std;
 using namespace NeoEngine;
diff -Nurpa neoengine.orig/neodevopengl/vertexbuffer.cpp neoengine/neodevopengl/vertexbuffer.cpp
--- neoengine.orig/neodevopengl/vertexbuffer.cpp	2004-06-12 08:02:02.000000000 +0400
+++ neoengine/neodevopengl/vertexbuffer.cpp	2008-12-17 01:08:40.000000000 +0300
@@ -25,6 +25,7 @@
  ***************************************************************************/
 
 #include "vertexbuffer.h"
+#include <cstring>
 
 using namespace NeoEngine;
 
