diff -ur neoengine.orig/neodevopengl/statistics.cpp neoengine/neodevopengl/statistics.cpp
--- neoengine.orig/neodevopengl/statistics.cpp	2003-12-21 13:49:12.000000000 +0200
+++ neoengine/neodevopengl/statistics.cpp	2009-08-10 11:33:21.000000000 +0300
@@ -24,6 +24,8 @@
 
  ***************************************************************************/
 
+#include <stdio.h>
+
 #include "statistics.h"
 #include "device.h"
 
