--- neofczip/Makefile.am.dynamic.old	2009-03-02 09:17:40.000000000 +0100
+++ neofczip/Makefile.am.dynamic	2009-03-02 09:17:49.000000000 +0100
@@ -1,11 +1,11 @@
 
 lib_LTLIBRARIES = libneofczip.la
 
-libneofczip_la_SOURCES = neofczip.cpp adler32.c compress.c crc32.c deflate.c gzio.c infback.c inffast.c inflate.c inftrees.c minigzip.c trees.c uncompr.c zutil.c
+libneofczip_la_SOURCES = neofczip.cpp
 
 EXTRA_DIST = zlib.h neofczip.dsp Makefile.am.static Makefile.am.dynamic
 
-libneofczip_la_LIBADD  = @NEOENGINE_EXTRA_LIBS@
+libneofczip_la_LIBADD  = @NEOENGINE_EXTRA_LIBS@ -lz
 
 # set the include path found by configure
 INCLUDES = -I..
--- neoicpng/Makefile.am.dynamic.old	2009-03-02 10:03:19.000000000 +0100
+++ neoicpng/Makefile.am.dynamic	2009-03-02 10:03:27.000000000 +0100
@@ -1,9 +1,7 @@
 
 lib_LTLIBRARIES = libneoicpng.la
 
-libneoicpng_la_SOURCES = neoicpng.cpp png.c pngerror.c pngget.c pngmem.c pngpread.c pngread.c pngrio.c \
-	pngrtran.c pngrutil.c pngset.c pngtrans.c pngwio.c pngwrite.c pngwtran.c pngwutil.c deflate.c \
-	inflate.c trees.c zutil.c infblock.c inftrees.c infcodes.c infutil.c inffast.c adler32.c crc32.c
+libneoicpng_la_SOURCES = neoicpng.cpp 
 
 EXTRA_DIST = pnggccrd.c pngvcrd.c deflate.h infblock.h infcodes.h inffast.h inffixed.h inftrees.h \
 	infutil.h pngasmrd.h pngconf.h png.h trees.h zconf.h zlib.h zutil.h \
@@ -11,7 +9,7 @@
 	neoicpng-static.dsp neoicpng-static.vcproj neoicpng-static.dev \
 	LICENSE Makefile.am.static Makefile.am.dynamic
 
-libneoicpng_la_LIBADD  = @NEOENGINE_EXTRA_LIBS@
+libneoicpng_la_LIBADD  = @NEOENGINE_EXTRA_LIBS@ -lpng
 
 # set the include path found by configure
 INCLUDES = -I..
--- neofcbzip2/Makefile.am.dynamic.old	2009-03-02 10:20:40.000000000 +0100
+++ neofcbzip2/Makefile.am.dynamic	2009-03-02 10:20:49.000000000 +0100
@@ -1,13 +1,13 @@
 
 lib_LTLIBRARIES = libneofcbzip2.la
 
-libneofcbzip2_la_SOURCES = neofcbzip2.cpp blocksort.c bzlib.c compress.c crctable.c decompress.c huffman.c randtable.c
+libneofcbzip2_la_SOURCES = neofcbzip2.cpp
 
 EXTRA_DIST = bzlib.h bzlib_private.h neofcbzip2.dsp neofcbzip2.vcproj neofcbzip2.dev \
 	neofcbzip2-static.dsp neofcbzip2-static.vcproj neofcbzip2-static.dev \
 	Makefile.am.static Makefile.am.dynamic
 
-libneofcbzip2_la_LIBADD  = @NEOENGINE_EXTRA_LIBS@
+libneofcbzip2_la_LIBADD  = @NEOENGINE_EXTRA_LIBS@ -lbz2
 
 # set the include path found by configure
 INCLUDES = -I..
--- neoicjpeg/Makefile.am.dynamic.old	2004-03-22 11:04:21.000000000 +0100
+++ neoicjpeg/Makefile.am.dynamic	2009-03-02 10:26:00.000000000 +0100
@@ -1,15 +1,7 @@
 
 lib_LTLIBRARIES = libneoicjpeg.la
 
-libneoicjpeg_la_SOURCES = neoicjpeg.cpp \
-	jcapimin.c  jcmarker.c  jdapimin.c  jdinput.c   jdtrans.c   jidctred.c  jquant1.c \
-	jcapistd.c  jcmaster.c  jdapistd.c  jdmainct.c  jerror.c    jquant2.c \
-	jccoefct.c  jcomapi.c   jdatadst.c  jdmarker.c  jfdctflt.c  jutils.c \
-	jccolor.c   jcparam.c   jdatasrc.c  jdmaster.c  jfdctfst.c  \
-	jcdctmgr.c  jcphuff.c   jdcoefct.c  jdmerge.c   jfdctint.c  jmemmgr.c \
-	jchuff.c    jcprepct.c  jdcolor.c   jdphuff.c   jidctflt.c  \
-	jcinit.c    jcsample.c  jddctmgr.c  jdpostct.c  jidctfst.c  jmemnobs.c \
-	jcmainct.c  jctrans.c   jdhuff.c    jdsample.c  jidctint.c
+libneoicjpeg_la_SOURCES = neoicjpeg.cpp
 
 EXTRA_DIST = README jchuff.h jdhuff.h jmemsys.h jpeglib.h \
 	jconfig.h jerror.h jmorecfg.h jversion.h jdct.h \
@@ -17,7 +9,7 @@
 	neoicjpeg-static.dsp neoicjpeg-static.vcproj neoicjpeg-static.dev \
 	Makefile.am.static Makefile.am.dynamic
 
-libneoicjpeg_la_LIBADD  = @NEOENGINE_EXTRA_LIBS@
+libneoicjpeg_la_LIBADD  = @NEOENGINE_EXTRA_LIBS@ -ljpeg
 
 # set the include path found by configure
 INCLUDES = -I..
