--- configure.ac.orig	2012-07-06 14:19:29.834778487 +0200
+++ configure.ac	2012-07-06 14:22:12.174921371 +0200
@@ -433,7 +433,7 @@
 ###########################################################
 GGL_MODULE_DIR=$libdir/$GGL_BASE_DIR/modules
 GGL_INCLUDE_DIR=$includedir/$GGL_BASE_DIR
-GGL_SYSDEPS_INCLUDE_DIR=$libdir/$GGL_BASE_DIR/include
+GGL_SYSDEPS_INCLUDE_DIR=$GGL_INCLUDE_DIR
 GGL_LIBEXEC_DIR=$libdir/$GGL_BASE_DIR
 GGL_RESOURCE_DIR=$datadir/$GGL_BASE_DIR
 
@@ -895,11 +895,15 @@
 
 # Check NetworkManager
 has_network_manager=no
-PKG_CHECK_MODULES(NETWORK_MANAGER,
+AC_ARG_WITH([networkmanager], AS_HELP_STRING([--without-networkmanager], [Build without networkmanager support (default: test)]))
+
+AS_IF([test "x$with_networkmanager" != "xno"], [
+  PKG_CHECK_MODULES(NETWORK_MANAGER,
                   [NetworkManager >= 0.7], [has_network_manager=yes],
                   [PKG_CHECK_MODULES(NETWORK_MANAGER,
                    [NetworkManager >= 0.6], [has_network_manager=old],
                    [has_network_manager=no])])
+])
 
 if test x$has_network_manager != xno; then
   PREDEFINED_MACROS="$PREDEFINED_MACROS -DHAVE_NETWORK_MANAGER"
