diff -Naur brltty-5.0/Autostart/Udev/udev.rules brltty-5.0.new/Autostart/Udev/udev.rules
--- brltty-5.0/Autostart/Udev/udev.rules	2014-01-27 10:39:20.000000000 -0800
+++ brltty-5.0.new/Autostart/Udev/udev.rules	2014-02-09 04:24:08.146802299 -0800
@@ -174,20 +174,22 @@
 # FreedomScientific [Focus Blue]
 ENV{PRODUCT}=="f4e/114/*", ENV{BRLTTY_BRAILLE_DRIVER}="fs", GOTO="brltty_usb_run"
 
+# Seika rules are commented out, because they conflict with USB-to-serial
+# converters.
 # Device: 10C4:EA60
 # Generic Identifier
 # Vendor: Cygnal Integrated Products, Inc.
 # Product: CP210x UART Bridge / myAVR mySmartUSB light
 # BrailleMemo [all models]
 # Seika [Braille Display]
-ENV{PRODUCT}=="10c4/ea60/*", ENV{BRLTTY_BRAILLE_DRIVER}="mm,sk", GOTO="brltty_usb_run"
+#ENV{PRODUCT}=="10c4/ea60/*", ENV{BRLTTY_BRAILLE_DRIVER}="mm,sk", GOTO="brltty_usb_run"
 
 # Device: 10C4:EA80
 # Generic Identifier
 # Vendor: Cygnal Integrated Products, Inc.
 # Product: CP210x UART Bridge
 # Seika [Note Taker]
-ENV{PRODUCT}=="10c4/ea80/*", ENV{BRLTTY_BRAILLE_DRIVER}="sk", GOTO="brltty_usb_run"
+#ENV{PRODUCT}=="10c4/ea80/*", ENV{BRLTTY_BRAILLE_DRIVER}="sk", GOTO="brltty_usb_run"
 
 # Device: 1C71:C005
 # HumanWare [all models]
