diff -Naur brltty-4.5/Hotplug/udev.rules brltty-4.5.new/Hotplug/udev.rules
--- brltty-4.5/Hotplug/udev.rules	2013-03-27 15:51:37.000000000 -0700
+++ brltty-4.5.new/Hotplug/udev.rules	2013-08-19 06:32:11.557621082 -0700
@@ -172,19 +172,20 @@
 # FreedomScientific [Focus Blue]
 ENV{PRODUCT}=="f4e/114/*", ENV{BRLTTY_DRIVER}="fs", GOTO="brltty_usb_run"
 
+# Seika rules are commented out.  They conflict with USB-to-serial converters.
 # Device: 10C4:EA60
 # Generic Identifier
 # Vendor: Cygnal Integrated Products, Inc.
 # Product: CP210x UART Bridge / myAVR mySmartUSB light
 # Seika [Braille Display]
-ENV{PRODUCT}=="10c4/ea60/*", ENV{BRLTTY_DRIVER}="sk", GOTO="brltty_usb_run"
+# ENV{PRODUCT}=="10c4/ea60/*", ENV{BRLTTY_DRIVER}="sk", GOTO="brltty_usb_run"
 
 # Device: 10C4:EA80
 # Generic Identifier
 # Vendor: Cygnal Integrated Products, Inc.
 # Product: CP210x UART Bridge
 # Seika [Note Taker]
-ENV{PRODUCT}=="10c4/ea80/*", ENV{BRLTTY_DRIVER}="sk", GOTO="brltty_usb_run"
+# ENV{PRODUCT}=="10c4/ea80/*", ENV{BRLTTY_DRIVER}="sk", GOTO="brltty_usb_run"
 
 # Device: 1C71:C005
 # HumanWare [all models]
