diff -Nur jtagger/com/googlepages/dronten/jtagger/gui/albumtab/AlbumOptionsPanel.java jtagger_patched/com/googlepages/dronten/jtagger/gui/albumtab/AlbumOptionsPanel.java
--- jtagger/com/googlepages/dronten/jtagger/gui/albumtab/AlbumOptionsPanel.java	2008-04-28 20:58:50.000000000 +0300
+++ jtagger_patched/com/googlepages/dronten/jtagger/gui/albumtab/AlbumOptionsPanel.java	2008-07-12 19:46:48.000000000 +0300
@@ -190,7 +190,7 @@
      *
      * @param actionEvent
      */
-    @Override public void actionPerformed(ActionEvent actionEvent) {
+    public void actionPerformed(ActionEvent actionEvent) {
         if (actionEvent.getSource() == aResetButton) {
             aArtistCheck.setSelected(false);
             aArtistInput.setText("");
diff -Nur jtagger/com/googlepages/dronten/jtagger/gui/dlg/MessageDialog.java jtagger_patched/com/googlepages/dronten/jtagger/gui/dlg/MessageDialog.java
--- jtagger/com/googlepages/dronten/jtagger/gui/dlg/MessageDialog.java	2008-04-28 20:58:50.000000000 +0300
+++ jtagger_patched/com/googlepages/dronten/jtagger/gui/dlg/MessageDialog.java	2008-07-12 19:56:29.000000000 +0300
@@ -56,7 +56,7 @@
     /**
      * @param actionEvent
      */
-    @Override public void actionPerformed(ActionEvent actionEvent) {
+    public void actionPerformed(ActionEvent actionEvent) {
         setVisible(false);
     }
 
@@ -81,4 +81,4 @@
     public void setText(String text) {
         aEditor.setText(text);
     }
-}
\ Dosya sonunda yenisatır yok.
+}
diff -Nur jtagger/com/googlepages/dronten/jtagger/gui/dlg/ProgressDialog.java jtagger_patched/com/googlepages/dronten/jtagger/gui/dlg/ProgressDialog.java
--- jtagger/com/googlepages/dronten/jtagger/gui/dlg/ProgressDialog.java	2008-04-28 20:58:50.000000000 +0300
+++ jtagger_patched/com/googlepages/dronten/jtagger/gui/dlg/ProgressDialog.java	2008-07-12 19:56:12.000000000 +0300
@@ -168,7 +168,7 @@
      *
      * @param actionEvent
      */
-    @Override public void actionPerformed(ActionEvent actionEvent) {
+    public void actionPerformed(ActionEvent actionEvent) {
         close();
     }
 
@@ -309,4 +309,4 @@
         aMajorProgress.setVisible(true);
         pack();
     }
-}
\ Dosya sonunda yenisatır yok.
+}
diff -Nur jtagger/com/googlepages/dronten/jtagger/gui/dlg/setup/GeneralSetup.java jtagger_patched/com/googlepages/dronten/jtagger/gui/dlg/setup/GeneralSetup.java
--- jtagger/com/googlepages/dronten/jtagger/gui/dlg/setup/GeneralSetup.java	2008-04-28 20:58:50.000000000 +0300
+++ jtagger_patched/com/googlepages/dronten/jtagger/gui/dlg/setup/GeneralSetup.java	2008-07-12 19:58:00.000000000 +0300
@@ -50,7 +50,7 @@
     /**
      * @param actionEvent
      */
-    @Override public void actionPerformed(ActionEvent actionEvent) {
+    public void actionPerformed(ActionEvent actionEvent) {
         if (actionEvent.getSource() == aBrowseDir) {
             JFileChooser chooser = new JFileChooser();
 
diff -Nur jtagger/com/googlepages/dronten/jtagger/gui/dlg/setup/MP3Setup.java jtagger_patched/com/googlepages/dronten/jtagger/gui/dlg/setup/MP3Setup.java
--- jtagger/com/googlepages/dronten/jtagger/gui/dlg/setup/MP3Setup.java	2008-04-28 20:58:50.000000000 +0300
+++ jtagger_patched/com/googlepages/dronten/jtagger/gui/dlg/setup/MP3Setup.java	2008-07-12 19:57:45.000000000 +0300
@@ -55,7 +55,7 @@
     /**
      * @param actionEvent
      */
-    @Override public void actionPerformed(ActionEvent actionEvent) {
+    public void actionPerformed(ActionEvent actionEvent) {
     }
 
 
diff -Nur jtagger/com/googlepages/dronten/jtagger/gui/dlg/setup/SetupDialog.java jtagger_patched/com/googlepages/dronten/jtagger/gui/dlg/setup/SetupDialog.java
--- jtagger/com/googlepages/dronten/jtagger/gui/dlg/setup/SetupDialog.java	2008-04-28 20:58:50.000000000 +0300
+++ jtagger_patched/com/googlepages/dronten/jtagger/gui/dlg/setup/SetupDialog.java	2008-07-12 19:57:06.000000000 +0300
@@ -76,7 +76,7 @@
      *
      * @param actionEvent
      */
-    @Override public void actionPerformed(ActionEvent actionEvent) {
+    public void actionPerformed(ActionEvent actionEvent) {
         for (BaseSetup panel : aSetupPanels) {
             panel.save();
         }
@@ -87,7 +87,7 @@
     /**
      * @param listSelectionEvent
      */
-    @Override public void valueChanged(ListSelectionEvent listSelectionEvent) {
+    public void valueChanged(ListSelectionEvent listSelectionEvent) {
         int index = aList.getSelectedIndex();
 
         if (aCurrentSetup != null) {
diff -Nur jtagger/com/googlepages/dronten/jtagger/gui/filenametab/FileOptionsPanel.java jtagger_patched/com/googlepages/dronten/jtagger/gui/filenametab/FileOptionsPanel.java
--- jtagger/com/googlepages/dronten/jtagger/gui/filenametab/FileOptionsPanel.java	2008-04-28 20:58:50.000000000 +0300
+++ jtagger_patched/com/googlepages/dronten/jtagger/gui/filenametab/FileOptionsPanel.java	2008-07-12 19:46:13.000000000 +0300
@@ -205,7 +205,7 @@
      *
      * @param actionEvent
      */
-    @Override public void actionPerformed(ActionEvent actionEvent) {
+    public void actionPerformed(ActionEvent actionEvent) {
         aUseTitle.setSelected(false);
         aUseText.setSelected(false);
         aInsertAlbum.setSelected(false);
diff -Nur jtagger/com/googlepages/dronten/jtagger/gui/leftpanel/DirTree.java jtagger_patched/com/googlepages/dronten/jtagger/gui/leftpanel/DirTree.java
--- jtagger/com/googlepages/dronten/jtagger/gui/leftpanel/DirTree.java	2008-04-28 20:58:50.000000000 +0300
+++ jtagger_patched/com/googlepages/dronten/jtagger/gui/leftpanel/DirTree.java	2008-07-12 20:39:09.000000000 +0300
@@ -150,7 +150,7 @@
      *
      * @param e
      */
-    @Override public void actionPerformed(ActionEvent e) {
+    public void actionPerformed(ActionEvent e) {
         if (aSelected == null || aSelected == aTop) {
             return;
         }
@@ -376,7 +376,7 @@
     /**
      * @param event
      */
-    @Override public void treeCollapsed(TreeExpansionEvent event) {
+    public void treeCollapsed(TreeExpansionEvent event) {
     }
 
 
@@ -385,7 +385,7 @@
      *
      * @param event
      */
-    @Override public void treeExpanded(TreeExpansionEvent event) {
+    public void treeExpanded(TreeExpansionEvent event) {
         //DefaultMutableTreeNode dirNode = (DefaultMutableTreeNode) event.getPath().getLastPathComponent();
 
         //addSubDirs(dirNode);
diff -Nur jtagger/com/googlepages/dronten/jtagger/gui/leftpanel/LeftPanel.java jtagger_patched/com/googlepages/dronten/jtagger/gui/leftpanel/LeftPanel.java
--- jtagger/com/googlepages/dronten/jtagger/gui/leftpanel/LeftPanel.java	2008-04-28 20:58:50.000000000 +0300
+++ jtagger_patched/com/googlepages/dronten/jtagger/gui/leftpanel/LeftPanel.java	2008-07-12 19:34:49.000000000 +0300
@@ -89,7 +89,7 @@
      *
      * @param path
      */
-    @Override public void valueChanged(String path) {
+    public void valueChanged(String path) {
         AudioThread.get().stopPlaying();
 
         if (path != null) {
diff -Nur jtagger/com/googlepages/dronten/jtagger/gui/tracktab/TrackOptionsPanel.java jtagger_patched/com/googlepages/dronten/jtagger/gui/tracktab/TrackOptionsPanel.java
--- jtagger/com/googlepages/dronten/jtagger/gui/tracktab/TrackOptionsPanel.java	2008-04-28 20:58:50.000000000 +0300
+++ jtagger_patched/com/googlepages/dronten/jtagger/gui/tracktab/TrackOptionsPanel.java	2008-07-12 19:47:43.000000000 +0300
@@ -202,7 +202,7 @@
      *
      * @param actionEvent Event message
      */
-    @Override public void actionPerformed(ActionEvent actionEvent) {
+    public void actionPerformed(ActionEvent actionEvent) {
         if (actionEvent.getSource() == aUpdateButton) {
             writeTags();
         }
diff -Nur jtagger/com/googlepages/dronten/jtagger/gui/tracktab/TrackTablePanel.java jtagger_patched/com/googlepages/dronten/jtagger/gui/tracktab/TrackTablePanel.java
--- jtagger/com/googlepages/dronten/jtagger/gui/tracktab/TrackTablePanel.java	2008-04-28 20:58:50.000000000 +0300
+++ jtagger_patched/com/googlepages/dronten/jtagger/gui/tracktab/TrackTablePanel.java	2008-07-12 19:45:27.000000000 +0300
@@ -112,7 +112,7 @@
      *
      * @param actionEvent
      */
-    @Override public void actionPerformed(ActionEvent actionEvent) {
+    public void actionPerformed(ActionEvent actionEvent) {
         try {
             if (actionEvent.getSource() == aDeleteButton) {
                 ListSelectionModel lsm = aTable.getSelectionModel();
diff -Nur jtagger/com/googlepages/dronten/jtagger/JTagger.java jtagger_patched/com/googlepages/dronten/jtagger/JTagger.java
--- jtagger/com/googlepages/dronten/jtagger/JTagger.java	2008-04-28 21:10:08.000000000 +0300
+++ jtagger_patched/com/googlepages/dronten/jtagger/JTagger.java	2008-07-12 19:32:58.000000000 +0300
@@ -116,7 +116,7 @@
      *
      * @param actionEvent Menu action event
      */
-    @Override public void actionPerformed(ActionEvent actionEvent) {
+    public void actionPerformed(ActionEvent actionEvent) {
         if (actionEvent.getSource() == aReloadItem) {
             reloadDirectory();
         }
diff -Nur jtagger/com/googlepages/dronten/jtagger/model/AlbumModel.java jtagger_patched/com/googlepages/dronten/jtagger/model/AlbumModel.java
--- jtagger/com/googlepages/dronten/jtagger/model/AlbumModel.java	2008-04-28 20:58:50.000000000 +0300
+++ jtagger_patched/com/googlepages/dronten/jtagger/model/AlbumModel.java	2008-07-12 20:45:14.000000000 +0300
@@ -49,7 +49,7 @@
      *
      * @param event
      */
-    @Override public void actionPerformed(ActionEvent event) {
+    public void actionPerformed(ActionEvent event) {
         if (event.getSource() == JTagger.get().getMainWindow().getAlbumPane().getAlbumTable().aTestRenameButton) {
             try {
                 AlbumRenamer renamer = JTagger.get().getMainWindow().getAlbumPane().getAlbumOptionsPanel().getRenamer();
@@ -120,7 +120,7 @@
     /**
      * @return column count
      */
-    @Override public int getColumnCount() {
+    public int getColumnCount() {
         return 6;
     }
 
@@ -159,7 +159,7 @@
     /**
      * @return number of rows
      */
-    @Override public int getRowCount() {
+    public int getRowCount() {
         return JTagger.get().getMainWindow().getTrackPane().getTrackTable().getTrackModel().getRowCount();
     }
 
@@ -169,7 +169,7 @@
      * @param col
      * @return get object to paint
      */
-    @Override public Object getValueAt(int row, int col) {
+    public Object getValueAt(int row, int col) {
         AudioTrack track = JTagger.get().getMainWindow().getTrackPane().getTrackTable().getTrackModel().getTrack(row);
         AudioTrack testTrack = JTagger.get().getMainWindow().getTrackPane().getTrackTable().getTrackModel().getTestTrack(row);
 
diff -Nur jtagger/com/googlepages/dronten/jtagger/model/FileModel.java jtagger_patched/com/googlepages/dronten/jtagger/model/FileModel.java
--- jtagger/com/googlepages/dronten/jtagger/model/FileModel.java	2008-04-28 20:58:50.000000000 +0300
+++ jtagger_patched/com/googlepages/dronten/jtagger/model/FileModel.java	2008-07-12 20:37:57.000000000 +0300
@@ -49,7 +49,7 @@
      *
      * @param event Action event
      */
-    @Override public void actionPerformed(ActionEvent event) {
+    public void actionPerformed(ActionEvent event) {
         if (event.getSource() == JTagger.get().getMainWindow().getFilePane().getFileTable().aTestRenameButton) {
             try {
                 FileRenamer renamer = JTagger.get().getMainWindow().getFilePane().getFileOptionsPanel().getRenamer();
@@ -120,7 +120,7 @@
     /**
      * @return Number of columns
      */
-    @Override public int getColumnCount() {
+    public int getColumnCount() {
         return 3;
     }
 
@@ -146,7 +146,7 @@
     /**
      * @return Number of rows
      */
-    @Override public int getRowCount() {
+    public int getRowCount() {
         return JTagger.get().getMainWindow().getTrackPane().getTrackTable().getTrackModel().getRowCount();
     }
 
@@ -156,7 +156,7 @@
      * @param col
      * @return Object to paint in cell
      */
-    @Override public Object getValueAt(int row, int col) {
+    public Object getValueAt(int row, int col) {
         AudioTrack track = JTagger.get().getMainWindow().getTrackPane().getTrackTable().getTrackModel().getTrack(row);
         AudioTrack testTrack = JTagger.get().getMainWindow().getTrackPane().getTrackTable().getTrackModel().getTestTrack(row);
 
diff -Nur jtagger/com/googlepages/dronten/jtagger/model/TitleModel.java jtagger_patched/com/googlepages/dronten/jtagger/model/TitleModel.java
--- jtagger/com/googlepages/dronten/jtagger/model/TitleModel.java	2008-04-28 20:58:50.000000000 +0300
+++ jtagger_patched/com/googlepages/dronten/jtagger/model/TitleModel.java	2008-07-12 20:54:34.000000000 +0300
@@ -49,7 +49,7 @@
      *
      * @param event Action event
      */
-    @Override public void actionPerformed(ActionEvent event) {
+    public void actionPerformed(ActionEvent event) {
         if (event.getSource() == JTagger.get().getMainWindow().getTitlePane().getTitleTable().aTestRenameButton) {
             try {
                 TitleRenamer renamer = JTagger.get().getMainWindow().getTitlePane().getTitleOptionsPanel().getRenamer();
@@ -120,7 +120,7 @@
     /**
      * @return column count
      */
-    @Override public int getColumnCount() {
+    public int getColumnCount() {
         return 3;
     }
 
@@ -150,7 +150,7 @@
     /**
      * @return number of rows
      */
-    @Override public int getRowCount() {
+    public int getRowCount() {
         return JTagger.get().getMainWindow().getTrackPane().getTrackTable().getTrackModel().getRowCount();
     }
 
@@ -160,7 +160,7 @@
      * @param col
      * @return get object to paint
      */
-    @Override public Object getValueAt(int row, int col) {
+    public Object getValueAt(int row, int col) {
         AudioTrack track = JTagger.get().getMainWindow().getTrackPane().getTrackTable().getTrackModel().getTrack(row);
         AudioTrack track_ = JTagger.get().getMainWindow().getTrackPane().getTrackTable().getTrackModel().getTestTrack(row);
 
diff -Nur jtagger/com/googlepages/dronten/jtagger/model/TrackModel.java jtagger_patched/com/googlepages/dronten/jtagger/model/TrackModel.java
--- jtagger/com/googlepages/dronten/jtagger/model/TrackModel.java	2008-04-28 20:58:50.000000000 +0300
+++ jtagger_patched/com/googlepages/dronten/jtagger/model/TrackModel.java	2008-07-12 20:30:40.000000000 +0300
@@ -100,7 +100,7 @@
      *
      * @return Columns
      */
-    @Override public int getColumnCount() {
+    public int getColumnCount() {
         return 3;
     }
 
@@ -149,7 +149,7 @@
      *
      * @return Number of rows
      */
-    @Override public int getRowCount() {
+    public int getRowCount() {
         return aAudioFiles.size();
     }
 
@@ -204,7 +204,7 @@
      * @param col
      * @return Object to paint
      */
-    @Override public Object getValueAt(int row, int col) {
+    public Object getValueAt(int row, int col) {
         AudioTrack track = getTrack(row);
         AudioTrack track_ = JTagger.get().getMainWindow().getTrackPane().getTrackTable().getTrackModel().getTestTrack(row);
 
