--- sys/conf/kern.pre.mk.old	2010-03-10 14:20:43 +0100
+++ sys/conf/kern.pre.mk	2010-03-10 14:20:56 +0100
@@ -93,7 +93,7 @@
 CFLAGS+= -fno-common -finline-limit=${INLINE_LIMIT}
 CFLAGS+= --param inline-unit-growth=100
 CFLAGS+= --param large-function-growth=1000
-WERROR?= -Werror
+WERROR?= -Wno-error
 .endif
 
 # XXX LOCORE means "don't declare C stuff" not "for locore.s".
--- sys/conf/kmod.mk.old	2010-03-10 14:21:06 +0100
+++ sys/conf/kmod.mk	2010-03-10 14:21:16 +0100
@@ -79,7 +79,7 @@
 . if !empty(CFLAGS:M-O[23s]) && empty(CFLAGS:M-fno-strict-aliasing)
 CFLAGS+=	-fno-strict-aliasing
 . endif
-WERROR?=	-Werror
+WERROR?=	-Wno-error
 .endif
 CFLAGS+=	${WERROR}
 CFLAGS+=	-D_KERNEL -D__FreeBSD_cc_version=700004
