--- sys/conf/ldscript.sparc64.old	2009-05-17 11:04:47 +0000
+++ sys/conf/ldscript.sparc64	2009-05-17 11:05:41 +0000
@@ -1,6 +1,6 @@
 /* $FreeBSD: src/sys/conf/ldscript.sparc64,v 1.4.30.1 2009/04/15 03:14:26 kensmith Exp $ */
-OUTPUT_FORMAT("elf64-sparc", "elf64-sparc",
-	      "elf64-sparc")
+OUTPUT_FORMAT("elf64-sparc-freebsd", "elf64-sparc-freebsd",
+	      "elf64-sparc-freebsd")
 OUTPUT_ARCH(sparc:v9)
 ENTRY(_start)
 SEARCH_DIR(/usr/lib);
