diff -uNr sys.old/conf/kern.mk sys/conf/kern.mk
--- sys.old/conf/kern.mk	2011-11-02 13:02:13.000000000 +0000
+++ sys/conf/kern.mk	2011-11-02 13:05:53.000000000 +0000
@@ -5,7 +5,7 @@
 #
 CWARNFLAGS?=	-Wall -Wredundant-decls -Wnested-externs -Wstrict-prototypes \
 		-Wmissing-prototypes -Wpointer-arith -Winline -Wcast-qual \
-		-Wundef -Wno-pointer-sign -fformat-extensions \
+		-Wundef -Wno-pointer-sign \
 		-Wmissing-include-dirs -fdiagnostics-show-option
 #
 # The following flags are next up for working on:
@@ -32,7 +32,7 @@
 #
 .if ${MACHINE_CPUARCH} == "i386"
 .if ${CC:T:Mclang} != "clang"
-CFLAGS+=	-mno-align-long-strings -mpreferred-stack-boundary=2 -mno-sse
+CFLAGS+=	-mpreferred-stack-boundary=2 -mno-sse
 .else
 CFLAGS+=	-mno-aes -mno-avx
 .endif
diff -uNr sys.old/dev/aic7xxx/aicasm/Makefile sys/dev/aic7xxx/aicasm/Makefile
--- sys.old/dev/aic7xxx/aicasm/Makefile	2011-11-02 13:02:11.000000000 +0000
+++ sys/dev/aic7xxx/aicasm/Makefile	2011-11-02 13:06:24.000000000 +0000
@@ -14,7 +14,7 @@
 SRCS=	${GENHDRS} ${CSRCS} ${YSRCS} ${LSRCS}
 CLEANFILES+= ${GENHDRS} ${YSRCS:R:C/(.*)/\1.output/g}
 DPADD=	${LIBL}
-LDADD=	-ll
+LDADD=	-lfl
 WARNS?=	5
 
 # Correct path for kernel builds
