--- snortalog.pl.orig	2005-04-13 04:42:06.000000000 -0400
+++ snortalog.pl	2005-04-14 00:36:20.000000000 -0400
@@ -397,6 +397,12 @@
 	$val_opt_p = $opt{p} if ( $opt_p == 1 ); 
 	$val_opt_g = $opt{g} if ( $opt_g == 1 ); 
 	$val_opt_l = $opt{l} if ( $opt_l == 1 ); 
+
+	if ( $opt_h == 1 && $opt_p == 1 ) {
+	    print "Error: You can only create a pdf or an html file, not both simultaneously.\n";
+	    exit();
+	}
+
 	#
 	# Test if kind of logs are selected else all arent chosen by default
 	#
