--- ./snortalog.pl.orig	2006-11-27 23:55:00.000000000 +0100
+++ ./snortalog.pl	2006-11-27 23:58:24.000000000 +0100
@@ -92,18 +92,18 @@
 # Graphical Tool Kit Librairies
 #
 $TK = 0 ;
-$TK = eval "use Tk;1;" ;
-$TK = eval "use Tk::NoteBook;2;" if ($TK == 1);
-$TK = eval "use Tk::Dialog;3;" if ($TK == 2);
+#$TK = eval "use Tk;1;" ;
+#$TK = eval "use Tk::NoteBook;2;" if ($TK == 1);
+#$TK = eval "use Tk::Dialog;3;" if ($TK == 2);
 
 #
 # GD Librairies for charts
 #
 $GD = 0 ;
-$GD = eval "use GD::Graph::pie;1;";
-$GD = eval "use GD::Graph::bars;2;" if ($GD == 1);
-$GD = eval "use GD::Graph::lines;3;" if ($GD == 2);
-$GD = eval "use GD::Graph::area;4;" if ($GD == 3);
+#$GD = eval "use GD::Graph::pie;1;";
+#$GD = eval "use GD::Graph::bars;2;" if ($GD == 1);
+#$GD = eval "use GD::Graph::lines;3;" if ($GD == 2);
+#$GD = eval "use GD::Graph::area;4;" if ($GD == 3);
 
 #
 # HTML and PDF manipulation libraries
