diff -ru wordwarvi-0.26.orig/Makefile wordwarvi-0.26/Makefile
--- wordwarvi-0.26.orig/Makefile	2009-08-09 10:33:02.000000000 -0400
+++ wordwarvi-0.26/Makefile	2010-02-08 17:45:30.977667366 -0500
@@ -1,5 +1,5 @@
 PREFIX=/usr
-DATADIR=${PREFIX}/share/wordwarvi
+DATADIR?=${PREFIX}/share/wordwarvi
 MANDIR?=${PREFIX}/share/man
 MANPAGEDIR=${MANDIR}/man6
 
@@ -21,18 +21,6 @@
 OGGOBJ=
 endif
 
-# DEBUG=-g
-# DEBUG=
-# PROFILE_FLAG=-pg
-#PROFILE_FLAG=
-#OPTIMIZE_FLAG=
-# OPTIMIZE_FLAG=-O3
-#OPTIMIZE_FLAG=-O3 -pedantic -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security
-OPTIMIZE_FLAG=-O3 -pedantic
-
-
-LDFLAGS=${PROFILE_FLAG}
-
 DEFINES=${SNDFLAGS} -DDATADIR=\"${DATADIR}/\"
 
 all:	wordwarvi wordwarvi.6.gz
@@ -50,32 +38,32 @@
 endif
 
 joystick.o:	joystick.c joystick.h Makefile
-	gcc ${DEBUG} ${PROFILE_FLAG} ${OPTIMIZE_FLAG} -pthread -Wall -c joystick.c
+	$(CC) ${DEBUG} ${PROFILE_FLAG} ${CFLAGS} -pthread -Wall -c joystick.c
 
 ogg_to_pcm.o:	ogg_to_pcm.c ogg_to_pcm.h Makefile
-	gcc ${DEBUG} ${PROFILE_FLAG} ${OPTIMIZE_FLAG} `pkg-config --cflags vorbisfile` \
+	$(CC) ${DEBUG} ${PROFILE_FLAG} ${CFLAGS} `pkg-config --cflags vorbisfile` \
 		-pthread -Wall -c ogg_to_pcm.c
 
 wwviaudio.o:	wwviaudio.c wwviaudio.h ogg_to_pcm.h my_point.h Makefile
-	gcc -Wall ${DEBUG} ${PROFILE_FLAG} ${OPTIMIZE_FLAG} \
+	$(CC) -Wall ${DEBUG} ${PROFILE_FLAG} ${CFLAGS} \
 		${DEFINES} \
 		-pthread `pkg-config --cflags vorbisfile` \
 		-c wwviaudio.c
 
 rumble.o:	rumble.c rumble.h Makefile
-	gcc ${DEBUG} ${PROFILE_FLAG} ${OPTIMIZE_FLAG} `pkg-config --cflags vorbisfile` \
+	$(CC) ${DEBUG} ${PROFILE_FLAG} ${CFLAGS} `pkg-config --cflags vorbisfile` \
 		-pthread -Wall -c rumble.c
 
 wwvi_font.o:	wwvi_font.c wwvi_font.h my_point.h Makefile
-	gcc ${DEBUG} ${PROFILE_FLAG} ${OPTIMIZE_FLAG} -pthread -Wall -c wwvi_font.c
+	$(CC) ${DEBUG} ${PROFILE_FLAG} ${CFLAGS} -pthread -Wall -c wwvi_font.c
 
 stamp:	stamp.c
-	gcc -o stamp stamp.c	
+	$(CC) $(LDFLAGS) -o stamp stamp.c	
 
 wordwarvi:	wordwarvi.c joystick.o rumble.o ${OGGOBJ} wwviaudio.o wwvi_font.o \
 		Makefile version.h stamp levels.h rumble.h
 	./stamp > stamp.h
-	gcc ${DEBUG} ${PROFILE_FLAG} ${OPTIMIZE_FLAG} ${SCREENSAVERFLAG} -pthread -Wall  ${DEFINES} \
+	$(CC) $(LDFLAGS) ${DEBUG} ${PROFILE_FLAG} ${CFLAGS} ${SCREENSAVERFLAG} -pthread -Wall  ${DEFINES} \
 		joystick.o \
 		rumble.o \
 		wwvi_font.o \
diff -ru wordwarvi-0.26.orig/wwviaudio.c wordwarvi-0.26/wwviaudio.c
--- wordwarvi-0.26.orig/wwviaudio.c	2009-08-09 10:33:02.000000000 -0400
+++ wordwarvi-0.26/wwviaudio.c	2010-02-08 17:45:07.127666854 -0500
@@ -124,7 +124,7 @@
 	strncpy(filebuf, filename, PATH_MAX);
 	rc = stat(filebuf, &statbuf);
 	if (rc != 0) {
-		snprintf(filebuf, PATH_MAX, "%s", filename);
+		snprintf(filebuf, PATH_MAX, "GENTOO_DATADIR%s", filename);
 		rc = stat(filebuf, &statbuf);
 		if (rc != 0) {
 			fprintf(stderr, "stat('%s') failed.\n", filebuf);
