diff -ru tvonscreen-1.0.141.orig/magazine.c tvonscreen-1.0.141/magazine.c
--- tvonscreen-1.0.141.orig/magazine.c	2009-02-25 16:39:18.024158328 +0100
+++ tvonscreen-1.0.141/magazine.c	2009-02-25 16:39:51.401402805 +0100
@@ -183,7 +183,7 @@
 		txt=channel->ShortName(true);
 		// logo: 64x48px
 
-		const char *ConfigDirectory=cPlugin::ConfigDirectory("logos");
+		const char *ConfigDirectory="/usr/share/vdr/channel-logos";
 		if (tvonscreenCfg.logos)
 			ConfigDirectory=tvonscreenCfg.logos;
 		char *fname=new char[strlen(ConfigDirectory) + 1 + strlen(txt) + strlen(".xpm") + 1];
