diff -ur Makefile.in Makefile.in.new
--- Makefile.in	2009-08-04 09:42:37.000000000 +0200
+++ Makefile.in	2009-10-08 20:53:16.000000000 +0200
@@ -58,8 +58,8 @@
 LDFLAGS = @MYLDFLAGS@
 CMDLDFLAGS = @MYCMDLDFLAGS@
 LIBS = @LIBS@
-LDENV = LD_RUN_PATH=/lib:/usr/lib:$(LIBDIR):$(HOME)/lib:/usr/local/lib:@MYRUNPATH@:.
-RUNENV = @MYLDLIBPATHENV@=.:/lib:/usr/lib:$(LIBDIR):$(HOME)/lib:/usr/local/lib:@MYRUNPATH@
+LDENV = 
+RUNENV = @MYLDLIBPATHENV@=.:$(LIBDIR)
 POSTCMD = @MYPOSTCMD@
 
 
@@ -120,8 +120,6 @@
 	cp -Rf $(LIBRARYFILES) $(DESTDIR)$(LIBDIR)
 	mkdir -p $(DESTDIR)$(BINDIR)
 	cp -Rf $(COMMANDFILES) $(DESTDIR)$(BINDIR)
-	mkdir -p $(DESTDIR)$(SBINDIR)
-	cp -Rf $(INITFILES) $(DESTDIR)$(SBINDIR)
 	mkdir -p $(DESTDIR)$(DATADIR)
 	cp -Rf $(SCREXTFILES) $(DOCUMENTFILES) $(DESTDIR)$(DATADIR)
 	mkdir -p $(DESTDIR)$(MAN1DIR)
diff -ur configure.in configure.in.new
--- configure.in	2009-09-17 17:26:04.000000000 +0200
+++ configure.in	2009-10-08 20:53:57.000000000 +0200
@@ -28,33 +28,15 @@
 MYPCFILES="tokyotyrant.pc"
 
 # Building flags
-MYCFLAGS="-std=c99 -Wall -fPIC -fsigned-char -O2"
+MYCFLAGS="-std=c99 -Wall -fPIC -fsigned-char"
 MYCPPFLAGS="-I. -I\$(INCLUDEDIR) -I$HOME/include -I/usr/local/include"
 MYCPPFLAGS="$MYCPPFLAGS -DNDEBUG -D_GNU_SOURCE=1 -D_REENTRANT -D__EXTENSIONS__"
-MYLDFLAGS="-L. -L\$(LIBDIR) -L$HOME/lib -L/usr/local/lib"
+MYLDFLAGS="-L. -L\$(LIBDIR)"
 MYCMDLDFLAGS=""
 MYRUNPATH="\$(LIBDIR)"
 MYLDLIBPATHENV="LD_LIBRARY_PATH"
 MYPOSTCMD="true"
 
-# Building paths
-PATH="$PATH:$HOME/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/bin:/sbin"
-PATH="$PATH:/opt/SUNWspro/bin:/usr/ccs/bin:/usr/xpg4/bin:/usr/xpg6/bin:/usr/ucb"
-CPATH="$HOME/include:/usr/local/include:$CPATH"
-LIBRARY_PATH="$HOME/lib:/usr/local/lib:$LIBRARY_PATH"
-LD_LIBRARY_PATH="$HOME/lib:/usr/local/lib:$LD_LIBRARY_PATH"
-PKG_CONFIG_PATH="$HOME/lib/pkgconfig:/usr/local/lib/pkgconfig:$PKG_CONFIG_PATH"
-export PATH CPATH LIBRARY_PATH LD_LIBRARY_PATH PKG_CONFIG_PATH
-if type pkg-config > /dev/null 2>&1
-then
-  PATH="$PATH:`pkg-config --variable=bindir tokyocabinet`"
-  CPATH="$CPATH:`pkg-config --variable=includedir tokyocabinet`"
-  LIBRARY_PATH="$LIBRARY_PATH:`pkg-config --variable=libdir tokyocabinet`"
-  LD_LIBRARY_PATH="$LD_LIBRARY_PATH:`pkg-config --variable=libdir tokyocabinet`"
-fi
-
-
-
 #================================================================
 # Options
 #================================================================
diff -ur tokyotyrant.pc.in tokyotyrant.pc.in.new
--- tokyotyrant.pc.in	2008-02-18 08:29:28.000000000 +0100
+++ tokyotyrant.pc.in.new	2009-10-08 20:54:12.000000000 +0200
@@ -10,5 +10,5 @@
 Name: Tokyo Tyrant
 Description: network interface of Tokyo Cabinet
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -ltokyotyrant @LIBS@
+Libs: -L${libdir} -ltokyotyrant
 Cflags: -I${includedir}
