Index: lib/CMakeLists.txt
===================================================================
--- lib/CMakeLists.txt	(revision 157576)
+++ lib/CMakeLists.txt	(working copy)
@@ -36,7 +36,7 @@
 append_if(libraries LIBCXX_HAS_C_LIB c)
 append_if(libraries LIBCXX_HAS_M_LIB m)
 append_if(libraries LIBCXX_HAS_RT_LIB rt)
-append_if(libraries LIBCXX_HAS_GCC_S_LIB gcc_s)
+append_if(libraries LIBCXX_HAS_LIBCXXRT cxxrt)
 
 target_link_libraries(cxx ${libraries})
 
Index: cmake/config-ix.cmake
===================================================================
--- cmake/config-ix.cmake	(revision 157576)
+++ cmake/config-ix.cmake	(working copy)
@@ -26,7 +26,7 @@
 check_library_exists(c printf "" LIBCXX_HAS_C_LIB)
 check_library_exists(m ccos "" LIBCXX_HAS_M_LIB)
 check_library_exists(rt clock_gettime "" LIBCXX_HAS_RT_LIB)
-check_library_exists(gcc_s __gcc_personality_v0 "" LIBCXX_HAS_GCC_S_LIB)
+check_library_exists(cxxrt  __cxa_get_globals "" LIBCXX_HAS_LIBCXXRT)
 
 # Check C++0x features
 if (LIBCXX_ENABLE_CXX0X)
