Index: libcxx-0.0_p20120523/lib/CMakeLists.txt
===================================================================
--- libcxx-0.0_p20120523.orig/lib/CMakeLists.txt
+++ libcxx-0.0_p20120523/lib/CMakeLists.txt
@@ -54,8 +54,8 @@ set_target_properties(cxx
   )
 
 install(TARGETS cxx
-  LIBRARY DESTINATION lib
-  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib${LIB_SUFFIX}
+  ARCHIVE DESTINATION lib${LIB_SUFFIX}
   )
 
 install(DIRECTORY ../include/
