diff --git a/liblangtag-gobject/Makefile.am b/liblangtag-gobject/Makefile.am
index 99c774f..a334925 100644
--- a/liblangtag-gobject/Makefile.am
+++ b/liblangtag-gobject/Makefile.am
@@ -173,12 +173,14 @@ LangTag_0_4_gir_INCLUDES =		\
 	GLib-2.0			\
 	GObject-2.0			\
 	$(NULL)
-LangTag_0_4_gir_CFLAGS =		\
-	$(INCLUDES)			\
+LangTag_0_4_gir_SCANNERFLAGS =		\
 	--symbol-prefix=lt		\
 	--identifier-prefix=Lt		\
 	-v				\
 	$(NULL)
+LangTag_0_4_gir_CFLAGS =		\
+	$(INCLUDES)			\
+	$(NULL)
 LangTag_0_4_gir_LIBS =			\
 	liblangtag-gobject.la		\
 	$(NULL)
