--- Makefile.in.orig	2008-08-29 11:26:41.000000000 +0100
+++ Makefile.in	2008-08-29 11:32:32.000000000 +0100
@@ -173,7 +173,7 @@
 
 quickplot_CXXFLAGS = -Wall $(GTKMM_CXXFLAGS) $(SNDFILE_CFLAGS) -DHTMLDIR=\"$(htmldir)\"
 
-quickplot_LDFLAGS = -Wall $(GTKMM_LIBS) $(SNDFILE_LIBS)
+quickplot_LDFLAGS = -Wall
 # -lefence
 
 # require automake 1.7
@@ -252,7 +252,7 @@
 	quickplot-main.$(OBJEXT) quickplot-parseGeometry.$(OBJEXT) \
 	quickplot-usage.$(OBJEXT)
 quickplot_OBJECTS = $(am_quickplot_OBJECTS)
-quickplot_LDADD = $(LDADD)
+quickplot_LDADD =  $(GTKMM_LIBS) $(SNDFILE_LIBS)
 quickplot_DEPENDENCIES =
 
 DEFAULT_INCLUDES =  -I. -I$(srcdir) -I.
