diff -Naur quickplot-0.8.13/App.h quickplot-0.8.13.new/App.h
--- quickplot-0.8.13/App.h	2006-08-04 11:33:36.000000000 -0400
+++ quickplot-0.8.13.new/App.h	2008-05-06 09:59:36.000000000 -0400
@@ -41,7 +41,7 @@
   
  public:
 
-  App(int *argv, char ***argv);
+  App(int *argc, char ***argv);
   virtual ~App(void);
 
 
diff -Naur quickplot-0.8.13/Graph.cpp quickplot-0.8.13.new/Graph.cpp
--- quickplot-0.8.13/Graph.cpp	2006-08-05 11:45:34.000000000 -0400
+++ quickplot-0.8.13.new/Graph.cpp	2008-05-06 09:59:36.000000000 -0400
@@ -42,6 +42,8 @@
 #define ZRO        ((value_t) 0.0)
 #define ABSVAL(x)  (((x) > 0)? (x) : (-(x)))
 
+#include <cstring>
+
 
 
 Graph::Graph(MainWindow *mainWindow_in):
diff -Naur quickplot-0.8.13/MainWindow.cpp quickplot-0.8.13.new/MainWindow.cpp
--- quickplot-0.8.13/MainWindow.cpp	2006-08-07 14:12:11.000000000 -0400
+++ quickplot-0.8.13.new/MainWindow.cpp	2008-05-06 09:59:36.000000000 -0400
@@ -40,6 +40,8 @@
 
 #include "quickplot_icon.xpm"
 
+#include <cstring>
+
 
 // from file: help_html.cpp which was generated from help.html.in
 extern const char *help_html;
diff -Naur quickplot-0.8.13/Plot.cpp quickplot-0.8.13.new/Plot.cpp
--- quickplot-0.8.13/Plot.cpp	2006-08-03 16:56:56.000000000 -0400
+++ quickplot-0.8.13.new/Plot.cpp	2008-05-06 09:59:36.000000000 -0400
@@ -31,6 +31,8 @@
 #include "Graph.h"
 #include "Globel.h"
 
+#include <cstring>
+
 
 // SPACE is the fraction of blank space on graphing area for the first
 // auto zoom level.  The blank space is split evenly between the top
