diff -Nuar multipath-tools-0.4.8.orig/devmap_name/Makefile multipath-tools-0.4.8/devmap_name/Makefile
--- multipath-tools-0.4.8.orig/devmap_name/Makefile	2007-08-02 14:05:37.000000000 -0700
+++ multipath-tools-0.4.8/devmap_name/Makefile	2008-06-20 16:20:19.342920167 -0700
@@ -10,7 +10,7 @@
 ifeq ($(strip $(BUILD)),klibc)
 	OBJS += $(libdm)
 else
-	LDFLAGS = -ldevmapper
+	LDFLAGS += -ldevmapper
 endif
 
 EXEC = devmap_name
@@ -32,11 +32,11 @@
 	install -d $(DESTDIR)$(bindir)
 	install -m 755 $(EXEC) $(DESTDIR)$(bindir)/
 	install -d $(DESTDIR)$(mandir)
-	install -m 644 $(EXEC).8.gz $(DESTDIR)$(mandir)
+	install -m 644 $(EXEC).8 $(DESTDIR)$(mandir)
 
 uninstall:
-	rm $(DESTDIR)$(bindir)/$(EXEC)
-	rm $(DESTDIR)$(mandir)/$(EXEC).8.gz
+	rm -f $(DESTDIR)$(bindir)/$(EXEC)
+	rm -f $(DESTDIR)$(mandir)/$(EXEC).8
 
 clean:
 	rm -f core *.o $(EXEC) *.gz
diff -Nuar multipath-tools-0.4.8.orig/kpartx/Makefile multipath-tools-0.4.8/kpartx/Makefile
--- multipath-tools-0.4.8.orig/kpartx/Makefile	2007-08-02 14:05:37.000000000 -0700
+++ multipath-tools-0.4.8/kpartx/Makefile	2008-06-20 16:30:07.571748249 -0700
@@ -13,7 +13,7 @@
 	       lopart.o xstrncpy.o devmapper.o dasd.o mac.o sun.o \
 	       $(MULTIPATHLIB)-$(BUILD).a $(libdm)
 else
-	LDFLAGS = -ldevmapper
+	LDFLAGS += -ldevmapper
 	OBJS = bsd.o dos.o kpartx.o solaris.o unixware.o dasd.o sun.o \
 	       gpt.o mac.o crc32.o lopart.o xstrncpy.o devmapper.o
 endif
@@ -44,11 +44,12 @@
 	install -d $(DESTDIR)/etc/udev/rules.d
-	install -m 644 kpartx.rules $(DESTDIR)/etc/udev/rules.d/
+	install -m 644 kpartx.rules $(DESTDIR)/etc/udev/rules.d/66-kpartx.rules
 	install -d $(DESTDIR)$(mandir)
-	install -m 644 $(EXEC).8.gz $(DESTDIR)$(mandir)
+	install -m 644 $(EXEC).8 $(DESTDIR)$(mandir)
 
 uninstall:
 	rm -f $(DESTDIR)$(bindir)/$(EXEC)
-	rm -f $(DESTDIR)$(mandir)/$(EXEC).8.gz
+	rm -f $(DESTDIR)$(mandir)/$(EXEC).8
+	rm -f $(DESTDIR)/etc/udev/rules.d/66-kpartx.rules
 
 clean:
 	rm -f core *.o $(EXEC) *.gz
diff -Nuar multipath-tools-0.4.8.orig/Makefile.inc multipath-tools-0.4.8/Makefile.inc
--- multipath-tools-0.4.8.orig/Makefile.inc	2007-08-02 14:05:37.000000000 -0700
+++ multipath-tools-0.4.8/Makefile.inc	2008-06-20 16:18:41.008583279 -0700
@@ -34,10 +34,10 @@
 CHECKERSLIB = $(checkersdir)/libcheckers
 MULTIPATHLIB = $(multipathdir)/libmultipath
 
-INSTALL_PROGRAM = install -s
+INSTALL_PROGRAM = install
 
-OPTFLAGS     = -pipe -g -Wall -Wunused -Wstrict-prototypes
-CFLAGS	     = $(OPTFLAGS)
+OPTFLAGS     = -Wall -Wunused -Wstrict-prototypes
+CFLAGS	     += $(OPTFLAGS)
 
 %.o:	%.c
 	$(CC) $(CFLAGS) -c -o $@ $<
diff -Nuar multipath-tools-0.4.8.orig/multipath/Makefile multipath-tools-0.4.8/multipath/Makefile
--- multipath-tools-0.4.8.orig/multipath/Makefile	2007-08-02 14:05:37.000000000 -0700
+++ multipath-tools-0.4.8/multipath/Makefile	2008-06-20 16:21:17.572126979 -0700
@@ -42,17 +42,17 @@
 	install -d $(DESTDIR)$(bindir)
 	$(INSTALL_PROGRAM) -m 755 $(EXEC) $(DESTDIR)$(bindir)/
 	install -d $(DESTDIR)/etc/udev/rules.d
-	install -m 644 multipath.rules $(DESTDIR)/etc/udev/rules.d/
+	install -m 644 multipath.rules $(DESTDIR)/etc/udev/rules.d/65-multipath.rules
 	install -d $(DESTDIR)$(mandir)
-	install -m 644 $(EXEC).8.gz $(DESTDIR)$(mandir)
+	install -m 644 $(EXEC).8 $(DESTDIR)$(mandir)
 	install -d $(DESTDIR)$(man5dir)
-	install -m 644 $(EXEC).conf.5.gz $(DESTDIR)$(man5dir)
+	install -m 644 $(EXEC).conf.5 $(DESTDIR)$(man5dir)
 
 uninstall:
-	rm $(DESTDIR)/etc/udev/rules.d/multipath.rules
-	rm $(DESTDIR)$(bindir)/$(EXEC)
-	rm $(DESTDIR)$(mandir)/$(EXEC).8.gz
-	rm $(DESTDIR)$(man5dir)/$(EXEC).conf.5.gz
+	rm -f $(DESTDIR)/etc/udev/rules.d/65-multipath.rules
+	rm -f $(DESTDIR)$(bindir)/$(EXEC)
+	rm -f $(DESTDIR)$(mandir)/$(EXEC).8
+	rm -f $(DESTDIR)$(man5dir)/$(EXEC).conf.5
 
 clean:
 	rm -f core *.o $(EXEC) *.gz
diff -Nuar multipath-tools-0.4.8.orig/multipathd/Makefile multipath-tools-0.4.8/multipathd/Makefile
--- multipath-tools-0.4.8.orig/multipathd/Makefile	2007-08-02 14:05:37.000000000 -0700
+++ multipath-tools-0.4.8/multipathd/Makefile	2008-06-20 16:21:51.421988986 -0700
@@ -7,7 +7,7 @@
 # basic flags setting
 #
 CFLAGS += -DDAEMON -I$(multipathdir) -I$(checkersdir)
-LDFLAGS = -lpthread -ldevmapper -lreadline -lncurses -laio
+LDFLAGS += -lpthread -ldevmapper -lreadline -lncurses -laio
 
 #
 # debuging stuff
@@ -48,12 +48,12 @@
 	$(INSTALL_PROGRAM) -m 755 $(EXEC) $(DESTDIR)$(bindir)
 	install -d $(DESTDIR)$(rcdir)
 	install -d $(DESTDIR)$(mandir)
-	install -m 644 $(EXEC).8.gz $(DESTDIR)$(mandir)
+	install -m 644 $(EXEC).8 $(DESTDIR)$(mandir)
 
 uninstall:
 	rm -f $(DESTDIR)$(bindir)/$(EXEC)
 	rm -f $(DESTDIR)$(rcdir)/$(EXEC)
-	rm -f $(DESTDIR)$(mandir)/$(EXEC).8.gz
+	rm -f $(DESTDIR)$(mandir)/$(EXEC).8
 
 clean:
 	$(MAKE) -C $(multipathdir) prepare DAEMON=1
diff -Nuar multipath-tools-0.4.8.orig/path_priority/pp_alua/Makefile multipath-tools-0.4.8/path_priority/pp_alua/Makefile
--- multipath-tools-0.4.8.orig/path_priority/pp_alua/Makefile	2007-08-02 14:05:37.000000000 -0700
+++ multipath-tools-0.4.8/path_priority/pp_alua/Makefile	2008-06-20 16:24:53.596722793 -0700
@@ -35,13 +35,13 @@
 klibc:	$(OBJS)
 	$(CC) -static -o $(EXEC) $(OBJS)
 
-install: $(EXEC) $(EXEC).8.gz
-	$(INSTALL) -s -m 755 $(EXEC) $(DESTDIR)$(bindir)/$(EXEC)
-	$(INSTALL) -m 644 $(EXEC).8.gz $(DESTDIR)$(mandir)/$(EXEC).8.gz
+install: $(EXEC) $(EXEC).8
+	$(INSTALL) -m 755 $(EXEC) $(DESTDIR)$(bindir)/$(EXEC)
+	$(INSTALL) -m 644 $(EXEC).8 $(DESTDIR)$(mandir)/$(EXEC).8
 
 uninstall:
-	rm $(DESTDIR)$(bindir)/$(EXEC)
-	rm $(DESTDIR)$(mandir)/$(EXEC).8.gz
+	rm -f $(DESTDIR)$(bindir)/$(EXEC)
+	rm -f $(DESTDIR)$(mandir)/$(EXEC).8
 
 clean:	
 	rm -f *.o *.gz $(EXEC)
diff -Nuar multipath-tools-0.4.8.orig/path_priority/pp_balance_units/Makefile multipath-tools-0.4.8/path_priority/pp_balance_units/Makefile
--- multipath-tools-0.4.8.orig/path_priority/pp_balance_units/Makefile	2007-08-02 14:05:37.000000000 -0700
+++ multipath-tools-0.4.8/path_priority/pp_balance_units/Makefile	2008-06-20 16:29:13.181754556 -0700
@@ -13,7 +13,7 @@
 	OBJS = pp_balance_units.o $(MULTIPATHLIB)-$(BUILD).a
 else
 	CFLAGS += -I$(multipathdir) -DDEBUG=$(DEBUG)
-	LDFLAGS = -ldevmapper
+	LDFLAGS += -ldevmapper
 	OBJS = pp_balance_units.o $(MULTIPATHLIB)-$(BUILD).a
 endif
 
@@ -38,7 +38,7 @@
 	$(INSTALL_PROGRAM) -m 755 $(EXEC) $(DESTDIR)$(bindir)/
 
 uninstall:
-	rm $(DESTDIR)$(bindir)/$(EXEC)
+	rm -f $(DESTDIR)$(bindir)/$(EXEC)
 
 clean:
 	rm -f core *.o $(EXEC) *.gz
diff -Nuar multipath-tools-0.4.8.orig/path_priority/pp_emc/Makefile multipath-tools-0.4.8/path_priority/pp_emc/Makefile
--- multipath-tools-0.4.8.orig/path_priority/pp_emc/Makefile	2007-08-02 14:05:37.000000000 -0700
+++ multipath-tools-0.4.8/path_priority/pp_emc/Makefile	2008-06-20 16:28:57.664254512 -0700
@@ -17,7 +17,8 @@
 	$(INSTALL_PROGRAM) -m 755 $(EXEC) $(DESTDIR)$(bindir)/$(EXEC)
 
 uninstall:
-	rm $(DESTDIR)$(bindir)/$(EXEC)
+	rm -f $(DESTDIR)$(bindir)/$(EXEC)
+
 clean:	
 	rm -f *.o $(EXEC)
 
diff -Nuar multipath-tools-0.4.8.orig/path_priority/pp_hds_modular/Makefile multipath-tools-0.4.8/path_priority/pp_hds_modular/Makefile
--- multipath-tools-0.4.8.orig/path_priority/pp_hds_modular/Makefile	2007-08-02 14:05:37.000000000 -0700
+++ multipath-tools-0.4.8/path_priority/pp_hds_modular/Makefile	2008-06-20 16:28:52.897167373 -0700
@@ -17,6 +17,7 @@
 	$(INSTALL_PROGRAM) -m 755 $(EXEC) $(DESTDIR)$(bindir)/$(EXEC)
 
 uninstall:
-	rm $(DESTDIR)$(bindir)/$(EXEC)
+	rm -f $(DESTDIR)$(bindir)/$(EXEC)
+
 clean:	
 	rm -f *.o $(EXEC)
diff -Nuar multipath-tools-0.4.8.orig/path_priority/pp_hp_sw/Makefile multipath-tools-0.4.8/path_priority/pp_hp_sw/Makefile
--- multipath-tools-0.4.8.orig/path_priority/pp_hp_sw/Makefile	2007-08-02 14:05:37.000000000 -0700
+++ multipath-tools-0.4.8/path_priority/pp_hp_sw/Makefile	2008-06-20 16:28:48.881865361 -0700
@@ -17,7 +17,7 @@
 	install -m 755 $(EXEC) $(DESTDIR)$(bindir)/$(EXEC)
 
 uninstall:
-	rm $(DESTDIR)$(bindir)/$(EXEC)
+	rm -f $(DESTDIR)$(bindir)/$(EXEC)
 clean:	
 	rm -f *.o $(EXEC)
 
diff -Nuar multipath-tools-0.4.8.orig/path_priority/pp_netapp/Makefile multipath-tools-0.4.8/path_priority/pp_netapp/Makefile
--- multipath-tools-0.4.8.orig/path_priority/pp_netapp/Makefile	2007-08-02 14:05:37.000000000 -0700
+++ multipath-tools-0.4.8/path_priority/pp_netapp/Makefile	2008-06-20 16:28:45.411765821 -0700
@@ -17,6 +17,7 @@
 	$(INSTALL_PROGRAM) -m 755 $(EXEC) $(DESTDIR)$(bindir)/$(EXEC)
 
 uninstall:
-	rm $(DESTDIR)$(bindir)/$(EXEC)
+	rm -f $(DESTDIR)$(bindir)/$(EXEC)
+
 clean:	
 	rm -f *.o $(EXEC)
diff -Nuar multipath-tools-0.4.8.orig/path_priority/pp_random/Makefile multipath-tools-0.4.8/path_priority/pp_random/Makefile
--- multipath-tools-0.4.8.orig/path_priority/pp_random/Makefile	2007-08-02 14:05:37.000000000 -0700
+++ multipath-tools-0.4.8/path_priority/pp_random/Makefile	2008-06-20 16:28:37.139873269 -0700
@@ -17,6 +17,7 @@
 	$(INSTALL_PROGRAM) -m 755 $(EXEC) $(DESTDIR)$(bindir)/$(EXEC)
 
 uninstall:
-	rm $(DESTDIR)$(bindir)/$(EXEC)
+	rm -f $(DESTDIR)$(bindir)/$(EXEC)
+
 clean:	
 	rm -f *.o $(EXEC)
diff -Nuar multipath-tools-0.4.8.orig/path_priority/pp_rdac/Makefile multipath-tools-0.4.8/path_priority/pp_rdac/Makefile
--- multipath-tools-0.4.8.orig/path_priority/pp_rdac/Makefile	2007-08-02 14:05:37.000000000 -0700
+++ multipath-tools-0.4.8/path_priority/pp_rdac/Makefile	2008-06-20 16:28:32.708817275 -0700
@@ -17,6 +17,7 @@
 	$(INSTALL_PROGRAM) -m 755 $(EXEC) $(DESTDIR)$(bindir)/$(EXEC)
 
 uninstall:
-	rm $(DESTDIR)$(bindir)/$(EXEC)
+	rm -f $(DESTDIR)$(bindir)/$(EXEC)
+
 clean:	
 	rm -f *.o $(EXEC)
