diff -Naur refe-0.8.0.orig/lib/refe/mygetopt.rb refe-0.8.0/lib/refe/mygetopt.rb
--- refe-0.8.0.orig/lib/refe/mygetopt.rb	2003-04-02 07:38:41.000000000 +0900
+++ refe-0.8.0/lib/refe/mygetopt.rb	2010-05-30 00:54:04.000000000 +0900
@@ -11,7 +11,7 @@
     alias orignew new
 
     def new( usage, descripter )
-      parser = orignew( * descripter.map {|line|
+      parser = orignew( * descripter.split(/^/).map {|line|
           line.strip!
           next if line.empty?
           display, shortopt, longopt, takearg, doc = line.split(/\s+/, 5)
