diff -Naur elinks-0.11.1.orig/configure.in elinks-0.11.1/configure.in
--- elinks-0.11.1.orig/configure.in	2006-01-29 14:10:33.000000000 +0100
+++ elinks-0.11.1/configure.in	2006-03-01 17:32:58.000000000 +0100
@@ -739,7 +739,7 @@
 	for luadir in "$withval" "" /usr /usr/local; do
 		for suffix in "" 50 51; do
 			if test "$cf_result" = no; then
-				LUA_LIBS="-llua$suffix -llualib$suffix -lm"
+				LUA_LIBS="`pkg-config --libs lua`"
 
 				if test ! -z "$luadir"; then
 					LUA_LIBS="-L$luadir/lib $LUA_LIBS"
@@ -754,10 +754,10 @@
 				AC_TRY_LINK([	#include <lua.h>
 						#include <lualib.h>],
 					    [	lua_State *L = lua_open();
-						lua_baselibopen(L);
-						lua_mathlibopen(L);
-						lua_strlibopen(L);
-						lua_tablibopen(L);
+						luaopen_base(L);
+						luaopen_math(L);
+						luaopen_string(L);
+						luaopen_table(L);
 						lua_pushboolean(L, 1);
 						lua_close(L);],
 					    cf_result=yes, cf_result=no)
diff -Naur elinks-0.11.1.orig/src/scripting/lua/core.c elinks-0.11.1/src/scripting/lua/core.c
--- elinks-0.11.1.orig/src/scripting/lua/core.c	2006-01-29 14:10:39.000000000 +0100
+++ elinks-0.11.1/src/scripting/lua/core.c	2006-03-01 17:41:43.000000000 +0100
@@ -633,7 +633,7 @@
 	if (file_can_read(file)) {
 		int oldtop = lua_gettop(S);
 
-		if (lua_dofile(S, file) != 0)
+		if (luaL_dofile(S, file) != 0)
 			sleep(3); /* Let some time to see error messages. */
 		lua_settop(S, oldtop);
 	}
@@ -755,7 +755,7 @@
 		int oldtop = lua_gettop(L);
 
 		if (prepare_lua(ses) == 0) {
-			lua_dostring(L, expr);
+			luaL_dostring(L, expr);
 			lua_settop(L, oldtop);
 			finish_lua();
 		}
diff -Naur elinks-0.11.1.orig/src/scripting/lua/hooks.c elinks-0.11.1/src/scripting/lua/hooks.c
--- elinks-0.11.1.orig/src/scripting/lua/hooks.c	2006-01-29 14:10:39.000000000 +0100
+++ elinks-0.11.1/src/scripting/lua/hooks.c	2006-03-01 17:41:07.000000000 +0100
@@ -200,7 +200,7 @@
 script_hook_quit(va_list ap, void *data)
 {
 	if (!prepare_lua(NULL)) {
-		lua_dostring(lua_state, "if quit_hook then quit_hook() end");
+		luaL_dostring(lua_state, "if quit_hook then quit_hook() end");
 		finish_lua();
 	}
 
