--- libfaketime-0.9.1/Makefile
+++ libfaketime-0.9.1/Makefile
@@ -1,6 +1,5 @@
 all:
 	$(MAKE) -C src all
-	$(MAKE) -C test all
 
 test:
 	$(MAKE) -C test all
--- libfaketime-0.9.1/src/Makefile
+++ libfaketime-0.9.1/src/Makefile
@@ -47,7 +47,7 @@
 #     the dynamic disabling of the FAKE_STAT calls.
 #
 
-CC = gcc
+CC ?= gcc
 INSTALL = install
 
 PREFIX = /usr/local
--- libfaketime-0.9.1/test/Makefile
+++ libfaketime-0.9.1/test/Makefile
@@ -1,7 +1,7 @@
-CC = gcc
+CC ?= gcc
 
-CFLAGS = -std=gnu99 -Wall -DFAKE_STAT
-LDFLAGS = -lrt
+CFLAGS += -std=gnu99 -Wall -DFAKE_STAT
+LIBS = -lrt
 
 SRC = timetest.c
 OBJ = ${SRC:.c=.o}
@@ -12,7 +12,7 @@
 	${CC} -c ${CFLAGS} $<
 
 timetest: ${OBJ}
-	${CC} -o $@ ${OBJ} ${LDFLAGS}
+	${CC} -o $@ ${OBJ} ${LDFLAGS} ${LIBS}
 
 test: timetest functest
 	@echo
