Index: facter-1.5.7/lib/facter/virtual.rb
===================================================================
--- facter-1.5.7.orig/lib/facter/virtual.rb
+++ facter-1.5.7/lib/facter/virtual.rb
@@ -72,7 +72,7 @@ Facter.add("is_virtual") do
 
     setcode do
         case Facter.value(:virtual)
-        when "xenu", "openvzve", "vmware"
+        when "xenu", "openvzve", "vmware", "vserver"
             true
         else 
             false
