Index: dep.in
===================================================================
--- udept-0.5.99.0.2.95/src/dep.in	(revision 909)
+++ udept-0.5.99.0.2.95/src/dep.in	(working copy)
@@ -1145,6 +1145,8 @@
 function usecomponent() {
 	# return a USE component, one flag per line, highest priority first
 	# TODO FIXME not correct yet - though OK for default case
+	# Added placeholders for default USE flag implementation
+	# See http://bugs.gentoo.org/show_bug.cgi?id=61732
 	local component="$1" cpv="$2"
 	case "$component" in
 		env) sed 's/[[:space:]]\+/\n/g' <<<"$USE" | tac;;
@@ -1154,8 +1156,11 @@
 				dep_satisfies "$cpv" $package && echo $use
 			done | sed 's/[[:space:]]\+/\n/g';;
 		conf) sed 's/[[:space:]]\+/\n/g' <<<"$__conf_USE";;
+		pkgprofile);;
 		auto) autouse;;
 		defaults);;
+		pkginternal);;
+		env.d);;
 		*) format_error <<<"Unknown USE component: $component" >&2;;
 	esac
 }
