--- a/Makefile.rules
+++ b/Makefile.rules
@@ -26,11 +26,9 @@
 .SUFFIXES: .deps/%.d .c
 
 %.o: %.c
-	@echo "  CC $<"; \
 	$(CC) $(CFLAGS) $(CPPFLAGS) -c -o $@ $<
 
 .deps/%.d: %.c
-	@echo "  DEP $<"; \
 	$(CC) $(DEPFLAGS) $< > $@.tmp; \
 	sed 's,\($*\)\.o[ :]*,\1.o $@ : ,g' < $@.tmp > $@; \
 	rm -f $@.tmp
--- a/src/GNUmakefile
+++ b/src/GNUmakefile
@@ -65,7 +65,6 @@
 all: bmon
 
 $(OUT): ../Makefile.opts $(OBJ)
-	@echo "  LD $(OUT)"; \
 	$(CC) $(CFLAGS) $(CPPFLAGS) -o $(OUT) $(OBJ) $(LDFLAGS) $(BMON_LIB)
 
 clean:
