--- GNUmakefile.	2005-04-05 17:01:33.000000000 +0200
+++ GNUmakefile	2006-08-19 02:51:37.374492723 +0200
@@ -50,7 +50,7 @@
 install:
 	./install-sh -d -m 0755 $(DESTDIR)$(prefix)/bin
 	./install-sh -d -m 0755 $(DESTDIR)$(mandir)/man1
-	./install-sh -c -s -m 0755 src/bmon $(DESTDIR)$(prefix)/bin
+	./install-sh -c -m 0755 src/bmon $(DESTDIR)$(prefix)/bin
 	./install-sh -c    -m 0644 man/bmon.1 $(DESTDIR)$(mandir)/man1
 
 show: Makefile.opts
