diff -ur ladspa_sdk_orig/src/makefile ladspa_sdk/src/makefile
--- ladspa_sdk_orig/src/makefile	2004-08-15 23:32:53.516055360 +0200
+++ ladspa_sdk/src/makefile	2004-08-15 23:35:21.579546288 +0200
@@ -48,6 +48,7 @@
 #
 
 test:	/tmp/test.wav ../snd/noise.wav always
+ifeq ($(shell which play), /usr/bin/play)
 	@echo ---------------------------------------------
 	@echo First listen to the white noise input signal:
 	@echo ---------------------------------------------
@@ -57,6 +58,7 @@
 	@echo -------------------------
 	@echo Should be a noise band around 6000Hz, repeated quietly after 1s.
 	play /tmp/test.wav
+endif
 
 install:	targets
 	mkdir -p	$(INSTALL_PLUGINS_DIR)				\
@@ -74,7 +74,7 @@
 	echo ; ../bin/analyseplugin -l ../plugins/sine.so ; echo
 	../bin/analyseplugin ../plugins/amp.so
 	../bin/analyseplugin ../plugins/noise.so
-	time ../bin/applyplugin	-s 1					\
+	../bin/applyplugin	-s 1					\
 				../snd/noise.wav /tmp/test.wav		\
 				../plugins/filter.so lpf 500		\
 				../plugins/filter.so lpf 500		\

