diff -ur ladspa_sdk.orig/src/applyplugin.c ladspa_sdk/src/applyplugin.c
--- ladspa_sdk.orig/src/applyplugin.c	2002-05-18 14:01:36 -0300
+++ ladspa_sdk/src/applyplugin.c	2006-10-31 18:25:23 -0300
@@ -6,7 +6,11 @@
 /*****************************************************************************/
 
 #include <dlfcn.h>
+#ifdef __FreeBSD__
+#include <machine/endian.h>
+#else
 #include <endian.h>
+#endif
 #include <errno.h>
 #include <math.h>
 #include <stdlib.h>
diff -ur ladspa_sdk.orig/src/makefile ladspa_sdk/src/makefile
--- ladspa_sdk.orig/src/makefile	2006-10-31 18:23:47 -0300
+++ ladspa_sdk/src/makefile	2006-10-31 18:24:51 -0300
@@ -15,7 +15,12 @@
 #
 
 INCLUDES	=	-I.
+# FreeBSD doesn't need -ldl
+ifeq ($(shell uname -s), FreeBSD)
+LIBRARIES	=	-lm
+else
 LIBRARIES	=	-ldl -lm
+endif
 CFLAGS		=	$(INCLUDES) -Wall -Werror -O3 -fPIC
 CXXFLAGS	=	$(CFLAGS)
 PLUGINS		=	../plugins/amp.so				\
