diff -Nru tinycobol-0.64/cobrun/Makefile.in tinycobol-0.64.new/cobrun/Makefile.in
--- tinycobol-0.64/cobrun/Makefile.in	2008-10-30 08:52:49.000000000 +0100
+++ tinycobol-0.64.new/cobrun/Makefile.in	2008-12-08 23:08:00.000000000 +0100
@@ -11,12 +11,12 @@
 datarootdir = ${prefix}/share
 pkgdatadir = ${datarootdir}/htcobol
 #cobdir_docdir=@cobdir_docdir@
-cobdir_docdir=${prefix}/doc/@cobdir_docdir@
+cobdir_docdir=${prefix}/share/doc/@cobdir_docdir@
 
 INSTALL=@INSTALL@
 INSTALL_DATA=@INSTALL_DATA@
 INSTALL_DIR=$(INSTALL) -d -m 755 
-INSTALL_PROGRAM=$(INSTALL) -s -m 755
+INSTALL_PROGRAM=$(INSTALL) -m 755
 
 INSTBIN=@cobbin_default@
 INSTRC=@cobdir_default@
@@ -37,7 +37,7 @@
 LIBS1=@htg_ld_args_defaults@
 LIBS2=-L../lib -lhtcobol @htg_ld_args_defaults@
 LDFLAGS=@LDFLAGS@
-CCXFLAGS=${INCLUDES}
+CCXFLAGS=${CFLAGS} ${INCLUDES}
 
 SRC01 = htcobrun.c
 
diff -Nru tinycobol-0.64/compiler/Makefile.in tinycobol-0.64.new/compiler/Makefile.in
--- tinycobol-0.64/compiler/Makefile.in	2008-10-30 08:52:49.000000000 +0100
+++ tinycobol-0.64.new/compiler/Makefile.in	2008-12-08 23:08:00.000000000 +0100
@@ -13,12 +13,12 @@
 datarootdir = ${prefix}/share
 pkgdatadir = ${datarootdir}/htcobol
 #cobdir_docdir=@cobdir_docdir@
-cobdir_docdir=${prefix}/doc/@cobdir_docdir@
+cobdir_docdir=${prefix}/share/doc/@cobdir_docdir@
 
 INSTALL=@INSTALL@
 INSTALL_DATA=@INSTALL_DATA@
 INSTALL_DIR=$(INSTALL) -d -m 755 
-INSTALL_PROGRAM=$(INSTALL) -s -m 755
+INSTALL_PROGRAM=$(INSTALL) -m 755
 
 RM= rm -f
 MKDIR=mkdir -p
@@ -38,7 +38,7 @@
 # -DDEBUG_COMPILER -DDEBUG_SCANNER
 #
 INCLUDES=@INCLUDES@
-CCXFLAGS=${INCLUDES} @CCXFLAGS@ -Wall -pedantic -Wno-long-long
+CCXFLAGS=${INCLUDES} ${CFLAGS} @CCXFLAGS@ -Wall -pedantic -Wno-long-long
 #CCXFLAGS1=${INCLUDES} @CCXFLAGS@
 CCXFLAGS1=${INCLUDES}
 LIBS=@LIBS@
diff -Nru tinycobol-0.64/info/Makefile.in tinycobol-0.64.new/info/Makefile.in
--- tinycobol-0.64/info/Makefile.in	2008-10-30 08:52:49.000000000 +0100
+++ tinycobol-0.64.new/info/Makefile.in	2008-12-08 23:08:00.000000000 +0100
@@ -16,15 +16,15 @@
 #datadir = ${datarootdir}
 datarootdir = ${prefix}/share
 pkgdatadir = ${datarootdir}/htcobol
-mandir = $(prefix)/man/man1
+mandir = $(prefix)/share/man/man1
 
 #cobdir_docdir=@cobdir_docdir@
-cobdir_docdir=${prefix}/doc/@cobdir_docdir@
+cobdir_docdir=${prefix}/share/doc/@cobdir_docdir@
 
 INSTALL=@INSTALL@
 INSTALL_DATA=@INSTALL_DATA@
 INSTALL_DIR=$(INSTALL) -d -m 755 
-INSTALL_PROGRAM=$(INSTALL) -s -m 755
+INSTALL_PROGRAM=$(INSTALL) -m 755
 
 #mandir=@mandir@
 #docdir=@docdir@
diff -Nru tinycobol-0.64/lib/Makefile.in tinycobol-0.64.new/lib/Makefile.in
--- tinycobol-0.64/lib/Makefile.in	2008-10-30 08:52:50.000000000 +0100
+++ tinycobol-0.64.new/lib/Makefile.in	2008-12-08 23:08:00.000000000 +0100
@@ -9,7 +9,7 @@
 datarootdir = ${prefix}/share
 pkgdatadir = ${datarootdir}/htcobol
 #cobdir_docdir=@cobdir_docdir@
-cobdir_docdir=${prefix}/doc/@cobdir_docdir@
+cobdir_docdir=${prefix}/share/doc/@cobdir_docdir@
 lib_dir=${exec_prefix}/lib
 
 INSTRC=@cobdir_default@
@@ -18,7 +18,7 @@
 INSTALL=@INSTALL@
 INSTALL_DATA=@INSTALL_DATA@
 INSTALL_DIR=$(INSTALL) -d -m 755 
-INSTALL_PROGRAM=$(INSTALL) -s -m 755
+INSTALL_PROGRAM=$(INSTALL) -m 755
 
 RANLIB=@RANLIB@
 AR=@AR@
@@ -36,8 +36,8 @@
 INCLUDES=@INCLUDES@
 LIBS=@htg_ld_args_defaults@
 LDFLAGS=@LDFLAGS@
-CCXFLAGS=${INCLUDES} @CCXFLAGS@ -Wall 
-CCXFLAGS1=${INCLUDES} @CCXFLAGS@ -Wall -fPIC -DPIC
+CCXFLAGS=${INCLUDES} ${CFLAGS} @CCXFLAGS@ -Wall 
+CCXFLAGS1=${INCLUDES} ${CFLAGS} @CCXFLAGS@ -Wall -fPIC -DPIC
 ACXFLAGS=@ACXFLAGS@
 
 #
--- tinycobol-0.64/compiler/Makefile.in	2009-07-17 19:46:28.000000000 +0200
+++ tinycobol-0.64/compiler/Makefile.in.new	2009-07-17 20:13:04.000000000 +0200
@@ -78,14 +78,16 @@
 #	strip $@
 
 
-scan.c: scan.l
+scan.c: scan.l htcobol.tab.c
 	$(LEX) -o$@ $< 
 
-htcobol.tab.c: htcobol.y scan.c
+htcobol.tab.c: htcobol.y
 	$(YACC) -dv -b htcobol $<  
 
 #htcobgen.o: htcoboly.h htcobol.c
 
+reswords.o: htcobol.tab.c reswords.c
+
 htcobol.o: htcoboly.h
 	$(CCX) $(CCXFLAGS) -c htcobol.c
 
--- tinycobol-0.64/Makefile.in	2009-07-17 19:35:54.000000000 +0200
+++ tinycobol-0.64/Makefile.in.new	2009-07-17 20:14:42.000000000 +0200
@@ -44,7 +44,7 @@
 all:
 	@for i in ${subdirs}; do \
 		echo Making all in $$i ; \
-		(cd $$i; ${MAKE} all) ; \
+		(cd $$i; ${MAKE} all) || exit ; \
 	done
 
 install: all
