If /proc isnt mounted, `top` will cause screen corruption ...

--- procps-3.2.3/top.c
+++ procps-3.2.3/top.c
@@ -3201,6 +3201,10 @@
 {
    (void)dont_care_argc;
    before(*argv);
+
+   if (access("/proc/stat", F_OK))
+      std_err("/proc is not mounted, required for output data");
+
                                         //                 +-------------+
    windows_stage1();                    //                 top (sic) slice
    configs_read();                      //                 > spread etc, <
