Slightly prettier error message

--- proc/readproc.c
+++ proc/readproc.c
@@ -907,3 +907,3 @@
     if(file2str("/proc/self", "stat", sbuf, sizeof sbuf) == -1){
-        fprintf(stderr, "Error, do this: mount -t proc none /proc\n");
+        fprintf(stderr, "Error, do this: mount -t proc proc /proc\n");
         _exit(47);
--- proc/sysinfo.c
+++ proc/sysinfo.c
@@ -31,4 +31,4 @@
 "  To mount /proc at boot you need an /etc/fstab line like:\n"	\
-"      /proc   /proc   proc    defaults\n"			\
-"  In the meantime, run \"mount /proc /proc -t proc\"\n"
+"      proc   /proc   proc    defaults\n"			\
+"  In the meantime, run \"mount proc /proc -t proc\"\n"
 
