--- a/src/contacts-main.c	2008-06-06 11:26:52.000000000 +0200
+++ b/src/contacts-main.c	2008-11-21 23:39:19.436028277 +0100
@@ -36,7 +36,9 @@
 #include "contacts-edit-pane.h"
 #include "contacts-ui.h"
 
+#ifdef HAVE_DBUS
 #include "contacts-dbus.h"
+#endif
 
 void
 contacts_update_treeview (ContactsData *data)
@@ -214,7 +216,9 @@
 	bacon_message_connection_set_callback (
 		mc, (BaconMessageReceivedFunc)contacts_bacon_cb, data);
 
+#ifdef HAVE_DBUS
 	contacts_dbus_init_for_session (data);
+#endif
 
 	/* Set critical errors to close application */
 	//g_log_set_always_fatal (G_LOG_LEVEL_CRITICAL);
