--- music-0.2.0-orig/lyrics.c	2007-04-10 03:56:26.000000000 +0200
+++ music-0.2.0/lyrics.c	2007-07-31 15:08:45.000000000 +0200
@@ -232,8 +232,8 @@
   char *q = NULL;
   char songFile[1024];
 
-  datei = config;
-  datei = datei + "/scripts/music_getlyrics.py";
+  
+  datei = "/usr/share/vdr/music/scripts/music_getlyrics.py";
 
   SetHelp(tr("Parent"),tr("Track-"),tr("Track+"),tr("Save"));
 
@@ -279,11 +279,7 @@
         mgr->Next();
         break;
       case kBlue:
-        if(user_is_admin)
 	  SaveExternLyrics(); 
-	else
-	  Skins.Message(mtError, tr("Operation denied !"));
-  	break;
       case kOk:
       case kRed:
       case kBack:
