diff -ruN music-0.2.0/mplayer.c~ music-0.2.0/mplayer.c
--- music-0.2.0/mplayer.c~
+++ music-0.2.0/mplayer.c
@@ -231,7 +231,12 @@
 #else
     Interface->Close();
 #endif
+#if VDRVERSNUM >= 10500
+    visible = modeOnly = false;
+    SetNeedsFastResponse(visible);
+#else
     needsFastResponse=visible=modeOnly=false;
+#endif
     }
 }
 
@@ -302,7 +307,13 @@
       Interface->Clear();
       if(MPlayerSetup.OsdPos>0) Interface->Fill(0,3,Interface->Width(),MPlayerSetup.OsdPos,clrTransparent);
 #endif
+#if VDRVERSNUM >= 10500
+      visible = true;
+      SetNeedsFastResponse(visible);
+      modeOnly=false;
+#else
       needsFastResponse=visible=true; modeOnly=false;
+#endif
       lastCurrent=lastTotal=-1;
       flush=true;
       }
