diff -Naur music-0.9.3-testing.orig//Makefile music-0.9.3-testing/Makefile
--- music-0.9.3-testing.orig//Makefile  2010-03-18 18:54:58.000000000 +0100
+++ music-0.9.3-testing/Makefile        2010-03-18 18:56:11.000000000 +0100
@@ -72,7 +72,7 @@
 APIVERSION = $(shell sed -ne '/define APIVERSION/s/^.*"\(.*\)".*$$/\1/p' $(VDRDIR)/config.h)

 ### Test whether VDR has locale support
-VDRLOCALE = $(shell grep '^LOCALEDIR' $(VDRDIR)/Makefile)
+VDRLOCALE = $(shell grep '^LOCDIR' $(VDRDIR)/Make.config)

 ### The name of the distribution archive:

