diff -ru linuxtv-dvb-apps-20070114-orig/lib/libdvbapi/dvbvideo.c linuxtv-dvb-apps-20070114/lib/libdvbapi/dvbvideo.c
--- linuxtv-dvb-apps-20070114-orig/lib/libdvbapi/dvbvideo.c	2007-07-18 10:03:47.902372374 +0200
+++ linuxtv-dvb-apps-20070114/lib/libdvbapi/dvbvideo.c	2007-07-18 10:04:15.404812842 +0200
@@ -25,6 +25,7 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <sys/ioctl.h>
+#include <linux/types.h>
 #include <linux/dvb/video.h>
 #include <errno.h>
 #include "dvbvideo.h"
