diff -NrU5 liferea-1.7.6.original/src/fl_sources/ttrss_source_feed.c liferea-1.7.6/src/fl_sources/ttrss_source_feed.c
--- liferea-1.7.6.original/src/fl_sources/ttrss_source_feed.c	2011-06-07 15:25:28.000000000 -0600
+++ liferea-1.7.6/src/fl_sources/ttrss_source_feed.c	2011-09-11 07:02:30.000000000 -0600
@@ -72,16 +72,12 @@
 				item_set_title (item, json_get_string (node, "title"));
 				item_set_source (item, json_get_string (node, "link"));
 				item_set_description (item, json_get_string (node, "content"));
 				item->time = json_get_int (node, "updated");
 				
-				if (json_get_bool (node, "unread")) {
+				if (json_get_bool (node, "unread"))
 					item->readStatus = FALSE;
-				}
-				else {
-					item->readStatus = TRUE;
-				}
 				if (json_get_bool (node, "marked"))
 					item->flagStatus = TRUE;
 					
 				items = g_list_append (items, (gpointer)item);
 				
diff -NrU5 liferea-1.7.6.original/src/itemset.c liferea-1.7.6/src/itemset.c
--- liferea-1.7.6.original/src/itemset.c	2011-06-07 15:25:28.000000000 -0600
+++ liferea-1.7.6/src/itemset.c	2011-09-11 07:02:30.000000000 -0600
@@ -77,11 +77,11 @@
 	GList		*oldItemIdIter = items;
 	itemPtr		oldItem = NULL;
 	gboolean	found, equal = FALSE;
 
 	/* determine if we should add it... */
-	debug2 (DEBUG_CACHE, "check new item for merging: \"%s\", %i", item_get_title (newItem), allowUpdates);
+	debug1 (DEBUG_CACHE, "check new item for merging: \"%s\"", item_get_title (newItem));
 		
 	/* compare to every existing item in this feed */
 	found = FALSE;
 	while (oldItemIdIter) {
 		oldItem = (itemPtr)(oldItemIdIter->data);
@@ -110,16 +110,10 @@
 
 		/* best case: they both have ids (position important: id check is useless without knowing if the items are different!) */
 		if (item_get_id (oldItem)) {			
 			if (0 == strcmp (item_get_id (oldItem), item_get_id (newItem))) {
 				found = TRUE;
-
-				/* found corresponding item, check if they are REALLY equal (eg, read status may have changed) */
-				if(oldItem->readStatus != newItem->readStatus) 
-					equal = FALSE;
-				if(oldItem->flagStatus != newItem->flagStatus)
-					equal = FALSE;
 				break;
 			} else {
 				/* different ids, but the content might be still equal (e.g. empty)
 				   so we need to explicitly unset the equal flag !!!  */
 				equal = FALSE;
@@ -158,13 +152,10 @@
 				oldItem->updateStatus = TRUE;
 				// FIXME: this does not remove metadata from DB
 				metadata_list_free (oldItem->metadata);
 				oldItem->metadata = newItem->metadata;
 				newItem->metadata = NULL;
-				oldItem->readStatus = newItem->readStatus;
-				oldItem->flagStatus = newItem->flagStatus;
-				
 				db_item_update (oldItem);
 				debug0 (DEBUG_CACHE, "-> item already existing and was updated");
 			} else {
 				debug0 (DEBUG_CACHE, "-> item updates not merged because of parser errors");
 			}
