--- umockdev-0.2.1/Makefile.am
+++ umockdev-0.2.1/Makefile.am
@@ -41,21 +41,7 @@
 # use this for running Python tests
 PYTEST = $(TESTS_ENVIRONMENT) $(PYTHON) -Wd -Werror::PendingDeprecationWarning -Werror::DeprecationWarning
 
-AM_CFLAGS = 	\
-	-Werror=missing-prototypes	\
-	-Werror=strict-prototypes	\
-	-Werror=nested-externs		\
-	-Werror=pointer-arith		\
-	-Werror=implicit-function-declaration \
-	-Werror=pointer-arith		\
-	-Werror=init-self		\
-	-Werror=format-security		\
-	-Werror=format=2		\
-	-Werror=missing-include-dirs	\
-	-Werror=unused-variable		\
-	-Werror=return-type		\
-	-Werror=uninitialized		\
-	-Wall
+AM_CFLAGS = -Wall
 
 # this allows us to use regexp literals without warnings
 AM_VALAFLAGS = --enable-experimental
@@ -82,9 +68,9 @@
 
 # this is an LD_PRELOAD library, so remove static library and libtool wrappers
 libumockdev-preload-install-hook:
-	rm $(DESTDIR)$(libdir)/libumockdev-preload.a
-	rm $(DESTDIR)$(libdir)/libumockdev-preload.so
-	rm $(DESTDIR)$(libdir)/libumockdev-preload.la
+	rm -f $(DESTDIR)$(libdir)/libumockdev-preload.a
+	rm -f $(DESTDIR)$(libdir)/libumockdev-preload.so
+	rm -f $(DESTDIR)$(libdir)/libumockdev-preload.la
 
 libumockdev-preload-uninstall-hook:
 	rm $(DESTDIR)$(libdir)/libumockdev-preload.so.*
