--- ack-2.04/ConfigDefault.pm
+++ ack-2.04/ConfigDefault.pm
@@ -180,6 +180,9 @@
 # Fortran http://en.wikipedia.org/wiki/Fortran
 --type-add=fortran:ext:f,f77,f90,f95,f03,for,ftn,fpp
 
+# Gentoo http://www.gentoo.org/
+--type-add=gentoo:ext:ebuild,eclass
+
 # Google Go http://golang.org/
 --type-add=go:ext:go
 
--- ack-2.04/t/ack-filetypes.t
+++ ack-2.04/t/ack-filetypes.t
@@ -25,6 +25,7 @@
 elisp
 erlang
 fortran
+gentoo
 go
 groovy
 haskell
