--- a/common/argus_auth.c
+++ b/common/argus_auth.c
@@ -127,7 +127,7 @@
 static int RaSimple(void *context, int, const char **, unsigned *);
 static int RaGetSecret(sasl_conn_t *, void *context, int, sasl_secret_t **);
 
-int ArgusInitializeAuthentication (struct ArgusInput *input, sasl_conn_t *)
+int RaSaslNegotiate (struct ArgusInput *input);
 int RaGetSaslString (int, char *, int);
 int RaSendSaslString (int, const char *, int, int);
 
@@ -433,7 +433,7 @@
    int in, out, retn = 0;
    sasl_conn_t *conn;
 
-   conn = input->conn;
+   conn = input->sasl_conn;
    in  = input->in;
    out = input->out;
 
