diff --git a/action.c b/action.c
index 259fb66..6b52d70 100644
--- a/action.c
+++ b/action.c
@@ -98,7 +98,7 @@
 #include <strings.h>
 #include <time.h>
 #include <errno.h>
-#include <json/json.h>
+#include <json.h>
 
 #include "dirty.h"
 #include "template.h"
diff --git a/configure.ac b/configure.ac
index d6d5c92..1970f60 100644
--- a/configure.ac
+++ b/configure.ac
@@ -33,7 +33,9 @@ PKG_PROG_PKG_CONFIG
 
 # modules we require
 PKG_CHECK_MODULES(LIBESTR, libestr >= 0.1.5)
-PKG_CHECK_MODULES([JSON_C], [json])
+PKG_CHECK_MODULES([JSON_C], [json],, [
+	PKG_CHECK_MODULES([JSON_C], [json-c])
+])
 
 case "${host}" in
   *-*-linux*)
@@ -821,7 +823,7 @@ if test "x$enable_rsyslogrt" = "xyes"; then
   RSRT_LIBS1="\$(top_builddir)/runtime/librsyslog.la"
 fi
 AM_CONDITIONAL(ENABLE_RSYSLOGRT, test x$enable_rsyslogrt = xyes)
-RSRT_CFLAGS="\$(RSRT_CFLAGS1) \$(LIBESTR_CFLAGS) \$(JSON_C_FLAGS)"
+RSRT_CFLAGS="\$(RSRT_CFLAGS1) \$(LIBESTR_CFLAGS) \$(JSON_C_CFLAGS)"
 RSRT_LIBS="\$(RSRT_LIBS1) \$(LIBESTR_LIBS) \$(JSON_C_LIBS)"
 AC_SUBST(RSRT_CFLAGS1)
 AC_SUBST(RSRT_LIBS1)
diff --git a/plugins/imkmsg/kmsg.c b/plugins/imkmsg/kmsg.c
index 822d3db..172ff4d 100644
--- a/plugins/imkmsg/kmsg.c
+++ b/plugins/imkmsg/kmsg.c
@@ -34,7 +34,7 @@
 #include <ctype.h>
 #include <sys/klog.h>
 #include <sys/sysinfo.h>
-#include <json/json.h>
+#include <json.h>
 
 #include "rsyslog.h"
 #include "srUtils.h"
diff --git a/plugins/mmaudit/mmaudit.c b/plugins/mmaudit/mmaudit.c
index 6b6b804..c7cff2c 100644
--- a/plugins/mmaudit/mmaudit.c
+++ b/plugins/mmaudit/mmaudit.c
@@ -43,7 +43,7 @@
 #include <errno.h>
 #include <unistd.h>
 #include <ctype.h>
-#include <json/json.h>
+#include <json.h>
 #include "conf.h"
 #include "syslogd-types.h"
 #include "template.h"
diff --git a/plugins/mmjsonparse/mmjsonparse.c b/plugins/mmjsonparse/mmjsonparse.c
index 35f69aa..b16aef0 100644
--- a/plugins/mmjsonparse/mmjsonparse.c
+++ b/plugins/mmjsonparse/mmjsonparse.c
@@ -35,7 +35,7 @@
 #include <errno.h>
 #include <unistd.h>
 #include <ctype.h>
-#include <json/json.h>
+#include <json.h>
 #include "conf.h"
 #include "syslogd-types.h"
 #include "template.h"
diff --git a/plugins/mmnormalize/mmnormalize.c b/plugins/mmnormalize/mmnormalize.c
index fcadc32..f93974a 100644
--- a/plugins/mmnormalize/mmnormalize.c
+++ b/plugins/mmnormalize/mmnormalize.c
@@ -40,7 +40,7 @@
 #include <unistd.h>
 #include <libestr.h>
 #include <libee/libee.h>
-#include <json/json.h>
+#include <json.h>
 #include <liblognorm.h>
 #include "conf.h"
 #include "syslogd-types.h"
diff --git a/plugins/ommongodb/ommongodb.c b/plugins/ommongodb/ommongodb.c
index dd99741..64d501d 100644
--- a/plugins/ommongodb/ommongodb.c
+++ b/plugins/ommongodb/ommongodb.c
@@ -33,7 +33,7 @@
 #include <stdint.h>
 #include <time.h>
 #include <mongo.h>
-#include <json/json.h>
+#include <json.h>
 /* For struct json_object_iter, should not be necessary in future versions */
 #include <json/json_object_private.h>
 
diff --git a/runtime/msg.c b/runtime/msg.c
index a227567..e683cdb 100644
--- a/runtime/msg.c
+++ b/runtime/msg.c
@@ -41,7 +41,7 @@
 #endif
 #include <netdb.h>
 #include <libestr.h>
-#include <json/json.h>
+#include <json.h>
 /* For struct json_object_iter, should not be necessary in future versions */
 #include <json/json_object_private.h>
 #if HAVE_MALLOC_H
diff --git a/runtime/msg.h b/runtime/msg.h
index 6faf066..ac220b6 100644
--- a/runtime/msg.h
+++ b/runtime/msg.h
@@ -30,7 +30,7 @@
 
 #include <pthread.h>
 #include <libestr.h>
-#include <json/json.h>
+#include <json.h>
 #include "obj.h"
 #include "syslogd-types.h"
 #include "template.h"
diff --git a/template.c b/template.c
index b675255..9cefa05 100644
--- a/template.c
+++ b/template.c
@@ -34,7 +34,7 @@
 #include <string.h>
 #include <ctype.h>
 #include <assert.h>
-#include <json/json.h>
+#include <json.h>
 #include "stringbuf.h"
 #include "syslogd-types.h"
 #include "template.h"
diff --git a/template.h b/template.h
index 318db6f..87a1c77 100644
--- a/template.h
+++ b/template.h
@@ -30,7 +30,7 @@
 #ifndef	TEMPLATE_H_INCLUDED
 #define	TEMPLATE_H_INCLUDED 1
 
-#include <json/json.h>
+#include <json.h>
 #include <libestr.h>
 #include "regexp.h"
 #include "stringbuf.h"
