https://bugs.gentoo.org/show_bug.cgi?id=417773
diff -rupN gpsdrive-2.11.org//src/CMakeLists.txt gpsdrive-2.11/src/CMakeLists.txt
--- gpsdrive-2.11.org//src/CMakeLists.txt	2012-06-19 20:02:50.960300653 +0200
+++ gpsdrive-2.11/src/CMakeLists.txt	2012-06-19 20:03:03.263146846 +0200
@@ -28,7 +28,7 @@ if (WITH_POSTGIS)
 endif (WITH_POSTGIS)
 
 if (WITH_MAPNIK)
-  find_package(Boost COMPONENTS filesystem REQUIRED)
+  find_package(Boost COMPONENTS filesystem thread REQUIRED)
   find_package(Mapnik REQUIRED)
   add_definitions(-DMAPNIK)
 endif (WITH_MAPNIK)
@@ -177,6 +177,7 @@ if (WITH_MAPNIK)
   set(GPSDRIVE_LINK_LIBRARIES
     ${GPSDRIVE_LINK_LIBRARIES}
     ${Boost_FILESYSTEM_LIBRARY}
+    ${Boost_THREAD_LIBRARY}
     ${MAPNIK_LIBRARIES}
   )
 endif (WITH_MAPNIK)
