--- gpsdrive-2.11/cmake/Modules/FindGTK2.cmake.gdk-pixbuf2	2010-07-28 18:21:37.000000000 +0100
+++ gpsdrive-2.11/cmake/Modules/FindGTK2.cmake	2010-07-28 18:23:09.000000000 +0100
@@ -192,6 +192,20 @@
     )
     #MESSAGE(STATUS "DEBUG: GTK2_ATK_INCLUDE_PATH = ${GTK2_ATK_INCLUDE_PATH}")
 
+    if (${CMAKE_MAJOR_VERSION} EQUAL 2 AND ${CMAKE_MINOR_VERSION} EQUAL 4)
+      pkgconfig(gdk-pixbuf-2.0 _GDKPIXBUF2_INCLUDEDIR _GDKPIXBUF2inkDir _GDKPIXBUF2_LDFLAGS _GDKPIXBUF2_CFLAGS)
+    else (${CMAKE_MAJOR_VERSION} EQUAL 2 AND ${CMAKE_MINOR_VERSION} EQUAL 4)
+      if (PKG_CONFIG_FOUND)
+        pkg_check_modules(_GDKPIXBUF2 gdk-pixbuf-2.0)
+      endif (PKG_CONFIG_FOUND)
+    endif (${CMAKE_MAJOR_VERSION} EQUAL 2 AND ${CMAKE_MINOR_VERSION} EQUAL 4)
+
+    FIND_PATH(GTK2_GDKPIXBUF2_INCLUDE_PATH gdk-pixbuf/gdk-pixbuf.h
+      ${_GDKPIXBUF2_INCLUDEDIR} /usr/include/gdk-pixbuf-2.0 /usr/include/gtk-2.0
+    )
+
+    #MESSAGE(STATUS "DEBUG:GTK2_GDKPIXBUF2_INCLUDE_PATH = ${GTK2_GDKPIXBUF2_INCLUDE_PATH}")
+
     FIND_LIBRARY(GTK2_GTKGL_LIBRARY
       NAMES
         gtkgl
@@ -329,6 +343,7 @@
                       ${GTK2_GLIB_INCLUDE_PATH}
                       ${GTK2_PANGO_INCLUDE_PATH}
                       ${GTK2_GDKCONFIG_INCLUDE_PATH}
+                      ${GTK2_GDKPIXBUF2_INCLUDE_PATH}
                       ${GTK2_ATK_INCLUDE_PATH}
                       ${GTK2_CAIRO_INCLUDE_PATH}
                     )
