diff -ru ctags-5.6-orig/parsers.h ctags-5.6/parsers.h
--- ctags-5.6-orig/parsers.h	2006-05-30 06:37:12.000000000 +0200
+++ ctags-5.6/parsers.h	2007-06-16 18:38:31.000000000 +0200
@@ -16,6 +16,7 @@
 
 /* Add the name of any new parser definition function here */
 #define PARSER_LIST \
+	AdaParser, \
 	AsmParser, \
 	AspParser, \
 	AwkParser, \
diff -ru ctags-5.6-orig/source.mak ctags-5.6/source.mak
--- ctags-5.6-orig/source.mak	2004-06-10 05:01:17.000000000 +0200
+++ ctags-5.6/source.mak	2007-06-16 18:37:40.000000000 +0200
@@ -8,6 +8,7 @@
 	strlist.h vstring.h
 
 SOURCES = \
+	ada.c \
 	args.c \
 	asm.c \
 	asp.c \
@@ -58,6 +59,7 @@
     argproc.c mac.c qdos.c
 
 OBJECTS = \
+	ada.$(OBJEXT) \
 	args.$(OBJEXT) \
 	asm.$(OBJEXT) \
 	asp.$(OBJEXT) \
