--- ctags-5.5.4/ruby.c.orig	2005-06-02 14:30:12.813161384 +0100
+++ ctags-5.5.4/ruby.c	2005-06-02 14:30:56.947451952 +0100
@@ -82,7 +82,7 @@
 		{
 		    while (isspace ((int) *cp))
 			++cp;
-		    while (isalnum ((int) *cp)  ||  *cp == '_')
+		    while (isalnum ((int) *cp)  ||  *cp == '_'  ||  *cp == ':')
 		    {
 			vStringPut (name, (int) *cp);
 			++cp;
@@ -99,7 +99,7 @@
 		{
 		    while (isspace ((int) *cp))
 			++cp;
-		    while (isalnum ((int) *cp)  ||  *cp == '_')
+		    while (isalnum ((int) *cp)  ||  *cp == '_'  ||  *cp == ':')
 		    {
 			vStringPut (name, (int) *cp);
 			++cp;
