Index: ebuild-branch/sh.c
===================================================================
--- ebuild-branch/sh.c	(revision 2)
+++ ebuild-branch/sh.c	(revision 3)
@@ -7,7 +7,8 @@
 *   GNU General Public License.
 *
 *   This module contains functions for generating tags for scripts for the
-*   Bourne shell (and its derivatives, the Korn and Z shells).
+*   Bourne shell (and its derivatives, the Korn and Z shells), along with some
+*   specialised kinds of shell scripts (Gentoo ebuilds and eclasses).
 */
 
 /*
@@ -101,7 +102,7 @@
 extern parserDefinition* ShParser (void)
 {
     static const char *const extensions [] = {
-	"sh", "SH", "bsh", "bash", "ksh", "zsh", NULL
+	"sh", "SH", "bsh", "bash", "ksh", "zsh", "ebuild", "eclass", NULL
     };
     parserDefinition* def = parserNew ("Sh");
     def->kinds      = ShKinds;
