--- src/Makefile.am.orig	2012-04-29 19:35:31.858232614 -0400
+++ src/Makefile.am	2012-04-29 19:36:49.389153634 -0400
@@ -11,7 +11,7 @@
 
 
 # define common linker flags
-AM_LDFLAGS = -lmenuw -lformw -lpanelw -lncursesw -lpthread -lcdio -liso9660 -lm -lburn
+LIBS = -lmenuw -lformw -lpanelw -lncursesw -lpthread -lcdio -liso9660 -lm -lburn
 
 
 # list source files
@@ -60,7 +60,8 @@
 	$(top_builddir)/gnulib/lib/libgnulib.a \
 	$(top_builddir)/src/external_tools/libcdwexternaltools.a \
 	$(top_builddir)/src/user_interface/widgets/libcdwwidgets.a \
-	$(top_builddir)/src/tasks/libcdwtasks.a
+	$(top_builddir)/src/tasks/libcdwtasks.a \
+	$(LIBS)
 
 cdw_SOURCES = $(cdw_source_files)
 
@@ -85,7 +86,8 @@
 	$(top_builddir)/gnulib/lib/libgnulib.a \
 	$(top_builddir)/src/external_tools/libcdwexternaltools_check.a \
 	$(top_builddir)/src/user_interface/widgets/libcdwwidgets_check.a \
-	$(top_builddir)/src/tasks/libcdwtasks_check.a
+	$(top_builddir)/src/tasks/libcdwtasks_check.a \
+	$(LIBS)
 
 cdw_tests_SOURCES = $(cdw_source_files)
 
