diff -ru cdw-0.6.0.orig/src/Makefile.am cdw-0.6.0/src/Makefile.am
--- cdw-0.6.0.orig/src/Makefile.am	2010-10-20 14:24:16.000000000 -0400
+++ cdw-0.6.0/src/Makefile.am	2010-11-05 03:49:25.609187912 -0400
@@ -10,10 +10,6 @@
 include $(top_srcdir)/Makefile.cdw.am
 
 
-# define common linker flags
-AM_LDFLAGS = -lmenuw -lformw -lpanelw -lncursesw -lpthread -lcdio -liso9660
-
-
 # list source files
 cdw_source_files = cddb.c main.c
 
@@ -56,7 +52,8 @@
 	$(top_builddir)/src/optical_file_systems/libcdwopticalfilesystems.a \
 	$(top_builddir)/src/configuration/libcdwconfiguration.a \
 	$(top_builddir)/src/user_interface/libcdwuserinterface.a \
-	$(top_builddir)/gnulib/lib/libgnulib.a
+	$(top_builddir)/gnulib/lib/libgnulib.a \
+	-lmenuw -lformw -lpanelw -lncursesw -lpthread -lcdio -liso9660 -lm
 
 cdw_SOURCES = $(cdw_source_files)
 
@@ -77,7 +74,8 @@
 	$(top_builddir)/src/optical_file_systems/libcdwopticalfilesystems_check.a \
 	$(top_builddir)/src/configuration/libcdwconfiguration_check.a \
 	$(top_builddir)/src/user_interface/libcdwuserinterface_check.a \
-	$(top_builddir)/gnulib/lib/libgnulib.a
+	$(top_builddir)/gnulib/lib/libgnulib.a \
+	-lmenuw -lformw -lpanelw -lncursesw -lpthread -lcdio -liso9660 -lm
 
 cdw_tests_SOURCES = $(cdw_source_files)
 
