--- usr/local/bin/qemulator	2007-03-03 22:40:27.000000000 +0100
+++ usr/local/bin/qemulator	2007-03-03 22:32:29.000000000 +0100
@@ -78,6 +78,7 @@
 if not os.path.exists(os.path.join(basedir, "main.py")):
     if os.path.exists(os.path.join(os.getcwd(), "main.py")):
         basedir = os.getcwd()
+basedir = "/usr/lib/qemulator"
 sys.path.insert(0, basedir)
 os.chdir(basedir)  
 #print "basedir: " + basedir
