diff -Naur mdbtools-0.6pre1-orig/src/gmdb2/sql.c mdbtools-0.6pre1/src/gmdb2/sql.c
--- src/gmdb2/sql.c	2004-02-13 13:49:52.000000000 -0500
+++ src/gmdb2/sql.c	2005-10-18 08:00:19.088362686 -0400
@@ -38,6 +38,8 @@
 
 static void gmdb_sql_tree_populate(MdbHandle *mdb, GladeXML *xml);
 static void gmdb_sql_load_query(GladeXML *xml, gchar *file_path);
+void gmdb_sql_save_as_cb(GtkWidget *w, GladeXML *xml);
+void gmdb_sql_save_query(GladeXML *xml, gchar *file_path);
 
 void
 gmdb_sql_close_all()
diff -Naur mdbtools-0.6pre1-orig/src/gmdb2/table_export.c mdbtools-0.6pre1/src/gmdb2/table_export.c
--- src/gmdb2/table_export.c	2003-01-14 12:37:44.000000000 -0500
+++ src/gmdb2/table_export.c	2005-10-18 08:00:19.089362678 -0400
@@ -37,6 +37,8 @@
 #define NEVER "Never"
 #define AUTOMAT "Automatic (where necessary)"
 
+void gmdb_table_export_populate_dialog(GladeXML *xml);
+
 void
 gmdb_print_quote(FILE *outfile, int need_quote, char quotechar, char *colsep, char *str)
 {
diff -Naur mdbtools-0.6pre1-orig/src/odbc/odbc.c mdbtools-0.6pre1/src/odbc/odbc.c
--- src/odbc/odbc.c	2004-05-30 04:06:43.000000000 -0400
+++ src/odbc/odbc.c	2005-10-18 08:00:19.090362669 -0400
@@ -46,6 +46,7 @@
 static SQLRETURN SQL_API _SQLFreeConnect(SQLHDBC hdbc);
 static SQLRETURN SQL_API _SQLFreeEnv(SQLHENV henv);
 static SQLRETURN SQL_API _SQLFreeStmt(SQLHSTMT hstmt, SQLUSMALLINT fOption);
+static void bind_columns(struct _hstmt *stmt);
 
 static void bind_columns (struct _hstmt*);
 
