--- mdbtools-cvs-20050624.orig/src/sql/Makefile.am	2005-01-15 06:02:06.000000000 +0100
+++ mdbtools-cvs-20050624/src/sql/Makefile.am	2010-01-12 13:04:01.000000000 +0100
@@ -1,5 +1,5 @@
 lib_LTLIBRARIES	=	libmdbsql.la
-libmdbsql_la_SOURCES=	mdbsql.c parser.y lexer.l 
+libmdbsql_la_SOURCES=	mdbsql.c parser.y lexer.l parser.h
 libmdbsql_la_LDFLAGS = -version-info 1:0:0
 DISTCLEANFILES = parser.c parser.h lexer.c
 AM_CPPFLAGS	=	-I$(top_srcdir)/include $(GLIB_CFLAGS)
@@ -7,5 +7,8 @@
 LDADD	=	../libmdb/libmdb.la 
 YACC = @YACC@ -d
 
+parser.h: parser.lo
+lexer.lo: parser.h
+
 dist-hook:
 	rm -f $(distdir)/parser.c $(distdir)/parser.h $(distdir)/lexer.c
