diff -u -r mdbtools/src/libmdb/file.c mdbtools.fix/src/libmdb/file.c
--- mdbtools/src/libmdb/file.c	2005-06-28 13:53:40.000000000 +0900
+++ mdbtools.fix/src/libmdb/file.c	2009-09-09 04:02:47.000000000 +0900
@@ -17,6 +17,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
+#include "config.h"
 #include "mdbtools.h"
 
 #ifdef DMALLOC
diff -u -r mdbtools/src/libmdb/iconv.c mdbtools.fix/src/libmdb/iconv.c
--- mdbtools/src/libmdb/iconv.c	2005-09-08 08:27:43.000000000 +0900
+++ mdbtools.fix/src/libmdb/iconv.c	2009-09-09 04:00:36.000000000 +0900
@@ -17,6 +17,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
+#include "config.h"
 #include "mdbtools.h"
 #include "errno.h"
 
