--- Makefile.orig	2006-12-16 00:49:34.000000000 -0500
+++ Makefile	2006-12-16 00:51:56.000000000 -0500
@@ -1,11 +1,11 @@
-CFLAGS+=-g -Os -D_FORTIFY_SOURCE=2 -Wall -W `pkg-config --cflags glib-2.0` 
+CFLAGS+=-Wall -W `pkg-config --cflags glib-2.0` 
 
 all: irqbalance
 
 LIBS=bitmap.o irqbalance.o cputree.o  procinterrupts.o irqlist.o placement.o activate.o network.o powermode.o numa.o classify.o
+LDLIBS=`pkg-config --libs glib-2.0`
 
-irqbalance: .depend $(LIBS)
-	gcc  -g -O2 -D_FORTIFY_SOURCE=2 -Wall  `pkg-config --libs glib-2.0` $(LIBS) -o irqbalance 
+irqbalance: $(LIBS)
 
 clean:
 	rm -f irqbalance *~ *.o .depend
@@ -13,5 +13,5 @@
 # rule for building dependency lists, and writing them to a file
 # named ".depend".
 .depend:
-	rm -f .depend
-	gccmakedep -f- -- $(CFLAGS) -- *.c > .depend
+	$(CC) $(CFLAGS) -MM *.c > .depend
+-include .depend
