diff --git a/Makefile.am b/Makefile.am
index 833bee6..c7780f8 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -38,6 +38,8 @@ install-data-local:
 	for instfile in $(noinst_HEADERS) module/$(LINUX_SYMBOLS); do \
 		$(INSTALL) -D $$instfile $$instdest/$$instfile; \
 	done
+	release=$(ZFS_META_VERSION)-$(ZFS_META_RELEASE); \
+	ln -s zfs-$$release/$(LINUX_VERSION) $(DESTDIR)/usr/src/zfs;
 endif
 
 ctags:
