diff -ur scripts.orig/Makefile.am scripts/Makefile.am
--- scripts.orig/Makefile.am	2009-10-15 21:47:20.000000000 +0200
+++ scripts/Makefile.am	2009-10-15 21:47:57.000000000 +0200
@@ -9,9 +9,7 @@
     get-fcc-rac.pl \
     get-gnis \
     get-pop \
-    get-maptools.sh \
     get-NWSdata \
-    get_shapelib.sh \
     gpx2shape \
     icontable.pl \
     inf2geo.pl \
@@ -39,8 +37,6 @@
     values \
     values.pl \
     waypoint-get.pl \
-    xastir-fixcfg.sh \
-    xastir-migrate.sh \
     Xastir_tigerpoly.py
 
 EXTRA_DIST = $(pkglib_SCRIPTS)
diff -u scripts.orig/permutations.pl scripts/permutations.pl
--- scripts.orig/permutations.pl	2005-03-25 21:53:04.000000000 +0100
+++ scripts/permutations.pl	2008-08-16 07:47:00.000000000 +0200
@@ -34,7 +34,7 @@
 #
 
 
-use lib "/usr/local/lib";
+use lib "/usr/lib";
 use Coordinate;         # WE7U's Coordinate.pm module
 
 
diff -u scripts.orig/test_coord.pl scripts/test_coord.pl
--- scripts.orig/test_coord.pl	2003-01-06 23:42:17.000000000 +0100
+++ scripts/test_coord.pl	2008-08-16 07:47:00.000000000 +0200
@@ -33,7 +33,7 @@
 
 #------------------------------------------------------------------------------------------------
 
-use lib "/usr/local/lib";
+use lib "/usr/lib";
 use Coordinate;		# Snag WE7U's Coordinate module
 
 
diff -u scripts.orig/toporama250k.pl scripts/toporama250k.pl
--- scripts.orig/toporama250k.pl	2008-08-16 07:42:13.000000000 +0200
+++ scripts/toporama250k.pl	2008-08-16 08:03:08.000000000 +0200
@@ -28,14 +28,14 @@
 # map files for the whole of Canada!
 #
 #
-# - "cd /usr/local/share/xastir/maps"
+# - "cd /usr/share/xastir/maps"
 #
 # - Assure you have write privileges in the directory above either
 #   by becoming root using the "su" command, using "sudo", or
 #   temporarily changing ownership and/or privileges on the
 #   "maps" directory.
 #
-# - "/usr/local/lib/xastir/toporama250k.pl"
+# - "/usr/lib/xastir/toporama250k.pl"
 #
 # - The script will create/populate this directory heirarchy:
 #   .../maps/toporama/images/
diff -u scripts.orig/toporama50k.pl scripts/toporama50k.pl
--- scripts.orig/toporama50k.pl	2008-08-16 07:42:13.000000000 +0200
+++ scripts/toporama50k.pl	2008-08-16 08:02:48.000000000 +0200
@@ -27,14 +27,14 @@
 # map files for the whole of Canada!
 #
 #
-# - "cd /usr/local/share/xastir/maps"
+# - "cd /usr/share/xastir/maps"
 #
 # - Assure you have write privileges in the directory above either
 #   by becoming root using the "su" command, using "sudo", or
 #   temporarily changing ownership and/or privileges on the
 #   "maps" directory.
 #
-# - "/usr/local/lib/xastir/toporama50k.pl"
+# - "/usr/lib/xastir/toporama50k.pl"
 #
 # - The script will create/populate this directory heirarchy:
 #   .../maps/toporama/images/
