--- config/Makefile.am.orig	2008-06-12 21:30:44.000000000 +0200
+++ config/Makefile.am	2009-06-06 18:03:32.000000000 +0200
@@ -57,5 +57,5 @@
 config_DATA = ${EXTRA_DIST}
 
 install-exec-hook:
-	rm -f \$(configdir)/nwsfz_ddmmyy.dbfawk
+	rm -f $(DESTDIR)$(configdir)/nwsfz_ddmmyy.dbfawk
 
--- Makefile.am.old	2010-05-17 21:49:11.000000000 +0200
+++ Makefile.am	2010-05-17 21:49:45.000000000 +0200
@@ -63,7 +63,7 @@
 # documentation is FHS compliant.
 
 install-exec-hook:
-	-rm -rf ${pkgdatadir}/doc
+	-rm -rf ${DESTDIR}${pkgdatadir}/doc
  
 mandir=${prefix}/share/man
 man_MANS=xastir.1 callpass.1 xastir_udp_client.1 testdbfawk.1
