diff -uNr BitchX.orig/dll/Makefile.in BitchX/dll/Makefile.in
--- BitchX.orig/dll/Makefile.in	2004-08-31 11:20:42.974299976 +0000
+++ BitchX/dll/Makefile.in	2004-08-31 11:23:19.113563200 +0000
@@ -166,7 +166,7 @@
 PLUGINS = @PLUGINS@
 
 .c.o:
-	$(CC) $(DEFS) $(CFLAGS) -c $<
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $<
 
 $(SHLIB_SUFFIX).o:
 	$(SHLIB_LD) $(SHLIB_CFLAGS) $< -o $@$(SHLIB_SUFFIX)
@@ -191,7 +191,7 @@
 dllinit.o:
 	@( \
 		if test x"$(PLUGINS)" != x""; then \
-			$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/dllinit.c; \
+			$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/dllinit.c; \
 		fi; \
 	)
 
diff -uNr BitchX.orig/dll/abot/Makefile.in BitchX/dll/abot/Makefile.in
--- BitchX.orig/dll/abot/Makefile.in	2004-08-31 11:20:42.969300736 +0000
+++ BitchX/dll/abot/Makefile.in	2004-08-31 11:23:19.094566088 +0000
@@ -165,7 +165,7 @@
 	  && ./config.status
 
 autobot.o: $(srcdir)/autobot.c
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/autobot.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/autobot.c
 
 autobot$(SHLIB_SUFFIX): autobot.o ../dllinit.o
 	$(SHLIB_LD) autobot.o ../dllinit.o $(SHLIB_CFLAGS) -o autobot$(SHLIB_SUFFIX)
diff -uNr BitchX.orig/dll/acro/Makefile.in BitchX/dll/acro/Makefile.in
--- BitchX.orig/dll/acro/Makefile.in	2004-08-31 11:20:42.977299520 +0000
+++ BitchX/dll/acro/Makefile.in	2004-08-31 11:23:19.130560616 +0000
@@ -165,7 +165,7 @@
 	  && ./config.status
 
 acro.o: $(srcdir)/acro.c
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/acro.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/acro.c
 
 acro$(SHLIB_SUFFIX): acro.o ../dllinit.o
 	$(SHLIB_LD) acro.o ../dllinit.o $(SHLIB_CFLAGS) -o acro$(SHLIB_SUFFIX)
diff -uNr BitchX.orig/dll/aim/Makefile.in BitchX/dll/aim/Makefile.in
--- BitchX.orig/dll/aim/Makefile.in	2004-08-31 11:20:43.013294048 +0000
+++ BitchX/dll/aim/Makefile.in	2004-08-31 11:23:19.166555144 +0000
@@ -169,20 +169,20 @@
 	  && ./config.status
 
 $(topdir)/source/compat.o: $(top_srcdir)/source/compat.c
-	$(CC) $(AIMDEFS) $(CFLAGS) \
+	$(CC) $(AIMDEFS) $(CFLAGS) $(SHLIB_CFLAGS) \
 		-c $(top_srcdir)/source/compat.c -o $(topdir)/source/compat.o
 
 toc.o: $(srcdir)/toc.c
-	$(CC) $(AIMDEFS) $(CFLAGS) -c $(srcdir)/toc.c
+	$(CC) $(AIMDEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/toc.c
 
 util.o: $(srcdir)/util.c
-	$(CC) $(AIMDEFS) $(CFLAGS) -c $(srcdir)/util.c
+	$(CC) $(AIMDEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/util.c
 
 cmd.o: $(srcdir)/cmd.c
-	$(CC) $(AIMDEFS) $(CFLAGS) -c $(srcdir)/cmd.c
+	$(CC) $(AIMDEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/cmd.c
 
 aim.o: $(srcdir)/aim.c
-	$(CC) $(AIMDEFS) $(CFLAGS) -c $(srcdir)/aim.c
+	$(CC) $(AIMDEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/aim.c
 
 aim$(SHLIB_SUFFIX): $(OBJS) 
 	$(SHLIB_LD) $(OBJS) $(SHLIB_CFLAGS) -o aim$(SHLIB_SUFFIX) -Ltoc -ltoc
diff -uNr BitchX.orig/dll/aim/toc/Makefile.in BitchX/dll/aim/toc/Makefile.in
--- BitchX.orig/dll/aim/toc/Makefile.in	2004-08-31 11:20:42.987298000 +0000
+++ BitchX/dll/aim/toc/Makefile.in	2004-08-31 11:23:19.148557880 +0000
@@ -159,7 +159,7 @@
 #DEBUG_CFLAGS = -g -DDEBUG_LIB_TOC
 
 .c.o:
-	$(CC) $(DEFS) $(CFLAGS) $(DEBUG_CFLAGS)  -c $<
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) $(DEBUG_CFLAGS)  -c $<
 
 LOBJS = toc.o network.o util.o server.o interface.o buddy.o ll.o misc.o
 
diff -uNr BitchX.orig/dll/amp/Makefile.in BitchX/dll/amp/Makefile.in
--- BitchX.orig/dll/amp/Makefile.in	2004-08-31 11:20:43.026292072 +0000
+++ BitchX/dll/amp/Makefile.in	2004-08-31 11:23:19.185552256 +0000
@@ -177,7 +177,7 @@
 	$(RM) *~ config.status config.cache config.log config.h Makefile
 
 ../dllinit.o:
-	$(CC) -DOS_$(OS) $(DEFS) $(CFLAGS) -o ../dllinit.o -c ../dllinit.c
+	$(CC) -DOS_$(OS) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -o ../dllinit.o -c ../dllinit.c
 
 amp$(SHLIB_SUFFIX): $(OBJ) ../dllinit.o
 	$(SHLIB_LD) -o amp$(SHLIB_SUFFIX) $(OBJ) ../dllinit.o
@@ -187,7 +187,7 @@
 OS := $(shell uname -s) 
 
 .c.o:
-	$(CC) -DOS_$(OS) $(DEFS) $(CFLAGS) -c $<
+	$(CC) -DOS_$(OS) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $<
 
 depend:
 	gcc -MM $(CPPFLAGS) $(DEFS) $(CFLAGS) *.c > .depend
diff -uNr BitchX.orig/dll/arcfour/Makefile.in BitchX/dll/arcfour/Makefile.in
--- BitchX.orig/dll/arcfour/Makefile.in	2004-08-31 11:20:43.061286752 +0000
+++ BitchX/dll/arcfour/Makefile.in	2004-08-31 11:23:19.203549520 +0000
@@ -165,10 +165,10 @@
 	  && ./config.status
 
 arcfour.o: $(srcdir)/arcfour.c
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/arcfour.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/arcfour.c
 
 md5.o: $(srcdir)/md5.c
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/md5.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/md5.c
 
 arcfour$(SHLIB_SUFFIX): arcfour.o md5.o ../dllinit.o
 	$(SHLIB_LD) arcfour.o md5.o ../dllinit.o $(SHLIB_CFLAGS) -o arcfour$(SHLIB_SUFFIX)
diff -uNr BitchX.orig/dll/autocycle/Makefile.in BitchX/dll/autocycle/Makefile.in
--- BitchX.orig/dll/autocycle/Makefile.in	2004-08-31 11:20:43.071285232 +0000
+++ BitchX/dll/autocycle/Makefile.in	2004-08-31 11:23:19.221546784 +0000
@@ -168,7 +168,7 @@
 	  && ./config.status
 
 autocycle.o: $(srcdir)/autocycle.c
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/autocycle.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/autocycle.c
 
 autocycle$(SHLIB_SUFFIX): autocycle.o ../dllinit.o
 	$(SHLIB_LD) autocycle.o ../dllinit.o $(SHLIB_CFLAGS) -o autocycle$(SHLIB_SUFFIX)
diff -uNr BitchX.orig/dll/blowfish/Makefile.in BitchX/dll/blowfish/Makefile.in
--- BitchX.orig/dll/blowfish/Makefile.in	2004-08-31 11:20:43.075284624 +0000
+++ BitchX/dll/blowfish/Makefile.in	2004-08-31 11:23:19.240543896 +0000
@@ -165,7 +165,7 @@
 	  && ./config.status
 
 blowfish.o: $(srcdir)/blowfish.c
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/blowfish.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/blowfish.c
 
 blowfish$(SHLIB_SUFFIX): blowfish.o ../dllinit.o
 	$(SHLIB_LD) blowfish.o ../dllinit.o $(SHLIB_CFLAGS) -o blowfish$(SHLIB_SUFFIX) $(TCL_LIBS)
diff -uNr BitchX.orig/dll/cavlink/Makefile.in BitchX/dll/cavlink/Makefile.in
--- BitchX.orig/dll/cavlink/Makefile.in	2004-08-31 11:20:43.090282344 +0000
+++ BitchX/dll/cavlink/Makefile.in	2004-08-31 11:23:19.257541312 +0000
@@ -165,7 +165,7 @@
 	  && ./config.status
 
 cavlink.o: $(srcdir)/cavlink.c
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/cavlink.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/cavlink.c
 
 cavlink$(SHLIB_SUFFIX): cavlink.o ../dllinit.o
 	$(SHLIB_LD) cavlink.o ../dllinit.o ${SHLIB_CFLAGS} -o cavlink$(SHLIB_SUFFIX)
diff -uNr BitchX.orig/dll/cdrom/Makefile.in BitchX/dll/cdrom/Makefile.in
--- BitchX.orig/dll/cdrom/Makefile.in	2004-08-31 11:20:43.096281432 +0000
+++ BitchX/dll/cdrom/Makefile.in	2004-08-31 11:23:19.275538576 +0000
@@ -165,7 +165,7 @@
 	  && ./config.status
 
 cdrom.o: $(srcdir)/cdrom.c
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/cdrom.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/cdrom.c
 
 cdrom$(SHLIB_SUFFIX): cdrom.o ../dllinit.o
 	$(SHLIB_LD) cdrom.o ../dllinit.o ${SHLIB_CFLAGS} -o cdrom$(SHLIB_SUFFIX)
diff -uNr BitchX.orig/dll/encrypt/Makefile.in BitchX/dll/encrypt/Makefile.in
--- BitchX.orig/dll/encrypt/Makefile.in	2004-08-31 11:20:43.100280824 +0000
+++ BitchX/dll/encrypt/Makefile.in	2004-08-31 11:23:19.293535840 +0000
@@ -165,7 +165,7 @@
 	  && ./config.status
 
 encrypt.o: $(srcdir)/encrypt.c
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/encrypt.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/encrypt.c
 
 encrypt$(SHLIB_SUFFIX): encrypt.o ../dllinit.o
 	$(SHLIB_LD) encrypt.o ../dllinit.o ${SHLIB_CFLAGS} -o encrypt$(SHLIB_SUFFIX)
diff -uNr BitchX.orig/dll/europa/Makefile.in BitchX/dll/europa/Makefile.in
--- BitchX.orig/dll/europa/Makefile.in	2004-08-31 11:20:43.137275200 +0000
+++ BitchX/dll/europa/Makefile.in	2004-08-31 11:23:19.311533104 +0000
@@ -170,7 +170,7 @@
 	  && ./config.status
 
 europa.o: $(srcdir)/europa.c
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/europa.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/europa.c
 
 europa$(SHLIB_SUFFIX): europa.o ../dllinit.o
 	$(SHLIB_LD) europa.o ../dllinit.o $(SHLIB_CFLAGS) -o europa$(SHLIB_SUFFIX) $(MYSQL_LIBS)
diff -uNr BitchX.orig/dll/fserv/Makefile.in BitchX/dll/fserv/Makefile.in
--- BitchX.orig/dll/fserv/Makefile.in	2004-08-31 11:20:43.236260152 +0000
+++ BitchX/dll/fserv/Makefile.in	2004-08-31 11:23:19.329530368 +0000
@@ -165,7 +165,7 @@
 	  && ./config.status
 
 fserv.o: $(srcdir)/fserv.c
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/fserv.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/fserv.c
 
 fserv$(SHLIB_SUFFIX): fserv.o ../dllinit.o
 	$(SHLIB_LD) fserv.o ../dllinit.o $(SHLIB_CFLAGS) -o fserv$(SHLIB_SUFFIX)
diff -uNr BitchX.orig/dll/hint/Makefile.in BitchX/dll/hint/Makefile.in
--- BitchX.orig/dll/hint/Makefile.in	2004-08-31 11:20:43.243259088 +0000
+++ BitchX/dll/hint/Makefile.in	2004-08-31 11:23:19.346527784 +0000
@@ -165,7 +165,7 @@
 	  && ./config.status
 
 hint.o: $(srcdir)/hint.c
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/hint.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/hint.c
 
 hint$(SHLIB_SUFFIX): hint.o ../dllinit.o
 	$(SHLIB_LD) hint.o ../dllinit.o $(SHLIB_CFLAGS) -o hint$(SHLIB_SUFFIX)
diff -uNr BitchX.orig/dll/identd/Makefile.in BitchX/dll/identd/Makefile.in
--- BitchX.orig/dll/identd/Makefile.in	2004-08-31 11:20:43.248258328 +0000
+++ BitchX/dll/identd/Makefile.in	2004-08-31 11:23:19.363525200 +0000
@@ -165,7 +165,7 @@
 	  && ./config.status
 
 identd.o: $(srcdir)/identd.c
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/identd.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/identd.c
 
 identd$(SHLIB_SUFFIX): identd.o ../dllinit.o
 	$(SHLIB_LD) identd.o ../dllinit.o $(SHLIB_CFLAGS) -o identd$(SHLIB_SUFFIX)
diff -uNr BitchX.orig/dll/nap/Makefile.in BitchX/dll/nap/Makefile.in
--- BitchX.orig/dll/nap/Makefile.in	2004-08-31 11:20:43.270254984 +0000
+++ BitchX/dll/nap/Makefile.in	2004-08-31 11:23:19.381522464 +0000
@@ -167,26 +167,26 @@
 	  && ./config.status
 
 $(topdir)/source/compat.o: $(top_srcdir)/source/compat.c
-	$(CC) $(DEFS) $(CFLAGS) \
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) \
 		-c $(top_srcdir)/source/compat.c -o $(topdir)/source/compat.o
 
 nap.o: $(srcdir)/nap.c
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/nap.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/nap.c
 
 napsend.o: $(srcdir)/napsend.c
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/napsend.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/napsend.c
 
 nap_file.o: $(srcdir)/nap_file.c
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/nap_file.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/nap_file.c
 
 napfunc.o: $(srcdir)/napfunc.c
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/napfunc.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/napfunc.c
 
 napother.o: $(srcdir)/napother.c
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/napother.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/napother.c
 
 md5.o: $(srcdir)/md5.c $(srcdir)/md5.h
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/md5.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/md5.c
 
 nap$(SHLIB_SUFFIX): $(OBJS) 
 	$(SHLIB_LD) $(OBJS) $(SHLIB_CFLAGS) -o nap$(SHLIB_SUFFIX)
diff -uNr BitchX.orig/dll/nicklist/Makefile.in BitchX/dll/nicklist/Makefile.in
--- BitchX.orig/dll/nicklist/Makefile.in	2004-08-31 11:20:43.297250880 +0000
+++ BitchX/dll/nicklist/Makefile.in	2004-08-31 11:23:19.399519728 +0000
@@ -168,7 +168,7 @@
 	  && ./config.status
 
 nicklist.o: $(srcdir)/nicklist.c
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/nicklist.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/nicklist.c
 
 nicklist$(SHLIB_SUFFIX): nicklist.o ../dllinit.o
 	$(SHLIB_LD) nicklist.o ../dllinit.o $(SHLIB_CFLAGS) -o nicklist$(SHLIB_SUFFIX)
diff -uNr BitchX.orig/dll/pkga/Makefile.in BitchX/dll/pkga/Makefile.in
--- BitchX.orig/dll/pkga/Makefile.in	2004-08-31 11:20:43.300250424 +0000
+++ BitchX/dll/pkga/Makefile.in	2004-08-31 11:23:19.416517144 +0000
@@ -165,7 +165,7 @@
 	  && ./config.status
 
 pkga.o: $(srcdir)/pkga.c
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/pkga.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/pkga.c
 
 pkga$(SHLIB_SUFFIX): pkga.o ../dllinit.o
 	$(SHLIB_LD) pkga.o ../dllinit.o $(SHLIB_CFLAGS) -o pkga$(SHLIB_SUFFIX)
diff -uNr BitchX.orig/dll/possum/Makefile.in BitchX/dll/possum/Makefile.in
--- BitchX.orig/dll/possum/Makefile.in	2004-08-31 11:20:43.304249816 +0000
+++ BitchX/dll/possum/Makefile.in	2004-08-31 11:23:19.433514560 +0000
@@ -165,13 +165,13 @@
 	  && ./config.status
 
 possum.o: $(srcdir)/possum.c
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/possum.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/possum.c
 
 head.o: $(srcdir)/head.c
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/head.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/head.c
 
 llist.o: $(srcdir)/llist.c
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/llist.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/llist.c
 
 possum$(SHLIB_SUFFIX): possum.o head.o llist.o ../dllinit.o
 	$(SHLIB_LD) possum.o llist.o head.o ../dllinit.o $(SHLIB_CFLAGS) -o possum$(SHLIB_SUFFIX)
diff -uNr BitchX.orig/dll/qbx/Makefile.in BitchX/dll/qbx/Makefile.in
--- BitchX.orig/dll/qbx/Makefile.in	2004-08-31 11:20:43.322247080 +0000
+++ BitchX/dll/qbx/Makefile.in	2004-08-31 11:23:19.451511824 +0000
@@ -164,7 +164,7 @@
 	  && ./config.status
 
 qbx.o: $(srcdir)/qbx.c
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/qbx.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/qbx.c
 
 qbx$(SHLIB_SUFFIX): qbx.o ../dllinit.o
 	$(SHLIB_LD) qbx.o ../dllinit.o $(SHLIB_CFLAGS) -o qbx$(SHLIB_SUFFIX)
diff -uNr BitchX.orig/dll/qmail/Makefile.in BitchX/dll/qmail/Makefile.in
--- BitchX.orig/dll/qmail/Makefile.in	2004-08-31 11:20:43.339244496 +0000
+++ BitchX/dll/qmail/Makefile.in	2004-08-31 11:23:19.468509240 +0000
@@ -165,7 +165,7 @@
 	  && ./config.status
 
 qmail.o: $(srcdir)/qmail.c
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/qmail.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/qmail.c
 
 qmail$(SHLIB_SUFFIX): qmail.o ../dllinit.o
 	$(SHLIB_LD) qmail.o ../dllinit.o $(SHLIB_CFLAGS) -o qmail$(SHLIB_SUFFIX)
diff -uNr BitchX.orig/dll/scan/Makefile.in BitchX/dll/scan/Makefile.in
--- BitchX.orig/dll/scan/Makefile.in	2004-08-31 11:20:43.344243736 +0000
+++ BitchX/dll/scan/Makefile.in	2004-08-31 11:23:19.486506504 +0000
@@ -165,7 +165,7 @@
 	  && ./config.status
 
 scan.o: $(srcdir)/scan.c
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/scan.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/scan.c
 
 scan$(SHLIB_SUFFIX): scan.o ../dllinit.o
 	$(SHLIB_LD) scan.o ../dllinit.o $(SHLIB_CFLAGS) -o scan$(SHLIB_SUFFIX)
diff -uNr BitchX.orig/dll/wavplay/Makefile.in BitchX/dll/wavplay/Makefile.in
--- BitchX.orig/dll/wavplay/Makefile.in	2004-08-31 11:20:43.350242824 +0000
+++ BitchX/dll/wavplay/Makefile.in	2004-08-31 11:23:19.504503768 +0000
@@ -165,7 +165,7 @@
 	  && ./config.status
 
 wavplay.o: $(srcdir)/wavplay.c
-	$(CC) $(DEFS) $(CFLAGS) -c $(srcdir)/wavplay.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -c $(srcdir)/wavplay.c
 
 wavplay$(SHLIB_SUFFIX): wavplay.o ../dllinit.o
 	$(SHLIB_LD) wavplay.o ../dllinit.o $(SHLIB_CFLAGS) -o wavplay$(SHLIB_SUFFIX)
diff -uNr BitchX.orig/dll/xmms/Makefile.in BitchX/dll/xmms/Makefile.in
--- BitchX.orig/dll/xmms/Makefile.in	2004-08-31 11:20:43.368240088 +0000
+++ BitchX/dll/xmms/Makefile.in	2004-08-31 11:23:19.521501184 +0000
@@ -167,7 +167,7 @@
 	  && ./config.status
 
 xmms.o: $(srcdir)/xmms.c
-	$(CC) $(DEFS) $(CFLAGS) `xmms-config --cflags` -c $(srcdir)/xmms.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) `xmms-config --cflags` -c $(srcdir)/xmms.c
 
 xmms$(SHLIB_SUFFIX): xmms.o ../dllinit.o
 	$(SHLIB_LD) xmms.o ../dllinit.o $(SHLIB_CFLAGS) -o xmms$(SHLIB_SUFFIX) `xmms-config --libs`
diff -uNr BitchX.orig/source/Makefile.in BitchX/source/Makefile.in
--- BitchX.orig/source/Makefile.in	2004-08-31 11:20:43.940153144 +0000
+++ BitchX/source/Makefile.in	2004-08-31 11:24:17.686658736 +0000
@@ -259,6 +259,9 @@
 wterm.o: wterm.c
 	$(CC) $(DEFS) $(CFLAGS) -DWTERM_C -c ./wterm.c
 
+compat.o: compat.c
+	$(CC) $(DEFS) $(CFLAGS) $(SHLIB_CFLAGS) -DWTERM_C -c ./compat.c
+
 wnetwork.c: $(srcdir)/network.c
 	$(RM) wnetwork.c
 	$(LN) $(srcdir)/network.c wnetwork.c
