diff -Naur discomatic-0.3.orig/src/folderutils.cpp discomatic-0.3/src/folderutils.cpp
--- discomatic-0.3.orig/src/folderutils.cpp	2003-10-17 22:54:37.000000000 +0200
+++ discomatic-0.3/src/folderutils.cpp	2008-12-15 19:15:55.000000000 +0100
@@ -6,6 +6,7 @@
 #include "project.h"
 #include "math.h"
 #include <sstream>
+#include <cstring>
 
 using namespace std;
 
diff -Naur discomatic-0.3.orig/src/process.cpp discomatic-0.3/src/process.cpp
--- discomatic-0.3.orig/src/process.cpp	2003-10-17 22:54:37.000000000 +0200
+++ discomatic-0.3/src/process.cpp	2008-12-15 19:18:33.000000000 +0100
@@ -1,5 +1,6 @@
 #include "process.h"
 #include <fstream>
+#include <cstring>
 
 gboolean read_input(GIOChannel *source, GIOCondition condition, gpointer data){
   ((Process*)data)->processInput(source);

