diff -ur truevision-0.5.5.orig/include/objparam.h truevision-0.5.5/include/objparam.h
--- truevision-0.5.5.orig/include/objparam.h	2008-12-21 16:57:25.000000000 +0100
+++ truevision-0.5.5/include/objparam.h	2008-12-21 16:58:49.000000000 +0100
@@ -465,7 +465,7 @@
 			TvWidget_entry( name, sname, tooltip, appref, val) {}
 		ObjParam_entry( ObjParam_entry & ref ) : ObjParam( ref ), TvWidget_entry( ref ) {}
 		void get_widget( GtkWidget *box, bool tt );
-		void get_widget( GtkWidget *row, bool tt, int row );
+		void get_widget( GtkWidget *tab, bool tt, int row );
 		void flush();
 		void swap_data( ObjParam *param );
 };
