--- Makefile.in	2005-10-07 15:03:22.000000000 +0100
+++ Makefile.in.new	2005-10-11 18:56:45.000000000 +0100
@@ -835,7 +835,10 @@
 	-export GNOME_MAGIC=/etc/gnome-vfs-mime-magic ; \
 	if [ -f $$GNOME_MAGIC ]; then \
 	if ! grep "TRUEVISION" $$GNOME_MAGIC; then \
-	echo -e "0\tstring\t\tTRUEVISION SCENE\t\t\tapplication/x-truevision" >> $$GNOME_MAGIC; \
+	cp /etc/gnome-vfs-mime-magic .;	\
+	echo -e "0\tstring\t\tTRUEVISION SCENE\t\t\tapplication/x-truevision" >> gnome-vfs-mime-magic;	\
+	mkdir $(DESTDIR)/etc; \
+	cp gnome-vfs-mime-magic $(DESTDIR)/etc/gnome-vfs-mime-magic; \
 	fi \
 	fi ; \
 	export DEFAULTS=$(prefix)/share/applications/defaults.list ; \
@@ -844,7 +847,6 @@
 	echo -e "application/x-truevision=truevision.desktop" >> $$DEFAULTS; \
 	fi \
 	fi ; \
-	update-mime-database "$(prefix)/share/mime"
 
 #	$(truevisionmaterials_DATA)
 #	$(truevisionpython_DATA)
