From 3ef38bdfe771bfff0247c668c044405c0f1b4145 Mon Sep 17 00:00:00 2001
From: Alexander Golubev <fatzer2@gmail.com>
Date: Mon, 3 Nov 2014 03:21:44 +0300
Subject: [PATCH 1/2] fix build with cmake-2.8.12 introduced changes in qt4
 macroses

---
 core/cmake/QutIMMacros.cmake | 90 +++++++++++++++++++++++++-------------------
 1 file changed, 52 insertions(+), 38 deletions(-)

diff --git a/core/cmake/QutIMMacros.cmake b/core/cmake/QutIMMacros.cmake
index 588f464..41e961d 100644
--- a/core/cmake/QutIMMacros.cmake
+++ b/core/cmake/QutIMMacros.cmake
@@ -31,43 +31,57 @@ if(NOT CPACK_INCLUDED)
 endif()
 
 #TODO separate to external macro
-MACRO(QUTIM_WRAP_CPP outfiles)
-	# get include dirs
-	QT4_GET_MOC_FLAGS(moc_flags)
-	QT4_EXTRACT_OPTIONS(moc_files moc_options ${ARGN})
-
-	FOREACH (it ${moc_files})
-		GET_FILENAME_COMPONENT(_abs_FILE ${it} ABSOLUTE)
-		GET_FILENAME_COMPONENT(_abs_PATH ${_abs_FILE} PATH)
-		GET_FILENAME_COMPONENT(_basename ${it} NAME_WE)
-
-		SET(_HAS_MOC false)
-
-		IF(EXISTS ${_abs_PATH}/${_basename}.cpp)
-			SET(_header ${_abs_PATH}/${_basename}.cpp)
-			FILE(READ ${_header} _contents)
-			STRING(REGEX MATCHALL "# *include +[^ ]+\\.moc[\">]" _match "${_contents}")
-			IF(_match)
-				SET(_HAS_MOC true)
-				FOREACH (_current_MOC_INC ${_match})
-					STRING(REGEX MATCH "[^ <\"]+\\.moc" _current_MOC "${_current_MOC_INC}")
-					SET(_moc    ${CMAKE_CURRENT_BINARY_DIR}/${_current_MOC})
-					QT4_CREATE_MOC_COMMAND(${_abs_FILE} ${_moc} "${_moc_INCS}" "")
-					MACRO_ADD_FILE_DEPENDENCIES(${_abs_FILE} ${_moc})
-				ENDFOREACH (_current_MOC_INC)
-			ENDIF()
-		ENDIF()
-		IF(NOT _HAS_MOC)
-			FILE(READ ${_abs_FILE} _contents)
-                        STRING(REGEX MATCHALL "Q_OBJECT|Q_GADGET" _match2 "${_contents}")
-			IF(_match2)
-				QT4_MAKE_OUTPUT_FILE(${_abs_FILE} moc_ cxx outfile)
-				QT4_CREATE_MOC_COMMAND(${_abs_FILE} ${outfile} "${moc_flags}" "${moc_options}")
-				SET(${outfiles} ${${outfiles}} ${outfile})
-			ENDIF()
-		ENDIF()
-	ENDFOREACH(it)
-ENDMACRO (QUTIM_WRAP_CPP)
+macro(QUTIM_WRAP_CPP outfiles)
+	if(NOT CMAKE_AUTOMOC)
+		# get include dirs
+		qt4_get_moc_flags(moc_flags)
+		if(CMAKE_VERSION VERSION_LESS "2.8.12")
+			qt4_extract_options(moc_files moc_options ${ARGN})
+		else()
+			qt4_extract_options(moc_files moc_options moc_target ${ARGN})
+		endif()
+
+		foreach(it ${moc_files})
+			get_filename_component(_abs_file ${it} ABSOLUTE)
+			get_filename_component(_abs_PATH ${_abs_file} PATH)
+			get_filename_component(_basename ${it} NAME_WE)
+
+			set(_HAS_MOC false)
+
+			if(EXISTS ${_abs_PATH}/${_basename}.cpp)
+				set(_header ${_abs_PATH}/${_basename}.cpp)
+				file(READ ${_header} _contents)
+				string(REGEX MATCHALL "# *include +[^ ]+\\.moc[\">]" _match "${_contents}")
+				if(_match)
+					set(_HAS_MOC true)
+					foreach(_current_MOC_INC ${_match})
+						string(REGEX MATCH "[^ <\"]+\\.moc" _current_MOC "${_current_MOC_INC}")
+						set(_moc    ${CMAKE_CURRENT_BINARY_DIR}/${_current_MOC})
+						if(CMAKE_VERSION VERSION_LESS "2.8.12")
+							qt4_create_moc_command(${_abs_file} ${_moc} "${_moc_INCS}" "")
+						else()
+							qt4_create_moc_command(${_abs_file} ${_moc} "${_moc_INCS}" "${moc_options}" "${moc_target}")
+						endif()
+						macro_add_file_dependencies(${_abs_file} ${_moc})
+					endforeach (_current_MOC_INC)
+				endif()
+			endif()
+			if(NOT _HAS_MOC)
+				file(READ ${_abs_file} _contents)
+				string(REGEX MATCHALL "Q_OBJECT|Q_GADGET" _match2 "${_contents}")
+				if(_match2)
+					qt4_make_output_file(${_abs_file} moc_ cxx outfile)
+					if(CMAKE_VERSION VERSION_LESS "2.8.12")
+						qt4_create_moc_command(${_abs_file} ${outfile} "${moc_flags}" "${moc_options}")
+					else()
+						qt4_create_moc_command(${_abs_file} ${outfile} "${moc_flags}" "${moc_options}" "${moc_target}")
+					endif()
+					set(${outfiles} ${${outfiles}} ${outfile})
+				endif()
+			endif()
+		endforeach(it)
+	endif()
+endmacro(QUTIM_WRAP_CPP)
 
 macro(QUTIM_ADD_ARTWORK_DIR art_dir)
 	FILE(GLOB ARTWORKS ${art_dir} "${art_dir}/*")
@@ -331,7 +345,7 @@ QUTIM_EXPORT_PLUGIN(${plugin_name}Plugin)
 #    set_target_properties( ${plugin_name} PROPERTIES COMPILE_FLAGS "-D${plugin_name}_MAKE" )
 	if( QUTIM_${plugin_name}_STATIC )
 	if( NOT EXISTS "${CMAKE_CURRENT_BINARY_DIR}/${plugin_name}helper.cpp" )
-		file( WRITE "${CMAKE_CURRENT_BINARY_DIR}/${plugin_name}helper.cpp"
+		file( WRITE "${CMAKE_CURRENT_BINARY_DIR}/${plugin_name}helper.cpp" 
 "#include <QtCore/QtPlugin>
 
 Q_IMPORT_PLUGIN(${plugin_name})
-- 
2.0.4

