Index: src/confuse.c
===================================================================
RCS file: /sources/confuse/confuse/src/confuse.c,v
retrieving revision 1.36
diff -u -B -r1.36 confuse.c
--- src/confuse.c	13 Oct 2007 20:01:45 -0000	1.36
+++ src/confuse.c	1 Sep 2008 11:57:30 -0000
@@ -41,6 +41,7 @@
 #define is_set(f, x) (((f) & (x)) == (f))
 
 #if defined(ENABLE_NLS) && defined(HAVE_GETTEXT)
+# include <locale.h>
 # include <libintl.h>
 # define _(str) dgettext(PACKAGE, str)
 #else
