diff -uNr gazpacho-0.7.0.orig/setup.py gazpacho-0.7.0/setup.py
--- gazpacho-0.7.0.orig/setup.py	2006-12-29 12:25:02.000000000 +0000
+++ gazpacho-0.7.0/setup.py	2006-12-29 12:26:12.000000000 +0000
@@ -8,9 +8,9 @@
 
 templates = []
 data_files = [
-    ('share/doc/gazpacho', ('AUTHORS', 'ChangeLog', 'CONTRIBUTORS',
+    ('share/doc/EBUILD_PF_HERE', ('AUTHORS', 'ChangeLog', 'CONTRIBUTORS',
                             'COPYING', 'README', 'NEWS')),
-    ('share/doc/gazpacho/examples', listfiles('examples', '*')),
+    ('share/doc/EBUILD_PF_HERE/examples', listfiles('examples', '*')),
     ('$datadir/catalogs', listfiles('catalogs', 'base.xml')),
     ('$datadir/pixmaps', listfiles('pixmaps', '*.png')),
     ('$datadir/pixmaps', listfiles('pixmaps', '*.ico')),
@@ -34,7 +34,7 @@
 
 resources = dict(locale='$prefix/share/locale')
 global_resources = dict(
-    doc='$prefix/share/doc/gazpacho',
+    doc='$prefix/share/doc/EBUILD_PF_HERE',
     pixmap='$datadir/pixmaps',
     glade='$datadir/glade',
     resource='$datadir/resources',
