--- bin/ktoolbar	2008-02-08 19:49:45.000000000 +0100
+++ bin/ktoolbar	2008-02-08 20:19:57.000000000 +0100
@@ -1,6 +1,10 @@
 #!/bin/sh
 
-javapathtowtk=`java-config --jdk-home`"/bin/"
+if [ "`uname -m`" = "x86_64" ]; then
+	javapathtowtk=`java-config --select-vm=emul-linux-x86-java --jdk-home`"/bin/"
+else
+	javapathtowtk=`java-config --jdk-home`"/bin/"
+fi
 
 PRG=$0
 
--- bin/prefs		2008-02-08 19:49:45.000000000 +0100
+++ bin/prefs		2008-02-08 20:19:57.000000000 +0100
@@ -1,6 +1,10 @@
 #!/bin/sh
 
-javapathtowtk=`java-config --jdk-home`"/bin/"
+if [ "`uname -m`" = "x86_64" ]; then
+	javapathtowtk=`java-config --select-vm=emul-linux-x86-java --jdk-home`"/bin/"
+else
+	javapathtowtk=`java-config --jdk-home`"/bin/"
+fi
 
 PRG=$0

--- bin/utils	2008-02-08 19:49:45.000000000 +0100
+++ bin/utils	2008-02-08 20:19:57.000000000 +0100
@@ -1,6 +1,10 @@
 #!/bin/sh
 
-javapathtowtk=`java-config --jdk-home`"/bin/"
+if [ "`uname -m`" = "x86_64" ]; then
+	javapathtowtk=`java-config --select-vm=emul-linux-x86-java --jdk-home`"/bin/"
+else
+	javapathtowtk=`java-config --jdk-home`"/bin/"
+fi
 
 PRG=$0

--- bin/wscompile	2008-02-08 19:49:45.000000000 +0100
+++ bin/wscompile	2008-02-08 20:19:57.000000000 +0100
@@ -1,6 +1,10 @@
 #!/bin/sh
 
-javapathtowtk=`java-config --jdk-home`"/bin/"
+if [ "`uname -m`" = "x86_64" ]; then
+	javapathtowtk=`java-config --select-vm=emul-linux-x86-java --jdk-home`"/bin/"
+else
+	javapathtowtk=`java-config --jdk-home`"/bin/"
+fi
 
 PRG=$0
 

--- bin/mekeytool	2008-02-08 19:49:45.000000000 +0100
+++ bin/mekeytool	2008-02-08 20:19:57.000000000 +0100
@@ -1,6 +1,10 @@
 #!/bin/sh
 
-javapathtowtk=`java-config --jdk-home`"/bin/"
+if [ "`uname -m`" = "x86_64" ]; then
+	javapathtowtk=`java-config --select-vm=emul-linux-x86-java --jdk-home`"/bin/"
+else
+	javapathtowtk=`java-config --jdk-home`"/bin/"
+fi
 
 PRG=$0
 
--- bin/emulator	2008-02-08 19:49:45.000000000 +0100
+++ bin/emulator	2008-02-08 20:19:57.000000000 +0100
@@ -1,6 +1,10 @@
 #!/bin/sh
 
-javapathtowtk=`java-config --jdk-home`"/bin/"
+if [ "`uname -m`" = "x86_64" ]; then
+	javapathtowtk=`java-config --select-vm=emul-linux-x86-java --jdk-home`"/bin/"
+else
+	javapathtowtk=`java-config --jdk-home`"/bin/"
+fi
 
 PRG=$0
 

--- bin/defaultdevice		2008-02-08 19:49:45.000000000 +0100
+++ bin/defaultdevice		2008-02-08 20:19:57.000000000 +0100
@@ -1,6 +1,10 @@
 #!/bin/sh
 
-javapathtowtk=`java-config --jdk-home`"/bin/"
+if [ "`uname -m`" = "x86_64" ]; then
+	javapathtowtk=`java-config --select-vm=emul-linux-x86-java --jdk-home`"/bin/"
+else
+	javapathtowtk=`java-config --jdk-home`"/bin/"
+fi
 
 PRG=$0
 

