diff -Naur gmime-2.2.23-pristine/gmime/gmime-utils.h gmime-2.2.23/gmime/gmime-utils.h
--- gmime-2.2.23-pristine/gmime/gmime-utils.h	2008-06-12 06:29:28.000000000 -0600
+++ gmime-2.2.23/gmime/gmime-utils.h	2010-02-03 19:01:18.145483975 -0700
@@ -106,7 +106,7 @@
  * Returns the number of output bytes needed to uuencode an input
  * buffer of size @x.
  **/
-#define GMIME_UUENCODE_LEN(x)      ((size_t) (((((x) + 2) / 45) * 62) + 62))
+#define GMIME_UUENCODE_LEN(x)      ((size_t) (((((x) + 2) / 45) * 62) + 64))
 
 
 /**
