diff -NrU5 gmime-2.2.23.orig/mono/AssemblyInfo.cs.in gmime-2.2.23/mono/AssemblyInfo.cs.in
--- gmime-2.2.23.orig/mono/AssemblyInfo.cs.in	2009-01-25 19:01:15.000000000 +0100
+++ gmime-2.2.23/mono/AssemblyInfo.cs.in	2009-01-25 19:02:08.000000000 +0100
@@ -1,6 +1,4 @@
 using System.Reflection;
 using System.Runtime.CompilerServices;
 
 [assembly:AssemblyVersion("@API_VERSION@")]
-[assembly:AssemblyDelaySign(true)]
-[assembly:AssemblyKeyFile("@srcdir@/gmime-sharp.pub")]
diff -NrU5 gmime-2.2.23.orig/mono/Makefile.am gmime-2.2.23/mono/Makefile.am
--- gmime-2.2.23.orig/mono/Makefile.am	2009-01-25 19:01:35.000000000 +0100
+++ gmime-2.2.23/mono/Makefile.am	2009-01-25 18:57:11.000000000 +0100
@@ -72,11 +72,11 @@
 	rm -f generated/* &&	\
 	$(GAPI_CODEGEN) --generate $(API) --outdir=generated --customdir=$(srcdir) --assembly-name=$(ASSEMBLY_NAME) 	\
 	&& touch generated-stamp
 
 $(ASSEMBLY): $(build_sources) generated-stamp $(ASSEMBLY_NAME).pub
-	$(CSC) --unsafe /out:$(ASSEMBLY) /target:library $(references) $(build_sources) generated/*.cs
+	$(CSC) --unsafe /out:$(ASSEMBLY) -keyfile:$(ASSEMBLY_NAME).snk /target:library $(references) $(build_sources) generated/*.cs
 
 install-data-local:
 	@if test -n '$(TARGET)'; then									\
 	  if test -n '$(DESTDIR)'; then								\
 	    echo "$(GACUTIL) /i $(ASSEMBLY) /f /package $(PACKAGE_SHARP) /root $(DESTDIR)$(prefix)/lib";		\
