From c9f92d53f591ae4e893fc81ccc49ea9d2f4f168c Mon Sep 17 00:00:00 2001
From: Jeffrey Stedfast <fejj@gnome.org>
Date: Mon, 23 Sep 2013 13:46:06 +0000
Subject: Minor code/comment cleanup

---
diff --git a/gmime/gmime-param.c b/gmime/gmime-param.c
index 4dd98d5..667adf7 100644
--- a/gmime/gmime-param.c
+++ b/gmime/gmime-param.c
@@ -396,7 +396,7 @@ rfc2184_param_charset (const char **in, char **langp)
 	if (langp)
 		*langp = NULL;
 	
-	while (*inptr != '\0' && *inptr != '\'')
+	while (*inptr && *inptr != '\'')
 		inptr++;
 	
 	if (*inptr != '\'')
@@ -408,7 +408,7 @@ rfc2184_param_charset (const char **in, char **langp)
 	charset[len] = '\0';
 	
 	lang = ++inptr;
-	while (*inptr != '\0' && *inptr != '\'')
+	while (*inptr && *inptr != '\'')
 		inptr++;
 	
 	if (*inptr == '\'') {
diff --git a/gmime/gmime-utils.c b/gmime/gmime-utils.c
index ada8faf..41d5639 100644
--- a/gmime/gmime-utils.c
+++ b/gmime/gmime-utils.c
@@ -2743,7 +2743,7 @@ header_fold_tokens (const char *field, const char *value, size_t vlen, rfc2047_t
 					g_string_insert_c (output, tab, '\n');
 					len = (lwsp - tab) + 1;
 				} else if (lwsp != 0) {
-					/* break just before the last lwsp character i*/
+					/* break just before the last lwsp character */
 					g_string_insert_c (output, lwsp, '\n');
 					len = 1;
 				} else if (len > 1) {
@@ -2769,7 +2769,7 @@ header_fold_tokens (const char *field, const char *value, size_t vlen, rfc2047_t
 				g_string_insert_c (output, tab, '\n');
 				len = (lwsp - tab) + 1;
 			} else if (lwsp != 0) {
-				/* break just before the last lwsp character i*/
+				/* break just before the last lwsp character */
 				g_string_insert_c (output, lwsp, '\n');
 				len = 1;
 			} else if (len > 1) {
--
cgit v0.9.2
