From b82cfae4954557f03a01f7b8c92cf1d59dea9033 Mon Sep 17 00:00:00 2001
From: Sergei Trofimovich <slyfox@gentoo.org>
Date: Wed, 7 Dec 2011 21:43:07 +0300
Subject: [PATCH] Makefile: fix paralell building (bdelta depends on it's
 shared library)

Signed-off-by: Sergei Trofimovich <slyfox@gentoo.org>
---
 trunk/src/Makefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/trunk/src/Makefile b/trunk/src/Makefile
index 7dd5085..1f4ff3e 100644
--- a/trunk/src/Makefile
+++ b/trunk/src/Makefile
@@ -11,7 +11,7 @@ all: $(ALL_TARGETS)
 libbdelta.so: libbdelta.cpp compatibility.h container.h checksum.h file.h
 	$(CXX) -shared -fPIC $(CXXFLAGS) $(LDFLAGS) $< -o $@
 
-bdelta: bdelta.cpp bdelta.h compatibility.h container.h file.h
+bdelta: bdelta.cpp bdelta.h compatibility.h container.h file.h libbdelta.so
 	$(CXX) $< -o $@ $(CXXFLAGS) $(LDFLAGS) -L. -lbdelta
 
 bpatch: bpatch.cpp compatibility.h bdelta.h file.h
-- 
1.7.8

