From 30073b511397646b9ef06ed657fda157dd8c6209 Mon Sep 17 00:00:00 2001
From: "Jason A. Donenfeld" <Jason@zx2c4.com>
Date: Thu, 13 Feb 2014 14:36:23 +0100
Subject: [PATCH] Link against pthread on all platforms.

Signed-off-by: Jason A. Donenfeld <Jason@zx2c4.com>
---
 src/gui/curses/CMakeLists.txt | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/gui/curses/CMakeLists.txt b/src/gui/curses/CMakeLists.txt
index 325c611..a8927bc 100644
--- a/src/gui/curses/CMakeLists.txt
+++ b/src/gui/curses/CMakeLists.txt
@@ -53,9 +53,7 @@ IF(${CMAKE_SYSTEM_NAME} STREQUAL "FreeBSD")
   ENDIF(HAVE_BACKTRACE)
 ENDIF(${CMAKE_SYSTEM_NAME} STREQUAL "FreeBSD")
 
-IF(${CMAKE_SYSTEM_NAME} STREQUAL "OpenBSD")
-  LIST(APPEND EXTRA_LIBS "pthread")
-ENDIF(${CMAKE_SYSTEM_NAME} STREQUAL "OpenBSD")
+LIST(APPEND EXTRA_LIBS "pthread")
 
 IF(ICONV_LIBRARY)
   LIST(APPEND EXTRA_LIBS ${ICONV_LIBRARY})
-- 
1.8.5.4

