diff --git a/Makefile b/Makefile
index 1a2b5f7..30526b2 100644
--- a/Makefile
+++ b/Makefile
@@ -115,32 +115,32 @@ all: $(PROGS) $(LIBTCC1) $(BCHECK_O) libtcc.a tcc-doc.html tcc.1 libtcc_test$(EX
 
 # Host Tiny C Compiler
 tcc$(EXESUF): $(NATIVE_FILES)
-	$(CC) -o $@ $< $(NATIVE_TARGET) $(CFLAGS) $(LIBS)
+	$(CC) -o $@ $(NATIVE_TARGET) $(CFLAGS) $(LDFLAGS) $< $(LIBS)
 
 # Cross Tiny C Compilers
 i386-tcc$(EXESUF): $(I386_FILES)
-	$(CC) -o $@ $< -DTCC_TARGET_I386 $(CFLAGS) $(LIBS)
+	$(CC) -o $@ -DTCC_TARGET_I386 $(CFLAGS) $(LDFLAGS) $< $(LIBS)
 
 i386-win32-tcc$(EXESUF): $(WIN32_FILES)
-	$(CC) -o $@ $< -DTCC_TARGET_PE $(CFLAGS) $(LIBS)
+	$(CC) -o $@ -DTCC_TARGET_PE $(CFLAGS) $(LDFLAGS) $< $(LIBS)
 
 x86_64-tcc$(EXESUF): $(X86_64_FILES)
-	$(CC) -o $@ $< -DTCC_TARGET_X86_64 $(CFLAGS) $(LIBS)
+	$(CC) -o $@ -DTCC_TARGET_X86_64 $(CFLAGS) $(LDFLAGS) $< $(LIBS)
 
 c67-tcc$(EXESUF): $(C67_FILES)
-	$(CC) -o $@ $< -DTCC_TARGET_C67 $(CFLAGS) $(LIBS)
+	$(CC) -o $@ -DTCC_TARGET_C67 $(CFLAGS) $(LDFLAGS) $< $(LIBS)
 
 arm-tcc-fpa$(EXESUF): $(ARM_FILES)
-	$(CC) -o $@ $< -DTCC_TARGET_ARM $(CFLAGS) $(LIBS)
+	$(CC) -o $@ -DTCC_TARGET_ARM $(CFLAGS) $(LDFLAGS) $< $(LIBS)
 
 arm-tcc-fpa-ld$(EXESUF): $(ARM_FILES)
-	$(CC) -o $@ $< -DTCC_TARGET_ARM -DLDOUBLE_SIZE=12 $(CFLAGS) $(LIBS)
+	$(CC) -o $@ -DTCC_TARGET_ARM -DLDOUBLE_SIZE=12 $(CFLAGS) $(LDFLAGS) $< $(LIBS)
 
 arm-tcc-vfp$(EXESUF): $(ARM_FILES)
-	$(CC) -o $@ $< -DTCC_TARGET_ARM -DTCC_ARM_VFP $(CFLAGS) $(LIBS)
+	$(CC) -o $@ -DTCC_TARGET_ARM -DTCC_ARM_VFP $(CFLAGS) $(LDFLAGS) $< $(LIBS)
 
 arm-tcc-vfp-eabi$(EXESUF): $(ARM_FILES)
-	$(CC) -o $@ $< -DTCC_TARGET_ARM -DTCC_ARM_EABI $(CFLAGS) $(LIBS)
+	$(CC) -o $@ -DTCC_TARGET_ARM -DTCC_ARM_EABI $(CFLAGS) $(LDFLAGS) $< $(LIBS)
 
 # libtcc generation and test
 libtcc.o: $(NATIVE_FILES)
@@ -150,14 +150,14 @@ libtcc.a: libtcc.o
 	$(AR) rcs $@ $^
 
 libtcc_test$(EXESUF): tests/libtcc_test.c libtcc.a
-	$(CC) -o $@ $^ -I. $(CFLAGS) $(LIBS)
+	$(CC) -o $@ -I. $(CFLAGS) $(LDFLAGS) $^ $(LIBS)
 
 libtest: libtcc_test$(EXESUF) $(LIBTCC1)
 	./libtcc_test$(EXESUF) lib_path=.
 
 # profiling version
 tcc_p$(EXESUF): $(NATIVE_FILES)
-	$(CC) -o $@ $< $(NATIVE_TARGET) $(CFLAGS_P) $(LIBS_P)
+	$(CC) -o $@ $(NATIVE_TARGET) $(CFLAGS_P) $(LDFLAGS) $< $(LIBS_P)
 
 # windows utilities
 tiny_impdef$(EXESUF): win32/tools/tiny_impdef.c
@@ -180,16 +180,16 @@ LIBTCC1_OBJS+=alloca86.o alloca86-bt.o
 endif
 
 %.o: %.c
-	$(LIBTCC1_CC) -o $@ -c $< -O2 -Wall
+	$(LIBTCC1_CC) $(CFLAGS) -c $< -o $@
 
 %.o: %.S
-	$(LIBTCC1_CC) -o $@ -c $<
+	$(LIBTCC1_CC) $(ASFLAGS) -c $< -o $@
 
 libtcc1.a: $(LIBTCC1_OBJS)
 	$(AR) rcs $@ $^
 
 bcheck.o: bcheck.c
-	$(CC) -o $@ -c $< -O2 -Wall
+	$(CC) $(CFLAGS) -c $< -o $@
 
 # install
 TCC_INCLUDES = stdarg.h stddef.h stdbool.h float.h varargs.h tcclib.h
