From becadab262724fa9b6652581bfa121b9eb738a21 Mon Sep 17 00:00:00 2001
From: Rok Mandeljc <rok.mandeljc@gmail.com>
Date: Sun, 22 Jan 2012 18:25:54 +0100
Subject: [PATCH] libMirage: Session: initialize pointer

---
 libmirage/src/libmirage/mirage-session.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/libmirage/src/libmirage/mirage-session.c b/libmirage/src/libmirage/mirage-session.c
index 7eb6ba9..1c42e9e 100644
--- a/libmirage/src/libmirage/mirage-session.c
+++ b/libmirage/src/libmirage/mirage-session.c
@@ -1404,7 +1404,7 @@ gboolean mirage_session_for_each_language (MIRAGE_Session *self, MIRAGE_Callback
 static gboolean set_cdtext_data (gint langcode, gint type, gint track, guint8 *data, gint len, MIRAGE_Session *self)
 {
     gboolean succeeded;
-    GObject *language;
+    GObject *language = NULL;
     
     if (track == 0) {
         /* Session */
-- 
1.7.8.4

