--- CMakeLists.txt
+++ CMakeLists.txt
@@ -6,7 +6,7 @@
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 set(CMAKE_AUTOMOC ON)
 
-add_definitions(-Wall -march=native -std=c++11)
+add_definitions(-Wall -std=c++11)
 
 if(NOT CMAKE_BUILD_TYPE)
         set(CMAKE_BUILD_TYPE Release)
@@ -19,7 +19,6 @@
 else()
         message(STATUS "Release build")
         add_definitions(-DNDEBUG)
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O2")
 endif()
 
 # PKG-CONFIG
@@ -52,26 +51,26 @@
 
 add_custom_target(uninstall COMMAND ${CMAKE_COMMAND} -P ${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake)
 
+if(USE_systemd)
 # login1
-pkg_check_modules(LOGIN1 "libsystemd-login")
+pkg_check_modules(LOGIN1 REQUIRED "libsystemd-login")
 
-if(LOGIN1_FOUND)
-    # do nothing
 else()
     set(LOGIN1_FOUND 0)
 endif()
 
+if(USE_upower)
 # upower
-pkg_check_modules(UPOWER "upower-glib")
+pkg_check_modules(UPOWER REQUIRED "upower-glib")
 
-if(UPOWER_FOUND)
-    # do nothing
 else()
     set(UPOWER_FOUND 0)
 endif()
 
+if(USE_systemd)
 # systemd
-pkg_check_modules(SYSTEMD "systemd")
+pkg_check_modules(SYSTEMD REQUIRED "systemd")
+endif()
 
 if(SYSTEMD_FOUND)
     execute_process(COMMAND ${PKG_CONFIG_EXECUTABLE} --variable=systemdsystemunitdir systemd OUTPUT_VARIABLE SYSTEMD_SYSTEM_UNIT_DIR)
