--- data/scripts/Xsession
+++ data/scripts/Xsession
@@ -74,13 +74,13 @@
     exec xterm -geometry 80x24-0-0
     ;;
   custom)
-    exec $HOME/.xsession
+    exec ck-launch-session $HOME/.xsession
     ;;
   default)
-    exec /usr/bin/startkde
+    exec ck-launch-session /usr/bin/startkde
     ;;
   *)
-    eval exec "$session"
+    eval ck-launch-session "$session"
     ;;
 esac
 exec xmessage -center -buttons OK:0 -default OK "Sorry, cannot execute $session. Check $DESKTOP_SESSION.desktop."
