http://bugs.gentoo.org/show_bug.cgi?id=278077

--- logitech_applet.c
+++ logitech_applet.c
@@ -65,6 +65,7 @@
     { VENDOR_LOGITECH, 0xC024, "M-BP82", "MX300 Optical Mouse", HAS_RES },
     { VENDOR_LOGITECH, 0xC025, "M-BP81A", "MX500 Optical Mouse", HAS_RES | HAS_SS | HAS_SSR },
     { VENDOR_LOGITECH, 0xC031, "M-UT58A", "iFeel Mouse (silver)", HAS_RES },
+    { VENDOR_LOGITECH, 0xC051, "M-BS81A", "MX518 Optical Mouse", HAS_RES },
     { VENDOR_LOGITECH, 0xC501, "C-BA4-MSE", "Mouse Receiver", HAS_CSR },
     { VENDOR_LOGITECH, 0xC502, "C-UA3-DUAL", "Dual Receiver", HAS_CSR | USE_CH2},
     { VENDOR_LOGITECH, 0xC504, "C-BD9-DUAL", "Cordless Freedom Optical", HAS_CSR | USE_CH2 },
@@ -182,7 +183,7 @@
  *  - 0x03 for 400cpi
  *  - 0x04 for 800cpi
  *  - 0x05 for 1200cpi
- *  - 0x06 for 1600cpi
+ *  - 0x06 for 1600cpi (actually 1800cpi on newer MX518)
  */
 int set_resolution(struct usb_device *dev, int resolution)
 {
