From: Julian Ospald <hasufell@gentoo.org>
Date: Thu Nov  1 21:16:18 UTC 2012
Subject: build system

respect flags

--- CMakeLists.txt
+++ CMakeLists.txt
@@ -158,12 +158,8 @@
 
 
 #set default flags values for all projects
-set(CMAKE_C_FLAGS "-pipe ${CMAKE_C_FLAGS}")
-set(CMAKE_C_FLAGS_RELEASE "-w -Os -fomit-frame-pointer ${CMAKE_C_FLAGS_RELEASE}")
-set(CMAKE_C_FLAGS_DEBUG "-Wall -O0 -g -DDEBUG ${CMAKE_C_FLAGS_DEBUG}")
-set(CMAKE_CXX_FLAGS ${CMAKE_C_FLAGS})
-set(CMAKE_CXX_FLAGS_RELEASE ${CMAKE_C_FLAGS_RELEASE})
-set(CMAKE_CXX_FLAGS_DEBUG ${CMAKE_C_FLAGS_DEBUG})
+set(CMAKE_C_FLAGS_DEBUG "-Wall -DDEBUG ${CMAKE_C_FLAGS_DEBUG}")
+set(CMAKE_CXX_FLAGS_DEBUG "-Wall -DDEBUG ${CMAKE_CXX_FLAGS_DEBUG}")
 
 #parse additional parameters
 if(FPFLAGS OR GHFLAGS)
