From: Julian Ospald <hasufell@gentoo.org>
Date: Thu Nov  1 21:52:40 UTC 2012
Subject: build system

fix parallel make

--- hedgewars/CMakeLists.txt
+++ hedgewars/CMakeLists.txt
@@ -193,7 +193,7 @@
 
 #this command is a workaround to some inlining issues present in older FreePascal versions and fixed in 2.6
 if(fpc_version LESS "020600")
-    add_custom_target(ENGINECLEAN COMMAND ${CMAKE_BUILD_TOOL} "clean" "${PROJECT_BINARY_DIR}" "${hedgewars_SOURCE_DIR}/hedgewars")
+	add_custom_target(ENGINECLEAN COMMAND $(MAKE) "clean" "${PROJECT_BINARY_DIR}" "${hedgewars_SOURCE_DIR}/hedgewars")
 endif()
 
 
@@ -265,4 +265,6 @@
     add_dependencies(${engine_output_name} ENGINECLEAN)
 endif()
 
+add_dependencies(${engine_output_name} avwrapper)
+
 install(PROGRAMS "${EXECUTABLE_OUTPUT_PATH}/${engine_output_name}${CMAKE_EXECUTABLE_SUFFIX}" DESTINATION ${target_dir})
