diff -r 164690ff1408 -r 4114ce5d885d share/hedgewars/Data/Maps/Basketball/map.lua
--- a/share/hedgewars/Data/Maps/Basketball/map.lua	Fri May 31 16:49:11 2013 +0200
+++ b/share/hedgewars/Data/Maps/Basketball/map.lua	Fri May 31 14:47:07 2013 -0400
@@ -1,6 +1,6 @@
 -- Hedgewars - Basketball for 2+ Players
 
-HedgewarsScriptLoad("Scripts/Locale.lua")()
+HedgewarsScriptLoad("/Scripts/Locale.lua")
 
 local score = {[0] = 0, [1] = 0, [2] = 0, [3] = 0, [4] = 0, [5] = 0}
 
diff -r 164690ff1408 -r 4114ce5d885d share/hedgewars/Data/Maps/CTF_Blizzard/map.lua
--- a/share/hedgewars/Data/Maps/CTF_Blizzard/map.lua	Fri May 31 16:49:11 2013 +0200
+++ b/share/hedgewars/Data/Maps/CTF_Blizzard/map.lua	Fri May 31 14:47:07 2013 -0400
@@ -91,7 +91,7 @@
 -- add support for more players
 -- re-enable sudden death, but set water rise to 0
 
-HedgewarsScriptLoad("Scripts/Locale.lua")()
+HedgewarsScriptLoad("/Scripts/Locale.lua")
 
 ---------------------------------------------------------------
 ----------lots of bad variables and things
diff -r 164690ff1408 -r 4114ce5d885d share/hedgewars/Data/Maps/Control/map.lua
--- a/share/hedgewars/Data/Maps/Control/map.lua	Fri May 31 16:49:11 2013 +0200
+++ b/share/hedgewars/Data/Maps/Control/map.lua	Fri May 31 14:47:07 2013 -0400
@@ -49,7 +49,7 @@
 --script begins
 -----------------
 
-HedgewarsScriptLoad("Scripts/Locale.lua")()
+HedgewarsScriptLoad("/Scripts/Locale.lua")
 
 ---------------------------------------------------------------
 ----------lots of bad variables and things
diff -r 164690ff1408 -r 4114ce5d885d share/hedgewars/Data/Maps/Knockball/map.lua
--- a/share/hedgewars/Data/Maps/Knockball/map.lua	Fri May 31 16:49:11 2013 +0200
+++ b/share/hedgewars/Data/Maps/Knockball/map.lua	Fri May 31 14:47:07 2013 -0400
@@ -1,6 +1,6 @@
 -- Hedgewars - Knockball for 2+ Players
 
-HedgewarsScriptLoad("Scripts/Locale.lua")()
+HedgewarsScriptLoad("/Scripts/Locale.lua")
 
 local score = {[0] = 0, [1] = 0, [2] = 0, [3] = 0, [4] = 0, [5] = 0}
 
diff -r 164690ff1408 -r 4114ce5d885d share/hedgewars/Data/Maps/TrophyRace/map.lua
--- a/share/hedgewars/Data/Maps/TrophyRace/map.lua	Fri May 31 16:49:11 2013 +0200
+++ b/share/hedgewars/Data/Maps/TrophyRace/map.lua	Fri May 31 14:47:07 2013 -0400
@@ -1,6 +1,6 @@
 -- Hedgewars - Roperace for 2+ Players
 
-HedgewarsScriptLoad("Scripts/Locale.lua")()
+HedgewarsScriptLoad("/Scripts/Locale.lua")
 
 -- store number of hedgehogs
 local numhhs = 0
