From b13a712224a5fa80e427f3ff46dbb4ae600097c4 Mon Sep 17 00:00:00 2001
From: hasufell <hasufell@gentoo.org>
Date: Wed, 25 Dec 2013 18:29:30 +0100
Subject: [PATCH] BUILD: make binary/libdir destinations modifiable

---
 cmake_modules/paths.cmake | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmake_modules/paths.cmake b/cmake_modules/paths.cmake
index ddf2623..aefe67a 100644
--- a/cmake_modules/paths.cmake
+++ b/cmake_modules/paths.cmake
@@ -14,8 +14,8 @@ set(CMAKE_LIBRARY_OUTPUT_DIRECTORY_DEBUG ${LIBRARY_OUTPUT_PATH})
 
 #resource paths
 if(UNIX AND NOT APPLE)
-    set(target_binary_install_dir "bin")
-    set(target_library_install_dir "lib")
+    set(target_binary_install_dir "bin" CACHE PATH "install dest for binaries")
+    set(target_library_install_dir "lib" CACHE PATH "install dest for libs")
 
     string(SUBSTRING "${DATA_INSTALL_DIR}" 0 1 sharepath_start)
     if (NOT (${sharepath_start} MATCHES "/"))
-- 
1.8.5.2

