--- src/utils/endian.h.old
+++ src/utils/endian.h
@@ -93,19 +93,6 @@
 	}
 }
 
-template<typename T>
-int writele(T *Bufo, EMUFILE*os)
-{
-	CTASSERT(sizeof(T)==1||sizeof(T)==2||sizeof(T)==4||sizeof(T)==8);
-	switch(sizeof(T)) {
-		case 1: return write8le((uint8*)Bufo,os);
-		case 2: return write16le((uint16*)Bufo,os);
-		case 4: return write32le((uint32*)Bufo,os);
-		case 8: return write64le((uint64*)Bufo,os);
-		default:
-			return 0;
-	}
-}
 
 #endif //__FCEU_ENDIAN
 
