--- SConstruct.old	2011-06-07 18:37:31.324415039 +0200
+++ SConstruct	2011-06-07 18:42:04.804434870 +0200
@@ -65,6 +65,9 @@
     env.Append(CCFLAGS =["-D_GTK_LITE"])
   if env['GTK_LITE']:
     env.Append(CCFLAGS =["-D_GTK_LITE"])
+  env.Append(LIBS =["-lz"])
+  env.Append(LIBS =["-ldl"])
+  env.Append(LIBS =["-lX11"])
 
   ### Lua platform defines
   ### Applies to all files even though only lua needs it, but should be ok
