--- src/input/bworld.cpp.old	2010-11-29 16:36:13.000000000 +0100
+++ src/input/bworld.cpp	2010-11-29 16:37:26.000000000 +0100
@@ -22,7 +22,7 @@
 #include "share.h"
 
 static int seq,ptr,bit,cnt,have;
-static uint8 bdata[20];
+static uint8 bdata[32];
 
 
 static uint8 Read(int w, uint8 ret)
--- src/drivers/common/cheat.cpp.old	2010-11-29 16:23:44.000000000 +0100
+++ src/drivers/common/cheat.cpp	2010-11-29 16:24:24.000000000 +0100
@@ -404,7 +404,7 @@
  clo:
 
  printf("\nSelection [%d]> ",def+1);
- fgets(tmp,256,stdin);
+ fgets(tmp,16,stdin);
  if(tmp[0]=='\n')
   return def;
  c=tolower(tmp[0]);
