diff -ur EnvisagePlugins-3.2.0.orig/setup_data.py EnvisagePlugins-3.2.0/setup_data.py
--- setup_data.py	2011-01-27 04:55:51.000000000 +0800
+++ setup_data.py	2013-04-14 17:58:50.450023411 +0800
@@ -2,6 +2,6 @@
     'name': 'EnvisagePlugins',
     'version': '3.2.0',
     'install_requires' : [
-        'EnvisageCore >= 3.2.0.dev',
+        'envisage >= 4.0.0',
     ],
 }
diff -ur EnvisagePlugins-3.2.0.orig/setup.py EnvisagePlugins-3.2.0/setup.py
--- setup.py	2011-01-27 04:56:07.000000000 +0800
+++ setup.py	2013-04-14 18:00:35.714015967 +0800
@@ -49,8 +49,7 @@
     author = 'Martin Chilvers, et. al.',
     author_email = 'martin@enthought.com',
     download_url = (
-        'http://www.enthought.com/repo/ETS/EnvisagePlugins-%s.tar.gz' %
-        INFO['version']),
+        'http://www.enthought.com/repo/ETS/EnvisagePlugins-%s.tar.gz'),
     classifiers = [c.strip() for c in """\
         Development Status :: 5 - Production/Stable
         Intended Audience :: Developers
