diff -ur egoboo-2.7.4.orig/game/Makefile.unix egoboo-2.7.4/game/Makefile.unix
--- egoboo-2.7.4.orig/game/Makefile.unix	2009-07-03 05:36:02.000000000 -0400
+++ egoboo-2.7.4/game/Makefile.unix	2010-02-21 06:11:32.096511264 -0500
@@ -1,5 +1,5 @@
-# note if you change the prefix also update egoboo.sh
-PREFIX	:= ${HOME}/.local
+# note if you change the prefix also update egoboo.sh
+PREFIX	:= ${HOME}/.local
 EGO_SRC  := camera.c char.c client.c clock.c configfile.c egoboo_endian.c \
 	egoboo_fileutil.c egoboo_math.c egoboo_setup.c egoboo_strutil.c \
 	enchant.c file_common.c file_linux.c font.c game.c graphic.c\
@@ -9,36 +9,35 @@
 	egoboo_console.c script_functions.c mad.c mpd.c egoboo_typedef.c \
 	ogl_texture.c ogl_extensions.c ogl_include.c ogl_debug.c \
 	SDL_extensions.c SDL_GL_extensions.c mesh.c
-
-EGO_OBJ  := ${EGO_SRC:.c=.o}
-ENET_SRC := ../enet/host.c ../enet/list.c ../enet/memory.c \
-                  ../enet/packet.c ../enet/peer.c ../enet/protocol.c \
-                  ../enet/unix.c
-ENET_OBJ := ${ENET_SRC:.c=.o}
-
-SDL_CONF  := sdl-config
-SDLCONF_I := $(shell ${SDL_CONF} --cflags)
-SDLCONF_L := $(shell ${SDL_CONF} --libs)
-
-CC      := gcc
+
+EGO_OBJ  := ${EGO_SRC:.c=.o}
+ENET_SRC := ../enet/host.c ../enet/list.c ../enet/memory.c \
+                  ../enet/packet.c ../enet/peer.c ../enet/protocol.c \
+                  ../enet/unix.c
+ENET_OBJ := ${ENET_SRC:.c=.o}
+
+SDL_CONF  := sdl-config
+SDLCONF_I := $(shell ${SDL_CONF} --cflags)
+SDLCONF_L := $(shell ${SDL_CONF} --libs)
+
 OPT     := -Os -Wall
-INC     := -I. -I../enet/include -I.. ${SDLCONF_I}
-CFLAGS  := ${OPT} ${INC}
-LDFLAGS := ${SDLCONF_L} -lSDL_ttf -lSDL_mixer -lGL -lGLU -lSDL_image
-
-EGO_BIN := egoboo
-
-all: ${EGO_BIN}
-   
-
-${EGO_BIN}: ${EGO_OBJ} ${ENET_OBJ}
-	${CC} -o $@ $^ ${LDFLAGS}
-
-install: ${EGO_BIN}
-	mkdir -p ${PREFIX}/bin
-	mkdir -p ${PREFIX}/libexec
-	install -m 755 ${EGO_BIN} ${PREFIX}/libexec
-	install -p -m 755 ${EGO_BIN}.sh ${PREFIX}/bin/${EGO_BIN}
-
-clean:
-	rm -f ${ENET_OBJ} ${EGO_OBJ} ${EGO_BIN}
+INC     := -I. -I.. ${SDLCONF_I} -DENET11
+CFLAGS  := ${CFLAGS} ${INC}
+LDFLAGS += ${SDLCONF_L} -lenet -lSDL_ttf -lSDL_mixer -lGL -lGLU -lSDL_image
+
+EGO_BIN := egoboo
+
+all: ${EGO_BIN}
+   
+
+${EGO_BIN}: ${EGO_OBJ}
+	${CC} -o $@ $^ ${LDFLAGS}
+
+install: ${EGO_BIN}
+	mkdir -p ${PREFIX}/bin
+	mkdir -p ${PREFIX}/libexec
+	install -m 755 ${EGO_BIN} ${PREFIX}/libexec
+	install -p -m 755 ${EGO_BIN}.sh ${PREFIX}/bin/${EGO_BIN}
+
+clean:
+	rm -f ${ENET_OBJ} ${EGO_OBJ} ${EGO_BIN}
