--- game/Makefile.unix.old	2011-06-18 20:29:27.450415076 +0200
+++ game/Makefile.unix	2011-06-18 20:29:43.400421594 +0200
@@ -23,7 +23,7 @@
 OPT     := -Os -Wall
 INC     := -I. -I.. ${SDLCONF_I} -DENET11
 CFLAGS  := ${CFLAGS} ${INC}
-LDFLAGS += ${SDLCONF_L} -lenet -lSDL_ttf -lSDL_mixer -lGL -lGLU -lSDL_image
+LDFLAGS += ${SDLCONF_L} -lenet -lSDL_ttf -lSDL_mixer -lGL -lGLU -lSDL_image -lm
 
 EGO_BIN := egoboo
 
