--- PIL/GifImagePlugin.py
+++ PIL/GifImagePlugin.py
@@ -364,6 +364,11 @@
         for i in range(maxcolor):
             s.append(o8(i) * 3)
 
+    if im.info.has_key('transparency'):
+        transparentIndex = im.info['transparency']
+        s.append('!' + o8(0xf9) + o8(4) + o8(1) + o8(0) + o8(0) +
+                 o8(transparentIndex) + o8(0))
+
     return s
 
 def getdata(im, offset = (0, 0), **params):
