diff -NrU5 xxvautotimer-0.1.2.orig/mysqlwrapped-1.4/StderrLog.cpp xxvautotimer-0.1.2/mysqlwrapped-1.4/StderrLog.cpp
--- xxvautotimer-0.1.2.orig/mysqlwrapped-1.4/StderrLog.cpp	2009-06-10 17:01:27.000000000 -0600
+++ xxvautotimer-0.1.2/mysqlwrapped-1.4/StderrLog.cpp	2009-06-10 17:04:09.000000000 -0600
@@ -23,10 +23,11 @@
 */
 #ifdef _WIN32
 #pragma warning(disable:4786)
 #endif
 
+#include <cstdio>
 #include <string>
 #include <map>
 #ifdef WIN32
 #include <config-win.h>
 #include <mysql.h>
