--- Makefile.in
+++ Makefile.in
@@ -1,7 +1,7 @@
 # ----------------------------------------------------------------------------
 
-CC = gcc
-CFLAGS = -Wall $(GTK_CFLAGS) -O2 -g
+CC ?= gcc
+CFLAGS ?= -Wall $(GTK_CFLAGS) -O2 -g
 
 # ----------------------------------------------------------------------------
 
@@ -11,13 +11,16 @@
 	  mount.so mail.so showdesktop.so
 
 all:	$(OBJECTS) $(MODULES)
-	$(CC) $(CFLAGS) -o suxpanel -Wl,-export-dynamic $(OBJECTS) $(GTK_LIBS) $(GTK_FLAGS)
+	$(CC) $(LDFLAGS) $(GTK_CFLAGS) $(CFLAGS) -o suxpanel -Wl,-export-dynamic $(OBJECTS) $(GTK_LIBS)
 
 %.so: %.c 
 	@echo "[01;32mcompile module: $< ($@)[0m"
-	$(CC) $(CFLAGS) -o $@ -shared $< $(GTK_CFLAGS) $(GTK_LIBS) \
+	$(CC) $(LDFLAGS) $(GTK_CFLAGS) $(CFLAGS) -fPIC -o $@ -shared $< $(GTK_LIBS) \
             $(WNCK_LIBS) $(WNCK_CFLAGS)
 
+%.o: %.c
+	$(CC) $(GTK_CFLAGS) $(WNCK_CFLAGS) $(CFLAGS) -c $< -o $@
+
 clean:
 	rm -rf *~ *.o *.so suxpanel
 
