--- cdriso/src/Linux/Makefile.orig	2003-08-14 01:26:32.970330336 -0400
+++ cdriso/src/Linux/Makefile	2003-08-14 01:26:44.489579144 -0400
@@ -6,7 +6,7 @@
 BUILD = 4
 PLUGIN = libcdriso-${VERSION}.${BUILD}.so
 CFG = cfgCdrIso
-CFLAGS = -fPIC -Wall -O2 -fomit-frame-pointer -I.. -I. -D__LINUX__
+CFLAGS = -fPIC -Wall $(OPTFLAGS) -I.. -I. -D__LINUX__
 OBJECTS = ../cdriso.o Config.o Linux.o
 CFGOBJS = conf.o interface.o support.o Config.o
 LIBS = -lz -lbz2
