From e6433e978c3be048a5ff640325fd589677d14e5b Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Fri, 8 Apr 2011 11:24:08 +0200
Subject: [PATCH 2/2] gentoo: gtk-doc needs to have the library built

---
 Makefile.am |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 3a1498e..099c054 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,4 +1,4 @@
-SUBDIRS = docs libdmapsharing tests
+SUBDIRS = libdmapsharing docs tests
 DIST_SUBDIRS = docs libdmapsharing m4 media tests
 
 ### all of the standard pc files we need to generate
-- 
1.7.4.1

