--- ext/nio4r/extconf.rb.old	2014-05-07 02:51:48.850414484 +0200
+++ ext/nio4r/extconf.rb	2014-05-07 02:52:10.680415068 +0200
@@ -28,6 +28,8 @@
   $defs << '-DHAVE_SYS_RESOURCE_H'
 end
 
+$LDFLAGS << ' -lev'
+
 dir_config 'nio4r_ext'
 create_makefile 'nio4r_ext'
 
@@ -41,4 +43,4 @@
 
   makefile_contents.gsub! 'LIBS = $(LIBRUBYARG_SHARED)', 'LIBS = -lws2_32 $(LIBRUBYARG_SHARED)'
   File.open('Makefile', 'w') { |f| f.write makefile_contents }
-end
\ No newline at end of file
+end
