commit 1d523ea89ab93eadd153983f3aefdcfcdede3c9a
Author: Carl Eugen Hoyos <cehoyos@ag.or.at>
Date:   Tue Mar 10 12:14:16 2015 +0100

    lavc/hevcdsp: Fix compilation for arm with --disable-neon.

diff --git a/libavcodec/hevcdsp.c b/libavcodec/hevcdsp.c
index 04af178..b7a4729 100644
--- a/libavcodec/hevcdsp.c
+++ b/libavcodec/hevcdsp.c
@@ -259,6 +259,6 @@ int i = 0;
 
     if (ARCH_X86)
         ff_hevc_dsp_init_x86(hevcdsp, bit_depth);
-    if (ARCH_ARM)
+    if (HAVE_NEON)
         ff_hevcdsp_init_arm(hevcdsp, bit_depth);
 }
