--- Makefile.orig	2005-11-12 09:14:51.000000000 -0500
+++ Makefile	2005-11-12 09:14:53.000000000 -0500
@@ -57,7 +57,7 @@ PUBLISHGNA-TARGET = $(PUBLISHGNA:%=$(PUB
 
 PUBLISHFILE-TARGET = $(PUBLISHFILE:%=$(PUBLISHFILE-DEST)/%)
 
-.PHONY: all install tar publish filepublish wwwpublish upload test album demo famille clean cvs arch website
+.PHONY: all install tar publish filepublish wwwpublish upload album demo famille clean cvs arch website
 
 all:
 	@echo "see README file to install bins"
@@ -111,9 +111,6 @@ gnaupload: gnapublish
 fileupload: filepublish
 	cd $(PUBLISHFILE-DEST) && rsync --delete -avr --rsh="ssh" . zubro@download.gna.org:/upload/bins || true
 
-test: install
-	time bins -p ~/share/pics/test ~/public_html/test
-
 album: install
 	time nice -14 bins -p ~/share/pics/album ~/public_html/album
 
