--- Makefile.orig	2003-08-19 01:31:28.963869296 -0400
+++ Makefile	2003-08-19 01:31:45.567345184 -0400
@@ -11 +11 @@
-	$(CC) $(CFLAGS) `libnet-config --defines` \
+	$(CC) $(CFLAGS) `libnet-1.0-config --defines` \
@@ -13 +13 @@
-		`libnet-config --libs` -lpcap
+		`libnet-1.0-config --libs` -lpcap
--- tcptraceroute.c.orig	2003-08-19 01:31:32.991257040 -0400
+++ tcptraceroute.c	2003-08-19 01:32:02.723737016 -0400
@@ -27,2 +27,2 @@
- *	gcc -O2 -Wall `libnet-config --defines` \
- *		-o tcptraceroute tcptraceroute.c `libnet-config --libs` -lpcap
+ *	gcc -O2 -Wall `libnet-1.0-config --defines` \
+ *		-o tcptraceroute tcptraceroute.c `libnet-1.0-config --libs` -lpcap
@@ -201 +201 @@
-#include <libnet.h>
+#include <libnet-1.0.h>
