--- tcptraceroute.c.old	2003-07-03 19:56:41.000000000 +0000
+++ tcptraceroute.c	2005-04-15 09:49:58.000000000 +0000
@@ -189,7 +189,7 @@
 };
 
 /* Various globals */
-u_long dst_ip, src_ip;
+unsigned int dst_ip, src_ip;
 u_short src_prt, dst_prt;
 char *device, *name, *dst, *src;
 char dst_name[TEXTSIZE], dst_prt_name[TEXTSIZE], filter[TEXTSIZE];
@@ -213,7 +213,7 @@
 /* interface linked list, built later by getinterfaces() */
 struct interface_entry {
 	char *name;
-	u_long addr;
+	unsigned int addr;
 	struct interface_entry *next;
 } *interfaces;
 
@@ -223,7 +223,7 @@
 	u_short id, src_prt;
 	struct timeval timestamp;
 	double delta;
-	u_long addr;
+	unsigned int addr;
 	char *state;
 	char *string;
 } proberecord;
@@ -451,7 +451,7 @@
  */
 
 #define IPTOSBUFFERS	12
-char *iptos(u_long in)
+char *iptos(unsigned int in)
 {
 	static char output[IPTOSBUFFERS][3*4+3+1];
 	static short which;
@@ -468,7 +468,7 @@
  * RFC1918 space.
  */
 
-char *iptohost(u_long in)
+char *iptohost(unsigned int in)
 {
 	u_char *p = (u_char *)&in;
 
@@ -538,7 +538,7 @@
 	struct ifconf ifc;
 	struct ifreq *ifrp, ifr;
 	int numreqs, i, s;
-	u_long addr;
+	unsigned int addr;
 	int salen;
 	char *x;
 
@@ -679,7 +679,7 @@
  * given destination address.
  */
 
-u_long findsrc(u_long dest)
+unsigned int findsrc(unsigned int dest)
 {
 	struct sockaddr_in sinsrc, sindest;
 	int s, size;
@@ -715,7 +715,7 @@
  * interface exists with an address of 127.0.0.1.
  */
 
-char *finddev(u_long with_src)
+char *finddev(unsigned int with_src)
 {
 	struct interface_entry *p;
 	char *device = NULL;
@@ -812,7 +812,7 @@
 	/* Variables to keep state between calls */
 	static char laststate[TEXTSIZE];
 	static int lastttl;
-	static u_long lastaddr;
+	static unsigned int lastaddr;
 	static int everprint;
 
 	int printflag = 0;
@@ -960,7 +960,7 @@
 void defaults(void)
 {
 	struct servent *serv;
-	u_long recommended_src;
+	unsigned int recommended_src;
 
 	getinterfaces();
 
