--- tcptraceroute-1.4.orig/tcptraceroute.c
+++ tcptraceroute-1.4/tcptraceroute.c
@@ -394,4 +394,4 @@
-    fatal("Usage: %s [-nNFSAE] [-i <interface>] [-f <first ttl>]
-       [-l <packet length>] [-q <number of queries>] [-t <tos>]
-       [-m <max ttl>] [-pP] <source port>] [-s <source address>]
-       [-w <wait time>] <host> [destination port] [packet length]\n\n", name);
+    fatal("Usage: %s [-nNFSAE] [-i <interface>] [-f <first ttl>]\n"\
+"       [-l <packet length>] [-q <number of queries>] [-t <tos>]\n"\
+"       [-m <max ttl>] [-pP] <source port>] [-s <source address>]\n"\
+"       [-w <wait time>] <host> [destination port] [packet length]\n\n", name);
@@ -1185,3 +1185,3 @@
-	safe_snprintf(filter, TEXTSIZE, "
-		(tcp and src host %s and src port %d and dst host %s)
-		or ((icmp[0] == 11 or icmp[0] == 3) and dst host %s)",
+	safe_snprintf(filter, TEXTSIZE, \
+"		(tcp and src host %s and src port %d and dst host %s)\n"\
+"		or ((icmp[0] == 11 or icmp[0] == 3) and dst host %s)",
@@ -2003,6 +2003,6 @@

 	defaults();
 	initcapture();
-	seteuid(getuid());
+	setuid(getuid());
 	return trace();
 }
