Move things around so hpn applies cleanly when using X509.

--- a/Makefile.in
+++ b/Makefile.in
@@ -46,11 +46,12 @@
 CC=@CC@
 LD=@LD@
 CFLAGS=@CFLAGS@
-CPPFLAGS=-I. -I$(srcdir) @CPPFLAGS@ @LDAP_CPPFLAGS@ $(PATHS) @DEFS@
+CPPFLAGS=-I. -I$(srcdir) @CPPFLAGS@ $(PATHS) @DEFS@
 LIBS=@LIBS@
 SSHDLIBS=@SSHDLIBS@
 LIBEDIT=@LIBEDIT@
 LIBLDAP=@LDAP_LDFLAGS@ @LDAP_LIBS@
+CPPFLAGS+=@LDAP_CPPFLAGS@
 AR=@AR@
 AWK=@AWK@
 RANLIB=@RANLIB@
--- a/servconf.c
+++ b/servconf.c
@@ -153,9 +153,6 @@ initialize_server_options(ServerOptions *options)
 	options->adm_forced_command = NULL;
 	options->chroot_directory = NULL;
 	options->zero_knowledge_password_authentication = -1;
-	options->revoked_keys_file = NULL;
-	options->trusted_user_ca_keys = NULL;
-	options->authorized_principals_file = NULL;
 
 	options->hostbased_algorithms = NULL;
 	options->pubkey_algorithms = NULL;
@@ -168,6 +165,9 @@ initialize_server_options(ServerOptions *options)
 	options->va.certificate_file = NULL;
 	options->va.responder_url = NULL;
 #endif /*def SSH_OCSP_ENABLED*/
+	options->revoked_keys_file = NULL;
+	options->trusted_user_ca_keys = NULL;
+	options->authorized_principals_file = NULL;
 }
 
 void
@@ -367,9 +367,6 @@ typedef enum {
 	sClientAliveCountMax, sAuthorizedKeysFile, sAuthorizedKeysFile2,
 	sGssAuthentication, sGssCleanupCreds, sAcceptEnv, sPermitTunnel,
 	sMatch, sPermitOpen, sForceCommand, sChrootDirectory,
-	sUsePrivilegeSeparation, sAllowAgentForwarding,
-	sZeroKnowledgePasswordAuthentication, sHostCertificate,
-	sRevokedKeys, sTrustedUserCAKeys, sAuthorizedPrincipalsFile,
 	sHostbasedAlgorithms,
 	sPubkeyAlgorithms,
 	sX509KeyAlgorithm,
@@ -380,6 +377,9 @@ typedef enum {
 	sCAldapVersion, sCAldapURL,
 	sVAType, sVACertificateFile,
 	sVAOCSPResponderURL,
+	sUsePrivilegeSeparation, sAllowAgentForwarding,
+	sZeroKnowledgePasswordAuthentication, sHostCertificate,
+	sRevokedKeys, sTrustedUserCAKeys, sAuthorizedPrincipalsFile,
 	sDeprecated, sUnsupported
 } ServerOpCodes;
 
