diff --git a/configure.ac b/configure.ac
index 2b60300..21b6112 100644
--- a/configure.ac
+++ b/configure.ac
@@ -725,6 +725,10 @@ mips-sony-bsd|mips-sony-newsos4)
 	AC_CHECK_HEADER([net/if_tap.h], ,
 	    AC_DEFINE([SSH_TUN_NO_L2], [1], [No layer 2 tunnel support]))
 	AC_DEFINE([BROKEN_GLOB], [1], [FreeBSD glob does not do what we need])
+	AC_DEFINE([DISABLE_UTMP], [1],
+		[Define if you don't want to use utmp])
+	AC_DEFINE([DISABLE_WTMP], [1],
+		[Define if you don't want to use wtmp])
 	;;
 *-*-bsdi*)
 	AC_DEFINE([SETEUID_BREAKS_SETUID])
