--- numpy/distutils/system_info.py	2011-10-10 20:45:44.261316565 +0300
+++ numpy/distutils/system_info.py	2011-10-10 20:50:22.977932228 +0300
@@ -255,13 +255,7 @@
       1 - display warning message
       2 - raise error
     """
-    cl = {'atlas':atlas_info,  # use lapack_opt or blas_opt instead
-          'atlas_threads':atlas_threads_info,                # ditto
-          'atlas_blas':atlas_blas_info,
-          'atlas_blas_threads':atlas_blas_threads_info,
-          'lapack_atlas':lapack_atlas_info,  # use lapack_opt instead
-          'lapack_atlas_threads':lapack_atlas_threads_info,  # ditto
-          'mkl':mkl_info,
+    cl = {'mkl':mkl_info,
           'lapack_mkl':lapack_mkl_info,      # use lapack_opt instead
           'blas_mkl':blas_mkl_info,          # use blas_opt instead
           'x11':x11_info,
