diff -urN jbigkit.old/libjbig/Makefile jbigkit/libjbig/Makefile
--- jbigkit.old/libjbig/Makefile	2012-03-12 20:42:50.774142386 +0100
+++ jbigkit/libjbig/Makefile	2012-03-12 20:43:20.147140210 +0100
@@ -14,7 +14,9 @@
 	SONAME = so
 endif
 
-all: libjbig.a libjbig.$(SONAME) libjbig85.a libjbig85.$(SONAME) tstcodec tstcodec85
+all: libjbig.$(SONAME) libjbig85.$(SONAME) tstcodec tstcodec85
+
+static: libjbig.a libjbig85.a
 
 tstcodec: tstcodec.o jbig.o jbig_ar.o
 	$(CC) $(LDFLAGS) $(CFLAGS) -o tstcodec $+
