Index: fbsd-6.1/libexec/rtld-elf/Makefile
===================================================================
--- fbsd-6.1.orig/libexec/rtld-elf/Makefile
+++ fbsd-6.1/libexec/rtld-elf/Makefile
@@ -7,6 +7,7 @@ SRCS=		rtld_start.S \
 MAN=		rtld.1
 CSTD?=		gnu99
 CFLAGS+=	-Wall -DFREEBSD_ELF -DIN_RTLD
+CFLAGS+=	${NOSSP_FLAGS}
 CFLAGS+=	-I${.CURDIR}/${MACHINE_ARCH} -I${.CURDIR}
 LDFLAGS+=	-nostdlib -e .rtld_start
 INSTALLFLAGS=	-C -b
