http://code.google.com/p/rar2fs/issues/detail?id=12#c16

--- configure.ac
+++ configure.ac
@@ -26,6 +26,7 @@
 
 AM_PROG_CC_C_O
 AC_PROG_CXX
+AC_PROG_MKDIR_P
 
 AC_USE_SYSTEM_EXTENSIONS
 AX_C_BIGENDIAN_CROSS
--- Makefile.am
+++ Makefile.am
@@ -35,6 +35,7 @@
 
 if LINUX
 install-exec-hook:
+	$(MKDIR_P) $(DESTDIR)$(sbindir)
 	rm -f $(DESTDIR)$(sbindir)/mount.rar2fs && \
 	$(LN_S) $(bindir)/rar2fs $(DESTDIR)$(sbindir)/mount.rar2fs
 
