--- build.xml	2009-03-22 23:03:51.000000000 +0200
+++ build.xml.fixed	2009-03-22 23:13:44.000000000 +0200
@@ -101,7 +101,7 @@
              packagenames="weka.*"
              Author="yes" 
              Public="yes" 
-             maxmemory="128m"/>
+             maxmemory="${javac_max_memory}"/>
     
     <!-- insert the links to our homepage and documentation.html -->
     <replace dir="${doc}" 
@@ -132,7 +132,7 @@
   </target>
 
   <!-- compile the testcases -->
-  <target name="compile_tests" depends="init_tests, compile" 
+  <target name="compile_tests" depends="init_tests" 
    description="Compile unit tests into build/testcases">
      <javac srcdir="${src-test}" 
        fork="yes" memoryMaximumSize="${javac_max_memory}"
