diff -Naur xapian-bindings-1.2.8.orig//configure.ac xapian-bindings-1.2.8/configure.ac
--- xapian-bindings-1.2.8.orig//configure.ac	2011-12-14 00:29:23.000000000 -0500
+++ xapian-bindings-1.2.8/configure.ac	2011-12-17 13:28:25.000000000 -0500
@@ -926,13 +926,15 @@
     if test -n "$LUA" ; then
       AC_MSG_RESULT([$lua_version])
       AC_ARG_VAR(LUA_LIB, [Directory to install xapian.so into])
-      AC_MSG_CHECKING(for Lua library to install xapian.so)
-      LUA_LIB=`$LUA -e 'require("package") print(package.cpath)' | cut -f 2 -d ';' | cut -f 1 -d '?'`
-      LUA_LIB=`expr X"$LUA_LIB" : X'\(.*\)/$'`
-      if test -z "$LUA_LIB"; then
-        AC_MSG_RESULT(not found)
-      else
-        AC_MSG_RESULT($LUA_LIB)
+      if test -z "$LUA_LIB" ; then
+        AC_MSG_CHECKING([for Lua library to install xapian.so])
+        LUA_LIB=`$LUA -e 'require("package") print(package.cpath)' | cut -f 2 -d ';' | cut -f 1 -d '?'`
+        LUA_LIB=`expr X"$LUA_LIB" : X'\(.*\)/$'`
+        if test -z "$LUA_LIB"; then
+          AC_MSG_RESULT(not found)
+        else
+          AC_MSG_RESULT($LUA_LIB)
+        fi
       fi
       AC_SUBST(LUA_LIB)
 
