--- a/frontends/htmlfe.c
+++ b/frontends/htmlfe.c
@@ -188,7 +188,7 @@
 										<TD ALIGN=\"RIGHT\">\n\
 											%llu\n\
 										</TD>\n\
-									</TR>
+									</TR>\
 ", host->bytes_sent, host->bytes_received);
 
 	fprintf(out, "\
--- a/frontends/traffic-togif
+++ b/frontends/traffic-togif
@@ -31,7 +31,7 @@
 my $y_size = 750;
 
 my $traffic_tops = "/usr/sbin/traffic-tops";
-my $gs = "/usr/bin/ghostscript";
+my $gs = "/usr/bin/gs";
 my $pnmcrop = "/usr/bin/pnmcrop";
 my $pnmscale = "/usr/bin/pnmscale";
 my $ppmtogif = "/usr/bin/ppmtogif";
--- a/Makefile	2003-06-20 09:07:17.000000000 +0200
+++ b/Makefile	2010-09-10 06:53:46.000000000 +0200
@@ -7,11 +7,11 @@
 OPT_FLAGS=$(RPM_OPT_FLAGS)
 WARNFLAGS=-Wall -Wno-conversion -Waggregate-return -Wmissing-prototypes -Wstrict-prototypes 
 DEBUGFLAGS=-g #-DDEBUG
-INCLUDES=-I/usr/include/pcap `glib-config --cflags`
+INCLUDES=-I/usr/include/pcap `pkg-config --cflags glib-2.0`
 DEFINES= -DHAVE_GETOPT_LONG
 CFLAGS+= -DVERSION="\"$(VERSION)"\" $(DEFINES) $(WARNFLAGS) $(DEBUGFLAGS) $(INCLUDES) $(OPT_FLAGS)
 LFLAGS=
-GLIB=`glib-config --libs`
+GLIB=`pkg-config --libs glib-2.0`
 #GLIB=/usr/lib/libglib.a
 #DEBUGLIBS=-lefence
 
@@ -25,7 +25,7 @@
 all: $(TARGETS) 
 	for x in $(SUBDIRS) ; \
 	do \
-		make -C $$x \
+		$(MAKE) -C $$x \
 			VERSION="$(VERSION)" OPT_FLAGS="$(OPT_FLAGS)" \
 			WARNFLAGS="$(WARNFLAGS)" INCLUDES="$(INCLUDES)" \
 			DEFINES="$(DEFINES)" GLIB="$(GLIB)" \
@@ -38,7 +38,7 @@
 install:
 	for x in $(SUBDIRS) ; \
 	do \
-		make -C $$x SBINDIR="$(SBINDIR)" MANDIR="$(MANDIR)" install || exit 1; \
+		$(MAKE) -C $$x SBINDIR="$(SBINDIR)" MANDIR="$(MANDIR)" install || exit 1; \
 	done
 	install -m 644 -o root -g root traffic-vis.8 $(MANDIR)/man8/
 
@@ -46,6 +46,6 @@
 	rm -f *.o core ~* *.a *.ps *.txt *.html test test.o *.tv
 	for x in $(SUBDIRS) ; \
 	do \
-		(cd $$x && make clean && cd ..) ; \
+		(cd $$x && $(MAKE) clean && cd ..) ; \
 	done
 
--- a/collector/Makefile	1999-02-11 04:23:50.000000000 +0100
+++ b/collector/Makefile	2010-09-10 06:53:46.000000000 +0200
@@ -1,4 +1,4 @@
-CFLAGS= -DVERSION="\"$(VERSION)"\" $(DEFINES) $(WARNFLAGS) $(DEBUGFLAGS) $(INCLUDES) -I.. $(OPT_FLAGS)
+CFLAGS+= -DVERSION="\"$(VERSION)"\" $(DEFINES) $(WARNFLAGS) $(DEBUGFLAGS) $(INCLUDES) -I.. $(OPT_FLAGS)
 OBJS=packet-parse.o packet-summary.o summary-output.o ../util.o
 TARGETS=traffic-collector
 LFLAGS=-L..
@@ -7,7 +7,7 @@
 all: $(TARGETS)
 
 traffic-collector: traffic-collector.o $(OBJS)
-	$(CC) -o $@ traffic-collector.o $(OBJS) $(LIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ traffic-collector.o $(OBJS) $(LIBS)
 
 install:
 	install -m 750 -o root -g root traffic-collector $(SBINDIR)
--- a/frontends/Makefile	1999-02-11 04:23:50.000000000 +0100
+++ b/frontends/Makefile	2010-09-10 06:53:46.000000000 +0200
@@ -1,4 +1,4 @@
-CFLAGS= -DVERSION="\"$(VERSION)\"" $(DEFINES) $(WARNFLAGS) $(DEBUGFLAGS) $(INCLUDES) -I.. $(OPT_FLAGS)
+CFLAGS+= -DVERSION="\"$(VERSION)\"" $(DEFINES) $(WARNFLAGS) $(DEBUGFLAGS) $(INCLUDES) -I.. $(OPT_FLAGS)
 OBJS=htmlfe.o psfe.o textfe.o
 TARGETS=traffic-tops traffic-totext traffic-tohtml
 LFLAGS=-L..
@@ -7,13 +7,13 @@
 all: $(OBJS) $(TARGETS)
 
 traffic-tops: traffic-tops.o psfe.o ../libreport.a
-	$(CC) -o $@ traffic-tops.o psfe.o $(LFLAGS) $(LIBS) -lm
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ traffic-tops.o psfe.o $(LFLAGS) $(LIBS) -lm
 
 traffic-totext: traffic-totext.o textfe.o ../libreport.a
-	$(CC) -o $@ traffic-totext.o textfe.o $(LFLAGS) $(LIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ traffic-totext.o textfe.o $(LFLAGS) $(LIBS)
 
 traffic-tohtml: traffic-tohtml.o htmlfe.o ../libreport.a
-	$(CC) -o $@ traffic-tohtml.o htmlfe.o $(LFLAGS) $(LIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ traffic-tohtml.o htmlfe.o $(LFLAGS) $(LIBS)
 
 install:
 	install -m 755 -o root -g root traffic-tops $(SBINDIR)
--- a/sort/Makefile	1999-02-11 04:23:50.000000000 +0100
+++ b/sort/Makefile	2010-09-10 06:56:39.000000000 +0200
@@ -1,13 +1,13 @@
-CFLAGS= -DVERSION="\"$(VERSION)\"" $(DEFINES) $(WARNFLAGS) $(DEBUGFLAGS) $(INCLUDES) -I.. $(OPT_FLAGS)
+CFLAGS+= -DVERSION="\"$(VERSION)\"" $(DEFINES) $(WARNFLAGS) $(DEBUGFLAGS) $(INCLUDES) -I.. $(OPT_FLAGS)
 OBJS=
 TARGETS=traffic-sort
-LFLAGS=-L..
-LIBS=-lreport $(GLIB) $(DEBUGLIBS)
+LFLAGS=
+LIBS=-L.. -lreport $(GLIB) $(DEBUGLIBS)
 
 all: $(TARGETS)
 
 traffic-sort: traffic-sort.o ../libreport.a
-	$(CC) $(LFLAGS) -o $@ traffic-sort.o $(LIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $(LFLAGS) -o $@ traffic-sort.o $(LIBS)
 
 install:
 	install -m 755 -o root -g root traffic-sort $(SBINDIR)
--- a/utils/Makefile	1999-02-11 04:23:50.000000000 +0100
+++ b/utils/Makefile	2010-09-10 06:53:46.000000000 +0200
@@ -1,4 +1,4 @@
-CFLAGS= -DVERSION="\"$(VERSION)\"" $(DEFINES) $(WARNFLAGS) $(DEBUGFLAGS) $(INCLUDES) -I.. $(OPT_FLAGS)
+CFLAGS+= -DVERSION="\"$(VERSION)\"" $(DEFINES) $(WARNFLAGS) $(DEBUGFLAGS) $(INCLUDES) -I.. $(OPT_FLAGS)
 OBJS=
 TARGETS=traffic-resolve traffic-exclude
 LFLAGS=-L..
@@ -7,10 +7,10 @@
 all: $(OBJS) $(TARGETS)
 
 traffic-exclude: traffic-exclude.o ../libreport.a
-	$(CC) -o $@ traffic-exclude.o $(LFLAGS) $(LIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ traffic-exclude.o $(LFLAGS) $(LIBS)
 
 traffic-resolve: traffic-resolve.o ../libreport.a
-	$(CC) -o $@ traffic-resolve.o $(LFLAGS) $(LIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ traffic-resolve.o $(LFLAGS) $(LIBS)
 
 install:
 	install -m 755 -o root -g root traffic-resolve $(SBINDIR)
