diff -Naur polarssl-0.12.0.old/library/Makefile polarssl-0.12.0/library/Makefile
--- polarssl-0.12.0.old/library/Makefile	2009-03-28 18:53:03.000000000 +0100
+++ polarssl-0.12.0/library/Makefile	2010-01-23 10:12:03.000000000 +0100
@@ -1,8 +1,7 @@
 
 # Also see "include/polarssl/config.h"
 
-CFLAGS	= -I../include -D_FILE_OFFSET_BITS=64 -Wall -Wdeclaration-after-statement
-OFLAGS	= -O
+CFLAGS	+= -I../include -D_FILE_OFFSET_BITS=64 -Wall -Wdeclaration-after-statement
 
 # MicroBlaze specific options:
 # CFLAGS += -mno-xl-soft-mul -mxl-barrel-shift
