--- old/library/Makefile	2011-11-16 17:41:41.000000000 -0500
+++ old/library/Makefile	2011-11-16 18:00:12.000000000 -0500
@@ -2,7 +2,7 @@ 
 # Also see "include/polarssl/config.h"
 
 CFLAGS	+= -I../include -D_FILE_OFFSET_BITS=64 -Wall -W -Wdeclaration-after-statement
-OFLAGS	= -O
+# OFLAGS	= -O
 
 # MicroBlaze specific options:
 # CFLAGS += -mno-xl-soft-mul -mxl-barrel-shift
@@ -32,7 +32,7 @@ 
 		sha4.o		ssl_cli.o	ssl_srv.o	\
 		ssl_tls.o	timing.o	version.o	\
 		x509parse.o	xtea.o
-	
+
 
 .SILENT:
 
--- old/Makefile	2011-11-16 17:41:41.000000000 -0500
+++ old/Makefile	2011-11-16 18:01:56.000000000 -0500
@@ -12,10 +12,10 @@ 
 install:
 	mkdir -p $(DESTDIR)/include/polarssl
 	cp -r include/polarssl $(DESTDIR)/include
-	
+
 	mkdir -p $(DESTDIR)/lib
 	cp library/libpolarssl.* $(DESTDIR)/lib
-	
+
 	mkdir -p $(DESTDIR)/bin
 	for p in programs/*/* ; do              \
 	    if [ -x $$p ] && [ ! -d $$p ] ;     \
--- old/programs/Makefile	2011-11-16 17:41:41.000000000 -0500
+++ old/programs/Makefile	2011-11-16 18:05:31.000000000 -0500
@@ -4,7 +4,7 @@ 
 # To compile with PKCS11: add "-lpkcs11-helper" to LDFLAGS
 
 CFLAGS	+= -I../include -D_FILE_OFFSET_BITS=64 -Wall -W -Wdeclaration-after-statement
-OFLAGS	= -O
+# OFLAGS	= -O
 LDFLAGS	+= -L../library -lpolarssl $(SYS_LDFLAGS)
 
 APPS =	aes/aescrypt2	aes/crypt_and_hash	\
--- old/tests/Makefile	2011-11-16 17:41:41.000000000 -0500
+++ old/tests/Makefile	2011-11-16 17:59:27.000000000 -0500
@@ -6,7 +6,7 @@ 
 CFLAGS	+= -I../include -D_FILE_OFFSET_BITS=64 -Wall -W -Wdeclaration-after-statement \
 			-Wno-unused-function -Wno-unused-value
 
-OFLAGS	= -O
+# OFLAGS	= -O
 LDFLAGS	+= -L../library -lpolarssl $(SYS_LDFLAGS)
 
 APPS =	test_suite_aes			test_suite_arc4			\
