diff -Naur mydns-1.1.0.orig/doc/Makefile.am mydns-1.1.0/doc/Makefile.am
--- mydns-1.1.0.orig/doc/Makefile.am	2004-02-28 03:28:06.000000000 +0900
+++ mydns-1.1.0/doc/Makefile.am	2008-12-28 09:48:57.000000000 +0900
@@ -33,25 +33,25 @@
 	fi
 
 mydns.8: sedfile ${@:=.in}
-	$(SHELL) @top_srcdir@/missing --run sed -f @SEDFILE@ ${@:=.in} | tr -s '\n' > $@
+	sed -f @SEDFILE@ ${@:=.in} | tr -s '\n' > $@
 
 mydnscheck.8: sedfile ${@:=.in}
-	$(SHELL) @top_srcdir@/missing --run sed -f @SEDFILE@ ${@:=.in} | tr -s '\n' > $@
+	sed -f @SEDFILE@ ${@:=.in} | tr -s '\n' > $@
 
 mydns.conf.5: sedfile ${@:=.in}
-	$(SHELL) @top_srcdir@/missing --run sed -f @SEDFILE@ ${@:=.in} | tr -s '\n' > $@
+	sed -f @SEDFILE@ ${@:=.in} | tr -s '\n' > $@
 
 mydns-conf.8: sedfile ${@:=.in}
-	$(SHELL) @top_srcdir@/missing --run sed -f @SEDFILE@ ${@:=.in} | tr -s '\n' > $@
+	sed -f @SEDFILE@ ${@:=.in} | tr -s '\n' > $@
 
 mydnsexport.8: sedfile ${@:=.in}
-	$(SHELL) @top_srcdir@/missing --run sed -f @SEDFILE@ ${@:=.in} | tr -s '\n' > $@
+	sed -f @SEDFILE@ ${@:=.in} | tr -s '\n' > $@
 
 mydnsimport.8: sedfile ${@:=.in}
-	$(SHELL) @top_srcdir@/missing --run sed -f @SEDFILE@ ${@:=.in} | tr -s '\n' > $@
+	sed -f @SEDFILE@ ${@:=.in} | tr -s '\n' > $@
 
 mydnsptrconvert.8: sedfile ${@:=.in}
-	$(SHELL) @top_srcdir@/missing --run sed -f @SEDFILE@ ${@:=.in} | tr -s '\n' > $@
+	sed -f @SEDFILE@ ${@:=.in} | tr -s '\n' > $@
 
 html:
 	if [ -x /usr/local/bin/mydns-texi2html ]; then /usr/local/bin/mydns-texi2html mydns.texi; fi
--- mydns-1.2.8/m4/mydns.m4.orig	2009-02-02 14:52:34.000000000 +0100
+++ mydns-1.2.8/m4/mydns.m4	2010-12-01 22:31:20.000000000 +0100
@@ -357,16 +357,12 @@
 					AC_MSG_RESULT([no])
 					if test -z "$PROFILE_ENABLED"; then
 						AC_C_CFLAGS(-finline-functions)
-					else
-						LDFLAGS="$LDFLAGS -static"
 					fi
 				fi
 			], [
 				AC_MSG_RESULT([no])
 				if test -z "$PROFILE_ENABLED"; then
 					AC_C_CFLAGS(-finline-functions)
-				else
-					LDFLAGS="$LDFLAGS -static"
 				fi
 			]
 		)
