diff -Naurp adplug-2.1-orig/src/a2m.cpp adplug-2.1/src/a2m.cpp
--- adplug-2.1-orig/src/a2m.cpp	2008-09-08 22:33:32.000000000 +0200
+++ adplug-2.1/src/a2m.cpp	2008-09-08 22:33:53.000000000 +0200
@@ -31,6 +31,8 @@
 
 #include "a2m.h"
 
+#include <cstring>
+
 const unsigned int Ca2mLoader::MAXFREQ = 2000,
 Ca2mLoader::MINCOPY = ADPLUG_A2M_MINCOPY,
 Ca2mLoader::MAXCOPY = ADPLUG_A2M_MAXCOPY,
diff -Naurp adplug-2.1-orig/src/adl.cpp adplug-2.1/src/adl.cpp
--- adplug-2.1-orig/src/adl.cpp	2008-09-08 22:33:32.000000000 +0200
+++ adplug-2.1/src/adl.cpp	2008-09-08 22:33:53.000000000 +0200
@@ -49,6 +49,7 @@
  */
 
 #include <inttypes.h>
+#include <cstring>
 #include <stdarg.h>
 #include <assert.h>
 
diff -Naurp adplug-2.1-orig/src/bmf.cpp adplug-2.1/src/bmf.cpp
--- adplug-2.1-orig/src/bmf.cpp	2008-09-08 22:33:32.000000000 +0200
+++ adplug-2.1/src/bmf.cpp	2008-09-08 22:33:53.000000000 +0200
@@ -43,6 +43,8 @@
 #include "bmf.h"
 #include "debug.h"
 
+#include <cstring>
+
 const unsigned char CxadbmfPlayer::bmf_adlib_registers[117] =
 {
   0x20, 0x23, 0x40, 0x43, 0x60, 0x63, 0x80, 0x83, 0xA0, 0xB0, 0xC0, 0xE0, 0xE3,
diff -Naurp adplug-2.1-orig/src/cff.cpp adplug-2.1/src/cff.cpp
--- adplug-2.1-orig/src/cff.cpp	2008-09-08 22:33:32.000000000 +0200
+++ adplug-2.1/src/cff.cpp	2008-09-08 22:33:53.000000000 +0200
@@ -28,6 +28,8 @@
 
 #include "cff.h"
 
+#include <cstring>
+
 /* -------- Public Methods -------------------------------- */
 
 CPlayer *CcffLoader::factory(Copl *newopl)
diff -Naurp adplug-2.1-orig/src/dro.cpp adplug-2.1/src/dro.cpp
--- adplug-2.1-orig/src/dro.cpp	2008-09-08 22:33:32.000000000 +0200
+++ adplug-2.1/src/dro.cpp	2008-09-08 22:33:53.000000000 +0200
@@ -27,6 +27,8 @@
 
 #include "dro.h"
 
+#include <cstring>
+
 /*** public methods *************************************/
 
 CPlayer *CdroPlayer::factory(Copl *newopl)
diff -Naurp adplug-2.1-orig/src/dtm.cpp adplug-2.1/src/dtm.cpp
--- adplug-2.1-orig/src/dtm.cpp	2008-09-08 22:33:32.000000000 +0200
+++ adplug-2.1/src/dtm.cpp	2008-09-08 22:33:53.000000000 +0200
@@ -24,6 +24,8 @@
 
 #include "dtm.h"
 
+#include <cstring>
+
 /* -------- Public Methods -------------------------------- */
 
 CPlayer *CdtmLoader::factory(Copl *newopl)
diff -Naurp adplug-2.1-orig/src/fmc.cpp adplug-2.1/src/fmc.cpp
--- adplug-2.1-orig/src/fmc.cpp	2008-09-08 22:33:32.000000000 +0200
+++ adplug-2.1/src/fmc.cpp	2008-09-08 22:33:53.000000000 +0200
@@ -21,6 +21,8 @@
 
 #include "fmc.h"
 
+#include <cstring>
+
 /* -------- Public Methods -------------------------------- */
 
 CPlayer *CfmcLoader::factory(Copl *newopl)
diff -Naurp adplug-2.1-orig/src/mad.cpp adplug-2.1/src/mad.cpp
--- adplug-2.1-orig/src/mad.cpp	2008-09-08 22:33:32.000000000 +0200
+++ adplug-2.1/src/mad.cpp	2008-09-08 22:33:53.000000000 +0200
@@ -21,6 +21,8 @@
 
 #include "mad.h"
 
+#include <cstring>
+
 /* -------- Public Methods -------------------------------- */
 
 CPlayer *CmadLoader::factory(Copl *newopl)
diff -Naurp adplug-2.1-orig/src/mkj.cpp adplug-2.1/src/mkj.cpp
--- adplug-2.1-orig/src/mkj.cpp	2008-09-08 22:33:32.000000000 +0200
+++ adplug-2.1/src/mkj.cpp	2008-09-08 22:33:53.000000000 +0200
@@ -24,6 +24,8 @@
 #include "mkj.h"
 #include "debug.h"
 
+#include <cstring>
+
 CPlayer *CmkjPlayer::factory(Copl *newopl)
 {
   return new CmkjPlayer(newopl);
diff -Naurp adplug-2.1-orig/src/msc.cpp adplug-2.1/src/msc.cpp
--- adplug-2.1-orig/src/msc.cpp	2008-09-08 22:33:32.000000000 +0200
+++ adplug-2.1/src/msc.cpp	2008-09-08 22:33:53.000000000 +0200
@@ -24,6 +24,8 @@
 #include "msc.h"
 #include "debug.h"
 
+#include <cstring>
+
 const unsigned char CmscPlayer::msc_signature [MSC_SIGN_LEN] = {
   'C', 'e', 'r', 'e', 's', ' ', '\x13', ' ',
   'M', 'S', 'C', 'p', 'l', 'a', 'y', ' ' };
diff -Naurp adplug-2.1-orig/src/mtk.cpp adplug-2.1/src/mtk.cpp
--- adplug-2.1-orig/src/mtk.cpp	2008-09-08 22:33:32.000000000 +0200
+++ adplug-2.1/src/mtk.cpp	2008-09-08 22:33:53.000000000 +0200
@@ -21,6 +21,8 @@
 
 #include "mtk.h"
 
+#include <cstring>
+
 /*** public methods **************************************/
 
 CPlayer *CmtkLoader::factory(Copl *newopl)
diff -Naurp adplug-2.1-orig/src/protrack.cpp adplug-2.1/src/protrack.cpp
--- adplug-2.1-orig/src/protrack.cpp	2008-09-08 22:33:32.000000000 +0200
+++ adplug-2.1/src/protrack.cpp	2008-09-08 22:33:53.000000000 +0200
@@ -28,6 +28,8 @@
 #include "protrack.h"
 #include "debug.h"
 
+#include <cstring>
+
 #define SPECIALARPLEN	256	// Standard length of special arpeggio lists
 #define JUMPMARKER	0x80	// Orderlist jump marker
 
diff -Naurp adplug-2.1-orig/src/rad.cpp adplug-2.1/src/rad.cpp
--- adplug-2.1-orig/src/rad.cpp	2008-09-08 22:33:32.000000000 +0200
+++ adplug-2.1/src/rad.cpp	2008-09-08 22:33:53.000000000 +0200
@@ -24,6 +24,8 @@
 
 #include "rad.h"
 
+#include <cstring>
+
 CPlayer *CradLoader::factory(Copl *newopl)
 {
   return new CradLoader(newopl);
diff -Naurp adplug-2.1-orig/src/rat.cpp adplug-2.1/src/rat.cpp
--- adplug-2.1-orig/src/rat.cpp	2008-09-08 22:33:32.000000000 +0200
+++ adplug-2.1/src/rat.cpp	2008-09-08 22:33:53.000000000 +0200
@@ -32,6 +32,8 @@
 #include "rat.h"
 #include "debug.h"
 
+#include <cstring>
+
 const unsigned char CxadratPlayer::rat_adlib_bases[18] =
 {
   0x00, 0x01, 0x02, 0x08, 0x09, 0x0A, 0x10, 0x11, 0x12,
diff -Naurp adplug-2.1-orig/src/raw.cpp adplug-2.1/src/raw.cpp
--- adplug-2.1-orig/src/raw.cpp	2008-09-08 22:33:32.000000000 +0200
+++ adplug-2.1/src/raw.cpp	2008-09-08 22:33:53.000000000 +0200
@@ -21,6 +21,8 @@
 
 #include "raw.h"
 
+#include <cstring>
+
 /*** public methods *************************************/
 
 CPlayer *CrawPlayer::factory(Copl *newopl)
diff -Naurp adplug-2.1-orig/src/rix.cpp adplug-2.1/src/rix.cpp
--- adplug-2.1-orig/src/rix.cpp	2008-09-08 22:33:32.000000000 +0200
+++ adplug-2.1/src/rix.cpp	2008-09-08 22:33:53.000000000 +0200
@@ -23,6 +23,9 @@
 #include "rix.h"
 #include "debug.h"
 
+#include <cstdlib>
+#include <cstring>
+
 const unsigned char CrixPlayer::adflag[] = {0,0,0,1,1,1,0,0,0,1,1,1,0,0,0,1,1,1};
 const unsigned char CrixPlayer::reg_data[] = {0,1,2,3,4,5,8,9,10,11,12,13,16,17,18,19,20,21};
 const unsigned char CrixPlayer::ad_C0_offs[] = {0,1,2,0,1,2,3,4,5,3,4,5,6,7,8,6,7,8};
diff -Naurp adplug-2.1-orig/src/rol.h adplug-2.1/src/rol.h
--- adplug-2.1-orig/src/rol.h	2008-09-08 22:33:32.000000000 +0200
+++ adplug-2.1/src/rol.h	2008-09-08 22:33:53.000000000 +0200
@@ -25,6 +25,7 @@
 
 #include <vector>
 #include <string>
+#include <cstring>
 
 #include "player.h"
 
diff -Naurp adplug-2.1-orig/src/s3m.cpp adplug-2.1/src/s3m.cpp
--- adplug-2.1-orig/src/s3m.cpp	2008-09-08 22:33:32.000000000 +0200
+++ adplug-2.1/src/s3m.cpp	2008-09-08 22:33:53.000000000 +0200
@@ -24,6 +24,8 @@
 
 #include "s3m.h"
 
+#include <cstring>
+
 const char Cs3mPlayer::chnresolv[] =	// S3M -> adlib channel conversion
   {-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,0,1,2,3,4,5,6,7,8,-1,-1,-1,-1,-1,-1,-1};
 
diff -Naurp adplug-2.1-orig/src/sa2.cpp adplug-2.1/src/sa2.cpp
--- adplug-2.1-orig/src/sa2.cpp	2008-09-08 22:33:32.000000000 +0200
+++ adplug-2.1/src/sa2.cpp	2008-09-08 22:33:53.000000000 +0200
@@ -25,6 +25,8 @@
 #include "sa2.h"
 #include "debug.h"
 
+#include <cstring>
+
 CPlayer *Csa2Loader::factory(Copl *newopl)
 {
   return new Csa2Loader(newopl);
diff -Naurp adplug-2.1-orig/src/sng.cpp adplug-2.1/src/sng.cpp
--- adplug-2.1-orig/src/sng.cpp	2008-09-08 22:33:32.000000000 +0200
+++ adplug-2.1/src/sng.cpp	2008-09-08 22:33:53.000000000 +0200
@@ -21,6 +21,8 @@
 
 #include "sng.h"
 
+#include <cstring>
+
 CPlayer *CsngPlayer::factory(Copl *newopl)
 {
   return new CsngPlayer(newopl);
diff -Naurp adplug-2.1-orig/test/playertest.cpp adplug-2.1/test/playertest.cpp
--- adplug-2.1-orig/test/playertest.cpp	2008-09-08 22:33:33.000000000 +0200
+++ adplug-2.1/test/playertest.cpp	2008-09-08 22:34:45.000000000 +0200
@@ -19,6 +19,7 @@
  * playertest.cpp - Test AdPlug replayers, by Simon Peter <dn.tlp@gmx.net>
  */
 
+#include <cstring>
 #include <stdlib.h>
 #include <stdio.h>
 #include <string>
