--- cgilib-0.5/cgitest.c	1999-08-17 10:21:20.000000000 -0300
+++ cgilib.cvs/cgitest.c	2005-05-03 00:38:50.000000000 -0300
@@ -23,6 +23,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 #include <cgi.h>
 
 s_cgi *cgi;
@@ -113,7 +114,7 @@
             cgiHeader();
 	    printf ("<html>\n<head><title>cgilib</title></title>\n\n<body bgcolor=\"#ffffff\">\n");
 	    printf ("<h1><a href=\"%s\">cgilib</a></h1>\n", URL);
-	    printf ("<h3>Cookie "Library" set</h3>\n");
+	    printf ("<h3>Cookie Library set</h3>\n");
 	    printf ("<p><br><p><br><a href=\"/cgi-bin/cgitest\">Test</a><p>\n");
 	    printf ("<p><br><p><br><a href=\"/cgi-bin/cgitest/redirect\">Redirect</a><p>\n");
 	    printf ("<p><br><p><br><a href=\"/cgi-bin/cgitest/listall\">List Everything</a><p>\n");
