--- mcl-0.53.00/samples/contrib/DynaComplete.py	1999-09-20 20:10:33.000000000 +0200
+++ mcl-0.53.00/samples/contrib/DynaComplete.py	1999-09-20 20:10:33.000000000 +0200
@@ -23,7 +23,7 @@
 	self.dict = { }
 	self.userwords = { }
 	self.lookup = None
-	self.wording = re.compile( '\s*(?P<word>[\w-#]+)' )
+	self.wording = re.compile( '\s*(?P<word>[\w\-#]+)' )
 	self.add_word( 'complete' )
 
     def add_word( self, inword ):
