--- Makefile.in
+++ Makefile.in
@@ -15,7 +15,7 @@
 LDFLAGS=@LDFLAGS@
 LIBS=@LIBS@
 prefix=@prefix@
-INSTALL_ROOT=@INSTALL_ROOT@
+libdir=@libdir@
 COMPILED_BY=@COMPILED_BY@
 MODULES=@WITH_PERL@ @WITH_PYTHON@
 
@@ -117,12 +117,13 @@
 	$(CC) -c $(MCL_CFLAGS) $(PYTHON_CPPFLAGS) $< -o $@
 
 install: $(EXE)
-	$(INSTALL) -d $(INSTALL_ROOT)$(prefix)/{bin,lib/mcl/{auto,contrib,plugins,sys}}
-	$(INSTALL) $(EXE) $(INSTALL_ROOT)$(prefix)/bin
-	$(INSTALL) o/plugins/*.so $(INSTALL_ROOT)$(prefix)/lib/mcl/plugins || : # There may be no plugins to install
-	$(INSTALL) samples/auto/*.* $(INSTALL_ROOT)$(prefix)/lib/mcl/auto/ # The *.* is desired, not a typo
-	$(INSTALL) samples/contrib/*.* $(INSTALL_ROOT)$(prefix)/lib/mcl/contrib/
-	$(INSTALL) samples/sys/*.* $(INSTALL_ROOT)$(prefix)/lib/mcl/sys/
+	$(INSTALL) -d $(DESTDIR)$(prefix)/bin
+	$(INSTALL) -d $(DESTDIR)$(libdir)/mcl/{auto,contrib,plugins,sys}
+	$(INSTALL) $(EXE) $(DESTDIR)$(prefix)/bin
+	-$(INSTALL) o/plugins/*.so $(DESTDIR)$(libdir)/mcl/plugins
+	$(INSTALL) samples/auto/*.* $(DESTDIR)$(libdir)/mcl/auto/
+	$(INSTALL) samples/contrib/*.* $(DESTDIR)$(libdir)/mcl/contrib/
+	$(INSTALL) samples/sys/*.* $(DESTDIR)$(libdir)/mcl/sys/
 
 
 include admin/Makefile
