--- Makefile.in	2004-03-30 11:32:20.422782923 +0000
+++ Makefile.in	2004-03-30 11:32:36.558806599 +0000
@@ -108,13 +108,13 @@
 	$(CC) $(LDFLAGS) -shared $< -o $@ $(PERL_LDFLAGS) $(LIBPTHREAD)
 
 o/plugins/PerlEmbeddedInterpreter.o: plugins/PerlEmbeddedInterpreter.cc
-	$(CC) -c $(MCL_CFLAGS) $(PERL_CPPFLAGS) $< -o $@
+	$(CC) -fPIC -c $(MCL_CFLAGS) $(PERL_CPPFLAGS) $< -o $@
 
 o/plugins/python.so: o/plugins/PythonEmbeddedInterpreter.o
 	$(CC) $(LDFLAGS) -shared $< -o $@ $(PYTHON_LDFLAGS) $(LIBPTHREAD)
 
 o/plugins/PythonEmbeddedInterpreter.o: plugins/PythonEmbeddedInterpreter.cc
-	$(CC) -c $(MCL_CFLAGS) $(PYTHON_CPPFLAGS) $< -o $@
+	$(CC) -fPIC -c $(MCL_CFLAGS) $(PYTHON_CPPFLAGS) $< -o $@
 
 install: $(EXE)
 	$(INSTALL) -d $(INSTALL_ROOT)$(prefix)/{bin,lib/mcl/{auto,contrib,plugins,sys}}
