esdiff -ru mcl-0.53.00/InputLine.cc mcl-0.53.00.new/InputLine.cc
--- mcl-0.53.00/InputLine.cc	2003-09-29 12:37:49.000000000 +0200
+++ mcl-0.53.00.new/InputLine.cc	2004-07-28 16:40:13.025986752 +0200
@@ -27,7 +27,7 @@
 
 History::History(int _id) : id (_id), current(0) {
 	max_history = config->getOption(opt_histsize);
-    strings = new (char*)[max_history];
+    strings = new char*[max_history];
     timestamps = new time_t[max_history];
 	
 	// Hmm, not sure about this
