--- OutputWindow.cc
+++ OutputWindow.cc
@@ -229,8 +229,8 @@
         highlight.len = strlen(s);
         
         // Show on the second line rather than under status bar
-        viewpoint = scrollback >? p-width;
-        viewpoint = viewpoint <? canvas;
+        viewpoint = scrollback > p-width ? scrollback : p-width;
+        viewpoint = viewpoint < canvas ? viewpoint : canvas;
         status->setf("Found string '%s'", s);
     }
 }
