--- src/caplayer.cpp.old	2007-08-30 02:52:11.000000000 +0200
+++ src/caplayer.cpp	2007-08-30 02:52:58.000000000 +0200
@@ -914,6 +914,8 @@
 void
 CAPlayer::display( int offsetX, int offsetY ) 
 {
+    if (frame >= CA_FPR)
+    	frame = CA_FPR - 1;
     sprite[frame]->draw ( (int)(x+offsetX - sprite[ frame ]->get_width()/2),
                                  (int)(y+offsetY - sprite[ frame ]->get_height()/2) );
 
