Index: softdevice-cvs/configure
===================================================================
--- softdevice-cvs.orig/configure
+++ softdevice-cvs/configure
@@ -173,9 +173,11 @@ test_ffmpeg()
   ffmpeg="yes";
 cat > ${TMPC} << EOF
 #define __STDC_CONSTANT_MACROS
+extern "C" {
 #include <stdlib.h>
 #include <avcodec.h>
 #include <avformat.h>
+}
 int main(void) {
 /*  if ( avcodec_build() != LIBAVCODEC_BUILD ) {
      fprintf(stderr,"Fatal Error! Libavcodec library build(%d) doesn't match avcodec.h build(%d)!!!\n",avcodec_build(),LIBAVCODEC_BUILD);
@@ -249,9 +251,11 @@ test_swscale()
 cat > ${TMPC} << EOF
 #define __STDC_CONSTANT_MACROS
 #include <stdlib.h>
+extern "C" {
 #include <avcodec.h>
 #include <avformat.h>
 #include <swscale.h>
+}
 int main(void) {
 /*  if ( avcodec_build() != LIBAVCODEC_BUILD ) {
      fprintf(stderr,"Fatal Error! Libavcodec library build(%d) doesn't match avcodec.h build(%d)!!!\n",avcodec_build(),LIBAVCODEC_BUILD);
Index: softdevice-cvs/VideoFilter.h
===================================================================
--- softdevice-cvs.orig/VideoFilter.h
+++ softdevice-cvs/VideoFilter.h
@@ -14,8 +14,10 @@
 
 #ifdef PP_LIBAVCODEC
 #include <stdint.h> //needed by postproc.h
+extern "C" {
   #include <postprocess.h>
   //#include <postproc/postprocess.h>
+}
 #endif //PP_LIBAVCODEC
 
 #include "video.h"
@@ -60,7 +62,9 @@ public:
 };
 
 #ifdef USE_SWSCALE
+extern "C" {
 #include <swscale.h>
+}
 #endif
 class cImageConvert : public cVideoFilter {
         sPicBuffer *outBuf;
