Index: softdevice-cvs/ShmClient.c
===================================================================
--- softdevice-cvs.orig/ShmClient.c
+++ softdevice-cvs/ShmClient.c
@@ -119,6 +119,16 @@ int main(int argc, char **argv) {
         xvRemote= new cShmRemote("softdevice-xv");
         //SetupStore.InitSetupStore();
         SetupStore->xvFullscreen=0;
+	if (argc>1) {
+		if (strcmp(argv[1], "-f") == 0) {
+			SetupStore->xvFullscreen=1;
+		} else if (strcmp(argv[1], "-h") == 0) {
+			printf ("Shared-Memory-Client for vdr-softdevice\n");
+			printf ("Options:\n");
+			printf ("  -f   Start fullscreen\n");
+			return 0;
+		}
+	}
 
         if ( !vout->Initialize()  ) {
                 fprintf(stderr,"Could not init video out!\n");
