# fixes segfault in soundwrapper if caleld without arguments
diff -Nur tucnak2-2.46.orig//src/soundwrapper.c tucnak2-2.46//src/soundwrapper.c
--- tucnak2-2.46.orig//src/soundwrapper.c	2011-05-26 08:47:36.000000000 +0200
+++ tucnak2-2.46//src/soundwrapper.c	2012-03-10 09:17:56.000000000 +0100
@@ -39,7 +39,7 @@
         return 1;
     }
 
-    if (argc >= 1){
+    if (argc > 1){
         freq = atoi(argv[1]);
         if (freq){ 
             arg=1193180/freq;
