From a8c7cf62811d688efeff49f8f9ab942b525ebfd4 Mon Sep 17 00:00:00 2001
From: Dan Fandrich <dan@coneharvesters.com>
Date: Wed, 26 Mar 2014 22:27:34 +0100
Subject: [PATCH] hostcheck: added a system include to define struct in_addr

---
 lib/hostcheck.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lib/hostcheck.c b/lib/hostcheck.c
index 162c522..42eb2ee 100644
--- a/lib/hostcheck.c
+++ b/lib/hostcheck.c
@@ -26,6 +26,10 @@
     defined(USE_GSKIT)
 /* these backends use functions from this file */
 
+#ifdef HAVE_NETINET_IN_H
+#include <netinet/in.h>
+#endif
+
 #include "hostcheck.h"
 #include "rawstr.h"
 #include "inet_pton.h"
-- 
1.9.1

