From 7bc33181d7e44aab35db01d4ac1560dcf408d5d9 Mon Sep 17 00:00:00 2001
From: Dominique Leuenberger <dimstar@opensuse.org>
Date: Mon, 21 Nov 2011 08:51:54 +0100
Subject: [PATCH] Build: link with libm. Fixes bgo#664439.

---
 configure.ac      |    1 +
 tools/Makefile.am |    7 ++++++-
 2 files changed, 7 insertions(+), 1 deletions(-)

diff --git a/configure.ac b/configure.ac
index b7d38c6..7ef4310 100644
--- a/configure.ac
+++ b/configure.ac
@@ -47,6 +47,7 @@ AC_ISC_POSIX
 AC_PROG_CC_STDC
 AC_STDC_HEADERS
 AC_C_BIGENDIAN
+LT_LIB_M
 
 GNOME_MAINTAINER_MODE_DEFINES
 GNOME_COMPILE_WARNINGS
diff --git a/tools/Makefile.am b/tools/Makefile.am
index ccb7e05..0bf1538 100644
--- a/tools/Makefile.am
+++ b/tools/Makefile.am
@@ -54,6 +54,7 @@ xpstopng_CFLAGS = \
 
 xpstopng_LDADD = \
 	libgxpstools.la		\
+	$(LIBM)                 \
 	$(LIBPNG_LIBS)
 endif # HAVE_LIBPNG
 
@@ -77,6 +78,7 @@ xpstojpeg_CFLAGS = \
 
 xpstojpeg_LDADD = \
 	libgxpstools.la		\
+	$(LIBM)                 \
 	$(LIBJPEG)
 endif # HAVE_LIBJPEG
 
@@ -99,6 +101,7 @@ xpstopdf_CFLAGS = \
 
 xpstopdf_LDADD = \
 	libgxpstools.la		\
+	$(LIBM)                 \
 	$(CAIRO_PDF_LIBS)
 endif # HAVE_CAIRO_PDF
 
@@ -121,6 +124,7 @@ xpstops_CFLAGS = \
 
 xpstops_LDADD = \
 	libgxpstools.la		\
+	$(LIBM)                 \
 	$(CAIRO_PS_LIBS)
 endif # HAVE_CAIRO_PS
 
@@ -143,5 +147,6 @@ xpstosvg_CFLAGS = \
 
 xpstosvg_LDADD = \
 	libgxpstools.la		\
+	$(LIBM)                 \
 	$(CAIRO_SVG_LIBS)
-endif # HAVE_CAIRO_SVG
\ No newline at end of file
+endif # HAVE_CAIRO_SVG
-- 
1.7.8.rc3

