diff -ru picpuz-2.1.1.orig/Makefile picpuz-2.1.1/Makefile
--- picpuz-2.1.1.orig/Makefile	2010-02-01 08:19:04.000000000 -0500
+++ picpuz-2.1.1/Makefile	2010-02-19 03:36:42.343749782 -0500
@@ -4,9 +4,8 @@
 VERSION = 2.1.1
 SOURCE = $(PROGRAM)-$(VERSION).cpp
 
-# replace CFLAGS and LFLAGS -O3 with -g for GDB debugging
-CFLAGS = -O3 -Wall -c `pkg-config --cflags gtk+-2.0`
-LFLAGS = -O3 -Wall -rdynamic `pkg-config --libs gtk+-2.0`
+CXXFLAGS += `pkg-config --cflags gtk+-2.0`
+LIBS = `pkg-config --libs gtk+-2.0`
 
 # use $PREFIX if defined, else assume /usr/local
 ifeq "$(PREFIX)" ""
@@ -20,14 +19,14 @@
 DOCDIR = $(PREFIX)/share/doc/$(PROGRAM)
 
 $(PROGRAM): $(PROGRAM).o zfuncs.o
-	@ g++ $(LFLAGS) -o $(PROGRAM) $(PROGRAM).o zfuncs.o
+	$(CXX) $(LDFLAGS) -o $(PROGRAM) $(PROGRAM).o zfuncs.o $(LIBS)
 
 $(PROGRAM).o: $(SOURCE)
-	@ g++ $(CFLAGS) -o $(PROGRAM).o $(SOURCE)		\
+	$(CXX) $(CXXFLAGS) -c -o $(PROGRAM).o $(SOURCE)		\
 	  -D "DATADIR=\"$(DATADIR)\"" -D "DOCDIR=\"$(DOCDIR)\""
 
 zfuncs.o: zfuncs.cpp
-	@ g++ $(CFLAGS) zfuncs.cpp	\
+	$(CXX) $(CXXFLAGS) -c zfuncs.cpp	\
 	  -D "DATADIR=\"$(DATADIR)\"" -D "DOCDIR=\"$(DOCDIR)\"" -D "BINDIR=\"$(BINDIR)\""
 
 install: $(PROGRAM)
