Index: nxcl/nxcl.cpp
===================================================================
--- nxcl/nxcl.cpp	(révision 520)
+++ nxcl/nxcl.cpp	(révision 545)
@@ -30,6 +30,7 @@
 #include <dbus/dbus.h>
 #include <X11/Xlib.h>
 }
+#include <stdlib.h>
 
 using namespace nxcl;
 using namespace std;
Index: lib/notQt.cpp
===================================================================
--- lib/notQt.cpp	(révision 520)
+++ lib/notQt.cpp	(révision 545)
@@ -29,6 +29,7 @@
 #include <sys/socket.h>	
 #include <signal.h>
 }
+#include <stdlib.h>
 
 #include "../config.h"
 #include "notQt.h"
Index: lib/nxclientlib.cpp
===================================================================
--- lib/nxclientlib.cpp	(révision 520)
+++ lib/nxclientlib.cpp	(révision 545)
@@ -43,6 +43,8 @@
     #include <sys/stat.h>
     #include <unistd.h>
 }
+#include <stdlib.h>
+#include <string.h>
 
 /*
  * On the location of nxproxy and nxssh binaries
