From: Julian Ospald <hasufell@gentoo.org>
Date: Thu Dec 27 22:58:59 UTC 2012
Subject: build system

	only enable Werror for development

--- CMakeLists.txt
+++ CMakeLists.txt
@@ -5,13 +5,16 @@
 
 cmake_minimum_required( VERSION 2.6 )
 
+# options
+option(WERROR "enable -Werror for development" OFF)
+option(ECHONEST_BUILD_TESTS "Build all unit tests" ON)
+
 set( CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DQT_NO_CAST_FROM_ASCII -DQT_NO_CAST_TO_ASCII" )
 
 find_package( Qt4 COMPONENTS QtCore QtNetwork QtTest REQUIRED )
 set( QT_DONT_USE_QTGUI TRUE )
 
-
 find_package(QJSON REQUIRED)
 
 include( ${QT_USE_FILE} )
@@ -26,7 +29,10 @@
 if (CMAKE_COMPILER_IS_GNUCXX)
     ADD_DEFINITIONS( -Wall -Wundef -Wcast-align -Wchar-subscripts -Wpointer-arith
        -Wwrite-strings -Wpacked -Wformat-security -Wmissing-format-attribute
-       -Wold-style-cast -Woverloaded-virtual -Wnon-virtual-dtor -Werror )
+       -Wold-style-cast -Woverloaded-virtual -Wnon-virtual-dtor )
+    if (WERROR)
+        ADD_DEFINITIONS( -Werror )
+    endif()
     if ( NOT WIN32 )
         add_definitions( -fvisibility=hidden )
     endif()
@@ -43,8 +49,6 @@
 endif(MSVC)
 
 
-option(ECHONEST_BUILD_TESTS "Build all unit tests" ON)
-
 if(ECHONEST_BUILD_TESTS)
   enable_testing()
 endif(ECHONEST_BUILD_TESTS)
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -40,10 +40,6 @@
 
 QT4_WRAP_CPP( ${LIBECHONEST_H} )
 
-if (CMAKE_COMPILER_IS_GNUCXX)
-    add_definitions( -Werror )
-endif (CMAKE_COMPILER_IS_GNUCXX)
-
 SET( OS_SPECIFIC_LINK_LIBRARIES "" )
 IF( APPLE OR MINGW )
   SET( OS_SPECIFIC_LINK_LIBRARIES ${QJSON_LIBRARIES} )
