diff -Naur ./src/hamster/widgets/activityentry.py ./src/hamster/widgets/activityentry.py
--- ./src/hamster/widgets/activityentry.py	2010-06-22 11:59:09.000000000 +0200
+++ ./src/hamster/widgets/activityentry.py	2010-07-21 13:12:15.000000000 +0200
@@ -20,10 +20,10 @@
 import gtk, gobject
 import datetime as dt
 
-from .hamster.configuration import runtime
+from ..configuration import runtime
 
-from .hamster import stuff
-from .hamster.stuff import format_duration
+from .. import stuff
+from ..stuff import format_duration
 
 class ActivityEntry(gtk.Entry):
     __gsignals__ = {
diff -Naur ./src/hamster/widgets/dateinput.py ./src/hamster/widgets/dateinput.py
--- ./src/hamster/widgets/dateinput.py	2010-06-04 18:53:24.000000000 +0200
+++ ./src/hamster/widgets/dateinput.py	2010-07-21 13:00:49.000000000 +0200
@@ -17,7 +17,7 @@
 # You should have received a copy of the GNU General Public License
 # along with Project Hamster.  If not, see <http://www.gnu.org/licenses/>.
 
-from .hamster.stuff import format_duration
+from ..stuff import format_duration
 import gtk
 import datetime as dt
 import calendar
diff -Naur ./src/hamster/widgets/dayline.py ./src/hamster/widgets/dayline.py
--- ./src/hamster/widgets/dayline.py	2010-06-22 11:59:09.000000000 +0200
+++ ./src/hamster/widgets/dayline.py	2010-07-21 13:02:06.000000000 +0200
@@ -20,8 +20,9 @@
 import gtk
 import gobject
 
-from .hamster import stuff
-from .hamster import graphics
+from .. import stuff
+from .. import graphics
+from ..configuration import conf
 
 import time
 import datetime as dt
diff -Naur ./src/hamster/widgets/facttree.py ./src/hamster/widgets/facttree.py
--- ./src/hamster/widgets/facttree.py	2010-06-22 11:59:09.000000000 +0200
+++ ./src/hamster/widgets/facttree.py	2010-07-21 13:02:43.000000000 +0200
@@ -20,9 +20,9 @@
 import gtk, gobject
 import datetime as dt
 
-from .hamster import stuff
-from .hamster.stuff import format_duration, format_activity
-from tags import Tag
+from .. import stuff
+from ..stuff import format_duration, format_activity
+from .tags import Tag
 
 import pango
 
diff -Naur ./src/hamster/widgets/__init__.py ./src/hamster/widgets/__init__.py
--- ./src/hamster/widgets/__init__.py	2010-06-22 11:59:09.000000000 +0200
+++ ./src/hamster/widgets/__init__.py	2010-07-21 12:59:34.000000000 +0200
@@ -21,21 +21,21 @@
 import gtk, pango
 
 # import our children
-from activityentry import ActivityEntry
-from dateinput import DateInput
-from timeinput import TimeInput
+from .activityentry import ActivityEntry
+from .dateinput import DateInput
+from .timeinput import TimeInput
 
-from timechart import TimeChart
+from .timechart import TimeChart
 
-from dayline import DayLine
+from .dayline import DayLine
 
-from tags import Tag
-from tags import TagBox
-from tags import TagsEntry
+from .tags import Tag
+from .tags import TagBox
+from .tags import TagsEntry
 
-from reportchooserdialog import ReportChooserDialog
+from .reportchooserdialog import ReportChooserDialog
 
-from facttree import FactTree
+from .facttree import FactTree
 
 # handy wrappers
 def add_hint(entry, hint):
diff -Naur ./src/hamster/widgets/tags.py ./src/hamster/widgets/tags.py
--- ./src/hamster/widgets/tags.py	2010-06-22 11:59:09.000000000 +0200
+++ ./src/hamster/widgets/tags.py	2010-07-21 13:18:47.000000000 +0200
@@ -21,9 +21,9 @@
 import pango, cairo
 from math import pi
 
-from .hamster import graphics
+from .. import graphics
 
-from .hamster.configuration import runtime
+from ..configuration import runtime
 
 class TagsEntry(gtk.Entry):
     __gsignals__ = {
diff -Naur ./src/hamster/widgets/timechart.py ./src/hamster/widgets/timechart.py
--- ./src/hamster/widgets/timechart.py	2010-06-22 11:59:09.000000000 +0200
+++ ./src/hamster/widgets/timechart.py	2010-07-21 13:16:51.000000000 +0200
@@ -20,7 +20,7 @@
 import os  # for locale
 import gtk, pango
 
-from .hamster import graphics
+from .. import graphics
 
 import time, datetime as dt
 import calendar
diff -Naur ./src/hamster/widgets/timeinput.py ./src/hamster/widgets/timeinput.py
--- ./src/hamster/widgets/timeinput.py	2010-06-04 18:53:24.000000000 +0200
+++ ./src/hamster/widgets/timeinput.py	2010-07-21 13:14:28.000000000 +0200
@@ -17,7 +17,7 @@
 # You should have received a copy of the GNU General Public License
 # along with Project Hamster.  If not, see <http://www.gnu.org/licenses/>.
 
-from .hamster.stuff import format_duration
+from ..stuff import format_duration
 import gtk
 from gtk import keysyms
 import datetime as dt
