diff --git a/qtiplot/qtiplot.pro b/qtiplot/qtiplot.pro
index 657ad0b..3655d66 100755
--- a/qtiplot/qtiplot.pro
+++ b/qtiplot/qtiplot.pro
@@ -235,8 +235,8 @@ win32: contains(CONFIG, Excel) {
 ###############################################################
 
 # At the very end: add global include- and lib path
-#unix:INCLUDEPATH += $$SYS_INCLUDEPATH
-#unix:LIBS += $$SYS_LIBS
+unix:INCLUDEPATH += $$SYS_INCLUDEPATH
+unix:LIBS += $$SYS_LIBS
 
 ###############################################################
 ############### Building QtiPlot as a browser plugin ##########
