=== modified file 'build.conf.example'
--- build.conf.example	2010-01-21 22:49:21 +0000
+++ build.conf.example	2010-02-14 15:04:17 +0000
@@ -69,6 +69,13 @@
 win32:QWT3D_LIBS = $$QTI_ROOT/3rdparty/qwtplot3d/lib/qwtplot3d.dll
 unix:QWT3D_LIBS = $$QTI_ROOT/3rdparty/qwtplot3d/lib/libqwtplot3d.a
 
+##########################################################
+## QTeXEngine - uncomment if you wish to use system copy
+# http://soft.proindependent.com/qtexengine/
+##########################################################
+#
+# QTEXENGINE_LIBS = -lQTeXEngine
+
 ###########################################################
 ## libxls - optional. you don't have to set these variables
 # http://libxls.sourceforge.net/

=== modified file 'qtiplot/qtiplot.pro'
--- qtiplot/qtiplot.pro	2010-01-21 22:49:21 +0000
+++ qtiplot/qtiplot.pro	2010-02-14 15:00:58 +0000
@@ -11,7 +11,6 @@
 # local copy included
 INCLUDEPATH       += ../3rdparty/liborigin
 INCLUDEPATH       += ../3rdparty/zlib
-INCLUDEPATH 	  += ../3rdparty/QTeXEngine/src
 
 # configurable
 INCLUDEPATH       += $$MUPARSER_INCLUDEPATH
@@ -24,6 +23,7 @@
 LIBS         += $$MUPARSER_LIBS
 LIBS         += $$QWT_LIBS
 LIBS         += $$QWT3D_LIBS
+LIBS         += $$QTEXENGINE_LIBS
 LIBS         += $$GSL_LIBS
 LIBS         += $$BOOST_LIBS
 
@@ -143,14 +143,16 @@
 include(src/table/table.pri)
 include(src/scripting/scripting.pri)
 
-###############################################################
-##################### TeX export: QTeXEngine ##################
-###############################################################
+isEmpty(QTEXENGINE_LIBS) {
+	###############################################################
+	##################### TeX export: QTeXEngine ##################
+	###############################################################
 
-INCLUDEPATH += ../3rdparty/QTeXEngine/src
-HEADERS 	+= ../3rdparty/QTeXEngine/src/QTeXEngine.h
-SOURCES     += ../3rdparty/QTeXEngine/src/QTeXPaintEngine.cpp
-SOURCES     += ../3rdparty/QTeXEngine/src/QTeXPaintDevice.cpp
+	INCLUDEPATH += ../3rdparty/QTeXEngine/src
+	HEADERS 	+= ../3rdparty/QTeXEngine/src/QTeXEngine.h
+	SOURCES     += ../3rdparty/QTeXEngine/src/QTeXPaintEngine.cpp
+	SOURCES     += ../3rdparty/QTeXEngine/src/QTeXPaintDevice.cpp
+}
 
 ###############################################################
 ##################### Scripting: PYTHON + SIP + PyQT ##########

