--- xchatosd-5.19/my_xosd.h.orig	2006-07-15 14:12:55.000000000 +0100
+++ xchatosd-5.19/my_xosd.h	2006-07-15 14:13:04.000000000 +0100
@@ -8,8 +8,8 @@
 
 class XOsd {
 	public:
-		XOsd::XOsd(int);
-		XOsd::~XOsd();
+		XOsd(int);
+		~XOsd();
 		int setPos(string);
 		int setVOffset(int);
 		int setHOffset(int);
--- xchatosd-5.19/xchatosd_conf.h.orig	2006-07-15 14:05:09.000000000 +0100
+++ xchatosd-5.19/xchatosd_conf.h	2006-07-15 14:05:40.000000000 +0100
@@ -11,8 +11,8 @@
 
 class XOsdConf {
 	public:
-		XOsdConf::XOsdConf();
-		XOsdConf::~XOsdConf();
+		XOsdConf();
+		~XOsdConf();
 		int read(string);
 		int write(string);
 		int getiValue(string);
