https://bugs.gentoo.org/show_bug.cgi?id=251449

--- xchatosd-5.19-orig/xchatosd_conf.cpp
+++ xchatosd-5.19/xchatosd_conf.cpp
@@ -1,4 +1,6 @@
 #include "xchatosd_conf.h"
+#include <cstdlib>
+#include <cstring>
 
 XOsdConf::XOsdConf()
 {
--- xchatosd-5.19-orig/xchatosd.cpp
+++ xchatosd-5.19/xchatosd.cpp
@@ -3,8 +3,7 @@
 #include <fstream>
 #include <iostream>
 #include <xosd.h>
-#include <stdio.h>
-#include <stdlib.h>
+#include <cstring>
 #include <langinfo.h>
 #include "xchatosd.h"
 #include "xchatosd_lang.h"
@@ -22,7 +21,7 @@ using namespace std;
 static xchat_plugin *ph;
 static XOsd *osd = NULL;
 static XOsdConf config;
-static char *codepage;
+static char const *codepage;
 #ifdef ICONV_LIB
 static iconv_t iconv_desc = (iconv_t) (-1);
 #endif
@@ -445,8 +444,8 @@ static int osd_notify_cb(char *word[], c
 
 extern "C" { int
 xchat_plugin_init(xchat_plugin * plugin_handle,
-		  char **plugin_name,
-		  char **plugin_desc, char **plugin_version, char *arg)
+		  char const **plugin_name,
+		  char const **plugin_desc, char const **plugin_version, char *arg)
 {
 	ph = plugin_handle;
 	XOSDconfRead();
