--- cdk-5.0-20090215.orig/aclocal.m4	2009-06-16 00:38:32.945387085 -0700
+++ cdk-5.0-20090215/aclocal.m4	2009-06-16 00:40:01.310004647 -0700
@@ -2425,7 +2425,7 @@
 		AC_MSG_ERROR(Cannot find libtool)
 	fi
 ])dnl
-	LIB_CREATE='${LIBTOOL} --mode=link ${CC} -rpath ${DESTDIR}${libdir} -version-info `cut -f1 ${srcdir}/VERSION` ${LIBTOOL_OPTS} -o'
+	LIB_CREATE='${LIBTOOL} --mode=link ${CC} -rpath ${DESTDIR}${libdir} -version-info `cut -f1 ${srcdir}/VERSION` ${LDFLAGS} ${LIBTOOL_OPTS} -o'
 	LIB_OBJECT='${OBJECTS:.o=.lo}'
 	LIB_SUFFIX=.la
 	LIB_CLEAN='${LIBTOOL} --mode=clean'
--- cdk-5.0-20090215.orig/configure	2009-06-16 00:38:32.964004839 -0700
+++ cdk-5.0-20090215/configure	2009-06-16 00:40:29.187005050 -0700
@@ -3869,7 +3869,7 @@
 echo "$as_me: error: Cannot find libtool" >&2;}
    { (exit 1); exit 1; }; }
 	fi
-	LIB_CREATE='${LIBTOOL} --mode=link ${CC} -rpath ${DESTDIR}${libdir} -version-info `cut -f1 ${srcdir}/VERSION` ${LIBTOOL_OPTS} -o'
+	LIB_CREATE='${LIBTOOL} --mode=link ${CC} -rpath ${DESTDIR}${libdir} -version-info `cut -f1 ${srcdir}/VERSION` ${LDFLAGS} ${LIBTOOL_OPTS} -o'
 	LIB_OBJECT='${OBJECTS:.o=.lo}'
 	LIB_SUFFIX=.la
 	LIB_CLEAN='${LIBTOOL} --mode=clean'
--- cdk-5.0-20090215.orig/Makefile.in	2009-06-16 00:38:32.951004532 -0700
+++ cdk-5.0-20090215/Makefile.in	2009-06-16 00:39:28.240044875 -0700
@@ -267,7 +267,7 @@
 all cdklib :: $(DEFAULT)
 
 $(CDKLIB) :: $(OBJECTS)
-	@echo creating $@; $(LIBTOOL_CREATE) $@ $(LIB_OBJECT)
+	@echo creating $@; $(LIBTOOL_CREATE) $@ $(LIB_OBJECT) $(LIBS)
 	@$(RANLIB) $@
 
 cdkshlib $(CDKSHLIB) :: $(CDKSHOBJS)
