From: Julian Ospald <hasufell@gentoo.org>
Date: Thu May 30 12:45:36 UTC 2013
Subject: provide options for unity/ayatana

--- geary-0.3.0/CMakeLists.txt
+++ geary-0.3.0/CMakeLists.txt
@@ -10,6 +10,8 @@
 
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake )
 
+include( CMakeDependentOption )
+
 #
 # Base bits
 #
@@ -31,6 +33,8 @@
 option(DEBUG "Build for debugging." OFF)
 option(ICON_UPDATE "Run gtk-update-icon-cache after the install." ON)
 option(DESKTOP_UPDATE "Run update-desktop-database after the install." ON)
+option(WITH_UNITY "Add unity support." OFF)
+cmake_dependent_option(WITH_LIBINDICATE "Enable minimal ayatana support through libappindicate" OFF "NOT WITH_UNITY" OFF)
 
 if (DEBUG)
     message(STATUS "Debug build")
@@ -45,9 +49,16 @@
 endif ()
 
 find_package(PkgConfig)
-pkg_check_modules(LIBINDICATE QUIET indicate-0.7>=0.6.1)
-pkg_check_modules(LIBUNITY QUIET unity>=5.12.0)
-pkg_check_modules(LIBMESSAGINGMENU QUIET messaging-menu>=12.10.2)
+
+if(WITH_UNITY)
+    pkg_check_modules(LIBINDICATE QUIET indicate-0.7>=0.6.1)
+    pkg_check_modules(LIBUNITY QUIET unity>=5.12.0)
+    pkg_check_modules(LIBMESSAGINGMENU QUIET messaging-menu>=12.10.2)
+endif()
+
+if(WITH_LIBINDICATE)
+    pkg_check_modules(LIBINDICATE QUIET indicate-0.7>=0.6.1)
+endif()
 
 # intl
 include(Gettext)
