--- src/Makefile.in~	2009-02-23 22:23:35.000000000 +0100
+++ src/Makefile.in	2010-08-08 02:03:16.000000000 +0200
@@ -46,6 +46,7 @@
 CGIDIR= 	$(exec_prefix)/cgi-bin
 CGIEXT=		@CGIEXT@
 MAILDIR=	$(exec_prefix)/mail
+LDFLAGS= @LDFLAGS@
 
 SHELL=		/bin/sh
 
@@ -91,13 +92,13 @@
 all: $(PROGRAMS)
 
 mailman: $(srcdir)/mail-wrapper.c $(COMMONOBJS)
-	$(CC) -I. $(MAIL_FLAGS) $(CFLAGS) $(COMMONOBJS) -o $@ $(srcdir)/mail-wrapper.c
+	$(CC) $(LDFLAGS) -I. $(MAIL_FLAGS) $(CFLAGS) $(COMMONOBJS) -o $@ $(srcdir)/mail-wrapper.c
 
 #addaliases: $(srcdir)/alias-wrapper.c $(COMMONOBJS)
 #	$(CC) -I. $(ALIAS_FLAGS) $(CFLAGS) -o $@ $(srcdir)/alias-wrapper.c
 
 $(CGI_PROGS): $(srcdir)/cgi-wrapper.c $(COMMONOBJS)
-	$(CC) -DSCRIPT="\"$@\"" -I. $(CGI_FLAGS) $(CFLAGS) $(COMMONOBJS) -o $@ $(srcdir)/cgi-wrapper.c
+	$(CC) $(LDFLAGS) -DSCRIPT="\"$@\"" -I. $(CGI_FLAGS) $(CFLAGS) $(COMMONOBJS) -o $@ $(srcdir)/cgi-wrapper.c
 
 common.o: $(srcdir)/common.c $(srcdir)/common.h Makefile
 	$(CC) -c -I. $(COMMON_FLAGS) $(CFLAGS) $(srcdir)/common.c
