*** audio.c	2009-04-04 11:53:29.000000000 -0700
--- ../line6usb/audio.c	2009-12-31 12:42:49.827996851 -0800
***************
*** 12,17 ****
--- 12,19 ----
  #include "driver.h"
  #include "audio.h"
  
+ #include <linux/version.h>
+ 
  #include <sound/core.h>
  #include <sound/initval.h>
  
***************
*** 28,37 ****
--- 30,48 ----
  	static int dev;
  	struct snd_card *card;
  
+ #if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 30)
  	card = snd_card_new(line6_index[dev], line6_id[dev], THIS_MODULE, 0);
  
  	if (card == NULL)
  		return -ENOMEM;
+ #else
+ 	int err;
+ 
+ 	err = snd_card_create(line6_index[dev], line6_id[dev], THIS_MODULE, 0,
+ 			      &card);
+ 	if (err < 0)
+ 		return err;
+ #endif
  
  	line6->card = card;
  
