--- nzbget-0.7.0/configure.ac.orig
+++ nzbget-0.7.0/configure.ac
@@ -204,7 +204,7 @@
 	[LIBVAL="no"])
 if test "$INCVAL" = "no" -o "$LIBVAL" = "no"; then
 	PKG_CHECK_MODULES(libxml2, libxml-2.0,
-		[LDFLAGS="${LDFLAGS} $libxml2_LIBS"]
+		[LIBS="${LIBS} $libxml2_LIBS"]
 		[CPPFLAGS="${CPPFLAGS} $libxml2_CFLAGS"]
 		[CFLAGS="${CFLAGS} $libxml2_CFLAGS"])
 fi
@@ -290,7 +290,7 @@
 		[LIBVAL="no"])
 	if test "$INCVAL" = "no" -o "$LIBVAL" = "no"; then
 		PKG_CHECK_MODULES(libsigc, sigc++-2.0,
-			[LDFLAGS="${LDFLAGS} $libsigc_LIBS"]
+			[LIBS="${LIBS} $libsigc_LIBS"]
 			[CPPFLAGS="${CPPFLAGS} $libsigc_CFLAGS"])
 	fi
 	
