--- sawfish-1.3.5.2/configure.in
+++ sawfish-1.3.5.2/configure.in
@@ -134,15 +134,15 @@
 dnl Checks for pango
 PKG_CHECK_MODULES(PANGO, pango >= 1.8.0,
   [if $PKG_CONFIG --exists pangoxft; then
-     AC_DEFINE(HAVE_PANGO_XFT, 1)
+     AC_DEFINE(HAVE_PANGO_XFT, 1, [Have pangoxft])
      PANGO_MODULES="pangoxft pangox"
      dnl for sawfish.gtk.widgets.font
      have_pango_xft=yes
      AC_SUBST(have_pango_xft)
    fi
    if test "x$PANGO_MODULES" != x; then
-     AC_DEFINE(HAVE_PANGO, 1)
+     AC_DEFINE(HAVE_PANGO, 1, [Have pango])
-     AC_DEFINE(HAVE_PANGO_X, 1)
+     AC_DEFINE(HAVE_PANGO_X, 1, [Have pangox])
      PANGO_LIBS=`$PKG_CONFIG --libs $PANGO_MODULES`
      PANGO_CFLAGS=`$PKG_CONFIG --cflags $PANGO_MODULES`
    fi], [: nothing])
