--- sawfish/lisp/sawfish/wm/util/prompt.jl
+++ sawfish/lisp/sawfish/wm/util/prompt.jl
@@ -121,10 +121,15 @@
   ;; Compilation hack: ensure that the compiler doesn't complain when
   ;; these are treated like functions and passed values.
   (eval-when-compile
-    (setq prompt-completion-fun (lambda (#!rest) nil)
-          prompt-validation-fun (lambda (#!rest) nil)
-          prompt-abbrev-fun (lambda (#!rest) nil)
-          prompt-display-fun (lambda (#!rest) nil)))
+    (progn
+      (defvar prompt-completion-fun nil)
+      (defvar prompt-validation-fun nil)
+      (defvar prompt-abbrev-fun nil)
+      (defvar prompt-display-fun nil)
+      (setq prompt-completion-fun (lambda (#!rest) nil)
+            prompt-validation-fun (lambda (#!rest) nil)
+            prompt-abbrev-fun (lambda (#!rest) nil)
+            prompt-display-fun (lambda (#!rest) nil))))
 
 
   ;; From merlin
