--- sawfish-1.6.2/configure.in
+++ sawfish-1.6.2/configure.in
@@ -81,11 +81,17 @@
 XFT_LIBS=""
 XFT_CFLAGS=""
 
+AC_ARG_WITH([xinerama], [AS_HELP_STRING([--with-xinerama], [Include Xinerama support])], [], [with_xinerama=yes])
+
+if test "$with_xinerama" != "no"; then
+
 PKG_CHECK_MODULES(libXineRama, xinerama >= 1.0
 		  ,XINERAMA_LIBS="-lXinerama"
 		  AC_DEFINE(HAVE_X11_EXTENSIONS_XINERAMA_H, 1, [Have xinerama])
 		  ,AC_MSG_ERROR([cannot locate libXineRama]))
 
+fi
+
 PKG_CHECK_MODULES(libXRandR, xrandr >= 1.0
 		  ,XRANDR_LIBS="-lXrandr"
 		  AC_DEFINE(HAVE_X11_EXTENSIONS_XRANDR_H, 1, [Have xrandr])
