diff --git a/Makefile.am b/Makefile.am
index 058c2cb..882fd23 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -9,7 +9,7 @@ EXTRA_DIST = LICENSE Changes libev.m4 autogen.sh \
 
 man_MANS = ev.3
 
-include_HEADERS = ev.h ev++.h event.h
+include_HEADERS = ev.h ev++.h
 
 lib_LTLIBRARIES = libev.la
 
diff --git a/configure.ac b/configure.ac
index 20665bb..f3eb882 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3,7 +3,6 @@ AC_CONFIG_SRCDIR([ev_epoll.c])
 
 AM_INIT_AUTOMAKE(libev,4.01) dnl also update ev.h!
 AC_CONFIG_HEADERS([config.h])
-AM_MAINTAINER_MODE
 
 AC_PROG_INSTALL
 AC_PROG_LIBTOOL
