diff -Naur libev-3.42.orig/Makefile.am libev-3.42/Makefile.am
--- libev-3.42.orig/Makefile.am	2008-05-11 09:12:36.000000000 +0900
+++ libev-3.42/Makefile.am	2008-06-22 09:23:05.000000000 +0900
@@ -9,7 +9,7 @@
 
 man_MANS = ev.3
 
-include_HEADERS = ev.h ev++.h event.h
+include_HEADERS = ev.h ev++.h
 
 lib_LTLIBRARIES = libev.la
 
diff -Naur libev-3.42.orig/configure.ac libev-3.42/configure.ac
--- libev-3.42.orig/configure.ac	2008-06-17 19:12:02.000000000 +0900
+++ libev-3.42/configure.ac	2008-06-18 01:23:21.000000000 +0900
@@ -3,15 +3,10 @@
 
 AM_INIT_AUTOMAKE(libev,3.42)
 AC_CONFIG_HEADERS([config.h])
-AM_MAINTAINER_MODE
 
 AC_PROG_INSTALL
 AC_PROG_LIBTOOL
 
-if test "x$GCC" = xyes ; then
-  CFLAGS="$CFLAGS -O3"
-fi
-
 m4_include([libev.m4])
 
 AC_CONFIG_FILES([Makefile])
