From 2a14564c8e1677c002ffac8643f706b334050988 Mon Sep 17 00:00:00 2001
From: Nick Schermer <nick@xfce.org>
Date: Sat, 08 Sep 2012 09:55:19 +0000
Subject: Protect against possible null pointers (bug #9109).

(cherry picked from commit 424418316733e3f8b9b17cd7345fe6d967b06ef8)
---
diff --git a/src/appfinder-category-model.c b/src/appfinder-category-model.c
index 483c4cc..3f3a060 100644
--- a/src/appfinder-category-model.c
+++ b/src/appfinder-category-model.c
@@ -589,6 +589,7 @@ xfce_appfinder_category_model_icon_theme_changed (XfceAppfinderCategoryModel *mo
   for (li = model->categories, idx = 0; li != NULL; li = li->next, idx++)
     {
       item = li->data;
+      appfinder_assert (item != NULL);
 
       if (item->pixbuf != NULL)
         {
diff --git a/src/appfinder-window.c b/src/appfinder-window.c
index 155e011..4f99c6a 100644
--- a/src/appfinder-window.c
+++ b/src/appfinder-window.c
@@ -1453,8 +1453,11 @@ xfce_appfinder_window_icon_theme_changed (XfceAppfinderWindow *window)
   window->icon_find = xfce_appfinder_model_load_pixbuf (GTK_STOCK_FIND, XFCE_APPFINDER_ICON_SIZE_48);
 
   /* drop cached pixbufs */
-  xfce_appfinder_model_icon_theme_changed (window->model);
-  xfce_appfinder_category_model_icon_theme_changed (window->category_model);
+  if (G_LIKELY (window->model != NULL))
+    xfce_appfinder_model_icon_theme_changed (window->model);
+
+  if (G_LIKELY (window->category_model != NULL))
+    xfce_appfinder_category_model_icon_theme_changed (window->category_model);
 
   /* update state */
   xfce_appfinder_window_entry_changed (window);
--
cgit v0.9.0.3
