diff -NrU5 bobotpp-2.2.2.orig/source/Channel.C bobotpp-2.2.2/source/Channel.C
--- bobotpp-2.2.2.orig/source/Channel.C	2008-04-18 16:19:56.000000000 +0200
+++ bobotpp-2.2.2/source/Channel.C	2008-04-18 16:31:08.000000000 +0200
@@ -23,10 +23,11 @@
 
 #include "Macros.H"
 #include "Utils.H"
 #include "Channel.H"
 #include "StringTokenizer.H"
+#include <cstdlib>
 #ifdef USESCRIPTS
 #include "Interp.H"
 #endif
 
 Channel::Channel(ServerConnection *c,
diff -NrU5 bobotpp-2.2.2.orig/source/DCCManager.H bobotpp-2.2.2/source/DCCManager.H
--- bobotpp-2.2.2.orig/source/DCCManager.H	2008-04-18 16:19:56.000000000 +0200
+++ bobotpp-2.2.2/source/DCCManager.H	2008-04-18 16:32:55.000000000 +0200
@@ -22,10 +22,11 @@
 #include <list>
 #include "DCCPerson.H"
 #include "DCCConnection.H"
 #include "UserCommands.H"
 #include "String.H"
+#include <sys/types.h>
 
 typedef std::map<String, DCCPerson*, std::less<String> > DCC_MAP;
 
 // Manages all active DCC connections and allows you to send a message
 // to each one using the string representation of the user's address
diff -NrU5 bobotpp-2.2.2.orig/source/Parser.C bobotpp-2.2.2/source/Parser.C
--- bobotpp-2.2.2.orig/source/Parser.C	2008-04-18 16:19:56.000000000 +0200
+++ bobotpp-2.2.2/source/Parser.C	2008-04-18 16:34:48.000000000 +0200
@@ -21,10 +21,11 @@
 #include "config.h"
 #endif
 
 #include <sys/types.h>
 #include <netinet/in.h>
+#include <cstdlib>
 
 #include "StringTokenizer.H"
 #include "Parser.H"
 #include "UserCommands.H"
 #include "Commands.H"
diff -NrU5 bobotpp-2.2.2.orig/source/ServerQueue.C bobotpp-2.2.2/source/ServerQueue.C
--- bobotpp-2.2.2.orig/source/ServerQueue.C	2008-04-18 16:19:56.000000000 +0200
+++ bobotpp-2.2.2/source/ServerQueue.C	2008-04-18 16:36:01.000000000 +0200
@@ -19,11 +19,11 @@
 
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif
 
-//#include <limits>
+#include <climits>
 #include "ServerQueue.H"
 #include "Utils.H"
 
 ServerQueue::ServerQueue(Socket * s, bool d)
   : Queue(s,d), penalty(0)
diff -NrU5 bobotpp-2.2.2.orig/source/ShitList.C bobotpp-2.2.2/source/ShitList.C
--- bobotpp-2.2.2.orig/source/ShitList.C	2008-04-18 16:19:56.000000000 +0200
+++ bobotpp-2.2.2/source/ShitList.C	2008-04-18 16:37:12.000000000 +0200
@@ -24,10 +24,11 @@
 #include <fstream>
 
 #include "ShitList.H"
 #include "StringTokenizer.H"
 #include "Utils.H"
+#include <cstdlib>
 
 ShitList::ShitList(String filename)
   : listFileName(filename)
 {
 #ifdef HAVE_STL_CLEAR
diff -NrU5 bobotpp-2.2.2.orig/source/String.C bobotpp-2.2.2/source/String.C
--- bobotpp-2.2.2.orig/source/String.C	2008-04-18 16:19:56.000000000 +0200
+++ bobotpp-2.2.2/source/String.C	2008-04-18 16:39:18.000000000 +0200
@@ -21,10 +21,11 @@
 
 #include <cstring>
 #include <cctype>
 #include <sstream>
 #include <iostream>
+#include <cstdlib>
 
 String::String()
 {
 }
 
diff -NrU5 bobotpp-2.2.2.orig/source/UserCommands.C bobotpp-2.2.2/source/UserCommands.C
--- bobotpp-2.2.2.orig/source/UserCommands.C	2008-04-18 16:19:56.000000000 +0200
+++ bobotpp-2.2.2/source/UserCommands.C	2008-04-18 16:40:39.000000000 +0200
@@ -30,10 +30,11 @@
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #include <netdb.h>
 #include <cstdlib>
+#include <cstring>
 #ifndef _X_OPEN_SOURCE
 #define _X_OPEN_SOURCE
 #endif
 #ifndef _X_OPEN_SOURCE_EXTENDED
 #define _X_OPEN_SOURCE_EXTENDED 1
diff -NrU5 bobotpp-2.2.2.orig/source/UserList.C bobotpp-2.2.2/source/UserList.C
--- bobotpp-2.2.2.orig/source/UserList.C	2008-04-18 16:19:56.000000000 +0200
+++ bobotpp-2.2.2/source/UserList.C	2008-04-18 16:41:08.000000000 +0200
@@ -17,10 +17,11 @@
 // Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
 // 02110-1301, USA.
 
 #include <fstream>
 #include <iostream>
+#include <cstdlib>
 
 #include "UserList.H"
 #include "StringTokenizer.H"
 #include "Utils.H"
 
