diff -ur uim-1.5.4.orig/sigscheme/src/read.c uim-1.5.4/sigscheme/src/read.c
--- uim-1.5.4.orig/sigscheme/src/read.c	2008-03-01 23:30:43.000000000 +0900
+++ uim-1.5.4/sigscheme/src/read.c	2008-12-10 02:09:21.000000000 +0900
@@ -918,6 +918,7 @@
     int err;
     ScmLBuf(char) lbuf;
     char init_buf[SCM_INITIAL_SYMBOL_BUF_SIZE];
+    init_buf[0] = '\0';
 
     CDBG((SCM_DBG_PARSER, "read_symbol"));
 
