diff -Naur uim-1.5.4.orig/notify/Makefile.am uim-1.5.4/notify/Makefile.am
--- uim-1.5.4.orig/notify/Makefile.am	2008-03-29 22:33:30.000000000 +0900
+++ uim-1.5.4/notify/Makefile.am	2008-10-30 01:38:43.000000000 +0900
@@ -14,6 +14,8 @@
 endif
 
 if KNOTIFY3
+QT_CXXFLAGS = $(UIM_QT_CXXFLAGS)
+QT_LDFLAGS  = $(UIM_QT_LDFLAGS)
 uimnotify_plugin_LTLIBRARIES += libuimnotify-knotify3.la
 libuimnotify_knotify3_la_SOURCES = uim-knotify3.cc
 libuimnotify_knotify3_la_CXXFLAGS = -I@KNOTIFY3_INCLUDE_DIR@ $(QT_CXXFLAGS)
diff -Naur uim-1.5.4.orig/notify/Makefile.in uim-1.5.4/notify/Makefile.in
--- uim-1.5.4.orig/notify/Makefile.in	2008-10-24 21:11:28.000000000 +0900
+++ uim-1.5.4/notify/Makefile.in	2008-10-30 01:45:46.000000000 +0900
@@ -369,6 +369,8 @@
 @LIBNOTIFY_TRUE@libuimnotify_libnotify_la_CFLAGS = @LIBNOTIFY_CFLAGS@
 @LIBNOTIFY_TRUE@libuimnotify_libnotify_la_LDFLAGS = -avoid-version -module
 @LIBNOTIFY_TRUE@libuimnotify_libnotify_la_LIBADD = @LIBNOTIFY_LIBS@ $(top_builddir)/replace/libreplace.la
+@KNOTIFY3_TRUE@QT_CXXFLAGS = $(UIM_QT_CXXFLAGS)
+@KNOTIFY3_TRUE@QT_LDFLAGS = $(UIM_QT_LDFLAGS)
 @KNOTIFY3_TRUE@libuimnotify_knotify3_la_SOURCES = uim-knotify3.cc
 @KNOTIFY3_TRUE@libuimnotify_knotify3_la_CXXFLAGS = -I@KNOTIFY3_INCLUDE_DIR@ $(QT_CXXFLAGS)
 @KNOTIFY3_TRUE@libuimnotify_knotify3_la_LDFLAGS = -avoid-version -module -L@KNOTIFY3_LIB_DIR@ $(QT_LDFLAGS)
