# include proper python headers 

--- charm-5.9/src/scripts/configure	2005-06-01 03:46:06.000000000 +0000
+++ charm-5.9-backup/src/scripts/configure	2005-09-27 14:28:48.000000000 +0000
@@ -1970,11 +1970,12 @@
 fi
 
 #### test if Python headers are installed ####
+PYTHON_VERSION=`python -V 2>&1 | awk {'print $2'} | awk -F. {'print $1"."$2'}`
 cat > $t <<EOT
-#include "python/Python.h"
-#include "python/compile.h"
-#include "python/eval.h"
-#include "python/node.h"
+#include "python${PYTHON_VERSION}/Python.h"
+#include "python${PYTHON_VERSION}/compile.h"
+#include "python${PYTHON_VERSION}/eval.h"
+#include "python${PYTHON_VERSION}/node.h"
 
 int main() {
     Py_Initialize();
@@ -1983,7 +1984,6 @@
     PyCodeObject *program = PyNode_Compile(programNode, "");
 }
 EOT
-PYTHON_VERSION=`python -V 2>&1 | awk {'print $2'} | awk -F. {'print $1"."$2'}`
 test_link "whether Python is installed" "yes" "no" "-lpython$PYTHON_VERSION -lpthread -lutil -ldl"
 
 cat >>confdefs.h <<_ACEOF
