diff -Naur charm-5.9/examples/pose/ASIM/Makefile charm-5.9-backup/examples/pose/ASIM/Makefile
--- charm-5.9/examples/pose/ASIM/Makefile	2004-10-22 04:38:26.000000000 +0000
+++ charm-5.9-backup/examples/pose/ASIM/Makefile	2005-09-27 03:27:08.000000000 +0000
@@ -12,8 +12,8 @@
 OPTS=-O#-DCMK_OPTIMIZE=1 -DCMK_MEMCHECKS_OFF=1
 
 CHARMBASE=../../..
-CHARMBIN=$(CHARMBASE)/bin
-CHARMINC=$(CHARMBASE)/include
+CHARMBIN=/usr/bin
+CHARMINC=/usr/include/charm-5.9
 
 CHARMC=$(CHARMBIN)/charmc -I$(CHARMINC)/pose $(OPTS)
 
diff -Naur charm-5.9/examples/pose/BigNetSim/Makefile charm-5.9-backup/examples/pose/BigNetSim/Makefile
--- charm-5.9/examples/pose/BigNetSim/Makefile	2004-10-17 22:48:10.000000000 +0000
+++ charm-5.9-backup/examples/pose/BigNetSim/Makefile	2005-09-27 03:27:08.000000000 +0000
@@ -16,8 +16,8 @@
 
 CHARMBASE=../../../
 #CHARMBASE=/expand8/home/gzheng/old/charm-6-8/net-linux/
-CHARMBIN=$(CHARMBASE)/bin
-CHARMINC=$(CHARMBASE)/include
+CHARMBIN=/usr/bin
+CHARMINC=/usr/include/charm-5.9
 
 ifeq ($(SEQUENTIAL),1)
 CHARMC=$(CHARMBIN)/charmc -I$(CHARMINC)/pose $(OPTS) -DSEQUENTIAL_POSE=1
diff -Naur charm-5.9/examples/pose/ChunkSim/Makefile charm-5.9-backup/examples/pose/ChunkSim/Makefile
--- charm-5.9/examples/pose/ChunkSim/Makefile	2004-10-12 17:12:26.000000000 +0000
+++ charm-5.9-backup/examples/pose/ChunkSim/Makefile	2005-09-27 03:27:08.000000000 +0000
@@ -12,8 +12,8 @@
 OPTS=-O#-DCMK_OPTIMIZE=1 -DCMK_MEMCHECKS_OFF=1
 
 CHARMBASE=../../../
-CHARMBIN=$(CHARMBASE)/bin
-CHARMINC=$(CHARMBASE)/include
+CHARMBIN=/usr/bin
+CHARMINC=/usr/include/charm-5.9
 
 CHARMC=$(CHARMBIN)/charmc -I$(CHARMINC)/pose $(OPTS)
 
@@ -109,4 +109,4 @@
 	$(CHARMBIN)/etrans.pl Worker
 
 test:
-	./pgm 1000 10 1 RANDOM SPARSE 50 -gf 344
\ No newline at end of file
+	./pgm 1000 10 1 RANDOM SPARSE 50 -gf 344
diff -Naur charm-5.9/examples/pose/HiSim/Makefile.common charm-5.9-backup/examples/pose/HiSim/Makefile.common
--- charm-5.9/examples/pose/HiSim/Makefile.common	2004-11-02 07:29:40.000000000 +0000
+++ charm-5.9-backup/examples/pose/HiSim/Makefile.common	2005-09-27 03:27:08.000000000 +0000
@@ -1,8 +1,8 @@
 
 CHARMBASE=../../../../
 #CHARMBASE=/expand8/home/gzheng/tmp/pose/charm/net-linux
-CHARMBIN=$(CHARMBASE)/bin
-CHARMINC=$(CHARMBASE)/include
+CHARMBIN=/usr/bin
+CHARMINC=/usr/include/charm-5.9
 ETRANS=$(CHARMBIN)/etrans.pl
 OPTS=-g
 
diff -Naur charm-5.9/examples/pose/LBSim/Makefile charm-5.9-backup/examples/pose/LBSim/Makefile
--- charm-5.9/examples/pose/LBSim/Makefile	2004-11-04 04:30:45.000000000 +0000
+++ charm-5.9-backup/examples/pose/LBSim/Makefile	2005-09-27 03:27:08.000000000 +0000
@@ -12,8 +12,8 @@
 OPTS=-O#-DCMK_OPTIMIZE=1 -DCMK_MEMCHECKS_OFF=1
 
 CHARMBASE=../../../
-CHARMBIN=$(CHARMBASE)/bin
-CHARMINC=$(CHARMBASE)/include
+CHARMBIN=/usr/bin
+CHARMINC=/usr/include/charm-5.9
 
 CHARMC=$(CHARMBIN)/charmc -I$(CHARMINC)/pose $(OPTS)
 
diff -Naur charm-5.9/examples/pose/Ring/Makefile charm-5.9-backup/examples/pose/Ring/Makefile
--- charm-5.9/examples/pose/Ring/Makefile	2005-02-08 18:58:05.000000000 +0000
+++ charm-5.9-backup/examples/pose/Ring/Makefile	2005-09-27 03:27:08.000000000 +0000
@@ -12,8 +12,8 @@
 OPTS=-O#-DCMK_OPTIMIZE=1 -DCMK_MEMCHECKS_OFF=1
 
 CHARMBASE=../../..
-CHARMBIN=$(CHARMBASE)/bin
-CHARMINC=$(CHARMBASE)/include
+CHARMBIN=/usr/bin/
+CHARMINC=/usr/include/charm-5.9
 
 CHARMC=$(CHARMBIN)/charmc -I$(CHARMINC)/pose $(OPTS)
 
diff -Naur charm-5.9/examples/pose/SimBenchmark/Makefile charm-5.9-backup/examples/pose/SimBenchmark/Makefile
--- charm-5.9/examples/pose/SimBenchmark/Makefile	2005-03-17 22:31:41.000000000 +0000
+++ charm-5.9-backup/examples/pose/SimBenchmark/Makefile	2005-09-27 03:27:08.000000000 +0000
@@ -12,8 +12,8 @@
 OPTS=-O#-DCMK_OPTIMIZE=1 -DCMK_MEMCHECKS_OFF=1
 
 CHARMBASE=../../..
-CHARMBIN=$(CHARMBASE)/bin
-CHARMINC=$(CHARMBASE)/include
+CHARMBIN=/usr/bin
+CHARMINC=/usr/include/charm-5.9
 
 CHARMC=$(CHARMBIN)/charmc -I$(CHARMINC)/pose $(OPTS)
 
