# compile with -fPIC to avoid TEXTRELs

--- charm-5.9/src/arch/net-linux/conv-mach.sh	2005-04-22 01:14:55.000000000 +0000
+++ charm-5.9-backup/src/arch/net-linux/conv-mach.sh	2005-09-27 18:07:10.000000000 +0000
@@ -1,8 +1,8 @@
 CMK_CPP_CHARM="/lib/cpp -P"
 CMK_CPP_C="gcc -E"
-CMK_CC="gcc "
-CMK_CXX="g++ "
-CMK_CXXPP="$CMK_CXX -x c++ -E "
+CMK_CC="gcc -fPIC"
+CMK_CXX="g++ -fPIC -Wno-deprecated"
+CMK_CXXPP="$CMK_CXX -E "
 CMK_CF77="g77 "
 CMK_CF90="f90 "
 CMK_CF90_FIXED="$CMK_CF90 -W132 "
@@ -10,12 +10,12 @@
 CMK_LIBS="-lckqt"
 #CMK_LD="$CMK_CC -Wl,--allow-multiple-definition "
 #CMK_LDXX="$CMK_CXX -Wl,--allow-multiple-definition "
-CMK_LD="$CMK_CC "
-CMK_LDXX="$CMK_CXX "
+CMK_LD="gcc "
+CMK_LDXX="g++ "
 CMK_LD_SHARED="-shared"
 CMK_LD_LIBRARY_PATH="-Wl,-rpath,$CHARMLIBSO/"
 CMK_XIOPTS=""
-CMK_F90LIBS="-L/usr/absoft/lib -L/opt/absoft/lib -lf90math -lfio -lU77 -lf77math "
+CMK_F90LIBS=""
 CMK_F77LIBS="-lg2c "
 CMK_MOD_NAME_ALLCAPS=1
 CMK_MOD_EXT="mod"
