diff -Naur charm-5.9/src/ck-com/MsgPacker.h charm-5.9-new/src/ck-com/MsgPacker.h
--- charm-5.9/src/ck-com/MsgPacker.h	2005-03-20 23:21:51.000000000 -0600
+++ charm-5.9-new/src/ck-com/MsgPacker.h	2006-04-19 17:16:05.000000000 -0500
@@ -83,7 +83,7 @@
     
     //Takes a queue of envelopes as char* ptrs and not charm message holders
     //Used by mesh streaming strategy
-    MsgPacker::MsgPacker(CkQ<char *> &msgq, int n_msgs);
+    MsgPacker(CkQ<char *> &msgq, int n_msgs);
     
     void getMessage(CombinedMessage *&msg, int &size);
     static void deliver(CombinedMessage *cmb_msg);
diff -Naur charm-5.9/src/ck-core/charisma.h charm-5.9-new/src/ck-core/charisma.h
--- charm-5.9/src/ck-core/charisma.h	2003-03-20 18:30:50.000000000 -0600
+++ charm-5.9-new/src/ck-core/charisma.h	2006-04-19 17:19:34.000000000 -0500
@@ -295,6 +295,7 @@
 class CharismaInPort
 {
   public:
+    virtual ~CharismaInPort() {}
     virtual void send(void *msg, int len) = 0;
     void _create(const char *name)
     {
@@ -307,6 +308,7 @@
   protected:
     CharismaInPort *inport;
   public:
+    virtual ~CharismaOutPort() {}
     virtual void emitData(void *data, int len)
     {
       inport->send(data, len);
diff -Naur charm-5.9/src/ck-core/cklocation.h charm-5.9-new/src/ck-core/cklocation.h
--- charm-5.9/src/ck-core/cklocation.h	2004-10-30 11:35:25.000000000 -0500
+++ charm-5.9-new/src/ck-core/cklocation.h	2006-04-19 17:21:24.000000000 -0500
@@ -431,6 +431,7 @@
 /// Abstract superclass of all array manager objects 
 class CkArrMgr {
 public:
+	virtual ~CkArrMgr() {}
 	/// Insert this initial element on this processor
 	virtual void insertInitial(const CkArrayIndex &idx,void *ctorMsg, int local=1)=0;
