--- Makefile.am.old	2010-10-18 10:28:31.000000000 +0200
+++ Makefile.am	2010-10-18 10:29:12.000000000 +0200
@@ -1,9 +1,4 @@
 cylindrix_CXXFLAGS = -Wall
-cylindrix_LDFLAGS = `allegro-config --libs`
-
-CXXFLAGS = $(cylindrix_CXXFLAGS)
-LDFLAGS = $(cylindrix_LDFLAGS)
-
 
 bin_PROGRAMS = cylindrix
 
--- configure.in.old	2010-10-18 10:56:50.000000000 +0200
+++ configure.in	2010-10-18 10:57:32.000000000 +0200
@@ -37,4 +37,5 @@
 	AC_DEFINE(__MSWINDOWS__,1,0) 
 fi
 
+LIBS=$(allegro-config --libs)
 AC_OUTPUT(Makefile)
