diff -Nuar --exclude '*~' audit-1.5.4.orig/swig/auditswig.i audit-1.5.4/swig/auditswig.i
--- audit-1.5.4.orig/swig/auditswig.i	2007-06-14 08:17:00.000000000 -0700
+++ audit-1.5.4/swig/auditswig.i	2007-07-14 23:43:41.790390709 -0700
@@ -26,6 +26,7 @@
 %define __signed__
 signed
 %enddef
+#define __attribute(x) /*nothing*/
 %include "/usr/include/asm/types.h"
 %include "/usr/include/linux/audit.h"
 #define __extension__ /*nothing*/
