--- audit-1.6.8/Makefile.am.orig	2008-02-20 14:33:51.034127088 -0800
+++ audit-1.6.8/Makefile.am	2008-02-20 14:33:57.640162864 -0800
@@ -21,8 +21,7 @@
 #   Rickard E. (Rik) Faith <faith@redhat.com>
 #
 
-SUBDIRS = lib auparse src/mt src audisp tools swig bindings init.d 
-	docs 
+SUBDIRS = lib auparse src/mt src audisp tools swig bindings init.d docs 
 EXTRA_DIST = ChangeLog AUTHORS NEWS README README-install audit.spec \
 	contrib/capp.rules contrib/nispom.rules contrib/lspp.rules \
 	contrib/skeleton.c contrib/avc_snap contrib/avc_syslog \
