 setup.py |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

--- a/setup.py
+++ b/setup.py
@@ -48,7 +48,7 @@ setup(name="pip",
           'https://github.com/pypa/virtualenv/tarball/master#egg=virtualenv-1.5.2.post2',
           ],
       packages=['pip', 'pip.commands', 'pip.vcs'],
-      entry_points=dict(console_scripts=['pip=pip:main', 'pip-%s=pip:main' % sys.version[:3]]),
+      entry_points=dict(console_scripts=['pip=pip:main']),
       test_suite='nose.collector',
       tests_require=['nose', 'virtualenv==1.5.2.post2', 'scripttest==1.1.1', 'mock'],
       zip_safe=False)
