--- pip-1.3.1/setup.py
+++ pip-1.3.1/setup.py
@@ -51,7 +51,7 @@
       license='MIT',
       packages=['pip', 'pip.commands', 'pip.vcs', 'pip.backwardcompat'],
       package_data={'pip': ['*.pem']},
-      entry_points=dict(console_scripts=['pip=pip:main', 'pip-%s=pip:main' % sys.version[:3]]),
+      entry_points=dict(console_scripts=['pip=pip:main']),
       test_suite='nose.collector',
       tests_require=tests_require,
       zip_safe=False,
