--- /tmp/GNS3-0.7-src/src/GNS3/Ui/ConfigurationPages/Page_PreferencesDynamips.py.old    2010-04-09 14:58:40.663821065 +0200
+++ /tmp/GNS3-0.7-src/src/GNS3/Ui/ConfigurationPages/Page_PreferencesDynamips.py        2010-04-09 14:59:24.282849072 +0200
@@ -55,6 +55,8 @@
         # Default path to dynamips executable
         if self.conf.path == '' and sys.platform.startswith('win'):
             self.conf.path = unicode('dynamips-wxp.exe')
+        else:
+            self.conf.path = unicode('/usr/bin/dynamips')
             
         # Default path to working directory
         if self.conf.workdir == '':

