--- Locale-Maketext-1.13/lib/Locale/Maketext.pm
+++ Locale-Maketext-1.13/lib/Locale/Maketext.pm
@@ -401,7 +401,7 @@
     my $module = $_[0];   # ASSUME sane module name!
     { no strict 'refs';
         return($tried{$module} = 1)
-        if defined(%{$module . '::Lexicon'}) or defined(@{$module . '::ISA'});
+        if %{$module . '::Lexicon'} or @{$module . '::ISA'};
         # weird case: we never use'd it, but there it is!
     }
 
