diff -ruN NX~/scripts/setup/nxnode NX/scripts/setup/nxnode
--- NX~/scripts/setup/nxnode	2010-08-04 12:34:28.000000000 +0200
+++ NX/scripts/setup/nxnode	2010-10-20 12:34:05.418016564 +0200
@@ -25,7 +25,7 @@
 
 WRAPPER="$NODE_ROOT/bin/nxuexec"
 
-CUPS_PATH="/usr /opt/sfw/cups /usr/local /usr/lib /opt/csw"
+CUPS_PATH="/usr /opt/sfw/cups /usr/local /usr/lib /opt/csw /usr/libexec"
 SMB_PATH="/sbin /bin /usr/sbin /usr/bin /usr/local/sbin /usr/local/bin"
 CUPS_CERTS_DIR="/var/run/cups/certs /var/lib/cups/certs /etc/cups/certs"
 
