diff -ruN work.orig/ganeti-2.0.3/lib/hypervisor/hv_kvm.py work/ganeti-2.0.3/lib/hypervisor/hv_kvm.py
--- a/ganeti-2.0.3/lib/hypervisor/hv_kvm.py	2009-09-23 09:56:34.000000000 -0700
+++ b/ganeti-2.0.3/lib/hypervisor/hv_kvm.py	2009-09-23 09:56:50.000000000 -0700
@@ -137,7 +137,7 @@
     script.write("else\n")
     script.write("  # Connect the interface to the bridge\n")
     script.write("  /sbin/ifconfig $INTERFACE 0.0.0.0 up\n")
-    script.write("  /usr/sbin/brctl addif $BRIDGE $INTERFACE\n")
+    script.write("  /sbin/brctl addif $BRIDGE $INTERFACE\n")
     script.write("fi\n\n")
     # As much as we'd like to put this in our _ROOT_DIR, that will happen to be
     # mounted noexec sometimes, so we'll have to find another place.
