http://bugs.gentoo.org/295093

--- guile-1.8.7.orig/test-suite/standalone/Makefile.am
+++ guile-1.8.7/test-suite/standalone/Makefile.am
@@ -28,7 +28,9 @@
 BUILT_SOURCES =
 EXTRA_DIST =
 
-TESTS_ENVIRONMENT = "${top_builddir}/pre-inst-guile-env"
+TESTS_ENVIRONMENT =					\
+builddir="$(builddir)"					\
+GUILE_AUTO_COMPILE=0 "${top_builddir}/pre-inst-guile-env"
 
 test_cflags =					\
   -I$(top_srcdir)/test-suite/standalone		\
--- guile-1.8.7.orig/test-suite/standalone/test-asmobs
+++ guile-1.8.7/test-suite/standalone/test-asmobs
@@ -2,7 +2,8 @@
 exec guile -q -s "$0" "$@"
 !#
 
-(load-extension "libtest-asmobs" "libtest_asmobs_init")
+(load-extension (string-append (getenv "builddir") "/libtest-asmobs")
+                "libtest_asmobs_init")
 
 (define (test x v)
   (if v
@@ -114,4 +115,4 @@
 
 ;; Local Variables:
 ;; mode: scheme
-;; End:
\ No newline at end of file
+;; End:
