diff --git a/test/test_httplib2.py b/test/test_httplib2.py
index 157fbc4..d7cecc5 100644
--- a/test/test_httplib2.py
+++ b/test/test_httplib2.py
@@ -1,5 +1,5 @@
 from wsgi_intercept import httplib2_intercept
-from socket import gaierror
+from socket import error
 import wsgi_intercept
 from test import wsgi_app
 import httplib2
@@ -30,7 +30,7 @@ def test_success():
 
 def test_bogus_domain():
     install()
-    py.test.raises(gaierror,
+    py.test.raises(error,
             'httplib2_intercept.HTTP_WSGIInterceptorWithTimeout("_nonexistant_domain_").connect()')
     uninstall()
 
