diff -urp hotplug-2004_09_23/etc/hotplug/net.agent hotplug-2004_09_23-ifrename/etc/hotplug/net.agent
--- hotplug-2004_09_23/etc/hotplug/net.agent	2004-09-21 01:02:34.000000000 +0200
+++ hotplug-2004_09_23-ifrename/etc/hotplug/net.agent	2005-07-19 20:12:15.000000000 +0200
@@ -53,9 +53,9 @@ add|register)
 	    # This needs to be done before ifup otherwise ifup will get
 	    # confused by the name changed and because iface need to be
 	    # down to change its name.
-	    if [ -x /usr/sbin/ifrename ] && [ -r /etc/iftab ]; then
+	    if [ -x /sbin/ifrename ] && [ -r /etc/iftab ]; then
 		debug_mesg invoke ifrename for $INTERFACE
-		NEWNAME=`/usr/sbin/ifrename -i $INTERFACE`
+		NEWNAME=`/sbin/ifrename -i $INTERFACE`
 		if [ -n "$NEWNAME" ]; then
 		    debug_mesg iface $INTERFACE is remapped to $NEWNAME
 		    INTERFACE=$NEWNAME
