--- qpsmtpd-0.40/plugins/check_spamhelo.badhelo_disconnect	2007-06-14 11:57:24.000000000 -0600
+++ qpsmtpd-0.40/plugins/check_spamhelo	2007-06-14 13:21:45.000000000 -0600
@@ -23,7 +23,7 @@
   for my $bad ($self->qp->config('badhelo')) {
     if ($host eq lc $bad) {
       $self->log(LOGDEBUG, "Denying HELO from host claiming to be $bad");
-      return (DENY, "Sorry, I don't believe that you are $host.");
+      return (DENY_DISCONNECT, "Sorry, I don't believe that you are $host.");
     }
   }
   return DECLINED;
