From 8b64193ac7b30faa2213d5be6ee477db3a1f70a6 Mon Sep 17 00:00:00 2001
From: Nikos Mavrogiannopoulos <nmav@gnutls.org>
Date: Mon, 3 Jun 2013 21:17:39 +0200
Subject: [PATCH] correctly place cflags

---
 lib/nettle/Makefile.am |   11 ++++++-----
 1 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/lib/nettle/Makefile.am b/lib/nettle/Makefile.am
index 0987c8e..3dcb928 100644
--- a/lib/nettle/Makefile.am
+++ b/lib/nettle/Makefile.am
@@ -18,17 +18,18 @@
 # You should have received a copy of the GNU Lesser General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>
 
-AM_CFLAGS = $(WERROR_CFLAGS) $(WSTACK_CFLAGS) $(WARN_CFLAGS)
+AM_CFLAGS = $(WERROR_CFLAGS) $(WSTACK_CFLAGS) $(WARN_CFLAGS) \
+	$(NETTLE_CFLAGS)		\
+	$(HOGWEED_CFLAGS)		\
+	$(GMP_CFLAGS)
+
 AM_CPPFLAGS = \
 	-I$(srcdir)/../../gl		\
 	-I$(builddir)/../../gl		\
 	-I$(srcdir)/../includes		\
 	-I$(builddir)/../includes	\
 	-I$(builddir)/../../gl          \
-	-I$(srcdir)/..			\
-	$(NETTLE_CFLAGS)		\
-	$(HOGWEED_CFLAGS)		\
-	$(GMP_CFLAGS)
+	-I$(srcdir)/..
 
 if ENABLE_MINITASN1
 AM_CPPFLAGS += -I$(srcdir)/../minitasn1
-- 
1.7.1

