--- rss2email.py~	2009-04-25 00:29:16.000000000 +0200
+++ rss2email.py	2009-07-26 15:39:16.787467661 +0200
@@ -619,7 +619,7 @@
 					subjecthdr = title
 					datehdr = time.strftime("%a, %d %b %Y %H:%M:%S -0000", datetime)
 					useragenthdr = "rss2email"
-					extraheaders = {'Date': datehdr, 'User-Agent': useragenthdr}
+					extraheaders = {'Date': datehdr, 'User-Agent': useragenthdr, 'X-rss-feed': f.url}
 					if BONUS_HEADER != '':
 						for hdr in BONUS_HEADER.strip().splitlines():
 							pos = hdr.strip().find(':')
