diff -U 0 src.orig/arp_main.c src/arp_main.c
--- src.orig/arp_main.c	2003-10-02 03:08:49.559975088 -0400
+++ src/arp_main.c	2003-10-02 03:09:22.732932032 -0400
@@ -18 +18 @@
-#include <libnet.h>
+#include <libnet-1.0.h>
diff -U 0 src.orig/arpg.c src/arpg.c
--- src.orig/arpg.c	2003-10-02 03:08:49.560974936 -0400
+++ src/arpg.c	2003-10-02 03:09:22.733931880 -0400
@@ -12 +12 @@
-#include <libnet.h>
+#include <libnet-1.0.h>
diff -U 0 src.orig/dhcp_main.c src/dhcp_main.c
--- src.orig/dhcp_main.c	2003-10-02 03:08:49.559975088 -0400
+++ src/dhcp_main.c	2003-10-02 03:09:22.745930056 -0400
@@ -14 +14 @@
-#include <libnet.h>
+#include <libnet-1.0.h>
diff -U 0 src.orig/discover_dispatch.c src/discover_dispatch.c
--- src.orig/discover_dispatch.c	2003-10-02 03:08:49.560974936 -0400
+++ src/discover_dispatch.c	2003-10-02 03:09:22.762927472 -0400
@@ -10 +10 @@
-#include <libnet.h>
+#include <libnet-1.0.h>
diff -U 0 src.orig/discover_main.c src/discover_main.c
--- src.orig/discover_main.c	2003-10-02 03:08:49.560974936 -0400
+++ src/discover_main.c	2003-10-02 03:09:22.766926864 -0400
@@ -8 +8 @@
-#include <libnet.h>
+#include <libnet-1.0.h>
diff -U 0 src.orig/fp.h src/fp.h
--- src.orig/fp.h	2003-10-02 03:08:49.561974784 -0400
+++ src/fp.h	2003-10-02 03:09:22.815919416 -0400
@@ -10 +10 @@
-#include <libnet.h>
+#include <libnet-1.0.h>
diff -U 0 src.orig/icmp_main.c src/icmp_main.c
--- src.orig/icmp_main.c	2003-10-02 03:08:49.559975088 -0400
+++ src/icmp_main.c	2003-10-02 03:09:22.777925192 -0400
@@ -14 +14 @@
-#include <libnet.h>
+#include <libnet-1.0.h>
diff -U 0 src.orig/network_raw.c src/network_raw.c
--- src.orig/network_raw.c	2003-10-02 03:08:49.560974936 -0400
+++ src/network_raw.c	2003-10-02 03:09:22.782924432 -0400
@@ -10 +10 @@
-#include <libnet.h>
+#include <libnet-1.0.h>
diff -U 0 src.orig/packets.c src/packets.c
--- src.orig/packets.c	2003-10-02 03:08:49.560974936 -0400
+++ src/packets.c	2003-10-02 03:09:22.790923216 -0400
@@ -6 +6 @@
-#include <libnet.h>
+#include <libnet-1.0.h>
diff -U 0 src.orig/thcrut.c src/thcrut.c
--- src.orig/thcrut.c	2003-10-02 03:08:49.561974784 -0400
+++ src/thcrut.c	2003-10-02 03:09:22.803921240 -0400
@@ -87 +87 @@
-#include <libnet.h>
+#include <libnet-1.0.h>
diff -U 0 src.orig/thcrut.h src/thcrut.h
--- src.orig/thcrut.h	2003-10-02 03:08:49.562974632 -0400
+++ src/thcrut.h	2003-10-02 03:09:22.825917896 -0400
@@ -7 +7 @@
-#include <libnet.h>
+#include <libnet-1.0.h>
diff -U 0 src.orig/thcrut_libnet.c src/thcrut_libnet.c
--- src.orig/thcrut_libnet.c	2003-10-02 03:08:49.559975088 -0400
+++ src/thcrut_libnet.c	2003-10-02 03:09:22.803921240 -0400
@@ -5 +5 @@
-#include <libnet.h>
+#include <libnet-1.0.h>
diff -U 0 src.orig/thcrut_libnet.h src/thcrut_libnet.h
--- src.orig/thcrut_libnet.h	2003-10-02 03:08:49.561974784 -0400
+++ src/thcrut_libnet.h	2003-10-02 03:09:22.825917896 -0400
@@ -5 +5 @@
-#include <libnet.h>
+#include <libnet-1.0.h>
--- configure.orig	2003-10-02 03:06:36.083266632 -0400
+++ configure	2003-10-02 03:08:01.425292672 -0400
@@ -1275,2 +1274,1 @@
-  CCOPTS='-O2 -Wall'
-  CFLAGS="$CCOPTS"
+  CCOPTS=""
@@ -2148 +2148 @@
-LIBS="-lnet  $LIBS"
+LIBS="-lnet-1.0  $LIBS"
@@ -2191,2 +2191,2 @@
-	CFLAGS="`libnet-config --cflags --defines | tr '\n' ' '` $CFLAGS"
-	LIBS="`libnet-config --libs` $LIBS"
+	CFLAGS="`libnet-1.0-config --cflags --defines | tr '\n' ' '` $CFLAGS"
+	LIBS="`libnet-1.0-config --libs` $LIBS"
diff -U 0 src.orig/Makefile.in src/Makefile.in
--- src.orig/Makefile.in	2003-10-02 03:16:07.823348928 -0400
+++ src/Makefile.in	2003-10-02 03:18:06.545300448 -0400
@@ -243,7 +243 @@
-	../Libnet-1.0.2a/include/libnet.h \
-	../Libnet-1.0.2a/include/libnet/libnet-headers.h \
-	../Libnet-1.0.2a/include/libnet/libnet-structures.h \
-	../Libnet-1.0.2a/include/libnet/libnet-macros.h \
-	../Libnet-1.0.2a/include/libnet/libnet-asn1.h \
-	../Libnet-1.0.2a/include/libnet/libnet-functions.h \
-	../Libnet-1.0.2a/include/libnet/libnet-ospf.h macvendor.h \
+	macvendor.h \
@@ -253,7 +247,2 @@
-arpg.o: arpg.c default.h ../config.h ../Libnet-1.0.2a/include/libnet.h \
-	../Libnet-1.0.2a/include/libnet/libnet-headers.h \
-	../Libnet-1.0.2a/include/libnet/libnet-structures.h \
-	../Libnet-1.0.2a/include/libnet/libnet-macros.h \
-	../Libnet-1.0.2a/include/libnet/libnet-asn1.h \
-	../Libnet-1.0.2a/include/libnet/libnet-functions.h \
-	../Libnet-1.0.2a/include/libnet/libnet-ospf.h thcrut.h state.h \
+arpg.o: arpg.c default.h ../config.h \
+	thcrut.h state.h \
@@ -266,7 +255 @@
-	../Libnet-1.0.2a/include/libnet.h \
-	../Libnet-1.0.2a/include/libnet/libnet-headers.h \
-	../Libnet-1.0.2a/include/libnet/libnet-structures.h \
-	../Libnet-1.0.2a/include/libnet/libnet-macros.h \
-	../Libnet-1.0.2a/include/libnet/libnet-asn1.h \
-	../Libnet-1.0.2a/include/libnet/libnet-functions.h \
-	../Libnet-1.0.2a/include/libnet/libnet-ospf.h thcrut.h state.h \
+	thcrut.h state.h \
@@ -276,7 +259 @@
-	../Libnet-1.0.2a/include/libnet.h \
-	../Libnet-1.0.2a/include/libnet/libnet-headers.h \
-	../Libnet-1.0.2a/include/libnet/libnet-structures.h \
-	../Libnet-1.0.2a/include/libnet/libnet-macros.h \
-	../Libnet-1.0.2a/include/libnet/libnet-asn1.h \
-	../Libnet-1.0.2a/include/libnet/libnet-functions.h \
-	../Libnet-1.0.2a/include/libnet/libnet-ospf.h default.h \
+	default.h \
@@ -287,7 +264,2 @@
-discover_main.o: discover_main.c ../Libnet-1.0.2a/include/libnet.h \
-	../Libnet-1.0.2a/include/libnet/libnet-headers.h \
-	../Libnet-1.0.2a/include/libnet/libnet-structures.h \
-	../Libnet-1.0.2a/include/libnet/libnet-macros.h \
-	../Libnet-1.0.2a/include/libnet/libnet-asn1.h \
-	../Libnet-1.0.2a/include/libnet/libnet-functions.h \
-	../Libnet-1.0.2a/include/libnet/libnet-ospf.h default.h \
+discover_main.o: discover_main.c \
+	default.h \
@@ -298,8 +270 @@
-fp.o: fp.c default.h ../config.h ../pcre-3.9/pcre.h fp.h \
-	../Libnet-1.0.2a/include/libnet.h \
-	../Libnet-1.0.2a/include/libnet/libnet-headers.h \
-	../Libnet-1.0.2a/include/libnet/libnet-structures.h \
-	../Libnet-1.0.2a/include/libnet/libnet-macros.h \
-	../Libnet-1.0.2a/include/libnet/libnet-asn1.h \
-	../Libnet-1.0.2a/include/libnet/libnet-functions.h \
-	../Libnet-1.0.2a/include/libnet/libnet-ospf.h
+fp.o: fp.c default.h ../config.h ../pcre-3.9/pcre.h fp.h
@@ -307,7 +272 @@
-	../Libnet-1.0.2a/include/libnet.h \
-	../Libnet-1.0.2a/include/libnet/libnet-headers.h \
-	../Libnet-1.0.2a/include/libnet/libnet-structures.h \
-	../Libnet-1.0.2a/include/libnet/libnet-macros.h \
-	../Libnet-1.0.2a/include/libnet/libnet-asn1.h \
-	../Libnet-1.0.2a/include/libnet/libnet-functions.h \
-	../Libnet-1.0.2a/include/libnet/libnet-ospf.h network.h \
+	network.h \
@@ -320,7 +279 @@
-	../Libnet-1.0.2a/include/libnet.h \
-	../Libnet-1.0.2a/include/libnet/libnet-headers.h \
-	../Libnet-1.0.2a/include/libnet/libnet-structures.h \
-	../Libnet-1.0.2a/include/libnet/libnet-macros.h \
-	../Libnet-1.0.2a/include/libnet/libnet-asn1.h \
-	../Libnet-1.0.2a/include/libnet/libnet-functions.h \
-	../Libnet-1.0.2a/include/libnet/libnet-ospf.h network_raw.h
+	network_raw.h
@@ -328,7 +281 @@
-	../Libnet-1.0.2a/include/libnet.h \
-	../Libnet-1.0.2a/include/libnet/libnet-headers.h \
-	../Libnet-1.0.2a/include/libnet/libnet-structures.h \
-	../Libnet-1.0.2a/include/libnet/libnet-macros.h \
-	../Libnet-1.0.2a/include/libnet/libnet-asn1.h \
-	../Libnet-1.0.2a/include/libnet/libnet-functions.h \
-	../Libnet-1.0.2a/include/libnet/libnet-ospf.h state.h \
+	state.h \
@@ -338,7 +285 @@
-	../Libnet-1.0.2a/include/libnet.h \
-	../Libnet-1.0.2a/include/libnet/libnet-headers.h \
-	../Libnet-1.0.2a/include/libnet/libnet-structures.h \
-	../Libnet-1.0.2a/include/libnet/libnet-macros.h \
-	../Libnet-1.0.2a/include/libnet/libnet-asn1.h \
-	../Libnet-1.0.2a/include/libnet/libnet-functions.h \
-	../Libnet-1.0.2a/include/libnet/libnet-ospf.h nmap_compat.h \
+	nmap_compat.h \
@@ -347,7 +288 @@
-	../Libnet-1.0.2a/include/libnet.h \
-	../Libnet-1.0.2a/include/libnet/libnet-headers.h \
-	../Libnet-1.0.2a/include/libnet/libnet-structures.h \
-	../Libnet-1.0.2a/include/libnet/libnet-macros.h \
-	../Libnet-1.0.2a/include/libnet/libnet-asn1.h \
-	../Libnet-1.0.2a/include/libnet/libnet-functions.h \
-	../Libnet-1.0.2a/include/libnet/libnet-ospf.h state.h \
+	state.h \
@@ -367,7 +302 @@
-	../Libnet-1.0.2a/include/libnet.h \
-	../Libnet-1.0.2a/include/libnet/libnet-headers.h \
-	../Libnet-1.0.2a/include/libnet/libnet-structures.h \
-	../Libnet-1.0.2a/include/libnet/libnet-macros.h \
-	../Libnet-1.0.2a/include/libnet/libnet-asn1.h \
-	../Libnet-1.0.2a/include/libnet/libnet-functions.h \
-	../Libnet-1.0.2a/include/libnet/libnet-ospf.h thcrut.h state.h \
+	thcrut.h state.h \
@@ -378,7 +307 @@
-thcrut_libnet.o: thcrut_libnet.c ../Libnet-1.0.2a/include/libnet.h \
-	../Libnet-1.0.2a/include/libnet/libnet-headers.h \
-	../Libnet-1.0.2a/include/libnet/libnet-structures.h \
-	../Libnet-1.0.2a/include/libnet/libnet-macros.h \
-	../Libnet-1.0.2a/include/libnet/libnet-asn1.h \
-	../Libnet-1.0.2a/include/libnet/libnet-functions.h \
-	../Libnet-1.0.2a/include/libnet/libnet-ospf.h
+thcrut_libnet.o: thcrut_libnet.c
@@ -387,7 +310 @@
-	../Libnet-1.0.2a/include/libnet.h \
-	../Libnet-1.0.2a/include/libnet/libnet-headers.h \
-	../Libnet-1.0.2a/include/libnet/libnet-structures.h \
-	../Libnet-1.0.2a/include/libnet/libnet-macros.h \
-	../Libnet-1.0.2a/include/libnet/libnet-asn1.h \
-	../Libnet-1.0.2a/include/libnet/libnet-functions.h \
-	../Libnet-1.0.2a/include/libnet/libnet-ospf.h state.h \
+	state.h \
