 configure.ac |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/configure.ac b/configure.ac
index 861e939..8fab1a1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -405,7 +405,7 @@ dnl MAYBE_WARN="-Wall -Wno-sign-compare -Werror -Wno-deprecated-declarations"
 if test "$red_target" = "windows"; then
 	MAYBE_WARN="-Wall -Wno-sign-compare -Wno-deprecated-declarations"
 else
-	MAYBE_WARN="-Wall -Wno-sign-compare -Werror -Wno-deprecated-declarations"
+	MAYBE_WARN="-Wall -Wno-sign-compare -Wno-deprecated-declarations"
 fi
 
 AC_ARG_ENABLE(werror,
