--- configure.ac.old	2012-02-03 14:10:59.163215573 +0100
+++ configure.ac	2012-02-03 14:12:18.003215948 +0100
@@ -333,11 +333,9 @@
 
 	dnl  Optional linux/joystick.h
 	AC_CHECK_HEADERS(linux/joystick.h, linux_joystick=yes)
-	AM_CONDITIONAL(LINUX_JOYSTICK, test x$linux_joystick = xyes)
 
 	dnl Optional linux/input.h
 	AC_CHECK_HEADERS(linux/input.h, linux_input=yes)
-	AM_CONDITIONAL(LINUX_INPUT, test x$linux_input = xyes)
 
 	if test "$WIN32" = "no" && test "$enable_clanDisplay" != "no"; then
 	CLANLIB_CHECK_LIB(fontconfig,  [`cat $srcdir/Setup/Tests/fontconfig.cpp`],  clanDisplay, [ *** Cannot find fontconfig (See http://fontconfig.org/ ) (Try libfontconfig1-dev or better) ], [-lfontconfig])
@@ -355,6 +353,9 @@
 	fi
 fi
 
+AM_CONDITIONAL(LINUX_JOYSTICK, test x$linux_joystick = xyes)
+AM_CONDITIONAL(LINUX_INPUT, test x$linux_input = xyes)
+
 have_xrender=no
 
 if test "$enable_clanDisplay" != "no"; then
@@ -507,13 +507,14 @@
 	fi
 
 	extra_LIBS_clanSound="$extra_LIBS_clanSound $sound_libs"
-	AM_CONDITIONAL(ALSA, test x$have_alsa = xyes)
 	fi
 
 	if test "$enable_clanSound" = "auto"; then enable_clanSound=yes; fi
 
 fi
 
+AM_CONDITIONAL(ALSA, test x$have_alsa = xyes)
+
 if test "$enable_clanSound" != "no"; then
 	enable_clanSound=yes
 	echo "Checking for clanMikmod stuff"
--- Documentation/Reference/Makefile.am.old	2012-02-06 16:00:50.034478851 +0100
+++ Documentation/Reference/Makefile.am	2012-02-06 16:01:01.524478906 +0100
@@ -10,7 +10,7 @@
 	mkdir doxyoutput;
 	mkdir reftest
 	doxygen ./clanlib.doxygen
-	make -C ../Utilities/ReferenceDocs
+	$(MAKE) -C ../Utilities/ReferenceDocs
 	../Utilities/ReferenceDocs/ReferenceDocs
 
 install-html:
