Index: core-108.00.01/_oasis
===================================================================
--- core-108.00.01.orig/_oasis
+++ core-108.00.01/_oasis
@@ -63,7 +63,7 @@ Executable test_runner
   Path:               lib_test
   MainIs:             test_runner.ml
   Build$:            flag(tests)
-  Custom:             true
+#  Custom:             true
 #  CompiledObject:     best
   Install:            false
   BuildDepends:       core,oUnit (>= 1.0.2)
