Index: core-109.07.00/myocamlbuild.ml
===================================================================
--- core-109.07.00.orig/myocamlbuild.ml
+++ core-109.07.00/myocamlbuild.ml
@@ -587,7 +587,7 @@ let setup_standard_build_flags () =
 
     (* enable warnings; make sure the '@' character isn't in the beginning;
        ms-dos interprets that character specially *)
-    flag ["compile"; "ocaml"] (S [A "-w"; A "A@Aemr-28"; A "-strict-sequence" ])
+    flag ["compile"; "ocaml"] (S [A "-strict-sequence" ])
 ;;
 
 let dispatch = function
