Index: core-111.28.00/lib/unix_stubs.c
===================================================================
--- core-111.28.00.orig/lib/unix_stubs.c
+++ core-111.28.00/lib/unix_stubs.c
@@ -434,7 +434,7 @@ static value core_stat_aux_64(struct sta
   CAMLparam0();
   CAMLlocal5(atime, mtime, ctime, offset, v);
 
-  #if defined _BSD_SOURCE || defined _SVID_SOURCE
+  #if defined _BSD_SOURCE || defined _SVID_SOURCE || (_POSIX_C_SOURCE - 0) >= 200809L || (_XOPEN_SOURCE - 0) >= 700
     atime = caml_copy_double((double) buf->st_atime + (buf->st_atim.tv_nsec / 1000000000.0f));
     mtime = caml_copy_double((double) buf->st_mtime + (buf->st_mtim.tv_nsec / 1000000000.0f));
     ctime = caml_copy_double((double) buf->st_ctime + (buf->st_ctim.tv_nsec / 1000000000.0f));
