--- zathura-pdf-mupdf-9999/pdf.c
+++ zathura-pdf-mupdf-9999/pdf.c
@@ -6,9 +6,9 @@
 #include <ctype.h>
 #include <girara/datastructures.h>
 #include <glib.h>
-#include <muxps.h>
-#include <mupdf.h>
-#include <fitz.h>
+#include <mupdf/fitz.h>
+#include <mupdf/pdf.h>
+#include <mupdf/xps.h>
 
 #include "pdf.h"
 
--- zathura-pdf-mupdf-9999/pdf.h
+++ zathura-pdf-mupdf-9999/pdf.h
@@ -5,7 +5,6 @@
 
 #include <stdbool.h>
 #include <zathura/plugin-api.h>
-#include <fitz.h>
 
 #if HAVE_CAIRO
 #include <cairo.h>
