--- oftpd-0.3.7.orig/src/ftp_session.c	2004-03-25 20:46:40.000000000 +0000
+++ oftpd-0.3.7/src/ftp_session.c	2006-12-27 11:38:56.219883456 +0000
@@ -683,10 +683,12 @@ static void do_lprt(ftp_session_t *f, co
 #ifdef INET6
     if ((SSFAM(host_port) != AF_INET) && (SSFAM(host_port) != AF_INET6)) {
         reply(f, 521, "Only IPv4 and IPv6 supported, address families (4,6)");
+        return;
     }
 #else
     if (SSFAM(host_port) != AF_INET) {
         reply(f, 521, "Only IPv4 supported, address family (4)");
+        return;
     }
 #endif
 
