*** Makefile.orig	Mon Nov 23 08:55:42 2009
--- Makefile	Mon Nov 23 08:57:48 2009
***************
*** 27,33 ****
  
  DEBUG := 0
  
! ifneq ($(shell [ -f $(KBUILD)/include/linux/aio_abi.h ] && echo 1),1)
  $(warning Building tp_smapi requires Linux kernel 2.6.19 or newer, and matching kernel headers.)
  $(warning You may need to override the following Make variables:)
  $(warning .   KVER=$(KVER))
--- 27,33 ----
  
  DEBUG := 0
  
! ifneq ($(shell [ -f $(KSRC)/include/linux/aio_abi.h ] && echo 1),1)
  $(warning Building tp_smapi requires Linux kernel 2.6.19 or newer, and matching kernel headers.)
  $(warning You may need to override the following Make variables:)
  $(warning .   KVER=$(KVER))
