diff -ur FreeWnn-1.1.1-a017.orig/Xsi/Wnn/uum/jhlp.c FreeWnn-1.1.1-a017/Xsi/Wnn/uum/jhlp.c
--- FreeWnn-1.1.1-a017.orig/Xsi/Wnn/uum/jhlp.c	Wed Feb 23 17:08:49 2000
+++ FreeWnn-1.1.1-a017/Xsi/Wnn/uum/jhlp.c	Wed Nov  1 19:33:54 2000
@@ -638,7 +638,7 @@
 #ifdef  SYSVR2
 #define index   strchr
 #endif /* SYSVR2 */
-  extern char *index ();
+/*  extern char *index ();*/
 
   strcpy (ostr, default_getoptstr);
   strcat (ostr, lang_db->getoptstr);
@@ -1720,7 +1720,7 @@
     }
 #endif /* defined(uniosu) */
 
-#ifndef sgi
+#if !defined(sgi) && !defined(linux)
   ptyname (nmbuf, ptynm, ptyno);
   if (chown (nmbuf, 0, 0) == ERROR)
     {
diff -ur FreeWnn-1.1.1-a017.orig/Xsi/Wnn/uum/wnnrc_op.c FreeWnn-1.1.1-a017/Xsi/Wnn/uum/wnnrc_op.c
--- FreeWnn-1.1.1-a017.orig/Xsi/Wnn/uum/wnnrc_op.c	Wed Feb 23 16:31:45 2000
+++ FreeWnn-1.1.1-a017/Xsi/Wnn/uum/wnnrc_op.c	Mon Oct 30 23:42:17 2000
@@ -133,7 +133,7 @@
   struct passwd *u;
   extern struct passwd *getpwnam ();
 #if defined(SYSVR2) && !defined(AIXV3)
-  extern char *strchr ();
+/*  extern char *strchr (); */
 #endif /* defined(SYSVR2) && !defined(AIXV3) */
 
   if (*s != '~' && *s != '@')
diff -u FreeWnn-1.1.1-a017.orig/Xsi/makerule.mk.in.~1~ FreeWnn-1.1.1-a017.orig/Xsi/makerule.mk.in
--- FreeWnn-1.1.1-a017.orig/Xsi/makerule.mk.in.~1~	Sun Feb 13 10:15:54 2000
+++ FreeWnn-1.1.1-a017.orig/Xsi/makerule.mk.in	Wed Nov  1 19:49:39 2000
@@ -90,7 +90,7 @@
 ALLDEFINES = $(INCLUDES) -DFUNCPROTO=15 -DNARROWPROTO $(DEFINES)
 CFLAGS = $(CDEBUGFLAGS) $(CCOPTIONS) $(ALLDEFINES)
 LINTFLAGS = $(LINTOPTS) -DLINT $(ALLDEFINES) $(DEPEND_DEFINES)
-LDOPTIONS = $(CDEBUGFLAGS) $(CCOPTIONS)
+LDOPTIONS = $(CDEBUGFLAGS) $(CCOPTIONS) -L/usr/lib/termcap
 
            CCLINK = $(CC)
 
@@ -114,7 +114,7 @@
       WNNBINDIR = ${PREFIX}/bin
       WNNINCDIR = @includedir@
       WNNLIBDIR = @libdir@
-    WNNTERM_LIB = -ltermcap
+    WNNTERM_LIB = -ltermcap -lcrypt
 
   PUBDICPLUSSRC = $(IMTOP)/PubdicPlus
   WNNCONSDICSRC = $(IMTOP)/Wnn-consortium/dic
