--- slim-1.3.5-old/CMakeLists.txt	2012-12-31 08:03:42.000000000 -0500
+++ slim-1.3.5/CMakeLists.txt	2013-02-06 09:05:06.000000000 -0500
@@ -171,6 +171,6 @@
 # configure
 install(FILES slim.conf DESTINATION ${SYSCONFDIR})
 # systemd service file
-install(FILES slim.service DESTINATION usr/lib/systemd/system)
+install(FILES slim.service DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/systemd/system)
 # themes directory
 subdirs(themes)
