--- CMakeLists.txt.old	2011-08-02 08:28:51.211645468 +0200
+++ CMakeLists.txt	2011-08-02 08:29:26.912641230 +0200
@@ -41,9 +41,9 @@
     IF (NOT OSX)
         OPTION(USE_X11 "Use X11 Clipboard functionality" ON)
     ENDIF ()
-    SET(PKG_DATADIR ${CMAKE_INSTALL_PREFIX}/share/mana)
+    SET(PKG_DATADIR @GENTOO_DATADIR@)
     SET(LOCALEDIR ${CMAKE_INSTALL_PREFIX}/share/locale)
-    SET(PKG_BINDIR ${CMAKE_INSTALL_PREFIX}/bin)
+    SET(PKG_BINDIR @GENTOO_BINDIR@)
 ENDIF (WIN32)
 
 ADD_SUBDIRECTORY(data)
--- mana.desktop.old	2011-08-02 11:31:56.049341650 +0200
+++ mana.desktop	2011-08-02 11:32:05.312340549 +0200
@@ -13,4 +13,4 @@
 Terminal=false
 Type=Application
 Icon=mana
-Categories=Application;Game;
+Categories=Game;
--- data/CMakeLists.txt.old	2011-08-02 12:15:38.520030381 +0200
+++ data/CMakeLists.txt	2011-08-02 12:15:44.971029615 +0200
@@ -1,6 +1,5 @@
 SET(DATA_DIR ${PKG_DATADIR}/data)
 
-ADD_SUBDIRECTORY(fonts)
 ADD_SUBDIRECTORY(graphics)
 ADD_SUBDIRECTORY(help)
 ADD_SUBDIRECTORY(icons)
