diff --exclude-from=/home/dang/bin/scripts/diffrc -up -ruN foomatic-filters-ppds-20070501.orig/install foomatic-filters-ppds-20070501/install
--- foomatic-filters-ppds-20070501.orig/install	2007-05-01 03:50:46.000000000 -0400
+++ foomatic-filters-ppds-20070501/install	2007-05-02 12:04:55.000000000 -0400
@@ -9,7 +9,7 @@
 prefix=/usr/local
 gzip=0
 moreoptions=1
-while [ $moreoptions == 1 ]
+while [ $moreoptions = 1 ]
   do
   case $1 in
       -p|--prefix)
@@ -46,10 +46,10 @@ done
 # Copy files
 echo "Copying files ..."
 mkdir -p $destdir$prefix
-cp -ax bin share $destdir$prefix
+cp -RPp bin share $destdir$prefix
 
 # Compress files
-if [ $gzip == 1 ]; then
+if [ $gzip = 1 ]; then
   echo "Compressing PPD files ..."
   gzip -rf $destdir$prefix/share/ppd
 fi
@@ -57,9 +57,9 @@ fi
 # Link for CUPS (do only if we have a CUPS daemon, PPD link to
 # /usr/share/cups/model not needed on pure CUPS clients)
 cups=0
-if ( which cupsd > /dev/null 2>&1 ); then
+if type cupsd > /dev/null 2> /dev/null; then
   echo "Setting link for CUPS ..."
-  if ( which cups-config > /dev/null 2>&1 ); then
+  if type cups-config > /dev/null 2> /dev/null; then
     # CUPS installed from source or "devel" package installed
     ppddir=`cups-config --datadir`/model
   else
@@ -74,7 +74,7 @@ if ( which cupsd > /dev/null 2>&1 ); the
 fi
 
 # Do we have PPR installed
-if ( which ppad > /dev/null 2>&1 ); then
+if type ppad > /dev/null 2>/dev/null; then
   echo "Setting link for PPR ..."
   if [ -d $destdir/usr/share/ppr/PPDFiles/ ]; then
     rm -f $destdir/usr/share/ppr/PPDFiles/foomatic-ppds
@@ -93,7 +93,7 @@ fi
 echo "Installation finished"
 echo ""
 
-if [ $cups == 1 ]; then
+if [ $cups = 1 ]; then
     echo "Please restart the CUPS daemon. Note that the CUPS daemon"
     echo "needs some minutes to register all PPD files."
     echo
