From a3b07a52a41b17dacbc1b703b6dd331cd1cf6e61 Mon Sep 17 00:00:00 2001
From: Jan Engelhardt <jengelh@medozas.de>
Date: Tue, 16 Aug 2011 07:28:21 +0200
Subject: [PATCH] build: make build of pmt-ehd dependent upon HAVE_LIBCRYPTSETUP

---
 src/Makefile.am   |    2 +-
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 347d475..38e8395 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -8,7 +8,7 @@ AM_CFLAGS = ${regular_CFLAGS} ${GCC_FVISIBILITY_HIDDEN} \
 moduledir		= @PAM_MODDIR@
 module_LTLIBRARIES	= pam_mount.la
 sbin_PROGRAMS		= pmvarrun
-if HAVE_LIBCRYPTO
+if HAVE_LIBCRYPTSETUP
 sbin_PROGRAMS		+= pmt-ehd
 endif
 ssbin_PROGRAMS		= mount.crypt
-- 
1.7.3.4

