--- data/pam-exherbo/gdm-launch-environment.pam~	2013-06-27 16:39:45.000000000 +0200
+++ data/pam-exherbo/gdm-launch-environment.pam	2013-10-12 20:44:03.838894768 +0200
@@ -8,7 +8,7 @@
 
 password    required    pam_deny.so
 
-session     required    pam_systemd.so kill-session-processes=1
+-session    optional    pam_systemd.so kill-session-processes=1
 session     optional    pam_keyinit.so force revoke
 session     required    pam_succeed_if.so audit quiet_success user = gdm
 session     required    pam_permit.so
