--- ipw3945.h.orig	2007-07-31 09:53:39.000000000 +0300
+++ ipw3945.h	2007-12-23 04:09:59.000000000 +0200
@@ -2414,6 +2414,18 @@ enum {
 #define IPW_ORD_TABLE_7_MASK              0x0000F700
 
 /*
+ * For backword Kernel compatibility (after 2.6.24)
+ */
+#if !defined(MAC_ARG)
+#define MAC_ARG(x) ((u8*)(x))[0],((u8*)(x))[1],((u8*)(x))[2],((u8*)(x))[3],((u8*)(x))[4],((u8*)(x))[5]
+#endif
+
+#if !defined(SET_MODULE_OWNER)
+#define SET_MODULE_OWNER(dev) do { } while (0)
+#endif
+
+
+/*
  * Table 0 Entries (all entries are 32 bits)
  */
 enum {
