diff -urNp ipw3945-1.2.2.org/Makefile ipw3945-1.2.2/Makefile
--- ipw3945-1.2.2.org/Makefile	2007-04-30 06:44:56.000000000 +0300
+++ ipw3945-1.2.2/Makefile	2007-11-14 19:23:07.000000000 +0200
@@ -111,7 +111,7 @@ IEEE80211_API := $(shell $(DIR)/snapshot
 EXTRA_CFLAGS += -DIPW3945_COMPAT=$(IEEE80211_API)
 
 ifeq ($(CONFIG_IPW3945_DEBUG),y)
-	EXTRA_CFLAGS += -g -Wa,-adhlms=$@.lst
+	EXTRA_CFLAGS += -g -Wa,-adhlms=$(T)/$@.lst
 endif
 
 ifdef KBUILD_EXTMOD
@@ -193,34 +193,6 @@ utils:
 	@[ ! -d util ] || make -C util IEEE80211_PATH=$(IEEE80211_PATH)
 
 check_inc:
-	@( [ "$(IEEE80211_DUPLICATE)" ] && echo -e \
-"\n WARNING: Your kernel contains ieee80211 symbol definitions and you\n"\
-"are not using the kernel's default ieee80211 subsystem.  (Perhaps you\n"\
-"used the out-of-tree ieee80211 subsystem's 'make install' or have\n"\
-"provided a path to the ieee80211 subsystem via IEEE80211_INC.)\n\n"\
-"If you wish to use the out-of-tree ieee80211 subsystem then it is\n"\
-"recommended to use that projects' \"make patch_kernel\" facility\n"\
-"and rebuild your kernel to update the Module symbol version information.\n"\
-"\n"\
-"Failure to do this may result in build warnings and unexpected\n"\
-"behavior when running modules which rely on the ieee80211 subsystem.\n\n"\ || \
-	exit 0)
-
-	@( [ "$(IEEE80211_DUPLICATE)" ] && \
-	   [ ! "$(IEEE80211_IGNORE_DUPLICATE)" ] && echo -e \
-" Aborting the build.  You can force the build to continue by adding:\n\n"\
-"\tIEEE80211_IGNORE_DUPLICATE=y\n\n"\
-"to your make command line.\n\n" && exit 1 || exit 0)
-
-	@( [ ! "$(IEEE80211_API)" ] && echo -e \
-"\n ERROR: A compatible subsystem was not found in the following path[s]:\n\n"\
-"\t$(IEEE80211_RES)\n\n"\
-"You need to install the ieee80211 subsystem from http://ieee80211.sf.net\n"\
-"and point this build to the location where you installed those sources, eg.:\n\n"\
-"\t% make IEEE80211_INC=/usr/src/ieee80211/\n\n"\
-"or use the 'make patch_kernel' within the ieee80211 subsystem to patch your\n"\
-"kernel sources.\n" && exit 1 || exit 0)
-
 	@echo -e \
 " Using ieee80211 subsystem version API v$(IEEE80211_API) from:\n\n" \
 "\tBase: $(IEEE80211_BASE)\n" \
@@ -241,7 +213,7 @@ codingstyle:
 
 clean:
 	@[ ! -d util ] || make -C util clean || true
-	rm -f *.{mod.c,mod,o,ko,rej,orig} .*.{stamp,flags,cmd} .lst *.lst *~ 
+	rm -f *.{mod.c,mod,o,ko,rej,orig} .*.{stamp,flags,cmd} $(T)/.lst $(T)/*.lst *~ 
 	rm -rf $(DIR)/tmp $(DIR)/.tmp_versions Module.symvers Modules.symvers
 	for file in *.{c,h}; do \
 		sed -i -e "s:\ *$$::g" -e "s:\t*$$::g" $$file; \
