--- configure.in~	2012-10-15 22:31:16.000000000 +0200
+++ configure.in	2012-10-20 09:09:09.126416968 +0200
@@ -235,13 +235,7 @@
 # Evolution.dll needs GMime#
 #
 if test "x$ENABLE_GNOME" != "xno"; then
-	if pkg-config --exists gmime-sharp-2.4; then
-		PKG_CHECK_MODULES(EVOLUTION, gmime-sharp-2.4, ENABLE_EVOLUTION="yes", ENABLE_EVOLUTION="no")
-	elif pkg-config --exists gmime-sharp-2.6; then
-		PKG_CHECK_MODULES(EVOLUTION, gmime-sharp-2.6, ENABLE_EVOLUTION="yes", ENABLE_EVOLUTION="no")
-	else
-		PKG_CHECK_MODULES(EVOLUTION, gmime-sharp >= 2.2.7, ENABLE_EVOLUTION="yes", ENABLE_EVOLUTION="no")
-	fi
+	PKG_CHECK_MODULES(EVOLUTION, gmime-sharp-2.6, ENABLE_EVOLUTION="yes", ENABLE_EVOLUTION="no")
 else
 	ENABLE_EVOLUTION=no
 fi
