--- a/debhelper/Makefile
+++ b/debhelper/Makefile
@@ -34,12 +34,17 @@ PERLLIBDIR=$(shell perl -MConfig -e 'pri
 
 POD2MAN=pod2man --utf8 -c Debhelper -r "$(VERSION)"
 
+ifeq ($(USE_NLS),yes)
 # l10n to be built is determined from .po files
-LANGS=$(notdir $(basename $(wildcard man/po4a/po/*.po)))
+LANGS?=$(notdir $(basename $(wildcard man/po4a/po/*.po)))
+else
+LANGS=
+endif
 
 build: version debhelper.7
 	find . -maxdepth 1 -type f -perm +100 -name "dh*" \
 		-exec $(POD2MAN) {} {}.1 \;
+ifeq ($(USE_NLS),yes)
 	po4a --previous -L UTF-8 man/po4a/po4a.cfg 
 	set -e; \
 	for lang in $(LANGS); do \
@@ -54,6 +59,7 @@ build: version debhelper.7
 				$(POD2MAN) --name="debhelper" --section=7 > debhelper.$$lang.7; \
 		fi; \
 	done
+endif
 
 version:
 	printf "package Debian::Debhelper::Dh_Version;\n\$$version='$(VERSION)';\n1" > \
