Fix build failure with automake 1.11.2.
https://bugs.gentoo.org/425208

--- gnome-games-2.28.2-orig/libgames-support/Makefile.am
+++ gnome-games-2.28.2/libgames-support/Makefile.am
@@ -288,7 +288,8 @@
 if BUILD_LIBGAMESCARDS
 if HAVE_RSVG
 
-pkglib_PROGRAMS = gnome-games-render-cards
+auxdir = $(pkglibdir)
+aux_PROGRAMS = gnome-games-render-cards
 
 gnome_games_render_cards_SOURCES = \
 	render-cards.c
