--- Csu-85/start.s
+++ Csu-85/start.s
@@ -22,7 +22,10 @@
  * @APPLE_LICENSE_HEADER_END@
  */
 
-#include <Availability.h>
+/* This is fun: We could use AvailabilityMacros.h to check if we're compiling
+ * for < 10.5 and not inclue Availability.h then. But we are compiling with
+ * deployment target 10.5 *on this host* where we still won't have
+ * Availability.h. So just drop the include */
 
 #if __ppc__ && __DYNAMIC__
 //
@@ -203,7 +206,7 @@
 	cmp	r4, #0
 	bne	Lapple			
 					// "apple" param now in r3
-#if __STATIC__ || ((__IPHONE_OS_VERSION_MIN_REQUIRED >= 30100) && !__ARM_ARCH_4T__)
+#if __STATIC__
 	bl	_main
 	b	_exit
 #else
