diff -ur xrootd-3.2.2.orig/src/XrdOss/XrdOssAio.cc xrootd-3.2.2/src/XrdOss/XrdOssAio.cc
--- xrootd-3.2.2.orig/src/XrdOss/XrdOssAio.cc	2012-06-18 14:39:12.000000000 +0200
+++ xrootd-3.2.2/src/XrdOss/XrdOssAio.cc	2012-07-11 20:41:40.792903194 +0200
@@ -15,10 +15,8 @@
 #include <signal.h>
 #include <stdio.h>
 #include <unistd.h>
-#ifdef _POSIX_ASYNCHRONOUS_IO
-#ifdef __FreeBSD__
 #include <fcntl.h>
-#endif
+#ifdef _POSIX_ASYNCHRONOUS_IO
 #ifdef __macos__
 #include <sys/aio.h>
 #else
