https://bugs.gentoo.org/show_bug.cgi?id=419451

Index: minidlna-1.0.25/Makefile
===================================================================
--- minidlna-1.0.25.orig/Makefile
+++ minidlna-1.0.25/Makefile
@@ -12,10 +12,7 @@
 #
 #CFLAGS = -Wall -O -D_GNU_SOURCE -g -DDEBUG
 #CFLAGS = -Wall -g -Os -D_GNU_SOURCE
-CFLAGS += -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 \
-	 -I/usr/include/ffmpeg \
-	 -I/usr/include/libavutil -I/usr/include/libavcodec -I/usr/include/libavformat \
-	 -I/usr/include/ffmpeg/libavutil -I/usr/include/ffmpeg/libavcodec -I/usr/include/ffmpeg/libavformat
+CFLAGS += -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64
 #STATIC_LINKING: CFLAGS += -DSTATIC
 #STATIC_LINKING: LDFLAGS = -static
 CC = gcc
Index: minidlna-1.0.25/metadata.c
===================================================================
--- minidlna-1.0.25.orig/metadata.c
+++ minidlna-1.0.25/metadata.c
@@ -30,9 +30,9 @@
 #include "image_utils.h"
 #include <jpeglib.h>
 #include <setjmp.h>
-#include <avutil.h>
-#include <avcodec.h>
-#include <avformat.h>
+#include <libavutil/avutil.h>
+#include <libavcodec/avcodec.h>
+#include <libavformat/avformat.h>
 #include "tagutils/tagutils.h"
 
 #include "upnpglobalvars.h"
Index: minidlna-1.0.25/minidlna.c
===================================================================
--- minidlna-1.0.25.orig/minidlna.c
+++ minidlna-1.0.25/minidlna.c
@@ -69,6 +69,7 @@
 
 #ifdef ENABLE_NLS
 #include <libintl.h>
+#include <locale.h>
 #endif
 
 #include "upnpglobalvars.h"
Index: minidlna-1.0.25/tagutils/tagutils-plist.c
===================================================================
--- minidlna-1.0.25.orig/tagutils/tagutils-plist.c
+++ minidlna-1.0.25/tagutils/tagutils-plist.c
@@ -27,7 +27,7 @@
 #include "misc.h"
 #include "tagutils.h"
 #include "textutils.h"
-#include "log.h"
+#include "../log.h"
 
 
 #define MAX_BUF 4096
