diff -urpN tf-40s1.orig/src/history.c tf-40s1/src/history.c
--- tf-40s1.orig/src/history.c	1999-03-06 23:43:24.000000000 +0100
+++ tf-40s1/src/history.c	2006-09-30 18:16:43.000000000 +0200
@@ -38,18 +38,6 @@
 #define LOCALSIZE      100	/* local history size */
 #define INPUTSIZE      100	/* command history buffer size */
 
-typedef struct History {	/* circular list of Alines, and logfile */
-    struct Aline **alines;
-    int size;			/* actual number of lines currently saved */
-    int maxsize;		/* maximum number of lines that can be saved */
-    int first;			/* position of first line in circular array */
-    int last;			/* position of last line in circular array */
-    int index;			/* current recall position */
-    int total;			/* total number of lines ever saved */
-    TFILE *logfile;
-    CONST char *logname;
-} History;
-
 #define empty(hist) (!(hist)->alines || !(hist)->size)
 
 static void     FDECL(alloc_history,(History *hist, int maxsize));
diff -urpN tf-40s1.orig/src/history.h tf-40s1/src/history.h
--- tf-40s1.orig/src/history.h	1999-03-06 23:43:24.000000000 +0100
+++ tf-40s1/src/history.h	2006-09-30 18:16:39.000000000 +0200
@@ -31,6 +31,18 @@ extern void   NDECL(free_histories);
 #define record_global(aline)  recordline(globalhist, (aline))
 #define record_local(aline)   recordline(localhist, (aline))
 
+typedef struct History {	/* circular list of Alines, and logfile */
+    struct Aline **alines;
+    int size;			/* actual number of lines currently saved */
+    int maxsize;		/* maximum number of lines that can be saved */
+    int first;			/* position of first line in circular array */
+    int last;			/* position of last line in circular array */
+    int index;			/* current recall position */
+    int total;			/* total number of lines ever saved */
+    TFILE *logfile;
+    CONST char *logname;
+} History;
+
 extern struct History globalhist[], localhist[];
 extern int log_count, norecord, nolog;
 
