------------------------------------------------------------------------
r1990 | ahuggel | 2010-01-02 02:15:28 +0100 (Sat, 02 Jan 2010) | 1 line

Minor fix. (Fulvio Senore)
------------------------------------------------------------------------

Index: exiv2/src/value.cpp
===================================================================
--- exiv2/src/value.cpp	(revision 1989)
+++ exiv2/src/value.cpp	(revision 1990)
@@ -160,7 +160,7 @@
     DataBuf Value::dataArea() const
     {
         return DataBuf(0, 0);
-    };
+    }
 
     DataValue::DataValue(TypeId typeId) : Value(typeId)
     {
