diff -U 3 -dHrN Normaliz2.8.orig/source/libnormaliz/Makefile Normaliz2.8/source/libnormaliz/Makefile
--- Normaliz2.8.orig/source/libnormaliz/Makefile	2012-11-11 19:05:18.610268519 -0800
+++ Normaliz2.8/source/libnormaliz/Makefile	2012-11-11 19:07:17.040437439 -0800
@@ -11,16 +11,16 @@
 all: default
 
 libnormaliz.o: $(LIBHEADERS) $(LIBSOURCES) libnormaliz-impl.cpp
-	$(CXX) $(CXXFLAGS) $(NORMFLAGS) -c libnormaliz-impl.cpp -o libnormaliz.o
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) $(NORMFLAGS) -c libnormaliz-impl.cpp -o libnormaliz.o
 
 cone_property.o: cone_property.h cone_property.cpp
-	$(CXX) $(CXXFLAGS) $(NORMFLAGS) -c cone_property.cpp -o $@
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) $(NORMFLAGS) -c cone_property.cpp -o $@
 
 hilbert_series.o: $(LIBHEADERS) HilbertSeries.cpp
-	$(CXX) $(CXXFLAGS) $(NORMFLAGS) -c HilbertSeries.cpp -o $@
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) $(NORMFLAGS) -c HilbertSeries.cpp -o $@
 
 libnormaliz-templated.o: $(LIBHEADERS) $(LIBSOURCES) libnormaliz-templated.cpp
-	$(CXX) $(CXXFLAGS) $(NORMFLAGS) -c libnormaliz-templated.cpp -o $@
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) $(NORMFLAGS) -c libnormaliz-templated.cpp -o $@
 
 libnormaliz.a: cone_property.o hilbert_series.o libnormaliz-templated.o
 	ar -cr $@ $^
diff -U 3 -dHrN Normaliz2.8.orig/source/Makefile Normaliz2.8/source/Makefile
--- Normaliz2.8.orig/source/Makefile	2012-11-11 19:05:18.610268519 -0800
+++ Normaliz2.8/source/Makefile	2012-11-11 19:06:14.265817851 -0800
@@ -17,13 +17,13 @@
 	$(MAKE) normaliz
 
 normaliz: $(SOURCES) $(HEADERS) libnormaliz/libnormaliz.a
-	$(CXX) $(CXXFLAGS) $(NORMFLAGS) Normaliz.cpp libnormaliz/libnormaliz.a $(GMPFLAGS) -o normaliz
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) $(NORMFLAGS) Normaliz.cpp libnormaliz/libnormaliz.a $(GMPFLAGS) -o normaliz
 
 normaliz1: $(SOURCES) $(HEADERS) $(LIBHEADERS) $(LIBSOURCES)
-	$(CXX) $(CXXFLAGS) $(NORMFLAGS) Normaliz-impl.cpp $(GMPFLAGS) -o normaliz1
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) $(NORMFLAGS) Normaliz-impl.cpp $(GMPFLAGS) -o normaliz1
 
 normaliz-pg: $(SOURCES) $(HEADERS) $(LIBHEADERS) $(LIBSOURCES)
-	$(CXX) $(CXXFLAGS) $(NORMFLAGS) -pg Normaliz-impl.cpp $(GMPFLAGS) -o normaliz-pg
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) $(NORMFLAGS) -pg Normaliz-impl.cpp $(GMPFLAGS) -o normaliz-pg
 
 
 #always go down the directory and let the make there check what has to be done
diff -U 3 -dHrN Normaliz2.8.orig/source/Makefile.configuration Normaliz2.8/source/Makefile.configuration
--- Normaliz2.8.orig/source/Makefile.configuration	2012-11-11 19:05:18.613268547 -0800
+++ Normaliz2.8/source/Makefile.configuration	2012-11-11 19:05:37.323453221 -0800
@@ -1,11 +1,9 @@
 ##
 ## Makefile configuration for normaliz
 ##
-CXX = g++ 
 #CXX = linux32 g++ -m32 -march=i686  #compile it for linux32
 CXXFLAGS += -Wall
 CXXFLAGS += -pedantic -Wno-long-long
-CXXFLAGS += -O3 -funroll-loops
 #CXXFLAGS += -I /usr/local/boost_1_45_0
 #CXXFLAGS += -g
 #CXXFLAGS += -pg
@@ -19,7 +17,6 @@
 endif
 
 ##for mac link libgcc not static
-NORMFLAGS = -static
 #NORMFLAGS = -static-libgcc
 
 GMPFLAGS = -lgmpxx -lgmp
