diff -U 3 -dHrN Normaliz2.10.org/source/Makefile Normaliz2.10/source/Makefile
--- Normaliz2.10.org/source/Makefile	2013-05-14 09:13:29.148646896 +0200
+++ Normaliz2.10/source/Makefile	2013-05-14 09:17:55.002658867 +0200
@@ -17,7 +17,7 @@
 	@$(MAKE) normaliz
 
 normaliz: $(SOURCES) $(HEADERS) libnormaliz/libnormaliz.a
-	$(CXX) $(CXXFLAGS) $(NORMFLAGS) Normaliz.cpp libnormaliz/libnormaliz.a $(GMPFLAGS) -o normaliz
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) $(NORMFLAGS) Normaliz.cpp libnormaliz/libnormaliz.a $(GMPFLAGS) -o normaliz
 
 normaliz1: $(SOURCES) $(HEADERS) $(LIBHEADERS) $(LIBSOURCES)
 	$(CXX) $(CXXFLAGS) $(NORMFLAGS) Normaliz-impl.cpp $(GMPFLAGS) -o normaliz1
diff -U 3 -dHrN Normaliz2.10.org/source/Makefile.configuration Normaliz2.10/source/Makefile.configuration
--- Normaliz2.10.org/source/Makefile.configuration	2013-05-14 09:13:29.149646895 +0200
+++ Normaliz2.10/source/Makefile.configuration	2013-05-14 09:13:44.469637334 +0200
@@ -1,11 +1,9 @@
 ##
 ## Makefile configuration for normaliz
 ##
-CXX = g++ 
 #CXX = linux32 g++ -m32 -march=i686  #compile it for linux32
 CXXFLAGS += -std=c++0x
 CXXFLAGS += -Wall -pedantic
-CXXFLAGS += -O3 -funroll-loops
 #CXXFLAGS += -I /usr/local/boost_1_45_0
 #CXXFLAGS += -g       ## debugging
 #CXXFLAGS += -pg      ## profiling
