diff -U 3 -dHrN Normaliz2.11/source/Makefile Normaliz2.11.new/source/Makefile
--- Normaliz2.11/source/Makefile	2014-02-24 12:44:10.000000000 +0100
+++ Normaliz2.11.new/source/Makefile	2014-05-09 15:26:52.885085512 +0200
@@ -20,7 +20,7 @@
 	$(CXX) $(CXXFLAGS) $(NORMFLAGS) -c Normaliz.cpp -o normaliz.o
 
 normaliz: $(SOURCES) $(HEADERS) normaliz.o libnormaliz/libnormaliz.a
-	$(CXX) $(CXXFLAGS) $(NORMFLAGS) normaliz.o libnormaliz/libnormaliz.a $(GMPFLAGS) -o normaliz
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) normaliz.o libnormaliz/libnormaliz.a $(GMPFLAGS) -o normaliz
 
 normaliz1: $(SOURCES) $(HEADERS) $(LIBHEADERS) $(LIBSOURCES)
 	$(CXX) $(CXXFLAGS) $(NORMFLAGS) Normaliz-impl.cpp $(GMPFLAGS) -o normaliz1
diff -U 3 -dHrN Normaliz2.11/source/Makefile.configuration Normaliz2.11.new/source/Makefile.configuration
--- Normaliz2.11/source/Makefile.configuration	2014-04-29 11:06:24.000000000 +0200
+++ Normaliz2.11.new/source/Makefile.configuration	2014-05-09 15:26:52.875085512 +0200
@@ -1,11 +1,9 @@
 ##
 ## Makefile configuration for normaliz
 ##
-CXX = g++ 
 #CXX = linux32 g++ -m32 -march=i686  #compile it for linux32
 CXXFLAGS += -std=c++0x
 CXXFLAGS += -Wall -pedantic
-CXXFLAGS += -O3 -funroll-loops
 #CXXFLAGS += -I /usr/local/boost_1_45_0
 #CXXFLAGS += -g       ## debugging
 #CXXFLAGS += -pg      ## profiling
@@ -19,7 +17,6 @@
 endif
 
 ## for distributing the executables link static (does not work for mac)
-CXXFLAGS += -static
 ## for almost static compilation on Mac use
 #CXXFLAGS += -static-libgcc
 ## make it compatible with older Mac versions
