From b9882960dab978b85cbd32fc9b31e588c8c817ff Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Wed, 10 Oct 2012 23:37:48 +0200
Subject: [PATCH] Drop deprecated g_thread_init

---
 components/editor/main.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/components/editor/main.c b/components/editor/main.c
index 0a0a37f..3c729a2 100644
--- a/components/editor/main.c
+++ b/components/editor/main.c
@@ -373,8 +373,6 @@ main (gint argc, gchar **argv)
 	bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
 	textdomain (GETTEXT_PACKAGE);
 
-	g_thread_init (NULL);
-
 	gtk_init (&argc, &argv);
 
 	editor = gtkhtml_editor_new ();
-- 
1.7.12

