--- setup.py
+++ setup.py
@@ -344,7 +344,7 @@
     mxincludedir = parser.get('build_ext', 'mx_include_dir')
 else:
     mxincludedir = os.path.join(get_python_inc(plat_specific=1), "mx")
-if os.path.exists(mxincludedir):
+if not use_pydatetime and os.path.exists(mxincludedir):
     include_dirs.append(mxincludedir)
     define_macros.append(('HAVE_MXDATETIME','1'))
     sources.append('adapter_mxdatetime.c')
