--- pngnq-1.0.orig/configure.ac	2009-05-20 14:05:52.000000000 +0200
+++ pngnq-1.0/configure.ac	2010-01-16 19:51:43.087613985 +0100
@@ -24,6 +24,10 @@
 AC_FUNC_STAT
 AC_HEADER_STDBOOL
 
+AC_SEARCH_LIBS([zlibVersion],[z])
+AC_SEARCH_LIBS([sqrt],[m])
+PKG_CHECK_MODULES([PNG], [libpng >= 1.2.0])
+
 # checks for library functions
 AC_CHECK_FUNCS([getopt])
 AC_CHECK_FUNCS([floor])
@@ -35,5 +39,5 @@
 AC_CHECK_FUNCS([strrchr])
 
 AC_CONFIG_HEADERS([src/config.h])
-AC_CONFIG_FILES([Makefile src/Makefile freegetopt/Makefile])
+AC_CONFIG_FILES([Makefile src/Makefile])
 AC_OUTPUT
--- pngnq-1.0.orig/Makefile.am	2009-05-20 14:05:52.000000000 +0200
+++ pngnq-1.0/Makefile.am	2010-01-16 19:50:25.127860135 +0100
@@ -1,3 +1,3 @@
-SUBDIRS = freegetopt src
+SUBDIRS = src
 
 dist_man_MANS = pngnq.1
--- pngnq-1.0.orig/src/Makefile.am	2009-05-20 14:04:15.000000000 +0200
+++ pngnq-1.0/src/Makefile.am	2010-01-16 20:06:48.150861688 +0100
@@ -1,6 +1,7 @@
-AM_LDFLAGS = ${shell libpng-config --ldflags} -lz
-AM_CFLAGS = ${shell libpng-config --I_opts} -Wall --pedantic -std=gnu99
+AM_CFLAGS = -Wall --pedantic -std=gnu99 @PNG_CFLAGS@
 
 bin_PROGRAMS = pngnq pngcomp
 pngnq_SOURCES = pngnq.c neuquant32.c rwpng.c  neuquant32.h rwpng.h errors.h
+pngnq_LDADD  = @PNG_LIBS@
 pngcomp_SOURCES = pngcomp.c rwpng.c colorspace.c  colorspace.h
+pngcomp_LDADD  = @PNG_LIBS@
