--- llpp-9999/work/llpp-9999/link.c
+++ llpp-9999/work/llpp-9999/link.c
@@ -30,12 +30,10 @@
 #include <caml/memory.h>
 #include <caml/unixsupport.h>
 
-#include <fitz.h>
-#include <mupdf.h>
-#include <mupdf-internal.h>
-#include <muxps.h>
-#include <muxps-internal.h>
-#include <mucbz.h>
+#include <mupdf/cbz.h>
+#include <mupdf/fitz.h>
+#include <mupdf/pdf.h>
+#include <mupdf/xps.h>
 
 #include FT_FREETYPE_H
 
