https://bugs.gentoo.org/420237

--- ecb-2.40-orig/ecb.el
+++ ecb-2.40/ecb.el
@@ -1312,7 +1312,6 @@
         (ecb-select-ecb-frame)
         (ecb-update-directories-buffer))
 
-    (let ((stack-trace-on-error stack-trace-on-error))
       ;; we activate only if all before-hooks return non nil
       (when (run-hook-with-args-until-failure 'ecb-before-activate-hook)
 
@@ -1602,7 +1601,7 @@
           (error
            (ecb-clean-up-after-activation-failure
             "Errors during the snapshot of the windows-configuration." err-obj)))
-        ))))
+        )))
 
 
 (defun ecb-deactivate ()
