diff -ru yaepg-0.0.2.1/yaepg.c yaepg-0.0.2.2/yaepg.c
--- yaepg-0.0.2.1/yaepg.c	2007-06-11 18:26:37.000000000 -0700
+++ yaepg-0.0.2.2/yaepg.c	2007-06-11 15:56:31.000000000 -0700
@@ -17,6 +17,7 @@
 static const tColor highlight		= (tColor) 0xFF505050;
 static const tColor lightgrey		= (tColor) 0xFFAAAAAA;
 
+#define fontYaepg			fontSml
 #define fontChanBox			fontYaepg
 #define fontTimeBox			fontYaepg
 #define fontEventBox			fontYaepg

