diff -ru yaepg-0.0.2.1-rev2-orig/yaepg.c yaepg-0.0.2.1-rev2/yaepg.c
--- yaepg-0.0.2.1-rev2-orig/yaepg.c	2007-07-02 10:37:25.934194467 +0200
+++ yaepg-0.0.2.1-rev2/yaepg.c	2007-07-02 10:37:46.436137435 +0200
@@ -1209,7 +1209,7 @@
 	cDateBox		*datebox;
 
 	cRecDlgBox		*recDlgBox;
-	uint64			msgboxStart;
+	uint64_t		msgboxStart;
 	cMessageBox		*msgbox;
 
 	/*
@@ -1235,7 +1235,7 @@
 	 * Direct Channel Change
 	 */
 	int			currChan;
-	uint64			lastInput;
+	uint64_t		lastInput;
 
 	/*
 	 * The last possible starting channel for the grid.
