From 6921fafa99726cec24cce0ce3b0be71944812ef7 Mon Sep 17 00:00:00 2001
From: Arno Rehn <arno@arnorehn.de>
Date: Thu, 15 Nov 2012 18:12:28 +0100
Subject: [PATCH] Apply a patch from Fedora to fix the build with cmake 2.8.10

---
 cmake/CMakeDetermineCSharpCompiler.cmake | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/cmake/CMakeDetermineCSharpCompiler.cmake b/cmake/CMakeDetermineCSharpCompiler.cmake
index 29c0eb8..862d9b4 100644
--- a/cmake/CMakeDetermineCSharpCompiler.cmake
+++ b/cmake/CMakeDetermineCSharpCompiler.cmake
@@ -84,6 +84,9 @@ endif (CMAKE_CSharp_COMPILER)
 get_filename_component(script_dir ${CMAKE_CURRENT_LIST_FILE} PATH)
 
 # configure variables set in this file for fast reload later on
-configure_file(${script_dir}/CMakeCSharpCompiler.cmake.in
-  ${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/CMakeCSharpCompiler.cmake IMMEDIATE @ONLY)
+ if(NOT CMAKE_PLATFORM_INFO_DIR) # pre-2.8.10
+   set(CMAKE_PLATFORM_INFO_DIR ${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY})
+ endif()
+ configure_file(${script_dir}/CMakeCSharpCompiler.cmake.in
+   ${CMAKE_PLATFORM_INFO_DIR}/CMakeCSharpCompiler.cmake IMMEDIATE @ONLY)
 set(CMAKE_CSharp_COMPILER_ENV_VAR "CSC")
-- 
1.8.0

