diff --git a/Makefile.am b/Makefile.am
index 0231ab8..2e26035 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -61,9 +61,12 @@ libfftw3@PREC_SUFFIX@_la_LDFLAGS = -no-undefined -version-info	\
 
 fftw3@PREC_SUFFIX@.pc: fftw.pc
 	cp -f fftw.pc fftw3@PREC_SUFFIX@.pc
+fftw3@PREC_SUFFIX@_threads.pc: fftw.pc
+	cp -f fftw.pc fftw3@PREC_SUFFIX@_threads.pc
+	sed 's|-lfftw3@PREC_SUFFIX@|-lfftw3@PREC_SUFFIX@_threads|g' \
+	-i fftw3@PREC_SUFFIX@_threads.pc
 pkgconfigdir = $(libdir)/pkgconfig
-pkgconfig_DATA = fftw3@PREC_SUFFIX@.pc
-
+pkgconfig_DATA = fftw3@PREC_SUFFIX@.pc fftw3@PREC_SUFFIX@_threads.pc
 WISDOM_DIR = /etc/fftw
 WISDOM = wisdom@PREC_SUFFIX@
 
