diff -urN kde-l10n-da-4.8.5.orig/docs/kdepim/korganizer/outlook-to-vcalendar.docbook kde-l10n-da-4.8.5/docs/kdepim/korganizer/outlook-to-vcalendar.docbook
--- kde-l10n-da-4.8.5.orig/docs/kdepim/korganizer/outlook-to-vcalendar.docbook	1970-01-01 01:00:00.000000000 +0100
+++ kde-l10n-da-4.8.5/docs/kdepim/korganizer/outlook-to-vcalendar.docbook	2012-08-17 17:10:27.213469844 +0200
@@ -0,0 +1,384 @@
+<chapter id="outlook-to-vcalendar-ws">
+
+<chapterinfo>
+<authorgroup>
+
+<author
+><firstname
+>Randy</firstname
+> <surname
+>Pearson</surname
+> <affiliation
+><address
+><email
+>blueboy@bamafolks.com</email
+></address
+></affiliation>
+<contrib
+>Oprindelig forfatter</contrib>
+</author>
+
+<author
+><firstname
+>Klaus</firstname
+> <surname
+>St&auml;rk</surname
+> <affiliation
+><address
+><email
+>staerk@kde.org</email
+></address
+></affiliation>
+<contrib
+>Yderligere tekster</contrib>
+</author>
+
+<othercredit role="reviewer"
+><firstname
+>Eric</firstname
+> <surname
+>Bischoff</surname
+> <affiliation
+><address
+><email
+>e.bischoff@noos.fr</email
+></address
+></affiliation>
+<contrib
+>Editor</contrib>
+</othercredit>
+
+
+</authorgroup>
+
+<abstract>
+<para
+>Denne udgave af &korganizer; workshop <quote
+>Outlook til vCalendar</quote
+> blev udgivet 17. oktober 2002. Se <filename
+>ChangeLog</filename
+> for detaljer.</para>
+<para
+>Bemærk at beskrivelserne og skærmaftrykkene svarer til version 3.1 af &korganizer;.</para>
+</abstract>
+
+<keywordset>
+<keyword
+>kdepim</keyword>
+<keyword
+>KOrganizer</keyword>
+<keyword
+>KOrganizer workshop</keyword>
+<keyword
+>Outlook</keyword>
+<keyword
+>vCalendar</keyword>
+<keyword
+>Outlook2VCal</keyword>
+</keywordset>
+<date
+>2002-10-17</date
+> <releaseinfo
+>3.1</releaseinfo
+> </chapterinfo>
+
+<title
+>KOrganizer workshop: Outlook til vCalendar</title>
+
+<sect1 id="outlook-to-vcalendar-overview">
+<title
+>Overblik</title>
+
+<para
+>Når du går væk fra &Microsoft; Outlook&reg; og begynder at bruge &kde;'s &korganizer; til at håndtere skemaer og aftaler, vil du formodentlig ønske at eksportere dine skemalægningsdata fra Outlook&reg; og importere dem til &korganizer;. &Microsoft;'s Outlook&reg; program har ganske vist et eksport-valg der kan gemme en aftale i industristandardens vCalendar-format. </para>
+
+<para
+>Uheldigvis vil Outlook&reg; kun eksportere aftalen du har valgt, en af gangen. Dette er helt fint hvis du kun ønsker at overføre nogle få aftaler, men overførsel af de 200-300 aftaler du måske har i din Outlook&reg; mappe ville ikke være særlig morsomt. Derfor lavede <ulink url="mailto:blueboy@bamafolks.com"
+>Randy Pearson</ulink
+> et lille program der hedder <application
+>Outlook2VCal</application
+> der kan skanne og eksportere alle aftalerne til en vCalendar-fil på én gang. </para>
+
+</sect1>
+
+<sect1 id="outlook-to-vcalendar-installation">
+<title
+>Installation</title>
+
+<para
+><application
+>Outlook2VCal 2.0</application
+>-programmet bliver <ulink url="http://korganizer.kde.org/importdata.html"
+>leveret som en komprimeret ZIP-fil</ulink
+>, der indeholder et <application
+>SETUP.EXE</application
+>-program. Træk simpelthen filerne ud ved brug af dit foretrukne redskab såsom <ulink url="http://www.winzip.com"
+>WinZip</ulink
+> og kør <application
+>SETUP.EXE</application
+>-programmet for at installere programmet på din Windows computer. Dette skal være den computer hvor du normalt kører &Microsoft; Outlook  97&reg;, 2000&reg; eller 2002&reg;. </para>
+
+<para
+>Afhængig af hvor opdateret dit system er, vil du måske, måske ikke blive bedt om at reboote under installationen. </para>
+
+</sect1>
+
+<sect1 id="outlook-to-vcalendar-export">
+<title
+>Brug af eksport</title>
+
+<para
+>Efter programmet er installeret, kan du køre ved at gå til <menuchoice
+><guimenu
+>Start</guimenu
+> <guisubmenu
+>Programmer</guisubmenu
+> <guisubmenu
+>Outlook2vCal</guisubmenu
+> <guimenuitem
+>Outlook2vCal</guimenuitem
+></menuchoice
+>. Du skulle så se denne skærm: </para>
+
+<screenshot id="screenshot-outlook2vcal-main">
+<screeninfo
+>Outlook2VCal hovedvindue</screeninfo>
+<mediaobject>
+<imageobject
+><imagedata fileref="o2v_main.png" format="PNG"/></imageobject>
+<textobject
+><phrase
+>Outlook2VCal hovedvindue</phrase
+></textobject>
+<caption
+><para
+>Outlook2VCal hovedvindue</para
+></caption>
+</mediaobject>
+</screenshot>
+
+
+<para
+>Programmet forsøger automatisk at kommunikere med Outlook&reg; ved brug af OLE-Automation. Hvis en fejl rapporteres, betyder dette sandsynligvis at du ikke har Outlook&reg; installeret. </para>
+
+<para
+>Du skal først beslutte om du ønsker at inkludere information om helligdage i den eksporterede fil eller ej. Normalt vil du nok ikke ønske at gøre dette, så <guilabel
+>Skip helligdage</guilabel
+>-feltet er som standard afkrydset. Dernæst klikkes på <guibutton
+>Export</guibutton
+>-knappen, hvilket vil vise en standard <guilabel
+>Gem fil</guilabel
+> dialog. Gå til det ønskede katalog og indtast filnavnet hvor du ønsker at lave den nye vCalendar datafil. </para>
+
+<screenshot id="screenshot-outlook2vcal-save">
+<screeninfo
+>Outlook2VCal dialog til at gemme fil</screeninfo>
+<mediaobject>
+<imageobject
+><imagedata fileref="o2v_save.png" format="PNG"/></imageobject>
+<textobject
+><phrase
+>Outlook2VCal dialog til at gemme fil</phrase
+></textobject>
+<caption
+><para
+>Outlook2VCal dialog til at gemme fil</para
+></caption>
+</mediaobject>
+</screenshot>
+
+<para
+>Afhængig af sikkerhedsindstillingerne i Outlook&reg;, vil du måske se denne skærm derefter. Hvis du gør, så sørg for at tillade adgang et minut eller to. </para>
+
+<screenshot id="screenshot-outlook2vcal-warning">
+<screeninfo
+>Outlook2VCal advarselsmeddelelse</screeninfo>
+<mediaobject>
+<imageobject
+><imagedata fileref="o2v_warning.png" format="PNG"/></imageobject>
+<textobject
+><phrase
+>Outlook2VCal advarselsmeddelelse</phrase
+></textobject>
+<caption
+><para
+>Outlook2VCal advarselsmeddelelse</para
+></caption>
+</mediaobject>
+</screenshot>
+
+<para
+>Programmet vil vise fremgangsbeskeder i sit hovedvindue mens det indlæser, konverterer og gemmer dine aftaler. Her er et endeligt billede af hvordan en succesrig kørsel ser ud: </para>
+
+<screenshot id="screenshot-outlook2vcal-progress">
+<screeninfo
+>Outlook2VCal fremgangsmeddelelse</screeninfo>
+<mediaobject>
+<imageobject
+><imagedata fileref="o2v_progress.png" format="PNG"/></imageobject>
+<textobject
+><phrase
+>Outlook2VCal fremgangsmeddelelse</phrase
+></textobject>
+<caption
+><para
+>Outlook2VCal fremgangsmeddelelse</para
+></caption>
+</mediaobject>
+</screenshot>
+
+</sect1>
+
+<sect1 id="outlook-to-vcalendar-import">
+<title
+>Brug af import</title>
+
+<para
+>Nu da du har eksporteret Outlook's&reg; aftaler, er det sidste skridt at indlæse aftalerne i &korganizer;. Overfør filen til computeren der kører &korganizer; (via netværk, floppy disk, hvad som helst). Kør nu &korganizer;. Om ønsket kan du lave en ny kalender, eller åbne en eksisterende. </para>
+
+<screenshot id="screenshot-outlook2vcal-importing1">
+<screeninfo
+>Outlook2VCal import af data nummer 1</screeninfo>
+<mediaobject>
+<imageobject
+><imagedata fileref="o2v_importing1.png" format="PNG"/></imageobject>
+<textobject
+><phrase
+>Outlook2VCal import af data nummer 1</phrase
+></textobject>
+<caption
+><para
+>Outlook2VCal import af data nummer 1</para
+></caption>
+</mediaobject>
+</screenshot>
+
+
+<para
+>Vælg nu menupunktet <menuchoice
+><guimenu
+>Fil</guimenu
+> <guisubmenu
+>Importér</guisubmenu
+> <guimenuitem
+>Importér kalender</guimenuitem
+></menuchoice
+>. Gå til eller indskriv navnet på vCalender-filen som du netop overførte over til maskinen. </para>
+
+<screenshot id="screenshot-outlook2vcal-importing2">
+<screeninfo
+>Outlook2VCal import af data nummer 2</screeninfo>
+<mediaobject>
+<imageobject
+><imagedata fileref="o2v_importing2.png" format="PNG"/></imageobject>
+<textobject
+><phrase
+>Outlook2VCal import af data nummer 2</phrase
+></textobject>
+<caption
+><para
+>Outlook2VCal import af data nummer 2</para
+></caption>
+</mediaobject>
+</screenshot>
+
+<para
+>&korganizer; spørger om du vil <guilabel
+>tilføje som ny kalender</guilabel
+>, hvilket tilføjer kalenderfilen som en ny lokal filressource, <guilabel
+>sammenflette med eksisterende kalender</guilabel
+>, hvilket sammenfletter kalenderindgangene med en eksisterende ressource eller <guilabel
+>åbne i separat vindue</guilabel
+>, hvilket lader dig vise og redigere kalenderen, men ikke tilføjer den i standardvisningen. For at tilføje kalenderindgangene, vælges en af de to første valgmuligheder, og der klikkes på <guibutton
+>O.k</guibutton
+>.</para>
+
+<para
+>Efter du har valgt <guibutton
+>O.k.</guibutton
+>-knappen, vil &korganizer; indlæse aftalerne fra filen og opdatere sin kalender. </para>
+
+<screenshot id="screenshot-outlook2vcal-importing3">
+<screeninfo
+>Outlook2VCal import af data nummer 3</screeninfo>
+<mediaobject>
+<imageobject
+><imagedata fileref="o2v_importing3.png" format="PNG"/></imageobject>
+<textobject
+><phrase
+>Outlook2VCal import af data nummer 3</phrase
+></textobject>
+<caption
+><para
+>Outlook2VCal import af data nummer 3</para
+></caption>
+</mediaobject>
+</screenshot>
+<para
+>Gem din ændrede kalender og nyd at bruge &korganizer;! </para>
+
+</sect1>
+
+<sect1 id="outlook-to-vcalendar-limitations">
+<title
+>Begrænsninger</title>
+
+<para
+>Den første udgave af dette program blev testet på &Microsoft; Windows 2000 Server&reg; og Professional&reg; kørende &Microsoft; Outlook 2000&reg;. Når det forsøgtes at bruge &Microsoft; Outlook 97&reg;, mislykkedes programmet fuldstændigt. </para>
+
+<para
+>Den anden version (2.0) rapporteres som (i det mindste) at virke med &Microsoft; Outlook  97&reg;, 2000&reg;, og 2002&reg;. </para>
+
+<para
+>Hvis nogen kender en måde at omgå dette problem, så send venligst en note til Randy Pearson (<email
+>blueboy@bamafolks.com</email
+>). </para>
+
+</sect1>
+
+<sect1 id="outlook-to-vcalendar-credits">
+<title
+>Medvirkende og licens</title>
+
+<para
+>Bidragydere:</para>
+
+<variablelist>
+<varlistentry>
+<term
+>Oprindelig forfatter</term>
+<listitem
+><para
+>Randy Pearson <email
+>blueboy@bamafolks.com</email
+></para
+></listitem>
+</varlistentry>
+
+<varlistentry>
+<term
+>Yderligere tekster</term>
+<listitem
+><para
+>Klaus St&auml;rk <email
+>staerk@kde.org</email
+></para
+></listitem>
+</varlistentry>
+
+<varlistentry>
+<term
+>Konversion til DocBook</term>
+<listitem
+><para
+>Eric Bischoff <email
+>e.bischoff@noos.fr</email
+></para
+></listitem>
+</varlistentry>
+</variablelist>
+
+
+</sect1>
+
+</chapter>
diff -urN kde-l10n-da-4.8.5.orig/docs/kdepim/korganizer/plugins-chapter.docbook kde-l10n-da-4.8.5/docs/kdepim/korganizer/plugins-chapter.docbook
--- kde-l10n-da-4.8.5.orig/docs/kdepim/korganizer/plugins-chapter.docbook	1970-01-01 01:00:00.000000000 +0100
+++ kde-l10n-da-4.8.5/docs/kdepim/korganizer/plugins-chapter.docbook	2012-08-17 17:10:20.829470030 +0200
@@ -0,0 +1,53 @@
+<chapter id="plugins-chapter">
+
+<title
+>Plugin</title>
+
+<para
+>&korganizer; giver mulighed for at udvide programmet med plugin. </para>
+
+<para
+>Plugin kan indstilles i &korganizer;s hovedindstillingsdialog. Vælg <menuchoice
+><guimenu
+>Opsætning</guimenu
+> <guimenuitem
+>Indstil &korganizer;</guimenuitem
+></menuchoice
+> eller <menuchoice
+><guimenu
+>Opsætning</guimenu
+> <guimenuitem
+>Indstil kalender</guimenuitem
+></menuchoice
+> inde i &kontact;, og klik på ikonen <guilabel
+>Plugin</guilabel
+> i ikonlisten i dialogens sidebjælke. </para>
+
+<sect1 id="plugins-chapter-availableplugins">
+<title
+>Tilgængelige plugin</title>
+
+<sect2 id="date-numbers-plugin">
+<title
+>Plugin for datonumre i kalendere</title>
+<para
+>Dette plugin tilføjer dagens nummer til dagen i året i agendavisningen. Første februar er for eksempel dag nummer 32. </para>
+</sect2>
+
+<sect2 id="jewish-calendar-plugin">
+<title
+>Plugin for jødisk kalender</title>
+<para
+>Pluginnet sørger for den jødiske kalenderdato udover den Gregorianska kalenderdato. Du kan vælge om du vil vise israelske helligdage, ugens parsha, Omers dag og/eller Chol HaMoed, i pluginnets indstillingsdialog. </para>
+</sect2>
+
+<sect2 id="journal-print-plugin">
+<title
+>Journal-udskriftsstil</title>
+<para
+>Dette plugin gør at du kan udskrive journal-indgange (dagbogsnoter). </para>
+</sect2>
+
+</sect1>
+
+</chapter>
