--- chess.h.old	2013-04-07 23:04:54.708245856 +0200
+++ chess.h	2013-04-07 23:05:20.052002160 +0200
@@ -140,13 +140,7 @@
 #  define QUEEN_VALUE                           1050
 #  define KING_VALUE                           40000
 #  define MAX_DRAFT                              256
-#  if defined(HAS_64BITS)
-typedef unsigned long BITBOARD;
-#  elif defined(NT_i386)
-typedef unsigned __int64 BITBOARD;
-#  else
 typedef unsigned long long BITBOARD;
-#  endif
 #  if defined(NT_i386)
 #    define BMF   "%I64u"
 #    define BMF6  "%6I64u"
