diff -Naur tlswrap-1.02/tlswrap.c tlswrap-1.02-new/tlswrap.c
--- tlswrap-1.02/tlswrap.c	2005-07-09 16:11:37.000000000 +0200
+++ tlswrap-1.02-new/tlswrap.c	2006-05-09 22:38:58.000000000 +0200
@@ -358,9 +358,9 @@
 
 	listen_fd = setup_listen(5, cfg_listenhost, cfg_listenport, 0);
 
-	fprintf(stderr,
+	fprintf(stdout,
 	    "TLSWrap %s (c) 2002-2005 Tomas Svensson <ts@unix1.net>\n", TLSWRAP_VERSION_TEXT);
-	fprintf(stderr, "Servicing up to %u clients on %s:%s\n", cfg_max_users, cfg_listenhost, cfg_listenport);
+	fprintf(stdout, "Servicing up to %u clients on %s:%s\n", cfg_max_users, cfg_listenhost, cfg_listenport);
 #if !defined __CYGWIN__ && !defined WIN32
 #ifdef __HAVE_DAEMON
 	if (!debug)
@@ -370,7 +370,7 @@
 		if ( (childpid = fork()) < 0)
 			sys_err("fork()");
 		else if (childpid != 0) {
-			fprintf(stderr, "Running as process %u\n", (unsigned int)childpid);
+			fprintf(stdout, "Running as process %u\n", (unsigned int)childpid);
 			exit(0); /* parent */
 		}
 		(void)setsid();
