diff -Nuar keyutils-1.2.orig/Makefile keyutils-1.2/Makefile
--- keyutils-1.2.orig/Makefile	2008-06-06 19:35:39.750214054 -0700
+++ keyutils-1.2/Makefile	2008-06-06 19:45:38.504078903 -0700
@@ -72,10 +72,10 @@
 	$(CC) $(CFLAGS) -fPIC -o $@ -c $<
 
 
-keyctl: keyctl.c keyutils.h Makefile
+keyctl: keyctl.c keyutils.h Makefile $(DEVELLIB)
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $< -L. -lkeyutils -Wl,-rpath,$(LIB)
 
-request-key: request-key.c keyutils.h Makefile
+request-key: request-key.c keyutils.h Makefile $(DEVELLIB)
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $< -L. -lkeyutils -Wl,-rpath,$(LIB)
 
 
@@ -91,7 +91,7 @@
 	$(INSTALL) -D $(LIBNAME) $(DESTDIR)$(LIBDIR)/$(LIBNAME)
 	$(LNS) $(LIBNAME) $(DESTDIR)$(LIBDIR)/$(SONAME)
 	mkdir -p $(DESTDIR)$(USRLIBDIR)
-	$(LNS) $(LIBDIR)/$(SONAME) $(DESTDIR)$(USRLIBDIR)/$(DEVELLIB)
+	$(LNS) $(SONAME) $(DESTDIR)$(LIBDIR)/$(DEVELLIB)
 	$(INSTALL) -D keyctl $(DESTDIR)$(BINDIR)/keyctl
 	$(INSTALL) -D request-key $(DESTDIR)$(SBINDIR)/request-key
 	$(INSTALL) -D request-key-debug.sh $(DESTDIR)$(SHAREDIR)/request-key-debug.sh
