diff -Nur a/Makefile b/Makefile
--- a/Makefile	2013-04-30 10:29:31.000000000 +0200
+++ b/Makefile	2013-05-10 21:07:51.189902648 +0200
@@ -22,7 +22,9 @@
 
 ifeq ($(WITH_ACL),yes)
 CFLAGS += -DWITH_ACL
+ifneq ($(OS_NAME),FreeBSD)
 LOADLIBES += -lacl
+endif
 # See pretest
 TEST_ACL=1
 else
diff -Nur a/config.c b/config.c
--- a/config.c	2013-05-10 21:03:29.309904119 +0200
+++ b/config.c	2013-05-10 21:05:13.989903531 +0200
@@ -1,6 +1,6 @@
 #include <sys/queue.h>
 /* Alloca is defined in stdlib.h in NetBSD */
-#ifndef __NetBSD__
+#if !defined(__NetBSD__) && !defined(__FreeBSD__)
 #include <alloca.h>
 #endif
 #include <limits.h>
@@ -24,6 +24,10 @@
 #include <fnmatch.h>
 #include <sys/mman.h>
 
+#if !defined(PATH_MAX) && defined(__FreeBSD__)
+#include <sys/param.h>
+#endif
+
 #include "basenames.h"
 #include "log.h"
 #include "logrotate.h"
diff -Nur a/logrotate.c b/logrotate.c
--- a/logrotate.c	2013-05-10 20:58:58.229905641 +0200
+++ b/logrotate.c	2013-05-10 21:06:07.459903230 +0200
@@ -1,6 +1,6 @@
 #include <sys/queue.h>
 /* alloca() is defined in stdlib.h in NetBSD */
-#ifndef __NetBSD__
+#if !defined(__NetBSD__) && !defined(__FreeBSD__)
 #include <alloca.h>
 #endif
 #include <limits.h>
@@ -41,6 +41,10 @@
 
 static acl_type prev_acl = NULL;
 
+#if !defined(PATH_MAX) && defined(__FreeBSD__)
+#include <sys/param.h>
+#endif
+
 #include "basenames.h"
 #include "log.h"
 #include "logrotate.h"
