diff -Nur a/logrotate.c b/logrotate.c
--- a/logrotate.c	2012-10-04 09:59:14.000000000 +0200
+++ b/logrotate.c	2012-11-09 20:50:01.659874980 +0100
@@ -1832,7 +1832,7 @@
 	}
 
 	/* Hack to hide earlier bug */
-	if ((year != 1900) && (year < 1996 || year > 2100)) {
+	if ((year != 1900) && (year < 1970 || year > 2100)) {
 	    message(MESS_ERROR,
 		    "bad year %d for file %s in state file %s\n", year,
 		    argv[0], stateFilename);
