diff -Nur a/logrotate.c b/logrotate.c
--- a/logrotate.c	2013-06-10 14:02:36.000000000 +0200
+++ b/logrotate.c	2013-06-18 18:34:09.329999708 +0200
@@ -2035,7 +2035,7 @@
 	}
 
 	/* Hack to hide earlier bug */
-	if ((year != 1900) && (year < 1996 || year > 2100)) {
+	if ((year != 1900) && (year < 1970 || year > 2100)) {
 	    message(MESS_ERROR,
 		    "bad year %d for file %s in state file %s\n", year,
 		    argv[0], stateFilename);
