diff -ur xen-4.3.0.orig/tools/debugger/gdbsx/gx/Makefile xen-4.3.0/tools/debugger/gdbsx/gx/Makefile
--- tools/debugger/gdbsx/gx/Makefile	2013-07-09 18:46:56.000000000 +0800
+++ tools/debugger/gdbsx/gx/Makefile	2013-07-23 16:14:53.930752663 +0800
@@ -3,6 +3,7 @@
 
 GX_OBJS := gx_comm.o gx_main.o gx_utils.o gx_local.o
 GX_HDRS := $(wildcard *.h)
+AR="ar"
 
 .PHONY: all
 all: gx_all.a
@@ -16,5 +17,5 @@
 #	$(CC) -c $(CFLAGS) -o $@ $<
 
 gx_all.a: $(GX_OBJS) Makefile $(GX_HDRS)
-	ar cr $@ $(GX_OBJS)        # problem with ld using -m32 
+	$(AR) cr $@ $(GX_OBJS)        # problem with ld using -m32 
 
