diff -ur xen-4.2.1.orig/tools/examples/xend-config.sxp xen-4.2.1/tools/examples/xend-config.sxp
--- tools/examples/xend-config.sxp	2012-12-17 23:00:23.000000000 +0800
+++ tools/examples/xend-config.sxp	2013-06-27 21:32:50.198221774 +0800
@@ -200,7 +200,7 @@
 #(network-script network-nat)
 #(vif-script     vif-nat)
 
-# dom0-min-mem is the lowest permissible memory level (in MB) for dom0.
+# dom0-min-mem is the lowest permissible memory level (in kb) for dom0.
 # This is a minimum both for auto-ballooning (as enabled by
 # enable-dom0-ballooning below) and for xm mem-set when applied to dom0.
 (dom0-min-mem 196)
diff -ur xen-4.2.1.orig/tools/python/xen/xend/tests/xend-config.sxp xen-4.2.1/tools/python/xen/xend/tests/xend-config.sxp
--- tools/python/xen/xend/tests/xend-config.sxp	2012-12-17 23:01:27.000000000 +0800
+++ tools/python/xen/xend/tests/xend-config.sxp	2013-06-27 21:32:02.726225130 +0800
@@ -116,7 +116,7 @@
 
 
 # Dom0 will balloon out when needed to free memory for domU.
-# dom0-min-mem is the lowest memory level (in MB) dom0 will get down to.
+# dom0-min-mem is the lowest memory level (in kb) dom0 will get down to.
 # If dom0-min-mem=0, dom0 will never balloon out.
 (dom0-min-mem 196)
 
