--- xen-4.2.0/tools/firmware/hvmloader/Makefile.orig	2012-05-27 21:57:04.481812859 +0100
+++ xen-4.2.0/tools/firmware/hvmloader/Makefile	2012-06-02 18:52:44.935034128 +0100
@@ -48,7 +48,7 @@
 else
 CIRRUSVGA_ROM := ../vgabios/VGABIOS-lgpl-latest.cirrus.bin
 endif
-ETHERBOOT_ROMS := $(addprefix ../etherboot/ipxe/src/bin/, $(addsuffix .rom, $(ETHERBOOT_NICS)))
+ETHERBOOT_ROMS := $(addprefix /usr/share/ipxe/, $(addsuffix .rom, $(ETHERBOOT_NICS)))
 endif
 
 ROMS := 
--- xen-4.2.0/Config.mk.orig	2012-05-27 21:57:04.479812884 +0100
+++ xen-4.2.0/Config.mk	2012-06-02 18:55:14.087169469 +0100
@@ -206,7 +206,7 @@
 # Wed Jun 26 16:30:45 2013 +0100
 # xen: Don't perform SMP setup.
 
-ETHERBOOT_NICS ?= rtl8139 8086100e
+ETHERBOOT_NICS ?= 10ec8139 8086100e
 
 # Specify which qemu-dm to use. This may be `ioemu' to use the old
 # Mercurial in-tree version, or a local directory, or a git URL.
--- xen-4.2.0/tools/firmware/Makefile.orig	2012-05-27 21:57:04.480812871 +0100
+++ xen-4.2.0/tools/firmware/Makefile	2012-06-02 19:03:52.254691484 +0100
@@ -10,7 +10,7 @@
 #SUBDIRS-$(CONFIG_SEABIOS) += seabios-dir
 SUBDIRS-$(CONFIG_ROMBIOS) += rombios
 SUBDIRS-$(CONFIG_ROMBIOS) += vgabios
-SUBDIRS-$(CONFIG_ROMBIOS) += etherboot
+#SUBDIRS-$(CONFIG_ROMBIOS) += etherboot
 SUBDIRS-y += hvmloader
 
 ovmf:
