# Fix gcc-4.6
diff -ur xen-4.2.0.orig/extras/mini-os/minios.mk xen-4.2.0/extras/mini-os/minios.mk
--- extras/mini-os/minios.mk	2012-09-17 18:21:17.000000000 +0800
+++ extras/mini-os/minios.mk	2012-12-05 14:01:10.653260260 +0800
@@ -6,7 +6,7 @@
 
 # Define some default flags.
 # NB. '-Wcast-qual' is nasty, so I omitted it.
-DEF_CFLAGS += -fno-builtin -Wall -Werror -Wredundant-decls -Wno-format -Wno-redundant-decls
+DEF_CFLAGS += -fno-builtin -Wall -Wredundant-decls -Wno-format -Wno-redundant-decls
 DEF_CFLAGS += $(call cc-option,$(CC),-fno-stack-protector,)
 DEF_CFLAGS += $(call cc-option,$(CC),-fgnu89-inline)
 DEF_CFLAGS += -Wstrict-prototypes -Wnested-externs -Wpointer-arith -Winline
diff -ur xen-4.2.0.orig/tools/libxc/Makefile xen-4.2.0/tools/libxc/Makefile
--- tools/libxc/Makefile       2012-09-17 18:21:18.000000000 +0800
+++ tools/libxc/Makefile       2012-12-05 14:01:10.653260260 +0800
@@ -85,7 +85,7 @@

 -include $(XEN_TARGET_ARCH)/Makefile

-CFLAGS   += -Werror -Wmissing-prototypes
+CFLAGS   += -Wmissing-prototypes
 CFLAGS   += -I. $(CFLAGS_xeninclude)

 # Needed for posix_fadvise64() in xc_linux.c
# Drop .config
diff -ur xen-4.2.0.orig/Config.mk xen-4.2.0/Config.mk
--- Config.mk	2012-09-17 18:23:12.000000000 +0800
+++ Config.mk	2012-12-05 14:01:10.641260261 +0800
@@ -7,7 +7,6 @@ Drop .config
 # fallback for older make
 realpath = $(wildcard $(foreach file,$(1),$(shell cd -P $(dir $(file)) && echo "$$PWD/$(notdir $(file))")))
 
--include $(XEN_ROOT)/.config
 
 # A debug build of Xen and tools?
 debug ?= n
@@ -24,7 +24,7 @@
 
 # Tools to run on system hosting the build
 HOSTCC      = gcc
-HOSTCFLAGS  = -Wall -Werror -Wstrict-prototypes -O2 -fomit-frame-pointer
+HOSTCFLAGS  = -Wstrict-prototypes -O2 -fomit-frame-pointer
 HOSTCFLAGS += -fno-strict-aliasing
 
 DISTDIR     ?= $(XEN_ROOT)/dist
@@ -156,7 +156,7 @@
 
 CFLAGS += -std=gnu99
 
-CFLAGS += -Wall -Wstrict-prototypes
+CFLAGS += -Wstrict-prototypes
 
 # Clang complains about macros that expand to 'if ( ( foo == bar ) ) ...'
 # and is over-zealous with the printf format lint
diff -ur xen-4.2.1.orig/tools/blktap2/drivers/Makefile xen-4.2.1/tools/blktap2/drivers/Makefile
--- tools/blktap2/drivers/Makefile	2012-12-17 23:00:11.000000000 +0800
+++ tools/blktap2/drivers/Makefile	2013-01-30 12:31:43.539941099 +0800
@@ -9,7 +9,7 @@
 LOCK_UTIL  = lock-util
 INST_DIR   = $(SBINDIR)
 
-CFLAGS    += -Werror -g
+CFLAGS    += -g
 CFLAGS    += -Wno-unused
 CFLAGS    += -fno-strict-aliasing
 CFLAGS    += -I$(BLKTAP_ROOT)/include -I$(BLKTAP_ROOT)/drivers
diff -ur xen-4.2.1.orig/tools/debugger/gdbsx/Rules.mk xen-4.2.1/tools/debugger/gdbsx/Rules.mk
--- tools/debugger/gdbsx/Rules.mk	2012-12-17 23:00:22.000000000 +0800
+++ tools/debugger/gdbsx/Rules.mk	2013-01-30 12:31:43.516941098 +0800
@@ -1,4 +1,4 @@
 include $(XEN_ROOT)/tools/Rules.mk
 
-CFLAGS   += -Werror -Wmissing-prototypes 
+CFLAGS   += -Wmissing-prototypes 
 # (gcc 4.3x and later)   -Wconversion -Wno-sign-conversion
diff -ur xen-4.2.1.orig/tools/libaio/harness/Makefile xen-4.2.1/tools/libaio/harness/Makefile
--- tools/libaio/harness/Makefile	2012-12-17 23:00:35.000000000 +0800
+++ tools/libaio/harness/Makefile	2013-01-30 12:31:43.541941099 +0800
@@ -4,7 +4,7 @@
 HARNESS_SRCS:=main.c
 # io_queue.c
 
-CFLAGS=-Wall -Werror -g -O -laio
+CFLAGS=-Wall -g -O -laio
 #-lpthread -lrt
 
 all: $(PROGS)
diff -ur xen-4.2.1.orig/tools/libfsimage/Rules.mk xen-4.2.1/tools/libfsimage/Rules.mk
--- tools/libfsimage/Rules.mk	2012-12-17 23:00:36.000000000 +0800
+++ tools/libfsimage/Rules.mk	2013-01-30 12:31:43.515941097 +0800
@@ -1,7 +1,7 @@
 include $(XEN_ROOT)/tools/Rules.mk
 
 CFLAGS += -Wno-unknown-pragmas -I$(XEN_ROOT)/tools/libfsimage/common/ -DFSIMAGE_FSDIR=\"$(FSDIR)\"
-CFLAGS += -Werror -D_GNU_SOURCE
+CFLAGS += -D_GNU_SOURCE
 LDFLAGS += -L../common/
 
 PIC_OBJS := $(patsubst %.c,%.opic,$(LIB_SRCS-y))
diff -ur xen-4.2.1.orig/tools/libxl/Makefile xen-4.2.1/tools/libxl/Makefile
--- tools/libxl/Makefile	2012-12-17 23:01:08.000000000 +0800
+++ tools/libxl/Makefile	2013-01-30 12:31:43.541941099 +0800
@@ -11,7 +11,7 @@
 XLUMAJOR = 1.0
 XLUMINOR = 1
 
-CFLAGS += -Werror -Wno-format-zero-length -Wmissing-declarations \
+CFLAGS += -Wno-format-zero-length -Wmissing-declarations \
 	-Wno-declaration-after-statement -Wformat-nonliteral
 CFLAGS += -I. -fPIC
 
diff -ur xen-4.2.1.orig/tools/qemu-xen/pc-bios/optionrom/Makefile xen-4.2.1/tools/qemu-xen/pc-bios/optionrom/Makefile
--- tools/qemu-xen/pc-bios/optionrom/Makefile	2012-09-11 02:10:52.000000000 +0800
+++ tools/qemu-xen/pc-bios/optionrom/Makefile	2013-01-30 12:31:43.528941098 +0800
@@ -9,7 +9,7 @@
 
 .PHONY : all clean build-all
 
-CFLAGS := -Wall -Wstrict-prototypes -Werror -fomit-frame-pointer -fno-builtin
+CFLAGS := -Wall -Wstrict-prototypes -fomit-frame-pointer -fno-builtin
 CFLAGS += -I$(SRC_PATH)
 CFLAGS += $(call cc-option, $(CFLAGS), -fno-stack-protector)
 QEMU_CFLAGS = $(CFLAGS)
diff -ur xen-4.2.1.orig/tools/xenstat/xentop/Makefile xen-4.2.1/tools/xenstat/xentop/Makefile
--- tools/xenstat/xentop/Makefile	2012-12-17 23:01:35.000000000 +0800
+++ tools/xenstat/xentop/Makefile	2013-01-30 12:31:43.535941098 +0800
@@ -18,7 +18,7 @@
 all install xentop:
 else
 
-CFLAGS += -DGCC_PRINTF -Wall -Werror $(CFLAGS_libxenstat)
+CFLAGS += -DGCC_PRINTF -Wall $(CFLAGS_libxenstat)
 LDLIBS += $(LDLIBS_libxenstat) $(CURSES_LIBS) $(SOCKET_LIBS)
 CFLAGS += -DHOST_$(XEN_OS)
 
diff -ur xen-4.2.1.orig/xen/arch/arm/Rules.mk xen-4.2.1/xen/arch/arm/Rules.mk
--- xen/arch/arm/Rules.mk	2012-12-17 23:01:37.000000000 +0800
+++ xen/arch/arm/Rules.mk	2013-01-30 12:31:43.498941097 +0800
@@ -9,7 +9,7 @@
 HAS_DEVICE_TREE := y
 
 CFLAGS += -fno-builtin -fno-common -Wredundant-decls
-CFLAGS += -iwithprefix include -Werror -Wno-pointer-arith -pipe
+CFLAGS += -iwithprefix include -Wno-pointer-arith -pipe
 CFLAGS += -I$(BASEDIR)/include
 
 # Prevent floating-point variables from creeping into Xen.
diff -ur xen-4.2.1.orig/xen/arch/x86/Rules.mk xen-4.2.1/xen/arch/x86/Rules.mk
--- xen/arch/x86/Rules.mk	2012-12-17 23:01:37.000000000 +0800
+++ xen/arch/x86/Rules.mk	2013-01-30 12:31:43.490941096 +0800
@@ -24,7 +24,7 @@
 endif
 
 CFLAGS += -fno-builtin -fno-common -Wredundant-decls
-CFLAGS += -iwithprefix include -Werror -Wno-pointer-arith -pipe
+CFLAGS += -iwithprefix include -Wno-pointer-arith -pipe
 CFLAGS += -I$(BASEDIR)/include 
 CFLAGS += -I$(BASEDIR)/include/asm-x86/mach-generic
 CFLAGS += -I$(BASEDIR)/include/asm-x86/mach-default
diff -ur xen-4.2.1.orig/xen/include/Makefile xen-4.2.1/xen/include/Makefile
--- xen/include/Makefile	2012-12-17 23:01:55.000000000 +0800
+++ xen/include/Makefile	2013-01-30 12:31:43.502941097 +0800
@@ -78,7 +78,7 @@
 all: headers.chk
 
 headers.chk: $(filter-out public/arch-% public/%ctl.h public/xsm/% public/%hvm/save.h, $(wildcard public/*.h public/*/*.h) $(public-y)) Makefile
-	for i in $(filter %.h,$^); do $(CC) -ansi -include stdint.h -Wall -W -Werror -S -o /dev/null -xc $$i || exit 1; echo $$i; done >$@.new
+	for i in $(filter %.h,$^); do $(CC) -ansi -include stdint.h -Wall -W -S -o /dev/null -xc $$i || exit 1; echo $$i; done >$@.new
 	mv $@.new $@
 
 endif
diff -ur xen-4.2.1.orig/tools/tests/mce-test/tools/Makefile xen-4.2.1/tools/tests/mce-test/tools/Makefile
--- tools/tests/mce-test/tools/Makefile  2012-12-17 23:01:35.000000000 +0800
+++ tools/tests/mce-test/tools/Makefile       2013-01-30 13:01:44.890020152 +0800
@@ -1,7 +1,7 @@
 XEN_ROOT=$(CURDIR)/../../../..
 include $(XEN_ROOT)/tools/Rules.mk
 
-CFLAGS += -Werror
+CFLAGS +=
 CFLAGS += $(CFLAGS_libxenctrl)
 CFLAGS += $(CFLAGS_libxenguest)
 CFLAGS += $(CFLAGS_libxenstore)
diff -ur xen-4.2.1.orig/tools/tests/mem-sharing/Makefile xen-4.2.1/tools/tests/mem-sharing/Makefile
--- tools/tests/mem-sharing/Makefile     2012-12-17 23:01:35.000000000 +0800
+++ tools/tests/mem-sharing/Makefile  2013-01-30 13:01:44.890020152 +0800
@@ -1,7 +1,7 @@
 XEN_ROOT=$(CURDIR)/../../..
 include $(XEN_ROOT)/tools/Rules.mk
 
-CFLAGS += -Werror
+CFLAGS +=
 
 CFLAGS += $(CFLAGS_libxenctrl)
 CFLAGS += $(CFLAGS_xeninclude)
diff -ur xen-4.2.1.orig/tools/tests/xen-access/Makefile xen-4.2.1/tools/tests/xen-access/Makefile
--- tools/tests/xen-access/Makefile      2012-12-17 23:01:35.000000000 +0800
+++ tools/tests/xen-access/Makefile   2013-01-30 13:01:44.891020152 +0800
@@ -1,7 +1,7 @@
 XEN_ROOT=$(CURDIR)/../../..
 include $(XEN_ROOT)/tools/Rules.mk
 
-CFLAGS += -Werror
+CFLAGS +=
 
 CFLAGS += $(CFLAGS_libxenctrl)
 CFLAGS += $(CFLAGS_libxenguest)
diff -ur xen-4.3.0.orig/tools/firmware/Rules.mk xen-4.3.0/tools/firmware/Rules.mk
--- tools/firmware/Rules.mk	2013-07-09 18:46:56.000000000 +0800
+++ ttools/firmware/Rules.mk	2013-07-20 20:28:57.811836505 +0800
@@ -11,8 +11,6 @@
 CFLAGS += -DNDEBUG
 endif
 
-CFLAGS += -Werror
-
 $(call cc-options-add,CFLAGS,CC,$(EMBEDDED_EXTRA_CFLAGS))
 
 # Extra CFLAGS suitable for an embedded type of environment.
diff -ur xen-4.3.0.orig/tools/memshr/Makefile xen-4.3.0/tools/memshr/Makefile
--- tools/memshr/Makefile	2013-07-09 18:46:56.000000000 +0800
+++ tools/memshr/Makefile	2013-07-22 23:04:20.165369499 +0800
@@ -3,7 +3,6 @@
 
 LIBMEMSHR-BUILD := libmemshr.a
 
-CFLAGS          += -Werror
 CFLAGS          += -Wno-unused
 CFLAGS          += $(CFLAGS_xeninclude)
 CFLAGS          += $(CFLAGS_libxenctrl)
diff -ur xen-4.3.0.orig/tools/misc/Makefile xen-4.3.0/tools/misc/Makefile
--- tools/misc/Makefile	2013-07-22 23:01:09.019357104 +0800
+++ tools/misc/Makefile	2013-07-22 23:03:43.645367131 +0800
@@ -1,8 +1,6 @@
 XEN_ROOT=$(CURDIR)/../..
 include $(XEN_ROOT)/tools/Rules.mk
 
-CFLAGS += -Werror
-
 CFLAGS += $(CFLAGS_libxenctrl)
 CFLAGS += $(CFLAGS_xeninclude)
 CFLAGS += $(CFLAGS_libxenstore)
diff -ur xen-4.3.0.orig/tools/xenpaging/Makefile xen-4.3.0/tools/xenpaging/Makefile
--- tools/xenpaging/Makefile	2013-07-09 18:46:56.000000000 +0800
+++ tools/xenpaging/Makefile	2013-07-22 23:04:51.893371557 +0800
@@ -11,7 +11,6 @@
 SRCS     += file_ops.c xenpaging.c policy_$(POLICY).c
 SRCS     += pagein.c
 
-CFLAGS   += -Werror
 CFLAGS   += -Wno-unused
 CFLAGS   += -g
 
diff -ur xen-4.3.0.orig/tools/xenpmd/Makefile xen-4.3.0/tools/xenpmd/Makefile
--- tools/xenpmd/Makefile	2013-07-09 18:46:56.000000000 +0800
+++ tools/xenpmd/Makefile	2013-07-22 23:05:13.357372949 +0800
@@ -1,7 +1,6 @@
 XEN_ROOT=$(CURDIR)/../..
 include $(XEN_ROOT)/tools/Rules.mk
 
-CFLAGS += -Werror
 CFLAGS += $(CFLAGS_libxenstore)
 
 LDLIBS += $(LDLIBS_libxenstore)

diff -ur xen-4.3.0.orig/tools/blktap2/control/Makefile xen-4.3.0/tools/blktap2/control/Makefile
--- tools/blktap2/control/Makefile	2013-07-09 18:46:56.000000000 +0800
+++ tools/blktap2/control/Makefile	2013-07-22 23:31:55.225476829 +0800
@@ -8,7 +8,6 @@
 
 IBIN               = tap-ctl
 
-CFLAGS            += -Werror
 CFLAGS            += -Wno-unused
 CFLAGS            += -I../include -I../drivers
 CFLAGS            += $(CFLAGS_xeninclude)
diff -ur xen-4.3.0.orig/tools/blktap2/lvm/Makefile xen-4.3.0/tools/blktap2/lvm/Makefile
--- tools/blktap2/lvm/Makefile	2013-07-09 18:46:56.000000000 +0800
+++ tools/blktap2/lvm/Makefile	2013-07-22 23:31:02.329473398 +0800
@@ -6,7 +6,6 @@
 TEST              := lvm-util
 endif
 
-CFLAGS            += -Werror
 CFLAGS            += -Wno-unused
 CFLAGS            += -I../include
 CFLAGS            += -D_GNU_SOURCE
diff -ur xen-4.3.0.orig/tools/blktap2/vhd/lib/Makefile xen-4.3.0/tools/blktap2/vhd/lib/Makefile
--- tools/blktap2/vhd/lib/Makefile	2013-07-09 18:46:56.000000000 +0800
+++ tools/blktap2/vhd/lib/Makefile	2013-07-22 23:32:23.889478688 +0800
@@ -12,7 +12,6 @@
 
 INST-DIR         = $(LIBDIR)
 
-CFLAGS          += -Werror
 CFLAGS          += -Wno-unused
 CFLAGS          += -I../../include
 CFLAGS          += -D_GNU_SOURCE
diff -ur xen-4.3.0.orig/tools/blktap2/vhd/Makefile xen-4.3.0/tools/blktap2/vhd/Makefile
--- tools/blktap2/vhd/Makefile	2013-07-09 18:46:56.000000000 +0800
+++ tools/blktap2/vhd/Makefile	2013-07-22 23:31:21.641474651 +0800
@@ -8,7 +8,6 @@
 IBIN               = vhd-util vhd-update
 INST_DIR           = $(SBINDIR)
 
-CFLAGS            += -Werror
 CFLAGS            += -Wno-unused
 CFLAGS            += -I../include
 CFLAGS            += -D_GNU_SOURCE
diff -ur xen-4.3.0.orig/tools/console/Makefile xen-4.3.0/tools/console/Makefile
--- tools/console/Makefile	2013-07-09 18:46:56.000000000 +0800
+++ tools/console/Makefile	2013-07-22 23:30:14.392470290 +0800
@@ -1,8 +1,6 @@
 XEN_ROOT=$(CURDIR)/../..
 include $(XEN_ROOT)/tools/Rules.mk
 
-CFLAGS  += -Werror
-
 CFLAGS  += $(CFLAGS_libxenctrl)
 CFLAGS  += $(CFLAGS_libxenstore)
 LDLIBS += $(LDLIBS_libxenctrl)
diff -ur xen-4.3.0.orig/tools/libxen/Makefile.dist xen-4.3.0/tools/libxen/Makefile.dist
--- tools/libxen/Makefile.dist	2013-07-09 18:46:56.000000000 +0800
+++ tools/libxen/Makefile.dist	2013-07-22 23:29:08.184465996 +0800
@@ -22,7 +22,7 @@
 CFLAGS = -Iinclude                     \
          $(shell xml2-config --cflags) \
          $(shell curl-config --cflags) \
-         -W -Wall -Wmissing-prototypes -Werror -std=c99 -O2 -fPIC
+         -W -Wall -Wmissing-prototypes -std=c99 -O2 -fPIC
 
 LDFLAGS = $(shell xml2-config --libs) \
           $(shell curl-config --libs)
diff -ur xen-4.3.0.orig/tools/ocaml/common.make xen-4.3.0/tools/ocaml/common.make
--- tools/ocaml/common.make	2013-07-09 18:46:56.000000000 +0800
+++ tools/ocaml/common.make	2013-07-22 23:28:35.280463862 +0800
@@ -9,7 +9,7 @@
 OCAMLYACC ?= ocamlyacc
 OCAMLFIND ?= ocamlfind
 
-CFLAGS += -fPIC -Werror -I$(shell ocamlc -where)
+CFLAGS += -fPIC -I$(shell ocamlc -where)
 
 OCAMLOPTFLAG_G := $(shell $(OCAMLOPT) -h 2>&1 | sed -n 's/^  *\(-g\) .*/\1/p')
 OCAMLOPTFLAGS = $(OCAMLOPTFLAG_G) -ccopt "$(LDFLAGS)" -dtypes $(OCAMLINCLUDE) -cc $(CC) -w F -warn-error F
diff -ur xen-4.3.0.orig/tools/python/setup.py xen-4.3.0/tools/python/setup.py
--- tools/python/setup.py	2013-07-09 18:46:56.000000000 +0800
+++ tools/python/setup.py	2013-07-22 23:32:49.305480336 +0800
@@ -4,7 +4,7 @@
 
 XEN_ROOT = "../.."
 
-extra_compile_args  = [ "-fno-strict-aliasing", "-Werror" ]
+extra_compile_args  = [ "-fno-strict-aliasing" ]
 
 PATH_XEN      = XEN_ROOT + "/tools/include"
 PATH_LIBXC    = XEN_ROOT + "/tools/libxc"
diff -ur xen-4.3.0.orig/tools/xcutils/Makefile xen-4.3.0/tools/xcutils/Makefile
--- tools/xcutils/Makefile	2013-07-09 18:46:56.000000000 +0800
+++ tools/xcutils/Makefile	2013-07-22 23:29:45.024468385 +0800
@@ -13,8 +13,6 @@
 
 PROGRAMS = xc_restore xc_save readnotes lsevtchn
 
-CFLAGS += -Werror
-
 CFLAGS_xc_restore.o := $(CFLAGS_libxenctrl) $(CFLAGS_libxenguest)
 CFLAGS_xc_save.o    := $(CFLAGS_libxenctrl) $(CFLAGS_libxenguest) $(CFLAGS_libxenstore)
 CFLAGS_readnotes.o  := $(CFLAGS_libxenctrl) $(CFLAGS_libxenguest)
diff -ur xen-4.3.0.orig/tools/xentrace/Makefile xen-4.3.0/tools/xentrace/Makefile
--- tools/xentrace/Makefile	2013-07-09 18:46:56.000000000 +0800
+++ tools/xentrace/Makefile	2013-07-22 23:30:28.216471186 +0800
@@ -1,8 +1,6 @@
 XEN_ROOT=$(CURDIR)/../..
 include $(XEN_ROOT)/tools/Rules.mk
 
-CFLAGS += -Werror
-
 CFLAGS += $(CFLAGS_libxenctrl)
 LDLIBS += $(LDLIBS_libxenctrl)
 
diff -ur xen-4.3.0.orig/tools/pygrub/setup.py xen-4.3.0/tools/pygrub/setup.py
--- tools/pygrub/setup.py	2013-07-09 18:46:56.000000000 +0800
+++ tools/pygrub/setup.py	2013-07-22 23:47:45.664538464 +0800
@@ -3,7 +3,7 @@
 import os
 import sys
 
-extra_compile_args  = [ "-fno-strict-aliasing", "-Werror" ]
+extra_compile_args  = [ "-fno-strict-aliasing" ]
 
 XEN_ROOT = "../.."
 
diff -ur xen-4.3.0.orig/tools/blktap/drivers/Makefile xen-4.3.0/tools/blktap/drivers/Makefile
--- tools/blktap/drivers/Makefile	2013-07-09 18:46:56.000000000 +0800
+++ tools/blktap/drivers/Makefile	2013-07-23 00:40:47.810744824 +0800
@@ -4,7 +4,6 @@
 IBIN         = blktapctrl tapdisk
 QCOW_UTIL    = img2qcow qcow2raw qcow-create
 
-CFLAGS   += -Werror
 CFLAGS   += -Wno-unused
 CFLAGS   += -I../lib
 CFLAGS   += $(CFLAGS_libxenctrl)
diff -ur xen-4.3.0.orig/tools/blktap/lib/Makefile xen-4.3.0/tools/blktap/lib/Makefile
--- tools/blktap/lib/Makefile	2013-07-09 18:46:56.000000000 +0800
+++ tools/blktap/lib/Makefile	2013-07-23 00:41:56.513749279 +0800
@@ -13,7 +13,6 @@
 SRCS     :=
 SRCS     += xenbus.c blkif.c xs_api.c
 
-CFLAGS   += -Werror
 CFLAGS   += -Wno-unused
 CFLAGS   += -fPIC
 # get asprintf():
diff -ur xen-4.3.0.orig/tools/configure xen-4.3.0/tools/configure
--- tools/configure	2013-07-09 18:46:56.000000000 +0800
+++ tools/configure	2013-07-23 00:39:37.435740260 +0800
@@ -7621,7 +7621,6 @@
 
     saved_CPPFLAGS="$CPPFLAGS"
 
-    CPPFLAGS="$CPPFLAGS -Werror"
     ac_fn_c_check_header_mongrel "$LINENO" "libutil.h" "ac_cv_header_libutil_h" "$ac_includes_default"
 if test "x$ac_cv_header_libutil_h" = x""yes; then :
 
diff -ur xen-4.3.0.orig/tools/flask/utils/Makefile xen-4.3.0/tools/flask/utils/Makefile
--- ools/flask/utils/Makefile	2013-07-09 18:46:56.000000000 +0800
+++ tools/flask/utils/Makefile	2013-07-23 00:40:03.929741978 +0800
@@ -1,7 +1,7 @@
 XEN_ROOT=$(CURDIR)/../../..
 include $(XEN_ROOT)/tools/Rules.mk
 
-CFLAGS += -Wall -g -Werror
+CFLAGS += -Wall
 CFLAGS += $(CFLAGS_libxenctrl)
 
 TESTDIR  = testsuite/tmp
diff -ur xen-4.3.0.orig/tools/misc/lomount/Makefile xen-4.3.0/tools/misc/lomount/Makefile
--- tools/misc/lomount/Makefile	2013-07-09 18:46:56.000000000 +0800
+++ tools/misc/lomount/Makefile	2013-07-23 00:42:05.378749854 +0800
@@ -1,8 +1,6 @@
 XEN_ROOT=$(CURDIR)/../../..
 include $(XEN_ROOT)/tools/Rules.mk
 
-CFLAGS  += -Werror
-
 .PHONY: all
 all: build
 
@@ -20,4 +18,4 @@
 lomount: lomount.o
 	$(CC) $(CFLAGS) -o $@ $< 
 
--include $(DEPS)
\ No newline at end of file
+-include $(DEPS)
diff -ur xen-4.3.0.orig/tools/qemu-xen-traditional/configure xen-4.3.0/tools/qemu-xen-traditional/configure
--- tools/qemu-xen-traditional/configure	2013-06-18 00:39:51.000000000 +0800
+++ tools/qemu-xen-traditional/configure	2013-07-23 00:43:39.570755962 +0800
@@ -1150,7 +1150,6 @@
 echo "sparse enabled    $sparse"
 echo "profiler          $profiler"
 echo "static build      $static"
-echo "-Werror enabled   $werror"
 if test "$darwin" = "yes" ; then
     echo "Cocoa support     $cocoa"
 fi
diff -ur xen-4.3.0.orig/tools/xenbackendd/Makefile xen-4.3.0/tools/xenbackendd/Makefile
--- tools/xenbackendd/Makefile	2013-07-09 18:46:56.000000000 +0800
+++ tools/xenbackendd/Makefile	2013-07-23 00:44:16.410758351 +0800
@@ -12,7 +12,6 @@
 XEN_ROOT=$(CURDIR)/../..
 include $(XEN_ROOT)/tools/Rules.mk
 
-CFLAGS  += -Werror
 CFLAGS  += $(CFLAGS_libxenstore)
 CPPFLAGS += -DXEN_SCRIPT_DIR="\"$(XEN_SCRIPT_DIR)\""
 LDLIBS  += $(LDLIBS_libxenstore)
diff -ur xen-4.3.0.orig/tools/xenmon/Makefile xen-4.3.0/tools/xenmon/Makefile
--- tools/xenmon/Makefile	2013-07-09 18:46:56.000000000 +0800
+++ tools/xenmon/Makefile	2013-07-23 00:44:51.114760602 +0800
@@ -13,7 +13,6 @@
 XEN_ROOT=$(CURDIR)/../..
 include $(XEN_ROOT)/tools/Rules.mk
 
-CFLAGS  += -Werror
 CFLAGS  += $(CFLAGS_libxenctrl)
 LDLIBS  += $(LDLIBS_libxenctrl)
 
diff -ur xen-4.3.0.orig/tools/xenstore/Makefile xen-4.3.0/tools/xenstore/Makefile
--- ools/xenstore/Makefile	2013-07-09 18:46:56.000000000 +0800
+++ tools/xenstore/Makefile	2013-07-23 00:45:25.938762860 +0800
@@ -4,7 +4,6 @@
 MAJOR = 3.0
 MINOR = 3
 
-CFLAGS += -Werror
 CFLAGS += -I.
 CFLAGS += $(CFLAGS_libxenctrl)
 
