--- Makefile.old	2011-05-19 09:48:26.698819456 +0200
+++ Makefile	2011-05-19 16:40:54.526390959 +0200
@@ -29,7 +29,7 @@
 sdl-cflags := $(shell sdl-config --cflags)
 sdl-ldflags := $(shell sdl-config --libs)
 
-ldflags := $(sdl-ldflags) -lSDL_image -lSDL_mixer $(LDFLAGS)
+LDLIBS := $(sdl-ldflags) -lSDL_image -lSDL_mixer -lm
 cflags := $(sdl-cflags) $(paths) $(CFLAGS)
 
 my_objects := args.o dust.o file.o mt.o rocks.o score.o sprite.o sound.o autopilot.o
@@ -89,7 +89,7 @@
 sprite.o: sprite.h common.h
 
 vor: $(objects)
-	$(CC) $^ $(ldflags) -o $@
+	$(LINK.c) $^ $(LOADLIBES) $(LDLIBS) -o $@
 
 include gfx.mk
 
