--- orig/fish_indent.c	2010-11-23 02:34:55.000000000 -0700
+++ fish-1.23.1/fish_indent.c	2010-11-23 02:35:29.000000000 -0700


@@ -171,23 +171,23 @@
 			case TOK_REDIRECT_IN:
 			case TOK_REDIRECT_FD:
 			{
-				sb_append( out, last );
+				/* sb_append( out, last ); */
 				switch( type )
 				{
 					case TOK_REDIRECT_OUT:
-						sb_append( out, L"> " );
+						sb_append( out, L" > " );
 						break;
 						
 					case TOK_REDIRECT_APPEND:
-						sb_append( out, L">> " );
+						sb_append( out, L" >> " );
 						break;
 						
 					case TOK_REDIRECT_IN:
-						sb_append( out, L"< " );
+						sb_append( out, L" < " );
 						break;
 						
 					case TOK_REDIRECT_FD:
-						sb_append( out, L">& " );
+						sb_append( out, L" >& " );
 						break;
 															
 				}
