--- braa-0.8/Makefile.orig	2004-01-12 09:15:51.000000000 +0100
+++ braa-0.8/Makefile	2009-01-15 03:58:52.000000000 +0100
@@ -4,8 +4,6 @@
 ###################################################################
 
 # Linux, FreeBSD, OpenBSD
-CFLAGS = 
-LDFLAGS = 
 
 # Solaris / SPARC
 # CFLAGS = -DSOLARIS_SPARC
@@ -24,9 +22,8 @@
 LIBS = 
 
 all: ${OBJECTS}
-	cc $(OBJECTS) -o $(OUT) $(LDFLAGS) $(LIBS)
-	strip $(OUT)
+	$(CC) $(CFLAGS) $(LDFLAGS) $(OBJECTS) -o $(OUT) $(LIBS)
 
 clean:
 	rm -rf $(OBJECTS) $(OUT)
-	
\ No newline at end of file
+	
