--- nitrogen-1.4.old/src/Makefile.am	2008-05-05 21:34:15.000000000 +0200
+++ nitrogen-1.4/src/Makefile.am	2008-12-03 20:14:34.000000000 +0100
@@ -1,6 +1,6 @@
 bin_PROGRAMS = nitrogen
 nitrogen_SOURCES = Config.h NWindow.h main.cc md5.h SetBG.cc main.h Thumbview.cc Config.cc NWindow.cc SetBG.h md5.c ArgParser.h ArgParser.cc Thumbview.h ImageCombo.cc ImageCombo.h Inotify.cc Inotify.h Util.h Util.cc
-nitrogen_LDFLAGS = @NITROGEN_LIBS@
+LIBS=@NITROGEN_LIBS@
 INCLUDES = @NITROGEN_CFLAGS@ \
 	-DNITROGEN_DATA_DIR=\"$(pkgdatadir)\"
 man_MANS = nitrogen.1
