 CMakeLists.txt |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a68497f..264f321 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -426,6 +426,11 @@ LIST(APPEND BALL_DEP_LIBRARIES
 	${PYTHON_LIBRARIES}
 )
 
+
+IF(UNIX)
+	LIST(APPEND BALL_DEP_LIBRARIES "pthread")
+ENDIF()
+
 IF(TBB_FOUND)
 	LIST(APPEND BALL_DEP_LIBRARIES ${TBB_LIBRARIES})
 ENDIF()
