From 67e11c7ad916c2e0b9af99153a01be378ea174c0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian.droege@collabora.co.uk>
Date: Sat, 10 Nov 2012 16:57:06 +0100
Subject: [PATCH] avcodecmap: Add mappings for the GBR color formats

---
 ext/libav/gstavcodecmap.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/ext/libav/gstavcodecmap.c b/ext/libav/gstavcodecmap.c
index abf8f92..5d7dd23 100644
--- a/ext/libav/gstavcodecmap.c
+++ b/ext/libav/gstavcodecmap.c
@@ -2182,6 +2182,11 @@ static const PixToFmt pixtofmttable[] = {
   {GST_VIDEO_FORMAT_I420_10BE, PIX_FMT_YUV420P10BE},
   {GST_VIDEO_FORMAT_I422_10LE, PIX_FMT_YUV422P10LE},
   {GST_VIDEO_FORMAT_I422_10BE, PIX_FMT_YUV422P10BE},
+  {GST_VIDEO_FORMAT_Y444_10LE, PIX_FMT_YUV444P10LE},
+  {GST_VIDEO_FORMAT_Y444_10BE, PIX_FMT_YUV444P10BE},
+  {GST_VIDEO_FORMAT_GBR, PIX_FMT_GBRP},
+  {GST_VIDEO_FORMAT_GBR_10LE, PIX_FMT_GBRP10LE},
+  {GST_VIDEO_FORMAT_GBR_10BE, PIX_FMT_GBRP10BE},
 };
 
 GstVideoFormat
-- 
1.8.1.2

