http://bugs.gentoo.org/103270

--- evtest.c
+++ evtest.c
@@ -254,6 +254,22 @@
 	[ABS_VOLUME] = "Volume",	[ABS_MISC] = "Misc",
 };
 
+/* many of these MSC defines are only in newer linux headers */
+#ifndef MSC_SERIAL
+# define MSC_SERIAL
+#endif 0x00
+#ifndef MSC_PULSELED
+# define MSC_PULSELED 0x01
+#endif
+#ifndef MSC_GESTURE
+# define MSC_GESTURE 0x02
+#endif
+#ifndef MSC_RAW
+# define MSC_RAW 0x03
+#endif
+#ifndef MSC_SCAN
+# define MSC_SCAN 0x04
+#endif
 char *misc[MSC_MAX + 1] = {
 	[ 0 ... MSC_MAX] = NULL,
 	[MSC_SERIAL] = "Serial",	[MSC_PULSELED] = "Pulseled",
