--- utils.orig/Makefile	2006-07-31 06:50:01.000000000 +0200
+++ utils/Makefile	2009-12-09 20:11:17.962976029 +0100
@@ -46,7 +45,7 @@
 
 ffcfstress: ffcfstress.c
 	$(CC) $(CFLAGS) -funsigned-char \
-	 -lm ffcfstress.c -o ffcfstress
+	 ffcfstress.c -o ffcfstress $(LDFLAGS) -lm
 
 ffmvforce.o: ffmvforce.c
 	$(CC) -c $(CFLAGS) $(CPPFLAGS) $^ -o $@ `sdl-config --cflags`
@@ -55,13 +54,13 @@
 	$(CC) $^ -o $@ $(LDFLAGS) $(CFLAGS) -lm `sdl-config --libs`
 
 jscal: jscal.o
-	$(CC) $(CFLAGS) $(CPPFLAGS) -lm $^ -o $@
+	$(CC) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) $^ -o $@ -lm
 
 gencodes: gencodes.c scancodes.h
 	$(CC) $(CFLAGS) $(CPPFLAGS) gencodes.c -o gencodes
 
 acceltest: acceltest.c
-	$(CC) $(CFLAGS) $(CPPFLAGS) -lm $^ -o $@
+	$(CC) $(CFLAGS) $(CPPFLAGS) $^ -o $@ -lm
 
 dist:
 	tar jcf ../joystick-`date +%Y%m%d`.tar.bz2 -C .. utils
