--- d2x-gl.sh.orig	2003-07-19 22:04:48.000000000 -0400
+++ d2x-gl.sh	2003-07-19 22:05:25.000000000 -0400
@@ -35,8 +35,7 @@
 # Finally, run the real game.
 # All switches can be overwritten by command-line arguments.
 #
-nice /usr/games/d2x-gl.real -tmap fp -nocdrom -nomovies \
+exec nice d2x-gl -tmap fp -nocdrom -nomovies \
 	-gl_mipmap -gl_alttexmerge -gl_reticle 2 \
 	-nofades \
-	$1 $2 $3 $4 $5
-
+	${@}
--- d2x-sdl.sh.orig	2003-07-19 22:05:04.000000000 -0400
+++ d2x-sdl.sh	2003-07-19 22:05:39.000000000 -0400
@@ -3,5 +3,5 @@
 # Run the real game.
 # All switches can be overwritten by command-line arguments.
 #
-nice /usr/games/d2x-sdl.real -nocdrom -nomovies -nofades $1 $2 $3 $4 $5
+exec nice d2x-sdl -nocdrom -nomovies -nofades ${@}
 
