--- main/multi.h.old	2012-09-01 14:33:13.668620792 +0200
+++ main/multi.h	2012-09-01 14:33:25.043385055 +0200
@@ -120,7 +120,7 @@
 
 #define MAX_NET_CREATE_OBJECTS  40
 
-#define MAX_MULTI_MESSAGE_LEN   120
+#define MAX_MULTI_MESSAGE_LEN   138
 
 // Exported functions
 
--- main/ai.c.old	2012-09-01 15:02:36.754081633 +0200
+++ main/ai.c	2012-09-01 20:16:07.846989673 +0200
@@ -1885,8 +1885,11 @@
 	fread( &Last_gate_time, sizeof(fix), 1, fp );
 	fread( &Gate_interval, sizeof(fix), 1, fp );
 	fread( &Boss_dying_start_time, sizeof(fix), 1, fp );
-	fread( &Boss_dying, sizeof(int), 1, fp );
-	fread( &Boss_dying_sound_playing, sizeof(int), 1, fp );
+	int temp;
+	fread( &temp, sizeof(int), 1, fp );
+	Boss_dying = temp;
+	fread( &temp, sizeof(int), 1, fp );
+	Boss_dying_sound_playing = temp;
 	fread( &Boss_hit_time, sizeof(fix), 1, fp );
 	// -- MK, 10/21/95, unused! -- fread( &Boss_been_hit, sizeof(int), 1, fp );
 
