diff --git a/Rakefile b/Rakefile
index fb71b42..f643cb1 100644
--- a/Rakefile
+++ b/Rakefile
@@ -217,6 +217,7 @@ end
 
 #### Custom testing tasks
 
+begin
 require 'rubygems' 
 require 'spec/rake/spectask'
 Rake::Task[:default].prerequisites.clear
@@ -241,6 +242,9 @@ task :ensure_diff_lcs do
     end
   end
 end
+rescue LoadError
+  $stderr.puts "RSpec not found, tests won't be available. Please install rspec with gem install rspec"
+end
 
 task :remove_other_platforms do
   Dir["lib/redcloth_scan.{bundle,so,jar,rb}"].each { |file| rm file }
