From 9b6b5c771c14f6c7fc4800273fab88471e3b9cb0 Mon Sep 17 00:00:00 2001
From: David Zeuthen <davidz@redhat.com>
Date: Tue, 29 Mar 2011 14:34:14 +0000
Subject: Use the right directory for the nautilus extension

Signed-off-by: David Zeuthen <davidz@redhat.com>
---
diff --git a/configure.ac b/configure.ac
index 66fa8e8..ad7b051 100644
--- a/configure.ac
+++ b/configure.ac
@@ -129,7 +129,7 @@ GNOME_KEYRING_REQUIRED=2.22
 GTK3_REQUIRED=2.90.7
 UNIQUE_REQUIRED=2.90.1
 LIBNOTIFY_REQUIRED=0.6.1
-NAUTILUS_REQUIRED=2.24.0
+NAUTILUS_REQUIRED=2.91.0
 AVAHI_UI_REQUIRED=0.6.25
 
 UDISKS_REQUIRED=1.0.0
diff --git a/src/nautilus-extension/Makefile.am b/src/nautilus-extension/Makefile.am
index 6c2c82c..c5a9875 100644
--- a/src/nautilus-extension/Makefile.am
+++ b/src/nautilus-extension/Makefile.am
@@ -34,7 +34,7 @@ CORE_LIBADD = 						\
 	$(NULL)
 
 
-nautilus_extensiondir = $(libdir)/nautilus/extensions-2.0
+nautilus_extensiondir = $(libdir)/nautilus/extensions-3.0
 nautilus_extension_LTLIBRARIES = libnautilus-gdu.la
 
 libnautilus_gdu_la_SOURCES =			\
--
cgit v0.9.0.2
