--- GNUmakefile.preamble.orig	2009-07-21 13:59:31.000000000 +0200
+++ GNUmakefile.preamble	2009-07-21 13:59:33.000000000 +0200
@@ -12,7 +12,7 @@
 ADDITIONAL_CFLAGS += 
 
 # Additional flags to pass to the linker
-ADDITIONAL_LDFLAGS += -lgsl -lgslcblas 
+ADDITIONAL_LDFLAGS +=
 
 # Additional include directories the compiler should search
 ADDITIONAL_INCLUDE_DIRS += -I/usr/include 
@@ -21,5 +21,5 @@
 ADDITIONAL_LIB_DIRS += -L/usr/lib 
 
 # Additional GUI libraries to link
-ADDITIONAL_GUI_LIBS += 
+ADDITIONAL_GUI_LIBS +=  -lgsl -lgslcblas 
 
