--- src/ccze-compat.h.old	2006-10-13 12:50:16 +0200
+++ src/ccze-compat.h	2006-10-13 12:49:21 +0200
@@ -29,6 +29,11 @@
 #include <stddef.h>
 #include <stdio.h>
 
+# ifndef __error_t_defined
+typedef int error_t;
+#  define __error_t_defined     1
+# endif
+
 #ifndef HAVE_STRNDUP
 char *strndup (const char *s, size_t size);
 #endif
--- src/ccze-compat.c.old	2006-10-13 12:51:35 +0200
+++ src/ccze-compat.c	2006-10-13 12:51:44 +0200
@@ -275,7 +275,7 @@
 		char **valuep)
 {
   int i = getsubopt (optionp, tokens, valuep);
-#if HAVE_SUBOPTARg
+#if HAVE_SUBOPTARG
   if (!*valuep && suboptarg)
     *valuep = strdup (suboptarg);
 #endif
