diff -Naur e-uae-0.8.28.orig/src/gfx-x11/xwin.c e-uae-0.8.28/src/gfx-x11/xwin.c
--- e-uae-0.8.28.orig/src/gfx-x11/xwin.c	2005-08-21 19:10:40.000000000 +0400
+++ e-uae-0.8.28/src/gfx-x11/xwin.c	2006-06-14 17:50:29.000000000 +0400
@@ -1215,7 +1215,7 @@
 
     if (! dgamode) {
 	if (! screen_is_picasso && refresh_necessary) {
-	    DO_PUTIMAGE (ami_dinfo.ximg, 0, 0, 0, 0, currprefs.gfx_width_fs, currprefs.gfx_height_fs);
+	    DO_PUTIMAGE (ami_dinfo.ximg, 0, 0, 0, 0, current_width, current_height);
 	    refresh_necessary = 0;
 	}
 	if (cursorOn && !currprefs.x11_hide_cursor) {
