diff -Naru e-uae-0.8.28.orig/src/gui-gtk/util.h e-uae-0.8.28/src/gui-gtk/util.h
--- e-uae-0.8.28.orig/src/gui-gtk/util.h	2005-08-30 18:00:57.000000000 +0400
+++ e-uae-0.8.28/src/gui-gtk/util.h	2006-07-17 14:31:59.000000000 +0400
@@ -34,7 +34,7 @@
 extern void add_table_padding (GtkWidget *table, int x, int y);
 extern void add_to_table(GtkWidget *table, GtkWidget *widget, int x, int y, int width, int xflags);
 
-#define GTKUTIL_TABLE_END	0
+#define GTKUTIL_TABLE_END	((GtkWidget *)0)
 #define GTKUTIL_ROW_END		((GtkWidget *)1)
 
 extern GtkWidget *gtkutil_add_table (GtkWidget *container, ...);
