commit 596f79fa65af66afdff4cf24b9135935bf511108
Author: Dmitry Yu Okunev <dyokunev@ut.mephi.ru>
Date:   Wed Jan 28 14:27:51 2015 +0300

    Fixed a segfault on non-set "HOME" environment variable

diff --git a/main.c b/main.c
index 792f1e4..176e1f4 100644
--- a/main.c
+++ b/main.c
@@ -1731,18 +1731,18 @@ int configs_parse(ctx_t *ctx_p, paramsource_t paramsource) {
 		char **config_path_p = config_paths, *config_path_real = xmalloc(PATH_MAX);
 		size_t config_path_real_size=PATH_MAX;
 
-		char *homedir = getenv("HOME");
-		size_t homedir_len = strlen(homedir);
+		char  *homedir     = getenv("HOME");
+		size_t homedir_len = (homedir == NULL ? 0 :strlen(homedir));
 
-		while(*config_path_p != NULL) {
+		while (*config_path_p != NULL) {
 			size_t config_path_len = strlen(*config_path_p);
 
-			if(config_path_len+homedir_len+3 > config_path_real_size) {
+			if (config_path_len+homedir_len+3 > config_path_real_size) {
 				config_path_real_size = config_path_len+homedir_len+3;
 				config_path_real      = xmalloc(config_path_real_size);
 			}
 
-			if(*config_path_p[0] != '/') {
+			if ((*config_path_p[0] != '/') && (homedir_len >= 0)) {
 				memcpy(config_path_real, homedir, homedir_len);
 				config_path_real[homedir_len] = '/';
 				memcpy(&config_path_real[homedir_len+1], *config_path_p, config_path_len+1);
@@ -1750,7 +1750,7 @@ int configs_parse(ctx_t *ctx_p, paramsource_t paramsource) {
 				memcpy(config_path_real, *config_path_p, config_path_len+1);
 
 			debug(1, "Trying config-file \"%s\" (case #1)", config_path_real);
-			if(!g_key_file_load_from_file(gkf, config_path_real, G_KEY_FILE_NONE, NULL)) {
+			if (!g_key_file_load_from_file(gkf, config_path_real, G_KEY_FILE_NONE, NULL)) {
 				debug(1, "Cannot open/parse file \"%s\"", config_path_real);
 				config_path_p++;
 				continue;
