diff -Naur gcl-2.6.7.orig/configure.in gcl-2.6.7/configure.in
--- gcl-2.6.7.orig/configure.in	2005-01-15 20:17:17.000000000 +0100
+++ gcl-2.6.7/configure.in	2007-02-13 20:40:40.000000000 +0100
@@ -788,9 +788,9 @@
 					    int main() {symbol_info t; return 0;}],
 					AC_MSG_RESULT(yes) 
 					AC_DEFINE(NEED_CONST),
-					AC_MSG_RESULT(cannot use bfd) exit 1;,
-					AC_MSG_RESULT(cannot use bfd) exit 1;),
-				AC_MSG_RESULT(cannot use bfd) exit 1;)
+					AC_MSG_RESULT(cannot use bfd); exit 1;,
+					AC_MSG_RESULT(cannot use bfd); exit 1;),
+				AC_MSG_RESULT(cannot use bfd); exit 1;)
 			,,-liberty))
 
 	AC_DEFINE(HAVE_LIBBFD)
@@ -1252,8 +1252,14 @@
 				return fpclass(f)>=FP_NZERO || fpclass(f)<FP_NZERO ? 0 : 1;
 				}],
 		AC_DEFINE(HAVE_IEEEFP) AC_MSG_RESULT(yes),
-		HAVE_IEEEFP=0 AC_MSG_RESULT(no),HAVE_IEEEFP=0 AC_MSG_RESULT(no))
-	,HAVE_ISNORMAL=0 AC_MSG_RESULT(no),HAVE_ISNORMAL=0 AC_MSG_RESULT(no))
+		HAVE_IEEEFP=0
+		AC_MSG_RESULT(no),
+		HAVE_IEEEFP=0
+		AC_MSG_RESULT(no)),
+		HAVE_ISNORMAL=0
+		AC_MSG_RESULT(no),
+		HAVE_ISNORMAL=0
+		AC_MSG_RESULT(no))
 
 AC_MSG_CHECKING([for isfinite])
 AC_TRY_RUN([#define _GNU_SOURCE
@@ -1271,8 +1277,14 @@
 				return finite(f) || !finite(f) ? 0 : 1;
 				}],
 		AC_DEFINE(HAVE_FINITE) AC_MSG_RESULT(yes),
-		HAVE_FINITE=0 AC_MSG_RESULT(no),HAVE_FINITE=0 AC_MSG_RESULT(no))
-	,HAVE_ISFINITE=0 AC_MSG_RESULT(no),HAVE_ISFINITE=0 AC_MSG_RESULT(no))
+		HAVE_FINITE=0
+		AC_MSG_RESULT(no),
+		HAVE_FINITE=0
+		AC_MSG_RESULT(no))
+	,HAVE_ISFINITE=0
+	AC_MSG_RESULT(no),
+	HAVE_ISFINITE=0
+	AC_MSG_RESULT(no))
 
 
 
