--- xpra-0.8.0/setup.py
+++ xpra-0.8.0/setup.py
@@ -582,8 +582,8 @@
             return {}
     if "install" in sys.argv:
         #prepare default [/usr/local]/etc configuration files:
-        if sys.prefix == '/usr':
-            etc_prefix = '/etc/xpra'
+        if sys.prefix[-4:] == '/usr':
+            etc_prefix = sys.prefix[:-4] + '/etc/xpra'
         else:
             etc_prefix = sys.prefix + '/etc/xpra'
         etc_files = ["etc/xpra/xorg.conf"]
