--- xpra-0.7.4/setup.py
+++ xpra-0.7.4/setup.py
@@ -459,8 +459,8 @@
 
         if 'install' in sys.argv:
             #prepare default [/usr/local]/etc configuration files:
-            if sys.prefix == '/usr':
-                etc_prefix = '/etc/xpra'
+            if sys.prefix[-4:] == '/usr':
+                etc_prefix = sys.prefix[:-4] + '/etc/xpra'
             else:
                 etc_prefix = sys.prefix + '/etc/xpra'
             etc_files = ["etc/xpra/xorg.conf"]
