diff --git a/lib/hdb/db3.c b/lib/hdb/db3.c
index 3b22c2e..a7bae76 100644
--- a/lib/hdb/db3.c
+++ b/lib/hdb/db3.c
@@ -268,7 +268,7 @@ DB_open(krb5_context context, HDB *db, int flags, mode_t mode)
     db_create(&d, NULL, 0);
     db->hdb_db = d;
 
-#if (DB_VERSION_MAJOR >= 4) && (DB_VERSION_MINOR >= 1)
+#if (DB_VERSION_MAJOR == 5) || (DB_VERSION_MAJOR == 4 && DB_VERSION_MINOR >= 1)
     ret = (*d->open)(db->hdb_db, NULL, fn, NULL, DB_BTREE, myflags, mode);
 #else
     ret = (*d->open)(db->hdb_db, fn, NULL, DB_BTREE, myflags, mode);
@@ -276,7 +276,7 @@ DB_open(krb5_context context, HDB *db, int flags, mode_t mode)
 
     if (ret == ENOENT) {
 	/* try to open without .db extension */
-#if (DB_VERSION_MAJOR >= 4) && (DB_VERSION_MINOR >= 1)
+#if (DB_VERSION_MAJOR == 5) || (DB_VERSION_MAJOR == 4 && DB_VERSION_MINOR >= 1)
 	ret = (*d->open)(db->hdb_db, NULL, db->hdb_name, NULL, DB_BTREE,
 			 myflags, mode);
 #else
