--- Makefile.orig	2004-06-23 14:49:14.966324312 +0000
+++ Makefile	2004-06-23 14:49:52.487620208 +0000
@@ -6,10 +6,10 @@
 OBJECTS = wmbutton.o wmb_libs.o
 
 .c.o:
-	gcc -O2 -c -Wall ${INCLUDES} ${FLAGS} $< -o $*.o
+	gcc -c -Wall ${INCLUDES} ${FLAGS} ${CFLAGS} $< -o $*.o
 
 ${TARGET}: ${OBJECTS}
-	gcc -O2 -o ${TARGET} ${OBJECTS} ${LIBINC} ${LIBS}
+	gcc -o ${TARGET} ${OBJECTS} ${LIBINC} ${LIBS} ${CFLAGS} 
 
 clean::
 	for i in ${OBJECTS}; do if [ -e $$i ] ; then rm $$i; fi; done 
