--- speech_tools/speech_class/EST_wave_io.cc.orig	2009-05-26 03:49:24.000000000 -0700
+++ speech_tools/speech_class/EST_wave_io.cc	2009-05-26 03:54:01.000000000 -0700
@@ -68,7 +68,7 @@
 
 int nist_get_param_int(const char *hdr, const char *field, int def_val)
 {
-    char *p;
+    const char *p;
     int val;
 
     if (((p=strstr(hdr,field)) != NULL) &&
@@ -84,7 +84,8 @@
 
 char *nist_get_param_str(const char *hdr, const char *field, const char *def_val)
 {
-    char *p,*val;
+    const char *p;
+    char *val;
     int size;
 
     if (((p=strstr(hdr,field)) != NULL) &&
