http://bugs.gentoo.org/326797

--- src/ptclib/pssl.cxx
+++ src/ptclib/pssl.cxx
@@ -888,7 +888,11 @@
   InitialisationMutex.Signal();
 
   // create the new SSL context
+  #if OPENSSL_VERSION_NUMBER >= 0x10000000L
+  const SSL_METHOD * meth = SSLv23_method();
+  #else
   SSL_METHOD * meth = SSLv23_method();
+  #endif
   context  = SSL_CTX_new(meth);
   if (context == NULL)
     PSSLAssert("Error creating context: ");
