--- src/ptlib/common/contain.cxx.orig	2005-11-30 13:47:41.000000000 +0100
+++ src/ptlib/common/contain.cxx	2007-09-27 10:01:49.000000000 +0200
@@ -2465,7 +2465,7 @@ PString & PString::vsprintf(const char *
   PAssert(SetSize(2000), POutOfMemory);
   ::vsprintf(theArray+len, fmt, arg);
 #else
-  PINDEX size = 0;
+  PINDEX size = len;
   do {
     size += 1000;
     PAssert(SetSize(size), POutOfMemory);
