--- pwlib-1.8.7/Makefile.in.orig	2004-06-24 12:46:48.000000000 +0000
+++ pwlib-1.8.7/Makefile.in	2005-09-09 15:48:10.000000000 +0000
@@ -268,14 +268,16 @@
 		ln -sf $(PTLIB_FILE) libpt.@SHAREDLIBEXT@ \
 	)
 ifeq (1, $(HAS_PLUGINS))
-	cd plugins/pwlib/device/ ; \
-	( for dir in ./* ;\
-		do mkdir -p $(DESTDIR)$(LIBDIR)/pwlib/devices/$$dir ; \
-		chmod 755 $(DESTDIR)$(LIBDIR)/pwlib/devices/$$dir ; \
-		(for fn in ./$$dir/*.so ; \
-			do $(INSTALL) -m 444 $$fn $(DESTDIR)$(LIBDIR)/pwlib/devices/$$dir; \
-		done ); \
-	done )
+	if [ -d plugins/pwlib/device ] ; then \
+		cd plugins/pwlib/device/ ; \
+		( for dir in ./* ;\
+			do mkdir -p $(DESTDIR)$(LIBDIR)/pwlib/devices/$$dir ; \
+			chmod 755 $(DESTDIR)$(LIBDIR)/pwlib/devices/$$dir ; \
+			(for fn in ./$$dir/*.so ; \
+				do $(INSTALL) -m 444 $$fn $(DESTDIR)$(LIBDIR)/pwlib/devices/$$dir; \
+			done ); \
+		done ) \
+	fi
 endif
 	$(INSTALL) -m 444 include/ptlib.h                $(DESTDIR)$(PREFIX)/include
 	$(INSTALL) -m 444 include/ptbuildopts.h          $(DESTDIR)$(PREFIX)/include
