Fix depends for a test file in the Makefile

--- a/Makefile	2012-07-04 19:53:31.000000000 +0200
+++ b/Makefile	2012-09-10 14:24:48.502757692 +0200
@@ -19,7 +19,7 @@
 include Makefile.config
 
 ifeq ($(USE_BATTERIES),yes)
-EXTLIB := batteries
+EXTLIB := camomile,batteries
 else
 EXTLIB := extlib
 endif
@@ -44,11 +44,15 @@
 
 GETLIB=-I +$(1) $(shell ocamlfind query $(1) -predicates byte -format "%d/%a")
 
+COMMA = ,
+
 FOR_P4  := \
 	$(call GETLIB,unix) \
 	$(call GETLIB,str) \
+	$(call GETLIB,num.core) \
+	$(call GETLIB,bigarray) \
 	$(call GETLIB,pcre) \
-	$(call GETLIB,$(EXTLIB)) \
+	$(shell ocamlfind query $(subst $(COMMA), ,$(EXTLIB)) -predicates byte -format "%d/%a") \
 	$(call GETLIB,calendar) \
 	$(call GETLIB,csv) \
 	./pgocaml.cma
