--- dwdiff-2.0.7/config.pkg
+++ dwdiff-2.0.7/config.pkg
@@ -104,6 +104,7 @@
 
 	if [ "yes" = "${with_dwfilter}" ] ; then
 		cat > .config.c <<EOF
+#include <sys/types.h>
 #include <unistd.h>
 
 int main(int argc, char *argv[]) {
@@ -114,8 +115,8 @@
 		clean .config.o
 
 		test_link "required functions for dwfilter" || {
-			checkfunction "fork" 'pid_t pid = fork();' "<unistd.h>"
-			checkfunction "open" 'execvp("test", NULL);' "<unistd.h>"
+			checkfunction "fork" 'pid_t pid = fork();' "<unistd.h>" "<sys/types.h>"
+			checkfunction "open" 'execvp("test", NULL);' "<unistd.h>" "<sys/types.h>"
 			check_message_result "!! A required function for dwfilter is not availabe. Try configuring with --without-dwfilter"
 			exit 1
 		}
