http://bugzilla.xfce.org/show_bug.cgi?id=4608

=== modified file 'libmailwatch-core/mailwatch-net-conn.c'
--- libmailwatch-core/mailwatch-net-conn.c	2008-11-13 11:42:40 +0000
+++ libmailwatch-core/mailwatch-net-conn.c	2008-11-13 11:42:48 +0000
@@ -195,7 +195,6 @@
     return 0;
 }
 
-#endif  /* defined(HAVE_SSL_SUPPORT) */
 
 
 
@@ -234,6 +233,7 @@
 
     return TRUE;
 }
+#endif  /* defined(HAVE_SSL_SUPPORT) */
 
 static XfceMailwatchNetConnStatus
 xfce_mailwatch_net_conn_do_connect(XfceMailwatchNetConn *net_conn,

=== modified file 'libmailwatch-core/mailwatch-utils.c'
--- libmailwatch-core/mailwatch-utils.c	2008-11-13 11:41:27 +0000
+++ libmailwatch-core/mailwatch-utils.c	2008-11-13 11:43:36 +0000
@@ -186,7 +186,7 @@
     return response_base64;
 #else
     g_warning("CRAM-MD5 computation unavailable: libmailwatch was not compiled with gnutls support.");
-    return NULL
+    return NULL;
 #endif
 }
 

