Index: burn-0.1.0-pre21/burn.c
===================================================================
--- burn-0.1.0-pre21.orig/burn.c
+++ burn-0.1.0-pre21/burn.c
@@ -45,8 +45,13 @@ namespace vdr_burn
 
 	string plugin::get_character_encoding()
 	{
-#ifdef UTF8PATCH
-		return "utf8";
+#if VDRVERSNUM >= 10503
+		if (cCharSetConv::SystemCharacterTable())
+			return cCharSetConv::SystemCharacterTable();
+		else
+			return "utf-8";
+#elif defined(UTF8PATCH)
+		return "utf-8";
 #else
 		return tr("iso8859-15");
 #endif
