--- i18n.c	2006-09-16 20:33:36.000000000 +0200
+++ i18n.c.patched	2007-01-03 13:37:08.000000000 +0100
@@ -210,7 +210,7 @@
 #endif
 #endif
 		},
-		{ "Total size: %1$.1f MB %2$s",
+		{ "$Total size: %1$.1f MB %2$s",
 		  "Gesamtgröße: %1$.1f MB %2$s",
 		  "",
 		  "",
@@ -782,7 +782,7 @@
 #endif
 #endif
 		},
-		{ "Job active (Writing: %1$d%%)",
+		{ "$Job active (Writing: %1$d%%)",
 		  "Auftrag aktiv (Schreibe: %1$d%%)",
 		  "",
 		  "",
@@ -1874,7 +1874,7 @@
 #endif
 #endif
 		},
-		{ "Recordings: %1$d, total size: %2$.1f MB",
+		{ "$Recordings: %1$d, total size: %2$.1f MB",
 		  "Aufzeichnungen: %1$d, Gesamtgröße: %2$.1f MB",
 		  "",
 		  "",
