--- menuburn.c	2006-09-16 20:33:36.000000000 +0200
+++ menuburn.c.patched	2007-01-03 13:37:41.000000000 +0100
@@ -226,7 +226,7 @@
 		{
 			const recording_list& recordings( job_.get_recordings() );
 			if ( recordings.size() > 0 )
-				return str( boost::format( tr("Recordings: %1$d, total size: %2$.1f MB") )
+				return str( boost::format( tr("$Recordings: %1$d, total size: %2$.1f MB") )
 							% recordings.size() % ( double( job_.get_tracks_size() ) / MEGABYTE(1) ) );
 			else
 				return tr("No recordings marked for burning");
@@ -566,7 +566,7 @@
 
 				int percent;
 				if (manager::get_active()->get_is_burning(percent))
-					Add( new menu::text_item( str( boost::format( tr("Job active (Writing: %1$d%%)") ) % percent ) ) );
+					Add( new menu::text_item( str( boost::format( tr("$Job active (Writing: %1$d%%)") ) % percent ) ) );
 				else
 					Add(new menu::text_item(tr("Job active (Converting)")));
 
