--- jobs.c.orig	2006-09-16 20:33:36.000000000 +0200
+++ jobs.c	2006-09-20 21:50:33.000000000 +0200
@@ -304,10 +304,10 @@
 		size_pair::size_type diskFree =
 				size_pair::size_type( get_disk_size_mb() ) * MEGABYTE(1) - get_tracks_size( cut_, track_info::streamtype_audio );
 		size_pair::size_type videoSize =
-				size_pair::size_type( double( get_tracks_size( cut_, track_info::streamtype_video ) ) * 1.06 );
+				size_pair::size_type( double( get_tracks_size( cut_, track_info::streamtype_video ) ) * 1.04 );
 
 		return videoSize > diskFree
-				 ? double( get_tracks_size( cut_, track_info::streamtype_video ) ) / diskFree + .12
+				 ? double( videoSize ) / diskFree
 				 : 1;
 	}
 
