diff -Naur burn-0.1.0-pre21.orig/burn.c burn-0.1.0-pre21/burn.c
--- burn-0.1.0-pre21.orig/burn.c	2006-09-17 18:36:06.572462419 +0200
+++ burn-0.1.0-pre21/burn.c	2006-09-17 18:40:10.319695668 +0200
@@ -80,7 +80,7 @@
 
 	bool plugin::Start()
 	{
-		m_configPath = ConfigDirectory(PLUGIN_NAME);
+		m_configPath = ("/usr/share/vdr/burn");
 
 		manager::start();
 		logger_vdr::start();
diff -Naur burn-0.1.0-pre21.orig/vdrburn-dvd.sh burn-0.1.0-pre21/vdrburn-dvd.sh
--- burn-0.1.0-pre21.orig/vdrburn-dvd.sh	2006-09-17 18:36:06.732472419 +0200
+++ burn-0.1.0-pre21/vdrburn-dvd.sh	2006-09-17 18:45:42.924482170 +0200
@@ -32,10 +32,6 @@
 	export JAVA_HOME=/opt/j2re1.4.2
 fi
 
-if [ -z $PROJECTX_HOME ]; then
-	export PROJECTX_HOME=/opt/ProjectX
-fi
-
 # Some versions of growisofs refuse to start when run with sudo, and may
 # misdetect this condition if vdr itself is started from a sudo session.
 unset SUDO_COMMAND
@@ -75,9 +71,8 @@
 			CUT="-cut $MPEG_DATA_PATH/px.cut"
 		fi
 
-		$JAVA_HOME/bin/java -Djava.awt.headless=true \
-				-jar $PROJECTX_HOME/ProjectX.jar \
-				-ini $CONFIG_PATH/ProjectX.ini \
+				/usr/bin/projectx_cli \
+				-ini /usr/share/vdr/burn/projectx-vdr.ini \
 				$CUT -id $USED_TRACKS,0x1f,0x20 \
 				-demux -out "$MPEG_DATA_PATH" -name vdrsync \
 				$(ls "$MPEG_TMP_PATH/convert/"[0-9][0-9][0-9].vdr)
 
diff -Naur burn-0.1.0-pre21.orig/burn.c burn-0.1.0-pre21/burn.c
--- burn-0.1.0-pre21.orig/burn.c	2006-09-19 21:23:48.871275190 +0200
+++ burn-0.1.0-pre21/burn.c	2006-09-19 21:31:52.173479689 +0200
@@ -85,7 +85,7 @@
 		manager::start();
 		logger_vdr::start();
 		RegisterI18n( i18n::get_phrases() );
-		gdwrapper::setup::set_font_path( m_configPath + "/fonts" );
+		gdwrapper::setup::set_font_path( "/usr/share/fonts/ttf-bitstream-vera" );
 		if ( !skin_list::get().load( get_config_path() ) )
 			return false;
 
