Index: burn-0.1.0-pre21/Makefile
===================================================================
--- burn-0.1.0-pre21.orig/Makefile
+++ burn-0.1.0-pre21/Makefile
@@ -133,7 +133,7 @@ all: libvdr-$(PLUGIN).so burn-buffers i1
 
 SUBDIRS:
 	@for dir in $(SUBDIRS); do \
-		make -C $$dir CXX="$(CXX)" CXXFLAGS="$(CXXFLAGS)" lib$$dir.a ; \
+		$(MAKE) -C $$dir CXX="$(CXX)" CXXFLAGS="$(CXXFLAGS)" lib$$dir.a ; \
 	done
 
 libvdr-$(PLUGIN).so: $(OBJS) SUBDIRS Makefile
@@ -165,7 +165,7 @@ clean:
 	@-rm -f *.o genindex/*.o $(DEPFILE) *.so *.tgz core* *~ burn-buffers \
 		scan-test test t/*.o
 	@for dir in $(SUBDIRS); do \
-		make -C $$dir clean ; \
+		$(MAKE) -C $$dir clean ; \
 	done
 
 ### Implicit rules:
