diff -ru vdrcd-0.0.10.1-orig/Makefile vdrcd-0.0.10.1/Makefile
--- vdrcd-0.0.10.1-orig/Makefile	2006-11-16 11:19:16.000000000 +0100
+++ vdrcd-0.0.10.1/Makefile	2006-11-16 11:19:46.000000000 +0100
@@ -40,7 +40,7 @@
 
 ### Includes and Defines (add further entries here):
 
-INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)/include -I../../../../DVB/include
+INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)/include
 
 DEFINES += -DPLUGIN_NAME_I18N='"$(PLUGIN)"' -D_GNU_SOURCE
 
