--- vdrcd.c	2006-01-10 14:55:32.000000000 +0100
+++ vdrcd.c_with_mp3ng	2006-01-10 14:55:21.000000000 +0100
@@ -50,9 +50,9 @@
        { "photo_cd",   "pcd",     false, State, true  },
        { "001.vdr",    NULL,      true,  State, false },
        { "001.vdr",    NULL,      false, Find,  false },
-       { "*.mp3",      "mp3",     false, Find,  false },
-       { "*.ogg",      "mp3",     false, Find,  false },
-       { "*.wav",      "mp3",     false, Find,  false },
+       { "*.mp3",      "mp3ng",     false, Find,  false },
+       { "*.ogg",      "mp3ng",     false, Find,  false },
+       { "*.wav",      "mp3ng",     false, Find,  false },
        { "*.avi",      "mplayer", false, Find,  false },
        { "*.avi-????", "mplayer", false, Find,  false },
 	   { "*.mpg",      "mplayer", false, Find,  false },
@@ -364,7 +364,7 @@
 				Skins.Message(mtError, tr("Drive not present in mplayersources.conf!"));
 #endif
 			}
-		} else if (strcmp(Typelist[type].PlugIn, "mp3") == 0) {
+		} else if (strcmp(Typelist[type].PlugIn, "mp3ng") == 0) {
 			const char *wanted = trP("MP3", type);
 
 			// Switch to Browse Mode if not already done...

--- vdrcd.c	2006-01-10 20:19:23.995886750 +0100
+++ vdrcd-gentoo.c	2006-01-10 20:26:33.378721500 +0100
@@ -397,9 +397,9 @@
 
 			if (last == status->Current())
 #if VDRVERSNUM < 10307
-				Interface->Error(tr("Drive not present in mp3sources.conf!"));
+				Interface->Error(tr("Drive not present in mp3ng/mp3ngsources!"));
 #else
-				Skins.Message(mtError, tr("Drive not present in mp3sources.conf!"));
+				Skins.Message(mtError, tr("Drive not present in mp3ng/mp3ngsources!"));
 #endif
 
 		}
