diff -ur unace/declare.h unace-fixed/declare.h
--- unace/declare.h	2002-08-24 01:17:19.000000000 +0200
+++ unace-fixed/declare.h	2005-06-05 12:23:11.000000000 +0200
@@ -7,6 +7,23 @@
 #ifndef __declare_h
 #define __declare_h
 
+#ifdef UNIX
+
+#include <inttypes.h>
+
+typedef uint16_t       USHORT;
+typedef int16_t        SHORT ;
+typedef uint16_t       UWORD ;
+typedef int16_t        WORD  ;
+typedef uint32_t       ULONG ;
+typedef int32_t        LONG  ;
+typedef uint8_t        UCHAR ;
+typedef int8_t         CHAR  ;
+typedef unsigned       UINT  ;
+typedef int            INT   ;
+
+#else /* UNIX */
+
 #ifdef AMIGA
 
 #include <exec/types.h>
@@ -27,6 +44,7 @@
 typedef unsigned       UINT  ;
 typedef int            INT   ;
 
+#endif /* UNIX */
 
 #endif /* __declare_h */
 
diff -ur unace/unace.c unace-fixed/unace.c
--- unace/unace.c	2005-06-05 12:21:43.000000000 +0200
+++ unace-fixed/unace.c	2005-06-05 12:24:39.000000000 +0200
@@ -516,7 +516,7 @@
             tpsize+= fhead.PSIZE;
          }
          if (!f_err)
-            printf("%02u.%02u.%02u|%02u:%02u|%c%c%9lu|%9lu|%4u%%|%c%s\n",
+            printf("%02u.%02u.%02u|%02u:%02u|%c%c%9u|%9u|%4u%%|%c%s\n",
                    ts_day (ti), ts_month(ti), ts_year(ti)%100,
                    ts_hour(ti), ts_min  (ti),
                    fhead.HEAD_FLAGS & ACE_SP_BEF   ? '<' : ' ',
@@ -529,7 +529,7 @@
    }
    if (!f_err)
    {
-      printf("\n                 %9lu|%9lu|%4u%%| %u file%s",
+      printf("\n                 %9u|%9u|%4u%%| %u file%s",
              psize,
              size,
              percentage(psize, size),
