--- lib/platform/CXWindowsEventQueueBuffer.cpp~	2009-02-23 01:37:24.000000000 +0100
+++ lib/platform/CXWindowsEventQueueBuffer.cpp	2009-02-23 01:55:05.000000000 +0100
@@ -117,7 +117,7 @@
 	// we want to give the cpu a chance s owe up this to 25ms
 #define TIMEOUT_DELAY 25
 
-	while( remaining > 0 && QLength(m_display)==0 && retval==0){
+	while( ((dtimeout < 0.0) || (remaining > 0)) && QLength(m_display)==0 && retval==0){
 #if HAVE_POLL
 	retval = poll(pfds, 1, TIMEOUT_DELAY); //16ms = 60hz, but we make it > to play nicely with the cpu
 #else
