--- CMakeLists.txt
+++ CMakeLists.txt
@@ -496,7 +496,7 @@
 #    RENAME "sumwars.png")
 #endif()
 
-install(FILES "AUTHORS" "COPYING" "README" # we intentionally avoid installing "INSTALL" here
+install(FILES "AUTHORS" "README" # we intentionally avoid installing "INSTALL" here
   DESTINATION "${SUMWARS_DOC_DIR}")
   
 install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/share/data"
@@ -516,7 +516,7 @@
   SEPARATE_ARGUMENTS(SUMWARS_LANGUAGES)
   foreach(i ${SUMWARS_LANGUAGES})
     install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/share/translation/${i}"
-      DESTINATION "${SUMWARS_SHARE_DIR}"
+      DESTINATION "${SUMWARS_SHARE_DIR}/translation"
       FILES_MATCHING PATTERN "*.mo"
       PATTERN ".hg" EXCLUDE
       PATTERN "xmlgettext" EXCLUDE) # both pattern excludes are there to prevent the directories from being created
