From 36cb68919303bc2b874a5faa879e269a35874eff Mon Sep 17 00:00:00 2001
From: Johannes Schickel <lordhoto@scummvm.org>
Date: Sat, 3 Nov 2012 21:57:08 +0100
Subject: [PATCH] TOOLS: Fix decompiler compilation with gcc 4.7.

---
 decompiler/refcounted.h |   10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/decompiler/refcounted.h b/decompiler/refcounted.h
index a496ff7..f4d9020 100644
--- a/decompiler/refcounted.h
+++ b/decompiler/refcounted.h
@@ -25,10 +25,8 @@
 
 class RefCounted;
 
-namespace boost {
 inline void intrusive_ptr_add_ref(RefCounted *p);
 inline void intrusive_ptr_release(RefCounted *p);
-} // End of namespace boost
 
 /**
  * Provides a base implementation of reference counting for use with boost::intrusive_ptr.
@@ -36,16 +34,14 @@
 class RefCounted {
 private:
 	long _refCount; ///< Reference count used for boost::intrusive_ptr.
-	friend void ::boost::intrusive_ptr_add_ref(RefCounted *p); ///< Allow access by reference counting methods in boost namespace.
-	friend void ::boost::intrusive_ptr_release(RefCounted *p); ///< Allow access by reference counting methods in boost namespace.
+	friend void ::intrusive_ptr_add_ref(RefCounted *p); ///< Allow access by reference counting methods.
+	friend void ::intrusive_ptr_release(RefCounted *p); ///< Allow access by reference counting methods.
 
 protected:
 	RefCounted() : _refCount(0) { }
 	virtual ~RefCounted() { }
 };
 
-namespace boost {
-
 /**
  * Add a reference to a pointer.
  */
@@ -61,6 +57,4 @@ inline void intrusive_ptr_release(RefCounted *p) {
 		delete p;
 }
 
-} // End of namespace boost
-
 #endif
-- 
1.7.10

