From 51222609f1af2eda0c223881dda3985435c3ad3f Mon Sep 17 00:00:00 2001
From: Colin Walters <walters@verbum.org>
Date: Sun, 08 Jan 2012 16:29:33 +0000
Subject: test-service-glib: Fix build with GLib 2.30

Only <glib.h> can be included directly.
---
diff --git a/test/core/test-service-glib.c b/test/core/test-service-glib.c
index 3355890..0568e40 100644
--- a/test/core/test-service-glib.c
+++ b/test/core/test-service-glib.c
@@ -11,7 +11,6 @@
 #include <string.h>
 #include <glib/gi18n.h>
 #include <glib-object.h>
-#include <glib/gquark.h>
 
 #include "my-object.h"
 #include "my-object-subclass.h"
--
cgit v0.9.0.2-2-gbebe
