--- youtube-servicemenu-1.12a/get_yt_video.desktop
+++ youtube-servicemenu-1.12a/get_yt_video.desktop
@@ -1,8 +1,9 @@
 [Desktop Entry]
+Name=Youtube Service Menu
+Type=Service
 ServiceTypes=text/html
-Actions=get_video;get_video_and_play;
+Actions=get-video;get-video-and-play;
 Icon=video-x-generic
-Encoding=UTF-8
 X-KDE-Submenu=Download &YouTube Video
 X-KDE-Submenu[de]=&YouTube Video herunterladen
 X-KDE-Submenu[el]=Λήψη του βίντεο από το YouTube
@@ -11,7 +12,7 @@
 X-KDE-Submenu[sk]=Stiahnuť &YouTube video
 X-KDE-Submenu[cs]=Stáhnout &YouTube video
 
-[Desktop Action get_video]
+[Desktop Action get-video]
 Name=Download
 Name[de]=Herunterladen
 Name[el]=Λήψη
@@ -24,7 +25,7 @@
 Icon=download
 Exec=get_yt_video.py --format=ask %u
 
-[Desktop Action get_video_and_play]
+[Desktop Action get-video-and-play]
 Name=Download and play
 Name[de]=Herunterladen und abspielen
 Name[el]=Λήψη και αναπαραγωγή
