--- proxychains-proxychains-4.2.0/configure
+++ proxychains-proxychains-4.2.0/configure
@@ -86,6 +86,7 @@
 echo CC?=$CC>config.mak
 [ -z "$CPPFLAGS" ] || echo CPPFLAGS?=$CPPFLAGS>>config.mak
 [ -z "$CFLAGS" ] || echo USER_CFLAGS?=$CFLAGS>>config.mak
+[ -z "$LDFLAGS" ] || echo USER_LDFLAGS?=$LDFLAGS>>config.mak
 echo prefix=$prefix>>config.mak
 echo exec_prefix=$exec_prefix>>config.mak
 echo bindir=$bindir>>config.mak
--- proxychains-proxychains-4.2.0/Makefile
+++ proxychains-proxychains-4.2.0/Makefile
@@ -17,8 +17,8 @@
 OBJS = $(SRCS:.c=.o)
 LOBJS = src/core.o src/common.o src/libproxychains.o
 
-CCFLAGS  = -Wall -O2 -g -std=c99 -D_GNU_SOURCE -pipe -DTHREAD_SAFE -Werror 
-LDFLAGS = -shared -fPIC -lpthread
+CCFLAGS = -Wall -std=c99 -D_GNU_SOURCE -DTHREAD_SAFE
+LIBS    = -lpthread -ldl
 INC     = 
 PIC     = -fPIC
 AR      = $(CROSS_COMPILE)ar
@@ -30,11 +30,11 @@
 
 -include config.mak
 
-LDSO_PATHNAME = libproxychains4.$(LDSO_SUFFIX)
+LDSO_PATHNAME = libproxychains.$(LDSO_SUFFIX)
 
 SHARED_LIBS = $(LDSO_PATHNAME)
 ALL_LIBS = $(SHARED_LIBS)
-PXCHAINS = proxychains4
+PXCHAINS = proxychains
 ALL_TOOLS = $(PXCHAINS)
 
 CCFLAGS+=$(USER_CFLAGS) $(MAC_CFLAGS)
@@ -63,9 +63,9 @@
 	$(CC) $(CCFLAGS) $(CFLAGS_MAIN) $(INC) $(PIC) -c -o $@ $<
 
 $(LDSO_PATHNAME): $(LOBJS)
-	$(CC) $(LDFLAGS) $(LD_SET_SONAME)$(LDSO_PATHNAME) -o $@ $(LOBJS)
+	$(CC) $(LDFLAGS) -shared $(LD_SET_SONAME)$(LDSO_PATHNAME) -o $@ $(LOBJS) $(LIBS)
 
 $(ALL_TOOLS): $(OBJS)
-	$(CC) src/main.o src/common.o -o $(PXCHAINS)
+	$(CC) $(LDFLAGS) src/main.o src/common.o -o $(PXCHAINS)
 
 .PHONY: all clean install install-config
