--- AstroMenaceSource/Core/VirtualFileSystem/VFS.cpp.orig
+++ AstroMenaceSource/Core/VirtualFileSystem/VFS.cpp
@@ -554,7 +554,7 @@
 	//+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 	// Проверяем сигнатуру "VFS_" - 4 байт
 	//+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
-	char Sign[4]; Sign[4] = 0;
+	char Sign[4]; memset(Sign, 0, sizeof(Sign));
 	if(SDL_RWread(TempVFS->File, &Sign, 4, 1) == -1)
 	{
 		// если файл меньше, значит ошибка
@@ -572,7 +572,7 @@
 	//+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 	// Bерсия - 4 байт
 	//+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
-	char Version[4]; Version[4] = 0;
+	char Version[4]; memset(Version, 0, sizeof(Version));
 	if(SDL_RWread(TempVFS->File, &Version, 4, 1) == -1)
 	{
 		fprintf(stderr, "VFS file corrupted: %s\n", Name);
