From 54077b9aeffa0b8e2338d55c2097a1758128c7dc Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Sun, 6 Oct 2013 17:37:38 +0200
Subject: [PATCH] Allow disabling gstreamer-egl

---
 configure.ac | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 4f6b85b..1814496 100644
--- a/configure.ac
+++ b/configure.ac
@@ -494,7 +494,7 @@ if test x"$EGL_WINDOW_SYSTEM" = x"auto"; then
 fi
 
 case "$EGL_WINDOW_SYSTEM" in
-  x11|none)
+  x11)
     PKG_CHECK_MODULES(EGL, egl, HAVE_EGL="yes", [
       HAVE_EGL="no"
       old_LIBS=$LIBS
@@ -594,6 +594,10 @@ case "$EGL_WINDOW_SYSTEM" in
     LIBS=$old_LIBS
     CFLAGS=$old_CFLAGS
     ;;
+  none|no)
+    HAVE_EGL="no"
+    AC_MSG_WARN([No EGL window system specified, will not build gstreamer-egl])
+    ;;
   *)
     AC_MSG_ERROR([invalid EGL window system specified])
     ;;
-- 
1.8.3.2

