diff --git a/prosody.cfg.lua.dist b/prosody.cfg.lua.dist
index e513b11..08ebca6 100644
--- a/prosody.cfg.lua.dist
+++ b/prosody.cfg.lua.dist
@@ -105,6 +105,8 @@ ssl = {
 -- default provider stores passwords in plaintext. If you do not trust your
 -- server please see http://prosody.im/doc/modules/mod_auth_internal_hashed
 -- for information about using the hashed backend.
+-- See http://prosody.im/doc/authentication for other possibilities including
+-- Cyrus SASL.
 
 authentication = "internal_plain"
 
@@ -123,12 +125,19 @@ authentication = "internal_plain"
 -- Logging configuration
 -- For advanced logging see http://prosody.im/doc/logging
 log = {
-	info = "prosody.log"; -- Change 'info' to 'debug' for verbose logging
-	error = "prosody.err";
-	-- "*syslog"; -- Uncomment this for logging to syslog
+	info = "/var/log/jabber/prosody.log"; -- Change 'info' to 'debug' for verbose logging
+	error = "/var/log/jabber/prosody.err";
+	-- "*syslog"; -- Uncomment this for logging to syslog; needs mod_posix
 	-- "*console"; -- Log to the console, useful for debugging with daemonize=false
 }
 
+-- Pidfile, used by prosodyctl and the init.d script
+pidfile = "/var/run/jabber/prosody.pid"
+
+-- User and group, used for daemon
+prosody_user = "jabber"
+prosody_group = "jabber"
+
 ----------- Virtual hosts -----------
 -- You need to add a VirtualHost entry for each domain you wish Prosody to serve.
 -- Settings under each VirtualHost entry apply *only* to that host.
