diff -Naur gsoap-2.7.orig/configure.in gsoap-2.7/configure.in
--- gsoap-2.7.orig/configure.in	2010-03-04 10:04:10.879898902 +0100
+++ gsoap-2.7/configure.in	2010-03-04 10:04:57.182889612 +0100
@@ -15,8 +15,8 @@
 AM_PROG_LEX
 AC_PROG_YACC
 AC_PROG_CPP
-AC_PROG_RANLIB
-#AC_PROG_LIBTOOL
+#AC_PROG_RANLIB
+AC_PROG_LIBTOOL
 AC_PROG_LN_S
 AC_PROG_AWK
 AC_PROG_INSTALL
@@ -158,12 +158,12 @@
   AC_MSG_RESULT(no)
   WSDL2H_EXTRA_FLAGS="-DWITH_OPENSSL -DWITH_GZIP"
   WSDL2H_EXTRA_LIBS="-lssl -lcrypto -lz"
-  WSDL2H_SOAP_CPP_LIB="libgsoapssl++.a"
+  WSDL2H_SOAP_CPP_LIB="libgsoapssl++.la"
 else
   AC_MSG_RESULT(yes)
   WSDL2H_EXTRA_FLAGS=
   WSDL2H_EXTRA_LIBS=
-  WSDL2H_SOAP_CPP_LIB="libgsoap++.a"
+  WSDL2H_SOAP_CPP_LIB="libgsoap++.la"
 fi
 AC_SUBST(WSDL2H_EXTRA_FLAGS)
 AC_SUBST(WSDL2H_EXTRA_LIBS)
diff -Naur gsoap-2.7.orig/gsoap/Makefile.am gsoap-2.7/gsoap/Makefile.am
--- gsoap-2.7.orig/gsoap/Makefile.am	2010-03-04 10:04:10.920894818 +0100
+++ gsoap-2.7/gsoap/Makefile.am	2010-03-04 10:06:28.000883644 +0100
@@ -31,20 +31,20 @@
 stdsoap2_ssl_cpp.cpp: stdsoap2.cpp
 	$(LN_S) -f $(srcdir)/stdsoap2.cpp stdsoap2_ssl_cpp.cpp
 
-lib_LIBRARIES = libgsoap.a libgsoap++.a libgsoapck.a libgsoapck++.a libgsoapssl.a libgsoapssl++.a
+lib_LTLIBRARIES = libgsoap.la libgsoap++.la libgsoapck.la libgsoapck++.la libgsoapssl.la libgsoapssl++.la
 
-libgsoap_a_SOURCES = stdsoap2.c
-libgsoap_a_CFLAGS = $(SOAPCPP2_DEBUG) $(SOAPCPP2_NONAMESPACES) -D$(platform)
-libgsoap___a_SOURCES = stdsoap2_cpp.cpp
-libgsoap___a_CXXFLAGS = $(SOAPCPP2_DEBUG) $(SOAPCPP2_NONAMESPACES) -D$(platform)
-libgsoapck_a_SOURCES = stdsoap2_ck.c
-libgsoapck_a_CFLAGS = $(SOAPCPP2_DEBUG) $(SOAPCPP2_NONAMESPACES) -D$(platform) -DWITH_COOKIES
-libgsoapck___a_SOURCES = stdsoap2_ck_cpp.cpp
-libgsoapck___a_CXXFLAGS = $(SOAPCPP2_DEBUG) $(SOAPCPP2_NONAMESPACES) -D$(platform) -DWITH_COOKIES
-libgsoapssl_a_SOURCES = stdsoap2_ssl.c
-libgsoapssl_a_CFLAGS = $(SOAPCPP2_DEBUG) $(SOAPCPP2_NONAMESPACES) -D$(platform) -DWITH_OPENSSL -DWITH_DOM -DWITH_GZIP
-libgsoapssl___a_SOURCES = stdsoap2_ssl_cpp.cpp
-libgsoapssl___a_CXXFLAGS = $(SOAPCPP2_DEBUG) $(SOAPCPP2_NONAMESPACES) -D$(platform) -DWITH_OPENSSL -DWITH_DOM -DWITH_GZIP
+libgsoap_la_SOURCES = stdsoap2.c
+libgsoap_la_CFLAGS = $(SOAPCPP2_DEBUG) $(SOAPCPP2_NONAMESPACES) -D$(platform)
+libgsoap___la_SOURCES = stdsoap2_cpp.cpp
+libgsoap___la_CXXFLAGS = $(SOAPCPP2_DEBUG) $(SOAPCPP2_NONAMESPACES) -D$(platform)
+libgsoapck_la_SOURCES = stdsoap2_ck.c
+libgsoapck_la_CFLAGS = $(SOAPCPP2_DEBUG) $(SOAPCPP2_NONAMESPACES) -D$(platform) -DWITH_COOKIES
+libgsoapck___la_SOURCES = stdsoap2_ck_cpp.cpp
+libgsoapck___la_CXXFLAGS = $(SOAPCPP2_DEBUG) $(SOAPCPP2_NONAMESPACES) -D$(platform) -DWITH_COOKIES
+libgsoapssl_la_SOURCES = stdsoap2_ssl.c
+libgsoapssl_la_CFLAGS = $(SOAPCPP2_DEBUG) $(SOAPCPP2_NONAMESPACES) -D$(platform) -DWITH_OPENSSL -DWITH_DOM -DWITH_GZIP
+libgsoapssl___la_SOURCES = stdsoap2_ssl_cpp.cpp
+libgsoapssl___la_CXXFLAGS = $(SOAPCPP2_DEBUG) $(SOAPCPP2_NONAMESPACES) -D$(platform) -DWITH_OPENSSL -DWITH_DOM -DWITH_GZIP
 
 BUILT_SOURCES = stdsoap2_cpp.cpp $(lib_LIBRARIES)
 
diff -Naur gsoap-2.7.orig/gsoap/samples/databinding/Makefile.am gsoap-2.7/gsoap/samples/databinding/Makefile.am
--- gsoap-2.7.orig/gsoap/samples/databinding/Makefile.am	2010-03-04 10:04:10.945913061 +0100
+++ gsoap-2.7/gsoap/samples/databinding/Makefile.am	2010-03-04 10:13:18.323885257 +0100
@@ -14,7 +14,7 @@
 WSDLINPUT=$(srcdir)/address.xsd
 SOAPHEADER=$(srcdir)/address.h
 SOAP_CPP_SRC=addressC.cpp
-SOAP_CPP_LIB=$(top_builddir)/gsoap/libgsoap++.a
+SOAP_CPP_LIB=$(top_builddir)/gsoap/libgsoap++.la
 
 $(SOAP_CPP_SRC) : $(WSDLINPUT)
 	$(WSDL) $(WSDL_FLAGS) $(WSDLINPUT)
diff -Naur gsoap-2.7.orig/gsoap/samples/Makefile.defines gsoap-2.7/gsoap/samples/Makefile.defines
--- gsoap-2.7.orig/gsoap/samples/Makefile.defines	2010-03-04 10:04:10.933894722 +0100
+++ gsoap-2.7/gsoap/samples/Makefile.defines	2010-03-04 10:11:37.388944663 +0100
@@ -15,13 +15,13 @@
 SOAP_C_CORE=soapC.c
 SOAP_C_CLIENT=soapClient.c $(SOAP_C_CORE)
 SOAP_C_SERVER=soapServer.c $(SOAP_C_CORE)
-SOAP_C_LIB=$(top_builddir)/gsoap/libgsoap.a
-SOAP_C_LIB_CK=$(top_builddir)/gsoap/libgsoapck.a
-SOAP_C_LIB_SSL=$(top_builddir)/gsoap/libgsoapssl.a
+SOAP_C_LIB=$(top_builddir)/gsoap/libgsoap.la
+SOAP_C_LIB_CK=$(top_builddir)/gsoap/libgsoapck.la
+SOAP_C_LIB_SSL=$(top_builddir)/gsoap/libgsoapssl.la
 
 SOAP_CPP_CORE=soapC.cpp
 SOAP_CPP_CLIENT=soapClient.cpp $(SOAP_CPP_CORE)
 SOAP_CPP_SERVER=soapServer.cpp $(SOAP_CPP_CORE)
-SOAP_CPP_LIB=$(top_builddir)/gsoap/libgsoap++.a
-SOAP_CPP_LIB_CK=$(top_builddir)/gsoap/libgsoapck++.a
-SOAP_CPP_LIB_SSL=$(top_builddir)/gsoap/libgsoapssl++.a
+SOAP_CPP_LIB=$(top_builddir)/gsoap/libgsoap++.la
+SOAP_CPP_LIB_CK=$(top_builddir)/gsoap/libgsoapck++.la
+SOAP_CPP_LIB_SSL=$(top_builddir)/gsoap/libgsoapssl++.la
