 configure.in |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.in b/configure.in
index 2573590..d9796ec 100644
--- a/configure.in
+++ b/configure.in
@@ -102,8 +102,8 @@ dnl        Check for libWINGs (replaces libPropList)
 dnl =====================================================
 
 AC_CHECK_LIB(WINGs, WMGetFromPLString,
-    [LIBPL_LIBS=-lWINGs LIBPL_INC_PATH="" missing=no],
-    [missing=yes])
+    [LIBPL_LIBS="`get-wings-flags --libs`" LIBPL_INC_PATH="`get-wings-flags --cflags`" missing=no],
+    [missing=yes], $XLFLAGS $XLIBS `get-wings-flags --libs`)
 
 if test "$missing" = yes; then
     echo "********** WARNING *************"
