diff -Naur mountapp-3.1.orig/src/properties.c mountapp-3.1/src/properties.c
--- mountapp-3.1.orig/src/properties.c	2012-07-04 11:36:00.177181490 +0200
+++ mountapp-3.1/src/properties.c	2012-07-04 11:36:57.007572387 +0200
@@ -448,7 +448,7 @@
         WMReleasePropList(mpProp);
     }
 
-    if (WMWritePropListToFile(props, getFileName(), FALSE) == FALSE)
+    if (WMWritePropListToFile(props, getFileName()) == FALSE)
         printf("Error occurred synchronizing proplist\n");
 
     WMReleasePropList(props);
