From: Julian Ospald <hasufell@gentoo.org>
Date: Wed Dec 25 22:55:09 UTC 2013
Subject: respect flags, fix installation

--- dustrac-1.6.2/CMakeLists.txt
+++ dustrac-1.6.2/CMakeLists.txt
@@ -43,7 +43,7 @@
 endif()
 
 set(CMAKE_VERBOSE_MAKEFILE OFF)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x -W -Wall -O3 -pedantic")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x -W -Wall")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fomit-frame-pointer -finline-functions -ffast-math")
 
 set(GAME_BINARY_NAME "dustrac-game")
--- dustrac-1.6.2/src/game/CMakeLists.txt
+++ dustrac-1.6.2/src/game/CMakeLists.txt
@@ -128,16 +128,11 @@
 
 # Copy desktop file
 if(ReleaseBuild AND ${CMAKE_SYSTEM_NAME} MATCHES "Linux")
-    set(DesktopFileSourcePath)
     if(USC)
-        set(DesktopFileSourcePath ${CMAKE_SOURCE_DIR}/src/dustrac-game.desktop.opt.in)
+        configure_file(${CMAKE_SOURCE_DIR}/src/dustrac-game.desktop.opt.in ${CMAKE_BINARY_DIR}/src/dustrac-game.desktop)
     else()
-        set(DesktopFileSourcePath ${CMAKE_SOURCE_DIR}/src/dustrac-game.desktop.in)
+        configure_file(${CMAKE_SOURCE_DIR}/src/dustrac-game.desktop.in ${CMAKE_BINARY_DIR}/src/dustrac-game.desktop)
     endif()
-    add_custom_target(desktop-file-game ALL
-        COMMAND cmake -E copy ${DesktopFileSourcePath}
-            ${CMAKE_BINARY_DIR}/dustrac-game.desktop
-    DEPENDS ${GAME_BINARY_NAME})
 endif()
 
 # Resource compilation for MinGW
--- dustrac-1.6.2/src/editor/CMakeLists.txt
+++ dustrac-1.6.2/src/editor/CMakeLists.txt
@@ -76,16 +76,11 @@
 
 # Copy desktop file
 if(ReleaseBuild AND ${CMAKE_SYSTEM_NAME} MATCHES "Linux")
-    set(DesktopFileSourcePath)
     if(USC)
-        set(DesktopFileSourcePath ${CMAKE_SOURCE_DIR}/src/dustrac-editor.desktop.opt.in)
+        configure_file(${CMAKE_SOURCE_DIR}/src/dustrac-editor.desktop.opt.in ${CMAKE_BINARY_DIR}/src/dustrac-editor.desktop)
     else()
-        set(DesktopFileSourcePath ${CMAKE_SOURCE_DIR}/src/dustrac-editor.desktop.in)
+        configure_file(${CMAKE_SOURCE_DIR}/src/dustrac-editor.desktop.in ${CMAKE_BINARY_DIR}/src/dustrac-editor.desktop)
     endif()
-    add_custom_target(desktop-file-editor ALL
-        COMMAND cmake -E copy ${DesktopFileSourcePath}
-            ${CMAKE_BINARY_DIR}/dustrac-editor.desktop
-    DEPENDS ${EDITOR_BINARY_NAME})
 endif()
 
 # Add the executable
--- dustrac-1.6.2/InstallLinux.cmake
+++ dustrac-1.6.2/InstallLinux.cmake
@@ -58,7 +58,7 @@
     install(FILES data/fonts.conf DESTINATION ${DATA_PATH})
     install(FILES data/meshes.conf DESTINATION ${DATA_PATH})
     install(FILES data/textures.conf DESTINATION ${DATA_PATH})
-    install(FILES AUTHORS CHANGELOG COPYING README DESTINATION ${DOC_PATH})
+    install(FILES AUTHORS CHANGELOG README DESTINATION ${DOC_PATH})
     install(DIRECTORY data/images DESTINATION ${DATA_PATH} FILES_MATCHING PATTERN "*.jpg")
     install(DIRECTORY data/images DESTINATION ${DATA_PATH} FILES_MATCHING PATTERN "*.png")
     install(DIRECTORY data/levels DESTINATION ${DATA_PATH} FILES_MATCHING PATTERN "*.trk")
@@ -69,8 +69,8 @@
 
     if(ReleaseBuild)
         # Install .desktop files
-        install(FILES dustrac-game.desktop DESTINATION share/applications)
-        install(FILES dustrac-editor.desktop DESTINATION share/applications)
+        install(FILES ${CMAKE_BINARY_DIR}/src/dustrac-game.desktop DESTINATION share/applications)
+        install(FILES ${CMAKE_BINARY_DIR}/src/dustrac-editor.desktop DESTINATION share/applications)
 
         # Install icons
         install(FILES data/icons/dustrac-game.png DESTINATION share/pixmaps)
