From: Julian Ospald <hasufell@gentoo.org>
Date: Wed Dec 25 23:21:50 UTC 2013
Subject: fix desktop files

--- dustrac-1.6.2/src/dustrac-editor.desktop.in
+++ dustrac-1.6.2/src/dustrac-editor.desktop.in
@@ -4,5 +4,5 @@
 Exec=dustrac-editor
 Icon=dustrac-editor
 Type=Application
-Categories=Game
+Categories=Game;
 StartupNotify=true
--- dustrac-1.6.2/src/dustrac-editor.desktop.opt.in
+++ dustrac-1.6.2/src/dustrac-editor.desktop.opt.in
@@ -4,5 +4,5 @@
 Exec=/opt/dustrac/dustrac-editor
 Icon=dustrac-editor
 Type=Application
-Categories=Game
+Categories=Game;
 StartupNotify=true
--- dustrac-1.6.2/src/dustrac-game.desktop.in
+++ dustrac-1.6.2/src/dustrac-game.desktop.in
@@ -4,5 +4,5 @@
 Exec=dustrac-game
 Icon=dustrac-game
 Type=Application
-Categories=Game
+Categories=Game;
 StartupNotify=true
--- dustrac-1.6.2/src/dustrac-game.desktop.opt.in
+++ dustrac-1.6.2/src/dustrac-game.desktop.opt.in
@@ -4,5 +4,5 @@
 Exec=/opt/dustrac/dustrac-game
 Icon=dustrac-game
 Type=Application
-Categories=Game
+Categories=Game;
 StartupNotify=true
