diff -ru dustrac-1.9.2.orig/CMakeLists.txt dustrac-1.9.2/CMakeLists.txt
--- dustrac-1.9.2.orig/CMakeLists.txt	2015-01-18 15:25:19.000000000 -0500
+++ dustrac-1.9.2/CMakeLists.txt	2015-02-26 12:28:00.270875467 -0500
@@ -68,8 +68,7 @@
 endif()
 
 if(CMAKE_COMPILER_IS_GNUCXX OR MINGW OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -W -Wall -O3 -pedantic")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fomit-frame-pointer -finline-functions -ffast-math")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -W -Wall -fomit-frame-pointer -finline-functions")
 elseif(MSVC)
     add_definitions(-DNOMINMAX)
 endif()
diff -ru dustrac-1.9.2.orig/InstallLinux.cmake dustrac-1.9.2/InstallLinux.cmake
--- dustrac-1.9.2.orig/InstallLinux.cmake	2015-01-18 15:25:19.000000000 -0500
+++ dustrac-1.9.2/InstallLinux.cmake	2015-02-26 12:28:27.246817132 -0500
@@ -58,7 +58,7 @@
     install(FILES data/fonts.conf DESTINATION ${DATA_PATH})
     install(FILES data/meshes.conf DESTINATION ${DATA_PATH})
     install(FILES data/textures.conf DESTINATION ${DATA_PATH})
-    install(FILES AUTHORS CHANGELOG COPYING README DESTINATION ${DOC_PATH})
+    install(FILES AUTHORS CHANGELOG README DESTINATION ${DOC_PATH})
     install(DIRECTORY data/images DESTINATION ${DATA_PATH} FILES_MATCHING PATTERN "*.jpg")
     install(DIRECTORY data/images DESTINATION ${DATA_PATH} FILES_MATCHING PATTERN "*.png")
     install(DIRECTORY data/levels DESTINATION ${DATA_PATH} FILES_MATCHING PATTERN "*.trk")
