From 43a981481ab9161a83dd200db39c093e2502f68a Mon Sep 17 00:00:00 2001
From: Bernd Buschinski <b.buschinski@googlemail.com>
Date: Wed, 4 Sep 2013 16:23:35 +0200
Subject: [PATCH] Fix toplevel bookmarks not showing in toolbar regression from
 d1963e23

---
 konqueror/src/konqbookmarkbar.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/konqueror/src/konqbookmarkbar.cpp b/konqueror/src/konqbookmarkbar.cpp
index 2141575..199e15f 100644
--- a/konqueror/src/konqbookmarkbar.cpp
+++ b/konqueror/src/konqbookmarkbar.cpp
@@ -184,10 +184,12 @@ void KBookmarkBar::fillBookmarkBar(const KBookmarkGroup & parent)
 
         if (!bm.isGroup())
         {
-	    if ( bm.isSeparator() )
+            if ( bm.isSeparator() )
+            {
                 if (m_toolBar) {
                     m_toolBar->addSeparator();
                 }
+            }
             else
             {
                 KAction *action = new KBookmarkAction( bm, m_pOwner, 0 );
-- 
1.8.3.2

