diff --git a/spec/support/shared/contexts/msf/util/exe.rb b/spec/support/shared/contexts/msf/util/exe.rb
index e1372a6..c1b6ea5 100644
--- a/spec/support/shared/contexts/msf/util/exe.rb
+++ b/spec/support/shared/contexts/msf/util/exe.rb
@@ -36,8 +36,8 @@ shared_context 'Msf::Util::Exe' do
       { :format => "psh",  :arch => "x86_64", :file_fp => /ASCII/  },
       { :format => "psh-net",  :arch => "x86", :file_fp => /ASCII/  },
       { :format => "psh-net",  :arch => "x86_64", :file_fp => /ASCII/  },
-      { :format => "war",  :arch => "x86", :file_fp => /zip/i  },
-      { :format => "war",  :arch => "x86_64", :file_fp => /zip/i  },
+      { :format => "war",  :arch => "x86", :file_fp => /zip|jar/i  },
+      { :format => "war",  :arch => "x86_64", :file_fp => /zip|jar/i  },
       { :format => "msi",  :arch => "x86", :file_fp => /(Composite Document)|(CDF V2 Document)/  },
       { :format => "msi",  :arch => "x64", :file_fp => /(Composite Document)|(CDF V2 Document)/  },
       { :format => "msi",  :arch => "x86_64", :file_fp => /(Composite Document)|(CDF V2 Document)/  },
@@ -51,29 +51,29 @@ shared_context 'Msf::Util::Exe' do
       { :format => "elf", :arch => "armle",  :file_fp => /ELF 32.*ARM/ },
       { :format => "elf", :arch => "mipsbe", :file_fp => /ELF 32-bit MSB executable, MIPS/ },
       { :format => "elf", :arch => "mipsle", :file_fp => /ELF 32-bit LSB executable, MIPS/ },
-      { :format => "war",  :arch => "x86", :file_fp => /zip/i  },
-      { :format => "war",  :arch => "x64", :file_fp => /zip/i  },
-      { :format => "war",  :arch => "armle", :file_fp => /zip/i  },
-      { :format => "war",  :arch => "mipsbe", :file_fp => /zip/i  },
-      { :format => "war",  :arch => "mipsle", :file_fp => /zip/i  },
+      { :format => "war",  :arch => "x86", :file_fp => /zip|jar/i  },
+      { :format => "war",  :arch => "x64", :file_fp => /zip|jar/i  },
+      { :format => "war",  :arch => "armle", :file_fp => /zip|jar/i  },
+      { :format => "war",  :arch => "mipsbe", :file_fp => /zip|jar/i  },
+      { :format => "war",  :arch => "mipsle", :file_fp => /zip|jar/i  },
     ],
     "bsd" => [
       { :format => "elf", :arch => "x86", :file_fp => /ELF 32.*BSD/ },
-      { :format => "war",  :arch => "x86", :file_fp => /zip/i  },
+      { :format => "war",  :arch => "x86", :file_fp => /zip|jar/i  },
     ],
     "solaris" => [
       { :format => "elf", :arch => "x86", :file_fp => /ELF 32/ },
-      { :format => "war",  :arch => "x86", :file_fp => /zip/i  },
+      { :format => "war",  :arch => "x86", :file_fp => /zip|jar/i  },
     ],
     "osx" => [
       { :format => "macho", :arch => "x86",   :file_fp => /Mach-O.*i386/  },
       { :format => "macho", :arch => "x64",   :file_fp => /Mach-O 64/     },
       { :format => "macho", :arch => "armle", :file_fp => /Mach-O.*(acorn|arm)/ },
       { :format => "macho", :arch => "ppc",   :file_fp => /Mach-O.*ppc/   },
-      { :format => "war",  :arch => "x86", :file_fp => /zip/i  },
-      { :format => "war",  :arch => "x64", :file_fp => /zip/i  },
-      { :format => "war",  :arch => "armle", :file_fp => /zip/i  },
-      { :format => "war",  :arch => "ppc", :file_fp => /zip/i  },
+      { :format => "war",  :arch => "x86", :file_fp => /zip|jar/i  },
+      { :format => "war",  :arch => "x64", :file_fp => /zip|jar/i  },
+      { :format => "war",  :arch => "armle", :file_fp => /zip|jar/i  },
+      { :format => "war",  :arch => "ppc", :file_fp => /zip|jar/i  },
     ],
   }
 
