--- man/autofs.8.in.old	2005-01-05 11:12:42.000000000 +0100
+++ man/autofs.8.in	2010-05-31 23:10:16.589319809 +0200
@@ -4,7 +4,7 @@
 @@INIT.D@@/autofs \- Control Script for automounter
 .SH SYNOPSIS
 .B @@INIT.D@@/autofs
-.I start|stop|reload|status
+.I start|stop|reload|stats|status
 .SH "DESCRIPTION"
 .B autofs
 control the operation of the
@@ -30,7 +30,7 @@ to find mount points on the system. For 
 .BR automount (8)
 process is started with the appropriate parameters. You can check the
 active mount points for the automounter with the
-.B @@INIT.D@@/autofs status
+.B @@INIT.D@@/autofs stats
 command. If the
 .I auto.master
 configuration file contains a line of the form
@@ -54,9 +54,12 @@ map is modified then the
 .B autofs
 script must be rerun to activate the changes.
 .P
-.B @@INIT.D@@/autofs status
+.B @@INIT.D@@/autofs stats
 will display the current configuration and a list of currently running
 automount daemons.
+.P
+.B @@INIT.D@@/autofs status
+will display whether automount service is running.
 .SH "SEE ALSO"
 .BR automount (8),
 .BR autofs (5),
@@ -64,4 +67,5 @@ automount daemons.
 .SH AUTHOR
 This manual page was written by Christoph Lameter <chris@waterf.org>,
 for the Debian GNU/Linux system.  Edited by H. Peter Anvin
-<hpa@transmeta.com>.
+<hpa@transmeta.com>.  Edited by D. Polke <dustin.polke@uni-siegen.de> to
+reflect Gentoo's init script implementation.
