--- gentoo/net-fs/autofs/files/autofs.init.orig
+++ gentoo/net-fs/autofs/files/autofs.init
@@ -11,11 +11,12 @@ DAEMON=/usr/sbin/automount
 SYSCONFDIR=/etc/autofs
 
 depend() {
-	need localmount	portmap
-	use ypbind nfs slapd portmap
+	need localmount
+	use ypbind nfs slapd portmap net
 }
 
-opts="start stop status stats reload restart"
+extra_commands="stats"
+extra_started_commands="reload"
 
 #
 # Check for all maps that are to be loaded
