=== modified file 'Makefile.rules'
--- Makefile.rules	2011-04-30 18:13:36 +0000
+++ Makefile.rules	2011-04-30 18:12:37 +0000
@@ -42,6 +42,10 @@
 AUTOFS_LDFLAGS += $(DMALLOCLIB)
 endif
 
+ifdef SASL
+CFLAGS +=  $(XML_FLAGS)
+endif
+
 # Standard rules
 
 .SUFFIXES: .c .o .s .so

=== modified file 'lib/defaults.c'
--- lib/defaults.c	2011-04-30 18:13:36 +0000
+++ lib/defaults.c	2011-04-30 18:12:37 +0000
@@ -19,7 +19,10 @@
 
 #include "list.h"
 #include "defaults.h"
+#include "config.h"
+#ifdef WITH_LDAP
 #include "lookup_ldap.h"
+#endif
 #include "log.h"
 #include "automount.h"
 
@@ -197,6 +200,7 @@
 	return 1;
 }
 
+#ifdef WITH_LDAP
 void defaults_free_uris(struct list_head *list)
 {
 	struct list_head *next;
@@ -252,9 +256,11 @@
 
 	return 1;
 }
+#endif
 
 struct list_head *defaults_get_uris(void)
 {
+#ifdef WITH_LDAP
 	FILE *f;
 	char buf[MAX_LINE_LEN];
 	char *res;
@@ -288,6 +294,9 @@
 
 	fclose(f);
 	return list;
+#else
+	return NULL;
+#endif
 }
 
 /*
@@ -450,6 +459,7 @@
 	return res;
 }
 
+#ifdef WITH_LDAP
 struct ldap_schema *defaults_get_default_schema(void)
 {
 	struct ldap_schema *schema;
@@ -645,6 +655,7 @@
 
 	return schema;
 }
+#endif
 
 unsigned int defaults_get_mount_nfs_default_proto(void)
 {

