--- autofs-4.1.3/daemon/automount.c.orig	2004-08-18 11:23:49.430470256 -0400
+++ autofs-4.1.3/daemon/automount.c	2004-08-18 11:24:45.047015256 -0400
@@ -848,10 +848,10 @@ static int st_prepare_shutdown(void)
 	alarm(0);
 
 	/* Prevent any new mounts */
-	ap.state = ST_SHUTDOWN_PENDING;
-
 	sigprocmask(SIG_SETMASK, &lock_sigs, NULL);
 
+	ap.state = ST_SHUTDOWN_PENDING;
+
 	/* Unmount everything */
 	exp = expire_proc(1);
 
