diff -ur libkarma-0.1.0.orig/karma-sharp/Makefile libkarma-0.1.0/karma-sharp/Makefile
--- libkarma-0.1.0.orig/karma-sharp/Makefile	2007-05-15 19:29:02.000000000 +0300
+++ libkarma-0.1.0/karma-sharp/Makefile	2009-05-12 13:58:55.000000000 +0300
@@ -26,18 +26,18 @@
 
 ifdef HAS_MONO
 install: $(ASSEMBLY) karma-sharp.pc
-	../install-sh -d $(DEST)/lib/karma-sharp
-	../install-sh -d $(DEST)/lib/pkgconfig
-	../install-sh -t $(DEST)/lib/karma-sharp $(ASSEMBLY)
-	../install-sh -t $(DEST)/lib/pkgconfig -m 0644 karma-sharp.pc
+	../install-sh -d $(DESTDIR)$(PREFIX)/lib/karma-sharp
+	../install-sh -d $(DESTDIR)$(PREFIX)/lib/pkgconfig
+	../install-sh -t $(DESTDIR)$(PREFIX)/lib/karma-sharp $(ASSEMBLY)
+	../install-sh -t $(DESTDIR)$(PREFIX)/lib/pkgconfig -m 0644 karma-sharp.pc
 else
 install:
 
 endif
 
 uninstall: FORCE
-	$(RM) -r $(DEST)/lib/karma-sharp
-	$(RM) $(DEST)/lib/pkgconfig/karma-sharp.pc
+	$(RM) -r $(PREFIX)/lib/karma-sharp
+	$(RM) $(PREFIX)/lib/pkgconfig/karma-sharp.pc
 
 $(ASSEMBLY): $(ASSEMBLY_SOURCES) $(ASSEMBLY_RESOURCES)
 	$(MCS) $(MCS_FLAGS) -unsafe -target:library -out:$@ $(ASSEMBLY_SOURCES) $(ASSEMBLY_RESOURCES:%=-resource:%) -r:Mono.Posix
diff -ur libkarma-0.1.0.orig/Makefile libkarma-0.1.0/Makefile
--- libkarma-0.1.0.orig/Makefile	2007-05-15 19:29:02.000000000 +0300
+++ libkarma-0.1.0/Makefile	2009-05-12 13:57:16.000000000 +0300
@@ -1,14 +1,12 @@
 LK_SRC=src
 TOOLSDIR=tools
-PREFIX=/usr/local
+PREFIX=/usr
 INSTALL?=./install-sh
 VERSION=0.1.0
-DEST=$(PREFIX)
 SHELL=/bin/bash
 
 export PREFIX
 export VERSION
-export DEST
 
 DISTNAME=libkarma-$(VERSION)
 DISTFILES= $(LK_SRC)/{Jamfile,Makefile,*[ch]} \
@@ -16,11 +14,10 @@
 	karma-sharp/{Makefile,karma-sharp.pc.in,Song.cs,Device.cs} \
 	Jamfile Makefile COPYING ChangeLog THANKS INSTALL* TODO \
 	install-sh karma-tools.spec
-DOCS=THANKS TODO ChangeLog
 
 INSTALL_DIR     := $(INSTALL) -d -o root -g root -m 0755
 INSTALL_FILE    := $(INSTALL)    -o root -g root -m 0644
-INSTALL_PROGRAM := $(INSTALL)    -o root -g root -m 0755 # -s
+INSTALL_PROGRAM := $(INSTALL)    -o root -g root -m 0755
 INSTALL_SCRIPT  := $(INSTALL)    -o root -g root -m 0755
 
 all: libkarma tools karma-sharp
@@ -31,14 +28,11 @@
 	cd $(LK_SRC) && $(MAKE) install
 	cd karma-sharp && $(MAKE) install
 	cd tools && $(MAKE) install
-	$(INSTALL_DIR) $(DESTDIR)$(PREFIX)/share/doc/libkarma
-	$(INSTALL_FILE) $(DOCS) $(DESTDIR)$(PREFIX)/share/doc/libkarma
 
 uninstall: 
 	cd $(LK_SRC) && $(MAKE) uninstall
 	cd karma-sharp && $(MAKE) uninstall
 	cd tools && $(MAKE) uninstall
-	rm -rf $(DESTDIR)$(PREFIX)/share/doc/libkarma
 
 libkarma: FORCE
 	cd $(LK_SRC) && $(MAKE)
diff -ur libkarma-0.1.0.orig/src/Makefile libkarma-0.1.0/src/Makefile
--- libkarma-0.1.0.orig/src/Makefile	2007-05-15 19:29:02.000000000 +0300
+++ libkarma-0.1.0/src/Makefile	2009-05-12 14:06:24.000000000 +0300
@@ -14,7 +14,7 @@
 LN_S?=ln -s
 
 CFLAGS+=-fPIC -D_REENTRANT -Wall -pedantic
-CFLAGS+=-ggdb -W -Wchar-subscripts -Wmissing-prototypes 
+CFLAGS+=-W -Wchar-subscripts -Wmissing-prototypes 
 CFLAGS+=-Wmissing-declarations -Wno-switch -Wredundant-decls -Wno-unused
 
 MAJOR?=$(shell VERSION=${VERSION} echo $${VERSION%%\.*})
@@ -37,20 +37,20 @@
 default: all
 
 install: shared-lib-stamp static-lib-stamp
-	../install-sh -m 0755 -d $(DEST)/lib
-	../install-sh -m 0755 -d $(DEST)/include
-	../install-sh -m 0755 -d $(DEST)/include/libkarma
-	../install-sh -m 0644 -t $(DEST)/lib/ $(LIBDIR)/$(TARGET).a
-	../install-sh -m 0755 -t $(PREFIX)/lib/ $(LIBDIR)/$(OUT_VERSION)
-	$(RM) $(PREFIX)/lib/$(OUT_MAJOR)
-	$(RM) $(PREFIX)/lib/$(OUT_BASE)
-	$(LN_S) $(OUT_VERSION) $(PREFIX)/lib/$(OUT_MAJOR)
-	$(LN_S) $(OUT_VERSION) $(PREFIX)/lib/$(OUT_BASE)
-	../install-sh -m 0644 -t $(DEST)/include/libkarma/ $(INST_HEADERS)
+	../install-sh -m 0755 -d $(DESTDIR)$(PREFIX)/lib
+	../install-sh -m 0755 -d $(DESTDIR)$(PREFIX)/include
+	../install-sh -m 0755 -d $(DESTDIR)$(PREFIX)/include/libkarma
+	../install-sh -m 0644 -t $(DESTDIR)$(PREFIX)/lib/ $(LIBDIR)/$(TARGET).a
+	../install-sh -m 0755 -t $(DESTDIR)$(PREFIX)/lib/ $(LIBDIR)/$(OUT_VERSION)
+	$(RM) $(DESTDIR)$(PREFIX)/lib/$(OUT_MAJOR)
+	$(RM) $(DESTDIR)$(PREFIX)/lib/$(OUT_BASE)
+	$(LN_S) $(OUT_VERSION) $(DESTDIR)$(PREFIX)/lib/$(OUT_MAJOR)
+	$(LN_S) $(OUT_VERSION) $(DESTDIR)$(PREFIX)/lib/$(OUT_BASE)
+	../install-sh -m 0644 -t $(DESTDIR)$(PREFIX)/include/libkarma/ $(INST_HEADERS)
 
 uninstall:
 	$(RM) $(PREFIX)/lib/$(TARGET).a $(PREFIX)/lib/$(OUT_MAJOR) $(PREFIX)/lib/$(OUT_BASE) $(PREFIX)/lib/$(OUT_VERSION)
-	$(RM) -r $(DEST)/include/libkarma
+	$(RM) -r $(PREFIX)/include/libkarma
 
 OBJS = $(SRCS:.c=.o)
 
diff -ur libkarma-0.1.0.orig/tools/Makefile libkarma-0.1.0/tools/Makefile
--- libkarma-0.1.0.orig/tools/Makefile	2007-05-15 19:29:02.000000000 +0300
+++ libkarma-0.1.0/tools/Makefile	2009-05-12 13:56:31.000000000 +0300
@@ -17,8 +17,8 @@
 
 SRCS=pathedit.c
 
-CC=gcc
-CFLAGS+=-Wall -pedantic -g
+CC?=gcc
+CFLAGS+=-Wall -pedantic
 LDFLAGS+=-ltag_c -lz -L $(LIBDIR)
 
 OBJS=$(SRCS:.c=.o)
@@ -43,13 +43,13 @@
 	$(CC) $(INCLUDES) $(CFLAGS) $@.c -o $@ -lusb
 
 install:
-	../install-sh -m 0755 -d $(DEST)/bin
-	../install-sh -m 0755 -t $(DEST)/bin/ $(TOOLS)
-	../install-sh -m 0755 -d $(PREFIX)/share/man/man1
-	../install-sh -m 0755 -t $(PREFIX)/share/man/man1 $(MANPAGES)
+	../install-sh -m 0755 -d $(DESTDIR)$(PREFIX)/bin
+	../install-sh -m 0755 -t $(DESTDIR)$(PREFIX)/bin/ $(TOOLS)
+	../install-sh -m 0755 -d $(DESTDIR)$(PREFIX)/share/man/man1
+	../install-sh -m 0755 -t $(DESTDIR)$(PREFIX)/share/man/man1 $(MANPAGES)
 
 uninstall:
-	cd $(DEST)/bin && $(RM) $(TOOLS)
+	cd $(PREFIX)/bin && $(RM) $(TOOLS)
 
 .PHONY: clean
 
