diff -ur pianobar-2013.05.19.orig/Makefile pianobar-2013.05.19/Makefile
--- pianobar-2013.05.19.orig/Makefile	2013-05-19 03:58:18.000000000 -0700
+++ pianobar-2013.05.19/Makefile	2013-05-25 13:45:21.555965614 -0700
@@ -58,10 +58,12 @@
 
 LIBWAITRESS_DIR:=src/libwaitress
 LIBWAITRESS_SRC:=${LIBWAITRESS_DIR}/waitress.c
+LIBWAITRESS_TSRC:=${LIBWAITRESS_DIR}/waitress-test.c
 LIBWAITRESS_HDR:=\
 		${LIBWAITRESS_DIR}/config.h \
 		${LIBWAITRESS_DIR}/waitress.h
 LIBWAITRESS_OBJ:=${LIBWAITRESS_SRC:.c=.o}
+LIBWAITRESS_TOBJ:=${LIBWAITRESS_TSRC:.c=.o}
 LIBWAITRESS_RELOBJ:=${LIBWAITRESS_SRC:.c=.lo}
 LIBWAITRESS_INCLUDE:=${LIBWAITRESS_DIR}
 
@@ -187,8 +189,8 @@
 # -Woverlength-strings: over-portability-ish
 
 waitress-test: CFLAGS+= -DTEST
-waitress-test: ${LIBWAITRESS_OBJ}
-	${CC} ${LDFLAGS} ${LIBWAITRESS_OBJ} ${LIBGNUTLS_LDFLAGS} -o waitress-test
+waitress-test: ${LIBWAITRESS_TOBJ}
+	${CC} ${LDFLAGS} ${LIBWAITRESS_TOBJ} ${LIBGNUTLS_LDFLAGS} -o waitress-test
 
 test: waitress-test
 	./waitress-test
