diff -urN obnam-1.2.orig/obnamlib/plugins/show_plugin.py obnam-1.2/obnamlib/plugins/show_plugin.py
--- obnam-1.2.orig/obnamlib/plugins/show_plugin.py	2012-10-06 18:45:31.000000000 +0200
+++ obnam-1.2/obnamlib/plugins/show_plugin.py	2012-10-11 09:54:28.757951340 +0200
@@ -234,6 +234,7 @@
                 self.show_diff_for_file(gen2, full, '+')
         for basename in sorted(set1):
             # This was only in gen1 - it got removed
+            full = os.path.join(dirname, basename)
             self.show_diff_for_file(gen1, full, '-')
 
         for subdir in subdirs:
