diff -urN obnam-1.2.orig/obnam.1.in obnam-1.2/obnam.1.in
--- obnam-1.2.orig/obnam.1.in	2012-10-06 18:45:31.000000000 +0200
+++ obnam-1.2/obnam.1.in	2012-10-11 10:18:55.991780255 +0200
@@ -59,6 +59,14 @@
 lists every backup generation for a given client, 
 plus some metadata about the generation.
 .IP \(bu
+.B diff
+compares two generations and lists files differing between them. Every output
+line will be prefixed either by a plus sign (+) for files that were added, a
+minus sign (-) for files that have been removed or an asterisk (*) for files
+that have changed.  If only one generation ID is specified on the command line
+that generation will be compared with its direct predecessor. If two IDs have
+been specified, all changes between those two generations will be listed.
+.IP \(bu
 .B genids
 lists the identifier for every backup generation for a given client.
 No other information is shown.
