--- chrome/chrome.gyp.orig	2013-07-24 21:16:37.701455700 +0000
+++ chrome/chrome.gyp	2013-07-24 21:17:07.871649032 +0000
@@ -216,6 +216,7 @@
             '../third_party/icu/icu.gyp:icui18n',
             '../third_party/icu/icu.gyp:icuuc',
             '../third_party/leveldatabase/leveldatabase.gyp:leveldatabase',
+            '../third_party/libusb/libusb.gyp:libusb',
           ],
           'include_dirs': [
             '..',
