--- build/linux/unbundle/v8.gyp.orig	2013-07-24 18:59:21.777276631 +0000
+++ build/linux/unbundle/v8.gyp	2013-07-24 18:59:44.847427272 +0000
@@ -26,7 +26,7 @@
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 {
-  'includes': ['../../build/common.gypi'],
+  'includes': ['../../build/toolchain.gypi', '../../build/features.gypi'],
   'targets': [
     {
       'target_name': 'v8',
