--- chrome/browser/metrics/compression_utils.cc.orig	2013-07-24 21:34:25.648251116 +0000
+++ chrome/browser/metrics/compression_utils.cc	2013-07-24 21:34:40.228343391 +0000
@@ -8,7 +8,6 @@
 
 #include "base/basictypes.h"
 #include "third_party/zlib/zlib.h"
-#include "third_party/zlib/zutil.h"
 
 namespace {
 
@@ -45,7 +44,7 @@
                             Z_DEFAULT_COMPRESSION,
                             Z_DEFLATED,
                             MAX_WBITS + kWindowBitsToGetGzipHeader,
-                            DEF_MEM_LEVEL,
+                            8,
                             Z_DEFAULT_STRATEGY,
                             ZLIB_VERSION,
                             sizeof(z_stream));
