--- build/linux/unbundle/icu.gyp.orig	2013-05-08 16:16:58.989728264 +0000
+++ build/linux/unbundle/icu.gyp	2013-05-08 16:18:07.910115715 +0000
@@ -7,6 +7,11 @@
     {
       'target_name': 'system_icu',
       'type': 'none',
+      'direct_dependent_settings': {
+        'defines': [
+          'U_USING_ICU_NAMESPACE=0',
+        ],
+      },
       'link_settings': {
         'ldflags': [
           '<!@(icu-config --ldflags)',
