--- content/browser/zygote_main_linux.cc.orig	2012-04-07 19:08:22.290236311 +0200
+++ content/browser/zygote_main_linux.cc	2012-04-07 19:09:04.232233552 +0200
@@ -807,7 +807,8 @@
 
 static bool EnterSandbox() {
   PreSandboxInit();
-  SkiaFontConfigUseIPCImplementation(kMagicSandboxIPCDescriptor);
+  SkiaFontConfigSetImplementation(
+      new FontConfigIPC(kMagicSandboxIPCDescriptor));
   return true;
 }
 
