--- chrome/chrome_common.gypi.orig	2013-01-19 05:55:34.000000000 +0100
+++ chrome/chrome_common.gypi	2013-01-19 05:56:04.000000000 +0100
@@ -31,6 +31,7 @@
         '<(DEPTH)/base/base.gyp:base_i18n',
         '<(DEPTH)/base/base.gyp:base_prefs',
         '<(DEPTH)/base/base.gyp:base_static',
+        '<(DEPTH)/base/third_party/nspr/nspr.gyp:nspr',
         '<(DEPTH)/build/temp_gyp/googleurl.gyp:googleurl',
         '<(DEPTH)/chrome/chrome_resources.gyp:chrome_resources',
         '<(DEPTH)/chrome/chrome_resources.gyp:chrome_strings',
