--- build/shim_headers.gypi.orig	2013-05-01 19:43:24.624304350 +0000
+++ build/shim_headers.gypi	2013-05-01 19:43:36.374370322 +0000
@@ -13,11 +13,11 @@
     'shim_headers_path': '<(SHARED_INTERMEDIATE_DIR)/shim_headers/<(_target_name)/<(_toolset)',
     'shim_generator_additional_args%': [],
   },
-  'include_dirs+++': [
+  'include_dirs++': [
     '<(shim_headers_path)',
   ],
   'all_dependent_settings': {
-    'include_dirs++++': [
+    'include_dirs+++': [
       '<(shim_headers_path)',
     ],
   },
