--- build/linux/unbundle/libvpx.gyp.orig	2013-05-07 23:57:58.518591272 +0000
+++ build/linux/unbundle/libvpx.gyp	2013-05-07 23:58:17.748706861 +0000
@@ -12,7 +12,7 @@
         ],
       },
       'variables': {
-        'headers_root_path': '<(libvpx_source)',
+        'headers_root_path': 'source/libvpx',
         'header_filenames': [
           'vpx/vpx_codec_impl_bottom.h',
           'vpx/vpx_image.h',
