--- chrome/chrome_browser.gypi.orig	2013-11-13 18:24:41.237282707 +0000
+++ chrome/chrome_browser.gypi	2013-11-13 18:26:03.357780490 +0000
@@ -3064,7 +3064,11 @@
             'browser/password_manager/native_backend_gnome_x.cc',
             'browser/password_manager/native_backend_gnome_x.h',
           ],
-        }],
+        }, {
+          'dependencies': [
+            '../build/linux/system.gyp:gnome_keyring',
+          ],
+	}],
         ['use_aura==1', {
           'sources/': [
             ['exclude', '^browser/automation/testing_automation_provider_win.cc'],
@@ -3153,11 +3157,6 @@
                 ],
               },
             }],
-            ['use_gnome_keyring==1', {
-              'dependencies': [
-                '../build/linux/system.gyp:gnome_keyring',
-              ],
-            }],
           ],
         }],
         ['input_speech==0', {
