--- build/linux/unbundle/harfbuzz.gyp.orig	2014-05-29 12:43:32.814569782 +0000
+++ build/linux/unbundle/harfbuzz.gyp	2014-05-29 12:44:40.685923614 +0000
@@ -33,6 +33,15 @@
           '<!@(pkg-config --libs-only-l <(harfbuzz_libraries))',
         ],
       },
+      'variables': {
+        'headers_root_path': 'src',
+        'header_filenames': [
+          'hb.h',
+        ],
+      },
+      'includes': [
+        '../../build/shim_headers.gypi',
+      ],
     },
   ],
 }
