--- third_party/libyuv/libyuv.gyp.orig	2014-01-16 23:27:36.335011975 +0000
+++ third_party/libyuv/libyuv.gyp	2014-01-16 23:28:05.845202472 +0000
@@ -24,8 +24,7 @@
             'HAVE_JPEG'
           ],
           'conditions': [
-            # Android uses libjpeg for system jpeg support.
-            [ 'OS == "android" and use_system_libjpeg == 1', {
+            [ 'use_system_libjpeg == 1', {
               'dependencies': [
                  '<(DEPTH)/third_party/libjpeg/libjpeg.gyp:libjpeg',
               ],
