--- build/linux/unbundle/zlib.gyp.orig	2013-05-08 00:04:11.690834349 +0000
+++ build/linux/unbundle/zlib.gyp	2013-05-08 00:04:20.240885738 +0000
@@ -43,6 +43,25 @@
           '-lminizip',
         ],
       },
-    }
+    },
+    {
+      'target_name': 'zip',
+      'type': 'static_library',
+      'dependencies': [
+        'minizip',
+        '../../base/base.gyp:base',
+      ],
+      'include_dirs': [
+        '../..',
+      ],
+      'sources': [
+        'google/zip.cc',
+        'google/zip.h',
+        'google/zip_internal.cc',
+        'google/zip_internal.h',
+        'google/zip_reader.cc',
+        'google/zip_reader.h',
+      ],
+    },
   ],
 }
