From 2a6a29863c735fd26bf12b1b55aff4f8bf736535 Mon Sep 17 00:00:00 2001
From: Alphat <AlphatPC@gmail.com>
Date: Sun, 9 Oct 2011 08:14:12 +0800
Subject: [PATCH 2/2] iasl-20110922-parallelmake-002

---
 compiler/Makefile |   39 ++++++++++++++-------------------------
 1 files changed, 14 insertions(+), 25 deletions(-)

diff --git a/compiler/Makefile b/compiler/Makefile
index c831064..b795e16 100644
--- a/compiler/Makefile
+++ b/compiler/Makefile
@@ -67,9 +67,9 @@ HEADERS = \
 
 OBJECTS = \
 	aslcompilerlex.o \
-	aslcompilerparse.o \
+	aslcompiler.y.o \
 	dtparserlex.o \
-	dtparserparse.o \
+	dtparser.y.o \
 	adfile.o \
 	adisasm.o \
 	adwalk.o \
@@ -203,17 +203,15 @@ OBJECTS = \
 
 INTERMEDIATES = \
 	aslcompilerlex.c \
-	aslcompilerparse.c \
+	aslcompiler.y.c \
 	dtparserlex.c \
-	dtparserparse.c
+	dtparser.y.c
 
 MISC = \
-    aslcompilerparse.h \
 	aslcompiler.y.h \
-	aslcompilerparse.output \
-    dtparserparse.h \
+	aslcompiler.y.output \
 	dtparser.y.h \
-	dtparserparse.output
+	dtparser.y.output
 
 CFLAGS+= \
     -D$(HOST) \
@@ -276,42 +274,33 @@ $(PROG) : $(INTERMEDIATES) $(MISC) $(OBJECTS)
 aslcompilerlex.c :   $(ASL_COMPILER)/aslcompiler.l
 	${LEX} ${LFLAGS} -PAslCompiler -o$@ $?
 
-aslcompilerparse.c aslcompilerparse.h aslcompilerparse.output : $(ASL_COMPILER)/aslcompiler.y
-	${YACC} ${YFLAGS} -pAslCompiler -oaslcompilerparse.c $?
+aslcompiler.y.c aslcompiler.y.h aslcompiler.y.output : $(ASL_COMPILER)/aslcompiler.y
+	${YACC} ${YFLAGS} -pAslCompiler -oaslcompiler.y.c $?
 
 dtparserlex.c :      $(ASL_COMPILER)/dtparser.l
 	${LEX} ${LFLAGS} -PDtParser -o$@ $?
 
-dtparserparse.c dtparserparse.h dtparserparse.output : $(ASL_COMPILER)/dtparser.y
-	${YACC} ${YFLAGS} -pDtParser -odtparserparse.c $?
+dtparser.y.c dtparser.y.h dtparser.y.output : $(ASL_COMPILER)/dtparser.y
+	${YACC} ${YFLAGS} -pDtParser -odtparser.y.c $?
 
 # Rename headers produced by bison/yacc
 
-dtparser.y.h: dtparserparse.h
-	@echo Copy intermediate file:
-	@cp -f -v dtparserparse.h dtparser.y.h
-
-aslcompiler.y.h : aslcompilerparse.h
-	@echo Copy intermediate file:
-	@cp -f -v aslcompilerparse.h aslcompiler.y.h
-
-
 #
 # Parsers and Lexers - final object files
 #
 # Cannot use the common compile warning flags since the C files are created
 # by the utilities above and they are not necessarily ANSI C, etc.
 #
-aslcompilerlex.o :   aslcompilerlex.c aslcompilerparse.h
+aslcompilerlex.o :   aslcompilerlex.c aslcompiler.y.h
 	$(CC) -c $(CFLAGS) -Wall -Werror -Wstrict-aliasing=0 -o$@ $<
 
-aslcompilerparse.o : aslcompilerparse.c
+aslcompiler.y.o : aslcompiler.y.c
 	$(CC) -c $(CFLAGS) -Wall -Werror -Wstrict-aliasing=0 -o$@ $?
 
-dtparserlex.o :      dtparserlex.c dtparserparse.h
+dtparserlex.o :      dtparserlex.c dtparser.y.h
 	$(CC) -c $(CFLAGS) -Wall -Werror -Wstrict-aliasing=0 -o$@ $<
 
-dtparserparse.o :    dtparserparse.c
+dtparser.y.o :    dtparser.y.c
 	$(CC) -c $(CFLAGS) -Wall -Werror -Wstrict-aliasing=0 -o$@ $?
 
 
-- 
1.7.7

