 tools/acpiexec/Makefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/tools/acpiexec/Makefile b/tools/acpiexec/Makefile
index ba6b534..8435985 100644
--- a/tools/acpiexec/Makefile
+++ b/tools/acpiexec/Makefile
@@ -269,7 +269,7 @@ CWARNINGFLAGS += \
 # Rules
 #
 $(PROG) : $(OBJECTS)
-	$(CC) $(LDFLAGS) $(OBJECTS) -o $(PROG)
+	$(CC) $(OBJECTS) $(LDFLAGS) -o $(PROG)
 	$(COPYPROG)
 
 %.o : %.c $(HEADERS) $(ACPICA_HEADERS)
