--- FuzzyOcr.scansets.orig	2006-12-26 23:45:35.000000000 +0100
+++ FuzzyOcr.scansets	2006-12-26 23:45:55.000000000 +0100
@@ -7,30 +7,30 @@
 
 
 # Standard Ocrad Scanset
-scanset ocrad {
-    command = $ocrad
-    args = -s5 $input
-}
+#scanset ocrad {
+#    command = $ocrad
+#    args = -s5 $input
+#}
 
 # Inverted Ocrad scanset
-scanset ocrad-invert {
-    command = $ocrad
-    args = -s5 -i $input
-}
+#scanset ocrad-invert {
+#    command = $ocrad
+#    args = -s5 -i $input
+#}
 
 # Inverted Ocrad scanset with decolorization
-scanset ocrad-decolorize-invert {
-    preprocessors = ppmtopgm, pamthreshold, pamtopnm
-    command = $ocrad
-    args = -s5 -i $input
-}
+#scanset ocrad-decolorize-invert {
+#    preprocessors = ppmtopgm, pamthreshold, pamtopnm
+#    command = $ocrad
+#    args = -s5 -i $input
+#}
 
 # Ocrad scanset with decolorization
-scanset ocrad-decolorize {
-    preprocessors = ppmtopgm, pamthreshold, pamtopnm
-    command = $ocrad
-    args = -s5 $input
-}
+#scanset ocrad-decolorize {
+#    preprocessors = ppmtopgm, pamthreshold, pamtopnm
+#    command = $ocrad
+#    args = -s5 $input
+#}
 
 # Standard Gocr Scanset
 scanset gocr {
