diff -r -u vdrrip-0.3.0/scripts/queuehandler.sh vdrrip-0.3.0.gentoo/scripts/queuehandler.sh
--- vdrrip-0.3.0/scripts/queuehandler.sh	2004-04-22 20:54:13.000000000 +0200
+++ vdrrip-0.3.0.gentoo/scripts/queuehandler.sh	2006-10-06 08:39:20.000000000 +0200
@@ -337,6 +337,11 @@
         log_info "demuxing all vdr-files from directory $dir"
 	evecho "demuxing vdr-files"
         execute "$vdrsync $dir -o $tempdir"
+	# vdrsync 0.1.2.2 developer version creates bd.mpa
+	if [ -e "$tempdir/bd.mpa" ]
+	then
+	  mv "$tempdir/bd.mpa" "$tempdir/bd.ac3"
+	fi
       fi
       ;;
     *)
Only in vdrrip-0.3.0.gentoo/scripts: queuehandler.sh.orig
