diff -r -u vdrrip-0.3.0/scripts/queuehandler.sh vdrrip-0.3.0.gentoo/scripts/queuehandler.sh
--- vdrrip-0.3.0/scripts/queuehandler.sh	2006-10-06 19:12:01.000000000 +0200
+++ vdrrip-0.3.0.gentoo/scripts/queuehandler.sh	2006-10-06 19:12:37.000000000 +0200
@@ -311,7 +311,7 @@
     length=`"$mp" -identify -frames 0 "$dir/001.vdr" 2>/dev/null | \
             grep ID_LENGTH | cut -d"=" -f2`
   fi
-  let local ss=length/2
+  let local ss=$(echo $length | sed -e s/[.,].*//)/2
   previewval="-ss $ss -endpos $previewlength"
 }
 
Only in vdrrip-0.3.0/scripts: queuehandler.sh.orig
