diff -r -u vdrrip-0.3.0.orig/scripts/queuehandler.sh vdrrip-0.3.0/scripts/queuehandler.sh
--- vdrrip-0.3.0.orig/scripts/queuehandler.sh	2006-10-11 09:59:27.000000000 +0200
+++ vdrrip-0.3.0/scripts/queuehandler.sh	2006-10-11 10:00:48.000000000 +0200
@@ -882,7 +882,7 @@
 #
 # echo $1 on stderr and write it to the syslog with priority user.error
 #
-  logger -s -p user.error -t [vdrrip-qh] "$1"
+  logger -s -p user.error -t vdrrip-qh "$1"
 
   if [ "$2" ]
   then
@@ -898,7 +898,7 @@
 #
 # echo $1 on stdout and write it to the syslog with priority user.info
 #
-  logger -s -p user.info -t [vdrrip-qh] "$1" 2>&1
+  logger -s -p user.info -t vdrrip-qh "$1" 2>&1
 }
 
 
@@ -908,7 +908,7 @@
 #
   if [ "$debug" = "1" ]
   then
-    logger -s -p user.debug -t [vdrrip-qh] "$1"
+    logger -s -p user.debug -t vdrrip-qh "$1"
   fi
 }
 
