diff -r -u vdrrip-0.3.0/scripts/queuehandler.sh vdrrip-0.3.0.gentoo/scripts/queuehandler.sh
--- vdrrip-0.3.0/scripts/queuehandler.sh	2004-04-22 20:54:13.000000000 +0200
+++ vdrrip-0.3.0.gentoo/scripts/queuehandler.sh	2006-10-06 00:08:00.000000000 +0200
@@ -92,6 +92,7 @@
     log_error "usage: $scriptname queuefile tempdir" 1
   fi
 
+  mkdir -p "$tempdir"
   if [ ! -d "$tempdir" ]
   then
     log_error "directory $tempdir doesn't exist, aborting !" 1
@@ -316,6 +317,11 @@
   if [ "$error" ]; then return
   elif [ "$dvd" ]; then return; fi
 
+  # Recreate $tempdir if removed by vdr housekeeping
+  mkdir -p "$tempdir"
+  if [ ! -d "$tempdir" ]; then return; fi
+  cd "$tempdir"
+
   case "$container" in
     "avi")
       # join all vdr-files to $tempdir/temp.vdr
