diff -r -u vdrrip-0.3.0/scripts/queuehandler.sh vdrrip-0.3.0.gentoo/scripts/queuehandler.sh
--- vdrrip-0.3.0/scripts/queuehandler.sh	2006-10-06 18:25:15.000000000 +0200
+++ vdrrip-0.3.0.gentoo/scripts/queuehandler.sh	2006-10-06 18:54:32.000000000 +0200
@@ -481,26 +481,40 @@
     local vopopts="-vop $vopopts"
   fi
 
+  if [ "$ppvalues" -a "${ppvalues:1:2}" != "hb" ]
+     then
+     local mintopts=""
+     local xintopts=""
+  else
+     local mintopts=":ildct:ilme"
+     local xintopts=":interlacing"
+  fi
+
   # encode in two passes 
   for pass in 1 2
   do
     if [ "$pass" = "1" ]
     then
       local ofile="-o /dev/null"
+      local mturbo=":turbo"
+      local xturbo=":turbo"
     else
       local ofile="-o $tempdir/$name.avi"
+    if [ "$useropts_lavc" ]; then log_info "\$useropts_lavc are set to \"$useropts_lavc\""; fi
+    if [ "$useropts_xvid" ]; then log_info "\$useropts_xvid are set to \"$useropts_xvid\""; fi
+    if [ "$mintopts" ]; then log_info "\Encoding an interlaced stream with \"$mintopts\""; fi
+    if [ "$xintopts" ]; then log_info "\Encoding an interlaced stream with \"$xintopts\""; fi
+      local mturbo=":$useropts_lavc$mintopts"
+      local xturbo=":$useropts_xvid$xintopts"
     fi
 
     # set mencoder video values
     case "$vcodec" in
       "lavc")
-        local vopts="-ovc lavc -lavcopts vcodec=mpeg4:vhq:vbitrate=$br_video:vqmin=$min_q:vqmax=$max_q:vpass=$pass -sws 2"
+        local vopts="-ovc lavc -lavcopts vcodec=mpeg4:vhq:vbitrate=$br_video:vqmin=$min_q:vqmax=$max_q:vpass=$pass$mturbo -sws 2"
         ;;
       "xvid")
-        local vopts="-ovc xvid -xvidencopts bitrate=$br_video:me_quality=6:pass=$pass -sws 2"
-        ;;
-      "divx4")
-        local vopts="-ovc divx4 -divx4opts br=$br_video:q=5:min_quant=$min_q:max_quant=$max_q:pass=$pass -sws 2"
+        local vopts="-ovc xvid -xvidencopts bitrate=$br_video:me_quality=6:pass=$pass$xturbo -sws 2"
         ;;
       *)
         log_error "unknown video codec $vcodec"
diff -r -u vdrrip-0.3.0/scripts/queuehandler.sh.conf vdrrip-0.3.0.gentoo/scripts/queuehandler.sh.conf
--- vdrrip-0.3.0/scripts/queuehandler.sh.conf	2004-04-22 20:54:11.000000000 +0200
+++ vdrrip-0.3.0.gentoo/scripts/queuehandler.sh.conf	2006-10-06 18:58:54.000000000 +0200
@@ -69,3 +69,14 @@
 
   vdrhostname="localhost"
 
+  # These are the options beside the bitrate and eventually the 
+  # interlace options passed to mencoder you can control the 
+  # quality of the encoding by changing them here
+
+  useropts_lavc=""
+  # For a high quality high bitrate encoding use:
+  #useropts_lavc="mbd=2:trell:cbp:mv0:cmp=3:subcmp=3"
+  useropts_xvid=""
+  # For a high quality high bitrate encoding use:
+  #useropts_xvid="me_quality=6:vhq=1:notrellis:max_bframes=2"
+
