diff -r -u vdrrip-0.3.0/movie.c vdrrip-0.3.0.gentoo/movie.c
--- vdrrip-0.3.0/movie.c	2004-04-21 18:47:10.000000000 +0200
+++ vdrrip-0.3.0.gentoo/movie.c	2006-10-06 09:38:31.000000000 +0200
@@ -21,8 +21,8 @@
 
 #define SAVEFILE "save.vdrrip"
 
-#define IDENTCMD "%s \'%s\'%s -identify -frames 0 2>/dev/null | sed -e \'s/[`\\!$\"]/\\&/g\'"
-#define CROPCMD "%s \'%s\'%s -vo null -ao null -really-quiet -ss %i -frames %i -vop cropdetect 2>/dev/null | grep \"crop=\" | sed \"s/.*crop\\(.*\\)).*/\\1/\" | sort | uniq -c | sort -r"
+#define IDENTCMD "%s \'%s\'%s -identify -vo null -ao null -frames 0 2>/dev/null | sed -e \'s/[`\\!$\"]/\\&/g\'"
+#define CROPCMD "%s \'%s\'%s -vo null -ao null -quiet -ss %i -frames %i -vop cropdetect 2>/dev/null | grep \"crop=\" | sed \"s/.*crop\\(.*\\)).*/\\1/\" | sort | uniq -c | sort -r"
 #define AUDIOCMD "%s \'%s/001.vdr\' -vo null -ao null -frames 0 -aid %i 2>/dev/null | grep AUDIO"
 #define AUDIOCMDDVD "%s %s -vo null -ao null -frames 0 -aid %i 2>/dev/null | grep AUDIO"
 #define MENCCMD "%s %s help 2>/dev/null"
@@ -517,7 +517,9 @@
 
     s = strcol(strgrep("ID_VIDEO_ASPECT", p), "=", 2);
     if (s) {
-      Aspect = atof(s);
+      /* FIXME newer mplayer does no longer return the aspect of dvb streams so it fixed to 4:3 this DOESN'T of course work with feeds or movies with different aspect ratio */ 
+      Aspect = atof(s); 
+      if (Aspect==0) {Aspect = 1.3333333333;} 
     } else {Aspect = -1;}
 
     CalcAspect = Aspect;
