diff -r -u vdrrip-0.3.0.org/movie.c vdrrip-0.3.0/movie.c
--- vdrrip-0.3.0.org/movie.c	2006-11-06 00:06:29.000000000 +0100
+++ vdrrip-0.3.0/movie.c	2006-11-06 00:04:35.000000000 +0100
@@ -21,8 +21,8 @@
 
 #define SAVEFILE "save.vdrrip"
 
-#define IDENTCMD "%s \'%s\'%s -identify -frames 0 2>/dev/null | sed -e \'s/[`\\!$\"]/\\&/g\'"
-#define CROPCMD "%s \'%s\'%s -vo null -ao null -really-quiet -ss %i -frames %i -vop cropdetect 2>/dev/null | grep \"crop=\" | sed \"s/.*crop\\(.*\\)).*/\\1/\" | sort | uniq -c | sort -r"
+#define IDENTCMD "%s \'%s\'%s -identify -vo null -ao null -frames 1 2>/dev/null | sed -e \'s/[`\\!$\"]/\\&/g\'"
+#define CROPCMD "%s \'%s\'%s -vo null -ao null -quiet -ss %i -frames %i -vop cropdetect 2>/dev/null | grep \"crop=\" | sed \"s/.*crop\\(.*\\)).*/\\1/\" | sort | uniq -c | sort -r"
 #define AUDIOCMD "%s \'%s/001.vdr\' -vo null -ao null -frames 0 -aid %i 2>/dev/null | grep AUDIO"
 #define AUDIOCMDDVD "%s %s -vo null -ao null -frames 0 -aid %i 2>/dev/null | grep AUDIO"
 #define MENCCMD "%s %s help 2>/dev/null"
@@ -515,11 +515,12 @@
     } else {Fps = -1;}
     FREE(s);
 
-    s = strcol(strgrep("ID_VIDEO_ASPECT", p), "=", 2);
-    if (s) {
-      Aspect = atof(s);
-    } else {Aspect = -1;}
-
+    Aspect = -1;
+    while(s = strcol(strgrep("ID_VIDEO_ASPECT", p), "=", 2)){
+       Aspect = atof(s);
+       if (Aspect != 0) break;
+    }   
+    if (Aspect == 0) {Aspect = 1.333333;}
     CalcAspect = Aspect;
 
     pclose(p);
