diff -ur cdcd-0.6.6/configure.ac cdcd-0.6.6-fbsd/configure.ac
--- cdcd-0.6.6/configure.ac	2004-04-04 00:54:57.000000000 +0200
+++ cdcd-0.6.6-fbsd/configure.ac	2005-08-17 11:42:39.000000000 +0200
@@ -40,6 +40,11 @@
     default_device=/dev/unspecified
     have_libcdaudio=1
     ;;
+   
+  *freebsd[567]*)
+    default_device=/dev/cd0
+    have_cdaudio=1
+    ;;
 
   *bsd*)
     default_device=/dev/wcd0a
@@ -69,8 +74,8 @@
 dnl Checks for header files.
 AC_HEADER_STDC
 AC_FUNC_ALLOCA
-AC_CHECK_HEADERS([string.h strings.h sys/ptyvar.h])
-AC_CHECK_HEADERS([errno.h locale.h malloc.h stdlib.h \
+AC_CHECK_HEADERS([string.h strings.h sys/ptyvar.h malloc.h])
+AC_CHECK_HEADERS([errno.h locale.h stdlib.h \
 readline/readline.h readline/history.h sys/socket.h unistd.h],, \
 [AC_MSG_ERROR([missing header file.])])
 
