--- Makefile.in	2005-07-04 12:46:31.000000000 +0200
+++ Makefile.in.new	2005-07-04 12:55:11.000000000 +0200
@@ -1,7 +1,7 @@
 CC = @CC@
-LIBDIR = -L/usr/X11R6/lib -L@XLIB@
+LIBDIR = -L/usr/X11R6/lib
 LIBS   = -lXpm -lXext -lX11
-CFLAGS = -O2 -Wall -I.. -I@XINC@
+CFLAGS = -O2 -Wall -I..
 RLFLAG = -DVERSION=\"@MK_YAWMPPP_RELEASE@\" -D@SYSDEF@
 OBJS  = yawmppp.o wmgeneral.o
 DOTHS = yawmppp.h wmgeneral.h ../misc.h ../list.h ../isprc.h ../ycommon.h
