--- a/youtube_dl/extractor/__init__.py
+++ b/youtube_dl/extractor/__init__.py
@@ -35,9 +35,7 @@
 from .myvideo import MyVideoIE
 from .nba import NBAIE
 from .photobucket import PhotobucketIE
-from .pornotube import PornotubeIE
 from .rbmaradio import RBMARadioIE
-from .redtube import RedTubeIE
 from .soundcloud import SoundcloudIE, SoundcloudSetIE
 from .spiegel import SpiegelIE
 from .stanfordoc import StanfordOpenClassroomIE
@@ -54,13 +52,8 @@
 from .vine import VineIE
 from .wimp import WimpIE
 from .worldstarhiphop import WorldStarHipHopIE
-from .xhamster import XHamsterIE
-from .xnxx import XNXXIE
-from .xvideos import XVideosIE
 from .yahoo import YahooIE, YahooSearchIE
-from .youjizz import YouJizzIE
 from .youku import YoukuIE
-from .youporn import YouPornIE
 from .youtube import YoutubeIE, YoutubePlaylistIE, YoutubeSearchIE, YoutubeUserIE, YoutubeChannelIE
 from .zdf import ZDFIE
 
@@ -90,7 +83,6 @@
         ComedyCentralIE(),
         EscapistIE(),
         CollegeHumorIE(),
-        XVideosIE(),
         SoundcloudSetIE(),
         SoundcloudIE(),
         InfoQIE(),
@@ -98,10 +90,6 @@
         StanfordOpenClassroomIE(),
         MTVIE(),
         YoukuIE(),
-        XNXXIE(),
-        YouJizzIE(),
-        PornotubeIE(),
-        YouPornIE(),
         GooglePlusIE(),
         ArteTvIE(),
         NBAIE(),
@@ -121,13 +109,11 @@
         ZDFIE(),
         TumblrIE(),
         BandcampIE(),
-        RedTubeIE(),
         InaIE(),
         HowcastIE(),
         VineIE(),
         FlickrIE(),
         TeamcocoIE(),
-        XHamsterIE(),
         HypemIE(),
         Vbox7IE(),
         GametrailersIE(),
