--- rtaudio-4.0.11/Makefile.in
+++ rtaudio-4.0.11/Makefile.in
@@ -24,13 +24,14 @@
 tests:
 	cd tests && $(MAKE) all
 
-$(LIBRARIES): $(OBJECTS)
+$(STATIC): $(OBJECTS)
 	$(AR) ruv $(STATIC) $(OBJECTS)
-	ranlib $(STATIC)
+	$(RANLIB) $(STATIC)
+
+$(SHARED): $(OBJECTS)
 	$(CC) -fPIC @libflags@ $(OBJECTS) @LIBS@
 	$(LN) -s @sharedname@ $(SHARED)
-
-#	$(CC) -shared $(OBJECTS) -o $(SHARED) @LIBS@
+	$(LN) -s @sharedname@ $(SHARED).$(MAJOR)
 
 %.o : %.cpp
 	$(CC) $(CFLAGS) $(DEFS) -c $(<) -o $@
