diff -Nru hsfmodem-7.80.02.05x86_64full.orig/diag/makefile hsfmodem-7.80.02.05x86_64full/diag/makefile
--- hsfmodem-7.80.02.05x86_64full.orig/diag/makefile	2006-01-13 20:23:50.000000000 +0100
+++ hsfmodem-7.80.02.05x86_64full/diag/makefile	2009-11-29 10:29:03.000000000 +0100
@@ -10,7 +10,7 @@
 include $(TOP)/config.mak
 -include $(TOP)/modules/imported/makeflags.mak
 
-CFLAGS =	-I$(TOP)/modules/imported/include -I$(TOP)/modules/include
+CFLAGS +=	-I$(TOP)/modules/imported/include -I$(TOP)/modules/include
 CFLAGS +=	$(IMPORTED_FRAMEWORK_DEFS) $(IMPORTED_GENERAL_DEFS)
 CFLAGS +=	-DCNXTTARGET="\"$(CNXTTARGET)\""
 
@@ -41,13 +41,13 @@
 	$(MKDIR) -p $@
 
 $(CNXTTARGET)diag: diag.o modemexpert.o
-	$(CC) -o $@ $^
+	$(CC) ${LDFLAGS} -o $@ $^
 
 $(CNXTTARGET)scr: scr.o
-	$(CC) -o $@ $^
+	$(CC) ${CFLAGS} ${LDFLAGS} -o $@ $^
 
 $(CNXTTARGET)dmp: dmp.o
-	$(CC) -o $@ $^
+	$(CC) ${CFLAGS} ${LDFLAGS} -o $@ $^
 
 clean:
 	rm -f *.o $(ALL)
diff -Nru hsfmodem-7.80.02.05x86_64full.orig/modules/Makefile hsfmodem-7.80.02.05x86_64full/modules/Makefile
--- hsfmodem-7.80.02.05x86_64full.orig/modules/Makefile	2009-10-22 00:35:59.000000000 +0200
+++ hsfmodem-7.80.02.05x86_64full/modules/Makefile	2009-11-29 10:29:03.000000000 +0100
@@ -158,7 +158,7 @@
 ifeq ($(ON_BUILD_SYSTEM)-$(CONFIG_SND_HDA_INTEL),no-)
 # On Ubuntu Gutsy (7.10), the HDA modules are not included in linux-image but in a seprarate package
 # and CONFIG_SND_HDA_INTEL is not enabled in the kernel .config file
-CONFIG_SND_HDA_INTEL=$(shell modprobe -n snd-hda-intel > /dev/null 2>&1 && echo y || echo n)
+# CONFIG_SND_HDA_INTEL=$(shell modprobe -n snd-hda-intel > /dev/null 2>&1 && echo y || echo n)
 endif
 
 MODULAR_HDA := $(shell test -e  ${REAL_KERNELSRC}/include/sound/hda_codec.h && echo yes || echo no)
diff -Nru hsfmodem-7.80.02.05x86_64full.orig/modules/Makefile.orig hsfmodem-7.80.02.05x86_64full/modules/Makefile.orig
diff -Nru hsfmodem-7.80.02.05x86_64full.orig/scripts/cnxtconfig.in hsfmodem-7.80.02.05x86_64full/scripts/cnxtconfig.in
--- hsfmodem-7.80.02.05x86_64full.orig/scripts/cnxtconfig.in	2009-10-22 00:35:59.000000000 +0200
+++ hsfmodem-7.80.02.05x86_64full/scripts/cnxtconfig.in	2009-11-29 10:29:03.000000000 +0100
@@ -2013,9 +2013,9 @@
 	if [ -d /etc/udev/rules.d ]; then
 		rm -f /etc/udev/rules.d/00-@CNXTTARGET@.rules
 		if [ "@CNXTDRIVER@" = "dgc" ]; then
-			echo 'KERNEL=="ttyACM0", SYMLINK="modem"' > /etc/udev/rules.d/00-@CNXTTARGET@.rules
+			echo 'KERNEL=="ttyACM0", SYMLINK="modem", GROUP="dialout"' > /etc/udev/rules.d/99-@CNXTTARGET@.rules
 		else
-		echo 'KERNEL=="ttyS@CNXTSERDEV@0", SYMLINK="modem"' > /etc/udev/rules.d/00-@CNXTTARGET@.rules
+		echo 'KERNEL=="ttyS@CNXTSERDEV@0", SYMLINK="modem", GROUP="dialout"' > /etc/udev/rules.d/99-@CNXTTARGET@.rules
 	fi
 	fi
 
diff -Nru hsfmodem-7.80.02.05x86_64full.orig/scripts/rccnxt.in hsfmodem-7.80.02.05x86_64full/scripts/rccnxt.in
--- hsfmodem-7.80.02.05x86_64full.orig/scripts/rccnxt.in	2004-11-22 22:59:29.000000000 +0100
+++ hsfmodem-7.80.02.05x86_64full/scripts/rccnxt.in	2009-11-29 10:29:03.000000000 +0100
@@ -1,63 +1,12 @@
-#
-# chkconfig: 2345 09 91
-# description: @CNXTDRVDSC@
-#
-### BEGIN INIT INFO
-# Provides: @CNXTTARGET@
-# Should-Start: hotplug coldplug
-# Required-Start: $local_fs
-# Required-Stop: $network
-# Default-Start: 3 5
-# Default-Stop: 0 1 2 6
-# Description: Start @CNXTDRVDSC@
-### END INIT INFO
 
-if [ -e /etc/runlevels ]; then
-	begincmd="ebegin"
-	endcmd="eend"
-else
-	begincmd="echo"
-	endcmd="/bin/true"
-fi
-
-
-start()
-{
-	${begincmd} "Starting @CNXTDRVDSC@"
+start() {
+	ebegin "Starting @CNXTDRVDSC@"
 	@CNXTSBINDIR@/@CNXTTARGET@config --rcstart
-	${endcmd} 0
+	eend 0
 }
 
-stop()
-{
-	${begincmd} "Stopping @CNXTDRVDSC@"
+stop() {
+	ebegin "Stopping @CNXTDRVDSC@"
 	@CNXTSBINDIR@/@CNXTTARGET@config --rcstop
-	${endcmd} 0
+	eend 0
 }
-
-restart()
-{
-	stop
-	start
-}
-
-if [ ! -e /etc/runlevels ]; then
-	case "$1" in
-      start)
-			start
-			;;
-      stop)
-			stop
-			;;
-      restart|force-reload|reload)
-			restart
-			;;
-      status)
-			exit 3
-			;;
-      *)
-			echo "Usage: $0 {start|stop|status|try-restart|restart|force-reload|reload}"
-			exit 1
-			;;
-	esac
-fi
