diff --git a/Makefile.in b/Makefile.in
index 4ac046d..c757dfa 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -53,8 +53,8 @@ CXXFLAGS = @MYCXXFLAGS@
 LDFLAGS = @MYLDFLAGS@
 CMDLDFLAGS = @MYCMDLDFLAGS@
 LIBS = @LIBS@
-LDENV = LD_RUN_PATH=/lib:/usr/lib:$(LIBDIR):$(HOME)/lib:/usr/local/lib:@MYRUNPATH@
-RUNENV = @MYLDLIBPATHENV@=.:/lib:/usr/lib:$(LIBDIR):$(HOME)/lib:/usr/local/lib:@MYRUNPATH@
+LDENV = 
+RUNENV = @MYLDLIBPATHENV@=.:$(LIBDIR)
 POSTCMD = @MYPOSTCMD@
 
 
diff --git a/configure.in b/configure.in
index ad63005..f0448d6 100644
--- a/configure.in
+++ b/configure.in
@@ -34,27 +34,17 @@ MYDOCUMENTFILES="COPYING ChangeLog doc kyotocabinet.idl"
 MYPCFILES="kyotocabinet.pc"
 
 # Building flags
-MYCFLAGS="-Wall -ansi -pedantic -fPIC -fsigned-char -O2"
+MYCFLAGS="-Wall -ansi -pedantic -fPIC -fsigned-char"
 MYCXXFLAGS="-Wall -fPIC -fsigned-char -O2"
-MYCPPFLAGS="-I. -I\$(INCLUDEDIR) -I$HOME/include -I/usr/local/include"
+MYCPPFLAGS="-I. -I\$(INCLUDEDIR)"
 MYCPPFLAGS="$MYCPPFLAGS -DNDEBUG -D_GNU_SOURCE=1"
 MYCPPFLAGS="$MYCPPFLAGS -D_FILE_OFFSET_BITS=64 -D_REENTRANT -D__EXTENSIONS__"
-MYLDFLAGS="-L. -L\$(LIBDIR) -L$HOME/lib -L/usr/local/lib"
+MYLDFLAGS="-L. -L\$(LIBDIR)"
 MYCMDLDFLAGS=""
 MYRUNPATH="\$(LIBDIR)"
 MYLDLIBPATHENV="LD_LIBRARY_PATH"
 MYPOSTCMD="true"
 
-# Building paths
-PATH="$PATH:$HOME/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/bin:/sbin"
-PATH="$PATH:/opt/SUNWspro/bin:/usr/ccs/bin:/usr/xpg4/bin:/usr/xpg6/bin:/usr/ucb"
-CPATH="$HOME/include:/usr/local/include:$CPATH"
-LIBRARY_PATH="$HOME/lib:/usr/local/lib:$LIBRARY_PATH"
-LD_LIBRARY_PATH="$HOME/lib:/usr/local/lib:$LD_LIBRARY_PATH"
-PKG_CONFIG_PATH="$HOME/lib/pkgconfig:/usr/local/lib/pkgconfig:$PKG_CONFIG_PATH"
-export PATH CPATH LIBRARY_PATH LD_LIBRARY_PATH PKG_CONFIG_PATH
-
-
 
 #================================================================
 # Options
diff --git a/kyotocabinet.pc.in b/kyotocabinet.pc.in
index f21c87b..718f2d8 100644
--- a/kyotocabinet.pc.in
+++ b/kyotocabinet.pc.in
@@ -10,5 +10,5 @@ datadir=@datadir@
 Name: Kyoto Cabinet
 Description: another modern implementation of DBM
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -lkyotocabinet @LIBS@
+Libs: -L${libdir} -lkyotocabinet
 Cflags: -I${includedir}
