diff --git a/Makefile.in b/Makefile.in
index 59e71a0..bd42c67 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -123,7 +123,6 @@ install :
 	cd man && cp -Rf $(MAN1FILES) $(DESTDIR)$(MAN1DIR)
 	mkdir -p $(DESTDIR)$(PCDIR)
 	cp -Rf $(PCFILES) $(DESTDIR)$(PCDIR)
-	-[ "$$UID" = 0 ] && PATH=/sbin:/usr/sbin:$(PATH) ldconfig || true
 	@printf '\n'
 	@printf '#================================================================\n'
 	@printf '# Thanks for using Kyoto Cabinet.\n'
