diff --git a/Makefile.in b/Makefile.in
index a813215..eefa4eb 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -117,8 +117,6 @@ install :
 	cp -Rf $(LIBRARYFILES) $(DESTDIR)$(LIBDIR)
 	mkdir -p $(DESTDIR)$(BINDIR)
 	cp -Rf $(COMMANDFILES) $(DESTDIR)$(BINDIR)
-	mkdir -p $(DESTDIR)$(DATADIR)
-	cp -Rf $(DOCUMENTFILES) $(DESTDIR)$(DATADIR)
 	mkdir -p $(DESTDIR)$(MAN1DIR)
 	cd man && cp -Rf $(MAN1FILES) $(DESTDIR)$(MAN1DIR)
 	mkdir -p $(DESTDIR)$(PCDIR)
