https://github.com/taglib/taglib/commit/7b80418122a42295ada34863609d82152baab03c

--- taglib-1.8/taglib/toolkit/taglib.h
+++ taglib-1.8/taglib/toolkit/taglib.h
@@ -27,7 +27,7 @@
 #define TAGLIB_H
 
 #define TAGLIB_MAJOR_VERSION 1
-#define TAGLIB_MINOR_VERSION 7
+#define TAGLIB_MINOR_VERSION 8
 #define TAGLIB_PATCH_VERSION 0
 
 #if defined(__GNUC__) && (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ > 1))
