--- GUI/xephem/tools/xephemdbd/request.c	2006-11-29 14:43:38.000000000 +0100
+++ GUI/xephem/tools/xephemdbd/request.c.fixed	2006-04-10 11:01:17.000000000 +0200
@@ -155,7 +155,7 @@
 {
 	char str1[32], str2[32];
 
-	fs_date (str1, mjd_day(mjd));
+	fs_date (str1, pref_get(PREF_DATE_FORMAT), mjd_day(mjd));
 	fs_sexa (str2, mjd_hr(mjd), 2, 3600);
 	printf ("UTC Time:  %s  %s  JD %13.5f\n", str1, str2, mjd+MJD0);
 
--- GUI/xephem/Makefile	2006-11-29 16:08:02.000000000 +0100
+++ GUI.new/xephem/Makefile	2006-11-29 16:08:56.000000000 +0100
@@ -15,7 +15,7 @@
 LIBLIB = -lastro -lip -llilxml -ljpegd -lpng -lz
 
 # MOTIFI is the directory containing the Xm directory of include files.
-# MOTIFL is the directory containing the libXm.a library.
+# MOTIFL is the directory containing libXm
 # MOTIF can be used to set one common directory if desired.
 # Defaults work ok for Linux and Mac OS X.
 ifdef MOTIF
@@ -34,7 +34,7 @@
 CLDFLAGS = -g
 CFLAGS=  $(LIBINC)     -Wall -I$(MOTIFI) -I/usr/X11R6/include -march=athlon-xp -O2 -pipe
 LDFLAGS = $(LIBLNK) $(CLDFLAGS) -L$(MOTIFL) -L/usr/X11R6/lib
-XLIBS = $(MOTIFL)/libXm.a -lXp -lXt -lXext -lXmu -lX11
+XLIBS = -lXm -lXp -lXt -lXext -lXmu -lX11
 LIBS = $(XLIBS) $(LIBLIB) -lm
 
 # for ppc Apple OS X to make universal (i386 and ppc binary), requires
