--- GUI/xephem/Makefile.orig	2006-05-13 19:06:06.000000000 -0400
+++ GUI/xephem/Makefile	2008-06-30 06:38:50.000000000 -0400
@@ -10,9 +10,9 @@
 # back here and compile and link all the *.c files into one executable, xephem.
 
 # -I and -L flags to find required supporting XEphem libraries
-LIBINC = -I../../libastro -I../../libip -I../../liblilxml -I../../libjpegd -I../../libpng -I../../libz
-LIBLNK = -L../../libastro -L../../libip -L../../liblilxml -L../../libjpegd -L../../libpng -L../../libz
-LIBLIB = -lastro -lip -llilxml -ljpegd -lpng -lz
+LIBINC = -I../../libastro -I../../libip -I../../liblilxml 
+LIBLNK = -L../../libastro -L../../libip -L../../liblilxml
+LIBLIB = -lastro -lip -llilxml -ljpeg -lpng -lz
 
 # MOTIFI is the directory containing the Xm directory of include files.
 # MOTIFL is the directory containing the libXm.a library.
@@ -34,7 +34,7 @@
 CLDFLAGS = -g
 CFLAGS = $(LIBINC) $(CLDFLAGS) -O2 -Wall -I$(MOTIFI) -I/usr/X11R6/include
 LDFLAGS = $(LIBLNK) $(CLDFLAGS) -L$(MOTIFL) -L/usr/X11R6/lib
-XLIBS = $(MOTIFL)/libXm.a -lXp -lXt -lXext -lXmu -lX11
+XLIBS = -lXm -lXp -lXt -lXext -lXmu -lX11
 LIBS = $(XLIBS) $(LIBLIB) -lm
 
 # for ppc Apple OS X to make universal (i386 and ppc binary), requires
@@ -187,10 +187,7 @@
 libs:
 	cd ../../libastro; make
 	cd ../../libip; make
-	cd ../../libjpegd; make
 	cd ../../liblilxml; make
-	cd ../../libpng; make
-	cd ../../libz; make
 
 clean:
 	touch x.o
