diff -Nur systemtap-2.0.orig/configure.ac systemtap-2.0/configure.ac
--- systemtap-2.0.orig/configure.ac	2012-11-04 22:04:07.726950494 +0000
+++ systemtap-2.0/configure.ac	2012-11-04 22:11:08.747068941 +0000
@@ -30,7 +30,7 @@
 AC_SUBST(CXXFLAGS)
 AC_SYS_LARGEFILE
 AM_GNU_GETTEXT(external)
-AM_GNU_GETTEXT_VERSION(0.17)
+AM_GNU_GETTEXT_VERSION(0.18)
 AC_CHECK_FUNCS(ppoll)
 
 dnl Handle the prologues option.
