diff -ur Djblets-0.7.15.orig/setup.py Djblets-0.7.15/setup.py
--- setup.py	2013-05-26 15:54:39.000000000 +0800
+++ setup.py	2013-06-15 19:00:25.729083436 +0800
@@ -94,7 +94,7 @@
       test_suite="dummy",
       license="MIT",
       description="A collection of useful classes and functions for Django",
-      packages=find_packages(),
+      packages=find_packages(exclude=['tests']),
       cmdclass=cmdclasses,
       install_requires=[
           'Django>=1.4.5,<1.5',
diff -ur Djblets-0.7.15.orig/Djblets.egg-info/top_level.txt Djblets-0.7.15/Djblets.egg-info/top_level.txt
--- Djblets.egg-info/top_level.txt	2013-05-26 15:54:43.000000000 +0800
+++ Djblets.egg-info/top_level.txt	2013-06-15 19:09:32.729083587 +0800
@@ -1,2 +1 @@
 djblets
-tests
