diff -Nur a/docs/conf/inspircd.conf.example b/docs/conf/inspircd.conf.example
--- a/docs/conf/inspircd.conf.example	2012-10-07 00:29:07.000000000 +0200
+++ b/docs/conf/inspircd.conf.example	2012-10-11 20:07:40.289924911 +0200
@@ -412,11 +412,11 @@
 
 # This file has all the information about oper classes, types and o:lines.
 # You *MUST* edit it.
-<include file="conf/opers.conf.example">
+<include file="/etc/inspircd/opers.conf.example">
 
 # This file has all the information about server links and ulined servers.
 # You *MUST* edit it if you intend to link servers.
-<include file="conf/links.conf.example">
+<include file="/etc/inspircd/links.conf.example">
 
 #-#-#-#-#-#-#-#-#-#-  MISCELLANEOUS CONFIGURATION  -#-#-#-#-#-#-#-#-#-#
 #                                                                     #
@@ -426,7 +426,8 @@
 #   motd - displayed on connect and when a user executes /MOTD
 #   rules - displayed when the user executes /RULES
 # Modules can also define their own files
-<files motd="conf/inspircd.motd.example" rules="conf/inspircd.rules.example">
+<files motd="/etc/inspircd/inspircd.motd.example"
+       rules="/etc/inspircd/inspircd.rules.example">
 
 # Example of an executable file include. Note this will be read on rehash,
 # not when the command is run.
@@ -472,7 +473,7 @@
 # the default of 'inspircd.pid' is used.                              #
 #                                                                     #
 
-#<pid file="/path/to/inspircd.pid">
+<pid file="/var/run/inspircd/inspircd.pid">
 
 #-#-#-#-#-#-#-#-#-#-#-#-#- BANLIST LIMITS #-#-#-#-#-#-#-#-#-#-#-#-#-#-#
 #                                                                     #
@@ -950,7 +951,7 @@
 # provide almost all the features of InspIRCd. :)                     #
 #                                                                     #
 # The default does nothing -- we include it for simplicity for you.   #
-<include file="conf/modules.conf.example">
+<include file="/etc/inspircd/modules.conf.example">
 
 # Here are some pre-built modules.conf files that closely match the
 # default configurations of some popular IRCd's. You still may want to
@@ -962,10 +963,10 @@
 # recommended that you make your own modules file based on modules.conf.example.
 
 # Settings similar to UnrealIRCd defaults.
-#<include file="conf/modules/modules.conf.unreal">
+#<include file="/etc/inspircd/modules/modules.conf.unreal">
 
 # Settings similar to Charybdis IRCd defaults.
-#<include file="conf/modules/modules.conf.charybdis">
+#<include file="/etc/inspircd/modules/modules.conf.charybdis">
 
 
 #########################################################################
diff -Nur a/make/template/main.mk b/make/template/main.mk
--- a/make/template/main.mk	2012-10-07 00:29:07.000000000 +0200
+++ b/make/template/main.mk	2012-10-11 22:20:37.139880112 +0200
@@ -34,7 +34,7 @@
 SYSTEM = @SYSTEM@
 BUILDPATH = @BUILD_DIR@
 SOCKETENGINE = @SOCKETENGINE@
-CXXFLAGS = -pipe -fPIC -DPIC
+MYCXXFLAGS = -fPIC -DPIC
 LDLIBS = -pthread -lstdc++
 LDFLAGS = 
 CORELDFLAGS = -rdynamic -L. $(LDFLAGS)
@@ -50,9 +50,9 @@
 INSTMODE_LIB = 0644
 
 @IFEQ $(CC) icc
-  CXXFLAGS += -Wshadow
+  MYCXXFLAGS += -Wshadow
 @ELSE
-  CXXFLAGS += -pedantic -Woverloaded-virtual -Wshadow -Wformat=2 -Wmissing-format-attribute -Wall
+  MYCXXFLAGS += -pedantic -Woverloaded-virtual -Wshadow -Wformat=2 -Wmissing-format-attribute
 @ENDIF
 
 
@@ -72,38 +72,38 @@
   LDLIBS += -lsocket -lnsl -lrt -lresolv
 @ENDIF
 @IFEQ $(SYSTEM) darwin
-  CXXFLAGS += -DDARWIN -frtti
+  MYCXXFLAGS += -DDARWIN -frtti
   LDLIBS += -ldl
   CORELDFLAGS = -dynamic -bind_at_load -L. $(LDFLAGS)
   PICLDFLAGS = -fPIC -shared -twolevel_namespace -undefined dynamic_lookup $(LDFLAGS)
 @ENDIF
 @IFEQ $(SYSTEM) interix
-  CXXFLAGS += -D_ALL_SOURCE -I/usr/local/include
+  MYCXXFLAGS += -D_ALL_SOURCE -I/usr/local/include
 @ENDIF
 
-@IFNDEF D
-  D=0
+@IFNDEF DEBUGLEVEL
+  DEBUGLEVEL=0
 @ENDIF
 
 DBGOK=0
-@IFEQ $(D) 0
-  CXXFLAGS += -O2 -g1
+@IFEQ $(DEBUGLEVEL) 0
+  MYCXXFLAGS += -O2 -g1
   HEADER = std-header
   DBGOK=1
 @ENDIF
-@IFEQ $(D) 1
-  CXXFLAGS += -O0 -g3 -Werror
+@IFEQ $(DEBUGLEVEL) 1
+  MYCXXFLAGS += -O0 -g3 -Werror
   HEADER = debug-header
   DBGOK=1
 @ENDIF
-@IFEQ $(D) 2
-  CXXFLAGS += -O2 -g3
+@IFEQ $(DEBUGLEVEL) 2
+  MYCXXFLAGS += -O2 -g3
   HEADER = debug-header
   DBGOK=1
 @ENDIF
 FOOTER = finishmessage
 
-CXXFLAGS += -Iinclude
+MYCXXFLAGS += -Iinclude
 
 @GNU_ONLY MAKEFLAGS += --no-print-directory
 
@@ -122,10 +122,12 @@
 @ENDIF
 
 @IFDEF PURE_STATIC
-  CXXFLAGS += -DPURE_STATIC
+  MYCXXFLAGS += -DPURE_STATIC
 @ENDIF
 
-@DO_EXPORT RUNCC RUNLD CXXFLAGS LDLIBS PICLDFLAGS VERBOSE SOCKETENGINE CORELDFLAGS
+MYCXXFLAGS += $(CXXFLAGS)
+
+@DO_EXPORT RUNCC RUNLD MYCXXFLAGS LDLIBS PICLDFLAGS VERBOSE SOCKETENGINE CORELDFLAGS
 @DO_EXPORT SOURCEPATH BUILDPATH PURE_STATIC SPLIT_CC
 
 # Default target
@@ -138,10 +140,10 @@
     @GNU_ONLY TARGET = modules/$(M:.so=).so
 @ENDIF
 
-@IFDEF T
+@IFDEF TGT
     HEADER =
     FOOTER = target
-    TARGET = $(T)
+    TARGET = $(TGT)
 @ENDIF
 
 @IFEQ $(DBGOK) 0
@@ -216,8 +218,8 @@
 	@-install -d -o $(INSTUID) -m $(INSTMODE_DIR) $(BASE)/data
 	@-install -d -o $(INSTUID) -m $(INSTMODE_DIR) $(BASE)/logs
 	@-install -d -m $(INSTMODE_DIR) $(BINPATH)
-	@-install -d -m $(INSTMODE_DIR) $(CONPATH)/examples/aliases
-	@-install -d -m $(INSTMODE_DIR) $(CONPATH)/examples/modules
+	@-install -d -m $(INSTMODE_DIR) $(CONPATH)/aliases
+	@-install -d -m $(INSTMODE_DIR) $(CONPATH)/modules
 	@-install -d -m $(INSTMODE_DIR) $(MODPATH)
 	[ $(BUILDPATH)/bin/ -ef $(BINPATH) ] || install -m $(INSTMODE_BIN) $(BUILDPATH)/bin/inspircd $(BINPATH)
 @IFNDEF PURE_STATIC
@@ -225,9 +227,9 @@
 @ENDIF
 	-install -m $(INSTMODE_BIN) @STARTSCRIPT@ $(BASE) 2>/dev/null
 	-install -m $(INSTMODE_LIB) tools/gdbargs $(BASE)/.gdbargs 2>/dev/null
-	-install -m $(INSTMODE_LIB) docs/conf/*.example $(CONPATH)/examples
-	-install -m $(INSTMODE_LIB) docs/conf/aliases/*.example $(CONPATH)/examples/aliases
-	-install -m $(INSTMODE_LIB) docs/conf/modules/*.conf.* $(CONPATH)/examples/modules
+	-install -m $(INSTMODE_LIB) docs/conf/*.example $(CONPATH)
+	-install -m $(INSTMODE_LIB) docs/conf/aliases/*.example $(CONPATH)/aliases
+	-install -m $(INSTMODE_LIB) docs/conf/modules/*.conf.* $(CONPATH)/modules
 	@echo ""
 	@echo "*************************************"
 	@echo "*        INSTALL COMPLETE!          *"
@@ -240,7 +242,7 @@
 	@echo '  Data:' $(DATPATH)
 	@echo 'To start the ircd, run:' $(BASE)/inspircd start
 	@echo 'Remember to create your config file:' $(CONPATH)/inspircd.conf
-	@echo 'Examples are available at:' $(CONPATH)/examples/
+	@echo 'Examples are available at:' $(CONPATH)
 
 @GNU_ONLY RCS_FILES = $(wildcard .git/index src/version.sh)
 @BSD_ONLY RCS_FILES = src/version.sh
diff -Nur a/make/unit-cc.pl b/make/unit-cc.pl
--- a/make/unit-cc.pl	2012-10-07 00:29:07.000000000 +0200
+++ b/make/unit-cc.pl	2012-10-11 22:18:58.459880658 +0200
@@ -97,7 +97,7 @@
 	my $libs = '';
 	my $binary = $ENV{RUNCC};
 	if ($do_compile) {
-		$flags = $ENV{CXXFLAGS};
+		$flags = $ENV{MYCXXFLAGS};
 		$flags =~ s/ -pedantic// if nopedantic($file);
 		$flags .= ' ' . getcompilerflags($file);
 
