diff --git a/include/vigra/noise_normalization.hxx b/include/vigra/noise_normalization.hxx
index 8758cde..344a601 100644
--- a/include/vigra/noise_normalization.hxx
+++ b/include/vigra/noise_normalization.hxx
@@ -583,6 +583,9 @@ void noiseVarianceListMedianCut(Vector1 const & noise, Vector2 & clusters,
         for(unsigned int k=0; k < clusters.size(); ++k)
         {
             int k1 = clusters[k][0], k2 = clusters[k][1]-1;
+            
+#if 0       // turned the "internal error" in a postcondition message
+            // for the most likely case
             std::string message("noiseVarianceListMedianCut(): internal error (");
             message += std::string("k: ") + asString(k) + ", ";
             message += std::string("k1: ") + asString(k1) + ", ";
@@ -590,6 +593,11 @@ void noiseVarianceListMedianCut(Vector1 const & noise, Vector2 & clusters,
             message += std::string("noise.size(): ") + asString(noise.size()) + ", ";
             message += std::string("clusters.size(): ") + asString(clusters.size()) + ").";
             vigra_invariant(k1 >= 0 && k1 < (int)noise.size() && k2 >= 0 && k2 < (int)noise.size(), message.c_str());
+#endif
+            
+            vigra_postcondition(k1 >= 0 && k1 < (int)noise.size() && 
+                                k2 >= 0 && k2 < (int)noise.size(), 
+                "noiseVarianceClustering(): Unable to find homogeneous regions.");
 
             double diff = noise[k2][0] - noise[k1][0];
             if(diff > diffMax)
