diff -urN vigra-1.7.1.old/include/vigra/random_forest/rf_visitors.hxx vigra-1.7.1/include/vigra/random_forest/rf_visitors.hxx
--- vigra-1.7.1.old/include/vigra/random_forest/rf_visitors.hxx	2011-03-31 19:32:42.000000000 +0200
+++ vigra-1.7.1/include/vigra/random_forest/rf_visitors.hxx	2011-03-31 20:15:28.000000000 +0200
@@ -860,6 +860,7 @@
     ArrayVector< int>           indices; 
     OOB_Error() : VisitorBase(), oob_breiman(0.0) {}
 
+#ifdef HasHDF5
     void save(std::string filen, std::string pathn)
     {
         if(*(pathn.end()-1) != '/')
@@ -869,6 +870,7 @@
         temp[0] = oob_breiman;
         writeHDF5(filename, (pathn + "breiman_error").c_str(), temp);
     }
+#endif // HasHDF5
     // negative value if sample was ib, number indicates how often.
     //  value >=0  if sample was oob, 0 means fail 1, corrrect
 
@@ -1056,6 +1058,7 @@
 
 	/** save to HDF5 file
 	 */
+#ifdef HasHDF5
     void save(std::string filen, std::string pathn)
     {
         if(*(pathn.end()-1) != '/')
@@ -1074,6 +1077,7 @@
         temp[0] = oob_per_tree2;
         writeHDF5(filename, (pathn + "ulli_error").c_str(), temp);
     }
+#endif // HasHDF5
     // negative value if sample was ib, number indicates how often.
     //  value >=0  if sample was oob, 0 means fail 1, corrrect
 
