diff -urN vigra-1.7.1.old/vigranumpy/docsrc/CMakeLists.txt vigra-1.7.1/vigranumpy/docsrc/CMakeLists.txt
--- vigra-1.7.1.old/vigranumpy/docsrc/CMakeLists.txt	2011-09-04 17:13:53.362610527 +0200
+++ vigra-1.7.1/vigranumpy/docsrc/CMakeLists.txt	2011-09-04 17:18:03.282607530 +0200
@@ -1,4 +1,4 @@
-IF(PYTHON_SPHINX)
+IF(PYTHON_SPHINX AND ENABLE_DOC)
 	# vigranumpy documentation generation using sphinx
 	IF(DEFINED VIGRANUMPY_DOCDIR)
 		MESSAGE(SEND_ERROR "VIGRANUMPY_DOCDIR already defined! Please delete from Cache.")
@@ -33,7 +33,7 @@
 			${VIGRANUMPY_DOCDIR}
 		COMMENT "Generating vigranumpy documentation")
 
-ELSE(PYTHON_SPHINX)
+ELSE(PYTHON_SPHINX AND ENABLE_DOC)
 	# no vigranumpy documentation if sphinx not available
 	ADD_CUSTOM_TARGET (doc_python 
 		${CMAKE_COMMAND} -E echo
@@ -41,7 +41,7 @@
 		"(sphinx-build not found)"
         VERBATIM)
 
-ENDIF(PYTHON_SPHINX)
+ENDIF(PYTHON_SPHINX AND ENABLE_DOC)
 
 ADD_DEPENDENCIES(doc_python vigranumpy)
 ADD_DEPENDENCIES(doc_python doc_cpp)
