--- fluxbox-1.3.6/util/fluxbox-generate_menu.in	2015-01-15 13:35:50.223367998 -0800
+++ fluxbox-1.3.6-copy/util/fluxbox-generate_menu.in	2015-01-15 13:38:21.789755711 -0800
@@ -331,7 +331,7 @@
     if [ -f "$entry_icon" ]; then
     # if icon exists and entry does not already exists, add it
         if ! grep -q -m 1 "^.${entry_exec}.[[:space:]]*<.*/${icon_base}\....>" $ICONMAPPING 2> /dev/null; then
-            printf "\"${entry_exec}\" \t <${entry_icon}>\n" >> $ICONMAPPING
+            printf "\"${execname}\" \t <${entry_icon}>\n" >> $ICONMAPPING
         else 
             : echo "#    mapping already exists for ${entry_exec}" >> $ICONMAPPING
         fi

