--- python/Makefile.am	2012-06-12 14:12:50.000000000 +0400
+++ python/Makefile.am	2012-09-17 16:21:55.230278577 +0400
@@ -35,7 +35,7 @@
 libhivexmod_la_SOURCES = hivex-py.c
 libhivexmod_la_CFLAGS = -Wall -I$(PYTHON_INCLUDEDIR) \
 		        -I$(top_srcdir)/lib -I$(top_builddir)/lib
-libhivexmod_la_LIBADD = $(top_builddir)/lib/libhivex.la
+libhivexmod_la_LIBADD = $(top_builddir)/lib/libhivex.la -lpython$(PYTHON_VERSION)
 libhivexmod_la_LDFLAGS = -avoid-version -shared
 
 TESTS_ENVIRONMENT = \
