--- hoe-2.6.0/lib/hoe/rubyforge.rb.account	2010-04-28 15:34:32.000000000 +0900
+++ hoe-2.6.0/lib/hoe/rubyforge.rb	2010-04-28 16:11:18.000000000 +0900
@@ -40,12 +40,20 @@
     end
 
     if Hoe.plugins.include? :publish then
+     begin
       path   = File.expand_path("~/.rubyforge/user-config.yml")
       config = YAML.load(File.read(path))
       base   = "/var/www/gforge-projects"
       dir    = "#{base}/#{rubyforge_name}/#{remote_rdoc_dir}"
 
       rdoc_locations << "#{config["username"]}@rubyforge.org:#{dir}"
+
+     rescue Errno::ENOENT => err
+      hoe_ver = Hoe::VERSION
+      msg = err.message
+      # puts "Hoe #{hoe_ver}: warning: #{msg}"
+     end
+
     end
   end
 end
