From: Julian Ospald <hasufell@gentoo.org>
Date: Tue Jan  8 16:31:27 UTC 2013
Subject: build system

	respect CFLAGS
	provide werror switch

--- gnurobots-1.2.0/configure.ac
+++ gnurobots-1.2.0/configure.ac
@@ -21,7 +21,7 @@
 AC_PREREQ([2.59])
 AM_CONFIG_HEADER([config.h])
 AC_CONFIG_SRCDIR([include/api.h])
-AM_INIT_AUTOMAKE
+AM_INIT_AUTOMAKE([-Wall])
 
 dnl Checks for programs.
 AC_PROG_CC
@@ -91,7 +91,13 @@
 dnl Checks for typedefs, structures, and compiler characteristics.
 AC_C_CONST
 
-CFLAGS="-pedantic-errors -Werror -Wall -g"
+AC_ARG_ENABLE([werror],
+    [AS_HELP_STRING([--enable-werror],
+        [enable -Werror flag (default: disabled)])],
+    [enable_werror=$enableval],
+    [enable_werror="no"])
+AS_IF([test "x$enable_werror" = "xyes"],
+      [CFLAGS="$CFLAGS -pedantic-errors -Werror"])
 
 dnl Done.
 AC_CONFIG_FILES([Makefile

