--- src/Makefile.am.old	2009-07-13 14:10:53.000000000 +0200
+++ src/Makefile.am	2009-07-13 14:11:19.000000000 +0200
@@ -28,5 +28,5 @@
 
 gnurobots_SOURCES = main.c api.c map.c grobot.c \
 		ui-window.c ui-cmdwin.c ui-arena.c
-gnurobots_LDFLAGS = $(GUILE_LDFLAGS) $(READLINE_LIBS) $(GTHREAD2_LIBS) \
+gnurobots_LDADD = $(GUILE_LDFLAGS) $(READLINE_LIBS) $(GTHREAD2_LIBS) \
 		$(GTK2_LIBS) $(VTE_LIBS) -lutil
