--- a/configure.in
+++ b/configure.in
@@ -30,7 +30,6 @@
 	[tirc_cv_debug="yes"], [tirc_cv_debug="no"])
 if test "$tirc_cv_debug" = yes; then
 	AC_DEFINE(DEBUGV)
-	OPTFLAG=${OPTFLAG-"-g"}
 else
 	dnl Defining NDEBUG disables the assert() macro.
 	AC_DEFINE(NDEBUG)
@@ -351,6 +350,10 @@
 		[tirc_cv_have_tparm=no])
 fi
 if test "$tirc_cv_have_tparm" = no; then
+	AC_CHECK_LIB(tinfo, tparm, [tirc_cv_have_tparm=tinfo],
+		[tirc_cv_have_tparm=no])
+fi
+if test "$tirc_cv_have_tparm" = no; then
 	AC_CHECK_LIB(curses, tparm, [tirc_cv_have_tparm=curses],
 		[tirc_cv_have_tparm=no])
 fi
@@ -715,7 +718,7 @@
 
 dnl Get the optimization and debug stuff out of CFLAGS where the darn stupid
 dnl autoconf dumps them into.
-CFLAGS=`echo $CFLAGS | sed -e 's/-O.//' -e 's/-g//'`
+dnl CFLAGS=`echo $CFLAGS | sed -e 's/-O.//' -e 's/-g//'`
 
 dnl Add efence if configured.
 if test "$tirc_cv_efence" = yes; then
