diff -urb builder-2.1.2/lib/builder/xchar.rb activesupport-2.3.2/lib/active_support/vendor/builder-2.1.2/builder/xchar.rb
--- builder-2.1.2/lib/builder/xchar.rb	2006-12-24 20:29:01.000000000 +0100
+++ activesupport-2.3.2/lib/active_support/vendor/builder-2.1.2/builder/xchar.rb	2008-10-19 20:32:21.000000000 +0200
@@ -10,7 +10,7 @@
 
 module Builder
   def self.check_for_name_collision(klass, method_name, defined_constant=nil)
-    if klass.instance_methods.include?(method_name)
+    if klass.instance_methods.include?(method_name.to_s)
       fail RuntimeError,
 	"Name Collision: Method '#{method_name}' is already defined in #{klass}"
     end
