http://overlays.gentoo.org/proj/java/changeset/8184

--- src/gjl
+++ src/gjl
@@ -23,7 +23,7 @@
 
 #Depreciated.
 def get_needed_target(pkg):
-    highest_target = None
+    highest_target = ""
     needs_jdk = False
     for pkg in get_needed_packages(pkg):
         pkg_needs_jdk = False
@@ -43,7 +43,7 @@
     return highest_target, needs_jdk
 
 def get_needed_target2(pkg):
-    highest_target = None
+    highest_target = ""
     needs_jdk = set()
     for pkg in get_needed_packages(pkg):
         target = pkg.target()
