--- setup
+++ setup
@@ -28,7 +28,7 @@
 ##-----------------------------------------------------------------------------
 ##	HERE ARE SOME USER-CONFIGURABLE VARIABLES
 
-EXTRA_CFLAGS="-DUNIT_ON_MAP -DNEW_AI -DUSE_LIBMODPLUG -DUSE_HP_FOR_XP -DNEW_FOW -DBUILDING_DESTROYED"
+EXTRA_CFLAGS="-DUNIT_ON_MAP -DNEW_AI -DUSE_LIBMODPLUG -DUSE_HP_FOR_XP -DNEW_FOW -DBUILDING_DESTROYED ${GENTOO_CFLAGS}"
 
 ## These could be used for experimental versions
 ## ---------------------------------------------
@@ -133,7 +133,7 @@
 #DEBUG_CFLAGS="-g -O0 -fsigned-char \$(PROFILE) -pipe -Wall -Werror \$(IFLAGS) \$(DFLAGS)"
 DEBUG_CFLAGS="-g -O1 -fsigned-char -Wall -Werror \$(IFLAGS) \$(DFLAGS)"
 #-- Production (default)
-CFLAGS="-O2 -pipe -fsigned-char -fomit-frame-pointer -fconserve-space -fexpensive-optimizations -ffast-math  \$(IFLAGS) \$(DFLAGS)"
+CFLAGS="-fsigned-char \$(IFLAGS) \$(DFLAGS)"
 #-- High Optimization
 HIGHOPT="-O3 -pipe -fsigned-char -fomit-frame-pointer -foptimize-sibling-calls -ffast-math -fstrength-reduce -fthread-jumps -fcse-follow-jumps -fcse-skip-blocks -frerun-cse-after-loop -frerun-loop-opt -fgcse -fgcse-lm -fgcse-sm -fdelete-null-pointer-checks -fexpensive-optimizations -fschedule-insns2 -funroll-loops -frename-registers -fno-cprop-registers \$(IFLAGS) \$(DFLAGS)"
 
@@ -459,7 +459,7 @@
   fi
   echo
   echo -n "Please enter selection: (s) "
-  read $N VIDSEL
+  VIDSEL=S
   echo
 fi
 
@@ -510,7 +510,7 @@
 	if [ "$WIN32" = "1" ]; then
 	  SDLLIB="\$(shell $SDL_CONFIG --libs) -lmingwex"
 	else
-	  SDLLIB="\$(shell $SDL_CONFIG --libs)"
+	  SDLLIB="\$(shell $SDL_CONFIG --libs) -lX11"
 	fi
 	echo "SDL		= -DUSE_SDL -DUSE_SDLA \$(SDL_CFLAGS)" >> $RULESFILE
 	echo "SDL_CFLAGS	= \$(shell $SDL_CONFIG --cflags)" >> $RULESFILE
@@ -551,7 +551,7 @@
 
 echo
 echo -n "Do you want to compile with sound? (y) "
-read $N ANSWER
+ANSWER=y
 echo
 case $ANSWER in
   [nN])
@@ -572,7 +572,7 @@
 	else
 	  echo
 	  echo -n "Do you want use arts daemon sound output? (n) "
-	  read $N ANSWERA
+	  ANSWERA=n
 	  echo
 	  case $ANSWERA in
 	    [yY])
@@ -595,7 +595,7 @@
 	    echo
 	    echo "Assuming you have a thread-safe X11 (libc6 or glibc),"
 	    echo -n "do you want to compile with threaded sound? (n) "
-	    read $N ANSWER2
+	    ANSWER2=y
 	    echo
 	    case $ANSWER2 in 
 	      [yY])
@@ -618,7 +618,7 @@
 	LIBCHECK="-lFLAC -lm"; check_for_libs
 	if test "$?" = 0 ; then 
 	echo -n "Would you like FLAC support? (n) "
-	read $N ANSWER
+	ANSWER=n
 	echo
 	case $ANSWER in 
 	    [yY])
@@ -638,7 +638,7 @@
 	LIBCHECK="-lvorbisfile -lvorbis -logg"; check_for_libs
 	if test "$?" = 0 ; then 
 	echo -n "Would you like OGG support? (n) "
-	read $N ANSWER
+	ANSWER=n
 	echo
 	case $ANSWER in 
 	    [yY])
@@ -671,7 +671,7 @@
 	LIBCHECK="-lmad"; check_for_libs
 	if test "$?" = 0 ; then 
 	echo -n "Would you like MAD MP3 support? (n) "
-	read $N ANSWER
+	ANSWER=n
 	echo
 	case $ANSWER in 
 	    [yY])
@@ -693,7 +693,7 @@
 	echo -n "or [S]DL "
 	fi
 	echo -n "CD Audio support? (n) "
-	read $N CDMode
+	CDMode=s
 	echo
 	case $CDMode in
 	    [sS])
@@ -777,7 +777,7 @@
 fi
 echo
 echo -n "Please enter selection ($COMPDEF): "
-read $N COMPOPT
+COMPOPT=O
 echo
 
 if test "$COMPOPT x" = " x" ; then COMPOPT=$COMPDEF; fi
@@ -827,7 +827,7 @@
     echo "optimization support, which may cause Freecraft to run much slower."
     echo
     echo -n "Would you like debug support? (n): "
-    read $N DEBUGASK
+    DEBUGASK=n
     case $DEBUGASK in
 	[yY])
 	    echo
@@ -867,7 +867,7 @@
     echo " H - High Optimization, not recommended."
     echo
     echo -n "Please enter selection (n): "
-    read $N OPTASK
+    OPTASK=N
     case $OPTASK in
 	[hH])
 	    echo
@@ -894,7 +894,7 @@
 	    echo "	Athlon XP, Athlon MP, Duron"
 	    echo
 	    echo -n "Please enter selection: "
-	    read $N TYPEASK
+	    TYPEASK=
 	    echo
 	    case $TYPEASK in
 		[iI])
@@ -910,7 +910,7 @@
 		    echo "  F - Pentium 4"
 		    echo
 		    echo -n "To choose none, press enter: "
-		    read $N PROCASK
+		    PROCASK=
 		    case $PROCASK in
 			[aA])
 			    CFLAGS="$CFLAGS -march=pentium -malign-double -mpreferred-stack-boundary=3"
@@ -949,7 +949,7 @@
 		    echo
 		    echo "To choose none, just press enter."
 		    echo -n "Please enter selection: "
-		    read $N PROCASK
+		    PROCASK=
 		    case $PROCASK in
 			[aA])
 			    CFLAGS="$CFLAGS -march=k6 -malign-double"
@@ -1156,7 +1156,7 @@
 echo "Done generating $RULESFILE."
 echo
 echo -n "Would you like to compile it now? (y) "
-read $N ANSWER
+ANSWER=n
 echo
 
 case $ANSWER in
