--- a/ipl/Makefile	2007-09-12 18:12:03.000000000 +0200
+++ b/ipl/Makefile	2009-05-07 06:16:20.000000000 +0200
@@ -11,17 +11,11 @@
 
 MACHINE	:= $(subst 64,,$(shell uname -m))
 
-ifneq (${MACHINE},parisc)
-# cross compilation
-CC	= hppa-linux-gcc
-LD	= hppa-linux-ld
-AR	= hppa-linux-ar
-endif
-
 # Source sharing with palo
 VPATH=../lib:.
 
-CFLAGS	= -I. -I../lib -I../include -O2 -mdisable-fpregs -Wall
+CFLAGS ?= -O2
+CFLAGS += -I. -I../lib -I../include -mdisable-fpregs -Wall
 LDFLAGS  = -N --section-start .init=0x60000 -e '$$START$$'
 
 all:	iplelf
--- a/palo/Makefile	2007-09-12 18:12:03.000000000 +0200
+++ b/palo/Makefile	2009-05-07 06:16:50.000000000 +0200
@@ -10,11 +10,11 @@
 ###############{
 
 OS := $(shell uname -s)
-CC=gcc
+
+CFLAGS ?= -g -O
+CFLAGS += -I../include -I../lib
 ifeq ($(strip ${OS}),HP-UX)
-CFLAGS=-g -O -I../include -I../lib -I$(PA)/include
-else
-CFLAGS=-g -O -I../include -I../lib
+CFLAGS += -I$(PA)/include
 endif
 
 # LDFLAGS=-Wl,-Bstatic 
