--- a/Makefile	2004-05-17 13:39:00.431714000 +0200
+++ b/Makefile	2004-05-17 13:39:45.221714000 +0200
@@ -56,7 +56,7 @@
     ROOT=root=/dev/nfs nfsroot=${NFSROOT} ip=bootp
 endif
 
-CMDLINE = 0/vmlinux HOME=/ TERM=linux $(ROOT) $(EISAIRQ) $(CONSOLE)
+CMDLINE = 0/vmlinux $(ROOT) $(EISAIRQ) $(CONSOLE)
 
 all: makepalo README palo.conf
 	
--- a/README	2004-05-17 13:39:00.431714000 +0200
+++ b/README	2004-05-17 13:40:09.341714000 +0200
@@ -42,13 +42,13 @@
        files to it, unmount.
      * Use palo to make the disk bootable:
         $ palo \
-             -c "5/vmlinux HOME=/ TERM=linux console=tty" \
+             -c "5/vmlinux console=tty" \
              -k path/to/vmlinux \
              -I /dev/your-disk
 
       On a self-hosted system, you can accomplish the same thing by placing
       the following into /etc/palo.conf:
-   --commandline=5/vmlinux HOME=/ TERM=linux console=tty
+   --commandline=5/vmlinux console=tty
    --recoverykernel=/boot/vmlinux
    --init-partitioned=/dev/your-disk
 
--- a/README.html	2004-05-17 13:39:00.431714000 +0200
+++ b/README.html	2004-05-17 13:40:27.801714000 +0200
@@ -52,13 +52,13 @@
     <p><li>Use <i>palo</i> to make the disk bootable:
     <pre>
 	$ palo \
-	     -c "5/vmlinux HOME=/ TERM=linux console=tty" \
+	     -c "5/vmlinux console=tty" \
 	     -k path/to/vmlinux \
 	     -I /dev/your-disk</pre>
 <center><table border=1 bgcolor=yellow width=80%><tr><td>
 On a self-hosted system, you can accomplish the same thing by placing
 the following into <tt>/etc/palo.conf</tt>:
-<pre>--commandline=5/vmlinux HOME=/ TERM=linux console=tty
+<pre>--commandline=5/vmlinux console=tty
 --recoverykernel=/boot/vmlinux
 --init-partitioned=/dev/your-disk</pre>
 and running <i>palo</i>.
--- a/ipl/ipl.c	2004-05-17 13:39:00.491714000 +0200
+++ b/ipl/ipl.c	2004-05-17 13:39:19.681714000 +0200
@@ -481,9 +481,6 @@
 	        chk_strcat(commandline, "ttyB0", CMDLINELEN, &ok);
             else
 	        chk_strcat(commandline, "ttyS0", CMDLINELEN, &ok);
-
-	    if (strstr(commandline, " TERM=") == 0)
-	        chk_strcat(commandline, " TERM=vt102", CMDLINELEN, &ok);
 	}
 	else
 	{
@@ -519,8 +516,6 @@
 	    }
 	    if (strstr(commandline, " sti_font=") == 0)
 	        chk_strcat(commandline, " sti_font=VGA8x16", CMDLINELEN, &ok);
-	    if (strstr(commandline, " TERM=") == 0)
-	        chk_strcat(commandline, " TERM=linux", CMDLINELEN, &ok);
 	}
     }
 
--- a/palo.conf	2004-05-17 13:39:00.431714000 +0200
+++ b/palo.conf	2004-05-17 13:41:01.541714000 +0200
@@ -8,7 +8,7 @@
 # recovery kernel, and the default dynamically-booted kernel.
 # --recoverykernel=/boot/vmlinux
 # --init-partitioned=/dev/sda
-# --commandline=3/boot/vmlinux HOME=/ TERM=linux root=/dev/sda3
+# --commandline=3/boot/vmlinux root=/dev/sda3
 
 #
 #Usage: palo [options]
@@ -18,8 +18,8 @@
 #		Maximum 127 characters.
 #    -k, --recoverykernel=path to recovery kernel (perhaps /boot/vmlinux)
 #    -b, --bootloader=path to boot loader
-		Defaults to /usr/share/palo/iplboot -- you usually don't
-		need to use -b or --bootloader
+#		Defaults to /usr/share/palo/iplboot -- you usually don't
+#		need to use -b or --bootloader
 #    -r, --ramdisk=path to initial ramdisk image
 #    -I, --init-partitioned=partitioned device or file
 #    		Initialize a pre-partitioned device, usually a hard disk.
