--- MusicBox/plugins/flac/Makefile.orig	2006-10-03 09:24:56.000000000 -0600
+++ MusicBox/plugins/flac/Makefile	2006-10-03 09:25:10.000000000 -0600
@@ -11,15 +11,15 @@
 
 sw_metadata.py _sw_metadata.so:	sw_metadata.i format.i
 	swig -python sw_metadata.i
-	$(CC) -g $(INCL) sw_metadata_wrap.c -shared $(LIBS) -o _sw_metadata.so
+	$(CC) -fPIC $(INCL) sw_metadata_wrap.c -shared $(LIBS) -o _sw_metadata.so
 
 decoder.py _decoder.so:	decoder.i format.i
 	swig -python decoder.i
-	$(CC) -g $(INCL) decoder_wrap.c -shared $(LIBS) -o _decoder.so
+	$(CC) -fPIC $(INCL) decoder_wrap.c -shared $(LIBS) -o _decoder.so
 
 encoder.py _encoder.so:	encoder.i format.i
 	swig -python encoder.i
-	$(CC) -g $(INCL) encoder_wrap.c -shared $(LIBS) -o _encoder.so
+	$(CC) -fPIC $(INCL) encoder_wrap.c -shared $(LIBS) -o _encoder.so
 
 clean:
 	rm -f sw_metadata_wrap.c _sw_metadata.so sw_metadata.py
