=== modified file 'plugins/video/H.263-1998/h263-1998.cxx'
--- plugins/video/H.263-1998/h263-1998.cxx	2010-12-06 15:18:06 +0000
+++ plugins/video/H.263-1998/h263-1998.cxx	2010-12-06 15:18:46 +0000
@@ -51,6 +51,8 @@
 
 #include "plugin-config.h"
 
+#define __STDC_CONSTANT_MACROS
+
 #include "h263-1998.h"
 #include <limits.h>
 #include <math.h>

=== modified file 'plugins/video/H.263-1998/h263pframe.h'
--- plugins/video/H.263-1998/h263pframe.h	2010-12-06 15:18:06 +0000
+++ plugins/video/H.263-1998/h263pframe.h	2010-12-06 15:19:56 +0000
@@ -25,6 +25,8 @@
 
 #include "plugin-config.h"
 
+#define __STDC_CONSTANT_MACROS
+
 #include <stdint.h>
 #include <vector>
 #include <string.h>

=== modified file 'plugins/video/common/trace.h'
--- plugins/video/common/trace.h	2010-12-06 15:18:06 +0000
+++ plugins/video/common/trace.h	2010-12-06 15:18:46 +0000
@@ -34,6 +34,8 @@
 
 #define TRACING 1
 
+#define __STDC_CONSTANT_MACROS
+
 #include <iostream>
 #include <iomanip>
 using namespace std;

