Fix LID plugin ldflags

https://bugs.gentoo.org/397681

--- plugins/LID/CAPI/Makefile.in
+++ plugins/LID/CAPI/Makefile.in
@@ -67,7 +67,7 @@
 OBJECTS = $(addprefix $(OBJDIR)/,$(patsubst %.cxx,%.o,$(notdir $(SRCS))))
 
 $(PLUGIN): $(OBJECTS)
-	$(Q_LD)$(CC) $(LDSO)$(SONAME) -o $@ $^ $(EXTRALIBS)
+	$(Q_LD)$(CC) $(LDFLAGS) $(LDSO)$(SONAME) -o $@ $^ $(EXTRALIBS)
 
 install:
 	mkdir -p $(DESTDIR)$(libdir)/$(LID_PLUGIN_DIR)
--- plugins/LID/IxJ/Makefile.in
+++ plugins/LID/IxJ/Makefile.in
@@ -72,7 +72,7 @@
 OBJECTS = $(addprefix $(OBJDIR)/,$(patsubst %.cxx,%.o,$(notdir $(SRCS))))
 
 $(PLUGIN): $(OBJECTS)
-	$(Q_LD)$(CC) $(LDSO)$(SONAME) -o $@ $^ $(EXTRALIBS)
+	$(Q_LD)$(CC) $(LDFLAGS) $(LDSO)$(SONAME) -o $@ $^ $(EXTRALIBS)
 
 install:
 	mkdir -p $(DESTDIR)$(libdir)/$(LID_PLUGIN_DIR)
--- plugins/LID/VPB/Makefile.in
+++ plugins/LID/VPB/Makefile.in
@@ -79,9 +79,9 @@
 
 ifeq (solaris,$(findstring solaris,$(target_os)))
 
-	$(Q_LD)$(CC) $(LDSO) $@ -o $@ $^ $(EXTRALIBS)
+	$(Q_LD)$(CC) $(LDFLAGS) $(LDSO) $@ -o $@ $^ $(EXTRALIBS)
 else	
-	$(Q_LD)$(CC) $(LDSO) -o $@ $^ $(EXTRALIBS)
+	$(Q_LD)$(CC) $(LDFLAGS) $(LDSO) -o $@ $^ $(EXTRALIBS)
 
 endif	
 
