--- brewtarget-1.2.4.orig/CMakeLists.txt	2012-03-14 00:08:19.969684464 +0100
+++ brewtarget-1.2.4/CMakeLists.txt	2012-03-14 00:11:39.289192627 +0100
@@ -66,29 +66,28 @@
 INCLUDE_DIRECTORIES("${CMAKE_BINARY_DIR}/QtDesignerPlugins")
 
 # ======================Find Qt4 (>= 4.6.0) for this project===================
-SET( EXT_LIBS "" )
-
 IF( ${BUILD_DESIGNER_PLUGINS} )
-   FIND_PACKAGE(Qt4 4.6.0 COMPONENTS QtCore QtGui QtNetwork QtSvg QtWebKit QtXml Phonon QtDesigner REQUIRED)
+   FIND_PACKAGE(Qt4 4.6.0 COMPONENTS QtCore QtGui QtNetwork QtSvg QtWebKit QtXml QtDesigner REQUIRED)
 ELSE()
-   FIND_PACKAGE(Qt4 4.6.0 COMPONENTS QtCore QtGui QtNetwork QtSvg QtWebkit QtXml Phonon REQUIRED)
+   FIND_PACKAGE(Qt4 4.6.0 COMPONENTS QtCore QtGui QtNetwork QtSvg QtWebkit QtXml REQUIRED)
 ENDIF()
 
 MESSAGE( STATUS "CMAKE_SYSTEM_INCLUDE_PATH: ${CMAKE_SYSTEM_INCLUDE_PATH}" )
 
 # Some distros seem to treat Qt and Phonon as separate, so try to find Phonon.
 IF( NOT QT_PHONON_FOUND )
-   MESSAGE( STATUS "Phonon not found as a part of Qt; trying another approach." )
-
-   FIND_LIBRARY( EXT_PHONON_LIB phonon )
-   FIND_PATH( EXT_PHONON_PATH "mediaobject.h" )
+   FIND_LIBRARY( EXT_PHONON_LIB phonon
+                 PATHS /usr/lib64/qt4 )
+   FIND_PATH( EXT_PHONON_PATH "mediaobject.h"
+              PATHS /usr/include/phonon
+                    /usr/include/qt4/phonon )
 
    IF( EXISTS ${EXT_PHONON_LIB} AND EXISTS ${EXT_PHONON_PATH} )
       MESSAGE( STATUS "Hooray! Phonon found." )
       MESSAGE( STATUS "Phonon dir: ${EXT_PHONON_PATH}")
       MESSAGE( STATUS "Phonon lib: ${EXT_PHONON_LIB}")
 
-      SET( EXT_LIBS ${EXT_LIBS} ${EXT_PHONON_LIB} )
+      SET( EXT_LIBRARIES ${EXT_PHONON_LIB} )
       INCLUDE_DIRECTORIES( ${EXT_PHONON_PATH} )
    ELSE()
       MESSAGE( FATAL_ERROR "Phonon not found. Run cmake again with the option -DCMAKE_PREFIX_PATH=/path/to/phonon" )
