diff --git a/capplets/common/Makefile.am b/capplets/common/Makefile.am
index 70ad44f..52ea6e6 100644
--- a/capplets/common/Makefile.am
+++ b/capplets/common/Makefile.am
@@ -48,7 +48,8 @@ libcommon_la_LIBADD =							\
 	$(METACITY_LIBS)						\
 	$(DBUS_LIBS)							\
 	$(GNOME_DESKTOP_LIBS)						\
-	$(GIO_LIBS)
+	$(GIO_LIBS)							\
+	$(GMODULE_LIBS)
 
 gnome_theme_test_SOURCES = \
 	gnome-theme-test.c
diff --git a/configure.ac b/configure.ac
index 4bc8d9b..f869f93 100644
--- a/configure.ac
+++ b/configure.ac
@@ -132,6 +132,7 @@ PKG_CHECK_MODULES(METACITY, libmetacity-private >= 2.23.1)
 PKG_CHECK_MODULES(TYPING, glib-2.0 > 2.11 gconf-2.0 gtk+-2.0)
 PKG_CHECK_MODULES(GSD_DBUS, gnome-settings-daemon)
 PKG_CHECK_MODULES(GIO, gio-2.0)
+PKG_CHECK_MODULES(GMODULE, gmodule-2.0)
 
 gtk_lib_dir=`$PKG_CONFIG --variable libdir gtk+-2.0`
 gtk_binary_version=`$PKG_CONFIG --variable gtk_binary_version gtk+-2.0`
diff --git a/libwindow-settings/Makefile.am b/libwindow-settings/Makefile.am
index ee6984d..0464597 100644
--- a/libwindow-settings/Makefile.am
+++ b/libwindow-settings/Makefile.am
@@ -17,7 +17,9 @@ libgnome_window_settings_la_LDFLAGS = 	\
 	-no-undefined			\
         -version-info 1:0:0
 
-libgnome_window_settings_la_LIBADD = $(GNOMECC_LIBS)
+libgnome_window_settings_la_LIBADD = \
+	$(GNOMECC_LIBS) \
+	$(GMODULE_LIBS)
 
 libgnome_window_settings_la_SOURCES = \
 	gnome-window-manager.c	\
