--- panels/network/cc-network-panel.c~	2014-04-14 11:23:25.000000000 +0200
+++ panels/network/cc-network-panel.c	2015-01-12 17:51:44.855061255 +0100
@@ -1294,30 +1294,18 @@
 panel_check_network_manager_version (CcNetworkPanel *panel)
 {
         const gchar *version;
-        gchar **split = NULL;
-        guint major = 0;
-        guint micro = 0;
-        guint minor = 0;
         gboolean ret = TRUE;
 
         /* parse running version */
         version = nm_client_get_version (panel->priv->client);
-        if (version != NULL) {
-                split = g_strsplit (version, ".", -1);
-                major = atoi (split[0]);
-                minor = atoi (split[1]);
-                micro = atoi (split[2]);
-        }
 
-        /* is it too new or old */
-        if (major > 0 || major > 9 || (minor <= 8 && micro < 992)) {
+        if (version == NULL) {
                 ret = FALSE;
 
                 /* do modal dialog in idle so we don't block startup */
                 panel->priv->nm_warning_idle = g_idle_add ((GSourceFunc)display_version_warning_idle, panel);
         }
 
-        g_strfreev (split);
         return ret;
 }
 
@@ -1364,8 +1352,9 @@
 {
         gboolean ret;
 
-        /* is the user compiling against a new version, but running an
-         * old daemon version? */
+        /* is the user compiling against a new version, but not running
+         * the daemon? */
+
         ret = panel_check_network_manager_version (panel);
         if (ret) {
                 manager_running (panel->priv->client, NULL, panel);
