diff -Naur setup-0.3.1-orig/i18n.cpp setup-0.3.1/i18n.cpp
--- setup-0.3.1-orig/i18n.cpp	2006-03-13 14:03:16.420823500 +0100
+++ setup-0.3.1/i18n.cpp	2006-03-13 14:33:37.790652000 +0100
@@ -12,24 +12,28 @@
 
 const tI18nPhrase Phrases[] = {
   {
-    "Setup - VDR",
-    "Setup - VDR",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
+    "Setup - VDR", // English
+    "Setup - VDR", // Deutsch
+    "", // Slovenski
+    "", // Italiano
+    "", // Nederlands
+    "", // Portugus
+    "", // 
+    "", // Norsk
+    "", // suomi
+    "", // Polski
+    "", //
+    "", //  // svenska
+    "", // Romaneste
+    "", // Magyar
+    "", // 
+    "", // Catal`
+    "", // Croatian
+    "", // Eesti
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
   {
     "Activate Plugins",
@@ -50,6 +54,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
 
   {
@@ -71,6 +79,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
   {
     "no",
@@ -91,6 +103,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
   {
     "yes",
@@ -111,6 +127,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
   {
     "Move",
@@ -131,6 +151,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
   {
     "Create",
@@ -151,6 +175,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
   {
     "Edit",
@@ -171,6 +199,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
   {
     "Create Menu",
@@ -191,6 +223,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
   {
     "Edit Menu",
@@ -211,6 +247,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
   {
     "Delete Menu?",
@@ -231,6 +271,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
   {
     "change menu name",
@@ -251,6 +295,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
   {
     "Delete",
@@ -271,6 +319,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
 
   {
@@ -292,6 +344,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
   
   {
@@ -313,6 +369,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
   {
     "After",
@@ -333,6 +393,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
   {
     "Into",
@@ -353,6 +417,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
   {
     "sort asc",
@@ -373,6 +441,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
   {
     "sort desc",
@@ -393,6 +465,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
   
   {
@@ -414,6 +490,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
   {
     "Plugins activate / deactivate",
@@ -434,6 +514,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
   {
     "Save",
@@ -454,6 +538,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
   {
     "Reboot",
@@ -474,10 +562,14 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
   {
     "Restart VDR",
-    "Starte VDR",
+    "Restarte VDR",
     "",
     "",
     "",
@@ -494,6 +586,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
   {
     "protected",
@@ -514,6 +610,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
   
   {
@@ -535,6 +635,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
 
   {
@@ -556,11 +660,15 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
 
   {
     "Error in configuration files",
-    "Fehler in MiniVDR-Konfigration",
+    "Fehler in VDR-Konfigration",
     "",
     "",
     "",
@@ -577,6 +685,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
 
   {
@@ -598,6 +710,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
 
   {
@@ -619,6 +735,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
   
   {
@@ -640,6 +760,10 @@
     "",
     "",
     "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
   },
   
   {
@@ -661,6 +785,10 @@
       "",
       "",
       "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
 },
 {
   "Execute",
@@ -681,6 +809,10 @@
   "",
   "",
   "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
 },
 
   // Standard VDR Setup pages:
@@ -935,6 +1067,10 @@
   "",
   "",
   "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
 },
 
 {
@@ -956,6 +1092,10 @@
   "",
   "",
   "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
 },
 
 
@@ -978,6 +1118,10 @@
   "",
   "",
   "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
 },
 
 {
@@ -999,6 +1143,10 @@
   "",
   "",
   "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
 },
 {
   "Up/Dn for new location - color key to move",
@@ -1019,6 +1167,10 @@
   "",
   "",
   "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
 },
 {
   "Help",
@@ -1039,6 +1191,10 @@
   "",
   "",
   "",
+	"", // Dansk
+#if VDRVERSNUM >= 10342
+    "", // Czech
+#endif
 },
 
 { "Really restart?",
@@ -1084,6 +1240,26 @@
     "Reboot?",
     "",
 },
-
+{ "Setup - Gentoo-VDR",
+  "Setup - Gentoo-VDR",
+  "",
+  "",
+  "",
+  "",
+  "",
+  "",
+  "",
+  "",
+  "",
+  "",
+  "",
+  "",
+  "",
+  "",
+  "",
+  "",
+  "Setup - Gentoo-VDR",
+  "",
+},
 { NULL }
 };
