--- perl-tk-804.028/pTk/Tcl-pTk
+++ perl-tk-804.028/pTk/Tcl-pTk
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 use open IO => ':bytes'; # Avoid UTF-8 issues with some perl5.8.0 (RedHat)
 use Carp;
 my $verbose = 0;
--- perl-tk-804.028.orig/pTk/mkVFunc
+++ perl-tk-804.028/pTk/mkVFunc
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 use strict;
 
 
--- perl-tk-804.028/TextList/reindex.pl
+++ perl-tk-804.028/TextList/reindex.pl
@@ -1,4 +1,4 @@
-#!/bin/perl
+#!/usr/bin/perl
 
 use lib qw(/home1/gbartels/textlist);
 use Tk;
