diff -Naur flite-1.2-release.orig/tools/find_sts_main.c flite-1.2-release/tools/find_sts_main.c
--- tools/find_sts_main.c	2002-12-26 17:18:30.000000000 +0000
+++ tools/find_sts_main.c	2004-05-31 10:28:14.399281736 +0000
@@ -40,6 +40,10 @@
 #include <stdio.h>
 #include <math.h>
 #include <string.h>
+
+/* to allow some normally const fields to be manipulated during building */
+#define const
+
 #include "cst_wave.h"
 #include "cst_track.h"
 #include "cst_sigpr.h"
