From: Julian Ospald <hasufell@gentoo.org>
Date: Sat Oct 27 22:39:31 UTC 2012
Subject: FHS

paths hardcoded until upstream fixed the logic

--- Common/Base/Ogre/BaseManager.cpp
+++ Common/Base/Ogre/BaseManager.cpp
@@ -49,13 +49,13 @@
 		mWindow(nullptr),
 		mExit(false),
 		mPluginCfgName("plugins.cfg"),
-		mResourceXMLName("resources.xml"),
+		mResourceXMLName("/etc/MYGUI/resources.xml"),
 		mResourceFileName("MyGUI_Core.xml")
 	{
 		#if MYGUI_PLATFORM == MYGUI_PLATFORM_APPLE
 			mResourcePath = macBundlePath() + "/Contents/Resources/";
 		#else
-			mResourcePath = "";
+			mResourcePath = "/etc/MYGUI/";
 		#endif
 	}
 
--- CMake/InstallResources.cmake
+++ CMake/InstallResources.cmake
@@ -24,7 +24,7 @@
 	else ()
 		install(FILES
 			${MYGUI_BINARY_DIR}/bin/${FILENAME}
-			DESTINATION "bin"
+			DESTINATION "/etc/MYGUI"
 		)
 	endif ()
 endfunction(install_file)
@@ -51,7 +51,7 @@
 	if (WIN32)
 		set(MYGUI_MEDIA_DIR "../../Media")
 	elseif (UNIX)
-		set(MYGUI_MEDIA_DIR "../share/MYGUI/Media")
+		set(MYGUI_MEDIA_DIR "${CMAKE_INSTALL_PREFIX}/share/MYGUI/Media")
 	else ()
 		set(MYGUI_MEDIA_DIR "../../Media")
 	endif ()
