diff -Naur cyassl-2.9.4.orig/configure.ac cyassl-2.9.4/configure.ac
--- cyassl-2.9.4.orig/configure.ac	2014-04-09 13:23:26.000000000 -0400
+++ cyassl-2.9.4/configure.ac	2014-06-09 13:38:59.988657720 -0400
@@ -13,7 +13,7 @@
 AC_CANONICAL_HOST
 AC_CANONICAL_BUILD
 
-AM_INIT_AUTOMAKE([1.11 -Wall -Werror -Wno-portability foreign tar-ustar subdir-objects no-define color-tests])
+AM_INIT_AUTOMAKE([1.11 foreign tar-ustar subdir-objects no-define color-tests])
 AC_PREREQ([2.63])
 
 AC_ARG_PROGRAM
@@ -1586,7 +1586,6 @@
 # OPTIMIZE FLAGS
 if test "$GCC" = "yes"
 then
-    AM_CFLAGS="$AM_CFLAGS -Wall -Wno-unused"
     if test "$ax_enable_debug" = "no"
     then
         if test "$ENABLED_FASTMATH" = "yes"
@@ -1603,7 +1602,7 @@
 fi
 
 LIB_SOCKET_NSL
-AX_HARDEN_CC_COMPILER_FLAGS
+dnl AX_HARDEN_CC_COMPILER_FLAGS
 
 # link to ws2_32 if on mingw
 case $host_os in
@@ -1612,8 +1611,8 @@
 esac
 
 # add user C_EXTRA_FLAGS back
-CFLAGS="$CFLAGS $USER_C_EXTRA_FLAGS"
-OPTION_FLAGS="$USER_C_EXTRA_FLAGS $AM_CFLAGS"
+CFLAGS="-fwrapv $CFLAGS"
+OPTION_FLAGS="$AM_CFLAGS"
 
 CREATE_HEX_VERSION
 AC_SUBST([AM_CPPFLAGS])
diff -Naur cyassl-2.9.4.orig/m4/ax_debug.m4 cyassl-2.9.4/m4/ax_debug.m4
--- cyassl-2.9.4.orig/m4/ax_debug.m4	2014-03-27 13:44:24.000000000 -0400
+++ cyassl-2.9.4/m4/ax_debug.m4	2014-06-09 13:34:14.291644716 -0400
@@ -43,19 +43,21 @@
 #  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 #  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-#serial 6
+#serial 6.1
 
 AC_DEFUN([AX_DEBUG],
     [AC_PREREQ([2.63])dnl
     AC_ARG_ENABLE([debug],
       [AS_HELP_STRING([--enable-debug],
         [Add debug code/turns off optimizations (yes|no) @<:@default=no@:>@])],
-      [ax_enable_debug=yes
-      AC_DEFINE([DEBUG],[1],[Define to 1 to enable debugging code.])],
-      [ax_enable_debug=no
-      AC_SUBST([MCHECK])
-      AC_DEFINE([DEBUG],[0],[Define to 1 to enable debugging code.])])
+      [ax_enable_debug=$enableval],
+      [ax_enable_debug=no])
+
+	AS_IF([test "x$ax_enable_debug" = xyes],
+		[AC_DEFINE([DEBUG],[1],[Define to 1 to enable debugging code.])],
+		[AC_SUBST([MCHECK])
+         AC_DEFINE([DEBUG],[0],[Define to 1 to enable debugging code.])])
 
     AC_MSG_CHECKING([for debug])
     AC_MSG_RESULT([$ax_enable_debug])
-    AM_CONDITIONAL([DEBUG],[test "x${ax_enable_debug}" = "xyes"])])
+    AM_CONDITIONAL([DEBUG],[test "x${ax_enable_debug}" = xyes])])
