Filtered from upstream commit bc0df844c7b20f30e456c493e086a9a871471550
Author: Eloy Duran <eloy.de.enige@gmail.com>
Date:   Mon Aug 31 23:36:56 2009 +0200

--- a/test/spec_bacon.rb
+++ b/test/spec_bacon.rb
@@ -1,5 +1,5 @@
 $-w,w = nil, $-w
-require File.join(File.dirname(__FILE__), '../lib/bacon')
+require File.expand_path('../../lib/bacon', __FILE__)
 $-w = w
 
 # Hooray for meta-testing.
--- a/test/spec_should.rb
+++ b/test/spec_should.rb
@@ -1,4 +1,4 @@
-require File.join(File.dirname(__FILE__), '../lib/bacon')
+require File.expand_path('../../lib/bacon', __FILE__)
 
 describe "#should shortcut for #it('should')" do
   
