diff -ru timeline-1.0.141-orig/checkerOsd.c timeline-1.0.141/checkerOsd.c
--- timeline-1.0.141-orig/checkerOsd.c	2006-11-16 14:39:23.000000000 +0100
+++ timeline-1.0.141/checkerOsd.c	2006-11-16 14:40:25.000000000 +0100
@@ -457,6 +457,9 @@
 	osd->DrawRectangle(areas[0].x1, areas[0].y1, areas[0].x2, areas[0].y2, clrGray50);
 	osd->DrawText(x0_graph+width_graph-cFont::GetFont(fontSml)->Width(navtxt),y0_graph,navtxt,clrWhite,clrBlue,cFont::GetFont(fontSml));
 
+	for(int i=0;i<10;i++)
+		quicktimer[i]=NULL;
+
 	if (Timers.Count()>0)
 	{
 		int graph_border=15;
@@ -501,8 +504,6 @@
 		cTimer *ct1;
 		struct paintdata *pd=new paintdata[2401];
 
-		for(int i=0;i<10;i++)
-			quicktimer[i]=NULL;
 		for(int i=0;i<MAXRECORDS;i++)
 			recordingtimer[i].timer=NULL;
 		for(int i=0;i<2401;i++)
