diff -ru timeline-1.0.141-orig/i18n.c timeline-1.0.141/i18n.c
--- timeline-1.0.141-orig/i18n.c	2006-11-16 14:06:30.000000000 +0100
+++ timeline-1.0.141/i18n.c	2006-11-16 14:09:05.000000000 +0100
@@ -33,7 +33,7 @@
     "",
   },
   { "Show timer overview and collisions",
-    "",
+    "Überblick über Timer/Kollisionen",
     "",
     "",
     "",
@@ -121,7 +121,7 @@
     "",
   },
   { "same input device",
-    "gleiche Eingangs-Device",
+    "gleiches Eingangs-Gerät",
     "",
     "",
     "",
