Remove dep on rdfpipe (from OpenSUSE)

--- configure.ac
+++ configure.ac
@@ -37,13 +37,4 @@
   AC_MSG_ERROR(You need the tool `rapper' from the `raptor-utils' package in order to compile Zeitgeist)
 fi
 
-# check for python-rdflib
-AC_MSG_CHECKING([for python-rdflib])
-echo "import rdflib" | python - 2>/dev/null
-if test $? -ne 0 ; then
-  AC_MSG_FAILURE([failed.  Please install the python-rdflib package.])
-else
-  AC_MSG_RESULT([yes])
-fi
-
 AC_OUTPUT
