 configure.ac |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/configure.ac b/configure.ac
index fc0b202..6db8686 100644
--- a/configure.ac
+++ b/configure.ac
@@ -33,6 +33,7 @@ SQLITE_REQUIRED=3.7
 ZEITGEIST_REQUIRED="glib-2.0 >= $GLIB_REQUIRED
                     gobject-2.0 >= $GLIB_REQUIRED
                     gio-unix-2.0 >= $GLIB_REQUIRED
+                    gmodule-2.0 >= $GLIB_REQUIRED
                     sqlite3 >= $SQLITE_REQUIRED"
 
 PKG_CHECK_MODULES(ZEITGEIST, [$ZEITGEIST_REQUIRED])
