--- zeitgeist-0.9.12/configure.ac	2013-04-13 10:09:15.000000000 +0200
+++ zeitgeist-0.9.12_mod/configure.ac	2013-04-22 21:14:47.604406313 +0200
@@ -189,6 +189,8 @@
 
 if test "x$enable_downloads_monitor" = "xyes" -a "x$enable_datahub" = "xyes"; then
     AC_DEFINE([DOWNLOADS_MONITOR_ENABLED], [1], [Is the XDG_DOWNLOAD_DIRECTORY file monitor enabled?])
+else
+    AC_DEFINE([DOWNLOADS_MONITOR_ENABLED], [0], [Is the XDG_DOWNLOAD_DIRECTORY file monitor enabled?])
 fi
 AM_CONDITIONAL(DOWNLOADS_MONITOR_ENABLED, test "x$enable_downloads_monitor" = "xyes" -a "x$enable_datahub" = "xyes")
 
