diff -Naur neatx.orig/Makefile.am neatx/Makefile.am
--- neatx.orig/Makefile.am	2012-08-23 12:01:14.290018757 +0200
+++ neatx/Makefile.am	2012-08-23 14:16:42.119497905 +0200
@@ -16,7 +16,7 @@
 appdir = $(pkgpythondir)/app
 docdir = $(datadir)/doc/$(PACKAGE)
 
-AM_CFLAGS = -Wall -Werror -DPKGLIBDIR=\"$(pkglibdir)\"
+AM_CFLAGS = -Wall -Werror -DPKGLIBEXECDIR=\"$(pkglibexecdir)\"
 
 DIRS = \
 	autotools \
@@ -70,14 +70,14 @@
 	lib/app/nxserver.py \
 	lib/app/nxserver_login.py
 
-dist_pkglib_SCRIPTS = \
+dist_pkglibexec_SCRIPTS = \
 	src/ttysetup
 
-nodist_pkglib_SCRIPTS = \
+nodist_pkglibexec_SCRIPTS = \
 	$(LOG_WRAPPER) \
 	$(PYTHON_BOOTSTRAP)
 
-pkglib_PROGRAMS = \
+pkglibexec_PROGRAMS = \
 	src/fdcopy
 
 dist_pkgdata_DATA = \
@@ -173,7 +173,7 @@
 	  echo "VERSION_FULL = '$(VERSION_FULL)'"; \
 	  echo "LOCALSTATEDIR = '$(localstatedir)'"; \
 	  echo "SYSCONFDIR = '$(sysconfdir)'"; \
-	  echo "PKGLIBDIR = '$(pkglibdir)'"; \
+	  echo "PKGLIBEXECDIR = '$(pkglibexecdir)'"; \
 	} > $@
 
 $(REPLACE_VARS_SED): Makefile stamp-directories
@@ -182,7 +182,7 @@
 	  echo 's#@SYSCONFDIR@#$(sysconfdir)#g'; \
 	  echo 's#@LOCALSTATEDIR@#$(localstatedir)#g'; \
 	  echo 's#@SBINDIR@#$(sbindir)#g'; \
-	  echo 's#@PKGLIBDIR@#$(pkglibdir)#g'; \
+	  echo 's#@PKGLIBEXECDIR@#$(pkglibexecdir)#g'; \
 	  echo 's#@PKGPYTHONDIR@#$(pkgpythondir)#g'; \
 	} > $@
 
@@ -199,14 +199,14 @@
 $(LOG_WRAPPER): Makefile stamp-directories $(GEN_LOG_WRAPPER)
 	set -e; \
 	progname=`basename $@ | sed -e 's/-wrapper//'`; \
-	$(GEN_LOG_WRAPPER) "$(pkglibdir)/$$progname" "$(DESTDIR)$(pythondir)" > $@ || rm -f $@
+	$(GEN_LOG_WRAPPER) "$(pkglibexecdir)/$$progname" "$(DESTDIR)$(pythondir)" > $@ || rm -f $@
 	chmod u+x $@
 
 # We need to create symlinks because "make distcheck" will not install Python
 # files when building.
 srclinks: stamp-directories
 	set -e; \
-	for i in $(pkgpython_PYTHON) $(app_PYTHON) $(dist_pkglib_SCRIPTS); do \
+	for i in $(pkgpython_PYTHON) $(app_PYTHON) $(dist_pkglibexec_SCRIPTS); do \
 		if test ! -f $$i -a -f $(abs_top_srcdir)/$$i; then \
 			$(LN_S) $(abs_top_srcdir)/$$i $$i; \
 		fi; \
diff -Naur neatx.orig/lib/constants.py neatx/lib/constants.py
--- neatx.orig/lib/constants.py	2012-08-23 12:01:14.290018757 +0200
+++ neatx/lib/constants.py	2012-08-23 12:01:23.405734335 +0200
@@ -40,14 +40,14 @@
 START_GNOME_COMMAND = "gnome-session"
 
 NXUSER = "nx"
-NXSERVER = _autoconf.PKGLIBDIR + "/nxserver"
-NXNODE = _autoconf.PKGLIBDIR + "/nxnode"
-NXNODE_WRAPPER = _autoconf.PKGLIBDIR + "/nxnode-wrapper"
-NXNC = _autoconf.PKGLIBDIR + "/nxnc"
-NXDIALOG = _autoconf.PKGLIBDIR + "/nxdialog"
+NXSERVER = _autoconf.PKGLIBEXECDIR + "/nxserver"
+NXNODE = _autoconf.PKGLIBEXECDIR + "/nxnode"
+NXNODE_WRAPPER = _autoconf.PKGLIBEXECDIR + "/nxnode-wrapper"
+NXNC = _autoconf.PKGLIBEXECDIR + "/nxnc"
+NXDIALOG = _autoconf.PKGLIBEXECDIR + "/nxdialog"
 NXAGENT_PKGNAME = "nxagent"
-FDCOPY = _autoconf.PKGLIBDIR + "/fdcopy"
-TTYSETUP = _autoconf.PKGLIBDIR + "/ttysetup"
+FDCOPY = _autoconf.PKGLIBEXECDIR + "/fdcopy"
+TTYSETUP = _autoconf.PKGLIBEXECDIR + "/ttysetup"
 
 NXAGENT_VERSION_SEP = ".-~"
 
