diff --git a/gsim.pro b/gsim.pro
index 60550e9..d159b19 100755
--- a/gsim.pro
+++ b/gsim.pro
@@ -1,36 +1,7 @@
-###################################################################################################
-#                  	Change this part according to your system configuration                   #
-###################################################################################################
-
-#Uncomment next line if you want to have a system console running alongside GSim on Windows
-#win32: CONFIG +=console
-
-#Uncomment next line if you want to use OpenGL
-#CONFIG+=use_opengl
-
-#Uncomment next line if you want to use EMF output
-#EMF output requires libEMF or Wine on X11 systems and MAC(?)
-CONFIG+=use_emf
-
-#DO NOT UNCOMMENT NEXT LINE. The scripting possibilty won't work at the moment
-#CONFIG+=use_script
-
-# Change directories locations for libcmatrix, MinUIT and muParser libraries
-unix:INCLUDEPATH += /home/vnmr1/libcmatrixR3/include \
-		    /home/vnmr1/Minuit2/include \
-		    /home/vnmr1/muparser/include
-win32:INCLUDEPATH += "C:\Documents and Settings\vadzorin\My Documents\gsim\libcmatrixR3\include" "C:\Documents and Settings\vadzorin\My Documents\gsim\Minuit2\include" "C:\Documents and Settings\vadzorin\My Documents\gsim\muparser\include"
-
-unix:LIBS += -lcmatrix -L/home/vnmr1/libcmatrixR3  -lMinuit2Base -L/home/vnmr1/Minuit2/lib -L/home/vnmr1/muparser/lib -lmuparser
-win32:LIBS += -lcmatrix -L"C:\Documents and Settings\vadzorin\My Documents\gsim\libcmatrixR3" -lMinuit2 -L"C:\Documents and Settings\vadzorin\My Documents\gsim\Minuit2\lib" -L"C:\Documents and Settings\vadzorin\My Documents\gsim\muparser\lib" -lmuparser
-
-#Change directories location for libEMF (if used)
-use_emf {
-	DEFINES+=USE_EMF_OUTPUT
-	unix:LIBS +=-lEMF -L/usr/include/libEmf
-	win32:LIBS +=-lgdi32
+GSIM_ROOT = .
+!include( $$GSIM_ROOT/build.conf ) {
+	message( "You need a build.conf file with local settings!" )
 }
-
 #################################################################################################
 #                             Don't change anything below this line                             #
 #################################################################################################
