diff -ru vdr-ffnetdev-0.1.0-orig/netosd.h vdr-ffnetdev-0.1.0/netosd.h
--- vdr-ffnetdev-0.1.0-orig/netosd.h	2008-12-17 15:46:49.194277519 +0100
+++ vdr-ffnetdev-0.1.0/netosd.h	2008-12-17 15:47:46.807603142 +0100
@@ -17,7 +17,7 @@
 	bool truecolor;
 protected:
 public:
-    cNetOSD(int XOfs, int XOfs);
+    cNetOSD(int Left, int Top);
     virtual ~cNetOSD();
     virtual eOsdError CanHandleAreas(const tArea *Areas, int NumAreas);
     virtual void Flush(void);
