diff -ru --exclude=po vdr-ffnetdev-0.1.0-orig/ffnetdev.c vdr-ffnetdev-0.1.0/ffnetdev.c
--- vdr-ffnetdev-0.1.0-orig/ffnetdev.c	2008-04-07 21:23:41.111987897 +0200
+++ vdr-ffnetdev-0.1.0/ffnetdev.c	2008-04-07 21:25:20.431993162 +0200
@@ -32,10 +32,10 @@
 	
 }
 
-cOsd * cNetOSDProvider::CreateOsd(int Left, int Top)
+cOsd * cNetOSDProvider::CreateOsd(int Left, int Top, uint level)
 {
     
-    osd = new cNetOSD(Left, Top);
+    osd = new cNetOSD(Left, Top, level);
     return osd;
 }
 
Nur in vdr-ffnetdev-0.1.0: libvdr-ffnetdev.so.1.6.0.
diff -ru --exclude=po vdr-ffnetdev-0.1.0-orig/netosd.c vdr-ffnetdev-0.1.0/netosd.c
--- vdr-ffnetdev-0.1.0-orig/netosd.c	2008-04-07 21:23:41.111987897 +0200
+++ vdr-ffnetdev-0.1.0/netosd.c	2008-04-07 21:24:58.971993422 +0200
@@ -9,7 +9,7 @@
 #include "osdworker.h"
 
 //////////////////////////////////////////////////////////////////////////////////////////////////
-cNetOSD::cNetOSD(int Left, int Top) : cOsd(Left, Top)
+cNetOSD::cNetOSD(int Left, int Top, uint level) : cOsd(Left, Top, level)
 {
 #ifdef DEBUG   
    fprintf(stderr,"[ffnetdev] NetOSD: Constructor cNetOSD.\n");
diff -ru --exclude=po vdr-ffnetdev-0.1.0-orig/netosd.h vdr-ffnetdev-0.1.0/netosd.h
--- vdr-ffnetdev-0.1.0-orig/netosd.h	2008-04-07 21:23:41.111987897 +0200
+++ vdr-ffnetdev-0.1.0/netosd.h	2008-04-07 21:25:05.401989955 +0200
@@ -17,7 +17,7 @@
 	bool truecolor;
 protected:
 public:
-    cNetOSD(int Left, int Top);
+    cNetOSD(int Left, int Top, uint level);
     virtual ~cNetOSD();
     virtual eOsdError CanHandleAreas(const tArea *Areas, int NumAreas);
     virtual void Flush(void);
@@ -31,7 +31,7 @@
     cNetOSD **NetOSD;
 public:
     cNetOSDProvider(void);
-    virtual cOsd *CreateOsd(int Left, int Top);
+    virtual cOsd *CreateOsd(int Left, int Top, uint level);
 };
 
 #endif //_NETOSD__H
