diff -ru vdr-ffnetdev-0.1.0-orig/remote.c vdr-ffnetdev-0.1.0/remote.c
--- vdr-ffnetdev-0.1.0-orig/remote.c	2007-06-24 14:50:34.525838339 +0200
+++ vdr-ffnetdev-0.1.0/remote.c	2007-06-24 14:50:55.027619351 +0200
@@ -27,7 +27,7 @@
 	return true;
 }
 
-bool cMyRemote::Put(uint64 Code, bool Repeat, bool Release)
+bool cMyRemote::Put(uint64_t Code, bool Repeat, bool Release)
 {
 	return cRemote::Put(Code, Repeat ,Release);
 }
diff -ru vdr-ffnetdev-0.1.0-orig/remote.h vdr-ffnetdev-0.1.0/remote.h
--- vdr-ffnetdev-0.1.0-orig/remote.h	2007-06-24 14:50:34.525838339 +0200
+++ vdr-ffnetdev-0.1.0/remote.h	2007-06-24 14:50:45.526795335 +0200
@@ -17,7 +17,7 @@
   cMyRemote(const char *Name);
   virtual bool Initialize(void);
   virtual bool Ready(void);
-  virtual bool Put(uint64 Code, bool Repeat = false, bool Release = false);
+  virtual bool Put(uint64_t Code, bool Repeat = false, bool Release = false);
 };
 
 
