From 7cda349c384c847fe838d418c1f83e85495d54d4 Mon Sep 17 00:00:00 2001
From: Aleksander Morgado <aleksander@lanedo.com>
Date: Tue, 23 Jul 2013 09:04:29 +0000
Subject: build: fix dbus activation file generation

https://bugzilla.gnome.org/show_bug.cgi?id=704711
---
diff --git a/configure.ac b/configure.ac
index 34ef0c7..242b248 100644
--- a/configure.ac
+++ b/configure.ac
@@ -265,7 +265,6 @@ data/Makefile
 data/ModemManager.pc
 data/mm-glib.pc
 data/org.freedesktop.ModemManager1.policy.in
-data/org.freedesktop.ModemManager1.service
 include/Makefile
 include/ModemManager-version.h
 build-aux/Makefile
diff --git a/data/Makefile.am b/data/Makefile.am
index 3a31ec5..70a92ee 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -35,6 +35,9 @@ endif
 dbusactivationdir = $(datadir)/dbus-1/system-services
 dbusactivation_DATA = org.freedesktop.ModemManager1.service
 dbusactivation_in_files = org.freedesktop.ModemManager1.service.in
+org.freedesktop.ModemManager1.service: org.freedesktop.ModemManager1.service.in
+	$(edit) $< >$@
+
 
 # Icon
 icondir=${datadir}/icons/hicolor/22x22/apps
--
cgit v0.9.0.2-2-gbebe
