--- mess/machine/corvushd.c.old	2011-06-30 11:41:55.987772189 +0200
+++ mess/machine/corvushd.c	2011-06-30 11:42:15.215986836 +0200
@@ -243,7 +243,7 @@
 		/* */
 		struct {
 			UINT8	status;						/* Status code returned by command executed */
-			UINT8	firmware[32];				/* Firmware message */
+			UINT8	firmware[33];				/* Firmware message */
 			UINT8	rom_version;				/* ROM Version */
 			struct {
 				UINT8	sectors_per_track;		/* Sectors/Track */
