diff -Nur a/gtk/Makefile b/gtk/Makefile
--- a/gtk/Makefile	2013-01-06 07:27:27.000000000 +0100
+++ b/gtk/Makefile	2013-01-21 19:37:26.051512802 +0100
@@ -15,14 +15,10 @@
 # along with this program; if not, see http://www.gnu.org/licenses/.
 ###############################################################################
 
-CXXFLAGS := -MMD -Wall -Wno-parentheses -Wno-write-strings -g -I/usr/X11R6/include -fno-exceptions -fno-rtti $(shell pkg-config --cflags gtk+-2.0) -DVERSION="\"$(shell cat ../VERSION)\""
-LDFLAGS = -L/usr/X11R6/lib
+CXXFLAGS := -MMD -Wall -Wno-parentheses -Wno-write-strings -I/usr/X11R6/include -fno-exceptions -fno-rtti $(shell pkg-config --cflags gtk+-2.0) -DVERSION="\"$(shell cat ../VERSION)\"" ${CXXFLAGS}
+LDFLAGS := -L/usr/X11R6/lib ${LDFLAGS}
 LIBS := -lXmu $(shell pkg-config --libs gtk+-2.0)
 
-ifeq "$(shell uname -s)" "Linux"
-LDFLAGS += -Wl,--hash-style=both
-endif
-
 SRCS = shell_main.cc shell_skin.cc skins.cc keymap.cc shell_loadimage.cc \
 	shell_spool.cc core_main.cc core_commands1.cc core_commands2.cc \
 	core_commands3.cc core_commands4.cc core_commands5.cc \
