diff -Nuar a/gtk/Makefile b/gtk/Makefile
--- a/gtk/Makefile	2013-01-06 07:27:27.000000000 +0100
+++ b/gtk/Makefile	2014-06-23 23:41:29.669913497 +0200
@@ -15,13 +15,9 @@
 # along with this program; if not, see http://www.gnu.org/licenses/.
 ###############################################################################
 
-CXXFLAGS := -MMD -Wall -Wno-parentheses -Wno-write-strings -g -I/usr/X11R6/include -fno-exceptions -fno-rtti $(shell pkg-config --cflags gtk+-2.0) -DVERSION="\"$(shell cat ../VERSION)\""
-LDFLAGS = -L/usr/X11R6/lib
-LIBS := -lXmu $(shell pkg-config --libs gtk+-2.0)
-
-ifeq "$(shell uname -s)" "Linux"
-LDFLAGS += -Wl,--hash-style=both
-endif
+CXXFLAGS := -MMD -Wall -Wno-parentheses -Wno-write-strings -I/usr/X11R6/include -fno-exceptions -fno-rtti $(shell pkg-config --cflags gtk+-2.0) -DVERSION="\"$(shell cat ../VERSION)\"" ${CXXFLAGS}
+LDFLAGS := -L/usr/X11R6/lib ${LDFLAGS}
+LIBS := -lXmu -lX11 -ldl -lpthread $(shell pkg-config --libs gtk+-2.0)
 
 SRCS = shell_main.cc shell_skin.cc skins.cc keymap.cc shell_loadimage.cc \
 	shell_spool.cc core_main.cc core_commands1.cc core_commands2.cc \
