--- darktable-1.0.4/src/CMakeLists.txt
+++ darktable-1.0.4/src/CMakeLists.txt
@@ -232,46 +232,11 @@
 # c++ code doesn't like -Werror (libraw)
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -fno-strict-aliasing")
 
-if(NOT APPLE AND NOT CMAKE_C_COMPILER_ID STREQUAL "Clang")
-  # apple build fails on deprecated warnings..
-  # and too many warnings reported by Clang for now
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror")
-endif(NOT APPLE AND NOT CMAKE_C_COMPILER_ID STREQUAL "Clang")
-
 if(NOT WIN32)
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pthread")
 endif(NOT WIN32)
 
-if(NOT BINARY_PACKAGE_BUILD)
-	MESSAGE("-- Checking for -march=native support")
-	CHECK_C_COMPILER_FLAG("-march=native" MARCHNATIVE)
-	if (MARCHNATIVE)
-		set(MARCH "-march=native")	
-	else()
-		MESSAGE("-- Checking for -mtune=native support")
-		CHECK_C_COMPILER_FLAG("-mtune=native" MTUNENATIVE)
-		if (MTUNENATIVE)
-			set(MARCH "-mtune=native")
-		else()
-			set(MARCH "-mtune=generic")
-		endif()
-	endif()
-else()
- 	set(MARCH "-mtune=generic")
-endif()
-
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -msse2 -g")
-set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -O3 -ffast-math -fexpensive-optimizations ${MARCH}")
-set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -O0 -g -D_DEBUG")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -msse2 -g")
-set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3 -ffast-math -fexpensive-optimizations ${MARCH}")
-set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -O0 -g -D_DEBUG")
-if(CMAKE_COMPILER_IS_GNUCC)
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mfpmath=sse")
-endif()
-if(CMAKE_COMPILER_IS_GNUCXX)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mfpmath=sse")
-endif()
+set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -D_DEBUG")
 
 #
 # Generate config.h
