--- darktable-1.1/CMakeLists.txt
+++ darktable-1.1/CMakeLists.txt
@@ -4,6 +4,7 @@
 
 option(USE_CAMERA_SUPPORT "Detect and use camera support if available." ON)
 option(USE_NLS "Build Native Language Support (using gettext)" ON)
+option(USE_COLORD "Enable colord support" ON)
 option(USE_GEO "Build geotagging parts" ON)
 option(USE_FLICKR "Enable Flickr support" ON)
 option(USE_GLIBJSON "Enable GlibJson support" ON)
--- darktable-1.1/src/CMakeLists.txt
+++ darktable-1.1/src/CMakeLists.txt
@@ -90,5 +90,3 @@
-include_directories(${CMAKE_CURRENT_SOURCE_DIR}/external/osm-gps-map/src)
-include_directories(${CMAKE_CURRENT_SOURCE_DIR}/external/colord-gtk/src)
 
 # initial compiler flags
 add_definitions("-DHAVE_CONFIG_H")
@@ -403,6 +401,7 @@
     include_directories(${LIBSOUP2_INCLUDE_DIRS})
     list(APPEND LIBS ${LIBSOUP2_LIBRARIES})
     add_definitions(${LIBSOUP2_DEFINITIONS})
+    include_directories(${CMAKE_CURRENT_SOURCE_DIR}/external/osm-gps-map/src)
     add_dependencies(lib_darktable osmgpsmap)
     add_library(osmgpsmap_static STATIC IMPORTED)
     set_target_properties(osmgpsmap_static PROPERTIES IMPORTED_LOCATION ${CMAKE_CURRENT_BINARY_DIR}/external/osm-gps-map/libosmgpsmap.a)
@@ -422,20 +421,23 @@
 target_link_libraries(lib_darktable ${LIBS})
 
 # libcolord-gtk:
-find_package(Colord)
-if(COLORD_FOUND)
-  include_directories(${COLORD_INCLUDE_DIRS})
-  list(APPEND LIBS ${COLORD_LIBRARIES})
-# we ship our own copy of libcolord-gtk - the one packaged in distros is linked against gtk3 :(
-  add_dependencies(lib_darktable colord-gtk)
-  add_library(colord-gtk_static STATIC IMPORTED)
-  set_target_properties(colord-gtk_static PROPERTIES IMPORTED_LOCATION ${CMAKE_CURRENT_BINARY_DIR}/external/colord-gtk/libcolord-gtk.a)
-  target_link_libraries(lib_darktable colord-gtk_static)
-  add_subdirectory(external/colord-gtk)
-  list(APPEND STATIC_LIBS colord-gtk_static)
-  add_dependencies(colord-gtk colord)
-  add_definitions("-DUSE_COLORDGTK")
-endif(COLORD_FOUND)
+if(USE_COLORD)
+  find_package(Colord)
+  if(COLORD_FOUND)
+    include_directories(${COLORD_INCLUDE_DIRS})
+    list(APPEND LIBS ${COLORD_LIBRARIES})
+    # we ship our own copy of libcolord-gtk - the one packaged in distros is linked against gtk3 :(
+    add_dependencies(lib_darktable colord-gtk)
+    include_directories(${CMAKE_CURRENT_SOURCE_DIR}/external/colord-gtk/src)
+    add_library(colord-gtk_static STATIC IMPORTED)
+    set_target_properties(colord-gtk_static PROPERTIES IMPORTED_LOCATION ${CMAKE_CURRENT_BINARY_DIR}/external/colord-gtk/libcolord-gtk.a)
+    target_link_libraries(lib_darktable colord-gtk_static)
+    add_subdirectory(external/colord-gtk)
+    list(APPEND STATIC_LIBS colord-gtk_static)
+    add_dependencies(colord-gtk colord)
+    add_definitions("-DUSE_COLORDGTK")
+  endif(COLORD_FOUND)
+endif(USE_COLORD)
 
 # libsquish for compressed thumbnails:
 add_dependencies(lib_darktable squish)
--- darktable-1.1/src/gui/gtk.c
+++ darktable-1.1/src/gui/gtk.c
@@ -932,8 +932,10 @@
 void dt_gui_gtk_cleanup(dt_gui_gtk_t *gui)
 {
   g_free(darktable.control->xprofile_data);
+#ifdef USE_COLORDGTK
   g_free(darktable.control->colord_profile_file);
   darktable.control->colord_profile_file = NULL;
+#endif
   darktable.control->xprofile_size = 0;
 }
 
