--- darktable-1.4.1/CMakeLists.txt
+++ darktable-1.4.1/CMakeLists.txt
@@ -26,6 +26,7 @@
 option(BINARY_PACKAGE_BUILD "Sets march optimization to generic" OFF)
 option(APRIL_FOOLS "Add the \"file manager\" (a shell) to lighttable mode" OFF)
 option(USE_XMLLINT "Run xmllint to test if darktableconfig.xml is valid" ON)
+option(USE_OPENEXR "Enable OpenEXR support" ON)
 option(USE_OPENJPEG "Enable JPEG 2000 support" ON)
 option(USE_WEBP "Enable WebP export support" ON)
 option(BUILD_CMSTEST "Build a test program to check your system's color management setup" OFF)
--- darktable-1.4.1/src/CMakeLists.txt
+++ darktable-1.4.1/src/CMakeLists.txt
@@ -152,13 +152,15 @@
   add_definitions(${INOTIFY_DEFINITIONS})
 endif(INOTIFY_FOUND)
 
-find_package(OpenEXR)
-if(OPENEXR_FOUND)
-  include_directories(${OpenEXR_INCLUDE_DIRS})
-  list(APPEND LIBS ${OpenEXR_LIBRARIES})
-  add_definitions(${OpenEXR_DEFINITIONS})
-  list(APPEND SOURCES "common/imageio_exr.cc")
-endif(OPENEXR_FOUND)
+if(USE_OPENEXR)
+  find_package(OpenEXR)
+  if(OPENEXR_FOUND)
+    include_directories(${OpenEXR_INCLUDE_DIRS})
+    list(APPEND LIBS ${OpenEXR_LIBRARIES})
+    add_definitions(${OpenEXR_DEFINITIONS})
+    list(APPEND SOURCES "common/imageio_exr.cc")
+  endif(OPENEXR_FOUND)
+endif(USE_OPENEXR)
 
 find_package(WebP 0.3.0)
 if(WEBP_FOUND)
