http://bugs.gentoo.org/show_bug.cgi?id=357877#c9

--- src/panflute/applet/applet.py
+++ src/panflute/applet/applet.py
@@ -352,7 +352,7 @@
 
         try:
             import pynotify
-            notification = pynotify.Notification (" ", "", None, None)
+            notification = pynotify.Notification (" ", "", None)
             notification.set_urgency (pynotify.URGENCY_LOW)
             return notification
         except ImportError, e:
