Index: reelchannelscan-0.4.3/csmenu.c
===================================================================
--- reelchannelscan-0.4.3.orig/csmenu.c
+++ reelchannelscan-0.4.3/csmenu.c
@@ -241,7 +241,7 @@ void cMenuChannelscan::Set()
 
   // avoid C/T-positions for SAT 
   // and take Astra as start position
-  if (srcTypes[sourceStat]==SAT  && (source==cSource::FromString("C") || source==cSource::FromString("T")))
+  if (srcTypes[sourceStat]==SAT  && (source==cSource::FromString("C") || source==cSource::FromString("T") || source==0))
     source = cSource::FromString("S19.2E"); 
 
    int blankLines = 4;
