Index: reelchannelscan-0.4.3/i18n.c
===================================================================
--- reelchannelscan-0.4.3.orig/i18n.c
+++ reelchannelscan-0.4.3/i18n.c
@@ -207,6 +207,28 @@
     "", // Dansk
     "", // Èesky (Czech)
   },
+  { "SearchMode$Auto NIT",
+    "NIT Automatik",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "", // Hrvatski (Croatian)
+    "", // Eesti
+    "", // Dansk
+    "", // Èesky (Czech)
+  },
   { "Manual",
     "Manuell",
     "",
@@ -317,6 +339,28 @@
     "", // Dansk
     "", // Èesky (Czech)
   },
+  { "Tuner",
+    "Tuner",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "", // Hrvatski (Croatian)
+    "", // Eesti
+    "", // Dansk
+    "", // Èesky (Czech)
+  },
   { "Scanning on transponder",
     "Durchsuche Transponder",
     "",
@@ -524,7 +568,7 @@
     "",
     "",
     "",
-    "Haku käynnissä %s.\tOdota hetkinen...",
+    "",
     "",
     "",
     "",
@@ -546,7 +590,7 @@
     "",
     "",
     "",
-    "Haku käynnissä %s.\tOdota hetkinen...",
+    "",
     "",
     "",
     "",
Index: reelchannelscan-0.4.3/csmenu.c
===================================================================
--- reelchannelscan-0.4.3.orig/csmenu.c
+++ reelchannelscan-0.4.3/csmenu.c
@@ -128,9 +128,11 @@ cMenuChannelscan::cMenuChannelscan(int C
   sBwItem[1] = "7 MHz";
   sBwItem[2] = "8 MHz";
 
+  /*
   sSRScanMode[0]= tr("Intelligent 6900/6875/6111");
   sSRScanMode[1]= tr("Try all 6900/6875/6111");
   sSRScanMode[2]= tr("Manual");
+  */
 
   srScanMode=0;
   
