diff -ruN gnome-alsamixer-0.9.6.orig/src/gam-mixer.h gnome-alsamixer-0.9.6/src/gam-mixer.h
--- gnome-alsamixer-0.9.6.orig/src/gam-mixer.h	2003-09-01 19:34:41.000000000 +0700
+++ gnome-alsamixer-0.9.6/src/gam-mixer.h	2005-04-11 22:07:14.094589008 +0700
@@ -21,7 +21,7 @@
 #ifndef __GAM_MIXER_H__
 #define __GAM_MIXER_H__
 
-#include <asoundlib.h>
+#include <alsa/asoundlib.h>
 #include <gtk/gtkvbox.h>
 #include <gtk/gtksizegroup.h>
 
diff -ruN gnome-alsamixer-0.9.6.orig/src/gam-slider.h gnome-alsamixer-0.9.6/src/gam-slider.h
--- gnome-alsamixer-0.9.6.orig/src/gam-slider.h	2003-09-08 19:39:12.000000000 +0700
+++ gnome-alsamixer-0.9.6/src/gam-slider.h	2005-04-11 22:07:14.094589008 +0700
@@ -21,7 +21,7 @@
 #ifndef __GAM_SLIDER_H__
 #define __GAM_SLIDER_H__
 
-#include <asoundlib.h>
+#include <alsa/asoundlib.h>
 #include <gtk/gtkhbox.h>
 #include <gtk/gtksizegroup.h>
 #include <gtk/gtklabel.h>
diff -ruN gnome-alsamixer-0.9.6.orig/src/gam-toggle.h gnome-alsamixer-0.9.6/src/gam-toggle.h
--- gnome-alsamixer-0.9.6.orig/src/gam-toggle.h	2003-08-28 00:08:03.000000000 +0700
+++ gnome-alsamixer-0.9.6/src/gam-toggle.h	2005-04-11 22:07:14.094589008 +0700
@@ -21,7 +21,7 @@
 #ifndef __GAM_TOGGLE_H__
 #define __GAM_TOGGLE_H__
 
-#include <asoundlib.h>
+#include <alsa/asoundlib.h>
 #include <gtk/gtkcheckbutton.h>
 
 G_BEGIN_DECLS
