diff --exclude='*~' --exclude='.*' -I '$Id:' -urN dvacm4-0.3.5.orig/m4/dvacm4.m4.in dvacm4-0.3.5/m4/dvacm4.m4.in
--- dvacm4-0.3.5.orig/m4/dvacm4.m4.in	2005-05-20 04:35:31.000000000 -0400
+++ dvacm4-0.3.5/m4/dvacm4.m4.in	2005-05-20 04:37:04.000000000 -0400
@@ -54,7 +54,7 @@
 dnl Append data to variable but only if grep "$data " $var fails.
 dnl
 dnl =====================================================================
-AC_DEFUN(DV_PREPEND_UNIQ,
+AC_DEFUN([DV_PREPEND_UNIQ],
 [
 { echo "${$2}" | grep -- "$1 " >/dev/null || $2="$1 ${$2}"; }dnl
 ])
@@ -71,7 +71,7 @@
 dnl of dirs. The first directory which contains file is returned in
 dnl the variabel directory
 dnl =====================================================================
-AC_DEFUN(DV_FIND_FILE,
+AC_DEFUN([DV_FIND_FILE],
 [
 $3=""
 for i in $2;
@@ -94,7 +94,7 @@
 dnl after linking with one of the libraries in $2 (without lib prefix). 
 dnl The library for which linking succeeds is added to $LIBS.
 dnl =====================================================================
-AC_DEFUN(DV_REQUIRE_CFUNC,
+AC_DEFUN([DV_REQUIRE_CFUNC],
 [
 AC_LANG_PUSH(C)
 AC_CHECK_FUNC($1,, [ AC_SEARCH_LIBS($1,"$2",,AC_MSG_ERROR([need $1 function])) ])
@@ -109,7 +109,7 @@
 dnl after linking with one of the libraries in $2 (without lib prefix). 
 dnl The library for which linking succeeds is added to $LIBS.
 dnl =====================================================================
-AC_DEFUN(DV_CHECK_CFUNC,
+AC_DEFUN([DV_CHECK_CFUNC],
 [
 AC_LANG_PUSH(C)
 AC_CHECK_FUNC($1,[ AC_DEFINE(HAVE_$1, [1], [C function $1]) ],
@@ -139,7 +139,7 @@
 dnl  else
 dnl   dv_check_lib_ok=no
 dnl =====================================================================
-AC_DEFUN(DV_CHECK_LTLIB,
+AC_DEFUN([DV_CHECK_LTLIB],
 [
 AC_LANG_PUSH(C++)
 dv_check_ltlib_ok=no;
@@ -210,7 +210,7 @@
 dnl =====================================================================
 dnl Tries DV_FIND_LIB and exits if not succesful.
 dnl =====================================================================
-AC_DEFUN(DV_REQUIRE_LTLIB,
+AC_DEFUN([DV_REQUIRE_LTLIB],
 [
 AC_MSG_CHECKING([whether required lib$3.la is available])
 
@@ -254,7 +254,7 @@
 dnl  else
 dnl   dv_check_lib_ok=no
 dnl =====================================================================
-AC_DEFUN(DV_CHECK_LIB,
+AC_DEFUN([DV_CHECK_LIB],
 [
 AC_LANG_PUSH(C++)
 dv_check_lib_ok=no;
@@ -323,7 +323,7 @@
 dnl =====================================================================
 dnl Tries DV_CHECK_LIB and exits if not succesful.
 dnl =====================================================================
-AC_DEFUN(DV_REQUIRE_LIB,
+AC_DEFUN([DV_REQUIRE_LIB],
 [
 AC_MSG_CHECKING([whether required lib$3.a is available])
 
@@ -352,7 +352,7 @@
 dnl =====================================================================
 dnl #define __restrict as empty string if it leads to compile errors.
 dnl =====================================================================
-AC_DEFUN(DV_CHECK_RESTRICT,
+AC_DEFUN([DV_CHECK_RESTRICT],
 [
 AC_MSG_CHECKING([whether __restrict is implemented correctly])
 
@@ -376,7 +376,7 @@
 dnl Find out where program-name is, checking directories in bin-dirs.
 dnl Set pathvar if found and abort if not found
 dnl =====================================================================
-AC_DEFUN(DV_REQUIRE_PROG,
+AC_DEFUN([DV_REQUIRE_PROG],
 [
 $1=""
 AM_PATH_PROG_WITH_TEST($1,$2,true, ,[$3])
@@ -385,7 +385,7 @@
 fi
 ])
 
-AC_DEFUN(DV_APPEND_UNIQ,
+AC_DEFUN([DV_APPEND_UNIQ],
 [
 { echo "${$2}" | grep -- "$1 " >/dev/null || $2="${$2} $1"; }dnl
 ])
