diff -ru bitstreamout-0.85-orig/types.h bitstreamout-0.85/types.h
--- bitstreamout-0.85-orig/types.h	2007-01-05 15:10:51.063602883 +0100
+++ bitstreamout-0.85/types.h	2007-01-05 15:11:55.619413470 +0100
@@ -26,7 +26,13 @@
 #include <sys/mman.h>
 #include <asm/types.h>
 #include <sys/types.h>
-#if defined(__i386__) || defined(__x86_64__)
+/*
+ * #if defined(__i386__) || defined(__x86_64__)
+ * #define HAVE_BITOPTS 1
+ * #endif
+ */
+
+#ifdef HAVE_BITOPTS
 # include <asm/bitops.h>
 #else
 # define clear_bit(n,addr)  (*(addr) &= ~(1 << (n)))
