From 6638f341240231001f6a3aeb92faeae4e9fefa7a Mon Sep 17 00:00:00 2001
From: Philippe Rouquier <bonfire-app@wanadoo.fr>
Date: Wed, 24 Nov 2010 14:29:01 +0000
Subject: Silenced a warning

---
diff --git a/libbrasero-burn/brasero-session.c b/libbrasero-burn/brasero-session.c
index fa67cca..73d8642 100644
--- a/libbrasero-burn/brasero-session.c
+++ b/libbrasero-burn/brasero-session.c
@@ -2059,7 +2059,7 @@ brasero_burn_session_get_src_medium (BraseroBurnSession *self)
 		return NULL;
 
 	track = priv->tracks->data;
-	if (!BRASERO_TRACK_DISC (track))
+	if (!BRASERO_IS_TRACK_DISC (track))
 		return NULL;
 
 	drive = brasero_track_disc_get_drive (BRASERO_TRACK_DISC (track));
--
cgit v0.9
