--- Python/configure.py
+++ Python/configure.py
@@ -312,7 +312,8 @@
             universal=pyqt.universal,
             arch=pyqt.arch,
             prot_is_public=opts.prot_is_public,
-            deployment_target=pyqt.deployment_target
+            deployment_target=pyqt.deployment_target,
+            strip=0
         )
     else:
         makefile = Makefile(
@@ -321,7 +322,8 @@
             install_dir=opts.qscimoddir,
             installs=installs,
             static=opts.static,
-            debug=opts.debug
+            debug=opts.debug,
+            strip=0
         )
 
     if qsci_define:
