diff -urNp nettle-1.13.org/configure.ac nettle-1.13/configure.ac
--- nettle-1.13.org/configure.ac	2005-10-06 22:29:11.000000000 +0300
+++ nettle-1.13/configure.ac	2006-10-31 21:19:24.000000000 +0200
@@ -152,7 +152,7 @@ case `uname -sr` in
     SHLIBFILE='$(SHLIBSONAME).$(SHLIBMINOR)'
     SHLIBLINK='$(CC) $(LDFLAGS) -shared -Wl,-soname=$(SHLIBSONAME)'
     # FIXME: It might make some sense to link with -lgmp
-    SHLIBLIBS=''
+    SHLIBLIBS='$(LIBS)'
     ;;
 esac
 
