diff -Naur nettle-2.4.orig//examples/Makefile.in nettle-2.4/examples/Makefile.in
--- nettle-2.4.orig//examples/Makefile.in	2011-09-03 12:51:07.000000000 +0000
+++ nettle-2.4/examples/Makefile.in	2012-05-08 23:49:01.580003605 +0000
@@ -68,7 +68,7 @@
 
 nettle-benchmark$(EXEEXT): nettle-benchmark.$(OBJEXT) nettle-openssl.$(OBJEXT) $(GETOPT_OBJS)
 	$(LINK) nettle-benchmark.$(OBJEXT) nettle-openssl.$(OBJEXT) io.$(OBJEXT) $(GETOPT_OBJS) \
-	-lnettle $(BENCH_LIBS) $(OPENSSL_LIBFLAGS) -o nettle-benchmark$(EXEEXT)
+	-lnettle -lm $(BENCH_LIBS) $(OPENSSL_LIBFLAGS) -o nettle-benchmark$(EXEEXT)
 
 $(TARGETS) : io.$(OBJEXT) ../libnettle.a
 
