diff -ur graphtft-0.2.2.orig/imlibrenderer/dvbrenderer/mpeg2encoder.c graphtft-0.2.2/imlibrenderer/dvbrenderer/mpeg2encoder.c
--- graphtft-0.2.2.orig/imlibrenderer/dvbrenderer/mpeg2encoder.c	2008-11-01 10:02:54.000000000 +0200
+++ graphtft-0.2.2/imlibrenderer/dvbrenderer/mpeg2encoder.c	2009-07-24 21:11:58.000000000 +0300
@@ -126,11 +126,11 @@
    int i;
    AVPicture avpsrc;
 
-   avpicture_fill(&avpsrc, buf, PIX_FMT_RGBA32, width, height);
+   avpicture_fill(&avpsrc, buf, PIX_FMT_RGBA, width, height);
 
 #ifndef HAVE_SWSCALE
 
-   img_convert(&pic, PIX_FMT_YUV420P, &avpsrc, PIX_FMT_RGBA32, width, height);
+   img_convert(&pic, PIX_FMT_YUV420P, &avpsrc, PIX_FMT_RGBA, width, height);
 
 #else
 
diff -ur graphtft-0.2.2.orig/imlibrenderer/fbrenderer/fbrenderer.c graphtft-0.2.2/imlibrenderer/fbrenderer/fbrenderer.c
--- graphtft-0.2.2.orig/imlibrenderer/fbrenderer/fbrenderer.c	2008-11-01 10:02:05.000000000 +0200
+++ graphtft-0.2.2/imlibrenderer/fbrenderer/fbrenderer.c	2009-07-24 21:11:39.000000000 +0300
@@ -192,7 +192,7 @@
 
 	switch (fb_vinfo.bits_per_pixel) 
    {
-		case 32: tell(4, "fbdevout.c: using 32 bit depth"); fb_type = PIX_FMT_RGBA32; break;
+		case 32: tell(4, "fbdevout.c: using 32 bit depth"); fb_type = PIX_FMT_RGBA; break;
 		case 24: tell(4, "fbdevout.c: using 24 bit depth"); fb_type = PIX_FMT_RGB24;  break;
 		case 16: tell(4, "fbdevout.c: using 16 bit depth"); fb_type = PIX_FMT_RGB565; break;
 		default: tell(4, "fbdevout.c: color depth not supported -> %i bits per pixel",
