Index: graphtft-30/imlibrenderer/dvbrenderer/mpeg2encoder.c
===================================================================
--- graphtft-30.orig/imlibrenderer/dvbrenderer/mpeg2encoder.c
+++ graphtft-30/imlibrenderer/dvbrenderer/mpeg2encoder.c
@@ -51,7 +51,6 @@ void mpeg_init(int video_width, int vide
    int i = 0;
    int status;
 
-   avcodec_init();
    avcodec_register_all();
     
    codec = avcodec_find_encoder(CODEC_ID_MPEG1VIDEO);
