diff -Naur graphtft-30.orig/dspitems.c graphtft-30/dspitems.c
--- graphtft-30.orig/dspitems.c	2012-04-07 03:09:26.000000000 +0200
+++ graphtft-30/dspitems.c	2012-04-07 03:11:03.000000000 +0200
@@ -397,7 +397,7 @@
          else if (strcasecmp(var, "Path") == 0)
             return Str::notNull(recording->FileName());
          else if (strcasecmp(var, "Time") == 0)
-            return formatDateTime(recording->start, fmt, buf, sizeof(buf));
+            return formatDateTime(recording->Start(), fmt, buf, sizeof(buf));
          else if (strcasecmp(var, "EventID") == 0)
             return Str::toStr(!recording->Info() ? na : (int)recording->Info()->EventID());
          else if (strcasecmp(var, "SubTitle") == 0)
diff -Naur graphtft-30.orig/graphtft.c graphtft-30/graphtft.c
--- graphtft-30.orig/graphtft.c	2012-04-07 03:16:05.000000000 +0200
+++ graphtft-30/graphtft.c	2012-04-07 03:19:52.000000000 +0200
@@ -311,7 +311,7 @@
    GraphTFTSetup.PluginConfPath = strdup(DATA_DIR);
 
 #if APIVERSNUM < 10507
-   RegisterI18n(Phrases);
+//   RegisterI18n(Phrases);
 #endif
 
    display = new cGraphTFTDisplay(THEMEVERSION);
diff -Naur graphtft-30.orig/graphtft.h graphtft-30/graphtft.h
--- graphtft-30.orig/graphtft.h	2012-04-07 03:16:05.000000000 +0200
+++ graphtft-30/graphtft.h	2012-04-07 03:21:04.000000000 +0200
@@ -16,7 +16,7 @@
 #include <vdr/config.h>
 
 #if APIVERSNUM < 10507
-#  include "i18n.h"
+//#  include "i18n.h"
 #  define trNOOP(s) (s)
 #endif
 
diff -Naur graphtft-30.orig/Makefile graphtft-30/Makefile
--- graphtft-30.orig/Makefile	2012-04-07 03:16:05.000000000 +0200
+++ graphtft-30/Makefile	2012-04-07 03:18:44.000000000 +0200
@@ -240,7 +240,7 @@
 ### The object files (add further files here):
 
 COMMONOBJS = $(PLUGIN).o dspitems.o display.o \
-	i18n.o setup.o osd.o scan.o theme.o common.o sysinfo.o \
+	setup.o osd.o scan.o theme.o common.o sysinfo.o \
    touchthread.o
 
 # transfer.o
@@ -311,10 +311,10 @@
 
 ### Targets:
 
-all: libvdr-$(PLUGIN).so i18n
+all: libvdr-$(PLUGIN).so
 	@cp libvdr-$(PLUGIN).so $(LIBDIR)/libvdr-$(PLUGIN).so.$(APIVERSION)
 
-alli: libvdr-$(PLUGIN).so i18n
+alli: libvdr-$(PLUGIN).so
 
 libvdr-$(PLUGIN).so: $(COMMONOBJS) $(IMLIBOBJS) $(DFBOBJS)
 	$(CXX) $(CXXFLAGS) $(LDFLAGS) -shared $^ $(LIBS) -o $@
diff -Naur graphtft-30.orig/setup.h graphtft-30/setup.h
--- graphtft-30.orig/setup.h	2012-04-07 03:16:05.000000000 +0200
+++ graphtft-30/setup.h	2012-04-07 03:21:41.000000000 +0200
@@ -22,7 +22,7 @@
 #include "theme.h"
 
 #if APIVERSNUM < 10507
-#  include "i18n.h"
+//#  include "i18n.h"
 #endif
 
 //***************************************************************************
