--- bash-3.0/config.h.in
+++ bash-3.0/config.h.in
@@ -197,7 +197,7 @@
 
 /* System paths */
 
-#define DEFAULT_MAIL_DIRECTORY "/usr/spool/mail"
+#define DEFAULT_MAIL_DIRECTORY "/var/spool/mail"
 
 /* Characteristics of the system's header files and libraries that affect
    the compilation environment. */
--- bash-3.0/config-bot.h
+++ bash-3.0/config-bot.h
@@ -178,4 +178,7 @@
 /******************************************************************/
 
 /* If you don't want bash to provide a default mail file to check. */
-/* #undef DEFAULT_MAIL_DIRECTORY */
+/* DP: - don't define a default DEFAULT_MAIL_DIRECTORY, because it
+ * DP:   can cause a timeout on NFS mounts.
+ */
+#undef DEFAULT_MAIL_DIRECTORY
