diff -Naur gonzui-1.2.orig/Makefile.am gonzui-1.2/Makefile.am
--- gonzui-1.2.orig/Makefile.am	2005-03-25 18:00:20.000000000 +0900
+++ gonzui-1.2/Makefile.am	2009-01-01 11:28:13.000000000 +0900
@@ -28,7 +28,7 @@
 		gonzui.rb.in > gonzui.rb
 	chmod -w gonzui.rb
 
-gonzuirc.sample.in: gonzuirc.sample.in.in
+gonzuirc.sample.in: gonzui.rb gonzuirc.sample.in.in
 	ruby gonzui-db --gonzuirc gonzuirc.sample.in.in --dump-config \
 		> gonzuirc.sample.in
 
diff -Naur gonzui-1.2.orig/configure.ac gonzui-1.2/configure.ac
--- gonzui-1.2.orig/configure.ac	2005-07-05 00:07:52.000000000 +0900
+++ gonzui-1.2/configure.ac	2009-01-01 11:28:13.000000000 +0900
@@ -9,7 +9,6 @@
 AC_PROG_CC
 AC_PROG_INSTALL
 AC_PROG_MAKE_SET
-AM_MAINTAINER_MODE
 AM_PATH_RUBY
 AM_RUN_LOG_DIRS
 AM_USER_GROUP
diff -Naur gonzui-1.2.orig/langscan/Makefile.am gonzui-1.2/langscan/Makefile.am
--- gonzui-1.2.orig/langscan/Makefile.am	2005-07-05 00:07:57.000000000 +0900
+++ gonzui-1.2/langscan/Makefile.am	2009-01-01 11:28:13.000000000 +0900
@@ -26,6 +26,7 @@
 		brainfuck.rb \
 		c.rb \
 		css.rb \
+		ebuild.rb \
 		elisp.rb \
 		java.rb \
 		javascript.rb \
diff -Naur gonzui-1.2.orig/langscan/python/Makefile.am gonzui-1.2/langscan/python/Makefile.am
--- gonzui-1.2.orig/langscan/python/Makefile.am	2005-03-20 21:08:58.000000000 +0900
+++ gonzui-1.2/langscan/python/Makefile.am	2009-01-01 11:28:39.000000000 +0900
@@ -27,7 +27,7 @@
 
 # We do this to workaround flex 2.5.4 issue because our .l
 # file could not be processed with flex 2.5.4.
-pythontok.c: pythontok.l
+pythontok.c: python.h pythontok.l
 	flex -d -opythontok.c pythontok.l || cp pythontok.c.dist pythontok.c
 	cmp -s pythontok.c pythontok.c.dist || cp pythontok.c pythontok.c.dist
 
