--- Makefile.am
+++ Makefile.am
@@ -28,7 +28,7 @@
 		gonzui.rb.in > gonzui.rb
 	chmod -w gonzui.rb
 
-gonzuirc.sample.in: gonzuirc.sample.in.in
+gonzuirc.sample.in: gonzui.rb gonzuirc.sample.in.in
 	ruby gonzui-db --gonzuirc gonzuirc.sample.in.in --dump-config \
 		> gonzuirc.sample.in
 
--- Makefile.in
+++ Makefile.in
@@ -671,7 +671,7 @@
 		gonzui.rb.in > gonzui.rb
 	chmod -w gonzui.rb
 
-gonzuirc.sample.in: gonzuirc.sample.in.in
+gonzuirc.sample.in: gonzui.rb gonzuirc.sample.in.in
 	ruby gonzui-db --gonzuirc gonzuirc.sample.in.in --dump-config \
 		> gonzuirc.sample.in
 
