diff -Naur gmetadom-0.2.6.orig/configure.ac gmetadom-0.2.6/configure.ac
--- gmetadom-0.2.6.orig/configure.ac	2007-10-15 18:42:23.000000000 +0900
+++ gmetadom-0.2.6/configure.ac	2008-01-17 23:41:17.000000000 +0900
@@ -312,7 +312,7 @@
 	HAVE_OCAMLC=no
 else
 	HAVE_OCAMLC=yes
-	OCAML_LIB=`$OCAMLC -v | tail -1 | cut -d ':' -f2 | tr -d ' '`
+	OCAML_LIB=`$OCAMLC -v | tail -n 1 | cut -d ':' -f2 | tr -d ' '`
 
 	if test $OCAML_INCLUDE_PREFIX = no; then
 		OCAML_CFLAGS=-I$OCAML_LIB
diff -Naur gmetadom-0.2.6.orig/src/gdome_caml/Makefile.am gmetadom-0.2.6/src/gdome_caml/Makefile.am
--- gmetadom-0.2.6.orig/src/gdome_caml/Makefile.am	2006-08-06 17:33:36.000000000 +0900
+++ gmetadom-0.2.6/src/gdome_caml/Makefile.am	2008-01-18 08:48:01.000000000 +0900
@@ -4,7 +4,7 @@
 ARCHIVE = mlgdome
 PKGNAME = gdome2
 OCAMLBUILDDIR = $(top_builddir)/src/gdome_caml
-OCAMLINSTALLDIR = $(DESTDIR)$(OCAML_LIB_PREFIX)/$(PKGNAME)
+OCAMLINSTALLDIR = $(DESTDIR)$(OCAML_LIB_PREFIX)/site-packages/$(PKGNAME)
 STUBSDIR = $(DESTDIR)$(OCAML_LIB_PREFIX)/stublibs
 OCAMLFLAGS = \
   -I $(OCAMLBUILDDIR)/types \
@@ -82,8 +82,8 @@
 	done
 	if [ "x$(OCAMLFIND)" != "x" ]; then \
 		mv $(OCAMLINSTALLDIR) $(OCAMLINSTALLDIR).saved &&	\
-		$(mkinstalldirs) $(DESTDIR)$(OCAML_LIB_PREFIX)/ && \
-		$(OCAMLFIND) install -destdir $(DESTDIR)$(OCAML_LIB_PREFIX)/ $(PKGNAME) META $(DLL) && \
+		$(mkinstalldirs) $(DESTDIR)$(OCAML_LIB_PREFIX)/site-packages/ && \
+		$(OCAMLFIND) install -ldconf ignore -destdir $(DESTDIR)$(OCAML_LIB_PREFIX)/site-packages/ $(PKGNAME) META $(DLL) && \
 		$(INSTALL_DATA) $(OCAMLINSTALLDIR).saved/* $(OCAMLINSTALLDIR)/ &&	\
 		rm -rf $(OCAMLINSTALLDIR).saved/;	\
 	else	\
@@ -92,7 +92,7 @@
 
 uninstall-local:
 	if [ "x$(OCAMLFIND)" != "x" ]; then	\
-	  $(OCAMLFIND) remove -destdir $(DESTDIR)$(OCAML_LIB_PREFIX)/ $(PKGNAME);	\
+	  $(OCAMLFIND) remove -destdir $(DESTDIR)$(OCAML_LIB_PREFIX)/site-packages/ $(PKGNAME);	\
 	else	\
 		rm -rf $(OCAMLINSTALLDIR);	\
 		rm $(STUBSDIR)/$(DLL);	\
diff -Naur gmetadom-0.2.6.orig/src/gdome_caml/basic/Makefile.am gmetadom-0.2.6/src/gdome_caml/basic/Makefile.am
--- gmetadom-0.2.6.orig/src/gdome_caml/basic/Makefile.am	2003-09-13 23:14:42.000000000 +0900
+++ gmetadom-0.2.6/src/gdome_caml/basic/Makefile.am	2008-01-18 08:57:33.000000000 +0900
@@ -1,6 +1,6 @@
 
 OCAMLBUILDDIR = $(top_builddir)/src/gdome_caml
-OCAMLINSTALLDIR = $(DESTDIR)$(OCAML_LIB_PREFIX)/gdome2
+OCAMLINSTALLDIR = $(DESTDIR)$(OCAML_LIB_PREFIX)/site-packages/gdome2
 OCAMLFLAGS = -I $(OCAMLBUILDDIR) -I $(OCAMLBUILDDIR)/core \
              -I $(OCAMLBUILDDIR)/types
 OCAMLC = @OCAMLC@ ${OCAMLFLAGS}
diff -Naur gmetadom-0.2.6.orig/src/gdome_caml/core/Makefile.am gmetadom-0.2.6/src/gdome_caml/core/Makefile.am
--- gmetadom-0.2.6.orig/src/gdome_caml/core/Makefile.am	2007-10-12 23:12:51.000000000 +0900
+++ gmetadom-0.2.6/src/gdome_caml/core/Makefile.am	2008-01-18 08:57:33.000000000 +0900
@@ -1,6 +1,6 @@
 
 OCAMLBUILDDIR = $(top_builddir)/src/gdome_caml
-OCAMLINSTALLDIR = $(DESTDIR)$(OCAML_LIB_PREFIX)/gdome2
+OCAMLINSTALLDIR = $(DESTDIR)$(OCAML_LIB_PREFIX)/site-packages/gdome2
 OCAMLFLAGS = -I $(OCAMLBUILDDIR)/types -I $(OCAMLBUILDDIR)/basic
 OCAMLC = @OCAMLC@ ${OCAMLFLAGS}
 OCAMLOPT = @OCAMLOPT@ ${OCAMLFLAGS}
diff -Naur gmetadom-0.2.6.orig/src/gdome_caml/events/Makefile.am gmetadom-0.2.6/src/gdome_caml/events/Makefile.am
--- gmetadom-0.2.6.orig/src/gdome_caml/events/Makefile.am	2007-10-12 23:12:51.000000000 +0900
+++ gmetadom-0.2.6/src/gdome_caml/events/Makefile.am	2008-01-18 08:57:33.000000000 +0900
@@ -1,6 +1,6 @@
 
 OCAMLBUILDDIR = $(top_builddir)/src/gdome_caml
-OCAMLINSTALLDIR = $(DESTDIR)$(OCAML_LIB_PREFIX)/gdome2
+OCAMLINSTALLDIR = $(DESTDIR)$(OCAML_LIB_PREFIX)/site-packages/gdome2
 OCAMLFLAGS = -I $(OCAMLBUILDDIR)/types -I $(OCAMLBUILDDIR)/basic
 OCAMLC = @OCAMLC@ ${OCAMLFLAGS}
 OCAMLOPT = @OCAMLOPT@ ${OCAMLFLAGS}
diff -Naur gmetadom-0.2.6.orig/src/gdome_caml/include/Makefile.am gmetadom-0.2.6/src/gdome_caml/include/Makefile.am
--- gmetadom-0.2.6.orig/src/gdome_caml/include/Makefile.am	2002-08-31 03:03:36.000000000 +0900
+++ gmetadom-0.2.6/src/gdome_caml/include/Makefile.am	2008-01-18 08:57:33.000000000 +0900
@@ -1,5 +1,5 @@
 
-OCAMLINSTALLDIR = $(DESTDIR)$(OCAML_LIB_PREFIX)/gdome2
+OCAMLINSTALLDIR = $(DESTDIR)$(OCAML_LIB_PREFIX)/site-packages/gdome2
 EXTRA_DIST = mlgdomevalue.h
 
 install-data-local: mlgdomevalue.h
diff -Naur gmetadom-0.2.6.orig/src/gdome_caml/ocaml/Makefile.am gmetadom-0.2.6/src/gdome_caml/ocaml/Makefile.am
--- gmetadom-0.2.6.orig/src/gdome_caml/ocaml/Makefile.am	2007-10-12 23:12:51.000000000 +0900
+++ gmetadom-0.2.6/src/gdome_caml/ocaml/Makefile.am	2008-01-18 08:57:33.000000000 +0900
@@ -1,6 +1,6 @@
 
 OCAMLBUILDDIR = $(top_builddir)/src/gdome_caml
-OCAMLINSTALLDIR = $(DESTDIR)$(OCAML_LIB_PREFIX)/gdome2
+OCAMLINSTALLDIR = $(DESTDIR)$(OCAML_LIB_PREFIX)/site-packages/gdome2
 
 OCAMLFLAGS = \
   -I $(OCAMLBUILDDIR) \
diff -Naur gmetadom-0.2.6.orig/src/gdome_caml/types/Makefile.am gmetadom-0.2.6/src/gdome_caml/types/Makefile.am
--- gmetadom-0.2.6.orig/src/gdome_caml/types/Makefile.am	2007-10-12 23:12:51.000000000 +0900
+++ gmetadom-0.2.6/src/gdome_caml/types/Makefile.am	2008-01-18 08:57:33.000000000 +0900
@@ -1,7 +1,7 @@
 
 OCAMLC = @OCAMLC@
 OCAMLDEP = @OCAMLDEP@
-OCAMLINSTALLDIR = $(DESTDIR)$(OCAML_LIB_PREFIX)/gdome2
+OCAMLINSTALLDIR = $(DESTDIR)$(OCAML_LIB_PREFIX)/site-packages/gdome2
 
 T_AUTO_MLI = \
   tNode.mli \
diff -Naur gmetadom-0.2.6.orig/src/gdome_cpp_smart/test/Makefile.am gmetadom-0.2.6/src/gdome_cpp_smart/test/Makefile.am
--- gmetadom-0.2.6.orig/src/gdome_cpp_smart/test/Makefile.am	2007-05-09 00:26:18.000000000 +0900
+++ gmetadom-0.2.6/src/gdome_cpp_smart/test/Makefile.am	2008-01-17 23:41:17.000000000 +0900
@@ -7,7 +7,6 @@
 
 LDADDS = \
   $(GDOME_LIBS) \
-  $(top_builddir)/src/gdome_cpp_smart/.libs/libgmetadom_gdome_cpp_smart.a \
   $(top_builddir)/src/gdome_cpp_smart/libgmetadom_gdome_cpp_smart.la \
   -lstdc++
 
