diff -ur vm-7.19.orig/base64-decode.c vm-7.19/base64-decode.c
--- vm-7.19.orig/base64-decode.c	1997-09-18 05:06:05.000000000 +0200
+++ vm-7.19/base64-decode.c	2007-02-24 08:57:57.000000000 +0100
@@ -3,6 +3,7 @@
 /* BASE64 on stdin -> converted data on stdout */
 
 #include <stdio.h>
+#include <stdlib.h>
 
 #ifdef _WIN32
 #ifndef WIN32
diff -ur vm-7.19.orig/base64-encode.c vm-7.19/base64-encode.c
--- vm-7.19.orig/base64-encode.c	1997-09-18 05:06:05.000000000 +0200
+++ vm-7.19/base64-encode.c	2007-02-24 08:57:54.000000000 +0100
@@ -7,6 +7,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 
 #ifdef _WIN32
 #ifndef WIN32
diff -ur vm-7.19.orig/qp-decode.c vm-7.19/qp-decode.c
--- vm-7.19.orig/qp-decode.c	2000-03-05 21:59:04.000000000 +0100
+++ vm-7.19/qp-decode.c	2007-02-24 08:57:46.000000000 +0100
@@ -4,6 +4,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <stdlib.h>
 
 #ifdef _WIN32
 #ifndef WIN32
diff -ur vm-7.19.orig/qp-encode.c vm-7.19/qp-encode.c
--- vm-7.19.orig/qp-encode.c	2000-11-30 22:19:52.000000000 +0100
+++ vm-7.19/qp-encode.c	2007-02-24 08:57:35.000000000 +0100
@@ -7,6 +7,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 
 #ifdef _WIN32
 #ifndef WIN32
