--- vm-digest.el~	2003-05-03 23:31:52.000000000 +0200
+++ vm-digest.el	2005-10-06 16:41:56.000000000 +0200
@@ -179,6 +179,7 @@
 		    (goto-char start)
 		    (while (= (following-char) ?\n)
 		      (delete-char 1)))
+		  (insert ?\n)
 		  (insert (vm-trailing-message-separator folder-type)))
 		 ((vm-mime-types-match "multipart/digest"
 				       (car (vm-mm-layout-type layout)))
@@ -202,6 +203,7 @@
 		      (goto-char start)
 		      (while (= (following-char) ?\n)
 			(delete-char 1)))
+		    (insert ?\n)
 		    (insert (vm-trailing-message-separator folder-type))
 		    (setq part-list (cdr part-list))))
 		 (t (error
