--- vm-8.0.12-orig/lisp/vm-pine.el	2008-11-05 21:04:35.000000000 +0100
+++ vm-8.0.12/lisp/vm-pine.el	2008-11-09 19:05:15.000000000 +0100
@@ -488,6 +488,7 @@
         (error "Folder buffer closed before deletion of source message."))))
 
 ;;-----------------------------------------------------------------------------
+;;;###autoload
 (defun vm-decode-postponed-mime-message ()
   "Replace the mime buttons by attachment buttons."
   (interactive)
