http://bugs.gentoo.org/256886

--- vm-8.0.12-orig/lisp/vm-reply.el	2008-11-05 21:04:35.000000000 +0100
+++ vm-8.0.12/lisp/vm-reply.el	2009-11-03 08:14:30.000000000 +0100
@@ -343,6 +343,12 @@
             (vm-insert-region-from-buffer (vm-buffer-of message)
                                           (vm-headers-of message)
                                           (vm-text-end-of message))
+	  (vm-insert-region-from-buffer (vm-buffer-of message)
+					(vm-headers-of message)
+					(vm-text-of message))
+	  (save-excursion
+	    (goto-char (point-min))
+	    (vm-decode-mime-message-headers))
           (vm-decode-mime-layout layout)
           (if vm-mime-yank-attachments
               (vm-decode-postponed-mime-message))))
