From 39f51ac87536459966c87050f66c3913fed654d8 Mon Sep 17 00:00:00 2001
From: Eugene Shalygin <eugene.shalygin@gmail.com>
Date: Mon, 18 Mar 2013 03:56:04 +1100
Subject: [PATCH] Fix crash when moving stream across devices.

REVIEW: 109293
BUG: 311167
---
 gui/mdwslider.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/gui/mdwslider.cpp b/gui/mdwslider.cpp
index af6e725..b09909d 100644
--- a/gui/mdwslider.cpp
+++ b/gui/mdwslider.cpp
@@ -1217,7 +1217,7 @@ void MDWSlider::showMoveMenu()
     KAction *a = new KAction(_mdwMoveActions);
     a->setText( i18n("Automatic According to Category") );
     _mdwMoveActions->addAction( QString("moveautomatic"), a);
-    connect(a, SIGNAL(triggered(bool)), SLOT(moveStreamAutomatic()));
+    connect(a, SIGNAL(triggered(bool)), SLOT(moveStreamAutomatic()), Qt::QueuedConnection);
     m_moveMenu->addAction( a );
 
     a = new KAction(_mdwMoveActions);
@@ -1229,7 +1229,7 @@ void MDWSlider::showMoveMenu()
     {
         a = new MDWMoveAction(md, _mdwMoveActions);
         _mdwMoveActions->addAction( QString("moveto") + md->id(), a);
-        connect(a, SIGNAL(moveRequest(QString)), SLOT(moveStream(QString)));
+        connect(a, SIGNAL(moveRequest(QString)), SLOT(moveStream(QString)), Qt::QueuedConnection);
         m_moveMenu->addAction( a );
     }
 }
-- 
1.8.1.5

