--- musa/m68kmake.c.orig	2006-05-13 12:52:38.000000000 +0100
+++ musa/m68kmake.c	2006-05-13 12:53:05.000000000 +0100
@@ -1029,10 +1029,10 @@
 	fprintf(filep, "/* ========================= OPCODE TABLE BUILDER ========================= */\n");
 	fprintf(filep, "/* ======================================================================== */\n\n");
 
-	fprintf(filep, "#include \"m68kops.h\"\n");
-	fprintf(filep, "#include \"m68kcpu.h\"\n");
 	fprintf(filep, "#include <stdlib.h>\n\n");
 	fprintf(filep, "#include <string.h>\n\n");
+	fprintf(filep, "#include \"m68kops.h\"\n");
+	fprintf(filep, "#include \"m68kcpu.h\"\n");
 
 	fprintf(filep, "extern void  (*m68k_instruction_jump_table[])(void); /* opcode handler jump table */\n\n");
 
