Index: omake-0.9.8.6/lib/build/OCaml.om
===================================================================
--- omake-0.9.8.6.orig/lib/build/OCaml.om
+++ omake-0.9.8.6/lib/build/OCaml.om
@@ -176,7 +176,7 @@ public.PREFIXED_OCAMLPACKS =\
 #
 declare OCAMLDEPFLAGS
 public.OCAMLPPFLAGS  =
-public.OCAMLFLAGS    = -warn-error A
+public.OCAMLFLAGS    =
 public.OCAMLCFLAGS   = -g
 public.OCAMLOPTFLAGS =
 public.OCAMLCPPFLAGS =
