From: Julian Ospald <hasufell@gentoo.org>
Date: Tue Oct 23 15:26:08 UTC 2012
Subject: fix desktop file

--- ffDiaporama.desktop
+++ ffDiaporama.desktop
@@ -1,6 +1,5 @@
 [Desktop Entry]
 Version=1.0
-Encoding=UTF-8
 Name=Movie creator ffDiaporama
 Name[fr]=Créateur de film ffDiaporama
 Name[it]=Movie creator ffDiaporama
@@ -16,10 +15,9 @@
 Comment[it]=Creatore di pellicole cinematografiche da foto e filmati video
 Comment[es]=Creador de películas a partir de fotos y clips de vídeo
 Comment[el]=Δημιουργός ταινιών από φωτογραφίες και βίντεο κλιπ
-Icon=/usr/share/ffDiaporama/application-ffDiaporama.png
+Icon=application-ffDiaporama
 Terminal=false
 Type=Application
-Path=/usr/share/ffDiaporama
-Exec=/usr/bin/ffDiaporama %F
+Exec=ffDiaporama %F
 MimeType=application/ffDiaporama;
-Categories=GNOME;AudioVideo;Qt;KDE;AudioVideo;
+Categories=GTK;GNOME;Qt;KDE;AudioVideo;
