--- liboping-1.6.1/src/Makefile.am	2011-03-06 10:03:21.000000000 +0100
+++ liboping-1.6.0/src/Makefile.am	2011-01-26 14:18:42.000000000 +0100
@@ -51,16 +51,3 @@
 noping_LDADD += -lrt
 endif
 endif # BUILD_WITH_LIBNCURSES
-
-install-exec-hook:
-	@if test "x0" = "x$$UID"; then \
-		if test "xLinux" = "x`uname -s`"; then \
-			echo "Setting CAP_NET_RAW capability on binaries."; \
-			setcap cap_net_raw=ep $(DESTDIR)$(bindir)/oping; \
-			setcap cap_net_raw=ep $(DESTDIR)$(bindir)/noping; \
-		else \
-			echo "Setting set-UID bit on binaries."; \
-			chmod u+s $(DESTDIR)$(bindir)/oping; \
-			chmod u+s $(DESTDIR)$(bindir)/noping; \
-		fi; \
-	fi
