From 11bfe822fc16a06d8eacacd36a069ccf9dbc1201 Mon Sep 17 00:00:00 2001
From: 李智豪 <zhihaoli.scut@gmail.com>
Date: Thu, 05 Nov 2009 22:30:21 +0000
Subject: Read correct argument when parsing alignment

---
diff --git a/src/plugin.c b/src/plugin.c
index 857d733..9a50cdc 100644
--- a/src/plugin.c
+++ b/src/plugin.c
@@ -289,7 +289,7 @@ plugin_new (NPMIMEType mime_type, NPP instance,
       guint j;
 
       for (j = 0; j < G_N_ELEMENTS (possibilities); j++) {
-	if (g_ascii_strcasecmp (argv[j], possibilities[j].name) == 0) {
+	if (g_ascii_strcasecmp (argv[i], possibilities[j].name) == 0) {
 	  align = possibilities[j].align;
 	  break;
 	}
--
cgit v0.8.3-6-g21f6
