--- Makefile.in.in.orig	2004-11-22 18:23:51.920290360 -0200
+++ Makefile.in.in	2004-11-22 18:24:39.144111248 -0200
@@ -123,7 +123,7 @@
 	    *)     destdir=$(localedir);; \
 	  esac; \
 	  lang=`echo $$cat | sed 's/\$(CATOBJEXT)$$//'`; \
-	  dir=$$destdir/$$lang/LC_MESSAGES; \
+	  dir=$(DESTDIR)/$$destdir/$$lang/LC_MESSAGES; \
 	  if test -r "$(MKINSTALLDIRS)"; then \
 	    $(MKINSTALLDIRS) $$dir; \
 	  else \
