--- src/main.cc.orig	2008-06-15 16:22:24.000000000 +0000
+++ src/main.cc	2008-06-15 16:22:40.000000000 +0000
@@ -32,6 +32,8 @@
 #include <string>
 #include <vector>
 #include <cerrno>
+#include <cstdlib>
+#include <cstring>
 
 #include <sys/stat.h> // for stat()
 #include <unistd.h> // for stat()

--- src/libbab.cc.orig	2008-06-15 16:28:41.000000000 +0000
+++ src/libbab.cc	2008-06-15 16:29:03.000000000 +0000
@@ -30,6 +30,8 @@
 #include <list>
 #include <cctype>
 #include <cerrno>
+#include <cstring>
+
 
 #ifdef ENABLE_NLS
 #include <libintl.h>

