--- gprolog-1.4.0.orig/src/Makefile.in	2011-04-08 18:35:22.000000000 +0200
+++ gprolog-1.4.0/src/Makefile.in	2012-07-04 09:46:49.000000000 +0200
@@ -81,10 +81,10 @@
 
 # --- Links --- #
 
-install-links: uninstall-links
+install-links:
 	if test $(LINKS_DIR) != none; then \
 	   ./mkinstalldirs $(LINKS_DIR); \
-	   (cd $(LINKS_DIR) ; $(LN_S) $(INSTALL_DIR)/bin/* .); \
+	   (cd $(LINKS_DIR); for i in $(BIN_FILES); do $(LN_S) $(INSTALL_DIR)/bin/$$i .; done); \
 	fi
 
 uninstall-links:
