diff -ur gprolog-1.4.2.orig/src/Makefile.in gprolog-1.4.2/src/Makefile.in
--- gprolog-1.4.2.orig/src/Makefile.in	2012-10-02 17:55:53.000000000 +0200
+++ gprolog-1.4.2/src/Makefile.in	2013-01-09 20:12:15.000000000 +0100
@@ -60,7 +60,6 @@
 install-system:
 	./mkinstalldirs $(INSTALL_DIR) $(INSTALL_DIR)/bin \
 	$(INSTALL_DIR)/include $(INSTALL_DIR)/lib
-	for i in $(TXT_FILES); do $(INSTALL_DATA) ../$$i $(INSTALL_DIR); done
 	for i in $(BIN_FILES); do $(INSTALL_PROGRAM) */$$i $(INSTALL_DIR)/bin; done
 	for i in $(OBJ_FILES); do $(INSTALL_DATA) */$$i $(INSTALL_DIR)/lib; done
 	for i in $(LIB_FILES); do $(INSTALL_DATA) */$$i $(INSTALL_DIR)/lib; done
