--- gprolog-1.4.0.orig/src/Makefile.in	2011-04-09 04:35:22.000000000 +1200
+++ gprolog-1.4.0/src/Makefile.in	2011-07-01 20:40:24.000000000 +1200
@@ -60,7 +60,6 @@
 install-system:
 	./mkinstalldirs $(INSTALL_DIR) $(INSTALL_DIR)/bin \
 	$(INSTALL_DIR)/include $(INSTALL_DIR)/lib
-	for i in $(TXT_FILES); do $(INSTALL_DATA) ../$$i $(INSTALL_DIR); done
 	for i in $(BIN_FILES); do $(INSTALL_PROGRAM) */$$i $(INSTALL_DIR)/bin; done
 	for i in $(OBJ_FILES); do $(INSTALL_DATA) */$$i $(INSTALL_DIR)/lib; done
 	for i in $(LIB_FILES); do $(INSTALL_DATA) */$$i $(INSTALL_DIR)/lib; done
