diff -ur gprolog-1.4.0.orig/src/Ma2Asm/alpha_any.c gprolog-1.4.0/src/Ma2Asm/alpha_any.c
--- gprolog-1.4.0.orig/src/Ma2Asm/alpha_any.c	2011-03-29 05:20:05.000000000 +1300
+++ gprolog-1.4.0/src/Ma2Asm/alpha_any.c	2011-07-05 19:38:52.000000000 +1200
@@ -870,7 +870,7 @@
  *                                                                         *
  *-------------------------------------------------------------------------*/
 int
-Call_C_Arg_Int(int offset, long int_val)
+Call_C_Arg_Int(int offset, PlLong int_val)
 {
   switch (offset)
     {
@@ -1433,7 +1433,7 @@
  *                                                                         *
  *-------------------------------------------------------------------------*/
 void
-Cmp_Ret_And_Int(long int_val)
+Cmp_Ret_And_Int(PlLong int_val)
 {
   Inst_Printf("lda", "$1,%ld", int_val);
   Inst_Printf("subq", "$0,$1,$1");
@@ -1551,7 +1551,7 @@
  *                                                                         *
  *-------------------------------------------------------------------------*/
 void
-Dico_Long(char *name, int global, VType vtype, long value)
+Dico_Long(char *name, int global, VType vtype, PlLong value)
 {
   switch (vtype)
     {
diff -ur gprolog-1.4.0.orig/src/Ma2Asm/mips_irix.c gprolog-1.4.0/src/Ma2Asm/mips_irix.c
--- gprolog-1.4.0.orig/src/Ma2Asm/mips_irix.c	2011-03-29 05:20:05.000000000 +1300
+++ gprolog-1.4.0/src/Ma2Asm/mips_irix.c	2011-07-05 19:39:30.000000000 +1200
@@ -510,7 +510,7 @@
  *                                                                         *
  *-------------------------------------------------------------------------*/
 int
-Call_C_Arg_Int(int offset, long int_val)
+Call_C_Arg_Int(int offset, PlLong int_val)
 {
   switch (offset)
     {
@@ -1125,7 +1125,7 @@
  *                                                                         *
  *-------------------------------------------------------------------------*/
 void
-Cmp_Ret_And_Int(long int_val)
+Cmp_Ret_And_Int(PlLong int_val)
 {
   Inst_Printf("li", "$24,%d", int_val);
   Inst_Printf("sub", "$12,$2,$24");	/* $2 - $24 -> $12 */
@@ -1238,7 +1238,7 @@
  *                                                                         *
  *-------------------------------------------------------------------------*/
 void
-Dico_Long(char *name, int global, VType vtype, long value)
+Dico_Long(char *name, int global, VType vtype, PlLong value)
 {
   switch (vtype)
     {
diff -ur gprolog-1.4.0.orig/src/Ma2Asm/powerpc_any.c gprolog-1.4.0/src/Ma2Asm/powerpc_any.c
--- gprolog-1.4.0.orig/src/Ma2Asm/powerpc_any.c	2011-03-29 05:20:05.000000000 +1300
+++ gprolog-1.4.0/src/Ma2Asm/powerpc_any.c	2011-07-05 19:17:16.000000000 +1200
@@ -500,7 +500,7 @@
  *                                                                         *
  *-------------------------------------------------------------------------*/
 int
-Call_C_Arg_Int(int offset, long int_val)
+Call_C_Arg_Int(int offset, PlLong int_val)
 {
   BEFORE_ARG;
 
@@ -835,7 +835,7 @@
  *                                                                         *
  *-------------------------------------------------------------------------*/
 void
-Cmp_Ret_And_Int(long int_val)
+Cmp_Ret_And_Int(PlLong int_val)
 {
   if (LITTLE_INT(int_val))
     Inst_Printf("cmpwi", CR(1) "," R(3) ",%ld", int_val);
@@ -958,7 +958,7 @@
  *                                                                         *
  *-------------------------------------------------------------------------*/
 void
-Dico_Long(char *name, int global, VType vtype, long value)
+Dico_Long(char *name, int global, VType vtype, PlLong value)
 {
   switch (vtype)
     {
diff -ur gprolog-1.4.0.orig/src/Ma2Asm/sparc_any.c gprolog-1.4.0/src/Ma2Asm/sparc_any.c
--- gprolog-1.4.0.orig/src/Ma2Asm/sparc_any.c	2011-03-29 05:20:05.000000000 +1300
+++ gprolog-1.4.0/src/Ma2Asm/sparc_any.c	2011-07-05 19:18:29.000000000 +1200
@@ -449,7 +449,7 @@
  *                                                                         *
  *-------------------------------------------------------------------------*/
 int
-Call_C_Arg_Int(int offset, long int_val)
+Call_C_Arg_Int(int offset, PlLong int_val)
 {
   BEFORE_ARG;
 
@@ -792,7 +792,7 @@
  *                                                                         *
  *-------------------------------------------------------------------------*/
 void
-Cmp_Ret_And_Int(long int_val)
+Cmp_Ret_And_Int(PlLong int_val)
 {
   if (LITTLE_INT(int_val))
     Inst_Printf("cmp", "%%o0,%ld", int_val);
@@ -912,7 +912,7 @@
  *                                                                         *
  *-------------------------------------------------------------------------*/
 void
-Dico_Long(char *name, int global, VType vtype, long value)
+Dico_Long(char *name, int global, VType vtype, PlLong value)
 {
 
   switch (vtype)
