--- gprolog-1.4.0.orig/src/Makefile.in	2011-04-09 04:35:22.000000000 +1200
+++ gprolog-1.4.0/src/Makefile.in	2011-07-02 08:25:22.000000000 +1200
@@ -97,7 +96,7 @@
 # --- Documentation --- #
 
 install-doc:
-	if test $(DOC_DIR) != none; then \
+	if test $(DOC_DIR) != $(DESTDIR)none; then \
 	   ./mkinstalldirs $(DOC_DIR); \
 	   (F=`cd ../doc; echo $(DOC_FILES)`; \
 	   for i in $$F; do $(INSTALL_DATA) ../doc/$$i $(DOC_DIR); done); \
@@ -113,7 +112,7 @@
 # --- HTML --- #
 
 install-html:
-	if test $(HTML_DIR) != none; then \
+	if test $(HTML_DIR) != $(DESTDIR)none; then \
 	   ./mkinstalldirs $(HTML_DIR); \
 	   (F=`cd ../doc/html_node; echo $(HTML_FILES)`; \
 	   for i in $$F; do $(INSTALL_DATA) ../doc/html_node/$$i $(HTML_DIR); done); \
@@ -129,7 +128,7 @@
 # --- Examples --- #
 
 install-examples:
-	if test $(EXAMPLES_DIR) != none; then \
+	if test $(EXAMPLES_DIR) != $(DESTDIR)none; then \
 	   ./mkinstalldirs $(EXAMPLES_DIR)/ExamplesPl; \
 	   (F=`cd ../examples/ExamplesPl; echo $(EXPL_FILES)`; \
 	   for i in $$F; do $(INSTALL_DATA) ../examples/ExamplesPl/$$i $(EXAMPLES_DIR)/ExamplesPl; done); \
