--- shorewall-lite-4.4.20.3/install.sh	2011-06-10 15:04:11.000000000 +0200
+++ install.sh	2011-06-14 12:09:52.000000000 +0200
@@ -199,7 +199,7 @@
     rm -rf ${DESTDIR}/etc/shorewall-lite
     rm -rf ${DESTDIR}/usr/share/shorewall-lite
     rm -rf ${DESTDIR}/var/lib/shorewall-lite
-    [ "$LIBEXEC" = share ] || rm -rf /usr/share/shorewall-lite/shorecap /usr/share/shorecap
+    [ "$LIBEXEC" = /usr/share ] || rm -rf /usr/share/shorewall-lite/shorecap /usr/share/shorecap
 fi
 
 #
