diff -uprN jexcelapi.orig/src/jxl/write/biff/CompoundFile.java jexcelapi/src/jxl/write/biff/CompoundFile.java
--- jexcelapi.orig/src/jxl/write/biff/CompoundFile.java	2005-01-23 15:39:35.389297104 +0100
+++ jexcelapi/src/jxl/write/biff/CompoundFile.java	2005-01-23 15:40:59.715477576 +0100
@@ -169,11 +169,11 @@ final class CompoundFile extends BaseCom
    */
   private static final class ReadPropertyStorage
   {
-    PropertyStorage propertyStorage;
+    jxl.biff.BaseCompoundFile.PropertyStorage propertyStorage;
     byte[] data;
     int number;
 
-    ReadPropertyStorage(PropertyStorage ps, byte[] d, int n)
+    ReadPropertyStorage(jxl.biff.BaseCompoundFile.PropertyStorage ps, byte[] d, int n)
     {
       propertyStorage = ps;
       data = d;
