diff -pruN kasumi-2.3.orig/KasumiAddWindow.cxx kasumi-2.3/KasumiAddWindow.cxx
--- kasumi-2.3.orig/KasumiAddWindow.cxx	2007-10-28 17:40:43.000000000 +0900
+++ kasumi-2.3/KasumiAddWindow.cxx	2008-02-19 16:59:45.000000000 +0900
@@ -25,6 +25,7 @@
 *********************************************************************/
 
 #include <gtk/gtk.h>
+#include <algorithm>
 #include <iostream>
 #include "KasumiAddWindow.hxx"
 #include "KasumiWord.hxx"
diff -pruN kasumi-2.3.orig/KasumiWord.cxx kasumi-2.3/KasumiWord.cxx
--- kasumi-2.3.orig/KasumiWord.cxx	2007-10-30 23:01:05.000000000 +0900
+++ kasumi-2.3/KasumiWord.cxx	2008-02-19 16:59:22.000000000 +0900
@@ -24,6 +24,7 @@
  *********************************************************************/
 
 #include <cstdlib>
+#include <cstring>
 #include <string>
 #include <map>
 #include <iconv.h>
diff -pruN kasumi-2.3.orig/main.cxx kasumi-2.3/main.cxx
--- kasumi-2.3.orig/main.cxx	2007-10-30 23:01:22.000000000 +0900
+++ kasumi-2.3/main.cxx	2008-02-19 16:59:22.000000000 +0900
@@ -28,6 +28,7 @@
 #include "config.h"
 #endif
 
+#include <cstring>
 #include <gtk/gtk.h>
 #include <stdlib.h>
 #include <time.h>
