diff -urN libav-0.8.5.orig//configure libav-0.8.5/configure
--- libav-0.8.5.orig//configure	2013-01-12 18:00:42.000000000 +0100
+++ libav-0.8.5/configure	2013-01-17 11:43:39.798920861 +0100
@@ -1062,6 +1062,8 @@
     asm_mod_y
     attribute_may_alias
     attribute_packed
+    cdio_paranoia_h
+    cdio_paranoia_paranoia_h
     closesocket
     cmov
     dcbzl
@@ -3020,8 +3022,9 @@
 
 enabled_any sndio_indev sndio_outdev && check_lib2 sndio.h sio_open -lsndio
 
-enabled libcdio &&
-    check_lib2 "cdio/cdda.h cdio/paranoia.h" cdio_cddap_open "-lcdio_paranoia -lcdio_cdda -lcdio"
+if enabled libcdio; then
+    check_lib2 "cdio/cdda.h cdio/paranoia.h" cdio_cddap_open -lcdio_paranoia -lcdio_cdda -lcdio || check_lib2 "cdio/paranoia/cdda.h cdio/paranoia/paranoia.h" cdio_cddap_open -lcdio_paranoia -lcdio_cdda -lcdio
+fi
 
 enabled x11grab                         &&
 check_header X11/Xlib.h                 &&
diff -urN libav-0.8.5.orig//libavdevice/libcdio.c libav-0.8.5/libavdevice/libcdio.c
--- libav-0.8.5.orig//libavdevice/libcdio.c	2013-01-12 18:00:42.000000000 +0100
+++ libav-0.8.5/libavdevice/libcdio.c	2013-01-17 11:44:28.314919094 +0100
@@ -23,8 +23,15 @@
  * libcdio CD grabbing
  */
 
+#include "config.h"
+
+#if HAVE_CDIO_PARANOIA_H
 #include <cdio/cdda.h>
 #include <cdio/paranoia.h>
+#elif HAVE_CDIO_PARANOIA_PARANOIA_H
+#include <cdio/paranoia/cdda.h>
+#include <cdio/paranoia/paranoia.h>
+#endif
 
 #include "libavutil/log.h"
 #include "libavutil/mem.h"
