diff -ur ../src.orig/sys/boot/efi/libefi/Makefile sys/boot/efi/libefi/Makefile
--- ../src.orig/sys/boot/efi/libefi/Makefile	2006-11-05 23:03:03.000000000 +0100
+++ sys/boot/efi/libefi/Makefile	2008-03-14 11:15:48.482695423 +0100
@@ -8,7 +8,7 @@
 
 CFLAGS+= -I${.CURDIR}/../include
 CFLAGS+= -I${.CURDIR}/../include/${MACHINE_ARCH:S/amd64/i386/}
-CFLAGS+= -I${.CURDIR}/../../../../lib/libstand
+CFLAGS+= -I/usr/include/libstand
 
 # Pick up the bootstrap header for some interface items
 CFLAGS+= -I${.CURDIR}/../../common
diff -ur ../src.orig/sys/boot/i386/boot2/Makefile sys/boot/i386/boot2/Makefile
--- ../src.orig/sys/boot/i386/boot2/Makefile	2007-05-19 07:07:47.000000000 +0200
+++ sys/boot/i386/boot2/Makefile	2008-03-14 11:21:16.247014910 +0100
@@ -24,7 +24,7 @@
 	-fno-guess-branch-probability \
 	-fomit-frame-pointer \
 	-fno-unit-at-a-time \
-	-mno-align-long-strings \
+	-march=i586 \
 	-mrtd \
 	-mno-mmx -mno-3dnow -mno-sse -mno-sse2 -mno-sse3 \
 	-D${BOOT2_UFS} \
diff -ur ../src.orig/sys/boot/i386/libfirewire/Makefile sys/boot/i386/libfirewire/Makefile
--- ../src.orig/sys/boot/i386/libfirewire/Makefile	2007-05-29 16:35:57.000000000 +0200
+++ sys/boot/i386/libfirewire/Makefile	2008-03-14 11:15:48.502696288 +0100
@@ -10,7 +10,7 @@
 CFLAGS+=	-D_BOOT
 
 CFLAGS+=	-I${.CURDIR}/../../common -I${.CURDIR}/../../.. -I.
-CFLAGS+=	-I${.CURDIR}/../../../../lib/libstand
+CFLAGS+=	-I/usr/include/libstand
 CFLAGS+=	-I${.CURDIR}/../btx/lib
 CFLAGS+=	-I${.CURDIR}/../libi386
 
diff -ur ../src.orig/sys/boot/i386/libi386/Makefile sys/boot/i386/libi386/Makefile
--- ../src.orig/sys/boot/i386/libi386/Makefile	2006-04-11 22:11:30.000000000 +0200
+++ sys/boot/i386/libi386/Makefile	2008-03-14 11:15:48.506695959 +0100
@@ -36,7 +36,7 @@
 		-I${.CURDIR}/../../../contrib/dev/acpica \
 		-I${.CURDIR}/../../.. -I.
 # the location of libstand
-CFLAGS+=	-I${.CURDIR}/../../../../lib/libstand/
+CFLAGS+=	-I/usr/include/libstand/
 
 .if ${MACHINE_ARCH} == "amd64"
 CLEANFILES+=	machine
diff -ur ../src.orig/sys/boot/ia64/common/Makefile sys/boot/ia64/common/Makefile
--- ../src.orig/sys/boot/ia64/common/Makefile	2006-11-05 23:03:03.000000000 +0100
+++ sys/boot/ia64/common/Makefile	2008-03-14 11:15:48.486695372 +0100
@@ -10,7 +10,7 @@
 CFLAGS+=	-I${.CURDIR}/../../efi/include
 CFLAGS+=	-I${.CURDIR}/../../efi/include/${MACHINE_ARCH}
 CFLAGS+=	-I${.CURDIR}/../../..
-CFLAGS+=	-I${.CURDIR}/../../../../lib/libstand
+CFLAGS+=	-I/usr/include/libstand
 
 .if ${MK_FORTH} != "no"
 BOOT_FORTH=	yes
diff -ur ../src.orig/sys/boot/ia64/efi/Makefile sys/boot/ia64/efi/Makefile
--- ../src.orig/sys/boot/ia64/efi/Makefile	2006-11-05 23:03:03.000000000 +0100
+++ sys/boot/ia64/efi/Makefile	2008-03-14 11:15:48.490695322 +0100
@@ -15,7 +15,7 @@
 CFLAGS+=	-I${.CURDIR}/../../efi/include
 CFLAGS+=	-I${.CURDIR}/../../efi/include/${MACHINE_ARCH}
 CFLAGS+=	-I${.CURDIR}/../../..
-CFLAGS+=	-I${.CURDIR}/../../../../lib/libstand
+CFLAGS+=	-I/usr/include/libstand
 
 LDSCRIPT=	${.CURDIR}/ldscript.${MACHINE_ARCH}
 LDFLAGS=	-Wl,-T${LDSCRIPT} -shared -symbolic
diff -ur ../src.orig/sys/boot/ia64/ski/Makefile sys/boot/ia64/ski/Makefile
--- ../src.orig/sys/boot/ia64/ski/Makefile	2006-11-05 23:03:04.000000000 +0100
+++ sys/boot/ia64/ski/Makefile	2008-03-14 11:15:48.494695551 +0100
@@ -14,7 +14,7 @@
 CFLAGS+=	-I${.CURDIR}/../common
 CFLAGS+=	-I${.CURDIR}/../../common
 CFLAGS+=	-I${.CURDIR}/../../..
-CFLAGS+=	-I${.CURDIR}/../../../../lib/libstand
+CFLAGS+=	-I/usr/include/libstand
 
 LDSCRIPT=	${.CURDIR}/ldscript.${MACHINE_ARCH}
 LDFLAGS=	-Wl,-T${LDSCRIPT}
diff -ur ../src.orig/sys/boot/ofw/libofw/Makefile sys/boot/ofw/libofw/Makefile
--- ../src.orig/sys/boot/ofw/libofw/Makefile	2007-06-17 02:17:15.000000000 +0200
+++ sys/boot/ofw/libofw/Makefile	2008-03-14 11:15:48.478695194 +0100
@@ -7,7 +7,7 @@
 	ofw_memory.c ofw_module.c ofw_net.c ofw_reboot.c \
 	ofw_time.c openfirm.c
 
-CFLAGS+=	-I${.CURDIR}/../../../../lib/libstand/
+CFLAGS+=	-I/usr/include/libstand/
 
 # Pick up the bootstrap header for some interface items
 CFLAGS+=	-I${.CURDIR}/../../common -I${.CURDIR}/../../.. -I.
diff -ur ../src.orig/sys/boot/pc98/libpc98/Makefile sys/boot/pc98/libpc98/Makefile
--- ../src.orig/sys/boot/pc98/libpc98/Makefile	2006-04-11 22:11:30.000000000 +0200
+++ sys/boot/pc98/libpc98/Makefile	2008-03-14 11:15:48.510696188 +0100
@@ -31,6 +31,6 @@
 		-I${.CURDIR}/../../i386/libi386 \
 		-I${.CURDIR}/../../.. -I.
 # the location of libstand
-CFLAGS+=	-I${.CURDIR}/../../../../lib/libstand/
+CFLAGS+=	-I/usr/include/libstand/
 
 .include <bsd.lib.mk>
diff -ur ../src.orig/sys/boot/powerpc/loader/Makefile sys/boot/powerpc/loader/Makefile
--- ../src.orig/sys/boot/powerpc/loader/Makefile	2006-03-17 19:54:36.000000000 +0100
+++ sys/boot/powerpc/loader/Makefile	2008-03-14 11:15:48.515695706 +0100
@@ -82,7 +82,7 @@
 CFLAGS+=	-I${.CURDIR}/../../ofw/libofw
 
 # where to get libstand from
-CFLAGS+=	-I${.CURDIR}/../../../../lib/libstand/
+CFLAGS+=	-I/usr/include/libstand/
 
 DPADD=		${LIBFICL} ${LIBOFW} ${LIBSTAND}
 LDADD=		${LIBFICL} ${LIBOFW} -lstand
diff -ur ../src.orig/sys/boot/sparc64/loader/Makefile sys/boot/sparc64/loader/Makefile
--- ../src.orig/sys/boot/sparc64/loader/Makefile	2006-03-17 19:54:36.000000000 +0100
+++ sys/boot/sparc64/loader/Makefile	2008-03-14 11:15:48.498695501 +0100
@@ -65,7 +65,7 @@
 CFLAGS+=	-I${.CURDIR}/../../ofw/libofw/
 
 # where to get libstand from
-CFLAGS+=	-I${.CURDIR}/../../../../lib/libstand/
+CFLAGS+=	-I/usr/include/libstand/
 
 DPADD=		${LIBFICL} ${LIBOFW} ${LIBSTAND}
 LDADD=		${LIBFICL} ${LIBOFW} -lstand
