diff -urp i2c-2.10.0/kernel/Module.mk i2c-2.10.0-module-path/kernel/Module.mk
--- i2c-2.10.0/kernel/Module.mk	2005-05-12 19:25:24.000000000 +0200
+++ i2c-2.10.0-module-path/kernel/Module.mk	2006-02-15 17:50:44.000000000 +0100
@@ -98,18 +98,18 @@ all :: all-kernel
 
 #
 # If $MODPREF/kernel exists, we presume the new (2.4.0) /lib/modules/x.y.z directory
-# layout, so we install in kernel/drivers/i2c/ and remove old versions in misc/
+# layout, so we install in drivers/i2c/ and remove old versions in misc/
 # Otherwise we install in misc/ as before.
 #
 install-kernel: all-kernel
 	if [ -n "$(KERNELTARGETS)" ] ; then \
-	  $(MKDIR) $(DESTDIR)$(MODPREF)/kernel/drivers/i2c ; \
-	  $(INSTALL) -m 644 $(KERNELTARGETS) $(DESTDIR)$(MODPREF)/kernel/drivers/i2c ; \
+	  $(MKDIR) $(DESTDIR)$(MODPREF)/drivers/i2c ; \
+	  $(INSTALL) -m 644 $(KERNELTARGETS) $(DESTDIR)$(MODPREF)/drivers/i2c ; \
 	  for i in $(KERNELTARGETS) ; do \
-	    $(RM) $(DESTDIR)$(MODPREF)/misc/`basename $$i` $(DESTDIR)$(MODPREF)/misc/`basename $$i`.gz $(DESTDIR)$(MODPREF)/kernel/drivers/i2c/`basename $$i`.gz ; \
+	    $(RM) $(DESTDIR)$(MODPREF)/misc/`basename $$i` $(DESTDIR)$(MODPREF)/misc/`basename $$i`.gz $(DESTDIR)$(MODPREF)/drivers/i2c/`basename $$i`.gz ; \
 	  done ; \
 	  for i in i2c-elektor.o i2c-pcf-epp.o ; do \
-	    $(RM) $(DESTDIR)$(MODPREF)/misc/`basename $$i` $(DESTDIR)$(MODPREF)/misc/`basename $$i`.gz $(DESTDIR)$(MODPREF)/kernel/drivers/i2c/`basename $$i` $(DESTDIR)$(MODPREF)/kernel/drivers/i2c/`basename $$i`.gz ; \
+	    $(RM) $(DESTDIR)$(MODPREF)/misc/`basename $$i` $(DESTDIR)$(MODPREF)/misc/`basename $$i`.gz $(DESTDIR)$(MODPREF)/drivers/i2c/`basename $$i` $(DESTDIR)$(MODPREF)/drivers/i2c/`basename $$i`.gz ; \
 	  done ; \
 	  $(RMDIR) $(DESTDIR)$(MODPREF)/misc 2> /dev/null || true ; \
 	fi
