diff --git a/texstudio.pro b/texstudio.pro
index 437a376..c2c6998 100644
--- a/texstudio.pro
+++ b/texstudio.pro
@@ -5,7 +5,10 @@ CONFIG = $$unique(CONFIG)
 CONFIG -= debug \
     debug_and_release \
     release
-CONFIG += qt debug_and_release precompile_header
+CONFIG += qt debug_and_release precompile_header \
+    link_pkgconfig
+PKGCONFIG = hunspell
+PHONON=
 exists(texmakerx_my.pri):include(texmakerx_my.pri)
 QT += network \
     xml \
@@ -55,24 +58,6 @@ HEADERS += texmaker.h \
     codesnippet.h \
     thesaurusdialog.h \
     xmltagslistwidget.h \
-    hunspell/affentry.hxx \
-    hunspell/affixmgr.hxx \
-    hunspell/atypes.hxx \
-    hunspell/baseaffix.hxx \
-    hunspell/csutil.hxx \
-    hunspell/dictmgr.hxx \
-    hunspell/hashmgr.hxx \
-    hunspell/htypes.hxx \
-    hunspell/hunspell.hxx \
-    hunspell/hunspell.h \
-    hunspell/langnum.hxx \
-    hunspell/license.hunspell \
-    hunspell/phonet.hxx \
-    hunspell/suggestmgr.hxx \
-    hunspell/license.myspell \
-    hunspell/filemgr.hxx \
-    hunspell/hunzip.hxx \
-    hunspell/w_char.hxx \
     qcodeedit/lib/qeditorinputbinding.h \
     qcodeedit/lib/qeditorinputbindinginterface.h \
     randomtextgenerator.h \
@@ -88,7 +73,6 @@ HEADERS += texmaker.h \
     latexdocument.h \
     unicodeinsertion.h \
     universalinputdialog.h \
-    hunspell/replist.hxx \
     scriptengine.h \
     insertgraphics.h \
     tmxtabwidget.h \
@@ -142,18 +126,6 @@ SOURCES += main.cpp \
     codesnippet.cpp \
     thesaurusdialog.cpp \
     xmltagslistwidget.cpp \
-    hunspell/affentry.cxx \
-    hunspell/affixmgr.cxx \
-    hunspell/csutil.cxx \
-    hunspell/dictmgr.cxx \
-    hunspell/hashmgr.cxx \
-    hunspell/hunspell.cxx \
-    hunspell/phonet.cxx \
-    hunspell/replist.cxx \
-    hunspell/suggestmgr.cxx \
-    hunspell/utf_info.cxx \
-    hunspell/filemgr.cxx \
-    hunspell/hunzip.cxx \
     encodingdialog.cpp \
     qcodeedit/lib/qeditorinputbinding.cpp \
     randomtextgenerator.cpp \
