 texstudio.pro | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/texstudio.pro b/texstudio.pro
index f801fc2..2b1d3df 100644
--- a/texstudio.pro
+++ b/texstudio.pro
@@ -10,8 +10,6 @@ QT += network \
     QT += phonon
     DEFINES += PHONON
 }
-contains($$list($$[QT_VERSION]), 4.3.*):message("qt 4.3.x")
-else:include(qtsingleapplication/qtsingleapplication.pri)
 
 # ##############################
 PRECOMPILED_HEADER = mostQtHeaders.h
@@ -567,3 +565,6 @@ unix {
   PKGCONFIG += hunspell
 }
 
+#use system qtsingleapplication
+CONFIG += qtsingleapplication
+
