 spellerutility.h |  2 +-
 texstudio.pro    | 37 ++++++-------------------------------
 2 files changed, 7 insertions(+), 32 deletions(-)

diff --git a/spellerutility.h b/spellerutility.h
index 1e1d6e2..083f412 100644
--- a/spellerutility.h
+++ b/spellerutility.h
@@ -13,7 +13,7 @@
 
 #include "mostQtHeaders.h"
 
-#include "hunspell/hunspell.hxx"
+#include <hunspell.hxx>
 class SpellerUtility: public QObject {
 	Q_OBJECT
 public:
diff --git a/texstudio.pro b/texstudio.pro
index 541b6b6..f801fc2 100644
--- a/texstudio.pro
+++ b/texstudio.pro
@@ -52,24 +52,6 @@ HEADERS += texmaker.h \
     debughelper.h \
     thesaurusdialog.h \
     xmltagslistwidget.h \
-    hunspell/affentry.hxx \
-    hunspell/affixmgr.hxx \
-    hunspell/atypes.hxx \
-    hunspell/baseaffix.hxx \
-    hunspell/csutil.hxx \
-    hunspell/dictmgr.hxx \
-    hunspell/hashmgr.hxx \
-    hunspell/htypes.hxx \
-    hunspell/hunspell.hxx \
-    hunspell/hunspell.h \
-    hunspell/langnum.hxx \
-    hunspell/license.hunspell \
-    hunspell/phonet.hxx \
-    hunspell/suggestmgr.hxx \
-    hunspell/license.myspell \
-    hunspell/filemgr.hxx \
-    hunspell/hunzip.hxx \
-    hunspell/w_char.hxx \
     qcodeedit/lib/qeditorinputbinding.h \
     qcodeedit/lib/qeditorinputbindinginterface.h \
     randomtextgenerator.h \
@@ -85,7 +67,6 @@ HEADERS += texmaker.h \
     latexdocument.h \
     unicodeinsertion.h \
     universalinputdialog.h \
-    hunspell/replist.hxx \
     scriptengine.h \
     insertgraphics.h \
     txstabwidget.h \
@@ -157,18 +138,6 @@ SOURCES += main.cpp \
     codesnippet.cpp \
     thesaurusdialog.cpp \
     xmltagslistwidget.cpp \
-    hunspell/affentry.cxx \
-    hunspell/affixmgr.cxx \
-    hunspell/csutil.cxx \
-    hunspell/dictmgr.cxx \
-    hunspell/hashmgr.cxx \
-    hunspell/hunspell.cxx \
-    hunspell/phonet.cxx \
-    hunspell/replist.cxx \
-    hunspell/suggestmgr.cxx \
-    hunspell/utf_info.cxx \
-    hunspell/filemgr.cxx \
-    hunspell/hunzip.cxx \
     encodingdialog.cpp \
     qcodeedit/lib/qeditorinputbinding.cpp \
     randomtextgenerator.cpp \
@@ -591,4 +560,10 @@ SOURCES+=synctex_parser_utils.c synctex_parser.c
 #QMAKE_CXXFLAGS_DEBUG += -Werror  -Wall -Wextra  -Winit-self -Wmain -Wmissing-include-dirs -Wtrigraphs -Wunused -Wunknown-pragmas  -Wundef  -Wpointer-arith -Wtype-limits -Wwrite-strings -Wclobbered  -Wempty-body -Wsign-compare -Waddress -Wlogical-op   -Winline
 QMAKE_CXXFLAGS_DEBUG += -Wall -Wextra  -Winit-self -Wmissing-include-dirs -Wtrigraphs -Wunused -Wunknown-pragmas  -Wundef  -Wpointer-arith  -Wwrite-strings -Wempty-body -Wsign-compare -Waddress   -Winline
 
+# enable use of pkgconfig
+unix {
+  CONFIG   += link_pkgconfig
+  # use system hunspell
+  PKGCONFIG += hunspell
+}
 
