From cb478565396b325a0860747a3c9c90789b13aea1 Mon Sep 17 00:00:00 2001
From: hannes <hannes@fedoraproject.org>
Date: Thu, 16 Feb 2012 20:49:50 +0000
Subject: [PATCH] Fix the testcase build failure

---
 tests/qsearchreplacepanel_t.cpp |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/tests/qsearchreplacepanel_t.cpp b/tests/qsearchreplacepanel_t.cpp
index e043f8b..22d6013 100644
--- tests/qsearchreplacepanel_t.cpp
+++ tests/qsearchreplacepanel_t.cpp
@@ -213,9 +213,9 @@ void QSearchReplacePanelTest::incrementalsearch(){
 				widget->cFind->lineEdit()->cursorForward(false,search.length());	
 				QTest::keyClick(widget->cFind->lineEdit(),search[search.length()-1].toLatin1());
 			}
-			QDocumentCursor s=ed->cursor().selectionStart();
-			QEQUAL2(s.lineNumber(), cy, search+" "+ed->cursor().selectedText()+"  "+QString::number(loop));
-			QEQUAL2(s.columnNumber(), cx, search+" "+ed->cursor().selectedText());
+			QDocumentCursor c=ed->cursor().selectionStart();
+			QEQUAL2(c.lineNumber(), cy, search+" "+ed->cursor().selectedText()+"  "+QString::number(loop));
+			QEQUAL2(c.columnNumber(), cx, search+" "+ed->cursor().selectedText());
 			QEQUAL2(ed->cursor().selectedText(), res, search+" "+ed->cursor().selectedText());
 			//searching shouldn't change highlighted selection
 			QCEMULTIEQUAL(getHighlightedSelection(ed), panel->getSearchScope(), ed->document()->cursor(scopey1,scopex1,scopey2,scopex2));			
-- 
1.7.7.6

