diff -urNp skype-4.2.0.13-orig/skype.desktop skype-4.2.0.13/skype.desktop
--- skype-4.2.0.13-orig/skype.desktop	2013-12-18 20:52:54.000000000 +0200
+++ skype-4.2.0.13/skype.desktop	2013-12-18 20:52:54.000000000 +0200
@@ -1,11 +1,11 @@
 [Desktop Entry]
 Name=Skype
 Comment=Skype Internet Telephony
 Exec=env PULSE_LATENCY_MSEC=60 skype %U
-Icon=skype.png
+Icon=skype
 Terminal=false
 Type=Application
 Encoding=UTF-8
-Categories=Network;Application;
+Categories=Network;InstantMessaging;Telephony;
 MimeType=x-scheme-handler/skype;
 X-KDE-Protocols=skype
