Remove hard-coded utility paths

--- a/src/lib/connections/bond
+++ b/src/lib/connections/bond
@@ -2,7 +2,7 @@
 
 . "$SUBR_DIR/ip"
 
-IFENSLAVE="/sbin/ifenslave"
+IFENSLAVE="ifenslave"
 
 bond_up() {
     if is_interface "$Interface"; then
--- a/src/lib/connections/bridge
+++ b/src/lib/connections/bridge
@@ -2,7 +2,7 @@
 
 . "$SUBR_DIR/ip"
 
-BRCTL="/usr/sbin/brctl"
+BRCTL="brctl"
 
 bridge_up() {
     if is_interface "$Interface"; then
--- a/src/lib/connections/pppoe
+++ b/src/lib/connections/pppoe
@@ -43,8 +43,8 @@ pppoe_up() {
     [[ -n ${PPPoEMAC} ]] && echo "pppoe-mac $(_quotestring "${PPPoEMAC}")" >> "${cfg}"
     [[ ${PPPoEIP6} == yes ]] && echo "+ipv6" >> "${cfg}"
 
-    /sbin/ip link set dev "${Interface}" up
-    /usr/sbin/pppd file "${cfg}"
+    ip link set dev "${Interface}" up
+    pppd file "${cfg}"
 
     if [[ $? -ne 0 ]]; then
         rm "${cfg}"
