--- xsd-regexp.el.orig	2006-01-27 00:31:24.000000000 +0100
+++ xsd-regexp.el	2006-01-27 00:32:20.328529600 +0100
@@ -290,7 +290,8 @@
 (defun xsdre-compile-single-char (ch)
   (if (memq ch '(?. ?* ?+ ?? ?\[ ?\] ?^ ?$ ?\\))
       (string ?\\ ch)
-    (string (decode-char 'ucs ch))))
+    (let ((utf-translate-cjk-mode nil))
+      (string (decode-char 'ucs ch)))))
   
 (defun xsdre-char-class-to-range-list (cc)
   "Return a range-list for a symbolic char-class."
@@ -403,7 +404,8 @@
       (setq range-list (cdr range-list)))
     (setq chars
 	  (mapcar (lambda (c)
-		    (decode-char 'ucs c))
+                    (let ((utf-translate-cjk-mode nil))
+                      (decode-char 'ucs c)))
 		  chars))
     (when caret
       (setq chars (cons ?^ chars)))
