--- buildgcc/Makefile.orig	2008-10-14 20:31:23.645424162 -0700
+++ buildgcc/Makefile	2008-10-14 20:36:36.484423984 -0700
@@ -36,14 +36,14 @@
 ##################################################
 
 # define compile options for each build
-CFLAGS   = -DNDEBUG -O3 $(CBASEFLAGS)
-CFLAGSd  = -g $(CBASEFLAGS)
-CFLAGSs  = -DNDEBUG -DASTYLE_LIB -O3 -fpic $(CBASEFLAGS)
-CFLAGSsd = -DASTYLE_LIB -g -fpic $(CBASEFLAGS)
-CFLAGSa  = -DNDEBUG -DASTYLE_LIB -O3 $(CBASEFLAGS)
-CFLAGSad = -DASTYLE_LIB -g $(CBASEFLAGS)
-CFLAGSsj  = -DNDEBUG -DASTYLE_JNI -O3 -fpic $(CBASEFLAGS) $(JAVAINCS)
-CFLAGSsjd = -DASTYLE_JNI -g -fpic $(CBASEFLAGS) $(JAVAINCS)
+CFLAGSb   = -DNDEBUG $(CFLAGS)
+CFLAGSd  = -g $(CFLAGS)
+CFLAGSs  = -DNDEBUG -DASTYLE_LIB -fPIC $(CFLAGS)
+CFLAGSsd = -DASTYLE_LIB -g -fPIC $(CFLAGS)
+CFLAGSa  = -DNDEBUG -DASTYLE_LIB $(CFLAGS)
+CFLAGSad = -DASTYLE_LIB -g $(CFLAGS)
+CFLAGSsj  = -DNDEBUG -DASTYLE_JNI -fPIC $(CFLAGS) $(JAVAINCS)
+CFLAGSsjd = -DASTYLE_JNI -g -fPIC $(CFLAGS) $(JAVAINCS)
 
 # object files are built from the source list $(SRC)
 # a suffix is added for each build
@@ -61,7 +61,7 @@
 # OBJ
 $(objdir)/%.o:  %.cpp  astyle.h
 	@ mkdir -p $(objdir)
-	$(CXX) $(CFLAGS) -c -o $@ $<
+	$(CXX) $(CFLAGSb) -c -o $@ $<
 
 # OBJd
 $(objdir)/%_d.o:  %.cpp  astyle.h
@@ -104,7 +104,7 @@
 release:  astyle
 astyle:  $(OBJ)
 	@ mkdir -p $(bindir)
-	$(CXX) -s -o $(bindir)/$@ $^
+	$(CXX) -o $(bindir)/$@ $^
 	@ echo
 
 debug:  astyled
@@ -116,7 +116,7 @@
 shared:  libastyle.so
 libastyle.so:  $(OBJs)
 	@ mkdir -p $(bindir)
-	$(CXX) -shared -Wl,-soname,$@  -s -o $(bindir)/$@ $^
+	$(CXX) -shared -Wl,-soname,$@  -o $(bindir)/$@ $^
 	@ echo
 
 shareddebug:  libastyled.so
--- buildgcc/Makefile.orig	2008-11-10 18:00:51.462464681 -0800
+++ buildgcc/Makefile	2008-11-10 18:31:21.586465413 -0800
@@ -140,7 +140,7 @@
 java:  libastylej.so
 libastylej.so:  $(OBJsj)
 	@ mkdir -p $(bindir)
-	$(CXX) -shared -Wl,-soname,$@  -s -o $(bindir)/$@ $^
+	$(CXX) -shared -Wl,-soname,$@  -o $(bindir)/$@ $^
 	@ echo
 
 javadebug:  libastylejd.so
