--- tests/probes/process/test_probes_process.xml.sh.orig	2012-04-05 15:49:38.746901398 +0200
+++ tests/probes/process/test_probes_process.xml.sh	2012-04-05 15:50:02.330775903 +0200
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
-TMP_P=(`LD_PRELOAD= ps -A -o pid -o ppid -o comm | awk '$1 != 2 && $2 != 2 {print $3}' | \
+TMP_P=(`LD_PRELOAD= ps -A -o pid -o ppid -o comm | grep -iE '^[a-z0-9A-Z \-]*$' |grep -vE 'cron|grep' | awk '$1 != 2 && $2 != 2 {print $3}' | \
         sed -n '2,30p'`)
 
 COUNTER=1
