--- cptutils-1.45/src/Common.mk.in
+++ cptutils-1.45/src/Common.mk.in
@@ -7,7 +7,7 @@
 CC        = @CC@
 CFLAGS   += @CFLAGS@ @DEFS@ 
 LDFLAGS  += 
-LDLIBS   += @LIBS@
+LIBS     += @LIBS@
 
 PYTHON    = @PYTHON@
 
--- cptutils-1.45/src/Progs.mk
+++ cptutils-1.45/src/Progs.mk
@@ -6,7 +6,7 @@
 CFLAGS  += -I../common/
 LDFLAGS += -L../common/ 
 LDLIBS  += -lcpt -lggr -lpsp -lsvg -lgpt -lpov -lodb \
-	   -lcss3 -lsao -lident -lfill
+	   -lcss3 -lsao -lident -lfill $(LIBS)
 
 TESTD    = ../t
 
