--- pinger-0.32c-orig/configure.in	2008-01-21 09:43:32.000000000 +0100
+++ pinger-0.32c/configure.in	2008-01-21 09:43:35.000000000 +0100
@@ -56,13 +56,15 @@
 
 AC_ARG_ENABLE(gtk, [  --enable-gtk            enable GTK+ support],
               [if test "$enableval" = no; then no_gtk=yes; fi])
-              AC_MSG_CHECKING([for GTK+ support])
+AC_MSG_CHECKING([for GTK+ support])
 AC_MSG_RESULT($enableval)
 dnl Check for additional external libraries etc..
+if test x$no_gtk != xyes; then
 AM_PATH_GTK_2_0(2.0.0, CFLAGS="$CFLAGS $GTK_CFLAGS"
 					 LIBS="$LIBS $GTK_LIBS"; HAVE_GTK_2=1;
 					 AC_DEFINE([HAVE_GTK_2], 1, Found GTK 2),
            HAVE_GTK_2=0)
+fi
 AC_CHECK_LIB(ncurses, initscr, HAVE_CURSES=1; LIBS="$LIBS -lncurses";
 	AC_DEFINE([HAVE_LIBNCURSES], [1], found ncurses library))
 AC_CHECK_LIB(intl, libintl_gettext, LIBS="$LIBS -lintl")
