diff -Naur nqc-3.1.r4.orig/compiler/Error.h nqc-3.1.r4/compiler/Error.h
--- nqc-3.1.r4.orig/compiler/Error.h	2005-03-16 17:52:44.000000000 +0000
+++ nqc-3.1.r4/compiler/Error.h	2006-05-08 06:25:30.000000000 +0000
@@ -118,7 +118,7 @@
 	};
 
 			Error(ErrorCode code, int data=0) : fCode(code), fData(data) {}
-			Error(ErrorCode code, const char *s) : fCode(code), fData((int)s) {}
+			Error(ErrorCode code, const char *s) : fCode(code), fData((long)s) {}
 
 	// raising errors
 	void	Raise(const LexLocation *loc) const;	// raise error at specified location
