diff -ruN nqc-2.5.r1/rcxlib/RCX_Image.h nqc-2.5.r1-fixed/rcxlib/RCX_Image.h
--- nqc-2.5.r1/rcxlib/RCX_Image.h	2003-04-13 19:19:57.000000000 +0200
+++ nqc-2.5.r1-fixed/rcxlib/RCX_Image.h	2005-12-14 01:35:14.172693272 +0100
@@ -78,7 +78,6 @@
 	void		Clear();
 	int			GetSize() const;
 
-private:
 	class	Chunk
 	{
 	public:
@@ -106,6 +105,7 @@
 		friend class RCX_Image;
 	};
 	
+private:
 	struct	Variable
 	{
 		int		fIndex;
