Use an external gdbus-codegen package

--- configure.ac
+++ configure.ac
@@ -367,7 +367,7 @@
 AC_SUBST(PERL_PATH)
 
 # Need suitable python path for greport
-AM_PATH_PYTHON(2.5,,PYTHON="/usr/bin/env python2.5")
+# AM_PATH_PYTHON(2.5,,PYTHON="/usr/bin/env python2.5")
 
 
 dnl ***********************
@@ -3857,8 +3857,6 @@
 gthread/Makefile
 gthread/tests/Makefile
 gio/Makefile
-gio/gdbus-2.0/codegen/Makefile
-gio/gdbus-2.0/codegen/config.py
 gio/xdgmime/Makefile
 gio/inotify/Makefile
 gio/libasyncns/Makefile
--- docs/reference/gio/Makefile.am
+++ docs/reference/gio/Makefile.am
@@ -138,8 +138,7 @@
 	gio-querymodules.1	\
 	glib-compile-schemas.1	\
 	gsettings.1		\
-	gdbus.1			\
-	gdbus-codegen.1
+	gdbus.1
 
 if ENABLE_MAN
 
--- gio/Makefile.am
+++ gio/Makefile.am
@@ -2,7 +2,7 @@
 
 NULL =
 
-SUBDIRS = gdbus-2.0/codegen
+SUBDIRS =
 
 if OS_UNIX
 SUBDIRS += libasyncns xdgmime
--- gio/tests/gdbus-object-manager-example/Makefile.am
+++ gio/tests/gdbus-object-manager-example/Makefile.am
@@ -21,10 +21,8 @@
 	gdbus-example-objectmanager-generated-org.gtk.GDBus.Example.ObjectManager.Cat.xml 	\
 	$(NULL)
 
-$(GDBUS_GENERATED) : gdbus-example-objectmanager.xml Makefile $(top_builddir)/gio/gdbus-2.0/codegen/gdbus-codegen
-	$(AM_V_GEN) UNINSTALLED_GLIB_SRCDIR=$(top_srcdir) \
-		UNINSTALLED_GLIB_BUILDDIR=$(top_builddir) \
-		$(PYTHON) $(top_builddir)/gio/gdbus-2.0/codegen/gdbus-codegen \
+$(GDBUS_GENERATED) : gdbus-example-objectmanager.xml Makefile
+	$(AM_V_GEN) gdbus-codegen \
 		--interface-prefix org.gtk.GDBus.Example.ObjectManager. \
 		--c-namespace Example \
 		--c-generate-object-manager \
--- gio/tests/Makefile.am
+++ gio/tests/Makefile.am
@@ -266,10 +266,8 @@
 gdbus_close_pending_LDADD = $(progs_ldadd)
 
 if OS_UNIX
-gdbus-test-codegen-generated.h gdbus-test-codegen-generated.c : test-codegen.xml Makefile $(top_builddir)/gio/gdbus-2.0/codegen/gdbus-codegen
-	$(AM_V_GEN) UNINSTALLED_GLIB_SRCDIR=$(top_srcdir) \
-		UNINSTALLED_GLIB_BUILDDIR=$(top_builddir) \
-		$(PYTHON) $(top_builddir)/gio/gdbus-2.0/codegen/gdbus-codegen \
+gdbus-test-codegen-generated.h gdbus-test-codegen-generated.c : test-codegen.xml Makefile
+	$(AM_V_GEN) gdbus-codegen \
 		--interface-prefix org.project. \
 		--generate-c-code gdbus-test-codegen-generated \
 		--c-generate-object-manager \
