From 81d0ebe29ccd852e969acda7394969e3f13eb136 Mon Sep 17 00:00:00 2001
From: Stef Walter <stefw@gnome.org>
Date: Thu, 31 Oct 2013 11:43:14 +0100
Subject: [PATCH] gmain: Fix use of uninitialized memory in sigaction structure

https://bugzilla.gnome.org/show_bug.cgi?id=711754


diff --git a/glib/gmain.c b/glib/gmain.c
index 9c37bb2..608c1a5 100644
--- a/glib/gmain.c
+++ b/glib/gmain.c
@@ -4978,6 +4978,7 @@ unref_unix_signal_handler_unlocked (int signum)
   if (unix_signal_refcount[signum] == 0)
     {
       struct sigaction action;
+      memset (&action, 0, sizeof (action));
       action.sa_handler = SIG_DFL;
       sigemptyset (&action.sa_mask);
       sigaction (signum, &action, NULL);
--
Gitg

