--- setup.py.org	2012-04-18 13:03:25.000000000 -0400
+++ setup.py	2012-04-18 13:04:53.000000000 -0400
@@ -96,15 +96,14 @@
                 include_dirs.append(e[2:])
         else:
             extra_compile_args.append(e)
-    libs = split_quoted(
-        os.popen("'%s' --libs" % CURL_CONFIG).read()+\
-        os.popen("'%s' --static-libs" % CURL_CONFIG).read())
+    libs = split_quoted(os.popen("'%s' --libs" % CURL_CONFIG).read())
     for e in libs:
         if e[:2] == "-l":
             libraries.append(e[2:])
-            if e[2:] == 'ssl':
+            if e[2:] in ('ssl', 'ssl3'):
                 define_macros.append(('HAVE_CURL_OPENSSL', 1))
             if e[2:] == 'gnutls':
+                libraries.append("gcrypt")
                 define_macros.append(('HAVE_CURL_GNUTLS', 1))
         elif e[:2] == "-L":
             library_dirs.append(e[2:])
