--- Makefile.orig	2010-08-13 21:39:07.000000000 -0400
+++ Makefile	2010-08-13 21:40:54.000000000 -0400
@@ -18,16 +18,16 @@
 BISON=/usr/bin/bison
 #YACC := $(shell if [ -x $(BYACC) ]; then echo $(BYACC); else echo $(BISON); fi)
 YACC=$(BISON)
-MKNOD=/bin/mknod
+MKNOD=true
 #for dietlibc
 #CC=/usr/bin/diet /usr/bin/gcc
 CC=/usr/bin/gcc
 FIND=/usr/bin/find
-STRIP=/usr/bin/strip
+STRIP=true
 LIBS := $(shell if [ "`uname -m`" != "sparc64" ]; then echo "-lfl" ; else echo "" ; fi)
 OPT_FLAGS := $(shell if [ "`uname -m`" != "sparc64" ] && [ "`uname -m`" != "x86_64" ]; then echo "-O2" ; else echo "-O2 -m64" ; fi)
 CFLAGS := $(OPT_FLAGS) -Wcast-qual -DGRSEC_DIR=\"$(GRSEC_DIR)\" -D_LARGEFILE64_SOURCE
-LDFLAGS=
+LDFLAGS+=
 INSTALL = /usr/bin/install -c
 
 # FHS
@@ -134,6 +134,7 @@
 			$(MKNOD) -m 0622 $(DESTDIR)/dev/grsec c 1 13 ; \
 		fi \
 	fi
+	@mkdir -p $(DESTDIR)/etc/udev/rules.d
 	@if [ -d $(DESTDIR)/etc/udev/rules.d ] ; then \
 		echo "ACTION!=\"add|change\", GOTO=\"permissions_end\"" > $(DESTDIR)/etc/udev/rules.d/80-grsec.rules ; \
 		echo "KERNEL==\"grsec\",          MODE=\"0622\"" >> $(DESTDIR)/etc/udev/rules.d/80-grsec.rules ; \
