Index: motion-3.2.12/ffmpeg.c
===================================================================
--- motion-3.2.12.orig/ffmpeg.c
+++ motion-3.2.12/ffmpeg.c
@@ -60,6 +60,10 @@
  */
 #define APPEND_PROTO "appfile"
 
+#ifndef guess_format
+#define guess_format av_guess_format
+#endif
+
 /* Some forward-declarations. */
 void ffmpeg_put_frame(struct ffmpeg *, AVFrame *);
 void ffmpeg_cleanups(struct ffmpeg *);
@@ -228,7 +232,9 @@ void ffmpeg_init()
     mpeg1_file_protocol.url_close = file_protocol.url_close;
 
     /* Register the append file protocol. */
-#if LIBAVFORMAT_BUILD >= (52<<16 | 31<<8)
+#if LIBAVFORMAT_BUILD >= (53<<16 )
+    av_register_protocol2(&mpeg1_file_protocol, sizeof(mpeg1_file_protocol));
+#elif LIBAVFORMAT_BUILD >= (52<<16 | 31<<8)
     av_register_protocol(&mpeg1_file_protocol);
 #else
     register_protocol(&mpeg1_file_protocol);
@@ -377,7 +383,7 @@ struct ffmpeg *ffmpeg_open(char *ffmpeg_
 
     ffmpeg->c     = c = AVSTREAM_CODEC_PTR(ffmpeg->video_st);
     c->codec_id   = ffmpeg->oc->oformat->video_codec;
-    c->codec_type = CODEC_TYPE_VIDEO;
+    c->codec_type = AVMEDIA_TYPE_VIDEO;
     is_mpeg1      = c->codec_id == CODEC_ID_MPEG1VIDEO;
 
     if (strcmp(ffmpeg_video_codec, "ffv1") == 0)
@@ -646,7 +652,7 @@ void ffmpeg_put_frame(struct ffmpeg *ffm
     if (ffmpeg->oc->oformat->flags & AVFMT_RAWPICTURE) {
         /* raw video case. The API will change slightly in the near future for that */
 #ifdef FFMPEG_AVWRITEFRAME_NEWAPI
-        pkt.flags |= PKT_FLAG_KEY;
+        pkt.flags |= AV_PKT_FLAG_KEY;
         pkt.data = (uint8_t *)pic;
         pkt.size = sizeof(AVPicture);
         ret = av_write_frame(ffmpeg->oc, &pkt);
@@ -667,7 +673,7 @@ void ffmpeg_put_frame(struct ffmpeg *ffm
 #ifdef FFMPEG_AVWRITEFRAME_NEWAPI
             pkt.pts = AVSTREAM_CODEC_PTR(ffmpeg->video_st)->coded_frame->pts;
             if (AVSTREAM_CODEC_PTR(ffmpeg->video_st)->coded_frame->key_frame) {
-                pkt.flags |= PKT_FLAG_KEY;
+                pkt.flags |= AV_PKT_FLAG_KEY;
             }
             pkt.data = ffmpeg->video_outbuf;
             pkt.size = out_size;
