Index: ROX-Filer/src/Makefile.in
===================================================================
--- ROX-Filer/src/Makefile.in	(revision 4834)
+++ ROX-Filer/src/Makefile.in	(working copy)
@@ -51,10 +51,12 @@
 ${PROG}: ${OBJECTS}
 	${CC} -o "${PROG}" ${OBJECTS} ${LDFLAGS}
 	mv "${PROG}" "${PLATFORM_DIR}"
+
+${PROG}.dbg: ${PROG}
 	-(cd "${PLATFORM_DIR}" && \
-	 objcopy --only-keep-debug ROX-Filer ROX-Filer.dbg && \
-	 strip ROX-Filer && \
-	 objcopy --add-gnu-debuglink=ROX-Filer.dbg ROX-Filer)
+	 objcopy --only-keep-debug ${PROG} ${PROG}.dbg && \
+	 strip ${PROG} && \
+	 objcopy --add-gnu-debuglink=${PROG}.dbg ${PROG})
 
 clean:
 	rm -f *.o Makefile.bak
