--- blitz-0.9.orig/blitz/funcs.h
+++ blitz-0.9/blitz/funcs.h
@@ -30,6 +30,7 @@
 #include <blitz/blitz.h>
 #include <blitz/promote.h>
 #include <blitz/prettyprint.h>
+#include <cstdlib>
 
 BZ_NAMESPACE(blitz)
     
--- blitz-0.9.orig/blitz/prettyprint.h
+++ blitz-0.9/blitz/prettyprint.h
@@ -27,6 +27,7 @@
 
 #ifndef BZ_PRETTYPRINT_H
 #define BZ_PRETTYPRINT_H
+#include <cstdlib>
 
 BZ_NAMESPACE(blitz)
 
