--- libsfml-2.0.0_pre20121126/src/SFML/Graphics/CMakeLists.txt
+++ libsfml-2.0.0_pre20121126/src/SFML/Graphics/CMakeLists.txt
@@ -102,7 +102,7 @@
 endif()
 
 # add preprocessor symbols
-add_definitions(-DGLEW_STATIC -DSTBI_FAILURE_USERMSG)
+add_definitions(-DSTBI_FAILURE_USERMSG)
 
 # ImageLoader.cpp must be compiled with the -fno-strict-aliasing
 # when gcc is used; otherwise saving PNGs may crash in stb_image_write
