--- xjavac-20041208/XJavac.java.vanilla	2010-01-15 11:08:48 +0100
+++ xjavac-20041208/XJavac.java	2010-01-15 11:09:31 +0100
@@ -66,6 +66,8 @@
                 bcp.createPathElement().setPath(bcpMember.toString());
                 bcpMember.replace(javaHome.length(), bcpMember.length(), "/lib/core.jar:");
                 bcp.createPathElement().setPath(bcpMember.toString());
+                bcpMember.replace(javaHome.length(), bcpMember.length(),  "/lib/vm.jar:");
+                bcp.createPathElement().setPath(bcpMember.toString());
                 bcpMember.replace(javaHome.length(), bcpMember.length(),  "/lib/graphics.jar:");
                 bcp.createPathElement().setPath(bcpMember.toString());
                 bcpMember.replace(javaHome.length(), bcpMember.length(),  "/lib/javaws.jar:");
@@ -96,7 +98,11 @@
             // for Apple and Blackdown: a Linux port of Sun Java
             else if( (vendor.indexOf("SUN") >= 0) || 
                      (vendor.indexOf("BLACKDOWN") >= 0) || 
-                     (vendor.indexOf("APPLE") >= 0) ) {
+                     (vendor.indexOf("APPLE") >= 0) ||
+		     (vendor.indexOf("HEWLETT-PACKARD") >= 0) ||
+		     (vendor.indexOf("BEA") >= 0) ||
+	             (vendor.indexOf("FREE SOFTWARE FOUNDATION") >= 0) ||
+		     (vendor.indexOf("THE FREEBSD FOUNDATION") >= 0) ) {
                 // we're on an SUN 1.4; fiddle with the bootclasspath.
                 // since we can't eviscerate XML-related info here,
                 // we must use the classpath
