--- XJavac.java.vanilla	2006-08-05 21:43:45.000000000 +0200
+++ XJavac.java	2006-08-05 21:45:27.000000000 +0200
@@ -57,7 +57,7 @@
             }
             // this is supposed to be provided by all JVM's from time immemorial
             String vendor = ((String)props.get("java.vendor")).toUpperCase(Locale.ENGLISH);
-            if(vendor.indexOf("IBM") >= 0){
+            if((vendor.indexOf("IBM") >= 0) && (isJDK14())){
                 // we're on an IBM 1.4; fiddle with the bootclasspath.
                 Path bcp = createBootclasspath();
                 String javaHome = System.getProperty("java.home");
@@ -131,4 +131,8 @@
             !version.equals(JavaEnvUtils.JAVA_1_1) &&
             !version.equals(JavaEnvUtils.JAVA_1_0));
     }
+    private boolean isJDK14() {
+        final String version = JavaEnvUtils.getJavaVersion();
+        return version.equals(JavaEnvUtils.JAVA_1_4);
+    }
 }
