--- src/unix/native/jni/Tray.c	2007-10-20 21:57:59.000000000 +1300
+++ src/unix/native/jni/Tray.c	2007-10-21 00:19:25.000000000 +1300
@@ -73,9 +73,9 @@
  */
 static void *awtHandle = NULL;
 
-#ifdef __linux__
-#define LIBARCH  "i386"
-#else
+#ifdef __x86_64
+#define LIBARCH "amd64"
+#endif
 #ifdef __i386
 #define LIBARCH "i386"
 #endif
@@ -85,7 +85,6 @@
 #ifdef __sparcv9
 #define LIBARCH "sparcv9"
 #endif
-#endif
 
 #define REFLECT_VOID_FUNCTION(name, arglist, paramlist)			\
 typedef name##_type arglist;						\
