From bbe56f4018ba01575b7fbb8a8c14d3e5bb822caa Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Olivier=20Cr=C3=AAte?= <tester@tester.ca>
Date: Sun, 16 May 2010 14:11:50 -0400
Subject: [PATCH] Disable libproxy support

---
 butterfly/connection.py |   16 ----------------
 1 files changed, 0 insertions(+), 16 deletions(-)

diff --git a/butterfly/connection.py b/butterfly/connection.py
index 359b01a..9a2f73b 100644
--- a/butterfly/connection.py
+++ b/butterfly/connection.py
@@ -100,7 +100,6 @@ class ButterflyConnection(telepathy.server.Connection,
             # If the HTTP proxy parameters have been set, don't try any
             # others proxies automatically.
             if 'http' not in self._proxies:
-                self._fill_suggested_proxies()
                 self._use_next_proxy()
 
             self._manager = weakref.proxy(manager)
@@ -148,21 +147,6 @@ class ButterflyConnection(telepathy.server.Connection,
         papyon.event.InviteEventInterface.__init__(self, self._msn_client)
         papyon.event.OfflineMessagesEventInterface.__init__(self, self._msn_client)
 
-    def _fill_suggested_proxies(self):
-        try:
-            import libproxy
-        except ImportError:
-            return
-
-        factory = libproxy.ProxyFactory()
-        proxies = factory.getProxies('http://gateway.messenger.msn.com/')
-
-        # Remove socks proxies that papyon doesn't support.
-        proxies = [p for p in proxies if p.startswith('http://') or p == 'direct://']
-
-        if proxies:
-            self._suggested_proxies = proxies
-
     def _use_next_proxy(self):
         if not self._suggested_proxies:
             return False
-- 
1.6.4.4

