--- jni/GNUmakefile.orig	2009-06-03 21:32:31.000000000 +0200
+++ jni/GNUmakefile	2009-06-03 21:34:56.000000000 +0200
@@ -43,7 +43,7 @@
 #   http://weblogs.java.net/blog/kellyohair/archive/2006/01/compilation_of_1.html
 JFLAGS = -fno-omit-frame-pointer -fno-strict-aliasing -DNDEBUG
 OFLAGS = -O2 $(JFLAGS)
-WFLAGS = -W -Werror -Wall -Wno-unused -Wno-parentheses
+WFLAGS = -Wall -Wno-unused -Wno-parentheses
 PICFLAGS = -fPIC
 SOFLAGS = # Filled in for each OS specifically
 LDFLAGS += $(SOFLAGS)
@@ -55,7 +55,7 @@
 
 JDK_INCLUDES= -I"$(JDK_HOME)/include" -I"$(JDK_HOME)/include/$(OS)"
 IFLAGS = -I"$(BUILD_DIR)" -I"$(JFFI_SRC_DIR)" -I"$(LIBFFI_BUILD_DIR)/include"
-CFLAGS = $(OFLAGS) $(WFLAGS) $(IFLAGS) $(PICFLAGS) $(JDK_INCLUDES)
+CFLAGS += $(OFLAGS) $(WFLAGS) $(IFLAGS) $(PICFLAGS) $(JDK_INCLUDES)
 CFLAGS += -D_REENTRANT -D_LARGEFILE64_SOURCE -D_GNU_SOURCE
 
 ifeq ($(OS),mingw32)
@@ -65,9 +65,6 @@
   ifneq ($(OS),$(BUILD_OS))
     CC ?= i386-mingw32-gcc
     LD ?= i386-mingw32-ld
-  else
-    CC = gcc
-    LD = ld
   endif
 endif
 ifneq ($(findstring cygwin,$(BUILD_OS)),)
