--- jni/GNUmakefile.orig	2009-06-03 21:52:34.000000000 +0200
+++ jni/GNUmakefile	2009-06-03 21:51:50.000000000 +0200
@@ -42,7 +42,7 @@
 
 # MacOS headers aren't completely warning free, so turn them off
 ifneq (($OS),darwin)
-  WFLAGS = -W -Werror -Wall -Wno-unused -Wno-parentheses -Wundef
+  WFLAGS = -Wall -Wno-unused -Wno-parentheses -Wundef
 endif
 PICFLAGS = -fPIC
 SOFLAGS = # Filled in for each OS specifically
@@ -54,7 +54,7 @@
 
 JDK_INCLUDES= -I"$(JDK_HOME)/include" -I"$(JDK_HOME)/include/$(OS)"
 IFLAGS = -I"$(BUILD_DIR)" -I"$(JFFI_SRC_DIR)" -I"$(LIBFFI_BUILD_DIR)/include"
-CFLAGS = $(OFLAGS) $(WFLAGS) $(IFLAGS) $(PICFLAGS) $(JDK_INCLUDES)
+CFLAGS += $(OFLAGS) $(WFLAGS) $(IFLAGS) $(PICFLAGS) $(JDK_INCLUDES)
 CFLAGS += -D_REENTRANT -D_LARGEFILE64_SOURCE -D_GNU_SOURCE
 
 ifeq ($(OS),mingw32)
@@ -64,9 +64,6 @@
   ifneq ($(OS),$(BUILD_OS))
     CC ?= i386-mingw32-gcc
     LD ?= i386-mingw32-ld
-  else
-    CC = gcc
-    LD = ld
   endif
 endif
 ifneq ($(findstring cygwin,$(BUILD_OS)),)
