--- jni/GNUmakefile.orig	2009-12-30 18:59:22.000000000 +1300
+++ jni/GNUmakefile	2009-12-30 19:02:10.000000000 +1300
@@ -42,7 +42,7 @@
 
 # MacOS headers aren't completely warning free, so turn them off
 ifneq ($(OS),darwin)
-  WFLAGS = -W -Werror -Wall -Wno-unused -Wno-parentheses -Wundef
+  WFLAGS = -Wno-unused -Wno-parentheses -Wundef
 endif
 PICFLAGS = -fPIC
 SOFLAGS = # Filled in for each OS specifically
@@ -54,8 +54,8 @@
 STRIP = strip -S
 
 JDK_INCLUDES = -I"$(JDK_HOME)/include" -I"$(JDK_HOME)/include/$(OS)"
-IFLAGS = -I"$(BUILD_DIR)" -I"$(BUILD_DIR)"/jni -I"$(JFFI_SRC_DIR)" -I"$(LIBFFI_BUILD_DIR)"/include
-CFLAGS = $(OFLAGS) $(WFLAGS) $(IFLAGS) $(PICFLAGS) $(JDK_INCLUDES)
+IFLAGS = -I"$(BUILD_DIR)" -I"$(BUILD_DIR)"/jni -I"$(JFFI_SRC_DIR)"
+CFLAGS += $(OFLAGS) $(WFLAGS) $(IFLAGS) $(PICFLAGS) $(JDK_INCLUDES)
 CFLAGS += -D_REENTRANT -D_LARGEFILE64_SOURCE -D_GNU_SOURCE
 
 ifeq ($(OS),cross-mingw32)
@@ -147,11 +147,11 @@
 endif
 
 LIBJFFI = $(BUILD_DIR)/$(PREFIX)$(LIBNAME)-$(VERSION).$(JNIEXT)
-FFI_CONFIGURE = $(LIBFFI_SRC_DIR)/configure --disable-static \
-	--with-pic=yes --disable-dependency-tracking
-ifdef CONFIGURE_HOST
-	FFI_CONFIGURE += --host=$(CONFIGURE_HOST)
-endif
+#FFI_CONFIGURE = $(LIBFFI_SRC_DIR)/configure --disable-static \
+#	--with-pic=yes --disable-dependency-tracking
+#ifdef CONFIGURE_HOST
+#	FFI_CONFIGURE += --host=$(CONFIGURE_HOST)
+#endif
 all:	$(LIBJFFI)
 
 debug:
@@ -211,12 +211,6 @@
 else
 $(LIBFFI):		
 	@mkdir -p $(LIBFFI_BUILD_DIR)
-	@if [ ! -f $(LIBFFI_BUILD_DIR)/Makefile ]; then \
-	    echo "Configuring libffi for $(PLATFORM)"; \
-	    cd $(LIBFFI_BUILD_DIR) && env CC="$(FFI_CC)" LD="$(FFI_LD)" CFLAGS="$(FFI_CFLAGS)" \
-		$(FFI_CONFIGURE) > /dev/null; \
-	fi
-	$(MAKE) -C $(LIBFFI_BUILD_DIR)
 endif
 clean::
 	# nothing to do - ant will delete the build dir
