From: Julian Ospald <hasufell@gentoo.org>
Date: Wed Jan 29 02:31:24 UTC 2014
Subject: make paths modifiable

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,7 +6,7 @@
 endif()
 
 set(CMAKE_MODULE_PATH "${SOLARUS_ENGINE_SOURCE_DIR}/cmake/modules/")
-set(SOLARUS_INSTALL_DESTINATION "bin")
+set(SOLARUS_INSTALL_DESTINATION "bin" CACHE PATH "binary install destination")
 
 option(SOLARUS_USE_LUAJIT "Use LuaJIT instead of default Lua (recommended)" ON)
 
