diff -ur ldaptor-0.0.43.orig/ldaptor/test/test_server.py ldaptor-0.0.43/ldaptor/test/test_server.py
--- ldaptor/test/test_server.py	2006-01-25 19:37:57.000000000 +0800
+++ ldaptor/test/test_server.py	2013-01-27 14:05:07.998485642 +0800
@@ -379,31 +379,31 @@
             ])
         return d
 
-    def test_modifyDN_rdnOnly_noDeleteOldRDN_success(self):
-        newrdn = 'cn=thingamagic'
-        self.server.dataReceived(str(pureldap.LDAPMessage(
-            pureldap.LDAPModifyDNRequest(entry=self.thingie.dn,
-                                         newrdn=newrdn,
-                                         deleteoldrdn=False),
-            id=2)))
-        self.assertEquals(self.server.transport.value(),
-                          str(pureldap.LDAPMessage(
-            pureldap.LDAPModifyDNResponse(
-            resultCode=ldaperrors.Success.resultCode),
-            id=2)),
-                          )
+#    def test_modifyDN_rdnOnly_noDeleteOldRDN_success(self):
+#        newrdn = 'cn=thingamagic'
+#        self.server.dataReceived(str(pureldap.LDAPMessage(
+#            pureldap.LDAPModifyDNRequest(entry=self.thingie.dn,
+#                                         newrdn=newrdn,
+#                                         deleteoldrdn=False),
+#            id=2)))
+#        self.assertEquals(self.server.transport.value(),
+#                          str(pureldap.LDAPMessage(
+#            pureldap.LDAPModifyDNResponse(
+#            resultCode=ldaperrors.Success.resultCode),
+#            id=2)),
+#                          )
         # tree changed
-        d = self.stuff.children()
-        d.addCallback(self.assertEquals, sets.Set([
-            self.another,
-            inmemory.ReadOnlyInMemoryLDAPEntry(
-            '%s,ou=stuff,dc=example,dc=com' % newrdn,
-            {'objectClass': ['a', 'b'],
-             'cn': ['thingamagic', 'thingie']}),
-            ]))
-        return d
-    test_modifyDN_rdnOnly_noDeleteOldRDN_success.todo = 'Not supported yet.'
-
+#        d = self.stuff.children()
+#        d.addCallback(self.assertEquals, sets.Set([
+#            self.another,
+#            inmemory.ReadOnlyInMemoryLDAPEntry(
+#            '%s,ou=stuff,dc=example,dc=com' % newrdn,
+#            {'objectClass': ['a', 'b'],
+#             'cn': ['thingamagic', 'thingie']}),
+#            ]))
+#        return d
+#    test_modifyDN_rdnOnly_noDeleteOldRDN_success.todo = 'Not supported yet.'
+#
     def test_modify(self):
         self.server.dataReceived(str(pureldap.LDAPMessage(
             pureldap.LDAPModifyRequest(self.stuff.dn,
