diff -uNtr ldaptor-0.0.43.orig/ldaptor/apps/webui/iwebui.py ldaptor-0.0.43/ldaptor/apps/webui/iwebui.py
--- ldaptor-0.0.43.orig/ldaptor/apps/webui/iwebui.py	2008-08-20 12:38:44.000000000 -0430
+++ ldaptor-0.0.43/ldaptor/apps/webui/iwebui.py	2008-08-20 12:44:53.000000000 -0430
@@ -1,4 +1,4 @@
-from twisted.python import components
+from zope import interface
 
-class ICurrentDN(components.Interface):
+class ICurrentDN(interface.Interface):
     """Marker interface for current DN for Ldaptor-webui."""
diff -uNtr ldaptor-0.0.43.orig/ldaptor/apps/webui/search.py ldaptor-0.0.43/ldaptor/apps/webui/search.py
--- ldaptor-0.0.43.orig/ldaptor/apps/webui/search.py	2008-08-20 12:38:44.000000000 -0430
+++ ldaptor-0.0.43/ldaptor/apps/webui/search.py	2008-08-20 12:37:42.000000000 -0430
@@ -1,6 +1,7 @@
 from zope.interface import implements
 from twisted.internet import defer
-from twisted.python import components
+#from twisted.python import components
+from zope import interface
 from webut.skin import iskin
 from ldaptor.protocols.ldap import ldapclient, ldapsyntax
 from ldaptor.protocols.ldap import distinguishedname, ldapconnector
@@ -17,7 +18,7 @@
 from nevow import rend, inevow, loaders, url, tags
 from formless import annotate, webform, iformless, configurable
 
-class IMove(components.Interface):
+class IMove(interface.Interface):
     """Entries being moved in the tree."""
     pass
 
diff -uNtr ldaptor-0.0.43.orig/ldaptor/interfaces.py ldaptor-0.0.43/ldaptor/interfaces.py
--- ldaptor-0.0.43.orig/ldaptor/interfaces.py	2008-08-20 12:38:44.000000000 -0430
+++ ldaptor-0.0.43/ldaptor/interfaces.py	2008-08-20 12:38:19.000000000 -0430
@@ -1,6 +1,7 @@
-from twisted.python import components
+#from twisted.python import interface
+from zope import interface
 
-class ILDAPEntry(components.Interface):
+class ILDAPEntry(interface.Interface):
     """
 
     Pythonic API for LDAP object access and modification.
@@ -126,7 +127,7 @@
         incorrect.
         """
 
-class IEditableLDAPEntry(components.Interface):
+class IEditableLDAPEntry(interface.Interface):
     """Interface definition for editable LDAP entries."""
 
     def __setitem__(self, key, value):
@@ -206,7 +207,7 @@
 
         """
 
-class IConnectedLDAPEntry(components.Interface):
+class IConnectedLDAPEntry(interface.Interface):
     """Interface definition for LDAP entries that are part of a bigger whole."""
 
     def namingContext(self):
@@ -334,7 +335,7 @@
 
         """
 
-class ILDAPConfig(components.Interface):
+class ILDAPConfig(interface.Interface):
     """Generic LDAP configuration retrieval."""
 
     def getBaseDN(self):
