diff -uNtr ldaptor-0.0.43.orig/ldaptor/usage.py ldaptor-0.0.43/ldaptor/usage.py
--- ldaptor-0.0.43.orig/ldaptor/usage.py	2008-08-20 12:38:44.000000000 -0430
+++ ldaptor-0.0.43/ldaptor/usage.py	2008-08-20 20:02:14.000000000 -0430
@@ -2,6 +2,9 @@
 from ldaptor.protocols import pureldap
 from ldaptor.protocols.ldap import distinguishedname
 
+class UsageError(usage.UsageError):
+    pass
+
 class Options(usage.Options):
     optParameters = ()
     def postOptions(self):
