From d65e37e05203a4e88b42680fd9504c54833f28a6 Mon Sep 17 00:00:00 2001
From: Luke Dashjr <luke-jr+git@utopios.org>
Date: Fri, 14 Jun 2013 18:06:02 +0000
Subject: [PATCH] Bugfix: opencl: Add missing include for fpgautils.h (needed
 for open_bitstream)

---
 ocl.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ocl.c b/ocl.c
index 15bd86c..12a3d29 100644
--- a/ocl.c
+++ b/ocl.c
@@ -34,6 +34,7 @@
 #define OMIT_OPENCL_API
 
 #include "findnonce.h"
+#include "fpgautils.h"
 #include "ocl.h"
 
 /* Platform API */
-- 
1.8.1.5

