From f7125dc965963bd9860dc92616e2499a02297587 Mon Sep 17 00:00:00 2001
From: Luke Dashjr <luke-jr+git@utopios.org>
Date: Thu, 26 Apr 2012 21:25:32 -0400
Subject: [PATCH] Bugfix: Fix building without OpenCL support

---
 driver-opencl.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/driver-opencl.c b/driver-opencl.c
index 82b8cf3..2e1d4cd 100644
--- a/driver-opencl.c
+++ b/driver-opencl.c
@@ -44,6 +44,7 @@
 /* TODO: cleanup externals ********************/
 
 
+#ifdef HAVE_OPENCL
 /* Platform API */
 CL_API_ENTRY cl_int CL_API_CALL
 (*clGetPlatformIDs)(cl_uint          /* num_entries */,
@@ -248,6 +249,7 @@ load_opencl_symbols() {
 	
 	return true;
 }
+#endif
 
 
 #ifdef HAVE_CURSES
-- 
1.7.3.4

