--- muttprint-0.72d/langinfo/Makefile
+++ muttprint-0.72d/langinfo/Makefile
@@ -21,13 +21,13 @@
 all: muttprint-langinfo
 
 muttprint-langinfo: main.o muttprint-langinfo.o muttprint-langinfo.h
-	$(CC) $(CCFLAGS) -o muttprint-langinfo muttprint-langinfo.o main.o
+	$(CC) $(LDFLAGS) -o muttprint-langinfo muttprint-langinfo.o main.o
 
 main.o: main.c muttprint-langinfo.h
-	$(CC) $(CCFLAGS) -c -o main.o main.c
+	$(CC) $(CFLAGS) -c -o main.o main.c
 
 muttprint-langinfo.o: muttprint-langinfo.c
-	$(CC) $(CCFLAGS) -c -o muttprint-langinfo.o muttprint-langinfo.c
+	$(CC) $(CFLAGS) -c -o muttprint-langinfo.o muttprint-langinfo.c
 	
 install: 
 	test -x muttprint-langinfo && \
