--- muttprint-0.72d/doc/manpages/de/muttprint.pod
+++ muttprint-0.72d/doc/manpages/de/muttprint.pod
@@ -22,7 +22,7 @@
 nur die wichtigen Header und nicht alle.
 
 Eine ausfürliche Benutzeranleitung finden Sie in 
-F<##DOCDIR##/muttprint/> im PDF- und HTML-Format.
+F<##DOCDIR##/muttprint/> im PDF--Format.
 
 Wie auch immer, Sie müssen folgende Zeile in Ihre F<##SYSCONFDIR##/Muttrc> oder
 F<~/.muttrc> eintragen:
--- muttprint-0.72d/doc/manpages/en/muttprint.pod
+++ muttprint-0.72d/doc/manpages/en/muttprint.pod
@@ -22,7 +22,7 @@
 whole plethora of them.
 
 For detailed information about Muttprint look read the User's guide in PDF 
-and HTML format at F<##DOCDIR##/muttprint/>.
+format at F<##DOCDIR##/muttprint/>.
 
 Anyway, you have to put the following line in your F<##SYSCONFDIR##/Muttrc> or 
 F<~/.muttrc>:
--- muttprint-0.72d/doc/manual/de/Makefile
+++ muttprint-0.72d/doc/manual/de/Makefile
@@ -39,7 +39,7 @@
 
 #
 # we will not make 'all' but only the 'standard' by default
-default: pdf html sed-clean
+default: pdf sed-clean
 all: ps dvi pdf html rtf sed-clean
 ps: manual-$(lang).ps
 pdf: manual-$(lang).pdf
@@ -88,10 +88,10 @@
 ##install: manual-$(lang).pdf manual-$(lang)/muttprint.html
 install: 
 	$(INSTALL) *.pdf $(docdir)/$(docdirname)
-	mkdir -p $(docdir)/$(docdirname)/manual-$(lang)
+	#mkdir -p $(docdir)/$(docdirname)/manual-$(lang)
 	# ignore the stylesheet-images/ directory because the files are not needed
-	$(INSTALL) manual-$(lang)/*.html $(docdir)/$(docdirname)/manual-$(lang)/
-	$(INSTALL) manual-$(lang)/*.css $(docdir)/$(docdirname)/manual-$(lang)/
+	#$(INSTALL) manual-$(lang)/*.html $(docdir)/$(docdirname)/manual-$(lang)/
+	#$(INSTALL) manual-$(lang)/*.css $(docdir)/$(docdirname)/manual-$(lang)/
 
 gzip:
 	$(gzip) *.ps *.txt
--- muttprint-0.72d/doc/manual/en/Makefile
+++ muttprint-0.72d/doc/manual/en/Makefile
@@ -39,7 +39,7 @@
 
 #
 # we will not make 'all' but only the 'standard' by default
-default: pdf html sed-clean
+default: pdf sed-clean
 all: ps dvi pdf html rtf sed-clean
 ps: manual-$(lang).ps
 pdf: manual-$(lang).pdf
@@ -88,10 +88,10 @@
 ##install: manual-$(lang).pdf manual-$(lang)/muttprint.html
 install: 
 	$(INSTALL) *.pdf $(docdir)/$(docdirname)
-	mkdir -p $(docdir)/$(docdirname)/manual-$(lang)
+	#mkdir -p $(docdir)/$(docdirname)/manual-$(lang)
 	# ignore the stylesheet-images/ directory because the files are not needed
-	$(INSTALL) manual-$(lang)/*.html $(docdir)/$(docdirname)/manual-$(lang)/
-	$(INSTALL) manual-$(lang)/*.css $(docdir)/$(docdirname)/manual-$(lang)/
+	#$(INSTALL) manual-$(lang)/*.html $(docdir)/$(docdirname)/manual-$(lang)/
+	#$(INSTALL) manual-$(lang)/*.css $(docdir)/$(docdirname)/manual-$(lang)/
 
 gzip:
 	$(gzip) *.ps *.txt
--- muttprint-0.72d/doc/manual/es/Makefile
+++ muttprint-0.72d/doc/manual/es/Makefile
@@ -39,7 +39,7 @@
 
 #
 # we will not make 'all' but only the 'standard' by default
-default: pdf html sed-clean
+default: pdf sed-clean
 all: ps dvi pdf html rtf sed-clean
 ps: manual-$(lang).ps
 pdf: manual-$(lang).pdf
@@ -88,10 +88,10 @@
 ##install: manual-$(lang).pdf manual-$(lang)/muttprint.html
 install: 
 	$(INSTALL) *.pdf $(docdir)/$(docdirname)
-	mkdir -p $(docdir)/$(docdirname)/manual-$(lang)
+	#mkdir -p $(docdir)/$(docdirname)/manual-$(lang)
 	# ignore the stylesheet-images/ directory because the files are not needed
-	$(INSTALL) manual-$(lang)/*.html $(docdir)/$(docdirname)/manual-$(lang)/
-	$(INSTALL) manual-$(lang)/*.css $(docdir)/$(docdirname)/manual-$(lang)/
+	#$(INSTALL) manual-$(lang)/*.html $(docdir)/$(docdirname)/manual-$(lang)/
+	#$(INSTALL) manual-$(lang)/*.css $(docdir)/$(docdirname)/manual-$(lang)/
 
 gzip:
 	$(gzip) *.ps *.txt
--- muttprint-0.72d/doc/manual/it/Makefile
+++ muttprint-0.72d/doc/manual/it/Makefile
@@ -39,7 +39,7 @@
 
 #
 # we will not make 'all' but only the 'standard' by default
-default: pdf html sed-clean
+default: pdf sed-clean
 all: ps dvi pdf html rtf sed-clean
 ps: manual-$(lang).ps
 pdf: manual-$(lang).pdf
@@ -88,10 +88,10 @@
 ##install: manual-$(lang).pdf manual-$(lang)/muttprint.html
 install: 
 	$(INSTALL) *.pdf $(docdir)/$(docdirname)
-	mkdir -p $(docdir)/$(docdirname)/manual-$(lang)
+	#mkdir -p $(docdir)/$(docdirname)/manual-$(lang)
 	# ignore the stylesheet-images/ directory because the files are not needed
-	$(INSTALL) manual-$(lang)/*.html $(docdir)/$(docdirname)/manual-$(lang)/
-	$(INSTALL) manual-$(lang)/*.css $(docdir)/$(docdirname)/manual-$(lang)/
+	#$(INSTALL) manual-$(lang)/*.html $(docdir)/$(docdirname)/manual-$(lang)/
+	#$(INSTALL) manual-$(lang)/*.css $(docdir)/$(docdirname)/manual-$(lang)/
 
 gzip:
 	$(gzip) *.ps *.txt
--- muttprint-0.72d/doc/manual/sl/Makefile
+++ muttprint-0.72d/doc/manual/sl/Makefile
@@ -39,7 +39,7 @@
 
 #
 # we will not make 'all' but only the 'standard' by default
-default: pdf html sed-clean
+default: pdf sed-clean
 all: ps dvi pdf html rtf sed-clean
 ps: manual-$(lang).ps
 pdf: manual-$(lang).pdf
@@ -88,10 +88,10 @@
 ##install: manual-$(lang).pdf manual-$(lang)/muttprint.html
 install: 
 	$(INSTALL) *.pdf $(docdir)/$(docdirname)
-	mkdir -p $(docdir)/$(docdirname)/manual-$(lang)
+	#mkdir -p $(docdir)/$(docdirname)/manual-$(lang)
 	# ignore the stylesheet-images/ directory because the files are not needed
-	$(INSTALL) manual-$(lang)/*.html $(docdir)/$(docdirname)/manual-$(lang)/
-	$(INSTALL) manual-$(lang)/*.css $(docdir)/$(docdirname)/manual-$(lang)/
+	#$(INSTALL) manual-$(lang)/*.html $(docdir)/$(docdirname)/manual-$(lang)/
+	#$(INSTALL) manual-$(lang)/*.css $(docdir)/$(docdirname)/manual-$(lang)/
 
 gzip:
 	$(gzip) *.ps *.txt

