--- muttprint.old	2005-08-04 10:46:25.000000000 +0200
+++ muttprint	2005-08-04 10:47:25.000000000 +0200
@@ -362,32 +362,7 @@
 	
 		#
 		# signature 
-		if (((/$Config{SIG_REGEXP}/o && !($Config{REM_SIG} eq "on")) ||
-			$signature_mode) && $Config{VERBATIMSIG} ne "raw") {
-
-			if (/$Config{SIG_REGEXP}/o) {
-				$signature_mode = TRUE;
-
-				# Leerzeile bei 2. Signatur
-				if (defined $signature) {
-					$signature .= "\n";
-				}
-			} 
-			else {
-				$signature .= $_;
-			}
-
-			# 2 Leerzeilen => Ende der Signatur
-			if (/^$/ and $sig_mod_counter == 0) {
-				$sig_mod_counter++;
-			} elsif (/^$/ and $sig_mod_counter == 1) {
-				$signature_mode = FALSE;
-			} else {
-				$sig_mod_counter = 0;
-			}
-
-			next;
-		}
+		last if (/$Config{SIG_REGEXP}/o && ($Config{REM_SIG} eq "on"));
 	
 		#
 		# and what's about Quoting?
