diff -urN WebMO.install.orig/cgi-bin/interfaces/gamess.int.disabled WebMO.install/cgi-bin/interfaces/gamess.int.disabled
--- WebMO.install.orig/cgi-bin/interfaces/gamess.int.disabled	2007-05-07 23:31:24.000000000 -0700
+++ WebMO.install/cgi-bin/interfaces/gamess.int.disabled	2007-05-07 23:32:19.000000000 -0700
@@ -2,7 +2,7 @@
 interfaceDescription{'gamess_admin'}="Edit preferences specific to Gamess"
 interfaceDescription{'gamess_name'}="Gamess"
 gamessVersion="2006"
-gamessGMSPATH="/usr/local/gamess"
+gamessGMSPATH="/usr/bin"
 gamessBinary="gamess.00.x"
 gamessDdikick="ddikick.x"
 cpuMin="0"
diff -urN WebMO.install.orig/cgi-bin/interfaces/mopac.int.disabled WebMO.install/cgi-bin/interfaces/mopac.int.disabled
--- WebMO.install.orig/cgi-bin/interfaces/mopac.int.disabled	2007-05-07 23:31:24.000000000 -0700
+++ WebMO.install/cgi-bin/interfaces/mopac.int.disabled	2007-05-07 23:31:43.000000000 -0700
@@ -2,8 +2,8 @@
 interfaceDescription{'mopac_admin'}="Edit preferences specific to Mopac"
 interfaceDescription{'mopac_name'}="Mopac"
 mopacVersion="7"
-mopacPath="/usr/local/mopac7/mopac.exe"
-mopacScratch="/scratch"
+mopacPath="/usr/bin/mopac7"
+mopacScratch="/tmp"
 mopacExternal="/usr/local/mopac7"
 cpuMin="0"
 cpuMax="100"
diff -urN WebMO.install.orig/cgi-bin/interfaces/tinker.int.disabled WebMO.install/cgi-bin/interfaces/tinker.int.disabled
--- WebMO.install.orig/cgi-bin/interfaces/tinker.int.disabled	2007-05-07 23:31:24.000000000 -0700
+++ WebMO.install/cgi-bin/interfaces/tinker.int.disabled	2007-05-07 23:31:43.000000000 -0700
@@ -2,9 +2,9 @@
 interfaceDescription{'tinker_admin'}="Edit preferences specific to Tinker"
 interfaceDescription{'tinker_name'}="Tinker"
 tinkerVersion="4.2"
-tinkerBinDir="/usr/local/tinker/bin"
-tinkerBinExt="x"
-tinkerParamDir="/usr/local/tinker/params"
+tinkerBinDir="/usr/bin"
+tinkerBinExt=""
+tinkerParamDir="/usr/share/tinker/params"
 cpuMin="0"
 cpuMax="100"
 cpuDefault="100"
