diff -ur WebMO.install.orig/cgi-bin/interfaces/gamess.int.disabled WebMO.install/cgi-bin/interfaces/gamess.int.disabled
--- WebMO.install.orig/cgi-bin/interfaces/gamess.int.disabled	2008-02-15 00:37:31.000000000 -0800
+++ WebMO.install/cgi-bin/interfaces/gamess.int.disabled	2008-02-15 00:37:56.000000000 -0800
@@ -3,6 +3,7 @@
 interfaceDescription{'gamess_name'}="Gamess"
 gamessVersion="2007"
 gamessGMSPATH="/usr/bin"
+gamessDATAPATH="/usr/share/gamess/ericfmt"
 gamessBinary="gamess.00.x"
 gamessDdikick="ddikick.x"
 cpuMin="0"
diff -ur WebMO.install.orig/cgi-bin/run_gamess.cgi WebMO.install/cgi-bin/run_gamess.cgi
--- WebMO.install.orig/cgi-bin/run_gamess.cgi	2008-02-15 00:37:31.000000000 -0800
+++ WebMO.install/cgi-bin/run_gamess.cgi	2008-02-15 00:38:12.000000000 -0800
@@ -30,7 +30,7 @@
 $ENV{'IRCDATA'} = $output_file.".irc";
 $ENV{'PUNCH'} = $output_file.".dat";
 $ENV{'INPUT'} = "$input_file.inp";
-$ENV{'ERICFMT'} = "$gamessGMSPATH/ericfmt.dat";
+$ENV{'ERICFMT'} = "$gamessDATAPATH/ericfmt.dat";
 $ENV{'PATH'} = $ENV{'PATH'}.":".$gamessGMSPATH;
 $ENV{'DDI_RSH'}=$serverShell;
 
diff -ur WebMO.install.orig/source/gamessmgr_admin.html WebMO.install/source/gamessmgr_admin.html
--- WebMO.install.orig/source/gamessmgr_admin.html	2008-02-15 00:37:31.000000000 -0800
+++ WebMO.install/source/gamessmgr_admin.html	2008-02-15 00:38:57.000000000 -0800
@@ -30,10 +30,14 @@
 							<TD><INPUT TYPE="text" NAME="gamessVersion" VALUE="<PERL>$gamessVersion</PERL>" SIZE=50></TD>
 						</TR>
 						<TR>
-							<TD><B>GAMESS directory</B></TD>
+							<TD><B>GAMESS binary directory</B></TD>
 							<TD><INPUT TYPE="text" NAME="gamessGMSPATH" VALUE="<PERL>$gamessGMSPATH</PERL>" SIZE=50></TD>
 						</TR>
 						<TR>
+							<TD><B>GAMESS data directory</B></TD>
+							<TD><INPUT TYPE="text" NAME="gamessDATAPATH" VALUE="<PERL>$gamessDATAPATH</PERL>" SIZE=50></TD>
+						</TR>
+						<TR>
 							<TD><B>GAMESS binary (name, not path)</B></TD>
 							<TD><INPUT TYPE="text" NAME="gamessBinary" VALUE="<PERL>$gamessBinary</PERL>" SIZE=50></TD>
 						</TR>
