diff -urN WebMO.install.orig/setup.pl WebMO.install/setup.pl
--- WebMO.install.orig/setup.pl	2005-11-21 07:40:28.000000000 -0800
+++ WebMO.install/setup.pl	2005-12-04 00:39:41.000000000 -0800
@@ -82,7 +82,7 @@
 #get a username
 my $username = `whoami`; chomp $username;
 my $homedir = $ENV{'HOME'} ne "" ? $ENV{'HOME'} : `echo ~`; chomp $homedir;
-if ($username eq "root")
+if ($username eq "root" && !$unattended_setup)
 {
 	print <<END;
 	
@@ -693,6 +693,7 @@
 print "finished\n";
 
 #run DIAGNOSE
+if (!$unattended_setup) {
 print "Running diagnostic tests...";
 system("$perlPath scripts/diagnose.pl $cgiBase/interfaces/globals.int");
 print "finished\n";
@@ -714,6 +715,7 @@
 		system("$browsers[0] file:$cwd/diagnose.html");
 	}
 }
+}
 
 #setup computational servers
 mkdir "$cgiBase/servers", 0777;
