--- ./Makefile.in.old	2009-07-30 12:17:10.000000000 +0000
+++ Makefile.in	2009-07-30 12:19:55.000000000 +0000
@@ -299,7 +299,7 @@
 Makefile: @srcdir@/Makefile.in $(MKDEPS)
 	./config.status
 
-yate: libyate.so $(OBJS) $(LIBS)
+yate: ${OBJS} libyate.so $(LIBS)
 	$(LINK) -o $@ $(LIBTHR) $^
 
 libyate.so: $(YLIB)
--- ./modules/Makefile.in.old	2009-07-30 12:21:07.000000000 +0000
+++ ./modules/Makefile.in	2009-07-30 12:25:18.000000000 +0000
@@ -14,7 +14,7 @@
 INCLUDES := -I.. -I@top_srcdir@
 CFLAGS += -Wall @MODULE_CFLAGS@ @INLINE_FLAGS@
 CXXFLAGS += -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
-LDFLAGS:= @LDFLAGS@ -L.. -lyate
+LDFLAGS += -L..
 MODFLAGS:= @MODULE_LDFLAGS@
 MODRELAX:= @MODULE_LDRELAX@
 MODSTRIP:= @MODULE_SYMBOLS@
@@ -106,7 +106,7 @@
 endif
 
 LOCALFLAGS =
-LOCALLIBS =
+LOCALLIBS = -lyate
 CCOMPILE = $(CC) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
 COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CXX) $(LDFLAGS)
@@ -198,74 +198,74 @@
 server/ysigchan.yate server/analog.yate: LOCALFLAGS = -I@top_srcdir@/libs/ysig
 server/wpcard.yate: LOCALFLAGS = -I@top_srcdir@/libs/ysig @WANPIPE_HWEC_INC@
 server/zapcard.yate: LOCALFLAGS = -I@top_srcdir@/libs/ysig @ZAP_FLAGS@
-server/ysigchan.yate server/wpcard.yate server/zapcard.yate server/analog.yate: LOCALLIBS = -lyatesig
+server/ysigchan.yate server/wpcard.yate server/zapcard.yate server/analog.yate: LOCALLIBS += -lyatesig
 
 server/analogdetect.yate: ../libs/ymodem/libyatemodem.a
 server/analogdetect.yate: LOCALFLAGS = -I@top_srcdir@/libs/ymodem
-server/analogdetect.yate: LOCALLIBS = -L../libs/ymodem -lyatemodem
+server/analogdetect.yate: LOCALLIBS += -L../libs/ymodem -lyatemodem
 
 h323chan.yate: LOCALFLAGS = -DPHAS_TEMPLATES -D_REENTRANT -DP_HAS_SEMAPHORES @H323_INC@
-h323chan.yate: LOCALLIBS = @H323_LIB@
+h323chan.yate: LOCALLIBS += @H323_LIB@
 
 server/pgsqldb.yate: LOCALFLAGS = @PGSQL_INC@
-server/pgsqldb.yate: LOCALLIBS = -lpq
+server/pgsqldb.yate: LOCALLIBS += -lpq
 
 server/mysqldb.yate: LOCALFLAGS = @MYSQL_INC@
-server/mysqldb.yate: LOCALLIBS = @MYSQL_LIB@
+server/mysqldb.yate: LOCALLIBS += @MYSQL_LIB@
 
-enumroute.yate: LOCALLIBS = @RESOLV_LIB@
+enumroute.yate: LOCALLIBS += @RESOLV_LIB@
 
-client/alsachan.yate: LOCALLIBS = -lasound
+client/alsachan.yate: LOCALLIBS += -lasound
 
 yiaxchan.yate: ../libs/yiax/libyateiax.a
 yiaxchan.yate: LOCALFLAGS = -I@top_srcdir@/libs/yiax
-yiaxchan.yate: LOCALLIBS = -L../libs/yiax -lyateiax
+yiaxchan.yate: LOCALLIBS += -L../libs/yiax -lyateiax
 
 yjinglechan.yate: ../libs/yxml/libyatexml.a ../libs/yjingle/libyatejingle.a
 yjinglechan.yate: LOCALFLAGS = -I@top_srcdir@/libs/yxml -I@top_srcdir@/libs/yjingle
-yjinglechan.yate: LOCALLIBS = -L../libs/yjingle -L../libs/yxml -lyatejingle -lyatexml
+yjinglechan.yate: LOCALLIBS += -L../libs/yjingle -L../libs/yxml -lyatejingle -lyatexml
 
 server/dbpbx.yate server/pbxassist.yate: ../libs/ypbx/libyatepbx.a
 server/dbpbx.yate server/pbxassist.yate: LOCALFLAGS = -I@top_srcdir@/libs/ypbx
-server/dbpbx.yate server/pbxassist.yate: LOCALLIBS = ../libs/ypbx/libyatepbx.a
+server/dbpbx.yate server/pbxassist.yate: LOCALLIBS += ../libs/ypbx/libyatepbx.a
 
 server/mgcpgw.yate server/mgcpca.yate: ../libs/ymgcp/libyatemgcp.a
 server/mgcpgw.yate server/mgcpca.yate: LOCALFLAGS = -I@top_srcdir@/libs/ymgcp
-server/mgcpgw.yate server/mgcpca.yate: LOCALLIBS = ../libs/ymgcp/libyatemgcp.a
+server/mgcpgw.yate server/mgcpca.yate: LOCALLIBS += ../libs/ymgcp/libyatemgcp.a
 
 ifeq (@SYSTEM_ILBC@,yes)
-ilbccodec.yate: LOCALLIBS = -lilbc
+ilbccodec.yate: LOCALLIBS += -lilbc
 else
 ilbccodec.yate: ../libs/ilbc/libilbc.a
-ilbccodec.yate: LOCALLIBS = ../libs/ilbc/libilbc.a
+ilbccodec.yate: LOCALLIBS += ../libs/ilbc/libilbc.a
 endif
 ilbccodec.yate: LOCALFLAGS = @ILBC_INC@
 
-gsmcodec.yate: LOCALLIBS = -lgsm
+gsmcodec.yate: LOCALLIBS += -lgsm
 gsmcodec.yate: LOCALFLAGS = @GSM_INC@
 
-speexcodec.yate: LOCALLIBS = -lspeex
+speexcodec.yate: LOCALLIBS += -lspeex
 speexcodec.yate: LOCALFLAGS = @SPEEX_INC@
 
 amrnbcodec.yate: LOCALFLAGS = @AMRNB_INC@
-amrnbcodec.yate: LOCALLIBS = @AMRNB_LIB@
+amrnbcodec.yate: LOCALLIBS += @AMRNB_LIB@
 
-faxchan.yate: LOCALLIBS = -lspandsp
+faxchan.yate: LOCALLIBS += -lspandsp
 faxchan.yate: LOCALFLAGS = @SPANDSP_INC@
 
 ysipchan.yate: ../libs/ysip/libyatesip.a
 ysipchan.yate: LOCALFLAGS = -I@top_srcdir@/libs/ysip
-ysipchan.yate: LOCALLIBS = ../libs/ysip/libyatesip.a
+ysipchan.yate: LOCALLIBS += ../libs/ysip/libyatesip.a
 
 yrtpchan.yate: ../libs/yrtp/libyatertp.a
 yrtpchan.yate: LOCALFLAGS = -I@top_srcdir@/libs/yrtp
-yrtpchan.yate: LOCALLIBS = ../libs/yrtp/libyatertp.a
+yrtpchan.yate: LOCALLIBS += ../libs/yrtp/libyatertp.a
 
 openssl.yate: LOCALFLAGS = @OPENSSL_INC@
-openssl.yate: LOCALLIBS = @OPENSSL_LIB@
+openssl.yate: LOCALLIBS += @OPENSSL_LIB@
 
 rmanager.yate: LOCALFLAGS = $(COREDUMP_INC)
-rmanager.yate: LOCALLIBS = $(COREDUMP_LIB)
+rmanager.yate: LOCALLIBS += $(COREDUMP_LIB)
 
 ../libyatesig.so ../libs/ysig/libyatesig.a:
 	$(MAKE) -C ../libs/ysig
--- ./engine/Makefile.in.old	2009-07-30 12:15:05.000000000 +0000
+++ ./engine/Makefile.in	2009-07-30 12:15:32.000000000 +0000
@@ -120,4 +120,4 @@
 	cd .. && ./config.status
 
 ../$(YLIB): $(LIBOBJS) $(LIBS)
-	$(LINK) -shared -o $@ -Wl,--soname=$(YLIB) $(LIBTHR) $^ $(LIBAUX)
+	$(LINK) -shared -o $@ -Wl,--soname=$(YLIB) $^ $(LIBTHR) $(LIBAUX)
