--- Makefile.in.old	2009-07-28 01:23:48.000000000 +0200
+++ Makefile.in	2009-07-28 01:00:34.000000000 +0200
@@ -7,12 +7,12 @@
 # override DEBUG at compile time to enable full debug or remove it all
 DEBUG :=
 
-CXX := @CXX@ -Wall
+CXX := @CXX@
 SED := sed
 DEFS :=
 LIBTHR:= -lpthread
 INCLUDES := -I. -I@srcdir@
-CFLAGS := -O2 @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CXXFLAGS += -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@
 LDCONFIG:=true
 
@@ -28,7 +28,7 @@
 OBJS := main.o
 
 CLEANS = $(PROGS) $(SLIBS) $(LIBS) $(OBJS) yatepaths.h core
-COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
+COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CXX) $(LDFLAGS)
 
 DOCGEN_F := $(INCS)
--- engine/Makefile.in.old	2009-07-28 01:04:58.000000000 +0200
+++ engine/Makefile.in	2009-07-28 01:05:53.000000000 +0200
@@ -4,15 +4,15 @@
 # override DEBUG at compile time to enable full debug or remove it all
 DEBUG :=
 
-CC  := @CC@ -Wall
-CXX := @CXX@ -Wall
+CC  := @CC@
+CXX := @CXX@
 SED := sed
 DEFS :=
 LIBAUX:= @DLOPEN_LIB@
 LIBTHR:= -lpthread
 INCLUDES := -I.. -I@top_srcdir@
-CFLAGS := -O2 @MODULE_CFLAGS@ @INLINE_FLAGS@
-CPPFLAGS := -O2 @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CFLAGS += -Wall @MODULE_CFLAGS@ @INLINE_FLAGS@
+CXXFLAGS += -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@
 LDCONFIG:=true
 
@@ -32,7 +32,7 @@
 LIBOBJS := $(CLSOBJS) $(ENGOBJS) $(TELOBJS) $(CLIOBJS)
 CLEANS = $(LIBOBJS) core
 CCOMPILE = $(CC) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
-COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CPPFLAGS)
+COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CXX) $(LDFLAGS)
 SCTPOPTS:=
 
--- clients/qt4/Makefile.in.old	2009-07-28 01:02:13.000000000 +0200
+++ clients/qt4/Makefile.in	2009-07-28 01:02:58.000000000 +0200
@@ -4,13 +4,13 @@
 # override DEBUG at compile time to enable full debug or remove it all
 DEBUG :=
 
-CXX := @CXX@ -Wall
+CXX := @CXX@
 AR  := ar
 MOC := @QT4_MOC@
 DEFS:=
 
 INCLUDES:=-I. -I@srcdir@ -I@top_srcdir@ @QT4_INC@
-CXXFLAGS:=$(CXXFLAGS) @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CXXFLAGS:= -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@ -L../.. -lyate
 INCFILES := @top_srcdir@/yateclass.h @top_srcdir@/yatecbase.h @srcdir@/qt4client.h
 
--- clients/Makefile.in.old	2009-07-28 01:01:13.000000000 +0200
+++ clients/Makefile.in	2009-07-28 01:02:00.000000000 +0200
@@ -7,12 +7,12 @@
 # override DEBUG at compile time to enable full debug or remove it all
 DEBUG :=
 
-CXX := @CXX@ -Wall
+CXX := @CXX@
 SED := sed
 DEFS :=
 LIBTHR:= -lpthread
 INCLUDES := -I.. -I@top_srcdir@
-CFLAGS := -O2 @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CXXFLAGS += -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@ -L.. -lyate
 INCFILES := @top_srcdir@/yatengine.h @top_srcdir@/yatephone.h ../yateversn.h
 
@@ -38,7 +38,7 @@
 
 LOCALFLAGS =
 LOCALLIBS =
-COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
+COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CXX) $(LDFLAGS)
 
 prefix = @prefix@
--- modules/test/Makefile.in.old	2009-07-28 01:07:12.000000000 +0200
+++ modules/test/Makefile.in	2009-07-28 01:08:05.000000000 +0200
@@ -1,11 +1,11 @@
 # Makefile
 # This file holds the make rules for the Telephony Engine test cases
 
-CXX := @CXX@ -Wall
+CXX := @CXX@
 SED := sed
 DEFS :=
 INCLUDES := -I@top_srcdir@
-CFLAGS := -O0 @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CXXFLAGS += -O0 -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@ -L../.. -lyate
 MODFLAGS:= @MODULE_LDFLAGS@
 MODSTRIP:= @MODULE_SYMBOLS@
@@ -17,7 +17,7 @@
 
 LOCALFLAGS =
 LOCALLIBS =
-COMPILE = $(CXX) $(DEFS) $(INCLUDES) $(CFLAGS)
+COMPILE = $(CXX) $(DEFS) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CXX) $(LDFLAGS)
 MODLINK = $(CXX) $(MODFLAGS) $(MODSTRIP) $(LDFLAGS)
 MODCOMP = $(COMPILE) $(MODFLAGS) $(MODSTRIP) $(LDFLAGS)
--- modules/Makefile.in.old	2009-07-28 01:09:55.000000000 +0200
+++ modules/Makefile.in	2009-07-28 01:10:29.000000000 +0200
@@ -7,13 +7,13 @@
 # override DEBUG at compile time to enable full debug or remove it all
 DEBUG :=
 
-CC  := @CC@ -Wall
-CXX := @CXX@ -Wall
+CC  := @CC@
+CXX := @CXX@
 SED := sed
 DEFS :=
 INCLUDES := -I.. -I@top_srcdir@
-CFLAGS := -O2 @MODULE_CFLAGS@ @INLINE_FLAGS@
-CPPFLAGS := -O2 @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CFLAGS += -Wall @MODULE_CFLAGS@ @INLINE_FLAGS@
+CXXFLAGS += -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@ -L.. -lyate
 MODFLAGS:= @MODULE_LDFLAGS@
 MODRELAX:= @MODULE_LDRELAX@
@@ -108,7 +108,7 @@
 LOCALFLAGS =
 LOCALLIBS =
 CCOMPILE = $(CC) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
-COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CPPFLAGS)
+COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CXX) $(LDFLAGS)
 MODLINK = $(CXX) $(MODFLAGS) $(MODSTRIP) $(LDFLAGS)
 MODCOMP = $(COMPILE) $(MODFLAGS) $(MODSTRIP) $(LDFLAGS)
--- libs/ysig/Makefile.in.old	2009-07-28 01:10:58.000000000 +0200
+++ libs/ysig/Makefile.in	2009-07-28 01:11:28.000000000 +0200
@@ -3,12 +3,12 @@
 
 DEBUG :=
 
-CXX  := @CXX@ -Wall
+CXX  := @CXX@
 AR  := ar
 DEFS :=
 LIBTHR := -lpthread
 INCLUDES := -I@top_srcdir@ -I../.. -I@srcdir@
-CFLAGS := -O2 @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CXXFLAGS += -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@ -L../.. -lyate
 INCFILES := @top_srcdir@/yateclass.h @srcdir@/yatesig.h
 
@@ -26,7 +26,7 @@
 
 LOCALFLAGS =
 LOCALLIBS =
-COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
+COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CC) $(LDFLAGS)
 
 prefix = @prefix@
--- libs/ymgcp/Makefile.in.old	2009-07-28 01:13:18.000000000 +0200
+++ libs/ymgcp/Makefile.in	2009-07-28 01:13:41.000000000 +0200
@@ -3,11 +3,11 @@
 
 DEBUG :=
 
-CXX  := @CXX@ -Wall
+CXX  := @CXX@
 AR  := ar
 DEFS :=
 INCLUDES := -I@top_srcdir@ -I../.. -I@srcdir@
-CFLAGS := -O2 @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CXXFLAGS += -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@ -L../.. -lyate
 INCFILES := @top_srcdir@/yateclass.h @srcdir@/yatemgcp.h
 
@@ -17,7 +17,7 @@
 
 LOCALFLAGS =
 LOCALLIBS =
-COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
+COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CC) $(LDFLAGS)
 
 prefix = @prefix@
--- libs/yrtp/Makefile.in.old	2009-07-28 01:15:02.000000000 +0200
+++ libs/yrtp/Makefile.in	2009-07-28 01:15:31.000000000 +0200
@@ -3,11 +3,11 @@
 
 DEBUG :=
 
-CXX  := @CXX@ -Wall
+CXX  := @CXX@
 AR  := ar
 DEFS :=
 INCLUDES := -I@top_srcdir@ -I../.. -I@srcdir@
-CFLAGS := -O2 @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CXXFLAGS += -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@ -L../.. -lyate
 INCFILES := @top_srcdir@/yateclass.h @srcdir@/yatertp.h
 
@@ -17,7 +17,7 @@
 
 LOCALFLAGS =
 LOCALLIBS =
-COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
+COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CC) $(LDFLAGS)
 
 prefix = @prefix@
--- libs/ilbc/Makefile.in.old	2009-07-28 01:15:47.000000000 +0200
+++ libs/ilbc/Makefile.in	2009-07-28 01:16:26.000000000 +0200
@@ -1,12 +1,12 @@
 # Makefile
 # This file holds the make rules for the libilbc 
 
-CC  := @CC@ -Wall
+CC  := @CC@
 AR  := ar
 SED := sed
 DEFS :=
 INCLUDES := -I@top_srcdir@
-CFLAGS := -O3 -funroll-loops -fomit-frame-pointer -fsigned-char @MODULE_CFLAGS@
+CFLAGS += -Wall  @MODULE_CFLAGS@
 LDFLAGS:= @LDFLAGS@
 
 PROGS=
--- libs/yjingle/Makefile.in.old	2009-07-28 01:16:54.000000000 +0200
+++ libs/yjingle/Makefile.in	2009-07-28 01:17:23.000000000 +0200
@@ -3,11 +3,11 @@
 
 DEBUG :=
 
-CXX  := @CXX@ -Wall
+CXX  := @CXX@
 AR  := ar
 DEFS :=
 INCLUDES := -I@top_srcdir@ -I../.. -I@srcdir@/../yxml -I@srcdir@
-CFLAGS := -O2 @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CXXFLAGS += -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@ -L../.. -lyate
 INCFILES := @top_srcdir@/yateclass.h @srcdir@/../yxml/tinystr.h @srcdir@/../yxml/tinyxml.h @srcdir@/xmlparser.h @srcdir@/xmpputils.h @srcdir@/yatejabber.h @srcdir@/yatejingle.h
 
@@ -17,7 +17,7 @@
 
 LOCALFLAGS =
 LOCALLIBS =
-COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
+COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CC) $(LDFLAGS)
 
 prefix = @prefix@
--- libs/ysip/Makefile.in.old	2009-07-28 01:17:40.000000000 +0200
+++ libs/ysip/Makefile.in	2009-07-28 01:17:59.000000000 +0200
@@ -3,11 +3,11 @@
 
 DEBUG :=
 
-CXX  := @CXX@ -Wall
+CXX  := @CXX@
 AR  := ar
 DEFS :=
 INCLUDES := -I@top_srcdir@ -I../.. -I@srcdir@
-CFLAGS := -O2 @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CXXFLAGS += -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@ -L../.. -lyate
 INCFILES := @top_srcdir@/yateclass.h @srcdir@/yatesip.h @srcdir@/util.h
 
@@ -17,7 +17,7 @@
 
 LOCALFLAGS =
 LOCALLIBS =
-COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
+COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CC) $(LDFLAGS)
 
 prefix = @prefix@
--- libs/ypbx/Makefile.in.old	2009-07-28 01:18:16.000000000 +0200
+++ libs/ypbx/Makefile.in	2009-07-28 01:18:39.000000000 +0200
@@ -3,11 +3,11 @@
 
 DEBUG :=
 
-CXX  := @CXX@ -Wall
+CXX  := @CXX@
 AR  := ar
 DEFS :=
 INCLUDES := -I@top_srcdir@ -I../.. -I@srcdir@
-CFLAGS := -O2 @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CXXFLAGS += -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@ -L../.. -lyate
 INCFILES := @top_srcdir@/yateclass.h @srcdir@/yatepbx.h
 
@@ -17,7 +17,7 @@
 
 LOCALFLAGS =
 LOCALLIBS =
-COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
+COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CC) $(LDFLAGS)
 
 prefix = @prefix@
--- libs/yxml/Makefile.in.old	2009-07-28 01:19:23.000000000 +0200
+++ libs/yxml/Makefile.in	2009-07-28 01:19:43.000000000 +0200
@@ -3,11 +3,11 @@
 
 DEBUG :=
 
-CXX  := @CXX@ -Wall
+CXX  := @CXX@
 AR  := ar
 DEFS :=
 INCLUDES := -I@top_srcdir@ -I../.. -I@srcdir@
-CFLAGS := -O2 @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CXXFLAGS += -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@ -L../.. -lyate
 INCFILES := @top_srcdir@/yateclass.h @srcdir@/tinystr.h @srcdir@/tinyxml.h
 
@@ -17,7 +17,7 @@
 
 LOCALFLAGS =
 LOCALLIBS =
-COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
+COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CC) $(LDFLAGS)
 
 prefix = @prefix@
--- libs/ymodem/Makefile.in.old	2009-07-28 01:20:00.000000000 +0200
+++ libs/ymodem/Makefile.in	2009-07-28 01:20:16.000000000 +0200
@@ -3,11 +3,11 @@
 
 DEBUG :=
 
-CXX  := @CXX@ -Wall
+CXX  := @CXX@
 AR  := ar
 DEFS :=
 INCLUDES := -I@top_srcdir@ -I../.. -I@srcdir@
-CFLAGS := -O2 @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CXXFLAGS += -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@ -L../.. -lyate
 INCFILES := @top_srcdir@/yateclass.h @srcdir@/yatemodem.h
 
@@ -17,7 +17,7 @@
 
 LOCALFLAGS =
 LOCALLIBS =
-COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
+COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CC) $(LDFLAGS)
 
 prefix = @prefix@
--- libs/yiax/Makefile.in.old	2009-07-28 01:20:44.000000000 +0200
+++ libs/yiax/Makefile.in	2009-07-28 01:21:01.000000000 +0200
@@ -3,11 +3,11 @@
 
 DEBUG :=
 
-CXX  := @CXX@ -Wall
+CXX  := @CXX@
 AR  := ar
 DEFS :=
 INCLUDES := -I@top_srcdir@ -I../.. -I@srcdir@
-CFLAGS := -O2 @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CXXFLAGS += -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@ -L../.. -lyate
 INCFILES := @top_srcdir@/yateclass.h @srcdir@/yateiax.h
 
@@ -17,7 +17,7 @@
 
 LOCALFLAGS =
 LOCALLIBS =
-COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
+COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CC) $(LDFLAGS)
 
 prefix = @prefix@
