--- configure.in
+++ configure.in
@@ -31,7 +31,6 @@
 dnl Don't assume we want high level optimisations, it is known to
 dnl produce broken code on certain architectures using certain compilers
 dnl --------------------------------------------------------------------
-CFLAGS="-O"
 AC_SUBST(CFLAGS)
 
 dnl --------------------------------------------------------------------
@@ -39,6 +38,7 @@
 dnl --------------------------------------------------------------------
 AC_PROG_CC
 AC_PROG_INSTALL
+AM_PROG_AS
 
 dnl --------------------------------------------------------------------
 dnl Check whether user wants simple warnings or advanced warnings
@@ -46,12 +46,10 @@
 AC_ARG_ENABLE(warnings, [  --enable-warnings       enable more compilation warning checks [default=no]], enable_warnings="yes", enable_warnings="no")
 
 if test "x$enable_warnings" = "xyes"; then
-  CFLAGS="$CFLAGS -pipe -W -Wall -ansi -pedantic -Wbad-function-cast \
+  CFLAGS="$CFLAGS -W -Wall -ansi -pedantic -Wbad-function-cast \
   -Wcast-align -Wcast-qual -Wchar-subscripts -Wconversion -Winline \
   -Wmissing-prototypes -Wnested-externs -Wpointer-arith \
   -Wredundant-decls -Wshadow -Wstrict-prototypes -Wwrite-strings"
-else
-  CFLAGS="$CFLAGS -pipe -Wall"
 fi
 AC_SUBST(CFLAGS)
 
@@ -144,12 +142,9 @@
 [  --without-ggi           disable check for GGI (General Graphics Interface)])
 
 if test "x$with_ggi" != "xno"; then
-  OCPPFLAGS="$CPPFLAGS"
-  CPPFLAGS="-I/usr/local/include"
-  AC_CHECK_HEADERS(ggi/gii.h ggi/ggi.h, CFLAGS="$CFLAGS -I/usr/local/include")
+  AC_CHECK_HEADERS(ggi/gii.h ggi/ggi.h)
   AC_CHECK_LIB(ggi, ggiInit)
   AC_CHECK_LIB(gii, giiInit)
-  CPPFLAGS="$OCPPFLAGS"
 fi
 
 dnl --------------------------------------------------------------------
