--- comptbl.c
+++ comptbl.c
@@ -28,6 +28,8 @@
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
+#include <stdlib.h>
+#include <string.h>
 
 #define ALLOC_SIZE 0x48d801     /* approximately 4.69MB */
 #define TBL_BASE ((unsigned char *)0x10000000)
--- gamegenie.c
+++ gamegenie.c
@@ -12,6 +12,7 @@
 
 #include <ctype.h>
 #include <stdio.h>
+#include <string.h>
 
 #include "gamegenie.h"
 
--- io.c
+++ io.c
@@ -21,6 +21,7 @@
 #endif /* HAVE_FEATURES_H */
 
 #include <stdio.h>
+#include <string.h>
 
 #include "consts.h"
 #include "globals.h"
--- mapper.c
+++ mapper.c
@@ -13,6 +13,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 
 #include "consts.h"
 #include "globals.h"
--- sound.c
+++ sound.c
@@ -116,6 +116,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
+#include <string.h>
 
 #ifdef HAVE_CONFIG_H
 #include "config.h"
