From: hasufell <julian.ospald@googlemail.com>
Date: Thu Apr 19 19:26:44 UTC 2012

fixed hardcoded DOCDIR

--- CMakeLists.txt
+++ CMakeLists.txt
@@ -332,8 +332,8 @@
     endif ()
     if (BUILD_DOCUMENTATION AND NOT NO_DOCS_PACKAGE)
         install(
-            DIRECTORY ${CMAKE_BINARY_DIR}/doc/GG
-            DESTINATION doc
+            DIRECTORY ${CMAKE_BINARY_DIR}/doc/GG/
+	    DESTINATION ${DOCDIR}
         )
     endif ()
 elseif (BUILD_DOCUMENTATION)
