--- orig/mpi.net-1.0.0/configure.ac	2008-10-06 15:52:38.000000000 +0200
+++ work/mpi.net-1.0.0/configure.ac	2010-03-01 00:02:49.000000000 +0100
@@ -164,19 +164,6 @@
   dnl We do not need the C bridge at all; everything can be done from C#
   need_cbridge="no"
 
-  real_CC="$CC"
-  CC="$MPICC"
-  AC_CHECK_LIB(pmpich,MPI_Init,[HAVE_LIBPMPICH=yes],[HAVE_LIBPMPICH=no],)
-  AC_CHECK_LIB(mpich,PMPI_Init,[HAVE_LIBMPICH=yes],[HAVE_LIBMPICH=no],)
-
-  if test "$HAVE_LIBPMPICH" = "no" ; then 
-    if test "$HAVE_LIBMPICH" = "yes" ; then 
-      AC_MSG_ERROR([ MPICH library is built statically. 
-      Please install MPICH built as a shared library by configuring MPICH with 
-      the --enable-sharedlibs=??? option)])
-    fi
-  fi
-  CC="real_CC"
 elif test "$MPI_KIND" = "openmpi" ; then
   dnl With Open MPI, we need to explicitly determine which libraries
   dnl the MPI.NET C bridge will need to link against, because libtool
@@ -202,9 +189,16 @@
 AM_CONDITIONAL(MPINET_CBRIDGE, test "$need_cbridge" = "yes")
 
 dnl Variable substitutions
-AC_SUBST(MCS ILASM MONO MPIEXEC PERL
-	 CSHARP_FLAGS CSHARP_DEFINES MPI_HEADER 
-	 MPI_SHARED_LIB_NAME MPI_FUNC_PREFIX)
+AC_SUBST(MCS)
+AC_SUBST(ILASM)
+AC_SUBST(MONO)
+AC_SUBST(MPIEXEC)
+AC_SUBST(PERL)
+AC_SUBST(CSHARP_FLAGS) 
+AC_SUBST(CSHARP_DEFINES)
+AC_SUBST(MPI_HEADER)
+AC_SUBST(MPI_SHARED_LIB_NAME)
+AC_SUBST(MPI_FUNC_PREFIX)
 
 AC_CONFIG_FILES([MPI/MPI.dll.config Tests/runtest.sh],[],[])
 
