In >=glib-2.32, only <glib.h> can be included directly.

--- a/deskbar/handlers/evolution/evolution.c
+++ b/deskbar/handlers/evolution/evolution.c
@@ -1,5 +1,4 @@
 #include <glib.h>
-#include <glib/gstring.h>
 #include <libebook/e-book.h>
 #include <pango/pango.h>
 #include <string.h>
--- a/deskbar/handlers/evolution/evolution.h
+++ b/deskbar/handlers/evolution/evolution.h
@@ -2,7 +2,7 @@
 #ifndef __DESKBAR_EVOLUTION_H__
 #define __DESKBAR_EVOLUTION_H__
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <gdk-pixbuf/gdk-pixbuf.h>
 
 G_BEGIN_DECLS
