# bug 410431

--- a/src/kadmin/ktutil/ktutil_funcs.c
+++ b/src/kadmin/ktutil/ktutil_funcs.c
@@ -64,7 +64,7 @@
     krb5_kt_list *list;
     int idx;
 {
-    krb5_kt_list lp, prev;
+    krb5_kt_list lp, prev = NULL;
     int i;
 
     for (lp = *list, i = 1; lp; prev = lp, lp = lp->next, i++) {
--- a/src/lib/kadm5/alt_prof.c
+++ b/src/lib/kadm5/alt_prof.c
@@ -164,7 +164,7 @@
     char **values;
     char *valp;
     int idx;
-    krb5_boolean val;
+    krb5_boolean val = 0;
 
     kret = krb5_aprof_getvals (acontext, hierarchy, &values);
     if (kret)
--- a/src/lib/krb5/unicode/ucstr.c
+++ b/src/lib/krb5/unicode/ucstr.c
@@ -109,7 +109,7 @@
             krb5_data ** newdataptr,
             unsigned flags)
 {
-    int i, j, len, clen, outpos, ucsoutlen, outsize;
+    int i, j, len, clen, outpos = 0, ucsoutlen, outsize;
     char *out = NULL, *outtmp, *s;
     krb5_ucs4 *ucs = NULL, *p, *ucsout = NULL;
     krb5_data *newdata;
diff --git a/src/util/profile/prof_init.c b/src/util/profile/prof_init.c
index 7dc5b47..cd90db8 100644
--- a/src/util/profile/prof_init.c
+++ b/src/util/profile/prof_init.c
@@ -255,7 +255,7 @@ copy_vtable_profile(profile_t profile, profile_t *ret_new_profile)
 {
     errcode_t err;
     void *cbdata;
-    profile_t new_profile;
+    profile_t new_profile = NULL;
 
     *ret_new_profile = NULL;
 
