Bug #454980
--- a/src/kadmin/dbutil/dump.c	2013-02-10 00:00:59.336080898 +0200
+++ b/src/kadmin/dbutil/dump.c	2013-02-10 00:02:00.559494398 +0200
@@ -1708,7 +1708,7 @@
     int                 alt_key_len, salt_len, alt_salt_len;
     char                *name;
     char                *mod_name;
-    int                 tmpint1, tmpint2, tmpint3;
+    int                 tmpint1 = 0, tmpint2 = 0, tmpint3 = 0;
     int                 error;
     const char          *try2read;
     int                 i;
