diff -Naur ParaView3/Applications/Client/CMakeLists.txt ParaView3.new/Applications/Client/CMakeLists.txt
--- ParaView3/Applications/Client/CMakeLists.txt	2010-01-04 10:24:37.000000000 -0500
+++ ParaView3.new/Applications/Client/CMakeLists.txt	2010-01-14 21:03:25.000000000 -0500
@@ -128,10 +128,10 @@
       "${CMAKE_CURRENT_BINARY_DIR}/CMake/tmp/pqClientDocFinder.txt"
       @ONLY 
       IMMEDIATE)
-  INSTALL(
-      FILES "${CMAKE_CURRENT_BINARY_DIR}/CMake/tmp/pqClientDocFinder.txt"
-      DESTINATION ${PV_INSTALL_BIN_DIR}
-      COMPONENT Runtime)
+    #  INSTALL(
+    #  FILES "${CMAKE_CURRENT_BINARY_DIR}/CMake/tmp/pqClientDocFinder.txt"
+    #  DESTINATION ${PV_INSTALL_BIN_DIR}
+    #  COMPONENT Runtime)
 ENDIF(NOT PV_INSTALL_NO_RUNTIME)
 
 IF(BUILD_TESTING)
