diff -Naur ParaView3/VTK/Infovis/Testing/Cxx/TestBoostAdapter.cxx ParaView3.new/VTK/Infovis/Testing/Cxx/TestBoostAdapter.cxx
--- ParaView3/VTK/Infovis/Testing/Cxx/TestBoostAdapter.cxx	2008-11-01 15:22:38.000000000 -0400
+++ ParaView3.new/VTK/Infovis/Testing/Cxx/TestBoostAdapter.cxx	2010-01-21 18:43:37.000000000 -0500
@@ -36,7 +36,7 @@
 #include <boost/graph/strong_components.hpp>
 #include <boost/graph/dijkstra_shortest_paths.hpp>
 #include <boost/graph/transitive_closure.hpp>
-#include <boost/property_map.hpp>
+#include <boost/property_map/property_map.hpp>
 #include <boost/vector_property_map.hpp>
 
 #include "vtkGraph.h"
diff -Naur ParaView3/VTK/Infovis/vtkBoostBreadthFirstSearch.cxx ParaView3.new/VTK/Infovis/vtkBoostBreadthFirstSearch.cxx
--- ParaView3/VTK/Infovis/vtkBoostBreadthFirstSearch.cxx	2009-03-13 12:11:50.000000000 -0400
+++ ParaView3.new/VTK/Infovis/vtkBoostBreadthFirstSearch.cxx	2010-01-21 18:43:16.000000000 -0500
@@ -46,7 +46,7 @@
 
 #include <boost/graph/visitors.hpp>
 #include <boost/graph/breadth_first_search.hpp>
-#include <boost/property_map.hpp>
+#include <boost/property_map/property_map.hpp>
 #include <boost/vector_property_map.hpp>
 #include <boost/pending/queue.hpp>
 
