diff -Naur ParaView3/Documentation/CMakeLists.txt ParaView3.new/Documentation/CMakeLists.txt
--- ParaView3/Documentation/CMakeLists.txt	2008-12-22 16:06:00.000000000 -0500
+++ ParaView3.new/Documentation/CMakeLists.txt	2010-01-16 11:02:15.000000000 -0500
@@ -155,7 +155,7 @@
 
 install(
     FILES "${ParaView_BINARY_DIR}/about.html"
-    DESTINATION ${PV_INSTALL_BIN_DIR}
+    DESTINATION ${PV_EXE_INSTALL}/Documentation
     COMPONENT Runtime)
 
 configure_file(
