diff -Naur ParaView3/Applications/OverView/Core/MainWindow.cxx ParaView3.new/Applications/OverView/Core/MainWindow.cxx
--- ParaView3/Applications/OverView/Core/MainWindow.cxx	2009-05-06 16:01:51.000000000 -0400
+++ ParaView3.new/Applications/OverView/Core/MainWindow.cxx	2010-01-14 18:52:54.000000000 -0500
@@ -933,7 +933,7 @@
   QString assistantExe;
   QString profileFile;
 
-  const char* assistantName = "assistant";
+  const char* assistantName = "paraview-assistant";
 #if defined(Q_WS_WIN)
   const char* binDir = "\\";
   const char* binDir1 = "\\..\\";
diff -Naur ParaView3/Qt/Components/pqClientMainWindow.cxx ParaView3.new/Qt/Components/pqClientMainWindow.cxx
--- ParaView3/Qt/Components/pqClientMainWindow.cxx	2009-09-22 13:37:21.000000000 -0400
+++ ParaView3.new/Qt/Components/pqClientMainWindow.cxx	2010-01-14 18:53:18.000000000 -0500
@@ -1041,7 +1041,7 @@
   QString assistantExe;
   QString profileFile;
 
-  const char* assistantName = "assistant";
+  const char* assistantName = "paraview-assistant";
 #if defined(Q_WS_WIN)
   const char* extString = ".exe";
   const char* binDir = "\\";
