--- apvlv-0.1.0-Source/src/ApvlvFile.cpp.orig
+++ apvlv-0.1.0-Source/src/ApvlvFile.cpp
@@ -285,7 +285,7 @@
   {
     PopplerRectangle rect = { x1, y1, x2, y2 };
     PopplerPage *page = poppler_document_get_page (mDoc, pn);
-    *out = poppler_page_get_text (page, POPPLER_SELECTION_WORD, &rect);
+    *out = poppler_page_get_selected_text (page, POPPLER_SELECTION_WORD, &rect);
     if (*out != NULL)
       {
         return true;
