Index: libgnome-keyring-2.32.0/egg/tests/test-dh.c
===================================================================
--- libgnome-keyring-2.32.0.orig/egg/tests/test-dh.c
+++ libgnome-keyring-2.32.0/egg/tests/test-dh.c
@@ -27,6 +27,7 @@
 
 #include "run-auto-test.h"
 
+#include "egg-libgcrypt.h"
 #include "egg-dh.h"
 #include "egg-secure-memory.h"
 
@@ -39,7 +40,7 @@ DEFINE_TEST(dh_perform)
 	gcry_mpi_t x2, X2;
 	gpointer k1, k2;
 	gboolean ret;
-
+	egg_libgcrypt_initialize ();
 	/* Load up the parameters */
 	if (!egg_dh_default_params ("ietf-ike-grp-modp-768", &p, &g))
 		g_assert_not_reached ();
