--- src/Capic/Gui/capiclientplayer.cpp.old	2010-12-03 12:13:25.000000000 +0100
+++ src/Capic/Gui/capiclientplayer.cpp	2010-12-03 12:14:35.000000000 +0100
@@ -20,12 +20,12 @@
 CapiClientPlayer::CapiClientPlayer(int id) : QObject(0), CapiPlayer(id) {
 	playerWidget = new PlayerInfo(0, 32);
 	playerWidget->setName(getName());
-	playerWidget->setMoney(QString(0));
+	playerWidget->setMoney(QString());
 	playerWidget->setAvatar(getAvatar());
 
 	auction = new PlayerAuction();
 	auction->setName(getName());
-	auction->setMoney(QString(0));
+	auction->setMoney(QString());
 	auction->setAvatar(getAvatar());
 
 	playerWidget->playerId = getId();
