--- yajl-2.0.4/src/CMakeLists.txt
+++ yajl-2.0.4/src/CMakeLists.txt
@@ -35,8 +35,6 @@
 # set the output path for libraries
 SET(LIBRARY_OUTPUT_PATH ${libDir})
 
-ADD_LIBRARY(yajl_s STATIC ${SRCS} ${HDRS} ${PUB_HDRS})
-
 ADD_LIBRARY(yajl SHARED ${SRCS} ${HDRS} ${PUB_HDRS})
 
 #### setup shared library version number
@@ -69,8 +67,6 @@
 
   EXEC_PROGRAM(${CMAKE_COMMAND} ARGS -E copy_if_different ${header} ${incDir})
 
-  ADD_CUSTOM_COMMAND(TARGET yajl_s POST_BUILD
-      COMMAND ${CMAKE_COMMAND} -E copy_if_different ${header} ${incDir})
 ENDFOREACH (header ${PUB_HDRS})
 
 INCLUDE_DIRECTORIES(${incDir}/..)
@@ -79,7 +75,6 @@
   # at build time you may specify the cmake variable LIB_SUFFIX to handle
   # 64-bit systems which use 'lib64'
   INSTALL(TARGETS yajl LIBRARY DESTINATION lib${LIB_SUFFIX})
-  INSTALL(TARGETS yajl_s ARCHIVE DESTINATION lib${LIB_SUFFIX})
   INSTALL(FILES ${PUB_HDRS} DESTINATION include/yajl)
   INSTALL(FILES ${incDir}/yajl_version.h DESTINATION include/yajl)
   INSTALL(FILES ${shareDir}/yajl.pc DESTINATION share/pkgconfig)
