diff --git a/wpa_supplicant/dbus/dbus_new_helpers.c b/wpa_supplicant/dbus/dbus_new_helpers.c
index 06749db..1cbdee9 100644
--- a/wpa_supplicant/dbus/dbus_new_helpers.c
+++ b/wpa_supplicant/dbus/dbus_new_helpers.c
@@ -810,7 +810,7 @@ void wpa_dbus_mark_property_changed(struct wpas_dbus_priv *iface,
 	const struct wpa_dbus_property_desc *dsc;
 	int i = 0;
 
-	if (iface == NULL)
+	if (iface == NULL || path == NULL)
 		return;
 
 	dbus_connection_get_object_path_data(iface->con, path,
diff --git a/wpa_supplicant/dbus/dbus_old.c b/wpa_supplicant/dbus/dbus_old.c
index 437a04d..9e6c762 100644
--- a/wpa_supplicant/dbus/dbus_old.c
+++ b/wpa_supplicant/dbus/dbus_old.c
@@ -377,7 +377,7 @@ void wpa_supplicant_dbus_notify_scan_results(struct wpa_supplicant *wpa_s)
 	DBusMessage *_signal;
 
 	/* Do nothing if the control interface is not turned on */
-	if (iface == NULL)
+	if (iface == NULL || wpa_s->dbus_path == NULL)
 		return;
 
 	_signal = dbus_message_new_signal(wpa_s->dbus_path,
@@ -414,7 +414,7 @@ void wpa_supplicant_dbus_notify_state_change(struct wpa_supplicant *wpa_s,
 	if (wpa_s->global == NULL)
 		return;
 	iface = wpa_s->global->dbus;
-	if (iface == NULL)
+	if (iface == NULL || wpa_s->dbus_path == NULL)
 		return;
 
 	/* Only send signal if state really changed */
@@ -473,7 +473,7 @@ void wpa_supplicant_dbus_notify_scanning(struct wpa_supplicant *wpa_s)
 	dbus_bool_t scanning = wpa_s->scanning ? TRUE : FALSE;
 
 	/* Do nothing if the control interface is not turned on */
-	if (iface == NULL)
+	if (iface == NULL || wpa_s->dbus_path == NULL)
 		return;
 
 	_signal = dbus_message_new_signal(wpa_s->dbus_path,
@@ -508,7 +508,7 @@ void wpa_supplicant_dbus_notify_wps_cred(struct wpa_supplicant *wpa_s,
 	if (wpa_s->global == NULL)
 		return;
 	iface = wpa_s->global->dbus;
-	if (iface == NULL)
+	if (iface == NULL || wpa_s->dbus_path == NULL)
 		return;
 
 	_signal = dbus_message_new_signal(wpa_s->dbus_path,
