diff -Naur a/src/eap_peer/Makefile b/src/eap_peer/Makefile
--- a/src/eap_peer/Makefile	2013-05-17 09:51:55.000000000 +0000
+++ b/src/eap_peer/Makefile	2013-05-17 09:54:03.010830904 +0000
@@ -36,6 +36,7 @@
 OBJS_both += ../utils/wpa_debug.o
 OBJS_both += ../utils/base64.o
 OBJS_both += ../utils/wpabuf.o
+OBJS_both += ../utils/eloop.o
 OBJS_both += ../crypto/md5.o
 ifneq ($(CONFIG_TLS), openssl)
 OBJS_both += ../crypto/sha1.o
@@ -48,6 +49,7 @@
 OBJS_both += ../crypto/aes-omac1.o
 OBJS_both += ../crypto/ms_funcs.o
 OBJS_both += ../crypto/sha256.o
+OBJS_both += ../crypto/random.o
 
 
 OBJS_both += ../eap_common/eap_peap_common.o
