--- dbus/fi.epitest.hostap.WPASupplicant.service        2010-09-07 10:43:39.000000000 -0500
+++ dbus/fi.epitest.hostap.WPASupplicant.service   2011-01-08 19:41:32.969507889 -0500
@@ -1,4 +1,4 @@
 [D-BUS Service]
 Name=fi.epitest.hostap.WPASupplicant
-Exec=/sbin/wpa_supplicant -u
+Exec=/usr/sbin/wpa_supplicant -u
 User=root
--- dbus/fi.w1.wpa_supplicant1.service  2010-09-07 10:43:39.000000000 -0500
+++ dbus/fi.w1.wpa_supplicant1.service     2011-01-08 19:42:12.420246349 -0500
@@ -1,4 +1,4 @@
 [D-BUS Service]
 Name=fi.w1.wpa_supplicant1
-Exec=/sbin/wpa_supplicant -u
+Exec=/usr/sbin/wpa_supplicant -u
 User=root
