Fixes issues with gdb-7.5 and/or split debug files.
From upstream mailing list.

--- a/kdbg/gdbdriver.cpp
+++ b/kdbg/gdbdriver.cpp
@@ -2191,6 +2191,9 @@
      *     Reading symbols from
      */
     while (strncmp(output, "Reading symbols from", 20) == 0 ||
+	   strncmp(output, "done.", 5) == 0 ||
+	   strncmp(output, "Missing separate debuginfo", 26) == 0 ||
+	   strncmp(output, "Try: ", 5) == 0 ||
 	   strncmp(output, "Using host libthread_db", 23) == 0 ||
 	   strncmp(output, "(no debugging symbols found)", 28) == 0)
     {
