From 83c3ce9b01311dd18da0eff717d46734d5e96b83 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Diego=20Elio=20'Flameeyes'=20Petten=C3=B2?= <flameeyes@gmail.com>
Date: Sat, 5 Sep 2009 21:54:24 +0200
Subject: [PATCH 2/2] Remove a call to set label_max_decimals on Gruff::Line that doesn't exist.

---
 test/test_line.rb |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/test/test_line.rb b/test/test_line.rb
index 4135b1e..1531931 100644
--- a/test/test_line.rb
+++ b/test/test_line.rb
@@ -462,7 +462,6 @@ class TestGruffLine < GruffTestCase
     g.legend_font_size = 12
     g.marker_font_size = 16
     g.hide_dots = false
-    g.label_max_decimals = 1
 
     g.write('test/output/line_jruby_error.png')
   end
-- 
1.6.4.2

