commit f5098443aba86fc59b02eace6ecad7530d72694b
Author: Serge Prikha <prikha@gmail.com>
Date:   Wed Jun 6 16:22:41 2012 +0400

    small fix on zero division

diff --git a/lib/gruff/base.rb b/lib/gruff/base.rb
index 590ef99..898afe0 100644
--- a/lib/gruff/base.rb
+++ b/lib/gruff/base.rb
@@ -1066,7 +1066,7 @@ module Gruff
     # Return a formatted string representing a number value that should be
     # printed as a label.
     def label(value)
-      label = if (@spread.to_f % @marker_count.to_f == 0) || !@y_axis_increment.nil?
+      label = if (@spread.to_f % (@marker_count.to_f==0 ? 1 : @marker_count.to_f) == 0) || !@y_axis_increment.nil?
         value.to_i.to_s
       elsif @spread > 10.0
         sprintf("%0i", value)
