--- clanbomber/ClientSetup.cpp.old	2009-01-22 10:16:04.000000000 +0100
+++ clanbomber/ClientSetup.cpp	2009-01-22 10:18:23.000000000 +0100
@@ -14,6 +14,7 @@
  *                                                                         *
  ***************************************************************************/
 
+#define DIRECTFB_ENABLE_DEPRECATED
 #include "ClientSetup.h"
 
 #include "ClanBomber.h"
--- clanbomber/ClanBomber.cpp.old	2009-01-22 10:26:30.000000000 +0100
+++ clanbomber/ClanBomber.cpp	2009-01-22 10:24:57.000000000 +0100
@@ -20,6 +20,7 @@
 #include <unistd.h>
 #include <math.h>
 
+#define DIRECTFB_ENABLE_DEPRECATED
 #include <directfb.h>
 
 #include "ClanBomber.h"
--- clanbomber/Client.cpp.old	2009-01-22 10:28:56.000000000 +0100
+++ clanbomber/Client.cpp	2009-01-22 10:30:36.000000000 +0100
@@ -204,7 +204,7 @@
     int bytes_received=-1;
     int buf[4096];
     int bufsize=sizeof(buf);
-    size_t addr_len=sizeof(struct sockaddr);
+    socklen_t addr_len=sizeof(struct sockaddr);
     memset((char*)&server_addr, 0, addr_len);
     int brecv=recvfrom(my_udp_socket, buf, bufsize, 0, (sockaddr*)&server_addr, &addr_len);
     bytes_received=brecv;
--- clanbomber/GameStatus.cpp.old	2009-01-22 10:32:43.000000000 +0100
+++ clanbomber/GameStatus.cpp	2009-01-22 10:33:13.000000000 +0100
@@ -19,6 +19,8 @@
 #include <stdio.h>
 #include <math.h>
 
+#define DIRECTFB_ENABLE_DEPRECATED
+
 #include <directfb.h>
 
 #include "ClanBomber.h"
--- clanbomber/Resources.h.old	2009-01-22 10:37:27.000000000 +0100
+++ clanbomber/Resources.h	2009-01-22 10:35:02.000000000 +0100
@@ -19,6 +19,7 @@
 #ifndef RESOURCES_H
 #define RESOURCES_H
 
+#define DIRECTFB_ENABLE_DEPRECATED
 #include <directfb.h>
 #include <fusionsound.h>
 
--- clanbomber/Server.cpp.old	2009-01-22 10:41:10.000000000 +0100
+++ clanbomber/Server.cpp	2009-01-22 10:41:34.000000000 +0100
@@ -295,7 +295,7 @@
     int bytes_received=-1;
     int buf[4096];
     int bufsize=sizeof(buf);
-    size_t addr_len=sizeof(struct sockaddr);
+    socklen_t addr_len=sizeof(struct sockaddr);
     memset((char*)&client_addr, 0, addr_len);
     int brecv=recvfrom(my_udp_socket, buf, bufsize, 0, (sockaddr*)&client_addr, &addr_len);
     bytes_received=brecv;
