--- src/Makefile.am.old	2011-06-20 09:20:15.000000000 +0200
+++ src/Makefile.am	2011-06-20 09:20:25.000000000 +0200
@@ -66,7 +66,7 @@
 manualdir = @htmldir@
 
 
-kildclient_LDADD = $(KILDCLIENT_LIBS) $(KILDCLIENT_PERL_LIBS) $(GTKSPELL_LIBS) $(LIBGNUTLS_LIBS) $(INTLLIBS)
+kildclient_LDADD = $(KILDCLIENT_LIBS) $(KILDCLIENT_PERL_LIBS) $(GTKSPELL_LIBS) $(LIBGNUTLS_LIBS) $(INTLLIBS) -lz
 
 localedir = @localedir@
 
