diff -ru kildclient-2.9.0.orig/Makefile.am kildclient-2.9.0/Makefile.am
--- kildclient-2.9.0.orig/Makefile.am	2009-02-01 07:46:53.000000000 -0500
+++ kildclient-2.9.0/Makefile.am	2009-11-02 17:35:21.625377149 -0500
@@ -11,11 +11,7 @@
           share
 endif
 
-pkgdocdir = $(datadir)/doc/kildclient
-
-pkgdoc_DATA = README AUTHORS NEWS COPYING
-
-appsdir = $(datadir)/applications
+appsdir = /usr/share/applications
 apps_DATA = kildclient.desktop
 
 EXTRA_DIST = getperlflags.pl \
diff -ru kildclient-2.9.0.orig/doc/C/Makefile.am kildclient-2.9.0/doc/C/Makefile.am
--- kildclient-2.9.0.orig/doc/C/Makefile.am	2009-02-01 07:46:53.000000000 -0500
+++ kildclient-2.9.0/doc/C/Makefile.am	2009-11-02 17:35:21.625377149 -0500
@@ -2,9 +2,9 @@
 
 SUBDIRS = images
 
-pkgdocdir = $(datadir)/doc/kildclient
+pkgdocdir = @docdir@
 
-manualdir = $(pkgdocdir)/html
+manualdir = @htmldir@
 
 #pkgdoc_DATA = kildclient.pdf
 
diff -ru kildclient-2.9.0.orig/kildclient.desktop kildclient-2.9.0/kildclient.desktop
--- kildclient-2.9.0.orig/kildclient.desktop	2009-01-11 14:45:06.000000000 -0500
+++ kildclient-2.9.0/kildclient.desktop	2009-11-02 17:35:21.625377149 -0500
@@ -4,7 +4,7 @@
 Exec=kildclient
 Icon=kildclient
 Terminal=false
-Categories=Network;RolePlaying;
+Categories=Game;Network;RolePlaying;
 
 Name=KildClient
 Name[eo]=KildClient
diff -ru kildclient-2.9.0.orig/po/Makefile.in.in kildclient-2.9.0/po/Makefile.in.in
--- kildclient-2.9.0.orig/po/Makefile.in.in	2009-05-23 17:35:41.000000000 -0400
+++ kildclient-2.9.0/po/Makefile.in.in	2009-11-02 17:35:21.625377149 -0500
@@ -28,9 +28,8 @@
 datarootdir = @datarootdir@
 datadir = @datadir@
 libdir = @libdir@
-localedir = $(libdir)/locale
-gnulocaledir = $(datadir)/locale
-gettextsrcdir = $(datadir)/glib-2.0/gettext/po
+localedir = @localedir@
+gnulocaledir = @localedir@
 subdir = po
 
 INSTALL = @INSTALL@
@@ -161,17 +160,6 @@
 	    fi; \
 	  fi; \
 	done
-	if test "$(PACKAGE)" = "glib"; then \
-	  if test -r "$(MKINSTALLDIRS)"; then \
-	    $(MKINSTALLDIRS) $(DESTDIR)$(gettextsrcdir); \
-	  else \
-	    $(SHELL) $(top_srcdir)/mkinstalldirs $(DESTDIR)$(gettextsrcdir); \
-	  fi; \
-	  $(INSTALL_DATA) $(srcdir)/Makefile.in.in \
-			  $(DESTDIR)$(gettextsrcdir)/Makefile.in.in; \
-	else \
-	  : ; \
-	fi
 
 # Define this as empty until I found a useful application.
 installcheck:
@@ -186,9 +174,6 @@
 	  rm -f $(DESTDIR)$(gnulocaledir)/$$lang/LC_MESSAGES/$(GETTEXT_PACKAGE)$(INSTOBJEXT); \
 	  rm -f $(DESTDIR)$(gnulocaledir)/$$lang/LC_MESSAGES/$(GETTEXT_PACKAGE)$(INSTOBJEXT).m; \
 	done
-	if test "$(PACKAGE)" = "glib"; then \
-	  rm -f $(DESTDIR)$(gettextsrcdir)/Makefile.in.in; \
-	fi
 
 check: all
 
diff -ru kildclient-2.9.0.orig/share/Makefile.am kildclient-2.9.0/share/Makefile.am
--- kildclient-2.9.0.orig/share/Makefile.am	2009-02-01 07:46:53.000000000 -0500
+++ kildclient-2.9.0/share/Makefile.am	2009-11-02 17:35:21.626376565 -0500
@@ -6,7 +6,7 @@
 		kildclient.hlp \
 		kcworld.dtd
 
-pixmapdir = $(datadir)/pixmaps
+pixmapdir = /usr/share/pixmaps
 pixmap_DATA = kildclient.png kildclient.xpm
 
 
diff -ru kildclient-2.9.0.orig/src/Makefile.am kildclient-2.9.0/src/Makefile.am
--- kildclient-2.9.0.orig/src/Makefile.am	2009-05-23 15:41:42.000000000 -0400
+++ kildclient-2.9.0/src/Makefile.am	2009-11-02 17:35:21.626376565 -0500
@@ -61,14 +61,14 @@
 
 EXTRA_DIST = $(pkgdata_DATA)
 
-pkgdocdir = $(datadir)/doc/kildclient
+pkgdocdir = @docdir@
 
-manualdir = $(pkgdocdir)/html
+manualdir = @htmldir@
 
 
 kildclient_LDADD = $(KILDCLIENT_LIBS) $(KILDCLIENT_PERL_LIBS) $(GTKSPELL_LIBS) $(LIBGNUTLS_LIBS) $(INTLLIBS)
 
-localedir = $(datadir)/locale
+localedir = @localedir@
 
 INCLUDES = -DLOCALEDIR=\"$(localedir)\" \
            -DPKGDATADIR=\"$(pkgdatadir)\" \
diff -ru kildclient-2.9.0.orig/src/mainwindow.c kildclient-2.9.0/src/mainwindow.c
--- kildclient-2.9.0.orig/src/mainwindow.c	2009-02-02 13:34:41.000000000 -0500
+++ kildclient-2.9.0/src/mainwindow.c	2009-11-02 17:36:29.851126172 -0500
@@ -169,9 +169,9 @@
   glade_xml_signal_autoconnect(gladexml);
 #ifndef __WIN32__
   gtk_window_set_icon_from_file(GTK_WINDOW(window),
-                                SYSDATADIR "/pixmaps/kildclient.png",
+                                "/usr/share/pixmaps/kildclient.png",
                                 NULL);
-  gtk_window_set_default_icon_from_file(SYSDATADIR "/pixmaps/kildclient.png",
+  gtk_window_set_default_icon_from_file("/usr/share/pixmaps/kildclient.png",
                                         NULL);
 #endif
 
