From 083ece20445bf6e1db26a74856185d0c9b1e8b5a Mon Sep 17 00:00:00 2001
From: Orion Poplawski <orion@cora.nwra.com>
Date: Tue, 4 Feb 2014 12:28:07 +0100
Subject: [PATCH] Fix password connection with libssh 0.6.0

---
 debian/changelog        |    3 ++-
 sshmasterconnection.cpp |    3 +++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/sshmasterconnection.cpp b/sshmasterconnection.cpp
index e0266db..827eefe 100755
--- a/sshmasterconnection.cpp
+++ b/sshmasterconnection.cpp
@@ -936,6 +936,9 @@ bool SshMasterConnection::userChallengeAuth()
 
 bool SshMasterConnection::userAuthWithPass()
 {
+    // Populate the userauth_list
+    ssh_userauth_none(my_ssh_session, NULL);
+
     int method = ssh_userauth_list(my_ssh_session, NULL);
 
     if (method& SSH_AUTH_METHOD_INTERACTIVE)
-- 
1.7.9.5

