diff -ur swish-e-2.4.7-orig/perl/Makefile.PL swish-e-2.4.7/perl/Makefile.PL
--- swish-e-2.4.7-orig/perl/Makefile.PL	2009-04-05 03:58:33.000000000 +0200
+++ swish-e-2.4.7/perl/Makefile.PL	2009-06-03 09:39:10.000000000 +0200
@@ -23,6 +23,7 @@
     NORECURS   => 1,              # keep it from recursing into subdirectories
     DIR        => [],
     XSPROTOARG => '-noprototypes',
+    CCFLAGS    => '-I../src',
     PREREQ_PM => { 'File::Spec' => '0.8', },
     test  => { TESTS => 't/*.t', },
     clean => {
