http://odamex.net/svn/revision.php?rev=3691

Index: ag-odalaunch/src/main.cpp
===================================================================
--- ag-odalaunch/src/main.cpp	(revision 3668)
+++ ag-odalaunch/src/main.cpp	(working copy)
@@ -52,7 +52,7 @@
 {
 	AG_Window *win;
 
-	if((win = AG_GuiDebugger(agWindowFocused)) != NULL)
+	if((win = (AG_Window*)AG_GuiDebugger(agWindowFocused)) != NULL)
 	{
 		AG_WindowShow(win);
 	}
