From 127c298ac43cbdf3b965002e7e9115571ca1d90d Mon Sep 17 00:00:00 2001
From: Julian Taylor <jtaylor.debian@googlemail.com>
Date: Tue, 8 May 2012 22:14:58 +0200
Subject: [PATCH] mask the result of g_io_channel_get_flags for the wanted
 flag

---
 src/tree.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/tree.c b/src/tree.c
index 721817e..b347b15 100644
--- a/src/tree.c
+++ b/src/tree.c
@@ -120,13 +120,13 @@ zenity_tree_handle_stdin (GIOChannel  *channel,
 
     string = g_string_new (NULL);
 
-    while (g_io_channel_get_flags(channel) != G_IO_FLAG_IS_READABLE)
+    while ((g_io_channel_get_flags(channel) & G_IO_FLAG_IS_READABLE) != G_IO_FLAG_IS_READABLE)
       ;
     do {
       gint status;
 
       do {
-        if (g_io_channel_get_flags(channel) == G_IO_FLAG_IS_READABLE)
+        if (g_io_channel_get_flags(channel) & G_IO_FLAG_IS_READABLE)
           status = g_io_channel_read_line_string (channel, string, NULL, &error);
         else
           return FALSE;
@@ -645,7 +645,7 @@ zenity_tree_dialog_response (GtkWidget *widget, int response, gpointer data)
       zen_data->exit_code = zenity_util_return_exit_code (ZENITY_ESC);
       break;
   }
-  if (channel != NULL && g_io_channel_get_flags (channel) == G_IO_FLAG_IS_READABLE)
+  if (channel != NULL && g_io_channel_get_flags (channel) & G_IO_FLAG_IS_READABLE)
     g_io_channel_shutdown (channel, TRUE, NULL);
 
   gtk_main_quit ();
-- 
1.7.8.6

