--- src/slicer/printlines.h.orig	2013-10-10 07:38:38.199031372 +0200
+++ src/slicer/printlines.h	2013-10-10 07:38:43.652364764 +0200
@@ -98,7 +98,7 @@
 
   int getCommands(Vector3d &lastpos, vector<Command> &commands,
 		  const double &minspeed, const double &movespeed,
-		  const double &minZspeed, const double &minZspeed,
+		  const double &minZspeed, const double &maxZspeed,
 		  const double &maxAOspeed, bool useTCommand) const;
 
   // // not used
