From 2a75e25ce2abbacd3c37f0a6ebd51e55d9f358d2 Mon Sep 17 00:00:00 2001
From: Alexandre Rostovtsev <tetromino@gentoo.org>
Date: Mon, 13 Aug 2012 01:37:42 -0400
Subject: [PATCH 1/2] Do not build 'Hello World' program

---
 src/Makefile.am |    4 ----
 1 files changed, 0 insertions(+), 4 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 49b56fe..8188a11 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -10,10 +10,6 @@ SUBDIRS =		\
 	eplugin		\
 	tests
 
-bin_PROGRAMS = evolution_kolab
-
-evolution_kolab_SOURCES = \
-	main.c
 
 AM_CPPFLAGS = 					\
 	-DPACKAGE_LOCALE_DIR=\"$(localedir)\"	\
-- 
1.7.8.6

