diff -ru analogtv-1.0.00-orig/Makefile analogtv-1.0.00/Makefile
--- analogtv-1.0.00-orig/Makefile	2006-07-24 12:07:50.916519443 +0200
+++ analogtv-1.0.00/Makefile	2006-07-24 12:18:40.059961942 +0200
@@ -93,8 +93,6 @@
 all: libvdr-$(PLUGIN).so
 .PHONY: all
 
-player-analogtv.o: player-analogtv.c
-	$(CC) $(CCFLAGS) -c $(DEFINES) $(INCLUDES) $(OPTFLAGS) player-analogtv.c
 
 libvdr-$(PLUGIN).so: $(OBJS) $(FFLIB)
 	$(CXX) $(CXXFLAGS) -shared $(OBJS) -ldvbmpegtools -o $@
diff -ru analogtv-1.0.00-orig/player-analogtv.c analogtv-1.0.00/player-analogtv.c
--- analogtv-1.0.00-orig/player-analogtv.c	2006-07-24 12:07:50.916519443 +0200
+++ analogtv-1.0.00/player-analogtv.c	2006-07-24 12:17:51.315176601 +0200
@@ -12,6 +12,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <stdarg.h>
 #include <string.h>
 #include <time.h>
@@ -26,6 +27,7 @@
 #include <sys/mman.h>
 #include <sys/sysinfo.h>
 #include <sys/soundcard.h>
+#include <sys/wait.h>
 
 #include <transform.h>
 #undef MIN
@@ -282,7 +284,7 @@
       }
 
       if (i < MAXSOURCES)
-        strcpy(sources[i], input.name);
+        strcpy(sources[i], (char*) input.name);
 
       d(1, "Detected Input#%d: %s", i, input.name);
     }
@@ -866,7 +868,7 @@
 {
   register int  f, r = 0, rv, rmax = 0, cc = 0, i;
   register int  dropped = 0, dropwarned = 1, skipped = 0;
-  auto     UC   buffer[BUFFSIZE], sx[128], cmd[256];
+  auto     char buffer[BUFFSIZE], sx[128], cmd[256];
   auto	   FILE *fi;
 
   for (i = 0; i < NUM_BUFS; i++)
@@ -936,7 +938,7 @@
       spes += r;
 
       if (r > 0)
-        kpes_to_ts(&p2t, buffer, r); // PES -> TS
+        kpes_to_ts(&p2t, (uint8_t*) buffer, r); // PES -> TS
     }
   }
 
