Index: cups/thread-private.h
===================================================================
--- cups/thread-private.h	(Revision 10432)
+++ cups/thread-private.h	(Revision 10433)
@@ -49,7 +49,7 @@
 typedef void *(__stdcall *_cups_thread_func_t)(void *arg);
 typedef struct _cups_mutex_s
 {
-  int			m_init;		/* Flag for on-demand initialization */		
+  int			m_init;		/* Flag for on-demand initialization */
   CRITICAL_SECTION	m_criticalSection;
 					/* Win32 Critical Section */
 } _cups_mutex_t;
@@ -62,6 +62,7 @@
 #    define _cupsThreadSetData(k,p) TlsSetValue(k,p)
 
 #  else
+typedef void	*(*_cups_thread_func_t)(void *arg);
 typedef char	_cups_mutex_t;
 typedef char	_cups_rwlock_t;
 typedef void	*_cups_threadkey_t;
