diff -r ef341d8811b2 src/configure.in
--- a/src/configure.in	Mon Apr 15 22:22:58 2013 +0200
+++ b/src/configure.in	Sun Apr 21 15:53:35 2013 -0700
@@ -1093,9 +1093,9 @@
     AC_CACHE_CHECK(Python's configuration directory,vi_cv_path_python3_conf,
     [
      vi_cv_path_python3_conf=
-     config_dir="config"
+     config_dir="config-${vi_cv_var_python3_version}"
      if test "${vi_cv_var_python3_abiflags}" != ""; then
-       config_dir="${config_dir}-${vi_cv_var_python3_version}${vi_cv_var_python3_abiflags}"
+       config_dir="${config_dir}${vi_cv_var_python3_abiflags}"
      fi
      for path in "${vi_cv_path_python3_pfx}" "${vi_cv_path_python3_epfx}"; do
        for subdir in lib64 lib share; do
