diff -ru vim71.orig/src/Makefile vim71/src/Makefile
--- vim71.orig/src/Makefile	2008-07-16 08:29:55 +0200
+++ vim71/src/Makefile	2008-07-16 08:53:44 +0200
@@ -1282,10 +1282,6 @@
 
 ALL_CFLAGS = $(PRE_DEFS) $(CFLAGS) $(PROFILE_CFLAGS) $(POST_DEFS)
 
-# Exclude $CFLAGS for osdef.sh, for Mac 10.4 some flags don't work together
-# with "-E".
-OSDEF_CFLAGS = $(PRE_DEFS) $(POST_DEFS)
-
 LINT_CFLAGS = -DLINT -I. $(PRE_DEFS) $(POST_DEFS) -Dinline= -D__extension__= -Dalloca=alloca
 
 LINT_EXTRA = -DUSE_SNIFF -DHANGUL_INPUT -D"__attribute__(x)="
@@ -2273,7 +2269,7 @@
 	    $(PERLLIB)/ExtUtils/typemap if_perl.xs >> $@
 
 auto/osdef.h: auto/config.h osdef.sh osdef1.h.in osdef2.h.in
-	CC="$(CC) $(OSDEF_CFLAGS)" srcdir=$(srcdir) sh $(srcdir)/osdef.sh
+	CC="$(CC) $(CFLAGS)" srcdir=$(srcdir) sh $(srcdir)/osdef.sh
 
 QUOTESED = sed -e 's/[\\"]/\\&/g' -e 's/\\"/"/' -e 's/\\";$$/";/'
 auto/pathdef.c: Makefile auto/config.mk
