diff -Naur parrot-5.5.0/config/auto/perldoc.pm parrot-5.5.0/config/auto/perldoc.pm
--- parrot-5.5.0/config/auto/perldoc.pm	2013-04-30 19:11:28.000000000 +0000
+++ parrot-5.5.0/config/auto/perldoc.pm	2013-06-26 15:44:31.031675583 +0000
@@ -38,7 +38,7 @@
 
     my $cmd = File::Spec->catfile($conf->data->get('scriptdirexp_provisional'), q{perldoc});
     my ( $fh, $filename ) = tempfile( UNLINK => 1 );
-    my $content = capture_output("$cmd -ud $filename perldoc") || undef;
+    my $content = capture_output("$cmd -u perldoc > $filename") || undef;
 
     return 1 unless defined( $self->_initial_content_check($conf, $content) );
 
@@ -67,7 +67,7 @@
         if ( $new_perldoc ) {
             $TEMP_pod_build .= <<"END";
 ops/$pod: ../src/ops/$ops
-\t\$(PERLDOC) -ud ops/$pod ../src/ops/$ops
+\t\$(PERLDOC) -u ../src/ops/$ops > ops/$pod
 \t\$(CHMOD) 0644 ops/$pod
 \t\$(ADDGENERATED) "docs/\$\@" "[doc]"
 
diff -Naur parrot-5.5.0/config/gen/makefiles/docs.in parrot-5.5.0/config/gen/makefiles/docs.in
--- parrot-5.5.0/config/gen/makefiles/docs.in	2013-06-15 17:36:41.000000000 +0000
+++ parrot-5.5.0/config/gen/makefiles/docs.in	2013-06-26 15:39:41.109881123 +0000
@@ -63,7 +63,7 @@
 	$(TOUCH) doc-prep
 
 packfile-c.pod: ../src/packfile/api.c
-#IF(new_perldoc):	$(PERLDOC) -ud packfile-c.pod ../src/packfile/api.c
+#IF(new_perldoc):	$(PERLDOC) -u ../src/packfile/api.c > packfile-c.pod
 #ELSE:	$(PERLDOC) -u ../src/packfile/api.c > packfile-c.pod
 	$(ADDGENERATED) "docs/$@" "[doc]"
 
