--- IMAPClient-0.11/imapclient/test/test_imapclient.py
+++ IMAPClient-0.11/imapclient/test/test_imapclient.py
@@ -10,7 +10,7 @@
 from datetime import datetime
 from mock import patch, sentinel, Mock
 
-from imapclient import six
+import six
 from imapclient.fixed_offset import FixedOffset
 from imapclient.imapclient import datetime_to_imap
 from imapclient.test.testable_imapclient import TestableIMAPClient as IMAPClient
--- IMAPClient-0.11/imapclient/test/test_imap_utf7.py
+++ IMAPClient-0.11/imapclient/test/test_imap_utf7.py
@@ -4,7 +4,7 @@
 
 from __future__ import unicode_literals
 
-from imapclient.six import text_type, PY3
+from six import text_type, PY3
 from imapclient.imap_utf7 import decode, encode
 from imapclient.test.util import unittest
 
