diff -ur iodine-0.6.0-rc1-original/src/tun.c iodine-0.6.0-rc1/src/tun.c
--- iodine-0.6.0-rc1-original/src/tun.c	2012-06-01 22:04:35.870962402 +0200
+++ iodine-0.6.0-rc1/src/tun.c	2012-06-02 04:32:08.967846671 +0200
@@ -455,7 +455,7 @@
 	}
 #ifndef WINDOWS32
 	snprintf(cmdline, sizeof(cmdline), 
-			"/sbin/ifconfig %s %s %s netmask %s",
+			"PATH=/bin:/sbin ifconfig %s %s %s netmask %s",
 			if_name,
 			ip,
 #ifdef FREEBSD
@@ -522,7 +522,7 @@
 
 	if (mtu > 200 && mtu <= 1500) {
 		snprintf(cmdline, sizeof(cmdline), 
-				"/sbin/ifconfig %s mtu %u",
+				"PATH=/bin:/sbin ifconfig %s mtu %u",
 				if_name,
 				mtu);
 		
