--- ksym_mod.c
+++ ksym_mod.c
@@ -95,14 +95,6 @@
 #include <sys/fcntl.h>
 #include <sys/stat.h>
 #include "module.h"
-#if !defined(__GLIBC__)
-#include <linux/time.h>
-#include <linux/linkage.h>
-#else /* __GLIBC__ */
-#include <linux/linkage.h>
-extern __off64_t lseek64 __P ((int __fd, __off64_t __offset, int __whence));
-extern int get_kernel_syms __P ((struct kernel_sym *__table));
-#endif /* __GLIBC__ */
 #include <stdarg.h>
 #include <paths.h>
 #include <linux/version.h>
