--- psi3.orig/src/lib/libint/Makefile.in	2003-11-23 11:07:51.000000000 -0800
+++ psi3/src/lib/libint/Makefile.in	2006-02-10 22:38:37.000000000 -0800
@@ -44,7 +44,10 @@
 $(MAKE) top_objdir=$(top_objdir_path); \
 rm -f Makefile; \
 if ! ./$(BUILD_BINARY); then \
- cd ..; rm -fR $(TEMP_DIR); exit 1; \
+ cd ..; \
+ rm -fR $(TEMP_DIR); \
+ echo "$(BUILD_BINARY) does not exist!"; \
+ exit 1; \
 fi; \
 cp Makefile.libint Makefile; \
 ${MAKE} top_objdir=$(top_objdir_path); \
@@ -65,7 +68,9 @@
 	$(MKDIRS) $(libdir)
 	$(INSTALL_DATA) $< $(libdir)
 	$(RANLIB) $(libdir)/`basename $(TARGET)`
+	echo Finished ranlib
 	$(MKDIRS) $(incdir)/libint
+	echo Finished mkdirs
 	$(INSTALL_INCLUDE) $(TARGET_HEADERS) $(incdir)/libint
 
 #
