--- psi3.orig/configure.in	2005-10-29 10:15:06.000000000 -0700
+++ psi3/configure.in	2006-02-10 20:56:59.000000000 -0800
@@ -104,9 +104,9 @@
 ######################
 
 # potential names of compilers
-FC_LIST="g77 f77 fc f2c"
-CC_LIST="gcc cc"
-CXX_LIST="g++ c++ cxx"
+FC_LIST="gfortran g77 f77 fc f2c ifc"
+CC_LIST="gcc cc icc"
+CXX_LIST="g++ c++ cxx icc"
 
 # set up os specific options
 case $os in
