--- psi3.orig/configure.in	2005-10-29 10:15:06.000000000 -0700
+++ psi3/configure.in	2006-02-10 20:56:59.000000000 -0800
@@ -333,7 +333,7 @@
 AC_ARG_WITH(blas,[  --with-blas             which blas library to use],[
 case $withval in
   yes)
-    BLAS='-lblas'
+    BLAS='-lblas -lpthread'
     ;;
   no)
     BLAS=''
@@ -344,16 +344,16 @@
 esac
 ])
 if test "X$BLAS" = X; then
-  AC_CHECK_LIB(essl, DGEMM, BLAS="-lessl")
+  AC_CHECK_LIB(essl, DGEMM, BLAS="-lessl -lpthread",, [-lpthread])
 fi
 if test "X$BLAS" = X; then
-  AC_CHECK_LIB(scs, DGEMM, BLAS="-lscs")
+  AC_CHECK_LIB(scs, DGEMM, BLAS="-lscs -lpthread",, [-lpthread])
 fi
 if test "X$BLAS" = X; then
-  AC_CHECK_LIB(atlas, DGEMM, BLAS="-latlas")
+  AC_CHECK_LIB(atlas, DGEMM, BLAS="-latlas -lpthread",, [-lpthread])
 fi
 if test "X$BLAS" = X; then
-  AC_CHECK_LIB(blas, DGEMM, BLAS="-lblas")
+  AC_CHECK_LIB(blas, DGEMM, BLAS="-lblas -lpthread",, [-lpthread])
 fi
 if test "X$BLAS" = X; then
  AC_MSG_ERROR("Did not find a BLAS library")
@@ -363,7 +363,7 @@
 AC_ARG_WITH(lapack,[  --with-lapack           which lapack library to use],[
 case $withval in
   yes)
-    LAPACK='-llapack'
+    LAPACK='-llapack -lpthread'
     ;;
   no)
     LAPACK=''
@@ -382,7 +382,7 @@
   LIBS=$CURRENT_LIBS
 fi
 if test "X$LAPACK" = X; then
-  AC_CHECK_LIB(lapack,DGEEV, LAPACK="-llapack",,$BLAS)
+  AC_CHECK_LIB(lapack,DGEEV, LAPACK="-llapack",,$BLAS -lpthread)
 fi
 if test "X$LAPACK" = X; then
  AC_MSG_ERROR("Did not find a LAPACK library")
