diff -Naur wxdfast-0.6.0-orig/configure.in wxdfast-0.6.0/configure.in
--- wxdfast-0.6.0-orig/configure.in	2007-03-11 16:45:25.000000000 -0600
+++ wxdfast-0.6.0/configure.in	2007-10-20 19:35:30.000000000 -0600
@@ -64,13 +64,19 @@
                 ])
 	fi
 
-AC_PATH_PROG([WXRC], [wxrc --version])
-	if test "x$WXRC" = "x"; then
-	AC_MSG_ERROR([
-		wxrc could not be found. This program gets build together with wxWidgets.
-		Please make sure wxrc is within your path.
-		])
-	fi
+AC_ARG_VAR(WXRC, Path to wxrc resource compiler)
+AC_MSG_CHECKING([for wxrc])
+if test "x$WXRC" = x ; then
+    WXRC=`$WX_CONFIG_WITH_ARGS --utility=wxrc`
+    if test "x$WXRC" = x ; then
+        AC_MSG_RESULT([not found])
+        AC_MSG_ERROR([
+            wxrc could not be found. This program gets build together with wxWidgets.
+            Please make sure wxrc is within your path.
+            ])
+    fi
+fi
+AC_MSG_RESULT([$WXRC])
 
 AC_OUTPUT([
 Makefile
