diff --git a/lib/files/cache/legacy.php b/lib/files/cache/legacy.php
index f114cf0..9556a26 100644
--- a/lib/files/cache/legacy.php
+++ b/lib/files/cache/legacy.php
@@ -97,7 +97,7 @@ function getEtag($path) {
 			$relativePath = '';
 		}
 		if(is_null($query)){
-			$query = \OC_DB::prepare('SELECT `propertyvalue` FROM `*PREFIX*properties` WHERE `userid` = ? AND propertypath = ? AND propertyname = "{DAV:}getetag"');
+			$query = \OC_DB::prepare('SELECT `propertyvalue` FROM `*PREFIX*properties` WHERE `userid` = ? AND `propertypath` = ? AND `propertyname` = \'{DAV:}getetag\'');
 		}
 		$result = $query->execute(array($user, '/' . $relativePath));
 		if ($row = $result->fetchRow()) {
