--- configure.in
+++ configure.in
@@ -54,8 +54,8 @@
 ACX_PTHREAD(have_pthreads=yes, have_pthreads=no)
 
 dnl Check for SSL
-AC_ARG_ENABLE([openssl], AC_HELP_STRING([--disable-openssl], [do not compile with OpenSSL support]), [want_ssl=no], [want_ssl=yes])
-if test "$want_ssl" = "yes"; then
+AC_ARG_ENABLE([openssl], AS_HELP_STRING([--disable-openssl], [do not compile with OpenSSL support]))
+if test "$enable_openssl" != "no"; then
 	PKG_CHECK_MODULES([OPENSSL], [openssl])
 	CFLAGS="$CFLAGS $OPENSSL_CFLAGS -DSOCK_SSL"
 	LIBS="$LIBS $OPENSSL_LIBS"
