--- bin/album2booh.old	2008-07-06 13:28:37.000000000 +0100
+++ bin/album2booh	2008-07-06 13:30:21.000000000 +0100
@@ -33,6 +33,7 @@
 # 	$ recode ISO-8859-1..UTF-8 captions.txt
 #
 
+require 'rubygems'
 require 'getoptlong'
 require 'gettext'
 include GetText
--- bin/booh.old	2008-07-06 13:28:37.000000000 +0100
+++ bin/booh	2008-07-06 13:30:40.000000000 +0100
@@ -19,6 +19,7 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
 
+require 'rubygems'
 require 'getoptlong'
 require 'tempfile'
 require 'thread'
--- bin/booh-backend.old	2008-07-06 13:28:37.000000000 +0100
+++ bin/booh-backend	2008-07-06 13:30:51.000000000 +0100
@@ -19,6 +19,7 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
 
+require 'rubygems'
 require 'getoptlong'
 require 'gettext'
 require 'gettext/locale'
--- bin/booh-classifier.old	2008-07-06 13:28:37.000000000 +0100
+++ bin/booh-classifier	2008-07-06 13:31:02.000000000 +0100
@@ -19,6 +19,7 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
 
+require 'rubygems'
 require 'getoptlong'
 require 'tempfile'
 
--- bin/booh-fix-whitebalance.old	2008-07-06 13:28:37.000000000 +0100
+++ bin/booh-fix-whitebalance	2008-07-06 13:31:13.000000000 +0100
@@ -19,6 +19,7 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
 
+require 'rubygems'
 require 'gtk2'
 require 'booh/libadds'
 
--- bin/booh-gamma-correction.old	2008-07-06 13:28:37.000000000 +0100
+++ bin/booh-gamma-correction	2008-07-06 13:31:24.000000000 +0100
@@ -19,6 +19,7 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
 
+require 'rubygems'
 require 'gtk2'
 require 'booh/libadds'
 
--- bin/webalbum2booh.old	2008-07-06 13:28:37.000000000 +0100
+++ bin/webalbum2booh	2008-07-06 13:34:13.000000000 +0100
@@ -19,6 +19,7 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
 
+require 'rubygems'
 require 'getoptlong'
 require 'gettext'
 include GetText
