From beca4dfb0e4d11d3729214967a1fe56ee5669831 Mon Sep 17 00:00:00 2001
From: Keith Packard <keithp@keithp.com>
Date: Fri, 15 Nov 2013 04:39:52 +0000
Subject: Handle new DamageUnregister API which has only one argument

API change in 1.15

Signed-off-by: Keith Packard <keithp@keithp.com>
---
diff --git a/src/drmmode_display.h b/src/drmmode_display.h
index adf4b99..745c484 100644
--- a/src/drmmode_display.h
+++ b/src/drmmode_display.h
@@ -37,6 +37,10 @@
 #define MODESETTING_OUTPUT_SLAVE_SUPPORT 1
 #endif
 
+#if XORG_VERSION_CURRENT >= XORG_VERSION_NUMERIC(1,14,99,2,0)
+#define DamageUnregister(d, dd) DamageUnregister(dd)
+#endif
+
 struct dumb_bo {
     uint32_t handle;
     uint32_t size;
--
cgit v0.9.0.2-2-gbebe
