--- dwb-2012.10.28/scripts/base.js
+++ dwb-2012.10.28/scripts/base.js
@@ -692,7 +692,7 @@ Object.freeze((function () {
         return __addSearchEngine();
       },
     submitSearchEngine : function (obj) {
-        return __submitSearchEngine(obj.searchString);
+        return __submitSearchEngine(JSON.parse(obj).searchString);
       },
     focusInput : function () {
         __focusInput();
