commit 47b95bf3289195bad9b9ec310246131888f26436
Author: David Edmundson <kde@davidedmundson.co.uk>
Date:   Mon May 20 18:15:06 2013 +0100

    Fix compile

diff --git a/lib/chat-widget.cpp b/lib/chat-widget.cpp
index bd8974b..c92aa2a 100644
--- a/lib/chat-widget.cpp
+++ b/lib/chat-widget.cpp
@@ -264,7 +264,7 @@ void ChatWidget::setTextChannel(const Tp::TextChannelPtr &newTextChannelPtr)
     setupChannelSignals();
 
     //if the UI is ready process any messages in queue
-    if (d->chatViewInitialized) {
+    if (d->chatviewlInitialised) {
         Q_FOREACH (const Tp::ReceivedMessage &message, d->channel->messageQueue()) {
             handleIncomingMessage(message, true);
         }
