diff -ruN vompserver-1.4/mvpreceiver.c vompserver-1.5/mvpreceiver.c
--- vompserver-1.4/mvpreceiver.c        2006-03-01 22:29:01.000000000 +0000
+++ vompserver-1.5/mvpreceiver.c        2007-01-16 18:53:27.000000000 +0000
@@ -25,8 +25,10 @@
 MVPReceiver::MVPReceiver(cChannel* channel, cDevice* device)
 #if VDRVERSNUM < 10300
 : cReceiver(channel->Ca(), 0, 7, channel->Vpid(), channel->Ppid(), channel->Apid1(), channel->Apid2(), channel->Dpid1(), channel->Dpid2(), channel->Tpid())
-#else
+#elif VDRVERSNUM < 10500
 : cReceiver(channel->Ca(), 0, channel->Vpid(), channel->Apids(), channel->Dpids(), channel->Spids())
+#else
+: cReceiver(channel->GetChannelID(), 0, channel->Vpid(), channel->Apids(), channel->Dpids(), channel->Spids())
 #endif
 {
   logger = Log::getInstance();
