--- configure.ac	2011-08-31 15:44:57.123536234 +0200
+++ configure.ac	2011-08-31 15:45:38.693016544 +0200
@@ -1940,10 +1940,6 @@
 # We must build without problems with future releases of libraries
 # and disabling deprecated API risks breaking the build
 
-if test "x$have_glib_2_32" != "xyes"; then
-  CPPFLAGS="${CPPFLAGS} -DG_DISABLE_DEPRECATED"
-fi
-
 if test "x$have_gtk_2_26" != "xyes"; then
   CPPFLAGS="${CPPFLAGS} -DGDK_DISABLE_DEPRECATED -DGTK_DISABLE_DEPRECATED"
 fi
