From fa9afae051cf2f8512765df8b3d385b360df03e9 Mon Sep 17 00:00:00 2001
From: Sebastian Pipping <sebastian@pipping.org>
Date: Fri, 4 May 2012 23:10:30 +0200
Subject: [PATCH] Fix configure option --with[out]-bzip2

---
 configure.ac |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/configure.ac b/configure.ac
index 6013f0b..8e50170 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1154,7 +1154,7 @@ AC_SUBST(Z_LIBS)
 # Check for libbzip2
 ####################
 
-AC_ARG_WITH(gs,  [  --without-bzip2         build without bzip2 support])
+AC_ARG_WITH(bzip2,  [  --without-bzip2         build without bzip2 support])
 
 have_libbzip2=no
 if test "x$with_bzip2" != xno; then
-- 
1.7.8.5

