commit c479f5cfe93eaefae36654c499e0e06fecb70724
Author: Loren Segal <lsegal@soen.ca>
Date:   Mon Aug 6 16:50:41 2012 -0400

    Fix test failures in server specs
    
    Fixes #576

diff --git a/spec/cli/server_spec.rb b/spec/cli/server_spec.rb
index 6a33b55..c63e145 100644
--- a/spec/cli/server_spec.rb
+++ b/spec/cli/server_spec.rb
@@ -32,7 +32,7 @@ describe YARD::CLI::Server do
       @libraries = {library.name => [library]}
     end
     unless @no_verify_libraries
-      @libraries.values.each {|libs| libs.each {|lib| File.should_receive(:exist?).at_least(1).times.with(File.expand_path(lib.yardoc_file)).and_return(true) } }
+      @libraries.values.each {|libs| libs.each {|lib| File.stub!(:exist?).with(File.expand_path(lib.yardoc_file)).and_return(true) } }
     end
     unless @no_adapter_mock
       @cli.stub!(:adapter).and_return(@adapter)
