=== modified file 'src/Makefile.am'
--- src/Makefile.am	2013-07-07 19:51:19 +0000
+++ src/Makefile.am	2013-07-07 22:55:28 +0000
@@ -70,6 +70,8 @@
 endif
 
 if HAVE_DIAG
+VALAFLAGS += \
+  -D DIAGNOSTIC
 gnomeccuidir = $(datadir)/gnome-control-center/ui/
 SUBDIRS = diagnostics
 endif

=== modified file 'src/activity-log-manager.vala'
--- src/activity-log-manager.vala	2013-07-07 21:51:26 +0000
+++ src/activity-log-manager.vala	2013-07-07 22:55:28 +0000
@@ -19,7 +19,10 @@
  * You should have received a copy of the GNU Lesser General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.";
  */
+ 
+#if DIAGNOSTIC
 extern Gtk.Widget whoopsie_daisy_preferences_new ();
+#endif
 
 namespace Alm {
 
@@ -35,13 +38,11 @@
 			Object(orientation: Gtk.Orientation.VERTICAL);
 			this.set_size_request(600, 450);
 			this.spacing = 6;
-
 			this.margin = 12;
 
 			blacklist = new Blacklist();
 			privacy_widget = new PrivacyWidget(blacklist);
-			whoopsie = whoopsie_daisy_preferences_new ();
-
+			
 			notebook = new Gtk.Notebook();
 			this.pack_start(notebook, true, true, 0);
 			var privacy_label = new Gtk.Label(_("Files & Applications"));
@@ -54,8 +55,11 @@
 				notebook.prepend_page(security_widget, security_label);
 			}
 
+#if DIAGNOSTIC
+			whoopsie = whoopsie_daisy_preferences_new ();
 			var whoopsie_label = new Gtk.Label(_("Diagnostics"));
 			notebook.append_page(whoopsie, whoopsie_label);
+#endif
 			this.show_all();
 		}
 

