 data/Makefile.am |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/data/Makefile.am b/data/Makefile.am
index 41a1bf0..1b90451 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -21,7 +21,7 @@ nobase_dist_data_DATA = \
 	icons/hicolor/48x48/apps/activity-log-manager.svg
 
 desktopdir = $(datadir)/applications
-desktop_in_files = activity-log-manager.desktop.in activity-log-manager-ccpanel.desktop.in
+desktop_in_files = activity-log-manager.desktop.in
 
 if HAVE_CCPANEL
 desktop_in_files+ = activity-log-manager-ccpanel.desktop.in
