diff --git a/GUI/kde/p7zip_compress.desktop b/GUI/kde/p7zip_compress.desktop
index 9010fbd..b965fa6 100644
--- a/GUI/kde/p7zip_compress.desktop
+++ b/GUI/kde/p7zip_compress.desktop
@@ -1,6 +1,8 @@
 [Desktop Entry]
-ServiceTypes=inode/directory
-Actions=CompressDialog;Compress7z;CompressZip;extractHere;ExtractSubdir;ExtractTo;test;Benchmark
+Type=Service
+ServiceTypes=KonqPopupMenu/Plugin
+MimeType=all/all;
+Actions=CompressDialog;Compress7z;CompressZip;Benchmark
 X-KDE-Priority=TopLevel
 X-KDE-Submenu=7-Zip
 X-KDE-Icon=p7zip
diff --git a/GUI/kde/p7zip_extract.desktop b/GUI/kde/p7zip_extract.desktop
index 9038c76..a7dca4d 100644
--- a/GUI/kde/p7zip_extract.desktop
+++ b/GUI/kde/p7zip_extract.desktop
@@ -1,10 +1,30 @@
 [Desktop Entry]
 X-SuSE-translate=true
-Encoding=UTF-8
-ServiceTypes=application/x-gzip,application/x-lha,application/x-tar,application/x-tgz,application/x-tbz,application/x-tbz2,application/x-zip,application/x-bzip,application/x-tzo,application/x-lzop,application/x-rar,application/x-rar-compressed,application/x-zoo,application/x-tarz,application/x-archive,application/x-bzip2,application/x-jar,application/x-deb,application/x-ace,application/x-7z,application/x-arc,application/x-arj,application/x-compress,application/x-cpio,application/x-pak
-Actions=extractHere
+Type=Service
+ServiceTypes=KonqPopupMenu/Plugin
+MimeType=application/x-gzip;application/x-lha;application/x-tar;application/x-compressed-tar;application/x-bzip-compressed-tar;application/zip;application/x-bzip;application/x-tzo;application/x-lzop;application/x-rar;application/x-zoo;application/x-tarz;application/x-archive;application/x-bzip2;application/x-java-archive;application/x-deb;application/x-ace;application/x-7z-compressed;application/x-arc;application/x-arj;application/x-compress;application/x-zip-compressed;application/x-cpio;application/x-pak;
+#MimeType=application/x-gzip;application/x-lha;application/x-tar;application/x-tgz;application/x-tbz;application/x-tbz2;application/x-zip;application/x-bzip;application/x-tzo;application/x-lzop;application/x-rar;application/x-rar-compressed;application/x-zoo;application/x-tarz;application/x-archive;application/x-bzip2;application/x-jar;application/x-deb;application/x-ace;application/x-7z;application/x-arc;application/x-arj;application/x-compress;application/x-cpio;application/x-pak;
+Actions=extractHere;ExtractSubdir;ExtractTo;test;
+X-KDE-Priority=TopLevel
+X-KDE-Submenu=7-Zip
+X-KDE-Icon=p7zip
 
 [Desktop Action extractHere]
-Name=Extract Here with 7-zip
+Name=Extract Here
 Exec=p7zipForFilemanager x %U
 Icon=p7zip
+
+[Desktop Action ExtractSubdir]
+Name=Extract to subdirectory
+Icon=p7zip
+Exec=p7zipForFilemanager xs %U
+
+[Desktop Action ExtractTo]
+Name=Extract to ...
+Icon=p7zip
+Exec=p7zipForFilemanager xd %U
+
+[Desktop Action test]
+Name=Test Archive
+Exec=p7zipForFilemanager t %U
+Icon=p7zip
diff --git a/GUI/kde/p7zip_extract_subdir.desktop b/GUI/kde/p7zip_extract_subdir.desktop
index 1800fa0..a711f3c 100644
--- a/GUI/kde/p7zip_extract_subdir.desktop
+++ b/GUI/kde/p7zip_extract_subdir.desktop
@@ -1,6 +1,5 @@
 [Desktop Entry]
 X-SuSE-translate=true
-Encoding=UTF-8
 ServiceTypes=application/x-gzip,application/x-lha,application/x-tar,application/x-tgz,application/x-tbz,application/x-tbz2,application/x-zip,application/x-bzip,application/x-tzo,application/x-lzop,application/x-rar,application/x-rar-compressed,application/x-zoo,application/x-tarz,application/x-archive,application/x-bzip2,application/x-jar,application/x-deb,application/x-ace,application/x-7z,application/x-arc,application/x-arj,application/x-compress,application/x-cpio,application/x-pak
 Actions=ExtractSubdir
 
diff --git a/GUI/kde/p7zip_extract_to.desktop b/GUI/kde/p7zip_extract_to.desktop
index bf13ec0..18d2b74 100644
--- a/GUI/kde/p7zip_extract_to.desktop
+++ b/GUI/kde/p7zip_extract_to.desktop
@@ -1,6 +1,5 @@
 [Desktop Entry]
 X-SuSE-translate=true
-Encoding=UTF-8
 ServiceTypes=application/x-gzip,application/x-lha,application/x-tar,application/x-tgz,application/x-tbz,application/x-tbz2,application/x-zip,application/x-bzip,application/x-tzo,application/x-lzop,application/x-rar,application/x-rar-compressed,application/x-zoo,application/x-tarz,application/x-archive,application/x-bzip2,application/x-jar,application/x-deb,application/x-ace,application/x-7z,application/x-arc,application/x-arj,application/x-compress,application/x-cpio,application/x-pak
 Actions=ExtractTo
 
diff --git a/GUI/kde/p7zip_test.desktop b/GUI/kde/p7zip_test.desktop
index 2c17acb..5212bb2 100644
--- a/GUI/kde/p7zip_test.desktop
+++ b/GUI/kde/p7zip_test.desktop
@@ -1,6 +1,5 @@
 [Desktop Entry]
 X-SuSE-translate=true
-Encoding=UTF-8
 ServiceTypes=application/x-gzip,application/x-lha,application/x-tar,application/x-tgz,application/x-tbz,application/x-tbz2,application/x-zip,application/x-bzip,application/x-tzo,application/x-lzop,application/x-rar,application/x-rar-compressed,application/x-zoo,application/x-tarz,application/x-archive,application/x-bzip2,application/x-jar,application/x-deb,application/x-ace,application/x-7z,application/x-arc,application/x-arj,application/x-compress,application/x-cpio,application/x-pak
 Actions=test
 
