--- Makefile.orig	2012-07-05 12:22:03.000000000 +0200
+++ Makefile	2012-07-09 06:11:49.281442066 +0200
@@ -8,7 +8,7 @@
 PREFIX=/usr/local
 MANPREFIX=${PREFIX}/share/man
 
-all:	$(LIBS) $(PROGRAMS) dnsdict6 thcping6
+all:	$(LIBS) $(PROGRAMS) dnsdict6
 
 dnsdict6:	dnsdict6.c $(LIBS)
 	$(CC) $(CFLAGS) -o $@ $^ $(LDFLAGS) -lpthread -lresolv
@@ -22,9 +22,9 @@
 strip:	all
 	strip $(PROGRAMS) dnsdict6 thcping6
 
-install: all strip
+install: all
 	install -m0755 -d ${DESTDIR}${PREFIX}/bin
-	install -m0755 $(PROGRAMS) dnsdict6 thcping6 *.sh ${DESTDIR}${PREFIX}/bin
+	install -m0755 $(PROGRAMS) dnsdict6 *.sh ${DESTDIR}${PREFIX}/bin
 	install -m0755 -d ${DESTDIR}${MANPREFIX}/man8
 	install -m0644 -D thc-ipv6.8 ${DESTDIR}${MANPREFIX}/man8
 
