--- data/stellarium.desktop.orig	2012-05-18 08:19:20.000000000 -0400
+++ data/stellarium.desktop	2012-08-03 03:34:27.064349264 -0400
@@ -1,6 +1,5 @@
 [Desktop Entry]
-Encoding=UTF-8
-Version=0.11.3
+Version=1.0
 Type=Application
 Name=Stellarium
 GenericName=Desktop planetarium
@@ -14,22 +13,22 @@
 Comment[fr]=Planétarium
 Comment[ru]=Планетарий
 
-Actions=Window;Fullscreen;Safemode
+Actions=Window;Fullscreen;Safemode;
 
 [Desktop Action Window]
 Name=Open in window
 Name[ru]=Открыть в окне
 Exec=stellarium -f no
-OnlyShowIn=Unity
+OnlyShowIn=Unity;
 
 [Desktop Action Fullscreen]
 Name=Open in fullscreen
 Name[ru]=Открыть в полноэкранном режиме
 Exec=stellarium -f yes
-OnlyShowIn=Unity
+OnlyShowIn=Unity;
 
 [Desktop Action Safemode]
 Name=Open in safe mode
 Name[ru]=Открыть в безопасном режиме
 Exec=stellarium --safe-mode
-OnlyShowIn=Unity
+OnlyShowIn=Unity;
