commit 6c3d14d7d4f2b34d448f982ca0d7d4be39955666
Author: hasufell <hasufell@gentoo.org>
Date:   Sat Sep 21 14:28:59 2013 +0200

    remove intrusive donate button

diff --git a/bin/gtk-youtube-viewer b/bin/gtk-youtube-viewer
index e33570a..34f5b07 100755
--- a/bin/gtk-youtube-viewer
+++ b/bin/gtk-youtube-viewer
@@ -353,7 +353,6 @@ my $app_icon_pixbuf    = 'Gtk2::Gdk::Pixbuf'->new_from_file("$icons_path/$execna
 my $user_icon_pixbuf   = 'Gtk2::Gdk::Pixbuf'->new_from_file_at_size("$icons_path/user.png", 16, 16);
 my $feed_icon_pixbuf   = 'Gtk2::Gdk::Pixbuf'->new_from_file_at_size("$icons_path/feed_icon.png", 16, 16);
 my $default_thumb      = 'Gtk2::Gdk::Pixbuf'->new_from_file_at_size("$icons_path/default_thumb.jpg", 120, 90);
-my $donate_icon_pixbuf = 'Gtk2::Gdk::Pixbuf'->new_from_file("$icons_path/donate.png");
 my $animation          = 'Gtk2::Gdk::PixbufAnimation'->new_from_file("$icons_path/spinner.gif");
 
 # Setting application title and icon
@@ -595,7 +594,6 @@ sub new_image_from_pixbuf {
 {
     $gui->get_object('username_list')->set_image(new_image_from_pixbuf('icon_from_pixbuf', $user_icon_pixbuf));
     $gui->get_object('channels_button')->set_image(new_image_from_pixbuf('icon_from_pixbuf', $user_icon_pixbuf));
-    $gui->get_object('donate_button')->set_image(new_image_from_pixbuf('icon_from_pixbuf', $donate_icon_pixbuf));
     $gui->get_object('button6')->set_image(new_image_from_pixbuf('icon_from_pixbuf', $feed_icon_pixbuf));
     $gui->get_object('button23')->set_image(new_image_from_pixbuf('icon_from_pixbuf', $feed_icon_pixbuf));
 }
diff --git a/share/gtk-youtube-viewer.glade b/share/gtk-youtube-viewer.glade
index 2fe6948..f7f8528 100644
--- a/share/gtk-youtube-viewer.glade
+++ b/share/gtk-youtube-viewer.glade
@@ -1178,23 +1178,6 @@ Unless the categoryID is valid, this field is ignored.</property>
                                         <child>
                                           <placeholder/>
                                         </child>
-                                        <child>
-                                          <object class="GtkButton" id="donate_button">
-                                            <property name="use_action_appearance">False</property>
-                                            <property name="visible">True</property>
-                                            <property name="can_focus">True</property>
-                                            <property name="receives_default">True</property>
-                                            <property name="image">image78</property>
-                                            <property name="relief">none</property>
-                                            <property name="image_position">right</property>
-                                            <signal name="clicked" handler="donate" swapped="no"/>
-                                          </object>
-                                          <packing>
-                                            <property name="expand">False</property>
-                                            <property name="fill">False</property>
-                                            <property name="position">1</property>
-                                          </packing>
-                                        </child>
                                       </object>
                                       <packing>
                                         <property name="expand">False</property>
