From: Julian Ospald <hasufell@gentoo.org>
Date: Mon Sep 24 23:14:08 UTC 2012
Subject: fix desktop file

https://github.com/trizen/youtube-viewer/issues/12

--- share/gtk-youtube-viewer.desktop
+++ share/gtk-youtube-viewer.desktop
@@ -1,11 +1,10 @@
 [Desktop Entry]
 Name=GTK Youtube Viewer
 Version=1.0
-Encoding=UTF-8
 Comment=A Gtk2 application to search and play Youtube videos with MPlayer.
 Exec=gtk-youtube-viewer
 Icon=gtk-youtube-viewer
 StartupNotify=false
 Terminal=false
 Type=Application
-Categories=AudioVideo
+Categories=AudioVideo;
