--- ext/charlock_holmes/extconf.rb.old	2014-04-19 04:10:03.261265135 +0200
+++ ext/charlock_holmes/extconf.rb	2014-04-19 04:11:01.717266700 +0200
@@ -45,40 +45,5 @@
   exit(1)
 end
 
-##
-# libmagic dependency
-#
-
-src = File.basename('file-5.08.tar.gz')
-dir = File.basename(src, '.tar.gz')
-
-Dir.chdir("#{CWD}/src") do
-  FileUtils.rm_rf(dir) if File.exists?(dir)
-
-  sys("tar zxvf #{src}")
-  Dir.chdir(dir) do
-    sys("./configure --prefix=#{CWD}/dst/ --disable-shared --enable-static --with-pic")
-    sys("patch -p0 < ../file-soft-check.patch")
-    sys("make -C src install")
-    sys("make -C magic install")
-  end
-end
-
-FileUtils.cp "#{CWD}/dst/lib/libmagic.a", "#{CWD}/libmagic_ext.a"
-
-$INCFLAGS[0,0] = " -I#{CWD}/dst/include "
-$LDFLAGS << " -L#{CWD} "
-
-dir_config 'magic'
-unless have_library 'magic_ext' and have_header 'magic.h'
-  STDERR.puts "\n\n"
-  STDERR.puts "***************************************************************************************"
-  STDERR.puts "********* error compiling and linking libmagic. please report issue on github *********"
-  STDERR.puts "***************************************************************************************"
-  exit(1)
-end
-
-$CFLAGS << ' -Wall -funroll-loops'
-$CFLAGS << ' -Wextra -O0 -ggdb3' if ENV['DEBUG']
-
+$LIBS << " -lz -lmagic -licuuc -licudata "
 create_makefile 'charlock_holmes/charlock_holmes'
