--- src/gnome-terminal.schemas.in	2003-02-21 20:06:56.000000000 -0600
+++ src/gnome-terminal.schemas.in	2003-02-21 20:07:13.000000000 -0600
@@ -341,7 +341,7 @@
       <applyto>/apps/gnome-terminal/profiles/Default/login_shell</applyto>
       <owner>gnome-terminal</owner>
       <type>bool</type>
-      <default>false</default>
+      <default>true</default>
       <locale name="C">
          <short>Whether to launch the command in the terminal as a login shell</short>
          <long>
