--- python/libmu_py/Makefile.in.org	2009-09-10 21:18:43.000000000 +0200
+++ python/libmu_py/Makefile.in	2010-01-21 19:51:58.000000000 +0100
@@ -122,7 +122,7 @@
 	"$(DESTDIR)$(lispdir)"
 LTLIBRARIES = $(lib_LTLIBRARIES) $(pythonexec_LTLIBRARIES)
 am__DEPENDENCIES_1 =
-c_api_la_DEPENDENCIES = $(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1)
+c_api_la_DEPENDENCIES = libmu_py.la
 am_c_api_la_OBJECTS = c_api.lo
 c_api_la_OBJECTS = $(am_c_api_la_OBJECTS)
 AM_V_lt = $(am__v_lt_$(V))
--- python/libmu_py/Makefile.in.org	2010-01-21 19:54:19.000000000 +0100
+++ python/libmu_py/Makefile.in	2010-01-21 20:29:56.000000000 +0100
@@ -886,7 +886,7 @@
 	  echo "rm -f \"$${dir}/so_locations\""; \
 	  rm -f "$${dir}/so_locations"; \
 	done
-install-pythonexecLTLIBRARIES: $(pythonexec_LTLIBRARIES)
+install-pythonexecLTLIBRARIES: $(pythonexec_LTLIBRARIES) install-libLTLIBRARIES
 	@$(NORMAL_INSTALL)
 	test -z "$(pythonexecdir)" || $(MKDIR_P) "$(DESTDIR)$(pythonexecdir)"
 	@list='$(pythonexec_LTLIBRARIES)'; test -n "$(pythonexecdir)" || list=; \
