diff -ur KGtk-0.10.1.orig/qt4/kqt4.cpp KGtk-0.10.1/qt4/kqt4.cpp
--- KGtk-0.10.1.orig/qt4/kqt4.cpp	2009-03-12 02:19:40.000000000 +0200
+++ KGtk-0.10.1/qt4/kqt4.cpp	2009-08-09 00:57:37.000000000 +0300
@@ -99,7 +99,7 @@
     {
         const char *a=useQt && QCoreApplication::arguments().count()
                         ? QCoreApplication::arguments()[0].toLatin1().constData() : getAppNameFromPid(getpid());
-        char       *slash;
+        const char *slash;
 
         // Was the cmdline app java? if so, try to use its parent name - just in case
         //   its run from a shell script, etc. - e.g. as eclipse does
