diff -ur createtorrent-1.1.3.orig/configure.in createtorrent-1.1.3/configure.in
--- createtorrent-1.1.3.orig/configure.in	2007-02-10 10:20:05.000000000 +0200
+++ createtorrent-1.1.3/configure.in	2007-04-13 22:05:13.000000000 +0300
@@ -2,7 +2,7 @@
 AM_CONFIG_HEADER(config.h)
 AM_INIT_AUTOMAKE(createtorrent,1.1.3)
 AC_PROG_CC
-AC_CHECK_LIB([ssl],[SHA1],,[AC_MSG_ERROR([error, OpenSSL required])])
+AC_CHECK_LIB(crypto, SHA1, LIBS="$LIBS -lcrypto")
 AC_HEADER_DIRENT
 AC_HEADER_STDC
 AC_CHECK_HEADER([openssl/sha.h],,[AC_MSG_ERROR([Error, OpenSSL header file sha.h not found.])])
