--- configure.ac.orig	2006-04-22 14:11:34.000000000 -0500
+++ configure.ac	2006-04-22 14:14:36.000000000 -0500
@@ -115,14 +115,25 @@
 	AC_MSG_ERROR([No gacutil tool found])
 fi
 
-dnl Check for the NUnit testing framework
-have_nunit="yes"
-AC_PATH_PROG(NUNIT, nunit-console)
-AC_PATH_PROG(XSLTPROC, xsltproc)
-if test "x$NUNIT" = "x" -o "x$XSLTPROC" = "x"; then
-	AC_MSG_WARN([Can't find NUnit and xsltproc. You will not be able to run unit tests])
-	have_nunit="no"
+AC_ARG_ENABLE([tests],
+              AC_HELP_STRING([--disable-tests], [Disable the unit tests]),,
+              [enable_tests=yes])
+
+if test "x$enable_tests" = "xyes" ; then
+   dnl Check for the NUnit testing framework
+   have_nunit="yes"
+
+   AC_PATH_PROG(NUNIT, nunit-console)
+   AC_PATH_PROG(XSLTPROC, xsltproc)
+
+   if test "x$NUNIT" = "x" -o "x$XSLTPROC" = "x"; then
+      AC_MSG_WARN([Can't find NUnit and xsltproc. You will not be able to run unit tests])
+      have_nunit="no"
+   fi
+else
+   have_nuint="no"
 fi
+
 AM_CONDITIONAL(HAVE_NUNIT, test "$have_nunit" = "yes")
 
 AC_SUBST(RUNTIME)
