http://bugs.gentoo.org/481360

--- nxcomp/Jpeg.cpp
+++ nxcomp/Jpeg.cpp
@@ -443,7 +443,7 @@
 
   JpegSetSrcManager(&cinfo, compressedData, compressedLen);
 
-  jpeg_read_header(&cinfo, 1);
+  jpeg_read_header(&cinfo, TRUE);
 
   if (jpegError) goto AbortDecompressJpeg16;
 
@@ -584,7 +584,7 @@
 
   JpegSetSrcManager(&cinfo, compressedData, compressedLen);
 
-  jpeg_read_header(&cinfo, 1);
+  jpeg_read_header(&cinfo, TRUE);
 
   if (jpegError) goto AbortDecompressJpeg24;
 
@@ -721,7 +721,7 @@
 
   JpegSetSrcManager(&cinfo, compressedData, compressedLen);
 
-  jpeg_read_header(&cinfo, 1);
+  jpeg_read_header(&cinfo, TRUE);
 
   if (jpegError) goto AbortDecompressJpeg32;
 
@@ -836,7 +836,7 @@
   jpegSrcManager.bytes_in_buffer = jpegBufferLen;
   jpegSrcManager.next_input_byte = (JOCTET *)jpegBufferPtr;
 
-  return 1;
+  return TRUE;
 }
 
 static void JpegSkipInputData(j_decompress_ptr cinfo, long num_bytes)
