diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -245,8 +245,31 @@
     AC_MSG_WARN([some Perl modules required to compile or test the Perl bindings are missing])
 fi
 
+AC_ARG_ENABLE([perl],
+       AS_HELP_STRING([--with-perl],[build Perl binding]))
+
+
+PERL=
+AS_IF([test "x$enable_perl" != "xno" ],[
+       AC_CHECK_PROG([PERL],[perl],[yes],[no])
+
+    for pm in Test::More ExtUtils::MakeMaker IO::Stringy; do
+        AC_MSG_CHECKING([for $pm])
+        if ! perl -M$pm -e1 >/dev/null 2>&1; then
+            AC_MSG_RESULT([no])
+            missing_perl_modules=yes
+        else
+            AC_MSG_RESULT([yes])
+        fi
+    done
+
+    AS_IF([test "x$missing_perl_modules" = "xyes"],
+        [AC_MSG_FAILURE([some Perl modules required to compile or test the Perl bindings are missing])],
+        )
+])
+
 AM_CONDITIONAL([HAVE_PERL],
-    [test "x$PERL" != "xno" && test "x$missing_perl_modules" != "xyes"])
+    [test "x$PERL" != "xno" && test "x$missing_perl_modules" != "xyes" -a "x$enable_perl" = "xyes"])
 
 dnl Check for Python (optional, for Python bindings).
 PYTHON_PREFIX=
