--- Makefile.orig	2009-01-04 15:50:20.457840024 +0100
+++ Makefile	2009-01-04 15:49:52.019837225 +0100
@@ -19,7 +19,7 @@
 conf-update.h: core.h helpers.h index.h modified.h config.h
 
 conf-update: core.o helpers.o conf-update.o index.o modified.o config.o
-	$(CC) $$(pkg-config --libs glib-2.0) -lncurses -lmenu -lcrypto $(LDFLAGS) -o conf-update config.o core.o helpers.o conf-update.o index.o modified.o
+	$(CC) $(LDFLAGS) -o conf-update config.o core.o helpers.o conf-update.o index.o modified.o $$(pkg-config --libs glib-2.0) -lncurses -lmenu -lcrypto
 
 .PHONY: clean
 
