--- conf-update-1.0/conf-update.c.orig	2006-10-09 13:23:33.000000000 +0200
+++ conf-update-1.0/conf-update.c	2010-09-23 08:58:54.982551103 +0200
@@ -133,12 +133,12 @@
 				esc_myfile = g_shell_quote(myfile);
 				highest = get_highest_update(protected, protected[i]);
 				esc_highest = g_shell_quote(highest);
-				cmd = (char *)calloc(strlen("diff -Nu % % | grep \"^[+-][^+-]\" | grep -v \"^[-+]#\" | grep -v \"^[-+][:space:]*$\" " ) + strlen(esc_highest) + strlen(esc_myfile), sizeof(char));
+				cmd = (char *)calloc(strlen("diff -Nu % % | grep \"^[+-][^+-]\" | grep -v \"^[-+]#\" | grep -v \"^[-+][[:space:]]*$\" " ) + strlen(esc_highest) + strlen(esc_myfile), sizeof(char));
 				strcpy(cmd, "diff -Nu ");
 				strcat(cmd, esc_myfile);
 				strcat(cmd, " ");
 				strcat(cmd, esc_highest);
-				strcat(cmd, " | grep \"^[+-][^+-]\" | grep -v \"^[-+]#\" | grep -v \"^[-+][:space:]*$\"");
+				strcat(cmd, " | grep \"^[+-][^+-]\" | grep -v \"^[-+]#\" | grep -v \"^[-+][[:space:]]*$\"");
 				
 				free(myfile);
 				free(esc_myfile);
