diff -ruN valgrind-3.3.0/coregrind/launcher-linux.c valgrind-3.3.0-fixed/coregrind/launcher-linux.c
--- valgrind-3.3.0/coregrind/launcher-linux.c	2007-12-11 00:18:46.000000000 +0100
+++ valgrind-3.3.0-fixed/coregrind/launcher-linux.c	2007-12-13 21:07:21.388704395 +0100
@@ -246,7 +246,7 @@
    else if (0==strcmp(VG_PLATFORM,"ppc32-linux"))
       default_platform = "ppc32-linux";
    else if (0==strcmp(VG_PLATFORM,"ppc64-linux"))
-      default_platform = "ppc32-linux";
+      default_platform = "ppc64-linux";
    else
       barf("Unknown VG_PLATFORM '%s'", VG_PLATFORM);
 
