diff -ruN valgrind-3.2.2/valgrind.pc.in valgrind-3.2.2-fixed/valgrind.pc.in
--- valgrind-3.2.2/valgrind.pc.in	2007-01-02 15:52:30.000000000 +0100
+++ valgrind-3.2.2-fixed/valgrind.pc.in	2007-01-25 18:54:13.478053000 +0100
@@ -4,13 +4,13 @@
 includedir=@includedir@/valgrind
 arch=@VG_ARCH@
 os=@VG_OS@
-platform=@VG_PLATFORM_PRI@
+platform=@VG_ARCH@-@VG_OS@
 valt_load_address=@VALT_LOAD_ADDRESS@
 
 Name: Valgrind
 Description: A dynamic binary instrumentation framework
 Version: @VERSION@
 Requires:
-Libs: -L${libdir}/valgrind/@VG_PLATFORM_PRI@ -lcoregrind -lvex -lgcc
+Libs: -L${libdir}/valgrind/@VG_ARCH@-@VG_OS@ -lcoregrind -lvex -lgcc
 Cflags: -I${includedir}
 
