diff -ruN valgrind-3.4.1/auxprogs/Makefile.am valgrind-3.4.1-fixed/auxprogs/Makefile.am
--- valgrind-3.4.1/auxprogs/Makefile.am	2009-03-01 22:55:53.000000000 +0100
+++ valgrind-3.4.1-fixed/auxprogs/Makefile.am	2009-08-08 19:56:32.000000000 +0200
@@ -64,14 +64,14 @@
 if BUILD_MPIWRAP_PRI
 noinst_PROGRAMS       += libmpiwrap-@VG_PLATFORM_PRI@.so
 libmpiwrap-@VG_PLATFORM_PRI@.so: libmpiwrap.c
-	$(MPI_CC) $(HACKY_FLAGS_PRI) \
+	$(MPI_CC) $(HACKY_FLAGS_PRI) $(LDFLAGS) \
 		-I../include \
 		-o libmpiwrap-@VG_PLATFORM_PRI@.so libmpiwrap.c
 endif
 if BUILD_MPIWRAP_SEC
 noinst_PROGRAMS       += libmpiwrap-@VG_PLATFORM_SEC@.so
 libmpiwrap-@VG_PLATFORM_SEC@.so: libmpiwrap.c
-	$(MPI_CC) $(HACKY_FLAGS_SEC) \
+	$(MPI_CC) $(HACKY_FLAGS_SEC) $(LDFLAGS) \
 		-I../include \
 		-o libmpiwrap-@VG_PLATFORM_SEC@.so libmpiwrap.c
 endif
