Upstream revision 678 fixes compilation with mono 2.6.

=== modified file 'Evolution/src/ContactItemSource.cs'
--- Evolution/src/ContactItemSource.cs	2009-06-22 04:05:16 +0000
+++ Evolution/src/ContactItemSource.cs	2010-02-12 01:08:47 +0000
@@ -32,8 +32,8 @@
 {
 	public struct ContactAttribute
 	{
-		public string Detail { get; protected set; }
-		public string Key { get; protected set; }
+		public string Detail { get; private set; }
+		public string Key { get; private set; }
 		
 		public ContactAttribute (string key, string detail)
 		{

