From 1df177f35a05db505577cdc929e63fde906a704b Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@gmail.com>
Date: Wed, 31 Jul 2013 00:51:03 +0000
Subject: nouveau: fix build on ppc by wrapping immintrin include.

Signed-off-by: Dave Airlie <airlied@redhat.com>
---
diff --git a/src/nouveau_xv.c b/src/nouveau_xv.c
index 5569b7c..e19378d 100644
--- a/src/nouveau_xv.c
+++ b/src/nouveau_xv.c
@@ -25,7 +25,9 @@
 #include "config.h"
 #endif
 
+#ifdef __SSE2__
 #include <immintrin.h>
+#endif
 
 #include "xf86xv.h"
 #include <X11/extensions/Xv.h>
--
cgit v0.9.0.2-2-gbebe
