diff --git a/src/src/main/java/jline/Terminal.java b/src/src/main/java/jline/Terminal.java
index b9b9061..3b22995 100644
--- a/src/src/main/java/jline/Terminal.java
+++ b/src/src/main/java/jline/Terminal.java
@@ -8,8 +8,6 @@ package jline;
 
 import java.io.*;
 
-import com.sun.jmx.snmp.ThreadContext;
-
 /**
  *  Representation of the input terminal for a platform. Handles
  *  any initialization that the platform may need to perform
