diff -uNr cddl.old/lib/libzfs/Makefile cddl/lib/libzfs/Makefile
--- cddl.old/lib/libzfs/Makefile	2013-08-09 17:37:00.000000000 -0400
+++ cddl/lib/libzfs/Makefile	2013-08-09 17:44:08.000000000 -0400
@@ -8,7 +8,7 @@
 LIB=	zfs
 DPADD=	${LIBMD} ${LIBPTHREAD} ${LIBUMEM} ${LIBUTIL} ${LIBM} ${LIBNVPAIR} \
 	${LIBZFS_CORE}
-LDADD=	-lmd -lpthread -lumem -lutil -lm -lnvpair -lzfs_core
+LDADD=	-lmd -lpthread -lumem -lutil -lm -lnvpair -lzfs_core -lgeom -luutil -lavl
 
 SRCS=	deviceid.c \
 	fsshare.c \
diff -uNr cddl.old/lib/libzpool/Makefile cddl/lib/libzpool/Makefile
--- cddl.old/lib/libzpool/Makefile	2013-08-09 17:46:23.000000000 -0400
+++ cddl/lib/libzpool/Makefile	2013-08-09 17:50:39.000000000 -0400
@@ -61,7 +61,7 @@
 LDFLAGS+=	-L${.CURDIR}/../../lib/libumem
 
 DPADD=		${LIBMD} ${LIBPTHREAD} ${LIBZ}
-LDADD=		-lmd -lpthread -lz
+LDADD=		-lmd -lpthread -lz -lumem -lavl
 
 # atomic.S doesn't like profiling.
 NO_PROFILE=
