--- xt-20051206/src/xt/java/com/jclark/xsl/trax/TransformerFactoryImpl.java.ark	2006-02-10 22:14:26.000000000 +0100
+++ xt-20051206/src/xt/java/com/jclark/xsl/trax/TransformerFactoryImpl.java	2006-02-11 02:03:10.000000000 +0100
@@ -384,6 +384,13 @@
 
     }
 
+    /**
+     *
+     */
+    public void setFeature(String name, boolean value) throws TransformerConfigurationException
+    {
+    }
+
     //
     //
     //
