From https://www.redhat.com/archives/libvir-list/2014-July/msg01076.html

diff --git a/src/util/virnuma.c b/src/util/virnuma.c
index 46f48d2..7a11a3b 100644
--- a/src/util/virnuma.c
+++ b/src/util/virnuma.c
@@ -390,7 +390,7 @@ virNumaGetMaxCPUs(void)
 }


-#ifdef HAVE_NUMA_BITMASK_ISBITSET
+#if WITH_NUMACTL && HAVE_NUMA_BITMASK_ISBITSET
 /**
  * virNumaNodeIsAvailable:
  * @node: node to check
@@ -485,7 +485,7 @@ virNumaGetDistances(int node ATTRIBUTE_UNUSED,
 {
     *distances = NULL;
     *ndistances = 0;
-    VIR_DEBUG("NUMA distance information isn't availble on this host");
+    VIR_DEBUG("NUMA distance information isn't available on this host");
     return 0;
 }
 #endif
