--- sharutils-4.6.3.orig/lib/inttostr.h
+++ sharutils-4.6.3/lib/inttostr.h
@@ -26,6 +26,10 @@
 # include <inttypes.h>
 #endif
 
+#if HAVE_STDINT_H
+# include <stdint.h>
+#endif
+
 #if HAVE_LIMITS_H
 # include <limits.h>
 #endif
--- sharutils-4.6.3.orig/lib/system.h
+++ sharutils-4.6.3/lib/system.h
@@ -24,6 +24,10 @@
 # include <inttypes.h>
 #endif
 
+#if HAVE_STDINT_H
+# include <stdint.h>
+#endif
+
 #ifndef HAVE_INTMAX_T
 #define HAVE_INTMAX_T
 typedef long intmax_t;
