diff -r -U2 fricas-1.1.7.orig/configure.ac fricas-1.1.7/configure.ac
--- fricas-1.1.7.orig/configure.ac	2012-06-19 08:17:01.000000000 +0700
+++ fricas-1.1.7/configure.ac	2012-08-05 11:18:22.000000000 +0700
@@ -243,4 +243,9 @@
 fi
 
+## Work-around for broken sbcl in Gentoo
+if test $FRICAS_LISP = sbcl ; then
+    FRICAS_LISP="$FRICAS_LISP --no-sysinit --no-userinit"
+fi
+
 AC_ARG_WITH([lisp-flavor],
             [AS_HELP_STRING([--with-lisp-flavor=F],
