diff -Nur esorex-3.9.6.orig/configure.ac esorex-3.9.6/configure.ac
--- esorex-3.9.6.orig/configure.ac	2012-08-05 08:13:03.074224198 -0700
+++ esorex-3.9.6/configure.ac	2012-08-05 08:57:19.609538901 -0700
@@ -32,14 +32,14 @@
 
 ESO_CHECK_DOCTOOLS
 
-AC_LIBLTDL_CONVENIENCE
+LT_INIT([dlopen,shared])
+AC_CHECK_HEADER([ltdl.h],
+    [AC_CHECK_LIB([ltdl], [lt_dladvise_init],
+        [LIBLTDL=-lltdl], [LIBLTDL=])],
+    [LIBLTDL=])
+
 AC_SUBST(INCLTDL)
 AC_SUBST(LIBLTDL)
-
-AC_ENABLE_STATIC(no)
-AC_ENABLE_SHARED(yes)
-
-AC_PROG_LIBTOOL
 AC_SUBST(LIBTOOL_DEPS)
 
 # Checks for libraries.
@@ -78,7 +78,6 @@
 AC_LTDL_SHLIBEXT
 
 
-AC_CONFIG_SUBDIRS(libltdl)
 
 AC_CONFIG_FILES(Makefile
                 Makefile.purify
diff -Nur esorex-3.9.6.orig/Makefile.am esorex-3.9.6/Makefile.am
--- esorex-3.9.6.orig/Makefile.am	2012-08-05 08:13:03.077224137 -0700
+++ esorex-3.9.6/Makefile.am	2012-08-05 08:58:00.417757040 -0700
@@ -25,7 +25,6 @@
 DISTCLEANFILES = *~
 
 
-COMPILE_FIRST = libltdl
 
 TOPSUBDIRS = src etc
 
