--- CMakeLists.txt	2011-10-08 04:02:28.000000000 +0200
+++ CMakeLists.txt	2011-10-20 02:18:25.705609820 +0200
@@ -51,7 +51,6 @@
 
 set(IW_SUPPORT_JPEG 0)
 set(IW_SUPPORT_PNG 0)
-set(IW_SUPPORT_WEBP 0)
 
 find_package(JPEG)
 if(JPEG_FOUND)
@@ -101,7 +102,6 @@
 endif()
 
 if(EXISTS "${LIBWEBP_LIBFILE}")
-  set(IW_SUPPORT_WEBP 1)
   set(WEBP_INCLUDE_DIR "${LIBWEBP_DIR}/src")
   set(WEBP_LIBRARIES "${LIBWEBP_LIBFILE}")
   message(STATUS "Found WebP: ${WEBP_LIBRARIES}")
@@ -132,8 +132,8 @@
 target_link_libraries(imagew imageworsener)
 
 if(IW_SUPPORT_WEBP)
-  include_directories(${WEBP_INCLUDE_DIR})
-  target_link_libraries(imagew ${WEBP_LIBRARIES})
+  include_directories(/usr/include/wepb)
+  target_link_libraries(imagew webp)
 endif()
 if(IW_SUPPORT_PNG)
   include_directories(${PNG_INCLUDE_DIR})
