--- tests/runtest	2011-10-08 04:02:28.000000000 +0200
+++ test/runtest	2011-10-20 02:36:57.839706231 +0200
@@ -58,7 +58,7 @@
 	mkdir actual
 fi
 
-rm -f actual/*.png actual/*.jpg actual/*.bmp actual/*.tif actual/*.miff actual/*.webp
+rm -f actual/*.png actual/*.jpg actual/*.bmp actual/*.tif actual/*.miff
 
 echo "Creating images..."
 
@@ -156,10 +156,6 @@
 $IW srcimg/g8a.png actual/miff32.miff -width 11 -depth 32
 $IW srcimg/rgb16.png actual/miff64.miff -width 11 -depth 64
 
-# Test writing WebP
-$IW srcimg/rgb16.png actual/webp1.webp -width 23
-$IW srcimg/g8.png actual/webp2.webp -width 24 -grayscale
-
 # Extra pixel density tests
 $IW srcimg/rgb8x1.png actual/dens-1.png $DCMPR $SCALE -filter mix
 $IW srcimg/rgb8x2.png actual/dens-2.png $DCMPR $SCALE -filter mix
@@ -200,7 +196,7 @@
 
 FAIL=0
 for ff in expected/*.png expected/*.jpg expected/*.bmp expected/*.tif \
- expected/*.miff expected/*.webp
+ expected/*.miff
 do
 	f=${ff#expected/}
 	#echo $f
