Index: src/planner-eds-plugin.c
===================================================================
--- src/planner-eds-plugin.c	(révision 909)
+++ src/planner-eds-plugin.c	(copie de travail)
@@ -55,6 +55,7 @@
 #include <libedataserver/e-source-group.h>
 #include <libedataserver/e-uid.h>
 /* Calendar */
+#define HANDLE_LIBICAL_MEMORY
 #include <libecal/e-cal.h>
 /* Addressbook */
 #include <libebook/e-book.h>
