From 0aad8e2d49567c78f0c0bd43ecc2547a7a9f3e7e Mon Sep 17 00:00:00 2001
From: Romain Perier <mrpouet@gentoo.org>
Date: Tue, 6 Oct 2009 11:41:11 +0200
Subject: [PATCH] Fix compile-warnings automagic support, which caused compilation to fail (-Werror)

---
 configure.ac |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/configure.ac b/configure.ac
index 4d83014..8500f3b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -238,7 +238,7 @@ set_compile_warnings=no
 AC_ARG_ENABLE(compile-warnings,
               [  --enable-compile-warnings
                           Enable verbose compiler warnings],
-              set_compile_warnings=yes)
+              set_compile_warnings=$enableval, set_compile_warnings=yes)
 
 if test "$GCC" = "yes" -a "$set_compile_warnings" != "no"; then
         WARN_CFLAGS="$WARN_CFLAGS -Wcast-align -Wsign-compare \
-- 
1.6.5.rc2

