--- Volume/mixer.py.orig	2006-10-02 17:52:17.000000000 -0600
+++ Volume/mixer.py	2006-10-02 17:51:53.000000000 -0600
@@ -51,7 +51,10 @@
 		id = 0
 		while (channel,id) in ALSA_CHANNELS:
 			id += 1
-		mixer = alsaaudio.Mixer(channel, id, MIXER_DEVICE.value)
+		try:
+			mixer = alsaaudio.Mixer(channel, id, MIXER_DEVICE.value)
+		except alsaaudio.ALSAAudioError:
+			continue
 		if len(mixer.volumecap()):
 			ALSA_CHANNELS.append((channel,id))
 except:
