--- gcal-3.01/doc/Makefile.in.old	2003-10-27 17:09:11.623164752 -0800
+++ gcal-3.01/doc/Makefile.in	2003-10-27 17:10:35.634393104 -0800
@@ -46,7 +46,7 @@
 # Extension (not including `.') for the installed manual page file names.
 manext = 1
 # Where to install the manual pages.
-mandir = @mandir@/man$(manext)
+mandir = @mandir@
 
 #### End of system configuration section. ####
 
@@ -555,7 +555,7 @@
 
 ALLDIRS = \
   $(infodir) \
-  $(mandir)
+  $(mandir)/man$(manext)/
 
 ALLDOCS = \
   $(de) \
@@ -802,7 +802,7 @@
 	$(PRE_INSTALL)
 	@for file in $(manFILES); \
 	do \
-	  rm -f $(mandir)/$$file.$(manext); \
+	  rm -f $(mandir)/man$(manext)/$$file.$(manext); \
 	done
 	$(NORMAL_INSTALL)
 	@case "$(doclang)" in \
@@ -835,8 +835,8 @@
 	      fi; \
 	    fi; \
 	  fi; \
-	  echo "Installing \`$$file.$(manext)' for LANGUAGE=$(doclang) users as \`$(mandir)/$$file.$(manext)'"; \
-	  $(INSTALL_DATA) $$file.$(manext) $(mandir)/$$file.$(manext); \
+	  echo "Installing \`$$file.$(manext)' for LANGUAGE=$(doclang) users as \`$(mandir)/man$(manext)/$$file.$(manext)'"; \
+	  $(INSTALL_DATA) $$file.$(manext) $(mandir)/man$(manext)/$$file.$(manext); \
 	  rm -f $$file.$(manext); \
 	done
 	@echo "LANGUAGE=$(doclang) manual pages installed"
@@ -893,7 +893,7 @@
 	$(NORMAL_UNINSTALL)
 	-for file in $(manFILES); \
 	do \
-	  rm -f $(mandir)/$$file.$(manext); \
+	  rm -f $(mandir)/man$(manext)/$$file.$(manext); \
 	done
 	-rm -f $(infodir)/$(PACKAGE).info*
 	@echo "Documentation files of distribution uninstalled"
