--- gcal-3.01-orig/po/Makefile.in.in	1999-11-08 01:27:46.000000000 +0100
+++ gcal-3.01/po/Makefile.in.in	2009-02-26 10:53:01.000000000 +0100
@@ -101,7 +101,7 @@
 $(srcdir)/$(PACKAGE).pot: $(POTFILES)
 	$(XGETTEXT) --default-domain=$(PACKAGE) --directory=$(top_srcdir) \
 	  --add-comments --keyword=_ --keyword=N_ \
-	  --files-from=$(srcdir)/POTFILES.in \
+	  --files-from=$(srcdir)/POTFILES.in --from-code=ISO-8859-1 \
 	&& test ! -f $(PACKAGE).po \
 	   || ( rm -f $(srcdir)/$(PACKAGE).pot \
 		&& mv $(PACKAGE).po $(srcdir)/$(PACKAGE).pot )
