Index: ROX-Lib2/python/rox/__init__.py
===================================================================
--- ROX-Lib2/python/rox/__init__.py	(revision 4953)
+++ ROX-Lib2/python/rox/__init__.py	(working copy)
@@ -197,8 +197,11 @@
 	try:
 		g.window_set_default_icon_list(g.gdk.pixbuf_new_from_file(_icon_path))
 	except:
-		# Older pygtk
-		_window_icon = g.gdk.pixbuf_new_from_file(_icon_path)
+		try:
+			# Older pygtk
+			_window_icon = g.gdk.pixbuf_new_from_file(_icon_path)
+		except:
+			pass
 del _icon_path
 
 class Window(g.Window):
