Call AC_CONFIG_AUX_DIR before it is used


--- a/configure.ac
+++ b/configure.ac
@@ -2,19 +2,14 @@
     m4_esyscmd([cat .tarball-version]),
     [bkorb@gnu.org])
 
+AC_CONFIG_AUX_DIR([build-aux])
+
 AS_BOX([Configuring AC_PACKAGE_TARNAME AC_PACKAGE_VERSION])
 AC_USE_SYSTEM_EXTENSIONS
 
-dnl we can't use AC_CONFIG_AUX_DIR here, because the heuristics
-dnl for finding install-sh in the generated configure script
-dnl consider this directory to be a subproject of gnulib proper,
-dnl and will only look for install-sh in . and .. :(
-dnl AC_CONFIG_AUX_DIR([build-aux])
-dnl
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_HEADER([config.h])
 AC_CONFIG_FILES([Makefile doc/Makefile lib/Makefile src/Makefile tests/Makefile])
-AC_CONFIG_AUX_DIR([build-aux])
 
 AM_INIT_AUTOMAKE([gnu check-news 1.5 dist-bzip2])
 gl_INIT
