Index: jruby-1.4.0/src/org/jruby/libraries/RbConfigLibrary.java
===================================================================
--- jruby-1.4.0.orig/src/org/jruby/libraries/RbConfigLibrary.java
+++ jruby-1.4.0/src/org/jruby/libraries/RbConfigLibrary.java
@@ -113,7 +113,7 @@ public class RbConfigLibrary implements 
         } else {
             normalizedHome = runtime.getJRubyHome();
         }
-        setConfig(configHash, "bindir", new NormalizedFile(normalizedHome, "bin").getPath());
+        setConfig(configHash, "bindir", "/usr/bin");
         setConfig(configHash, "RUBY_INSTALL_NAME", jrubyScript());
         setConfig(configHash, "ruby_install_name", jrubyScript());
         setConfig(configHash, "SHELL", jrubyShell());
