From 95648e0369e74d65582ee3ef44b0a0a338f82cf0 Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@gentoo.org>
Date: Tue, 12 Mar 2013 12:08:56 -0400
Subject: [PATCH] equery: respect PORTAGE_CONFIGROOT too

Signed-off-by: Mike Frysinger <vapier@gentoo.org>
---
 pym/gentoolkit/package.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pym/gentoolkit/package.py b/pym/gentoolkit/package.py
index 2a103de..da9d65c 100644
--- a/pym/gentoolkit/package.py
+++ b/pym/gentoolkit/package.py
@@ -61,6 +61,7 @@ from gentoolkit.eprefix import EPREFIX
 def _NewPortageConfig(local_config):
 	ret = portage.config(local_config=local_config,
 			eprefix=EPREFIX if EPREFIX else None,
+			config_root=os.environ.get('PORTAGE_CONFIGROOT', None),
 			target_root=os.environ.get('ROOT', None))
 	ret.lock()
 	return ret
-- 
1.8.1.2

