--- sflphone-client-gnome/src/sflnotify.c
+++ sflphone-client-gnome/src/sflnotify.c
@@ -30,6 +30,10 @@
 
 #include <sflnotify.h>
 
+#ifndef NOTIFY_CHECK_VERSION
+#define NOTIFY_CHECK_VERSION(x,y,z) 0
+#endif
+
 GnomeNotification *_gnome_notification;
 
 void create_new_gnome_notification (gchar *title, gchar *body, NotifyUrgency urgency, gint timeout, GnomeNotification **notif)
@@ -43,12 +47,20 @@
         notify_init ("SFLphone");
 
         // Set struct fields
+#if NOTIFY_CHECK_VERSION (0, 7, 0)
+        _notif->notification = notify_notification_new (title, body, NULL);
+#else
         _notif->notification = notify_notification_new (title, body, NULL, NULL);
+#endif
         //_notif->icon = gdk_pixbuf_new_from_file_at_size (LOGO, 120, 120, NULL);
         _notif->icon = gdk_pixbuf_new_from_file (LOGO_SMALL, NULL);
 #if GTK_CHECK_VERSION(2,10,0)
+#if NOTIFY_CHECK_VERSION (0, 7, 0)
+        // notify_notification_attach_to_status_icon function was removed
+#else
         notify_notification_attach_to_status_icon (_notif->notification , get_status_icon() );
 #endif
+#endif
 
         notify_notification_set_urgency (_notif->notification, urgency);
         
