From 3ed300124a425d43ee627ff35b35aa1168ba01d0 Mon Sep 17 00:00:00 2001
From: Tristan Matthews <tristan.matthews@savoirfairelinux.com>
Date: Tue, 6 Mar 2012 10:19:59 -0500
Subject: [PATCH] * #9144: Fixes "Only <glib.h> can be included directly"
 error

---
 plugins/addressbook/evolution/eds.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/plugins/addressbook/evolution/eds.h b/plugins/addressbook/evolution/eds.h
index 0ce1783..5bbf71e 100644
--- a/plugins/addressbook/evolution/eds.h
+++ b/plugins/addressbook/evolution/eds.h
@@ -38,7 +38,7 @@
 #ifndef __EDS_H__
 #define __EDS_H__
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <gdk-pixbuf/gdk-pixbuf.h>
 #include <libebook/e-book.h>
 
-- 
1.7.8.6

