diff -Nur ivtv-1.1.0/utils/ivtvctl.c ivtv-1.1.0-headers/utils/ivtvctl.c
--- ivtv-1.1.0/utils/ivtvctl.c	2008-05-24 08:21:16.000000000 -0400
+++ ivtv-1.1.0-headers/utils/ivtvctl.c	2008-07-03 12:11:38.000000000 -0400
@@ -63,7 +63,7 @@
 
 #define VIDIOC_INT_RESET            	_IOW ('d', 102, __u32)
 
-#include <linux/ivtv.h>
+#include "linux/ivtv.h"
 
 #define __stringify_1(x)	#x
 #define __stringify(x)		__stringify_1(x)
diff -Nur ivtv-1.1.0/utils/ivtvplay.cc ivtv-1.1.0-headers/utils/ivtvplay.cc
--- ivtv-1.1.0/utils/ivtvplay.cc	2008-05-24 08:21:16.000000000 -0400
+++ ivtv-1.1.0-headers/utils/ivtvplay.cc	2008-07-03 12:10:43.000000000 -0400
@@ -41,7 +41,7 @@
 #include <linux/types.h>
 #include <linux/videodev2.h>
 #include <linux/dvb/video.h>
-#include <linux/ivtv.h>
+#include "linux/ivtv.h"
 
 typedef unsigned long W32;
 typedef unsigned long long W64;
diff -Nur ivtv-1.1.0/utils/ivtv-radio.c ivtv-1.1.0-headers/utils/ivtv-radio.c
--- ivtv-1.1.0/utils/ivtv-radio.c	2008-05-24 08:21:16.000000000 -0400
+++ ivtv-1.1.0-headers/utils/ivtv-radio.c	2008-07-03 12:10:08.000000000 -0400
@@ -39,7 +39,7 @@
 
 #include <linux/videodev2.h>
 #include <linux/dvb/video.h>
-#include <linux/ivtv.h>
+#include "linux/ivtv.h"
 
 #define RADIO_DEV "/dev/radio0"
 #define AUDIO_IN_DEV "/dev/video24"
