--- src/makes/plg.mk
+++ src/makes/plg.mk
@@ -4,1 +4,1 @@
-CFLAGS = -g -Wall -fPIC -O4 -fomit-frame-pointer -ffast-math $(INCLUDE)
+CFLAGS = -Wall -fPIC $(OPTFLAGS) $(INCLUDE)
--- src/makes/mk.fpse
+++ src/makes/mk.fpse
@@ -7,1 +7,1 @@
-INCLUDE += -I./fpse `gtk-config --cflags`
+INCLUDE += -I./fpse `gtk-config --cflags` `sdl-config --cflags`
@@ -10,1 +10,1 @@
-LIBS += -lSDL
+LIBS += `sdl-config --libs`
