diff -ur src-orig/draw.c src/draw.c
--- src-orig/draw.c	2006-01-06 03:19:55.000000000 -0500
+++ src/draw.c	2006-01-06 03:20:33.000000000 -0500
@@ -4400,6 +4400,7 @@
 
 #ifdef USE_XF86VM
 
+#undef BOOL
 #include <X11/extensions/xf86vmode.h>
 static XF86VidModeModeInfo **modes=0;
 static int           iOldMode=0;
@@ -5705,7 +5706,13 @@
 #else
      pDstR=(unsigned long *)(pDst-(ddx+dga2Fix));
 #endif
-     for(x=0;x<ddx2;x++) *((unsigned long*)pDst)++=*pDstR++;
+     for(x=0;x<ddx2;x++)
+      {
+       *((unsigned long*)pDst) = *pDstR;
+	pDst += sizeof (unsigned long);
+	pDstR += sizeof (unsigned long);
+      }
+
     }
    else
     {
Only in src-orig/: draw.c.orig
diff -ur src-orig/gpu.h src/gpu.h
--- src-orig/gpu.h	2006-01-06 03:19:55.000000000 -0500
+++ src/gpu.h	2006-01-06 03:19:33.000000000 -0500
@@ -63,4 +63,4 @@
 
 /////////////////////////////////////////////////////////////////////////////
 
-#endif // _GPU_INTERNALS_H
\ No newline at end of file
+#endif // _GPU_INTERNALS_H
diff -ur src-orig/menu.h src/menu.h
--- src-orig/menu.h	2006-01-06 03:19:55.000000000 -0500
+++ src/menu.h	2006-01-06 03:19:33.000000000 -0500
@@ -33,4 +33,4 @@
 void BuildDispMenu(int iInc);
 void SwitchDispMenu(int iStep);
 
-#endif // _GPU_MENU_H_
\ No newline at end of file
+#endif // _GPU_MENU_H_
diff -ur src-orig/prim.c src/prim.c
--- src-orig/prim.c	2006-01-06 03:19:55.000000000 -0500
+++ src/prim.c	2006-01-06 03:19:33.000000000 -0500
@@ -269,10 +269,12 @@
 
 // x -466 1023 180 1023
 // y   20 -228 222 -100
-
+
+/*
 // 0 __1
 //  \ / \
 //   2___3
+*/
 
 __inline BOOL CheckCoord4()
 {
diff -ur src-orig/prim.h src/prim.h
--- src-orig/prim.h	2006-01-06 03:19:55.000000000 -0500
+++ src/prim.h	2006-01-06 03:19:33.000000000 -0500
@@ -30,4 +30,4 @@
 void UploadScreen (long Position);
 void PrepareFullScreenUpload (long Position);
 
-#endif // _PRIMDRAW_H_
\ No newline at end of file
+#endif // _PRIMDRAW_H_
diff -ur src-orig/psemu.h src/psemu.h
--- src-orig/psemu.h	2006-01-06 03:19:55.000000000 -0500
+++ src/psemu.h	2006-01-06 03:19:33.000000000 -0500
@@ -33,4 +33,4 @@
 #define PSE_LT_SPU	4
 
 
-#endif // _PSEMU_PLUGIN_DEFS_H
\ No newline at end of file
+#endif // _PSEMU_PLUGIN_DEFS_H
