diff --git a/phoenix.pro b/phoenix.pro
index 8232c54..241eb61 100644
--- a/phoenix.pro
+++ b/phoenix.pro
@@ -115,7 +115,6 @@ RC_FILE = fritzing.rc
 RESOURCES += phoenixresources.qrc
 	include(pri/kitchensink.pri)
 	include(pri/mainwindow.pri)
-	include(pri/quazip.pri)
 	include(pri/partsbinpalette.pri)
 	include(pri/partseditor.pri)
 	include(pri/referencemodel.pri)
diff --git a/pri/utils.pri b/pri/utils.pri
index 94f18c1..ab0de4f 100644
--- a/pri/utils.pri
+++ b/pri/utils.pri
@@ -68,4 +68,4 @@ src/utils/textutils.cpp \
 src/utils/zoomslider.cpp 
 
 
-
+LIBS += -lquazip
diff --git a/src/autoroute/panelizer.cpp b/src/autoroute/panelizer.cpp
index 38db81a..9bdeb2f 100644
--- a/src/autoroute/panelizer.cpp
+++ b/src/autoroute/panelizer.cpp
@@ -25,22 +25,22 @@ $Date: 2013/01/28 22:58:53 $
 ********************************************************************/
 
 #include "panelizer.h"
-#include "../../debugdialog.h"
-#include "../../sketch/pcbsketchwidget.h"
-#include "../../utils/textutils.h"
-#include "../../utils/graphicsutils.h"
-#include "../../utils/folderutils.h"
-#include "../../utils/folderutils.h"
-#include "../../items/resizableboard.h"
-#include "../../items/logoitem.h"
-#include "../../items/groundplane.h"
-#include "../../fsvgrenderer.h"
-#include "../../fapplication.h"
-#include "../../svg/gerbergenerator.h"
-#include "../../referencemodel/referencemodel.h"
-#include "../../version/version.h"
-#include "../../processeventblocker.h"
-#include "../../connectors/connectoritem.h"
+#include "../debugdialog.h"
+#include "../sketch/pcbsketchwidget.h"
+#include "../utils/textutils.h"
+#include "../utils/graphicsutils.h"
+#include "../utils/folderutils.h"
+#include "../utils/folderutils.h"
+#include "../items/resizableboard.h"
+#include "../items/logoitem.h"
+#include "../items/groundplane.h"
+#include "../fsvgrenderer.h"
+#include "../fapplication.h"
+#include "../svg/gerbergenerator.h"
+#include "../referencemodel/referencemodel.h"
+#include "../version/version.h"
+#include "../processeventblocker.h"
+#include "../connectors/connectoritem.h"
 
 #include "cmrouter/tileutils.h"
 
diff --git a/src/autoroute/panelizer.h b/src/autoroute/panelizer.h
index a21dbf2..f6f36a4 100644
--- a/src/autoroute/panelizer.h
+++ b/src/autoroute/panelizer.h
@@ -30,8 +30,8 @@ $Date: 2013/01/28 22:58:53 $
 #include <QString>
 #include <QSizeF>
 
-#include "../../mainwindow/mainwindow.h"
-#include "../../items/itembase.h"
+#include "../mainwindow/mainwindow.h"
+#include "../items/itembase.h"
 #include "cmrouter/tile.h"
 
 
diff --git a/src/utils/folderutils.cpp b/src/utils/folderutils.cpp
index bba0b9e..a84da8f 100644
--- a/src/utils/folderutils.cpp
+++ b/src/utils/folderutils.cpp
@@ -34,8 +34,8 @@ $Date: 2013/01/28 22:58:53 $
 #include <QCryptographicHash>
 
 #include "../debugdialog.h"
-#include "../lib/quazip/quazip.h"
-#include "../lib/quazip/quazipfile.h"
+#include <quazip/quazip.h>
+#include <quazip/quazipfile.h>
 #include "../lib/qtsysteminfo/QtSystemInfo.h"
 
 
@@ -552,4 +552,4 @@ void FolderUtils::copyBin(const QString & dest, const QString & source) {
 #endif
 
     );
-}
\ No newline at end of file
+}
