--- Makefile
+++ Makefile
@@ -10,15 +10,15 @@
 all:
 	@echo "There is nothing to compile."
 
-install: uninstall correctpath
+install: install-doc
 	@echo Installing fvwm-crystal $(RELEASE) to $(prefix)
-	mkdir -p $(prefix)/bin $(prefix)/share/fvwm-crystal/fvwm 
+	mkdir -p $(DESTDIR)$(prefix)/bin $(DESTDIR)$(prefix)/share/fvwm-crystal/fvwm 
 
-	install -m 755 bin/fvwm-crystal.apps bin/fvwm-crystal.wallpaper bin/fvwm-crystal bin/fvwm-crystal.generate-menu bin/fvwm-crystal.infoline bin/fvwm-crystal.mplayer-wrapper $(prefix)/bin
-	cp -d -r fvwm/* $(prefix)/share/fvwm-crystal/fvwm/
+	install -m 755 bin/fvwm-crystal.apps bin/fvwm-crystal.wallpaper bin/fvwm-crystal bin/fvwm-crystal.generate-menu bin/fvwm-crystal.infoline bin/fvwm-crystal.mplayer-wrapper $(DESTDIR)$(prefix)/bin
+	cp -d -r fvwm/* $(DESTDIR)$(prefix)/share/fvwm-crystal/fvwm/
 
-	mkdir -p $(prefix)/share/man/man1
-	cp -d -r man/* $(prefix)/share/man/man1
+	mkdir -p $(DESTDIR)$(prefix)/share/man/man1
+	cp -d -r man/* $(DESTDIR)$(prefix)/share/man/man1
 
 uninstall:
 	@echo "Uninstalling previously installed fvwm-crystal"
@@ -188,17 +188,12 @@
 	tar czf "../fvwm-crystal-minimal-$(RELEASE).tar.gz" "fvwm-crystal-minimal-$(RELEASE)"
 	rm -rf "fvwm-crystal-minimal-$(RELEASE)"
 
-# It is needed to adjust some path inside fvwm-crystal.generate-menu since this file must know the install path
-correctpath:
-	sed -i 's:FC_MENUBASEROOT="/usr/share:FC_MENUBASEROOT="$(prefix)/share:' bin/fvwm-crystal.generate-menu
-	sed -i 's:FC_ICONBASEROOT="/usr/share:FC_ICONBASEROOT="$(prefix)/share:' bin/fvwm-crystal.generate-menu
-
 uninstall-doc:
 	-rm -rf $(docdir)
 	-rm -rf $(addondir)
 
 install-doc:
-	mkdir -p $(addondir) $(docdir)
-	install -m 644 addons/* $(addondir)
-	install -m 644 AUTHORS COPYING NEWS ChangeLog README INSTALL $(docdir)
-	install -m 644 doc/* $(docdir)
+	mkdir -p $(DESTDIR)$(addondir) $(DESTDIR)$(docdir)
+	install -m 644 addons/* $(DESTDIR)$(addondir)
+	install -m 644 AUTHORS NEWS ChangeLog README $(DESTDIR)$(docdir)
+	install -m 644 doc/* $(DESTDIR)$(docdir)
