diff -Nru pam_radius-1.3.17.orig/Makefile pam_radius-1.3.17/Makefile
--- pam_radius-1.3.17.orig/Makefile	2008-06-28 08:43:43.000000000 +0200
+++ pam_radius-1.3.17/Makefile	2008-06-28 08:42:57.000000000 +0200
@@ -8,6 +8,7 @@
 
 VERSION=1.3.17
 
+LD=$(CC)
 ######################################################################
 #
 # If we're really paranoid, use these flags
@@ -15,7 +16,7 @@
 #
 #  If you're not using GCC, then you'll have to change the CFLAGS.
 #
-CFLAGS = -Wall -fPIC
+CFLAGS += -Wall -fPIC -fno-strict-aliasing
 #
 # On Irix, use this with MIPSPRo C Compiler, and don't forget to export CC=cc
 # gcc on Irix does not work yet for pam_radius
@@ -55,7 +56,7 @@
 #	gcc -shared pam_radius_auth.o md5.o -lpam -lc -o pam_radius_auth.so
 #
 pam_radius_auth.so: pam_radius_auth.o md5.o
-	ld -Bshareable pam_radius_auth.o md5.o -lpam -o pam_radius_auth.so
+	$(LD) $(LDFLAGS) -shared $(CFLAGS) pam_radius_auth.o md5.o -lpam -o pam_radius_auth.so
 
 ######################################################################
 #
diff -Nru pam_radius-1.3.17.orig/pam_radius_auth.c pam_radius-1.3.17/pam_radius_auth.c
--- pam_radius-1.3.17.orig/pam_radius_auth.c	2007-03-26 11:36:13.000000000 +0200
+++ pam_radius-1.3.17/pam_radius_auth.c	2008-06-28 08:43:24.000000000 +0200
@@ -1062,7 +1062,7 @@
 pam_sm_authenticate(pam_handle_t *pamh,int flags,int argc,CONST char **argv)
 {
   CONST char *user;
-  CONST char **userinfo;
+  CONST char *userinfo;
   char *password = NULL;
   CONST char *rhost;
   char *resp2challenge = NULL;
diff -Nru pam_radius-1.3.17.orig/USAGE pam_radius-1.3.17/USAGE
--- pam_radius-1.3.17.orig/USAGE	2008-06-28 08:43:43.000000000 +0200
+++ pam_radius-1.3.17/USAGE	2008-06-28 08:42:57.000000000 +0200
@@ -3,9 +3,9 @@
 
   The pam configuration can be:
 ...
-auth       sufficient   /lib/security/pam_radius_auth.so [options]
+auth       sufficient   pam_radius_auth.so [options]
 ...
-account    sufficient   /lib/security/pam_radius_auth.so
+account    sufficient   pam_radius_auth.so
 
 ---------------------------------------------------------------------------
 
