--- configure.in	2012-08-30 16:29:25.000000000 +0800
+++ configure.in.new	2013-01-30 14:48:16.504664907 +0800
@@ -31,8 +31,8 @@
 AC_PROG_CC
 if test "$GCC" = yes
 then
-  CFLAGS="${CFLAGS} ${EXTRA_GCFLAGS}-pipe -Wall -Wshadow -Wbad-function-cast -Wmissing-prototypes -Wstrict-prototypes -Wcast-align -Wcast-qual -Wpointer-arith -Wwrite-strings -Wmissing-declarations -Wnested-externs -Wundef -pedantic -fno-common"
-  LDFLAGS="${LDFLAGS} ${EXTRA_GLDFLAGS}-g"
+  CFLAGS="${CFLAGS} ${EXTRA_GCFLAGS} -Wall -Wshadow -Wbad-function-cast -Wmissing-prototypes -Wstrict-prototypes -Wcast-align -Wcast-qual -Wpointer-arith -Wwrite-strings -Wmissing-declarations -Wnested-externs -Wundef -pedantic -fno-common"
+  LDFLAGS="${LDFLAGS} ${EXTRA_GLDFLAGS}"
 else
   CFLAGS="${CFLAGS} ${EXTRA_CFLAGS}"
   LDFLAGS="${LDFLAGS} ${EXTRA_LDFLAGS}"
@@ -66,7 +66,6 @@
 fi
 
 AC_CHECK_HEADERS(tgmath.h,have_tgmath_h=yes)
-AC_CHECK_FUNCS(lrint)
 AC_CHECK_FUNCS(nanosleep)
 
 AC_CHECK_FUNCS(tgetent,have_tgetent=yes)
