--- code/qcommon/files.c
+++ code/qcommon/files.c
@@ -31,7 +31,7 @@
 
 #include "q_shared.h"
 #include "qcommon.h"
-#include "unzip.h"
+#include <minizip/unzip.h>
 
 extern cvar_t *com_quiet;
 
--- Makefile
+++ Makefile
@@ -1001,7 +1001,7 @@
 else
   LIBS += -lz
 endif
-
+LIBS += -lz -lminizip
 ifdef DEFAULT_BASEDIR
   BASE_CFLAGS += -DDEFAULT_BASEDIR=\\\"$(DEFAULT_BASEDIR)\\\"
 endif
@@ -1498,8 +1498,6 @@
   $(B)/client/q_math.o \
   $(B)/client/q_shared.o \
   \
-  $(B)/client/unzip.o \
-  $(B)/client/ioapi.o \
   $(B)/client/puff.o \
   $(B)/client/vm.o \
   $(B)/client/vm_interpreted.o \
@@ -1790,8 +1778,6 @@
   $(B)/ded/q_math.o \
   $(B)/ded/q_shared.o \
   \
-  $(B)/ded/unzip.o \
-  $(B)/ded/ioapi.o \
   $(B)/ded/vm.o \
   $(B)/ded/vm_interpreted.o \
   \
