applied on top of ioq3-1807-urt-251210-git-nobumpy.patch

--- Makefile
+++ Makefile
@@ -242,7 +242,7 @@
     CURL_LIBS=$(shell pkg-config --silence-errors --libs libcurl)
     OPENAL_CFLAGS=$(shell pkg-config --silence-errors --cflags openal)
     OPENAL_LIBS=$(shell pkg-config --silence-errors --libs openal)
-    SDL_CFLAGS=$(shell pkg-config --silence-errors --cflags sdl|sed 's/-Dmain=SDL_main//')
+    SDL_CFLAGS=$(shell pkg-config --silence-errors --cflags freetype2 sdl|sed 's/-Dmain=SDL_main//')
     SDL_LIBS=$(shell pkg-config --silence-errors --libs sdl)
   endif
   # Use sdl-config if all else fails
@@ -305,9 +305,9 @@
   endif
 
   BASE_CFLAGS = -Wall -fno-strict-aliasing -Wimplicit -Wstrict-prototypes \
-    -pipe -DUSE_ICON
+    -DUSE_ICON
   CLIENT_CFLAGS = $(SDL_CFLAGS)
-  SERVER_CFLAGS =
+  SERVER_CFLAGS = $(shell pkg-config --silence-errors --cflags sdl|sed 's/-Dmain=SDL_main//')
 
   ifeq ($(USE_OPENAL),1)
     CLIENT_CFLAGS += -DUSE_OPENAL
