diff -u openmcu_v2_2_0.orig/main.cxx openmcu_v2_2_0/main.cxx
--- openmcu_v2_2_0.orig/main.cxx	2006-09-14 02:05:07.000000000 +0200
+++ openmcu_v2_2_0/main.cxx	2006-09-14 02:05:39.000000000 +0200
@@ -54,9 +54,9 @@
 #include <ptlib.h>
 #include "mcu.h"
 
-static const char DefaultConnectingWAVFile[] = "connecting.wav";
-static const char DefaultEnteringWAVFile[]   = "entering.wav";
-static const char DefaultLeavingWAVFile[]    = "leaving.wav";
+static const char DefaultConnectingWAVFile[] = "/usr/share/openmcu/sounds/connecting.wav";
+static const char DefaultEnteringWAVFile[]   = "/usr/share/openmcu/sounds/entering.wav";
+static const char DefaultLeavingWAVFile[]    = "/usr/share/openmcu/sounds/leaving.wav";
 
 static const char ConnectingWAVFileKey[]  = "Connecting WAV File";
 static const char EnteringWAVFileKey[]    = "Entering WAV File";
Only in openmcu_v2_2_0.orig: main.cxx.orig
diff -u openmcu_v2_2_0.orig/mcu.cxx openmcu_v2_2_0/mcu.cxx
--- openmcu_v2_2_0.orig/mcu.cxx	2006-09-14 02:06:59.000000000 +0200
+++ openmcu_v2_2_0/mcu.cxx	2006-09-14 02:09:25.000000000 +0200
@@ -80,7 +80,7 @@
 static const char DefaultRoomKey[]          = "Default room";
 static const char DefaultRoomTimeLimitKey[] = "Room time limit";
 
-static const char DefaultCallLogFilename[] = "mcu_log.txt"; 
+static const char DefaultCallLogFilename[] = "/var/log/openmcu/mcu_log.txt"; 
 static const char DefaultRoom[]            = "room101";
 
 #if OPENMCU_VIDEO
@@ -151,8 +151,8 @@
 #endif
   exeDir.Change();
 
-  httpNameSpace.AddResource(new PHTTPDirectory("data", "data"));
-  httpNameSpace.AddResource(new PServiceHTTPDirectory("html", "html"));
+  httpNameSpace.AddResource(new PHTTPDirectory("data", "/usr/share/openmcu/data"));
+  httpNameSpace.AddResource(new PServiceHTTPDirectory("html", "/usr/share/openmcu/html"));
 
   manager  = CreateConferenceManager();
   endpoint = CreateEndPoint(*manager);
@@ -226,7 +226,7 @@
 
 #if P_SSL
   // SSL certificate file.
-  PString certificateFile = cfg.GetString(HTTPCertificateFileKey, "server.pem");
+  PString certificateFile = cfg.GetString(HTTPCertificateFileKey, "/etc/openmcu/server.pem");
   rsrc->Add(new PHTTPStringField(HTTPCertificateFileKey, 25, certificateFile));
   if (!SetServerCertificate(certificateFile, TRUE)) {
     PSYSTEMLOG(Fatal, "BMAC\tCould not load certificate \"" << certificateFile << '"');
