--- a/configure.in
+++ b/configure.in
@@ -144,10 +144,10 @@
 AC_CHECK_HEADERS(unistd.h stdarg.h)
 
 dnl Checks for typedefs, structures, and compiler characteristics.
 
 dnl Checks for library functions.
-AC_CHECK_FUNCS(getopt getopt_long setsid setpgid setpgrp putenv vsnprintf usleep)
+AC_CHECK_FUNCS(getline getopt getopt_long setsid setpgid setpgrp putenv vsnprintf usleep)
 
 AC_TYPE_SIGNAL
 
 AC_OUTPUT(Makefile doc/Makefile src/Makefile contrib/Makefile)
