diff --exclude='*~' -Naur etherape-0.9.3.orig/configure.in etherape-0.9.3/configure.in
--- etherape-0.9.3.orig/configure.in	2005-08-19 20:06:22.000000000 -0300
+++ etherape-0.9.3/configure.in	2005-08-19 20:07:57.000000000 -0300
@@ -79,7 +79,8 @@
 
 AC_CHECK_FUNC(res_mkquery, , 
   AC_CHECK_LIB(bind, res_mkquery, , 
-   AC_CHECK_LIB(resolv, res_mkquery, , [AC_MSG_ERROR([No resolver library found])] )))
+   AC_CHECK_LIB(resolv, res_mkquery, ,
+    AC_CHECK_LIB(resolv, __res_mkquery, , [AC_MSG_ERROR([No resolver library found])] ))))
 else
 AC_CHECK_FUNC(gethostbyaddr_r, , 
   AC_CHECK_LIB(bind, gethostbyaddr_r, , 
