Index: libxmlpatch-0.3.1/diff/Makefile.am
===================================================================
--- libxmlpatch-0.3.1.orig/diff/Makefile.am
+++ libxmlpatch-0.3.1/diff/Makefile.am
@@ -29,6 +29,7 @@ INCLUDES = $(DEPS_XML_CFLAGS) $(DEPS_GLI
 lib_LTLIBRARIES = libxmldiff.la
 
 libxmldiff_la_SOURCES = xml_diff.c 
+libxmldiff_la_LIBADD = $(DEPS_GLIB_LIBS) 
 
 libxmldiff_includedir = $(includedir)/libxmldiff
 libxmldiff_include_HEADERS = xml_diff.h
