From 04d18bfe6b82b07c68553b2da7958f8f53208610 Mon Sep 17 00:00:00 2001
From: Jiří Techet <techet@gmail.com>
Date: Sun, 29 Sep 2013 15:11:52 +0000
Subject: Make sure Maemo is disabled when --disable-maemo is used

---
diff --git a/configure.ac b/configure.ac
index 7981a01..4f3e69c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -125,7 +125,7 @@ fi
 
 AC_ARG_ENABLE(maemo,
   AS_HELP_STRING([--enable-maemo],[compile with maemo optimizations]),
-    [enable_maemo=yes])
+    enable_maemo=$enableval, enable_maemo=no)
 
 # -----------------------------------------------------------
 # Enable gtk
--
cgit v0.9.2
