diff -Nur statsvn-0.4.1/src/net/sf/statsvn/Main.java statsvn-0.4.1_patched/src/net/sf/statsvn/Main.java
--- statsvn-0.4.1/src/net/sf/statsvn/Main.java	2008-09-19 07:19:16.000000000 +0300
+++ statsvn-0.4.1_patched/src/net/sf/statsvn/Main.java	2008-09-19 07:19:42.000000000 +0300
@@ -295,7 +295,7 @@
 		System.setProperty("java.awt.headless", "true");
 
 		final ReportConfig config = new ReportConfig(content, ConfigurationOptions.getProjectName(), ConfigurationOptions.getOutputDir(), ConfigurationOptions
-		        .getMarkupSyntax(), ConfigurationOptions.getCssHandler(), ConfigurationOptions.getCharSet());
+		        .getMarkupSyntax(), ConfigurationOptions.getCssHandler());
 		config.setWebRepository(ConfigurationOptions.getWebRepository());
 		config.setWebBugtracker(ConfigurationOptions.getWebBugtracker());
 		config.setNonDeveloperLogins(ConfigurationOptions.getNonDeveloperLogins());
