diff --git a/lib/bindery/html.py b/lib/bindery/html.py
index 0d1b953..d838982 100644
--- a/lib/bindery/html.py
+++ b/lib/bindery/html.py
@@ -13,7 +13,7 @@ import itertools
 from functools import *
 from itertools import *
 
-from amara.thirdparty import html5lib
+import html5lib
 from amara import tree
 from amara.lib import inputsource
 from amara.bindery import nodes
diff --git a/setup.py b/setup.py
index b6d0094..d3f8440 100755
--- a/setup.py
+++ b/setup.py
@@ -282,11 +282,6 @@ setup(name='Amara',
                 'amara.xslt.tree',
                 'amara.xslt.xpatterns',
                 'amara.xupdate',
-                'amara.thirdparty',
-                'amara.thirdparty.httplib2',
-                'amara.thirdparty.html5lib',
-                'amara.thirdparty.html5lib.treebuilders',
-                # Test support; would be possible to break out into separate
                 # package at some point.
                 'amara.test',
                 'amara.test.xpath',
-- 
1.7.8.5

