diff --git a/lib/src/expat/expat.c b/lib/src/expat/expat.c
index 4a9bf65..8e8e0a5 100644
--- a/lib/src/expat/expat.c
+++ b/lib/src/expat/expat.c
@@ -64,7 +64,7 @@ Expat wrapper library";
 #define XmlString_SHARED
 #include "xmlstring.h"
 #include "cStringIO.h"
-#include "lib/expat.h"          /* Expat library */
+#include <expat.h>          /* Expat library */
 #include "xmlchar.h"            /* XML_Char to PyUnicode support */
 #include "stack.h"              /* Stack_* routines */
 #include "hash_table.h"         /* XML_Char-keyed HashTable */
diff --git a/lib/src/expat/xmlchar.h b/lib/src/expat/xmlchar.h
index c1378b0..be4f3ac 100644
--- a/lib/src/expat/xmlchar.h
+++ b/lib/src/expat/xmlchar.h
@@ -6,7 +6,7 @@ extern "C" {
 #endif
 
 #include "Python.h"
-#include "lib/expat_external.h"
+#include <expat_external.h>
 
 #ifndef XML_UNICODE
 #error ExpatReader requires a Unicode-enabled Expat
diff --git a/setup.py b/setup.py
index 6a0e650..f3856e3 100755
--- a/setup.py
+++ b/setup.py
@@ -303,12 +303,11 @@ setup(name='Amara',
           Extension('amara._expat',
                     define_macros=[('HAVE_EXPAT_CONFIG_H', None),
                                    ('Expat_BUILDING_MODULE', None),
+                                   ('XML_UNICODE_WCHAR_T', None),
                                    ],
                     include_dirs=['lib/src', 'lib/src/expat'],
+                    libraries=['expatw'],
                     sources=[# Expat XML parser
-                             'lib/src/expat/lib/xmlparse.c',
-                             'lib/src/expat/lib/xmlrole.c',
-                             'lib/src/expat/lib/xmltok.c',
                              # Miscellaneous supporting routines
                              'lib/src/expat/util.c',
                              # XML_Char <-> PyUnicode
