diff -Nru pptp-1.7.1.orig/pptp.c pptp-1.7.1/pptp.c
--- pptp-1.7.1.orig/pptp.c	2006-02-13 05:07:42.000000000 +0200
+++ pptp-1.7.1/pptp.c	2007-03-18 09:23:16.000000000 +0200
@@ -43,11 +43,7 @@
 #include "pptp_callmgr.h"
 #include "pptp_gre.h"
 #include "version.h"
-#if defined(__linux__)
-#include <linux/prctl.h>
-#else
 #include "inststr.h"
-#endif
 #include "util.h"
 #include "pptp_quirks.h"
 #include "pqueue.h"
@@ -355,10 +351,6 @@
 
     snprintf(buf, sizeof(buf), "pptp: GRE-to-PPP gateway on %s", 
             ttyname(tty_fd));
-#ifdef PR_SET_NAME
-    rc = prctl(PR_SET_NAME, "pptpgw", 0, 0, 0);
-    if (rc != 0) perror("prctl");
-#endif
     inststr(argc, argv, envp, buf);
     if (sigsetjmp(env, 1)!= 0) goto shutdown;
 
