--- configure.in.old	2008-09-16 23:42:41.600170750 +0200
+++ configure.in	2008-09-16 23:46:35.563175050 +0200
@@ -32,7 +32,9 @@
 
 # Find Python compiler and linker flags
 PYTHON_LIBS=[$($PYTHON -c "from distutils import sysconfig
-print '/'.join(sysconfig.get_config_vars('LIBPL','LDLIBRARY'))+' '+sysconfig.get_config_var('LIBS')+' '+sysconfig.get_config_var('LINKFORSHARED')")]
+import os
+print os.path.join(*sysconfig.get_config_vars('LIBDIR', 'LDLIBRARYDIR', 'LDLIBRARY')),
+sysconfig.get_config_var('LIBS'), sysconfig.get_config_var('LINKFORSHARED')")]
 PYTHON_CFLAGS=[$($PYTHON -c "from distutils import sysconfig
 print sysconfig.get_python_inc()")]
 PYTHON_CFLAGS="-I$PYTHON_CFLAGS"
