diff -ru vo-aacenc-0.1.1.orig/aacenc/src/asm/ARMV7/PrePostMDCT_v7.s vo-aacenc-0.1.1/aacenc/src/asm/ARMV7/PrePostMDCT_v7.s
--- vo-aacenc-0.1.1.orig/aacenc/src/asm/ARMV7/PrePostMDCT_v7.s	2011-06-18 17:45:24.000000000 +0200
+++ vo-aacenc-0.1.1/aacenc/src/asm/ARMV7/PrePostMDCT_v7.s	2011-06-18 17:45:42.000000000 +0200
@@ -21,6 +21,7 @@
 @
 @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
 
+	.fpu neon
 	.section .text
 	.global	PreMDCT
 
@@ -132,4 +133,4 @@
 	ldmia     sp!, {r4 - r11, pc}
 
 	@ENDP  		@ |PostMDCT|
-	.end
\ No newline at end of file
+	.end
diff -ru vo-aacenc-0.1.1.orig/aacenc/src/asm/ARMV7/R4R8First_v7.s vo-aacenc-0.1.1/aacenc/src/asm/ARMV7/R4R8First_v7.s
--- vo-aacenc-0.1.1.orig/aacenc/src/asm/ARMV7/R4R8First_v7.s	2011-06-18 17:45:24.000000000 +0200
+++ vo-aacenc-0.1.1/aacenc/src/asm/ARMV7/R4R8First_v7.s	2011-06-18 17:45:57.000000000 +0200
@@ -21,6 +21,7 @@
 @
 @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
 
+	.fpu neon
 	.section .text
 	.global	Radix8First
 
@@ -143,4 +144,4 @@
 	ldmia    		sp!, {r4 - r11, pc}
 
 	@ENDP  @ |Radix4First|
-	.end
\ No newline at end of file
+	.end
diff -ru vo-aacenc-0.1.1.orig/aacenc/src/asm/ARMV7/Radix4FFT_v7.s vo-aacenc-0.1.1/aacenc/src/asm/ARMV7/Radix4FFT_v7.s
--- vo-aacenc-0.1.1.orig/aacenc/src/asm/ARMV7/Radix4FFT_v7.s	2011-06-18 17:45:24.000000000 +0200
+++ vo-aacenc-0.1.1/aacenc/src/asm/ARMV7/Radix4FFT_v7.s	2011-06-18 17:45:49.000000000 +0200
@@ -21,6 +21,7 @@
 @
 @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
 
+	.fpu neon
 	.section .text
 	.global	Radix4FFT
 
@@ -140,4 +141,4 @@
 	ldmia   			sp!, {r4 - r11, pc}
 
 	@ENDP  @ |Radix4FFT|
-	.end
\ No newline at end of file
+	.end

