--- bin/radiant.orig	2010-08-11 13:46:51.021730420 +0200
+++ bin/radiant	2010-08-11 13:47:11.678230437 +0200
@@ -4,7 +4,7 @@
 
 $LOAD_PATH << File.join(File.dirname(__FILE__), '..')
 
-require 'lib/ruby_version_check'
+require 'ruby_version_check'
 Signal.trap("INT") { puts; exit }
 
 require 'lib/radiant'
@@ -15,7 +15,7 @@
 
 app_path = ARGV.first
 
-require 'lib/rails_generator'
+require 'rails_generator'
 require 'rails_generator/scripts/generate'
 
 class Rails::Generator::Base
--- lib/tasks/database.rake.~1~	2010-08-14 09:59:40.322355364 +0200
+++ lib/tasks/database.rake	2010-08-14 10:01:28.405760795 +0200
@@ -28,4 +28,9 @@
       :database_template => ENV['DATABASE_TEMPLATE']
     )
   end
-end
\ No newline at end of file
+
+  desc "Migrate schema to latest version"
+  task :migrate do
+    ActiveRecord::Migrator.migrate("#{RADIANT_ROOT}/db/migrate/")
+  end
+end
