--- db/migrate/20081203140407_add_indexes.rb.orig	2010-06-01 09:34:51.588558179 +0200
+++ db/migrate/20081203140407_add_indexes.rb	2010-06-01 09:35:27.728808367 +0200
@@ -1,11 +1,11 @@
 class AddIndexes < ActiveRecord::Migration
   def self.up
-    add_index :pages,       :class_name,            :name => :pages_class_name
-    add_index :pages,       :parent_id,             :name => :pages_parent_id
-    add_index :pages,       %w{slug parent_id},     :name => :pages_child_slug
-    add_index :pages,       %w{virtual status_id},  :name => :pages_published
+    add_index :pages,       :class_name,            :name => 'pages_class_name'
+    add_index :pages,       :parent_id,             :name => 'pages_parent_id'
+    add_index :pages,       %w{slug parent_id},     :name => 'pages_child_slug'
+    add_index :pages,       %w{virtual status_id},  :name => 'pages_published'
     
-    add_index :page_parts,  %w{page_id name},       :name => :parts_by_page
+    add_index :page_parts,  %w{page_id name},       :name => 'parts_by_page'
   end
 
   def self.down
