--- colubrid/application.py	2006-08-30 06:03:31.000000000 -0700
+++ colubrid/application.py	2008-03-09 04:37:59.204620929 -0700
@@ -354,6 +354,8 @@
             return RoutesRequest(self, e, s, c)
         super(RoutesApplication, self).__init__(environ, start_response,
                                                 create_request)
+
+    def process_request(self):
         path = self.request.environ.get('PATH_INFO') or '/'
         match = self._routes_mapper.match(path)
         if match is None:
@@ -367,5 +369,4 @@
             del match['action']
         del match['controller']
 
-        # XXX: can't return from __init__
         return handler(app, **match)
