--- src/output/formatstring.cc
+++ src/output/formatstring.cc
@@ -240,9 +240,8 @@
 		if(some_overlay_used)
 			*some_overlay_used = true;
 		if(overlay_translations) {
-			overlay = (*overlay_translations)[index];
-			if(overlay != 0) {
-				number = 0;
+			number = (*overlay_translations)[index];
+			if(number == 0) {
 				for(vector<ExtendedVersion::Overlay>::iterator it(overlay_translations->begin());
 					likely(it != overlay_translations->end()); ++it) {
 					if(number < *it)
