diff -ru freenx-server_orig/Makefile freenx-server/Makefile
--- freenx-server_orig/Makefile	2009-03-29 14:07:27.000000000 +0200
+++ freenx-server/Makefile	2010-05-09 09:18:36.000000000 +0200
@@ -6,7 +6,7 @@
 PROGRAMS=nxacl.sample nxcheckload.sample nxcups-gethost nxdesktop_helper nxdialog nxkeygen nxloadconfig nxnode nxnode-login nxprint nxserver nxserver-helper/nxserver-helper nxsetup nxviewer_helper nxviewer-passwd/nxpasswd/nxpasswd nx-session-launcher/nx-session-launcher nx-session-launcher/nx-session-launcher-suid nxserver-usermode nxserver-suid/nxserver-suid
 
 all:
-	cd nxviewer-passwd && xmkmf && make Makefiles && make depend
+	cd nxviewer-passwd && xmkmf && $(MAKE) Makefiles && $(MAKE) depend
 	source nxloadconfig &&\
 	export PATH_BIN PATH_LIB CUPS_BACKEND NX_VERSION NX_ETC_DIR &&\
 	for i in $(SUBDIRS) ; \
