--- ./nxnode-login	2009-04-20 16:54:24.000000000 +0200
+++ /usr/bin/nxnode-login	2009-04-20 16:54:27.000000000 +0200
@@ -73,6 +73,7 @@
 	expect {
 		"Are you sure you want to continue connecting (yes/no)?" { send "yes\r" }
 		"assword*:"  { sleep 0.3; send "$password\r" }
+		"SSH passphrase:"  { sleep 0.3; send "$password\r" }
 		"Permission denied*" { exit 1 }
 		"su: Authentication failure" { exit 1 }
 		"NX> 1000 NXNODE - Version" { 
