diff -Naur freenx-0.7.1.orig/node.conf.sample freenx-0.7.1/node.conf.sample
--- freenx-0.7.1.orig/node.conf.sample	2007-10-31 13:47:45.000000000 +0100
+++ freenx-0.7.1/node.conf.sample	2007-10-31 13:48:35.000000000 +0100
@@ -319,12 +319,12 @@
 # of available ppd drivers via the $COMMAND_FOOMATIC command.
 
 #ENABLE_FOOMATIC="1"
-#COMMAND_FOOMATIC="/usr/lib/cups/driver/foomatic-ppdfile"
+#COMMAND_FOOMATIC="/usr/bin/foomatic-ppdfile"
 
 # CUPS_BACKEND and CUPS_ETC are the corresponding paths of your CUPS 
 # installation.
 
-#CUPS_BACKEND="/usr/lib/cups/backend"
+#CUPS_BACKEND="/usr/libexec/cups/backend"
 #CUPS_IPP_BACKEND="$CUPS_BACKEND/nxipp"
 #CUPS_DEFAULT_SOCK="/var/run/cups/cups.sock"
 #CUPS_ETC="/etc/cups"
diff -Naur freenx-0.7.1.orig/nxloadconfig freenx-0.7.1/nxloadconfig
--- freenx-0.7.1.orig/nxloadconfig	2007-10-31 13:47:45.000000000 +0100
+++ freenx-0.7.1/nxloadconfig	2007-10-31 13:48:43.000000000 +0100
@@ -155,9 +155,9 @@
 ENABLE_CUPS_SEAMLESS="0"
 CUPS_SEAMLESS_DELAY="10"
 ENABLE_FOOMATIC="1"
-COMMAND_FOOMATIC="/usr/lib/cups/driver/foomatic-ppdfile"
+COMMAND_FOOMATIC="/usr/bin/foomatic-ppdfile"
 
-CUPS_BACKEND="/usr/lib/cups/backend"
+CUPS_BACKEND="/usr/libexec/cups/backend"
 CUPS_IPP_BACKEND="$CUPS_BACKEND/nxipp"
 CUPS_DEFAULT_SOCK="/var/run/cups/cups.sock"
 CUPS_ETC="/etc/cups/"
