--- a/etc/initsystem/prepare-dirs
+++ b/etc/initsystem/prepare-dirs
@@ -31,3 +31,7 @@
 mkdir -p $ICINGA2_RUN_DIR/icinga2/cmd
 chown $ICINGA2_USER:$ICINGA2_COMMAND_GROUP $ICINGA2_RUN_DIR/icinga2/cmd
 chmod 2750 $ICINGA2_RUN_DIR/icinga2/cmd
+
+mkdir -p $ICINGA2_STATE_DIR/cache/icinga2
+chown $ICINGA2_USER:$ICINGA2_COMMAND_GROUP $ICINGA2_STATE_DIR/cache/icinga2
+chmod 750 $ICINGA2_STATE_DIR/cache/icinga2
