diff -ur libecap-0.0.2.orig/src/libecap/common/name.cc libecap-0.0.2.new/src/libecap/common/name.cc
--- libecap-0.0.2.orig/src/libecap/common/name.cc	2008-05-14 04:18:00.000000000 +0600
+++ libecap-0.0.2.new/src/libecap/common/name.cc	2009-07-04 19:07:51.086046555 +0600
@@ -1,6 +1,6 @@
 #include <libecap/common/name.h>
 #include <libecap/common/errors.h>
-#include <limits>
+#include <limits.h>
 
 libecap::Name::Id libecap::Name::TheLastId = libecap::Name::nameUnidentified;
 
