Index: Macaulay2/emacs/M2.el
===================================================================
--- Macaulay2/emacs/M2.el	(revision 13991)
+++ Macaulay2/emacs/M2.el	(revision 14000)
@@ -102,7 +102,7 @@
 (mapcar
  (function
   (lambda (syntax-table)
-    (modify-syntax-entry ?\\ "\\" syntax-table)
+    (modify-syntax-entry ?\\ "." syntax-table)
     (modify-syntax-entry ?-  ". 12" syntax-table)
     (modify-syntax-entry ?\n ">" syntax-table)
     (modify-syntax-entry ?\^m ">" syntax-table)
