diff -urN OpenCPN-2.5.0-Source.old/CMakeLists.txt OpenCPN-2.5.0-Source/CMakeLists.txt
--- OpenCPN-2.5.0-Source.old/CMakeLists.txt	2011-07-31 16:21:15.000000000 +0200
+++ OpenCPN-2.5.0-Source/CMakeLists.txt	2011-07-31 16:22:09.000000000 +0200
@@ -49,7 +49,7 @@
 IF (NOT WIN32)
   # default
   SET (ARCH "i386")
-  SET (LIB_INSTALL_DIR "lib")
+  SET (LIB_INSTALL_DIR "lib${LIB_SUFFIX}")
   IF (EXISTS /etc/debian_version)
     SET (PACKAGE_FORMAT "DEB")
     SET (PACKAGE_DEPS "libwxgtk2.8-0 (>= 2.8.7.1), libglu1-mesa (>= 7.0.0), libgl1-mesa-glx (>= 7.0.0), zlib1g, bzip2")
