diff -ur ripperX-2.7.2.orig/plugins/Makefile.in ripperX-2.7.2/plugins/Makefile.in
--- ripperX-2.7.2.orig/plugins/Makefile.in	2008-02-18 07:38:28.000000000 +0200
+++ ripperX-2.7.2/plugins/Makefile.in	2008-03-02 07:16:23.000000000 +0200
@@ -615,7 +615,7 @@
 all: ripperX_plugin_tester $(PLUGINS)
 
 .c.o:
-	$(CC) $(CFLAGS) $(LDFLAGS) -c $< -o $@
+	$(CC) $(CFLAGS) -c $< -o $@
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
