diff -ur cmucl-19c-release-20051115.orig.orig/src/code/save.lisp cmucl-19c-release-20051115.orig/src/code/save.lisp
--- cmucl-19c-release-20051115.orig.orig/src/code/save.lisp	2005-12-07 12:27:38.000000000 -0600
+++ cmucl-19c-release-20051115.orig/src/code/save.lisp	2005-12-07 12:29:30.000000000 -0600
@@ -297,11 +297,9 @@
 	))
 
 (setf (getf *herald-items* :bugs)
-      '("For support see http://www.cons.org/cmucl/support.html Send bug reports to the debian BTS."
-        terpri
-	"or to "
-	"pvaneynd@debian.org" terpri
-       "type (help) for help, (quit) to exit, and (demo) to see the demos" terpri
+      '("For support see http://www.cons.org/cmucl/support.html" terpri
+	"Send bug reports to http://bugs.gentoo.org" terpri
+	"Type (help) for help or (quit) to exit." terpri
         terpri
 	"Loaded subsystems:"))
 
