--- pypy-pypy-release-1.7/pypy/translator/c/gcc/trackgcroot.py.bak	2011-11-29 14:22:29.000000000 +0100
+++ pypy-pypy-release-1.7/pypy/translator/c/gcc/trackgcroot.py	2011-11-29 14:22:55.000000000 +0100
@@ -1694,6 +1694,7 @@
             }
             """
         elif self.format in ('elf64', 'darwin64'):
+            print >> output, "\t.section .note.GNU-stack,\"\",%progbits"
             print >> output, "\t.text"
             print >> output, "\t.globl %s" % _globalname('pypy_asm_stackwalk')
             _variant(elf64='.type pypy_asm_stackwalk, @function',
