diff -Naur libcilkrts.orig/Makefile.am libcilkrts/Makefile.am
--- libcilkrts.orig/Makefile.am	2013-01-13 21:32:38.477690223 -0700
+++ libcilkrts/Makefile.am	2013-01-13 21:33:08.697795224 -0700
@@ -38,12 +38,12 @@
 
 # Compiler and linker flags.
 GENERAL_FLAGS = -I$(top_srcdir)/include -DIN_CILK_RUNTIME=1
-GENERAL_FLAGS += -fpic -O3
+GENERAL_FLAGS += -fpic
 GENERAL_FLAGS += -DBUILD_USER=\"$(USER)\"
 GENERAL_FLAGS += -DBUILD_HOST=\"`hostname`\"
 
 # Enable Intel Cilk Plus extension
-GENERAL_FLAGS += -fcilkplus
+#GENERAL_FLAGS += -fcilkplus
 
 AM_CFLAGS = $(GENERAL_FLAGS) -std=c99
 AM_CPPFLAGS = $(GENERAL_FLAGS)
