From 38aa888cc88bb27930818ec8dc55033a072cbfd5 Mon Sep 17 00:00:00 2001
From: Daiki Ueno <ueno@gnu.org>
Date: Wed, 16 Oct 2013 07:34:48 +0000
Subject: autopoint: use 'm4' as the fallback macro directory

---
diff --git a/gettext-tools/misc/autopoint.in b/gettext-tools/misc/autopoint.in
index 72b268b..d8c7d74 100644
--- a/gettext-tools/misc/autopoint.in
+++ b/gettext-tools/misc/autopoint.in
@@ -347,6 +347,10 @@ if test -z "$m4dir" && test -f Makefile.am; then
   done
 fi
 
+if test -z "$m4dir"; then
+  m4dir=m4
+fi
+
 # Check whether to omit the intl/ directory.
 omitintl=
 xargs=`func_trace AM_GNU_GETTEXT "$configure_in"`
--
cgit v0.9.0.2
