From 75cb1897a9eabd6acf280cade2ecffb1554380db Mon Sep 17 00:00:00 2001
From: Ritesh Raj Sarraf <rrs@researchut.com>
Date: Mon, 5 Nov 2012 04:07:26 +0530
Subject: [PATCH 01/20] Do not ship the board-specific/ folder in default
 installs

---
 install.sh | 1 -
 1 file changed, 1 deletion(-)

diff --git a/install.sh b/install.sh
index 322e784..536a63a 100755
--- a/install.sh
+++ b/install.sh
@@ -112,7 +112,6 @@ $INSTALL -d -m 755 "$DESTDIR/usr/share/laptop-mode-tools/modules"
 $INSTALL -d -m 755 "$DESTDIR/usr/share/laptop-mode-tools/module-helpers"
 $INSTALL -d -m 755 "$DESTDIR/usr/lib/tmpfiles.d"
 $INSTALL -d -m 755 "$DESTDIR/etc/laptop-mode/conf.d"
-$INSTALL -d -m 755 "$DESTDIR/etc/laptop-mode/conf.d/board-specific"
 $INSTALL -d -m 755 "$DESTDIR/etc/laptop-mode/modules"
 $INSTALL -d -m 755 "$DESTDIR/usr/sbin"
 $INSTALL -d -m 755 "$DESTDIR/lib/udev"
-- 
1.8.1.5

