commit e41f89d507ec92339bf8751e1f5fd72233ff299d
Author: Sergey Popov <admin@pinkbyte.ru>
Date:   Thu Jan 10 15:08:00 2013 +0400

    Fix distribution detection - return string instead of functions

diff --git a/lib/ansible/module_common.py b/lib/ansible/module_common.py
index 3db8439..882e5cd 100644
--- a/lib/ansible/module_common.py
+++ b/lib/ansible/module_common.py
@@ -96,10 +96,10 @@ def get_distribution():
     ''' return the distribution name '''
     if platform.system() == 'Linux':
         try:
-            distribution = platform.linux_distribution()[0].capitalize
+            distribution = platform.linux_distribution()[0].capitalize()
         except:
             # FIXME: MethodMissing, I assume?
-            distribution = platform.dist()[0].capitalize
+            distribution = platform.dist()[0].capitalize()
     else:
         distribution = None
     return distribution
