diff -urN spindown-0.4.0.orig/Makefile spindown-0.4.0/Makefile
--- spindown-0.4.0.orig/Makefile	2009-04-18 18:32:59.000000000 +0200
+++ spindown-0.4.0/Makefile	2009-06-09 19:52:47.931991569 +0200
@@ -4,7 +4,7 @@
 OBJS = main.o diskset.o disk.o spindown.o iniparser.o dictionary.o log.o spindownd.o\
         exceptions.o
 CC = g++
-CFLAGS =-O1
+CFLAGS +=
 SRC = src/
 INPARSER = $(SRC)ininiparser3.0b/
 SRCDIR = spindown-$(VERSION)
