From 75d5cdaa4be0b9b25da13655db77e2caebdbf570 Mon Sep 17 00:00:00 2001
From: Nick Schermer <nick@xfce.org>
Date: Fri, 27 Jan 2012 20:31:12 +0000
Subject: Show dialog in settings category.

---
diff --git a/xfce4-notifyd-config/xfce4-notifyd-config.desktop.in b/xfce4-notifyd-config/xfce4-notifyd-config.desktop.in
index 51d4827..9fa3ac2 100644
--- a/xfce4-notifyd-config/xfce4-notifyd-config.desktop.in
+++ b/xfce4-notifyd-config/xfce4-notifyd-config.desktop.in
@@ -8,5 +8,5 @@ Terminal=false
 StartupNotify=true
 Type=Application
 OnlyShowIn=XFCE;
-Categories=X-XFCE;Settings;DesktopSettings;
+Categories=XFCE;GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-PersonalSettings;
 X-XfcePluggable=true
--
cgit v0.9.0.3
