diff -ur mccp4-0.7.1.orig/ccp4_sysdep.h mccp4-0.7.1/ccp4_sysdep.h
--- mccp4-0.7.1.orig/ccp4_sysdep.h	2008-03-12 22:39:16.000000000 -0700
+++ mccp4-0.7.1/ccp4_sysdep.h	2008-03-12 22:40:31.000000000 -0700
@@ -199,7 +199,7 @@
 #  define NATIVEIT DFNTI_IBO
 #endif
 
-#if defined(MIPSEL) || defined(i386) || defined(i860) || defined(__ia64__)
+#if defined(MIPSEL) || defined(i386) || defined(i860) || defined(__ia64__) || defined(__x86_64__)
 #  define NATIVEIT DFNTI_IBO
 #  define NATIVEFT DFNTF_LEIEEE
 #endif
