--- Makefile.orig	2006-03-08 20:36:02.000000000 +0100
+++ Makefile	2009-08-26 00:34:02.000000000 +0200
@@ -2,9 +2,7 @@
 # $Id: fwlogwatch-1.1-make.patch,v 1.1 2009/08/25 22:37:06 jer Exp $
 
 # Linux
-CC = gcc
-CFLAGS = -DHAVE_ZLIB -DHAVE_GETTEXT -pipe -O2 -Wall #-pedantic -Wpointer-arith #-g #-p
-LDFLAGS = #-g #-static -p
+CFLAGS += -DHAVE_ZLIB -DHAVE_GETTEXT
 LIBS = -lcrypt -lz #-ladns #-lc_p
 
 # Solaris
@@ -39,7 +37,7 @@
 LFLAGS = -B #-f #-p -p -d
 
 INSTALL = install
-INSTALL_PROGRAM = $(INSTALL) -s -m 0755
+INSTALL_PROGRAM = $(INSTALL) -m 0755
 INSTALL_SCRIPT = $(INSTALL) -m 0755
 INSTALL_DATA = $(INSTALL) -m 0644
 INSTALL_DIR = /usr/local
