--- makesnortsam.sh.orig	2005-01-25 07:49:43.432290588 -0500
+++ makesnortsam.sh	2005-01-25 07:51:53.722343371 -0500
@@ -11,15 +11,13 @@
 #        Under Solaris, the OPSEC stuff is linked dynamically.
 #        On other platforms, statically.
 
-
+source ${ROOT}sbin/functions.sh
 systype=`uname`
 
 case "$1" in
 	[oO][pP][sS][eE][cC])
 
-		echo "-------------------------------------------------------------------------------"
-		echo "Building SnortSam (release) with OPSEC libraries"
-		echo "-------------------------------------------------------------------------------"
+		einfo "Building SnortSam (release) with OPSEC libraries"
 
 		rm -f snortsam
 		rm -f snortsam-debug
@@ -58,9 +56,7 @@
 		esac
 
 
-		echo "-------------------------------------------------------------------------------"
-		echo "Building SnortSam (debug) with OPSEC libraries"
-		echo "-------------------------------------------------------------------------------"
+		einfo "Building SnortSam (debug) with OPSEC libraries"
 
 		rm -f *.o
 
@@ -97,9 +93,7 @@
 		cd ..
 		;;
 	*)
-		echo "-------------------------------------------------------------------------------"
-		echo "Building SnortSam (release)"
-		echo "-------------------------------------------------------------------------------"
+		einfo "Building SnortSam (release)"
 
 		rm -f snortsam
 		rm -f snortsam-debug
@@ -138,9 +132,7 @@
 		esac
 
 
-		echo "-------------------------------------------------------------------------------"
-		echo "Building SnortSam (debug)"
-		echo "-------------------------------------------------------------------------------"
+		einfo "Building SnortSam (debug)"
 
 		rm -f *.o
 
