--- ./dll/aim/toc/interface.c
+++ ./dll/aim/toc/interface.c
@@ -8,6 +8,7 @@
 
 #include <stdlib.h>
 #include <stdio.h>
+#include <string.h>
 #include <time.h>
 #include "toc.h"
 
--- ./dll/possum/llist.c
+++ ./dll/possum/llist.c
@@ -1,4 +1,5 @@
 #include <stdlib.h>
+#include <string.h>
 #include "llist.h"
 
 llist *lmake(size_t size) {
--- ./dll/europa/europa.c
+++ ./dll/europa/europa.c
@@ -5,6 +5,8 @@
   the file COPYING for details.
  */
 
+#include <string.h>
+
 #define MOD_VERSION "0.01"
 #define MOD_NAME "Europa"
 #include "europa.h"
--- ./dll/arcfour/md5.c
+++ ./dll/arcfour/md5.c
@@ -1,5 +1,6 @@
 /* MD5 Message Digest algorithm */
 
+#include <string.h>
 #include "md5.h"
 
 /* Prototypes */
--- contrib/vh1.c
+++ contrib/vh1.c
@@ -30,6 +30,8 @@
 #include <netinet/in.h>
 #if !defined(linux)
 #include <netinet/in_var.h>
+#else
+#include <arpa/inet.h>
 #endif
 #include <netdb.h>
 
