Only in BitchX-work: .config.h
diff -ur BitchX/include/irc_std.h BitchX-work/include/irc_std.h
--- BitchX/include/irc_std.h	2003-04-10 18:09:07.000000000 -0700
+++ BitchX-work/include/irc_std.h	2007-09-23 11:33:40.000000000 -0700
@@ -98,7 +98,7 @@
 # define __N
 # define __inline__
 #else
-# if (__GNUC__ >= 2) && (__GNUC_MINOR__ >= 7)
+# if (__GNUC__ > 2) || ((__GNUC__ == 2) && (__GNUC_MINOR__ >= 7))
 #  define __A(x) __attribute__ ((format (printf, x, x + 1)))
 #  define __N    __attribute__ ((noreturn))
 # else
Only in BitchX/include: module.h.orig
Only in BitchX/include: modval.h.orig
