Index: vdr-1.5.13/Makefile
===================================================================
--- vdr-1.5.13.orig/Makefile
+++ vdr-1.5.13/Makefile
@@ -20,6 +20,7 @@ PREFIX  ?= /usr/local
 MANDIR   = $(PREFIX)/share/man
 BINDIR   = $(PREFIX)/bin
 LOCDIR   = ./locale
+INCDIR   = $(PREFIX)/include
 LIBS     = -ljpeg -lpthread -ldl -lcap -lrt -lfreetype -lfontconfig
 INCLUDES = -I/usr/include/freetype2
 
@@ -161,7 +162,7 @@ clean-plugins:
 
 # Install the files:
 
-install: install-bin install-conf install-doc install-plugins install-i18n
+install: install-bin install-conf install-doc install-plugins install-i18n install-header
 
 # VDR binary:
 
@@ -192,6 +193,13 @@ install-plugins: plugins
 	@mkdir -p $(DESTDIR)$(PLUGINLIBDIR)
 	@cp --remove-destination $(PLUGINDIR)/lib/lib*-*.so.$(APIVERSION) $(DESTDIR)$(PLUGINLIBDIR)
 
+# Header files:
+
+install-header:
+	@mkdir -p $(DESTDIR)$(INCDIR)/vdr $(DESTDIR)$(INCDIR)/vdr/libsi
+	@cp *.h Make.config $(DESTDIR)$(INCDIR)/vdr
+	@cp libsi/*.h $(DESTDIR)$(INCDIR)/vdr/libsi
+
 # Source documentation:
 
 srcdoc:
