--- comm-1.9.1/mozilla/config/system-headers
+++ comm-1.9.1/mozilla/config/system-headers
@@ -200,7 +200,6 @@
 ctype.h
 curl/curl.h
 curl/easy.h
-curl/types.h
 curses.h
 cxxabi.h
 DateTimeUtils.h
--- comm-1.9.1/mozilla/js/src/config/system-headers
+++ comm-1.9.1/mozilla/js/src/config/system-headers
@@ -200,7 +200,6 @@
 ctype.h
 curl/curl.h
 curl/easy.h
-curl/types.h
 curses.h
 cxxabi.h
 DateTimeUtils.h
--- comm-1.9.1/mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/http_upload.cc
+++ comm-1.9.1/mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/http_upload.cc
@@ -31,7 +31,6 @@
 #include <dlfcn.h>
 #include <curl/curl.h>
 #include <curl/easy.h>
-#include <curl/types.h>
 
 #include "common/linux/http_upload.h"
 
