--- comm-release/mozilla/config/system-headers
+++ comm-release/mozilla/config/system-headers
@@ -207,7 +207,6 @@
 ctype.h
 curl/curl.h
 curl/easy.h
-curl/types.h
 curses.h
 cxxabi.h
 DateTimeUtils.h
--- comm-release/mozilla/js/src/config/system-headers
+++ comm-release/mozilla/js/src/config/system-headers
@@ -207,7 +207,6 @@
 ctype.h
 curl/curl.h
 curl/easy.h
-curl/types.h
 curses.h
 cxxabi.h
 DateTimeUtils.h
--- comm-release/mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/http_upload.cc
+++ comm-release/mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/http_upload.cc
@@ -33,7 +33,6 @@
 #include <dlfcn.h>
 #include <curl/curl.h>
 #include <curl/easy.h>
-#include <curl/types.h>
 
 namespace {
 
--- comm-release/mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/libcurl_wrapper.cc
+++ comm-release/mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/libcurl_wrapper.cc
@@ -29,7 +29,6 @@
 
 #include <curl/curl.h>
 #include <curl/easy.h>
-#include <curl/types.h>
 #include <dlfcn.h>
 
 #include <iostream>
