From ca6778ba2a6c4fed24d5468213f8ed918f3dcbe4 Mon Sep 17 00:00:00 2001
From: Mike Gorse <mgorse@suse.com>
Date: Fri, 7 Dec 2012 11:21:19 -0600
Subject: [PATCH 3/3] Fix deregistering of applications

---
 registryd/registry.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/registryd/registry.c b/registryd/registry.c
index a64e094..e283a90 100644
--- a/registryd/registry.c
+++ b/registryd/registry.c
@@ -290,9 +290,10 @@ handle_disconnection (DBusConnection *bus, DBusMessage *message, void *user_data
           for (i = 0; i < reg->apps->len; i++)
             {
               SpiReference *ref  = g_ptr_array_index (reg->apps, i);
-              while (!g_strcmp0 (old, ref->name))
+              if (!g_strcmp0 (old, ref->name))
                 {
                   remove_application (reg, bus, i);
+                  i--;
                 }
             } 
 
-- 
1.8.0

