--- libinklevel/Makefile.orig	2006-10-04 19:18:54.000000000 +0200
+++ libinklevel/Makefile	2006-10-04 19:20:18.000000000 +0200
@@ -1,4 +1,5 @@
 DESTDIR = /usr/local/
+LIBDIR = $(DESTDIR)/lib
 CC = gcc
 CFLAGS = -Wall -O2 -fPIC -DPIC -I.
 
@@ -7,12 +8,12 @@
 install: all
 	mkdir -p $(DESTDIR)/include
 	install -m 644 -o root -g root inklevel.h $(DESTDIR)/include/inklevel.h
-	mkdir -p $(DESTDIR)/lib
+	mkdir -p $(LIBDIR) 
 	install -m 755 -o root -g root libinklevel.so.2.0.6.5 \
-	$(DESTDIR)/lib/libinklevel.so.2.0.6.5
-	cd $(DESTDIR)/lib && rm -fr libinklevel.so && \
+	$(LIBDIR)/libinklevel.so.2.0.6.5
+	cd $(LIBDIR) && rm -fr libinklevel.so && \
 	ln -s libinklevel.so.2.0.6.5 libinklevel.so
-	ldconfig $(DESTDIR)/lib
+	ldconfig $(LIBDIR)
 
 clean:
 	rm -fr *.o *.so.2.0.6.5 *~
