diff -u libinklevel-0.7.1.orig/Makefile libinklevel-0.7.1/Makefile
--- libinklevel-0.7.1.orig/Makefile	2007-05-01 13:23:54.000000000 +0200
+++ libinklevel-0.7.1/Makefile	2007-05-01 13:25:37.000000000 +0200
@@ -1,4 +1,5 @@
 DESTDIR = /usr/local/
+LIBDIR = lib
 CFLAGS = -Wall -O2 -fPIC -DPIC -I.
 
 all: libinklevel.so.4.0.7.1
@@ -6,14 +7,14 @@
 install: all
 	mkdir -p $(DESTDIR)/include
 	install -m 644 -o root -g root inklevel.h $(DESTDIR)/include/inklevel.h
-	mkdir -p $(DESTDIR)/lib
+	mkdir -p $(DESTDIR)/${LIBDIR}
 	install -m 755 -o root -g root libinklevel.so.4.0.7.1 \
-	$(DESTDIR)/lib/libinklevel.so.4.0.7.1
-	cd $(DESTDIR)/lib && rm -fr libinklevel.so && \
+	$(DESTDIR)/${LIBDIR}/libinklevel.so.4.0.7.1
+	cd $(DESTDIR)/${LIBDIR} && rm -fr libinklevel.so && \
 	ln -s libinklevel.so.4.0.7.1 libinklevel.so
-	cd $(DESTDIR)/lib && rm -fr libinklevel.so.4 && \
+	cd $(DESTDIR)/${LIBDIR} && rm -fr libinklevel.so.4 && \
 	ln -s libinklevel.so.4.0.7.1 libinklevel.so.4
-	/sbin/ldconfig $(DESTDIR)/lib
+	/sbin/ldconfig $(DESTDIR)/${LIBDIR}
 
 clean:
 	rm -fr *.o *.so.4.0.7.1 *~
