diff -Naur libirman-0.4.2.orig/Makefile.in libirman-0.4.2/Makefile.in
--- libirman-0.4.2.orig/Makefile.in	2004-09-24 02:58:09.312603000 -0700
+++ libirman-0.4.2/Makefile.in	2004-09-24 02:59:22.620612916 -0700
@@ -112,22 +112,22 @@
 	@echo notice: the test programs are only installed in the current directory \(`pwd`\)
 
 install-dirs:
-	$(top_srcdir)/mkinstalldirs $(libdir)
-	$(top_srcdir)/mkinstalldirs $(bindir)
-	$(top_srcdir)/mkinstalldirs $(sbindir)
-	$(top_srcdir)/mkinstalldirs $(sysconfdir)
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)$(libdir)
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)$(bindir)
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)$(sbindir)
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)$(sysconfdir)
 
 install-normal:
-	$(INSTALL) -m 755 workmanir $(bindir)/workmanir
-	$(INSTALL) -m 755 libirman.a $(libdir)/libirman.a
-	$(INSTALL) -m 755 libirman.so $(libdir)/libirman.so
-	$(INSTALL) -m 644 irman.conf $(sysconfdir)/irman.conf
-	$(INSTALL) -m 644 irman.h $(includedir)/irman.h
+	$(INSTALL) -m 755 workmanir $(DESTDIR)$(bindir)/workmanir
+	$(INSTALL) -m 755 libirman.a $(DESTDIR)$(libdir)/libirman.a
+	$(INSTALL) -m 755 libirman.so $(DESTDIR)$(libdir)/libirman.so
+	$(INSTALL) -m 644 irman.conf $(DESTDIR)$(sysconfdir)/irman.conf
+	$(INSTALL) -m 644 irman.h $(DESTDIR)$(includedir)/irman.h
 
 install-swtest:
-	$(INSTALL) -m 755 workmanir $(bindir)/workmanir
-	$(INSTALL) -m 755 libirman_sw.a $(libdir)/libirman_sw.a
-	$(INSTALL) -m 755 libirman_sw.so $(libdir)/libirman_sw.so
+	$(INSTALL) -m 755 workmanir $(DESTDIR)$(bindir)/workmanir
+	$(INSTALL) -m 755 libirman_sw.a $(DESTDIR)$(libdir)/libirman_sw.a
+	$(INSTALL) -m 755 libirman_sw.so $(DESTDIR)$(libdir)/libirman_sw.so
 
 uninstall:
 	$(RM) $(bindir)/workmanir
