diff -Naur libirman-0.4.2.orig/Makefile.in libirman-0.4.2/Makefile.in
--- libirman-0.4.2.orig/Makefile.in	1999-03-28 15:42:41.000000000 -0800
+++ libirman-0.4.2/Makefile.in	2004-07-04 23:39:43.101677411 -0700
@@ -16,7 +16,7 @@
 CC = @CC@
 TAR = tar
 
-CFLAGS = @CFLAGS@
+CFLAGS = @CFLAGS@ -fPIC -DPIC
 CPPFLAGS = -I. @CPPFLAGS@ @DEFS@
 LDFLAGS = @LDFLAGS@ -L. -lirman
 LDFLAGS_SW = @LDFLAGS@ -L. -lirman_sw
@@ -58,9 +58,9 @@
 both: standard swtest
 
 
-libirman: libirman.a
+libirman: libirman.a libirman.so
 
-libirman_sw: libirman_sw.a
+libirman_sw: libirman_sw.a libirman_sw.so
 
 
 test_io: libirman.a $(IO_OBJS)
@@ -85,10 +85,16 @@
 	$(AR) $(ARFLAGS) libirman.a $(IROBJS)
 	$(RANLIB) libirman.a
 
+libirman.so: $(IROBJS)
+	$(CC) -shared -Wl,-shared,-soname,libirman.so -o libirman.so $(IROBJS)
+
 libirman_sw.a: $(IROBJS_SW)
 	$(AR) $(ARFLAGS) libirman_sw.a $(IROBJS_SW)
 	$(RANLIB) libirman_sw.a
 
+libirman_sw.so: $(IROBJS_SW)
+	$(CC) -shared -Wl,-shared,-soname,libirman_sw.so -o libirman_sw.so $(IROBJS_SW)
+
 %.o: %.c irman.h config.h
 	$(CC) $(CPPFLAGS) $(CFLAGS) -c $< -o $@
 
@@ -96,7 +102,7 @@
 	$(CC) $(CPPFLAGS) $(CFLAGS) -c $< -o $@
 
 ircmd.o: ircmd.c irman.h config.h
-	$(CC) $(CPPFLAGS) -DIR_SYSCONF_DIR=\"$(sysconfdir)\" -c $(srcdir)/ircmd.c -o ircmd.o
+	$(CC) $(CFLAGS) $(CPPFLAGS) -DIR_SYSCONF_DIR=\"$(sysconfdir)\" -c $(srcdir)/ircmd.c -o ircmd.o
 irfunc_sw.o: irfunc.c irman.h
 	$(CC) $(CPPFLAGS) -DIR_SOFTWARE_TEST $(CFLAGS) -c $(srcdir)/irfunc.c -o irfunc_sw.o
 
@@ -114,18 +120,22 @@
 install-normal:
 	$(INSTALL) -m 755 workmanir $(bindir)/workmanir
 	$(INSTALL) -m 755 libirman.a $(libdir)/libirman.a
+	$(INSTALL) -m 755 libirman.so $(libdir)/libirman.so
 	$(INSTALL) -m 644 irman.conf $(sysconfdir)/irman.conf
 	$(INSTALL) -m 644 irman.h $(includedir)/irman.h
 
 install-swtest:
 	$(INSTALL) -m 755 workmanir $(bindir)/workmanir
 	$(INSTALL) -m 755 libirman_sw.a $(libdir)/libirman_sw.a
+	$(INSTALL) -m 755 libirman_sw.so $(libdir)/libirman_sw.so
 
 uninstall:
 	$(RM) $(bindir)/workmanir
 	$(RM) $(bindir)/workmanir_sw
 	$(RM) $(libdir)/libirman.a
+	$(RM) $(libdir)/libirman.so
 	$(RM) $(libdir)/libirman_sw.a
+	$(RM) $(libdir)/libirman_sw.so
 	$(RM) $(sysconfdir)/irman.conf
 
 dist:
