Index: libirman-0.4.2/workmanir.c
===================================================================
--- libirman-0.4.2.orig/workmanir.c
+++ libirman-0.4.2/workmanir.c
@@ -148,7 +148,7 @@ int main(int argc, char **argv)
 
   errno = 0;  
   if (ir_init(filename) < 0) {
-    fprintf(stderr, "%s: error initialising Irman: `%s'\n", ir_strerror(errno));
+    fprintf(stderr, "%s: error initialising Irman: `%s'\n", progname, ir_strerror(errno));
     exit(1);
   }
 
