diff -uNr ./modules/filter/email.cpp ./modules/filter/email.cpp
--- ./modules/filter/email.cpp	2004-04-24 12:39:56.371216800 +0300
+++ ./modules/filter/email.cpp	2004-04-24 12:40:05.521825696 +0300
@@ -33,7 +33,8 @@
         memset(data, 0, sizeof(bool)*256);
 	return no_err;
       }
-      bool have(char c) {
+      bool have(FilterChar::Chr c) {
+	if (c > 255) return false;
         return data[static_cast<unsigned char>(c)];
       }
       QuoteChars() {clear();}
