diff -Naur soft-flower-gdm.old/GdmGreeterTheme.desktop soft-flower-gdm/GdmGreeterTheme.desktop
--- soft-flower-gdm.old/GdmGreeterTheme.desktop	2008-11-23 17:30:03.000000000 +0100
+++ soft-flower-gdm/GdmGreeterTheme.desktop	2008-11-23 17:30:27.000000000 +0100
@@ -2,15 +2,12 @@
 Version=1.0
 Encoding=UTF-8
 Name=No name
-Name[en_AU]=GdmGreeterTheme.desktop
+Name[en_US]=GdmGreeterTheme.desktop
 
 [GdmGreeterTheme]
 Greeter=soft-flower.xml
 Name=Soft Flower Gnome
 Description=Simple Gnome GDM - based on the Simple Ubuntu Blue gdm theme by mandha
-http://www.gnome-look.org/content/show.php?content=29943
-icons and xml based on Ubuntu World GDM by Stemp.
-http://gnome-look.org/content/show.php?content=24719
-Author= blueshome
-Copyright= GPL
+Author=blueshome
+Copyright=GPL
 Screenshot=screenshot.png
