Index: plan-1.9/src/Makefile
===================================================================
--- plan-1.9.orig/src/Makefile
+++ plan-1.9/src/Makefile
@@ -142,7 +142,7 @@ FBSD_C	= -ansi -I/usr/X11R6/include -DBS
 	  -DSTDLIBMALLOC
 OBSD_C	= -ansi -I/usr/X11R6/include -DBSD -DRABBITS -DNOLOCK -DNOREGEX\
 	  -DNO_MOTIF -DSTDLIBMALLOC
-LINUX_C	= -W -Wall -I/usr/X11R6/include -DNOLOCK -DLINUX
+LINUX_C	= ${LDFLAGS} -I/usr/X11R6/include -DNOLOCK -DLINUX
 SVR4_C	= -DSVR4 -DSYSV -DOLDTZ -DUSERAND -DPID_T=long -DNOMSEP
 NEWS_C	= -DNEWSOS4 -DNOMSEP -DSUN -DBSD -DRABBITS -DNGROUPS_MAX=16\
 	  -DPID_T="int" -Dconst=""
