Index: plan-1.9/src/Makefile
===================================================================
--- plan-1.9.orig/src/Makefile
+++ plan-1.9/src/Makefile
@@ -562,7 +562,6 @@ install:
 	cp ../misc/plan_cal.ps ../language/* $(DESTDIR)$(SHARE)
 	-(test x'$(JAPAN)' != x && cp ../misc/plan_calSJ.ps $(DESTDIR)$(SHARE)) || exit 0
 	cp ../language/plan.help $(DESTDIR)$(SHARE)
-	-strip $(DESTDIR)$(DIR)/plan $(DESTDIR)$(DIR)/pland $(DESTDIR)$(DIR)/notifier $(DESTDIR)$(DIR)/netplan
 	-chmod 755 $(DESTDIR)$(DIR)/plan $(DESTDIR)$(DIR)/pland $(DESTDIR)$(DIR)/notifier $(DESTDIR)$(DIR)/netplan
 	-chmod 644 $(DESTDIR)$(SHARE)/plan.help $(DESTDIR)$(SHARE)/plan_cal.ps $(DESTDIR)$(SHARE)/plan.lang.*
 	-(test x'$(JAPAN)' != x && chmod 644 $(DESTDIR)$(SHARE)/plan_calSJ.ps) || exit 0
