--- pflogsumm.pl.orig	2006-11-17 01:33:02.000000000 +0100
+++ pflogsumm.pl	2006-11-17 01:33:50.000000000 +0100
@@ -696,7 +696,7 @@
 	    }
 	}
 	elsif((($addr, $relay, $delay, $status, $toRmdr) = $logRmdr =~
-		/to=<([^>]*)>, (?:orig_to=<[^>]*>, )?relay=([^,]+), delay=([^,]+), status=(\S+)(.*)$/o) >= 4)
+		/to=<([^>]*)>, (?:orig_to=<[^>]*>, )?relay=([^,]+), (?:conn_use=[^,]+, )?delay=([^,]+), (?:delays=[^,]+, )?(?:dsn=[^,]+, )?status=(\S+)(.*)$/o) >= 4)
 	{
 
 	    if($opts{'m'} && $addr =~ /^(.*!)*([^!]+)!([^!@]+)@([^\.]+)$/o) {
