diff -Naur webserver-1.2.103.orig/configure.ac webserver-1.2.103/configure.ac
--- webserver-1.2.103.orig/configure.ac	2013-01-26 17:52:44.000000000 -0500
+++ webserver-1.2.103/configure.ac	2013-07-15 08:36:01.000000000 -0400
@@ -1597,12 +1597,6 @@
 admin/static/images/flags/Makefile
 admin/static/images/wizards/Makefile
 admin/static/images/other/Makefile
-admin/CTK/Makefile
-admin/CTK/CTK/Makefile
-admin/CTK/static/css/Makefile
-admin/CTK/static/images/Makefile
-admin/CTK/static/js/Makefile
-admin/CTK/static/Makefile
 po/Makefile
 ])
 
