http://lists.freedesktop.org/archives/xdg/2012-April/012348.html

--- src/validate.c
+++ src/validate.c
@@ -499,6 +499,7 @@
   { "Core",                   FALSE, FALSE, FALSE, { NULL } },
   { "KDE",                    FALSE, FALSE, FALSE, { "Qt", NULL } },
   { "GNOME",                  FALSE, FALSE, FALSE, { "GTK", NULL } },
+  { "XFCE",                   FALSE, FALSE, FALSE, { "GTK", NULL } },
   { "GTK",                    FALSE, FALSE, FALSE, { NULL } },
   { "Qt",                     FALSE, FALSE, FALSE, { NULL } },
   { "Motif",                  FALSE, FALSE, FALSE, { NULL } },
