diff -ur mutella-0.4.5.old/mutella/uiremote.cpp mutella-0.4.5/mutella/uiremote.cpp
--- mutella-0.4.5.old/mutella/uiremote.cpp	2009-07-09 04:38:28.583840188 +0300
+++ mutella-0.4.5/mutella/uiremote.cpp	2009-07-09 04:40:48.955839846 +0300
@@ -575,11 +575,11 @@
 	{
 		// login -- HTTP-Basic so far
 		bool bLoginOK = false;
-		char * pszAuth = NULL;
+		const char * pszAuth = NULL;
 		if ( (pszAuth = strstr(szHandshake, "Authorization:")) )
 		{
 			// extract authorization line
-			char * pRN = strstr(pszAuth, "\r\n");
+			const char * pRN = strstr(pszAuth, "\r\n");
 			CString sAuthLine(pszAuth + strlen("Authorization:"), pRN-pszAuth-strlen("Authorization:"));
 			int nPos = sAuthLine.find("Basic");
 			if (nPos>=0)
@@ -616,7 +616,7 @@
 	const char* pszCookie;
 	if( (pszCookie = strstr(szHandshake, "Cookie:")) )
 	{
-		char * pRN = strstr(pszCookie, "\r\n");
+		const char * pRN = strstr(pszCookie, "\r\n");
 		CString sCookieLine(pszCookie + strlen("Cookie:"), pRN-pszCookie-strlen("Cookie:"));
 		// we will split the line by "; " combination
 		list<CString> Cookies;
