diff -rupN mutella-0.4.5b/mutella/common.h mutella-0.4.5/mutella/common.h
--- mutella-0.4.5b/mutella/common.h	2009-01-26 02:01:11.000000000 -0400
+++ mutella-0.4.5/mutella/common.h	2009-01-26 02:05:37.000000000 -0400
@@ -50,7 +50,7 @@ CString restore_string(CString s, bool b
 bool QueryMatch(CString Result, const CString& Query);
 // fast functions which do not copy the string
 void MakeWordList(LPSTR szQuery, vector<char*>& QWords);
-void MakeWordList(LPSTR szQuery, vector<char*>& QWords, vector<char*>& QWords);
+void MakeWordList(LPSTR szQuery, vector<char*>& QWordsInc, vector<char*>& QWords);
 bool MatchWordList(const CString& ResultLower, const vector<char*>& QWords, bool bMatchAll = true);
 
 // slower but 'storage-independent' versions
diff -rupN mutella-0.4.5b/mutella/tstring.h mutella-0.4.5/mutella/tstring.h
--- mutella-0.4.5b/mutella/tstring.h	2009-01-26 02:01:11.000000000 -0400
+++ mutella-0.4.5/mutella/tstring.h	2009-01-26 02:04:00.000000000 -0400
@@ -23,6 +23,7 @@
 
 #include <stdarg.h>
 #include <iostream>
+#include <climits>
 
 #ifdef TSTRING_DEBUG
 
diff -rupN mutella-0.4.5b/mutella/conversions.h mutella-0.4.5/mutella/conversions.h
--- mutella-0.4.5b/mutella/conversions.h	2009-01-26 02:17:42.000000000 -0400
+++ mutella-0.4.5/mutella/conversions.h	2009-01-26 02:18:25.000000000 -0400
@@ -19,9 +19,9 @@
     copyright            : (C) 2001 by
     email                : maksik@gmx.co.uk
  ***************************************************************************/
-
 #if !defined(__CONVERSIONS_H_INCLUDED__)
 #define __CONVERSIONS_H_INCLUDED__
+#include <cstring>
 
 CString DWrdtoStr(DWORD);     // Convert DWORD to a CString
 
diff -rupN mutella-0.4.5b/mutella/gnumarkedfiles.cpp mutella-0.4.5/mutella/gnumarkedfiles.cpp
--- mutella-0.4.5b/mutella/gnumarkedfiles.cpp	2009-01-26 02:17:42.000000000 -0400
+++ mutella-0.4.5/mutella/gnumarkedfiles.cpp	2009-01-26 02:18:46.000000000 -0400
@@ -10,6 +10,7 @@
 #include "config.h"
 #endif
 
+#include <cstring>
 #include "mutella.h"
 #include "structures.h"
 
