--- Makefile    2003-04-21 00:14:50.000000000 +0200
+++ Makefile.new        2003-05-09 14:39:11.000000000 +0200
@@ -92,8 +92,8 @@
 # The database directory can be either in the user's $HOME (or relative to any
 # other env variable) or at a global place like /var/spool (world-writable)
 # -> uncomment 2 lines below to put the databases in the user's home
-#USER_DBDIR  = .nxtvdb
-#DEFS       += -DEPG_DB_ENV=\"HOME\" -DEPG_DB_DIR=\"$(USER_DBDIR)\"
+USER_DBDIR  = .nxtvdb
+DEFS       += -DEPG_DB_ENV=\"HOME\" -DEPG_DB_DIR=\"$(USER_DBDIR)\"
 ifndef USER_DBDIR
 SYS_DBDIR    = /usr/tmp/nxtvdb
 DEFS        += -DEPG_DB_DIR=\"$(SYS_DBDIR)\"
