diff -Naur nxtvepg-2.8.0.orig/Makefile nxtvepg-2.8.0/Makefile
--- nxtvepg-2.8.0.orig/Makefile	2008-03-30 16:19:54.000000000 +0200
+++ nxtvepg-2.8.0/Makefile	2008-03-30 16:21:00.000000000 +0200
@@ -209,8 +209,8 @@
 $(BUILD_DIR)/vbirec: $(VBIREC_OBJS)
 	$(CC) -o $@ $(VBIREC_OBJS) $(LDFLAGS) $(GUILIBS) $(ACQLIBS)
 
-.PHONY: install
-install: daemon Nxtvepg.ad nxtvepgd.1
+.PHONY: install-common
+install-common: daemon nxtvepgd.1
 	test -d $(bindir) || install -d $(bindir)
 	test -d $(mandir) || install -d $(mandir)
 	test -d $(resdir)/app-defaults || install -d $(resdir)/app-defaults
@@ -218,10 +218,16 @@
 	test -d $(INST_DB_DIR) || install -d $(INST_DB_DIR)
 	chmod $(INST_DB_PERM) $(INST_DB_DIR)
 endif
-	install -c -m 0755 $(BUILD_DIR)/nxtvepg $(bindir)
-	install -c -m 0755 $(BUILD_DIR)/nxtvepgd $(bindir)
 	install -c -m 0644 nxtvepg.1   $(mandir)
+
+.PHONY: install-daemon
+install-daemon: install-common
+	install -c -m 0755 $(BUILD_DIR)/nxtvepgd $(bindir)
 	install -c -m 0644 nxtvepgd.1  $(mandir)
+
+.PHONY: install
+install: install-common install-daemon Nxtvepg.ad
+	install -c -m 0755 $(BUILD_DIR)/nxtvepg $(bindir)
 	install -c -m 0644 Nxtvepg.ad  $(resdir)/app-defaults/Nxtvepg
 
 .SUFFIXES: .c .o .tcl
