diff -Naur nxtvepg-2.8.0.orig/Makefile nxtvepg-2.8.0/Makefile
--- nxtvepg-2.8.0.orig/Makefile	2008-03-30 16:36:52.000000000 +0200
+++ nxtvepg-2.8.0/Makefile	2008-03-30 16:36:58.000000000 +0200
@@ -86,7 +86,7 @@
 ACQLIBS += -lpthread
 
 # use UTF-8 internally instead of Latin-1 (EXPERIMENTAL)
-#DEFS   += -DUSE_UTF8 -DXMLTV_OUTPUT_UTF8
+DEFS   += -DUSE_UTF8 -DXMLTV_OUTPUT_UTF8
 
 # enable support for importing XMLTV files
 DEFS   += -DUSE_XMLTV_IMPORT
