diff --git a/src/vi.c b/src/vi.c
index e52ea4d..e0ab94b 100644
--- a/src/vi.c
+++ b/src/vi.c
@@ -918,14 +918,14 @@ vi_comment_out(EditLine *el, Int c)
  * NB: posix implies that we should enter insert mode, however
  * this is against historical precedent...
  */
-#ifdef __weak_reference
+#if defined(__weak_reference) && defined(__NetBSD__)
 extern char *get_alias_text(const char *) __weak_reference(get_alias_text);
 #endif
 protected el_action_t
 /*ARGSUSED*/
 vi_alias(EditLine *el, Int c)
 {
-#ifdef __weak_reference
+#if defined(__weak_reference) && defined(__NetBSD__)
 	char alias_name[3];
 	char *alias_text;
 
