--- Makefile.am.old
+++ Makefile.am
@@ -42,10 +42,9 @@
 DISTCLEANFILES = stamp-h
 
 AM_CPPFLAGS=-DINFODIR='"$(infodir)"' @X_CFLAGS@  -DSYSCONFDIR='"$(sysconfdir)"' \
-	    -DBITMAPDIR='"$(datadir)/games/xboard/bitmaps/default"' \
+	    -DBITMAPDIR='"$(bitmapsdir)"' \
 	    -DLOCALEDIR='"$(localedir)"'
 
-ACLOCAL_AMFLAGS = -I m4
 
 info_TEXINFOS =  xboard.texi
 xboard_TEXINFOS =  copyright.texi 
@@ -59,15 +59,15 @@
 
 ### icon files
 
-icondir = $(datadir)/icons/hicolor/48x48/apps
+icondir = /usr/share/icons/hicolor/48x48/apps
 dist_icon_DATA = xboard.png
-svgicondir = $(datadir)/icons/hicolor/scalable/apps
+svgicondir = /usr/share/icons/hicolor/scalable/apps
 dist_svgicon_DATA = xboard.svg
 
 
 ### desktop files for KDE and gnome
 
-Applicationsdir = $(datadir)/applications
+Applicationsdir = /usr/share/applications
 Applications_DATA = xboard.desktop xboard-fen-viewer.desktop  xboard-pgn-viewer.desktop  xboard-tourney.desktop
 
 ### mime file
@@ -76,7 +76,7 @@
 
 ### directory and files for bitmaps
 
-bitmapsdir = $(datadir)/games/xboard/bitmaps/default
+bitmapsdir = $(datadir)/bitmaps/default
 dist_bitmaps_DATA = \
   bitmaps/a33o.bm    bitmaps/c45s.bm        bitmaps/icon_outline.bm  bitmaps/m49o.bm   bitmaps/p45o.bm   bitmaps/r29o.bm \
   bitmaps/a33s.bm    bitmaps/c49o.bm        bitmaps/icon_white.bm    bitmaps/m49s.bm   bitmaps/p45s.bm   bitmaps/r29s.bm \
@@ -145,12 +145,12 @@
 
 ### directory and files for pixmaps
 
-pixmapsdir = $(datadir)/games/xboard/pixmaps/textures
+pixmapsdir = $(datadir)/pixmaps/textures
 dist_pixmaps_DATA = pixmaps/wood_d.xpm pixmaps/wood_l.xpm pixmaps/cross32.xpm pixmaps/cross48.xpm pixmaps/board32.xpm pixmaps/board48.xpm
 
 ### directory and files for sound files
 
-soundsdir = $(datadir)/games/xboard/sounds
+soundsdir = $(datadir)/sounds
 dist_sounds_DATA = \
   sounds/cymbal.wav     sounds/pop2.wav   sounds/slap.wav     sounds/ding1.wav      sounds/laser.wav \
   sounds/woodthunk.wav  sounds/gong.wav   sounds/penalty.wav  sounds/honkhonk.wav   sounds/phone.wav
--- xboard.conf.in.old
+++ xboard.conf.in
@@ -67,8 +67,8 @@
 ;
 ; Engines & adjudicatons in engine-engine games
 ;
--firstChessProgram fairymax
--secondChessProgram fairymax
+-firstChessProgram gnuchess
+-secondChessProgram gnuchess
 -firstChessProgramNames {fairymax
 "Fruit 2.1" -fcp fruit -fUCI
 "Crafty" -fcp crafty
--- xboard.h.old
+++ xboard.h
@@ -119,8 +119,8 @@
 
 #define BORDER_X_OFFSET 3
 #define BORDER_Y_OFFSET 27
-#define FIRST_CHESS_PROGRAM	"fairymax"
-#define SECOND_CHESS_PROGRAM	"fairymax"
+#define FIRST_CHESS_PROGRAM	"gnuchess"
+#define SECOND_CHESS_PROGRAM	"gnuchess"
 #define FIRST_DIRECTORY         "."
 #define SECOND_DIRECTORY        "."
 #define SOUND_BELL              ""
