commit 0b6d8acf55b4d8721e854cfba025e621f61ff5c2
Author: hasufell <hasufell@gentoo.org>
Date:   Fri Sep 6 18:15:38 2013 +0200

    set gnuchess as default engine

--- a/gtk/xboard.h
+++ b/gtk/xboard.h
@@ -118,8 +118,8 @@ typedef struct {
 
 #define BORDER_X_OFFSET 3
 #define BORDER_Y_OFFSET 27
-#define FIRST_CHESS_PROGRAM	"fairymax"
-#define SECOND_CHESS_PROGRAM	"fairymax"
+#define FIRST_CHESS_PROGRAM	"gnuchess"
+#define SECOND_CHESS_PROGRAM	"gnuchess"
 #define FIRST_DIRECTORY         "."
 #define SECOND_DIRECTORY        "."
 #define SOUND_BELL              ""
--- a/xaw/xboard.h
+++ b/xaw/xboard.h
@@ -118,8 +118,8 @@ typedef struct {
 
 #define BORDER_X_OFFSET 3
 #define BORDER_Y_OFFSET 27
-#define FIRST_CHESS_PROGRAM	"fairymax"
-#define SECOND_CHESS_PROGRAM	"fairymax"
+#define FIRST_CHESS_PROGRAM	"gnuchess"
+#define SECOND_CHESS_PROGRAM	"gnuchess"
 #define FIRST_DIRECTORY         "."
 #define SECOND_DIRECTORY        "."
 #define SOUND_BELL              ""
--- a/xboard.conf.in
+++ b/xboard.conf.in
@@ -67,8 +67,8 @@
 ;
 ; Engines & adjudicatons in engine-engine games
 ;
--firstChessProgram fairymax
--secondChessProgram fairymax
+-firstChessProgram gnuchess
+-secondChessProgram gnuchess
 -firstChessProgramNames {fairymax
 "Fruit 2.1" -fcp fruit -fUCI
 "Crafty" -fcp crafty
