--- native/Makefile.orig	2009-06-03 22:09:02.000000000 +0200
+++ native/Makefile	2009-06-03 22:14:56.000000000 +0200
@@ -49,7 +49,6 @@
 LIBPFX=lib
 LIBSFX=.so
 JNISFX=$(LIBSFX)
-CC=gcc
 LD=gcc
 # Default to Sun recommendations for JNI compilation
 #COPT=-O2 -fomit-frame-pointer
@@ -61,8 +60,8 @@
 COUT=-o $@
 CINCLUDES=$(JAVA_INCLUDES) -I"$(JAVAH)" -I$(FFI_BUILD)/include
 CDEFINES=-D_REENTRANT
-PCFLAGS=-W -Wall -Wno-unused -Wno-parentheses
-CFLAGS=$(PCFLAGS) $(COPT) $(CDEBUG) $(CDEFINES) $(CINCLUDES) \
+PCFLAGS=-Wall -Wno-unused -Wno-parentheses
+CFLAGS+=$(PCFLAGS) $(COPT) $(CDEBUG) $(CDEFINES) $(CINCLUDES) \
        -DVERSION='"$(VERSION)"' -DCHECKSUM='"$(CHECKSUM)"'
 # -static-libgcc avoids gcc library incompatibilities across linux systems
 LDFLAGS=-o $@ -shared -static-libgcc
