diff --git a/src/pubsubpublishfactory.cpp b/src/pubsubpublishfactory.cpp
index 22d90b4..130f022 100644
--- a/src/pubsubpublishfactory.cpp
+++ b/src/pubsubpublishfactory.cpp
@@ -116,7 +116,7 @@ void PublishFactory::serialize(Payload *extension, QXmlStreamWriter *writer)
 		Logger::warning() << "Invalid stanza extension at PubSub::Publish";
 		return;
 	}
-	writer->writeStartElement(QLatin1String("pusbsub"));
+	writer->writeStartElement(QLatin1String("pubsub"));
 	writer->writeDefaultNamespace(NS_PUBSUB);
 	writer->writeStartElement(QLatin1String("publish"));
 	writer->writeAttribute(QLatin1String("node"), node);