diff -Naurb redmine-1.4.1.orig/config/boot.rb redmine-1.4.1/config/boot.rb
--- redmine-1.4.1.orig/config/boot.rb	2012-04-20 19:01:56.000000000 +0900
+++ redmine-1.4.1/config/boot.rb	2012-04-25 13:00:01.702911764 +0900
@@ -41,11 +41,6 @@
   class Boot
     def run
       load_initializer
-      Rails::Initializer.class_eval do
-        def load_gems
-          @bundler_loaded ||= Bundler.require :default, Rails.env
-        end
-      end
       Rails::Initializer.run(:set_load_path)
     end
   end
