--- python-selinux-orig/Makefile	2006-02-12 04:37:05.000000000 +0300
+++ python-selinux/Makefile	2011-02-22 19:06:02.000000000 +0300
@@ -23,7 +23,7 @@
 	$(PYREXC) $<
 
 %.so: %.c
-	$(CC) -fPIC -shared -o $@ -I /usr/include/python${PYVER} $< -lselinux
+	$(CC) -fPIC -shared -o $@ -I /usr/include/python${PYVER} ${CFLAGS} ${LDFLAGS} $< -lselinux
 
 install: $(all_so)
 	mkdir -p ${DESTDIR}/usr/lib/python${PYVER}/site-packages
