--- netwatch-1.3.0/netwatch.c.old	2009-01-03 04:08:50.000000000 +0100
+++ netwatch-1.3.0/netwatch.c	2009-03-19 19:44:12.268281240 +0100
@@ -2758,7 +2758,7 @@ updatecurrent (HOSTINFO * work, struct i
       work->plog = open (nam, O_APPEND | O_WRONLY);
       if (work->plog < 0)
       {
-	work->plog = open (nam, O_APPEND | O_CREAT | O_WRONLY);
+	work->plog = open (nam, O_APPEND | O_CREAT | O_WRONLY, S_IRUSR | S_IWUSR);
 	st = write (work->plog, &starttime, sizeof (starttime));
 	st = write (work->plog, &simmagic, sizeof (simmagic));
       }
