Index: src/view_dir_list.c
===================================================================
--- src/view_dir_list.c	(revision 1917)
+++ src/view_dir_list.c	(working copy)
@@ -309,7 +309,7 @@
 		base = remove_level_from_path(vd->dir_fd->path);
 		if (strcmp(base, dir_fd->path) == 0)
 			{
-			old_path = g_strdup(vd->dir_fd->name);
+			old_path = g_strdup(vd->dir_fd->path);
 			}
 		g_free(base);
 		}
@@ -329,7 +329,7 @@
 		while (work && !found)
 			{
 			FileData *fd = work->data;
-			if (strcmp(old_path, fd->name) == 0) found = fd;
+			if (strcmp(old_path, fd->path) == 0) found = fd;
 			work = work->next;
 			}
 
