--- src/tools/bdfgname.c
+++ src/tools/bdfgname.c
@@ -47,9 +47,9 @@
 
 static int
 #ifdef __STDC__
-getline(FILE *in, char *buf, int limit)
+get_line(FILE *in, char *buf, int limit)
 #else
-getline(in, buf, limit)
+get_line(in, buf, limit)
 FILE *in;
 char *buf;
 int limit;
@@ -99,11 +99,11 @@
 
     while (!feof(in)) {
         pos = ftell(in);
-        (void) getline(in, buf, 256);
+        (void) get_line(in, buf, 256);
         while (!feof(in) && (buf[0] == 0 || buf[0] == '#')) {
             buf[0] = 0;
             pos = ftell(in);
-            (void) getline(in, buf, 256);
+            (void) get_line(in, buf, 256);
         }
 
         if (buf[0] == 0)
@@ -170,11 +170,11 @@
 
     while (!feof(in)) {
         pos = ftell(in);
-        (void) getline(in, buf, 256);
+        (void) get_line(in, buf, 256);
         while (!feof(in) && (buf[0] == 0 || buf[0] == '#')) {
             buf[0] = 0;
             pos = ftell(in);
-            (void) getline(in, buf, 256);
+            (void) get_line(in, buf, 256);
         }
 
         c = _bdf_atol(buf, 0, 16);
