diff -r -U1 fotoxx-14.04.orig/Makefile fotoxx-14.04/Makefile
--- fotoxx-14.04.orig/Makefile	2014-04-01 13:09:32.000000000 +0700
+++ fotoxx-14.04/Makefile	2014-04-03 03:13:14.752618591 +0700
@@ -5,5 +5,4 @@
 # defaults for parameters that may be pre-defined
-CXXFLAGS ?= -O2 -Wall -ggdb -Wno-deprecated-declarations
-
-LDFLAGS ?= -rdynamic 
+CXXFLAGS += -Wall -Wno-deprecated-declarations
+LDFLAGS += -rdynamic
 PREFIX ?= /usr
@@ -16,3 +15,3 @@
 LOCALESDIR = $(SHAREDIR)/locales
-DOCDIR = $(PREFIX)/share/doc/fotoxx
+DOCDIR = $(PREFIX)/share/doc/$(PF)
 MANDIR = $(PREFIX)/share/man/man1
@@ -20,4 +19,4 @@
 
-CFLAGS = $(CXXFLAGS) -c `pkg-config --cflags gtk+-3.0`
-LIBS = `pkg-config --libs gtk+-3.0`
+CFLAGS = $(CXXFLAGS) -c -DPF=\"$(PF)\" `pkg-config --cflags gtk+-3.0`
+LIBS = `pkg-config --libs gtk+-3.0` -lpthread
 
@@ -79,3 +78,2 @@
 	rm -f  $(DESTDIR)$(MANDIR)/fotoxx.1.gz
-	xdg-desktop-menu uninstall $(DESTDIR)$(MENUFILE)
 	rm -f  $(DESTDIR)$(MENUFILE)
@@ -99,5 +97,2 @@
 	rm -f fotoxx.1.gz
-	# menu (desktop) file
-	cp -f desktop $(DESTDIR)$(MENUFILE)
-	xdg-desktop-menu install --novendor $(DESTDIR)$(MENUFILE)
 
@@ -108,3 +103,2 @@
 	rm -f  $(DESTDIR)$(MANDIR)/fotoxx.1.gz
-	xdg-desktop-menu uninstall $(DESTDIR)$(MENUFILE)
 	rm -f  $(DESTDIR)$(MENUFILE)
diff -r -U1 fotoxx-14.04.orig/zfuncs.cc fotoxx-14.04/zfuncs.cc
--- fotoxx-14.04.orig/zfuncs.cc	2014-04-01 13:09:32.000000000 +0700
+++ fotoxx-14.04/zfuncs.cc	2014-04-03 03:16:43.485604505 +0700
@@ -3995,3 +3995,3 @@
    strncatv(zlocalesdir,199,work,"/share/",zappname,"/locales",null);      //  /prefix/share/appname/locales
-   strncatv(zdocdir,199,work,"/share/doc/",zappname,null);                 //  /prefix/share/doc/appname
+   strncatv(zdocdir,199,work,"/share/doc/",PF,null);                       //  /prefix/share/doc/$(PF)
    
