diff -r -U1 fotoxx-14.06.1.orig/Makefile fotoxx-14.06.1/Makefile
--- fotoxx-14.06.1.orig/Makefile	2014-06-06 17:40:42.000000000 +0700
+++ fotoxx-14.06.1/Makefile	2014-06-12 01:33:21.967129937 +0700
@@ -5,5 +5,4 @@
 # defaults for parameters that may be pre-defined
-CXXFLAGS ?= -O2 -Wall -ggdb 
-
-LDFLAGS ?= -rdynamic 
+CXXFLAGS += -Wall
+LDFLAGS += -rdynamic
 PREFIX ?= /usr
@@ -16,3 +15,3 @@
 LOCALESDIR = $(SHAREDIR)/locales
-DOCDIR = $(PREFIX)/share/doc/fotoxx
+DOCDIR = $(PREFIX)/share/doc/$(PF)
 MANDIR = $(PREFIX)/share/man/man1
@@ -21,4 +20,4 @@
 
-CFLAGS = $(CXXFLAGS) -c `pkg-config --cflags gtk+-3.0`
-LIBS = `pkg-config --libs gtk+-3.0`
+CFLAGS = $(CXXFLAGS) -c -DPF=\"$(PF)\" `pkg-config --cflags gtk+-3.0`
+LIBS = `pkg-config --libs gtk+-3.0` -lpthread
 
@@ -80,3 +79,2 @@
 	rm -f  $(DESTDIR)$(MANDIR)/fotoxx.1.gz
-	xdg-desktop-menu uninstall $(DESTDIR)$(MENUFILE)
 	rm -f  $(DESTDIR)$(MENUFILE)
@@ -87,2 +85,3 @@
 	mkdir -p  $(DESTDIR)$(DOCDIR)
+	mkdir -p  $(DESTDIR)$(APPDATADIR)
 	mkdir -p  $(DESTDIR)$(MANDIR)
@@ -101,6 +100,2 @@
 	rm -f fotoxx.1.gz
-	# menu (desktop) file
-	cp -f desktop $(DESTDIR)$(MENUFILE)
-	# chmod +x $(DESTDIR)$(MENUFILE)
-	xdg-desktop-menu install --novendor $(DESTDIR)$(MENUFILE)
 
@@ -111,3 +106,2 @@
 	rm -f  $(DESTDIR)$(MANDIR)/fotoxx.1.gz
-	xdg-desktop-menu uninstall $(DESTDIR)$(MENUFILE)
 	rm -f  $(DESTDIR)$(MENUFILE)
diff -r -U1 fotoxx-14.06.1.orig/zfuncs.cc fotoxx-14.06.1/zfuncs.cc
--- fotoxx-14.06.1.orig/zfuncs.cc	2014-06-06 17:40:42.000000000 +0700
+++ fotoxx-14.06.1/zfuncs.cc	2014-06-12 01:31:16.672135206 +0700
@@ -3864,3 +3864,3 @@
    strncatv(zlocalesdir,199,work,"/share/",zappname,"/locales",null);      //  /prefix/share/appname/locales
-   strncatv(zdocdir,199,work,"/share/doc/",zappname,null);                 //  /prefix/share/doc/appname
+   strncatv(zdocdir,199,work,"/share/doc/",PF,null);                       //  /prefix/share/doc/$(PF)
    
