Don't strip - bug #240764

--- Makefile.in.orig	2002-09-19 02:00:40.000000000 +0200
+++ Makefile.in	2008-10-10 13:17:39.000000000 +0200
@@ -66,7 +66,7 @@
 
 install-$(TARGET): $(TARGET)
 	$(INSTALL) -d -m 755 $(bindir)
-	$(INSTALL) -s -m 755 $(TARGET) $(bindir)/$(TARGET)
+	$(INSTALL) -m 755 $(TARGET) $(bindir)/$(TARGET)
 	$(INSTALL) -d -m 755 $(mandir)/man1
 	$(INSTALL) -m 644 ifstat.1 $(mandir)/man1/ifstat.1
 
