--- kf-0.1.4.1/configure	2004-06-12 22:40:49.912614680 +0200
+++ kf-0.1.4.1/configure	2004-06-12 22:41:36.264568112 +0200
@@ -1216,28 +1216,28 @@
   else
      PKG_CONFIG_MIN_VERSION=0.9.0
      if $PKG_CONFIG --atleast-pkgconfig-version $PKG_CONFIG_MIN_VERSION; then
-        echo $ac_n "checking for gtkspell-2.0 >= 2.0.5""... $ac_c" 1>&6
-echo "configure:1221: checking for gtkspell-2.0 >= 2.0.5" >&5
+        echo $ac_n "checking for gtkspell-2.0 >= 2.0.4""... $ac_c" 1>&6
+echo "configure:1221: checking for gtkspell-2.0 >= 2.0.4" >&5
 
-        if $PKG_CONFIG --exists "gtkspell-2.0 >= 2.0.5" ; then
+        if $PKG_CONFIG --exists "gtkspell-2.0 >= 2.0.4" ; then
             echo "$ac_t""yes" 1>&6
             succeeded=yes
 
             echo $ac_n "checking GTKSPELL_CFLAGS""... $ac_c" 1>&6
 echo "configure:1228: checking GTKSPELL_CFLAGS" >&5
-            GTKSPELL_CFLAGS=`$PKG_CONFIG --cflags "gtkspell-2.0 >= 2.0.5"`
+            GTKSPELL_CFLAGS=`$PKG_CONFIG --cflags "gtkspell-2.0 >= 2.0.4"`
             echo "$ac_t""$GTKSPELL_CFLAGS" 1>&6
 
             echo $ac_n "checking GTKSPELL_LIBS""... $ac_c" 1>&6
 echo "configure:1233: checking GTKSPELL_LIBS" >&5
-            GTKSPELL_LIBS=`$PKG_CONFIG --libs "gtkspell-2.0 >= 2.0.5"`
+            GTKSPELL_LIBS=`$PKG_CONFIG --libs "gtkspell-2.0 >= 2.0.4"`
             echo "$ac_t""$GTKSPELL_LIBS" 1>&6
         else
             GTKSPELL_CFLAGS=""
             GTKSPELL_LIBS=""
             ## If we have a custom action on failure, don't print errors, but 
             ## do set a variable so people can do so.
-            GTKSPELL_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "gtkspell-2.0 >= 2.0.5"`
+            GTKSPELL_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "gtkspell-2.0 >= 2.0.4"`
             
         fi
 
@@ -1258,7 +1258,7 @@
 #	PACKAGE_LIBS="$PACKAGE_LIBS $GTKSPELL_LIBS"
 #	PACKAGE_CFLAGS="$PACKAGE_CFLAGS $GTKSPELL_CFLAGS"
 	if test "x$have_gtkspell" = "xyes"; then
-		MODULES="$MODULES gtkspell-2.0 >= 2.0.5"
+		MODULES="$MODULES gtkspell-2.0 >= 2.0.4"
 		cat >> confdefs.h <<\EOF
 #define HAVE_GTKSPELL 1
 EOF
