--- src/Makefile.in.orig	2005-01-08 09:36:54.863238032 +0930
+++ src/Makefile.in	2005-01-08 09:38:04.867595752 +0930
@@ -40,10 +40,11 @@
 updatedd.o: updatedd.h
 
 install:
-	$(INSTALL) -m 755 $(bin) $(bindir)
+	$(INSTALL) -d $(DESTDIR)$(bindir)
+	$(INSTALL) -m 755 $(bin) $(DESTDIR)$(bindir)
 
-	$(INSTALL) -d $(mandir)/man1
-	$(INSTALL) -m 644 updatedd.1 $(mandir)/man1
+	$(INSTALL) -d $(DESTDIR)$(mandir)/man1
+	$(INSTALL) -m 644 updatedd.1 $(DESTDIR)$(mandir)/man1
 
 uninstall:
 	$(RM) $(bindir)/$(bin)
--- Documentation/Makefile.in.orig	2005-01-08 09:44:02.846174736 +0930
+++ Documentation/Makefile.in	2005-01-08 09:44:51.121835720 +0930
@@ -31,9 +31,9 @@
 	rm -f Makefile
 
 install:
-	$(INSTALL) -d $(datadir)/doc/updatedd
-	$(INSTALL) updatedd-*.pdf $(datadir)/doc/updatedd
-	$(INSTALL) updatedd-pppd-rc $(datadir)/doc/updatedd
+	$(INSTALL) -d $(DESTDIR)$(datadir)/doc/updatedd
+	$(INSTALL) updatedd-*.pdf $(DESTDIR)$(datadir)/doc/updatedd
+	$(INSTALL) updatedd-pppd-rc $(DESTDIR)$(datadir)/doc/updatedd
 
 uninstall:
 	$(RM) -r $(datadir)/doc/updatedd
--- ./src/plugins/Makefile.in.orig	2005-01-08 09:46:28.027514976 +0930
+++ ./src/plugins/Makefile.in	2005-01-08 09:46:52.125721824 +0930
@@ -41,8 +41,8 @@
 		$(LIBERROR) -o $*.so
 
 install:
-	$(INSTALL) -d $(libdir)/updatedd
-	$(INSTALL) -m 755 *.so $(libdir)/updatedd
+	$(INSTALL) -d $(DESTDIR)$(libdir)/updatedd
+	$(INSTALL) -m 755 *.so $(DESTDIR)$(libdir)/updatedd
 
 uninstall:
 	$(RM) -r $(libdir)/updatedd
--- ./scripts/Makefile.in.orig	2005-01-08 09:47:38.972691872 +0930
+++ ./scripts/Makefile.in	2005-01-08 09:47:59.305241760 +0930
@@ -30,8 +30,8 @@
 	$(RM) Makefile
 
 install:
-	$(INSTALL) -d $(datadir)/updatedd
-	$(INSTALL) *.pl $(datadir)/updatedd
+	$(INSTALL) -d $(DESTDIR)$(datadir)/updatedd
+	$(INSTALL) *.pl $(DESTDIR)$(datadir)/updatedd
 
 uninstall:
 	$(RM) -r $(datadir)/updatedd
--- ./updatedd-wrapper/Makefile.in.orig	2005-01-08 09:48:22.544508520 +0930
+++ ./updatedd-wrapper/Makefile.in	2005-01-08 09:50:04.877951472 +0930
@@ -37,18 +37,19 @@
 	$(RM) Makefile
 
 install:
-	$(INSTALL) -m 755 $(bin) $(bindir)
+	$(INSTALL) -d $(DESTDIR)/$(bindir)
+	$(INSTALL) -m 755 $(bin) $(DESTDIR)$(bindir)
 
-	if test ! -e $(sysconfdir)/$(conf); then		\
-		$(INSTALL) -d $(sysconfdir);			\
-		$(INSTALL) -m 600 $(conf) $(sysconfdir);	\
+	if test ! -e $(DESTDIR)$(sysconfdir)/$(conf); then		\
+		$(INSTALL) -d $(DESTDIR)$(sysconfdir);			\
+		$(INSTALL) -m 600 $(conf) $(DESTDIR)$(sysconfdir);	\
 	fi
 
-	$(INSTALL) -d $(mandir)/man1
-	$(INSTALL) -d $(mandir)/man5
+	$(INSTALL) -d $(DESTDIR)$(mandir)/man1
+	$(INSTALL) -d $(DESTDIR)$(mandir)/man5
 
-	$(INSTALL) -m 644 updatedd-wrapper.1 $(mandir)/man1
-	$(INSTALL) -m 644 updatedd-wrapper.conf.5 $(mandir)/man5
+	$(INSTALL) -m 644 updatedd-wrapper.1 $(DESTDIR)$(mandir)/man1
+	$(INSTALL) -m 644 updatedd-wrapper.conf.5 $(DESTDIR)$(mandir)/man5
 
 uninstall:
 	$(RM) $(bindir)/$(bin)
--- configure.orig	2005-01-08 09:55:16.094639344 +0930
+++ configure	2005-01-08 09:55:28.810706208 +0930
@@ -1959,13 +1959,13 @@
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-g "
   else
     CFLAGS="-g"
   fi
 else
   if test "$GCC" = yes; then
-    CFLAGS="-O2"
+    CFLAGS=""
   else
     CFLAGS=
   fi
@@ -3566,9 +3566,9 @@
 #
 
 if test "$CC" = "gcc"; then
-	CFLAGS="-Wall -Wwrite-strings -O2 $CFLAGS"
+	CFLAGS="-Wall -Wwrite-strings  $CFLAGS"
 else
-	CFLAGS="-O2 $CFLAGS"
+	CFLAGS=" $CFLAGS"
 fi
 
 # ------------------] PERL [
