--- prime-1.0.0.1.orig/src/prime.src	2005-03-07 16:51:34.000000000 +0900
+++ prime-1.0.0.1/src/prime.src	2010-05-27 18:44:37.000000000 +0900
@@ -12,8 +12,10 @@
 # the GNU General Public License version 2.
 #
 
+require 'rbconfig'
+
 PRIME_BINDIR = '%bindir%'
-PRIME_LIBDIR = '%rubydir%'
+PRIME_LIBDIR = Config::CONFIG["sitelibdir"]
 $LOAD_PATH.unshift(PRIME_LIBDIR) unless $LOAD_PATH.member?(PRIME_LIBDIR)
 
 require 'getoptlong'
