--- Imakefile_	2003-12-06 18:09:23.845935720 +0100
+++ Imakefile	2003-12-06 18:10:02.620041160 +0100
@@ -1,5 +1,5 @@
-/**/# Imakefile for xkeycaps;
-/**/# Copyright (c) 1991, 1992, 1993, 1994, 1995, 1999 Jamie Zawinski.
+/* Imakefile for xkeycaps;
+   Copyright (c) 1991, 1992, 1993, 1994, 1995, 1999 Jamie Zawinski. */
 
 /*
  *  To specify a different default keyboard (for when the vendor display
@@ -353,8 +353,8 @@
             TAR = tar
        COMPRESS = compress
    COMPRESS_EXT = Z
-/**/#      COMPRESS = gzip --verbose --best
-/**/#  COMPRESS_EXT = gz
+/* COMPRESS = gzip --verbose --best */
+/* COMPRESS_EXT = gz */
 
 ComplexProgramTarget(xkeycaps)
 
