commit 054055b5115af6046dc9b4396b7911221a3b765c
Author: hasufell <julian.ospald@googlemail.com>
Date:   Sat Jun 8 13:32:21 2013 +0200

    CMAKE: add option to choose between system luajit/lua

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 51ed278..3d7e815 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -180,19 +180,28 @@ if(USE_FREETYPE)
 	set(CGUITTFONT_LIBRARY cguittfont)
 endif(USE_FREETYPE)
 
-
-find_library(LUA_LIBRARY luajit
-		NAMES luajit-5.1)
-find_path(LUA_INCLUDE_DIR luajit.h
-	NAMES luajit.h
-	PATH_SUFFIXES luajit-2.0)
-message (STATUS "LuaJIT library: ${LUA_LIBRARY}")
-message (STATUS "LuaJIT headers: ${LUA_INCLUDE_DIR}")
+option(USE_LUAJIT "Use luajit instead of lua" OFF)
+
+if(USE_LUAJIT)
+	find_library(LUA_LIBRARY luajit
+			NAMES libluajit-5.1.so luajit-5.1)
+	find_path(LUA_INCLUDE_DIR luajit.h
+		NAMES luajit.h
+		PATH_SUFFIXES luajit-2.0)
+	message (STATUS "LuaJIT library: ${LUA_LIBRARY}")
+	message (STATUS "LuaJIT headers: ${LUA_INCLUDE_DIR}")
+else(USE_LUAJIT)
+	find_package(Lua51)
+endif(USE_LUAJIT)
 
 if(LUA_LIBRARY AND LUA_INCLUDE_DIR) 
-	message (STATUS "LuaJIT found.")
+	if(USE_LUAJIT)
+		message (STATUS "LuaJIT found.")
+	else(USE_LUAJIT)
+		message (STATUS "Lua51 found.")
+	endif(USE_LUAJIT)
 else(LUA_LIBRARY AND LUA_INCLUDE_DIR)
-	message (STATUS "LuaJIT not found, using bundled Lua.")
+	message (STATUS "LuaJIT/Lua51 not found, using bundled Lua.")
 	set(LUA_INCLUDE_DIR "${PROJECT_SOURCE_DIR}/lua/src")
 	set(LUA_LIBRARY "lua")
 	add_subdirectory(lua)
