--- svgalib/init.c.old	2010-01-04 16:38:38.000000000 +0100
+++ svgalib/init.c	2010-01-04 16:39:09.000000000 +0100
@@ -44,7 +44,7 @@
 #include <sys/stat.h>
 #include <sys/vt.h>
 #include <unistd.h>
-#include <asm/io.h>		/*for waiting for retrace */
+#include <sys/io.h>		/*for waiting for retrace */
 #include <sys/ioctl.h>		/*for waiting for retrace */
 extern void     fadein1 ();
 extern GraphicsContext *starbackground;
