diff --git a/plparse/totem-pl-parser.c b/plparse/totem-pl-parser.c
index aeaf3ea..dd00037 100644
--- a/plparse/totem-pl-parser.c
+++ b/plparse/totem-pl-parser.c
@@ -1835,8 +1835,10 @@ totem_pl_parser_parse_internal (TotemPlParser *parser,
 
 	/* Not a directory on http though */
 	if (g_strcmp0 (mimetype, "inode/directory") == 0 &&
-	    g_file_has_uri_scheme (file, "http"))
-		g_clear_pointer (&mimetype, g_free);
+	    g_file_has_uri_scheme (file, "http")) {
+		g_free (mimetype);
+		mimetype = NULL;
+	    }
 
 	DEBUG(file, g_print ("_get_mime_type_for_name for '%s' returned '%s'\n", uri, mimetype));
 	if (mimetype == NULL || strcmp (UNKNOWN_TYPE, mimetype) == 0
