diff -ru ChessDB-3.6.18.orig/Makefile.conf ChessDB-3.6.18/Makefile.conf
--- ChessDB-3.6.18.orig/Makefile.conf	2007-04-25 06:09:51.000000000 -0400
+++ ChessDB-3.6.18/Makefile.conf	2009-09-15 13:58:13.390702234 -0400
@@ -227,14 +227,14 @@
 #   for all users, and put extra files in $SHAREDIR, type "make install".
 #
 install: all
-	install -m 755 -d $(SHAREDIR)
+	install -m 755 -d $(DESTDIR)$(SHAREDIR)
 	install -m 755 -d $(DESTDIR)$(BINDIR)
-	install -m 755 -d $(SOUNDSDIR)
-	install -m 755 -d $(TBDIR)
-	install -m 755 -d $(MANDIR)/man6
-	-install -m 644 -p $(SOUNDS)  $(SOUNDSDIR)
-#	-install -m 644 -p $(TABLEBASES)  $(TBDIR)
-	install -m 644 -p $(MAN6) $(MANDIR)/man6
+	install -m 755 -d $(DESTDIR)$(SOUNDSDIR)
+	install -m 755 -d $(DESTDIR)$(TBDIR)
+	install -m 755 -d $(DESTDIR)$(MANDIR)/man6
+	-install -m 644 -p $(SOUNDS)  $(DESTDIR)$(SOUNDSDIR)
+#	-install -m 644 -p $(TABLEBASES)  $(DESTDIR)$(TBDIR)
+	install -m 644 -p $(MAN6) $(DESTDIR)$(MANDIR)/man6
 	install -m 755 chessdb $(SCRIPTS) $(EXECS) $(DESTDIR)$(BINDIR)
 	-install -m 755 -d $(DESTDIR)$(SHAREDIR)
 	-install -m 644 -p chessdb.eco ratings.ssp $(DESTDIR)$(SHAREDIR)
@@ -339,7 +339,7 @@
 
 scmerge: src/scmerge.o src/misc.o src/index.o src/date.o src/namebase.o \
           src/gfile.o src/bytebuf.o src/textbuf.o src/myassert.o \
-          src/stralloc.o
+          src/stralloc.o $(OBJS)
 	$(LINK) -o scmerge src/scmerge.o $(OBJS) -lz
 
 pgnchessdb: src/pgnchessdb.o $(OBJS)
diff -ru ChessDB-3.6.18.orig/tcl/start.tcl ChessDB-3.6.18/tcl/start.tcl
--- ChessDB-3.6.18.orig/tcl/start.tcl	2007-04-21 17:51:42.000000000 -0400
+++ ChessDB-3.6.18/tcl/start.tcl	2009-09-15 13:59:18.354832575 -0400
@@ -17,7 +17,7 @@
 # The "\" at the end of the comment line below is necessary! It means
 #   that the "exec" line is a comment to Tcl/Tk, but not to /bin/sh.
 # The next line restarts using tkchessdb: \
-exec tkchessdb "$0" "$@"
+TCLLIBPATH="/usr/lib/snack2.2 $TCLLIBPATH" exec tkchessdb "$0" "$@"
 
 # For the above to work, tkchessdb must be in a directory in your PATH.
 # Alternatively, you can set the first line to start tkchessdb directly
@@ -147,13 +147,13 @@
 if {$windowsOS} {
   set ecoFile [file join $chessdbDataDir "chessdb.eco"]
 } else {
-  set ecoFile "/usr/local/share/chessdb/chessdb.eco"
+  set ecoFile "@GENTOO_DATADIR@/chessdb.eco"
 }
 
 if {$windowsOS} {
   set defaultTBDIR [file nativename [file join $::chessdbExeDir "tablebases"]]
 } else {
-  set defaultTBDIR "/usr/local/share/chessdb/tablebases"
+  set defaultTBDIR "@GENTOO_DATADIR@/tablebases"
 }
 
 # boardSizes: a list of the available board sizes.
@@ -692,8 +692,8 @@
   set ::utils::sound::soundFolder [file nativename [file join $::chessdbExeDir sounds]]
   set spellCheckFile [file nativename [file join $chessdbDataDir "ratings.ssp"] ]
 } else {
-  set spellCheckFile "/usr/local/share/chessdb/ratings.ssp"
-  set ::utils::sound::soundFolder "/usr/local/share/chessdb/sounds"
+  set spellCheckFile "@GENTOO_DATADIR@/ratings.ssp"
+  set ::utils::sound::soundFolder "@GENTOO_DATADIR@/sounds"
 }
 
 # Engines data:
