Index: ctrlproxy-3.0.8/Makefile
===================================================================
--- ctrlproxy-3.0.8.orig/Makefile
+++ ctrlproxy-3.0.8/Makefile
@@ -103,7 +103,7 @@ dep_files = $(patsubst %.o, %.d, $(objs)
 
 linestack-cmd$(EXEEXT): src/linestack-cmd.o $(objs) $(LIBIRC)
 	@echo Linking $@
-	@$(LD) $(LIBS) -lreadline -rdynamic -o $@ $^
+	@$(LD) $(LDFLAGS) -rdynamic -o $@ $^ $(LIBS) -lreadline
 
 ctrlproxy$(EXEEXT): src/main.o $(objs) $(LIBIRC)
 	@echo Linking $@
