 Include/cholmod_internal.h |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/Include/cholmod_internal.h b/Include/cholmod_internal.h
index 36f20a2..2cd6dd2 100644
--- a/Include/cholmod_internal.h
+++ b/Include/cholmod_internal.h
@@ -53,9 +53,7 @@
 #define NDEBUG
 #endif
 
-/* Uncomment this line to enable debugging.  CHOLMOD will be very slow.
 #undef NDEBUG
- */
 
 #ifdef MATLAB_MEX_FILE
 #include "mex.h"
