diff -ru berusky-1.5.2.orig/Makefile.am berusky-1.5.2/Makefile.am
--- berusky-1.5.2.orig/Makefile.am	2012-05-17 06:30:00.000000000 -0400
+++ berusky-1.5.2/Makefile.am	2012-11-15 17:26:23.271928805 -0500
@@ -5,8 +5,7 @@
 
 SUBDIRS = po src data
 
-beruskydocdir = ${prefix}/doc/berusky
-beruskydoc_DATA = \
+EXTRA_DIST = \
 	README\
 	COPYING\
 	AUTHORS\
@@ -15,8 +14,6 @@
 	NEWS\
 	TODO
 
-EXTRA_DIST = $(beruskydoc_DATA)
-
 # Copy all the spec files. Of cource, only one is actually used.
 dist-hook:
 	for specfile in *.spec; do \
diff -ru berusky-1.5.2.orig/data/Makefile.am berusky-1.5.2/data/Makefile.am
--- berusky-1.5.2.orig/data/Makefile.am	2012-05-17 06:30:00.000000000 -0400
+++ berusky-1.5.2/data/Makefile.am	2012-11-15 17:27:12.966230390 -0500
@@ -3,10 +3,3 @@
 ## Created by Anjuta - will be overwritten
 ## If you don't want it to overwrite it,
 ## 	Please disable it in the Anjuta project configuration
-
-berusky_datadir = $(prefix)/@NO_PREFIX_PACKAGE_DATA_DIR@/@PACKAGE@
-
-berusky_data_DATA = \
-	berusky.ini
-
-EXTRA_DIST = $(berusky_data_DATA)
diff -ru berusky-1.5.2.orig/data/berusky.ini berusky-1.5.2/data/berusky.ini
--- berusky-1.5.2.orig/data/berusky.ini	2012-05-17 06:30:00.000000000 -0400
+++ berusky-1.5.2/data/berusky.ini	2012-11-15 17:27:52.708872231 -0500
@@ -1,9 +1,9 @@
 # Configuration for berusky game
 
 # Game data location
-level_data = /usr/share/berusky/Levels
-game_data = /usr/share/berusky/GameData
-graphics_data = /usr/share/berusky/Graphics
+level_data = @GENTOO_DATADIR@/Levels
+game_data = @GENTOO_DATADIR@/GameData
+graphics_data = @GENTOO_DATADIR@/Graphics
 
 # Graphics settings for game and editor
 # game runs in 640x480
@@ -17,7 +17,7 @@
 
 # Game binary & tmp dir
 # they're used by editor for "run level" command
-game_binary = /usr/bin/berusky
+game_binary = @GENTOO_BINDIR@/berusky
 tmp_data = /var/tmp
 
 # Dir for users levels (it's the default level dir for level editor)
diff -ru berusky-1.5.2.orig/src/defines.h berusky-1.5.2/src/defines.h
--- berusky-1.5.2.orig/src/defines.h	2012-09-22 15:21:33.000000000 -0400
+++ berusky-1.5.2/src/defines.h	2012-11-15 17:28:33.043493961 -0500
@@ -445,7 +445,7 @@
 #define INI_USER_DIRECTORY "~/.berusky"
 #define INI_USER_LEVELS    "~/.berusky/User"
 #define INI_USER_PROFILES  "~/.berusky/Profiles"
-#define INI_FILE_GLOBAL    "/var/games/berusky/"INI_FILE_NAME
+#define INI_FILE_GLOBAL    "@GENTOO_DATADIR@/"INI_FILE_NAME
 #define INI_FILE_USER      "~/.berusky/"INI_FILE_NAME
 #define INI_FILE_LOCAL     "./"INI_FILE_NAME
 #elif WINDOWS
