Index: trunk/distrib/gentoo/initpost
===================================================================
--- trunk/distrib/gentoo/initpost (revision 2712)
+++ trunk/distrib/gentoo/initpost (revision 2715)
@@ -66,6 +66,6 @@
 	# force /lib/rc/sh even if we don't have it in older stages
 	$_CHROOT_SH mkdir /lib 2>/dev/null || :
-	$_CHROOT_SH mkdir /lib/rc 2/dev/null || :
-	$_CHROOT_SH mkdir /lib/rc/sh 2/dev/null || :
+	$_CHROOT_SH mkdir /lib/rc 2>/dev/null || :
+	$_CHROOT_SH mkdir /lib/rc/sh 2>/dev/null || :
 
 	$_CAT "$__DISTRIBDIR"/gentoo/init-vserver.sh | \
