Index: scripts/vserver-init.functions
===================================================================
--- scripts/vserver-init.functions	(revision 2711)
+++ scripts/vserver-init.functions	(revision 2712)
@@ -21,7 +21,7 @@
     _postResult() { echo; }
     lockfile=/var/lock/subsys/$LOCKFILE
 elif test -e /etc/gentoo-release; then
-    . /sbin/functions.sh
+    . /etc/init.d/functions.sh
     _beginResult() { ebegin "$@"; }
     _postResult() { :; }
     success() { eend 0; }
Index: distrib/gentoo/initpost
===================================================================
--- distrib/gentoo/initpost	(revision 2711)
+++ distrib/gentoo/initpost	(revision 2712)
@@ -63,6 +63,11 @@
 if test "$initstyle" == "gentoo"; then
 	echo ">>> Installing special init-style magic ... "
 
+	# force /lib/rc/sh even if we don't have it in older stages
+	$_CHROOT_SH mkdir /lib 2>/dev/null || :
+	$_CHROOT_SH mkdir /lib/rc 2/dev/null || :
+	$_CHROOT_SH mkdir /lib/rc/sh 2/dev/null || :
+
 	$_CAT "$__DISTRIBDIR"/gentoo/init-vserver.sh | \
 	$_CHROOT_SH truncate   /lib/rc/sh/init-vserver.sh
 	$_CHROOT_SH chmod 0755 /lib/rc/sh/init-vserver.sh
