diff -Nru util-vserver-0.30.215.orig/Makefile.in util-vserver-0.30.215/Makefile.in
--- util-vserver-0.30.215.orig/Makefile.in	2008-03-17 10:00:13.000000000 +0100
+++ util-vserver-0.30.215/Makefile.in	2008-06-18 19:24:54.000000000 +0200
@@ -1496,7 +1496,7 @@
 	$(lib_internal_testsuite_sigbus_gnu_SOURCES) \
 	$(src_capchroot_SOURCES) $(src_chain_echo_SOURCES) \
 	$(src_chbind_compat_SOURCES) $(src_chcontext_compat_SOURCES) \
-	src/check-unixfile.c $(src_chroot_sh_SOURCES) \
+	$(src_chroot_sh_SOURCES) \
 	$(src_chxid_SOURCES) $(src_exec_cd_SOURCES) \
 	$(src_exec_ulimit_SOURCES) $(src_fakerunlevel_SOURCES) \
 	$(src_filetime_SOURCES) $(src_h2ext_SOURCES) \
@@ -1818,7 +1818,7 @@
 	ensc_fmt/lib_libvserver_a-% lib/testsuite/cflags \
 	lib/testsuite/personality lib/testsuite/fmt \
 	src/chcontext-compat src/capchroot src/chain-echo \
-	src/check-unixfile src/chroot-sh src/filetime src/lockfile \
+	src/chroot-sh src/filetime src/lockfile \
 	src/readlink src/secure-mount src/save_ctxinfo src/mask2prefix \
 	src/chbind-compat src/exec-cd src/fakerunlevel \
 	src/keep-ctx-alive src/exec-ulimit src/vps src/lsxid src/chxid \
@@ -1828,7 +1828,7 @@
 	src/vsched src/vshelper-sync src/vrsetup src/vserver-stat \
 	src/vserver-info src/vdlimit src/vdu src/vwait src/ncontext \
 	src/nattribute src/naddress src/vdevmap src/vsysctl src/vclone \
-	src/h2ext src/vtag src/vspace src/vmemctrl src/tunctl \
+	src/vtag src/vspace src/vmemctrl src/tunctl \
 	$(am__append_3) $(tests_dietprogs) \
 	lib_internal/libinternal-diet.a \
 	lib_internal/testsuite/isnumber lib_internal/testsuite/sigbus \
diff -Nru util-vserver-0.30.215.orig/src/Makefile-files util-vserver-0.30.215/src/Makefile-files
--- util-vserver-0.30.215.orig/src/Makefile-files	2008-03-17 09:59:45.000000000 +0100
+++ util-vserver-0.30.215/src/Makefile-files	2008-06-18 17:49:42.000000000 +0200
@@ -38,7 +38,6 @@
 DIETPROGS +=		src/chcontext-compat \
 			src/capchroot \
 			src/chain-echo \
-			src/check-unixfile \
 			src/chroot-sh \
 			src/filetime \
 			src/lockfile \
@@ -81,7 +80,6 @@
 			src/vdevmap \
 			src/vsysctl \
 			src/vclone \
-			src/h2ext \
 			src/vtag \
 			src/vspace \
 			src/vmemctrl \
