--- pptpctrl.c.orig	2013-10-25 15:10:53.632625057 +0400
+++ pptpctrl.c	2013-10-25 15:11:12.175626673 +0400
@@ -759,8 +759,8 @@
 			syslog(LOG_DEBUG, "CTRL (PPPD Launcher): remote address = %s", pppaddrs[1]);
 	}
 	
+	char pppInterfaceIPs[33];
 	if (*pppaddrs[0] || *pppaddrs[1]) {
-		char pppInterfaceIPs[33];
 		sprintf(pppInterfaceIPs, "%s:%s", pppaddrs[0], pppaddrs[1]);
 		pppd_argv[an++] = pppInterfaceIPs;
 	}
