diff --git a/bin/test.pl b/bin/test.pl
index 145be47..81c50bd 100755
--- a/bin/test.pl
+++ b/bin/test.pl
@@ -1,5 +1,5 @@
 #!/usr/bin/env perl
 chdir "./test";
-$r=system "../bin/t_coffee test.pep -method fast_pair -outfile=new_reference_test.aln -outorder=input>/dev/null";
+$r=system "../t_coffee_source/t_coffee test.pep -method fast_pair -outfile=new_reference_test.aln -outorder=input>/dev/null";
 if ($r==0){print "\nInstallation of t_coffee Successful\n";}
 else {print "\nInstallation of t_coffee Not Successful\n";}
