--- libtrace-3.0.14/Makefile.am
+++ libtrace-3.0.14/Makefile.am
@@ -5,9 +5,7 @@
 ACLOCAL_AMFLAGS = -I m4
 AUTOMAKE_OPTIONS = 1.9 foreign
 
-if HAS_DOXYGEN
 man_MANS = docs/doxygen/man/man3/trace*.3 docs/doxygen/man/man3/libtrace*.3 
-endif
 EXTRA_DIST=libtrace.sln libtrace.vcproj test-convert.vcproj \
 		   test-format.vcproj test-pcap-bpf.vcproj
 
@@ -26,10 +24,6 @@
 		sed -i "s/SVN_REVISION.*/SVN_REVISION $$r/" $(distdir)/lib/libtrace.h.in; \
 	fi
 
-if HAS_DOXYGEN
-docs/doxygen/man/man3/*.3: docs
-else
 docs/doxygen/man/man3/*.3: 
-endif
 
 install-man-hook: docs
