--- vdr-1.7.32/PLUGINS/src/burn/scanner.c	2011-09-07 21:37:44.000000000 +0200
+++ vdr-1.7.33/PLUGINS/src/burn/scanner.c	2012-12-08 20:37:44.462711233 +0100
@@ -405,12 +405,12 @@
 				if (!PmtFound) {
 					if ( Pid == 0)
 						PatPmtParser.ParsePat(DataPtr, TS_SIZE);
-					else if (Pid == PatPmtParser.PmtPid())
+					else if (PatPmtParser.IsPmtPid(Pid))
 						PatPmtParser.ParsePmt(DataPtr, TS_SIZE);
 					else if (PatPmtParser.GetVersions(PatVersion, PmtVersion)) {
 						PmtFound = true;
 						int streams = 0;
-						logger::debug(format( "PID found: PMT PID=0x{0}, Vpid=0x{1}, Vtype=0x{2}") % format::base( PatPmtParser.PmtPid(), 16 ) % format::base( PatPmtParser.Vpid(), 16 ) % format::base( PatPmtParser.Vtype(), 16 ));
+						logger::debug(format( "PID found: Vpid=0x{1}, Vtype=0x{2}") % format::base( PatPmtParser.Vpid(), 16 ) % format::base( PatPmtParser.Vtype(), 16 ));
 						if ( PatPmtParser.Vpid() && (PatPmtParser.Vtype() == 2)) { // accept only MPEG2
 							track_info track( PatPmtParser.Vpid(), track_info::streamtype_video );
 							m_tracks.push_back( track );
