--- a/curses.c
+++ b/curses.c
@@ -314,6 +314,7 @@
     mpls = net_mpls(at);
 
     if( addrcmp( (void *) addr, (void *) &unspec_addr, af ) != 0 ) {
+#ifdef ENABLE_IPV6
       struct in6_addr addr6 = *addr;
 
       if (PrintAS) {
@@ -352,6 +353,7 @@
 
               printw("[AS%s] ", chas);
       }
+#endif /* ENABLE_IPV6 */
       name = dns_lookup(addr);
       if (! net_up(at))
 	attron(A_BOLD);
