http://bugs.gentoo.org/401311

--- src/ffmpeg2theora.c
+++ src/ffmpeg2theora.c
@@ -522,8 +522,8 @@
     AVStream *vstream = NULL;
     AVCodec *acodec = NULL;
     AVCodec *vcodec = NULL;
-    pp_mode_t *ppMode = NULL;
-    pp_context_t *ppContext = NULL;
+    pp_mode *ppMode = NULL;
+    pp_context *ppContext = NULL;
     int sws_flags = this->resize_method;
     float frame_aspect = 0;
     double fps = 0.0;
