--- src/cpufreqd_acpi_battery.c.orig	2010-04-18 15:57:41.000000000 +0200
+++ src/cpufreqd_acpi_battery.c	2012-04-07 14:26:54.409641936 +0200
@@ -36,6 +36,7 @@
 #define PRESENT		"present"
 #define STATUS		"status"
 #define CURRENT_NOW	"current_now"
+#define POWER_NOW	"power_now"
 
 struct battery_info {
 	int capacity;
@@ -146,8 +147,12 @@
 	if (!binfo->status)
 		return -1;
 	binfo->current_now = get_class_device_attribute(binfo->cdev, CURRENT_NOW);
-	if (!binfo->current_now)
-		return -1;
+	if (!binfo->current_now) {
+		/* try the "power_now" name */
+		binfo->current_now = get_class_device_attribute(binfo->cdev, POWER_NOW);
+		if (!binfo->current_now)
+			return -1;
+	}
 
 	/* read the last full capacity, this is not going to change
 	 * very often, so no need to poke it later */
@@ -311,6 +316,10 @@
 
 	/* Read battery informations */
 	for (i = 0; i < bat_dir_num; i++) {
+		if (!info[i].open) {
+			clog(LOG_INFO, "Skipping %s (closed)\n", info[i].cdev->name);
+			continue;
+		}
 
 		if (read_int(info[i].present, &info[i].is_present) != 0) {
 			clog(LOG_INFO, "Skipping %s\n", info[i].cdev->name);
