--- m4/i18n.mk.old	2012-04-22 20:47:31.285053576 +0200
+++ m4/i18n.mk	2012-04-22 20:51:29.911102673 +0200
@@ -21,11 +21,9 @@
 	done
 
 $(STAMP): $(POFILES)
-	@for j in $(POFILES); do \
+	@for j in $(LINGUAS); do \
 		echo "converting $$j"; \
-		k=$$(basename $$j); \
-		country=`echo $$k | cut -d "." -f 1`; \
-		$(MSGFMT) -o $$country.mo $$j; \
+		[[ ! -f $$j.po ]] || $(MSGFMT) -o $$j.mo $$j.po; \
 	done
 	@touch $(STAMP)
 
@@ -35,12 +35,14 @@
 
 install-data-local:
 	@for j in $(MOFILES); do \
+		if [[ -f $$j ]]; then \
 		k=$$(basename $$j); \
 		country=`echo $$k | cut -d "." -f 1`; \
 		dir=${DESTDIR}$(prefix)/share/locale/$$country/LC_MESSAGES; \
 		echo "installing $$dir/$(CATALOG).mo ($$k)"; \
 		install -d $$dir; \
 		cp $$j $$dir/$(CATALOG).mo; \
+		fi \
 	done
 
 uninstall-local:
