--- gnubg.orig/Makefile.am
+++ gnubg/Makefile.am
@@ -2,7 +2,7 @@
 
 ACLOCAL_AMFLAGS = -I m4
 
-AM_CPPFLAGS = -DAC_DOCDIR='"$(datadir)/doc/gnubg/"' -DAC_DATADIR='"$(datadir)"' -DAC_PKGDATADIR='"$(datadir)/gnubg"'
+AM_CPPFLAGS = -DAC_DOCDIR='"@docdir@"' -DAC_DATADIR='"$(datadir)"' -DAC_PKGDATADIR='"$(datadir)/gnubg"'
 
 SUBDIRS = lib doc met po m4 sounds board3d textures scripts flags fonts non-src pixmaps .
 
--- gnubg.orig/doc/Makefile.am
+++ gnubg/doc/Makefile.am
@@ -1,6 +1,6 @@
-AM_CPPFLAGS = -DAC_DOCDIR='"$(datadir)/doc/gnubg/"' -DAC_DATADIR='"$(datadir)"' -DAC_PKGDATADIR='"$(datadir)/gnubg"'
-gnubg_helpdir = $(datadir)/doc/gnubg/
-gnubg_imagedir = $(datadir)/doc/gnubg/images
+AM_CPPFLAGS = -DAC_DOCDIR='"@docdir@"' -DAC_DATADIR='"$(datadir)"' -DAC_PKGDATADIR='"$(datadir)/gnubg"'
+gnubg_helpdir = @docdir@ 
+gnubg_imagedir = @docdir@/images
 gnubg_help_DATA =  gnubg.html allabout.html
 gnubg_image_DATA = images/1540d81e.png images/16d7944c.png images/1805dc7d.png \
 		   images/20bc52ca.png images/20d56757.png images/234924dc.png images/26e34ea5.png \
--- gnubg.orig/pixmaps/Makefile.am
+++ gnubg/pixmaps/Makefile.am
@@ -1,5 +1,5 @@
-gnubgiconsdir = $(datadir)
-gnubgpixmapsdir = $(pkgdatadir)/pixmaps
+gnubgiconsdir = /usr/share
+gnubgpixmapsdir = /usr/share/pixmaps
 
 ICONS_16_SCALABLE = icons/hicolor/16x16/apps/scalable/gnubg.svg
 ICONS_16 = icons/hicolor/16x16/apps/gnubg.png
