diff -r cd828dc1ffd9 tests/96F47C6A.asc
--- a/tests/96F47C6A.asc	Tue Sep 02 17:54:40 2014 +0200
+++ b/tests/96F47C6A.asc	Tue Sep 02 17:55:56 2014 +0200
@@ -1,5 +1,4 @@
 -----BEGIN PGP PUBLIC KEY BLOCK-----
-Version: GnuPG v1.4.12 (GNU/Linux)
 
 mI0EUAlt9AEEAMVygQA7sGE7xS5lGk8bPK7vtBuNAb9ETjXmC8jPLZ35KTviRq1c
 NfFl0J66ObgcLEQjl84GMQ6Du6qLDAAgwi/2TqzmoGljXEiOd+lePBOhEV7WgkTx
diff -r cd828dc1ffd9 tests/test_gpg.py
--- a/tests/test_gpg.py	Tue Sep 02 17:54:40 2014 +0200
+++ b/tests/test_gpg.py	Tue Sep 02 17:55:56 2014 +0200
@@ -150,6 +150,7 @@
         self.assertTrue(self.gpg.import_data(open(os.path.dirname(__file__) + '/96F47C6A.asc').read()))
         k1 = open(os.path.dirname(__file__) + '/96F47C6A.asc').read()
         self.gpg.context.set_option('armor')
+        self.gpg.context.set_option('no-emit-version')
         self.gpg.context.set_option('export-options', 'export-minimal')
         k2 = self.gpg.export_data('96F47C6A')
         self.assertEqual(k1,k2)
