--- macaddr.h.orig	2009-03-18 00:08:27.000000000 -0700
+++ macaddr.h	2009-03-18 00:09:30.000000000 -0700
@@ -73,7 +73,7 @@
 			longmac |= (uint64_t) bs_in[5] << ((MAC_LEN - 5 - 1) * 8);
 
             // If it has a mask component, get that
-            char *in_mask = strchr(in, '/');
+            const char *in_mask = strchr(in, '/');
             if (in_mask != NULL) {
                 longmask = 0;
 
