--- a/bindings/ruby/extconf.rb
+++ b/bindings/ruby/extconf.rb
@@ -4,13 +4,13 @@
 require 'mkmf'
 
 if /linux/ =~ RUBY_PLATFORM
-   $LDFLAGS += '-Wl,--rpath -Wl,$(EPREFIX)/lib'
+   $LDFLAGS += ' -Wl,--rpath -Wl,$(EPREFIX)/lib'
 elsif /solaris/ =~ RUBY_PLATFORM
-   $LDFLAGS += '-R$(EPREFIX)/lib'
+   $LDFLAGS += ' -R$(EPREFIX)/lib'
 elsif /hpux/ =~ RUBY_PLATFORM
-   $LDFLAGS += '+b$(EPREFIX)/lib'
+   $LDFLAGS += ' +b$(EPREFIX)/lib'
 elsif /aix/ =~ RUBY_PLATFORM
-   $LDFLAGS += '-Wl,-blibpath:$(EPREFIX)/lib'
+   $LDFLAGS += ' -Wl,-blibpath:$(EPREFIX)/lib'
 end
 
 dir_config("rrd","../../src","../../src/.libs")
