https://bugs.gentoo.org/509332

--- a/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp.
+++ b/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp
@@ -423,9 +423,9 @@
         // mp3 being one of them sadly. Most importantly: allow the libsndfile decoder to do
         // its thing.
         //
-        if( file->type() == CODEC_ID_WMAV1 ||
-            file->type() == CODEC_ID_WMAV2 ||
-            file->type() == CODEC_ID_AAC )
+        if( file->type() == AV_CODEC_ID_WMAV1 ||
+            file->type() == AV_CODEC_ID_WMAV2 ||
+            file->type() == AV_CODEC_ID_AAC )
 #endif
             return file;
     }
