--- multiskkserv-ctl.c.orig	2002-01-13 06:10:13.000000000 +0900
+++ multiskkserv-ctl.c	2004-11-23 18:19:27.670353257 +0900
@@ -168,7 +168,7 @@
   char rbuf[SKKSERV_REQUEST_SIZE];
 
   if ((sock = socket_connect(remote, &sstr, port, (char *)SKKSERV_SERVICE, family)) < 0) {
-    fprintf(stderr, __FUNCTION__ ": cannot make a connection.\n");
+    fprintf(stderr, "%s: cannot make a connection.\n", __FUNCTION__);
     return;
   }
 
