diff -ur cmucl.orig/src/lisp/Config.linux_gencgc cmucl/src/lisp/Config.linux_gencgc
--- cmucl.orig/src/lisp/Config.linux_gencgc	2006-07-18 21:54:31.000000000 -0500
+++ cmucl/src/lisp/Config.linux_gencgc	2006-10-21 11:59:40.000000000 -0500
@@ -21,7 +21,7 @@
 RUNTIME = $(GENCGC) $(LINKAGE)
 # __NO_CTYPE so builds on glibc 2.3 will run on (some) older glibc's.
 CPPFLAGS = -D__NO_CTYPE -I. -I$(PATH1) -I- -I/usr/X11R6/include $(RUNTIME)
-CFLAGS = -rdynamic -Wstrict-prototypes -Wall -g $(RUNTIME)
+CFLAGS = @CFLAGS@ -rdynamic -Wstrict-prototypes -Wall -g $(RUNTIME)
 ASFLAGS = -g -DGENCGC -DLINKAGE_TABLE
 NM = $(PATH1)/linux-nm
 UNDEFSYMPATTERN = -Xlinker -u -Xlinker &
diff -ur cmucl.orig/src/motif/server/Config.x86 cmucl/src/motif/server/Config.x86
--- cmucl.orig/src/motif/server/Config.x86	2004-04-26 07:07:36.000000000 -0500
+++ cmucl/src/motif/server/Config.x86	2006-10-21 11:59:13.000000000 -0500
@@ -1,6 +1,6 @@
-CFLAGS = -O2 -I/usr/X11R6/include -I. -I$(VPATH)
+CFLAGS = @CFLAGS@ -I/usr/X11R6/include -I. -I$(VPATH)
 LDFLAGS = -L/usr/X11R6/lib
-LIBS = -static -lXm -dynamic -lXt -lXext -lX11 -lSM -lICE -lXp -ldl -lpthread
+LIBS = -lXm -lXt -lXext -lX11 -lSM -lICE -lXp -ldl -lpthread
 # This def assumes you are building in the same or parallel
 # tree to the CVS souce layout. Sites may need to customize
 # this path.
