diff -ur cmucl.orig/src/code/print.lisp cmucl/src/code/print.lisp
--- cmucl.orig/src/code/print.lisp	2009-07-06 17:42:34.000000000 +0200
+++ cmucl/src/code/print.lisp	2009-11-09 01:47:01.000000000 +0100
@@ -2190,8 +2190,11 @@
     (set-bit #\_ extension-attribute)
     (set-bit #\. dot-attribute)
     (set-bit #\/ slash-attribute)
+    (set-bit #\` funny-attribute)
+    (set-bit #\| funny-attribute)
 
-    ;; Make anything not explicitly allowed funny...
+    ;; For everything else, derive the attribute from the Unicode
+    ;; properties of the character.
     (dotimes (i char-code-limit)
       (when (zerop (aref character-attributes i))
 	(let* ((char (code-char i)))
