diff -ur cmucl.orig/src/code/misc.lisp cmucl/src/code/misc.lisp
--- cmucl.orig/src/code/misc.lisp	2010-04-20 19:57:45.000000000 +0200
+++ cmucl/src/code/misc.lisp	2010-10-15 02:15:53.064623104 +0200
@@ -181,8 +181,10 @@
 
 (defun lisp-implementation-version ()
   "Returns a string describing the implementation version."
-  (format nil "~A (~X~A)" *lisp-implementation-version* c:byte-fasl-file-version
-	  #+unicode (intl:gettext " Unicode") #-unicode ""))
+  (format nil "~X~A~A"
+          c:byte-fasl-file-version
+          #+unicode "-unicode" #-unicode ""
+          *lisp-implementation-version*))
 
 (defun machine-instance ()
   "Returns a string giving the name of the local machine."
