Index: tools/make-main-dist.sh
===================================================================
RCS file: /project/cmucl/cvsroot/src/tools/make-main-dist.sh,v
retrieving revision 1.18
diff -u -w -u -r1.18 make-main-dist.sh
--- cmucl/src/tools/make-main-dist.sh	13 Aug 2009 14:22:56 -0000	1.18
+++ cmucl/src/tools/make-main-dist.sh	30 Sep 2009 14:26:41 -0000
@@ -29,6 +29,7 @@
 fi
 
 DESTDIR=${INSTALL_DIR:-release-$$}
+DOCDIR=${DOCDIR:-doc/cmucl}
 MANDIR=${MANDIR:-man/man1}
 TARGET="`echo $1 | sed 's:/*$::'`"
 VERSION=$2
@@ -81,11 +82,11 @@
 # set -x
 echo Installing main components
 install -d ${GROUP} ${OWNER} -m 0755 $DESTDIR/bin
-install -d ${GROUP} ${OWNER} -m 0755 $DESTDIR/doc/cmucl
 install -d ${GROUP} ${OWNER} -m 0755 $DESTDIR/lib/cmucl
 install -d ${GROUP} ${OWNER} -m 0755 $DESTDIR/lib/cmucl/lib
 install -d ${GROUP} ${OWNER} -m 0755 $DESTDIR/lib/cmucl/lib/subsystems
 install -d ${GROUP} ${OWNER} -m 0755 $DESTDIR/lib/cmucl/lib/ext-formats
+install -d ${GROUP} ${OWNER} -m 0755 $DESTDIR/${DOCDIR}
 install -d ${GROUP} ${OWNER} -m 0755 $DESTDIR/${MANDIR}
 install ${GROUP} ${OWNER} -m 0755 $TARGET/lisp/lisp $DESTDIR/bin/
 if [ "$EXECUTABLE" = "true" ]
@@ -126,11 +127,11 @@
 	$DESTDIR/${MANDIR}/
 install ${GROUP} ${OWNER} -m 0644 src/general-info/lisp.1 \
 	$DESTDIR/${MANDIR}/
-install ${GROUP} ${OWNER} -m 0644 src/general-info/README $DESTDIR/doc/cmucl/
+install ${GROUP} ${OWNER} -m 0644 src/general-info/README $DESTDIR/${DOCDIR}
 if [ -f src/general-info/release-$VERSION.txt ] 
 then
 	install ${GROUP} ${OWNER} -m 0644 src/general-info/release-$VERSION.txt \
-		$DESTDIR/doc/cmucl/
+		$DESTDIR/${DOCDIR}
 fi
 
 if [ -z "$INSTALL_DIR" ]; then
