--- mednafen.orig/src/Makefile.am
+++ mednafen/src/Makefile.am
@@ -5,7 +5,7 @@
 
 bin_PROGRAMS	=	mednafen
 mednafen_SOURCES 	= 	debug.cpp error.cpp mempatcher.cpp settings.cpp endian.cpp mednafen.cpp file.cpp general.cpp md5.cpp memory.cpp netplay.cpp state.cpp movie.cpp player.cpp PSFLoader.cpp tests.cpp qtrecord.cpp cdplay.cpp okiadpcm.cpp FileWrapper.cpp Stream.cpp MemoryStream.cpp FileStream.cpp
-mednafen_LDADD 		= 	trio/libtrio.a drivers/libmdfnsdl.a sexyal/libsexyal.a @SDL_LIBS@
+mednafen_LDADD 		= 	trio/libtrio.a drivers/libmdfnsdl.a sexyal/libsexyal.a @SDL_LIBS@ -lminizip
 mednafen_DEPENDENCIES	=	trio/libtrio.a drivers/libmdfnsdl.a sexyal/libsexyal.a
 
 SUBDIRS			+=	sexyal drivers
--- mednafen.orig/src/compress/Makefile.am.inc
+++ mednafen/src/compress/Makefile.am.inc
@@ -1 +1 @@
-mednafen_SOURCES	+=	compress/minilzo.c compress/quicklz.c compress/blz.cpp compress/ioapi.c compress/unzip.c
+mednafen_SOURCES	+=	compress/minilzo.c compress/quicklz.c compress/blz.cpp compress/ioapi.c
--- mednafen.orig/src/file.cpp
+++ mednafen/src/file.cpp
@@ -33,7 +33,7 @@
 
 
 #include <zlib.h>
-#include "compress/unzip.h"
+#include <minizip/unzip.h>
 
 #include "file.h"
 #include "general.h"
