--- sources/a68k/src/A68kdef.h.org	2004-05-19 15:21:48.555233864 +0200
+++ sources/a68k/src/A68kdef.h	2004-05-19 15:22:12.527589512 +0200
@@ -35,7 +35,7 @@
 #include <fcntl.h> */
 #else
 /* extern char *malloc(); */ /* (2.71.F3d) should be declared in stdlib.h */
-extern int  open(), creat(), read(), write(), close(), unlink();
+/* extern int  open(), creat(), read(), write(), close(), unlink(); */
 #if defined(__MACH__) && defined(__APPLE__) /* Darwin / OS X */
 /* On Darwin, the implicit declaration below won't work, so we need to include
    <unistd.h>. - v.2.71.F3j, Kevin Kofler (reported by Christian Walther) */
