--- unix/configure.orig	2006-05-12 00:08:06.000000000 -0400
+++ unix/configure	2006-05-12 00:08:22.000000000 -0400
@@ -5552,7 +5552,7 @@
 	    # results, and the version is kept in special file).
 	
 	    if test -r /etc/.relid -a "X`uname -n`" = "X`uname -s`" ; then
-		system=MP-RAS-`awk '{print }' /etc/.relid'`
+		system=MP-RAS-`awk '{print }' /etc/.relid`
 	    fi
 	    if test "`uname -s`" = "AIX" ; then
 		system=AIX-`uname -v`.`uname -r`
