diff -r a4fab18949f0 -r 0374b3dbc4ae dbserver.ml
--- a/dbserver.ml	Fri Jan 25 18:16:13 2013 -0600
+++ b/dbserver.ml	Tue Apr 16 15:36:36 2013 -0500
@@ -293,8 +293,8 @@
           let keys = lookup_keys request.search in
           let count = List.length keys in
           let keys = truncate request.limit keys in
+          let keys = clean_keys request keys in
           let hashes = List.map ~f:KeyHash.hash keys in
-          let keys = clean_keys request keys in
           if request.machine_readable then
             ("text/plain",
              count,
diff -r a4fab18949f0 -r 0374b3dbc4ae parsePGP.ml
--- a/parsePGP.ml	Fri Jan 25 18:16:13 2013 -0600
+++ b/parsePGP.ml	Tue Apr 16 15:36:36 2013 -0500
@@ -153,7 +153,7 @@
 
 let parse_ecdh_pubkey cin =
    let length = cin#read_int_size 1 in
-   let oid = sprintf "%x" (cin#read_int_size length) in
+   let oid = sprintf "%Lx" (cin#read_int64_size length) in
    let mpi = read_mpi cin in
    let kdf_length = cin#read_int_size 1 in
    let kdf_res = cin#read_int_size 1 in
@@ -166,7 +166,7 @@
 
  let parse_ecdsa_pubkey cin =
    let length = cin#read_int_size 1 in
-   let oid = sprintf "%x" (cin#read_int_size length) in
+   let oid = sprintf "%Lx" (cin#read_int64_size length) in
    let psize = oid_to_psize oid
    in
    psize
