diff -u zaphistory-0.9.4-orig/zaphistoryosd.c zaphistory-0.9.4/zaphistoryosd.c
--- zaphistory-0.9.4-orig/zaphistoryosd.c	2006-05-17 21:56:05.000000000 +0200
+++ zaphistory-0.9.4/zaphistoryosd.c	2006-05-17 21:58:28.000000000 +0200
@@ -205,8 +205,11 @@
     if (schedule == NULL)
 	return osUnknown;
 
-    return AddSubMenu( new cMenuEvent(schedule->GetEventAround(time(NULL)) ) );
-    
+    const cEvent *event = schedule->GetEventAround(time(NULL));
+    if (event)
+	return AddSubMenu( new cMenuEvent(event) );
+    else
+	return osUnknown;
 }
 
 eOSState cMenuZappedChannels::ProcessKey(eKeys Key)
