diff -ur unrtf-0.19.3/convert.c.old unrtf-0.19.3/convert.c
--- unrtf-0.19.3/convert.c.old	2004-12-25 21:08:41.000000000 -0500
+++ unrtf-0.19.3/convert.c	2004-12-25 20:40:26.000000000 -0500
@@ -341,7 +341,7 @@
 				while(w2) {
 					tmp = word_string (w2);
 					if (tmp && tmp[0] != '\\')
-						strcat(name,tmp);
+						strncat(name,tmp,sizeof(name) - strlen(name) - 1);
 
 					w2=w2->next;
 				}
