diff -Nuarp unrtf-0.21.1.orig/charmaps/Makefile.am unrtf-0.21.1/charmaps/Makefile.am
--- unrtf-0.21.1.orig/charmaps/Makefile.am	2010-01-17 07:33:12.000000000 +0000
+++ unrtf-0.21.1/charmaps/Makefile.am	2010-02-01 06:37:10.184989205 +0000
@@ -1,2 +1,2 @@
-outputsdir = ${libdir}/${PACKAGE}
+outputsdir = ${pkgdatadir}
 dist_outputs_DATA = SYMBOL.charmap
diff -Nuarp unrtf-0.21.1.orig/outputs/Makefile.am unrtf-0.21.1/outputs/Makefile.am
--- unrtf-0.21.1.orig/outputs/Makefile.am	2010-01-17 07:32:48.000000000 +0000
+++ unrtf-0.21.1/outputs/Makefile.am	2010-02-01 06:37:20.968321648 +0000
@@ -1,2 +1,2 @@
-outputsdir = ${libdir}/${PACKAGE}
+outputsdir = ${pkgdatadir}
 dist_outputs_DATA = html.conf latex.conf rtf.conf text.conf vt.conf
diff -Nuarp unrtf-0.21.1.orig/tests/Makefile.am unrtf-0.21.1/tests/Makefile.am
--- unrtf-0.21.1.orig/tests/Makefile.am	2010-02-01 06:30:50.085007020 +0000
+++ unrtf-0.21.1/tests/Makefile.am	2010-02-01 06:31:07.848358870 +0000
@@ -61,7 +61,7 @@ CLEANFILES = $(testresults_html) $(testr
 # add the test scripts and the test data to the distributed files
 EXTRA_DIST = $(TESTS) $(testdata)
 
-TESTS_ENVIRONMENT = UNRTF=$(top_builddir)/src/unrtf
+TESTS_ENVIRONMENT = UNRTF="$(top_builddir)/src/unrtf -P $(top_builddir)/outputs:$(top_builddir)/charmaps"
 
 # ensure that the distributed scripts are executable before distribution.
 # Could be necessary if checking out CVS.
