--- a/configure.ac
+++ b/configure.ac
@@ -199,20 +199,6 @@
 dnl Build static libraries by default. Furthermore, fortran shared libraries
 dnl are unsupported. Disallow a user from enabling both shared libraries and
 dnl fortran.
-if test "X${enable_shared}" != "Xyes"; then
-    enable_shared="no"
-fi
-
-if test "X${enable_shared}" = "Xyes"; then
-    if test "X${BUILD_FORTRAN}" = "Xyes"; then
-        AC_MSG_ERROR([Cannot build shared fortran libraries. Please configure with --disable-fortran flag.])
-    fi
-fi
-
-AC_PROG_INSTALL
-AC_PROG_LN_S
-
-AC_CHECK_PROGS([AR], [ar xar], [:], [$PATH])
 
 AC_CHECK_PROGS([YACC], ['bison -y' byacc yacc], [none], [])
 
@@ -389,7 +375,7 @@
 
 dnl ----------------------------------------------------------------------
 dnl Is the szip library present?
-AC_SUBST(USE_COMP_SZIP) USE_COMP_SZIP="no"
+AC_SUBST([USE_COMP_SZIP]) USE_COMP_SZIP="no"
 AC_ARG_WITH([szlib],
             [AC_HELP_STRING([--with-szlib=DIR],
                             [Use szlib library [default=no]])],,
@@ -702,7 +688,7 @@
 dnl ======================================================================
 dnl Libtool initialization
 dnl ======================================================================
-LT_INIT(dlopen disable-shared) 
+LT_INIT(dlopen) 
 LT_OUTPUT
 
 dnl ======================================================================
@@ -780,11 +766,6 @@
 dnl ------------------------------------------------------------------------
 dnl Check to see if libtool has enabled shared libraries. Set a conditional
 dnl as some Makefiles will build based on availability of shared libraries.
-if (./libtool --features | grep '^enable shared libraries' > /dev/null); then
-  enable_shared=yes
-else
-  enable_shared=no
-fi
 
 dnl ------------------------------------------------------------------------
 dnl Specify shared library extension the host machine should recognize.
@@ -814,10 +795,6 @@
 dnl Report anything stripped as a flag in CFLAGS and 
 dnl only the compiler in CC_VERSION.
 CC_NOFLAGS=`echo $CC | sed 's/ -.*//'`
-CFLAGS_TO_ADD=`echo $CC | grep - | sed 's/.* -/-/'`
-if test -n $CFLAGS_TO_ADD; then
-  CFLAGS="$CFLAGS_TO_ADD$CFLAGS"
-fi
 
 if `echo $CC_NOFLAGS | grep ^/ >/dev/null 2>&1`; then
   CC_VERSION="$CC_NOFLAGS"
@@ -841,26 +818,7 @@
 dnl Strip anything that looks like a flag off of $F77
 dnl Report anything stripped as a flag in RFLAGS and 
 dnl only the compiler in F77_VERSION.
-F77_NOFLAGS=`echo $F77 | sed 's/ -.*//'`
-FFLAGS_TO_ADD=`echo $F77 | grep - | sed 's/.* -/-/'`
-if test -n $FFLAGS_TO_ADD; then
- FFLAGS="$FFLAGS_TO_ADD$FFLAGS"
-fi
-
-if `echo $F77_NOFLAGS | grep ^/ >/dev/null 2>&1`; then
-  F77_VERSION="$F77_NOFLAGS"
-else
-  F77_VERSION="$F77";
-  for x in `echo $PATH | sed -e 's/:/ /g'`; do
-    if test -x $x/$F77_NOFLAGS; then
-      F77_VERSION="$x/$F77_NOFLAGS"
-      break
-    fi
-  done
-fi
-if test -n "$fc_version_info"; then
-  F77_VERSION="$F77_VERSION ( $fc_version_info)"
-fi
+F77_VERSION="$F77";
 
 
 dnl This part doesn't work yet since HDF4 config files do not contain
--- a/hdf/test/fortestF.f
+++ b/hdf/test/fortestF.f
@@ -203,7 +203,7 @@
 
 	if (cmd .NE. 'Test' .AND. cmd .NE. 'test') then
 	 print *, 'Unknown Command: ', cmd, param
-	 print *, 'Try one of "Skip", "Test", "Verbosity" or "Cleanup"'
+	 print *, 'Try one of Skip, Test, Verbosity or Cleanup'
 	 retcode = -1
 	 return
 	endif
--- a/mfhdf/libsrc/Makefile.am
+++ b/mfhdf/libsrc/Makefile.am
@@ -32,10 +32,11 @@
 else
 libmfhdf_la_SOURCES = $(CSOURCES)
 endif
+libmfhdf_la_LIBADD = $(top_builddir)/hdf/src/libdf.la
 
 if HDF_BUILD_XDR
 XDRLIB = ../xdr/libxdr.la
-libmfhdf_la_LIBADD = $(XDRLIB)
+libmfhdf_la_LIBADD += $(XDRLIB)
 endif
 
 if HDF_BUILD_NETCDF
@@ -50,11 +51,7 @@
 # fortran has been disabled in configure. The Fortran linker gets confused 
 # when shared libraries are enabled, so when Fortran is not enabled,
 # this replaces the Fortran link macro with the C link macro.
-if HDF_BUILD_FORTRAN
-    # do nothing
-else
-    F77LINK = $(LINK)
-endif
+
 
 #############################################################################
 ##                          And the cleanup                                ##
