--- MagickThumbnail/options.py.orig	2006-10-03 14:53:45.000000000 -0600
+++ MagickThumbnail/options.py	2006-10-03 14:55:56.000000000 -0600
@@ -8,7 +8,11 @@
 import rox.options
 import rox.OptionsBox
 
-rox.setup_app_options('MagickThumbnail')
+appname = 'MagickThumbnail'
+sitename = 'kulinarna.maczewski.dyndns.org'
+
+rox.choices.migrate(appname, sitename)
+rox.setup_app_options(appname, site=sitename)
 
 tsize=rox.options.Option('tsize', 128)
 frameit=rox.options.Option('frameit', 1)
