--- cnet-2.0.10/Makefile.orig	2006-03-06 20:03:14.000000000 +0100
+++ cnet-2.0.10/Makefile	2009-01-15 05:51:23.000000000 +0100
@@ -7,14 +7,14 @@
 # It is presumed that BINDIR and MANDIR exist, so you may need
 # to create them, and set their permissions, if necessary.
 #
-PREFIX		= /usr/local
+PREFIX		= /usr/
 #
 BINDIR		= $(PREFIX)/bin
-CNETDIR		= $(PREFIX)/cnetlib
-MANDIR		= $(PREFIX)/man/man1
+CNETDIR		= $(PREFIX)/lib/cnetlib
+MANDIR		= $(PREFIX)/share/man/man1
 MANEXT		= 1
 MANFILE		= $(MANDIR)/$(NAME).$(MANEXT)
-WEBDIR		= /home/httpd/html/cnet
+WEBDIR		= $(CNETDIR)/html
 
 # ---------------------------------------------------------------------
 
--- cnet-2.0.10/src/Makefile.common.orig	2004-05-10 21:26:53.000000000 +0200
+++ cnet-2.0.10/src/Makefile.common	2009-01-15 05:51:23.000000000 +0100
@@ -18,7 +18,6 @@
 
 $(NAME):	$(OBJ)
 		$(LD) $(LDFLAGS) -o $(NAME) $(OBJ) $(LIBDIRS) $(XLIBS) $(LIBS)
-		@$(STRIP) $(NAME)
 		@chmod 755 $(NAME)
 
 
--- cnet-2.0.10/src/Makefile.linux.orig	2004-05-10 21:27:19.000000000 +0200
+++ cnet-2.0.10/src/Makefile.linux	2009-01-15 05:51:23.000000000 +0100
@@ -26,12 +26,10 @@
 
 # Cannot use the -pedantic option if requiring 'long long' syntax.
 #ANSICC		= gcc -ansi -pedantic
-ANSICC		= gcc -ansi
+ANSICC		= $(CC) -ansi
 INCLUDES	= $(LIBELFINC)
 
-CFLAGS		= -O -Wall -fPIC $(INCLUDES) $(XINCLUDES)
-
-LD		= gcc -rdynamic
+LD		= $(CC) -rdynamic
 LIBDIRS		= -L/usr/X11R6/lib $(XLIBDIRS) $(LIBELFDIR)
 LIBS		= -lelf -ldl -lm
 STRIP		= strip
