--- pugixml-1.2/scripts/CMakeLists.txt
+++ pugixml-1.2/scripts/CMakeLists.txt
@@ -12,7 +12,7 @@
     add_library(pugixml STATIC ${SOURCES})
 endif()
 
-set_target_properties(pugixml PROPERTIES VERSION 1.2 SOVERSION 1.2)
+set_target_properties(pugixml PROPERTIES VERSION 1.2 SOVERSION 1)
 
-install(TARGETS pugixml LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
+install(TARGETS pugixml LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 install(FILES ${HEADERS} DESTINATION include)
