--- Config.orig	2005-06-02 20:28:35.000000000 +0200
+++ Config	2005-10-16 15:13:32.000000000 +0200
@@ -24,9 +24,9 @@
 #	Name of local configuration file (location is $HOME.)
 YAPS_LCFGFILE 	= .yapsrc
 #	Directory for installing the binary
-YAPS_BINDIR	= /usr/local/bin
+YAPS_BINDIR	= /usr/bin
 #	Directory for optional support files
-YAPS_LIBDIR	= /usr/local/lib/yaps
+YAPS_LIBDIR	= /usr/lib/yaps
 #	User/Group/Mode for yaps
 YAPS_USER	= bin
 YAPS_GROUP	= uucp
@@ -43,12 +43,12 @@
 ifdef	SLANG
 LDEFS	+= -DSCRIPT_SLANG
 LLIBS	+= -lslang
-MATH	= True
+#MATH	= True
 endif
 ifdef	LUA
 LDEFS	+= -DSCRIPT_LUA
-LLIBS	+= -llualib -llua
-MATH	= True
+LLIBS	+= `pkg-config --libs lua`
+#MATH	= True
 endif
 ifdef	MATH
 LLIBS	+= -lm
--- Makefile.orig	2005-06-02 20:33:27.000000000 +0200
+++ Makefile	2005-10-16 15:13:32.000000000 +0200
@@ -33,19 +33,19 @@
 	lynx -cfg=/dev/null -nolist -dump $< > $@
 
 install:	$(DSTFLE) $(CFGFILE)
-	if [ ! -d $(YAPS_LIBDIR) ]; then \
-		install -d -m 755 -o $(YAPS_USER) -g $(YAPS_GROUP) $(YAPS_LIBDIR) ; \
+	if [ ! -d $(DESTDIR)$(YAPS_LIBDIR) ]; then \
+		install -d -m 755 $(DESTDIR)$(YAPS_LIBDIR) ; \
 	fi
 	@if [ -f contrib/Makefile ]; then	\
 		$(MAKE) -C contrib install ;	\
 	fi
 
 $(DSTFLE):	yaps
-	install -o $(YAPS_USER) -g $(YAPS_GROUP) -m $(YAPS_MODE) -s yaps $@
+	install -m $(YAPS_MODE) -s yaps $(DESTDIR)$@
 
 $(CFGFILE):	yaps.rc
 	@if [ ! -f $@ ]; then \
-		install -o $(YAPS_RCUSER) -g $(YAPS_RCGROUP) -m $(YAPS_RCMODE) -s yaps.rc $@ ; \
+		install -m $(YAPS_RCMODE) -s yaps.rc $(DESTDIR)$@ ; \
 	fi
 
 clean:
--- yaps.rc.orig	2005-06-02 20:28:35.000000000 +0200
+++ yaps.rc	2005-10-16 15:13:32.000000000 +0200
@@ -9,26 +9,28 @@
 #	This is a comma seperated list of available services. This
 #	is required for autodetect the matching service for a
 #	pager-id
-services	D1,D2,E+
-+services	Telmi,Telmi-Fun,Telmi-Family,Telmi-Top,Telmi-Pro
-+services	Skyper,Quix
+services        D1-0160, D1-0170, D1-0171, D1-0175
++services       D2-0172, D2-0173, D2-0174, D2-0174
++services       E+0177, E+0178
++services       Telmi,Telmi-Fun,Telmi-Family,Telmi-Top,Telmi-Pro
++services       Skyper,Quix,Cityruf
 #
 #	This is the default caller-id. If the protocol/service allowes
 #	it, this is inserted in the message (and the receiver gets this
 #	as the sender's number)
-call-id		+1-(555)-1234
+#call-id		0123-4567890
 #
 #	If you forget to sign your message, you can automatically append
 #	this string to each message.
-signature	(It's me!)
+#signature	NAME
 #
 #	You can either switch the use of the signature here on for all
 #	services or in each service section for that particular service.
-use-signature	True
+#use-signature	True
 #
 #	To get more verbose output when running yaps increase this value.
 #	Currently 4 is the highest supported value.
-verbose		0
+verbose		4
 #
 #	If you'd like to log every sending session define this. It should
 #	point to a writeable file. If you install the software setgid, then
@@ -88,12 +90,12 @@
 #
 #	Most services limit the length of a message to a specific length.
 #	Enter the value here to check valid messages
-#max-size	160
+max-size	160
 
 #
 #	If a message is longer than max-size, you can set this to split
 #	it off into several single messages of allowed length
-#may-split	True
+may-split	True
 
 #
 #	If a message is longer than max-size, you can set this to truncate
@@ -107,7 +109,7 @@
 
 #
 #	If the service allows using your caller-id set this to True.
-#use-call-id	True
+use-call-id	True
 
 #
 #	If the service is reachable over a phonenumber where the pager-id
@@ -186,7 +188,8 @@
 
 #
 #	Set this to the modem device you wish to use
-#device		/dev/modem
+device		/dev/ttyS0
+#device		/dev/ttyI5
 #device		/dev/cua1
 #device		/dev/cua/a
 
@@ -194,7 +197,7 @@
 #	Set this to the prefix of the lockfile use for the device. There
 #	are several places, and two common formats:
 #lock-prefix	/usr/spool/uucp/LCK..
-#lock-prefix	/var/spool/lock/LCK..
+lock-prefix	/var/spool/lock/LCK..
 #lock-prefix	/var/spool/locks/LK.
 
 #
@@ -223,7 +226,57 @@
 #
 #	D1 (by Deutsche Telekom)
 #
-[D1]
+[D1-0160]
+protocol	tap
+valid-pid	0160|0049160|49160
+change-pid	0160
+convert		*no-8bit,cv-d1
+cv-d1		{
+	Ä	[
+	Ö	\
+	Ü	~
+	ä	{
+	ö	|
+	ü	~
+	ß	^]
+}
++cost		timetable=Wk0800-1800=5.7;Wk1800-0800=11.4;Ss=11.4
+phone		01712092522
+# use this number for isdn-dialin:
+# phone           01712521001
+max-size	160
+may-split	True
+use-call-id	False
+tap-old		True
+#
+#----------------------------------------------------------
+#
+[D1-0170]
+protocol	tap
+valid-pid	0170|0049170|49170
+change-pid	0170
+convert		*no-8bit,cv-d1
+cv-d1		{
+	Ä	[
+	Ö	\
+	Ü	~
+	ä	{
+	ö	|
+	ü	~
+	ß	^]
+}
++cost		timetable=Wk0800-1800=5.7;Wk1800-0800=11.4;Ss=11.4
+phone		01712092522
+# use this number for isdn-dialin:
+# phone           01712521001
+max-size	160
+may-split	True
+use-call-id	False
+tap-old		True
+#
+#----------------------------------------------------------
+#
+[D1-0171]
 protocol	tap
 valid-pid	0171|0049171|49171
 change-pid	0171
@@ -239,21 +292,100 @@
 }
 +cost		timetable=Wk0800-1800=5.7;Wk1800-0800=11.4;Ss=11.4
 phone		01712092522
+# use this number for isdn-dialin:
+# phone           01712521001
+max-size	160
+may-split	True
+use-call-id	False
+tap-old		True
+#
+#----------------------------------------------------------
+#
+[D1-0175]
+protocol	tap
+valid-pid	0175|0049175|49175
+change-pid	0175
+convert		*no-8bit,cv-d1
+cv-d1		{
+	Ä	[
+	Ö	\
+	Ü	~
+	ä	{
+	ö	|
+	ü	~
+	ß	^]
+}
++cost		timetable=Wk0800-1800=5.7;Wk1800-0800=11.4;Ss=11.4
+phone		01712092522
+# use this number for isdn-dialin:
+# phone           01712521001
 max-size	160
 may-split	True
 use-call-id	False
 tap-old		True
 #
+#----------------------------------------------------------
+#
+
+
+
+#
 #	D2 (by Mannesmann Mobilfunk)
 #
-[D2]
-modem		standard-d2
+[D2-0162]
+modem		standard
+protocol	ucp
+valid-pid	0162|0049162|49162
+change-pid	0162
+convert		*no-8bit,cv-default
++cost		timetable=Wk0800-1800=5.7;Wk1800-0800=11.4;Ss=11.4
+# phone		01722278020
+# use this number for isdn-dialin:
+phone           01722278010
+max-size	160
+may-split	True
+use-call-id	True
+can-delay	True
+can-expire	True
+can-rds		True
+check-call-id	>1111111111111111<
+check-pager-id	>1111111111111111<
+ucp-extend	True
+#
+#----------------------------------------------------------
+#
+[D2-0172]
+modem		standard
 protocol	ucp
 valid-pid	0172|0049172|49172
 change-pid	0172
 convert		*no-8bit,cv-default
 +cost		timetable=Wk0800-1800=5.7;Wk1800-0800=11.4;Ss=11.4
-phone		01722278020
+# phone		01722278020
+# use this number for isdn-dialin:
+phone           01722278010
+max-size	160
+may-split	True
+use-call-id	True
+can-delay	True
+can-expire	True
+can-rds		True
+check-call-id	>1111111111111111<
+check-pager-id	>1111111111111111<
+ucp-extend	True
+#
+#----------------------------------------------------------
+#
+[D2-0173]
+modem		standard
+protocol	ucp
+valid-pid	0173|0049173|49173|173
+change-pid	0173
+convert		*no-8bit,cv-default
++cost		timetable=Wk0800-1800=5.7;Wk1800-0800=11.4;Ss=11.4
+# phone		01722278020
+# use this number for isdn-dialin:
+phone           01722278010
 max-size	160
 may-split	True
 use-call-id	True
@@ -264,10 +396,39 @@
 check-pager-id	>1111111111111111<
 ucp-extend	True
 #
+#----------------------------------------------------------
+#
+[D2-0174]
+modem		standard
+protocol	ucp
+valid-pid	0174|0049174|49174|174
+change-pid	0174
+convert		*no-8bit,cv-default
++cost		timetable=Wk0800-1800=5.7;Wk1800-0800=11.4;Ss=11.4
+# phone		01722278020
+# use this number for isdn-dialin:
+phone           01722278010
+max-size	160
+may-split	True
+use-call-id	True
+can-delay	True
+can-expire	True
+can-rds		True
+check-call-id	>1111111111111111<
+check-pager-id	>1111111111111111<
+ucp-extend	True
+#
+#----------------------------------------------------------
+#
+
+
+
+#
+#
 #	E+ (by ???)
 #	!! UNTESTED !!
 #
-[E+]
+[E+0177]
 protocol	tap
 valid-pid	0177|0049177|49177
 change-pid	0177
@@ -278,6 +439,22 @@
 may-split	True
 use-call-id	False
 #
+#----------------------------------------------------------
+#
+[E+0178]
+protocol        tap
+valid-pid       0178|0049178|49178|0178
+#change-pid     0178
+convert         *no-8bit,cv-default
++cost           timetable=Wk0800-1800=5.7;Wk1800-0800=11.4;Ss=11.4
+phone           01771167
+max-size        160
+may-split       True
+use-call-id     False
+#
+#----------------------------------------------------------
+#
+#
 #	Telmi, Telmi-* (by Deutsche Funkruf)
 #	!! ONLY TESTED FOR TELMI-FUN !!
 #
@@ -326,6 +503,20 @@
 phone		016593
 max-size	80
 use-call-id	True
+#
+#		Cityruf
+#	!!Used for emessage.de also!!
+#
+[Cityruf]
+protocol       tap
+valid-pid      0168|0164
+change-pid     -
+convert        *no-8bit,cv-default,cv-extend
+phone          016902
+max-size       80
+use-call-id            False
+insert-pager-id        False
+truncate       True
 
 #########################################
 #	modem section			#
@@ -335,10 +526,10 @@
 #	Locking
 #
 #	Older Unix versions
-lock-prefix	/usr/spool/uucp/LCK..
+#lock-prefix	/usr/spool/uucp/LCK..
 #	Newer Unix versions
 #lock-prefix	/var/spool/uucp/LCK..
-#lock-prefix	/var/lock/LCK..
+lock-prefix	/var/lock/LCK..
 #	SysV4
 #lock-prefix	/var/spool/locks/LK.
 #lock-method	sysv4
@@ -346,22 +537,38 @@
 #	Device
 #
 #	Typical a symbolic link to the real device
-device		/dev/modem
+device		/dev/ttyS0
 #	"normal" call-out device
 #device		/dev/cua1
 #	Solaris convention
 #device		/dev/cua/a
+# use the following device for isdn-dialout:
+#device		/dev/ttyI6
+# use the following device for CAPI
+# syntax: CAPI/<controller>/<MSN>
+#device		CAPI/1/12345
 
 #
+#
 #	Modem talk
 #
-init		\\r !200D ATZ\r <OK ATE0Q0V1\r <OK
-dial		ATD%L\r <60CONNECT|OK|BUSY|NO\sDIALTONE|NO\sCARRIER
-reset		ATZ\r <OK
+#init		\\r !200D AT&F2\r <OK ATE0Q0V1M1L1X3\r <OK
+init		\\r !200D AT&F2\r <OK ATV1M1L1X3\r <OK
+
+# use the following init-string for isdn:
+# and change MSN to your MSN (what else...:-))
+#init            \\r !200D ATZ\r <OK AT&E9879356\r <OK
+
+dial		ATDT0%L\r <60CONNECT|OK|BUSY|NO\sDIALTONE|NO\sCARRIER
+reset		ATH\r <OK
 timeout		10
+
+
 #	Customizing an existing modem entry
-[standard-d2] standard
-local-init	AT\%C2\r <OK
+#[standard-d2] standard
+#local-init	AT\%C2\r <OK
+
+
 #
 #	alias section
 #
