diff -Naur wavplay-1.4.orig/wavplay.h wavplay-1.4/wavplay.h
--- wavplay-1.4.orig/wavplay.h	1999-12-04 01:06:42.000000000 +0100
+++ wavplay-1.4/wavplay.h	2004-09-06 19:57:03.741888416 +0200
@@ -500,7 +500,7 @@
 extern int UnlockDSP(int ipc,int playrecx,ErrFunc erf);
 
 extern WAVFILE *WavOpenForRead(const char *Pathname,ErrFunc erf);
-extern WAVFILE *WavOpenForWrite(const char *Pathname,OprMode m,UInt32 sample_rate,UInt16 bits,UInt32 samples,ErrFunc erf);
+extern WAVFILE *WavOpenForWrite(const char *Pathname,Chan chmode,UInt32 sample_rate,UInt16 bits, UInt32 samples,ErrFunc erf);
 extern void WavReadOverrides(WAVFILE *wfile,WavPlayOpts *wavopts);
 extern int WavClose(WAVFILE *wfile,ErrFunc erf);
 
