--- criu-0.6/Makefile
+++ criu-0.6/Makefile
@@ -87,9 +87,6 @@
 
 ifeq ($(DEBUG),1)
 	DEFINES += -DCR_DEBUG
-	CFLAGS	+= -O0 -ggdb3
-else
-	CFLAGS	+= -O2
 endif
 
 CFLAGS		+= $(WARNINGS) $(DEFINES)
--- criu-0.6/test/zdtm/lib/Makefile
+++ criu-0.6/test/zdtm/lib/Makefile
@@ -1,6 +1,6 @@
 include ../Makefile.inc
 
-CFLAGS	= -g -O2 -Wall -Werror -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=0
+CFLAGS	= -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=0
 
 LIBDIR	= .
 LIB	= libzdtmtst.a
--- criu-0.6/test/zdtm/live/static/Makefile
+++ criu-0.6/test/zdtm/live/static/Makefile
@@ -3,7 +3,7 @@
 LIBDIR	= ../../lib
 LIB	= $(LIBDIR)/libzdtmtst.a
 override CPPFLAGS += -I$(LIBDIR)
-CFLAGS	= -g -O2 -Wall -Werror -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=0
+CFLAGS	= -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=0
 
 TST_NOFILE	=				\
 		busyloop00			\
--- criu-0.6/test/zdtm/live/streaming/Makefile
+++ criu-0.6/test/zdtm/live/streaming/Makefile
@@ -3,7 +3,7 @@
 LIBDIR	= ../../lib
 LIB	= $(LIBDIR)/libzdtmtst.a
 override CPPFLAGS += -I$(LIBDIR)
-CFLAGS	= -g -O2 -Wall -Werror -fno-strict-aliasing
+CFLAGS	= -Wall -fno-strict-aliasing
 
 TST_NOFILE	=	\
 		pipe_loop00	\
--- criu-0.6/test/zdtm/live/transition/Makefile
+++ criu-0.6/test/zdtm/live/transition/Makefile
@@ -3,7 +3,7 @@
 LIBDIR	= ../../lib
 LIB	= $(LIBDIR)/libzdtmtst.a
 override CPPFLAGS += -I$(LIBDIR)
-CFLAGS	= -g -O2 -Wall -Werror
+CFLAGS	= -Wall
 
 TST_NOFILE	=	\
 		ipc	 	\
