--- criu-1.2/Makefile
+++ criu-1.2/Makefile
@@ -256,6 +256,7 @@
 	$(Q) mkdir -p $(DESTDIR)$(SYSTEMDUNITDIR)
 	$(Q) install -m 644 scripts/sd/criu.socket $(DESTDIR)$(SYSTEMDUNITDIR)
 	$(Q) install -m 644 scripts/sd/criu.service $(DESTDIR)$(SYSTEMDUNITDIR)
+	$(Q) mkdir -p $(DESTDIR)$(LOGROTATEDIR)
 	$(Q) install -m 644 scripts/logrotate.d/criu-service $(DESTDIR)$(LOGROTATEDIR)
 
 install-man:
--- criu-1.2/Makefile.inc
+++ criu-1.2/Makefile.inc
@@ -15,15 +15,9 @@
 SBINDIR		:= $(PREFIX)/sbin
 MANDIR		:= $(PREFIX)/share/man
 SYSTEMDUNITDIR	:= $(PREFIX)/lib/systemd/system/
-LOGROTATEDIR	:= $(PREFIX)/etc/logrotate.d/
+LOGROTATEDIR	:= $(SYSCONFDIR)/logrotate.d/
 LIBDIR		:= $(PREFIX)/lib
-# For recent Debian/Ubuntu with multiarch support
-DEB_HOST_MULTIARCH ?= $(shell dpkg-architecture \
-			-qDEB_HOST_MULTIARCH 2>/dev/null)
-ifneq "$(DEB_HOST_MULTIARCH)" ""
-LIBDIR		:= $(PREFIX)/lib/$(DEB_HOST_MULTIARCH)
-# For most other systems
-else ifeq "$(shell uname -m)" "x86_64"
+ifeq "$(shell uname -m)" "x86_64"
 LIBDIR		:= $(PREFIX)/lib64
 endif
 
