--- etc/shtool.orig	2004-02-15 15:58:06.714608714 +0100
+++ etc/shtool	2004-02-15 15:58:45.627450465 +0100
@@ -532,13 +532,13 @@
     if [ ".$opt_e" = .yes -a ".`echo $text | egrep '%d'`" != . ]; then
         if [ ".$domainname" = . ]; then
             if [ -f /etc/resolv.conf ]; then
-                domainname="`egrep '^[ 	]*domain' /etc/resolv.conf | head -1 |\
+                domainname="`egrep '^[ 	]*domain' /etc/resolv.conf | head -n 1 |\
                              sed -e 's/.*domain//' \
                                  -e 's/^[ 	]*//' -e 's/^ *//' -e 's/^	*//' \
                                  -e 's/^\.//' -e 's/^/./' |\
                              awk '{ printf("%s", $1); }'`"
                 if [ ".$domainname" = . ]; then
-                    domainname="`egrep '^[ 	]*search' /etc/resolv.conf | head -1 |\
+                    domainname="`egrep '^[ 	]*search' /etc/resolv.conf | head -n 1 |\
                                  sed -e 's/.*search//' \
                                      -e 's/^[ 	]*//' -e 's/^ *//' -e 's/^	*//' \
                                      -e 's/ .*//' -e 's/	.*//' \
@@ -2100,7 +2100,7 @@
         if [ -r "$FILE" ]; then
             #   grep out current information
             id=`grep 'Version [0-9]*.[0-9]*[.abps][0-9]* ([0-9]*-[a-zA-Z]*-[0-9]*)' $FILE | \
-                head -1 | \
+                head -n 1 | \
                 sed -e 's%.*Version \([0-9]*\)\.\([0-9]*\)\([.abps]\)\([0-9]*\) (\([0-9]*-[a-zA-Z]*-[0-9]*\)).*%\1:\2:\3:\4:\5%'`
             version=`echo $id | awk -F: '{ print $1 }'`
             revision=`echo $id | awk -F: '{ print $2 }'`
