diff -Nru gcdial-0.7b.orig/src/rcfile.c gcdial-0.7b/src/rcfile.c
--- gcdial-0.7b.orig/src/rcfile.c	2000-04-16 12:44:22.000000000 +0200
+++ gcdial-0.7b/src/rcfile.c	2009-06-20 11:55:50.000000000 +0200
@@ -19,7 +19,7 @@
 
 static int get_settings(FILE *fp, char **server, char **port, char **username,
 		char **password);
-static int getline(FILE *fp, char **argument);
+static int my_getline(FILE *fp, char **argument);
 static int getline_keyword(const char *buf);
 static int check_rcfile_perms(void);
 
@@ -99,7 +99,7 @@
 	int line = 0;
 	int error = 0;
 
-	if (RC_FATAL==(ret = getline(fp, &arg)))
+	if (RC_FATAL==(ret = my_getline(fp, &arg)))
 		return -1;
 	while (ret!=RC_EOF) {
 		++line;
@@ -147,7 +147,7 @@
 				break;
 			} /* switch(ret) */
 		} /* arg !NULL */
-		if (RC_FATAL==(ret = getline(fp, &arg)))
+		if (RC_FATAL==(ret = my_getline(fp, &arg)))
 			return -1;
 	} /* while(ret!=RC_EOF) */
 	if (error)
@@ -155,7 +155,7 @@
 	return 0;
 }
 
-int getline(FILE *fp, char **argument)
+int my_getline(FILE *fp, char **argument)
 {
 	char *buf;
 	int com_len;
