--- gcompris-12.05.orig/Makefile.am
+++ gcompris-12.05/Makefile.am
@@ -17,7 +17,7 @@
 # endif
 
 
-icondir=$(datadir)/pixmaps
+icondir=/usr/share/pixmaps
 icon_DATA = gcompris.png gcompris-edit.png
 
 if BUILD_MINGW32
@@ -25,7 +25,7 @@
 Applications_in_files =
 Applications_DATA =
 else
-Applicationsdir = $(datadir)/applications/
+Applicationsdir = /usr/share/applications/
 Applications_in_files = gcompris.desktop.in gcompris-edit.desktop.in
 Applications_DATA = gcompris.desktop gcompris-edit.desktop
 endif
--- gcompris-12.05.orig/configure.ac
+++ gcompris-12.05/configure.ac
@@ -204,10 +204,8 @@
           USE_DEBUG="$enableval", USE_DEBUG="no")
 if test "x$USE_DEBUG" == "xyes" ; then
     AC_DEFINE_UNQUOTED(DEBUG, 1, "Enable debug messages.")
-    CFLAGS="$CFLAGS $OSX_CFLAGS -Wall -g"
-else
-    CFLAGS="$CFLAGS $OSX_CFLAGS -Wall -O2"
 fi
+CFLAGS="$CFLAGS $OSX_CFLAGS -Wall"
 
 dnl disable activation code
 AC_ARG_ENABLE(disable-activation-code,
@@ -338,28 +336,28 @@
   fi
 else
   dnl Set PACKAGE_LOCALE_DIR in config.h.
-  AC_DEFINE_UNQUOTED(PACKAGE_LOCALE_DIR, "${myprefix}/${DATADIRNAME}/locale", [GCompris locale directory])
+  AC_DEFINE_UNQUOTED(PACKAGE_LOCALE_DIR, "/usr/share/locale", [GCompris locale directory])
 
   dnl Set PACKAGE_DATA_DIR
   PACKAGE_DATA_DIR="boards"
   AC_SUBST(PACKAGE_DATA_DIR)
-  AC_DEFINE_UNQUOTED(PACKAGE_DATA_DIR, "${myprefix}/${DATADIRNAME}", [GCompris data directory])
+  AC_DEFINE_UNQUOTED(PACKAGE_DATA_DIR, "${datadir}", [GCompris data directory])
 
   dnl Plugin Directory
   my_libdir=`eval echo "${libdir}"`
-  AC_DEFINE_UNQUOTED(PACKAGE_CLIB_DIR, "${my_libdir}", [Gcompris plugins directory])
+  AC_DEFINE_UNQUOTED(PACKAGE_CLIB_DIR, "${libdir}", [Gcompris plugins directory])
   plugindir=$libdir/gcompris
   AC_SUBST(plugindir)
 
   dnl Python plugin Directory
-  PYTHON_PLUGIN_DIR="${myprefix}/${DATADIRNAME}/${PACKAGE}/python"
+  PYTHON_PLUGIN_DIR="${datadir}/${PACKAGE}/python"
   AC_SUBST(PYTHON_PLUGIN_DIR)
 
   dnl System GCompris config directory
   AC_DEFINE_UNQUOTED(SYSTEM_CONFIG_DIR, "/etc", [System GCompris config directory])
 
   dnl Test for gnuchess
-  AC_PATH_PROGS(GNUCHESS, gnuchess gnome-gnuchess, no, [/usr/bin:/usr/games:/usr/local/bin:$PATH])
+  AC_PATH_PROGS(GNUCHESS, gnuchess gnome-gnuchess, no, [/usr/games/bin:/usr/bin:/usr/games:/usr/local/bin:$PATH])
   if test x$GNUCHESS = xno; then
     AC_MSG_ERROR([Couldn't find gnuchess, please install the gnuchess package version]
                  [ 5 or above or define the environment variable GNUCHESS.])
--- gcompris-12.05.orig/docs/C/Makefile.am
+++ gcompris-12.05/docs/C/Makefile.am
@@ -10,7 +10,7 @@
 	     overview.dia \
 	     $(man_MANS)
 
-gcompris_helpdir = $(datadir)/gnome/help/$(PACKAGE)/C
+gcompris_helpdir = /usr/share/gnome/help/$(PACKAGE)/C
 
 all: $(doc)
 
--- gcompris-12.05.orig/docs/eu/Makefile.am
+++ gcompris-12.05/docs/eu/Makefile.am
@@ -8,7 +8,7 @@
 	     overview.dia \
 	     gcompris.texi
 
-gcompris_helpdir = $(datadir)/gnome/help/$(PACKAGE)/eu
+gcompris_helpdir = /usr/share/gnome/help/$(PACKAGE)/eu
 
 all: $(doc)
 
--- gcompris-12.05.orig/docs/fr/Makefile.am
+++ gcompris-12.05/docs/fr/Makefile.am
@@ -8,7 +8,7 @@
 	     overview.dia \
 	     gcompris.texi
 
-gcompris_helpdir = $(datadir)/gnome/help/$(PACKAGE)/fr
+gcompris_helpdir = /usr/share/gnome/help/$(PACKAGE)/fr
 
 all: $(doc)
 
--- gcompris-12.05.orig/po/Makefile.in.in
+++ gcompris-12.05/po/Makefile.in.in
@@ -34,7 +34,7 @@
 datarootdir = @datarootdir@
 libdir = @libdir@
 DATADIRNAME = @DATADIRNAME@
-itlocaledir = $(prefix)/$(DATADIRNAME)/locale
+itlocaledir = /usr/share/locale
 subdir = po
 install_sh = @install_sh@
 # Automake >= 1.8 provides @mkdir_p@.
--- gcompris-12.05.orig/src/gcompris/gcompris.c
+++ gcompris-12.05/src/gcompris/gcompris.c
@@ -1341,7 +1341,7 @@
     properties->package_plugin_dir = g_strconcat(pkg_clib_dir, "/gcompris", NULL);
     properties->package_python_plugin_dir = g_strconcat(pkg_data_dir, "/gcompris/python",
 							NULL);
-    properties->system_icon_dir = g_strconcat(pkg_data_dir, "/pixmaps", NULL);
+    properties->system_icon_dir = g_strconcat("/usr/share", "/pixmaps", NULL);
     properties->menu_dir = g_strdup(properties->package_data_dir);
     g_free(pkg_data_dir);
     g_free(pkg_clib_dir);
