--- src/logfile/adif.c.orig	2009-05-17 13:09:29.000000000 +0200
+++ src/logfile/adif.c	2009-05-17 15:28:32.000000000 +0200
@@ -449,6 +449,25 @@
 			}
 		}
 
+		/* export all QSL-info's as 'Y' for 'y', 'Y' or 'x', 
+		 * drop entry for 'n' or 'N' and 
+		 * as 'V' (invalid) for all other values of nonzero length
+		 */
+		else if (fld == QSLOUT || fld == QSLIN)
+		{
+			qfield = q[fld];	
+			if (qfield_len == 1 && (qfield[0] == 'Y' || qfield[0] == 'y' || qfield[0] == 'X' || qfield[0] == 'x'))
+				qfield[0] = 'Y';
+			else if (qfield_len == 1 && (qfield[0] == 'N' || qfield[0] == 'n'))
+				continue;
+			else
+			{
+				qfield[0] = 'I';
+				qfield[1] = '\0';
+				qfield_len = 1;
+			}
+		}
+
 		else if (fld == RST || fld == MYRST)
 		{
 			if (mode == MODE_SSB || mode == MODE_AM || mode == MODE_FM)
