diff --git a/Makefile b/Makefile
index de0aa1a..abf042f 100644
--- a/Makefile
+++ b/Makefile
@@ -86,7 +86,7 @@ endif
 ARCH_TYPE = $(shell arch)
 
 LLVM_CXXFLAGS=$(shell $(LLVM_CONFIG) --cppflags)
-LLVM_VERSION=LLVM_$(shell $(LLVM_CONFIG) --version | sed -e s/\\./_/ -e s/svn// -e s/\.0//)
+LLVM_VERSION=LLVM_$(shell $(LLVM_CONFIG) --version | sed -e 's/svn//' -e 's/\./_/' -e 's/\..*//')
 LLVM_VERSION_DEF=-D$(LLVM_VERSION)
 
 LLVM_COMPONENTS = engine ipo bitreader bitwriter instrumentation linker
