diff -Naur ispc-ispc-c6ec130.orig/ispc.cpp ispc-ispc-c6ec130/ispc.cpp
--- ispc-ispc-c6ec130.orig/ispc.cpp	2012-06-29 09:38:38.000000000 -0600
+++ ispc-ispc-c6ec130/ispc.cpp	2012-07-20 13:24:38.000000000 -0600
@@ -40,6 +40,7 @@
 #include "util.h"
 #include "llvmutil.h"
 #include <stdio.h>
+#include <unistd.h>
 #ifdef ISPC_IS_WINDOWS
 #include <windows.h>
 #include <direct.h>
diff -Naur ispc-ispc-c6ec130.orig/lex.ll ispc-ispc-c6ec130/lex.ll
--- ispc-ispc-c6ec130.orig/lex.ll	2012-06-29 09:38:38.000000000 -0600
+++ ispc-ispc-c6ec130/lex.ll	2012-07-20 13:37:15.000000000 -0600
@@ -41,6 +41,7 @@
 #include "parse.hh"
 #include <stdlib.h>
 #include <stdint.h>
+#include <unistd.h>
 
 static uint64_t lParseBinary(const char *ptr, SourcePos pos, char **endPtr);
 static int lParseInteger(bool dotdotdot);
diff -Naur ispc-ispc-c6ec130.orig/main.cpp ispc-ispc-c6ec130/main.cpp
--- ispc-ispc-c6ec130.orig/main.cpp	2012-06-29 09:38:38.000000000 -0600
+++ ispc-ispc-c6ec130/main.cpp	2012-07-20 13:25:18.000000000 -0600
@@ -41,6 +41,9 @@
 #include "type.h"
 #include <stdio.h>
 #include <stdlib.h>
+#include <sys/types.h>
+#include <unistd.h>
+
 #ifdef ISPC_IS_WINDOWS
   #include <time.h>
 #endif // ISPC_IS_WINDOWS
