Index: nagios-plugins-1.4.14/plugins/check_ldap.c
===================================================================
--- nagios-plugins-1.4.14.orig/plugins/check_ldap.c
+++ nagios-plugins-1.4.14/plugins/check_ldap.c
@@ -38,6 +38,10 @@ const char *email = "nagiosplug-devel@li
 #include <lber.h>
 #include <ldap.h>
 
+ldap_init LDAP_P(( /* deprecated, use ldap_create or ldap_initialize */
+	LDAP_CONST char *host,
+	int port ));
+
 enum {
 	UNDEFINED = 0,
 #ifdef HAVE_LDAP_SET_OPTION
