diff -Naur nagios-plugins-1.4.10.orig/configure.in nagios-plugins-1.4.10/configure.in
--- nagios-plugins-1.4.10.orig/configure.in	2007-09-29 02:44:36.000000000 +0200
+++ nagios-plugins-1.4.10/configure.in	2007-09-29 11:20:19.000000000 +0200
@@ -210,26 +210,6 @@
 LIBS="$_SAVEDLIBS"
 CPPFLAGS="$_SAVEDCPPFLAGS"
 
-dnl Check for radius libraries
-_SAVEDLIBS="$LIBS"
-AC_CHECK_LIB(radiusclient,rc_read_config)
-if test "$ac_cv_lib_radiusclient_rc_read_config" = "yes"; then
-  EXTRAS="$EXTRAS check_radius"
-	RADIUSLIBS="-lradiusclient"
-  AC_SUBST(RADIUSLIBS)
-else
-  AC_CHECK_LIB(radiusclient-ng,rc_read_config)
-  if test "$ac_cv_lib_radiusclient_ng_rc_read_config" = "yes"; then
-    EXTRAS="$EXTRAS check_radius"
-  	  RADIUSLIBS="-lradiusclient-ng"
-    AC_SUBST(RADIUSLIBS)
-  else
-    AC_MSG_WARN([Skipping radius plugin])
-    AC_MSG_WARN([install radius libs to compile this plugin (see REQUIREMENTS).])
-  fi
-fi
-LIBS="$_SAVEDLIBS"
-
 dnl Check for LDAP libraries
 _SAVEDLIBS="$LIBS"
 AC_CHECK_LIB(ldap,main,,,-llber)
