Index: nagios-plugins-1.4.15/configure.in
===================================================================
--- nagios-plugins-1.4.15.orig/configure.in
+++ nagios-plugins-1.4.15/configure.in
@@ -257,23 +257,6 @@ fi
 LIBS="$_SAVEDLIBS"
 CPPFLAGS="$_SAVEDCPPFLAGS"
 
-dnl Check for LDAP libraries
-_SAVEDLIBS="$LIBS"
-AC_CHECK_LIB(ldap,main,,,-llber)
-if test "$ac_cv_lib_ldap_main" = "yes"; then
-  LDAPLIBS="-lldap -llber"\
-  LDAPINCLUDE="-I/usr/include/ldap"
-  AC_SUBST(LDAPLIBS)
-  AC_SUBST(LDAPINCLUDE)
-  AC_CHECK_FUNCS(ldap_set_option)
-  EXTRAS="$EXTRAS check_ldap"
-	AC_CHECK_FUNCS(ldap_init ldap_set_option ldap_get_option ldap_start_tls_s)
-else
-  AC_MSG_WARN([Skipping LDAP plugin])
-  AC_MSG_WARN([install LDAP libs to compile this plugin (see REQUIREMENTS).])
-fi
-LIBS="$_SAVEDLIBS"
-
 dnl Check for headers used by check_ide_smart
 AC_CHECK_HEADER(linux/hdreg.h, FOUNDINCLUDE=yes, FOUNDINCLUDE=no)
 if test "$FOUNDINCLUDE" = "yes" ; then
@@ -305,6 +288,8 @@ fi
 AC_ARG_WITH([ipv6],
 	[AS_HELP_STRING([--with-ipv6], [support IPv6 @<:@default=check@:>@])],
 	[], [with_ipv6=check])
+AC_ARG_WITH([ldap],[AS_HELP_STRING([--with-ldap], [support Openldap @<:@default=check@:>@])],
+	[], [with_ldap=check])
 
 dnl Check for AF_INET6 support - unistd.h required for Darwin
 if test "$with_ipv6" != "no"; then
@@ -333,6 +318,24 @@ if test "$with_ipv6" != "no"; then
 	with_ipv6="$np_cv_sys_ipv6"
 fi
 
+dnl Check for LDAP libraries
+if test "$with_ldap" != "no"; then
+
+	_SAVEDLIBS="$LIBS"
+	AC_CHECK_LIB(ldap,main,,,-llber)
+	if test "$ac_cv_lib_ldap_main" = "yes"; then
+		LDAPLIBS="-lldap -llber"\
+		LDAPINCLUDE="-I/usr/include/ldap"
+		AC_SUBST(LDAPLIBS)
+		AC_SUBST(LDAPINCLUDE)
+		AC_CHECK_FUNCS(ldap_set_option)
+		EXTRAS="$EXTRAS check_ldap"
+		AC_CHECK_FUNCS(ldap_init ldap_set_option ldap_get_option ldap_start_tls_s)
+	else
+		AC_MSG_ERROR([Cannot find openldap libraries.])
+	fi
+fi
+LIBS="$_SAVEDLIBS"
 
 dnl Checks for Kerberos. Must come before openssl checks for Redhat EL 3
 AC_CHECK_HEADERS(krb5.h,FOUNDINCLUDE=yes,FOUNDINCLUDE=no)
