--- configure.in.orig	2009-02-26 14:05:55.000000000 +0000
+++ configure.in	2009-02-26 14:07:02.000000000 +0000
@@ -94,6 +94,8 @@
 PKG_ARCH=`uname -p`
 REV_DATESTAMP=`date '+%Y.%m.%d.%H.%M'`
 REV_TIMESTAMP=`date '+%Y%m%d%H%M%S'`
+first_ip=`/sbin/ifconfig | egrep "inet addr:" | sed -n '1p' | sed 's/ Bcast.*$//' | sed 's/^\s*inet addr://'`
+
 
 AC_SUBST(PKG_ARCH)
 AC_SUBST(REV_DATESTAMP)
@@ -869,14 +871,14 @@
 	fi
 
 elif [[ "z$ac_cv_uname_s" = "zUnixWare" ]] && \
-	$PATH_TO_PING -n -s 127.0.0.1 56 1 2>/dev/null | \
+	$PATH_TO_PING -n -s $first_ip 56 1 2>/dev/null | \
 	egrep -i "^round-trip|^rtt" >/dev/null
 then
         with_ping_command="$PATH_TO_PING -n -U -c %d %s"
 	ac_cv_ping_packets_first=yes
 	AC_MSG_RESULT([$with_ping_command])
 
-elif $PATH_TO_PING -n -U -w 10 -c 1 127.0.0.1 2>/dev/null | \
+elif $PATH_TO_PING -n -U -w 10 -c 1 $first_ip 2>/dev/null | \
 	egrep -i "^round-trip|^rtt" >/dev/null
 then
         with_ping_command="$PATH_TO_PING -n -U -w %d -c %d %s"
@@ -884,52 +886,52 @@
   ac_cv_ping_has_timeout=yes
 	AC_MSG_RESULT([$with_ping_command])
 
-elif $PATH_TO_PING -n -U -c 1 127.0.0.1 2>/dev/null | \
+elif $PATH_TO_PING -n -U -c 1 $first_ip 2>/dev/null | \
 	egrep -i "^round-trip|^rtt" >/dev/null
 then
         with_ping_command="$PATH_TO_PING -n -U -c %d %s"
 	ac_cv_ping_packets_first=yes
 	AC_MSG_RESULT([$with_ping_command])
 
-elif $PATH_TO_PING -n -c 1 127.0.0.1 2>/dev/null | \
+elif $PATH_TO_PING -n -c 1 $first_ip 2>/dev/null | \
 	egrep -i "^round-trip|^rtt" >/dev/null
 then
 	with_ping_command="$PATH_TO_PING -n -c %d %s"
 	ac_cv_ping_packets_first=yes
 	AC_MSG_RESULT([$with_ping_command])
 
-elif $PATH_TO_PING -n 127.0.0.1 -c 1 2>/dev/null | \
+elif $PATH_TO_PING -n $first_ip -c 1 2>/dev/null | \
 	egrep -i "^round-trip|^rtt" >/dev/null
 then
 	with_ping_command="$PATH_TO_PING -n %s -c %d"
 	AC_MSG_RESULT([$with_ping_command])
 
-elif $PATH_TO_PING 127.0.0.1 -n 1 2>/dev/null | \
+elif $PATH_TO_PING $first_ip -n 1 2>/dev/null | \
 	egrep -i "^round-trip|^rtt" >/dev/null
 then
 	with_ping_command="$PATH_TO_PING %s -n %d"
 	AC_MSG_RESULT([$with_ping_command])
 
-elif $PATH_TO_PING -n -s 127.0.0.1 56 1 2>/dev/null | \
+elif $PATH_TO_PING -n -s $first_ip 56 1 2>/dev/null | \
 	egrep -i "^round-trip|^rtt" >/dev/null
 then
 	with_ping_command="$PATH_TO_PING -n -s %s 56 %d"
 	AC_MSG_RESULT([$with_ping_command])
 
-elif $PATH_TO_PING -n -h 127.0.0.1 -s 56 -c 1 2>/dev/null | \
+elif $PATH_TO_PING -n -h $first_ip -s 56 -c 1 2>/dev/null | \
 	egrep -i "^round-trip|^rtt" >/dev/null
 then
 	with_ping_command="$PATH_TO_PING -n -h %s -s 56 -c %d"
 	AC_MSG_RESULT([$with_ping_command])
 
-elif $PATH_TO_PING -n -s 56 -c 1 127.0.0.1 2>/dev/null | \
+elif $PATH_TO_PING -n -s 56 -c 1 $first_ip 2>/dev/null | \
 	egrep -i "^round-trip|^rtt" >/dev/null
 then
 	with_ping_command="$PATH_TO_PING -n -s 56 -c %d %s"
 	ac_cv_ping_packets_first=yes
 	AC_MSG_RESULT([$with_ping_command])
 
-elif $PATH_TO_PING -n -c 1 127.0.0.1 2>/dev/null | \
+elif $PATH_TO_PING -n -c 1 $first_ip 2>/dev/null | \
 	egrep -i "^round-trip|^rtt" >/dev/null
 then
 	with_ping_command="$PATH_TO_PING -n -c %d %s"
@@ -1132,7 +1134,7 @@
 	if test -n "$PATH_TO_NSLOOKUP"
 	then
 		AC_MSG_CHECKING(for nslookup syntax)
-		if $PATH_TO_NSLOOKUP -sil 127.0.0.1 2>&1 | grep ["Invalid option: sil"] >/dev/null
+		if $PATH_TO_NSLOOKUP -sil $first_ip 2>&1 | grep ["Invalid option: sil"] >/dev/null
 		then
 			ac_cv_nslookup_command="$PATH_TO_NSLOOKUP"
 			AC_MSG_RESULT([$ac_cv_nslookup_command])
