diff -Naur nagios-plugins-1.4.10.orig/contrib/check_dns_random.pl nagios-plugins-1.4.10/contrib/check_dns_random.pl
--- nagios-plugins-1.4.10.orig/contrib/check_dns_random.pl	2003-01-29 06:27:20.000000000 +0100
+++ nagios-plugins-1.4.10/contrib/check_dns_random.pl	2007-09-29 10:57:38.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/perl -w
 # ------------------------------------------------------------------------------
 # File Name:            check_dns_random.pl
 # Author:               Richard Mayhew - South Africa
@@ -33,9 +33,9 @@
 
 my $host = shift || &usage;
 
-my $domainfile = "/usr/local/nagios/etc/domains.list";
+my $domainfile = "/etc/nagios/domains.list";
 my $wc = `/usr/bin/wc -l $domainfile`;
-my $check = "/usr/local/nagios/libexec/check_dns";
+my $check = "/usr/nagios/libexec/check_dns";
 my $x = 0;
 my $srv_file = "";
 my $z = "";
@@ -49,8 +49,7 @@
                 my @data = split(/\n/,$srv_file);
 
 chomp $wc;
-$wc =~ s/ //g;
-$wc =~ s/domains//g;
+$wc =~ s/([[:digit:]]+) .*/$1/g;
 
 $x = rand $wc;
 ($z,$y) = split(/\./,$x);
diff -Naur nagios-plugins-1.4.10.orig/contrib/check_linux_raid.pl nagios-plugins-1.4.10/contrib/check_linux_raid.pl
--- nagios-plugins-1.4.10.orig/contrib/check_linux_raid.pl	2007-08-28 05:19:45.000000000 +0200
+++ nagios-plugins-1.4.10/contrib/check_linux_raid.pl	2007-09-29 10:57:50.000000000 +0200
@@ -23,7 +23,7 @@
 #	  WARNING md0 status=[UUU_U], recovery=46.4%, finish=123.0min
 
 use strict;
-use lib "/usr/local/nagios/libexec";
+use lib "/usr/nagios/libexec";
 use utils qw(%ERRORS);
 
 # die with an error if we're not on Linux
