--- compface-1.5.1.orig/Makefile.in
+++ compface-1.5.1/Makefile.in
@@ -71,18 +71,16 @@
 		$(RM) -rf `cat .fname` .fname
 
 install: 	$(NAME) $(UNNAME) $(LIBNAME)
-		$(INSTALL_PROGRAM) $(srcdir)/$(NAME) $(EXECUTABLE) 
-		-chmod 0755 $(EXECUTABLE)
-		-strip $(EXECUTABLE)
-		$(INSTALL_PROGRAM) $(srcdir)/$(UNNAME) $(UNEXECUTABLE)
-		-chmod 0755 $(UNEXECUTABLE)
-		-strip $(UNEXECUTABLE)
-		$(INSTALL_DATA) $(srcdir)/$(NAME).1 $(MAN1DIR)/$(NAME).1
-		cd $(MAN1DIR) && $(RM) ./$(UNNAME).1 && $(LN_S) $(NAME).1 $(UNNAME).1
-		$(INSTALL_DATA) $(srcdir)/$(NAME).3 $(MAN3DIR)/$(NAME).3
-		cd $(MAN3DIR) && $(RM) ./$(UNNAME).3 && $(LN_S) $(NAME).3 $(UNNAME).3
-		for hdr in $(INSTALLHEADERS); do $(INSTALL_DATA) $(srcdir)/$${hdr} $(INCLUDEDIR)/$${hdr}; done
-		$(INSTALL_DATA) $(srcdir)/$(LIBNAME) $(LIBRARY)
+		$(INSTALL_PROGRAM) $(srcdir)/$(NAME) $(DESTDIR)/$(EXECUTABLE) 
+		-chmod 0755 $(DESTDIR)/$(EXECUTABLE)
+		$(INSTALL_PROGRAM) $(srcdir)/$(UNNAME) $(DESTDIR)/$(UNEXECUTABLE)
+		-chmod 0755 $(DESTDIR)/$(UNEXECUTABLE)
+		$(INSTALL_DATA) $(srcdir)/$(NAME).1 $(DESTDIR)/$(MAN1DIR)/$(NAME).1
+		$(LN_S) $(NAME).1 $(DESTDIR)/$(MAN1DIR)/$(UNNAME).1
+		$(INSTALL_DATA) $(srcdir)/$(NAME).3 $(DESTDIR)/$(MAN3DIR)/$(NAME).3
+		$(LN_S) $(NAME).3 $(DESTDIR)/$(MAN3DIR)/$(UNNAME).3
+		for hdr in $(INSTALLHEADERS); do $(INSTALL_DATA) $(srcdir)/$${hdr} $(DESTDIR)/$(INCLUDEDIR)/$${hdr}; done
+		$(INSTALL_DATA) $(srcdir)/$(LIBNAME) $(DESTDIR)/$(LIBRARY)
 
 shar :
 		shar.script $(OTHERS) $(HDRS) $(SOURCES) > $(NAME).sh
