From 202a70158973fee73432816d2b7612161824cc29 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Wed, 15 Aug 2012 14:02:07 +0200
Subject: [PATCH] Disable udev targets, for udev-190.

---
 Makefile.am | 190 +-----------------------------------------------------------
 1 file changed, 1 insertion(+), 189 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 64dfe81..a378007 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -309,8 +309,6 @@ dist_systemunit_DATA = \
 	units/systemd-ask-password-wall.path \
 	units/systemd-ask-password-console.path \
 	units/syslog.target \
-	units/systemd-udevd-control.socket \
-	units/systemd-udevd-kernel.socket \
 	units/system-update.target
 
 nodist_systemunit_DATA = \
@@ -340,9 +338,6 @@ nodist_systemunit_DATA = \
 	units/systemd-kexec.service \
 	units/systemd-fsck@.service \
 	units/systemd-fsck-root.service \
-	units/systemd-udevd.service \
-	units/systemd-udev-trigger.service \
-	units/systemd-udev-settle.service \
 	units/debug-shell.service
 
 dist_userunit_DATA = \
@@ -1643,12 +1638,6 @@ SUBDIRS += \
 	docs/libudev
 endif
 
-include_HEADERS += \
-	src/libudev/libudev.h
-
-lib_LTLIBRARIES += \
-	-ludev
-
 libudev_la_SOURCES =\
 	src/libudev/libudev-private.h \
 	src/libudev/libudev.c \
@@ -1671,9 +1660,6 @@ libudev_la_LDFLAGS = \
 libudev_la_LIBADD = \
 	libsystemd-shared.la
 
-pkgconfiglib_DATA += \
-	src/libudev/libudev.pc
-
 EXTRA_DIST += \
 	src/libudev/libudev.pc.in \
 	src/libudev/libudev.sym
@@ -1694,12 +1680,7 @@ libudev-install-move-hook:
 libudev-uninstall-move-hook:
 	rm -f $(DESTDIR)$(rootlibdir)/libudev.so*
 
-INSTALL_EXEC_HOOKS += libudev-install-move-hook
-UNINSTALL_EXEC_HOOKS += libudev-uninstall-move-hook
-
 # ------------------------------------------------------------------------------
-noinst_LTLIBRARIES += \
-	libudev-private.la
 
 libudev_private_la_SOURCES =\
 	$(libudev_la_SOURCES) \
@@ -1714,15 +1695,6 @@ libudev_private_la_LIBADD = \
 	libsystemd-shared.la
 
 # ------------------------------------------------------------------------------
-MANPAGES += \
-	man/udev.7 \
-	man/udevadm.8 \
-	man/systemd-udevd.service.8
-
-MANPAGES_ALIAS += \
-	man/systemd-udevd.8 \
-	man/systemd-udevd-control.socket.8 \
-	man/systemd-udevd-kernel.socket.8
 
 man/systemd-udevd.8: man/systemd-udevd.service.8
 man/systemd-udevd-control.socket.8: man/systemd-udevd.service.8
@@ -1734,28 +1706,7 @@ udev-confdirs:
 INSTALL_DATA_HOOKS += udev-confdirs
 
 dist_udevrules_DATA += \
-	rules/99-systemd.rules \
-	rules/42-usb-hid-pm.rules \
-	rules/50-udev-default.rules \
-	rules/60-persistent-storage-tape.rules \
-	rules/60-persistent-serial.rules \
-	rules/60-persistent-input.rules \
-	rules/60-persistent-alsa.rules \
-	rules/60-persistent-storage.rules \
-	rules/64-btrfs.rules \
-	rules/75-net-description.rules \
-	rules/75-tty-description.rules \
-	rules/78-sound-card.rules \
-	rules/80-drivers.rules \
-	rules/95-udev-late.rules
-
-udevconfdir = $(sysconfdir)/udev
-dist_udevconf_DATA = \
-	src/udev/udev.conf
-
-sharepkgconfigdir = $(datadir)/pkgconfig
-sharepkgconfig_DATA = \
-	src/udev/udev.pc
+	rules/99-systemd.rules
 
 EXTRA_DIST += \
 	rules/99-systemd.rules.in \
@@ -1783,17 +1734,6 @@ systemd-install-hook:
 	ln -sf ../systemd-udevd.service $(DESTDIR)$(systemunitdir)/sysinit.target.wants/systemd-udevd.service
 	ln -sf ../systemd-udev-trigger.service $(DESTDIR)$(systemunitdir)/sysinit.target.wants/systemd-udev-trigger.service
 
-INSTALL_DATA_HOOKS += systemd-install-hook
-
-bin_PROGRAMS += \
-	udevadm
-
-rootlibexec_PROGRAMS += \
-	systemd-udevd
-
-noinst_LTLIBRARIES += \
-	libudev-core.la
-
 libudev_core_la_SOURCES = \
 	src/udev/udev.h \
 	src/udev/udev-event.c \
@@ -1860,14 +1800,6 @@ udevadm_LDADD = \
 	libsystemd-shared.la
 
 # ------------------------------------------------------------------------------
-TESTS += \
-	test/udev-test.pl \
-	test/rules-test.sh
-
-noinst_PROGRAMS += \
-	test-libudev \
-	test-udev
-
 test_libudev_SOURCES = \
 	src/test/test-libudev.c
 
@@ -1917,8 +1849,6 @@ ata_id_LDADD = \
 	libudev-private.la \
 	libsystemd-shared.la
 
-udevlibexec_PROGRAMS += \
-	ata_id
 
 # ------------------------------------------------------------------------------
 cdrom_id_SOURCES = \
@@ -1928,12 +1858,6 @@ cdrom_id_LDADD = \
 	-ludev \
 	libsystemd-shared.la
 
-udevlibexec_PROGRAMS += \
-	cdrom_id
-
-dist_udevrules_DATA += \
-	rules/60-cdrom_id.rules
-
 # ------------------------------------------------------------------------------
 collect_SOURCES = \
 	src/udev/collect/collect.c
@@ -1941,9 +1865,6 @@ collect_SOURCES = \
 collect_LDADD = \
 	libudev-private.la
 
-udevlibexec_PROGRAMS += \
-	collect
-
 # ------------------------------------------------------------------------------
 scsi_id_SOURCES =\
 	src/udev/scsi_id/scsi_id.c \
@@ -1955,9 +1876,6 @@ scsi_id_LDADD = \
 	libudev-private.la \
 	libsystemd-shared.la
 
-udevlibexec_PROGRAMS += \
-	scsi_id
-
 EXTRA_DIST += \
 	src/udev/scsi_id/README
 
@@ -1968,12 +1886,6 @@ v4l_id_SOURCES = \
 v4l_id_LDADD = \
 	-ludev
 
-udevlibexec_PROGRAMS += \
-	v4l_id
-
-dist_udevrules_DATA += \
-	rules/60-persistent-v4l.rules
-
 # ------------------------------------------------------------------------------
 accelerometer_SOURCES = \
 	src/udev/accelerometer/accelerometer.c
@@ -1982,12 +1894,6 @@ accelerometer_LDADD = \
 	-ludev -lm \
 	libsystemd-shared.la
 
-udevlibexec_PROGRAMS += \
-	accelerometer
-
-dist_udevrules_DATA += \
-	rules/61-accelerometer.rules
-
 # ------------------------------------------------------------------------------
 if ENABLE_GUDEV
 if ENABLE_GTK_DOC
@@ -2168,99 +2074,11 @@ nodist_keymap_SOURCES = \
 BUILT_SOURCES += \
 	$(nodist_keymap_SOURCES)
 
-udevlibexec_PROGRAMS += \
-	keymap
-
-dist_doc_DATA += \
-	src/udev/keymap/README.keymap.txt
-
-dist_udevrules_DATA += \
-	src/udev/keymap/95-keymap.rules \
-	src/udev/keymap/95-keyboard-force-release.rules
-
-dist_udevhome_SCRIPTS = \
-	src/udev/keymap/findkeyboards \
-	src/udev/keymap/keyboard-force-release.sh
-
-TESTS += \
-	src/udev/keymap/check-keymaps.sh
-
 CLEANFILES += \
 	src/udev/keymap/keys.txt \
 	src/udev/keymap/keys-from-name.gperf \
 	src/udev/keymap/keyboard-force-release.sh
 
-udevkeymapdir = $(udevlibexecdir)/keymaps
-dist_udevkeymap_DATA = \
-	keymaps/acer \
-	keymaps/acer-aspire_5720 \
-	keymaps/acer-aspire_8930 \
-	keymaps/acer-aspire_5920g \
-	keymaps/acer-aspire_6920 \
-	keymaps/acer-travelmate_c300 \
-	keymaps/asus \
-	keymaps/compaq-e_evo \
-	keymaps/dell \
-	keymaps/dell-latitude-xt2 \
-	keymaps/everex-xt5000 \
-	keymaps/fujitsu-amilo_li_2732 \
-	keymaps/fujitsu-amilo_pa_2548 \
-	keymaps/fujitsu-amilo_pro_edition_v3505 \
-	keymaps/fujitsu-amilo_pro_v3205 \
-	keymaps/fujitsu-amilo_si_1520 \
-	keymaps/fujitsu-esprimo_mobile_v5 \
-	keymaps/fujitsu-esprimo_mobile_v6 \
-	keymaps/genius-slimstar-320 \
-	keymaps/hewlett-packard \
-	keymaps/hewlett-packard-2510p_2530p \
-	keymaps/hewlett-packard-compaq_elitebook \
-	keymaps/hewlett-packard-pavilion \
-	keymaps/hewlett-packard-presario-2100 \
-	keymaps/hewlett-packard-tablet \
-	keymaps/hewlett-packard-tx2 \
-	keymaps/ibm-thinkpad-usb-keyboard-trackpoint \
-	keymaps/inventec-symphony_6.0_7.0 \
-	keymaps/lenovo-3000 \
-	keymaps/lenovo-ideapad \
-	keymaps/lenovo-thinkpad-usb-keyboard-trackpoint \
-	keymaps/lenovo-thinkpad_x6_tablet \
-	keymaps/lenovo-thinkpad_x200_tablet \
-	keymaps/lg-x110 \
-	keymaps/logitech-wave \
-	keymaps/logitech-wave-cordless \
-	keymaps/logitech-wave-pro-cordless \
-	keymaps/maxdata-pro_7000 \
-	keymaps/medion-fid2060 \
-	keymaps/medionnb-a555 \
-	keymaps/micro-star \
-	keymaps/module-asus-w3j \
-	keymaps/module-ibm \
-	keymaps/module-lenovo \
-	keymaps/module-sony \
-	keymaps/module-sony-old \
-	keymaps/module-sony-vgn \
-	keymaps/module-sony-vpc \
-	keymaps/olpc-xo \
-	keymaps/onkyo \
-	keymaps/oqo-model2 \
-	keymaps/samsung-other \
-	keymaps/samsung-90x3a \
-	keymaps/samsung-sq1us \
-	keymaps/samsung-sx20s \
-	keymaps/toshiba-satellite_a100 \
-	keymaps/toshiba-satellite_a110 \
-	keymaps/toshiba-satellite_m30x \
-	keymaps/zepto-znote
-
-udevkeymapforcereldir = $(udevlibexecdir)/keymaps/force-release
-dist_udevkeymapforcerel_DATA = \
-	keymaps-force-release/dell-touchpad \
-	keymaps-force-release/dell-xps \
-	keymaps-force-release/hp-other \
-	keymaps-force-release/samsung-other \
-	keymaps-force-release/samsung-90x3a \
-	keymaps-force-release/common-volume-keys
-
 src/udev/keymap/keys.txt: Makefile
 	$(AM_V_at)$(MKDIR_P) $(dir $@)
 	$(AM_V_GEN)$(CPP) $(CFLAGS) $(AM_CPPFLAGS) $(CPPFLAGS) -dM -include linux/input.h - < /dev/null | $(AWK) '/^#define[ \t]+KEY_[^ ]+[ \t]+[0-9]/ { if ($$2 != "KEY_MAX") { print $$2 } }' | sed 's/^KEY_COFFEE$$/KEY_SCREENLOCK/' > $@
@@ -2288,12 +2106,6 @@ mtd_probe_SOURCES =  \
 mtd_probe_CPPFLAGS = \
 	$(AM_CPPFLAGS)
 
-dist_udevrules_DATA += \
-	rules/75-probe_mtd.rules
-
-udevlibexec_PROGRAMS += \
-	mtd_probe
-
 # ------------------------------------------------------------------------------
 libsystemd_id128_la_SOURCES = \
 	src/libsystemd-id128/sd-id128.c
-- 
1.7.12

