From 39640ff78085e151b9f9fe03bde76efb47facdd0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Fri, 18 Jan 2013 17:35:54 +0100
Subject: [PATCH] Disable udev & libsystemd-daemon for systemd ebuild.

---
 Makefile.am | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index fb7d8fe..09cf7c2 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -293,8 +293,6 @@ dist_systemunit_DATA = \
 	units/systemd-ask-password-wall.path \
 	units/systemd-ask-password-console.path \
 	units/syslog.target \
-	units/systemd-udevd-control.socket \
-	units/systemd-udevd-kernel.socket \
 	units/system-update.target
 
 nodist_systemunit_DATA = \
@@ -324,9 +322,6 @@ nodist_systemunit_DATA = \
 	units/systemd-kexec.service \
 	units/systemd-fsck@.service \
 	units/systemd-fsck-root.service \
-	units/systemd-udevd.service \
-	units/systemd-udev-trigger.service \
-	units/systemd-udev-settle.service \
 	units/debug-shell.service
 
 if HAVE_KMOD
@@ -1658,6 +1653,13 @@ systemd_tty_ask_password_agent_LDADD = \
 	libsystemd-label.la \
 	libsystemd-shared.la
 
+dist_udevrules_DATA += \
+	rules/99-systemd.rules
+
+# disable udev & systemd-daemon which are installed by sys-fs/udev
+# ENABLE_GUDEV is something that won't be set in systemd ebuild
+if ENABLE_GUDEV
+
 # ------------------------------------------------------------------------------
 libsystemd_daemon_la_SOURCES = \
 	src/libsystemd-daemon/sd-daemon.c
@@ -2417,6 +2419,9 @@ dist_udevrules_DATA += \
 udevlibexec_PROGRAMS += \
 	mtd_probe
 
+endif
+# (udev disable)
+
 # ------------------------------------------------------------------------------
 libsystemd_id128_la_SOURCES = \
 	src/libsystemd-id128/sd-id128.c
-- 
1.8.1.1

