From 74f96de2cdf89c75f1a619364d90c787ec63d2f5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Wed, 2 Oct 2013 10:23:49 +0200
Subject: [PATCH] Update Makefile.in.

---
 Makefile.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index f0b7893..02bd630 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -1422,7 +1422,7 @@ libsystemd_login_internal_la_OBJECTS =  \
 @ENABLE_LOGIND_TRUE@libsystemd_login_la_DEPENDENCIES =  \
 @ENABLE_LOGIND_TRUE@	libsystemd-shared.la \
 @ENABLE_LOGIND_TRUE@	libsystemd-daemon-internal.la \
-@ENABLE_LOGIND_TRUE@	$(am__DEPENDENCIES_1)
+@ENABLE_LOGIND_TRUE@	libsystemd-label.la $(am__DEPENDENCIES_1)
 am__libsystemd_login_la_SOURCES_DIST = src/login/sd-login.c \
 	src/systemd/sd-login.h src/login/login-shared.c \
 	src/login/login-shared.h
@@ -6096,6 +6096,7 @@ dist_catalog_DATA = \
 @ENABLE_LOGIND_TRUE@libsystemd_login_la_LIBADD = \
 @ENABLE_LOGIND_TRUE@	libsystemd-shared.la \
 @ENABLE_LOGIND_TRUE@	libsystemd-daemon-internal.la \
+@ENABLE_LOGIND_TRUE@	libsystemd-label.la \
 @ENABLE_LOGIND_TRUE@	$(RT_LIBS)
 
 @ENABLE_LOGIND_TRUE@libsystemd_login_internal_la_SOURCES = \
-- 
1.8.3.2

