Index: ocamlnet-3.6.6/src/netsys/netsys_posix.ml
===================================================================
--- ocamlnet-3.6.6.orig/src/netsys/netsys_posix.ml
+++ ocamlnet-3.6.6/src/netsys/netsys_posix.ml
@@ -705,7 +705,7 @@ type at_flag = AT_EACCESS | AT_SYMLINK_N
 IFDEF HAVE_O_SHARE_DELETE THEN
 type open_flag1 = Unix.open_flag =
     O_RDONLY | O_WRONLY | O_RDWR | O_NONBLOCK | O_APPEND | O_CREAT | O_TRUNC
-  | O_EXCL | O_NOCTTY | O_DSYNC | O_SYNC | O_RSYNC | O_SHARE_DELETE
+  | O_EXCL | O_NOCTTY | O_DSYNC | O_SYNC | O_RSYNC | O_SHARE_DELETE | O_CLOEXEC
 ELSE
 type open_flag1 = Unix.open_flag =
     O_RDONLY | O_WRONLY | O_RDWR | O_NONBLOCK | O_APPEND | O_CREAT | O_TRUNC
