From 9e24c121e502ebd8e88821044d78fda187537861 Mon Sep 17 00:00:00 2001
From: Dennis Schridde <devurandom@gmx.net>
Date: Tue, 18 Sep 2012 15:20:25 +0200
Subject: [PATCH] Fix compilation with Lua 5.2 (issue #146)

---
 inotify.c |    4 ++--
 lsyncd.c  |   10 +++++-----
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/inotify.c b/inotify.c
index 432e530..ac6f7d9 100644
--- a/inotify.c
+++ b/inotify.c
@@ -129,7 +129,7 @@
 /**
  * Cores inotify functions.
  */
-static const luaL_reg linotfylib[] = {
+static const luaL_Reg linotfylib[] = {
 		{"addwatch",   l_addwatch   },
 		{"rmwatch",    l_rmwatch    },
 		{NULL, NULL}
@@ -352,7 +352,7 @@
 register_inotify(lua_State *L)
 {
 	lua_pushstring(L, "inotify");
-	luaL_register(L, "inotify", linotfylib);
+	luaL_newlib(L, linotfylib);
 }
 
 /**
diff --git a/lsyncd.c b/lsyncd.c
index fbb5d33..f7c61c2 100644
--- a/lsyncd.c
+++ b/lsyncd.c
@@ -841,12 +841,12 @@ struct pipemsg {
 		if (lua_istable(L, i)) {
 			int tlen;
 			int it;
-			lua_checkstack(L, lua_gettop(L) + lua_objlen(L, i) + 1);
+			lua_checkstack(L, lua_gettop(L) + lua_rawlen(L, i) + 1);
 			// move table to top of stack
 			lua_pushvalue(L, i);
 			lua_remove(L, i);
 			argc--;
-			tlen = lua_objlen(L, -1);
+			tlen = lua_rawlen(L, -1);
 			for (it = 1; it <= tlen; it++) {
 				lua_pushinteger(L, it);
 				lua_gettable(L, -2);
@@ -1263,7 +1263,7 @@ struct pipemsg {
 	return 0;
 }
 
-static const luaL_reg lsyncdlib[] = {
+static const luaL_Reg lsyncdlib[] = {
 		{"configure",     l_configure     },
 		{"exec",          l_exec          },
 		{"log",           l_log           },
@@ -1368,7 +1368,7 @@ struct pipemsg {
 void
 register_lsyncd(lua_State *L)
 {
-	luaL_register(L, "lsyncd", lsyncdlib);
+	luaL_newlib(L, lsyncdlib);
 	lua_setglobal(L, "lysncd");
 
 	// creates the metatable for jiffies userdata
@@ -1676,7 +1676,7 @@ struct pipemsg {
 	int argp = 1;
 
 	// load Lua
-	L = lua_open();
+	L = luaL_newstate();
 	luaL_openlibs(L);
 	{
 		// checks the lua version
-- 
1.7.10

