diff -Nur director-0.2.8.old/i18n.c director-0.2.8/i18n.c
--- director-0.2.8.old/i18n.c	2005-09-27 13:26:50.000000000 +0200
+++ director-0.2.8/i18n.c	2007-05-03 20:10:35.000000000 +0200
@@ -128,5 +128,28 @@
 #endif
 #endif
   },
+{ "Director",
+    "Director",
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+#if VDRVERSNUM && VDRVERSNUM >= 10313
+    "", // Eesti
+#if VDRVERSNUM >= 10316
+    "", // Dansk
+#endif
+#endif
+  },
   { NULL }
   };
