--- trunk/1.8.x/ccs-tools/ccstools/usr_sbin/Makefile	2011/08/20 04:19:39	5357
+++ trunk/1.8.x/ccs-tools/ccstools/usr_sbin/Makefile	2011/09/28 08:53:43	5494
@@ -19,7 +19,7 @@
 	ln -sf libccstools.so.2.0.2 libccstools.so
 
 .c:
-	$(CC) $(CFLAGS) -o $@ -lccstools -L. $<
+	$(CC) $(CFLAGS) -o $@ $< -lccstools -L. 
 
 ccs-editpolicy: ccstools.h editpolicy*.c readline.h /usr/include/curses.h libccstools.so
 	$(CC) $(CFLAGS) -o ccs-editpolicy editpolicy*.c -lncurses -lccstools -L. -DCOLOR_ON
