diff --git a/sbin/Makefile b/sbin/Makefile
index cc61f13..3bdb292 100644
--- a/sbin/Makefile
+++ b/sbin/Makefile
@@ -9,7 +9,7 @@ install: all
 	$(INSTALL) -m 0700 $(BUILD_FILES) $(INSTALLDIR)$(SBINDIR)
 
 .c:
-	$(CC) $(CFLAGS) -o $@ $<
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $<
 
 clean:
 	rm -f -- $(BUILD_FILES)
diff --git a/usr_lib_ccs/Makefile b/usr_lib_ccs/Makefile
index 8181201..bee5eae 100644
--- a/usr_lib_ccs/Makefile
+++ b/usr_lib_ccs/Makefile
@@ -11,7 +11,7 @@ install: all
 	$(INSTALL) -m 0644 ../README.ccs ../COPYING.ccs $(INSTALLDIR)/$(USRLIBDIR)/ccs/
 
 .c:
-	$(CC) $(CFLAGS) -o $@ $<
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $<
 
 clean:
 	rm -f -- $(BUILD_FILES)
diff --git a/usr_sbin/Makefile b/usr_sbin/Makefile
index 69a8d13..cd03d76 100644
--- a/usr_sbin/Makefile
+++ b/usr_sbin/Makefile
@@ -15,17 +15,17 @@
 	sleep 10

 libccstools.so: ccstools.c ccstools.h
-	$(CC) $(CFLAGS) -fPIC ccstools.c -shared -Wl,-soname,libccstools.so.3 -o libccstools.so.3.0.1
+	$(CC) $(CFLAGS) $(LDFLAGS) -fPIC ccstools.c -shared -Wl,-soname,libccstools.so.3 -o libccstools.so.3.0.1
 	ln -sf libccstools.so.3.0.1 libccstools.so
 
 .c:
-	$(CC) $(CFLAGS) -o $@ $< -lccstools -L. 
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $< -lccstools -L. 
 
 ccs-editpolicy: ccstools.h editpolicy*.c readline.h /usr/include/curses.h libccstools.so
-	$(CC) $(CFLAGS) -o ccs-editpolicy editpolicy*.c -lncurses -lccstools -L. -DCOLOR_ON
+	$(CC) $(CFLAGS) $(LDFLAGS) -o ccs-editpolicy editpolicy*.c -lncurses -lccstools -L. -DCOLOR_ON
 
 ccs-queryd: ccstools.h ccs-queryd.c readline.h /usr/include/curses.h libccstools.so
-	$(CC) $(CFLAGS) -o ccs-queryd ccs-queryd.c -lncurses -lccstools -L.
+	$(CC) $(CFLAGS) $(LDFLAGS) -o ccs-queryd ccs-queryd.c -lncurses -lccstools -L.
 
 install: all
 	mkdir -p -m 0755 $(INSTALLDIR)$(USRLIBDIR)
