--- yoshimi-0.060.12.orig/desktop/yoshimi.desktop.in
+++ yoshimi-0.060.12/desktop/yoshimi.desktop.in
@@ -5,8 +5,8 @@
 Version=@YOSHIMI_VERSION@
 Type=Application
 Categories=AudioVideo;X-Sound;Audio;
-Exec=@CMAKE_INSTALL_PREFIX@/bin/yoshimi
-TryExec=@CMAKE_INSTALL_PREFIX@/bin/yoshimi
+Exec=yoshimi
+TryExec=yoshimi
 Terminal=false
 StartupNotify=true
-Icon=@CMAKE_INSTALL_PREFIX@/share/yoshimi/yoshimi.png
+Icon=yoshimi
--- yoshimi-0.060.12.orig/src/CMakeLists.txt
+++ yoshimi-0.060.12/src/CMakeLists.txt
@@ -197,6 +197,11 @@
     message (STATUS "Building for ${CMAKE_BUILD_TYPE}, flags: ${CMAKE_CXX_FLAGS_RELEASE}")
 endif (BuildForDebug)
 
+configure_file(
+    "${CMAKE_CURRENT_SOURCE_DIR}/../desktop/yoshimi.desktop.in"
+    "${CMAKE_CURRENT_BINARY_DIR}/yoshimi.desktop"
+    IMMEDIATE @ONLY)
+
 set (DSP_sources
     DSP/FFTwrapper.cpp  DSP/AnalogFilter.cpp  DSP/FormantFilter.cpp
     DSP/SVFilter.cpp  DSP/Filter.cpp  DSP/Unison.cpp
@@ -368,6 +373,11 @@
         WORLD_READ WORLD_EXECUTE
 )
 
+install (FILES ${CMAKE_CURRENT_BINARY_DIR}/yoshimi.desktop
+    DESTINATION share/applications)
+install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/../desktop/yoshimi.png
+    DESTINATION share/pixmaps)
+
 set_directory_properties (PROPERTIES
     ADDITIONAL_MAKE_CLEAN_FILES "${CMAKE_SOURCE_DIR}/*UI.c* ${CMAKE_SOURCE_DIR}/src/*UI.h"
 )
