GNU DESTDIR and multilib directory support

--- Makefile
+++ Makefile
@@ -22,10 +22,13 @@
 #EXTRA_LDFLAGS=
 
 # Use these with Debian GNU/Linux for example
-#TARGET_OS=LINUX
-#PLUGIN_INSTALL_DIRECTORY=/usr/lib/xchat/plugins
-#FRONTEND_INSTALL_DIRECTORY=/usr/bin
-#CWIRC_EXTENSIONS_DIRECTORY=/usr/lib/cwirc/extensions
+TARGET_OS=LINUX
+DESTDIR=
+PREFIX=/usr
+LIBDIR=$(PREFIX)/lib
+PLUGIN_INSTALL_DIRECTORY=$(DESTDIR)$(LIBDIR)/xchat/plugins
+FRONTEND_INSTALL_DIRECTORY=$(DESTDIR)$(PREFIX)/bin
+CWIRC_EXTENSIONS_DIRECTORY=$(DESTDIR)$(LIBDIR)/cwirc/extensions
 #EXTRA_LDFLAGS=
 
 # Use these with FreeBSD 5.1 for example
