--- tcprules.c.orig	2005-09-12 00:44:11.000000000 +0200
+++ tcprules.c	2005-09-12 00:58:11.000000000 +0200
@@ -127,6 +127,9 @@
     for (;;) {
       int tmp;
       tmp = byte_chr(x + colon,len - colon,':');
+	  if(colon == 0 && tmp == len) {
+		strerr_die2x(111, FATAL, "Unable to find colon on non-empty line.");
+	  }
       colon += tmp;
       if (colon == len) continue;
       if (byte_equal(x+colon+1,4,"deny") || byte_equal(x+colon+1,5,"allow")) break;
