Author: Ralph Sennhauser <sera@gentoo.org>

Fix parallel unpacking #453608

diff --git a/Makefile.am b/Makefile.am
index 3f96d92..714cefa 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1289,7 +1289,7 @@ endif
 clean-extract-cacao: clean-cacao
 	rm -f stamps/extract-cacao.stamp
 
-stamps/extract-jamvm.stamp:
+stamps/extract-jamvm.stamp: stamps/download-jamvm.stamp
 if BUILD_JAMVM
 	set -e ; \
 	if ! test -d jamvm ; \
