--- pcfclock-0.44.orig/configure	2004-05-17 19:14:08.000000000 +0200
+++ pcfclock-0.44/configure	2010-07-20 23:56:19.324330498 +0200
@@ -2962,6 +2962,9 @@
       if test -f $ac_dir/include/linux/autoconf.h; then
         ac_file_linux_autoconf_h=$ac_dir/include/linux/autoconf.h
       fi
+      if test -f $ac_dir/include/generated/autoconf.h; then
+        ac_file_linux_autoconf_h=$ac_dir/include/generated/autoconf.h
+      fi
       if test x$ac_file_linux_version_h != x && test x$ac_file_linux_autoconf_h != x; then
         have_linux=yes
 	break
