From 5adc20f669b4759ac431da7a992217637c22f104 Mon Sep 17 00:00:00 2001
From: Alon Bar-Lev <alon.barlev@gmail.com>
Date: Mon, 9 Mar 2015 23:22:02 +0200
Subject: [PATCH] build: tests: fix Test_choice_ocsp on separate builddir

Signed-off-by: Alon Bar-Lev <alon.barlev@gmail.com>
---
 tests/Makefile.am | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/Makefile.am b/tests/Makefile.am
index 4b326f0..a00ebc7 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -54,6 +54,8 @@ TESTS_ENVIRONMENT = \
 	ASN1INDEF2=$(srcdir)/TestIndef2.p12 \
 	ASN1INDEF3=$(srcdir)/TestIndef3.der \
 	ASN1ENCODING=$(srcdir)/Test_encoding.asn \
+	ASN1CHOICE_OCSP=$(srcdir)/pkix.asn \
+	ASN1CHOICE_OCSP_DATA=$(srcdir)/ocsp.der \
 	THREADSAFETY_FILES=`find $(top_srcdir)/lib -name \*.c` \
 	EXEEXT=$(EXEEXT) \
 	$(VALGRIND)
-- 
2.0.5

