--- libtasn1-3.0/configure.ac
+++ libtasn1-3.0/configure.ac
@@ -21,7 +21,7 @@
 AC_CONFIG_AUX_DIR([build-aux])
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_HEADERS(config.h)
-AM_INIT_AUTOMAKE([1.10 -Wall -Werror -Wno-override])
+AM_INIT_AUTOMAKE([1.10 -Wall -Wno-override])
 AM_SILENT_RULES([yes])
 
 # Library code modified:                              REVISION++
--- libtasn1-3.0/lib/Makefile.am
+++ libtasn1-3.0/lib/Makefile.am
@@ -19,7 +19,7 @@
 SUBDIRS = gllib
 
 AM_CFLAGS = $(WARN_CFLAGS) $(WERROR_CFLAGS) $(CFLAG_VISIBILITY)
-AM_CPPFLAGS = -I$(builddir)/gllib -DASN1_BUILDING
+AM_CPPFLAGS = -I$(top_srcdir)/lib/gllib -DASN1_BUILDING
 
 include_HEADERS = libtasn1.h
 
