--- ax25/axgetput/Makefile.am.orig	2008-08-04 00:52:33.000000000 +0200
+++ ax25/axgetput/Makefile.am	2009-09-20 18:17:18.000000000 +0200
@@ -13,7 +13,7 @@
 
 axgetput_SOURCES = axgetput.c axgetput.h util.h proto_bin.h util.c proto_bin.c includes.h
 
-install-exec-local:
+install-exec-hook:
 	(cd $(DESTDIR)$(bindir) && ln -sf axgetput bget && ln -sf axgetput bput)
 
 install-data-hook:                                                             
--- 6pack/Makefile.am.orig	2009-09-25 16:13:13.000000000 +0200
+++ 6pack/Makefile.am	2009-09-25 16:14:03.000000000 +0200
@@ -15,6 +15,6 @@
 AX25_SYSCONFDIR=${sysconfdir}/ax25/
 AX25_LOCALSTATEDIR=${localstatedir}/ax25/
 
-install-exec-local:
+install-exec-hook:
 	(cd $(DESTDIR)$(sbindir))
 
--- kiss/Makefile.am.orig	2009-09-25 16:15:07.000000000 +0200
+++ kiss/Makefile.am	2009-09-25 16:15:21.000000000 +0200
@@ -15,6 +15,6 @@
 AX25_SYSCONFDIR=${sysconfdir}/ax25/
 AX25_LOCALSTATEDIR=${localstatedir}/ax25/
 
-install-exec-local:
+install-exec-hook:
 	(cd $(DESTDIR)$(sbindir) ; ln -sf kissattach spattach)
 
