diff -urN Xsettings-client-0.10.orig/Makefile.am Xsettings-client-0.10/Makefile.am
--- Xsettings-client-0.10.orig/Makefile.am	2006-10-24 21:21:19.000000000 -0700
+++ Xsettings-client-0.10/Makefile.am	2006-10-24 21:39:34.000000000 -0700
@@ -9,4 +9,4 @@
 
 libXsettings_client_la_SOURCES= $(source_c) $(source_h)
 
-libXsettings_client_la_LIBADD      = @X_LIBS@
\ No newline at end of file
+libXsettings_client_la_LIBADD      = @X_LIBS@ -lX11
