--- configure.orig	2010-03-06 22:25:15.645376914 +0200
+++ configure	2010-03-06 22:26:36.806376713 +0200
@@ -25,10 +25,6 @@
 infodir='$(datadir)/info'
 mandir='$(datadir)/man'
 sysconfdir='$(prefix)/etc'
-CXX=
-CPPFLAGS=
-CXXFLAGS='-Wall -W -O2'
-LDFLAGS=
 
 # Loop over all args
 while [ -n "$1" ] ; do
@@ -101,6 +97,14 @@
 	esac
 done
 
+# Defaults if the user did not select any
+if [ x"${CFLAGS+set}" != xset ] ; then
+	CFLAGS='-Wall -W -O2'
+fi
+if [ x"${CXXFLAGS+set}" != xset ] ; then
+	CXXFLAGS='-Wall -W -O2'
+fi
+
 # Find the source files, if location was not specified.
 srcdirtext=
 if [ -z "${srcdir}" ] ; then
