--- a/Source/filter.c	2010-09-15 02:17:55.000000000 +0200
+++ b/Source/filter.c	2010-09-15 02:23:46.000000000 +0200
@@ -22,6 +22,7 @@
 
 #include <stdlib.h>
 #include <ctype.h>
+#include <string.h> /* memcpy() */
 
 #include <fnmatch.h>
 
--- a/Source/main.c	2010-09-15 02:17:56.000000000 +0200
+++ b/Source/main.c	2010-09-15 02:24:00.000000000 +0200
@@ -37,6 +37,7 @@
 #include <pwd.h>
 #include <stdlib.h>
 #include <getopt.h>
+#include <grp.h> /* initgroups() */
 
 #include "defines.h"
 #include "configuration.h"
