Index: work/src/tree.c
===================================================================
--- work.orig/src/tree.c
+++ work/src/tree.c
@@ -91,7 +91,8 @@ void tree_apply()
     gboolean err;
 
     if (obc_config_file)
-        p = g_strdup(obc_config_file);
+        p = g_build_filename(obt_paths_config_home(paths), "openbox",
+                             obc_config_file, NULL);
     else
         p = g_build_filename(obt_paths_config_home(paths), "openbox",
                              "rc.xml", NULL);
