diff -burN cpulimit-9999~/Makefile cpulimit-9999/Makefile
--- cpulimit-9999~/Makefile	2010-06-20 02:55:22.046518587 +0100
+++ cpulimit-9999/Makefile	2010-06-20 03:05:25.439517679 +0100
@@ -6,7 +6,7 @@
 all::	$(TARGETS)
 
 cpulimit:	cpulimit.c $(LIBS)
-	$(CC) -o cpulimit cpulimit.c $(LIBS) $(CFLAGS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o cpulimit cpulimit.c $(LIBS) -lrt
 
 process.o: process.c process.h
 	$(CC) -c process.c $(CFLAGS)
@@ -19,4 +19,3 @@
 
 clean:
 	rm -f *~ *.o $(TARGETS)
-
