diff -ur wbar-1.3.3.orig/Makefile wbar-1.3.3/Makefile
--- wbar-1.3.3.orig/Makefile	2007-09-26 19:24:10.000000000 +0300
+++ wbar-1.3.3/Makefile	2008-01-13 14:42:20.000000000 +0200
@@ -1,6 +1,6 @@
 TARGET=wbar
-CXXFLAGS=`imlib2-config --cflags` -Wall -O2 #-DCOS_ZOOM #-DLINEAR_TRASL -DNO_EXPAND #-DAVGFILTER
-LDFLAGS=`imlib2-config --libs`  -Wl,-O2
+CXXFLAGS+=`imlib2-config --cflags` -Wall #-DCOS_ZOOM #-DLINEAR_TRASL -DNO_EXPAND #-DAVGFILTER
+LIBS=`imlib2-config --libs` -lX11
 PREFIX=/usr/share/wbar
 
 	
@@ -14,7 +14,7 @@
 $(objects): $(headers) Makefile
 
 $(TARGET): $(objects) 
-	g++ $(LDFLAGS) -o $(@) $(objects)
+	$(CXX) $(LDFLAGS) $(objects) $(LIBS) -o $(@)
 	#strip $(@)
 
 install: $(TARGET)
@@ -23,16 +23,17 @@
 		exit 1; \
 	fi
 
-	install -d $(PREFIX)
+	install -d $(DESTDIR)$(PREFIX)
 	awk '{if($$1 ~ /i:/ || ($$1 ~ /t:/ && NR<4)) print $$1" $(PREFIX)/"$$2; else print $$0;}' \
-		./dot.wbar > $(PREFIX)/dot.wbar
-	cp -a ./iconpack $(PREFIX)/iconpack
+		./dot.wbar > $(DESTDIR)$(PREFIX)/dot.wbar
+	cp -a ./iconpack $(DESTDIR)$(PREFIX)/iconpack
 
 	#ln -fs $(PREFIX)/iconpack/comic.ttf $(PREFIX)/iconpack/wbar.nuvoux/font.ttf
 	#ln -fs $(PREFIX)/iconpack/comic.ttf $(PREFIX)/iconpack/wbar.ice/font.ttf
-	ln -fs $(PREFIX)/iconpack/comic.ttf $(PREFIX)/iconpack/wbar.osx/font.ttf
+	ln -fs ../comic.ttf $(DESTDIR)$(PREFIX)/iconpack/wbar.osx/font.ttf
 	
-	install ./wbar /usr/bin
+	install -d $(DESTDIR)/usr/bin
+	install ./wbar $(DESTDIR)/usr/bin
 
 uninstall:
 	if [ "`whoami`" != "root" ]; then \
