From: Julian Ospald <hasufell@gentoo.org>
Date: Sun Aug 26 16:34:39 UTC 2012
Subject: fix desktop file destination and QA issues

diff --git a/configure.ac b/configure.ac
index 0fc821c..2db7c02 100644
--- a/configure.ac
+++ b/configure.ac
@@ -83,6 +83,7 @@ AC_DEFINE_UNQUOTED(PIDOF_BIN, "$PIDOF_BIN ", [ program to retreive wbar pid ])
 
 AC_CONFIG_FILES([Makefile \
 share/Makefile \
+share/wbar.desktop.in \
 etc/Makefile \
 etc/wbar.cfg \
 etc/wbar.desktop.in \
diff --git a/etc/Makefile.am b/etc/Makefile.am
index 1bcb238..a433e0c 100644
--- a/etc/Makefile.am
+++ b/etc/Makefile.am
@@ -3,7 +3,7 @@ CLEANFILES = *~ $(wbardesktop:.desktop.in.in=.desktop)
 defaultdir = @DEFAULT_CONFIGDIR@
 default_DATA = $(wbarcfg:.cfg.in=.cfg)
 
-autostartdir = @DEFAULT_CONFIGDIR@
+autostartdir = $(sysconfdir)/xdg/autostart
 autostart_DATA = $(wbardesktop:.desktop.in.in=.desktop)
 
 @INTLTOOL_DESKTOP_RULE@
diff --git a/po/POTFILES.in b/po/POTFILES.in
index 8472c3c..8ed7725 100644
--- a/po/POTFILES.in
+++ b/po/POTFILES.in
@@ -1,4 +1,5 @@
 etc/wbar.desktop.in
+share/wbar.desktop.in
 share/wbar.glade
 src/config/Functions.cc
 src/config/Functions.h
diff --git a/po/POTFILES.skip b/po/POTFILES.skip
index 327f139..a9fbd01 100644
--- a/po/POTFILES.skip
+++ b/po/POTFILES.skip
@@ -1,2 +1,2 @@
 etc/wbar.desktop.in.in
-
+share/wbar.desktop.in.in
diff --git a/share/Makefile.am b/share/Makefile.am
index 87c736b..091935c 100644
--- a/share/Makefile.am
+++ b/share/Makefile.am
@@ -1,8 +1,16 @@
-CLEANFILES = *~
+CLEANFILES = *~ $(wbardesktop:.desktop.in.in=.desktop)
 
 gladedir = @GLADE_PACKAGE_DIR@
 glade_DATA = @PACKAGE@.glade
 
-EXTRA_DIST = \
-$(glade_DATA)
+desktopfilesdir = $(prefix)/share/applications
+desktopfiles_DATA = $(wbardesktop:.desktop.in.in=.desktop)
+
+@INTLTOOL_DESKTOP_RULE@
 
+wbardesktop = wbar.desktop.in.in
+
+EXTRA_DIST = \
+$(glade_DATA) \
+$(wbardesktop) \
+$(wbardesktop:.desktop.in.in=.desktop.in)
diff --git a/share/wbar.desktop.in.in b/share/wbar.desktop.in.in
new file mode 100644
index 0000000..48fc33c
--- /dev/null
+++ b/share/wbar.desktop.in.in
@@ -0,0 +1,12 @@
+[Desktop Entry]
+_Name=Warlock Bar
+Type=Application
+Exec=@BINDIR@/@PACKAGE_NAME@
+Terminal=false
+Icon=@PIXMAPDIR@/@PACKAGE_NAME@.png
+_Comment=A light and fast launch bar.
+Categories=Utility;
+X-GNOME-Bugzilla-Bugzilla=GNOME
+X-GNOME-Bugzilla-Product=@PACKAGE_NAME@
+X-GNOME-Bugzilla-Component=general
+X-Ubuntu-Gettext-Domain=@PACKAGE_NAME@
