--- makefile.linux	2003-11-16 22:38:20.000000000 -0500
+++ Makefile	2010-08-09 13:10:59.000000000 -0400
@@ -1,14 +1,18 @@
+OBJS = Public.o ChainWalkContext.o ChainWalkSet.o CrackEngine.o HashAlgorithm.o HashSet.o HashRoutine.o MemoryPool.o RainbowCrack.o RainbowTableDump.o RainbowTableGenerate.o RainbowTableSort.o
 
 all: rtgen rtdump rtsort rcrack
 
-rtgen:
-	g++ Public.cpp ChainWalkContext.cpp HashAlgorithm.cpp HashRoutine.cpp RainbowTableGenerate.cpp -lssl -O3 -o rtgen
+rtgen: $(OBJS)
+	$(CXX) $(LDFLAGS) Public.o ChainWalkContext.o HashAlgorithm.o HashRoutine.o RainbowTableGenerate.o -lssl -lcrypto -o rtgen
 
-rtdump:
-	g++ Public.cpp ChainWalkContext.cpp HashAlgorithm.cpp HashRoutine.cpp RainbowTableDump.cpp -lssl -o rtdump
+rtdump: $(OBJS)
+	$(CXX) $(LDFLAGS) Public.o ChainWalkContext.o HashAlgorithm.o HashRoutine.o RainbowTableDump.o -lssl -lcrypto -o rtdump
 
-rtsort:
-	g++ Public.cpp RainbowTableSort.cpp -o rtsort
+rtsort: $(OBJS)
+	$(CXX) $(LDFLAGS) Public.o RainbowTableSort.o -o rtsort
 
-rcrack:
-	g++ Public.cpp ChainWalkContext.cpp HashAlgorithm.cpp HashRoutine.cpp HashSet.cpp MemoryPool.cpp ChainWalkSet.cpp CrackEngine.cpp RainbowCrack.cpp -lssl -O3 -o rcrack
+rcrack: $(OBJS)
+	$(CXX) $(LDFLAGS) Public.o ChainWalkContext.o HashAlgorithm.o HashRoutine.o HashSet.o MemoryPool.o ChainWalkSet.o CrackEngine.o RainbowCrack.o -lssl -lcrypto -o rcrack
+
+clean:
+	rm -f $(OBJS) rtgen rtdump rtsort rcrack
