--- rainbowcrack-1.2-src/src/makefile.linux.orig	2009-01-03 15:46:11.220007191 +0100
+++ rainbowcrack-1.2-src/src/makefile.linux	2009-01-03 15:45:56.786005299 +0100
@@ -2,13 +2,13 @@
 all: rtgen rtdump rtsort rcrack
 
 rtgen:
-	$(CXX) Public.cpp ChainWalkContext.cpp HashAlgorithm.cpp HashRoutine.cpp RainbowTableGenerate.cpp -lssl ${CXXFLAGS} -o rtgen
+	$(CXX) Public.cpp ChainWalkContext.cpp HashAlgorithm.cpp HashRoutine.cpp RainbowTableGenerate.cpp -lssl -lcrypto ${CXXFLAGS} -o rtgen
 
 rtdump:
-	$(CXX) Public.cpp ChainWalkContext.cpp HashAlgorithm.cpp HashRoutine.cpp RainbowTableDump.cpp -lssl ${CXXFLAGS} -o rtdump
+	$(CXX) Public.cpp ChainWalkContext.cpp HashAlgorithm.cpp HashRoutine.cpp RainbowTableDump.cpp -lssl -lcrypto ${CXXFLAGS} -o rtdump
 
 rtsort:
 	$(CXX) Public.cpp RainbowTableSort.cpp ${CXXFLAGS} -o rtsort
 
 rcrack:
-	$(CXX) Public.cpp ChainWalkContext.cpp HashAlgorithm.cpp HashRoutine.cpp HashSet.cpp MemoryPool.cpp ChainWalkSet.cpp CrackEngine.cpp RainbowCrack.cpp -lssl ${CXXFLAGS} -o rcrack
+	$(CXX) Public.cpp ChainWalkContext.cpp HashAlgorithm.cpp HashRoutine.cpp HashSet.cpp MemoryPool.cpp ChainWalkSet.cpp CrackEngine.cpp RainbowCrack.cpp -lssl -lcrypto ${CXXFLAGS} -o rcrack
