--- makefile.linux	2003-11-16 22:38:20.000000000 -0600
+++ Makefile	2004-12-22 00:59:23.000000000 -0600
@@ -2,13 +2,13 @@
 all: rtgen rtdump rtsort rcrack
 
 rtgen:
-	g++ Public.cpp ChainWalkContext.cpp HashAlgorithm.cpp HashRoutine.cpp RainbowTableGenerate.cpp -lssl -O3 -o rtgen
+	$(CXX) Public.cpp ChainWalkContext.cpp HashAlgorithm.cpp HashRoutine.cpp RainbowTableGenerate.cpp -lssl ${CXXFLAGS} -o rtgen
 
 rtdump:
-	g++ Public.cpp ChainWalkContext.cpp HashAlgorithm.cpp HashRoutine.cpp RainbowTableDump.cpp -lssl -o rtdump
+	$(CXX) Public.cpp ChainWalkContext.cpp HashAlgorithm.cpp HashRoutine.cpp RainbowTableDump.cpp -lssl ${CXXFLAGS} -o rtdump
 
 rtsort:
-	g++ Public.cpp RainbowTableSort.cpp -o rtsort
+	$(CXX) Public.cpp RainbowTableSort.cpp ${CXXFLAGS} -o rtsort
 
 rcrack:
-	g++ Public.cpp ChainWalkContext.cpp HashAlgorithm.cpp HashRoutine.cpp HashSet.cpp MemoryPool.cpp ChainWalkSet.cpp CrackEngine.cpp RainbowCrack.cpp -lssl -O3 -o rcrack
+	$(CXX) Public.cpp ChainWalkContext.cpp HashAlgorithm.cpp HashRoutine.cpp HashSet.cpp MemoryPool.cpp ChainWalkSet.cpp CrackEngine.cpp RainbowCrack.cpp -lssl ${CXXFLAGS} -o rcrack
