diff -urNp rainbowcrack-1.2-src.org/src/ChainWalkContext.cpp rainbowcrack-1.2-src/src/ChainWalkContext.cpp
--- rainbowcrack-1.2-src.org/src/ChainWalkContext.cpp	2003-11-23 00:33:44.000000000 +0200
+++ rainbowcrack-1.2-src/src/ChainWalkContext.cpp	2006-11-01 21:05:13.000000000 +0200
@@ -58,7 +58,7 @@ bool CChainWalkContext::LoadCharset(stri
 	}
 
 	vector<string> vLine;
-	if (ReadLinesFromFile("charset.txt", vLine))
+	if (ReadLinesFromFile("charset.txt", vLine) || ReadLinesFromFile("@@SHARE@@/charset.txt", vLine))
 	{
 		int i;
 		for (i = 0; i < vLine.size(); i++)
