diff -aur rainbowcrack-1.2-src.old/src/ChainWalkContext.cpp rainbowcrack-1.2-src/src/ChainWalkContext.cpp
--- rainbowcrack-1.2-src.old/src/ChainWalkContext.cpp	2008-12-11 20:55:23.000000000 +0100
+++ rainbowcrack-1.2-src/src/ChainWalkContext.cpp	2008-12-11 20:56:15.000000000 +0100
@@ -10,6 +10,7 @@
 
 #include "ChainWalkContext.h"
 
+#include <cstring>
 #include <ctype.h>
 #include <openssl/rand.h>
 #ifdef _WIN32
diff -aur rainbowcrack-1.2-src.old/src/ChainWalkSet.cpp rainbowcrack-1.2-src/src/ChainWalkSet.cpp
--- rainbowcrack-1.2-src.old/src/ChainWalkSet.cpp	2008-12-11 20:55:23.000000000 +0100
+++ rainbowcrack-1.2-src/src/ChainWalkSet.cpp	2008-12-11 20:59:07.000000000 +0100
@@ -8,6 +8,7 @@
 	#pragma warning(disable : 4786)
 #endif
 
+#include <cstring>
 #include "ChainWalkSet.h"
 
 CChainWalkSet::CChainWalkSet()
diff -aur rainbowcrack-1.2-src.old/src/RainbowCrack.cpp rainbowcrack-1.2-src/src/RainbowCrack.cpp
--- rainbowcrack-1.2-src.old/src/RainbowCrack.cpp	2008-12-11 20:55:23.000000000 +0100
+++ rainbowcrack-1.2-src/src/RainbowCrack.cpp	2008-12-11 20:59:48.000000000 +0100
@@ -18,6 +18,7 @@
 	#include <unistd.h>
 #endif
 
+#include <cstring>
 #include <openssl/md4.h>
 #ifdef _WIN32
 	#pragma comment(lib, "libeay32.lib")
diff -aur rainbowcrack-1.2-src.old/src/RainbowTableDump.cpp rainbowcrack-1.2-src/src/RainbowTableDump.cpp
--- rainbowcrack-1.2-src.old/src/RainbowTableDump.cpp	2008-12-11 20:55:23.000000000 +0100
+++ rainbowcrack-1.2-src/src/RainbowTableDump.cpp	2008-12-11 20:57:57.000000000 +0100
@@ -4,6 +4,7 @@
    Copyright (C) Zhu Shuanglei <shuanglei@hotmail.com>
 */
 
+#include <algorithm>
 #include "ChainWalkContext.h"
 
 int main(int argc, char* argv[])
diff -aur rainbowcrack-1.2-src.old/src/RainbowTableGenerate.cpp rainbowcrack-1.2-src/src/RainbowTableGenerate.cpp
--- rainbowcrack-1.2-src.old/src/RainbowTableGenerate.cpp	2008-12-11 20:55:23.000000000 +0100
+++ rainbowcrack-1.2-src/src/RainbowTableGenerate.cpp	2008-12-11 20:57:29.000000000 +0100
@@ -15,6 +15,8 @@
 #endif
 #include <time.h>
 
+#include <cstring>
+#include <algorithm>
 #include "ChainWalkContext.h"
 
 void Usage()
diff -aur rainbowcrack-1.2-src.old/src/RainbowTableSort.cpp rainbowcrack-1.2-src/src/RainbowTableSort.cpp
--- rainbowcrack-1.2-src.old/src/RainbowTableSort.cpp	2008-12-11 20:55:23.000000000 +0100
+++ rainbowcrack-1.2-src/src/RainbowTableSort.cpp	2008-12-11 20:58:43.000000000 +0100
@@ -4,6 +4,7 @@
    Copyright (C) Zhu Shuanglei <shuanglei@hotmail.com>
 */
 
+#include <cstdlib>
 #include "Public.h"
 
 #define ASSUMED_MIN_MEMORY 32 * 1024 * 1024
