Index: shorturl-0.8.7/lib/shorturl.rb
===================================================================
--- shorturl-0.8.7.orig/lib/shorturl.rb
+++ shorturl-0.8.7/lib/shorturl.rb
@@ -39,8 +39,10 @@ class Service
   def call(url)
     Net::HTTP.start(@hostname, @port) { |http|
       response = case @method
-                 when :post: http.post(@action, "#{@field}=#{CGI.escape(url)}")
-                 when :get: http.get("#{@action}?#{@field}=#{CGI.escape(url)}")
+                 when :post then
+                   http.post(@action, "#{@field}=#{CGI.escape(url)}")
+                 when :get then
+                   http.get("#{@action}?#{@field}=#{CGI.escape(url)}")
                  end
       if response.code == @code.to_s
         @response_block ? @response_block.call(response) : @block.call(response.read_body)
