--- nut-18.7/fltk/Makefile
+++ nut-18.7/fltk/Makefile
@@ -5,7 +5,7 @@
 NUTDIR := \".nutdb\"
 FOODDIR := \"$(libdir)\"
 EXECUTABLE := Nut
-LIBS := `fltk-config --libs`
+LIBS := `fltk-config --ldflags`
 
 OS := $(shell uname -s)
 ifeq ($(OS),HP-UX)
@@ -31,7 +31,7 @@
 OBJS := ../anameal.o ../db.o ../food.o ../meal.o ../options.o ../prtmenu.o ../ranking.o ../recmeal.o ../trendy.o ../util.o $(patsubst %.cc,%.o,$(SOURCE))
 
 nut:	$(OBJS)
-	$(CXX) $(OPT) $(LDFLAGS) `fltk-config --ldflags` -o $(EXECUTABLE) $(OBJS) $(LIBS)
+	$(CXX) $(OPT) $(LDFLAGS) -o $(EXECUTABLE) $(OBJS) $(LIBS)
 
 deps:
 	makedepend -Y.. ../anameal.c ../db.c ../food.c ../meal.c ../options.c ../prtmenu.c ../recmeal.c ../util.c *.cc >/dev/null 2>&1
