 scribus/plugins/pixmapexport/export.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/scribus/plugins/pixmapexport/export.cpp b/scribus/plugins/pixmapexport/export.cpp
index 6ff5e0a..d9e273a 100644
--- a/scribus/plugins/pixmapexport/export.cpp
+++ b/scribus/plugins/pixmapexport/export.cpp
@@ -96,7 +96,7 @@ bool PixmapExportPlugin::run(ScribusDoc* doc, QString target)
 	Q_ASSERT(target.isEmpty());
 	Q_ASSERT(!doc->masterPageMode());
 	QSharedPointer<ExportBitmap> ex( new ExportBitmap() );
-	QSharedPointer<ExportForm>  dia( new ExportForm(doc->scMW(), doc, ex->pageDPI, ex->quality, ex->bitmapType) );
+	QSharedPointer<ExportForm>  dia( new ExportForm(0, doc, ex->pageDPI, ex->quality, ex->bitmapType) );
 
 	// interval widgets handling
 	QString tmp;
