diff -ru amarok-0.0.2-orig/menuAmarok.cpp amarok-0.0.2/menuAmarok.cpp
--- amarok-0.0.2-orig/menuAmarok.cpp	2008-12-17 15:06:48.487227787 +0100
+++ amarok-0.0.2/menuAmarok.cpp	2008-12-17 15:07:39.700550051 +0100
@@ -720,7 +720,7 @@
 
 
 // unfortunately these playlists are not managed by the amarok DB in v1.4 or older
-char *AMAROK_STATIC_PLAYLISTS []=
+const char *AMAROK_STATIC_PLAYLISTS []=
     {
         "50 Random Tracks",
         "All Collection",
diff -ru amarok-0.0.2-orig/Sockets.cpp amarok-0.0.2/Sockets.cpp
--- amarok-0.0.2-orig/Sockets.cpp	2008-12-17 15:06:48.487227787 +0100
+++ amarok-0.0.2/Sockets.cpp	2008-12-17 15:07:20.357219764 +0100
@@ -20,6 +20,7 @@
 #ifndef WIN32
 // this is for Linux
 #include <unistd.h>
+#include <cstring>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
