From d5604be9e5a148ced73dee2ede49b38f97119baf Mon Sep 17 00:00:00 2001
From: Yanko Kaneti <yaneti@declera.com>
Date: Mon, 28 Jan 2013 15:49:51 +0200
Subject: [PATCH] xi2: Fix access beyond array boundaries

---
 clutter/x11/clutter-device-manager-xi2.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/clutter/x11/clutter-device-manager-xi2.c b/clutter/x11/clutter-device-manager-xi2.c
index 616dee7..bd9473e 100644
--- a/clutter/x11/clutter-device-manager-xi2.c
+++ b/clutter/x11/clutter-device-manager-xi2.c
@@ -94,13 +94,13 @@ translate_valuator_class (Display             *xdisplay,
       atoms_initialized = TRUE;
     }
 
-  for (i = CLUTTER_INPUT_AXIS_IGNORE;
-       i < CLUTTER_INPUT_AXIS_LAST;
+  for (i = 0;
+       i < N_AXIS_ATOMS;
        i += 1)
     {
       if (clutter_input_axis_atoms[i] == class->label)
         {
-          axis = i;
+          axis = i + 1;
           break;
         }
     }
-- 
1.8.1

