diff -ru channelscan-0.1.0-orig/channelscan.c channelscan-0.1.0/channelscan.c
--- channelscan-0.1.0-orig/channelscan.c	2006-08-09 15:43:41.658270206 +0200
+++ channelscan-0.1.0/channelscan.c	2006-08-09 15:43:55.115608737 +0200
@@ -86,7 +86,7 @@
                              unsigned int &UNCChip  );
   bool Signal(int Frequenz, char *Pol, int Symbolrate);
   void LoadSatFile(int iPosition);
-  void cMainMenu::ShowSatData(const char cSelectedFile[100]);
+  void ShowSatData(const char cSelectedFile[100]);
 };
 
 cMainMenu::cMainMenu(void) {
