diff -Naur channelscan-0.1.0.orig/channelscan.c channelscan-0.1.0/channelscan.c
--- channelscan-0.1.0.orig/channelscan.c	2009-01-06 20:46:26.000000000 +0100
+++ channelscan-0.1.0/channelscan.c	2009-01-06 20:48:32.000000000 +0100
@@ -170,7 +170,7 @@
 		Pol = &cTemp;
 		Symbolrate = oTransp.iSymbolrate[i];
                 *SChannel=*OldChannel;
-                SChannel->cChannel::SetSatTransponderData(oTransp.iSource,Frequenz,*Pol,Symbolrate,FEC_AUTO);
+                SChannel->cChannel::SetSatTransponderData(oTransp.iSource,Frequenz,*Pol,Symbolrate,FEC_AUTO,0,0,0);
                 cDevice::GetDevice(cardnr)->SwitchChannel(SChannel,HasSwitched);
                 PFilter = new PatFilter();
                 SFilter = new SdtFilter(PFilter);
