diff -Naurp ekg-1.8rc1-orig/configure.in ekg-1.8rc1/configure.in
--- ekg-1.8rc1-orig/configure.in	2008-12-11 16:11:33.000000000 +0100
+++ ekg-1.8rc1/configure.in	2008-12-11 16:21:32.000000000 +0100
@@ -115,7 +115,7 @@ AC_ARG_ENABLE(openssl,
 
 if test "x$have_openssl" = "xyes"; then
 	OBJS="$OBJS simlite.o"
-	LDFLAGS="$OPENSSL_LIBS $LDFLAGS"
+	LIBS="$LIBS $OPENSSL_LIBS"
 	CFLAGS="$CFLAGS $OPENSSL_INCLUDES"
 fi
 
@@ -209,7 +209,7 @@ if test "x$enable_ui_ncurses" != "xno"; 
 		AC_MSG_ERROR([You need ncurses and ncurses-devel packages.])
 	fi
 
-	LDFLAGS="$LDFLAGS $CURSES_LIBS"
+	LIBS="$LIBS $CURSES_LIBS"
 	CFLAGS="$CFLAGS $CURSES_INCLUDES"
 	echo "CURSES_INCLUDES=$CURSES_INCLUDES"
 fi
@@ -220,7 +220,7 @@ if test "x$enable_ui_gtk" != "xno"; then
 		AC_DEFINE(WITH_UI_GTK, 1, [define if you want ui-gtk])
 		OBJS="$OBJS ui-gtk.o ui-gtk-maingui.o ui-gtk-xtext.o ui-gtk-chanview.o ui-gtk-palette.o ui-gtk-bindings.o"
 
-		LDFLAGS="$LDFLAGS $GTK_LIBS"
+		LIBS="$LIBS $GTK_LIBS"
 		CFLAGS="$CFLAGS $GTK_CFLAGS"
 	fi
 fi
@@ -233,11 +233,11 @@ if test "x$enable_ui_readline" = "xyes";
 		TERMCAP_LIB=""
 	else
 		AC_CHECK_LIB(termcap, tgetent, [TERMCAP_LIB="-ltermcap"])
-		LDFLAGS="$LDFLAGS $TERMCAP_LIB"
+		LIBS="$LIBS $TERMCAP_LIB"
 	fi
 		if test "x$TERMCAP_LIB" = "x" -a "x$CURSES_LIBS" = "x"; then
 		AC_CHECK_NCURSES
-		LDFLAGS="$LDFLAGS $CURSES_LIBS"
+		LIBS="$LIBS $CURSES_LIBS"
 		CFLAGS="$CFLAGS $CURSES_INCLUDES"
 	fi
      
@@ -247,7 +247,7 @@ if test "x$enable_ui_readline" = "xyes";
 		AC_MSG_ERROR([You need both readline and readline-devel packages.])
 	fi
 
-	LDFLAGS="$READLINE_LIBS $LDFLAGS"
+	LIBS="$LIBS $READLINE_LIBS"
 	CFLAGS="$CFLAGS $READLINE_INCLUDES"
 
 	AC_CHECK_FUNCS([rl_set_prompt rl_filename_completion_function rl_get_screen_size rl_set_key rl_bind_key_in_map])
@@ -279,7 +279,7 @@ if test "x$with_libgsm" != "xno"; then
 		AC_CHECK_HEADERS(linux/soundcard.h,
 		[
 			AC_DEFINE(HAVE_VOIP, 1, [define if you want VOIP support])
-			LDFLAGS="-lgsm $LDFLAGS"
+			LIBS="$LIBS -lgsm"
 			OBJS="$OBJS voice.o"
 		])
 	fi
@@ -298,7 +298,7 @@ if test "x$with_libungif" != "xno"; then
 		AC_CHECK_HEADERS(gif_lib.h,
 		[
 			AC_DEFINE(HAVE_LIBUNGIF, 1, [define if you have libungif])
-			LDFLAGS="-lungif $LDFLAGS"
+			LIBS="$LIBS -lungif"
 			have_libungif=yes
 		])
 	])
@@ -317,7 +317,7 @@ if test "x$with_libjpeg" != "xno"; then
 		AC_CHECK_HEADERS(jpeglib.h,
 		[
 			AC_DEFINE(HAVE_LIBJPEG, 1, [define if you have libjpeg])
-			LDFLAGS="-ljpeg $LDFLAGS"
+			LIBS="$LIBS -ljpeg"
 			have_libjpeg=yes
 		])
 	])
@@ -331,7 +331,7 @@ AC_CHECK_PYTHON
   
 if test "x$have_python" = "xyes"; then
 	OBJS="$OBJS python.o"
-	LDFLAGS="$PYTHON_LIBS $LDFLAGS"
+	LIBS="$LIBS $PYTHON_LIBS"
 	CFLAGS="$CFLAGS $PYTHON_INCLUDES"
 fi
 
@@ -348,7 +348,7 @@ if test "x$with_zlib" != "xno"; then
 		AC_CHECK_HEADERS(zlib.h,
 		[
 			AC_DEFINE(HAVE_ZLIB, 1, [define if you want compressed logs])
-			LDFLAGS="-lz $LDFLAGS"
+			LIBS="$LIBS -lz"
 			have_zlib=yes
 		])
 	])
@@ -392,7 +392,7 @@ AC_ARG_ENABLE(aspell,
  	    if test "x$have_aspell_libs" = "xyes"; then
 		if test "x$ui_ncurses_default" = "xyes"; then    
  		    AC_DEFINE(WITH_ASPELL, 1, [define if you want aspell support])
- 		    LDFLAGS="$LDFLAGS -laspell"		
+ 		    LIBS="$LIBS -laspell"
 		fi
 	    else 
 	        enable_aspell=no
