diff -Naurp ekg-1.8rc1-orig/src/ui-gtk-palette.c ekg-1.8rc1/src/ui-gtk-palette.c
--- ekg-1.8rc1-orig/src/ui-gtk-palette.c	2010-01-27 22:00:51.000000000 +0100
+++ ekg-1.8rc1/src/ui-gtk-palette.c	2010-01-27 22:01:13.000000000 +0100
@@ -46,6 +46,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 
+#include <gtk/gtk.h>
 #include <gdk-pixbuf/gdk-pixbuf.h>
 #include <gdk-pixbuf/gdk-pixdata.h>
 #include <gtk/gtkstock.h>
