handle newer sort syntax

http://bugs.gentoo.org/154029

--- test/00/t0010a.sh
+++ test/00/t0010a.sh
@@ -91,7 +91,7 @@
 LC_ALL=C
 export LC_ALL
 
-sort +4 < test.out > test.out2
+sort -k5 < test.out > test.out2
 if test $? -ne 0 ; then fail; fi
 
 cat > ok << 'fubar'
--- test/00/t0011a.sh
+++ test/00/t0011a.sh
@@ -82,7 +82,7 @@
 $bin/tardy -list outfile /dev/null 2> test.out
 if test $? -ne 0 ; then fail; fi
 
-sort +4 < test.out > test.out2
+sort -k5 < test.out > test.out2
 if test $? -ne 0 ; then fail; fi
 
 cat > ok << 'fubar'
@@ -110,7 +110,7 @@
 $bin/tardy -list outfile /dev/null 2> test.out
 if test $? -ne 0 ; then fail; fi
 
-sort +4 < test.out > test.out2
+sort -k5 < test.out > test.out2
 if test $? -ne 0 ; then fail; fi
 
 cat > ok << 'fubar'
