--- tardy-1.27/test/00/t0075a.sh
+++ tardy-1.27/test/00/t0075a.sh
@@ -19,16 +19,17 @@
 
 TEST_SUBJECT="tarballs >2GB"
 . test_prelude
+export TZ=UTC
 
 cat > test.ok << 'fubar'
--rw-r--r-- root/root       256 1970-01-02 10:00 a.bogus
--rw-r--r-- root/root       256 1970-01-02 10:00 b.bogus
--rw-r--r-- root/root       256 1970-01-02 10:00 c.bogus
--rw-r--r-- root/root       256 1970-01-02 10:00 d.bogus
--rw-r--r-- root/root       256 1970-01-02 10:00 e.bogus
--rw-r--r-- root/root       256 1970-01-02 10:00 f.bogus
--rw-r--r-- root/root       256 1970-01-02 10:00 h.bogus
--rw-r--r-- root/root       256 1970-01-02 10:00 i.bogus
+-rw-r--r-- root/root       256 1970-01-02 00:00 a.bogus
+-rw-r--r-- root/root       256 1970-01-02 00:00 b.bogus
+-rw-r--r-- root/root       256 1970-01-02 00:00 c.bogus
+-rw-r--r-- root/root       256 1970-01-02 00:00 d.bogus
+-rw-r--r-- root/root       256 1970-01-02 00:00 e.bogus
+-rw-r--r-- root/root       256 1970-01-02 00:00 f.bogus
+-rw-r--r-- root/root       256 1970-01-02 00:00 h.bogus
+-rw-r--r-- root/root       256 1970-01-02 00:00 i.bogus
 fubar
 if test $? -ne 0 ; then no_result; fi
 
