diff --exclude='*~' -Naur tardy-1.12.orig/common/filenamelist/filter/progress.cc tardy-1.12/common/filenamelist/filter/progress.cc
--- tardy-1.12.orig/common/filenamelist/filter/progress.cc	2008-06-22 15:08:50.000000000 +0000
+++ tardy-1.12/common/filenamelist/filter/progress.cc	2008-06-22 15:09:36.000000000 +0000
@@ -27,6 +27,7 @@
 #include <ac/unistd.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <string.h>
 
 #include <filenamelist/filter/progress.h>
 
