diff -urN mongodb-src-r2.2.0.orig/src/SConscript.client mongodb-src-r2.2.0/src/SConscript.client
--- mongodb-src-r2.2.0.orig/src/SConscript.client	2013-03-08 10:32:46.000000000 +0200
+++ mongodb-src-r2.2.0/src/SConscript.client	2013-03-08 10:34:04.208437762 +0200
@@ -136,9 +136,9 @@
 # install
 prefix = GetOption("prefix")
 
-env.Install(prefix + "/lib", '${LIBPREFIX}mongoclient${LIBSUFFIX}')
+env.InstallAs(prefix + "/lib", '${LIBPREFIX}mongoclient${LIBSUFFIX}')
 if has_option( "sharedclient" ):
-    env.Install(prefix + "/lib", '${SHLIBPREFIX}mongoclient${SHLIBSUFFIX}')
+    env.InstallAs(prefix + "/lib", '${SHLIBPREFIX}mongoclient${SHLIBSUFFIX}')
 
 for x in clientHeaderDirectories:
     env.Install(prefix + "/include/mongo/" + x,
