diff -urN mongodb-src-r2.2.0.orig/src/SConscript.client mongodb-src-r2.2.0/src/SConscript.client
--- mongodb-src-r2.2.0.orig/src/SConscript.client	2012-08-28 08:28:11.000000000 +0300
+++ mongodb-src-r2.2.0/src/SConscript.client	2013-03-08 10:32:46.776436389 +0200
@@ -3,7 +3,7 @@
 # This SConscript describes build and install rules for the Mongo C++ driver and associated exmaple
 # programs.
 
-Import('env clientEnv')
+Import('env clientEnv has_option')
 
 clientSource = [
     'mongo/bson/oid.cpp',
@@ -96,9 +96,11 @@
     clientHeaders.extend(Glob('mongo/%s/*.hpp' % path))
 
 mongoclient_lib = env.Library('mongoclient', clientSource),
+if has_option( "sharedclient" ):
+    mongoclient_slib = env.SharedLibrary('mongoclient', clientSource)
 mongoclient_install = env.Install('#/', [
         mongoclient_lib,
-        #env.SharedLibrary('mongoclient', clientSource),
+        mongoclient_slib,
         ])
 env.Alias('mongoclient', mongoclient_install)
 
@@ -135,6 +137,8 @@
 prefix = GetOption("prefix")
 
 env.Install(prefix + "/lib", '${LIBPREFIX}mongoclient${LIBSUFFIX}')
+if has_option( "sharedclient" ):
+    env.Install(prefix + "/lib", '${SHLIBPREFIX}mongoclient${SHLIBSUFFIX}')
 
 for x in clientHeaderDirectories:
     env.Install(prefix + "/include/mongo/" + x,
