diff --git a/src/Makefile.am b/src/Makefile.am
index 8b07f68..678ebbd 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -2,9 +2,7 @@
 
 AUTOMAKE_OPTIONS = 1.6
 
-AM_CPPFLAGS = -DGFS_MODULES_DIR=\"$(libdir)/gerris\"
-
-INCLUDES = -I$(top_srcdir) -I$(includedir) \
+AM_CPPFLAGS = -DGFS_MODULES_DIR=\"$(libdir)/gerris\" \
            -DG_LOG_DOMAIN=\"Gfs\" $(GTS_CFLAGS)
 
 bin_SCRIPTS = gfs-config
@@ -114,12 +112,14 @@ libgfs3D_la_LDFLAGS = $(NO_UNDEFINED)\
         -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)\
 	-release $(LT_RELEASE) -export-dynamic
 libgfs3D_la_SOURCES = $(SRC)
+libgfs3D_la_LIBADD = $(GTS_LIBS)
 
 libgfs2D_la_LDFLAGS = $(NO_UNDEFINED)\
         -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)\
 	-release $(LT_RELEASE) -export-dynamic
 libgfs2D_la_SOURCES = $(SRC)
-libgfs2D_la_CFLAGS = $(AM_CFLAGS) -DFTT_2D=1
+libgfs2D_la_CPPFLAGS = $(AM_CPPFLAGS) -DFTT_2D=1
+libgfs2D_la_LIBADD = $(GTS_LIBS)
 
 CLEANFILES = $(BUILT_SOURCES)
 
@@ -131,12 +131,12 @@ EXTRA_DIST = \
 bin_PROGRAMS = gerris2D gerris3D
 
 gerris2D_SOURCES = gerris.c
-gerris2D_CFLAGS = $(AM_CFLAGS) -DFTT_2D=1
-gerris2D_LDADD = $(GFS2D_LIBS)
+gerris2D_CPPFLAGS = $(AM_CPPFLAGS) -DFTT_2D=1
+gerris2D_LDADD = libgfs2D.la
 gerris2D_DEPENDENCIES = libgfs2D.la
 
 gerris3D_SOURCES = gerris.c
-gerris3D_LDADD = $(GFS3D_LIBS)
+gerris3D_LDADD = libgfs3D.la
 gerris3D_DEPENDENCIES = libgfs3D.la
 
 if DARCS_CONTROLLED
