--- config.tcl.orig	2009-08-27 18:57:52.000000000 -0400
+++ config.tcl	2009-08-27 18:58:13.000000000 -0400
@@ -233,7 +233,7 @@
 		::config::setKey autoconnect 0			;#Automatically connect when amsn starts
 		::config::setKey libtls_temp ""			;#TLS
 		::config::setKey lineflushlog 1			;#Flush log files after each line
-		::config::setKey autocheckver 1			;#Automatically check for newer versions on startup
+		::config::setKey autocheckver 0			;#Automatically check for newer versions on startup
 		::config::setKey truncatenames 1		;#Truncate nicknames longer than window width in windows' title
 		::config::setKey truncatenicks 0		;#Truncate nicknames longer than window width in chat windows
 		::config::setKey keepalive 1				;#Keep alive connection (ping every minute)
@@ -244,7 +244,7 @@
 		::config::setKey start_ns_server "messenger.hotmail.com:1863"
 		::config::setKey default_gateway_server "gateway.messenger.hotmail.com"
 		::config::setKey start_gateway_server "gateway.messenger.hotmail.com"		
-		::config::setKey activeautoupdate 1		;#Active the auto update
+		::config::setKey activeautoupdate 0		;#Active the auto update
 		::config::setKey allowbadwords 1		;#Disable censure on nicks
 		::config::setKey enablebanner 1		;#Show or Hide AMSN Banner (By default Show)
 		::config::setKey startontray 0		;#Start amsn on tray icon only (hide contact list)
