--- a/src/Socket.cpp
+++ b/src/Socket.cpp
@@ -31,6 +31,8 @@
 #include <arpa/inet.h>
 #include "Socket.h"
 
+#include <cstring>
+
 CSocket::CSocket()
 : _reactor(NULL)
 , _handle(-1)
--- a/src/bdict.h
+++ b/src/bdict.h
@@ -31,6 +31,8 @@
 #include <map>
 #include "bobject.h"
 
+#include <cstring>
+
 class CBDict:public CBObject
 {
 public:
--- a/src/binteger.h
+++ b/src/binteger.h
@@ -31,6 +31,8 @@
 #include <stdio.h>
 #include "bobject.h"
 
+#include <cstdlib>
+
 class CBInteger:public CBObject
 {
 public:
--- a/src/bstring.h
+++ b/src/bstring.h
@@ -27,6 +27,7 @@
 #define _BSTRING_H_
 
 #include<string>
+#include<cstdlib>
 
 class CBString:public CBObject
 {
--- a/src/utils.h
+++ b/src/utils.h
@@ -44,6 +44,8 @@
 #include <glib.h>
 #include "sha1.h"
 
+#include <cstring>
+
 #define _(string)			gettext(string)
 #define N(string)			string
 
