diff -ur BitStormLite-0.2p.orig/src/bdict.h BitStormLite-0.2p/src/bdict.h
--- BitStormLite-0.2p.orig/src/bdict.h	2007-05-17 16:05:06.000000000 +0300
+++ BitStormLite-0.2p/src/bdict.h	2009-07-25 22:53:28.000000000 +0300
@@ -26,6 +26,7 @@
 #ifndef _BDICT_H_
 #define _BDICT_H_
 
+#include <stdio.h>
 #include <iostream>
 #include <string>
 #include <map>
diff -ur BitStormLite-0.2p.orig/src/BitStorm.h BitStormLite-0.2p/src/BitStorm.h
--- BitStormLite-0.2p.orig/src/BitStorm.h	2007-05-17 16:06:12.000000000 +0300
+++ BitStormLite-0.2p/src/BitStorm.h	2009-07-25 22:51:46.000000000 +0300
@@ -25,7 +25,7 @@
 
 #ifndef _BITSTORM_H_
 #define _BITSTORM_H_
- 
+ #include <stdint.h>
  #include <string>
  #include <list>
  #include <map>
diff -ur BitStormLite-0.2p.orig/src/BTTask.h BitStormLite-0.2p/src/BTTask.h
--- BitStormLite-0.2p.orig/src/BTTask.h	2007-05-17 16:06:52.000000000 +0300
+++ BitStormLite-0.2p/src/BTTask.h	2009-07-25 22:51:57.000000000 +0300
@@ -26,6 +26,7 @@
 #ifndef BTTASK_H_
 #define BTTASK_H_
 
+#include <stdint.h>
 #include <pthread.h>
 #include <string>
 #include <list>
