--- src/gui/gtkhexxagonboard.cpp
+++ src/gui/gtkhexxagonboard.cpp
@@ -120,8 +120,8 @@
             posx += width  * offset / 2;
             posy += height * offset / 2;
 
-            int x = (int) round(posx) + xoff;
-            int y = (int) round(posy) + yoff;
+            int displayx = (int) round(posx) + xoff;
+            int displayy = (int) round(posy) + yoff;
 
             int type;
             LookUp lookup(bbMask);
@@ -155,10 +155,10 @@
 
             Glib::RefPtr<Gdk::GC> gc = Gdk::GC::create(back);
             gc->set_clip_mask(mask[type]);
-            gc->set_clip_origin(x, y);
+            gc->set_clip_origin(displayx, displayy);
 
             if(back && map_board[type])
-                back->draw_drawable(gc, map_board[type], 0, 0, x, y, -1, -1);
+                back->draw_drawable(gc, map_board[type], 0, 0, displayx, displayy, -1, -1);
 
             cellNo++;
         }
@@ -205,12 +205,12 @@
             posx += width  * offset / 2;
             posy += height * offset / 2;
             
-            int x = (int) round(posx + w / 2) + xoff;
-            int y = (int) round(posy + h / 2) + yoff;
+            int displayx = (int) round(posx + w / 2) + xoff;
+            int displayy = (int) round(posy + h / 2) + yoff;
         
             // Now we got the place for cell cellNo
-            double dx = inx - x;
-            double dy = iny - y;
+            double dx = inx - displayx;
+            double dy = iny - displayy;
             double a = (w/2)*(w/2);
             double b  = (h/2)*(h/2);
             
